# Comparing `tmp/wgd-2.0.32.tar.gz` & `tmp/wgd-2.0.33.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "dist/wgd-2.0.32.tar", last modified: Mon Apr 22 14:09:05 2024, max compression
+gzip compressed data, was "dist/wgd-2.0.33.tar", last modified: Wed Apr 24 15:17:57 2024, max compression
```

## Comparing `wgd-2.0.32.tar` & `wgd-2.0.33.tar`

### file list

```diff
@@ -1,31 +1,31 @@
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-22 14:09:05.829514 wgd-2.0.32/
--rwxrwxrwx   0 heche     (1000) heche     (1000)    35071 2022-09-25 05:52:37.000000 wgd-2.0.32/LICENSE
--rwxrwxrwx   0 heche     (1000) heche     (1000)    95782 2024-04-22 14:09:05.830515 wgd-2.0.32/PKG-INFO
--rwxrwxrwx   0 heche     (1000) heche     (1000)    95544 2024-04-22 14:07:27.000000 wgd-2.0.32/README.md
--rwxrwxrwx   0 heche     (1000) heche     (1000)    62872 2024-04-22 13:51:18.000000 wgd-2.0.32/cli.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)      103 2024-04-22 14:09:05.831764 wgd-2.0.32/setup.cfg
--rwxrwxrwx   0 heche     (1000) heche     (1000)     1949 2024-04-22 09:38:13.000000 wgd-2.0.32/setup.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-22 14:09:05.805148 wgd-2.0.32/test/
--rwxrwxrwx   0 heche     (1000) heche     (1000)     4521 2023-06-06 22:05:06.000000 wgd-2.0.32/test/test_core.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-22 14:09:05.821514 wgd-2.0.32/wgd/
--rwxrwxrwx   0 heche     (1000) heche     (1000)      711 2022-09-25 05:52:37.000000 wgd-2.0.32/wgd/__init__.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    31126 2022-12-22 23:06:45.000000 wgd-2.0.32/wgd/beast.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)     1685 2022-09-30 07:30:21.000000 wgd-2.0.32/wgd/cluster.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    12579 2023-11-20 15:14:14.000000 wgd-2.0.32/wgd/codeml.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   157655 2024-04-22 13:37:59.000000 wgd-2.0.32/wgd/core.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    17245 2024-04-22 09:47:26.000000 wgd-2.0.32/wgd/mcmctree.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    14140 2023-03-01 13:14:50.000000 wgd-2.0.32/wgd/mix.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   119232 2023-12-30 20:22:21.000000 wgd-2.0.32/wgd/peak.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)     4722 2023-06-06 22:05:06.000000 wgd-2.0.32/wgd/postplot.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    54236 2024-04-16 08:22:14.000000 wgd-2.0.32/wgd/ratecorrect.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    18305 2024-03-04 04:24:50.000000 wgd-2.0.32/wgd/syn.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    21638 2023-10-09 15:32:16.000000 wgd-2.0.32/wgd/tree.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    31523 2023-06-23 20:25:14.000000 wgd-2.0.32/wgd/utils.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   176561 2024-03-06 03:00:56.000000 wgd-2.0.32/wgd/viz.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-22 14:09:05.828514 wgd-2.0.32/wgd.egg-info/
--rwxrwxrwx   0 heche     (1000) heche     (1000)    95782 2024-04-22 14:09:05.000000 wgd-2.0.32/wgd.egg-info/PKG-INFO
--rwxrwxrwx   0 heche     (1000) heche     (1000)      416 2024-04-22 14:09:05.000000 wgd-2.0.32/wgd.egg-info/SOURCES.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)        1 2024-04-22 14:09:05.000000 wgd-2.0.32/wgd.egg-info/dependency_links.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)       32 2024-04-22 14:09:05.000000 wgd-2.0.32/wgd.egg-info/entry_points.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)      774 2024-04-22 14:09:05.000000 wgd-2.0.32/wgd.egg-info/requires.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)        8 2024-04-22 14:09:05.000000 wgd-2.0.32/wgd.egg-info/top_level.txt
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-24 15:17:57.617244 wgd-2.0.33/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    35071 2022-09-25 05:52:37.000000 wgd-2.0.33/LICENSE
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    96706 2024-04-24 15:17:57.617244 wgd-2.0.33/PKG-INFO
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    96468 2024-04-24 15:16:44.000000 wgd-2.0.33/README.md
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    62340 2024-04-24 15:16:17.000000 wgd-2.0.33/cli.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      103 2024-04-24 15:17:57.617244 wgd-2.0.33/setup.cfg
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     1949 2024-04-24 13:31:56.000000 wgd-2.0.33/setup.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-24 15:17:57.483681 wgd-2.0.33/test/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     4521 2023-06-06 22:05:06.000000 wgd-2.0.33/test/test_core.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-24 15:17:57.608965 wgd-2.0.33/wgd/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      711 2022-09-25 05:52:37.000000 wgd-2.0.33/wgd/__init__.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    31126 2022-12-22 23:06:45.000000 wgd-2.0.33/wgd/beast.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     1685 2022-09-30 07:30:21.000000 wgd-2.0.33/wgd/cluster.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    12579 2023-11-20 15:14:14.000000 wgd-2.0.33/wgd/codeml.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   157655 2024-04-22 13:37:59.000000 wgd-2.0.33/wgd/core.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    17245 2024-04-22 09:47:26.000000 wgd-2.0.33/wgd/mcmctree.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    14140 2023-03-01 13:14:50.000000 wgd-2.0.33/wgd/mix.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   119934 2024-04-24 15:10:03.000000 wgd-2.0.33/wgd/peak.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     4722 2023-06-06 22:05:06.000000 wgd-2.0.33/wgd/postplot.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    54236 2024-04-16 08:22:14.000000 wgd-2.0.33/wgd/ratecorrect.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    18305 2024-03-04 04:24:50.000000 wgd-2.0.33/wgd/syn.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    21638 2023-10-09 15:32:16.000000 wgd-2.0.33/wgd/tree.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    31523 2023-06-23 20:25:14.000000 wgd-2.0.33/wgd/utils.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   176561 2024-03-06 03:00:56.000000 wgd-2.0.33/wgd/viz.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-24 15:17:57.617244 wgd-2.0.33/wgd.egg-info/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    96706 2024-04-24 15:17:57.000000 wgd-2.0.33/wgd.egg-info/PKG-INFO
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      416 2024-04-24 15:17:57.000000 wgd-2.0.33/wgd.egg-info/SOURCES.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)        1 2024-04-24 15:17:57.000000 wgd-2.0.33/wgd.egg-info/dependency_links.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)       32 2024-04-24 15:17:57.000000 wgd-2.0.33/wgd.egg-info/entry_points.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      774 2024-04-24 15:17:57.000000 wgd-2.0.33/wgd.egg-info/requires.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)        8 2024-04-24 15:17:57.000000 wgd-2.0.33/wgd.egg-info/top_level.txt
```

### Comparing `wgd-2.0.32/LICENSE` & `wgd-2.0.33/LICENSE`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/PKG-INFO` & `wgd-2.0.33/README.md`

 * *Files 1% similar despite different names*

```diff
@@ -1,5987 +1,6030 @@
-00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
-00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
-00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 320a  Version: 2.0.32.
-00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
-00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
-00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
-00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
-00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
-00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
-00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
-000000a0: 650a 4c69 6365 6e73 653a 2047 504c 0a44  e.License: GPL.D
-000000b0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
-000000c0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
-000000d0: 726b 646f 776e 0a4c 6963 656e 7365 2d46  rkdown.License-F
-000000e0: 696c 653a 204c 4943 454e 5345 0a0a 3c64  ile: LICENSE..<d
-000000f0: 6976 2061 6c69 676e 3d22 6365 6e74 6572  iv align="center
-00000100: 223e 0a0a 2320 6077 6764 2076 3260 203a  ">..# `wgd v2` :
-00000110: 2061 2073 7569 7465 2074 6f6f 6c20 6f66   a suite tool of
-00000120: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
-00000130: 6e64 2074 696d 696e 670a 0a5b 215b 4275  nd timing..[![Bu
-00000140: 696c 6420 5374 6174 7573 5d28 6874 7470  ild Status](http
-00000150: 733a 2f2f 6170 702e 7472 6176 6973 2d63  s://app.travis-c
-00000160: 692e 636f 6d2f 6865 6368 652d 7073 622f  i.com/heche-psb/
-00000170: 7767 642e 7376 673f 6272 616e 6368 3d70  wgd.svg?branch=p
-00000180: 6879 6c6f 6461 7469 6e67 295d 2868 7474  hylodating)](htt
-00000190: 7073 3a2f 2f61 7070 2e74 7261 7669 732d  ps://app.travis-
-000001a0: 6369 2e63 6f6d 2f68 6563 6865 2d70 7362  ci.com/heche-psb
-000001b0: 2f77 6764 290a 5b21 5b44 6f63 756d 656e  /wgd).[![Documen
-000001c0: 7461 7469 6f6e 2053 7461 7475 735d 2868  tation Status](h
-000001d0: 7474 7073 3a2f 2f72 6561 6474 6865 646f  ttps://readthedo
-000001e0: 6373 2e6f 7267 2f70 726f 6a65 6374 732f  cs.org/projects/
-000001f0: 7767 6476 322f 6261 6467 652f 3f76 6572  wgdv2/badge/?ver
-00000200: 7369 6f6e 3d6c 6174 6573 7429 5d28 6874  sion=latest)](ht
-00000210: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
-00000220: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
-00000230: 7465 7374 2f3f 6261 6467 653d 6c61 7465  test/?badge=late
-00000240: 7374 290a 5b21 5b6c 6963 656e 7365 5d28  st).[![license](
-00000250: 6874 7470 733a 2f2f 696d 672e 7368 6965  https://img.shie
-00000260: 6c64 732e 696f 2f70 7970 692f 6c2f 7767  lds.io/pypi/l/wg
-00000270: 642e 7376 6729 5d28 6874 7470 733a 2f2f  d.svg)](https://
-00000280: 7079 7069 2e70 7974 686f 6e2e 6f72 672f  pypi.python.org/
-00000290: 7079 7069 2f77 6764 290a 5b21 5b4c 6174  pypi/wgd).[![Lat
-000002a0: 6573 7420 5079 5049 2076 6572 7369 6f6e  est PyPI version
-000002b0: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
-000002c0: 6965 6c64 732e 696f 2f70 7970 692f 762f  ields.io/pypi/v/
-000002d0: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
-000002e0: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
-000002f0: 672f 7079 7069 2f77 6764 290a 5b21 5b69  g/pypi/wgd).[![i
-00000300: 6e73 7461 6c6c 2077 6974 6820 6269 6f63  nstall with bioc
-00000310: 6f6e 6461 5d28 6874 7470 733a 2f2f 696d  onda](https://im
-00000320: 672e 7368 6965 6c64 732e 696f 2f62 6164  g.shields.io/bad
-00000330: 6765 2f69 6e73 7461 6c6c 2532 3077 6974  ge/install%20wit
-00000340: 682d 6269 6f63 6f6e 6461 2d62 7269 6768  h-bioconda-brigh
-00000350: 7467 7265 656e 2e73 7667 3f73 7479 6c65  tgreen.svg?style
-00000360: 3d66 6c61 7429 5d28 6874 7470 3a2f 2f62  =flat)](http://b
-00000370: 696f 636f 6e64 612e 6769 7468 7562 2e69  ioconda.github.i
-00000380: 6f2f 7265 6369 7065 732f 7767 642f 5245  o/recipes/wgd/RE
-00000390: 4144 4d45 2e68 746d 6c29 0a5b 215b 446f  ADME.html).[![Do
-000003a0: 776e 6c6f 6164 735d 2868 7474 7073 3a2f  wnloads](https:/
-000003b0: 2f70 6570 792e 7465 6368 2f62 6164 6765  /pepy.tech/badge
-000003c0: 2f77 6764 295d 2868 7474 7073 3a2f 2f70  /wgd)](https://p
-000003d0: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
-000003e0: 2f77 6764 290a 0a2a 2a48 656e 6763 6869  /wgd)..**Hengchi
-000003f0: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
-00000400: 6165 6e65 706f 656c 2c20 5976 6573 2056  aenepoel, Yves V
-00000410: 616e 2064 6520 5065 6572 2a2a 0a0a 5b2a  an de Peer**..[*
-00000420: 2a42 696f 696e 666f 726d 6174 6963 7320  *Bioinformatics 
-00000430: 2620 4576 6f6c 7574 696f 6e61 7279 2047  & Evolutionary G
-00000440: 656e 6f6d 6963 7320 4772 6f75 702a 2a5d  enomics Group**]
-00000450: 2868 7474 7073 3a2f 2f77 7777 2e76 616e  (https://www.van
-00000460: 6465 7065 6572 6c61 622e 6f72 672f 292a  depeerlab.org/)*
-00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
-00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
-00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
-000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
-000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
-000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
-000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
-000004e0: 6c61 7469 6f6e 2920 7c20 0a5b 2a2a 5061  lation) | .[**Pa
-000004f0: 7261 6d65 7465 7273 2a2a 5d28 2370 6172  rameters**](#par
-00000500: 616d 6574 6572 7329 207c 200a 5b2a 2a55  ameters) | .[**U
-00000510: 7361 6765 2a2a 5d28 2375 7361 6765 2920  sage**](#usage) 
-00000520: 7c20 0a5b 2a2a 496c 6c75 7374 7261 7469  | .[**Illustrati
-00000530: 6f6e 2a2a 5d28 2369 6c6c 7573 7472 6174  on**](#illustrat
-00000540: 696f 6e29 207c 0a5b 2a2a 446f 6375 6d65  ion) |.[**Docume
-00000550: 6e74 6174 696f 6e2a 2a5d 2868 7474 7073  ntation**](https
-00000560: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
-00000570: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
-00000580: 742f 3f62 6164 6765 3d6c 6174 6573 7429  t/?badge=latest)
-00000590: 207c 0a5b 2a2a 4369 7461 7469 6f6e 2a2a   |.[**Citation**
-000005a0: 5d28 2363 6974 6174 696f 6e29 0a0a 3c2f  ](#citation)..</
-000005b0: 6469 763e 0a0a 6077 6764 2076 3260 2069  div>..`wgd v2` i
-000005c0: 7320 6120 7079 7468 6f6e 2070 6163 6b61  s a python packa
-000005d0: 6765 2075 7067 7261 6465 6420 6672 6f6d  ge upgraded from
-000005e0: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
-000005f0: 6764 6020 7061 636b 6167 6520 6169 6d69  gd` package aimi
-00000600: 6e67 2066 6f72 2074 6865 2069 6e66 6572  ng for the infer
-00000610: 656e 6365 2061 6e64 2074 696d 696e 6720  ence and timing 
-00000620: 6f66 2061 6e63 6965 6e74 2077 686f 6c65  of ancient whole
-00000630: 2d67 656e 6f6d 6520 6475 706c 6963 6174  -genome duplicat
-00000640: 696f 6e20 2857 4744 2920 6576 656e 7473  ion (WGD) events
-00000650: 2e20 466f 7220 7468 6520 7072 6f70 6f73  . For the propos
-00000660: 6520 6f66 2069 6c6c 7573 7472 6174 696e  e of illustratin
-00000670: 6720 7468 6520 7072 696e 6369 706c 6520  g the principle 
-00000680: 616e 6420 7573 6167 6520 6f66 2060 7767  and usage of `wg
-00000690: 6420 7632 602c 2077 6520 636f 6d70 696c  d v2`, we compil
-000006a0: 6564 2074 6869 7320 646f 6375 6d65 6e74  ed this document
-000006b0: 6174 696f 6e2e 2042 656c 6f77 2077 6520  ation. Below we 
-000006c0: 6669 7273 7420 6761 7665 2061 6e20 696e  first gave an in
-000006d0: 7472 6f64 7563 7469 6f6e 206f 7665 7220  troduction over 
-000006e0: 7468 6520 7363 6f70 6520 616e 6420 6d65  the scope and me
-000006f0: 6368 616e 6973 6d20 6f66 2060 7767 6420  chanism of `wgd 
-00000700: 7632 6020 616e 6420 7468 656e 2074 6865  v2` and then the
-00000710: 2070 7261 6374 6963 616c 2069 6e66 6f72   practical infor
-00000720: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00000730: 6720 746f 2074 6865 2069 6e73 7461 6c6c  g to the install
-00000740: 6174 696f 6e20 616e 6420 7573 6167 652e  ation and usage.
-00000750: 2041 6e20 6578 616d 706c 6172 2077 6f72   An examplar wor
-00000760: 6b66 6c6f 7720 6973 2070 726f 7669 6465  kflow is provide
-00000770: 6420 696e 2074 6865 2074 7574 6f72 6961  d in the tutoria
-00000780: 6c20 7365 6374 696f 6e20 6f6e 2068 6f77  l section on how
-00000790: 2074 6f20 7365 656b 2065 7669 6465 6e63   to seek evidenc
-000007a0: 6520 666f 7220 6120 7075 7461 7469 7665  e for a putative
-000007b0: 2057 4744 2065 7665 6e74 2061 6e64 2070   WGD event and p
-000007c0: 6572 666f 726d 2070 726f 7065 7220 7469  erform proper ti
-000007d0: 6d69 6e67 2077 6974 6820 6120 6672 6573  ming with a fres
-000007e0: 686c 7920 6f62 7461 696e 6564 2067 656e  hly obtained gen
-000007f0: 6f6d 6520 6173 7365 6d62 6c79 2069 6e20  ome assembly in 
-00000800: 6861 6e64 2e20 466f 7220 7468 6f73 6520  hand. For those 
-00000810: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000820: 6564 2069 6e20 6d6f 7265 2074 6865 6f72  ed in more theor
-00000830: 6574 6963 616c 2064 6574 6169 6c73 2c20  etical details, 
-00000840: 7765 2072 6563 6f6d 6d65 6e64 2074 7572  we recommend tur
-00000850: 6e69 6e67 2074 6f20 6f75 7220 7061 7065  ning to our pape
-00000860: 7220 616e 6420 626f 6f6b 2063 6861 7074  r and book chapt
-00000870: 6572 2066 6f72 206d 6f72 6520 6465 7461  er for more deta
-00000880: 696c 6564 2064 6573 6372 6970 7469 6f6e  iled description
-00000890: 2061 6e64 2069 6e73 6967 6874 6675 6c20   and insightful 
-000008a0: 6469 7363 7573 7369 6f6e 732e 2054 6865  discussions. The
-000008b0: 206b 6579 2069 6d70 726f 7665 6420 6665   key improved fe
-000008c0: 6174 7572 6573 206f 6620 6077 6764 2076  atures of `wgd v
-000008d0: 3260 2061 7265 2064 656d 6f6e 7374 7261  2` are demonstra
-000008e0: 7465 6420 696e 206f 7572 206c 6174 6573  ted in our lates
-000008f0: 7420 6d61 6e75 7363 7269 7074 206f 6e20  t manuscript on 
-00000900: 5b42 696f 696e 666f 726d 6174 6963 735d  [Bioinformatics]
-00000910: 2868 7474 7073 3a2f 2f64 6f69 2e6f 7267  (https://doi.org
-00000920: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
-00000930: 726d 6174 6963 732f 6274 6165 3237 3229  rmatics/btae272)
-00000940: 2e20 4966 2079 6f75 2075 7365 2060 7767  . If you use `wg
-00000950: 6420 7632 6020 696e 2079 6f75 7220 7265  d v2` in your re
-00000960: 7365 6172 6368 2c20 706c 6561 7365 2063  search, please c
-00000970: 6974 6520 7573 2061 7320 7375 6767 6573  ite us as sugges
-00000980: 7465 6420 696e 205b 4369 7461 7469 6f6e  ted in [Citation
-00000990: 5d28 2363 6974 6174 696f 6e29 2073 6563  ](#citation) sec
-000009a0: 7469 6f6e 2e0a 0a23 2320 496e 7472 6f64  tion...## Introd
-000009b0: 7563 7469 6f6e 0a0a 506f 6c79 706c 6f69  uction..Polyploi
-000009c0: 6469 7a61 7469 6f6e 732c 2074 6865 2065  dizations, the e
-000009d0: 766f 6c75 7469 6f6e 6172 7920 7072 6f63  volutionary proc
-000009e0: 6573 7320 7468 6174 2074 6865 2065 6e74  ess that the ent
-000009f0: 6972 6520 6765 6e6f 6d65 206f 6620 616e  ire genome of an
-00000a00: 206f 7267 616e 6973 6d20 6973 2064 7570   organism is dup
-00000a10: 6c69 6361 7465 642c 2061 6c73 6f20 6e61  licated, also na
-00000a20: 6d65 6420 6173 2077 686f 6c65 2d67 656e  med as whole-gen
-00000a30: 6f6d 6520 6475 706c 6963 6174 696f 6e73  ome duplications
-00000a40: 2028 5747 4473 292c 206f 6363 7572 2072   (WGDs), occur r
-00000a50: 6563 7572 7265 6e74 6c79 2061 6372 6f73  ecurrently acros
-00000a60: 7320 7468 6520 7472 6565 206f 6620 6c69  s the tree of li
-00000a70: 6665 2e20 5468 6572 6520 6172 6520 7477  fe. There are tw
-00000a80: 6f20 6d6f 6465 7320 6f66 2070 6f6c 7970  o modes of polyp
-00000a90: 6c6f 6964 697a 6174 696f 6e73 2c20 6175  loidizations, au
-00000aa0: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
-00000ab0: 6f6e 7320 616e 6420 616c 6c6f 706f 6c79  ons and allopoly
-00000ac0: 706c 6f69 6469 7a61 7469 6f6e 732e 2041  ploidizations. A
-00000ad0: 7574 6f70 6f6c 7970 6c6f 6964 697a 6174  utopolyploidizat
-00000ae0: 696f 6e73 2061 7265 2074 6865 2064 7570  ions are the dup
-00000af0: 6c69 6361 7469 6f6e 206f 6620 7468 6520  lication of the 
-00000b00: 7361 6d65 2067 656e 6f6d 652c 2072 6573  same genome, res
-00000b10: 756c 7469 6e67 2069 6e20 7477 6f20 6964  ulting in two id
-00000b20: 656e 7469 6361 6c20 7375 6267 656e 6f6d  entical subgenom
-00000b30: 6573 2061 7420 7468 6520 7469 6d65 2069  es at the time i
-00000b40: 7420 656d 6572 6765 642e 2057 6869 6c65  t emerged. While
-00000b50: 2074 6865 2061 6c6c 6f70 6f6c 7970 6c6f   the allopolyplo
-00000b60: 6964 697a 6174 696f 6e73 2061 7265 206e  idizations are n
-00000b70: 6f72 6d61 6c6c 7920 6163 6869 6576 6564  ormally achieved
-00000b80: 2069 6e20 7477 6f20 7374 6570 732c 2066   in two steps, f
-00000b90: 6972 7374 2074 6865 2068 7962 7269 6469  irst the hybridi
-00000ba0: 7a61 7469 6f6e 2062 6574 7765 656e 2074  zation between t
-00000bb0: 776f 2064 6966 6665 7265 6e74 2073 7065  wo different spe
-00000bc0: 6369 6573 2c20 7265 7375 6c74 696e 6720  cies, resulting 
-00000bd0: 696e 2074 6865 2061 7269 7369 6e67 206f  in the arising o
-00000be0: 6620 7472 616e 7369 656e 7420 686f 6d6f  f transient homo
-00000bf0: 706c 6f69 6479 2c73 6563 6f6e 6420 7468  ploidy,second th
-00000c00: 6520 6475 706c 6963 6174 696f 6e20 6f66  e duplication of
-00000c10: 2074 6865 2068 6f6d 6f70 6c6f 6964 792c   the homoploidy,
-00000c20: 2072 6573 756c 7469 6e67 2069 6e20 7468   resulting in th
-00000c30: 6520 656d 6572 6765 6e63 6520 6f66 2061  e emergence of a
-00000c40: 6c6c 6f70 6f6c 7970 6c6f 6964 792e 2044  llopolyploidy. D
-00000c50: 7565 2074 6f20 7468 6520 756e 7374 6162  ue to the unstab
-00000c60: 696c 6974 7920 616e 6420 756e 6261 6c61  ility and unbala
-00000c70: 6e63 6564 2074 6574 7261 736f 6d69 6320  nced tetrasomic 
-00000c80: 696e 6865 7269 7461 6e63 652c 2066 6f72  inheritance, for
-00000c90: 2069 6e73 7461 6e63 6520 7468 6520 6e75   instance the nu
-00000ca0: 636c 6561 722d 6379 746f 706c 6173 6d69  clear-cytoplasmi
-00000cb0: 6320 696e 636f 6d70 6174 6962 696c 6974  c incompatibilit
-00000cc0: 792c 2074 6865 2070 6f6c 7970 6c6f 6964  y, the polyploid
-00000cd0: 7920 6765 6e6f 6d65 2077 696c 6c20 7468  y genome will th
-00000ce0: 656e 2065 7870 6572 6965 6e63 6520 6120  en experience a 
-00000cf0: 7072 6f63 6573 7320 6361 6c6c 6564 2064  process called d
-00000d00: 6970 6c6f 6964 697a 6174 696f 6e2c 2061  iploidization, a
-00000d10: 6c73 6f20 6e61 6d65 6420 6173 2066 7261  lso named as fra
-00000d20: 6374 696f 6e61 7469 6f6e 2c20 6475 7269  ctionation, duri
-00000d30: 6e67 2077 6869 6368 2061 206c 6172 6765  ng which a large
-00000d40: 2070 6f72 7469 6f6e 206f 6620 6765 6e65   portion of gene
-00000d50: 2064 7570 6c69 6361 7465 7320 7769 6c6c   duplicates will
-00000d60: 2067 6574 206c 6f73 7420 616e 6420 6f6e   get lost and on
-00000d70: 6c79 2061 2066 7261 6374 696f 6e20 6361  ly a fraction ca
-00000d80: 6e20 6265 2072 6574 6169 6e65 642e 2054  n be retained. T
-00000d90: 6865 2074 7261 6365 7320 6f66 2070 6f6c  he traces of pol
-00000da0: 7970 6c6f 6964 697a 6174 696f 6e73 2063  yploidizations c
-00000db0: 616e 2062 6520 7468 7573 2075 6e65 6172  an be thus unear
-00000dc0: 7468 6564 2066 726f 6d20 7468 6573 6520  thed from these 
-00000dd0: 7265 7461 696e 6564 2067 656e 6520 6475  retained gene du
-00000de0: 706c 6963 6174 6573 2e20 5468 7265 6520  plicates. Three 
-00000df0: 6170 7072 6f61 6368 6573 2062 6173 6564  approaches based
-00000e00: 206f 6e20 6765 6e65 2064 7570 6c69 6361   on gene duplica
-00000e10: 7465 732c 206e 616d 656c 792c 202a 4b2a  tes, namely, *K*
-00000e20: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
-00000e30: 686f 642c 2067 656e 6520 7472 6565 202d  hod, gene tree -
-00000e40: 2073 7065 6369 6573 2074 7265 6520 7265   species tree re
-00000e50: 636f 6e63 696c 6961 7469 6f6e 206d 6574  conciliation met
-00000e60: 686f 6420 616e 6420 7379 6e74 656e 7920  hod and synteny 
-00000e70: 6d65 7468 6f64 2c20 6172 6520 636f 6d6d  method, are comm
-00000e80: 6f6e 6c79 2075 7365 6420 696e 2064 6574  only used in det
-00000e90: 6563 7469 6e67 2065 7669 6465 6e63 6520  ecting evidence 
-00000ea0: 666f 7220 5747 4473 2e20 5468 6520 6765  for WGDs. The ge
-00000eb0: 6e65 2074 7265 6520 2d20 7370 6563 6965  ne tree - specie
-00000ec0: 7320 7472 6565 2072 6563 6f6e 6369 6c69  s tree reconcili
-00000ed0: 6174 696f 6e20 6d65 7468 6f64 2069 7320  ation method is 
-00000ee0: 6e6f 7420 7769 7468 696e 2074 6865 2073  not within the s
-00000ef0: 636f 7065 206f 6620 6077 6764 2076 3260  cope of `wgd v2`
-00000f00: 2c20 6275 7420 7765 206b 696e 646c 7920  , but we kindly 
-00000f10: 7265 6665 7220 7265 6164 6572 7320 7768  refer readers wh
-00000f20: 6f20 6172 6520 696e 7465 7265 7374 6564  o are interested
-00000f30: 2074 6f20 7468 6520 7068 796c 6f67 656e   to the phylogen
-00000f40: 6f6d 6963 2070 726f 6772 616d 2064 6576  omic program dev
-00000f50: 656c 6f70 6564 2062 7920 4172 7468 7572  eloped by Arthur
-00000f60: 205a 7761 656e 6570 6f65 6c20 6e61 6d65   Zwaenepoel name
-00000f70: 6420 5b57 4841 4c45 5d28 6874 7470 733a  d [WHALE](https:
-00000f80: 2f2f 6769 7468 7562 2e63 6f6d 2f61 727a  //github.com/arz
-00000f90: 7761 2f57 6861 6c65 2e6a 6c29 2061 6e64  wa/Whale.jl) and
-00000fa0: 2074 6865 2061 7373 6f63 6961 7465 6420   the associated 
-00000fb0: 5b70 6170 6572 5d28 6874 7470 733a 2f2f  [paper](https://
-00000fc0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
-00000fd0: 6d6f 6c62 6576 2f6d 737a 3038 3829 2066  molbev/msz088) f
-00000fe0: 6f72 206d 6f72 6520 7465 6368 6e69 6361  or more technica
-00000ff0: 6c20 616e 6420 7468 656f 7265 7469 6361  l and theoretica
-00001000: 6c20 6465 7461 696c 732e 0a0a 5468 6520  l details...The 
-00001010: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00001020: 6d65 7468 6f64 2069 7320 6573 7461 626c  method is establ
-00001030: 6973 6865 6420 6f6e 2061 206d 6f64 656c  ished on a model
-00001040: 206f 6620 6765 6e65 2066 616d 696c 7920   of gene family 
-00001050: 6576 6f6c 7574 696f 6e20 7468 6174 2065  evolution that e
-00001060: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
-00001070: 6973 2061 6c6c 6f77 6564 2074 6f20 6576  is allowed to ev
-00001080: 6f6c 7665 2076 6961 2067 656e 6520 6475  olve via gene du
-00001090: 706c 6963 6174 696f 6e20 616e 6420 6c6f  plication and lo
-000010a0: 7373 2e20 4e6f 7465 2074 6861 7420 7468  ss. Note that th
-000010b0: 6520 6765 6e65 2066 616d 696c 7920 6865  e gene family he
-000010c0: 7265 2069 7320 6173 7375 6d65 6420 746f  re is assumed to
-000010d0: 2062 6520 7468 6520 636c 7573 7465 7220   be the cluster 
-000010e0: 6f66 2061 6c6c 2067 656e 6573 2064 6573  of all genes des
-000010f0: 6365 6e64 6564 2066 726f 6d20 616e 2061  cended from an a
-00001100: 6e63 6573 7472 616c 2067 656e 6520 696e  ncestral gene in
-00001110: 2061 2073 696e 676c 6520 6765 6e6f 6d65   a single genome
-00001120: 2e20 5265 636f 7665 7269 6e67 2074 6865  . Recovering the
-00001130: 2067 656e 6520 7472 6565 206f 6620 7375   gene tree of su
-00001140: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
-00001150: 6e66 6f72 6d73 2074 6865 2074 696d 696e  nforms the timin
-00001160: 672c 2073 6369 6c69 6365 7420 7468 6520  g, scilicet the 
-00001170: 6167 652c 206f 6620 6765 6e65 2064 7570  age, of gene dup
-00001180: 6c69 6361 7469 6f6e 2065 7665 6e74 732e  lication events.
-00001190: 2054 6865 2061 6765 2072 6566 6572 6564   The age refered
-000011a0: 2068 6572 652c 2069 7320 6e6f 7420 696e   here, is not in
-000011b0: 2072 6561 6c20 6765 6f6c 6f67 6963 616c   real geological
-000011c0: 2074 696d 652c 2062 7574 2069 6e20 7468   time, but in th
-000011d0: 6520 756e 6974 206f 6620 6576 6f6c 7574  e unit of evolut
-000011e0: 696f 6e61 7279 2064 6973 7461 6e63 652c  ionary distance,
-000011f0: 2069 2e65 2e2c 2074 6865 206e 756d 6265   i.e., the numbe
-00001200: 7220 6f66 2073 7562 7374 6974 7574 696f  r of substitutio
-00001210: 6e73 2070 6572 2073 6974 652e 2057 6865  ns per site. Whe
-00001220: 6e20 7468 6520 6576 6f6c 7574 696f 6e61  n the evolutiona
-00001230: 7279 2072 6174 6520 7265 6d61 696e 7320  ry rate remains 
-00001240: 6170 7072 6f78 696d 6174 656c 7920 636f  approximately co
-00001250: 6e73 7461 6e74 2c20 7468 6520 6576 6f6c  nstant, the evol
-00001260: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001270: 6520 6973 2074 6865 6e20 7375 7070 6f73  e is then suppos
-00001280: 6564 2074 6f20 6265 2070 726f 706f 7274  ed to be proport
-00001290: 696f 6e61 6c20 746f 2074 6865 2072 6561  ional to the rea
-000012a0: 6c20 6576 6f6c 7574 696f 6e61 7279 2074  l evolutionary t
-000012b0: 696d 652e 2054 6865 2073 796e 6f6e 796d  ime. The synonym
-000012c0: 6f75 7320 6469 7374 616e 6365 202a 4b2a  ous distance *K*
-000012d0: 3c73 7562 3e53 3c2f 7375 623e 2c20 7468  <sub>S</sub>, th
-000012e0: 6520 6e75 6d62 6572 206f 6620 7379 6e6f  e number of syno
-000012f0: 6e79 6d6f 7573 2073 7562 7374 6974 7574  nymous substitut
-00001300: 696f 6e73 2070 6572 2073 796e 6f6e 796d  ions per synonym
-00001310: 6f75 7320 7369 7465 2c20 6973 2073 7563  ous site, is suc
-00001320: 6820 6361 6e64 6964 6174 6520 7468 6174  h candidate that
-00001330: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
-00001340: 7469 7475 7469 6f6e 7320 776f 756c 6420  titutions would 
-00001350: 6e6f 7420 696e 6375 7220 7468 6520 6368  not incur the ch
-00001360: 616e 6765 206f 6620 616d 696e 6f20 6163  ange of amino ac
-00001370: 6964 2061 6e64 2061 7265 2074 6875 7320  id and are thus 
-00001380: 7265 6761 7264 6564 2061 7320 6e65 7574  regarded as neut
-00001390: 7261 6c2c 2077 6869 6368 2061 6363 6f72  ral, which accor
-000013a0: 6469 6e67 2074 6f20 7468 6520 6e65 7574  ding to the neut
-000013b0: 7261 6c20 7468 656f 7279 2073 686f 756c  ral theory shoul
-000013c0: 6420 6f63 6375 7220 696e 2063 6f6e 7374  d occur in const
-000013d0: 616e 7420 7261 7465 2e20 4769 7665 6e20  ant rate. Given 
-000013e0: 6120 6d6f 6465 6c20 6f66 2067 656e 6520  a model of gene 
-000013f0: 6661 6d69 6c79 2074 6861 7420 616c 6c6f  family that allo
-00001400: 7773 2074 6865 2067 656e 6520 746f 2064  ws the gene to d
-00001410: 7570 6c69 6361 7465 2061 6e64 2067 6574  uplicate and get
-00001420: 206c 6f73 7420 696e 2061 2066 6978 6564   lost in a fixed
-00001430: 2072 6174 652c 206f 6e65 2063 616e 2064   rate, one can d
-00001440: 6572 6976 6520 7468 6174 2074 6865 2070  erive that the p
-00001450: 726f 6261 6269 6c69 7479 2064 656e 7369  robability densi
-00001460: 7479 2066 756e 6374 696f 6e20 6f66 2074  ty function of t
-00001470: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-00001480: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00001490: 696f 6e20 6f66 2072 6574 6169 6e65 6420  ion of retained 
-000014a0: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
-000014b0: 6973 2061 2071 7561 7369 2d65 7870 6f6e  is a quasi-expon
-000014c0: 656e 7469 616c 2066 756e 6374 696f 6e20  ential function 
-000014d0: 7468 6174 206d 6f73 7420 7265 7461 696e  that most retain
-000014e0: 6564 2067 656e 6520 6475 706c 6963 6174  ed gene duplicat
-000014f0: 6573 2061 7265 2072 6563 656e 746c 7920  es are recently 
-00001500: 626f 726e 6564 2077 6974 6820 7e30 2061  borned with ~0 a
-00001510: 6765 2077 6869 6c65 2061 7320 7468 6520  ge while as the 
-00001520: 6167 6520 676f 696e 6720 6f6c 6465 7220  age going older 
-00001530: 7468 6520 6173 736f 6369 6174 6564 206e  the associated n
-00001540: 756d 6265 7220 6f66 2072 6574 6169 6e65  umber of retaine
-00001550: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-00001560: 7320 6465 6361 7920 7175 6173 692d 6578  s decay quasi-ex
-00001570: 706f 6e65 6e74 6961 6c6c 792e 2054 6865  ponentially. The
-00001580: 7265 666f 7265 2c20 7468 6520 6f63 6375  refore, the occu
-00001590: 7261 6e63 6520 6f66 206c 6172 6765 2d73  rance of large-s
-000015a0: 6361 6c65 2067 656e 6520 6475 706c 6963  cale gene duplic
-000015b0: 6174 696f 6e20 6576 656e 7473 2c20 666f  ation events, fo
-000015c0: 7220 696e 7374 616e 6520 5747 4473 2c20  r instane WGDs, 
-000015d0: 7769 7468 2076 6172 6965 6420 7265 7465  with varied rete
-000015e0: 6e74 696f 6e20 7261 7465 2c20 7769 6c6c  ntion rate, will
-000015f0: 206c 6561 7665 2061 6e20 6167 6520 7065   leave an age pe
-00001600: 616b 2066 726f 6d20 7468 6520 6275 7273  ak from the burs
-00001610: 7420 6f66 2067 656e 6520 6475 706c 6963  t of gene duplic
-00001620: 6174 6573 2069 6e20 6120 7368 6f72 7420  ates in a short 
-00001630: 7469 6d65 2d66 7261 6d65 2075 706f 6e20  time-frame upon 
-00001640: 7468 6520 696e 6974 6961 6c20 6167 6520  the initial age 
-00001650: 6469 7374 7269 6275 7469 6f6e 2c20 616e  distribution, an
-00001660: 6420 6361 6e20 6265 2075 6e76 6569 6c65  d can be unveile
-00001670: 6420 6672 6f6d 206d 6978 7475 7265 206d  d from mixture m
-00001680: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
-00001690: 2e20 486f 7765 7665 722c 2057 4744 7320  . However, WGDs 
-000016a0: 6964 656e 7469 6669 6564 2066 726f 6d20  identified from 
-000016b0: 7468 6520 7061 7261 6c6f 676f 7573 202a  the paralogous *
-000016c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-000016d0: 6765 2064 6973 7472 6962 7574 696f 6e73  ge distributions
-000016e0: 2063 616e 206f 6e6c 7920 696e 666f 726d   can only inform
-000016f0: 2074 6865 2057 4744 2074 696d 696e 6720   the WGD timing 
-00001700: 696e 2074 6865 2074 696d 652d 7363 616c  in the time-scal
-00001710: 6520 6f66 2074 6861 7420 7370 6563 6966  e of that specif
-00001720: 6963 2073 7065 6369 6573 2c20 7768 6963  ic species, whic
-00001730: 6820 6973 206e 6f74 2063 6f6d 7061 7261  h is not compara
-00001740: 626c 6520 696e 2074 6865 2070 6879 6c6f  ble in the phylo
-00001750: 6765 6e65 7469 6320 636f 6e74 6578 742e  genetic context.
-00001760: 204f 6e6c 7920 7769 7468 2074 6865 206f   Only with the o
-00001770: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00001780: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00001790: 6973 7472 6962 7574 696f 6e73 2c20 7768  istributions, wh
-000017a0: 6963 6820 636f 6e76 6572 7420 7468 6520  ich convert the 
-000017b0: 6573 7469 6d61 7465 6420 626f 6479 2066  estimated body f
-000017c0: 726f 6d20 7061 7261 6c6f 6775 6573 2074  rom paralogues t
-000017d0: 6f20 6f72 7468 6f6c 6f67 7565 7320 616e  o orthologues an
-000017e0: 6420 696e 666f 726d 2074 6865 2072 656c  d inform the rel
-000017f0: 6174 6976 6520 7469 6d69 6e67 206f 6620  ative timing of 
-00001800: 7370 6563 6961 7469 6f6e 2065 7665 6e74  speciation event
-00001810: 732c 2063 616e 2077 6520 6465 6369 7068  s, can we deciph
-00001820: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
-00001830: 7469 6320 706c 6163 656d 656e 7420 6f66  tic placement of
-00001840: 2057 4744 7320 6166 7465 7220 7072 6f70   WGDs after prop
-00001850: 6572 2072 6174 6520 636f 7272 6563 7469  er rate correcti
-00001860: 6f6e 2e20 6077 6764 2076 3260 2069 7320  on. `wgd v2` is 
-00001870: 7375 6368 2070 726f 6772 616d 2074 6861  such program tha
-00001880: 7420 6865 6c70 7320 7573 6572 7320 636f  t helps users co
-00001890: 6e73 7472 7563 7420 7061 7261 6c6f 676f  nstruct paralogo
-000018a0: 7573 2061 6e64 206f 7274 686f 6c6f 676f  us and orthologo
-000018b0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-000018c0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000018d0: 696f 6e73 2061 6e64 2072 6561 6c69 7a65  ions and realize
-000018e0: 2062 6f74 6820 7468 6520 6964 656e 7469   both the identi
-000018f0: 6669 6361 7469 6f6e 2061 6e64 2070 6c61  fication and pla
-00001900: 6365 6d65 6e74 206f 6620 5747 4473 2e0a  cement of WGDs..
-00001910: 0a49 6e20 7468 6520 7072 656d 6973 6520  .In the premise 
-00001920: 6f66 2070 6879 6c6f 6765 6e65 7469 6361  of phylogenetica
-00001930: 6c6c 7920 6c6f 6361 7465 6420 5747 4473  lly located WGDs
-00001940: 2c20 7468 6520 6162 736f 6c75 7465 2061  , the absolute a
-00001950: 6765 2028 696e 2067 656f 6c6f 6769 6361  ge (in geologica
-00001960: 6c20 7469 6d65 2920 6f66 2057 4744 7320  l time) of WGDs 
-00001970: 6361 6e20 616c 736f 2062 6520 696e 6665  can also be infe
-00001980: 7272 6564 2066 726f 6d20 7468 6f73 6520  rred from those 
-00001990: 5747 442d 7265 7461 696e 6564 2067 656e  WGD-retained gen
-000019a0: 6520 6475 706c 6963 6174 6573 2c20 616c  e duplicates, al
-000019b0: 7468 6f75 6768 2074 6865 7265 2068 6173  though there has
-000019c0: 2062 6565 6e20 6e6f 2065 6173 7920 6f72   been no easy or
-000019d0: 2073 7472 6169 6768 7466 6f72 7761 7264   straightforward
-000019e0: 2070 6970 656c 696e 6520 666f 7220 7468   pipeline for th
-000019f0: 6973 206a 6f62 2073 6f20 6661 722e 2049  is job so far. I
-00001a00: 6e20 6077 6764 2076 3260 2c20 7765 2064  n `wgd v2`, we d
-00001a10: 6576 656c 6f70 6564 2061 2066 6561 7369  eveloped a feasi
-00001a20: 626c 6520 696e 7465 6772 6174 6564 2070  ble integrated p
-00001a30: 6970 6c65 6c69 6e65 2066 6f72 2061 6273  ipleline for abs
-00001a40: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
-00001a50: 5747 4473 2e20 5468 6520 7069 7065 6c69  WGDs. The pipeli
-00001a60: 6e65 2063 616e 2062 6520 726f 7567 686c  ne can be roughl
-00001a70: 7920 6469 7669 6465 6420 696e 746f 2074  y divided into t
-00001a80: 6872 6565 206d 6169 6e20 7374 6570 732e  hree main steps.
-00001a90: 2031 2920 5468 6520 636f 6e73 7472 7563   1) The construc
-00001aa0: 7469 6f6e 206f 6620 616e 6368 6f72 202a  tion of anchor *
-00001ab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-00001ac0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
-00001ad0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
-00001ae0: 6f66 2063 7265 6469 6162 6c65 202a 4b2a  of crediable *K*
-00001af0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
-00001b00: 6765 2061 646f 7074 6564 2066 6f72 2070  ge adopted for p
-00001b10: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-00001b20: 6e67 2c20 7573 696e 6720 6077 6764 2064  ng, using `wgd d
-00001b30: 6d64 602c 2060 7767 6420 6b73 6460 2c20  md`, `wgd ksd`, 
-00001b40: 6077 6764 2073 796e 6020 616e 6420 6077  `wgd syn` and `w
-00001b50: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
-00001b60: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
-00001b70: 2063 6f6e 7369 6465 7220 6765 6e6f 6d65   consider genome
-00001b80: 2061 7373 656d 626c 7920 6265 6361 7573   assembly becaus
-00001b90: 6520 666f 7220 7472 616e 7363 7269 7074  e for transcript
-00001ba0: 6f6d 6520 6173 7365 6d62 6c79 2069 7427  ome assembly it'
-00001bb0: 7320 696d 706f 7373 6962 6c65 2074 6f20  s impossible to 
-00001bc0: 6469 7374 696e 6775 6973 6820 5747 442d  distinguish WGD-
-00001bd0: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001be0: 6573 2066 726f 6d20 736d 616c 6c2d 7363  es from small-sc
-00001bf0: 616c 6520 6475 706c 6963 6174 696f 6e2d  ale duplication-
-00001c00: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001c10: 6573 2c20 7768 6963 6820 6861 7070 656e  es, which happen
-00001c20: 6564 2069 6e20 6120 636f 6e74 696e 756f  ed in a continuo
-00001c30: 7573 2074 696d 652d 6672 616d 6520 696e  us time-frame in
-00001c40: 7374 6561 6420 6f66 206f 6e6c 7920 6120  stead of only a 
-00001c50: 7365 7061 7261 7465 2073 686f 7274 2074  separate short t
-00001c60: 696d 652d 6672 616d 6520 616e 6420 7468  ime-frame and th
-00001c70: 7573 2072 6566 6c65 6374 7320 7468 6520  us reflects the 
-00001c80: 6475 7261 7469 6f6e 206f 6620 7468 6174  duration of that
-00001c90: 2062 7261 6e63 6820 7261 7468 6572 2074   branch rather t
-00001ca0: 6861 6e20 7468 6520 7469 6d65 2061 7420  han the time at 
-00001cb0: 7768 6963 6820 5747 4420 6f63 6375 7272  which WGD occurr
-00001cc0: 6564 2e20 3229 2054 6865 2066 6f72 6d75  ed. 2) The formu
-00001cd0: 6c61 7469 6f6e 206f 6620 6120 7374 6172  lation of a star
-00001ce0: 7469 6e67 2074 7265 6520 7573 6564 2069  ting tree used i
-00001cf0: 6e20 7468 6520 7068 796c 6f67 656e 6574  n the phylogenet
-00001d00: 6963 2064 6174 696e 672c 2063 6f6d 706f  ic dating, compo
-00001d10: 7365 6420 6f66 2061 2066 6577 2073 7065  sed of a few spe
-00001d20: 6369 6573 2061 6e64 2061 6e6e 6f74 6174  cies and annotat
-00001d30: 6564 2077 6974 6820 666f 7373 696c 2063  ed with fossil c
-00001d40: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00001d50: 6d61 7469 6f6e 2e20 5468 6973 2073 7465  mation. This ste
-00001d60: 7020 6973 2065 7373 656e 7469 616c 2066  p is essential f
-00001d70: 6f72 2074 6865 2072 6573 756c 7420 6f66  or the result of
-00001d80: 2061 6273 6f6c 7574 6520 5747 4420 6461   absolute WGD da
-00001d90: 7469 6e67 2074 6861 7420 7765 2073 7567  ting that we sug
-00001da0: 6765 7374 2075 7365 7273 206f 6620 7461  gest users of ta
-00001db0: 6b69 6e67 2067 7265 6174 2063 6175 7469  king great cauti
-00001dc0: 6f6e 2074 6f20 6173 7375 7265 2074 6865  on to assure the
-00001dd0: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
-00001de0: 6e64 2070 726f 7065 7220 626f 756e 6473  nd proper bounds
-00001df0: 2066 6f72 2066 6f73 7369 6c20 6361 6c69   for fossil cali
-00001e00: 6272 6174 696f 6e73 2e20 3329 2054 6865  brations. 3) The
-00001e10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-00001e20: 206f 7274 686f 6772 6f75 7073 2063 6f6e   orthogroups con
-00001e30: 7369 7374 696e 6720 6f66 2063 6f6c 6c69  sisting of colli
-00001e40: 6e65 6172 2064 7570 6c69 6361 7465 7320  near duplicates 
-00001e50: 6f66 2074 6865 2066 6f63 616c 2073 7065  of the focal spe
-00001e60: 6369 6573 2061 6e64 2074 6865 6972 2072  cies and their r
-00001e70: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
-00001e80: 6974 7320 2852 4248 7329 2061 6761 696e  its (RBHs) again
-00001e90: 7374 206f 7468 6572 2073 7065 6369 6573  st other species
-00001ea0: 2069 6e20 7468 6520 7374 6172 7469 6e67   in the starting
-00001eb0: 2074 7265 6520 616e 6420 7468 6520 7068   tree and the ph
-00001ec0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-00001ed0: 6720 7573 696e 6720 6120 6d6f 6c65 6375  g using a molecu
-00001ee0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
-00001ef0: 616d 2066 6f72 2069 6e73 7461 6e63 6520  am for instance 
-00001f00: 606d 636d 6374 7265 6560 2c20 7669 6120  `mcmctree`, via 
-00001f10: 7072 6f67 7261 6d20 6077 6764 2064 6d64  program `wgd dmd
-00001f20: 6020 616e 6420 6077 6764 2066 6f63 7573  ` and `wgd focus
-00001f30: 602e 2057 6520 7265 636f 6d6d 656e 6420  `. We recommend 
-00001f40: 7468 6520 7573 6167 6520 6f66 2042 6179  the usage of Bay
-00001f50: 6573 6961 6e20 6d6f 6c65 6375 6c61 7220  esian molecular 
-00001f60: 6461 7469 6e67 2070 726f 6772 616d 2060  dating program `
-00001f70: 6d63 6d63 7472 6565 602c 2077 6869 6368  mcmctree`, which
-00001f80: 2070 726f 7669 6465 7320 6120 7661 7269   provides a vari
-00001f90: 6574 7920 6f66 2064 6966 6665 7265 6e74  ety of different
-00001fa0: 2073 7562 7374 6974 7574 696f 6e20 616e   substitution an
-00001fb0: 6420 7261 7465 206d 6f64 656c 732e 204e  d rate models. N
-00001fc0: 6f6e 6574 6865 6c65 7373 2c20 7765 2075  onetheless, we u
-00001fd0: 7267 6520 7573 6572 7320 746f 2073 6574  rge users to set
-00001fe0: 2070 7269 6f72 2064 6973 7472 6962 7574   prior distribut
-00001ff0: 696f 6e20 6f66 2064 6966 6665 7265 6e74  ion of different
-00002000: 2070 6172 616d 6574 6572 7320 7769 7468   parameters with
-00002010: 2063 6175 7469 6f6e 2061 6e64 2061 7373   caution and ass
-00002020: 7572 6520 6164 6571 7561 7465 2073 616d  ure adequate sam
-00002030: 706c 696e 6720 6f66 2064 6966 6665 7265  pling of differe
-00002040: 6e74 2070 6172 616d 6574 6572 732e 0a0a  nt parameters...
-00002050: 2323 2049 6e73 7461 6c6c 6174 696f 6e0a  ## Installation.
-00002060: 0a54 6865 2065 6173 6965 7374 2077 6179  .The easiest way
-00002070: 2074 6f20 696e 7374 616c 6c20 6077 6764   to install `wgd
-00002080: 2076 3260 2069 7320 7573 696e 6720 5059   v2` is using PY
-00002090: 5049 2e20 4e6f 7465 2074 6861 7420 6966  PI. Note that if
-000020a0: 2079 6f75 2077 616e 7420 746f 2067 6574   you want to get
-000020b0: 2074 6865 206c 6174 6573 7420 7570 6461   the latest upda
-000020c0: 7465 2c20 7765 2073 7567 6765 7374 2069  te, we suggest i
-000020d0: 6e73 7461 6c6c 696e 6720 6672 6f6d 2074  nstalling from t
-000020e0: 6865 2073 6f75 7263 652c 2073 696e 6365  he source, since
-000020f0: 2074 6865 2075 7064 6174 6520 6f6e 2050   the update on P
-00002100: 5950 4920 7769 6c6c 2062 6520 6465 6c61  YPI will be dela
-00002110: 7965 6420 636f 6d70 6172 6564 2074 6f20  yed compared to 
-00002120: 6865 7265 206f 6620 736f 7572 6365 2e0a  here of source..
-00002130: 0a60 6060 0a70 6970 2069 6e73 7461 6c6c  .```.pip install
-00002140: 2077 6764 0a60 6060 0a0a 546f 2069 6e73   wgd.```..To ins
-00002150: 7461 6c6c 2060 7767 6420 7632 6020 696e  tall `wgd v2` in
-00002160: 2061 2076 6972 7475 616c 2065 6e76 6972   a virtual envir
-00002170: 6f6e 6d65 6e74 2c20 7468 6520 666f 6c6c  onment, the foll
-00002180: 6f77 696e 6720 636f 6d6d 616e 6420 6c69  owing command li
-00002190: 6e65 7320 636f 756c 6420 6265 2075 7365  nes could be use
-000021a0: 642e 0a0a 6060 600a 6769 7420 636c 6f6e  d...```.git clon
-000021b0: 6520 3c77 6764 2072 6570 6f3e 0a63 6420  e <wgd repo>.cd 
-000021c0: 7767 640a 7669 7274 7561 6c65 6e76 202d  wgd.virtualenv -
-000021d0: 703d 7079 7468 6f6e 3320 454e 5620 286f  p=python3 ENV (o
-000021e0: 7220 7079 7468 6f6e 3320 2d6d 2076 656e  r python3 -m ven
-000021f0: 7620 454e 5629 0a73 6f75 7263 6520 454e  v ENV).source EN
-00002200: 562f 6269 6e2f 6163 7469 7661 7465 0a70  V/bin/activate.p
-00002210: 6970 2069 6e73 7461 6c6c 202d 7220 7265  ip install -r re
-00002220: 7175 6972 656d 656e 7473 2e74 7874 0a70  quirements.txt.p
-00002230: 6970 2069 6e73 7461 6c6c 202e 0a60 6060  ip install ..```
-00002240: 0a0a 5768 656e 206d 6574 2077 6974 6820  ..When met with 
-00002250: 7065 726d 6973 7369 6f6e 2070 726f 626c  permission probl
-00002260: 656d 2069 6e20 696e 7374 616c 6c61 7469  em in installati
-00002270: 6f6e 2c20 706c 6561 7365 2074 7279 2074  on, please try t
-00002280: 6865 2066 6f6c 6c6f 7769 6e67 2063 6f6d  he following com
-00002290: 6d61 6e64 206c 696e 652e 0a0a 6060 600a  mand line...```.
-000022a0: 7069 7020 696e 7374 616c 6c20 2d65 202e  pip install -e .
-000022b0: 0a60 6060 0a0a 4966 206d 756c 7469 706c  .```..If multipl
-000022c0: 7920 7665 7273 696f 6e73 206f 6620 6077  y versions of `w
-000022d0: 6764 6020 7765 7265 2069 6e73 7461 6c6c  gd` were install
-000022e0: 6564 2069 6e20 7468 6520 7379 7374 656d  ed in the system
-000022f0: 2c20 706c 6561 7365 2061 6464 2074 6865  , please add the
-00002300: 2072 6967 6874 2070 6174 6820 6f66 2069   right path of i
-00002310: 6e74 6572 6573 7465 6420 7665 7273 696f  nterested versio
-00002320: 6e20 696e 746f 2074 6865 2065 6e76 6972  n into the envir
-00002330: 6f6e 6d65 6e74 2076 6172 6961 626c 6573  onment variables
-00002340: 2c20 666f 7220 6578 616d 706c 650a 0a60  , for example..`
-00002350: 6060 0a65 7870 6f72 7420 5041 5448 3d22  ``.export PATH="
-00002360: 2450 4154 483a 7e2f 2e6c 6f63 616c 2f62  $PATH:~/.local/b
-00002370: 696e 2f77 6764 220a 6060 600a 0a4e 6f74  in/wgd".```..Not
-00002380: 6520 7468 6174 2074 6865 2076 6572 7369  e that the versi
-00002390: 6f6e 206f 6620 606e 756d 7079 6020 6973  on of `numpy` is
-000023a0: 2069 6d70 6f72 7461 6e74 2028 666f 7220   important (for 
-000023b0: 6d61 6e79 206f 7468 6572 2070 6163 6b61  many other packa
-000023c0: 6765 7320 6172 6520 7468 6520 7361 6d65  ges are the same
-000023d0: 206f 6620 636f 7572 7365 292c 2065 7370   of course), esp
-000023e0: 6563 6961 6c6c 7920 666f 7220 6066 6173  ecially for `fas
-000023f0: 7463 6c75 7374 6572 6020 7061 636b 6167  tcluster` packag
-00002400: 652e 2049 6e20 6f75 7220 7465 7374 2c20  e. In our test, 
-00002410: 7468 6520 606e 756d 7079 6020 312e 3139  the `numpy` 1.19
-00002420: 2e30 2077 6f72 6b73 2066 696e 6520 6f6e  .0 works fine on
-00002430: 2060 7079 7468 6f6e 332e 362f 3860 2e20   `python3.6/8`. 
-00002440: 4966 2079 6f75 206d 6574 2073 6f6d 6520  If you met some 
-00002450: 6572 726f 7273 206f 7220 7761 726e 696e  errors or warnin
-00002460: 6773 2061 626f 7574 2060 6e75 6d70 7960  gs about `numpy`
-00002470: 2c20 6d61 7962 6520 636f 6e73 6964 6572  , maybe consider
-00002480: 696e 6720 7072 652d 696e 7374 616c 6c20  ing pre-install 
-00002490: 606e 756d 7079 6020 6173 2031 2e31 392e  `numpy` as 1.19.
-000024a0: 3020 6f72 206f 7468 6572 2063 6c6f 7365  0 or other close
-000024b0: 2d62 7920 7665 7273 696f 6e73 2062 6566  -by versions bef
-000024c0: 6f72 6520 796f 7520 696e 7374 616c 6c20  ore you install 
-000024d0: 6077 6764 602e 0a0a 2323 2050 6172 616d  `wgd`...## Param
-000024e0: 6574 6572 730a 0a54 6865 7265 2061 7265  eters..There are
-000024f0: 2037 206d 6169 6e20 7072 6f67 7261 6d73   7 main programs
-00002500: 2069 6e20 6077 6764 2076 3260 3a20 6064   in `wgd v2`: `d
-00002510: 6d64 602c 6066 6f63 7573 602c 606b 7364  md`,`focus`,`ksd
-00002520: 602c 606d 6978 602c 6070 6561 6b60 2c60  `,`mix`,`peak`,`
-00002530: 7379 6e60 2c60 7669 7a60 2e20 4865 7265  syn`,`viz`. Here
-00002540: 6166 7465 7220 7765 2077 696c 6c20 7072  after we will pr
-00002550: 6f76 6964 6520 6120 6465 7461 696c 6564  ovide a detailed
-00002560: 2065 6c75 6369 6461 7469 6f6e 206f 6e20   elucidation on 
-00002570: 6561 6368 206f 6620 7468 6520 7072 6f67  each of the prog
-00002580: 7261 6d20 616e 6420 6974 7320 6173 736f  ram and its asso
-00002590: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
-000025a0: 732e 2050 6c65 6173 6520 7265 6665 7220  s. Please refer 
-000025b0: 746f 2074 6865 205b 5573 6167 655d 2823  to the [Usage](#
-000025c0: 7573 6167 6529 2066 6f72 2074 6865 2073  usage) for the s
-000025d0: 6365 6e61 7269 6f73 2074 6f20 7768 6963  cenarios to whic
-000025e0: 6820 6561 6368 2070 6172 616d 6574 6572  h each parameter
-000025f0: 2061 7070 6c69 6573 2e0a 0a54 6865 2070   applies...The p
-00002600: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
-00002610: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
-00002620: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
-00002630: 7768 6f6c 6520 7061 7261 6e6f 6d65 2c20  whole paranome, 
-00002640: 5242 4873 2028 5265 6369 7072 6f63 616c  RBHs (Reciprocal
-00002650: 2042 6573 7420 4869 7473 292c 204d 5242   Best Hits), MRB
-00002660: 4873 2028 4d75 6c74 6970 6c65 2052 6563  Hs (Multiple Rec
-00002670: 6970 726f 6361 6c20 4265 7374 2048 6974  iprocal Best Hit
-00002680: 7329 2c20 6f72 7468 6f67 726f 7570 7320  s), orthogroups 
-00002690: 616e 6420 736f 6d65 206f 7468 6572 206f  and some other o
-000026a0: 7274 686f 6772 6f75 702d 7265 6c61 7465  rthogroup-relate
-000026b0: 6420 6675 6e63 7469 6f6e 732c 2069 6e63  d functions, inc
-000026c0: 6c75 6469 6e67 2063 6972 6375 6d73 6372  luding circumscr
-000026d0: 6970 7469 6f6e 206f 6620 6e65 7374 6564  iption of nested
-000026e0: 2073 696e 676c 652d 636f 7079 206f 7274   single-copy ort
-000026f0: 686f 6772 6f75 7073 2028 4e53 4f47 7329  hogroups (NSOGs)
-00002700: 2c20 756e 6269 6173 6564 2074 6573 7420  , unbiased test 
-00002710: 6f66 2073 696e 676c 652d 636f 7079 206f  of single-copy o
-00002720: 7274 686f 6772 6f75 7073 2028 534f 4773  rthogroups (SOGs
-00002730: 2920 6f76 6572 206d 6973 7369 6e67 2069  ) over missing i
-00002740: 6e70 6172 616c 6f67 732c 2063 6f6e 7374  nparalogs, const
-00002750: 7275 6374 696f 6e20 6f66 2042 5553 434f  ruction of BUSCO
-00002760: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
-00002770: 6f70 7920 6f72 7468 6f67 726f 7570 7320  opy orthogroups 
-00002780: 2853 4f47 7329 2c61 6e64 2074 6865 2063  (SOGs),and the c
-00002790: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
-000027a0: 656e 6365 2069 6e66 6572 656e 6365 206f  ence inference o
-000027b0: 6620 7068 796c 6f67 656e 792e 0a60 6060  f phylogeny..```
-000027c0: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-000027d0: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
-000027e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000027f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002800: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002810: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002820: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
-00002830: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
-00002840: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
-00002850: 2064 6566 6175 6c74 2077 6764 5f64 6d64   default wgd_dmd
-00002860: 0a2d 742c 202d 2d74 6d70 6469 722c 2074  .-t, --tmpdir, t
-00002870: 6865 2074 656d 706f 7261 7279 2077 6f72  he temporary wor
-00002880: 6b69 6e67 2064 6972 6563 746f 7279 2c20  king directory, 
-00002890: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
-000028a0: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
-000028b0: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
-000028c0: 6420 6265 2061 7373 6967 6e65 6420 7261  d be assigned ra
-000028d0: 6e64 6f6d 206e 616d 6573 2069 6e20 6375  ndom names in cu
-000028e0: 7272 656e 7420 6469 7265 6374 6f72 7920  rrent directory 
-000028f0: 616e 6420 6175 746f 6d61 7465 6c79 2072  and automately r
-00002900: 656d 6f76 6564 2061 7420 7468 6520 636f  emoved at the co
-00002910: 6d70 6c65 7469 6f6e 206f 6620 7072 6f67  mpletion of prog
-00002920: 7261 6d2c 2065 6c73 6520 7468 6520 746d  ram, else the tm
-00002930: 7064 6972 2077 6f75 6c64 2062 6520 6b65  pdir would be ke
-00002940: 7074 0a2d 702c 202d 2d70 726f 742c 2066  pt.-p, --prot, f
-00002950: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00002960: 6865 7220 7573 696e 6720 7072 6f74 6569  her using protei
-00002970: 6e20 6f72 206e 7563 6c65 6f74 6964 6520  n or nucleotide 
-00002980: 7365 7175 656e 6365 730a 2d63 2c20 2d2d  sequences.-c, --
-00002990: 6373 636f 7265 2c20 7468 6520 632d 7363  cscore, the c-sc
-000029a0: 6f72 6520 746f 2072 6573 7472 6963 7420  ore to restrict 
-000029b0: 7468 6520 686f 6d6f 6c6f 6720 7369 6d69  the homolog simi
-000029c0: 6c61 7269 7479 206f 6620 4d52 4248 732c  larity of MRBHs,
-000029d0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
-000029e0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
-000029f0: 2c20 7468 6520 632d 7363 6f72 6520 6675  , the c-score fu
-00002a00: 6e63 696f 6e20 776f 756c 646e 2774 2062  ncion wouldn't b
-00002a10: 6520 6163 7469 7661 7465 642c 2065 6c73  e activated, els
-00002a20: 6520 6578 7065 6374 696e 6720 6120 6465  e expecting a de
-00002a30: 6369 6d61 6c20 7769 7468 696e 2074 6865  cimal within the
-00002a40: 2072 616e 6765 206f 6620 3020 616e 6420   range of 0 and 
-00002a50: 310a 2d49 2c20 2d2d 696e 666c 6174 696f  1.-I, --inflatio
-00002a60: 6e2c 2074 6865 2069 6e66 6c61 7469 6f6e  n, the inflation
-00002a70: 2066 6163 746f 7220 666f 7220 4d43 4c20   factor for MCL 
-00002a80: 7072 6f67 7261 6d2c 2064 6566 6175 6c74  program, default
-00002a90: 2032 2e30 2c20 7769 7468 2068 6967 6865   2.0, with highe
-00002aa0: 7220 7661 6c75 6520 6c65 6164 696e 6720  r value leading 
-00002ab0: 746f 206d 6f72 6520 6275 7420 736d 616c  to more but smal
-00002ac0: 6c65 7220 636c 7573 7465 7273 0a2d 652c  ler clusters.-e,
-00002ad0: 202d 2d65 7661 6c2c 2074 6865 2065 2d76   --eval, the e-v
-00002ae0: 616c 7565 2063 7574 2d6f 6666 2066 6f72  alue cut-off for
-00002af0: 2073 696d 696c 6172 6974 7920 696e 2064   similarity in d
-00002b00: 6961 6d6f 6e64 2061 6e64 2f6f 7220 686d  iamond and/or hm
-00002b10: 6d65 722c 2064 6566 6175 6c74 2031 652d  mer, default 1e-
-00002b20: 3130 0a2d 2d74 6f5f 7374 6f70 2c20 666c  10.--to_stop, fl
-00002b30: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00002b40: 6572 2074 6f20 7472 616e 736c 6174 6520  er to translate 
-00002b50: 7468 726f 7567 6820 5354 4f50 2063 6f64  through STOP cod
-00002b60: 6f6e 732c 2069 6620 7468 6520 666c 6167  ons, if the flag
-00002b70: 2077 6173 2073 6574 2c20 7472 616e 736c   was set, transl
-00002b80: 6174 696f 6e20 7769 6c6c 2062 6520 7465  ation will be te
-00002b90: 726d 696e 6174 6564 2061 7420 7468 6520  rminated at the 
-00002ba0: 6669 7273 7420 696e 2d66 7261 6d65 2073  first in-frame s
-00002bb0: 746f 7020 636f 646f 6e2c 2065 6c73 6520  top codon, else 
-00002bc0: 6120 6675 6c6c 2074 7261 6e73 6c61 7469  a full translati
-00002bd0: 6f6e 2063 6f6e 7469 6e75 696e 6720 6f6e  on continuing on
-00002be0: 2070 6173 7369 6e67 2061 6e79 2073 746f   passing any sto
-00002bf0: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
-00002c00: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
-00002c10: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00002c20: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00002c30: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
-00002c40: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
-00002c50: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
-00002c60: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
-00002c70: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
-00002c80: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
-00002c90: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
-00002ca0: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
-00002cb0: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
-00002cc0: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
-00002cd0: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
-00002ce0: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
-00002cf0: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
-00002d00: 616e 736c 6174 6564 0a2d 662c 202d 2d66  anslated.-f, --f
-00002d10: 6f63 7573 2c20 7468 6520 7370 6563 6965  ocus, the specie
-00002d20: 7320 746f 2062 6520 6d65 7267 6564 206f  s to be merged o
-00002d30: 6e20 6c6f 6361 6c20 4d52 4248 732c 2064  n local MRBHs, d
-00002d40: 6566 6175 6c74 204e 6f6e 652c 2069 6620  efault None, if 
-00002d50: 4e6f 6e65 2077 6173 2067 6976 656e 2c20  None was given, 
-00002d60: 7468 6520 6c6f 6361 6c20 4d52 4248 7320  the local MRBHs 
-00002d70: 776f 756c 646e 2774 2062 6520 696e 6665  wouldn't be infe
-00002d80: 7272 6564 0a2d 6170 2c20 2d2d 616e 6368  rred.-ap, --anch
-00002d90: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
-00002da0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-00002db0: 2066 696c 6520 6672 6f6d 2069 2d61 6468   file from i-adh
-00002dc0: 6f72 6520 666f 7220 636f 6e73 7472 7563  ore for construc
-00002dd0: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
-00002de0: 6f75 7073 2077 6974 6820 616e 6368 6f72  oups with anchor
-00002df0: 2070 6169 7273 2c20 6465 6661 756c 7420   pairs, default 
-00002e00: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
-00002e10: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
-00002e20: 7473 2064 6174 6166 696c 6520 7573 6564  ts datafile used
-00002e30: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
-00002e40: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
-00002e50: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
-00002e60: 2064 6566 6175 6c74 204e 6f6e 650a 2d6c   default None.-l
-00002e70: 652c 202d 2d6c 6973 7465 6c65 6d65 6e74  e, --listelement
-00002e80: 732c 2074 6865 206c 6973 7473 6567 6d65  s, the listsegme
-00002e90: 6e74 7320 6461 7461 2066 696c 6520 7573  nts data file us
-00002ea0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
-00002eb0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-00002ec0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
-00002ed0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
-00002ee0: 2d67 742c 202d 2d67 656e 6574 6162 6c65  -gt, --genetable
-00002ef0: 2c20 7468 6520 6765 6e65 2074 6162 6c65  , the gene table
-00002f00: 2064 6174 6166 696c 6520 7573 6564 2069   datafile used i
-00002f10: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
-00002f20: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
-00002f30: 2069 6620 696e 6974 6961 7465 642c 2064   if initiated, d
-00002f40: 6566 6175 6c74 204e 6f6e 650a 2d63 6f63  efault None.-coc
-00002f50: 2c20 2d2d 636f 6c6c 696e 6561 7263 6f61  , --collinearcoa
-00002f60: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
-00002f70: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-00002f80: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
-00002f90: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
-00002fa0: 656e 6365 2061 6e61 6c79 7369 732c 2069  ence analysis, i
-00002fb0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00002fc0: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00002fd0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00002fe0: 7465 640a 2d6b 662c 202d 2d6b 6565 7066  ted.-kf, --keepf
-00002ff0: 6173 7461 2c20 666c 6167 206f 7074 696f  asta, flag optio
-00003000: 6e2c 2077 6865 7468 6572 2074 6f20 6f75  n, whether to ou
-00003010: 7470 7574 2074 6865 2073 6571 7565 6e63  tput the sequenc
-00003020: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
-00003030: 204d 5242 4873 2c20 6966 2074 6865 2066   MRBHs, if the f
-00003040: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
-00003050: 2073 6571 7565 6e63 6573 206f 6620 4d52   sequences of MR
-00003060: 4248 7320 776f 756c 6420 6265 2069 6e20  BHs would be in 
-00003070: 6f75 7470 7574 0a2d 6b64 2c20 2d2d 6b65  output.-kd, --ke
-00003080: 6570 6475 706c 6963 6174 6573 2c20 666c  epduplicates, fl
-00003090: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-000030a0: 6572 2074 6f20 616c 6c6f 7720 7468 6520  er to allow the 
-000030b0: 7361 6d65 2067 656e 6520 746f 206f 6363  same gene to occ
-000030c0: 7572 2069 6e20 6469 6666 6572 656e 7420  ur in different 
-000030d0: 4d52 4248 7320 286f 6e6c 7920 6d65 616e  MRBHs (only mean
-000030e0: 696e 6766 756c 2077 6865 6e20 7468 6520  ingful when the 
-000030f0: 6373 636f 7265 2077 6173 2075 7365 6429  cscore was used)
-00003100: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00003110: 7320 7365 742c 2074 6865 2073 616d 6520  s set, the same 
-00003120: 6765 6e65 2063 6f75 6c64 2062 6520 6173  gene could be as
-00003130: 7369 676e 6564 2074 6f20 6469 6666 6572  signed to differ
-00003140: 656e 7420 4d52 4248 730a 2d67 6d2c 202d  ent MRBHs.-gm, -
-00003150: 2d67 6c6f 6261 6c6d 7262 682c 2066 6c61  -globalmrbh, fla
-00003160: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003170: 7220 746f 2069 6e69 7469 6174 6520 676c  r to initiate gl
-00003180: 6f62 616c 204d 5242 4873 2063 6f6e 7374  obal MRBHs const
-00003190: 7275 6374 696f 6e2c 2069 6620 7468 6520  ruction, if the 
-000031a0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-000031b0: 6520 2d2d 666f 6375 7320 6f70 7469 6f6e  e --focus option
-000031c0: 2077 6f75 6c64 2062 6520 6967 6e6f 7265   would be ignore
-000031d0: 6420 616e 6420 6f6e 6c79 2067 6c6f 6261  d and only globa
-000031e0: 6c20 4d52 4248 7320 776f 756c 6420 6265  l MRBHs would be
-000031f0: 2062 7569 6c74 0a2d 6e2c 202d 2d6e 7468   built.-n, --nth
-00003200: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
-00003210: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
-00003220: 7573 652c 2064 6566 6175 6c74 2034 0a2d  use, default 4.-
-00003230: 6f69 2c20 2d2d 6f72 7468 6f69 6e66 6572  oi, --orthoinfer
-00003240: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00003250: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00003260: 7465 206f 7274 686f 6772 6f75 7020 696e  te orthogroup in
-00003270: 6665 726e 6563 652c 2069 6620 7468 6520  fernece, if the 
-00003280: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00003290: 6520 6f72 7468 6f67 726f 7570 2069 6e66  e orthogroup inf
-000032a0: 6572 6e65 6365 2070 726f 6772 616d 2077  ernece program w
-000032b0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-000032c0: 640a 2d6f 6f2c 202d 2d6f 6e6c 796f 7274  d.-oo, --onlyort
-000032d0: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
-000032e0: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
-000032f0: 2063 6f6e 6475 6374 206f 7274 686f 6772   conduct orthogr
-00003300: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
-00003310: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003320: 6574 2c20 6f6e 6c79 2074 6865 206f 7274  et, only the ort
-00003330: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
-00003340: 6520 7069 7065 6c69 6e65 2077 6f75 6c64  e pipeline would
-00003350: 2062 6520 7065 7266 6f72 6d65 6420 7768   be performed wh
-00003360: 696c 6520 7468 6520 6f74 6865 7220 616e  ile the other an
-00003370: 616c 7973 6973 2077 6f75 6c64 6e27 7420  alysis wouldn't 
-00003380: 6265 2069 6e69 7469 6174 6564 0a2d 676e  be initiated.-gn
-00003390: 2c20 2d2d 6765 746e 736f 672c 2066 6c61  , --getnsog, fla
-000033a0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000033b0: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
-000033c0: 6520 7365 6172 6368 696e 6720 666f 7220  e searching for 
-000033d0: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
-000033e0: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
-000033f0: 2028 4e53 4f47 7329 2028 6f6e 6c79 206d   (NSOGs) (only m
-00003400: 6561 6e69 6e67 6675 6c20 7768 656e 2074  eaningful when t
-00003410: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
-00003420: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
-00003430: 2077 6173 2061 6374 6976 6174 6564 292c   was activated),
-00003440: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003450: 2073 6574 2c20 6164 6469 7469 6f6e 616c   set, additional
-00003460: 204e 534f 4773 2061 6e61 6c79 7369 7320   NSOGs analysis 
-00003470: 776f 756c 6420 6265 2070 6572 666f 726d  would be perform
-00003480: 6564 2062 6573 6964 6573 2074 6865 2062  ed besides the b
-00003490: 6173 6963 206f 7274 686f 6772 6f75 7020  asic orthogroup 
-000034a0: 696e 6665 726e 6563 650a 2d74 7265 652c  infernece.-tree,
-000034b0: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
-000034c0: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
-000034d0: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-000034e0: 6d20 746f 2069 6e76 6f6b 6520 286f 6e6c  m to invoke (onl
-000034f0: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
-00003500: 6e20 7468 6520 636f 6c6c 696e 6561 7220  n the collinear 
-00003510: 636f 616c 6573 6365 6e63 652c 2067 656e  coalescence, gen
-00003520: 652d 746f 2d66 616d 696c 7920 6173 7369  e-to-family assi
-00003530: 676e 6d65 6e74 206f 7220 4e53 4f47 7320  gnment or NSOGs 
-00003540: 616e 616c 7973 6973 2077 6572 6520 6163  analysis were ac
-00003550: 7469 7661 7465 6429 2c20 6465 6661 756c  tivated), defaul
-00003560: 7420 6661 7374 7472 6565 0a2d 7473 2c20  t fasttree.-ts, 
-00003570: 2d2d 7472 6565 7365 742c 2074 6865 2070  --treeset, the p
-00003580: 6172 616d 6574 6572 7320 7365 7474 696e  arameters settin
-00003590: 6720 666f 7220 6765 6e65 2074 7265 6520  g for gene tree 
-000035a0: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
-000035b0: 6c74 204e 6f6e 652c 2074 6869 7320 6f70  lt None, this op
-000035c0: 7469 6f6e 2063 616e 2062 6520 7072 6f76  tion can be prov
-000035d0: 6964 6564 206d 756c 7469 706c 6520 7469  ided multiple ti
-000035e0: 6d65 730a 2d6d 632c 202d 2d6d 736f 6763  mes.-mc, --msogc
-000035f0: 7574 2c20 7468 6520 7261 7469 6f20 6375  ut, the ratio cu
-00003600: 746f 6666 2066 6f72 206d 6f73 746c 7920  toff for mostly 
-00003610: 7369 6e67 6c65 2d63 6f70 7920 6661 6d69  single-copy fami
-00003620: 6c79 2028 6d65 616e 696e 6766 756c 2077  ly (meaningful w
-00003630: 6865 6e20 6163 7469 7661 7469 6e67 2074  hen activating t
-00003640: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
-00003650: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
-00003660: 2920 616e 6420 7370 6563 6965 7320 7265  ) and species re
-00003670: 7072 6573 656e 7461 7469 6f6e 2069 6e20  presentation in 
-00003680: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
-00003690: 6365 6e63 6520 616e 616c 7973 6973 2c20  cence analysis, 
-000036a0: 6465 6661 756c 7420 302e 382e 0a2d 6761  default 0.8..-ga
-000036b0: 2c20 2d2d 6765 6e65 6173 7369 676e 2c20  , --geneassign, 
-000036c0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-000036d0: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
-000036e0: 2074 6865 2067 656e 652d 746f 2d66 616d   the gene-to-fam
-000036f0: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
-00003700: 6e61 6c79 7369 732c 2069 6620 7468 6520  nalysis, if the 
-00003710: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00003720: 6520 616e 616c 7973 6973 2077 6f75 6c64  e analysis would
-00003730: 2062 6520 696e 6974 6961 7465 640a 2d73   be initiated.-s
-00003740: 612c 202d 2d73 6571 3261 7373 6967 6e2c  a, --seq2assign,
-00003750: 2074 6865 2071 7565 7269 6564 2073 6571   the queried seq
-00003760: 7565 6e63 6573 2064 6174 6120 6669 6c65  uences data file
-00003770: 2069 6e20 6765 6e65 2d74 6f2d 6661 6d69   in gene-to-fami
-00003780: 6c79 2061 7373 6967 6e6d 656e 7420 616e  ly assignment an
-00003790: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
-000037a0: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
-000037b0: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
-000037c0: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
-000037d0: 0a2d 6661 2c20 2d2d 6661 6d32 6173 7369  .-fa, --fam2assi
-000037e0: 676e 2c20 7468 6520 7175 6572 6965 6420  gn, the queried 
-000037f0: 6661 6d69 6c69 7920 6461 7461 2066 696c  familiy data fil
-00003800: 6520 696e 2067 656e 652d 746f 2d66 616d  e in gene-to-fam
-00003810: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
-00003820: 6e61 6c79 7369 732c 2064 6566 6175 6c74  nalysis, default
-00003830: 204e 6f6e 650a 2d63 632c 202d 2d63 6f6e   None.-cc, --con
-00003840: 6361 742c 2066 6c61 6720 6f70 7469 6f6e  cat, flag option
-00003850: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-00003860: 7469 6174 6520 7468 6520 636f 6e63 6174  tiate the concat
-00003870: 656e 6174 696f 6e20 7069 7065 6c69 6e65  enation pipeline
-00003880: 2066 6f72 206f 7274 686f 6772 6f75 7020   for orthogroup 
-00003890: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
-000038a0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000038b0: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
-000038c0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
-000038d0: 2d74 652c 202d 2d74 6573 7473 6f67 2c20  -te, --testsog, 
-000038e0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-000038f0: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
-00003900: 2074 6865 2075 6e62 6961 7365 6420 7465   the unbiased te
-00003910: 7374 206f 6620 7369 6e67 6c65 2d63 6f70  st of single-cop
-00003920: 7920 6765 6e65 2066 616d 696c 6965 732c  y gene families,
-00003930: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003940: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00003950: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00003960: 6961 7465 640a 2d62 732c 202d 2d62 696e  iated.-bs, --bin
-00003970: 732c 2074 6865 206e 756d 6265 7220 6f66  s, the number of
-00003980: 2062 696e 7320 6469 7669 6465 6420 696e   bins divided in
-00003990: 2074 6865 2067 656e 6520 6c65 6e67 7468   the gene length
-000039a0: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
-000039b0: 6465 6661 756c 7420 3130 300a 2d6e 702c  default 100.-np,
-000039c0: 202d 2d6e 6f72 6d61 6c69 7a65 6470 6572   --normalizedper
-000039d0: 6365 6e74 2c20 7468 6520 7065 7263 656e  cent, the percen
-000039e0: 7461 6765 206f 6620 7570 7065 7220 6869  tage of upper hi
-000039f0: 7473 2075 7365 6420 666f 7220 6765 6e65  ts used for gene
-00003a00: 206c 656e 6774 6820 6e6f 726d 616c 697a   length normaliz
-00003a10: 6174 696f 6e2c 2064 6566 6175 6c74 2035  ation, default 5
-00003a20: 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d 616c  .-nn, --nonormal
-00003a30: 697a 6174 696f 6e2c 2066 6c61 6720 6f70  ization, flag op
-00003a40: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003a50: 2063 616c 6c20 6f66 6620 7468 6520 6e6f   call off the no
-00003a60: 726d 616c 697a 6174 696f 6e2c 2069 6620  rmalization, if 
-00003a70: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003a80: 2c20 6e6f 206e 6f72 6d61 6c69 7a61 7469  , no normalizati
-00003a90: 6f6e 2077 6f75 6c64 2062 6520 636f 6e64  on would be cond
-00003aa0: 7563 7465 640a 2d62 736f 672c 202d 2d62  ucted.-bsog, --b
-00003ab0: 7573 636f 736f 672c 2066 6c61 6720 6f70  uscosog, flag op
-00003ac0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003ad0: 2069 6e69 7469 6174 6520 7468 6520 6275   initiate the bu
-00003ae0: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
-00003af0: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-00003b00: 6c79 2061 6e61 6c79 7369 732c 2069 6620  ly analysis, if 
-00003b10: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003b20: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
-00003b30: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00003b40: 640a 2d62 686d 6d2c 202d 2d62 7573 636f  d.-bhmm, --busco
-00003b50: 686d 6d2c 2074 6865 2048 4d4d 2070 726f  hmm, the HMM pro
-00003b60: 6669 6c65 2064 6174 6166 696c 6520 696e  file datafile in
-00003b70: 2074 6865 2062 7573 636f 2d67 7569 6465   the busco-guide
-00003b80: 6420 7369 6e67 6c65 2d63 6f70 7920 6765  d single-copy ge
-00003b90: 6e65 2066 616d 696c 7920 616e 616c 7973  ne family analys
-00003ba0: 6973 2c20 6465 6661 756c 7420 4e6f 6e65  is, default None
-00003bb0: 0a2d 6263 7466 2c20 2d2d 6275 7363 6f63  .-bctf, --buscoc
-00003bc0: 7574 6f66 662c 2074 6865 2048 4d4d 2073  utoff, the HMM s
-00003bd0: 636f 7265 2063 7574 6f66 6620 6461 7461  core cutoff data
-00003be0: 6669 6c65 2069 6e20 7468 6520 6275 7363  file in the busc
-00003bf0: 6f2d 6775 6964 6564 2073 696e 676c 652d  o-guided single-
-00003c00: 636f 7079 2067 656e 6520 6661 6d69 6c79  copy gene family
-00003c10: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
-00003c20: 6c74 204e 6f6e 650a 2d6f 6620 2c2d 2d6f  lt None.-of ,--o
-00003c30: 6766 6f72 6d61 742c 2066 6c61 6720 6f70  gformat, flag op
-00003c40: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003c50: 2061 6464 2069 6e64 6578 2074 6f20 7468   add index to th
-00003c60: 6520 5242 4820 6661 6d69 6c69 6573 0a60  e RBH families.`
-00003c70: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00003c80: 6077 6764 2066 6f63 7573 6020 6361 6e20  `wgd focus` can 
-00003c90: 7265 616c 697a 6520 7468 6520 636f 6e63  realize the conc
-00003ca0: 6174 656e 6174 696f 6e2d 6261 7365 6420  atenation-based 
-00003cb0: 616e 6420 636f 616c 6573 6365 6e63 652d  and coalescence-
-00003cc0: 6261 7365 6420 7068 796c 6f67 656e 6574  based phylogenet
-00003cd0: 6963 2069 6e66 6572 656e 6365 2061 6e64  ic inference and
-00003ce0: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
-00003cf0: 7469 6e67 206f 6620 5747 4473 2065 7463  ting of WGDs etc
-00003d00: 2e0a 6060 600a 7767 6420 666f 6375 7320  ..```.wgd focus 
-00003d10: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-00003d20: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
-00003d30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003d40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003d50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003d60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003d70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
-00003d80: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
-00003d90: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
-00003da0: 2064 6566 6175 6c74 2077 6764 5f66 6f63   default wgd_foc
-00003db0: 7573 0a2d 742c 202d 2d74 6d70 6469 722c  us.-t, --tmpdir,
-00003dc0: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
-00003dd0: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
-00003de0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
-00003df0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
-00003e00: 6e2c 2074 6865 2074 6d70 6469 7220 7769  n, the tmpdir wi
-00003e10: 6c6c 2062 6520 6173 7369 676e 6564 2072  ll be assigned r
-00003e20: 616e 646f 6d20 6e61 6d65 7320 696e 2063  andom names in c
-00003e30: 7572 7265 6e74 2064 6972 6563 746f 7279  urrent directory
-00003e40: 2061 6e64 2061 7574 6f6d 6174 656c 7920   and automately 
-00003e50: 7265 6d6f 7665 6420 6174 2074 6865 2063  removed at the c
-00003e60: 6f6d 706c 6574 696f 6e20 6f66 2070 726f  ompletion of pro
-00003e70: 6772 616d 2c20 656c 7365 2074 6865 2074  gram, else the t
-00003e80: 6d70 6469 7220 776f 756c 6420 6265 206b  mpdir would be k
-00003e90: 6570 740a 2d6e 2c20 2d2d 6e74 6872 6561  ept.-n, --nthrea
-00003ea0: 6473 2c20 7468 6520 6e75 6d62 6572 206f  ds, the number o
-00003eb0: 6620 7468 7265 6164 7320 746f 2075 7365  f threads to use
-00003ec0: 2c20 6465 6661 756c 7420 340a 2d2d 746f  , default 4.--to
-00003ed0: 5f73 746f 702c 2066 6c61 6720 6f70 7469  _stop, flag opti
-00003ee0: 6f6e 2c20 7768 6574 6865 7220 746f 2074  on, whether to t
-00003ef0: 7261 6e73 6c61 7465 2074 6872 6f75 6768  ranslate through
-00003f00: 2053 544f 5020 636f 646f 6e73 2c20 6966   STOP codons, if
-00003f10: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00003f20: 742c 2074 7261 6e73 6c61 7469 6f6e 2077  t, translation w
-00003f30: 696c 6c20 6265 2074 6572 6d69 6e61 7465  ill be terminate
-00003f40: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
-00003f50: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00003f60: 6f6e 2c20 656c 7365 2061 2066 756c 6c20  on, else a full 
-00003f70: 7472 616e 736c 6174 696f 6e20 636f 6e74  translation cont
-00003f80: 696e 7569 6e67 206f 6e20 7061 7374 2061  inuing on past a
-00003f90: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
-00003fa0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00003fb0: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
-00003fc0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003fd0: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
-00003fe0: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
-00003ff0: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
-00004000: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
-00004010: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
-00004020: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
-00004030: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
-00004040: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
-00004050: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
-00004060: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
-00004070: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00004080: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
-00004090: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
-000040a0: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
-000040b0: 2d73 7472 6970 5f67 6170 732c 2066 6c61  -strip_gaps, fla
-000040c0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000040d0: 7220 746f 2064 726f 7020 616c 6c20 6761  r to drop all ga
-000040e0: 7073 2069 6e20 6d75 6c74 6970 6c65 2073  ps in multiple s
-000040f0: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
-00004100: 742c 2069 6620 7468 6520 666c 6167 2077  t, if the flag w
-00004110: 6173 2073 6574 2c20 616c 6c20 6761 7073  as set, all gaps
-00004120: 2077 6f75 6c64 2062 6520 6472 6f70 7065   would be droppe
-00004130: 640a 2d61 2c20 2d2d 616c 6967 6e65 722c  d.-a, --aligner,
-00004140: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
-00004150: 2070 726f 6772 616d 2074 6f20 7573 652c   program to use,
-00004160: 2064 6566 6175 6c74 206d 6166 6674 0a2d   default mafft.-
-00004170: 7472 6565 2c20 2d2d 7472 6565 5f6d 6574  tree, --tree_met
-00004180: 686f 642c 2077 6869 6368 2067 656e 6520  hod, which gene 
-00004190: 7472 6565 2069 6e66 6572 656e 6365 2070  tree inference p
-000041a0: 726f 6772 616d 2074 6f20 696e 766f 6b65  rogram to invoke
-000041b0: 2c20 6465 6661 756c 7420 6661 7374 7472  , default fasttr
-000041c0: 6565 0a2d 7473 2c20 2d2d 7472 6565 7365  ee.-ts, --treese
-000041d0: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
-000041e0: 7320 7365 7474 696e 6720 666f 7220 6765  s setting for ge
-000041f0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
-00004200: 652c 2064 6566 6175 6c74 204e 6f6e 652c  e, default None,
-00004210: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
-00004220: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
-00004230: 7469 706c 6520 7469 6d65 730a 2d2d 636f  tiple times.--co
-00004240: 6e63 6174 656e 6174 696f 6e2c 2066 6c61  ncatenation, fla
-00004250: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004260: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
-00004270: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
-00004280: 6261 7365 6420 7370 6563 6965 7320 7472  based species tr
-00004290: 6565 2069 6e66 6572 656e 6365 2c20 6966  ee inference, if
-000042a0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-000042b0: 742c 2063 6f6e 6361 7465 6e61 7469 6f6e  t, concatenation
-000042c0: 2d62 6173 6564 2073 7065 6369 6573 2074  -based species t
-000042d0: 7265 6520 776f 756c 6420 6265 2069 6e66  ree would be inf
-000042e0: 6572 6564 0a2d 2d63 6f61 6c65 7363 656e  ered.--coalescen
-000042f0: 6365 2c20 666c 6167 206f 7074 696f 6e2c  ce, flag option,
-00004300: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00004310: 6961 7465 2074 6865 2063 6f61 6c65 7363  iate the coalesc
-00004320: 656e 6365 2d62 6173 6564 2073 7065 6369  ence-based speci
-00004330: 6573 2074 7265 6520 696e 6665 7265 6e63  es tree inferenc
-00004340: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
-00004350: 6173 2073 6574 2c20 636f 616c 6573 6365  as set, coalesce
-00004360: 6e63 652d 6261 7365 6420 7370 6563 6965  nce-based specie
-00004370: 7320 7472 6565 2077 6f75 6c64 2062 6520  s tree would be 
-00004380: 696e 6665 7265 640a 2d73 702c 202d 2d73  infered.-sp, --s
-00004390: 7065 6369 6573 7472 6565 2c20 7370 6563  peciestree, spec
-000043a0: 6965 7320 7472 6565 2064 6174 6166 696c  ies tree datafil
-000043b0: 6520 666f 7220 6461 7469 6e67 2c20 6465  e for dating, de
-000043c0: 6661 756c 7420 4e6f 6e65 0a2d 642c 202d  fault None.-d, -
-000043d0: 2d64 6174 696e 672c 2077 6869 6368 206d  -dating, which m
-000043e0: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
-000043f0: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
-00004400: 6465 6661 756c 7420 6e6f 6e65 0a2d 6473  default none.-ds
-00004410: 2c20 2d2d 6461 7469 6e67 7365 742c 2074  , --datingset, t
-00004420: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
-00004430: 7474 696e 6720 666f 7220 6461 7469 6e67  tting for dating
-00004440: 2070 726f 6772 616d 2c20 6465 6661 756c   program, defaul
-00004450: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
-00004460: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
-00004470: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
-00004480: 6573 0a2d 6e73 2c20 2d2d 6e73 6974 6573  es.-ns, --nsites
-00004490: 2c20 7468 6520 6e73 6974 6573 2069 6e66  , the nsites inf
-000044a0: 6f72 6d61 7469 6f6e 2066 6f72 2072 3873  ormation for r8s
-000044b0: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
-000044c0: 204e 6f6e 650a 2d6f 742c 202d 2d6f 7574   None.-ot, --out
-000044d0: 6772 6f75 702c 2074 6865 206f 7574 6772  group, the outgr
-000044e0: 6f75 7020 696e 666f 726d 6174 696f 6e20  oup information 
-000044f0: 666f 7220 7238 7320 6461 7469 6e67 2c20  for r8s dating, 
-00004500: 6465 6661 756c 7420 4e6f 6e65 0a2d 7074  default None.-pt
-00004510: 2c20 2d2d 7061 7274 6974 696f 6e2c 2066  , --partition, f
-00004520: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004530: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00004540: 7061 7274 6974 696f 6e20 6461 7469 6e67  partition dating
-00004550: 2061 6e61 6c79 7369 7320 666f 7220 636f   analysis for co
-00004560: 646f 6e2c 2069 6620 7468 6520 666c 6167  don, if the flag
-00004570: 2077 6173 2073 6574 2c20 616e 2061 6464   was set, an add
-00004580: 6974 696f 6e61 6c20 7061 7274 6974 696f  itional partitio
-00004590: 6e20 6461 7469 6e67 2061 6e61 6c79 7369  n dating analysi
-000045a0: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
-000045b0: 6174 6564 0a2d 616d 2c20 2d2d 6161 6d6f  ated.-am, --aamo
-000045c0: 6465 6c2c 2077 6869 6368 2070 726f 7465  del, which prote
-000045d0: 696e 206d 6f64 656c 2074 6f20 6265 2075  in model to be u
-000045e0: 7365 6420 696e 206d 636d 6374 7265 652c  sed in mcmctree,
-000045f0: 2064 6566 6175 6c74 2070 6f69 7373 6f6e   default poisson
-00004600: 0a2d 6b73 2c20 666c 6167 206f 7074 696f  .-ks, flag optio
-00004610: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00004620: 6974 6961 7465 204b 7320 6361 6c63 756c  itiate Ks calcul
-00004630: 6174 696f 6e20 666f 7220 686f 6d6f 6c6f  ation for homolo
-00004640: 6775 6573 2069 6e20 7468 6520 7072 6f76  gues in the prov
-00004650: 6964 6564 206f 7274 686f 6c6f 676f 7573  ided orthologous
-00004660: 2067 656e 6520 6661 6d69 6c79 0a2d 2d61   gene family.--a
-00004670: 6e6e 6f74 6174 696f 6e2c 2077 6869 6368  nnotation, which
-00004680: 2061 6e6e 6f74 6174 696f 6e20 7072 6f67   annotation prog
-00004690: 7261 6d20 746f 2075 7365 2c20 6465 6661  ram to use, defa
-000046a0: 756c 7420 4e6f 6e65 0a2d 2d70 6169 7277  ult None.--pairw
-000046b0: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
-000046c0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-000046d0: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
-000046e0: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
-000046f0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00004700: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
-00004710: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
-00004720: 7374 696d 6174 6564 0a2d 6564 2c20 2d2d  stimated.-ed, --
-00004730: 6567 676e 6f67 6461 7461 2c20 7468 6520  eggnogdata, the 
-00004740: 6567 676e 6f67 2061 6e6e 6f74 6174 696f  eggnog annotatio
-00004750: 6e20 6461 7461 6669 6c65 2c20 6465 6661  n datafile, defa
-00004760: 756c 7420 4e6f 6e65 0a2d 2d70 6661 6d2c  ult None.--pfam,
-00004770: 2077 6869 6368 206f 7074 696f 6e20 746f   which option to
-00004780: 2075 7365 2066 6f72 2070 6661 6d20 616e   use for pfam an
-00004790: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
-000047a0: 7420 4e6f 6e65 0a2d 2d64 6d6e 622c 2074  t None.--dmnb, t
-000047b0: 6865 2064 6961 6d6f 6e64 2064 6174 6162  he diamond datab
-000047c0: 6173 6520 666f 7220 616e 6e6f 7461 7469  ase for annotati
-000047d0: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
-000047e0: 0a2d 2d68 6d6d 2c20 7468 6520 484d 4d20  .--hmm, the HMM 
-000047f0: 7072 6f66 696c 6520 666f 7220 616e 6e6f  profile for anno
-00004800: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
-00004810: 4e6f 6e65 0a2d 2d65 7661 6c75 652c 2074  None.--evalue, t
-00004820: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
-00004830: 6666 2066 6f72 2061 6e6e 6f74 6174 696f  ff for annotatio
-00004840: 6e2c 2064 6566 6175 6c74 2031 652d 3130  n, default 1e-10
-00004850: 0a2d 2d65 7865 7061 7468 2c20 7468 6520  .--exepath, the 
-00004860: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
-00004870: 7270 726f 7363 616e 2065 7865 6375 7461  rproscan executa
-00004880: 626c 652c 2064 6566 6175 6c74 204e 6f6e  ble, default Non
-00004890: 650a 2d66 2c20 2d2d 666f 7373 696c 2c20  e.-f, --fossil, 
-000048a0: 7468 6520 666f 7373 696c 2063 616c 6962  the fossil calib
-000048b0: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
-000048c0: 6f6e 2069 6e20 4265 6173 742c 2064 6566  on in Beast, def
-000048d0: 6175 6c74 2028 2763 6c61 6465 313b 636c  ault ('clade1;cl
-000048e0: 6164 6532 272c 2027 7461 7861 312c 7461  ade2', 'taxa1,ta
-000048f0: 7861 323b 7461 7861 332c 7461 7861 3427  xa2;taxa3,taxa4'
-00004900: 2c20 2734 3b35 272c 2027 302e 353b 302e  , '4;5', '0.5;0.
-00004910: 3627 2c20 2734 3030 3b35 3030 2729 0a2d  6', '400;500').-
-00004920: 7268 2c20 2d2d 726f 6f74 6865 6967 6874  rh, --rootheight
-00004930: 2c20 7468 6520 726f 6f74 2068 6569 6768  , the root heigh
-00004940: 7420 6361 6c69 6272 6174 696f 6e20 696e  t calibration in
-00004950: 666f 2069 6e20 4265 6173 742c 2064 6566  fo in Beast, def
-00004960: 6175 6c74 2028 342c 302e 352c 3430 3029  ault (4,0.5,400)
-00004970: 0a2d 6373 2c20 2d2d 6368 6169 6e73 6574  .-cs, --chainset
-00004980: 2c20 7468 6520 7061 7261 6d65 7465 7273  , the parameters
-00004990: 206f 6620 4d43 4d43 2063 6861 696e 2069   of MCMC chain i
-000049a0: 6e20 4265 6173 742c 2064 6566 6175 6c74  n Beast, default
-000049b0: 2028 3130 3030 302c 3130 3029 0a2d 2d62   (10000,100).--b
-000049c0: 6561 7374 6c67 6a61 722c 2074 6865 2070  eastlgjar, the p
-000049d0: 6174 6820 746f 2062 6561 7374 4c47 2e6a  ath to beastLG.j
-000049e0: 6172 2c20 6465 6661 756c 7420 4e6f 6e65  ar, default None
-000049f0: 0a2d 2d62 6561 676c 652c 2066 6c61 6720  .--beagle, flag 
-00004a00: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00004a10: 746f 2075 7365 2062 6561 676c 6520 696e  to use beagle in
-00004a20: 2042 6561 7374 2c20 6966 2074 6865 2066   Beast, if the f
-00004a30: 6c61 6720 7761 7320 7365 742c 2062 6561  lag was set, bea
-00004a40: 676c 6520 776f 756c 6420 6265 2075 7365  gle would be use
-00004a50: 640a 2d2d 7072 6f74 636f 6364 6174 696e  d.--protcocdatin
-00004a60: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00004a70: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00004a80: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
-00004a90: 7465 696e 2d63 6f6e 6361 7465 6e61 7469  tein-concatenati
-00004aa0: 6f6e 2d62 6173 6564 2064 6174 696e 6720  on-based dating 
-00004ab0: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
-00004ac0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00004ad0: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
-00004ae0: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
-00004af0: 2d70 726f 7464 6174 696e 672c 2066 6c61  -protdating, fla
-00004b00: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004b10: 7220 746f 206f 6e6c 7920 696e 6974 6961  r to only initia
-00004b20: 7465 2074 6865 2070 726f 7465 696e 2d62  te the protein-b
-00004b30: 6173 6564 2064 6174 696e 6720 616e 616c  ased dating anal
-00004b40: 7973 6973 2c20 6966 2074 6865 2066 6c61  ysis, if the fla
-00004b50: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
-00004b60: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
-00004b70: 2069 6e69 7469 6174 6564 0a60 6060 0a0a   initiated.```..
-00004b80: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
-00004b90: 206b 7364 6020 6361 6e20 7265 616c 697a   ksd` can realiz
-00004ba0: 6520 7468 6520 636f 6e73 7472 7563 7469  e the constructi
-00004bb0: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
-00004bc0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00004bd0: 6275 7469 6f6e 2061 6e64 2072 6174 6520  bution and rate 
-00004be0: 636f 7272 6563 7469 6f6e 2e0a 6060 600a  correction..```.
-00004bf0: 7767 6420 6b73 6420 6661 6d69 6c69 6573  wgd ksd families
-00004c00: 2073 6571 7565 6e63 6573 2028 6f70 7469   sequences (opti
-00004c10: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
-00004c20: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004c30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004c40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004c50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004c60: 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574 6469  ----.-o, --outdi
-00004c70: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
-00004c80: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
-00004c90: 2077 6764 5f6b 7364 0a2d 742c 202d 2d74   wgd_ksd.-t, --t
-00004ca0: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
-00004cb0: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
-00004cc0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
-00004cd0: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
-00004ce0: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
-00004cf0: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
-00004d00: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
-00004d10: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
-00004d20: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
-00004d30: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
-00004d40: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
-00004d50: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
-00004d60: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
-00004d70: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
-00004d80: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
-00004d90: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
-00004da0: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
-00004db0: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
-00004dc0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004dd0: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
-00004de0: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
-00004df0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
-00004e00: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
-00004e10: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
-00004e20: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
-00004e30: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
-00004e40: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
-00004e50: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
-00004e60: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
-00004e70: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
-00004e80: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
-00004e90: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
-00004ea0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004eb0: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
-00004ec0: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
-00004ed0: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
-00004ee0: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
-00004ef0: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
-00004f00: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
-00004f10: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
-00004f20: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
-00004f30: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
-00004f40: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
-00004f50: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
-00004f60: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
-00004f70: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
-00004f80: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
-00004f90: 6174 6564 0a2d 2d70 6169 7277 6973 652c  ated.--pairwise,
-00004fa0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004fb0: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00004fc0: 6520 7061 6972 7769 7365 204b 7320 6573  e pairwise Ks es
-00004fd0: 7469 6d61 7469 6f6e 2c20 6966 2074 6865  timation, if the
-00004fe0: 2066 6c61 6720 7761 7320 7365 742c 2070   flag was set, p
-00004ff0: 6169 7277 6973 6520 4b73 2076 616c 7565  airwise Ks value
-00005000: 7320 776f 756c 6420 6265 2065 7374 696d  s would be estim
-00005010: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
-00005020: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00005030: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
-00005040: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
-00005050: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
-00005060: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
-00005070: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
-00005080: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
-00005090: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
-000050a0: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
-000050b0: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
-000050c0: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
-000050d0: 6166 6674 200a 2d74 7265 652c 202d 2d74  afft .-tree, --t
-000050e0: 7265 655f 6d65 7468 6f64 2c20 7768 6963  ree_method, whic
-000050f0: 6820 6765 6e65 2074 7265 6520 696e 6665  h gene tree infe
-00005100: 7265 6e63 6520 7072 6f67 7261 6d20 746f  rence program to
-00005110: 2069 6e76 6f6b 652c 2064 6566 6175 6c74   invoke, default
-00005120: 2066 6173 7474 7265 650a 2d2d 7472 6565   fasttree.--tree
-00005130: 5f6f 7074 696f 6e73 2c20 6f70 7469 6f6e  _options, option
-00005140: 7320 696e 2074 7265 6520 696e 6665 7265  s in tree infere
-00005150: 6e63 6520 6173 2061 2063 6f6d 6d61 2073  nce as a comma s
-00005160: 6570 6172 6174 6564 2073 7472 696e 672c  eparated string,
-00005170: 2064 6566 6175 6c74 204e 6f6e 650a 2d2d   default None.--
-00005180: 6e6f 6465 5f61 7665 7261 6765 2c20 666c  node_average, fl
-00005190: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-000051a0: 6572 2074 6f20 696e 6974 6961 7465 206e  er to initiate n
-000051b0: 6f64 652d 6176 6572 6167 6520 7761 7920  ode-average way 
-000051c0: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
-000051d0: 2069 6e73 7465 6164 206f 6620 6e6f 6465   instead of node
-000051e0: 2d77 6569 6768 7465 642c 2069 6620 7468  -weighted, if th
-000051f0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00005200: 7468 6520 6e6f 6465 2d61 7665 7261 6769  the node-averagi
-00005210: 6e67 2064 652d 7265 6475 6e64 616e 6379  ng de-redundancy
-00005220: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00005230: 7465 640a 2d73 722c 202d 2d73 7061 6972  ted.-sr, --spair
-00005240: 2c20 7468 6520 7370 6563 6965 7320 7061  , the species pa
-00005250: 6972 2074 6f20 6265 2070 6c6f 7474 6564  ir to be plotted
-00005260: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
-00005270: 7468 6973 206f 7074 696f 6e20 6361 6e20  this option can 
-00005280: 6265 2070 726f 7669 6465 6420 6d75 6c74  be provided mult
-00005290: 6970 6c65 2074 696d 6573 0a2d 7370 2c20  iple times.-sp, 
-000052a0: 2d2d 7370 6563 6965 7374 7265 652c 2074  --speciestree, t
-000052b0: 6865 2073 7065 6369 6573 2074 7265 6520  he species tree 
-000052c0: 746f 2070 6572 666f 726d 2072 6174 6520  to perform rate 
-000052d0: 636f 7272 6563 7469 6f6e 2c20 6465 6661  correction, defa
-000052e0: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
-000052f0: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
+00000000: 3c64 6976 2061 6c69 676e 3d22 6365 6e74  <div align="cent
+00000010: 6572 223e 0a0a 2320 6077 6764 2076 3260  er">..# `wgd v2`
+00000020: 203a 2061 2073 7569 7465 2074 6f6f 6c20   : a suite tool 
+00000030: 6f66 2057 4744 2069 6e66 6572 656e 6365  of WGD inference
+00000040: 2061 6e64 2074 696d 696e 670a 0a5b 215b   and timing..[![
+00000050: 4275 696c 6420 5374 6174 7573 5d28 6874  Build Status](ht
+00000060: 7470 733a 2f2f 6170 702e 7472 6176 6973  tps://app.travis
+00000070: 2d63 692e 636f 6d2f 6865 6368 652d 7073  -ci.com/heche-ps
+00000080: 622f 7767 642e 7376 673f 6272 616e 6368  b/wgd.svg?branch
+00000090: 3d70 6879 6c6f 6461 7469 6e67 295d 2868  =phylodating)](h
+000000a0: 7474 7073 3a2f 2f61 7070 2e74 7261 7669  ttps://app.travi
+000000b0: 732d 6369 2e63 6f6d 2f68 6563 6865 2d70  s-ci.com/heche-p
+000000c0: 7362 2f77 6764 290a 5b21 5b44 6f63 756d  sb/wgd).[![Docum
+000000d0: 656e 7461 7469 6f6e 2053 7461 7475 735d  entation Status]
+000000e0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
+000000f0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
+00000100: 732f 7767 6476 322f 6261 6467 652f 3f76  s/wgdv2/badge/?v
+00000110: 6572 7369 6f6e 3d6c 6174 6573 7429 5d28  ersion=latest)](
+00000120: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+00000130: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+00000140: 6c61 7465 7374 2f3f 6261 6467 653d 6c61  latest/?badge=la
+00000150: 7465 7374 290a 5b21 5b6c 6963 656e 7365  test).[![license
+00000160: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+00000170: 6965 6c64 732e 696f 2f70 7970 692f 6c2f  ields.io/pypi/l/
+00000180: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
+00000190: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
+000001a0: 672f 7079 7069 2f77 6764 290a 5b21 5b4c  g/pypi/wgd).[![L
+000001b0: 6174 6573 7420 5079 5049 2076 6572 7369  atest PyPI versi
+000001c0: 6f6e 5d28 6874 7470 733a 2f2f 696d 672e  on](https://img.
+000001d0: 7368 6965 6c64 732e 696f 2f70 7970 692f  shields.io/pypi/
+000001e0: 762f 7767 642e 7376 6729 5d28 6874 7470  v/wgd.svg)](http
+000001f0: 733a 2f2f 7079 7069 2e70 7974 686f 6e2e  s://pypi.python.
+00000200: 6f72 672f 7079 7069 2f77 6764 290a 5b21  org/pypi/wgd).[!
+00000210: 5b69 6e73 7461 6c6c 2077 6974 6820 6269  [install with bi
+00000220: 6f63 6f6e 6461 5d28 6874 7470 733a 2f2f  oconda](https://
+00000230: 696d 672e 7368 6965 6c64 732e 696f 2f62  img.shields.io/b
+00000240: 6164 6765 2f69 6e73 7461 6c6c 2532 3077  adge/install%20w
+00000250: 6974 682d 6269 6f63 6f6e 6461 2d62 7269  ith-bioconda-bri
+00000260: 6768 7467 7265 656e 2e73 7667 3f73 7479  ghtgreen.svg?sty
+00000270: 6c65 3d66 6c61 7429 5d28 6874 7470 3a2f  le=flat)](http:/
+00000280: 2f62 696f 636f 6e64 612e 6769 7468 7562  /bioconda.github
+00000290: 2e69 6f2f 7265 6369 7065 732f 7767 642f  .io/recipes/wgd/
+000002a0: 5245 4144 4d45 2e68 746d 6c29 0a5b 215b  README.html).[![
+000002b0: 446f 776e 6c6f 6164 735d 2868 7474 7073  Downloads](https
+000002c0: 3a2f 2f70 6570 792e 7465 6368 2f62 6164  ://pepy.tech/bad
+000002d0: 6765 2f77 6764 295d 2868 7474 7073 3a2f  ge/wgd)](https:/
+000002e0: 2f70 6570 792e 7465 6368 2f70 726f 6a65  /pepy.tech/proje
+000002f0: 6374 2f77 6764 290a 0a2a 2a48 656e 6763  ct/wgd)..**Hengc
+00000300: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
+00000310: 5a77 6165 6e65 706f 656c 2c20 5976 6573  Zwaenepoel, Yves
+00000320: 2056 616e 2064 6520 5065 6572 2a2a 0a0a   Van de Peer**..
+00000330: 5b2a 2a42 696f 696e 666f 726d 6174 6963  [**Bioinformatic
+00000340: 7320 2620 4576 6f6c 7574 696f 6e61 7279  s & Evolutionary
+00000350: 2047 656e 6f6d 6963 7320 4772 6f75 702a   Genomics Group*
+00000360: 2a5d 2868 7474 7073 3a2f 2f77 7777 2e76  *](https://www.v
+00000370: 616e 6465 7065 6572 6c61 622e 6f72 672f  andepeerlab.org/
+00000380: 292a 2a2c 2056 4942 2d55 4765 6e74 2043  )**, VIB-UGent C
+00000390: 656e 7465 7220 666f 7220 506c 616e 7420  enter for Plant 
+000003a0: 5379 7374 656d 7320 4269 6f6c 6f67 792a  Systems Biology*
+000003b0: 2a0a 0a5b 2a2a 496e 7472 6f64 7563 7469  *..[**Introducti
+000003c0: 6f6e 2a2a 5d28 2369 6e74 726f 6475 6374  on**](#introduct
+000003d0: 696f 6e29 207c 200a 5b2a 2a49 6e73 7461  ion) | .[**Insta
+000003e0: 6c6c 6174 696f 6e2a 2a5d 2823 696e 7374  llation**](#inst
+000003f0: 616c 6c61 7469 6f6e 2920 7c20 0a5b 2a2a  allation) | .[**
+00000400: 5061 7261 6d65 7465 7273 2a2a 5d28 2370  Parameters**](#p
+00000410: 6172 616d 6574 6572 7329 207c 200a 5b2a  arameters) | .[*
+00000420: 2a55 7361 6765 2a2a 5d28 2375 7361 6765  *Usage**](#usage
+00000430: 2920 7c20 0a5b 2a2a 496c 6c75 7374 7261  ) | .[**Illustra
+00000440: 7469 6f6e 2a2a 5d28 2369 6c6c 7573 7472  tion**](#illustr
+00000450: 6174 696f 6e29 207c 0a5b 2a2a 446f 6375  ation) |.[**Docu
+00000460: 6d65 6e74 6174 696f 6e2a 2a5d 2868 7474  mentation**](htt
+00000470: 7073 3a2f 2f77 6764 7632 2e72 6561 6474  ps://wgdv2.readt
+00000480: 6865 646f 6373 2e69 6f2f 656e 2f6c 6174  hedocs.io/en/lat
+00000490: 6573 742f 3f62 6164 6765 3d6c 6174 6573  est/?badge=lates
+000004a0: 7429 207c 0a5b 2a2a 4369 7461 7469 6f6e  t) |.[**Citation
+000004b0: 2a2a 5d28 2363 6974 6174 696f 6e29 0a0a  **](#citation)..
+000004c0: 3c2f 6469 763e 0a0a 6077 6764 2076 3260  </div>..`wgd v2`
+000004d0: 2069 7320 6120 7079 7468 6f6e 2070 6163   is a python pac
+000004e0: 6b61 6765 2075 7067 7261 6465 6420 6672  kage upgraded fr
+000004f0: 6f6d 2074 6865 206f 7269 6769 6e61 6c20  om the original 
+00000500: 6077 6764 6020 7061 636b 6167 6520 6169  `wgd` package ai
+00000510: 6d69 6e67 2066 6f72 2074 6865 2069 6e66  ming for the inf
+00000520: 6572 656e 6365 2061 6e64 2074 696d 696e  erence and timin
+00000530: 6720 6f66 2061 6e63 6965 6e74 2077 686f  g of ancient who
+00000540: 6c65 2d67 656e 6f6d 6520 6475 706c 6963  le-genome duplic
+00000550: 6174 696f 6e20 2857 4744 2920 6576 656e  ation (WGD) even
+00000560: 7473 2e20 466f 7220 7468 6520 7072 6f70  ts. For the prop
+00000570: 6f73 6520 6f66 2069 6c6c 7573 7472 6174  ose of illustrat
+00000580: 696e 6720 7468 6520 7072 696e 6369 706c  ing the principl
+00000590: 6520 616e 6420 7573 6167 6520 6f66 2060  e and usage of `
+000005a0: 7767 6420 7632 602c 2077 6520 636f 6d70  wgd v2`, we comp
+000005b0: 696c 6564 2074 6869 7320 646f 6375 6d65  iled this docume
+000005c0: 6e74 6174 696f 6e2e 2042 656c 6f77 2077  ntation. Below w
+000005d0: 6520 6669 7273 7420 6761 7665 2061 6e20  e first gave an 
+000005e0: 696e 7472 6f64 7563 7469 6f6e 206f 7665  introduction ove
+000005f0: 7220 7468 6520 7363 6f70 6520 616e 6420  r the scope and 
+00000600: 6d65 6368 616e 6973 6d20 6f66 2060 7767  mechanism of `wg
+00000610: 6420 7632 6020 616e 6420 7468 656e 2074  d v2` and then t
+00000620: 6865 2070 7261 6374 6963 616c 2069 6e66  he practical inf
+00000630: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
+00000640: 696e 6720 746f 2074 6865 2069 6e73 7461  ing to the insta
+00000650: 6c6c 6174 696f 6e20 616e 6420 7573 6167  llation and usag
+00000660: 652e 2041 6e20 6578 616d 706c 6172 2077  e. An examplar w
+00000670: 6f72 6b66 6c6f 7720 6973 2070 726f 7669  orkflow is provi
+00000680: 6465 6420 696e 2074 6865 2074 7574 6f72  ded in the tutor
+00000690: 6961 6c20 7365 6374 696f 6e20 6f6e 2068  ial section on h
+000006a0: 6f77 2074 6f20 7365 656b 2065 7669 6465  ow to seek evide
+000006b0: 6e63 6520 666f 7220 6120 7075 7461 7469  nce for a putati
+000006c0: 7665 2057 4744 2065 7665 6e74 2061 6e64  ve WGD event and
+000006d0: 2070 6572 666f 726d 2070 726f 7065 7220   perform proper 
+000006e0: 7469 6d69 6e67 2077 6974 6820 6120 6672  timing with a fr
+000006f0: 6573 686c 7920 6f62 7461 696e 6564 2067  eshly obtained g
+00000700: 656e 6f6d 6520 6173 7365 6d62 6c79 2069  enome assembly i
+00000710: 6e20 6861 6e64 2e20 466f 7220 7468 6f73  n hand. For thos
+00000720: 6520 7768 6f20 6172 6520 696e 7465 7265  e who are intere
+00000730: 7374 6564 2069 6e20 6d6f 7265 2074 6865  sted in more the
+00000740: 6f72 6574 6963 616c 2064 6574 6169 6c73  oretical details
+00000750: 2c20 7765 2072 6563 6f6d 6d65 6e64 2074  , we recommend t
+00000760: 7572 6e69 6e67 2074 6f20 6f75 7220 7061  urning to our pa
+00000770: 7065 7220 616e 6420 626f 6f6b 2063 6861  per and book cha
+00000780: 7074 6572 2066 6f72 206d 6f72 6520 6465  pter for more de
+00000790: 7461 696c 6564 2064 6573 6372 6970 7469  tailed descripti
+000007a0: 6f6e 2061 6e64 2069 6e73 6967 6874 6675  on and insightfu
+000007b0: 6c20 6469 7363 7573 7369 6f6e 732e 2054  l discussions. T
+000007c0: 6865 206b 6579 2069 6d70 726f 7665 6420  he key improved 
+000007d0: 6665 6174 7572 6573 206f 6620 6077 6764  features of `wgd
+000007e0: 2076 3260 2061 7265 2064 656d 6f6e 7374   v2` are demonst
+000007f0: 7261 7465 6420 696e 206f 7572 206c 6174  rated in our lat
+00000800: 6573 7420 6d61 6e75 7363 7269 7074 206f  est manuscript o
+00000810: 6e20 5b42 696f 696e 666f 726d 6174 6963  n [Bioinformatic
+00000820: 735d 2868 7474 7073 3a2f 2f64 6f69 2e6f  s](https://doi.o
+00000830: 7267 2f31 302e 3130 3933 2f62 696f 696e  rg/10.1093/bioin
+00000840: 666f 726d 6174 6963 732f 6274 6165 3237  formatics/btae27
+00000850: 3229 2e20 4966 2079 6f75 2075 7365 2060  2). If you use `
+00000860: 7767 6420 7632 6020 696e 2079 6f75 7220  wgd v2` in your 
+00000870: 7265 7365 6172 6368 2c20 706c 6561 7365  research, please
+00000880: 2063 6974 6520 7573 2061 7320 7375 6767   cite us as sugg
+00000890: 6573 7465 6420 696e 205b 4369 7461 7469  ested in [Citati
+000008a0: 6f6e 5d28 2363 6974 6174 696f 6e29 2073  on](#citation) s
+000008b0: 6563 7469 6f6e 2e0a 0a23 2320 496e 7472  ection...## Intr
+000008c0: 6f64 7563 7469 6f6e 0a0a 506f 6c79 706c  oduction..Polypl
+000008d0: 6f69 6469 7a61 7469 6f6e 732c 2074 6865  oidizations, the
+000008e0: 2065 766f 6c75 7469 6f6e 6172 7920 7072   evolutionary pr
+000008f0: 6f63 6573 7320 7468 6174 2074 6865 2065  ocess that the e
+00000900: 6e74 6972 6520 6765 6e6f 6d65 206f 6620  ntire genome of 
+00000910: 616e 206f 7267 616e 6973 6d20 6973 2064  an organism is d
+00000920: 7570 6c69 6361 7465 642c 2061 6c73 6f20  uplicated, also 
+00000930: 6e61 6d65 6420 6173 2077 686f 6c65 2d67  named as whole-g
+00000940: 656e 6f6d 6520 6475 706c 6963 6174 696f  enome duplicatio
+00000950: 6e73 2028 5747 4473 292c 206f 6363 7572  ns (WGDs), occur
+00000960: 2072 6563 7572 7265 6e74 6c79 2061 6372   recurrently acr
+00000970: 6f73 7320 7468 6520 7472 6565 206f 6620  oss the tree of 
+00000980: 6c69 6665 2e20 5468 6572 6520 6172 6520  life. There are 
+00000990: 7477 6f20 6d6f 6465 7320 6f66 2070 6f6c  two modes of pol
+000009a0: 7970 6c6f 6964 697a 6174 696f 6e73 2c20  yploidizations, 
+000009b0: 6175 746f 706f 6c79 706c 6f69 6469 7a61  autopolyploidiza
+000009c0: 7469 6f6e 7320 616e 6420 616c 6c6f 706f  tions and allopo
+000009d0: 6c79 706c 6f69 6469 7a61 7469 6f6e 732e  lyploidizations.
+000009e0: 2041 7574 6f70 6f6c 7970 6c6f 6964 697a   Autopolyploidiz
+000009f0: 6174 696f 6e73 2061 7265 2074 6865 2064  ations are the d
+00000a00: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
+00000a10: 6520 7361 6d65 2067 656e 6f6d 652c 2072  e same genome, r
+00000a20: 6573 756c 7469 6e67 2069 6e20 7477 6f20  esulting in two 
+00000a30: 6964 656e 7469 6361 6c20 7375 6267 656e  identical subgen
+00000a40: 6f6d 6573 2061 7420 7468 6520 7469 6d65  omes at the time
+00000a50: 2069 7420 656d 6572 6765 642e 2057 6869   it emerged. Whi
+00000a60: 6c65 2074 6865 2061 6c6c 6f70 6f6c 7970  le the allopolyp
+00000a70: 6c6f 6964 697a 6174 696f 6e73 2061 7265  loidizations are
+00000a80: 206e 6f72 6d61 6c6c 7920 6163 6869 6576   normally achiev
+00000a90: 6564 2069 6e20 7477 6f20 7374 6570 732c  ed in two steps,
+00000aa0: 2066 6972 7374 2074 6865 2068 7962 7269   first the hybri
+00000ab0: 6469 7a61 7469 6f6e 2062 6574 7765 656e  dization between
+00000ac0: 2074 776f 2064 6966 6665 7265 6e74 2073   two different s
+00000ad0: 7065 6369 6573 2c20 7265 7375 6c74 696e  pecies, resultin
+00000ae0: 6720 696e 2074 6865 2061 7269 7369 6e67  g in the arising
+00000af0: 206f 6620 7472 616e 7369 656e 7420 686f   of transient ho
+00000b00: 6d6f 706c 6f69 6479 2c73 6563 6f6e 6420  moploidy,second 
+00000b10: 7468 6520 6475 706c 6963 6174 696f 6e20  the duplication 
+00000b20: 6f66 2074 6865 2068 6f6d 6f70 6c6f 6964  of the homoploid
+00000b30: 792c 2072 6573 756c 7469 6e67 2069 6e20  y, resulting in 
+00000b40: 7468 6520 656d 6572 6765 6e63 6520 6f66  the emergence of
+00000b50: 2061 6c6c 6f70 6f6c 7970 6c6f 6964 792e   allopolyploidy.
+00000b60: 2044 7565 2074 6f20 7468 6520 756e 7374   Due to the unst
+00000b70: 6162 696c 6974 7920 616e 6420 756e 6261  ability and unba
+00000b80: 6c61 6e63 6564 2074 6574 7261 736f 6d69  lanced tetrasomi
+00000b90: 6320 696e 6865 7269 7461 6e63 652c 2066  c inheritance, f
+00000ba0: 6f72 2069 6e73 7461 6e63 6520 7468 6520  or instance the 
+00000bb0: 6e75 636c 6561 722d 6379 746f 706c 6173  nuclear-cytoplas
+00000bc0: 6d69 6320 696e 636f 6d70 6174 6962 696c  mic incompatibil
+00000bd0: 6974 792c 2074 6865 2070 6f6c 7970 6c6f  ity, the polyplo
+00000be0: 6964 7920 6765 6e6f 6d65 2077 696c 6c20  idy genome will 
+00000bf0: 7468 656e 2065 7870 6572 6965 6e63 6520  then experience 
+00000c00: 6120 7072 6f63 6573 7320 6361 6c6c 6564  a process called
+00000c10: 2064 6970 6c6f 6964 697a 6174 696f 6e2c   diploidization,
+00000c20: 2061 6c73 6f20 6e61 6d65 6420 6173 2066   also named as f
+00000c30: 7261 6374 696f 6e61 7469 6f6e 2c20 6475  ractionation, du
+00000c40: 7269 6e67 2077 6869 6368 2061 206c 6172  ring which a lar
+00000c50: 6765 2070 6f72 7469 6f6e 206f 6620 6765  ge portion of ge
+00000c60: 6e65 2064 7570 6c69 6361 7465 7320 7769  ne duplicates wi
+00000c70: 6c6c 2067 6574 206c 6f73 7420 616e 6420  ll get lost and 
+00000c80: 6f6e 6c79 2061 2066 7261 6374 696f 6e20  only a fraction 
+00000c90: 6361 6e20 6265 2072 6574 6169 6e65 642e  can be retained.
+00000ca0: 2054 6865 2074 7261 6365 7320 6f66 2070   The traces of p
+00000cb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
+00000cc0: 2063 616e 2062 6520 7468 7573 2075 6e65   can be thus une
+00000cd0: 6172 7468 6564 2066 726f 6d20 7468 6573  arthed from thes
+00000ce0: 6520 7265 7461 696e 6564 2067 656e 6520  e retained gene 
+00000cf0: 6475 706c 6963 6174 6573 2e20 5468 7265  duplicates. Thre
+00000d00: 6520 6170 7072 6f61 6368 6573 2062 6173  e approaches bas
+00000d10: 6564 206f 6e20 6765 6e65 2064 7570 6c69  ed on gene dupli
+00000d20: 6361 7465 732c 206e 616d 656c 792c 202a  cates, namely, *
+00000d30: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
+00000d40: 6574 686f 642c 2067 656e 6520 7472 6565  ethod, gene tree
+00000d50: 202d 2073 7065 6369 6573 2074 7265 6520   - species tree 
+00000d60: 7265 636f 6e63 696c 6961 7469 6f6e 206d  reconciliation m
+00000d70: 6574 686f 6420 616e 6420 7379 6e74 656e  ethod and synten
+00000d80: 7920 6d65 7468 6f64 2c20 6172 6520 636f  y method, are co
+00000d90: 6d6d 6f6e 6c79 2075 7365 6420 696e 2064  mmonly used in d
+00000da0: 6574 6563 7469 6e67 2065 7669 6465 6e63  etecting evidenc
+00000db0: 6520 666f 7220 5747 4473 2e20 5468 6520  e for WGDs. The 
+00000dc0: 6765 6e65 2074 7265 6520 2d20 7370 6563  gene tree - spec
+00000dd0: 6965 7320 7472 6565 2072 6563 6f6e 6369  ies tree reconci
+00000de0: 6c69 6174 696f 6e20 6d65 7468 6f64 2069  liation method i
+00000df0: 7320 6e6f 7420 7769 7468 696e 2074 6865  s not within the
+00000e00: 2073 636f 7065 206f 6620 6077 6764 2076   scope of `wgd v
+00000e10: 3260 2c20 6275 7420 7765 206b 696e 646c  2`, but we kindl
+00000e20: 7920 7265 6665 7220 7265 6164 6572 7320  y refer readers 
+00000e30: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
+00000e40: 6564 2074 6f20 7468 6520 7068 796c 6f67  ed to the phylog
+00000e50: 656e 6f6d 6963 2070 726f 6772 616d 2064  enomic program d
+00000e60: 6576 656c 6f70 6564 2062 7920 4172 7468  eveloped by Arth
+00000e70: 7572 205a 7761 656e 6570 6f65 6c20 6e61  ur Zwaenepoel na
+00000e80: 6d65 6420 5b57 4841 4c45 5d28 6874 7470  med [WHALE](http
+00000e90: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f61  s://github.com/a
+00000ea0: 727a 7761 2f57 6861 6c65 2e6a 6c29 2061  rzwa/Whale.jl) a
+00000eb0: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
+00000ec0: 6420 5b70 6170 6572 5d28 6874 7470 733a  d [paper](https:
+00000ed0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
+00000ee0: 332f 6d6f 6c62 6576 2f6d 737a 3038 3829  3/molbev/msz088)
+00000ef0: 2066 6f72 206d 6f72 6520 7465 6368 6e69   for more techni
+00000f00: 6361 6c20 616e 6420 7468 656f 7265 7469  cal and theoreti
+00000f10: 6361 6c20 6465 7461 696c 732e 0a0a 5468  cal details...Th
+00000f20: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00000f30: 3e20 6d65 7468 6f64 2069 7320 6573 7461  > method is esta
+00000f40: 626c 6973 6865 6420 6f6e 2061 206d 6f64  blished on a mod
+00000f50: 656c 206f 6620 6765 6e65 2066 616d 696c  el of gene famil
+00000f60: 7920 6576 6f6c 7574 696f 6e20 7468 6174  y evolution that
+00000f70: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
+00000f80: 7920 6973 2061 6c6c 6f77 6564 2074 6f20  y is allowed to 
+00000f90: 6576 6f6c 7665 2076 6961 2067 656e 6520  evolve via gene 
+00000fa0: 6475 706c 6963 6174 696f 6e20 616e 6420  duplication and 
+00000fb0: 6c6f 7373 2e20 4e6f 7465 2074 6861 7420  loss. Note that 
+00000fc0: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
+00000fd0: 6865 7265 2069 7320 6173 7375 6d65 6420  here is assumed 
+00000fe0: 746f 2062 6520 7468 6520 636c 7573 7465  to be the cluste
+00000ff0: 7220 6f66 2061 6c6c 2067 656e 6573 2064  r of all genes d
+00001000: 6573 6365 6e64 6564 2066 726f 6d20 616e  escended from an
+00001010: 2061 6e63 6573 7472 616c 2067 656e 6520   ancestral gene 
+00001020: 696e 2061 2073 696e 676c 6520 6765 6e6f  in a single geno
+00001030: 6d65 2e20 5265 636f 7665 7269 6e67 2074  me. Recovering t
+00001040: 6865 2067 656e 6520 7472 6565 206f 6620  he gene tree of 
+00001050: 7375 6368 2067 656e 6520 6661 6d69 6c79  such gene family
+00001060: 2069 6e66 6f72 6d73 2074 6865 2074 696d   informs the tim
+00001070: 696e 672c 2073 6369 6c69 6365 7420 7468  ing, scilicet th
+00001080: 6520 6167 652c 206f 6620 6765 6e65 2064  e age, of gene d
+00001090: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
+000010a0: 732e 2054 6865 2061 6765 2072 6566 6572  s. The age refer
+000010b0: 6564 2068 6572 652c 2069 7320 6e6f 7420  ed here, is not 
+000010c0: 696e 2072 6561 6c20 6765 6f6c 6f67 6963  in real geologic
+000010d0: 616c 2074 696d 652c 2062 7574 2069 6e20  al time, but in 
+000010e0: 7468 6520 756e 6974 206f 6620 6576 6f6c  the unit of evol
+000010f0: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
+00001100: 652c 2069 2e65 2e2c 2074 6865 206e 756d  e, i.e., the num
+00001110: 6265 7220 6f66 2073 7562 7374 6974 7574  ber of substitut
+00001120: 696f 6e73 2070 6572 2073 6974 652e 2057  ions per site. W
+00001130: 6865 6e20 7468 6520 6576 6f6c 7574 696f  hen the evolutio
+00001140: 6e61 7279 2072 6174 6520 7265 6d61 696e  nary rate remain
+00001150: 7320 6170 7072 6f78 696d 6174 656c 7920  s approximately 
+00001160: 636f 6e73 7461 6e74 2c20 7468 6520 6576  constant, the ev
+00001170: 6f6c 7574 696f 6e61 7279 2064 6973 7461  olutionary dista
+00001180: 6e63 6520 6973 2074 6865 6e20 7375 7070  nce is then supp
+00001190: 6f73 6564 2074 6f20 6265 2070 726f 706f  osed to be propo
+000011a0: 7274 696f 6e61 6c20 746f 2074 6865 2072  rtional to the r
+000011b0: 6561 6c20 6576 6f6c 7574 696f 6e61 7279  eal evolutionary
+000011c0: 2074 696d 652e 2054 6865 2073 796e 6f6e   time. The synon
+000011d0: 796d 6f75 7320 6469 7374 616e 6365 202a  ymous distance *
+000011e0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2c20  K*<sub>S</sub>, 
+000011f0: 7468 6520 6e75 6d62 6572 206f 6620 7379  the number of sy
+00001200: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
+00001210: 7574 696f 6e73 2070 6572 2073 796e 6f6e  utions per synon
+00001220: 796d 6f75 7320 7369 7465 2c20 6973 2073  ymous site, is s
+00001230: 7563 6820 6361 6e64 6964 6174 6520 7468  uch candidate th
+00001240: 6174 2073 796e 6f6e 796d 6f75 7320 7375  at synonymous su
+00001250: 6273 7469 7475 7469 6f6e 7320 776f 756c  bstitutions woul
+00001260: 6420 6e6f 7420 696e 6375 7220 7468 6520  d not incur the 
+00001270: 6368 616e 6765 206f 6620 616d 696e 6f20  change of amino 
+00001280: 6163 6964 2061 6e64 2061 7265 2074 6875  acid and are thu
+00001290: 7320 7265 6761 7264 6564 2061 7320 6e65  s regarded as ne
+000012a0: 7574 7261 6c2c 2077 6869 6368 2061 6363  utral, which acc
+000012b0: 6f72 6469 6e67 2074 6f20 7468 6520 6e65  ording to the ne
+000012c0: 7574 7261 6c20 7468 656f 7279 2073 686f  utral theory sho
+000012d0: 756c 6420 6f63 6375 7220 696e 2063 6f6e  uld occur in con
+000012e0: 7374 616e 7420 7261 7465 2e20 4769 7665  stant rate. Give
+000012f0: 6e20 6120 6d6f 6465 6c20 6f66 2067 656e  n a model of gen
+00001300: 6520 6661 6d69 6c79 2074 6861 7420 616c  e family that al
+00001310: 6c6f 7773 2074 6865 2067 656e 6520 746f  lows the gene to
+00001320: 2064 7570 6c69 6361 7465 2061 6e64 2067   duplicate and g
+00001330: 6574 206c 6f73 7420 696e 2061 2066 6978  et lost in a fix
+00001340: 6564 2072 6174 652c 206f 6e65 2063 616e  ed rate, one can
+00001350: 2064 6572 6976 6520 7468 6174 2074 6865   derive that the
+00001360: 2070 726f 6261 6269 6c69 7479 2064 656e   probability den
+00001370: 7369 7479 2066 756e 6374 696f 6e20 6f66  sity function of
+00001380: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+00001390: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+000013a0: 7574 696f 6e20 6f66 2072 6574 6169 6e65  ution of retaine
+000013b0: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
+000013c0: 7320 6973 2061 2071 7561 7369 2d65 7870  s is a quasi-exp
+000013d0: 6f6e 656e 7469 616c 2066 756e 6374 696f  onential functio
+000013e0: 6e20 7468 6174 206d 6f73 7420 7265 7461  n that most reta
+000013f0: 696e 6564 2067 656e 6520 6475 706c 6963  ined gene duplic
+00001400: 6174 6573 2061 7265 2072 6563 656e 746c  ates are recentl
+00001410: 7920 626f 726e 6564 2077 6974 6820 7e30  y borned with ~0
+00001420: 2061 6765 2077 6869 6c65 2061 7320 7468   age while as th
+00001430: 6520 6167 6520 676f 696e 6720 6f6c 6465  e age going olde
+00001440: 7220 7468 6520 6173 736f 6369 6174 6564  r the associated
+00001450: 206e 756d 6265 7220 6f66 2072 6574 6169   number of retai
+00001460: 6e65 6420 6765 6e65 2064 7570 6c69 6361  ned gene duplica
+00001470: 7465 7320 6465 6361 7920 7175 6173 692d  tes decay quasi-
+00001480: 6578 706f 6e65 6e74 6961 6c6c 792e 2054  exponentially. T
+00001490: 6865 7265 666f 7265 2c20 7468 6520 6f63  herefore, the oc
+000014a0: 6375 7261 6e63 6520 6f66 206c 6172 6765  curance of large
+000014b0: 2d73 6361 6c65 2067 656e 6520 6475 706c  -scale gene dupl
+000014c0: 6963 6174 696f 6e20 6576 656e 7473 2c20  ication events, 
+000014d0: 666f 7220 696e 7374 616e 6520 5747 4473  for instane WGDs
+000014e0: 2c20 7769 7468 2076 6172 6965 6420 7265  , with varied re
+000014f0: 7465 6e74 696f 6e20 7261 7465 2c20 7769  tention rate, wi
+00001500: 6c6c 206c 6561 7665 2061 6e20 6167 6520  ll leave an age 
+00001510: 7065 616b 2066 726f 6d20 7468 6520 6275  peak from the bu
+00001520: 7273 7420 6f66 2067 656e 6520 6475 706c  rst of gene dupl
+00001530: 6963 6174 6573 2069 6e20 6120 7368 6f72  icates in a shor
+00001540: 7420 7469 6d65 2d66 7261 6d65 2075 706f  t time-frame upo
+00001550: 6e20 7468 6520 696e 6974 6961 6c20 6167  n the initial ag
+00001560: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
+00001570: 616e 6420 6361 6e20 6265 2075 6e76 6569  and can be unvei
+00001580: 6c65 6420 6672 6f6d 206d 6978 7475 7265  led from mixture
+00001590: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
+000015a0: 6973 2e20 486f 7765 7665 722c 2057 4744  is. However, WGD
+000015b0: 7320 6964 656e 7469 6669 6564 2066 726f  s identified fro
+000015c0: 6d20 7468 6520 7061 7261 6c6f 676f 7573  m the paralogous
+000015d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000015e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+000015f0: 6e73 2063 616e 206f 6e6c 7920 696e 666f  ns can only info
+00001600: 726d 2074 6865 2057 4744 2074 696d 696e  rm the WGD timin
+00001610: 6720 696e 2074 6865 2074 696d 652d 7363  g in the time-sc
+00001620: 616c 6520 6f66 2074 6861 7420 7370 6563  ale of that spec
+00001630: 6966 6963 2073 7065 6369 6573 2c20 7768  ific species, wh
+00001640: 6963 6820 6973 206e 6f74 2063 6f6d 7061  ich is not compa
+00001650: 7261 626c 6520 696e 2074 6865 2070 6879  rable in the phy
+00001660: 6c6f 6765 6e65 7469 6320 636f 6e74 6578  logenetic contex
+00001670: 742e 204f 6e6c 7920 7769 7468 2074 6865  t. Only with the
+00001680: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
+00001690: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+000016a0: 2064 6973 7472 6962 7574 696f 6e73 2c20   distributions, 
+000016b0: 7768 6963 6820 636f 6e76 6572 7420 7468  which convert th
+000016c0: 6520 6573 7469 6d61 7465 6420 626f 6479  e estimated body
+000016d0: 2066 726f 6d20 7061 7261 6c6f 6775 6573   from paralogues
+000016e0: 2074 6f20 6f72 7468 6f6c 6f67 7565 7320   to orthologues 
+000016f0: 616e 6420 696e 666f 726d 2074 6865 2072  and inform the r
+00001700: 656c 6174 6976 6520 7469 6d69 6e67 206f  elative timing o
+00001710: 6620 7370 6563 6961 7469 6f6e 2065 7665  f speciation eve
+00001720: 6e74 732c 2063 616e 2077 6520 6465 6369  nts, can we deci
+00001730: 7068 6572 2074 6865 2070 6879 6c6f 6765  pher the phyloge
+00001740: 6e65 7469 6320 706c 6163 656d 656e 7420  netic placement 
+00001750: 6f66 2057 4744 7320 6166 7465 7220 7072  of WGDs after pr
+00001760: 6f70 6572 2072 6174 6520 636f 7272 6563  oper rate correc
+00001770: 7469 6f6e 2e20 6077 6764 2076 3260 2069  tion. `wgd v2` i
+00001780: 7320 7375 6368 2070 726f 6772 616d 2074  s such program t
+00001790: 6861 7420 6865 6c70 7320 7573 6572 7320  hat helps users 
+000017a0: 636f 6e73 7472 7563 7420 7061 7261 6c6f  construct paralo
+000017b0: 676f 7573 2061 6e64 206f 7274 686f 6c6f  gous and ortholo
+000017c0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000017d0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+000017e0: 7574 696f 6e73 2061 6e64 2072 6561 6c69  utions and reali
+000017f0: 7a65 2062 6f74 6820 7468 6520 6964 656e  ze both the iden
+00001800: 7469 6669 6361 7469 6f6e 2061 6e64 2070  tification and p
+00001810: 6c61 6365 6d65 6e74 206f 6620 5747 4473  lacement of WGDs
+00001820: 2e0a 0a49 6e20 7468 6520 7072 656d 6973  ...In the premis
+00001830: 6520 6f66 2070 6879 6c6f 6765 6e65 7469  e of phylogeneti
+00001840: 6361 6c6c 7920 6c6f 6361 7465 6420 5747  cally located WG
+00001850: 4473 2c20 7468 6520 6162 736f 6c75 7465  Ds, the absolute
+00001860: 2061 6765 2028 696e 2067 656f 6c6f 6769   age (in geologi
+00001870: 6361 6c20 7469 6d65 2920 6f66 2057 4744  cal time) of WGD
+00001880: 7320 6361 6e20 616c 736f 2062 6520 696e  s can also be in
+00001890: 6665 7272 6564 2066 726f 6d20 7468 6f73  ferred from thos
+000018a0: 6520 5747 442d 7265 7461 696e 6564 2067  e WGD-retained g
+000018b0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
+000018c0: 616c 7468 6f75 6768 2074 6865 7265 2068  although there h
+000018d0: 6173 2062 6565 6e20 6e6f 2065 6173 7920  as been no easy 
+000018e0: 6f72 2073 7472 6169 6768 7466 6f72 7761  or straightforwa
+000018f0: 7264 2070 6970 656c 696e 6520 666f 7220  rd pipeline for 
+00001900: 7468 6973 206a 6f62 2073 6f20 6661 722e  this job so far.
+00001910: 2049 6e20 6077 6764 2076 3260 2c20 7765   In `wgd v2`, we
+00001920: 2064 6576 656c 6f70 6564 2061 2066 6561   developed a fea
+00001930: 7369 626c 6520 696e 7465 6772 6174 6564  sible integrated
+00001940: 2070 6970 6c65 6c69 6e65 2066 6f72 2061   pipleline for a
+00001950: 6273 6f6c 7574 6520 6461 7469 6e67 206f  bsolute dating o
+00001960: 6620 5747 4473 2e20 5468 6520 7069 7065  f WGDs. The pipe
+00001970: 6c69 6e65 2063 616e 2062 6520 726f 7567  line can be roug
+00001980: 686c 7920 6469 7669 6465 6420 696e 746f  hly divided into
+00001990: 2074 6872 6565 206d 6169 6e20 7374 6570   three main step
+000019a0: 732e 2031 2920 5468 6520 636f 6e73 7472  s. 1) The constr
+000019b0: 7563 7469 6f6e 206f 6620 616e 6368 6f72  uction of anchor
+000019c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000019d0: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+000019e0: 6420 7468 6520 6465 6c69 6e65 6174 696f  d the delineatio
+000019f0: 6e20 6f66 2063 7265 6469 6162 6c65 202a  n of crediable *
+00001a00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
+00001a10: 616e 6765 2061 646f 7074 6564 2066 6f72  ange adopted for
+00001a20: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+00001a30: 7469 6e67 2c20 7573 696e 6720 6077 6764  ting, using `wgd
+00001a40: 2064 6d64 602c 2060 7767 6420 6b73 6460   dmd`, `wgd ksd`
+00001a50: 2c20 6077 6764 2073 796e 6020 616e 6420  , `wgd syn` and 
+00001a60: 6077 6764 2070 6561 6b60 2e20 4e6f 7465  `wgd peak`. Note
+00001a70: 2074 6861 7420 6865 7265 2077 6520 6f6e   that here we on
+00001a80: 6c79 2063 6f6e 7369 6465 7220 6765 6e6f  ly consider geno
+00001a90: 6d65 2061 7373 656d 626c 7920 6265 6361  me assembly beca
+00001aa0: 7573 6520 666f 7220 7472 616e 7363 7269  use for transcri
+00001ab0: 7074 6f6d 6520 6173 7365 6d62 6c79 2069  ptome assembly i
+00001ac0: 7427 7320 696d 706f 7373 6962 6c65 2074  t's impossible t
+00001ad0: 6f20 6469 7374 696e 6775 6973 6820 5747  o distinguish WG
+00001ae0: 442d 6465 7269 7665 6420 6475 706c 6963  D-derived duplic
+00001af0: 6174 6573 2066 726f 6d20 736d 616c 6c2d  ates from small-
+00001b00: 7363 616c 6520 6475 706c 6963 6174 696f  scale duplicatio
+00001b10: 6e2d 6465 7269 7665 6420 6475 706c 6963  n-derived duplic
+00001b20: 6174 6573 2c20 7768 6963 6820 6861 7070  ates, which happ
+00001b30: 656e 6564 2069 6e20 6120 636f 6e74 696e  ened in a contin
+00001b40: 756f 7573 2074 696d 652d 6672 616d 6520  uous time-frame 
+00001b50: 696e 7374 6561 6420 6f66 206f 6e6c 7920  instead of only 
+00001b60: 6120 7365 7061 7261 7465 2073 686f 7274  a separate short
+00001b70: 2074 696d 652d 6672 616d 6520 616e 6420   time-frame and 
+00001b80: 7468 7573 2072 6566 6c65 6374 7320 7468  thus reflects th
+00001b90: 6520 6475 7261 7469 6f6e 206f 6620 7468  e duration of th
+00001ba0: 6174 2062 7261 6e63 6820 7261 7468 6572  at branch rather
+00001bb0: 2074 6861 6e20 7468 6520 7469 6d65 2061   than the time a
+00001bc0: 7420 7768 6963 6820 5747 4420 6f63 6375  t which WGD occu
+00001bd0: 7272 6564 2e20 3229 2054 6865 2066 6f72  rred. 2) The for
+00001be0: 6d75 6c61 7469 6f6e 206f 6620 6120 7374  mulation of a st
+00001bf0: 6172 7469 6e67 2074 7265 6520 7573 6564  arting tree used
+00001c00: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
+00001c10: 6574 6963 2064 6174 696e 672c 2063 6f6d  etic dating, com
+00001c20: 706f 7365 6420 6f66 2061 2066 6577 2073  posed of a few s
+00001c30: 7065 6369 6573 2061 6e64 2061 6e6e 6f74  pecies and annot
+00001c40: 6174 6564 2077 6974 6820 666f 7373 696c  ated with fossil
+00001c50: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
+00001c60: 6f72 6d61 7469 6f6e 2e20 5468 6973 2073  ormation. This s
+00001c70: 7465 7020 6973 2065 7373 656e 7469 616c  tep is essential
+00001c80: 2066 6f72 2074 6865 2072 6573 756c 7420   for the result 
+00001c90: 6f66 2061 6273 6f6c 7574 6520 5747 4420  of absolute WGD 
+00001ca0: 6461 7469 6e67 2074 6861 7420 7765 2073  dating that we s
+00001cb0: 7567 6765 7374 2075 7365 7273 206f 6620  uggest users of 
+00001cc0: 7461 6b69 6e67 2067 7265 6174 2063 6175  taking great cau
+00001cd0: 7469 6f6e 2074 6f20 6173 7375 7265 2074  tion to assure t
+00001ce0: 6865 2074 7265 6520 746f 706f 6c6f 6779  he tree topology
+00001cf0: 2061 6e64 2070 726f 7065 7220 626f 756e   and proper boun
+00001d00: 6473 2066 6f72 2066 6f73 7369 6c20 6361  ds for fossil ca
+00001d10: 6c69 6272 6174 696f 6e73 2e20 3329 2054  librations. 3) T
+00001d20: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+00001d30: 6f66 206f 7274 686f 6772 6f75 7073 2063  of orthogroups c
+00001d40: 6f6e 7369 7374 696e 6720 6f66 2063 6f6c  onsisting of col
+00001d50: 6c69 6e65 6172 2064 7570 6c69 6361 7465  linear duplicate
+00001d60: 7320 6f66 2074 6865 2066 6f63 616c 2073  s of the focal s
+00001d70: 7065 6369 6573 2061 6e64 2074 6865 6972  pecies and their
+00001d80: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
+00001d90: 2068 6974 7320 2852 4248 7329 2061 6761   hits (RBHs) aga
+00001da0: 696e 7374 206f 7468 6572 2073 7065 6369  inst other speci
+00001db0: 6573 2069 6e20 7468 6520 7374 6172 7469  es in the starti
+00001dc0: 6e67 2074 7265 6520 616e 6420 7468 6520  ng tree and the 
+00001dd0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+00001de0: 696e 6720 7573 696e 6720 6120 6d6f 6c65  ing using a mole
+00001df0: 6375 6c61 7220 6461 7469 6e67 2070 726f  cular dating pro
+00001e00: 6772 616d 2066 6f72 2069 6e73 7461 6e63  gram for instanc
+00001e10: 6520 606d 636d 6374 7265 6560 2c20 7669  e `mcmctree`, vi
+00001e20: 6120 7072 6f67 7261 6d20 6077 6764 2064  a program `wgd d
+00001e30: 6d64 6020 616e 6420 6077 6764 2066 6f63  md` and `wgd foc
+00001e40: 7573 602e 2057 6520 7265 636f 6d6d 656e  us`. We recommen
+00001e50: 6420 7468 6520 7573 6167 6520 6f66 2042  d the usage of B
+00001e60: 6179 6573 6961 6e20 6d6f 6c65 6375 6c61  ayesian molecula
+00001e70: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
+00001e80: 2060 6d63 6d63 7472 6565 602c 2077 6869   `mcmctree`, whi
+00001e90: 6368 2070 726f 7669 6465 7320 6120 7661  ch provides a va
+00001ea0: 7269 6574 7920 6f66 2064 6966 6665 7265  riety of differe
+00001eb0: 6e74 2073 7562 7374 6974 7574 696f 6e20  nt substitution 
+00001ec0: 616e 6420 7261 7465 206d 6f64 656c 732e  and rate models.
+00001ed0: 204e 6f6e 6574 6865 6c65 7373 2c20 7765   Nonetheless, we
+00001ee0: 2075 7267 6520 7573 6572 7320 746f 2073   urge users to s
+00001ef0: 6574 2070 7269 6f72 2064 6973 7472 6962  et prior distrib
+00001f00: 7574 696f 6e20 6f66 2064 6966 6665 7265  ution of differe
+00001f10: 6e74 2070 6172 616d 6574 6572 7320 7769  nt parameters wi
+00001f20: 7468 2063 6175 7469 6f6e 2061 6e64 2061  th caution and a
+00001f30: 7373 7572 6520 6164 6571 7561 7465 2073  ssure adequate s
+00001f40: 616d 706c 696e 6720 6f66 2064 6966 6665  ampling of diffe
+00001f50: 7265 6e74 2070 6172 616d 6574 6572 732e  rent parameters.
+00001f60: 0a0a 2323 2049 6e73 7461 6c6c 6174 696f  ..## Installatio
+00001f70: 6e0a 0a54 6865 2065 6173 6965 7374 2077  n..The easiest w
+00001f80: 6179 2074 6f20 696e 7374 616c 6c20 6077  ay to install `w
+00001f90: 6764 2076 3260 2069 7320 7573 696e 6720  gd v2` is using 
+00001fa0: 5059 5049 2e20 4e6f 7465 2074 6861 7420  PYPI. Note that 
+00001fb0: 6966 2079 6f75 2077 616e 7420 746f 2067  if you want to g
+00001fc0: 6574 2074 6865 206c 6174 6573 7420 7570  et the latest up
+00001fd0: 6461 7465 2c20 7765 2073 7567 6765 7374  date, we suggest
+00001fe0: 2069 6e73 7461 6c6c 696e 6720 6672 6f6d   installing from
+00001ff0: 2074 6865 2073 6f75 7263 652c 2073 696e   the source, sin
+00002000: 6365 2074 6865 2075 7064 6174 6520 6f6e  ce the update on
+00002010: 2050 5950 4920 7769 6c6c 2062 6520 6465   PYPI will be de
+00002020: 6c61 7965 6420 636f 6d70 6172 6564 2074  layed compared t
+00002030: 6f20 6865 7265 206f 6620 736f 7572 6365  o here of source
+00002040: 2e0a 0a60 6060 0a70 6970 2069 6e73 7461  ...```.pip insta
+00002050: 6c6c 2077 6764 0a60 6060 0a0a 546f 2069  ll wgd.```..To i
+00002060: 6e73 7461 6c6c 2060 7767 6420 7632 6020  nstall `wgd v2` 
+00002070: 696e 2061 2076 6972 7475 616c 2065 6e76  in a virtual env
+00002080: 6972 6f6e 6d65 6e74 2c20 7468 6520 666f  ironment, the fo
+00002090: 6c6c 6f77 696e 6720 636f 6d6d 616e 6420  llowing command 
+000020a0: 6c69 6e65 7320 636f 756c 6420 6265 2075  lines could be u
+000020b0: 7365 642e 0a0a 6060 600a 6769 7420 636c  sed...```.git cl
+000020c0: 6f6e 6520 3c77 6764 2072 6570 6f3e 0a63  one <wgd repo>.c
+000020d0: 6420 7767 640a 7669 7274 7561 6c65 6e76  d wgd.virtualenv
+000020e0: 202d 703d 7079 7468 6f6e 3320 454e 5620   -p=python3 ENV 
+000020f0: 286f 7220 7079 7468 6f6e 3320 2d6d 2076  (or python3 -m v
+00002100: 656e 7620 454e 5629 0a73 6f75 7263 6520  env ENV).source 
+00002110: 454e 562f 6269 6e2f 6163 7469 7661 7465  ENV/bin/activate
+00002120: 0a70 6970 2069 6e73 7461 6c6c 202d 7220  .pip install -r 
+00002130: 7265 7175 6972 656d 656e 7473 2e74 7874  requirements.txt
+00002140: 0a70 6970 2069 6e73 7461 6c6c 202e 0a60  .pip install ..`
+00002150: 6060 0a0a 5768 656e 206d 6574 2077 6974  ``..When met wit
+00002160: 6820 7065 726d 6973 7369 6f6e 2070 726f  h permission pro
+00002170: 626c 656d 2069 6e20 696e 7374 616c 6c61  blem in installa
+00002180: 7469 6f6e 2c20 706c 6561 7365 2074 7279  tion, please try
+00002190: 2074 6865 2066 6f6c 6c6f 7769 6e67 2063   the following c
+000021a0: 6f6d 6d61 6e64 206c 696e 652e 0a0a 6060  ommand line...``
+000021b0: 600a 7069 7020 696e 7374 616c 6c20 2d65  `.pip install -e
+000021c0: 202e 0a60 6060 0a0a 4966 206d 756c 7469   ..```..If multi
+000021d0: 706c 7920 7665 7273 696f 6e73 206f 6620  ply versions of 
+000021e0: 6077 6764 6020 7765 7265 2069 6e73 7461  `wgd` were insta
+000021f0: 6c6c 6564 2069 6e20 7468 6520 7379 7374  lled in the syst
+00002200: 656d 2c20 706c 6561 7365 2061 6464 2074  em, please add t
+00002210: 6865 2072 6967 6874 2070 6174 6820 6f66  he right path of
+00002220: 2069 6e74 6572 6573 7465 6420 7665 7273   interested vers
+00002230: 696f 6e20 696e 746f 2074 6865 2065 6e76  ion into the env
+00002240: 6972 6f6e 6d65 6e74 2076 6172 6961 626c  ironment variabl
+00002250: 6573 2c20 666f 7220 6578 616d 706c 650a  es, for example.
+00002260: 0a60 6060 0a65 7870 6f72 7420 5041 5448  .```.export PATH
+00002270: 3d22 2450 4154 483a 7e2f 2e6c 6f63 616c  ="$PATH:~/.local
+00002280: 2f62 696e 2f77 6764 220a 6060 600a 0a4e  /bin/wgd".```..N
+00002290: 6f74 6520 7468 6174 2074 6865 2076 6572  ote that the ver
+000022a0: 7369 6f6e 206f 6620 606e 756d 7079 6020  sion of `numpy` 
+000022b0: 6973 2069 6d70 6f72 7461 6e74 2028 666f  is important (fo
+000022c0: 7220 6d61 6e79 206f 7468 6572 2070 6163  r many other pac
+000022d0: 6b61 6765 7320 6172 6520 7468 6520 7361  kages are the sa
+000022e0: 6d65 206f 6620 636f 7572 7365 292c 2065  me of course), e
+000022f0: 7370 6563 6961 6c6c 7920 666f 7220 6066  specially for `f
+00002300: 6173 7463 6c75 7374 6572 6020 7061 636b  astcluster` pack
+00002310: 6167 652e 2049 6e20 6f75 7220 7465 7374  age. In our test
+00002320: 2c20 7468 6520 606e 756d 7079 6020 312e  , the `numpy` 1.
+00002330: 3139 2e30 2077 6f72 6b73 2066 696e 6520  19.0 works fine 
+00002340: 6f6e 2060 7079 7468 6f6e 332e 362f 3860  on `python3.6/8`
+00002350: 2e20 4966 2079 6f75 206d 6574 2073 6f6d  . If you met som
+00002360: 6520 6572 726f 7273 206f 7220 7761 726e  e errors or warn
+00002370: 696e 6773 2061 626f 7574 2060 6e75 6d70  ings about `nump
+00002380: 7960 2c20 6d61 7962 6520 636f 6e73 6964  y`, maybe consid
+00002390: 6572 696e 6720 7072 652d 696e 7374 616c  ering pre-instal
+000023a0: 6c20 606e 756d 7079 6020 6173 2031 2e31  l `numpy` as 1.1
+000023b0: 392e 3020 6f72 206f 7468 6572 2063 6c6f  9.0 or other clo
+000023c0: 7365 2d62 7920 7665 7273 696f 6e73 2062  se-by versions b
+000023d0: 6566 6f72 6520 796f 7520 696e 7374 616c  efore you instal
+000023e0: 6c20 6077 6764 602e 0a0a 2323 2050 6172  l `wgd`...## Par
+000023f0: 616d 6574 6572 730a 0a54 6865 7265 2061  ameters..There a
+00002400: 7265 2037 206d 6169 6e20 7072 6f67 7261  re 7 main progra
+00002410: 6d73 2069 6e20 6077 6764 2076 3260 3a20  ms in `wgd v2`: 
+00002420: 6064 6d64 602c 6066 6f63 7573 602c 606b  `dmd`,`focus`,`k
+00002430: 7364 602c 606d 6978 602c 6070 6561 6b60  sd`,`mix`,`peak`
+00002440: 2c60 7379 6e60 2c60 7669 7a60 2e20 4865  ,`syn`,`viz`. He
+00002450: 7265 6166 7465 7220 7765 2077 696c 6c20  reafter we will 
+00002460: 7072 6f76 6964 6520 6120 6465 7461 696c  provide a detail
+00002470: 6564 2065 6c75 6369 6461 7469 6f6e 206f  ed elucidation o
+00002480: 6e20 6561 6368 206f 6620 7468 6520 7072  n each of the pr
+00002490: 6f67 7261 6d20 616e 6420 6974 7320 6173  ogram and its as
+000024a0: 736f 6369 6174 6564 2070 6172 616d 6574  sociated paramet
+000024b0: 6572 732e 2050 6c65 6173 6520 7265 6665  ers. Please refe
+000024c0: 7220 746f 2074 6865 205b 5573 6167 655d  r to the [Usage]
+000024d0: 2823 7573 6167 6529 2066 6f72 2074 6865  (#usage) for the
+000024e0: 2073 6365 6e61 7269 6f73 2074 6f20 7768   scenarios to wh
+000024f0: 6963 6820 6561 6368 2070 6172 616d 6574  ich each paramet
+00002500: 6572 2061 7070 6c69 6573 2e0a 0a54 6865  er applies...The
+00002510: 2070 726f 6772 616d 2060 7767 6420 646d   program `wgd dm
+00002520: 6460 2063 616e 2072 6561 6c69 7a65 2074  d` can realize t
+00002530: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
+00002540: 6620 7768 6f6c 6520 7061 7261 6e6f 6d65  f whole paranome
+00002550: 2c20 5242 4873 2028 5265 6369 7072 6f63  , RBHs (Reciproc
+00002560: 616c 2042 6573 7420 4869 7473 292c 204d  al Best Hits), M
+00002570: 5242 4873 2028 4d75 6c74 6970 6c65 2052  RBHs (Multiple R
+00002580: 6563 6970 726f 6361 6c20 4265 7374 2048  eciprocal Best H
+00002590: 6974 7329 2c20 6f72 7468 6f67 726f 7570  its), orthogroup
+000025a0: 7320 616e 6420 736f 6d65 206f 7468 6572  s and some other
+000025b0: 206f 7274 686f 6772 6f75 702d 7265 6c61   orthogroup-rela
+000025c0: 7465 6420 6675 6e63 7469 6f6e 732c 2069  ted functions, i
+000025d0: 6e63 6c75 6469 6e67 2063 6972 6375 6d73  ncluding circums
+000025e0: 6372 6970 7469 6f6e 206f 6620 6e65 7374  cription of nest
+000025f0: 6564 2073 696e 676c 652d 636f 7079 206f  ed single-copy o
+00002600: 7274 686f 6772 6f75 7073 2028 4e53 4f47  rthogroups (NSOG
+00002610: 7329 2c20 756e 6269 6173 6564 2074 6573  s), unbiased tes
+00002620: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
+00002630: 206f 7274 686f 6772 6f75 7073 2028 534f   orthogroups (SO
+00002640: 4773 2920 6f76 6572 206d 6973 7369 6e67  Gs) over missing
+00002650: 2069 6e70 6172 616c 6f67 732c 2063 6f6e   inparalogs, con
+00002660: 7374 7275 6374 696f 6e20 6f66 2042 5553  struction of BUS
+00002670: 434f 2d67 7569 6465 6420 7369 6e67 6c65  CO-guided single
+00002680: 2d63 6f70 7920 6f72 7468 6f67 726f 7570  -copy orthogroup
+00002690: 7320 2853 4f47 7329 2c61 6e64 2074 6865  s (SOGs),and the
+000026a0: 2063 6f6c 6c69 6e65 6172 2063 6f61 6c65   collinear coale
+000026b0: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
+000026c0: 206f 6620 7068 796c 6f67 656e 792e 0a60   of phylogeny..`
+000026d0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
+000026e0: 6e63 6573 2028 6f70 7469 6f6e 290a 2d2d  nces (option).--
+000026f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002700: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002710: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002720: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002730: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+00002740: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
+00002750: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
+00002760: 792c 2064 6566 6175 6c74 2077 6764 5f64  y, default wgd_d
+00002770: 6d64 0a2d 742c 202d 2d74 6d70 6469 722c  md.-t, --tmpdir,
+00002780: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
+00002790: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
+000027a0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+000027b0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
+000027c0: 6e2c 2074 6865 2074 6d70 6469 7220 776f  n, the tmpdir wo
+000027d0: 756c 6420 6265 2061 7373 6967 6e65 6420  uld be assigned 
+000027e0: 7261 6e64 6f6d 206e 616d 6573 2069 6e20  random names in 
+000027f0: 6375 7272 656e 7420 6469 7265 6374 6f72  current director
+00002800: 7920 616e 6420 6175 746f 6d61 7465 6c79  y and automately
+00002810: 2072 656d 6f76 6564 2061 7420 7468 6520   removed at the 
+00002820: 636f 6d70 6c65 7469 6f6e 206f 6620 7072  completion of pr
+00002830: 6f67 7261 6d2c 2065 6c73 6520 7468 6520  ogram, else the 
+00002840: 746d 7064 6972 2077 6f75 6c64 2062 6520  tmpdir would be 
+00002850: 6b65 7074 0a2d 702c 202d 2d70 726f 742c  kept.-p, --prot,
+00002860: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00002870: 6574 6865 7220 7573 696e 6720 7072 6f74  ether using prot
+00002880: 6569 6e20 6f72 206e 7563 6c65 6f74 6964  ein or nucleotid
+00002890: 6520 7365 7175 656e 6365 730a 2d63 2c20  e sequences.-c, 
+000028a0: 2d2d 6373 636f 7265 2c20 7468 6520 632d  --cscore, the c-
+000028b0: 7363 6f72 6520 746f 2072 6573 7472 6963  score to restric
+000028c0: 7420 7468 6520 686f 6d6f 6c6f 6720 7369  t the homolog si
+000028d0: 6d69 6c61 7269 7479 206f 6620 4d52 4248  milarity of MRBH
+000028e0: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
+000028f0: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
+00002900: 656e 2c20 7468 6520 632d 7363 6f72 6520  en, the c-score 
+00002910: 6675 6e63 696f 6e20 776f 756c 646e 2774  funcion wouldn't
+00002920: 2062 6520 6163 7469 7661 7465 642c 2065   be activated, e
+00002930: 6c73 6520 6578 7065 6374 696e 6720 6120  lse expecting a 
+00002940: 6465 6369 6d61 6c20 7769 7468 696e 2074  decimal within t
+00002950: 6865 2072 616e 6765 206f 6620 3020 616e  he range of 0 an
+00002960: 6420 310a 2d49 2c20 2d2d 696e 666c 6174  d 1.-I, --inflat
+00002970: 696f 6e2c 2074 6865 2069 6e66 6c61 7469  ion, the inflati
+00002980: 6f6e 2066 6163 746f 7220 666f 7220 4d43  on factor for MC
+00002990: 4c20 7072 6f67 7261 6d2c 2064 6566 6175  L program, defau
+000029a0: 6c74 2032 2e30 2c20 7769 7468 2068 6967  lt 2.0, with hig
+000029b0: 6865 7220 7661 6c75 6520 6c65 6164 696e  her value leadin
+000029c0: 6720 746f 206d 6f72 6520 6275 7420 736d  g to more but sm
+000029d0: 616c 6c65 7220 636c 7573 7465 7273 0a2d  aller clusters.-
+000029e0: 652c 202d 2d65 7661 6c2c 2074 6865 2065  e, --eval, the e
+000029f0: 2d76 616c 7565 2063 7574 2d6f 6666 2066  -value cut-off f
+00002a00: 6f72 2073 696d 696c 6172 6974 7920 696e  or similarity in
+00002a10: 2064 6961 6d6f 6e64 2061 6e64 2f6f 7220   diamond and/or 
+00002a20: 686d 6d65 722c 2064 6566 6175 6c74 2031  hmmer, default 1
+00002a30: 652d 3130 0a2d 2d74 6f5f 7374 6f70 2c20  e-10.--to_stop, 
+00002a40: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00002a50: 7468 6572 2074 6f20 7472 616e 736c 6174  ther to translat
+00002a60: 6520 7468 726f 7567 6820 5354 4f50 2063  e through STOP c
+00002a70: 6f64 6f6e 732c 2069 6620 7468 6520 666c  odons, if the fl
+00002a80: 6167 2077 6173 2073 6574 2c20 7472 616e  ag was set, tran
+00002a90: 736c 6174 696f 6e20 7769 6c6c 2062 6520  slation will be 
+00002aa0: 7465 726d 696e 6174 6564 2061 7420 7468  terminated at th
+00002ab0: 6520 6669 7273 7420 696e 2d66 7261 6d65  e first in-frame
+00002ac0: 2073 746f 7020 636f 646f 6e2c 2065 6c73   stop codon, els
+00002ad0: 6520 6120 6675 6c6c 2074 7261 6e73 6c61  e a full transla
+00002ae0: 7469 6f6e 2063 6f6e 7469 6e75 696e 6720  tion continuing 
+00002af0: 6f6e 2070 6173 7369 6e67 2061 6e79 2073  on passing any s
+00002b00: 746f 7020 636f 646f 6e73 2077 6f75 6c64  top codons would
+00002b10: 2062 6520 696e 6974 6961 7465 640a 2d2d   be initiated.--
+00002b20: 6364 732c 2066 6c61 6720 6f70 7469 6f6e  cds, flag option
+00002b30: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
+00002b40: 7920 7472 616e 736c 6174 6520 7468 6520  y translate the 
+00002b50: 636f 6d70 6c65 7465 2043 4453 2074 6861  complete CDS tha
+00002b60: 7420 7374 6172 7473 2077 6974 6820 6120  t starts with a 
+00002b70: 7661 6c69 6420 7374 6172 7420 636f 646f  valid start codo
+00002b80: 6e20 616e 6420 6f6e 6c79 2063 6f6e 7461  n and only conta
+00002b90: 696e 7320 6120 7369 6e67 6c65 2069 6e2d  ins a single in-
+00002ba0: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
+00002bb0: 2061 7420 7468 6520 656e 6420 616e 6420   at the end and 
+00002bc0: 6d75 7374 2062 6520 6469 7669 6461 626c  must be dividabl
+00002bd0: 6520 6279 2074 6872 6565 2c20 6966 2074  e by three, if t
+00002be0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00002bf0: 206f 6e6c 7920 7468 6520 636f 6d70 6c65   only the comple
+00002c00: 7465 2043 4453 2077 6f75 6c64 2062 6520  te CDS would be 
+00002c10: 7472 616e 736c 6174 6564 0a2d 662c 202d  translated.-f, -
+00002c20: 2d66 6f63 7573 2c20 7468 6520 7370 6563  -focus, the spec
+00002c30: 6965 7320 746f 2062 6520 6d65 7267 6564  ies to be merged
+00002c40: 206f 6e20 6c6f 6361 6c20 4d52 4248 732c   on local MRBHs,
+00002c50: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
+00002c60: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
+00002c70: 2c20 7468 6520 6c6f 6361 6c20 4d52 4248  , the local MRBH
+00002c80: 7320 776f 756c 646e 2774 2062 6520 696e  s wouldn't be in
+00002c90: 6665 7272 6564 0a2d 6170 2c20 2d2d 616e  ferred.-ap, --an
+00002ca0: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
+00002cb0: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
+00002cc0: 7461 2066 696c 6520 6672 6f6d 2069 2d61  ta file from i-a
+00002cd0: 6468 6f72 6520 666f 7220 636f 6e73 7472  dhore for constr
+00002ce0: 7563 7469 6e67 2074 6865 206f 7274 686f  ucting the ortho
+00002cf0: 6772 6f75 7073 2077 6974 6820 616e 6368  groups with anch
+00002d00: 6f72 2070 6169 7273 2c20 6465 6661 756c  or pairs, defaul
+00002d10: 7420 4e6f 6e65 0a2d 736d 2c20 2d2d 7365  t None.-sm, --se
+00002d20: 676d 656e 7473 2c20 7468 6520 7365 676d  gments, the segm
+00002d30: 656e 7473 2064 6174 6166 696c 6520 7573  ents datafile us
+00002d40: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
+00002d50: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+00002d60: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
+00002d70: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
+00002d80: 2d6c 652c 202d 2d6c 6973 7465 6c65 6d65  -le, --listeleme
+00002d90: 6e74 732c 2074 6865 206c 6973 7473 6567  nts, the listseg
+00002da0: 6d65 6e74 7320 6461 7461 2066 696c 6520  ments data file 
+00002db0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
+00002dc0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
+00002dd0: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
+00002de0: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
+00002df0: 650a 2d67 742c 202d 2d67 656e 6574 6162  e.-gt, --genetab
+00002e00: 6c65 2c20 7468 6520 6765 6e65 2074 6162  le, the gene tab
+00002e10: 6c65 2064 6174 6166 696c 6520 7573 6564  le datafile used
+00002e20: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
+00002e30: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+00002e40: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
+00002e50: 2064 6566 6175 6c74 204e 6f6e 650a 2d63   default None.-c
+00002e60: 6f63 2c20 2d2d 636f 6c6c 696e 6561 7263  oc, --collinearc
+00002e70: 6f61 6c65 7363 656e 6365 2c20 666c 6167  oalescence, flag
+00002e80: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+00002e90: 2074 6f20 696e 6974 6961 7465 2074 6865   to initiate the
+00002ea0: 2063 6f6c 6c69 6e65 6172 2063 6f61 6c65   collinear coale
+00002eb0: 7363 656e 6365 2061 6e61 6c79 7369 732c  scence analysis,
+00002ec0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00002ed0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+00002ee0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+00002ef0: 6961 7465 640a 2d6b 662c 202d 2d6b 6565  iated.-kf, --kee
+00002f00: 7066 6173 7461 2c20 666c 6167 206f 7074  pfasta, flag opt
+00002f10: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00002f20: 6f75 7470 7574 2074 6865 2073 6571 7565  output the seque
+00002f30: 6e63 6520 696e 666f 726d 6174 696f 6e20  nce information 
+00002f40: 6f66 204d 5242 4873 2c20 6966 2074 6865  of MRBHs, if the
+00002f50: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00002f60: 6865 2073 6571 7565 6e63 6573 206f 6620  he sequences of 
+00002f70: 4d52 4248 7320 776f 756c 6420 6265 2069  MRBHs would be i
+00002f80: 6e20 6f75 7470 7574 0a2d 6b64 2c20 2d2d  n output.-kd, --
+00002f90: 6b65 6570 6475 706c 6963 6174 6573 2c20  keepduplicates, 
+00002fa0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00002fb0: 7468 6572 2074 6f20 616c 6c6f 7720 7468  ther to allow th
+00002fc0: 6520 7361 6d65 2067 656e 6520 746f 206f  e same gene to o
+00002fd0: 6363 7572 2069 6e20 6469 6666 6572 656e  ccur in differen
+00002fe0: 7420 4d52 4248 7320 286f 6e6c 7920 6d65  t MRBHs (only me
+00002ff0: 616e 696e 6766 756c 2077 6865 6e20 7468  aningful when th
+00003000: 6520 6373 636f 7265 2077 6173 2075 7365  e cscore was use
+00003010: 6429 2c20 6966 2074 6865 2066 6c61 6720  d), if the flag 
+00003020: 7761 7320 7365 742c 2074 6865 2073 616d  was set, the sam
+00003030: 6520 6765 6e65 2063 6f75 6c64 2062 6520  e gene could be 
+00003040: 6173 7369 676e 6564 2074 6f20 6469 6666  assigned to diff
+00003050: 6572 656e 7420 4d52 4248 730a 2d67 6d2c  erent MRBHs.-gm,
+00003060: 202d 2d67 6c6f 6261 6c6d 7262 682c 2066   --globalmrbh, f
+00003070: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00003080: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00003090: 676c 6f62 616c 204d 5242 4873 2063 6f6e  global MRBHs con
+000030a0: 7374 7275 6374 696f 6e2c 2069 6620 7468  struction, if th
+000030b0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+000030c0: 7468 6520 2d2d 666f 6375 7320 6f70 7469  the --focus opti
+000030d0: 6f6e 2077 6f75 6c64 2062 6520 6967 6e6f  on would be igno
+000030e0: 7265 6420 616e 6420 6f6e 6c79 2067 6c6f  red and only glo
+000030f0: 6261 6c20 4d52 4248 7320 776f 756c 6420  bal MRBHs would 
+00003100: 6265 2062 7569 6c74 0a2d 6e2c 202d 2d6e  be built.-n, --n
+00003110: 7468 7265 6164 732c 2074 6865 206e 756d  threads, the num
+00003120: 6265 7220 6f66 2074 6872 6561 6473 2074  ber of threads t
+00003130: 6f20 7573 652c 2064 6566 6175 6c74 2034  o use, default 4
+00003140: 0a2d 6f69 2c20 2d2d 6f72 7468 6f69 6e66  .-oi, --orthoinf
+00003150: 6572 2c20 666c 6167 206f 7074 696f 6e2c  er, flag option,
+00003160: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00003170: 6961 7465 206f 7274 686f 6772 6f75 7020  iate orthogroup 
+00003180: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
+00003190: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+000031a0: 7468 6520 6f72 7468 6f67 726f 7570 2069  the orthogroup i
+000031b0: 6e66 6572 6e65 6365 2070 726f 6772 616d  nfernece program
+000031c0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+000031d0: 7465 640a 2d6f 6f2c 202d 2d6f 6e6c 796f  ted.-oo, --onlyo
+000031e0: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
+000031f0: 6e2c 2077 6865 7468 6572 2074 6f20 6f6e  n, whether to on
+00003200: 6c79 2063 6f6e 6475 6374 206f 7274 686f  ly conduct ortho
+00003210: 6772 6f75 7020 696e 6665 726e 6563 652c  group infernece,
+00003220: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00003230: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
+00003240: 7274 686f 6772 6f75 7020 696e 6665 726e  rthogroup infern
+00003250: 6563 6520 7069 7065 6c69 6e65 2077 6f75  ece pipeline wou
+00003260: 6c64 2062 6520 7065 7266 6f72 6d65 6420  ld be performed 
+00003270: 7768 696c 6520 7468 6520 6f74 6865 7220  while the other 
+00003280: 616e 616c 7973 6973 2077 6f75 6c64 6e27  analysis wouldn'
+00003290: 7420 6265 2069 6e69 7469 6174 6564 0a2d  t be initiated.-
+000032a0: 676e 2c20 2d2d 6765 746e 736f 672c 2066  gn, --getnsog, f
+000032b0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+000032c0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+000032d0: 7468 6520 7365 6172 6368 696e 6720 666f  the searching fo
+000032e0: 7220 6e65 7374 6564 2073 696e 676c 652d  r nested single-
+000032f0: 636f 7079 2067 656e 6520 6661 6d69 6c69  copy gene famili
+00003300: 6573 2028 4e53 4f47 7329 2028 6f6e 6c79  es (NSOGs) (only
+00003310: 206d 6561 6e69 6e67 6675 6c20 7768 656e   meaningful when
+00003320: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
+00003330: 696e 6665 726e 6563 6520 7069 7065 6c69  infernece pipeli
+00003340: 6e65 2077 6173 2061 6374 6976 6174 6564  ne was activated
+00003350: 292c 2069 6620 7468 6520 666c 6167 2077  ), if the flag w
+00003360: 6173 2073 6574 2c20 6164 6469 7469 6f6e  as set, addition
+00003370: 616c 204e 534f 4773 2061 6e61 6c79 7369  al NSOGs analysi
+00003380: 7320 776f 756c 6420 6265 2070 6572 666f  s would be perfo
+00003390: 726d 6564 2062 6573 6964 6573 2074 6865  rmed besides the
+000033a0: 2062 6173 6963 206f 7274 686f 6772 6f75   basic orthogrou
+000033b0: 7020 696e 6665 726e 6563 650a 2d74 7265  p infernece.-tre
+000033c0: 652c 202d 2d74 7265 655f 6d65 7468 6f64  e, --tree_method
+000033d0: 2c20 7768 6963 6820 6765 6e65 2074 7265  , which gene tre
+000033e0: 6520 696e 6665 7265 6e63 6520 7072 6f67  e inference prog
+000033f0: 7261 6d20 746f 2069 6e76 6f6b 6520 286f  ram to invoke (o
+00003400: 6e6c 7920 6d65 616e 696e 6766 756c 2077  nly meaningful w
+00003410: 6865 6e20 7468 6520 636f 6c6c 696e 6561  hen the collinea
+00003420: 7220 636f 616c 6573 6365 6e63 652c 2067  r coalescence, g
+00003430: 656e 652d 746f 2d66 616d 696c 7920 6173  ene-to-family as
+00003440: 7369 676e 6d65 6e74 206f 7220 4e53 4f47  signment or NSOG
+00003450: 7320 616e 616c 7973 6973 2077 6572 6520  s analysis were 
+00003460: 6163 7469 7661 7465 6429 2c20 6465 6661  activated), defa
+00003470: 756c 7420 6661 7374 7472 6565 0a2d 7473  ult fasttree.-ts
+00003480: 2c20 2d2d 7472 6565 7365 742c 2074 6865  , --treeset, the
+00003490: 2070 6172 616d 6574 6572 7320 7365 7474   parameters sett
+000034a0: 696e 6720 666f 7220 6765 6e65 2074 7265  ing for gene tre
+000034b0: 6520 696e 6665 7265 6e63 652c 2064 6566  e inference, def
+000034c0: 6175 6c74 204e 6f6e 652c 2074 6869 7320  ault None, this 
+000034d0: 6f70 7469 6f6e 2063 616e 2062 6520 7072  option can be pr
+000034e0: 6f76 6964 6564 206d 756c 7469 706c 6520  ovided multiple 
+000034f0: 7469 6d65 730a 2d6d 632c 202d 2d6d 736f  times.-mc, --mso
+00003500: 6763 7574 2c20 7468 6520 7261 7469 6f20  gcut, the ratio 
+00003510: 6375 746f 6666 2066 6f72 206d 6f73 746c  cutoff for mostl
+00003520: 7920 7369 6e67 6c65 2d63 6f70 7920 6661  y single-copy fa
+00003530: 6d69 6c79 2028 6d65 616e 696e 6766 756c  mily (meaningful
+00003540: 2077 6865 6e20 6163 7469 7661 7469 6e67   when activating
+00003550: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
+00003560: 696e 6665 726e 6563 6520 7069 7065 6c69  infernece pipeli
+00003570: 6e65 2920 616e 6420 7370 6563 6965 7320  ne) and species 
+00003580: 7265 7072 6573 656e 7461 7469 6f6e 2069  representation i
+00003590: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
+000035a0: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
+000035b0: 2c20 6465 6661 756c 7420 302e 382e 0a2d  , default 0.8..-
+000035c0: 6761 2c20 2d2d 6765 6e65 6173 7369 676e  ga, --geneassign
+000035d0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+000035e0: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+000035f0: 7465 2074 6865 2067 656e 652d 746f 2d66  te the gene-to-f
+00003600: 616d 696c 7920 6173 7369 676e 6d65 6e74  amily assignment
+00003610: 2061 6e61 6c79 7369 732c 2069 6620 7468   analysis, if th
+00003620: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00003630: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
+00003640: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+00003650: 2d73 612c 202d 2d73 6571 3261 7373 6967  -sa, --seq2assig
+00003660: 6e2c 2074 6865 2071 7565 7269 6564 2073  n, the queried s
+00003670: 6571 7565 6e63 6573 2064 6174 6120 6669  equences data fi
+00003680: 6c65 2069 6e20 6765 6e65 2d74 6f2d 6661  le in gene-to-fa
+00003690: 6d69 6c79 2061 7373 6967 6e6d 656e 7420  mily assignment 
+000036a0: 616e 616c 7973 6973 2c20 6465 6661 756c  analysis, defaul
+000036b0: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
+000036c0: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
+000036d0: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
+000036e0: 6573 0a2d 6661 2c20 2d2d 6661 6d32 6173  es.-fa, --fam2as
+000036f0: 7369 676e 2c20 7468 6520 7175 6572 6965  sign, the querie
+00003700: 6420 6661 6d69 6c69 7920 6461 7461 2066  d familiy data f
+00003710: 696c 6520 696e 2067 656e 652d 746f 2d66  ile in gene-to-f
+00003720: 616d 696c 7920 6173 7369 676e 6d65 6e74  amily assignment
+00003730: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
+00003740: 6c74 204e 6f6e 650a 2d63 632c 202d 2d63  lt None.-cc, --c
+00003750: 6f6e 6361 742c 2066 6c61 6720 6f70 7469  oncat, flag opti
+00003760: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
+00003770: 6e69 7469 6174 6520 7468 6520 636f 6e63  nitiate the conc
+00003780: 6174 656e 6174 696f 6e20 7069 7065 6c69  atenation pipeli
+00003790: 6e65 2066 6f72 206f 7274 686f 6772 6f75  ne for orthogrou
+000037a0: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
+000037b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000037c0: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
+000037d0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+000037e0: 640a 2d74 652c 202d 2d74 6573 7473 6f67  d.-te, --testsog
+000037f0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00003800: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00003810: 7465 2074 6865 2075 6e62 6961 7365 6420  te the unbiased 
+00003820: 7465 7374 206f 6620 7369 6e67 6c65 2d63  test of single-c
+00003830: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
+00003840: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+00003850: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
+00003860: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
+00003870: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
+00003880: 696e 732c 2074 6865 206e 756d 6265 7220  ins, the number 
+00003890: 6f66 2062 696e 7320 6469 7669 6465 6420  of bins divided 
+000038a0: 696e 2074 6865 2067 656e 6520 6c65 6e67  in the gene leng
+000038b0: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
+000038c0: 2c20 6465 6661 756c 7420 3130 300a 2d6e  , default 100.-n
+000038d0: 702c 202d 2d6e 6f72 6d61 6c69 7a65 6470  p, --normalizedp
+000038e0: 6572 6365 6e74 2c20 7468 6520 7065 7263  ercent, the perc
+000038f0: 656e 7461 6765 206f 6620 7570 7065 7220  entage of upper 
+00003900: 6869 7473 2075 7365 6420 666f 7220 6765  hits used for ge
+00003910: 6e65 206c 656e 6774 6820 6e6f 726d 616c  ne length normal
+00003920: 697a 6174 696f 6e2c 2064 6566 6175 6c74  ization, default
+00003930: 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d   5.-nn, --nonorm
+00003940: 616c 697a 6174 696f 6e2c 2066 6c61 6720  alization, flag 
+00003950: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00003960: 746f 2063 616c 6c20 6f66 6620 7468 6520  to call off the 
+00003970: 6e6f 726d 616c 697a 6174 696f 6e2c 2069  normalization, i
+00003980: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00003990: 6574 2c20 6e6f 206e 6f72 6d61 6c69 7a61  et, no normaliza
+000039a0: 7469 6f6e 2077 6f75 6c64 2062 6520 636f  tion would be co
+000039b0: 6e64 7563 7465 640a 2d62 736f 672c 202d  nducted.-bsog, -
+000039c0: 2d62 7573 636f 736f 672c 2066 6c61 6720  -buscosog, flag 
+000039d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000039e0: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
+000039f0: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
+00003a00: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
+00003a10: 6d69 6c79 2061 6e61 6c79 7369 732c 2069  mily analysis, i
+00003a20: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00003a30: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+00003a40: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00003a50: 7465 640a 2d62 686d 6d2c 202d 2d62 7573  ted.-bhmm, --bus
+00003a60: 636f 686d 6d2c 2074 6865 2048 4d4d 2070  cohmm, the HMM p
+00003a70: 726f 6669 6c65 2064 6174 6166 696c 6520  rofile datafile 
+00003a80: 696e 2074 6865 2062 7573 636f 2d67 7569  in the busco-gui
+00003a90: 6465 6420 7369 6e67 6c65 2d63 6f70 7920  ded single-copy 
+00003aa0: 6765 6e65 2066 616d 696c 7920 616e 616c  gene family anal
+00003ab0: 7973 6973 2c20 6465 6661 756c 7420 4e6f  ysis, default No
+00003ac0: 6e65 0a2d 6263 7466 2c20 2d2d 6275 7363  ne.-bctf, --busc
+00003ad0: 6f63 7574 6f66 662c 2074 6865 2048 4d4d  ocutoff, the HMM
+00003ae0: 2073 636f 7265 2063 7574 6f66 6620 6461   score cutoff da
+00003af0: 7461 6669 6c65 2069 6e20 7468 6520 6275  tafile in the bu
+00003b00: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
+00003b10: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
+00003b20: 6c79 2061 6e61 6c79 7369 732c 2064 6566  ly analysis, def
+00003b30: 6175 6c74 204e 6f6e 650a 2d6f 6620 2c2d  ault None.-of ,-
+00003b40: 2d6f 6766 6f72 6d61 742c 2066 6c61 6720  -ogformat, flag 
+00003b50: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00003b60: 746f 2061 6464 2069 6e64 6578 2074 6f20  to add index to 
+00003b70: 7468 6520 5242 4820 6661 6d69 6c69 6573  the RBH families
+00003b80: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+00003b90: 6d20 6077 6764 2066 6f63 7573 6020 6361  m `wgd focus` ca
+00003ba0: 6e20 7265 616c 697a 6520 7468 6520 636f  n realize the co
+00003bb0: 6e63 6174 656e 6174 696f 6e2d 6261 7365  ncatenation-base
+00003bc0: 6420 616e 6420 636f 616c 6573 6365 6e63  d and coalescenc
+00003bd0: 652d 6261 7365 6420 7068 796c 6f67 656e  e-based phylogen
+00003be0: 6574 6963 2069 6e66 6572 656e 6365 2061  etic inference a
+00003bf0: 6e64 2070 6879 6c6f 6765 6e65 7469 6320  nd phylogenetic 
+00003c00: 6461 7469 6e67 206f 6620 5747 4473 2065  dating of WGDs e
+00003c10: 7463 2e0a 6060 600a 7767 6420 666f 6375  tc..```.wgd focu
+00003c20: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+00003c30: 6e63 6573 2028 6f70 7469 6f6e 290a 2d2d  nces (option).--
+00003c40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003c50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003c60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003c70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003c80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+00003c90: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
+00003ca0: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
+00003cb0: 792c 2064 6566 6175 6c74 2077 6764 5f66  y, default wgd_f
+00003cc0: 6f63 7573 0a2d 742c 202d 2d74 6d70 6469  ocus.-t, --tmpdi
+00003cd0: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
+00003ce0: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
+00003cf0: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
+00003d00: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
+00003d10: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
+00003d20: 7769 6c6c 2062 6520 6173 7369 676e 6564  will be assigned
+00003d30: 2072 616e 646f 6d20 6e61 6d65 7320 696e   random names in
+00003d40: 2063 7572 7265 6e74 2064 6972 6563 746f   current directo
+00003d50: 7279 2061 6e64 2061 7574 6f6d 6174 656c  ry and automatel
+00003d60: 7920 7265 6d6f 7665 6420 6174 2074 6865  y removed at the
+00003d70: 2063 6f6d 706c 6574 696f 6e20 6f66 2070   completion of p
+00003d80: 726f 6772 616d 2c20 656c 7365 2074 6865  rogram, else the
+00003d90: 2074 6d70 6469 7220 776f 756c 6420 6265   tmpdir would be
+00003da0: 206b 6570 740a 2d6e 2c20 2d2d 6e74 6872   kept.-n, --nthr
+00003db0: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
+00003dc0: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
+00003dd0: 7365 2c20 6465 6661 756c 7420 340a 2d2d  se, default 4.--
+00003de0: 746f 5f73 746f 702c 2066 6c61 6720 6f70  to_stop, flag op
+00003df0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00003e00: 2074 7261 6e73 6c61 7465 2074 6872 6f75   translate throu
+00003e10: 6768 2053 544f 5020 636f 646f 6e73 2c20  gh STOP codons, 
+00003e20: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00003e30: 7365 742c 2074 7261 6e73 6c61 7469 6f6e  set, translation
+00003e40: 2077 696c 6c20 6265 2074 6572 6d69 6e61   will be termina
+00003e50: 7465 6420 6174 2074 6865 2066 6972 7374  ted at the first
+00003e60: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
+00003e70: 6f64 6f6e 2c20 656c 7365 2061 2066 756c  odon, else a ful
+00003e80: 6c20 7472 616e 736c 6174 696f 6e20 636f  l translation co
+00003e90: 6e74 696e 7569 6e67 206f 6e20 7061 7374  ntinuing on past
+00003ea0: 2061 6e79 2073 746f 7020 636f 646f 6e73   any stop codons
+00003eb0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00003ec0: 7465 640a 2d2d 6364 732c 2066 6c61 6720  ted.--cds, flag 
+00003ed0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00003ee0: 746f 206f 6e6c 7920 7472 616e 736c 6174  to only translat
+00003ef0: 6520 7468 6520 636f 6d70 6c65 7465 2043  e the complete C
+00003f00: 4453 2074 6861 7420 7374 6172 7473 2077  DS that starts w
+00003f10: 6974 6820 6120 7661 6c69 6420 7374 6172  ith a valid star
+00003f20: 7420 636f 646f 6e20 616e 6420 6f6e 6c79  t codon and only
+00003f30: 2063 6f6e 7461 696e 7320 6120 7369 6e67   contains a sing
+00003f40: 6c65 2069 6e2d 6672 616d 6520 7374 6f70  le in-frame stop
+00003f50: 2063 6f64 6f6e 2061 7420 7468 6520 656e   codon at the en
+00003f60: 6420 616e 6420 6d75 7374 2062 6520 6469  d and must be di
+00003f70: 7669 6461 626c 6520 6279 2074 6872 6565  vidable by three
+00003f80: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00003f90: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
+00003fa0: 636f 6d70 6c65 7465 2043 4453 2077 6f75  complete CDS wou
+00003fb0: 6c64 2062 6520 7472 616e 736c 6174 6564  ld be translated
+00003fc0: 0a2d 2d73 7472 6970 5f67 6170 732c 2066  .--strip_gaps, f
+00003fd0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00003fe0: 6865 7220 746f 2064 726f 7020 616c 6c20  her to drop all 
+00003ff0: 6761 7073 2069 6e20 6d75 6c74 6970 6c65  gaps in multiple
+00004000: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
+00004010: 656e 742c 2069 6620 7468 6520 666c 6167  ent, if the flag
+00004020: 2077 6173 2073 6574 2c20 616c 6c20 6761   was set, all ga
+00004030: 7073 2077 6f75 6c64 2062 6520 6472 6f70  ps would be drop
+00004040: 7065 640a 2d61 2c20 2d2d 616c 6967 6e65  ped.-a, --aligne
+00004050: 722c 2077 6869 6368 2061 6c69 676e 6d65  r, which alignme
+00004060: 6e74 2070 726f 6772 616d 2074 6f20 7573  nt program to us
+00004070: 652c 2064 6566 6175 6c74 206d 6166 6674  e, default mafft
+00004080: 0a2d 7472 6565 2c20 2d2d 7472 6565 5f6d  .-tree, --tree_m
+00004090: 6574 686f 642c 2077 6869 6368 2067 656e  ethod, which gen
+000040a0: 6520 7472 6565 2069 6e66 6572 656e 6365  e tree inference
+000040b0: 2070 726f 6772 616d 2074 6f20 696e 766f   program to invo
+000040c0: 6b65 2c20 6465 6661 756c 7420 6661 7374  ke, default fast
+000040d0: 7472 6565 0a2d 7473 2c20 2d2d 7472 6565  tree.-ts, --tree
+000040e0: 7365 742c 2074 6865 2070 6172 616d 6574  set, the paramet
+000040f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
+00004100: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
+00004110: 6e63 652c 2064 6566 6175 6c74 204e 6f6e  nce, default Non
+00004120: 652c 2074 6869 7320 6f70 7469 6f6e 2063  e, this option c
+00004130: 616e 2062 6520 7072 6f76 6964 6564 206d  an be provided m
+00004140: 756c 7469 706c 6520 7469 6d65 730a 2d2d  ultiple times.--
+00004150: 636f 6e63 6174 656e 6174 696f 6e2c 2066  concatenation, f
+00004160: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004170: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00004180: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+00004190: 6e2d 6261 7365 6420 7370 6563 6965 7320  n-based species 
+000041a0: 7472 6565 2069 6e66 6572 656e 6365 2c20  tree inference, 
+000041b0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+000041c0: 7365 742c 2063 6f6e 6361 7465 6e61 7469  set, concatenati
+000041d0: 6f6e 2d62 6173 6564 2073 7065 6369 6573  on-based species
+000041e0: 2074 7265 6520 776f 756c 6420 6265 2069   tree would be i
+000041f0: 6e66 6572 6564 0a2d 2d63 6f61 6c65 7363  nfered.--coalesc
+00004200: 656e 6365 2c20 666c 6167 206f 7074 696f  ence, flag optio
+00004210: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00004220: 6974 6961 7465 2074 6865 2063 6f61 6c65  itiate the coale
+00004230: 7363 656e 6365 2d62 6173 6564 2073 7065  scence-based spe
+00004240: 6369 6573 2074 7265 6520 696e 6665 7265  cies tree infere
+00004250: 6e63 652c 2069 6620 7468 6520 666c 6167  nce, if the flag
+00004260: 2077 6173 2073 6574 2c20 636f 616c 6573   was set, coales
+00004270: 6365 6e63 652d 6261 7365 6420 7370 6563  cence-based spec
+00004280: 6965 7320 7472 6565 2077 6f75 6c64 2062  ies tree would b
+00004290: 6520 696e 6665 7265 640a 2d73 702c 202d  e infered.-sp, -
+000042a0: 2d73 7065 6369 6573 7472 6565 2c20 7370  -speciestree, sp
+000042b0: 6563 6965 7320 7472 6565 2064 6174 6166  ecies tree dataf
+000042c0: 696c 6520 666f 7220 6461 7469 6e67 2c20  ile for dating, 
+000042d0: 6465 6661 756c 7420 4e6f 6e65 0a2d 642c  default None.-d,
+000042e0: 202d 2d64 6174 696e 672c 2077 6869 6368   --dating, which
+000042f0: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
+00004300: 6720 7072 6f67 7261 6d20 746f 2075 7365  g program to use
+00004310: 2c20 6465 6661 756c 7420 6e6f 6e65 0a2d  , default none.-
+00004320: 6473 2c20 2d2d 6461 7469 6e67 7365 742c  ds, --datingset,
+00004330: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
+00004340: 7365 7474 696e 6720 666f 7220 6461 7469  setting for dati
+00004350: 6e67 2070 726f 6772 616d 2c20 6465 6661  ng program, defa
+00004360: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
+00004370: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
+00004380: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
+00004390: 696d 6573 0a2d 6e73 2c20 2d2d 6e73 6974  imes.-ns, --nsit
+000043a0: 6573 2c20 7468 6520 6e73 6974 6573 2069  es, the nsites i
+000043b0: 6e66 6f72 6d61 7469 6f6e 2066 6f72 2072  nformation for r
+000043c0: 3873 2064 6174 696e 672c 2064 6566 6175  8s dating, defau
+000043d0: 6c74 204e 6f6e 650a 2d6f 742c 202d 2d6f  lt None.-ot, --o
+000043e0: 7574 6772 6f75 702c 2074 6865 206f 7574  utgroup, the out
+000043f0: 6772 6f75 7020 696e 666f 726d 6174 696f  group informatio
+00004400: 6e20 666f 7220 7238 7320 6461 7469 6e67  n for r8s dating
+00004410: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00004420: 7074 2c20 2d2d 7061 7274 6974 696f 6e2c  pt, --partition,
+00004430: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004440: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00004450: 6520 7061 7274 6974 696f 6e20 6461 7469  e partition dati
+00004460: 6e67 2061 6e61 6c79 7369 7320 666f 7220  ng analysis for 
+00004470: 636f 646f 6e2c 2069 6620 7468 6520 666c  codon, if the fl
+00004480: 6167 2077 6173 2073 6574 2c20 616e 2061  ag was set, an a
+00004490: 6464 6974 696f 6e61 6c20 7061 7274 6974  dditional partit
+000044a0: 696f 6e20 6461 7469 6e67 2061 6e61 6c79  ion dating analy
+000044b0: 7369 7320 776f 756c 6420 6265 2069 6e69  sis would be ini
+000044c0: 7469 6174 6564 0a2d 616d 2c20 2d2d 6161  tiated.-am, --aa
+000044d0: 6d6f 6465 6c2c 2077 6869 6368 2070 726f  model, which pro
+000044e0: 7465 696e 206d 6f64 656c 2074 6f20 6265  tein model to be
+000044f0: 2075 7365 6420 696e 206d 636d 6374 7265   used in mcmctre
+00004500: 652c 2064 6566 6175 6c74 2070 6f69 7373  e, default poiss
+00004510: 6f6e 0a2d 6b73 2c20 666c 6167 206f 7074  on.-ks, flag opt
+00004520: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00004530: 696e 6974 6961 7465 204b 7320 6361 6c63  initiate Ks calc
+00004540: 756c 6174 696f 6e20 666f 7220 686f 6d6f  ulation for homo
+00004550: 6c6f 6775 6573 2069 6e20 7468 6520 7072  logues in the pr
+00004560: 6f76 6964 6564 206f 7274 686f 6c6f 676f  ovided orthologo
+00004570: 7573 2067 656e 6520 6661 6d69 6c79 0a2d  us gene family.-
+00004580: 2d61 6e6e 6f74 6174 696f 6e2c 2077 6869  -annotation, whi
+00004590: 6368 2061 6e6e 6f74 6174 696f 6e20 7072  ch annotation pr
+000045a0: 6f67 7261 6d20 746f 2075 7365 2c20 6465  ogram to use, de
+000045b0: 6661 756c 7420 4e6f 6e65 0a2d 2d70 6169  fault None.--pai
+000045c0: 7277 6973 652c 2066 6c61 6720 6f70 7469  rwise, flag opti
+000045d0: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
+000045e0: 6e69 7469 6174 6520 7061 6972 7769 7365  nitiate pairwise
+000045f0: 204b 7320 6573 7469 6d61 7469 6f6e 2c20   Ks estimation, 
+00004600: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00004610: 7365 742c 2070 6169 7277 6973 6520 4b73  set, pairwise Ks
+00004620: 2076 616c 7565 7320 776f 756c 6420 6265   values would be
+00004630: 2065 7374 696d 6174 6564 0a2d 6564 2c20   estimated.-ed, 
+00004640: 2d2d 6567 676e 6f67 6461 7461 2c20 7468  --eggnogdata, th
+00004650: 6520 6567 676e 6f67 2061 6e6e 6f74 6174  e eggnog annotat
+00004660: 696f 6e20 6461 7461 6669 6c65 2c20 6465  ion datafile, de
+00004670: 6661 756c 7420 4e6f 6e65 0a2d 2d70 6661  fault None.--pfa
+00004680: 6d2c 2077 6869 6368 206f 7074 696f 6e20  m, which option 
+00004690: 746f 2075 7365 2066 6f72 2070 6661 6d20  to use for pfam 
+000046a0: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
+000046b0: 756c 7420 4e6f 6e65 0a2d 2d64 6d6e 622c  ult None.--dmnb,
+000046c0: 2074 6865 2064 6961 6d6f 6e64 2064 6174   the diamond dat
+000046d0: 6162 6173 6520 666f 7220 616e 6e6f 7461  abase for annota
+000046e0: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
+000046f0: 6e65 0a2d 2d68 6d6d 2c20 7468 6520 484d  ne.--hmm, the HM
+00004700: 4d20 7072 6f66 696c 6520 666f 7220 616e  M profile for an
+00004710: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
+00004720: 7420 4e6f 6e65 0a2d 2d65 7661 6c75 652c  t None.--evalue,
+00004730: 2074 6865 2065 2d76 616c 7565 2063 7574   the e-value cut
+00004740: 2d6f 6666 2066 6f72 2061 6e6e 6f74 6174  -off for annotat
+00004750: 696f 6e2c 2064 6566 6175 6c74 2031 652d  ion, default 1e-
+00004760: 3130 0a2d 2d65 7865 7061 7468 2c20 7468  10.--exepath, th
+00004770: 6520 7061 7468 2074 6f20 7468 6520 696e  e path to the in
+00004780: 7465 7270 726f 7363 616e 2065 7865 6375  terproscan execu
+00004790: 7461 626c 652c 2064 6566 6175 6c74 204e  table, default N
+000047a0: 6f6e 650a 2d66 2c20 2d2d 666f 7373 696c  one.-f, --fossil
+000047b0: 2c20 7468 6520 666f 7373 696c 2063 616c  , the fossil cal
+000047c0: 6962 7261 7469 6f6e 2069 6e66 6f72 6d61  ibration informa
+000047d0: 7469 6f6e 2069 6e20 4265 6173 742c 2064  tion in Beast, d
+000047e0: 6566 6175 6c74 2028 2763 6c61 6465 313b  efault ('clade1;
+000047f0: 636c 6164 6532 272c 2027 7461 7861 312c  clade2', 'taxa1,
+00004800: 7461 7861 323b 7461 7861 332c 7461 7861  taxa2;taxa3,taxa
+00004810: 3427 2c20 2734 3b35 272c 2027 302e 353b  4', '4;5', '0.5;
+00004820: 302e 3627 2c20 2734 3030 3b35 3030 2729  0.6', '400;500')
+00004830: 0a2d 7268 2c20 2d2d 726f 6f74 6865 6967  .-rh, --rootheig
+00004840: 6874 2c20 7468 6520 726f 6f74 2068 6569  ht, the root hei
+00004850: 6768 7420 6361 6c69 6272 6174 696f 6e20  ght calibration 
+00004860: 696e 666f 2069 6e20 4265 6173 742c 2064  info in Beast, d
+00004870: 6566 6175 6c74 2028 342c 302e 352c 3430  efault (4,0.5,40
+00004880: 3029 0a2d 6373 2c20 2d2d 6368 6169 6e73  0).-cs, --chains
+00004890: 6574 2c20 7468 6520 7061 7261 6d65 7465  et, the paramete
+000048a0: 7273 206f 6620 4d43 4d43 2063 6861 696e  rs of MCMC chain
+000048b0: 2069 6e20 4265 6173 742c 2064 6566 6175   in Beast, defau
+000048c0: 6c74 2028 3130 3030 302c 3130 3029 0a2d  lt (10000,100).-
+000048d0: 2d62 6561 7374 6c67 6a61 722c 2074 6865  -beastlgjar, the
+000048e0: 2070 6174 6820 746f 2062 6561 7374 4c47   path to beastLG
+000048f0: 2e6a 6172 2c20 6465 6661 756c 7420 4e6f  .jar, default No
+00004900: 6e65 0a2d 2d62 6561 676c 652c 2066 6c61  ne.--beagle, fla
+00004910: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00004920: 7220 746f 2075 7365 2062 6561 676c 6520  r to use beagle 
+00004930: 696e 2042 6561 7374 2c20 6966 2074 6865  in Beast, if the
+00004940: 2066 6c61 6720 7761 7320 7365 742c 2062   flag was set, b
+00004950: 6561 676c 6520 776f 756c 6420 6265 2075  eagle would be u
+00004960: 7365 640a 2d2d 7072 6f74 636f 6364 6174  sed.--protcocdat
+00004970: 696e 672c 2066 6c61 6720 6f70 7469 6f6e  ing, flag option
+00004980: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
+00004990: 7920 696e 6974 6961 7465 2074 6865 2070  y initiate the p
+000049a0: 726f 7465 696e 2d63 6f6e 6361 7465 6e61  rotein-concatena
+000049b0: 7469 6f6e 2d62 6173 6564 2064 6174 696e  tion-based datin
+000049c0: 6720 616e 616c 7973 6973 2c20 6966 2074  g analysis, if t
+000049d0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+000049e0: 2074 6865 2061 6e61 6c79 7369 7320 776f   the analysis wo
+000049f0: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
+00004a00: 0a2d 2d70 726f 7464 6174 696e 672c 2066  .--protdating, f
+00004a10: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004a20: 6865 7220 746f 206f 6e6c 7920 696e 6974  her to only init
+00004a30: 6961 7465 2074 6865 2070 726f 7465 696e  iate the protein
+00004a40: 2d62 6173 6564 2064 6174 696e 6720 616e  -based dating an
+00004a50: 616c 7973 6973 2c20 6966 2074 6865 2066  alysis, if the f
+00004a60: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00004a70: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
+00004a80: 6265 2069 6e69 7469 6174 6564 0a60 6060  be initiated.```
+00004a90: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
+00004aa0: 6764 206b 7364 6020 6361 6e20 7265 616c  gd ksd` can real
+00004ab0: 697a 6520 7468 6520 636f 6e73 7472 7563  ize the construc
+00004ac0: 7469 6f6e 206f 6620 2a4b 2a3c 7375 623e  tion of *K*<sub>
+00004ad0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00004ae0: 7269 6275 7469 6f6e 2061 6e64 2072 6174  ribution and rat
+00004af0: 6520 636f 7272 6563 7469 6f6e 2e0a 6060  e correction..``
+00004b00: 600a 7767 6420 6b73 6420 6661 6d69 6c69  `.wgd ksd famili
+00004b10: 6573 2073 6571 7565 6e63 6573 2028 6f70  es sequences (op
+00004b20: 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d  tion).----------
+00004b30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004b40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004b50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004b60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004b70: 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574  ------.-o, --out
+00004b80: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
+00004b90: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
+00004ba0: 6c74 2077 6764 5f6b 7364 0a2d 742c 202d  lt wgd_ksd.-t, -
+00004bb0: 2d74 6d70 6469 722c 2074 6865 2074 656d  -tmpdir, the tem
+00004bc0: 706f 7261 7279 2077 6f72 6b69 6e67 2064  porary working d
+00004bd0: 6972 6563 746f 7279 2c20 6465 6661 756c  irectory, defaul
+00004be0: 7420 4e6f 6e65 2c20 6966 204e 6f6e 6520  t None, if None 
+00004bf0: 7761 7320 6769 7665 6e2c 2074 6865 2074  was given, the t
+00004c00: 6d70 6469 7220 7769 6c6c 2062 6520 6173  mpdir will be as
+00004c10: 7369 676e 6564 2072 616e 646f 6d20 6e61  signed random na
+00004c20: 6d65 7320 696e 2063 7572 7265 6e74 2064  mes in current d
+00004c30: 6972 6563 746f 7279 2061 6e64 2061 7574  irectory and aut
+00004c40: 6f6d 6174 656c 7920 7265 6d6f 7665 6420  omately removed 
+00004c50: 6174 2074 6865 2063 6f6d 706c 6574 696f  at the completio
+00004c60: 6e20 6f66 2070 726f 6772 616d 2c20 656c  n of program, el
+00004c70: 7365 2074 6865 2074 6d70 6469 7220 776f  se the tmpdir wo
+00004c80: 756c 6420 6265 206b 6570 740a 2d6e 2c20  uld be kept.-n, 
+00004c90: 2d2d 6e74 6872 6561 6473 2c20 7468 6520  --nthreads, the 
+00004ca0: 6e75 6d62 6572 206f 6620 7468 7265 6164  number of thread
+00004cb0: 7320 746f 2075 7365 2c20 6465 6661 756c  s to use, defaul
+00004cc0: 7420 340a 2d2d 746f 5f73 746f 702c 2066  t 4.--to_stop, f
+00004cd0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004ce0: 6865 7220 746f 2074 7261 6e73 6c61 7465  her to translate
+00004cf0: 2074 6872 6f75 6768 2053 544f 5020 636f   through STOP co
+00004d00: 646f 6e73 2c20 6966 2074 6865 2066 6c61  dons, if the fla
+00004d10: 6720 7761 7320 7365 742c 2074 7261 6e73  g was set, trans
+00004d20: 6c61 7469 6f6e 2077 696c 6c20 6265 2074  lation will be t
+00004d30: 6572 6d69 6e61 7465 6420 6174 2074 6865  erminated at the
+00004d40: 2066 6972 7374 2069 6e2d 6672 616d 6520   first in-frame 
+00004d50: 7374 6f70 2063 6f64 6f6e 2c20 656c 7365  stop codon, else
+00004d60: 2061 2066 756c 6c20 7472 616e 736c 6174   a full translat
+00004d70: 696f 6e20 636f 6e74 696e 7569 6e67 206f  ion continuing o
+00004d80: 6e20 7061 7374 2061 6e79 2073 746f 7020  n past any stop 
+00004d90: 636f 646f 6e73 2077 6f75 6c64 2062 6520  codons would be 
+00004da0: 696e 6974 6961 7465 640a 2d2d 6364 732c  initiated.--cds,
+00004db0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004dc0: 6574 6865 7220 746f 206f 6e6c 7920 7472  ether to only tr
+00004dd0: 616e 736c 6174 6520 7468 6520 636f 6d70  anslate the comp
+00004de0: 6c65 7465 2043 4453 2074 6861 7420 7374  lete CDS that st
+00004df0: 6172 7473 2077 6974 6820 6120 7661 6c69  arts with a vali
+00004e00: 6420 7374 6172 7420 636f 646f 6e20 616e  d start codon an
+00004e10: 6420 6f6e 6c79 2063 6f6e 7461 696e 7320  d only contains 
+00004e20: 6120 7369 6e67 6c65 2069 6e2d 6672 616d  a single in-fram
+00004e30: 6520 7374 6f70 2063 6f64 6f6e 2061 7420  e stop codon at 
+00004e40: 7468 6520 656e 6420 616e 6420 6d75 7374  the end and must
+00004e50: 2062 6520 6469 7669 6461 626c 6520 6279   be dividable by
+00004e60: 2074 6872 6565 2c20 6966 2074 6865 2066   three, if the f
+00004e70: 6c61 6720 7761 7320 7365 742c 206f 6e6c  lag was set, onl
+00004e80: 7920 7468 6520 636f 6d70 6c65 7465 2043  y the complete C
+00004e90: 4453 2077 6f75 6c64 2062 6520 7472 616e  DS would be tran
+00004ea0: 736c 6174 6564 0a2d 2d70 6169 7277 6973  slated.--pairwis
+00004eb0: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
+00004ec0: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00004ed0: 6174 6520 7061 6972 7769 7365 204b 7320  ate pairwise Ks 
+00004ee0: 6573 7469 6d61 7469 6f6e 2c20 6966 2074  estimation, if t
+00004ef0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00004f00: 2070 6169 7277 6973 6520 4b73 2076 616c   pairwise Ks val
+00004f10: 7565 7320 776f 756c 6420 6265 2065 7374  ues would be est
+00004f20: 696d 6174 6564 0a2d 2d73 7472 6970 5f67  imated.--strip_g
+00004f30: 6170 732c 2066 6c61 6720 6f70 7469 6f6e  aps, flag option
+00004f40: 2c20 7768 6574 6865 7220 746f 2064 726f  , whether to dro
+00004f50: 7020 616c 6c20 6761 7073 2069 6e20 6d75  p all gaps in mu
+00004f60: 6c74 6970 6c65 2073 6571 7565 6e63 6520  ltiple sequence 
+00004f70: 616c 6967 6e6d 656e 742c 2069 6620 7468  alignment, if th
+00004f80: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00004f90: 616c 6c20 6761 7073 2077 6f75 6c64 2062  all gaps would b
+00004fa0: 6520 6472 6f70 7065 640a 2d61 2c20 2d2d  e dropped.-a, --
+00004fb0: 616c 6967 6e65 722c 2077 6869 6368 2061  aligner, which a
+00004fc0: 6c69 676e 6d65 6e74 2070 726f 6772 616d  lignment program
+00004fd0: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
+00004fe0: 206d 6166 6674 200a 2d74 7265 652c 202d   mafft .-tree, -
+00004ff0: 2d74 7265 655f 6d65 7468 6f64 2c20 7768  -tree_method, wh
+00005000: 6963 6820 6765 6e65 2074 7265 6520 696e  ich gene tree in
+00005010: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+00005020: 746f 2069 6e76 6f6b 652c 2064 6566 6175  to invoke, defau
+00005030: 6c74 2066 6173 7474 7265 650a 2d2d 7472  lt fasttree.--tr
+00005040: 6565 5f6f 7074 696f 6e73 2c20 6f70 7469  ee_options, opti
+00005050: 6f6e 7320 696e 2074 7265 6520 696e 6665  ons in tree infe
+00005060: 7265 6e63 6520 6173 2061 2063 6f6d 6d61  rence as a comma
+00005070: 2073 6570 6172 6174 6564 2073 7472 696e   separated strin
+00005080: 672c 2064 6566 6175 6c74 204e 6f6e 650a  g, default None.
+00005090: 2d2d 6e6f 6465 5f61 7665 7261 6765 2c20  --node_average, 
+000050a0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+000050b0: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+000050c0: 206e 6f64 652d 6176 6572 6167 6520 7761   node-average wa
+000050d0: 7920 6f66 2064 652d 7265 6475 6e64 616e  y of de-redundan
+000050e0: 6379 2069 6e73 7465 6164 206f 6620 6e6f  cy instead of no
+000050f0: 6465 2d77 6569 6768 7465 642c 2069 6620  de-weighted, if 
+00005100: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00005110: 2c20 7468 6520 6e6f 6465 2d61 7665 7261  , the node-avera
+00005120: 6769 6e67 2064 652d 7265 6475 6e64 616e  ging de-redundan
+00005130: 6379 2077 6f75 6c64 2062 6520 696e 6974  cy would be init
+00005140: 6961 7465 640a 2d73 722c 202d 2d73 7061  iated.-sr, --spa
+00005150: 6972 2c20 7468 6520 7370 6563 6965 7320  ir, the species 
+00005160: 7061 6972 2074 6f20 6265 2070 6c6f 7474  pair to be plott
+00005170: 6564 2c20 6465 6661 756c 7420 4e6f 6e65  ed, default None
+00005180: 2c20 7468 6973 206f 7074 696f 6e20 6361  , this option ca
+00005190: 6e20 6265 2070 726f 7669 6465 6420 6d75  n be provided mu
+000051a0: 6c74 6970 6c65 2074 696d 6573 0a2d 7370  ltiple times.-sp
+000051b0: 2c20 2d2d 7370 6563 6965 7374 7265 652c  , --speciestree,
+000051c0: 2074 6865 2073 7065 6369 6573 2074 7265   the species tre
+000051d0: 6520 746f 2070 6572 666f 726d 2072 6174  e to perform rat
+000051e0: 6520 636f 7272 6563 7469 6f6e 2c20 6465  e correction, de
+000051f0: 6661 756c 7420 4e6f 6e65 2c20 6966 204e  fault None, if N
+00005200: 6f6e 6520 7761 7320 6769 7665 6e2c 2074  one was given, t
+00005210: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+00005220: 6f6e 2061 6e61 6c79 7369 7320 776f 756c  on analysis woul
+00005230: 6420 6265 2063 616c 6c65 6420 6f66 660a  d be called off.
+00005240: 2d72 772c 202d 2d72 6577 6569 6768 742c  -rw, --reweight,
+00005250: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00005260: 6574 6865 7220 746f 2072 6563 616c 6375  ether to recalcu
+00005270: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
+00005280: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
+00005290: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+000052a0: 7320 7365 742c 2074 6865 2077 6569 6768  s set, the weigh
+000052b0: 7420 776f 756c 6420 6265 2072 6563 616c  t would be recal
+000052c0: 6375 6c61 7465 640a 2d6f 722c 202d 2d6f  culated.-or, --o
+000052d0: 6e6c 7972 6f6f 746f 7574 2c20 666c 6167  nlyrootout, flag
+000052e0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+000052f0: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
 00005300: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00005310: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
-00005320: 6265 2063 616c 6c65 6420 6f66 660a 2d72  be called off.-r
-00005330: 772c 202d 2d72 6577 6569 6768 742c 2066  w, --reweight, f
-00005340: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005350: 6865 7220 746f 2072 6563 616c 6375 6c61  her to recalcula
-00005360: 7465 2074 6865 2077 6569 6768 7420 7065  te the weight pe
-00005370: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
-00005380: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00005390: 7365 742c 2074 6865 2077 6569 6768 7420  set, the weight 
-000053a0: 776f 756c 6420 6265 2072 6563 616c 6375  would be recalcu
-000053b0: 6c61 7465 640a 2d6f 722c 202d 2d6f 6e6c  lated.-or, --onl
-000053c0: 7972 6f6f 746f 7574 2c20 666c 6167 206f  yrootout, flag o
-000053d0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000053e0: 6f20 6f6e 6c79 2063 6f6e 6475 6374 2072  o only conduct r
-000053f0: 6174 6520 636f 7272 6563 7469 6f6e 2075  ate correction u
-00005400: 7369 6e67 2074 6865 206f 7574 6772 6f75  sing the outgrou
-00005410: 7020 6174 2072 6f6f 7420 6173 206f 7574  p at root as out
-00005420: 6772 6f75 702c 2069 6620 7468 6520 666c  group, if the fl
-00005430: 6167 2077 6173 2073 6574 2c20 6f6e 6c79  ag was set, only
-00005440: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
-00005450: 2072 6f6f 7420 776f 756c 6420 6265 2075   root would be u
-00005460: 7365 6420 6173 206f 7574 6772 6f75 700a  sed as outgroup.
-00005470: 2d65 706b 2c20 2d2d 6578 7472 6170 6172  -epk, --extrapar
-00005480: 616e 6f6d 656b 732c 2065 7874 7261 2070  anomeks, extra p
-00005490: 6172 616e 6f6d 6520 4b73 2064 6174 6120  aranome Ks data 
-000054a0: 746f 2070 6c6f 7420 696e 2074 6865 206d  to plot in the m
-000054b0: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
-000054c0: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
-000054d0: 6e65 0a2d 6170 2c20 2d2d 616e 6368 6f72  ne.-ap, --anchor
-000054e0: 706f 696e 7473 2c20 616e 6368 6f72 706f  points, anchorpo
-000054f0: 696e 7473 2e74 7874 2066 696c 6520 746f  ints.txt file to
-00005500: 2070 6c6f 7420 616e 6368 6f72 204b 7320   plot anchor Ks 
-00005510: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00005520: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
-00005530: 6661 756c 7420 4e6f 6e65 0a2d 706b 2c20  fault None.-pk, 
-00005540: 2d2d 706c 6f74 6b64 652c 2066 6c61 6720  --plotkde, flag 
-00005550: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00005560: 746f 2070 6c6f 7420 6b64 6520 6375 7276  to plot kde curv
-00005570: 6520 6f66 206f 7274 686f 6c6f 676f 7573  e of orthologous
-00005580: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005590: 206f 7665 7220 6869 7374 6f67 7261 6d20   over histogram 
-000055a0: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-000055b0: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
-000055c0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-000055d0: 742c 2074 6865 206b 6465 2063 7572 7665  t, the kde curve
-000055e0: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
-000055f0: 640a 2d70 6167 2c20 2d2d 706c 6f74 6170  d.-pag, --plotap
-00005600: 676d 6d2c 2066 6c61 6720 6f70 7469 6f6e  gmm, flag option
-00005610: 2c20 7768 6574 6865 7220 746f 2070 6572  , whether to per
-00005620: 666f 726d 2061 6e64 2070 6c6f 7420 6d69  form and plot mi
-00005630: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
-00005640: 6620 616e 6368 6f72 204b 7320 696e 2074  f anchor Ks in t
-00005650: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
-00005660: 7269 6275 7469 6f6e 2c20 6966 2074 6865  ribution, if the
-00005670: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00005680: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
-00005690: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
-000056a0: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
-000056b0: 640a 2d70 656d 2c20 2d2d 706c 6f74 656c  d.-pem, --plotel
-000056c0: 6d6d 2c20 666c 6167 206f 7074 696f 6e2c  mm, flag option,
-000056d0: 2077 6865 7468 6572 2074 6f20 7065 7266   whether to perf
-000056e0: 6f72 6d20 616e 6420 706c 6f74 2065 6c6d  orm and plot elm
-000056f0: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
-00005700: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
-00005710: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
-00005720: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00005730: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00005740: 7365 742c 2074 6865 2065 6c6d 6d20 6d69  set, the elmm mi
-00005750: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
-00005760: 6620 7061 7261 6e6f 6d65 204b 7320 776f  f paranome Ks wo
-00005770: 756c 6420 6265 2070 6c6f 7474 6564 0a2d  uld be plotted.-
-00005780: 632c 202d 2d63 6f6d 706f 6e65 6e74 732c  c, --components,
-00005790: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
-000057a0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
-000057b0: 6f6e 656e 7473 2074 6f20 6669 7420 696e  onents to fit in
-000057c0: 2061 6e63 686f 7220 4b73 206d 6978 7475   anchor Ks mixtu
-000057d0: 7265 206d 6f64 656c 696e 672c 2064 6566  re modeling, def
-000057e0: 6175 6c74 2028 312c 3429 0a2d 786c 2c20  ault (1,4).-xl, 
-000057f0: 2d2d 786c 696d 2c20 7468 6520 7820 6178  --xlim, the x ax
-00005800: 6973 206c 696d 6974 206f 6620 4b73 2064  is limit of Ks d
-00005810: 6973 7472 6962 7574 696f 6e0a 2d79 6c2c  istribution.-yl,
-00005820: 202d 2d79 6c69 6d2c 2074 6865 2079 2061   --ylim, the y a
-00005830: 7869 7320 6c69 6d69 7420 6f66 204b 7320  xis limit of Ks 
-00005840: 6469 7374 7269 6275 7469 6f6e 0a2d 6164  distribution.-ad
-00005850: 6f2c 202d 2d61 646a 7573 746f 7274 686f  o, --adjustortho
-00005860: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00005870: 6865 7468 6572 2074 6f20 6164 6a75 7374  hether to adjust
-00005880: 2074 6865 2068 6973 746f 6772 616d 2068   the histogram h
-00005890: 6569 6768 7420 6f66 206f 7274 686f 6c6f  eight of ortholo
-000058a0: 676f 7573 204b 7320 6173 2074 6f20 6d61  gous Ks as to ma
-000058b0: 7463 6820 7468 6520 6865 6967 6874 206f  tch the height o
-000058c0: 6620 7061 7261 6c6f 676f 7573 204b 732c  f paralogous Ks,
-000058d0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-000058e0: 2073 6574 2c20 7468 6520 6164 6a75 7374   set, the adjust
-000058f0: 6d65 6e74 2077 6f75 6c64 2062 6520 636f  ment would be co
-00005900: 6e64 7563 7465 640a 2d61 6466 2c20 2d2d  nducted.-adf, --
-00005910: 6164 6a75 7374 6661 6374 6f72 2c20 7468  adjustfactor, th
-00005920: 6520 6164 6a75 7374 6d65 6e74 2066 6163  e adjustment fac
-00005930: 746f 7220 6f66 206f 7274 686f 6c6f 676f  tor of orthologo
-00005940: 7573 204b 732c 2064 6566 6175 6c74 2030  us Ks, default 0
-00005950: 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c 7068  .5.-oa, --okalph
-00005960: 612c 2074 6865 206f 7061 6369 7479 206f  a, the opacity o
-00005970: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
-00005980: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
-00005990: 206d 6978 6564 2070 6c6f 742c 2064 6566   mixed plot, def
-000059a0: 6175 6c74 2030 2e35 0a2d 6661 2c20 2d2d  ault 0.5.-fa, --
-000059b0: 666f 6375 7332 616c 6c2c 2073 6574 2066  focus2all, set f
-000059c0: 6f63 616c 2073 7065 6369 6573 2061 6e64  ocal species and
-000059d0: 206c 6574 2073 7065 6369 6573 2070 6169   let species pai
-000059e0: 7220 746f 2062 6520 6265 7477 6565 6e20  r to be between 
-000059f0: 666f 6361 6c20 616e 6420 616c 6c20 7468  focal and all th
-00005a00: 6520 7265 6d61 696e 696e 6720 7370 6563  e remaining spec
-00005a10: 6965 732c 2064 6566 6175 6c74 204e 6f6e  ies, default Non
-00005a20: 650a 2d6b 732c 202d 2d6b 7374 7265 652c  e.-ks, --kstree,
-00005a30: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005a40: 6574 6865 7220 746f 2069 6e66 6572 204b  ether to infer K
-00005a50: 7320 7472 6565 2c20 6966 2074 6865 2066  s tree, if the f
-00005a60: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
-00005a70: 204b 7320 7472 6565 2069 6e66 6572 656e   Ks tree inferen
-00005a80: 6365 2061 6e61 6c79 7369 7320 776f 756c  ce analysis woul
-00005a90: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
-00005aa0: 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e 6361  ock, --onlyconca
-00005ab0: 746b 7374 7265 652c 2066 6c61 6720 6f70  tkstree, flag op
-00005ac0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00005ad0: 206f 6e6c 7920 696e 6665 7220 4b73 2074   only infer Ks t
-00005ae0: 7265 6520 756e 6465 7220 636f 6e63 6174  ree under concat
-00005af0: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
-00005b00: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005b10: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
-00005b20: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
-00005b30: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
-00005b40: 6d65 6e74 2077 6f75 6c64 2062 6520 6361  ment would be ca
-00005b50: 6c63 756c 6174 6564 0a2d 6373 2c20 2d2d  lculated.-cs, --
-00005b60: 636c 6173 7369 632c 2066 6c61 6720 6f70  classic, flag op
-00005b70: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00005b80: 2064 7261 7720 6d69 7865 6420 4b73 2070   draw mixed Ks p
-00005b90: 6c6f 7420 696e 2061 2063 6c61 7373 6963  lot in a classic
-00005ba0: 206d 616e 6e65 7220 7768 6572 6520 7468   manner where th
-00005bb0: 6520 6675 6c6c 206f 7274 686f 6c6f 676f  e full orthologo
-00005bc0: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
-00005bd0: 6f6e 2069 7320 6472 6177 6564 2c20 6966  on is drawed, if
-00005be0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00005bf0: 742c 2074 6865 2063 6c61 7373 6963 206d  t, the classic m
-00005c00: 6978 6564 204b 7320 706c 6f74 2077 6f75  ixed Ks plot wou
-00005c10: 6c64 2062 6520 6472 6177 6e0a 2d74 612c  ld be drawn.-ta,
-00005c20: 202d 2d74 6f70 6172 726f 772c 2066 6c61   --toparrow, fla
-00005c30: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005c40: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
-00005c50: 6172 726f 7720 746f 2062 6520 6174 2074  arrow to be at t
-00005c60: 6865 2074 6f70 206f 6620 7468 6520 706c  he top of the pl
-00005c70: 6f74 2c20 696e 7374 6561 6420 6f66 2062  ot, instead of b
-00005c80: 6569 6e67 2063 6f6f 7264 696e 6174 6564  eing coordinated
-00005c90: 2061 7320 7468 6520 4b44 4520 6f66 2074   as the KDE of t
-00005ca0: 6865 206f 7274 686f 6c6f 676f 7573 204b  he orthologous K
-00005cb0: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00005cc0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00005cd0: 7365 742c 2074 6865 2061 7272 6f77 2077  set, the arrow w
-00005ce0: 6f75 6c64 2062 6520 7365 7420 6174 2074  ould be set at t
-00005cf0: 6865 2074 6f70 0a2d 6273 2c20 2d2d 626f  he top.-bs, --bo
-00005d00: 6f74 7374 7261 702c 2074 6865 206e 756d  otstrap, the num
-00005d10: 6265 7220 6f66 2062 6f6f 7473 7472 6170  ber of bootstrap
-00005d20: 2072 6570 6c69 6361 7465 7320 6f66 206f   replicates of o
-00005d30: 7274 686f 6c6f 6720 4b73 2064 6973 7472  rtholog Ks distr
-00005d40: 6962 7574 696f 6e20 696e 206d 6978 6564  ibution in mixed
-00005d50: 2070 6c6f 740a 6060 600a 0a54 6865 2070   plot.```..The p
-00005d60: 726f 6772 616d 2060 7767 6420 6d69 7860  rogram `wgd mix`
-00005d70: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
-00005d80: 206d 6978 7475 7265 206d 6f64 656c 2063   mixture model c
-00005d90: 6c75 7374 6572 696e 6720 616e 616c 7973  lustering analys
-00005da0: 6973 206f 6620 2a4b 2a3c 7375 623e 533c  is of *K*<sub>S<
-00005db0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00005dc0: 6275 7469 6f6e 2e0a 6060 600a 7767 6420  bution..```.wgd 
-00005dd0: 6d69 7820 6b73 5f64 6174 6166 696c 6520  mix ks_datafile 
-00005de0: 286f 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d  (option).-------
-00005df0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005e00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005e10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005e20: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005e30: 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20 2d2d  ---------.-f, --
-00005e40: 6669 6c74 6572 732c 2074 6865 2063 7574  filters, the cut
-00005e50: 6f66 6620 616c 6967 6e6d 656e 7420 6c65  off alignment le
-00005e60: 6e67 7468 2c20 6465 6661 756c 7420 3330  ngth, default 30
-00005e70: 300a 2d72 2c20 2d2d 6b73 5f72 616e 6765  0.-r, --ks_range
-00005e80: 2c20 7468 6520 4b73 2072 616e 6765 2074  , the Ks range t
-00005e90: 6f20 6265 2063 6f6e 7369 6465 7265 642c  o be considered,
-00005ea0: 2064 6566 6175 6c74 2028 302c 2035 290a   default (0, 5).
-00005eb0: 2d62 2c20 2d2d 6269 6e73 2c20 7468 6520  -b, --bins, the 
-00005ec0: 6e75 6d62 6572 206f 6620 6269 6e73 2069  number of bins i
-00005ed0: 6e20 4b73 2064 6973 7472 6962 7574 696f  n Ks distributio
-00005ee0: 6e2c 2064 6566 6175 6c74 2035 300a 2d6f  n, default 50.-o
-00005ef0: 2c20 2d2d 6f75 7464 6972 2c20 7468 6520  , --outdir, the 
-00005f00: 6f75 7470 7574 2064 6972 6563 746f 7279  output directory
-00005f10: 2c20 6465 6661 756c 7420 7767 645f 6d69  , default wgd_mi
-00005f20: 780a 2d2d 6d65 7468 6f64 2c20 7768 6963  x.--method, whic
-00005f30: 6820 6d69 7874 7572 6520 6d6f 6465 6c20  h mixture model 
-00005f40: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
-00005f50: 676d 6d0a 2d6e 2c20 2d2d 636f 6d70 6f6e  gmm.-n, --compon
-00005f60: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
-00005f70: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
-00005f80: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
-00005f90: 6974 2c20 6465 6661 756c 7420 2831 2c20  it, default (1, 
-00005fa0: 3429 0a2d 672c 202d 2d67 616d 6d61 2c20  4).-g, --gamma, 
-00005fb0: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
-00005fc0: 7465 7220 666f 7220 6267 6d6d 206d 6f64  ter for bgmm mod
-00005fd0: 656c 732c 2064 6566 6175 6c74 2030 2e30  els, default 0.0
-00005fe0: 3031 0a2d 6e69 2c20 2d2d 6e5f 696e 6974  01.-ni, --n_init
-00005ff0: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
-00006000: 6b2d 6d65 616e 7320 696e 6974 6961 6c69  k-means initiali
-00006010: 7a61 7469 6f6e 732c 2064 6566 6175 6c74  zations, default
-00006020: 2032 3030 0a2d 6d69 2c20 2d2d 6d61 785f   200.-mi, --max_
-00006030: 6974 6572 2c20 7468 6520 6d61 7869 6d75  iter, the maximu
-00006040: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
-00006050: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
-00006060: 3230 300a 6060 600a 0a54 6865 2070 726f  200.```..The pro
-00006070: 6772 616d 2060 7767 6420 7065 616b 6020  gram `wgd peak` 
-00006080: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
-00006090: 7365 6172 6368 206f 6620 6372 6564 6961  search of credia
-000060a0: 626c 6520 2a4b 2a3c 7375 623e 533c 2f73  ble *K*<sub>S</s
-000060b0: 7562 3e20 7261 6e67 6520 7573 6564 2069  ub> range used i
-000060c0: 6e20 5747 4420 6461 7469 6e67 2e0a 6060  n WGD dating..``
-000060d0: 600a 7767 6420 7065 616b 206b 735f 6461  `.wgd peak ks_da
-000060e0: 7461 6669 6c65 2028 6f70 7469 6f6e 290a  tafile (option).
-000060f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006100: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006110: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006120: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006130: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006140: 0a2d 6170 2c20 2d2d 616e 6368 6f72 706f  .-ap, --anchorpo
-00006150: 696e 7473 2c20 7468 6520 616e 6368 6f72  ints, the anchor
-00006160: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
-00006170: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00006180: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
-00006190: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
-000061a0: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
-000061b0: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
-000061c0: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
-000061d0: 7473 6567 6d65 6e74 7320 6461 7461 6669  tsegments datafi
-000061e0: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
-000061f0: 200a 2d6d 702c 202d 2d6d 756c 7469 706c   .-mp, --multipl
-00006200: 6963 6f6e 7061 6972 732c 2074 6865 206d  iconpairs, the m
-00006210: 756c 7469 706c 6963 6f6e 7061 6972 7320  ultipliconpairs 
-00006220: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-00006230: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
-00006240: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
-00006250: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
-00006260: 6c74 2077 6764 5f70 6561 6b0a 2d61 662c  lt wgd_peak.-af,
-00006270: 202d 2d61 6c69 676e 6669 6c74 6572 2c20   --alignfilter, 
-00006280: 6375 746f 6666 2066 6f72 2061 6c69 676e  cutoff for align
-00006290: 6d65 6e74 2069 6465 6e74 6974 792c 206c  ment identity, l
-000062a0: 656e 6774 6820 616e 6420 636f 7665 7261  ength and covera
-000062b0: 6765 2c20 6465 6661 756c 7420 302e 302c  ge, default 0.0,
-000062c0: 2030 2c20 302e 300a 2d72 2c20 2d2d 6b73   0, 0.0.-r, --ks
-000062d0: 7261 6e67 652c 2072 616e 6765 206f 6620  range, range of 
-000062e0: 4b73 2074 6f20 6265 2061 6e61 6c79 7a65  Ks to be analyze
-000062f0: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
-00006300: 290a 2d62 772c 202d 2d62 696e 5f77 6964  ).-bw, --bin_wid
-00006310: 7468 2c20 6261 6e64 7769 6474 6820 6f66  th, bandwidth of
-00006320: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00006330: 2c20 6465 6661 756c 7420 302e 310a 2d69  , default 0.1.-i
-00006340: 632c 202d 2d77 6569 6768 7473 5f6f 7574  c, --weights_out
-00006350: 6c69 6572 735f 696e 636c 7564 6564 2c20  liers_included, 
-00006360: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00006370: 7468 6572 2074 6f20 696e 636c 7564 6520  ther to include 
-00006380: 4b73 206f 7574 6c69 6572 732c 2069 6620  Ks outliers, if 
-00006390: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-000063a0: 2c20 4b73 206f 7574 6c69 6572 7320 776f  , Ks outliers wo
-000063b0: 756c 6420 6265 2069 6e63 6c75 6465 6420  uld be included 
-000063c0: 696e 2074 6865 2061 6e61 6c79 7369 730a  in the analysis.
-000063d0: 2d6d 2c20 2d2d 6d65 7468 6f64 2c20 7768  -m, --method, wh
-000063e0: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
-000063f0: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
-00006400: 7420 676d 6d0a 2d2d 7365 6564 2c20 7261  t gmm.--seed, ra
-00006410: 6e64 6f6d 2073 6565 6420 6769 7665 6e20  ndom seed given 
-00006420: 746f 2069 6e69 7469 616c 697a 6174 696f  to initializatio
-00006430: 6e2c 2064 6566 6175 6c74 2032 3335 3238  n, default 23528
-00006440: 3930 0a2d 6569 2c20 2d2d 656d 5f69 7465  90.-ei, --em_ite
-00006450: 722c 2074 6865 206e 756d 6265 7220 6f66  r, the number of
-00006460: 2045 4d20 6974 6572 6174 696f 6e73 2074   EM iterations t
-00006470: 6f20 7065 7266 6f72 6d2c 2064 6566 6175  o perform, defau
-00006480: 6c74 2032 3030 0a2d 6e69 2c20 2d2d 6e5f  lt 200.-ni, --n_
-00006490: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
-000064a0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-000064b0: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
-000064c0: 6175 6c74 2032 3030 0a2d 6e2c 202d 2d63  ault 200.-n, --c
-000064d0: 6f6d 706f 6e65 6e74 732c 2074 6865 2072  omponents, the r
-000064e0: 616e 6765 206f 6620 7468 6520 6e75 6d62  ange of the numb
-000064f0: 6572 206f 6620 636f 6d70 6f6e 656e 7473  er of components
-00006500: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
-00006510: 2028 312c 2034 290a 2d67 2c20 2d2d 6761   (1, 4).-g, --ga
-00006520: 6d6d 612c 2074 6865 2067 616d 6d61 2070  mma, the gamma p
-00006530: 6172 616d 6574 6572 2066 6f72 2062 676d  arameter for bgm
-00006540: 6d20 6d6f 6465 6c73 2c20 6465 6661 756c  m models, defaul
-00006550: 7420 3165 2d33 0a2d 2d62 6f6f 7473 2c20  t 1e-3.--boots, 
-00006560: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
-00006570: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
-00006580: 6573 206f 6620 6b64 652c 2064 6566 6175  es of kde, defau
-00006590: 6c74 2032 3030 0a2d 2d77 6569 6768 7465  lt 200.--weighte
-000065a0: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
-000065b0: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
-000065c0: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
-000065d0: 686f 6420 6f66 2064 652d 7265 6475 6e64  hod of de-redund
-000065e0: 616e 6379 2c20 6966 2074 6865 2066 6c61  ancy, if the fla
-000065f0: 6720 7761 7320 7365 742c 2074 6865 206e  g was set, the n
-00006600: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
-00006610: 686f 6420 776f 756c 6420 6265 2075 7365  hod would be use
-00006620: 640a 2d70 2c20 2d2d 706c 6f74 2c20 7468  d.-p, --plot, th
-00006630: 6520 706c 6f74 7469 6e67 206d 6574 686f  e plotting metho
-00006640: 6420 746f 2062 6520 7573 6564 2c20 6465  d to be used, de
-00006650: 6661 756c 7420 6964 656e 7469 6361 6c0a  fault identical.
-00006660: 2d62 6d2c 202d 2d62 775f 6d65 7468 6f64  -bm, --bw_method
-00006670: 2c20 7468 6520 6261 6e64 7769 6474 6820  , the bandwidth 
-00006680: 6d65 7468 6f64 2074 6f20 6265 2075 7365  method to be use
-00006690: 6420 696e 2061 6e61 6c79 7a69 6e67 2074  d in analyzing t
-000066a0: 6865 2070 6561 6b20 6f66 2057 4744 2064  he peak of WGD d
-000066b0: 6174 6573 2c20 6465 6661 756c 7420 7369  ates, default si
-000066c0: 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65 646f  lverman.--n_medo
-000066d0: 6964 732c 2074 6865 206e 756d 6265 7220  ids, the number 
-000066e0: 6f66 206d 6564 6f69 6473 2074 6f20 6669  of medoids to fi
-000066f0: 742c 2064 6566 6175 6c74 2032 0a2d 6b6d  t, default 2.-km
-00006700: 2c20 2d2d 6b64 656d 6574 686f 642c 2074  , --kdemethod, t
-00006710: 6865 206b 6465 206d 6574 686f 6420 746f  he kde method to
-00006720: 2062 6520 7573 6564 2069 6e20 616e 616c   be used in anal
-00006730: 797a 696e 6720 7468 6520 7065 616b 206f  yzing the peak o
-00006740: 6620 5747 4420 6461 7465 732c 206b 6d65  f WGD dates, kme
-00006750: 646f 6964 7320 616e 616c 7973 6973 206f  doids analysis o
-00006760: 7220 7468 6520 6261 7369 6320 4b73 2070  r the basic Ks p
-00006770: 6c6f 7474 696e 672c 2064 6566 6175 6c74  lotting, default
-00006780: 2073 6369 7079 0a2d 2d6e 5f63 6c75 7374   scipy.--n_clust
-00006790: 6572 732c 2074 6865 206e 756d 6265 7220  ers, the number 
-000067a0: 6f66 2063 6c75 7374 6572 7320 746f 2070  of clusters to p
-000067b0: 6c6f 7420 456c 626f 7720 6c6f 7373 2066  lot Elbow loss f
-000067c0: 756e 6374 696f 6e2c 2064 6566 6175 6c74  unction, default
-000067d0: 2035 0a2d 2d6b 6d65 646f 6964 732c 2066   5.--kmedoids, f
-000067e0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000067f0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00006800: 4b2d 4d65 646f 6964 7320 636c 7573 7465  K-Medoids cluste
-00006810: 7269 6e67 2061 6e61 6c79 7369 732c 2069  ring analysis, i
-00006820: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00006830: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00006840: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00006850: 7465 640a 2d67 642c 202d 2d67 7569 6465  ted.-gd, --guide
-00006860: 2c20 7468 6520 7265 6769 6d65 2072 6573  , the regime res
-00006870: 6964 696e 6720 616e 6368 6f72 732c 2064  iding anchors, d
-00006880: 6566 6175 6c74 2073 6567 6d65 6e74 0a2d  efault segment.-
-00006890: 7072 6374 2c20 2d2d 7072 6f6d 696e 656e  prct, --prominen
-000068a0: 6365 5f63 7574 6f66 662c 2074 6865 2070  ce_cutoff, the p
-000068b0: 726f 6d69 6e65 6e63 6520 6375 746f 6666  rominence cutoff
-000068c0: 206f 6620 6163 6365 7074 6162 6c65 2070   of acceptable p
-000068d0: 6561 6b73 2069 6e20 7065 616b 2066 696e  eaks in peak fin
-000068e0: 6469 6e67 2073 7465 7073 2c20 6465 6661  ding steps, defa
-000068f0: 756c 7420 302e 310a 2d72 682c 202d 2d72  ult 0.1.-rh, --r
-00006900: 656c 5f68 6569 6768 742c 2074 6865 2072  el_height, the r
-00006910: 656c 6174 6976 6520 6865 6967 6874 2061  elative height a
-00006920: 7420 7768 6963 6820 7468 6520 7065 616b  t which the peak
-00006930: 2077 6964 7468 2069 7320 6d65 6173 7572   width is measur
-00006940: 6564 2c20 6465 6661 756c 7420 302e 340a  ed, default 0.4.
-00006950: 2d6b 642c 202d 2d6b 7374 6f64 6174 652c  -kd, --kstodate,
-00006960: 2074 6865 2072 616e 6765 206f 6620 4b73   the range of Ks
-00006970: 2074 6f20 6265 2064 6174 6564 2069 6e20   to be dated in 
-00006980: 6865 7572 6973 7469 6320 7365 6172 6368  heuristic search
-00006990: 2c20 6465 6661 756c 7420 2830 2e35 2c20  , default (0.5, 
-000069a0: 312e 3529 0a2d 786c 2c20 2d2d 786c 696d  1.5).-xl, --xlim
-000069b0: 2c20 7468 6520 7820 6178 6973 206c 696d  , the x axis lim
-000069c0: 6974 206f 6620 474d 4d20 4b73 2064 6973  it of GMM Ks dis
-000069d0: 7472 6962 7574 696f 6e0a 2d79 6c2c 202d  tribution.-yl, -
-000069e0: 2d79 6c69 6d2c 2074 6865 2079 2061 7869  -ylim, the y axi
-000069f0: 7320 6c69 6d69 7420 6f66 2047 4d4d 204b  s limit of GMM K
-00006a00: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-00006a10: 2d6d 616e 7561 6c73 6574 2c20 666c 6167  -manualset, flag
-00006a20: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00006a30: 2074 6f20 6f75 7470 7574 2061 6e63 686f   to output ancho
-00006a40: 7220 7061 6972 7320 7769 7468 206d 616e  r pairs with man
-00006a50: 7561 6c6c 7920 7365 7420 4b73 2072 616e  ually set Ks ran
-00006a60: 6765 2c20 6966 2074 6865 2066 6c61 6720  ge, if the flag 
-00006a70: 7761 7320 7365 742c 206d 616e 7561 6c6c  was set, manuall
-00006a80: 7920 7365 7420 4b73 2072 616e 6765 2077  y set Ks range w
-00006a90: 6f75 6c64 2062 6520 6f75 7470 7574 7465  ould be outputte
-00006aa0: 640a 2d2d 6369 2c20 7468 6520 636f 6e66  d.--ci, the conf
-00006ab0: 6964 656e 6365 206c 6576 656c 206f 6620  idence level of 
-00006ac0: 6c6f 672d 6e6f 726d 616c 2064 6973 7472  log-normal distr
-00006ad0: 6962 7574 696f 6e20 746f 2064 6174 652c  ibution to date,
-00006ae0: 2064 6566 6175 6c74 2039 350a 2d2d 6864   default 95.--hd
-00006af0: 722c 2074 6865 2068 6967 6865 7374 2064  r, the highest d
-00006b00: 656e 7369 7479 2072 6567 696f 6e20 2848  ensity region (H
-00006b10: 4452 2920 6170 706c 6965 6420 696e 2074  DR) applied in t
-00006b20: 6865 2073 6567 6d65 6e74 2d67 7569 6465  he segment-guide
-00006b30: 6420 616e 6368 6f72 2070 6169 7220 4b73  d anchor pair Ks
-00006b40: 2064 6973 7472 6962 7574 696f 6e2c 2064   distribution, d
-00006b50: 6566 6175 6c74 2039 350a 2d2d 6865 7572  efault 95.--heur
-00006b60: 6973 7469 632c 2066 6c61 6720 6f70 7469  istic, flag opti
-00006b70: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00006b80: 6e69 7469 6174 6520 6865 7572 6973 7469  nitiate heuristi
-00006b90: 6320 6d65 7468 6f64 206f 6620 6465 6669  c method of defi
-00006ba0: 6e69 6e67 2043 4920 666f 7220 6461 7469  ning CI for dati
-00006bb0: 6e67 2c20 6966 2074 6865 2066 6c61 6720  ng, if the flag 
-00006bc0: 7761 7320 7365 742c 2074 6865 2068 6575  was set, the heu
-00006bd0: 7269 7374 6963 206d 6574 686f 6420 776f  ristic method wo
-00006be0: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
-00006bf0: 0a2d 6b63 2c20 2d2d 6b73 6375 746f 6666  .-kc, --kscutoff
-00006c00: 2c20 7468 6520 4b73 2073 6174 7572 6174  , the Ks saturat
-00006c10: 696f 6e20 6375 746f 6666 2069 6e20 6461  ion cutoff in da
-00006c20: 7469 6e67 2c20 6465 6661 756c 7420 350a  ting, default 5.
-00006c30: 2d2d 7368 6f77 6369 2c20 666c 6167 206f  --showci, flag o
-00006c40: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-00006c50: 6f20 7368 6f77 2043 4920 666f 7220 7468  o show CI for th
-00006c60: 6520 6f72 6967 696e 616c 2061 6e63 686f  e original ancho
-00006c70: 7220 4b73 2047 4d4d 2061 6e61 6c79 7369  r Ks GMM analysi
-00006c80: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00006c90: 6173 2073 6574 2c20 6974 2077 6f75 6c64  as set, it would
-00006ca0: 2062 6520 7368 6f77 6564 0a2d 2d6b 6565   be showed.--kee
-00006cb0: 7074 6d70 6669 672c 2066 6c61 6720 6f70  ptmpfig, flag op
-00006cc0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00006cd0: 206b 6565 7020 7465 6d70 6f72 6172 7920   keep temporary 
-00006ce0: 6669 6775 7265 7320 696e 2070 6561 6b20  figures in peak 
-00006cf0: 6669 6e64 696e 6720 7072 6f63 6573 732c  finding process,
-00006d00: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00006d10: 2073 6574 2c20 7468 6f73 6520 6669 6775   set, those figu
-00006d20: 7265 7320 776f 756c 6420 6265 206b 6570  res would be kep
-00006d30: 740a 6060 600a 0a54 6865 2070 726f 6772  t.```..The progr
-00006d40: 616d 2060 7767 6420 7379 6e60 2063 616e  am `wgd syn` can
-00006d50: 2072 6561 6c69 7a65 2074 6865 2069 6e74   realize the int
-00006d60: 7261 2d20 616e 6420 696e 7465 722d 7370  ra- and inter-sp
-00006d70: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
-00006d80: 6e66 6572 656e 6365 2e0a 6060 600a 7767  nference..```.wg
-00006d90: 6420 7379 6e20 6661 6d69 6c69 6573 2067  d syn families g
-00006da0: 6666 7320 286f 7074 696f 6e29 0a2d 2d2d  ffs (option).---
-00006db0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006dc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006dd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006de0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006df0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d6b  -------------.-k
-00006e00: 732c 202d 2d6b 735f 6469 7374 7269 6275  s, --ks_distribu
-00006e10: 7469 6f6e 2c20 6b73 2064 6973 7472 6962  tion, ks distrib
-00006e20: 7574 696f 6e20 6461 7461 6669 6c65 2c20  ution datafile, 
-00006e30: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
-00006e40: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
-00006e50: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
-00006e60: 2064 6566 6175 6c74 2077 6764 5f73 796e   default wgd_syn
-00006e70: 0a2d 662c 202d 2d66 6561 7475 7265 2c20  .-f, --feature, 
-00006e80: 7468 6520 6665 6174 7572 6520 666f 7220  the feature for 
-00006e90: 7061 7273 696e 6720 6765 6e65 2049 4473  parsing gene IDs
-00006ea0: 2066 726f 6d20 4746 4620 6669 6c65 732c   from GFF files,
-00006eb0: 2064 6566 6175 6c74 2067 656e 650a 2d61   default gene.-a
-00006ec0: 2c20 2d2d 6174 7472 6962 7574 652c 2074  , --attribute, t
-00006ed0: 6865 2061 7474 7269 6275 7465 2066 6f72  he attribute for
-00006ee0: 2070 6172 7369 6e67 2074 6865 2067 656e   parsing the gen
-00006ef0: 6520 4944 7320 6672 6f6d 2074 6865 2047  e IDs from the G
-00006f00: 4646 2066 696c 6573 2c20 6465 6661 756c  FF files, defaul
-00006f10: 7420 4944 0a2d 6174 672c 202d 2d61 6464  t ID.-atg, --add
-00006f20: 6974 696f 6e61 6c67 6666 696e 666f 2c20  itionalgffinfo, 
-00006f30: 7468 6520 6665 6174 7572 6520 616e 6420  the feature and 
-00006f40: 6174 7472 6962 7574 6520 696e 666f 726d  attribute inform
-00006f50: 6174 696f 6e20 6f66 2061 6464 6974 696f  ation of additio
-00006f60: 6e61 6c20 6766 6633 2066 696c 6573 2069  nal gff3 files i
-00006f70: 6620 6469 6666 6572 656e 7420 696e 2074  f different in t
-00006f80: 6865 2066 6f72 6d61 7420 6f66 2028 6665  he format of (fe
-00006f90: 6174 7572 653b 6174 7472 6962 7574 6529  ature;attribute)
-00006fa0: 272c 2064 6566 6175 6c74 204e 6f6e 650a  ', default None.
-00006fb0: 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c 2074  -ml, --minlen, t
-00006fc0: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
-00006fd0: 6820 6f66 2061 2073 6361 6666 6f6c 6420  h of a scaffold 
-00006fe0: 746f 2062 6520 696e 636c 7564 6564 2069  to be included i
-00006ff0: 6e20 646f 7470 6c6f 742c 2064 6566 6175  n dotplot, defau
-00007000: 6c74 202d 312c 2069 6620 2d31 2077 6173  lt -1, if -1 was
-00007010: 2073 6574 2c20 7468 6520 3130 2520 6f66   set, the 10% of
-00007020: 2074 6865 206c 6f6e 6765 7374 2073 6361   the longest sca
-00007030: 6666 6f6c 6420 776f 756c 6420 6265 2073  ffold would be s
-00007040: 6574 0a2d 6d73 2c20 2d2d 6d61 7873 697a  et.-ms, --maxsiz
-00007050: 652c 2074 6865 206d 6178 696d 756d 2066  e, the maximum f
-00007060: 616d 696c 7920 7369 7a65 2074 6f20 6265  amily size to be
-00007070: 2069 6e63 6c75 6465 642c 2064 6566 6175   included, defau
-00007080: 6c74 2032 3030 0a2d 722c 202d 2d6b 735f  lt 200.-r, --ks_
-00007090: 7261 6e67 652c 2074 6865 204b 7320 7261  range, the Ks ra
-000070a0: 6e67 6520 696e 2063 6f6c 6f72 6564 2064  nge in colored d
-000070b0: 6f74 706c 6f74 2c20 6465 6661 756c 7420  otplot, default 
-000070c0: 2830 2c20 3529 0a2d 2d69 6164 686f 7265  (0, 5).--iadhore
-000070d0: 5f6f 7074 696f 6e73 2c20 7468 6520 7061  _options, the pa
-000070e0: 7261 6d65 7465 7220 7365 7474 696e 6720  rameter setting 
-000070f0: 696e 2069 6164 686f 7265 2c20 6465 6661  in iadhore, defa
-00007100: 756c 7420 6173 2061 2073 7472 696e 6720  ult as a string 
-00007110: 6f66 206c 656e 6774 6820 7a65 726f 0a2d  of length zero.-
-00007120: 6d67 2c20 2d2d 6d69 6e73 6567 6c65 6e2c  mg, --minseglen,
-00007130: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
-00007140: 6774 6820 6f66 2073 6567 6d65 6e74 7320  gth of segments 
-00007150: 746f 2069 6e63 6c75 6465 2069 6e20 7261  to include in ra
-00007160: 7469 6f20 6966 203c 3d20 312c 2064 6566  tio if <= 1, def
-00007170: 6175 6c74 2031 3030 3030 0a2d 6b72 2c20  ault 10000.-kr, 
-00007180: 2d2d 6b65 6570 7265 6475 6e2c 2066 6c61  --keepredun, fla
-00007190: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000071a0: 7220 746f 206b 6565 7020 7265 6475 6e64  r to keep redund
-000071b0: 616e 7420 6d75 6c74 6970 6c69 636f 6e73  ant multiplicons
-000071c0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-000071d0: 7320 7365 742c 2074 6865 2072 6564 756e  s set, the redun
-000071e0: 6461 6e74 206d 756c 7469 706c 6963 6f6e  dant multiplicon
-000071f0: 7320 776f 756c 6420 6265 206b 6570 740a  s would be kept.
-00007200: 2d6d 676e 2c20 2d2d 6d69 6e67 656e 656e  -mgn, --mingenen
-00007210: 756d 2c20 7468 6520 6d69 6e69 6d75 6d20  um, the minimum 
-00007220: 6e75 6d62 6572 206f 6620 6765 6e65 7320  number of genes 
-00007230: 666f 7220 6120 7365 676d 656e 7420 746f  for a segment to
-00007240: 2062 6520 636f 6e73 6964 6572 6564 2c20   be considered, 
-00007250: 6465 6661 756c 7420 3330 0a2d 6473 2c20  default 30.-ds, 
-00007260: 2d2d 646f 7473 697a 652c 2074 6865 2064  --dotsize, the d
-00007270: 6f74 2073 697a 6520 696e 2064 6f74 2070  ot size in dot p
-00007280: 6c6f 742c 2064 6566 6175 6c74 2030 2e33  lot, default 0.3
-00007290: 0a2d 6161 2c20 2d2d 6170 616c 7068 612c  .-aa, --apalpha,
-000072a0: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-000072b0: 616e 6368 6f72 2064 6f74 732c 2064 6566  anchor dots, def
-000072c0: 6175 6c74 2031 0a2d 6861 2c20 2d2d 686f  ault 1.-ha, --ho
-000072d0: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-000072e0: 7479 206f 6620 686f 6d6f 6c6f 6720 646f  ty of homolog do
-000072f0: 7473 2c20 6465 6661 756c 7420 300a 2d73  ts, default 0.-s
-00007300: 7274 2c20 2d2d 7368 6f77 7265 616c 7469  rt, --showrealti
-00007310: 636b 2c20 666c 6167 206f 7074 696f 6e2c  ck, flag option,
-00007320: 2077 6865 7468 6572 2074 6f20 7368 6f77   whether to show
-00007330: 2074 6865 2072 6561 6c20 7469 636b 2069   the real tick i
-00007340: 6e20 6765 6e65 7320 6f72 2062 6173 6573  n genes or bases
-00007350: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007360: 7320 7365 742c 2074 6865 2072 6561 6c20  s set, the real 
-00007370: 7469 636b 2077 6f75 6c64 2062 6520 7368  tick would be sh
-00007380: 6f77 6564 0a2d 746c 732c 202d 2d74 6963  owed.-tls, --tic
-00007390: 6b6c 6162 656c 7369 7a65 2c20 7468 6520  klabelsize, the 
-000073a0: 6c61 6265 6c20 7369 7a65 206f 6620 7469  label size of ti
-000073b0: 636b 2c20 6465 6661 756c 7420 350a 2d67  ck, default 5.-g
-000073c0: 722c 202d 2d67 6973 7472 622c 2066 6c61  r, --gistrb, fla
-000073d0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000073e0: 7220 746f 2075 7365 2067 6973 745f 7261  r to use gist_ra
-000073f0: 696e 626f 7720 6173 2063 6f6c 6f72 206d  inbow as color m
-00007400: 6170 206f 6620 646f 7470 6c6f 740a 2d6e  ap of dotplot.-n
-00007410: 2c20 2d2d 6e74 6872 6561 6473 2c20 7468  , --nthreads, th
-00007420: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
-00007430: 6164 7320 746f 2075 7365 2069 6e20 7379  ads to use in sy
-00007440: 6e74 656e 7920 696e 6665 7265 6e63 652c  nteny inference,
-00007450: 2064 6566 6175 6c74 2034 0a60 6060 0a0a   default 4.```..
-00007460: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
-00007470: 2076 697a 6020 6361 6e20 7265 616c 697a   viz` can realiz
-00007480: 6520 7468 6520 7669 7375 616c 697a 6174  e the visualizat
-00007490: 696f 6e20 6f66 202a 4b2a 3c73 7562 3e53  ion of *K*<sub>S
-000074a0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-000074b0: 6962 7574 696f 6e20 616e 6420 7379 6e74  ibution and synt
-000074c0: 656e 792e 0a60 6060 0a77 6764 2076 697a  eny..```.wgd viz
-000074d0: 2028 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d   (option).------
-000074e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000074f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007500: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007510: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007520: 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 642c 202d  ----------.-d, -
-00007530: 2d64 6174 6166 696c 652c 2074 6865 204b  -datafile, the K
-00007540: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
-00007550: 756c 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f  ult None.-o, --o
-00007560: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
-00007570: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
-00007580: 6175 6c74 2077 6764 5f76 697a 0a2d 7372  ault wgd_viz.-sr
-00007590: 2c20 2d2d 7370 6169 722c 2074 6865 2073  , --spair, the s
-000075a0: 7065 6369 6573 2070 6169 7220 746f 2062  pecies pair to b
-000075b0: 6520 706c 6f74 7465 642c 2064 6566 6175  e plotted, defau
-000075c0: 6c74 204e 6f6e 652c 2074 6869 7320 6f70  lt None, this op
-000075d0: 7469 6f6e 2063 616e 2062 6520 7072 6f76  tion can be prov
-000075e0: 6964 6564 206d 756c 7469 706c 6520 7469  ided multiple ti
-000075f0: 6d65 730a 2d66 612c 202d 2d66 6f63 7573  mes.-fa, --focus
-00007600: 3261 6c6c 2c20 7365 7420 666f 6361 6c20  2all, set focal 
-00007610: 7370 6563 6965 7320 616e 6420 6c65 7420  species and let 
-00007620: 7370 6563 6965 7320 7061 6972 2074 6f20  species pair to 
-00007630: 6265 2062 6574 7765 656e 2066 6f63 616c  be between focal
-00007640: 2061 6e64 2061 6c6c 2074 6865 2072 656d   and all the rem
-00007650: 6169 6e69 6e67 2073 7065 6369 6573 2c20  aining species, 
-00007660: 6465 6661 756c 7420 4e6f 6e65 0a2d 6773  default None.-gs
-00007670: 2c20 2d2d 6773 6d61 702c 2074 6865 2067  , --gsmap, the g
-00007680: 656e 6520 6e61 6d65 2d73 7065 6369 6573  ene name-species
-00007690: 206e 616d 6520 6d61 702c 2064 6566 6175   name map, defau
-000076a0: 6c74 204e 6f6e 650a 2d73 702c 202d 2d73  lt None.-sp, --s
-000076b0: 7065 6369 6573 7472 6565 2c20 7468 6520  peciestree, the 
-000076c0: 7370 6563 6965 7320 7472 6565 2074 6f20  species tree to 
-000076d0: 7065 7266 6f72 6d20 7261 7465 2063 6f72  perform rate cor
-000076e0: 7265 6374 696f 6e2c 2064 6566 6175 6c74  rection, default
-000076f0: 204e 6f6e 652c 2069 6620 4e6f 6e65 2077   None, if None w
-00007700: 6173 2067 6976 656e 2c20 7468 6520 7261  as given, the ra
-00007710: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
-00007720: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-00007730: 6361 6c6c 6564 206f 6666 0a2d 706b 2c20  called off.-pk, 
-00007740: 2d2d 706c 6f74 6b64 652c 2066 6c61 6720  --plotkde, flag 
-00007750: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00007760: 746f 2070 6c6f 7420 6b64 6520 6375 7276  to plot kde curv
-00007770: 6520 7570 6f6e 2068 6973 746f 6772 616d  e upon histogram
-00007780: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007790: 7320 7365 742c 206b 6465 2063 7572 7665  s set, kde curve
-000077a0: 2077 6f75 6c64 2062 6520 6164 6465 640a   would be added.
-000077b0: 2d72 772c 202d 2d72 6577 6569 6768 742c  -rw, --reweight,
-000077c0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-000077d0: 6574 6865 7220 746f 2072 6563 616c 6375  ether to recalcu
-000077e0: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
-000077f0: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
-00007800: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007810: 7320 7365 742c 2074 6865 2077 6569 6768  s set, the weigh
-00007820: 7420 776f 756c 6420 6265 2072 6563 616c  t would be recal
-00007830: 6375 6c61 7465 640a 2d6f 722c 202d 2d6f  culated.-or, --o
-00007840: 6e6c 7972 6f6f 746f 7574 2c20 666c 6167  nlyrootout, flag
-00007850: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00007860: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
-00007870: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00007880: 2075 7369 6e67 2074 6865 206f 7574 6772   using the outgr
-00007890: 6f75 7020 6174 2072 6f6f 7420 6173 206f  oup at root as o
-000078a0: 7574 6772 6f75 702c 2069 6620 7468 6520  utgroup, if the 
-000078b0: 666c 6167 2077 6173 2073 6574 2c20 6f6e  flag was set, on
-000078c0: 6c79 2074 6865 206f 7574 6772 6f75 7020  ly the outgroup 
-000078d0: 6174 2072 6f6f 7420 776f 756c 6420 6265  at root would be
-000078e0: 2075 7365 6420 6173 206f 7574 6772 6f75   used as outgrou
-000078f0: 700a 2d69 7465 722c 202d 2d65 6d5f 6974  p.-iter, --em_it
-00007900: 6572 6174 696f 6e73 2c20 7468 6520 6d61  erations, the ma
-00007910: 7869 6d75 6d20 454d 2069 7465 7261 7469  ximum EM iterati
-00007920: 6f6e 732c 2064 6566 6175 6c74 2032 3030  ons, default 200
-00007930: 0a2d 696e 6974 2c20 2d2d 656d 5f69 6e69  .-init, --em_ini
-00007940: 7469 616c 697a 6174 696f 6e73 2c20 7468  tializations, th
-00007950: 6520 6d61 7869 6d75 6d20 454d 2069 6e69  e maximum EM ini
-00007960: 7469 616c 697a 6174 696f 6e73 2c20 6465  tializations, de
-00007970: 6661 756c 7420 3230 300a 2d70 7263 742c  fault 200.-prct,
-00007980: 202d 2d70 726f 6d69 6e65 6e63 655f 6375   --prominence_cu
-00007990: 746f 6666 2c20 7468 6520 7072 6f6d 696e  toff, the promin
-000079a0: 656e 6365 2063 7574 6f66 6620 6f66 2061  ence cutoff of a
-000079b0: 6363 6570 7461 626c 6520 7065 616b 732c  cceptable peaks,
-000079c0: 2064 6566 6175 6c74 2030 2e31 0a2d 7268   default 0.1.-rh
-000079d0: 2c20 2d2d 7265 6c5f 6865 6967 6874 2c20  , --rel_height, 
-000079e0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-000079f0: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00007a00: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00007a10: 6561 7375 7265 642c 2064 6566 6175 6c74  easured, default
-00007a20: 2030 2e34 0a2d 736d 2c20 2d2d 7365 676d   0.4.-sm, --segm
-00007a30: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
-00007a40: 7473 2064 6174 6166 696c 652c 2064 6566  ts datafile, def
-00007a50: 6175 6c74 204e 6f6e 650a 2d6d 6c2c 202d  ault None.-ml, -
-00007a60: 2d6d 696e 6c65 6e2c 2074 6865 206d 696e  -minlen, the min
-00007a70: 696d 756d 206c 656e 6774 6820 6f66 2061  imum length of a
-00007a80: 2073 6361 6666 6f6c 6420 746f 2062 6520   scaffold to be 
-00007a90: 696e 636c 7564 6564 2069 6e20 646f 7470  included in dotp
-00007aa0: 6c6f 742c 2064 6566 6175 6c74 202d 312c  lot, default -1,
-00007ab0: 2069 6620 2d31 2077 6173 2073 6574 2c20   if -1 was set, 
-00007ac0: 7468 6520 3130 2520 6f66 2074 6865 206c  the 10% of the l
-00007ad0: 6f6e 6765 7374 2073 6361 6666 6f6c 6473  ongest scaffolds
-00007ae0: 2077 696c 6c20 6265 2073 6574 0a2d 6d73   will be set.-ms
-00007af0: 2c20 2d2d 6d61 7873 697a 652c 2074 6865  , --maxsize, the
-00007b00: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-00007b10: 7369 7a65 2074 6f20 6265 2069 6e63 6c75  size to be inclu
-00007b20: 6465 642c 2064 6566 6175 6c74 2032 3030  ded, default 200
-00007b30: 0a2d 6170 2c20 2d2d 616e 6368 6f72 706f  .-ap, --anchorpo
-00007b40: 696e 7473 2c20 7468 6520 616e 6368 6f72  ints, the anchor
-00007b50: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
-00007b60: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00007b70: 6d74 2c20 2d2d 6d75 6c74 6970 6c69 636f  mt, --multiplico
-00007b80: 6e2c 2074 6865 206d 756c 7469 706c 6963  n, the multiplic
-00007b90: 6f6e 7320 6461 7461 6669 6c65 2c20 6465  ons datafile, de
-00007ba0: 6661 756c 7420 4e6f 6e65 0a2d 6774 2c20  fault None.-gt, 
-00007bb0: 2d2d 6765 6e65 7461 626c 652c 2074 6865  --genetable, the
-00007bc0: 2067 656e 6520 7461 626c 6520 6461 7461   gene table data
-00007bd0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007be0: 6e65 0a2d 6d67 2c20 2d2d 6d69 6e73 6567  ne.-mg, --minseg
-00007bf0: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
-00007c00: 206c 656e 6774 6820 6f66 2073 6567 6d65   length of segme
-00007c10: 6e74 7320 746f 2069 6e63 6c75 6465 2c20  nts to include, 
-00007c20: 696e 2072 6174 696f 2069 6620 3c3d 2031  in ratio if <= 1
-00007c30: 2c20 6465 6661 756c 7420 3130 3030 300a  , default 10000.
-00007c40: 2d6d 676e 2c20 2d2d 6d69 6e67 656e 656e  -mgn, --mingenen
-00007c50: 756d 2c20 7468 6520 6d69 6e69 6d75 6d20  um, the minimum 
-00007c60: 6e75 6d62 6572 206f 6620 6765 6e65 7320  number of genes 
-00007c70: 666f 7220 6120 7365 676d 656e 7420 746f  for a segment to
-00007c80: 2062 6520 636f 6e73 6964 6572 6564 2c20   be considered, 
-00007c90: 6465 6661 756c 7420 3330 0a2d 6b72 2c20  default 30.-kr, 
-00007ca0: 2d2d 6b65 6570 7265 6475 6e2c 2066 6c61  --keepredun, fla
-00007cb0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00007cc0: 7220 746f 206b 6565 7020 7265 6475 6e64  r to keep redund
-00007cd0: 616e 7420 6d75 6c74 6970 6c69 636f 6e73  ant multiplicons
-00007ce0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007cf0: 7320 7365 742c 2074 6865 2072 6564 756e  s set, the redun
-00007d00: 6461 6e74 206d 756c 7469 706c 6963 6f6e  dant multiplicon
-00007d10: 7320 776f 756c 6420 6265 206b 6570 740a  s would be kept.
-00007d20: 2d65 706b 2c20 2d2d 6578 7472 6170 6172  -epk, --extrapar
-00007d30: 616e 6f6d 656b 732c 2065 7874 7261 2070  anomeks, extra p
-00007d40: 6172 616e 6f6d 6520 4b73 2064 6174 6120  aranome Ks data 
-00007d50: 746f 2070 6c6f 7420 696e 2074 6865 206d  to plot in the m
-00007d60: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
-00007d70: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
-00007d80: 6e65 0a2d 7061 672c 202d 2d70 6c6f 7461  ne.-pag, --plota
-00007d90: 7067 6d6d 2c20 666c 6167 206f 7074 696f  pgmm, flag optio
-00007da0: 6e2c 2077 6865 7468 6572 2074 6f20 636f  n, whether to co
-00007db0: 6e64 7563 7420 616e 6420 706c 6f74 206d  nduct and plot m
-00007dc0: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00007dd0: 6f66 2061 6e63 686f 7220 4b73 2069 6e20  of anchor Ks in 
-00007de0: 7468 6520 6d69 7865 6420 4b73 2064 6973  the mixed Ks dis
-00007df0: 7472 6962 7574 696f 6e2c 2069 6620 7468  tribution, if th
-00007e00: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00007e10: 7468 6520 6d69 7874 7572 6520 6d6f 6465  the mixture mode
-00007e20: 6c69 6e67 206f 6620 616e 6368 6f72 204b  ling of anchor K
-00007e30: 7320 776f 756c 6420 6265 2063 6f6e 6475  s would be condu
-00007e40: 6374 6564 2061 6e64 2070 6c6f 7474 6564  cted and plotted
-00007e50: 0a2d 7065 6d2c 202d 2d70 6c6f 7465 6c6d  .-pem, --plotelm
-00007e60: 6d2c 2066 6c61 6720 6f70 7469 6f6e 2c20  m, flag option, 
-00007e70: 7768 6574 6865 7220 746f 2063 6f6e 6475  whether to condu
-00007e80: 6374 2061 6e64 2070 6c6f 7420 656c 6d6d  ct and plot elmm
-00007e90: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00007ea0: 6720 6f66 2070 6172 616e 6f6d 6520 4b73  g of paranome Ks
-00007eb0: 2069 6e20 7468 6520 6d69 7865 6420 4b73   in the mixed Ks
-00007ec0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
-00007ed0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00007ee0: 6574 2c20 7468 6520 656c 6d6d 206d 6978  et, the elmm mix
-00007ef0: 7475 7265 206d 6f64 656c 696e 6720 6f66  ture modeling of
-00007f00: 2070 6172 616e 6f6d 6520 4b73 2077 6f75   paranome Ks wou
-00007f10: 6c64 2062 6520 636f 6e64 7563 7465 6420  ld be conducted 
-00007f20: 616e 6420 706c 6f74 7465 640a 2d63 2c20  and plotted.-c, 
-00007f30: 2d2d 636f 6d70 6f6e 656e 7473 2c20 7468  --components, th
-00007f40: 6520 7261 6e67 6520 6f66 2074 6865 206e  e range of the n
-00007f50: 756d 6265 7220 6f66 2063 6f6d 706f 6e65  umber of compone
-00007f60: 6e74 7320 746f 2066 6974 2069 6e20 616e  nts to fit in an
-00007f70: 6368 6f72 204b 7320 6d69 7874 7572 6520  chor Ks mixture 
-00007f80: 6d6f 6465 6c69 6e67 2c20 6465 6661 756c  modeling, defaul
-00007f90: 7420 2831 2c34 290a 2d70 7379 2c20 2d2d  t (1,4).-psy, --
-00007fa0: 706c 6f74 7379 6e2c 2066 6c61 6720 6f70  plotsyn, flag op
-00007fb0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00007fc0: 2069 6e69 7469 6174 6520 7468 6520 7379   initiate the sy
-00007fd0: 6e74 656e 7920 706c 6f74 2c20 6f6e 6c79  nteny plot, only
-00007fe0: 2077 6865 6e20 7468 6520 666c 6167 2077   when the flag w
-00007ff0: 6173 2073 6574 2c20 7468 6520 7379 6e74  as set, the synt
-00008000: 656e 7920 706c 6f74 2077 6f75 6c64 2062  eny plot would b
-00008010: 6520 7072 6f64 7563 6564 0a2d 6473 2c20  e produced.-ds, 
-00008020: 2d2d 646f 7473 697a 652c 2074 6865 2064  --dotsize, the d
-00008030: 6f74 2073 697a 6520 696e 2064 6f74 2070  ot size in dot p
-00008040: 6c6f 742c 2064 6566 6175 6c74 2030 2e33  lot, default 0.3
-00008050: 0a2d 6161 2c20 2d2d 6170 616c 7068 612c  .-aa, --apalpha,
-00008060: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-00008070: 616e 6368 6f72 2064 6f74 732c 2064 6566  anchor dots, def
-00008080: 6175 6c74 2031 0a2d 6861 2c20 2d2d 686f  ault 1.-ha, --ho
-00008090: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-000080a0: 7479 206f 6620 686f 6d6f 6c6f 6720 646f  ty of homolog do
-000080b0: 7473 2c20 6465 6661 756c 7420 300a 2d73  ts, default 0.-s
-000080c0: 7274 2c20 2d2d 7368 6f77 7265 616c 7469  rt, --showrealti
-000080d0: 636b 2c20 666c 6167 206f 7074 696f 6e2c  ck, flag option,
-000080e0: 2077 6865 7468 6572 2074 6f20 7368 6f77   whether to show
-000080f0: 2074 6865 2072 6561 6c20 7469 636b 2069   the real tick i
-00008100: 6e20 6765 6e65 7320 6f72 2062 6173 6573  n genes or bases
-00008110: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00008120: 7320 7365 742c 2074 6865 2072 6561 6c20  s set, the real 
-00008130: 7469 636b 2077 6f75 6c64 2062 6520 7368  tick would be sh
-00008140: 6f77 6564 0a2d 746c 732c 202d 2d74 6963  owed.-tls, --tic
-00008150: 6b6c 6162 656c 7369 7a65 2c20 7468 6520  klabelsize, the 
-00008160: 6c61 6265 6c20 7369 7a65 206f 6620 7469  label size of ti
-00008170: 636b 2c20 6465 6661 756c 7420 350a 2d78  ck, default 5.-x
-00008180: 6c2c 202d 2d78 6c69 6d2c 2074 6865 2078  l, --xlim, the x
-00008190: 2061 7869 7320 6c69 6d69 7420 6f66 204b   axis limit of K
-000081a0: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-000081b0: 796c 2c20 2d2d 796c 696d 2c20 7468 6520  yl, --ylim, the 
-000081c0: 7920 6178 6973 206c 696d 6974 206f 6620  y axis limit of 
-000081d0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
-000081e0: 2d61 646f 2c20 2d2d 6164 6a75 7374 6f72  -ado, --adjustor
-000081f0: 7468 6f2c 2066 6c61 6720 6f70 7469 6f6e  tho, flag option
-00008200: 2c20 7768 6574 6865 7220 746f 2061 646a  , whether to adj
-00008210: 7573 7420 7468 6520 6869 7374 6f67 7261  ust the histogra
-00008220: 6d20 6865 6967 6874 206f 6620 6f72 7468  m height of orth
-00008230: 6f6c 6f67 6f75 7320 4b73 2061 7320 746f  ologous Ks as to
-00008240: 206d 6174 6368 2074 6865 2068 6569 6768   match the heigh
-00008250: 7420 6f66 2070 6172 616c 6f67 6f75 7320  t of paralogous 
-00008260: 4b73 2c20 6966 2074 6865 2066 6c61 6720  Ks, if the flag 
-00008270: 7761 7320 7365 742c 2074 6865 2061 646a  was set, the adj
-00008280: 7573 746d 656e 7420 776f 756c 6420 6265  ustment would be
-00008290: 2063 6f6e 6475 6374 6564 0a2d 6164 662c   conducted.-adf,
-000082a0: 202d 2d61 646a 7573 7466 6163 746f 722c   --adjustfactor,
-000082b0: 2074 6865 2061 646a 7573 746d 656e 7420   the adjustment 
-000082c0: 6661 6374 6f72 206f 6620 6f72 7468 6f6c  factor of orthol
-000082d0: 6f67 6f75 7320 4b73 2c20 6465 6661 756c  ogous Ks, defaul
-000082e0: 7420 302e 350a 2d6f 612c 202d 2d6f 6b61  t 0.5.-oa, --oka
-000082f0: 6c70 6861 2c20 7468 6520 6f70 6163 6974  lpha, the opacit
-00008300: 7920 6f66 206f 7274 686f 6c6f 676f 7573  y of orthologous
-00008310: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00008320: 2069 6e20 6d69 7865 6420 706c 6f74 2c20   in mixed plot, 
-00008330: 6465 6661 756c 7420 302e 350a 2d63 732c  default 0.5.-cs,
-00008340: 202d 2d63 6c61 7373 6963 2c20 666c 6167   --classic, flag
-00008350: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00008360: 2074 6f20 6472 6177 206d 6978 6564 204b   to draw mixed K
-00008370: 7320 706c 6f74 2069 6e20 6120 636c 6173  s plot in a clas
-00008380: 7369 6320 6d61 6e6e 6572 2077 6865 7265  sic manner where
-00008390: 2074 6865 2066 756c 6c20 6f72 7468 6f6c   the full orthol
-000083a0: 6f67 6f75 7320 4b73 2064 6973 7472 6962  ogous Ks distrib
-000083b0: 7574 696f 6e20 6973 2064 7261 7765 642c  ution is drawed,
-000083c0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-000083d0: 2073 6574 2c20 7468 6520 636c 6173 7369   set, the classi
-000083e0: 6320 6d69 7865 6420 4b73 2070 6c6f 7420  c mixed Ks plot 
-000083f0: 776f 756c 6420 6265 2064 7261 776e 0a2d  would be drawn.-
-00008400: 7461 2c20 2d2d 746f 7061 7272 6f77 2c20  ta, --toparrow, 
-00008410: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00008420: 7468 6572 2074 6f20 6164 6a75 7374 2074  ther to adjust t
-00008430: 6865 2061 7272 6f77 2074 6f20 6265 2061  he arrow to be a
-00008440: 7420 7468 6520 746f 7020 6f66 2074 6865  t the top of the
-00008450: 2070 6c6f 742c 2069 6e73 7465 6164 206f   plot, instead o
-00008460: 6620 6265 696e 6720 636f 6f72 6469 6e61  f being coordina
-00008470: 7465 6420 6173 2074 6865 204b 4445 206f  ted as the KDE o
-00008480: 6620 7468 6520 6f72 7468 6f6c 6f67 6f75  f the orthologou
-00008490: 7320 4b73 2064 6973 7472 6962 7574 696f  s Ks distributio
-000084a0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-000084b0: 6173 2073 6574 2c20 7468 6520 6172 726f  as set, the arro
-000084c0: 7720 776f 756c 6420 6265 2073 6574 2061  w would be set a
-000084d0: 7420 7468 6520 746f 700a 2d6e 612c 202d  t the top.-na, -
-000084e0: 2d6e 6f64 6561 7665 7261 6765 642c 2066  -nodeaveraged, f
-000084f0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00008500: 6865 7220 746f 2075 7365 206e 6f64 652d  her to use node-
-00008510: 6176 6572 6167 6564 206d 6574 686f 6420  averaged method 
-00008520: 666f 7220 6465 2d72 6564 756e 6461 6e63  for de-redundanc
-00008530: 792c 2069 6620 7468 6520 666c 6167 2077  y, if the flag w
-00008540: 6173 2073 6574 2c20 7468 6520 6e6f 6465  as set, the node
-00008550: 2d61 7665 7261 6765 6420 6d65 7468 6f64  -averaged method
-00008560: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00008570: 7465 640a 2d62 732c 202d 2d62 6f6f 7473  ted.-bs, --boots
-00008580: 7472 6170 2c20 7468 6520 6e75 6d62 6572  trap, the number
-00008590: 206f 6620 626f 6f74 7374 7261 7020 7265   of bootstrap re
-000085a0: 706c 6963 6174 6573 206f 6620 6f72 7468  plicates of orth
-000085b0: 6f6c 6f67 204b 7320 6469 7374 7269 6275  olog Ks distribu
-000085c0: 7469 6f6e 2069 6e20 6d69 7865 6420 706c  tion in mixed pl
-000085d0: 6f74 0a2d 6772 2c20 2d2d 6769 7374 7262  ot.-gr, --gistrb
-000085e0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-000085f0: 6865 7468 6572 2074 6f20 7573 6520 6769  hether to use gi
-00008600: 7374 5f72 6169 6e62 6f77 2061 7320 636f  st_rainbow as co
-00008610: 6c6f 7220 6d61 7020 6f66 2064 6f74 706c  lor map of dotpl
-00008620: 6f74 0a2d 6e2c 202d 2d6e 7468 7265 6164  ot.-n, --nthread
-00008630: 732c 2074 6865 206e 756d 6265 7220 6f66  s, the number of
-00008640: 2074 6872 6561 6473 2074 6f20 7573 6520   threads to use 
-00008650: 696e 2062 6f6f 7473 7472 6170 2073 616d  in bootstrap sam
-00008660: 706c 696e 672c 2064 6566 6175 6c74 2031  pling, default 1
-00008670: 0a60 6060 0a0a 2323 2055 7361 6765 0a0a  .```..## Usage..
-00008680: 4865 7265 2077 6520 7072 6f76 6964 6564  Here we provided
-00008690: 2074 6865 2062 6173 6963 2075 7361 6765   the basic usage
-000086a0: 2066 6f72 2065 6163 6820 7072 6f67 7261   for each progra
-000086b0: 6d20 616e 6420 7468 6520 7265 6c65 7661  m and the releva
-000086c0: 6e74 2070 6172 616d 6574 6572 7320 616e  nt parameters an
-000086d0: 6420 7375 6767 6573 7469 6f6e 7320 6f6e  d suggestions on
-000086e0: 2070 6172 616d 6574 6572 697a 6174 696f   parameterizatio
-000086f0: 6e2e 2041 2072 656d 696e 6465 7220 7468  n. A reminder th
-00008700: 6174 2074 6865 2067 6976 656e 2063 6f72  at the given cor
-00008710: 6573 2061 6e64 2074 6872 6561 6473 2063  es and threads c
-00008720: 616e 2073 6967 6e69 6669 6361 6e74 6c79  an significantly
-00008730: 2069 6d70 6163 7420 7468 6520 7275 6e20   impact the run 
-00008740: 7469 6d65 2061 6e64 2074 6875 7320 7765  time and thus we
-00008750: 2061 6464 6564 2073 6f6d 6520 7265 706f   added some repo
-00008760: 7274 2069 6e66 6f72 6d61 7469 6f6e 2070  rt information p
-00008770: 6572 7461 696e 696e 6720 746f 2074 6865  ertaining to the
-00008780: 2073 7973 7465 6d20 6f66 2075 7365 7273   system of users
-00008790: 2074 6f20 6661 6369 6c69 7461 7465 2074   to facilitate t
-000087a0: 6865 2065 6666 6963 6965 6e74 2073 6574  he efficient set
-000087b0: 7469 6e67 206f 6620 7468 7265 6164 7320  ting of threads 
-000087c0: 616e 6420 6d65 6d6f 7279 2e20 5468 6520  and memory. The 
-000087d0: 6c6f 6769 6361 6c20 4350 5573 2072 6570  logical CPUs rep
-000087e0: 6f72 7465 6420 7265 7072 6573 656e 7473  orted represents
-000087f0: 2074 6865 206e 756d 6265 7220 6f66 2070   the number of p
-00008800: 6879 7369 6361 6c20 636f 7265 7320 6d75  hysical cores mu
-00008810: 6c74 6970 6c69 6564 2062 7920 7468 6520  ltiplied by the 
-00008820: 6e75 6d62 6572 206f 6620 7468 7265 6164  number of thread
-00008830: 7320 7468 6174 2063 616e 2072 756e 206f  s that can run o
-00008840: 6e20 6561 6368 2063 6f72 652c 2061 6c73  n each core, als
-00008850: 6f20 6b6e 6f77 6e20 6173 2048 7970 6572  o known as Hyper
-00008860: 2054 6872 6561 6469 6e67 2e20 5468 6520   Threading. The 
-00008870: 6e75 6d62 6572 206f 6620 6c6f 6769 6361  number of logica
-00008880: 6c20 4350 5573 206d 6179 206e 6f74 206e  l CPUs may not n
-00008890: 6563 6573 7361 7269 6c79 2062 6520 6571  ecessarily be eq
-000088a0: 7569 7661 6c65 6e74 2074 6f20 7468 6520  uivalent to the 
-000088b0: 6163 7475 616c 206e 756d 6265 7220 6f66  actual number of
-000088c0: 2043 5055 7320 7468 6520 6375 7272 656e   CPUs the curren
-000088d0: 7420 7072 6f63 6573 7320 6361 6e20 7573  t process can us
-000088e0: 652e 2054 6865 2061 7661 696c 6162 6c65  e. The available
-000088f0: 206d 656d 6f72 7920 7265 6665 7273 2074   memory refers t
-00008900: 6f20 7468 6520 6d65 6d6f 7279 2074 6861  o the memory tha
-00008910: 7420 6361 6e20 6265 2067 6976 656e 2069  t can be given i
-00008920: 6e73 7461 6e74 6c79 2074 6f20 7072 6f63  nstantly to proc
-00008930: 6573 7365 7320 7769 7468 6f75 7420 7468  esses without th
-00008940: 6520 7379 7374 656d 2067 6f69 6e67 2069  e system going i
-00008950: 6e74 6f20 7377 6170 2061 6e64 2072 6566  nto swap and ref
-00008960: 6c65 6374 7320 7468 6520 6163 7475 616c  lects the actual
-00008970: 206d 656d 6f72 7920 6176 6169 6c61 626c   memory availabl
-00008980: 652e 2054 6865 2066 7265 6520 6d65 6d6f  e. The free memo
-00008990: 7279 2072 6566 6572 7320 746f 2074 6865  ry refers to the
-000089a0: 206d 656d 6f72 7920 6e6f 7420 6265 696e   memory not bein
-000089b0: 6720 7573 6564 2061 7420 616c 6c20 287a  g used at all (z
-000089c0: 6572 6f65 6429 2074 6861 7420 6973 2072  eroed) that is r
-000089d0: 6561 6469 6c79 2061 7661 696c 6162 6c65  eadily available
-000089e0: 2e20 5468 6520 6465 7363 7269 7074 696f  . The descriptio
-000089f0: 6e20 6162 6f76 6520 7265 6665 7273 2074  n above refers t
-00008a00: 6f20 7468 6520 646f 6375 6d65 6e74 6174  o the documentat
-00008a10: 696f 6e20 6f66 2060 7073 7574 696c 602e  ion of `psutil`.
-00008a20: 0a0a 2323 2320 7767 6420 646d 640a 0a2a  ..### wgd dmd..*
-00008a30: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-00008a40: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-00008a50: 6d65 2a2a 0a60 6060 0a77 6764 2064 6d64  me**.```.wgd dmd
-00008a60: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00008a70: 6c65 6120 2d49 2032 202d 6520 3165 2d31  lea -I 2 -e 1e-1
-00008a80: 3020 2d62 7320 3130 3020 2d6e 7020 3520  0 -bs 100 -np 5 
-00008a90: 282d 6e6e 2920 282d 2d74 6f5f 7374 6f70  (-nn) (--to_stop
-00008aa0: 2920 282d 2d63 6473 2920 282d 6e20 3429  ) (--cds) (-n 4)
-00008ab0: 2028 2d6f 2077 6764 5f64 6d64 2920 282d   (-o wgd_dmd) (-
-00008ac0: 7420 776f 726b 696e 675f 746d 7029 0a60  t working_tmp).`
-00008ad0: 6060 200a 0a4e 6f74 6520 7468 6174 2077  `` ..Note that w
-00008ae0: 6520 646f 6e27 7420 7072 6f76 6964 6520  e don't provide 
-00008af0: 7468 6520 6461 7461 206f 6620 7468 6973  the data of this
-00008b00: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
-00008b10: 2028 6364 7329 2066 696c 6520 6041 7175   (cds) file `Aqu
-00008b20: 696c 6567 6961 5f63 6f65 7275 6c65 6160  ilegia_coerulea`
-00008b30: 2062 7574 2069 7420 6361 6e20 6265 2065   but it can be e
-00008b40: 6173 696c 7920 646f 776e 6c6f 6164 6564  asily downloaded
-00008b50: 2061 7420 5b50 6879 746f 7a6f 6d65 5d28   at [Phytozome](
-00008b60: 6874 7470 733a 2f2f 7068 7974 6f7a 6f6d  https://phytozom
-00008b70: 652d 6e65 7874 2e6a 6769 2e64 6f65 2e67  e-next.jgi.doe.g
-00008b80: 6f76 2f69 6e66 6f2f 4163 6f65 7275 6c65  ov/info/Acoerule
-00008b90: 615f 7633 5f31 2920 2873 616d 6520 666f  a_v3_1) (same fo
-00008ba0: 7220 6f74 6865 7220 5573 6167 6520 646f  r other Usage do
-00008bb0: 6329 2e20 4e6f 7465 2074 6861 7420 696e  c). Note that in
-00008bc0: 2074 6865 2069 7373 7565 7320 736f 6d65   the issues some
-00008bd0: 2075 7365 7273 2064 6964 6e27 7420 646f   users didn't do
-00008be0: 776e 6c6f 6164 2074 6865 2060 4163 6f65  wnload the `Acoe
-00008bf0: 7275 6c65 615f 3332 325f 7633 2e31 2e63  rulea_322_v3.1.c
-00008c00: 6473 5f70 7269 6d61 7279 5472 616e 7363  ds_primaryTransc
-00008c10: 7269 7074 4f6e 6c79 2e66 612e 677a 6020  riptOnly.fa.gz` 
-00008c20: 6275 7420 696e 7374 6561 6420 7468 6520  but instead the 
-00008c30: 6041 636f 6572 756c 6561 5f33 3232 5f76  `Acoerulea_322_v
-00008c40: 332e 312e 6364 732e 6661 2e67 7a60 2066  3.1.cds.fa.gz` f
-00008c50: 696c 652e 2046 6f72 2074 6865 2063 6f6e  ile. For the con
-00008c60: 7374 7275 6374 696f 6e20 6f66 2077 686f  struction of who
-00008c70: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00008c80: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00008c90: 7269 6275 7469 6f6e 2c20 2a2a 6f6e 6c79  ribution, **only
-00008ca0: 206f 6e65 2074 7261 6e73 6372 6970 7420   one transcript 
-00008cb0: 2874 6865 2070 7269 6d61 7279 206f 6e65  (the primary one
-00008cc0: 2920 7065 7220 6765 6e65 2073 686f 756c  ) per gene shoul
-00008cd0: 6420 6265 2069 6e63 6c75 6465 642a 2a20  d be included** 
-00008ce0: 7375 6368 2074 6861 7420 7468 6520 2a4b  such that the *K
-00008cf0: 2a3c 7375 623e 533c 2f73 7562 3e20 6973  *<sub>S</sub> is
-00008d00: 2072 6561 6c6c 7920 696e 6469 6361 7469   really indicati
-00008d10: 6e67 2074 6865 2061 6765 206f 6620 6765  ng the age of ge
-00008d20: 6e65 2064 7570 6c69 6361 7469 6f6e 2065  ne duplication e
-00008d30: 7665 6e74 2c20 696e 7374 6561 6420 6f66  vent, instead of
-00008d40: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008d50: 6963 696e 672e 2054 7261 6e73 6372 6970  icing. Transcrip
-00008d60: 746f 6d65 2064 6174 6120 7368 6f75 6c64  tome data should
-00008d70: 2062 6520 6361 7265 6675 6c6c 7920 7472   be carefully tr
-00008d80: 6561 7465 6420 7769 7468 2064 652d 7265  eated with de-re
-00008d90: 6475 6e64 616e 6379 2073 6f20 6173 2074  dundancy so as t
-00008da0: 6f20 7265 6475 6365 2074 6865 2066 616c  o reduce the fal
-00008db0: 7365 2070 6f73 6974 6976 6520 6475 706c  se positive dupl
-00008dc0: 6963 6174 696f 6e20 6275 6d70 2063 6175  ication bump cau
-00008dd0: 7365 6420 6279 2070 6572 7661 7369 7665  sed by pervasive
-00008de0: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008df0: 6963 696e 672e 2048 6572 6520 7468 6520  icing. Here the 
-00008e00: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
-00008e10: 2070 6172 616d 6574 6572 2c20 6769 7665   parameter, give
-00008e20: 6e20 6279 2060 2d49 6020 6f72 2060 2d2d  n by `-I` or `--
-00008e30: 696e 666c 6174 696f 6e60 2c20 6166 6665  inflation`, affe
-00008e40: 6374 7320 7468 6520 6772 616e 756c 6172  cts the granular
-00008e50: 6974 7920 6f72 2072 6573 6f6c 7574 696f  ity or resolutio
-00008e60: 6e20 6f66 2074 6865 2063 6c75 7374 6572  n of the cluster
-00008e70: 696e 6720 6f75 7463 6f6d 6520 616e 6420  ing outcome and 
-00008e80: 696d 706c 6963 6974 6c79 2063 6f6e 7472  implicitly contr
-00008e90: 6f6c 6c73 2074 6865 206e 756d 6265 7220  olls the number 
-00008ea0: 6f66 2063 6c75 7374 6572 732c 2077 6974  of clusters, wit
-00008eb0: 6820 6c6f 7720 7661 6c75 6573 2073 7563  h low values suc
-00008ec0: 6820 6173 2031 2e33 206f 7220 312e 3420  h as 1.3 or 1.4 
-00008ed0: 6c65 6164 696e 6720 746f 2066 6577 6572  leading to fewer
-00008ee0: 2062 7574 206c 6172 6765 7220 636c 7573   but larger clus
-00008ef0: 7465 7273 2061 6e64 2068 6967 6820 7661  ters and high va
-00008f00: 6c75 6573 2073 7563 6820 6173 2035 206f  lues such as 5 o
-00008f10: 7220 3620 6c65 6164 696e 6720 746f 206d  r 6 leading to m
-00008f20: 6f72 6520 6275 7420 736d 616c 6c65 7220  ore but smaller 
-00008f30: 636c 7573 7465 7273 2e20 5765 2073 6574  clusters. We set
-00008f40: 2074 6865 2064 6566 6175 6c74 2076 616c   the default val
-00008f50: 7565 2061 7320 3220 6173 2073 7567 6765  ue as 2 as sugge
-00008f60: 7374 6564 2062 7920 5b4d 434c 5d28 6874  sted by [MCL](ht
-00008f70: 7470 733a 2f2f 6d69 6361 6e73 2e6f 7267  tps://micans.org
-00008f80: 2f6d 636c 2f29 2e20 5468 6520 652d 7661  /mcl/). The e-va
-00008f90: 6c75 6520 6375 742d 6f66 6620 666f 7220  lue cut-off for 
-00008fa0: 7365 7175 656e 6365 2073 696d 696c 6172  sequence similar
-00008fb0: 6974 792c 2067 6976 656e 2062 7920 602d  ity, given by `-
-00008fc0: 6560 206f 7220 602d 2d65 7661 6c60 2c20  e` or `--eval`, 
-00008fd0: 7768 6963 6820 6465 6e6f 7465 7320 7468  which denotes th
-00008fe0: 6520 6578 7065 6374 6564 2076 616c 7565  e expected value
-00008ff0: 206f 6620 7468 6520 6869 7420 7175 616e   of the hit quan
-00009000: 7469 6669 6573 2074 6865 206e 756d 6265  tifies the numbe
-00009010: 7220 6f66 2061 6c69 676e 6d65 6e74 7320  r of alignments 
-00009020: 6f66 2073 696d 696c 6172 206f 7220 6265  of similar or be
-00009030: 7474 6572 2071 7561 6c69 7479 2074 6861  tter quality tha
-00009040: 7420 796f 7520 6578 7065 6374 2074 6f20  t you expect to 
-00009050: 6669 6e64 2073 6561 7263 6869 6e67 2074  find searching t
-00009060: 6869 7320 7175 6572 7920 6167 6169 6e73  his query agains
-00009070: 7420 6120 6461 7461 6261 7365 206f 6620  t a database of 
-00009080: 7261 6e64 6f6d 2073 6571 7565 6e63 6573  random sequences
-00009090: 2074 6865 2073 616d 6520 7369 7a65 2061   the same size a
-000090a0: 7320 7468 6520 6163 7475 616c 2074 6172  s the actual tar
-000090b0: 6765 7420 6461 7461 6261 7365 2c20 6973  get database, is
-000090c0: 2074 6865 206b 6579 2070 6172 616d 6574   the key paramet
-000090d0: 6572 206d 6561 7375 7269 6e67 2074 6865  er measuring the
-000090e0: 2073 6967 6e69 6669 6361 6e63 6520 6f66   significance of
-000090f0: 2061 2068 6974 2c20 7768 6963 6820 6973   a hit, which is
-00009100: 2073 6574 2068 6572 6520 6173 2064 6566   set here as def
-00009110: 6175 6c74 2031 652d 3130 2e20 4e6f 7465  ault 1e-10. Note
-00009120: 2074 6861 7420 5b44 4941 4d4f 4e44 5d28   that [DIAMOND](
-00009130: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00009140: 6f6d 2f62 6275 6368 6669 6e6b 2f64 6961  om/bbuchfink/dia
-00009150: 6d6f 6e64 2f77 696b 692f 312e 2d54 7574  mond/wiki/1.-Tut
-00009160: 6f72 6961 6c29 2069 7473 656c 6620 6279  orial) itself by
-00009170: 2064 6566 6175 6c74 206f 6e6c 7920 7265   default only re
-00009180: 706f 7274 7320 616c 6c20 616c 6967 6e6d  ports all alignm
-00009190: 656e 7473 2077 6974 6820 652d 7661 6c75  ents with e-valu
-000091a0: 6520 3c20 302e 3030 312e 2054 6865 2070  e < 0.001. The p
-000091b0: 6572 6365 6e74 6167 6520 6f66 2075 7070  ercentage of upp
-000091c0: 6572 2068 6974 7320 7573 6564 2066 6f72  er hits used for
-000091d0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
-000091e0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
-000091f0: 6e20 6279 2060 2d6e 7060 206f 7220 602d  n by `-np` or `-
-00009200: 2d6e 6f72 6d61 6c69 7a65 6470 6572 6365  -normalizedperce
-00009210: 6e74 602c 2077 6869 6368 2064 6574 6572  nt`, which deter
-00009220: 6d69 6e65 7320 7468 6520 7570 7065 7220  mines the upper 
-00009230: 7065 7263 656e 7469 6c65 206f 6620 6869  percentile of hi
-00009240: 7473 2070 6572 2062 696e 2028 6361 7465  ts per bin (cate
-00009250: 676f 7269 7a65 6420 6279 2067 656e 6520  gorized by gene 
-00009260: 6c65 6e67 7468 2920 7573 6564 2069 6e20  length) used in 
-00009270: 7468 6520 6669 7420 6f66 206c 696e 6561  the fit of linea
-00009280: 7220 7265 6772 6573 7369 6f6e 2c20 636f  r regression, co
-00009290: 6e73 6964 6572 696e 6720 7468 6174 206e  nsidering that n
-000092a0: 6f74 2061 6c6c 2068 6974 7320 7065 7220  ot all hits per 
-000092b0: 6269 6e20 7368 6f77 2061 7070 6172 656e  bin show apparen
-000092c0: 7420 6c69 6e65 6172 2072 656c 6174 696f  t linear relatio
-000092d0: 6e73 6869 702c 2069 7320 7365 7420 6173  nship, is set as
-000092e0: 2064 6566 6175 6c74 2035 2c20 696e 6469   default 5, indi
-000092f0: 6361 7469 6e67 2074 6865 2075 7361 6765  cating the usage
-00009300: 206f 6620 746f 7020 3525 2068 6974 7320   of top 5% hits 
-00009310: 7065 7220 6269 6e2e 2054 6865 206e 756d  per bin. The num
-00009320: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
-00009330: 6465 6420 696e 2067 656e 6520 6c65 6e67  ded in gene leng
-00009340: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
-00009350: 2c20 6769 7665 6e20 6279 2060 2d62 7360  , given by `-bs`
-00009360: 206f 7220 602d 2d62 696e 7360 2c20 6465   or `--bins`, de
-00009370: 7465 726d 696e 6573 2074 6865 206e 756d  termines the num
-00009380: 6265 7220 6f66 2062 696e 7320 746f 2063  ber of bins to c
-00009390: 6174 6567 6f72 697a 6520 7468 6520 6765  ategorize the ge
-000093a0: 6e65 206c 656e 6774 682c 2069 7320 7365  ne length, is se
-000093b0: 7420 6173 2064 6566 6175 6c74 2031 3030  t as default 100
-000093c0: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
-000093d0: 602d 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f  `-nn` or `--nono
-000093e0: 726d 616c 697a 6174 696f 6e60 2063 616e  rmalization` can
-000093f0: 2062 6520 7365 7420 746f 2063 616c 6c20   be set to call 
-00009400: 6f66 6620 7468 6520 6e6f 726d 616c 697a  off the normaliz
-00009410: 6174 696f 6e20 7072 6f63 6573 732c 2061  ation process, a
-00009420: 6c74 686f 7567 6820 6974 2773 2073 7567  lthough it's sug
-00009430: 6765 7374 6564 2074 6f20 636f 6e64 7563  gested to conduc
-00009440: 7420 7468 6520 6e6f 726d 616c 697a 6174  t the normalizat
-00009450: 696f 6e20 746f 2061 6371 7569 7265 206d  ion to acquire m
-00009460: 6f72 6520 6163 6375 7261 7465 2067 656e  ore accurate gen
-00009470: 6520 6661 6d69 6c79 2063 6c75 7374 6572  e family cluster
-00009480: 696e 6720 7265 7375 6c74 2e20 5468 6520  ing result. The 
-00009490: 7061 7261 6d65 7465 7273 2060 2d2d 746f  parameters `--to
-000094a0: 5f73 746f 7060 2061 6e64 2060 2d2d 6364  _stop` and `--cd
-000094b0: 7360 2063 6f6e 7472 6f6c 2074 6865 2062  s` control the b
-000094c0: 6568 6176 696f 7572 206f 6620 7472 616e  ehaviour of tran
-000094d0: 736c 6174 696e 6720 636f 6469 6e67 2073  slating coding s
-000094e0: 6571 7565 6e63 6520 696e 746f 2061 6d69  equence into ami
-000094f0: 6e6f 2061 6369 6420 7365 7175 656e 6365  no acid sequence
-00009500: 2e20 4966 2074 6865 2060 2d2d 746f 5f73  . If the `--to_s
-00009510: 746f 7060 2077 6173 2073 6574 2c20 7468  top` was set, th
-00009520: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009530: 756c 6420 6265 2074 6572 6d69 6e61 7465  uld be terminate
-00009540: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
-00009550: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00009560: 6f6e 2c20 6f74 6865 7277 6973 6520 7468  on, otherwise th
-00009570: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009580: 756c 6420 7369 6d70 6c79 2073 6b69 7020  uld simply skip 
-00009590: 616e 7920 7374 6f70 2063 6f64 6f6e 732e  any stop codons.
-000095a0: 2049 6620 7468 6520 602d 2d63 6473 6020   If the `--cds` 
-000095b0: 7761 7320 7365 742c 2073 6571 7565 6e63  was set, sequenc
-000095c0: 6573 2074 6861 7420 646f 6573 6e27 7420  es that doesn't 
-000095d0: 7374 6172 7420 7769 7468 2061 2076 616c  start with a val
-000095e0: 6964 2073 7461 7274 2063 6f64 6f6e 2c20  id start codon, 
-000095f0: 6f72 2063 6f6e 7461 696e 7320 6d6f 7265  or contains more
-00009600: 2074 6861 6e20 6f6e 6520 696e 2d66 7261   than one in-fra
-00009610: 6d65 2073 746f 7020 636f 646f 6e2c 206f  me stop codon, o
-00009620: 7220 6973 206e 6f74 2064 6976 6964 6162  r is not dividab
-00009630: 6c65 2062 7920 332c 2077 6f75 6c64 2062  le by 3, would b
-00009640: 6520 7369 6d70 6c79 2064 726f 7070 6564  e simply dropped
-00009650: 2c20 7375 6368 2074 6861 7420 6f6e 6c79  , such that only
-00009660: 2073 7472 6963 7420 636f 6d70 6c65 7465   strict complete
-00009670: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
-00009680: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00009690: 6465 6420 696e 2074 6865 2073 7562 7365  ded in the subse
-000096a0: 7175 656e 7420 616e 616c 7973 6973 2e20  quent analysis. 
-000096b0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
-000096c0: 7261 6c6c 656c 2074 6872 6561 6473 2062  rallel threads b
-000096d0: 7920 7468 6520 6f70 7469 6f6e 2060 2d6e  y the option `-n
-000096e0: 6020 6f72 2060 2d2d 6e74 6872 6561 6473  ` or `--nthreads
-000096f0: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
-00009700: 6163 6365 6c65 7261 7465 2074 6865 2063  accelerate the c
-00009710: 616c 6375 6c61 7469 6f6e 2077 6974 6869  alculation withi
-00009720: 6e20 6064 6961 6d6f 6e64 602e 2054 6865  n `diamond`. The
-00009730: 2064 6972 6563 746f 7279 206f 6620 6f75   directory of ou
-00009740: 7470 7574 206f 7220 696e 7465 726d 6564  tput or intermed
-00009750: 6961 7465 2066 696c 6573 2069 7320 6465  iate files is de
-00009760: 7465 726d 696e 6564 2062 7920 7468 6520  termined by the 
-00009770: 7061 7261 6d65 7465 7220 602d 6f60 206f  parameter `-o` o
-00009780: 7220 602d 2d6f 7574 6469 7260 2c20 616e  r `--outdir`, an
-00009790: 6420 602d 7460 206f 7220 602d 2d74 6d70  d `-t` or `--tmp
-000097a0: 6469 7260 2c20 7768 6963 6820 7769 6c6c  dir`, which will
-000097b0: 2062 6520 6372 6561 7465 6420 6279 2074   be created by t
-000097c0: 6865 2070 726f 6772 616d 2069 7473 656c  he program itsel
-000097d0: 6620 616e 6420 6265 206f 7665 7277 7269  f and be overwri
-000097e0: 7474 656e 2069 6620 7468 6520 666f 6c64  tten if the fold
-000097f0: 6572 2068 6173 2061 6c72 6561 6479 2062  er has already b
-00009800: 6565 6e20 6372 6561 7465 642e 204e 6f74  een created. Not
-00009810: 6520 7468 6174 2074 6865 2073 6f66 7477  e that the softw
-00009820: 6172 6520 6064 6961 6d6f 6e64 6020 7368  are `diamond` sh
-00009830: 6f75 6c64 2062 6520 7072 652d 696e 7374  ould be pre-inst
-00009840: 616c 6c65 6420 616e 6420 7365 7420 696e  alled and set in
-00009850: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-00009860: 2070 6174 6820 696e 2061 6c6c 2074 6865   path in all the
-00009870: 2061 6e61 6c79 7369 7320 7065 7266 6f72   analysis perfor
-00009880: 6d65 6420 6279 2060 7767 6420 646d 6460  med by `wgd dmd`
-00009890: 2065 7863 6570 7420 666f 7220 7468 6520   except for the 
-000098a0: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
-000098b0: 6365 6e63 6520 616e 616c 7973 6973 2e0a  cence analysis..
-000098c0: 0a2a 2a57 6520 7375 6767 6573 7420 7468  .**We suggest th
-000098d0: 6174 2074 6865 2064 6566 6175 6c74 2073  at the default s
-000098e0: 6574 7469 6e67 2069 6e20 7768 6963 6820  etting in which 
-000098f0: 7468 6520 696e 666c 6174 696f 6e20 6661  the inflation fa
-00009900: 6374 6f72 2069 7320 7365 7420 6173 2032  ctor is set as 2
-00009910: 2c20 652d 7661 6c75 6520 6375 742d 6f66  , e-value cut-of
-00009920: 6620 6173 2031 652d 3130 2061 6e64 206f  f as 1e-10 and o
-00009930: 7468 6572 2070 6172 616d 6574 6572 7320  ther parameters 
-00009940: 696e 2064 6566 6175 6c74 2069 7320 6120  in default is a 
-00009950: 676f 6f64 2073 7461 7274 696e 6720 706f  good starting po
-00009960: 696e 742c 2075 6e6c 6573 7320 796f 7520  int, unless you 
-00009970: 7370 6563 6966 6963 616c 6c79 2077 616e  specifically wan
-00009980: 7420 746f 2065 7870 6c6f 7265 2074 6865  t to explore the
-00009990: 2065 6666 6563 7473 206f 6620 6469 6666   effects of diff
-000099a0: 6572 656e 7420 7061 7261 6d65 7465 7273  erent parameters
-000099b0: 2e2a 2a20 5375 6368 2074 6861 7420 7468  .** Such that th
-000099c0: 6520 636f 6d6d 616e 6420 666f 7220 7468  e command for th
-000099d0: 6520 6465 6c69 6e65 6174 696f 6e20 6f66  e delineation of
-000099e0: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
-000099f0: 6973 2073 696d 706c 7920 6173 2062 656c  is simply as bel
-00009a00: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-00009a10: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00009a20: 6c65 610a 6060 600a 0a2a 2a54 6865 2064  lea.```..**The d
-00009a30: 656c 696e 6561 7469 6f6e 206f 6620 5242  elineation of RB
-00009a40: 4873 2a2a 0a60 6060 0a77 6764 2064 6d64  Hs**.```.wgd dmd
-00009a50: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-00009a60: 6e63 6532 202d 6520 3165 2d31 3020 2d62  nce2 -e 1e-10 -b
-00009a70: 7320 3130 3020 2d6e 7020 3520 282d 6e6e  s 100 -np 5 (-nn
-00009a80: 2920 282d 6e20 3429 2028 2d63 2030 2e39  ) (-n 4) (-c 0.9
-00009a90: 2920 282d 2d6f 6766 6f72 6d61 7429 2028  ) (--ogformat) (
-00009aa0: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
-00009ab0: 7329 2028 2d6f 2077 6764 5f64 6d64 2920  s) (-o wgd_dmd) 
-00009ac0: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-00009ad0: 0a60 6060 0a0a 546f 2064 656c 696e 6561  .```..To delinea
-00009ae0: 7465 2052 4248 7320 6265 7477 6565 6e20  te RBHs between 
-00009af0: 7477 6f20 6364 7320 7365 7175 656e 6365  two cds sequence
-00009b00: 2066 696c 6573 2c20 7468 6520 7265 6c65   files, the rele
-00009b10: 7661 6e74 2070 6172 616d 6574 6572 2069  vant parameter i
-00009b20: 7320 6d6f 7374 6c79 2074 6865 2073 616d  s mostly the sam
-00009b30: 6520 6173 2077 686f 6c65 2070 6172 616e  e as whole paran
-00009b40: 6f6d 6520 696e 6665 7265 6e63 652c 2065  ome inference, e
-00009b50: 7863 6570 7420 666f 7220 7468 6520 7061  xcept for the pa
-00009b60: 7261 6d65 7465 7220 602d 6360 206f 7220  rameter `-c` or 
-00009b70: 602d 2d63 7363 6f72 6560 2c20 7768 6963  `--cscore`, whic
-00009b80: 6820 7261 6e67 6573 2062 6574 7765 656e  h ranges between
-00009b90: 2030 2061 6e64 2031 2061 6e64 2069 7320   0 and 1 and is 
-00009ba0: 7573 6564 2074 6f20 7265 6c61 7820 7468  used to relax th
-00009bb0: 6520 7369 6d69 6c61 7269 7479 2063 7574  e similarity cut
-00009bc0: 6f66 6620 6672 6f6d 2074 6865 2065 7863  off from the exc
-00009bd0: 6c75 7369 7665 2072 6563 6970 726f 6361  lusive reciproca
-00009be0: 6c20 6265 7374 2068 6974 7320 746f 2061  l best hits to a
-00009bf0: 2063 6572 7461 696e 2072 6174 696f 2061   certain ratio a
-00009c00: 7320 746f 2074 6865 2062 6573 7420 6869  s to the best hi
-00009c10: 7473 2e20 466f 7220 696e 7374 616e 6365  ts. For instance
-00009c20: 2c20 6966 2074 6865 2067 656e 6520 6231  , if the gene b1
-00009c30: 2066 726f 6d20 6765 6e6f 6d65 2042 2068   from genome B h
-00009c40: 6173 2074 6865 2062 6573 7420 6869 7420  as the best hit 
-00009c50: 6765 6e65 2061 3120 6672 6f6d 2067 656e  gene a1 from gen
-00009c60: 6f6d 6520 4120 7769 7468 2074 6865 2062  ome A with the b
-00009c70: 6974 2073 636f 7265 2061 7320 3130 302c  it score as 100,
-00009c80: 2077 6869 6368 2069 7320 6120 7363 6f72   which is a scor
-00009c90: 696e 6720 6d61 7472 6978 2069 6e64 6570  ing matrix indep
-00009ca0: 656e 6465 6e74 206d 6561 7375 7265 206f  endent measure o
-00009cb0: 6620 7468 6520 286c 6f63 616c 2920 7369  f the (local) si
-00009cc0: 6d69 6c61 7269 7479 206f 6620 7468 6520  milarity of the 
-00009cd0: 7477 6f20 616c 6967 6e65 6420 7365 7175  two aligned sequ
-00009ce0: 656e 6365 732c 2077 6974 6820 6c61 7267  ences, with larg
-00009cf0: 6572 2076 616c 7565 7320 696e 6469 6361  er values indica
-00009d00: 7469 6e67 2068 6967 6865 7220 7369 6d69  ting higher simi
-00009d10: 6c61 7269 7469 6573 2c20 6769 7665 6e20  larities, given 
-00009d20: 7468 6520 602d 6320 302e 3960 2c20 6765  the `-c 0.9`, ge
-00009d30: 6e65 7320 6672 6f6d 2067 656e 6f6d 6520  nes from genome 
-00009d40: 4120 7768 6963 6820 6861 7320 7468 6520  A which has the 
-00009d50: 6269 7420 7363 6f72 6520 7769 7468 2067  bit score with g
-00009d60: 656e 6520 6231 2068 6967 6865 7220 7468  ene b1 higher th
-00009d70: 616e 2030 2e39 7831 3030 2077 696c 6c20  an 0.9x100 will 
-00009d80: 616c 736f 2062 6520 7772 6974 7465 6e20  also be written 
-00009d90: 696e 2074 6865 2072 6573 756c 7420 6669  in the result fi
-00009da0: 6c65 2c20 7768 6963 6820 696e 2061 2073  le, which in a s
-00009db0: 656e 7365 2061 7265 206e 6f74 2052 4248  ense are not RBH
-00009dc0: 7320 616e 796d 6f72 6520 6f66 2063 6f75  s anymore of cou
-00009dd0: 7273 652c 2062 7574 2074 6865 2068 6967  rse, but the hig
-00009de0: 686c 7920 7369 6d69 6c61 7220 686f 6d6f  hly similar homo
-00009df0: 6c6f 6775 6520 7061 6972 732e 2049 6620  logue pairs. If 
-00009e00: 6d6f 7265 2074 6861 6e20 3220 7365 7175  more than 2 sequ
-00009e10: 656e 6365 2066 696c 6573 2077 6572 6520  ence files were 
-00009e20: 7072 6f76 6964 6564 2c20 6576 6572 7920  provided, every 
-00009e30: 7061 6972 2d77 6973 6520 5242 4873 2077  pair-wise RBHs w
-00009e40: 6f75 6c64 2062 6520 6361 6c63 756c 6174  ould be calculat
-00009e50: 6564 2065 7863 6570 7420 666f 7220 7175  ed except for qu
-00009e60: 6572 7969 6e67 2074 6865 2073 616d 6520  erying the same 
-00009e70: 7365 7175 656e 6365 2069 7473 656c 662e  sequence itself.
-00009e80: 2054 6865 206e 756d 6265 7220 6f66 2070   The number of p
-00009e90: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-00009ea0: 746f 2062 6f6f 7374 6572 2074 6865 2072  to booster the r
-00009eb0: 756e 6e69 6e67 2073 7065 6564 2063 616e  unning speed can
-00009ec0: 2062 6520 7365 7420 6279 2074 6865 206f   be set by the o
-00009ed0: 7074 696f 6e20 602d 6e60 206f 7220 602d  ption `-n` or `-
-00009ee0: 2d6e 7468 7265 6164 7360 2077 6869 6368  -nthreads` which
-00009ef0: 2069 7320 7375 6767 6573 7465 6420 746f   is suggested to
-00009f00: 2062 6520 7365 7420 6173 2028 4e2d 3129   be set as (N-1)
-00009f10: 4e2f 3220 7768 6572 6520 4e20 6973 2074  N/2 where N is t
-00009f20: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
-00009f30: 2066 696c 6573 2074 6f20 6163 6869 6576   files to achiev
-00009f40: 6520 7468 6520 6869 6768 6573 7420 6566  e the highest ef
-00009f50: 6669 6369 656e 6379 2e20 5468 6520 6f70  ficiency. The op
-00009f60: 7469 6f6e 2060 2d2d 6f67 666f 726d 6174  tion `--ogformat
-00009f70: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
-00009f80: 6164 6420 696e 6465 7820 2866 6f72 2069  add index (for i
-00009f90: 6e73 7461 6e63 6520 4746 3030 3030 3030  nstance GF000000
-00009fa0: 3031 2920 746f 2074 6865 206f 7574 7075  01) to the outpu
-00009fb0: 7420 5242 4820 6765 6e65 2066 616d 696c  t RBH gene famil
-00009fc0: 6965 7320 7768 6963 6820 6361 6e20 6265  ies which can be
-00009fd0: 2066 7572 7468 6572 2075 7365 6420 696e   further used in
-00009fe0: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00009ff0: 7375 623e 2063 616c 6375 6c61 7469 6f6e  sub> calculation
-0000a000: 2062 7920 6077 6764 206b 7364 602e 0a0a   by `wgd ksd`...
-0000a010: 2a2a 5468 6520 7375 6767 6573 7465 6420  **The suggested 
-0000a020: 636f 6d6d 616e 6420 746f 2073 7461 7274  command to start
-0000a030: 2077 6974 6820 6973 2061 6c73 6f20 756e   with is also un
-0000a040: 6465 7220 7468 6520 6465 6661 756c 7420  der the default 
-0000a050: 7365 7474 696e 6720 7769 7468 2074 6865  setting with the
-0000a060: 2063 6f6d 6d61 6e64 2073 686f 776e 2062   command shown b
-0000a070: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-0000a080: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
-0000a090: 7175 656e 6365 320a 6060 600a 0a2a 2a54  quence2.```..**T
-0000a0a0: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
-0000a0b0: 6620 6c6f 6361 6c20 4d52 4248 732a 2a0a  f local MRBHs**.
-0000a0c0: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000a0d0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000a0e0: 7365 7175 656e 6365 3320 2d66 2073 6571  sequence3 -f seq
-0000a0f0: 7565 6e63 6531 202d 6520 3165 2d31 3020  uence1 -e 1e-10 
-0000a100: 2d62 7320 3130 3020 2d6e 7020 3520 282d  -bs 100 -np 5 (-
-0000a110: 6e6e 2920 282d 6e20 3429 2028 2d6b 6629  nn) (-n 4) (-kf)
-0000a120: 2028 2d6b 6429 2028 2d63 2030 2e39 2920   (-kd) (-c 0.9) 
-0000a130: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
-0000a140: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
-0000a150: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
-0000a160: 290a 6060 600a 0a2a 2a54 6865 2064 6973  ).```..**The dis
-0000a170: 7469 6e63 7469 6f6e 2062 6574 7765 656e  tinction between
-0000a180: 206c 6f63 616c 2061 6e64 2067 6c6f 6261   local and globa
-0000a190: 6c20 2868 6572 6575 6e64 6572 2920 4d52  l (hereunder) MR
-0000a1a0: 4248 7320 6973 2074 6861 7420 6c6f 6361  BHs is that loca
-0000a1b0: 6c20 4d52 4248 7320 6172 6520 7468 6520  l MRBHs are the 
-0000a1c0: 7265 7375 6c74 7320 6f66 206d 6572 6765  results of merge
-0000a1d0: 6420 5242 4873 206f 6e20 6120 6a6f 696e  d RBHs on a join
-0000a1e0: 7420 666f 6361 6c20 7370 6563 6965 732c  t focal species,
-0000a1f0: 2066 6f72 2069 6e73 7461 6e63 6520 696e   for instance in
-0000a200: 2061 2074 6872 6565 2073 7065 6369 6573   a three species
-0000a210: 2073 7973 7465 6d20 2841 2c28 422c 4329   system (A,(B,C)
-0000a220: 292c 2074 6865 206c 6f63 616c 204d 5242  ), the local MRB
-0000a230: 4873 206f 6620 4120 6f6e 6c79 2072 6571  Hs of A only req
-0000a240: 7569 7265 2074 6865 2063 616c 6375 6c61  uire the calcula
-0000a250: 7469 6f6e 206f 6620 5242 4873 2062 6574  tion of RBHs bet
-0000a260: 7765 656e 2041 2061 6e64 2043 2028 6465  ween A and C (de
-0000a270: 6e6f 7465 6420 6173 2041 4329 2061 6e64  noted as AC) and
-0000a280: 2041 4220 616e 6420 7468 656e 2074 6865   AB and then the
-0000a290: 206d 6572 6769 6e67 206f 6620 4142 2061   merging of AB a
-0000a2a0: 6e64 2041 4320 6174 2074 6865 2061 7869  nd AC at the axi
-0000a2b0: 7320 6f66 2041 2c20 7768 696c 6520 7468  s of A, while th
-0000a2c0: 6520 676c 6f61 626c 204d 5242 4873 2061  e gloabl MRBHs a
-0000a2d0: 7265 2069 6e64 6570 656e 6465 6e74 206f  re independent o
-0000a2e0: 6620 666f 6361 6c20 7370 6563 6965 7320  f focal species 
-0000a2f0: 696e 2074 6861 7420 6974 206a 7573 7420  in that it just 
-0000a300: 6361 6c6c 7320 7468 6520 6361 6c63 756c  calls the calcul
-0000a310: 6174 696f 6e73 206f 6620 616c 6c20 706f  ations of all po
-0000a320: 7373 6962 6c65 2073 7065 6369 6573 2070  ssible species p
-0000a330: 6169 7220 286e 6f74 2073 656c 6620 746f  air (not self to
-0000a340: 2073 656c 6620 7370 6563 6965 7320 7061   self species pa
-0000a350: 6972 292c 2073 7563 6820 7468 6174 2041  ir), such that A
-0000a360: 422c 2041 432c 2061 6e64 2042 4320 6172  B, AC, and BC ar
-0000a370: 6520 616c 6c20 746f 2062 6520 6361 6c63  e all to be calc
-0000a380: 756c 6174 6564 2061 6e64 206d 6572 6765  ulated and merge
-0000a390: 642e 2a2a 0a0a 5477 6f20 7479 7065 7320  d.**..Two types 
-0000a3a0: 6f66 204d 5242 4873 2061 7320 696e 7465  of MRBHs as inte
-0000a3b0: 7072 6574 6174 6564 2061 626f 7665 2063  pretated above c
-0000a3c0: 616e 2062 6520 6465 6c69 6e65 6174 6564  an be delineated
-0000a3d0: 2062 7920 6077 6764 2064 6d64 602c 2074   by `wgd dmd`, t
-0000a3e0: 6865 206c 6f63 616c 204d 5242 4873 2061  he local MRBHs a
-0000a3f0: 6e64 2074 6865 2067 6c6f 6261 6c20 4d52  nd the global MR
-0000a400: 4248 732e 2054 6865 206c 6f63 616c 204d  BHs. The local M
-0000a410: 5242 4873 2061 7265 2063 6f6e 7374 7275  RBHs are constru
-0000a420: 6374 6564 2062 7920 6d65 7267 696e 6720  cted by merging 
-0000a430: 616c 6c20 7468 6520 7265 6c65 7661 6e74  all the relevant
-0000a440: 2052 4248 7320 6f6e 6c79 2077 6974 6820   RBHs only with 
-0000a450: 7468 6520 666f 6361 6c20 7370 6563 6965  the focal specie
-0000a460: 732c 2077 6869 6368 2069 7320 7365 7420  s, which is set 
-0000a470: 6279 2060 2d66 6020 6f72 2060 2d2d 666f  by `-f` or `--fo
-0000a480: 6375 7360 2e20 5468 6520 7061 7261 6d65  cus`. The parame
-0000a490: 7465 7220 602d 6b66 6020 6f72 2060 2d2d  ter `-kf` or `--
-0000a4a0: 6b65 6570 6661 7374 6160 2063 616e 2062  keepfasta` can b
-0000a4b0: 6520 7365 7420 746f 2072 6574 6169 6e20  e set to retain 
-0000a4c0: 7468 6520 7365 7175 656e 6365 2069 6e66  the sequence inf
-0000a4d0: 6f72 6d61 7469 6f6e 206f 6620 6561 6368  ormation of each
-0000a4e0: 204d 5242 482e 2054 6865 2070 6172 616d   MRBH. The param
-0000a4f0: 6574 6572 2060 2d6b 6460 206f 7220 602d  eter `-kd` or `-
-0000a500: 2d6b 6565 7064 7570 6c69 6361 7465 7360  -keepduplicates`
-0000a510: 2064 6574 6572 6d69 6e65 7320 7768 6574   determines whet
-0000a520: 6865 7220 7468 6520 7361 6d65 2067 656e  her the same gen
-0000a530: 6573 2063 616e 2061 7070 6561 7220 696e  es can appear in
-0000a540: 2064 6966 6665 7265 6e74 206c 6f63 616c   different local
-0000a550: 204d 5242 4873 2e20 4e6f 726d 616c 6c79   MRBHs. Normally
-0000a560: 2074 6865 7265 2077 696c 6c20 6265 206e   there will be n
-0000a570: 6f20 6475 706c 6963 6174 6573 2069 6e20  o duplicates in 
-0000a580: 7468 6520 6c6f 6361 6c20 4d52 4248 7320  the local MRBHs 
-0000a590: 6275 7420 6966 2075 7365 7273 2073 6574  but if users set
-0000a5a0: 2074 6865 2060 2d63 6020 6173 2030 2e39   the `-c` as 0.9
-0000a5b0: 2028 6f72 2061 6e79 2076 616c 7565 2073   (or any value s
-0000a5c0: 6d61 6c6c 6572 2074 6861 6e20 3129 2c20  maller than 1), 
-0000a5d0: 6974 2773 206c 696b 656c 7920 7468 6174  it's likely that
-0000a5e0: 2074 6865 2073 616d 6520 6765 6e65 2077   the same gene w
-0000a5f0: 696c 6c20 6861 7665 2063 6861 6e63 6520  ill have chance 
-0000a600: 746f 2061 7070 6561 7220 6d75 6c74 6970  to appear multip
-0000a610: 6c65 2074 696d 6573 2069 6e20 6469 6666  le times in diff
-0000a620: 6572 656e 7420 6c6f 6361 6c20 4d52 4248  erent local MRBH
-0000a630: 732e 2054 6861 7420 6973 2074 6f20 7361  s. That is to sa
-0000a640: 792c 2074 6865 2070 6172 616d 6574 6572  y, the parameter
-0000a650: 2060 2d6b 6460 2069 7320 6d65 616e 696e   `-kd` is meanin
-0000a660: 6766 756c 206f 6e6c 7920 7768 656e 2069  gful only when i
-0000a670: 7427 7320 7365 7420 746f 6765 7468 6572  t's set together
-0000a680: 2077 6974 6820 7468 6520 7061 7261 6d65   with the parame
-0000a690: 7465 7220 602d 6360 2e20 5468 6520 6e75  ter `-c`. The nu
-0000a6a0: 6d62 6572 206f 6620 7061 7261 6c6c 656c  mber of parallel
-0000a6b0: 2074 6872 6561 6473 2069 7320 7375 6767   threads is sugg
-0000a6c0: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
-0000a6d0: 6173 2074 6865 206e 756d 6265 7220 6f66  as the number of
-0000a6e0: 2063 6473 2066 696c 6573 206d 696e 7573   cds files minus
-0000a6f0: 2031 2e0a 0a2a 2a41 2073 7567 6765 7374   1...**A suggest
-0000a700: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000a710: 6973 2075 6e64 6572 2074 6865 2064 6566  is under the def
-0000a720: 6175 6c74 2070 6172 616d 6574 6572 2077  ault parameter w
-0000a730: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000a740: 7368 6f77 6e20 6173 2062 656c 6f77 2e2a  shown as below.*
-0000a750: 2a0a 0a60 6060 0a77 6764 2064 6d64 2073  *..```.wgd dmd s
-0000a760: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000a770: 6532 2073 6571 7565 6e63 6533 202d 6620  e2 sequence3 -f 
-0000a780: 7365 7175 656e 6365 310a 6060 600a 0a2a  sequence1.```..*
-0000a790: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-0000a7a0: 206f 6620 676c 6f62 616c 204d 5242 4873   of global MRBHs
-0000a7b0: 2a2a 0a60 6060 0a77 6764 2064 6d64 2073  **.```.wgd dmd s
-0000a7c0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000a7d0: 6532 2073 6571 7565 6e63 6533 202d 676d  e2 sequence3 -gm
-0000a7e0: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-0000a7f0: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-0000a800: 6e20 3429 2028 2d6b 6629 2028 2d6b 6429  n 4) (-kf) (-kd)
-0000a810: 2028 2d63 2030 2e39 2920 282d 2d74 6f5f   (-c 0.9) (--to_
-0000a820: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
-0000a830: 6f20 7767 645f 646d 6429 2028 2d74 2077  o wgd_dmd) (-t w
-0000a840: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
-0000a850: 0a54 6865 2067 6c6f 6261 6c20 4d52 4248  .The global MRBH
-0000a860: 7320 6973 2063 6f6e 7374 7275 6374 6564  s is constructed
-0000a870: 2062 7920 6578 6861 7573 7469 7665 6c79   by exhaustively
-0000a880: 206d 6572 6769 6e67 2061 6c6c 2074 6865   merging all the
-0000a890: 2070 6f73 7369 626c 6520 7061 6972 2d77   possible pair-w
-0000a8a0: 6973 6520 5242 4873 2065 7863 6570 7420  ise RBHs except 
-0000a8b0: 666f 7220 7175 6572 7969 6e67 2074 6865  for querying the
-0000a8c0: 2073 6571 7565 6e63 6520 6974 7365 6c66   sequence itself
-0000a8d0: 2c20 7768 6963 6820 6361 6e20 6265 2069  , which can be i
-0000a8e0: 6e69 7469 6174 6564 2062 7920 6164 6420  nitiated by add 
-0000a8f0: 7468 6520 666c 6167 2060 2d67 6d60 206f  the flag `-gm` o
-0000a900: 7220 602d 2d67 6c6f 6261 6c6d 7262 6860  r `--globalmrbh`
-0000a910: 2e20 5468 6520 7265 7374 206f 6620 7265  . The rest of re
-0000a920: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
-0000a930: 7320 7374 6179 7320 7468 6520 7361 6d65  s stays the same
-0000a940: 2061 7320 7468 6520 6c6f 6361 6c20 4d52   as the local MR
-0000a950: 4248 732e 2054 6865 206e 756d 6265 7220  BHs. The number 
-0000a960: 6f66 2070 6172 616c 6c65 6c20 7468 7265  of parallel thre
-0000a970: 6164 7320 6973 2073 7567 6765 7374 6564  ads is suggested
-0000a980: 2074 6f20 6265 2073 6574 2061 7320 284e   to be set as (N
-0000a990: 2d31 294e 2f32 2074 6f6f 2077 6865 7265  -1)N/2 too where
-0000a9a0: 204e 2069 7320 7468 6520 6e75 6d62 6572   N is the number
-0000a9b0: 206f 6620 6364 7320 6669 6c65 7320 746f   of cds files to
-0000a9c0: 2061 6368 6965 7665 2074 6865 2068 6967   achieve the hig
-0000a9d0: 6865 7374 2065 6666 6963 6965 6e63 792e  hest efficiency.
-0000a9e0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-0000a9f0: 7374 6172 7469 6e67 2072 756e 2069 7320  starting run is 
-0000aa00: 756e 6465 7220 7468 6520 6465 6661 756c  under the defaul
-0000aa10: 7420 7061 7261 6d65 7465 7220 7769 7468  t parameter with
-0000aa20: 2074 6865 2063 6f6d 6d61 6e64 2073 686f   the command sho
-0000aa30: 776e 2061 7320 6265 6c6f 772e 2a2a 0a0a  wn as below.**..
-0000aa40: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000aa50: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000aa60: 7365 7175 656e 6365 3320 2d67 6d0a 6060  sequence3 -gm.``
-0000aa70: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
-0000aa80: 7469 6f6e 206f 6620 6f72 7468 6f67 726f  tion of orthogro
-0000aa90: 7570 732a 2a0a 6060 600a 7767 6420 646d  ups**.```.wgd dm
-0000aaa0: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
-0000aab0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000aac0: 2d6f 6920 2d6f 6f20 2d65 2031 652d 3130  -oi -oo -e 1e-10
-0000aad0: 202d 6273 2031 3030 202d 6e70 2035 2028   -bs 100 -np 5 (
-0000aae0: 2d6e 6e29 2028 2d63 6329 2028 2d74 6529  -nn) (-cc) (-te)
-0000aaf0: 2028 2d6d 6320 302e 3829 2028 2d67 6e29   (-mc 0.8) (-gn)
-0000ab00: 2028 2d74 7265 6520 2766 6173 7474 7265   (-tree 'fasttre
-0000ab10: 6527 2920 282d 7473 2027 2d66 6173 7465  e') (-ts '-faste
-0000ab20: 7374 2729 2028 2d6e 2034 2920 282d 2d74  st') (-n 4) (--t
-0000ab30: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000ab40: 282d 6f20 7767 645f 646d 6429 2028 2d74  (-o wgd_dmd) (-t
-0000ab50: 2077 6f72 6b69 6e67 5f74 6d70 290a 6060   working_tmp).``
-0000ab60: 600a 0a49 6e20 6077 6764 2076 3260 2c20  `..In `wgd v2`, 
-0000ab70: 7765 2061 6c73 6f20 696d 706c 656d 656e  we also implemen
-0000ab80: 7465 6420 616e 2061 6c67 6f72 6974 686d  ted an algorithm
-0000ab90: 206f 6620 6465 6c69 6e65 6174 696e 6720   of delineating 
-0000aba0: 6f72 7468 6f67 726f 7570 732c 2077 6869  orthogroups, whi
-0000abb0: 6368 2063 616e 2062 6520 696e 6974 6961  ch can be initia
-0000abc0: 7465 6420 7769 7468 2074 6865 2070 6172  ted with the par
-0000abd0: 616d 6574 6572 2060 2d6f 6960 206f 7220  ameter `-oi` or 
-0000abe0: 602d 2d6f 7274 686f 696e 6665 7260 2e20  `--orthoinfer`. 
-0000abf0: 5477 6f20 7761 7973 206f 6620 6465 6c69  Two ways of deli
-0000ac00: 6e65 6174 696f 6e20 6361 6e20 6265 2063  neation can be c
-0000ac10: 686f 7365 6e2c 2074 6865 2063 6f6e 6361  hosen, the conca
-0000ac20: 7465 6e61 7469 6f6e 2077 6179 2028 7365  tenation way (se
-0000ac30: 7420 6279 2074 6865 2070 6172 616d 6574  t by the paramet
-0000ac40: 6572 2060 2d63 6360 206f 7220 602d 2d63  er `-cc` or `--c
-0000ac50: 6f6e 6361 7460 2920 6f72 2074 6865 206e  oncat`) or the n
-0000ac60: 6f6e 2d63 6f6e 6361 7465 6e61 7469 6f6e  on-concatenation
-0000ac70: 2028 6465 6661 756c 7429 2077 6179 2e20   (default) way. 
-0000ac80: 496e 2062 7269 6566 2c20 7468 6520 636f  In brief, the co
-0000ac90: 6e63 6174 656e 6174 696f 6e20 7761 7920  ncatenation way 
-0000aca0: 6f66 2064 656c 696e 6561 7469 6e67 206f  of delineating o
-0000acb0: 7274 686f 6772 6f75 7073 2073 7461 7274  rthogroups start
-0000acc0: 7320 7769 7468 2063 6f6e 6361 7465 6e61  s with concatena
-0000acd0: 7469 6e67 2061 6c6c 2074 6865 2073 6571  ting all the seq
-0000ace0: 7565 6e63 6573 2069 6e74 6f20 6120 7369  uences into a si
-0000acf0: 6e67 6c65 2073 6571 7565 6e63 6520 6669  ngle sequence fi
-0000ad00: 6c65 2061 6e64 2074 6865 6e20 696e 6665  le and then infe
-0000ad10: 7272 696e 6720 7468 6520 7768 6f6c 6520  rring the whole 
-0000ad20: 7061 7261 6e6f 6d65 206f 6620 7468 6973  paranome of this
-0000ad30: 2073 696e 676c 6520 7365 7175 656e 6365   single sequence
-0000ad40: 2066 696c 6520 7769 7468 2074 6865 2063   file with the c
-0000ad50: 6c75 7374 6572 696e 6720 7265 7375 6c74  lustering result
-0000ad60: 7320 6d61 7070 6564 2062 6163 6b20 746f  s mapped back to
-0000ad70: 2074 6865 2062 656c 6f6e 6769 6e67 2073   the belonging s
-0000ad80: 7065 6369 6573 2e20 5768 696c 6520 7468  pecies. While th
-0000ad90: 6520 6e6f 6e2d 636f 6e63 6174 656e 6174  e non-concatenat
-0000ada0: 696f 6e20 7761 7920 7374 6172 7473 2077  ion way starts w
-0000adb0: 6974 6820 7265 7370 6563 7469 7665 2070  ith respective p
-0000adc0: 6169 722d 7769 7365 2064 6961 6d6f 6e64  air-wise diamond
-0000add0: 2073 6561 7263 6820 2869 6e63 6c75 6469   search (includi
-0000ade0: 6e67 2071 7565 7279 696e 6720 7468 6520  ng querying the 
-0000adf0: 7361 6d65 2073 6571 7565 6e63 6520 6974  same sequence it
-0000ae00: 7365 6c66 2920 616e 6420 7468 656e 2061  self) and then a
-0000ae10: 6c6c 2074 6865 2073 6571 7565 6e63 6520  ll the sequence 
-0000ae20: 7369 6d69 6c61 7269 7479 2074 6162 6c65  similarity table
-0000ae30: 7320 7769 6c6c 2062 6520 636f 6e63 6174  s will be concat
-0000ae40: 656e 6174 6564 2061 6e64 2063 6c75 7374  enated and clust
-0000ae50: 6572 6564 2069 6e74 6f20 6f72 7468 6f67  ered into orthog
-0000ae60: 726f 7570 732e 2053 6f6d 6520 6f74 6865  roups. Some othe
-0000ae70: 7220 706f 7373 6962 6c79 2075 7365 6675  r possibly usefu
-0000ae80: 6c20 706f 7374 2d63 6c75 7374 6572 696e  l post-clusterin
-0000ae90: 6720 6675 6e63 7469 6f6e 7320 6361 6e20  g functions can 
-0000aea0: 6265 2069 6e69 7469 6174 6564 2c20 696e  be initiated, in
-0000aeb0: 636c 7564 696e 6720 7468 6520 7061 7261  cluding the para
-0000aec0: 6d65 7465 7220 602d 7465 6020 6f72 2060  meter `-te` or `
-0000aed0: 2d2d 7465 7374 736f 6760 2c20 7768 6963  --testsog`, whic
-0000aee0: 6820 6361 6e20 6265 2073 6574 2074 6f20  h can be set to 
-0000aef0: 7374 6172 7420 7468 6520 756e 6269 6173  start the unbias
-0000af00: 6564 2074 6573 7420 6f66 2073 696e 676c  ed test of singl
-0000af10: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-0000af20: 6c69 6573 2028 6e6f 7465 2074 6861 7420  lies (note that 
-0000af30: 7468 6973 2066 756e 6374 696f 6e20 6e65  this function ne
-0000af40: 6564 7320 6068 6d6d 6572 6020 2876 332e  eds `hmmer` (v3.
-0000af50: 3162 3229 2074 6f20 6265 2069 6e73 7461  1b2) to be insta
-0000af60: 6c6c 6564 2069 6e20 7468 6520 656e 7669  lled in the envi
-0000af70: 726f 6e6d 656e 7420 7061 7468 292c 2074  ronment path), t
-0000af80: 6865 2070 6172 616d 6574 6572 2060 2d6d  he parameter `-m
-0000af90: 6360 206f 7220 602d 2d6d 736f 6763 7574  c` or `--msogcut
-0000afa0: 602c 2072 616e 6769 6e67 2062 6574 7765  `, ranging betwe
-0000afb0: 656e 2030 2074 6f20 312c 2077 6869 6368  en 0 to 1, which
-0000afc0: 2063 616e 2062 6520 7365 7420 746f 2073   can be set to s
-0000afd0: 6561 7263 6820 7468 6520 736f 2d63 616c  earch the so-cal
-0000afe0: 6c65 6420 6d6f 7374 6c79 2073 696e 676c  led mostly singl
-0000aff0: 652d 636f 7079 2066 616d 696c 7920 7768  e-copy family wh
-0000b000: 6963 6820 6861 7320 6869 6768 6572 2074  ich has higher t
-0000b010: 6861 6e20 6365 7274 6169 6e20 6375 742d  han certain cut-
-0000b020: 6f66 6620 7065 7263 656e 7461 6765 206f  off percentage o
-0000b030: 6620 7370 6563 6965 7320 636f 7665 7261  f species covera
-0000b040: 6765 2c20 7468 6520 7061 7261 6d65 7465  ge, the paramete
-0000b050: 7220 602d 676e 6020 6f72 2060 2d2d 6765  r `-gn` or `--ge
-0000b060: 746e 736f 6760 2c20 7768 6963 6820 6361  tnsog`, which ca
-0000b070: 6e20 6265 2073 6574 2074 6f20 7365 6172  n be set to sear
-0000b080: 6368 2066 6f72 206e 6573 7465 6420 7369  ch for nested si
-0000b090: 6e67 6c65 2d63 6f70 7920 6765 6e65 2066  ngle-copy gene f
-0000b0a0: 616d 696c 6965 7320 284e 534f 4773 2920  amilies (NSOGs) 
-0000b0b0: 7768 6963 6820 6973 206f 7269 6769 6e61  which is origina
-0000b0c0: 6c6c 7920 6d75 6c74 6979 2d63 6f70 7920  lly multiy-copy 
-0000b0d0: 6275 7420 6861 7320 6120 286d 6f73 746c  but has a (mostl
-0000b0e0: 7929 2073 696e 676c 652d 636f 7079 2062  y) single-copy b
-0000b0f0: 7261 6e63 6820 2877 6869 6368 2072 6571  ranch (which req
-0000b100: 7569 7265 7320 7468 6520 6368 6f73 656e  uires the chosen
-0000b110: 2074 7265 652d 696e 6665 7265 6e63 6520   tree-inference 
-0000b120: 7072 6f67 7261 6d20 7365 7420 6279 2060  program set by `
-0000b130: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
-0000b140: 655f 6d65 7468 6f64 6020 746f 2062 6520  e_method` to be 
-0000b150: 7072 652d 696e 7374 616c 6c65 6420 696e  pre-installed in
-0000b160: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-0000b170: 2070 6174 6820 7769 7468 2074 6865 2070   path with the p
-0000b180: 6172 616d 6574 6572 7320 7365 7474 696e  arameters settin
-0000b190: 6720 666f 7220 6765 6e65 2074 7265 6520  g for gene tree 
-0000b1a0: 696e 6665 7265 6e63 6520 636f 6e74 726f  inference contro
-0000b1b0: 6c6c 6564 2062 7920 602d 7473 6020 6f72  lled by `-ts` or
-0000b1c0: 2060 2d2d 7472 6565 7365 7460 292e 2054   `--treeset`). T
-0000b1d0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000b1e0: 646d 6460 2077 6f75 6c64 2073 7469 6c6c  dmd` would still
-0000b1f0: 2063 6f6e 6475 6374 2074 6865 2052 4248   conduct the RBH
-0000b200: 7320 6361 6c63 756c 6174 696f 6e20 756e  s calculation un
-0000b210: 6c65 7373 2074 6865 2070 6172 616d 6574  less the paramet
-0000b220: 6572 2060 2d6f 6f60 206f 7220 602d 2d6f  er `-oo` or `--o
-0000b230: 6e6c 796f 7274 686f 6020 7761 7320 7365  nlyortho` was se
-0000b240: 742e 2049 6620 6f6e 6520 6f6e 6c79 2077  t. If one only w
-0000b250: 616e 7473 2074 6f20 696e 6665 7220 7468  ants to infer th
-0000b260: 6520 6f72 7468 6f67 726f 7570 732c 2069  e orthogroups, i
-0000b270: 7427 7320 7375 6767 6573 7465 6420 746f  t's suggested to
-0000b280: 2061 6464 2074 6865 2066 6c61 6720 602d   add the flag `-
-0000b290: 6f6f 6020 746f 206a 7573 7420 696d 706c  oo` to just impl
-0000b2a0: 656d 656e 7420 7468 6520 6f72 7468 6f67  ement the orthog
-0000b2b0: 726f 7570 7320 6465 6c69 6e65 6174 696f  roups delineatio
-0000b2c0: 6e20 616e 616c 7973 6973 2e20 5468 6520  n analysis. The 
-0000b2d0: 6e75 6d62 6572 206f 6620 7061 7261 6c6c  number of parall
-0000b2e0: 656c 2074 6872 6561 6473 2069 7320 7375  el threads is su
-0000b2f0: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000b300: 7420 6173 2028 4e2b 3129 4e2f 3220 7768  t as (N+1)N/2 wh
-0000b310: 6572 6520 4e20 6973 2074 6865 206e 756d  ere N is the num
-0000b320: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
-0000b330: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-0000b340: 6869 6768 6573 7420 6566 6669 6369 656e  highest efficien
-0000b350: 6379 2073 696e 6365 2074 6865 2073 656c  cy since the sel
-0000b360: 662d 636f 6d70 6172 6973 6f6e 2069 7320  f-comparison is 
-0000b370: 616c 736f 2069 6e63 6c75 6465 642e 0a0a  also included...
-0000b380: 2a2a 5468 6520 6465 6661 756c 7420 7365  **The default se
-0000b390: 7474 696e 6720 6f66 2070 6172 616d 6574  tting of paramet
-0000b3a0: 6572 7320 6973 2061 2072 6561 736f 6e61  ers is a reasona
-0000b3b0: 626c 6520 7374 6172 7469 6e67 2070 6f69  ble starting poi
-0000b3c0: 6e74 2077 6974 6820 7468 6520 636f 6d6d  nt with the comm
-0000b3d0: 616e 6420 6173 2062 656c 6f77 2e2a 2a0a  and as below.**.
-0000b3e0: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-0000b3f0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-0000b400: 2073 6571 7565 6e63 6533 202d 6f69 202d   sequence3 -oi -
-0000b410: 6f6f 0a60 6060 0a0a 2a2a 5468 6520 636f  oo.```..**The co
-0000b420: 6c6c 696e 6561 7220 636f 616c 6573 6365  llinear coalesce
-0000b430: 6e63 6520 696e 6665 7265 6e63 6520 6f66  nce inference of
-0000b440: 2070 6879 6c6f 6765 6e79 2a2a 0a60 6060   phylogeny**.```
-0000b450: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000b460: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000b470: 7565 6e63 6533 202d 6170 2061 7064 6174  uence3 -ap apdat
-0000b480: 6120 2d73 6d20 736d 6461 7461 202d 6c65  a -sm smdata -le
-0000b490: 206c 6564 6174 6120 2d67 7420 6774 6461   ledata -gt gtda
-0000b4a0: 7461 202d 636f 6320 282d 7472 6565 2027  ta -coc (-tree '
-0000b4b0: 6661 7374 7472 6565 2729 2028 2d74 7320  fasttree') (-ts 
-0000b4c0: 272d 6661 7374 6573 7427 2920 282d 6e20  '-fastest') (-n 
-0000b4d0: 3429 2028 2d2d 746f 5f73 746f 7029 2028  4) (--to_stop) (
-0000b4e0: 2d2d 6364 7329 2028 2d6f 2077 6764 5f64  --cds) (-o wgd_d
-0000b4f0: 6d64 2920 282d 7420 776f 726b 696e 675f  md) (-t working_
-0000b500: 746d 7029 0a60 6060 0a0a 4120 6e6f 7665  tmp).```..A nove
-0000b510: 6c20 7068 796c 6f67 656e 6574 6963 2069  l phylogenetic i
-0000b520: 6e66 6572 656e 6365 206d 6574 686f 6420  nference method 
-0000b530: 6e61 6d65 6420 2263 6f6c 6c69 6e65 6172  named "collinear
-0000b540: 2063 6f61 6c65 7363 656e 6365 2069 6e66   coalescence inf
-0000b550: 6572 656e 6365 2220 6973 2061 6c73 6f20  erence" is also 
-0000b560: 696d 706c 656d 656e 7465 6420 696e 2060  implemented in `
-0000b570: 7767 6420 7632 602e 2046 6f72 2074 6869  wgd v2`. For thi
-0000b580: 7320 616e 616c 7973 6973 2c20 7573 6572  s analysis, user
-0000b590: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-0000b5a0: 6520 7468 6520 616e 6368 6f72 2070 6f69  e the anchor poi
-0000b5b0: 6e74 7320 6669 6c65 2062 7920 602d 6170  nts file by `-ap
-0000b5c0: 6020 6f72 2060 2d2d 616e 6368 6f72 706f  ` or `--anchorpo
-0000b5d0: 696e 7473 602c 2074 6865 2063 6f6c 6c69  ints`, the colli
-0000b5e0: 6e65 6172 2073 6567 6d65 6e74 7320 6669  near segments fi
-0000b5f0: 6c65 2062 7920 602d 736d 6020 6f72 2060  le by `-sm` or `
-0000b600: 2d2d 7365 676d 656e 7473 602c 2074 6865  --segments`, the
-0000b610: 206c 6973 7473 6567 6d65 6e74 7320 6669   listsegments fi
-0000b620: 6c65 2062 7920 602d 6c65 6020 6f72 2060  le by `-le` or `
-0000b630: 2d2d 6c69 7374 656c 656d 656e 7473 602c  --listelements`,
-0000b640: 2061 6e64 2074 6865 2067 656e 6520 7461   and the gene ta
-0000b650: 626c 6520 6669 6c65 2062 7920 602d 6774  ble file by `-gt
-0000b660: 6020 6f72 2060 2d2d 6765 6e65 7461 626c  ` or `--genetabl
-0000b670: 6560 2c20 616c 6c20 6f66 2077 6869 6368  e`, all of which
-0000b680: 2063 616e 2062 6520 7072 6f64 7563 6564   can be produced
-0000b690: 2069 6e20 7468 6520 7072 6f67 7261 6d20   in the program 
-0000b6a0: 6077 6764 2073 796e 602e 2054 6865 2070  `wgd syn`. The p
-0000b6b0: 6172 616d 6574 6572 2060 2d63 6f63 6020  arameter `-coc` 
-0000b6c0: 6f72 2060 2d2d 636f 6c6c 696e 6561 7263  or `--collinearc
-0000b6d0: 6f61 6c65 7363 656e 6365 6020 6e65 6564  oalescence` need
-0000b6e0: 7320 746f 2062 6520 7365 7420 746f 2073  s to be set to s
-0000b6f0: 7461 7274 2074 6869 7320 616e 616c 7973  tart this analys
-0000b700: 6973 2e20 5468 6520 7472 6565 2d69 6e66  is. The tree-inf
-0000b710: 6572 656e 6365 2070 726f 6772 616d 2061  erence program a
-0000b720: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
-0000b730: 6420 7061 7261 6d65 7465 7273 2063 616e  d parameters can
-0000b740: 2062 6520 7365 7420 6a75 7374 2061 7320   be set just as 
-0000b750: 6162 6f76 6520 6279 2060 2d74 7265 6560  above by `-tree`
-0000b760: 206f 7220 602d 2d74 7265 655f 6d65 7468   or `--tree_meth
-0000b770: 6f64 6020 616e 6420 602d 7473 6020 6f72  od` and `-ts` or
-0000b780: 2060 2d2d 7472 6565 7365 7460 2e20 506c   `--treeset`. Pl
-0000b790: 6561 7365 2061 6c73 6f20 6d61 6b65 2073  ease also make s
-0000b7a0: 7572 6520 7468 6520 6368 6f73 656e 2074  ure the chosen t
-0000b7b0: 7265 652d 696e 6665 7265 6e63 6520 7072  ree-inference pr
-0000b7c0: 6f67 7261 6d20 6973 2069 6e73 7461 6c6c  ogram is install
-0000b7d0: 6564 2069 6e20 7468 6520 656e 7669 726f  ed in the enviro
-0000b7e0: 6e6d 656e 7420 7061 7468 2e20 5468 6520  nment path. The 
-0000b7f0: 7072 6f67 7261 6d20 6061 7374 7261 6c2d  program `astral-
-0000b800: 7072 6f60 2069 7320 7265 7175 6972 6564  pro` is required
-0000b810: 2074 6f20 6265 2069 6e73 7461 6c6c 6564   to be installed
-0000b820: 2069 6e20 7468 6520 656e 7669 726f 6e6d   in the environm
-0000b830: 656e 7420 7061 7468 2074 6f6f 2e20 4e6f  ent path too. No
-0000b840: 7465 2074 6861 7420 7468 6572 6520 7368  te that there sh
-0000b850: 6f75 6c64 2062 6520 6e6f 2064 7570 6c69  ould be no dupli
-0000b860: 6361 7465 6420 6765 6e65 2049 4473 2069  cated gene IDs i
-0000b870: 6e20 7468 6520 7365 7175 656e 6365 2066  n the sequence f
-0000b880: 696c 652e 2054 6865 2070 6172 616c 6c65  ile. The paralle
-0000b890: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
-0000b8a0: 7265 2074 6f20 6163 6365 6c65 7261 7465  re to accelerate
-0000b8b0: 2074 6865 2073 6571 7565 6e63 6520 616c   the sequence al
-0000b8c0: 6967 6e6d 656e 7420 616e 6420 6765 6e65  ignment and gene
-0000b8d0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000b8e0: 666f 7220 6561 6368 2067 656e 6520 6661  for each gene fa
-0000b8f0: 6d69 6c79 2061 6e64 2074 6875 7320 7375  mily and thus su
-0000b900: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000b910: 7420 6173 206d 7563 6820 6173 2074 6865  t as much as the
-0000b920: 206e 756d 6265 7220 6f66 2067 656e 6520   number of gene 
-0000b930: 6661 6d69 6c69 6573 2e0a 0a2a 2a41 2073  families...**A s
-0000b940: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
-0000b950: 6720 7275 6e20 6f66 2074 6869 7320 616e  g run of this an
-0000b960: 616c 7973 6973 2069 7320 7769 7468 2074  alysis is with t
-0000b970: 6865 2073 696d 706c 6520 636f 6d6d 616e  he simple comman
-0000b980: 6420 6265 6c6f 772e 2a2a 0a0a 6060 600a  d below.**..```.
-0000b990: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
-0000b9a0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000b9b0: 656e 6365 3320 2d61 7020 6170 6461 7461  ence3 -ap apdata
-0000b9c0: 202d 736d 2073 6d64 6174 6120 2d6c 6520   -sm smdata -le 
-0000b9d0: 6c65 6461 7461 202d 6774 2067 7464 6174  ledata -gt gtdat
-0000b9e0: 6120 2d63 6f63 0a60 6060 0a0a 2323 2320  a -coc.```..### 
-0000b9f0: 7767 6420 666f 6375 730a 0a2a 2a54 6865  wgd focus..**The
-0000ba00: 2063 6f6e 6361 7465 6e61 7469 6f6e 2d62   concatenation-b
-0000ba10: 6173 6564 2f63 6f61 6c65 7363 656e 6365  ased/coalescence
-0000ba20: 2d62 6173 6564 2070 6879 6c6f 6765 6e65  -based phylogene
-0000ba30: 7469 6320 696e 6665 7265 6e63 652a 2a0a  tic inference**.
-0000ba40: 6060 600a 7767 6420 666f 6375 7320 6661  ```.wgd focus fa
-0000ba50: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
-0000ba60: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000ba70: 6e63 6533 2028 2d2d 636f 6e63 6174 656e  nce3 (--concaten
-0000ba80: 6174 696f 6e29 2028 2d2d 636f 616c 6573  ation) (--coales
-0000ba90: 6365 6e63 6529 2028 2d74 7265 6520 2766  cence) (-tree 'f
-0000baa0: 6173 7474 7265 6527 2920 282d 7473 2027  asttree') (-ts '
-0000bab0: 2d66 6173 7465 7374 2729 2028 2d6e 2034  -fastest') (-n 4
-0000bac0: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
-0000bad0: 2d63 6473 2920 282d 6f20 7767 645f 666f  -cds) (-o wgd_fo
-0000bae0: 6375 7329 2028 2d74 2077 6f72 6b69 6e67  cus) (-t working
-0000baf0: 5f74 6d70 290a 6060 600a 0a54 6865 2070  _tmp).```..The p
-0000bb00: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
-0000bb10: 7360 2069 6d70 6c65 6d65 6e74 6564 2074  s` implemented t
-0000bb20: 776f 2062 6173 6963 2070 6879 6c6f 6765  wo basic phyloge
-0000bb30: 6e65 7469 6320 696e 6665 7265 6e63 6520  netic inference 
-0000bb40: 6d65 7468 6f64 732c 2069 2e65 2e2c 2063  methods, i.e., c
-0000bb50: 6f6e 6361 7465 6e61 7469 6f6e 2d62 6173  oncatenation-bas
-0000bb60: 6564 2061 6e64 2063 6f61 6c65 7363 656e  ed and coalescen
-0000bb70: 6365 2d62 6173 6564 206d 6574 686f 6473  ce-based methods
-0000bb80: 2e20 546f 2069 6e69 7469 6174 6520 7468  . To initiate th
-0000bb90: 6573 6520 616e 616c 7973 6973 2c20 7573  ese analysis, us
-0000bba0: 6572 7320 6e65 6564 2074 6f20 7365 7420  ers need to set 
-0000bbb0: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-0000bbc0: 602d 2d63 6f6e 6361 7465 6e61 7469 6f6e  `--concatenation
-0000bbd0: 6020 6f72 2060 2d2d 636f 616c 6573 6365  ` or `--coalesce
-0000bbe0: 6e63 6560 2e20 5468 6520 636f 6e63 6174  nce`. The concat
-0000bbf0: 656e 6174 696f 6e2d 6261 7365 6420 6d65  enation-based me
-0000bc00: 7468 6f64 2069 6e63 6c75 6465 7320 6120  thod includes a 
-0000bc10: 6665 7720 6d61 6a6f 7220 7374 6570 732c  few major steps,
-0000bc20: 2069 2e65 2e2c 2074 6865 206d 756c 7469   i.e., the multi
-0000bc30: 706c 6520 7365 7175 656e 6365 2061 6c69  ple sequence ali
-0000bc40: 676e 6d65 6e74 2028 4d53 4129 206f 6620  gnment (MSA) of 
-0000bc50: 6561 6368 2067 656e 6520 6661 6d69 6c79  each gene family
-0000bc60: 2c20 7468 6520 636f 6e63 6174 656e 6174  , the concatenat
-0000bc70: 696f 6e20 6f66 2061 6c6c 2067 656e 6520  ion of all gene 
-0000bc80: 6661 6d69 6c69 6573 2061 6e64 2074 6865  families and the
-0000bc90: 6e20 7468 6520 6765 6e65 2074 7265 6520  n the gene tree 
-0000bca0: 2861 6c73 6f20 7370 6563 6965 7320 7472  (also species tr
-0000bcb0: 6565 2069 6e20 7468 6973 2063 6173 6529  ee in this case)
-0000bcc0: 2069 6e66 6572 656e 6365 2e20 5468 6520   inference. The 
-0000bcd0: 636f 616c 6573 6365 6e63 652d 6261 7365  coalescence-base
-0000bce0: 6420 6d65 7468 6f64 2077 696c 6c20 696e  d method will in
-0000bcf0: 7374 6561 6420 7065 7266 6f72 6d20 7468  stead perform th
-0000bd00: 6520 4d53 4120 6f66 2065 6163 6820 6765  e MSA of each ge
-0000bd10: 6e65 2066 616d 696c 7920 616e 6420 7468  ne family and th
-0000bd20: 6520 6765 6e65 2074 7265 6520 696e 6665  e gene tree infe
-0000bd30: 7265 6e63 6520 6261 7365 6420 6f6e 2065  rence based on e
-0000bd40: 6163 6820 4d53 412c 2061 6e64 2074 6865  ach MSA, and the
-0000bd50: 6e20 696e 6665 7220 7468 6520 7370 6563  n infer the spec
-0000bd60: 6965 7320 7472 6565 2062 6173 6564 206f  ies tree based o
-0000bd70: 6e20 7468 6573 6520 696e 6469 7669 6475  n these individu
-0000bd80: 616c 2067 656e 6520 7472 6565 732e 2054  al gene trees. T
-0000bd90: 6865 2074 7265 652d 696e 6665 7265 6e63  he tree-inferenc
-0000bda0: 6520 7072 6f67 7261 6d20 616e 6420 7468  e program and th
-0000bdb0: 6520 6173 736f 6369 6174 6564 2070 6172  e associated par
-0000bdc0: 616d 6574 6572 7320 6361 6e20 6265 2073  ameters can be s
-0000bdd0: 6574 206a 7573 7420 6173 2061 626f 7665  et just as above
-0000bde0: 2062 7920 602d 7472 6565 6020 6f72 2060   by `-tree` or `
-0000bdf0: 2d2d 7472 6565 5f6d 6574 686f 6460 2061  --tree_method` a
-0000be00: 6e64 2060 2d74 7360 206f 7220 602d 2d74  nd `-ts` or `--t
-0000be10: 7265 6573 6574 602e 2050 6c65 6173 6520  reeset`. Please 
-0000be20: 616c 736f 206d 616b 6520 7375 7265 2074  also make sure t
-0000be30: 6865 2063 686f 7365 6e20 7472 6565 2d69  he chosen tree-i
-0000be40: 6e66 6572 656e 6365 2070 726f 6772 616d  nference program
-0000be50: 2069 7320 696e 7374 616c 6c65 6420 696e   is installed in
-0000be60: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-0000be70: 2070 6174 682e 2054 6865 2070 726f 6772   path. The progr
-0000be80: 616d 2060 6173 7472 616c 2d70 726f 6020  am `astral-pro` 
-0000be90: 6973 2072 6571 7569 7265 6420 746f 2062  is required to b
-0000bea0: 6520 696e 7374 616c 6c65 6420 696e 2074  e installed in t
-0000beb0: 6865 2065 6e76 6972 6f6e 6d65 6e74 2070  he environment p
-0000bec0: 6174 6820 6966 2074 6865 2063 6f61 6c65  ath if the coale
-0000bed0: 7363 656e 6365 2d62 6173 6564 206d 6574  scence-based met
-0000bee0: 686f 6420 6973 2063 686f 7365 6e2e 2054  hod is chosen. T
-0000bef0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000bf00: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000bf10: 6163 6365 6c65 7261 7465 2074 6865 2073  accelerate the s
-0000bf20: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
-0000bf30: 7420 616e 6420 6765 6e65 2074 7265 6520  t and gene tree 
-0000bf40: 696e 6665 7265 6e63 6520 666f 7220 6561  inference for ea
-0000bf50: 6368 2067 656e 6520 6661 6d69 6c79 2074  ch gene family t
-0000bf60: 6f6f 2061 6e64 2074 6875 7320 7375 6767  oo and thus sugg
-0000bf70: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
-0000bf80: 6173 206d 7563 6820 6173 2074 6865 206e  as much as the n
-0000bf90: 756d 6265 7220 6f66 2067 656e 6520 6661  umber of gene fa
-0000bfa0: 6d69 6c69 6573 2e0a 0a2a 2a41 2073 7567  milies...**A sug
-0000bfb0: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
-0000bfc0: 7275 6e20 6f66 2074 6869 7320 616e 616c  run of this anal
-0000bfd0: 7973 6973 2069 7320 7769 7468 2074 6865  ysis is with the
-0000bfe0: 2073 696d 706c 6520 636f 6d6d 616e 6420   simple command 
-0000bff0: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000c000: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000c010: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000c020: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000c030: 2d63 6f6e 6361 7465 6e61 7469 6f6e 0a77  -concatenation.w
-0000c040: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
-0000c050: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000c060: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000c070: 2d2d 636f 616c 6573 6365 6e63 650a 6060  --coalescence.``
-0000c080: 600a 0a2a 2a54 6865 2066 756e 6374 696f  `..**The functio
-0000c090: 6e61 6c20 616e 6e6f 7461 7469 6f6e 206f  nal annotation o
-0000c0a0: 6620 6765 6e65 2066 616d 696c 6965 732a  f gene families*
-0000c0b0: 2a0a 6060 600a 7767 6420 666f 6375 7320  *.```.wgd focus 
-0000c0c0: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-0000c0d0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000c0e0: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
-0000c0f0: 696f 6e20 6567 676e 6f67 202d 6564 2065  ion eggnog -ed e
-0000c100: 6464 6174 6120 2d2d 646d 6e62 2064 6264  ddata --dmnb dbd
-0000c110: 6174 610a 6060 600a 0a54 6865 2070 726f  ata.```..The pro
-0000c120: 6772 616d 2060 7767 6420 666f 6375 7360  gram `wgd focus`
-0000c130: 2061 6c73 6f20 6164 6465 6420 736f 6d65   also added some
-0000c140: 2077 7261 7070 696e 6720 6675 6e63 7469   wrapping functi
-0000c150: 6f6e 7320 666f 7220 6675 6e63 7469 6f6e  ons for function
-0000c160: 616c 2061 6e6e 6f74 6174 696f 6e20 6f66  al annotation of
-0000c170: 2067 656e 6520 6661 6d69 6c69 6573 206f   gene families o
-0000c180: 6e20 7468 6520 686f 6f64 206f 6620 6461  n the hood of da
-0000c190: 7461 6261 7365 7320 616e 6420 736f 6674  tabases and soft
-0000c1a0: 7761 7265 7320 666f 7220 696e 7374 616e  wares for instan
-0000c1b0: 6365 2060 4567 674e 4f47 6020 616e 6420  ce `EggNOG` and 
-0000c1c0: 6045 6767 4e4f 472d 6d61 7070 6572 602e  `EggNOG-mapper`.
-0000c1d0: 2046 6f72 2074 6865 2061 6e6e 6f74 6174   For the annotat
-0000c1e0: 696f 6e20 7573 696e 6720 6045 6767 4e4f  ion using `EggNO
-0000c1f0: 472d 6d61 7070 6572 602c 2075 7365 7273  G-mapper`, users
-0000c200: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
-0000c210: 2074 6865 2070 6174 6820 746f 2074 6865   the path to the
-0000c220: 2065 6767 4e4f 4720 616e 6e6f 7461 7469   eggNOG annotati
-0000c230: 6f6e 2064 6174 6162 6173 6520 7669 6120  on database via 
-0000c240: 7468 6520 6f70 7469 6f6e 2060 2d65 6460  the option `-ed`
-0000c250: 2c20 7468 6520 7061 7468 2074 6f20 7468  , the path to th
-0000c260: 6520 6469 616d 6f6e 642d 636f 6d70 6174  e diamond-compat
-0000c270: 6962 6c65 2064 6174 6162 6173 6520 7669  ible database vi
-0000c280: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
-0000c290: 646d 6e64 5f64 6260 2061 6e64 2074 6865  dmnd_db` and the
-0000c2a0: 206f 7074 696f 6e20 602d 2d61 6e6e 6f74   option `--annot
-0000c2b0: 6174 696f 6e60 2073 6574 2061 7320 2265  ation` set as "e
-0000c2c0: 6767 6e6f 6722 2e20 5468 6520 6d61 6e6e  ggnog". The mann
-0000c2d0: 6572 2068 6f77 2050 4641 4d20 616e 6e6f  er how PFAM anno
-0000c2e0: 7461 7469 6f6e 2077 696c 6c20 6265 2070  tation will be p
-0000c2f0: 6572 666f 726d 6564 2063 616e 2062 6520  erformed can be 
-0000c300: 636f 6e74 726f 6c6c 6564 2076 6961 2074  controlled via t
-0000c310: 6865 206f 7074 696f 6e20 6070 6661 6d60  he option `pfam`
-0000c320: 2c20 6569 7468 6572 2022 6e6f 6e65 222c  , either "none",
-0000c330: 2022 7265 616c 6967 6e22 206f 7220 2264   "realign" or "d
-0000c340: 656e 6f76 6f22 2c20 7468 6520 6465 7461  enovo", the deta
-0000c350: 696c 6564 2065 7870 6c61 6e61 7469 6f6e  iled explanation
-0000c360: 2063 616e 2062 6520 666f 756e 6420 6174   can be found at
-0000c370: 2074 6865 2077 696b 6920 6f66 205b 4567   the wiki of [Eg
-0000c380: 674e 4f47 2d6d 6170 7065 725d 2868 7474  gNOG-mapper](htt
-0000c390: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-0000c3a0: 6567 676e 6f67 6462 2f65 6767 6e6f 672d  eggnogdb/eggnog-
-0000c3b0: 6d61 7070 6572 2f77 696b 6929 2e20 506c  mapper/wiki). Pl
-0000c3c0: 6561 7365 2070 7265 2d69 6e73 7461 6c6c  ease pre-install
-0000c3d0: 2074 6865 2060 4567 674e 4f47 2d6d 6170   the `EggNOG-map
-0000c3e0: 7065 7260 2070 7974 686f 6e20 7061 636b  per` python pack
-0000c3f0: 6167 6520 6966 2075 7369 6e67 2074 6869  age if using thi
-0000c400: 7320 6675 6e63 7469 6f6e 2e20 466f 7220  s function. For 
-0000c410: 7468 6520 616e 6e6f 7461 7469 6f6e 2075  the annotation u
-0000c420: 7369 6e67 2060 686d 6d73 6361 6e60 2c20  sing `hmmscan`, 
-0000c430: 7768 6174 2069 7320 696d 706c 656d 656e  what is implemen
-0000c440: 7465 6420 696e 2060 7767 6420 7632 6020  ted in `wgd v2` 
-0000c450: 6973 2061 2073 696d 706c 6520 6275 6e64  is a simple bund
-0000c460: 6c65 2066 756e 6374 696f 6e20 746f 2070  le function to p
-0000c470: 6572 666f 726d 2060 686d 6d73 6361 6e60  erform `hmmscan`
-0000c480: 2061 6e61 6c79 7369 7320 666f 7220 6120   analysis for a 
-0000c490: 6769 7665 6e20 686d 6d20 7072 6f66 696c  given hmm profil
-0000c4a0: 6520 616e 6420 7365 7420 6f66 2067 656e  e and set of gen
-0000c4b0: 6520 6661 6d69 6c69 6573 2073 7563 6820  e families such 
-0000c4c0: 7468 6174 2075 7365 7273 206e 6565 6420  that users need 
-0000c4d0: 746f 2073 6574 2074 6865 206f 7074 696f  to set the optio
-0000c4e0: 6e20 602d 2d61 6e6e 6f74 6174 696f 6e60  n `--annotation`
-0000c4f0: 2061 7320 2268 6d6d 7066 616d 2220 616e   as "hmmpfam" an
-0000c500: 6420 7072 6f76 6964 6520 7468 6520 686d  d provide the hm
-0000c510: 6d70 726f 6669 6c65 2076 6961 2074 6865  mprofile via the
-0000c520: 206f 7074 696f 6e20 602d 2d68 6d6d 602e   option `--hmm`.
-0000c530: 2046 6f72 2074 6865 2061 6e6e 6f74 6174   For the annotat
-0000c540: 696f 6e20 7573 696e 6720 6069 6e74 6572  ion using `inter
-0000c550: 7072 6f73 6361 6e60 2c20 7573 6572 7320  proscan`, users 
-0000c560: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-0000c570: 7468 6520 7061 7468 2074 6f20 7468 6520  the path to the 
-0000c580: 696e 7465 7270 726f 7363 616e 2069 6e73  interproscan ins
-0000c590: 7461 6c6c 6174 696f 6e20 666f 6c64 6572  tallation folder
-0000c5a0: 2077 6865 7265 2074 6865 7265 2069 7320   where there is 
-0000c5b0: 6120 6069 6e74 6572 7072 6f73 6361 6e2e  a `interproscan.
-0000c5c0: 7368 6020 6669 6c65 2076 6961 2074 6865  sh` file via the
-0000c5d0: 206f 7074 696f 6e20 602d 2d65 7865 7061   option `--exepa
-0000c5e0: 7468 6020 616e 6420 7365 7420 7468 6520  th` and set the 
-0000c5f0: 7468 6520 6f70 7469 6f6e 2060 2d2d 616e  the option `--an
-0000c600: 6e6f 7461 7469 6f6e 6020 6173 2022 696e  notation` as "in
-0000c610: 7465 7270 726f 7363 616e 222e 2054 6865  terproscan". The
-0000c620: 2070 6172 616c 6c65 6c20 7468 7265 6164   parallel thread
-0000c630: 7320 6865 7265 2061 7265 2074 6f20 7061  s here are to pa
-0000c640: 7261 6c6c 656c 697a 6520 7468 6520 616e  rallelize the an
-0000c650: 616c 7973 6973 2066 6f72 2065 6163 6820  alysis for each 
-0000c660: 6765 6e65 2066 616d 696c 7920 616e 6420  gene family and 
-0000c670: 7468 7573 2073 7567 6765 7374 6564 2074  thus suggested t
-0000c680: 6f20 6265 2073 6574 2061 7320 6d75 6368  o be set as much
-0000c690: 2061 7320 7468 6520 6e75 6d62 6572 206f   as the number o
-0000c6a0: 6620 6765 6e65 2066 616d 696c 6965 732e  f gene families.
-0000c6b0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-0000c6c0: 7374 6172 7469 6e67 2072 756e 206f 6620  starting run of 
-0000c6d0: 7468 6973 2061 6e61 6c79 7369 7320 6361  this analysis ca
-0000c6e0: 6e20 6265 2077 6974 6820 7468 6520 636f  n be with the co
-0000c6f0: 6d6d 616e 6420 6265 6c6f 772e 2a2a 0a0a  mmand below.**..
-0000c700: 6060 600a 7767 6420 666f 6375 7320 6661  ```.wgd focus fa
-0000c710: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
-0000c720: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000c730: 6e63 6533 202d 2d61 6e6e 6f74 6174 696f  nce3 --annotatio
-0000c740: 6e20 6567 676e 6f67 202d 6564 2065 6464  n eggnog -ed edd
-0000c750: 6174 6120 2d2d 646d 6e62 2064 6264 6174  ata --dmnb dbdat
-0000c760: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
-0000c770: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
-0000c780: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000c790: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
-0000c7a0: 686d 6d70 6661 6d20 2d2d 686d 6d20 686d  hmmpfam --hmm hm
-0000c7b0: 6d64 6174 610a 7767 6420 666f 6375 7320  mdata.wgd focus 
-0000c7c0: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-0000c7d0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000c7e0: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
-0000c7f0: 696f 6e20 696e 7465 7270 726f 7363 616e  ion interproscan
-0000c800: 202d 2d65 7865 7061 7468 2024 5041 5448   --exepath $PATH
-0000c810: 0a60 6060 0a0a 2a2a 5468 6520 7068 796c  .```..**The phyl
-0000c820: 6f67 656e 6574 6963 2064 6174 696e 6720  ogenetic dating 
-0000c830: 6f66 2057 4744 732a 2a0a 6060 600a 7767  of WGDs**.```.wg
-0000c840: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000c850: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000c860: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000c870: 6420 6d63 6d63 7472 6565 202d 7370 2073  d mcmctree -sp s
-0000c880: 7064 6174 6120 282d 2d70 726f 7463 6f63  pdata (--protcoc
-0000c890: 6461 7469 6e67 2920 282d 2d70 6172 7469  dating) (--parti
-0000c8a0: 7469 6f6e 2920 282d 2d61 616d 6f64 656c  tion) (--aamodel
-0000c8b0: 206c 6729 2028 2d64 7320 2762 7572 6e69   lg) (-ds 'burni
-0000c8c0: 6e20 3d20 3230 3030 2729 2028 2d64 7320  n = 2000') (-ds 
-0000c8d0: 2773 616d 7066 7265 7120 3d20 3130 3030  'sampfreq = 1000
-0000c8e0: 2729 2028 2d64 7320 276e 7361 6d70 6c65  ') (-ds 'nsample
-0000c8f0: 203d 2032 3030 3030 2729 2020 282d 6e20   = 20000')  (-n 
-0000c900: 3429 2028 2d2d 746f 5f73 746f 7029 2028  4) (--to_stop) (
-0000c910: 2d2d 6364 7329 2028 2d6f 2077 6764 5f66  --cds) (-o wgd_f
-0000c920: 6f63 7573 2920 282d 7420 776f 726b 696e  ocus) (-t workin
-0000c930: 675f 746d 7029 0a60 6060 0a0a 5468 6520  g_tmp).```..The 
-0000c940: 6162 736f 6c75 7465 2064 6174 696e 6720  absolute dating 
-0000c950: 6f66 2057 4744 7320 6973 2061 2073 7065  of WGDs is a spe
-0000c960: 6369 6669 6320 7069 7065 6c69 6e65 2069  cific pipeline i
-0000c970: 6d70 6c65 6d65 6e74 6564 2069 6e20 6077  mplemented in `w
-0000c980: 6764 2076 3260 2075 7369 6e67 2074 6865  gd v2` using the
-0000c990: 206d 6574 686f 6420 6f66 2070 6879 6c6f   method of phylo
-0000c9a0: 6765 6e65 7469 6320 6461 7469 6e67 2e20  genetic dating. 
-0000c9b0: 5468 6520 6661 6d69 6c69 6573 2075 7365  The families use
-0000c9c0: 6420 696e 2074 6869 7320 7374 6570 2063  d in this step c
-0000c9d0: 616e 2062 6520 7072 6f64 7563 6564 2066  an be produced f
-0000c9e0: 726f 6d20 6077 6764 2064 6d64 6020 616e  rom `wgd dmd` an
-0000c9f0: 6420 6077 6764 2070 6561 6b60 2e20 4e6f  d `wgd peak`. No
-0000ca00: 7465 2074 6861 7420 6865 7265 2077 6520  te that here we 
-0000ca10: 6f6e 6c79 2064 6973 6375 7373 2068 6f77  only discuss how
-0000ca20: 2074 6f20 6461 7465 2057 4744 7320 7769   to date WGDs wi
-0000ca30: 7468 2067 656e 6f6d 6520 6173 7365 6d62  th genome assemb
-0000ca40: 6c79 2c20 696e 7374 6561 6420 6f66 2074  ly, instead of t
-0000ca50: 7261 6e73 6372 6970 746f 6d65 2061 7373  ranscriptome ass
-0000ca60: 656d 626c 7920 2877 6869 6368 2077 696c  embly (which wil
-0000ca70: 6c20 6265 2064 6973 6375 7373 6564 2069  l be discussed i
-0000ca80: 6e20 6120 7365 7061 7261 7465 2073 6563  n a separate sec
-0000ca90: 7469 6f6e 2068 6572 6575 6e64 6572 292e  tion hereunder).
-0000caa0: 2054 6865 2061 7373 756d 7074 696f 6e20   The assumption 
-0000cab0: 7765 206d 6164 6520 6865 7265 2069 7320  we made here is 
-0000cac0: 7468 6174 206e 6f74 2061 6c6c 2061 6e63  that not all anc
-0000cad0: 686f 7220 7061 6972 7320 2863 6f6c 6c69  hor pairs (colli
-0000cae0: 6e65 6172 2064 7570 6c69 6361 7465 7329  near duplicates)
-0000caf0: 2061 7265 2073 7569 7461 626c 6520 666f   are suitable fo
-0000cb00: 7220 7068 796c 6f67 656e 6574 6963 2064  r phylogenetic d
-0000cb10: 6174 696e 672c 2066 6f72 2069 6e73 7461  ating, for insta
-0000cb20: 6e63 6520 7468 6f73 6520 6661 7374 6c79  nce those fastly
-0000cb30: 206f 7220 736c 6f77 6c79 2065 766f 6c76   or slowly evolv
-0000cb40: 696e 6720 6765 6e65 2064 7570 6c69 6361  ing gene duplica
-0000cb50: 7465 732c 2062 6563 6175 7365 2074 6865  tes, because the
-0000cb60: 7927 7265 2070 726f 6e65 2074 6f20 6769  y're prone to gi
-0000cb70: 7665 2062 6961 7365 6420 6461 7469 6e67  ve biased dating
-0000cb80: 2065 7374 696d 6174 696f 6e2e 2053 6f20   estimation. So 
-0000cb90: 6173 2074 6f20 7265 7472 6965 7665 2074  as to retrieve t
-0000cba0: 6865 2022 7265 6c69 6162 6c65 2220 616e  he "reliable" an
-0000cbb0: 6368 6f72 2070 6169 7273 2c20 7765 2069  chor pairs, we i
-0000cbc0: 6d70 6c65 6d65 6e74 6564 2073 6f6d 6520  mplemented some 
-0000cbd0: 6d65 7468 6f64 7320 6f66 2069 6465 6e74  methods of ident
-0000cbe0: 6966 7969 6e67 2063 7265 6469 6162 6c65  ifying crediable
-0000cbf0: 2061 6e63 686f 7220 7061 6972 7320 6261   anchor pairs ba
-0000cc00: 7365 6420 6f6e 2074 6865 6972 204b 7320  sed on their Ks 
-0000cc10: 7661 6c75 6573 2061 6e64 2f6f 7220 7265  values and/or re
-0000cc20: 7369 6469 6e67 2063 6f6c 6c69 6e65 6172  siding collinear
-0000cc30: 2073 6567 6d65 6e74 732e 2046 6f72 2061   segments. For a
-0000cc40: 2067 656e 6f6d 6520 7769 7468 2063 6c65   genome with cle
-0000cc50: 6172 2073 6967 6e61 6c73 206f 6620 7075  ar signals of pu
-0000cc60: 7461 7469 7665 2057 4744 732c 2073 7563  tative WGDs, suc
-0000cc70: 6820 6173 2074 6865 202a 4171 7569 6c65  h as the *Aquile
-0000cc80: 6769 6120 636f 6572 756c 6561 2a20 696e  gia coerulea* in
-0000cc90: 2074 6865 2065 7861 6d70 6c65 2c20 6120   the example, a 
-0000cca0: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-0000ccb0: 2074 6861 7420 6170 706c 6965 7320 7468   that applies th
-0000ccc0: 6520 7072 696e 6369 706c 6520 6f66 2068  e principle of h
-0000ccd0: 6f77 205b 6b73 7261 7465 735d 2868 7474  ow [ksrates](htt
-0000cce0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-0000ccf0: 5649 422d 5053 422f 6b73 7261 7465 7329  VIB-PSB/ksrates)
-0000cd00: 2066 696e 6420 7468 6520 696e 6974 6961   find the initia
-0000cd10: 6c20 7065 616b 7320 616e 6420 7468 6569  l peaks and thei
-0000cd20: 7220 7061 7261 6d65 7465 7273 2077 6173  r parameters was
-0000cd30: 2069 6d70 6c65 6d65 6e74 6564 2074 6f20   implemented to 
-0000cd40: 6669 6e64 2074 6865 2039 3525 2063 6f6e  find the 95% con
-0000cd50: 6669 6465 6e63 6520 6c65 7665 6c20 6f66  fidence level of
-0000cd60: 2074 6865 2061 7373 756d 6564 206c 6f67   the assumed log
-0000cd70: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-0000cd80: 696f 6e20 6f66 2074 6865 2061 6e63 686f  ion of the ancho
-0000cd90: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-0000cda0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000cdb0: 6962 7574 696f 6e20 746f 2066 696c 7465  ibution to filte
-0000cdc0: 7220 616e 6368 6f72 2070 6169 7273 2077  r anchor pairs w
-0000cdd0: 6974 6820 746f 6f20 6869 6768 206f 7220  ith too high or 
-0000cde0: 6c6f 7720 2a4b 2a3c 7375 623e 533c 2f73  low *K*<sub>S</s
-0000cdf0: 7562 3e20 6167 6573 2c20 7468 6520 6578  ub> ages, the ex
-0000ce00: 616d 706c 6172 2063 6f6d 6d61 6e64 206f  amplar command o
-0000ce10: 6620 7768 6963 6820 6973 2073 686f 7765  f which is showe
-0000ce20: 6420 696e 2074 6865 2060 7767 6420 7065  d in the `wgd pe
-0000ce30: 616b 6020 7365 6374 696f 6e20 6f66 205b  ak` section of [
-0000ce40: 496c 6c75 7374 7261 7469 6f6e 5d28 6874  Illustration](ht
-0000ce50: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000ce60: 2f68 6563 6865 2d70 7362 2f77 6764 3f74  /heche-psb/wgd?t
-0000ce70: 6162 3d72 6561 646d 652d 6f76 2d66 696c  ab=readme-ov-fil
-0000ce80: 6523 696c 6c75 7374 7261 7469 6f6e 292e  e#illustration).
-0000ce90: 2049 6620 7468 6973 2068 6575 7269 7374   If this heurist
-0000cea0: 6963 206d 6574 686f 6420 6661 696c 6564  ic method failed
-0000ceb0: 2074 6f20 6769 7665 2072 6561 736f 6e61   to give reasona
-0000cec0: 626c 6520 7265 7375 6c74 732c 2077 6869  ble results, whi
-0000ced0: 6368 2069 7320 7573 7561 6c6c 7920 6475  ch is usually du
-0000cee0: 6520 746f 2074 6865 2065 6666 6563 7420  e to the effect 
-0000cef0: 6f66 206d 756c 7469 706c 6520 6164 6a61  of multiple adja
-0000cf00: 6365 6e74 2057 4744 7320 7468 6174 2062  cent WGDs that b
-0000cf10: 6c75 7273 2074 6865 2070 6561 6b2d 6669  lurs the peak-fi
-0000cf20: 6e64 696e 6720 7072 6f63 6573 732c 2075  nding process, u
-0000cf30: 7365 7273 2063 616e 2074 7572 6e20 746f  sers can turn to
-0000cf40: 2074 6865 2063 6f6c 6c69 6e65 6172 2073   the collinear s
-0000cf50: 6567 6d65 6e74 732d 6775 6964 6564 2061  egments-guided a
-0000cf60: 6e63 686f 7220 7061 6972 2063 6c75 7374  nchor pair clust
-0000cf70: 6572 696e 6720 696d 706c 656d 656e 7465  ering implemente
-0000cf80: 6420 696e 2060 7767 6420 7632 602c 2069  d in `wgd v2`, i
-0000cf90: 6e20 7768 6963 6820 6120 636f 6c6c 696e  n which a collin
-0000cfa0: 6561 7220 7365 676d 656e 742d 7769 7365  ear segment-wise
-0000cfb0: 2047 4d4d 2063 6c75 7374 6572 696e 6720   GMM clustering 
-0000cfc0: 7769 6c6c 2062 6520 6669 7273 7420 636f  will be first co
-0000cfd0: 6e64 7563 7465 6420 6261 7365 6420 6f6e  nducted based on
-0000cfe0: 2074 6865 2022 736f 2d63 616c 6c65 6422   the "so-called"
-0000cff0: 2073 6567 6d65 6e74 202a 4b2a 3c73 7562   segment *K*<sub
-0000d000: 3e53 3c2f 7375 623e 2061 6765 2072 6570  >S</sub> age rep
-0000d010: 7265 7365 6e74 6564 2062 7920 7468 6520  resented by the 
-0000d020: 6d65 6469 616e 202a 4b2a 3c73 7562 3e53  median *K*<sub>S
-0000d030: 3c2f 7375 623e 2061 6765 206f 6620 616c  </sub> age of al
-0000d040: 6c20 7468 6520 7265 7369 6469 6e67 2073  l the residing s
-0000d050: 796e 7465 6c6f 6773 2028 6e6f 7465 2074  yntelogs (note t
-0000d060: 6861 7420 7468 6520 6765 6e65 2064 7570  hat the gene dup
-0000d070: 6c69 6361 7465 2070 6169 7273 2061 646f  licate pairs ado
-0000d080: 7074 6564 2069 6e20 7468 6973 2073 7465  pted in this ste
-0000d090: 7020 6973 2066 726f 6d20 7468 6520 6669  p is from the fi
-0000d0a0: 6c65 2060 6d75 6c74 6970 6c69 636f 6e5f  le `multiplicon_
-0000d0b0: 7061 6972 732e 7478 7460 2077 6869 6368  pairs.txt` which
-0000d0c0: 2063 6f6e 7461 696e 7320 7468 6520 6675   contains the fu
-0000d0d0: 6c6c 2073 6574 206f 6620 7379 6e74 656c  ll set of syntel
-0000d0e0: 6f67 7329 2c20 696e 7374 6561 6420 6f66  ogs), instead of
-0000d0f0: 2074 6865 2073 6d61 6c6c 6572 2067 656e   the smaller gen
-0000d100: 6520 7365 7420 6f66 2061 6e63 686f 7220  e set of anchor 
-0000d110: 7061 6972 732e 2054 6865 2064 6973 7469  pairs. The disti
-0000d120: 6e63 7469 6f6e 2062 6574 7765 656e 2061  nction between a
-0000d130: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
-0000d140: 7379 6e74 656c 6f67 7320 6973 2074 6861  syntelogs is tha
-0000d150: 7420 7468 6520 6c61 7474 6572 2072 6566  t the latter ref
-0000d160: 6572 7320 746f 206d 756c 7469 706c 6520  ers to multiple 
-0000d170: 7365 7473 206f 6620 6765 6e65 7320 6465  sets of genes de
-0000d180: 7269 7665 6420 6672 6f6d 2074 6865 2073  rived from the s
-0000d190: 616d 6520 616e 6365 7374 7261 6c20 6765  ame ancestral ge
-0000d1a0: 6e6f 6d69 6320 7265 6769 6f6e 2077 6869  nomic region whi
-0000d1b0: 6c65 2074 6865 2066 6f72 6d65 7220 696d  le the former im
-0000d1c0: 706c 6965 7320 7468 6520 6c61 7474 6572  plies the latter
-0000d1d0: 2062 7574 2072 6571 7569 7265 7320 6164   but requires ad
-0000d1e0: 6469 7469 6f6e 616c 6c79 2074 6865 2063  ditionally the c
-0000d1f0: 6f6e 7365 7276 6564 2067 656e 6520 6f72  onserved gene or
-0000d200: 6465 722c 2062 6f74 6820 6f66 2077 6869  der, both of whi
-0000d210: 6368 2061 7265 2c20 7769 7468 696e 2061  ch are, within a
-0000d220: 2067 656e 6f6d 652c 2061 7373 756d 6564   genome, assumed
-0000d230: 2074 6f20 6265 206f 7269 6769 6e61 7465   to be originate
-0000d240: 6420 6672 6f6d 2074 6865 2064 7570 6c69  d from the dupli
-0000d250: 6361 7469 6f6e 206f 6620 6120 636f 6d6d  cation of a comm
-0000d260: 6f6e 2061 6e63 6573 7472 616c 2067 656e  on ancestral gen
-0000d270: 6f6d 6963 2072 6567 696f 6e20 616e 6420  omic region and 
-0000d280: 6173 2073 7563 6820 6465 656d 6564 2065  as such deemed e
-0000d290: 7669 6465 6e63 6520 666f 7220 5747 442e  vidence for WGD.
-0000d2a0: 2054 6865 6e20 7468 6520 7379 6e74 656c   Then the syntel
-0000d2b0: 6f67 7320 7769 6c6c 2062 6520 6d61 7070  ogs will be mapp
-0000d2c0: 6564 2062 6163 6b20 6163 636f 7264 696e  ed back accordin
-0000d2d0: 6720 746f 2074 6865 2063 6c75 7374 6572  g to the cluster
-0000d2e0: 696e 6720 7265 7375 6c74 7320 6f66 2074  ing results of t
-0000d2f0: 6865 6972 2061 6666 696c 6961 7465 6420  heir affiliated 
-0000d300: 7365 676d 656e 7473 2e20 5369 6e63 6520  segments. Since 
-0000d310: 7468 6520 4761 7573 7369 616e 2073 6861  the Gaussian sha
-0000d320: 7065 206f 6620 7468 6520 7365 676d 656e  pe of the segmen
-0000d330: 7420 636c 7573 7465 7220 646f 6573 6e27  t cluster doesn'
-0000d340: 7420 6e65 6365 7373 6172 696c 7920 696d  t necessarily im
-0000d350: 6974 6174 6520 7468 6520 7368 6170 6520  itate the shape 
-0000d360: 6f66 2074 6865 2072 6573 6964 696e 6720  of the residing 
-0000d370: 7379 6e74 656c 6f67 732c 2073 6f20 6173  syntelogs, so as
-0000d380: 2074 6f20 7265 7472 6965 7665 2074 6865   to retrieve the
-0000d390: 2022 7265 6c69 6162 6c65 2220 6765 6e65   "reliable" gene
-0000d3a0: 2073 6574 7320 666f 7220 7068 796c 6f67   sets for phylog
-0000d3b0: 656e 6574 6963 2064 6174 696e 672c 2077  enetic dating, w
-0000d3c0: 6520 6164 706f 7465 6420 7468 6520 2839  e adpoted the (9
-0000d3d0: 3525 2920 6869 6768 6573 7420 6465 6e73  5%) highest dens
-0000d3e0: 6974 7920 7265 6769 6f6e 2028 4844 5229  ity region (HDR)
-0000d3f0: 206f 6620 7468 6520 7379 6e74 656c 6f67   of the syntelog
-0000d400: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d410: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000d420: 6e20 666f 7220 7468 6520 7068 796c 6f67  n for the phylog
-0000d430: 656e 6574 6963 2064 6174 696e 672c 2074  enetic dating, t
-0000d440: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-0000d450: 6620 7468 6520 2839 3525 2920 4844 5220  f the (95%) HDR 
-0000d460: 696e 2074 6865 2066 756e 6374 696f 6e20  in the function 
-0000d470: 5b63 616c 6375 6c61 7465 4850 445d 2868  [calculateHPD](h
-0000d480: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000d490: 6d2f 6865 6368 652d 7073 622f 7767 642f  m/heche-psb/wgd/
-0000d4a0: 626c 6f62 2f70 6879 6c6f 6461 7469 6e67  blob/phylodating
-0000d4b0: 2f77 6764 2f70 6561 6b2e 7079 234c 3137  /wgd/peak.py#L17
-0000d4c0: 3537 2920 7365 656b 7320 7468 6520 7368  57) seeks the sh
-0000d4d0: 6f72 7465 7374 202a 4b2a 3c73 7562 3e53  ortest *K*<sub>S
-0000d4e0: 3c2f 7375 623e 2072 616e 6765 2028 612c  </sub> range (a,
-0000d4f0: 6229 2077 6869 6368 2073 6174 6973 6669  b) which satisfi
-0000d500: 6573 2074 6865 2072 6571 7569 7265 6d65  es the requireme
-0000d510: 6e74 206f 6620 7370 616e 6e69 6e67 206d  nt of spanning m
-0000d520: 6f72 6520 7468 616e 2028 3935 2529 206f  ore than (95%) o
-0000d530: 6620 616c 6c20 7468 6520 2a4b 2a3c 7375  f all the *K*<su
-0000d540: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-0000d550: 2e20 4f6e 2074 6865 2070 7265 6d69 7365  . On the premise
-0000d560: 206f 6620 6964 656e 7469 6669 6564 2061   of identified a
-0000d570: 6e63 686f 7220 7061 6972 7320 286e 6f74  nchor pairs (not
-0000d580: 6520 7468 6174 2074 6865 2073 796e 7465  e that the synte
-0000d590: 6c6f 6773 2061 7265 2061 6c73 6f20 7265  logs are also re
-0000d5a0: 6665 7272 6564 2061 7320 616e 6368 6f72  ferred as anchor
-0000d5b0: 2070 6169 7273 2068 6572 6561 6674 6572   pairs hereafter
-0000d5c0: 292c 2077 6520 696d 706c 656d 656e 7465  ), we implemente
-0000d5d0: 6420 696e 2074 6865 2070 726f 6772 616d  d in the program
-0000d5e0: 2060 7767 6420 646d 6460 2074 6865 2073   `wgd dmd` the s
-0000d5f0: 6f2d 6361 6c6c 6564 2061 6e63 686f 722d  o-called anchor-
-0000d600: 6177 6172 6520 6c6f 6361 6c20 4d52 4248  aware local MRBH
-0000d610: 7320 6f72 206f 7274 686f 6772 6f75 7073  s or orthogroups
-0000d620: 2c20 696e 2077 6869 6368 2074 6865 206f  , in which the o
-0000d630: 7269 6769 6e61 6c20 6c6f 6361 6c20 4d52  riginal local MR
-0000d640: 4248 7320 6172 6520 6675 7274 6865 7220  BHs are further 
-0000d650: 6d65 7267 6564 2077 6974 6820 616e 6368  merged with anch
-0000d660: 6f72 2070 6169 7273 2073 7563 6820 7468  or pairs such th
-0000d670: 6174 2065 6163 6820 6f72 7468 6f67 726f  at each orthogro
-0000d680: 7570 2063 6f6e 7461 696e 7320 7468 6520  up contains the 
-0000d690: 616e 6368 6f72 2070 6169 7220 616e 6420  anchor pair and 
-0000d6a0: 7468 6520 6f72 7468 6f6c 6f67 7565 732e  the orthologues.
-0000d6b0: 2057 6974 6820 7468 6973 206f 7274 686f   With this ortho
-0000d6c0: 6772 6f75 702c 2075 7365 7273 2074 6865  group, users the
-0000d6d0: 6e20 6e65 6564 206f 6e65 2073 7461 7274  n need one start
-0000d6e0: 696e 6720 7472 6565 2066 696c 6520 2861  ing tree file (a
-0000d6f0: 7320 7368 6f77 6e20 696e 2074 6865 205b  s shown in the [
-0000d700: 496c 6c75 7374 7261 7469 6f6e 5d28 6874  Illustration](ht
-0000d710: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000d720: 2f68 6563 6865 2d70 7362 2f77 6764 2f74  /heche-psb/wgd/t
-0000d730: 7265 652f 7068 796c 6f64 6174 696e 673f  ree/phylodating?
-0000d740: 7461 623d 7265 6164 6d65 2d6f 762d 6669  tab=readme-ov-fi
-0000d750: 6c65 2369 6c6c 7573 7472 6174 696f 6e29  le#illustration)
-0000d760: 2070 6172 7429 2069 6e64 6963 6174 696e   part) indicatin
-0000d770: 6720 7468 6520 7472 6565 2074 6f70 6f6c  g the tree topol
-0000d780: 6f67 7920 616e 6420 666f 7373 696c 2063  ogy and fossil c
-0000d790: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-0000d7a0: 6d61 7469 6f6e 2066 6f72 2074 6865 2066  mation for the f
-0000d7b0: 696e 616c 2070 6879 6c6f 6765 6e65 7469  inal phylogeneti
-0000d7c0: 6320 6461 7469 6e67 2075 7369 6e67 2074  c dating using t
-0000d7d0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000d7e0: 666f 6375 7360 2e20 5573 6572 7320 6361  focus`. Users ca
-0000d7f0: 6e20 7365 7420 7468 6520 666c 6167 206f  n set the flag o
-0000d800: 7074 696f 6e20 602d 2d70 726f 7463 6f63  ption `--protcoc
-0000d810: 6461 7469 6e67 6020 746f 206f 6e6c 7920  dating` to only 
-0000d820: 636f 6e64 7563 7420 7468 6520 6461 7469  conduct the dati
-0000d830: 6e67 206f 6620 7468 6520 636f 6e63 6174  ng of the concat
-0000d840: 656e 6174 696f 6e20 7072 6f74 6569 6e20  enation protein 
-0000d850: 4d53 4120 6f72 2074 6865 2066 6c61 6720  MSA or the flag 
-0000d860: 6f70 7469 6f6e 2060 2d2d 7072 6f74 6461  option `--protda
-0000d870: 7469 6e67 6020 746f 206f 6e6c 7920 636f  ting` to only co
-0000d880: 6e64 7563 7420 7468 6520 6461 7469 6e67  nduct the dating
-0000d890: 206f 6620 7468 6520 7072 6f74 6569 6e20   of the protein 
-0000d8a0: 4d53 4173 2c20 6e6f 7465 6420 7468 6174  MSAs, noted that
-0000d8b0: 2074 6865 7365 2074 776f 206f 7074 696f   these two optio
-0000d8c0: 6e73 206f 6e6c 7920 776f 726b 2066 6f72  ns only work for
-0000d8d0: 2074 6865 2060 6d63 6d63 7472 6565 6020   the `mcmctree` 
-0000d8e0: 6f70 7469 6f6e 2073 6f20 6661 722e 2054  option so far. T
-0000d8f0: 6865 2066 6c61 6720 6f70 7469 6f6e 2060  he flag option `
-0000d900: 2d2d 7061 7274 6974 696f 6e60 2063 616e  --partition` can
-0000d910: 2062 6520 7365 7420 746f 2070 6572 666f   be set to perfo
-0000d920: 726d 2060 6d63 6d63 7472 6565 6020 616e  rm `mcmctree` an
-0000d930: 616c 7973 6973 2075 7369 6e67 2074 6865  alysis using the
-0000d940: 2070 6172 7469 7469 6f6e 6564 2064 6174   partitioned dat
-0000d950: 6120 2869 2e65 2e2c 2031 7374 2c20 326e  a (i.e., 1st, 2n
-0000d960: 6420 616e 6420 3372 6420 706f 7369 7469  d and 3rd positi
-0000d970: 6f6e 206f 6620 636f 646f 6e29 2069 6e73  on of codon) ins
-0000d980: 7465 6164 206f 6620 7573 696e 6720 7468  tead of using th
-0000d990: 6520 636f 646f 6e20 6173 2061 2077 686f  e codon as a who
-0000d9a0: 6c65 2e20 5468 6520 6f70 7469 6f6e 2060  le. The option `
-0000d9b0: 2d2d 6161 6d6f 6465 6c60 2063 616e 2062  --aamodel` can b
-0000d9c0: 6520 7365 7420 746f 2064 6574 6572 6d69  e set to determi
-0000d9d0: 6e65 2074 6865 2061 6d69 6e6f 2061 6369  ne the amino aci
-0000d9e0: 6420 6d6f 6465 6c20 6170 706c 6965 6420  d model applied 
-0000d9f0: 696e 2060 6d63 6d63 7472 6565 6020 616e  in `mcmctree` an
-0000da00: 616c 7973 6973 2e20 5468 6520 6f70 7469  alysis. The opti
-0000da10: 6f6e 2060 2d64 7360 2063 616e 2062 6520  on `-ds` can be 
-0000da20: 7573 6564 2074 6f20 7365 7420 7061 7261  used to set para
-0000da30: 6d65 7465 7273 2066 6f72 2074 6865 206d  meters for the m
-0000da40: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
-0000da50: 7072 6f67 7261 6d2e 2054 6865 2070 6172  program. The par
-0000da60: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000da70: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000da80: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000da90: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000daa0: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000dab0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000dac0: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000dad0: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000dae0: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000daf0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000db00: 7469 6e67 2072 756e 2063 616e 2062 6520  ting run can be 
-0000db10: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
-0000db20: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
-0000db30: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
-0000db40: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000db50: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000db60: 2d64 206d 636d 6374 7265 6520 2d73 7020  -d mcmctree -sp 
-0000db70: 7370 6461 7461 0a77 6764 2066 6f63 7573  spdata.wgd focus
-0000db80: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000db90: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000dba0: 7175 656e 6365 3320 2d64 2072 3873 202d  quence3 -d r8s -
-0000dbb0: 7370 2073 7064 6174 6120 2d2d 6e73 6974  sp spdata --nsit
-0000dbc0: 6573 206e 7369 7465 696e 666f 0a77 6764  es nsiteinfo.wgd
-0000dbd0: 2066 6f63 7573 2066 616d 696c 6965 7320   focus families 
-0000dbe0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
-0000dbf0: 6365 3220 7365 7175 656e 6365 3320 2d64  ce2 sequence3 -d
-0000dc00: 2062 6561 7374 202d 2d66 6f73 7369 6c20   beast --fossil 
-0000dc10: 666f 7373 696c 696e 666f 202d 2d72 6f6f  fossilinfo --roo
-0000dc20: 7468 6569 6768 7420 726f 6f74 6865 6967  theight rootheig
-0000dc30: 6874 696e 666f 202d 2d63 6861 696e 7365  htinfo --chainse
-0000dc40: 7420 6368 6169 6e73 6574 696e 666f 202d  t chainsetinfo -
-0000dc50: 2d62 6561 7374 6c67 6a61 7220 2450 4154  -beastlgjar $PAT
-0000dc60: 480a 6060 600a 0a23 2323 2077 6764 206b  H.```..### wgd k
-0000dc70: 7364 0a0a 2a2a 5468 6520 636f 6e73 7472  sd..**The constr
-0000dc80: 7563 7469 6f6e 206f 6620 7768 6f6c 6520  uction of whole 
-0000dc90: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-0000dca0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000dcb0: 7472 6962 7574 696f 6e2a 2a0a 6060 600a  tribution**.```.
-0000dcc0: 7767 6420 6b73 6420 6661 6d69 6c69 6573  wgd ksd families
-0000dcd0: 2073 6571 7565 6e63 6520 282d 6f20 7767   sequence (-o wg
-0000dce0: 645f 6b73 6420 2d74 2077 6764 5f6b 7364  d_ksd -t wgd_ksd
-0000dcf0: 5f74 6d70 202d 2d6e 7468 7265 6164 7320  _tmp --nthreads 
-0000dd00: 3420 2d2d 746f 5f73 746f 7020 2d2d 6364  4 --to_stop --cd
-0000dd10: 7320 2d2d 7061 6972 7769 7365 202d 2d73  s --pairwise --s
-0000dd20: 7472 6970 5f67 6170 7320 2d2d 616c 6967  trip_gaps --alig
-0000dd30: 6e65 7220 6d61 6666 7420 2d2d 7472 6565  ner mafft --tree
-0000dd40: 5f6d 6574 686f 6420 6661 7374 7472 6565  _method fasttree
-0000dd50: 202d 2d6e 6f64 655f 6176 6572 6167 6529   --node_average)
-0000dd60: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
-0000dd70: 6d20 6077 6764 206b 7364 602c 2061 7320  m `wgd ksd`, as 
-0000dd80: 696d 7069 6564 2062 7920 6974 7320 6e61  impied by its na
-0000dd90: 6d65 2c20 6973 2066 6f72 2074 6865 2063  me, is for the c
-0000dda0: 6f6e 7374 7275 6374 696f 6e20 6f66 202a  onstruction of *
-0000ddb0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000ddc0: 6765 2064 6973 7472 6962 7574 696f 6e2e  ge distribution.
-0000ddd0: 2045 7863 6570 7420 666f 7220 7468 6520   Except for the 
-0000dde0: 6166 6f72 656d 656e 7469 6f6e 6564 2070  aforementioned p
-0000ddf0: 6172 616d 6574 6572 7320 7375 6368 2061  arameters such a
-0000de00: 7320 602d 2d74 6f5f 7374 6f70 6020 616e  s `--to_stop` an
-0000de10: 6420 602d 2d63 6473 602c 2074 6865 7265  d `--cds`, there
-0000de20: 2061 7265 2073 6f6d 6520 696d 706f 7274   are some import
-0000de30: 616e 7420 7061 7261 6d65 7465 7273 2074  ant parameters t
-0000de40: 6861 7420 6861 7665 2063 7275 6369 616c  hat have crucial
-0000de50: 2069 6d70 6163 7420 6f6e 2074 6865 202a   impact on the *
-0000de60: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
-0000de70: 7374 696d 6174 696f 6e2e 2054 6865 206f  stimation. The o
-0000de80: 7074 696f 6e20 602d 2d70 6169 7277 6973  ption `--pairwis
-0000de90: 6560 2069 7320 6120 7665 7279 2069 6d70  e` is a very imp
-0000dea0: 6f72 7461 6e74 2070 6172 616d 6574 6572  ortant parameter
-0000deb0: 2066 6f72 2074 6865 202a 4b2a 3c73 7562   for the *K*<sub
-0000dec0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-0000ded0: 696f 6e2c 2077 6974 6820 7768 6963 6820  ion, with which 
-0000dee0: 7468 6520 6043 4f44 454d 4c60 2077 696c  the `CODEML` wil
-0000def0: 6c20 6361 6c63 756c 6174 6520 7468 6520  l calculate the 
-0000df00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000df10: 666f 7220 6561 6368 2067 656e 6520 7061  for each gene pa
-0000df20: 6972 2073 6570 6172 6174 656c 7920 6261  ir separately ba
-0000df30: 7365 6420 6f6e 2074 6865 2061 6c69 676e  sed on the align
-0000df40: 6d65 6e74 206f 6620 6f6e 6c79 2074 6865  ment of only the
-0000df50: 7365 2074 776f 2067 656e 6573 2069 6e73  se two genes ins
-0000df60: 7465 6164 206f 6620 7468 6520 7768 6f6c  tead of the whol
-0000df70: 6520 616c 6967 6e6d 656e 7420 6f66 2074  e alignment of t
-0000df80: 6865 2066 616d 696c 792c 2073 7563 6820  he family, such 
-0000df90: 7468 6174 206c 6573 7320 6761 7073 2061  that less gaps a
-0000dfa0: 7265 2065 7870 6563 7465 6420 616e 6420  re expected and 
-0000dfb0: 7468 7573 2074 6865 2061 6c69 676e 6d65  thus the alignme
-0000dfc0: 6e74 2069 6e20 7468 6520 636f 6e73 6964  nt in the consid
-0000dfd0: 6572 6174 696f 6e20 6f66 2060 434f 4445  eration of `CODE
-0000dfe0: 4d4c 6020 7769 6c6c 2062 6520 6c6f 6e67  ML` will be long
-0000dff0: 6572 2028 6265 6361 7573 6520 6043 4f44  er (because `COD
-0000e000: 454d 4c60 2077 696c 6c20 6175 746f 6d61  EML` will automa
-0000e010: 7465 6c79 2073 6b69 7020 6576 6572 7920  tely skip every 
-0000e020: 636f 6c75 6d6e 2077 6974 6820 6761 702c  column with gap,
-0000e030: 2072 6567 6172 646c 6573 7320 6f66 2077   regardless of w
-0000e040: 6865 7468 6572 2069 7427 7320 616e 206f  hether it's an o
-0000e050: 7665 7261 6c6c 206f 7220 7061 7274 6961  verall or partia
-0000e060: 6c20 6761 7029 2c20 7769 7468 6f75 7420  l gap), without 
-0000e070: 7768 6963 6820 7468 6520 6043 4f44 454d  which the `CODEM
-0000e080: 4c60 2077 696c 6c20 6361 6c63 756c 6174  L` will calculat
-0000e090: 6520 7468 6520 2a4b 2a3c 7375 623e 533c  e the *K*<sub>S<
-0000e0a0: 2f73 7562 3e20 6261 7365 6420 6f6e 2074  /sub> based on t
-0000e0b0: 6865 2077 686f 6c65 2061 6c69 676e 6d65  he whole alignme
-0000e0c0: 6e74 206f 6620 7468 6520 6661 6d69 6c79  nt of the family
-0000e0d0: 2c20 7768 6963 6820 6d69 6768 7420 6861  , which might ha
-0000e0e0: 7665 206e 6f20 2a4b 2a3c 7375 623e 533c  ve no *K*<sub>S<
-0000e0f0: 2f73 7562 3e20 7265 7375 6c74 2061 7420  /sub> result at 
-0000e100: 616c 6c20 6966 2074 6865 2073 7472 6970  all if the strip
-0000e110: 7065 6420 616c 6967 6e6d 656e 7420 6c65  ped alignment le
-0000e120: 6e67 7468 2028 7265 6d6f 7669 6e67 2061  ngth (removing a
-0000e130: 6c6c 2067 6170 2d63 6f6e 7461 696e 696e  ll gap-containin
-0000e140: 6720 636f 6c75 6d6e 7329 2077 6173 207a  g columns) was z
-0000e150: 6572 6f2c 2061 2063 6175 7365 206f 6620  ero, a cause of 
-0000e160: 6469 6666 6572 656e 7420 6e75 6d62 6572  different number
-0000e170: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
-0000e180: 7562 3e20 6573 7469 6d61 7465 7320 6265  ub> estimates be
-0000e190: 7477 6565 6e20 2270 6169 7277 6973 6520  tween "pairwise 
-0000e1a0: 6d6f 6465 2220 616e 6420 226e 6f6e 2d70  mode" and "non-p
-0000e1b0: 6169 7277 6973 6520 6d6f 6465 222e 2049  airwise mode". I
-0000e1c0: 7427 7320 6469 6666 6963 756c 7420 746f  t's difficult to
-0000e1d0: 2073 6179 2077 6869 6368 206d 6f64 6520   say which mode 
-0000e1e0: 6973 206d 6f72 6520 6964 6561 6c2c 2061  is more ideal, a
-0000e1f0: 6c74 686f 7567 6820 7468 6520 226e 6f6e  lthough the "non
-0000e200: 2d70 6169 7277 6973 6520 6d6f 6465 2220  -pairwise mode" 
-0000e210: 2864 6566 6175 6c74 2073 6574 7469 6e67  (default setting
-0000e220: 2920 7768 6963 6820 7275 6e73 206f 6e20  ) which runs on 
-0000e230: 7468 6520 7768 6f6c 6520 616c 6967 6e6d  the whole alignm
-0000e240: 656e 7420 696e 7374 6561 6420 6f66 2061  ent instead of a
-0000e250: 206c 6f63 616c 2061 6c69 676e 6d65 6e74   local alignment
-0000e260: 2c20 6d69 6768 7420 6265 206d 6f72 6520  , might be more 
-0000e270: 6269 6f6c 6f67 6963 616c 6c79 2063 6f6e  biologically con
-0000e280: 7365 7276 6564 2069 6e20 7468 6174 2069  served in that i
-0000e290: 7420 6173 7375 7265 7320 7468 6520 6576  t assures the ev
-0000e2a0: 6f6c 7574 696f 6e20 6f66 2065 6163 6820  olution of each 
-0000e2b0: 636f 6c75 6d6e 2074 6f20 6265 2073 7461  column to be sta
-0000e2c0: 7274 6564 2066 726f 6d20 7468 6520 726f  rted from the ro
-0000e2d0: 6f74 206f 6620 7468 6520 6661 6d69 6c79  ot of the family
-0000e2e0: 2061 6e64 2061 6c6c 2074 6865 2067 656e   and all the gen
-0000e2f0: 6520 6475 706c 6963 6174 6573 2061 7265  e duplicates are
-0000e300: 2074 616b 656e 2069 6e74 6f20 6163 636f   taken into acco
-0000e310: 756e 7420 696e 2074 6865 202a 4b2a 3c73  unt in the *K*<s
-0000e320: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
-0000e330: 6174 696f 6e20 7072 6f63 6573 732e 2054  ation process. T
-0000e340: 6865 206f 7074 696f 6e20 602d 2d73 7472  he option `--str
-0000e350: 6970 5f67 6170 7360 2063 616e 2072 656d  ip_gaps` can rem
-0000e360: 6f76 6520 616c 6c20 7468 6520 6761 702d  ove all the gap-
-0000e370: 636f 6e74 6169 6e69 6e67 2063 6f6c 756d  containing colum
-0000e380: 6e73 2c20 7769 7468 206f 7220 7769 7468  ns, with or with
-0000e390: 6f75 7420 7768 6963 6820 7468 6520 7265  out which the re
-0000e3a0: 7375 6c74 206f 6620 226e 6f6e 2d70 6169  sult of "non-pai
-0000e3b0: 7277 6973 6520 6d6f 6465 2220 776f 6e27  rwise mode" won'
-0000e3c0: 7420 6265 2061 6666 6563 7465 642c 2077  t be affected, w
-0000e3d0: 6869 6c65 2077 6974 6820 7768 6963 6820  hile with which 
-0000e3e0: 7468 6520 7265 7375 6c74 206f 6620 2270  the result of "p
-0000e3f0: 6169 7277 6973 6520 6d6f 6465 2220 7769  airwise mode" wi
-0000e400: 6c6c 2062 6520 616c 7465 7265 642e 2054  ll be altered. T
-0000e410: 6865 206f 7074 696f 6e20 602d 2d61 6c69  he option `--ali
-0000e420: 676e 6572 6020 616e 6420 602d 2d61 6c6e  gner` and `--aln
-0000e430: 5f6f 7074 696f 6e73 6020 7768 6963 6820  _options` which 
-0000e440: 6465 6369 6465 2077 6869 6368 2061 6c69  decide which ali
-0000e450: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
-0000e460: 6f20 6265 2075 7365 6420 616e 6420 7768  o be used and wh
-0000e470: 6963 6820 7061 7261 6d65 7465 7220 746f  ich parameter to
-0000e480: 2062 6520 7365 742c 2077 696c 6c20 6861   be set, will ha
-0000e490: 7665 2069 6d70 6163 7420 6f6e 2074 6865  ve impact on the
-0000e4a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000e4b0: 2072 6573 756c 7473 2c20 6e6f 7465 6420   results, noted 
-0000e4c0: 7468 6174 2074 6865 2064 6566 6175 6c74  that the default
-0000e4d0: 2070 726f 6772 616d 2069 7320 606d 6166   program is `maf
-0000e4e0: 6674 6020 616e 6420 7468 6520 7061 7261  ft` and the para
-0000e4f0: 6d65 7465 7220 6973 2060 2d2d 6175 746f  meter is `--auto
-0000e500: 602e 2054 6865 206f 7074 696f 6e20 602d  `. The option `-
-0000e510: 2d74 7265 655f 6d65 7468 6f64 6020 616e  -tree_method` an
-0000e520: 6420 602d 2d74 7265 655f 6f70 7469 6f6e  d `--tree_option
-0000e530: 7360 2064 6563 6964 6520 7768 6963 6820  s` decide which 
-0000e540: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000e550: 6e63 6520 7072 6f67 7261 6d20 746f 2062  nce program to b
-0000e560: 6520 7573 6564 2061 6e64 2077 6869 6368  e used and which
-0000e570: 2070 6172 616d 6574 6572 2074 6f20 6265   parameter to be
-0000e580: 2073 6574 2c20 776f 6e27 7420 6166 6665   set, won't affe
-0000e590: 6374 2074 6865 202a 4b2a 3c73 7562 3e53  ct the *K*<sub>S
-0000e5a0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
-0000e5b0: 6e20 6974 7365 6c66 2062 7574 2074 6865  n itself but the
-0000e5c0: 2072 6573 756c 7420 6f66 2064 652d 7265   result of de-re
-0000e5d0: 6475 6e64 616e 6379 2c20 6e6f 7465 6420  dundancy, noted 
-0000e5e0: 7468 6174 2077 6520 696d 706c 656d 656e  that we implemen
-0000e5f0: 7465 6420 6120 6275 696c 742d 696e 2067  ted a built-in g
-0000e600: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
-0000e610: 6365 206d 6574 686f 6420 6261 7365 6420  ce method based 
-0000e620: 6f6e 2074 6865 2041 7665 7261 6765 204c  on the Average L
-0000e630: 696e 6b61 6765 2043 6c75 7374 6572 696e  inkage Clusterin
-0000e640: 6720 2841 4c43 2920 2874 6875 7320 6120  g (ALC) (thus a 
-0000e650: 6469 7374 616e 6365 2d62 6173 6564 2074  distance-based t
-0000e660: 7265 6529 2077 6974 6820 7468 6520 602d  ree) with the `-
-0000e670: 2d74 7265 655f 6d65 7468 6f64 6020 7365  -tree_method` se
-0000e680: 7420 6173 2022 636c 7573 7465 7222 2e20  t as "cluster". 
-0000e690: 5477 6f20 6d65 7468 6f64 7320 6f66 2064  Two methods of d
-0000e6a0: 652d 7265 6475 6e64 616e 6379 2061 7265  e-redundancy are
-0000e6b0: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
-0000e6c0: 6077 6764 2076 3260 2c20 6e61 6d65 6c79  `wgd v2`, namely
-0000e6d0: 206e 6f64 652d 7765 6967 6874 6564 2061   node-weighted a
-0000e6e0: 6e64 206e 6f64 652d 6176 6572 6167 6564  nd node-averaged
-0000e6f0: 206d 6574 686f 6473 2e20 5468 6520 6e6f   methods. The no
-0000e700: 6465 2d77 6569 6768 7465 6420 6d65 7468  de-weighted meth
-0000e710: 6f64 2061 6368 6965 7665 7320 7468 6520  od achieves the 
-0000e720: 6465 2d72 6564 756e 6461 6e63 7920 7669  de-redundancy vi
-0000e730: 6120 7765 6967 6869 6e67 2074 6865 202a  a weighing the *
-0000e740: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
-0000e750: 616c 7565 2061 7373 6f63 6961 7465 6420  alue associated 
-0000e760: 7769 7468 2065 6163 6820 6765 6e65 2070  with each gene p
-0000e770: 6169 7220 7375 6368 2074 6861 7420 7468  air such that th
-0000e780: 6520 7765 6967 6874 206f 6620 6120 7369  e weight of a si
-0000e790: 6e67 6c65 2064 7570 6c69 6361 7469 6f6e  ngle duplication
-0000e7a0: 2065 7665 6e74 2073 756d 7320 7570 2074   event sums up t
-0000e7b0: 6f20 3120 286e 6f74 6564 2074 6861 7420  o 1 (noted that 
-0000e7c0: 7468 6520 6e75 6d62 6572 206f 6620 2a4b  the number of *K
-0000e7d0: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-0000e7e0: 7469 6d61 7469 6f6e 2072 656d 6169 6e73  timation remains
-0000e7f0: 2074 6865 2073 616d 6529 2077 6869 6c65   the same) while
-0000e800: 2074 6865 206e 6f64 652d 6176 6572 6167   the node-averag
-0000e810: 6564 206d 6574 686f 6420 7265 616c 697a  ed method realiz
-0000e820: 6573 2074 6865 2064 652d 7265 6475 6e64  es the de-redund
-0000e830: 616e 6379 2076 6961 2063 616c 6375 6c61  ancy via calcula
-0000e840: 7469 6e67 2070 6572 2067 656e 6520 7472  ting per gene tr
-0000e850: 6565 206e 6f64 6520 6f6e 6520 6176 6572  ee node one aver
-0000e860: 6167 6564 202a 4b2a 3c73 7562 3e53 3c2f  aged *K*<sub>S</
-0000e870: 7375 623e 2076 616c 7565 2074 6f20 7265  sub> value to re
-0000e880: 7072 6573 656e 7420 7468 6520 6167 6520  present the age 
-0000e890: 6f66 2065 6163 6820 6475 706c 6963 6174  of each duplicat
-0000e8a0: 696f 6e20 6576 656e 742e 2054 6865 206f  ion event. The o
-0000e8b0: 7074 696f 6e20 602d 2d6e 6f64 655f 6176  ption `--node_av
-0000e8c0: 6572 6167 6560 2063 616e 2062 6520 7365  erage` can be se
-0000e8d0: 7420 746f 2063 686f 6f73 6520 7468 6520  t to choose the 
-0000e8e0: 6e6f 6465 2d61 7665 7261 6765 6420 7761  node-averaged wa
-0000e8f0: 7920 6f66 2064 652d 7265 6475 6e64 616e  y of de-redundan
-0000e900: 6379 2e20 4469 6666 6572 656e 7420 6d65  cy. Different me
-0000e910: 7468 6f64 7320 6f66 2064 652d 7265 6475  thods of de-redu
-0000e920: 6e64 616e 6379 2077 696c 6c20 6861 7665  ndancy will have
-0000e930: 2069 6d70 6163 7420 6f6e 2074 6865 2064   impact on the d
-0000e940: 6574 6563 7469 6f6e 206f 6620 5747 4420  etection of WGD 
-0000e950: 7369 676e 616c 732c 2077 6869 6368 2068  signals, which h
-0000e960: 6173 2062 6565 6e20 696e 7665 7374 6967  as been investig
-0000e970: 6174 6564 2069 6e20 7468 6973 205b 6c69  ated in this [li
-0000e980: 7465 7261 7475 7265 5d28 6874 7470 733a  terature](https:
-0000e990: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-0000e9a0: 332f 6762 652f 6576 7932 3030 292e 2054  3/gbe/evy200). T
-0000e9b0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000e9c0: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000e9d0: 7061 7261 6c6c 656c 697a 6520 7468 6520  parallelize the 
-0000e9e0: 616e 616c 7973 6973 2066 6f72 2065 6163  analysis for eac
-0000e9f0: 6820 6765 6e65 2066 616d 696c 7920 616e  h gene family an
-0000ea00: 6420 7468 7573 2073 7567 6765 7374 6564  d thus suggested
-0000ea10: 2074 6f20 6265 2073 6574 2061 7320 6d75   to be set as mu
-0000ea20: 6368 2061 7320 7468 6520 6e75 6d62 6572  ch as the number
-0000ea30: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
-0000ea40: 732e 0a0a 2a2a 4120 7375 6767 6573 7465  s...**A suggeste
-0000ea50: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
-0000ea60: 616e 2075 7365 2063 6f6d 6d61 6e64 2061  an use command a
-0000ea70: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-0000ea80: 6764 206b 7364 2066 616d 696c 6965 7320  gd ksd families 
-0000ea90: 7365 7175 656e 6365 0a60 6060 0a0a 2a2a  sequence.```..**
-0000eaa0: 5468 6520 636f 6e73 7472 7563 7469 6f6e  The construction
-0000eab0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-0000eac0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000ead0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000eae0: 2a2a 0a60 6060 0a77 6764 206b 7364 2066  **.```.wgd ksd f
-0000eaf0: 616d 696c 6965 7320 7365 7175 656e 6365  amilies sequence
-0000eb00: 3120 7365 7175 656e 6365 3220 282d 2d72  1 sequence2 (--r
-0000eb10: 6577 6569 6768 7429 0a60 6060 0a0a 4672  eweight).```..Fr
-0000eb20: 6f6d 2070 6172 616c 6f67 6f75 7320 746f  om paralogous to
-0000eb30: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
-0000eb40: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000eb50: 2064 6973 7472 6962 7574 696f 6e2c 2075   distribution, u
-0000eb60: 7365 7273 206f 6e6c 7920 6e65 6564 2074  sers only need t
-0000eb70: 6f20 7072 6f76 6964 6520 6d6f 7265 2063  o provide more c
-0000eb80: 6473 2066 696c 6573 2e20 4e6f 7465 2074  ds files. Note t
-0000eb90: 6861 7420 7769 7468 206f 7274 686f 6c6f  hat with ortholo
-0000eba0: 676f 7573 2067 656e 6520 6661 6d69 6c69  gous gene famili
-0000ebb0: 6573 2074 6865 2077 6569 6768 7469 6e67  es the weighting
-0000ebc0: 206d 6574 686f 6420 6361 6e20 6265 2073   method can be s
-0000ebd0: 6574 2074 6f20 6265 2063 616c 6375 6c61  et to be calcula
-0000ebe0: 7465 6420 7065 7220 7370 6563 6965 7320  ted per species 
-0000ebf0: 7061 6972 2069 6e73 7465 6164 206f 6620  pair instead of 
-0000ec00: 636f 6e73 6964 6572 696e 6720 7468 6520  considering the 
-0000ec10: 7768 6f6c 6520 6661 6d69 6c79 2062 6563  whole family bec
-0000ec20: 6175 7365 2077 6865 6e20 706c 6f74 7469  ause when plotti
-0000ec30: 6e67 206f 7274 686f 6c6f 676f 7573 202a  ng orthologous *
-0000ec40: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000ec50: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000ec60: 6265 7477 6565 6e20 7477 6f20 7370 6563  between two spec
-0000ec70: 6965 7320 7468 6520 7765 6967 6874 2063  ies the weight c
-0000ec80: 616c 6375 6c61 7465 6420 6672 6f6d 2074  alculated from t
-0000ec90: 6869 7320 7370 6563 6966 6963 2073 7065  his specific spe
-0000eca0: 6369 6573 2070 6169 7220 7368 6f75 6c64  cies pair should
-0000ecb0: 2062 6520 636f 6e73 6572 7665 6420 7768   be conserved wh
-0000ecc0: 696c 6520 7468 6520 6f6e 6520 6361 6c63  ile the one calc
-0000ecd0: 756c 6174 6564 2066 726f 6d20 7468 6520  ulated from the 
-0000ece0: 7768 6f6c 6520 6661 6d69 6c79 2077 696c  whole family wil
-0000ecf0: 6c20 7661 7279 2077 6974 6820 7468 6520  l vary with the 
-0000ed00: 6e75 6d62 6572 206f 6620 7370 6563 6965  number of specie
-0000ed10: 732e 2054 6f20 696e 6974 6961 7465 2074  s. To initiate t
-0000ed20: 6865 2077 6569 6768 7469 6e67 2070 6572  he weighting per
-0000ed30: 2073 7065 6369 6573 2070 6169 722c 2074   species pair, t
-0000ed40: 6865 206f 7074 696f 6e20 602d 2d72 6577  he option `--rew
-0000ed50: 6569 6768 7460 2063 616e 2062 6520 7365  eight` can be se
-0000ed60: 742e 0a0a 2a2a 4120 7375 6767 6573 7465  t...**A suggeste
-0000ed70: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
-0000ed80: 616e 2075 7365 2063 6f6d 6d61 6e64 2061  an use command a
-0000ed90: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-0000eda0: 6764 206b 7364 2066 616d 696c 6965 7320  gd ksd families 
-0000edb0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
-0000edc0: 6365 320a 6060 600a 0a2a 2a54 6865 2063  ce2.```..**The c
-0000edd0: 6f6e 7374 7275 6374 696f 6e20 6f66 202a  onstruction of *
-0000ede0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000edf0: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000ee00: 7769 7468 2072 6174 6520 636f 7272 6563  with rate correc
-0000ee10: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
-0000ee20: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
-0000ee30: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000ee40: 7365 7175 656e 6365 3320 2d2d 666f 6375  sequence3 --focu
-0000ee50: 7332 616c 6c20 7365 7175 656e 6365 3120  s2all sequence1 
-0000ee60: 2d73 7020 7370 6461 7461 0a60 6060 0a0a  -sp spdata.```..
-0000ee70: 496e 7370 6972 6564 2062 7920 7468 6520  Inspired by the 
-0000ee80: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-0000ee90: 616c 676f 7269 7468 6d20 696e 205b 6b73  algorithm in [ks
-0000eea0: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000eeb0: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000eec0: 422f 6b73 7261 7465 7329 2c20 7765 2069  B/ksrates), we i
-0000eed0: 6d70 6c65 6d65 6e74 6564 2074 6865 2072  mplemented the r
-0000eee0: 6174 6520 636f 7272 6563 7469 6f6e 2061  ate correction a
-0000eef0: 6e61 6c79 7369 7320 616c 736f 2069 6e20  nalysis also in 
-0000ef00: 6077 6764 2076 3260 2c20 7768 6963 6820  `wgd v2`, which 
-0000ef10: 6973 206d 6f73 746c 7920 7468 6520 7361  is mostly the sa
-0000ef20: 6d65 2061 7320 5b6b 7372 6174 6573 5d28  me as [ksrates](
-0000ef30: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000ef40: 6f6d 2f56 4942 2d50 5342 2f6b 7372 6174  om/VIB-PSB/ksrat
-0000ef50: 6573 2920 6275 7420 6469 6666 6572 7320  es) but differs 
-0000ef60: 696e 2074 6865 2063 616c 6375 6c61 7469  in the calculati
-0000ef70: 6f6e 206f 6620 7468 6520 7374 616e 6461  on of the standa
-0000ef80: 7264 2064 6576 6961 7469 6f6e 206f 6620  rd deviation of 
-0000ef90: 7265 7363 616c 6564 202a 4b2a 3c73 7562  rescaled *K*<sub
-0000efa0: 3e53 3c2f 7375 623e 2061 6765 732e 2054  >S</sub> ages. T
-0000efb0: 6f20 7065 7266 6f72 6d20 7468 6520 7261  o perform the ra
-0000efc0: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
-0000efd0: 616c 7973 6973 2c20 7573 6572 7320 6361  alysis, users ca
-0000efe0: 6e20 7573 6520 626f 7468 2074 6865 2060  n use both the `
-0000eff0: 7767 6420 6b73 6460 2061 6e64 2060 7767  wgd ksd` and `wg
-0000f000: 6420 7669 7a60 2070 726f 6772 616d 2e20  d viz` program. 
-0000f010: 466f 7220 7468 6520 6077 6764 206b 7364  For the `wgd ksd
-0000f020: 6020 7072 6f67 7261 6d2c 2075 7365 7273  ` program, users
-0000f030: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
-0000f040: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
-0000f050: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-0000f060: 2d2d 7370 6563 6965 7374 7265 6560 206f  --speciestree` o
-0000f070: 6e20 7768 6963 6820 7261 7465 2063 6f72  n which rate cor
-0000f080: 7265 6374 696f 6e20 6361 6e20 6265 2063  rection can be c
-0000f090: 6f6e 6475 6374 6564 2e20 4e6f 7465 2074  onducted. Note t
-0000f0a0: 6861 7420 756e 6e65 6365 7373 6172 7920  hat unnecessary 
-0000f0b0: 6272 6163 6b65 7473 206d 6967 6874 206c  brackets might l
-0000f0c0: 6561 6420 746f 2075 6e65 7870 6563 7465  ead to unexpecte
-0000f0d0: 6420 6572 726f 7273 2c20 666f 7220 696e  d errors, for in
-0000f0e0: 7374 616e 6365 2061 2074 7265 6520 6028  stance a tree `(
-0000f0f0: 412c 2842 2c43 2929 3b60 2073 686f 756c  A,(B,C));` shoul
-0000f100: 6420 6e6f 7420 6265 2072 6570 7265 7365  d not be represe
-0000f110: 6e74 6564 2061 7320 6028 412c 2828 422c  nted as `(A,((B,
-0000f120: 4329 2929 3b60 2e20 5468 6520 7365 7420  C)));`. The set 
-0000f130: 6f66 2073 7065 6369 6573 2070 6169 7273  of species pairs
-0000f140: 2074 6f20 6265 2073 686f 776e 2069 7320   to be shown is 
-0000f150: 666c 6578 6962 6c65 2074 6861 7420 7468  flexible that th
-0000f160: 6520 6d6f 7374 2063 6f6e 7665 6e69 656e  e most convenien
-0000f170: 7420 6f70 7469 6f6e 2069 7320 602d 2d66  t option is `--f
-0000f180: 6f63 7573 3261 6c6c 6020 7768 6963 6820  ocus2all` which 
-0000f190: 7369 6d70 6c79 2073 686f 7773 2061 6c6c  simply shows all
-0000f1a0: 2074 6865 2070 6f73 7369 626c 6520 666f   the possible fo
-0000f1b0: 6361 6c2d 7369 7374 6572 2073 7065 6369  cal-sister speci
-0000f1c0: 6573 2070 6169 7273 2c20 6f72 2075 7365  es pairs, or use
-0000f1d0: 7273 2063 616e 206d 616e 7561 6c6c 7920  rs can manually 
-0000f1e0: 7365 7420 7468 6520 7370 6563 6965 7320  set the species 
-0000f1f0: 7061 6972 7320 7669 6120 7468 6520 6f70  pairs via the op
-0000f200: 7469 6f6e 2060 2d2d 7370 6169 7260 2e20  tion `--spair`. 
-0000f210: 4e6f 7465 2074 6861 7420 6966 2074 6865  Note that if the
-0000f220: 2073 7065 6369 6573 2070 6169 7273 2077   species pairs w
-0000f230: 6572 6520 6d61 6e75 616c 6c79 2073 6574  ere manually set
-0000f240: 2c20 6974 2077 6f75 6c64 2062 6520 6e65  , it would be ne
-0000f250: 6564 6564 2074 6f20 7365 7420 7468 6520  eded to set the 
-0000f260: 6f70 7469 6f6e 2060 2d2d 636c 6173 7369  option `--classi
-0000f270: 6360 2e20 5468 6520 6f70 7469 6f6e 2060  c`. The option `
-0000f280: 6f6e 6c79 726f 6f74 6f75 7460 2063 616e  onlyrootout` can
-0000f290: 2062 6520 7365 7420 746f 206f 6e6c 7920   be set to only 
-0000f2a0: 636f 6e73 6964 6572 206f 7574 6772 6f75  consider outgrou
-0000f2b0: 7020 6174 2074 6865 2072 6f6f 742c 2069  p at the root, i
-0000f2c0: 6e73 7465 6164 206f 6620 616c 6c20 7468  nstead of all th
-0000f2d0: 6520 706f 7373 6962 6c65 206f 7574 6772  e possible outgr
-0000f2e0: 6f75 7073 2070 6572 2066 6f63 616c 2d73  oups per focal-s
-0000f2f0: 6973 7465 7220 7370 6563 6965 7320 7061  ister species pa
-0000f300: 6972 2c20 7768 6963 6820 6861 7320 696d  ir, which has im
-0000f310: 7061 6374 206f 6e20 7468 6520 6669 6e61  pact on the fina
-0000f320: 6c20 636f 7272 6563 7465 6420 2a4b 2a3c  l corrected *K*<
-0000f330: 7375 623e 533c 2f73 7562 3e20 6167 6573  sub>S</sub> ages
-0000f340: 2e20 536f 6d65 206f 7468 6572 206f 7074  . Some other opt
-0000f350: 696f 6e73 2077 6869 6368 2068 6176 6520  ions which have 
-0000f360: 6e6f 2069 6d70 6163 7420 6f6e 2074 6865  no impact on the
-0000f370: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-0000f380: 2062 7574 2061 6464 206d 6f72 6520 6c61   but add more la
-0000f390: 7965 7273 206f 7220 6368 616e 6765 2074  yers or change t
-0000f3a0: 6865 2061 7070 6561 7261 6e63 6520 6f6e  he appearance on
-0000f3b0: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
-0000f3c0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000f3d0: 6973 7472 6962 7574 696f 6e2c 2069 6e63  istribution, inc
-0000f3e0: 6c75 6465 2060 706c 6f74 6170 676d 6d60  lude `plotapgmm`
-0000f3f0: 2061 6e64 2060 706c 6f74 656c 6d6d 6020   and `plotelmm` 
-0000f400: 6574 632e 2054 6865 206f 7074 696f 6e20  etc. The option 
-0000f410: 6070 6c6f 7461 7067 6d6d 6020 6361 6e20  `plotapgmm` can 
-0000f420: 6265 2073 6574 2074 6f20 6361 6c6c 2074  be set to call t
-0000f430: 6865 2047 4d4d 2061 6e61 6c79 7369 7320  he GMM analysis 
-0000f440: 7570 6f6e 2074 6865 2061 6e63 686f 7220  upon the anchor 
-0000f450: 7061 6972 202a 4b2a 3c73 7562 3e53 3c2f  pair *K*<sub>S</
-0000f460: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000f470: 7574 696f 6e20 616e 6420 706c 6f74 2074  ution and plot t
-0000f480: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
-0000f490: 7375 6c74 2075 706f 6e20 7468 6520 6d69  sult upon the mi
-0000f4a0: 7865 6420 2a4b 2a3c 7375 623e 533c 2f73  xed *K*<sub>S</s
-0000f4b0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
-0000f4c0: 7469 6f6e 2c20 7768 6963 6820 6861 7320  tion, which has 
-0000f4d0: 746f 2062 6520 636f 2d73 6574 2077 6974  to be co-set wit
-0000f4e0: 6820 7468 6520 6f70 7469 6f6e 2060 2d2d  h the option `--
-0000f4f0: 616e 6368 6f72 706f 696e 7473 6020 7072  anchorpoints` pr
-0000f500: 6f76 6964 696e 6720 7468 6520 616e 6368  oviding the anch
-0000f510: 6f72 2070 6169 7273 2069 6e66 6f72 6d61  or pairs informa
-0000f520: 7469 6f6e 2e20 5468 6520 6f70 7469 6f6e  tion. The option
-0000f530: 2060 2d2d 706c 6f74 656c 6d6d 6020 6361   `--plotelmm` ca
-0000f540: 6e20 6265 2073 6574 2074 6f20 6361 6c6c  n be set to call
-0000f550: 2074 6865 2045 4c4d 4d20 616e 616c 7973   the ELMM analys
-0000f560: 6973 2075 706f 6e20 7468 6520 7061 7261  is upon the para
-0000f570: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-0000f580: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000f590: 6962 7574 696f 6e2e 0a0a 5468 6572 6520  ibution...There 
-0000f5a0: 6172 6520 3231 2063 6f6c 756d 6e73 2069  are 21 columns i
-0000f5b0: 6e20 7468 6520 7265 7375 6c74 2060 2a6b  n the result `*k
-0000f5c0: 732e 7473 7660 2066 696c 6520 6265 7369  s.tsv` file besi
-0000f5d0: 6465 7320 7468 6520 696e 6465 7820 636f  des the index co
-0000f5e0: 6c75 6d6e 7320 6070 6169 7260 2061 7320  lumns `pair` as 
-0000f5f0: 7468 6520 756e 6971 7565 2069 6465 6e74  the unique ident
-0000f600: 6966 6965 7220 666f 7220 6561 6368 2067  ifier for each g
-0000f610: 656e 6520 7061 6972 2e20 5468 6520 604e  ene pair. The `N
-0000f620: 602c 2060 5360 2c20 6064 4e60 2c20 6064  `, `S`, `dN`, `d
-0000f630: 4e2f 6453 602c 2060 6453 602c 2060 6c60  N/dS`, `dS`, `l`
-0000f640: 2061 6e64 2060 7460 2061 7265 2066 726f   and `t` are fro
-0000f650: 6d20 7468 6520 636f 6465 6d6c 2072 6573  m the codeml res
-0000f660: 756c 7473 2c20 7265 7072 6573 656e 7469  ults, representi
-0000f670: 6e67 2074 6865 204e 2065 7374 696d 6174  ng the N estimat
-0000f680: 652c 2074 6865 2053 2065 7374 696d 6174  e, the S estimat
-0000f690: 652c 2074 6865 2064 4e20 6573 7469 6d61  e, the dN estima
-0000f6a0: 7465 2c20 7468 6520 644e 2f64 5320 286f  te, the dN/dS (o
-0000f6b0: 6d65 6761 2920 6573 7469 6d61 7465 2c20  mega) estimate, 
-0000f6c0: 7468 6520 6453 2065 7374 696d 6174 652c  the dS estimate,
-0000f6d0: 2074 6865 206c 6f67 2d6c 696b 656c 6968   the log-likelih
-0000f6e0: 6f6f 6420 616e 6420 7468 6520 7420 6573  ood and the t es
-0000f6f0: 7469 6d61 7465 2c20 7265 7370 6563 7469  timate, respecti
-0000f700: 7665 6c79 2e20 5468 6520 6061 6c69 676e  vely. The `align
-0000f710: 6d65 6e74 636f 7665 7261 6765 602c 2060  mentcoverage`, `
-0000f720: 616c 6967 6e6d 656e 7469 6465 6e74 6974  alignmentidentit
-0000f730: 7960 2061 6e64 2060 616c 6967 6e6d 656e  y` and `alignmen
-0000f740: 746c 656e 6774 6860 2061 7265 2074 6865  tlength` are the
-0000f750: 2069 6e66 6f72 6d61 7469 6f6e 2070 6572   information per
-0000f760: 7461 696e 696e 6720 746f 2074 6865 2061  taining to the a
-0000f770: 6c69 676e 6d65 6e74 2066 6f72 2065 6163  lignment for eac
-0000f780: 6820 6661 6d69 6c79 2c20 7265 7072 6573  h family, repres
-0000f790: 656e 7469 6e67 2074 6865 2072 6174 696f  enting the ratio
-0000f7a0: 206f 6620 7468 6520 7374 7269 7070 6564   of the stripped
-0000f7b0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
-0000f7c0: 6820 636f 6d70 6172 6564 2074 6f20 7468  h compared to th
-0000f7d0: 6520 6675 6c6c 2061 6c69 676e 6d65 6e74  e full alignment
-0000f7e0: 206c 656e 6774 682c 2074 6865 2072 6174   length, the rat
-0000f7f0: 696f 206f 6620 636f 6c75 6d6e 7320 7769  io of columns wi
-0000f800: 7468 2069 6465 6e74 6963 616c 206e 7563  th identical nuc
-0000f810: 6c65 6f74 6964 6573 2063 6f6d 7061 7265  leotides compare
-0000f820: 6420 746f 2074 6865 206f 7665 7261 6c6c  d to the overall
-0000f830: 2063 6f6c 756d 6e73 206f 6620 7468 6520   columns of the 
-0000f840: 7374 7269 7070 6564 2061 6c69 676e 6d65  stripped alignme
-0000f850: 6e74 2c20 616e 6420 7468 6520 6c65 6e67  nt, and the leng
-0000f860: 7468 206f 6620 7468 6520 6675 6c6c 2061  th of the full a
-0000f870: 6c69 676e 6d65 6e74 2c20 7265 7370 6563  lignment, respec
-0000f880: 7469 7665 6c79 2e0a 2323 2320 7767 6420  tively..### wgd 
-0000f890: 6d69 780a 0a2a 2a54 6865 206d 6978 7475  mix..**The mixtu
-0000f8a0: 7265 206d 6f64 656c 2063 6c75 7374 6572  re model cluster
-0000f8b0: 696e 6720 616e 616c 7973 6973 206f 6620  ing analysis of 
-0000f8c0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000f8d0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000f8e0: 2a2a 0a60 6060 0a77 6764 206d 6978 206b  **.```.wgd mix k
-0000f8f0: 7364 6174 6120 282d 2d6e 5f69 6e69 7420  sdata (--n_init 
-0000f900: 3230 3020 2d2d 6d61 785f 6974 6572 2032  200 --max_iter 2
-0000f910: 3030 202d 2d6b 735f 7261 6e67 6520 3020  00 --ks_range 0 
-0000f920: 3520 2d2d 6669 6c74 6572 7320 3330 3020  5 --filters 300 
-0000f930: 2d2d 6269 6e73 2035 3020 2d2d 636f 6d70  --bins 50 --comp
-0000f940: 6f6e 656e 7473 2031 2034 202d 2d67 616d  onents 1 4 --gam
-0000f950: 6d61 2030 2e30 3031 290a 6060 600a 0a54  ma 0.001).```..T
-0000f960: 6869 7320 7061 7274 206f 6620 4761 7573  his part of Gaus
-0000f970: 7369 616e 206d 6978 7475 7265 206d 6f64  sian mixture mod
-0000f980: 656c 696e 6720 2847 4d4d 2920 616e 616c  eling (GMM) anal
-0000f990: 7973 6973 2069 7320 696e 6865 7269 7465  ysis is inherite
-0000f9a0: 6420 6672 6f6d 2074 6865 206f 7269 6769  d from the origi
-0000f9b0: 6e61 6c20 6077 6764 6020 7072 6f67 7261  nal `wgd` progra
-0000f9c0: 6d2c 2062 7574 2077 7269 7465 7320 6164  m, but writes ad
-0000f9d0: 6469 7469 6f6e 616c 6c79 2074 6865 2070  ditionally the p
-0000f9e0: 726f 6261 6269 6c69 7479 206f 6620 6561  robability of ea
-0000f9f0: 6368 202a 4b2a 3c73 7562 3e53 3c2f 7375  ch *K*<sub>S</su
-0000fa00: 623e 2076 616c 7565 2069 6e74 6f20 7468  b> value into th
-0000fa10: 6520 6669 6e61 6c20 6461 7461 6672 616d  e final datafram
-0000fa20: 652e 2042 6173 6963 616c 6c79 2c20 7573  e. Basically, us
-0000fa30: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-0000fa40: 6964 6520 7769 7468 2061 2028 6e6f 726d  ide with a (norm
-0000fa50: 616c 6c79 2066 726f 6d20 7768 6f6c 652d  ally from whole-
-0000fa60: 7061 7261 6e6f 6d65 206f 7220 616e 6368  paranome or anch
-0000fa70: 6f72 2d70 6169 7273 2920 2a4b 2a3c 7375  or-pairs) *K*<su
-0000fa80: 623e 533c 2f73 7562 3e20 6461 7461 6669  b>S</sub> datafi
-0000fa90: 6c65 2061 6e64 2074 6865 2047 4d4d 2061  le and the GMM a
-0000faa0: 6e61 6c79 7369 7320 7769 6c6c 2062 6520  nalysis will be 
-0000fab0: 636f 6e64 7563 7465 6420 7570 6f6e 2074  conducted upon t
-0000fac0: 6865 2064 6174 6166 696c 652e 2053 6f6d  he datafile. Som
-0000fad0: 6520 7061 7261 6d65 7465 7273 2063 616e  e parameters can
-0000fae0: 2061 6666 6563 7420 7468 6520 7265 7375   affect the resu
-0000faf0: 6c74 732c 2069 6e63 6c75 6469 6e67 2060  lts, including `
-0000fb00: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
-0000fb10: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-0000fb20: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-0000fb30: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
-0000fb40: 6175 6c74 2032 3030 292c 2060 2d2d 6d61  ault 200), `--ma
-0000fb50: 785f 6974 6572 602c 2077 6869 6368 2073  x_iter`, which s
-0000fb60: 6574 7320 7468 6520 6d61 7869 6d75 6d20  ets the maximum 
-0000fb70: 6e75 6d62 6572 206f 6620 6974 6572 6174  number of iterat
-0000fb80: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
-0000fb90: 3029 2c20 602d 2d6d 6574 686f 6460 2c20  0), `--method`, 
-0000fba0: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
-0000fbb0: 2077 6869 6368 2063 6c75 7374 6572 696e   which clusterin
-0000fbc0: 6720 6d65 7468 6f64 2074 6f20 7573 6520  g method to use 
-0000fbd0: 2864 6566 6175 6c74 2067 6d6d 292c 2060  (default gmm), `
-0000fbe0: 2d2d 6761 6d6d 6160 2c20 7768 6963 6820  --gamma`, which 
-0000fbf0: 7365 7473 2074 6865 2067 616d 6d61 2070  sets the gamma p
-0000fc00: 6172 616d 6574 6572 2066 6f72 2074 6865  arameter for the
-0000fc10: 2062 676d 6d20 6d6f 6465 6c20 2864 6566   bgmm model (def
-0000fc20: 6175 6c74 2030 2e30 3031 292c 2060 2d2d  ault 0.001), `--
-0000fc30: 636f 6d70 6f6e 656e 7473 602c 2077 6869  components`, whi
-0000fc40: 6368 2073 6574 7320 7468 6520 7261 6e67  ch sets the rang
-0000fc50: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-0000fc60: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-0000fc70: 2066 6974 2028 6465 6661 756c 7420 3120   fit (default 1 
-0000fc80: 3429 2c20 616e 6420 7468 6520 6461 7461  4), and the data
-0000fc90: 2066 696c 7465 7269 6e67 2070 6172 616d   filtering param
-0000fca0: 6574 6572 7320 602d 2d66 696c 7465 7273  eters `--filters
-0000fcb0: 6020 7768 6963 6820 6669 6c74 6572 7320  ` which filters 
-0000fcc0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
-0000fcd0: 6967 6e6d 656e 7420 6c65 6e67 7468 2c20  ignment length, 
-0000fce0: 602d 2d6b 735f 7261 6e67 6560 2077 6869  `--ks_range` whi
-0000fcf0: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
-0000fd00: 6261 7365 6420 6f6e 202a 4b2a 3c73 7562  based on *K*<sub
-0000fd10: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-0000fd20: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
-0000fd30: 7220 602d 2d62 696e 7360 2077 6869 6368  r `--bins` which
-0000fd40: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-0000fd50: 206f 6620 6269 6e73 2069 6e20 2a4b 2a3c   of bins in *K*<
-0000fd60: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-0000fd70: 7269 6275 7469 6f6e 2028 6465 6661 756c  ribution (defaul
-0000fd80: 7420 3530 292e 0a0a 2a2a 4120 7375 6767  t 50)...**A sugg
-0000fd90: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000fda0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-0000fdb0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-0000fdc0: 6f77 2a2a 0a0a 6060 600a 7767 6420 6d69  ow**..```.wgd mi
-0000fdd0: 7820 6b73 6461 7461 0a60 6060 0a0a 2323  x ksdata.```..##
-0000fde0: 2320 7767 6420 7065 616b 0a0a 2a2a 5468  # wgd peak..**Th
-0000fdf0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
-0000fe00: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
-0000fe10: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
-0000fe20: 2069 6e20 5747 4420 6461 7469 6e67 2a2a   in WGD dating**
-0000fe30: 0a60 6060 0a77 6764 2070 6561 6b20 6b73  .```.wgd peak ks
-0000fe40: 6461 7461 202d 6170 2061 7064 6174 6120  data -ap apdata 
-0000fe50: 2d73 6d20 736d 6461 7461 202d 6c65 206c  -sm smdata -le l
-0000fe60: 6564 6174 6120 2d6d 7020 6d70 6461 7461  edata -mp mpdata
-0000fe70: 202d 2d68 6575 7269 7374 6963 2028 2d2d   --heuristic (--
-0000fe80: 616c 6967 6e66 696c 7465 7220 302e 3020  alignfilter 0.0 
-0000fe90: 3020 302e 3020 2d2d 6b73 7261 6e67 6520  0 0.0 --ksrange 
-0000fea0: 3020 3520 2d2d 6269 6e5f 7769 6474 6820  0 5 --bin_width 
-0000feb0: 302e 3120 2d2d 6775 6964 6520 7365 676d  0.1 --guide segm
-0000fec0: 656e 7420 2d2d 7072 6f6d 696e 656e 6365  ent --prominence
-0000fed0: 5f63 7574 6f66 6620 302e 3120 2d2d 7265  _cutoff 0.1 --re
-0000fee0: 6c5f 6865 6967 6874 2030 2e34 202d 2d63  l_height 0.4 --c
-0000fef0: 6920 3935 202d 2d68 6472 2039 3520 2d2d  i 95 --hdr 95 --
-0000ff00: 6b73 6375 746f 6666 2035 290a 6060 600a  kscutoff 5).```.
-0000ff10: 0a41 7320 6d65 6e74 696f 6e65 6420 7072  .As mentioned pr
-0000ff20: 6576 696f 7573 6c79 2c20 6120 6865 7572  eviously, a heur
-0000ff30: 6973 7469 6320 6d65 7468 6f64 2061 6e64  istic method and
-0000ff40: 2061 2063 6f6c 6c69 6e65 6172 2073 6567   a collinear seg
-0000ff50: 6d65 6e74 732d 6775 6964 6564 2061 6e63  ments-guided anc
-0000ff60: 686f 7220 7061 6972 2063 6c75 7374 6572  hor pair cluster
-0000ff70: 696e 6720 666f 7220 7468 6520 7365 6172  ing for the sear
-0000ff80: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
-0000ff90: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000ffa0: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
-0000ffb0: 4420 6461 7469 6e67 2061 7265 2069 6d70  D dating are imp
-0000ffc0: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
-0000ffd0: 2076 3260 2e20 5573 6572 7320 6e65 6564   v2`. Users need
-0000ffe0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000fff0: 616e 6368 6f72 2070 6f69 6e74 732c 2073  anchor points, s
-00010000: 6567 6d65 6e74 732c 206c 6973 7473 6567  egments, listseg
-00010010: 6d65 6e74 732c 206d 756c 7469 706c 6963  ments, multiplic
-00010020: 6f6e 7061 6972 7320 6461 7461 6669 6c65  onpairs datafile
-00010030: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
-00010040: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-00010050: 636c 7573 7465 7269 6e67 2066 756e 6374  clustering funct
-00010060: 696f 6e2e 2053 6f6d 6520 7061 7261 6d65  ion. Some parame
-00010070: 7465 7273 2074 6861 7420 6361 6e20 696d  ters that can im
-00010080: 7061 6374 2074 6865 2072 6573 756c 7473  pact the results
-00010090: 2069 6e63 6c75 6465 2060 2d2d 616c 6967   include `--alig
-000100a0: 6e66 696c 7465 7260 2c20 7768 6963 6820  nfilter`, which 
-000100b0: 6669 6c74 6572 7320 7468 6520 6461 7461  filters the data
-000100c0: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
-000100d0: 656e 7420 6964 656e 7469 7479 2c20 6c65  ent identity, le
-000100e0: 6e67 7468 2061 6e64 2063 6f76 6572 6167  ngth and coverag
-000100f0: 652c 2060 2d2d 6b73 7261 6e67 6560 2c20  e, `--ksrange`, 
-00010100: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
-00010110: 616e 6765 206f 6620 4b73 2074 6f20 6265  ange of Ks to be
-00010120: 2061 6e61 6c79 7a65 642c 2060 2d2d 6269   analyzed, `--bi
-00010130: 6e5f 7769 6474 6860 2c20 7768 6963 6820  n_width`, which 
-00010140: 7365 7473 2074 6865 2062 616e 6477 6964  sets the bandwid
-00010150: 7468 206f 6620 2a4b 2a3c 7375 623e 533c  th of *K*<sub>S<
-00010160: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00010170: 6f6e 2c20 602d 2d77 6569 6768 7473 5f6f  on, `--weights_o
-00010180: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
-00010190: 6020 7768 6963 6820 6465 7465 726d 696e  ` which determin
-000101a0: 6573 2077 6865 7468 6572 2074 6f20 696e  es whether to in
-000101b0: 636c 7564 6520 2a4b 2a3c 7375 623e 533c  clude *K*<sub>S<
-000101c0: 2f73 7562 3e20 6f75 746c 6965 7273 2028  /sub> outliers (
-000101d0: 7768 6f73 6520 7661 6c75 6520 6973 206f  whose value is o
-000101e0: 7665 7220 3529 2069 6e20 616e 616c 7973  ver 5) in analys
-000101f0: 6973 2c20 602d 2d6d 6574 686f 6460 2077  is, `--method` w
-00010200: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-00010210: 7768 6963 6820 636c 7573 7465 7269 6e67  which clustering
-00010220: 206d 6574 686f 6420 746f 2075 7365 2028   method to use (
-00010230: 6465 6661 756c 7420 676d 6d29 2c20 602d  default gmm), `-
-00010240: 2d73 6565 6460 2077 6869 6368 2073 6574  -seed` which set
-00010250: 7320 7468 6520 7261 6e64 6f6d 2073 6565  s the random see
-00010260: 6420 6769 7665 6e20 746f 2069 6e69 7469  d given to initi
-00010270: 616c 697a 6174 696f 6e20 2864 6566 6175  alization (defau
-00010280: 6c74 2032 3335 3238 3930 292c 2060 2d2d  lt 2352890), `--
-00010290: 6e5f 696e 6974 602c 2077 6869 6368 2073  n_init`, which s
-000102a0: 6574 7320 7468 6520 6e75 6d62 6572 206f  ets the number o
-000102b0: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
-000102c0: 6c69 7a61 7469 6f6e 7320 2864 6566 6175  lizations (defau
-000102d0: 6c74 2032 3030 292c 2060 2d2d 656d 5f69  lt 200), `--em_i
-000102e0: 7465 7260 2c20 7768 6963 6820 7365 7473  ter`, which sets
-000102f0: 2074 6865 206d 6178 696d 756d 206e 756d   the maximum num
-00010300: 6265 7220 6f66 2069 7465 7261 7469 6f6e  ber of iteration
-00010310: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
-00010320: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
-00010330: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
-00010340: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-00010350: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
-00010360: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
-00010370: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
-00010380: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
-00010390: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-000103a0: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-000103b0: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
-000103c0: 3120 3429 2c20 602d 2d77 6569 6768 7465  1 4), `--weighte
-000103d0: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
-000103e0: 6e65 7320 7768 6574 6865 7220 746f 2075  nes whether to u
-000103f0: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00010400: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-00010410: 6564 756e 6461 6e63 792c 2060 2d2d 6775  edundancy, `--gu
-00010420: 6964 6560 2077 6869 6368 2064 6574 6572  ide` which deter
-00010430: 6d69 6e65 7320 7768 6963 6820 7265 6769  mines which regi
-00010440: 6d65 2072 6573 6964 696e 6720 616e 6368  me residing anch
-00010450: 6f72 7320 746f 2062 6520 7573 6564 2028  ors to be used (
-00010460: 6465 6661 756c 7420 7365 676d 656e 7429  default segment)
-00010470: 2c20 602d 2d70 726f 6d69 6e65 6e63 655f  , `--prominence_
-00010480: 6375 746f 6666 6020 7768 6963 6820 7365  cutoff` which se
-00010490: 7473 2074 6865 2070 726f 6d69 6e65 6e63  ts the prominenc
-000104a0: 6520 6375 746f 6666 206f 6620 6163 6365  e cutoff of acce
-000104b0: 7074 6162 6c65 2070 6561 6b73 2069 6e20  ptable peaks in 
-000104c0: 7065 616b 2066 696e 6469 6e67 2070 726f  peak finding pro
-000104d0: 6365 7373 2c20 602d 2d72 656c 5f68 6569  cess, `--rel_hei
-000104e0: 6768 7460 2077 6869 6368 2073 6574 7320  ght` which sets 
-000104f0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-00010500: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00010510: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00010520: 6561 7375 7265 642c 2060 2d2d 6b73 746f  easured, `--ksto
-00010530: 6461 7465 6020 7768 6963 6820 6d61 6e75  date` which manu
-00010540: 616c 6c79 2073 6574 7320 7468 6520 7261  ally sets the ra
-00010550: 6e67 6520 6f66 202a 4b2a 3c73 7562 3e53  nge of *K*<sub>S
-00010560: 3c2f 7375 623e 2074 6f20 6265 2064 6174  </sub> to be dat
-00010570: 6564 2069 6e20 6865 7572 6973 7469 6320  ed in heuristic 
-00010580: 7365 6172 6368 2061 6e64 206e 6565 6473  search and needs
-00010590: 2074 6f20 6265 2063 6f2d 7365 7420 7769   to be co-set wi
-000105a0: 7468 206f 7074 696f 6e20 602d 2d6d 616e  th option `--man
-000105b0: 7561 6c73 6574 602c 2060 2d2d 786c 696d  ualset`, `--xlim
-000105c0: 6020 616e 6420 602d 2d79 6c69 6d60 2064  ` and `--ylim` d
-000105d0: 6574 6572 6d69 6e69 6e67 2074 6865 2078  etermining the x
-000105e0: 2061 6e64 2079 2061 7869 7320 6c69 6d69   and y axis limi
-000105f0: 7420 6f66 2047 4d4d 202a 4b2a 3c73 7562  t of GMM *K*<sub
-00010600: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00010610: 7574 696f 6e2c 2060 2d2d 6369 6020 7365  ution, `--ci` se
-00010620: 7474 696e 6720 7468 6520 636f 6e66 6964  tting the confid
-00010630: 656e 6365 206c 6576 656c 206f 6620 6c6f  ence level of lo
-00010640: 672d 6e6f 726d 616c 2064 6973 7472 6962  g-normal distrib
-00010650: 7574 696f 6e20 746f 2064 6174 6520 2864  ution to date (d
-00010660: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
-00010670: 6472 6020 7365 7474 696e 6720 7468 6520  dr` setting the 
-00010680: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
-00010690: 7265 6769 6f6e 2028 4844 5229 2061 7070  region (HDR) app
-000106a0: 6c69 6564 2069 6e20 7468 6520 7365 676d  lied in the segm
-000106b0: 656e 742d 6775 6964 6564 2061 6e63 686f  ent-guided ancho
-000106c0: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-000106d0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
-000106e0: 696f 6e2c 2060 2d2d 6865 7572 6973 7469  ion, `--heuristi
-000106f0: 6360 2064 6574 6572 6d69 6e69 6e67 2077  c` determining w
-00010700: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00010710: 7465 2068 6575 7269 7374 6963 206d 6574  te heuristic met
-00010720: 686f 6420 6f66 2064 6566 696e 696e 6720  hod of defining 
-00010730: 4349 2066 6f72 2064 6174 696e 672c 2060  CI for dating, `
-00010740: 2d2d 6b73 6375 746f 6666 6020 7365 7474  --kscutoff` sett
-00010750: 696e 6720 7468 6520 2a4b 2a3c 7375 623e  ing the *K*<sub>
-00010760: 533c 2f73 7562 3e20 7361 7475 7261 7469  S</sub> saturati
-00010770: 6f6e 2063 7574 6f66 6620 696e 2064 6174  on cutoff in dat
-00010780: 696e 6720 2864 6566 6175 6c74 2035 2920  ing (default 5) 
-00010790: 616e 6420 602d 2d73 686f 7763 6960 2064  and `--showci` d
-000107a0: 6574 6572 6d69 6e69 6e67 2077 6865 7468  etermining wheth
-000107b0: 6572 2074 6f20 7368 6f77 2043 4920 666f  er to show CI fo
-000107c0: 7220 7468 6520 6f72 6967 696e 616c 2061  r the original a
-000107d0: 6e63 686f 7220 4b73 2047 4d4d 2061 6e61  nchor Ks GMM ana
-000107e0: 6c79 7369 732e 0a0a 2a2a 4120 7375 6767  lysis...**A sugg
-000107f0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-00010800: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-00010810: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00010820: 6f77 2a2a 0a0a 6060 600a 7767 6420 7065  ow**..```.wgd pe
-00010830: 616b 206b 7364 6174 6120 2d61 7020 6170  ak ksdata -ap ap
-00010840: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
-00010850: 2d6c 6520 6c65 6461 7461 202d 6d70 206d  -le ledata -mp m
-00010860: 7064 6174 6120 2d2d 6865 7572 6973 7469  pdata --heuristi
-00010870: 630a 6060 600a 0a23 2323 2077 6764 2073  c.```..### wgd s
-00010880: 796e 0a0a 2a2a 5468 6520 696e 7472 612d  yn..**The intra-
-00010890: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-000108a0: 2069 6e66 6572 656e 6365 2a2a 0a60 6060   inference**.```
-000108b0: 0a77 6764 2073 796e 2066 616d 696c 6965  .wgd syn familie
-000108c0: 7320 6766 6620 282d 2d6b 735f 6469 7374  s gff (--ks_dist
-000108d0: 7269 6275 7469 6f6e 206b 7364 6174 6120  ribution ksdata 
-000108e0: 2d66 2067 656e 6520 2d61 2049 4420 2d2d  -f gene -a ID --
-000108f0: 6d69 6e6c 656e 202d 3120 2d2d 6d69 6e73  minlen -1 --mins
-00010900: 6567 6c65 6e20 3130 3030 3020 2d2d 6d69  eglen 10000 --mi
-00010910: 6e67 656e 656e 756d 2033 3029 0a60 6060  ngenenum 30).```
-00010920: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
-00010930: 6764 2073 796e 6020 6973 206d 6169 6e6c  gd syn` is mainl
-00010940: 7920 6465 616c 696e 6720 7769 7468 2063  y dealing with c
-00010950: 6f6c 6c69 6e65 6172 6974 7920 6f72 2073  ollinearity or s
-00010960: 796e 7465 6e79 2028 626f 7468 2072 6566  ynteny (both ref
-00010970: 6572 7265 6420 746f 2061 7320 7379 6e74  erred to as synt
-00010980: 656e 7920 6865 7265 6166 7465 7229 2061  eny hereafter) a
-00010990: 6e61 6c79 7369 732e 2054 776f 2069 6e70  nalysis. Two inp
-000109a0: 7574 2066 696c 6573 2061 7265 2065 7373  ut files are ess
-000109b0: 656e 7469 616c 2c20 7468 6520 6765 6e65  ential, the gene
-000109c0: 2066 616d 696c 7920 6669 6c65 2061 6e64   family file and
-000109d0: 2074 6865 2067 6666 3320 6669 6c65 2e20   the gff3 file. 
-000109e0: 5468 6520 6765 6e65 2066 616d 696c 7920  The gene family 
-000109f0: 6669 6c65 2069 7320 696e 2074 6865 2066  file is in the f
-00010a00: 6f72 6d61 7420 6173 2060 4f72 7468 6f46  ormat as `OrthoF
-00010a10: 696e 6465 7260 2e20 5468 6520 736f 6674  inder`. The soft
-00010a20: 7761 7265 2060 692d 6164 686f 7265 6020  ware `i-adhore` 
-00010a30: 6973 2061 2070 7265 7265 7175 6973 6974  is a prerequisit
-00010a40: 652e 2057 6974 6820 6465 6661 756c 7420  e. With default 
-00010a50: 7061 7261 6d65 7465 7273 2c20 7468 6520  parameters, the 
-00010a60: 7072 6f67 7261 6d20 6261 7369 6361 6c6c  program basicall
-00010a70: 7920 636f 6e64 7563 7473 2031 2920 6669  y conducts 1) fi
-00010a80: 6c74 6572 696e 6720 6765 6e65 2066 616d  ltering gene fam
-00010a90: 696c 6965 7320 6261 7365 6420 6f6e 206d  ilies based on m
-00010aa0: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-00010ab0: 7a65 2032 2920 7265 7472 6965 7669 6e67  ze 2) retrieving
-00010ac0: 2067 656e 6520 706f 7369 7469 6f6e 2061   gene position a
-00010ad0: 6e64 2073 6361 6666 6f6c 6420 696e 666f  nd scaffold info
-00010ae0: 726d 6174 696f 6e20 6672 6f6d 2067 6666  rmation from gff
-00010af0: 3320 6669 6c65 2033 2920 7072 6f64 7563  3 file 3) produc
-00010b00: 696e 6720 7468 6520 636f 6e66 6967 7572  ing the configur
-00010b10: 6174 696f 6e20 6669 6c65 2061 6e64 2061  ation file and a
-00010b20: 7373 6f63 6961 7465 6420 6461 7461 6669  ssociated datafi
-00010b30: 6c65 7320 666f 7220 6069 2d61 6468 6f72  les for `i-adhor
-00010b40: 6560 2034 2920 6361 6c6c 696e 6720 6069  e` 4) calling `i
-00010b50: 2d61 6468 6f72 6560 2067 6976 656e 2074  -adhore` given t
-00010b60: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
-00010b70: 7420 746f 2069 6e66 6572 2073 796e 7465  t to infer synte
-00010b80: 6e79 2035 2920 7669 7375 616c 697a 696e  ny 5) visualizin
-00010b90: 6720 7468 6520 7379 6e74 656e 7920 696e  g the synteny in
-00010ba0: 2022 646f 7470 6c6f 7422 2069 6e20 7468   "dotplot" in th
-00010bb0: 6520 756e 6974 206f 6620 6765 6e65 7320  e unit of genes 
-00010bc0: 616e 6420 6261 7365 732c 2069 6e20 2253  and bases, in "S
-00010bd0: 796e 6465 7074 6822 2070 6c6f 7420 7368  yndepth" plot sh
-00010be0: 6f77 696e 6720 7468 6520 6469 7374 7269  owing the distri
-00010bf0: 6275 7469 6f6e 206f 6620 6469 6666 6572  bution of differ
-00010c00: 656e 7420 6361 7465 676f 7269 6573 206f  ent categories o
-00010c10: 6620 636f 6c6c 696e 6561 7269 7479 2072  f collinearity r
-00010c20: 6174 696f 7320 7769 7468 696e 2061 6e64  atios within and
-00010c30: 2062 6574 7765 656e 2073 7065 6369 6573   between species
-00010c40: 2c20 696e 2022 6475 7053 7461 636b 2220  , in "dupStack" 
-00010c50: 706c 6f74 2073 686f 7769 6e67 206d 756c  plot showing mul
-00010c60: 7469 706c 6963 6f6e 7320 7769 7468 2064  tiplicons with d
-00010c70: 6966 6665 7265 6e74 206d 756c 7469 706c  ifferent multipl
-00010c80: 6963 6174 696f 6e20 6c65 7665 6c73 2e20  ication levels. 
-00010c90: 3629 2069 6620 7769 7468 202a 4b2a 3c73  6) if with *K*<s
-00010ca0: 7562 3e53 3c2f 7375 623e 2064 6174 612c  ub>S</sub> data,
-00010cb0: 2061 2022 2a4b 2a3c 7375 623e 533c 2f73   a "*K*<sub>S</s
-00010cc0: 7562 3e20 646f 7470 6c6f 7422 2077 6974  ub> dotplot" wit
-00010cd0: 6820 646f 7473 2061 6e6e 6f74 6174 6564  h dots annotated
-00010ce0: 2069 6e20 2a4b 2a3c 7375 623e 533c 2f73   in *K*<sub>S</s
-00010cf0: 7562 3e20 7661 6c75 6573 2061 6e64 2061  ub> values and a
-00010d00: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00010d10: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
-00010d20: 7468 2061 6e63 686f 7220 7061 6972 7320  th anchor pairs 
-00010d30: 6465 6e6f 7465 6420 7769 6c6c 2062 6520  denoted will be 
-00010d40: 7072 6f64 7563 6564 2e20 5468 6520 6765  produced. The ge
-00010d50: 6e65 2069 6e66 6f72 6d61 7469 6f6e 2069  ne information i
-00010d60: 6e20 7468 6520 6765 6e65 2066 616d 696c  n the gene famil
-00010d70: 7920 6669 6c65 2061 6e64 2067 6666 3320  y file and gff3 
-00010d80: 6669 6c65 2073 686f 756c 6420 6265 206d  file should be m
-00010d90: 6174 6368 6564 2077 6869 6368 2072 6571  atched which req
-00010da0: 7569 7265 7320 7573 6572 7320 746f 2073  uires users to s
-00010db0: 6574 2070 726f 7065 7220 602d 2d66 6561  et proper `--fea
-00010dc0: 7475 7265 6020 616e 6420 602d 2d61 7474  ture` and `--att
-00010dd0: 7269 6275 7465 602e 2054 6865 206d 6178  ribute`. The max
-00010de0: 696d 756d 2066 616d 696c 7920 7369 7a65  imum family size
-00010df0: 2074 6f20 6265 2069 6e63 6c75 6465 6420   to be included 
-00010e00: 6361 6e20 6265 2073 6574 2076 6961 2074  can be set via t
-00010e10: 6865 206f 7074 696f 6e20 602d 2d6d 6178  he option `--max
-00010e20: 7369 7a65 602c 206e 6f74 6564 2074 6861  size`, noted tha
-00010e30: 7420 7468 6973 2066 696c 7465 7269 6e67  t this filtering
-00010e40: 2069 7320 6d61 696e 6c79 2074 6f20 6472   is mainly to dr
-00010e50: 6f70 2074 686f 7365 2068 7567 6520 7461  op those huge ta
-00010e60: 6e64 656d 2064 7570 6c69 6361 7465 7320  ndem duplicates 
-00010e70: 6661 6d69 6c79 2061 6e64 2074 7261 6e73  family and trans
-00010e80: 706f 7361 626c 6520 656c 656d 656e 7473  posable elements
-00010e90: 2028 5445 7329 2066 616d 696c 792c 2061   (TEs) family, a
-00010ea0: 6e64 206e 6f74 206d 616e 6461 746f 7279  nd not mandatory
-00010eb0: 2e20 5573 6572 7320 6361 6e20 6669 6c74  . Users can filt
-00010ec0: 6572 2074 686f 7365 2066 7261 676d 656e  er those fragmen
-00010ed0: 7461 7279 2073 6361 6666 6f6c 6473 2076  tary scaffolds v
-00010ee0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00010ef0: 2d6d 696e 6c65 6e60 2e20 5468 6520 6d69  -minlen`. The mi
-00010f00: 6e69 6d75 6d20 6c65 6e67 7468 2061 6e64  nimum length and
-00010f10: 206e 756d 6265 7220 6f66 2067 656e 6573   number of genes
-00010f20: 2066 6f72 2061 2073 6567 6d65 6e74 2074   for a segment t
-00010f30: 6f20 6265 2063 6f6e 7369 6465 7265 6420  o be considered 
-00010f40: 6361 6e20 6265 2073 6574 2076 6961 2074  can be set via t
-00010f50: 6865 206f 7074 696f 6e20 602d 2d6d 696e  he option `--min
-00010f60: 7365 676c 656e 6020 616e 6420 602d 2d6d  seglen` and `--m
-00010f70: 696e 6765 6e65 6e75 6d60 2e20 5265 6475  ingenenum`. Redu
-00010f80: 6e64 616e 7420 6d75 6c74 6970 6c69 636f  ndant multiplico
-00010f90: 6e73 2063 616e 2062 6520 6b65 7074 2062  ns can be kept b
-00010fa0: 7920 7365 7420 7468 6520 666c 6167 206f  y set the flag o
-00010fb0: 7074 696f 6e20 602d 2d6b 6565 7072 6564  ption `--keepred
-00010fc0: 756e 602e 0a0a 2a2a 4120 7375 6767 6573  un`...**A sugges
-00010fd0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
-00010fe0: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
-00010ff0: 2073 696d 706c 7920 6173 2062 656c 6f77   simply as below
-00011000: 2a2a 0a0a 6060 600a 7767 6420 7379 6e20  **..```.wgd syn 
-00011010: 6661 6d69 6c69 6573 2067 6666 0a60 6060  families gff.```
-00011020: 0a0a 2a2a 5468 6520 696e 7465 722d 7370  ..**The inter-sp
-00011030: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
-00011040: 6e66 6572 656e 6365 2a2a 0a60 6060 0a77  nference**.```.w
-00011050: 6764 2073 796e 2066 616d 696c 6965 7320  gd syn families 
-00011060: 6766 6631 2067 6666 3220 282d 2d61 6464  gff1 gff2 (--add
-00011070: 6974 696f 6e61 6c67 6666 696e 666f 2022  itionalgffinfo "
-00011080: 6d52 4e41 3b4e 616d 6522 202d 2d61 6464  mRNA;Name" --add
-00011090: 6974 696f 6e61 6c67 6666 696e 666f 2022  itionalgffinfo "
-000110a0: 6765 6e65 3b49 4422 290a 6060 600a 0a46  gene;ID").```..F
-000110b0: 6f72 206d 756c 7469 2d73 7065 6369 6573  or multi-species
-000110c0: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-000110d0: 6365 2c20 6966 2075 7365 7273 2068 6176  ce, if users hav
-000110e0: 6520 6766 6633 2066 696c 6573 2077 6869  e gff3 files whi
-000110f0: 6368 2068 6176 6520 6469 6666 6572 656e  ch have differen
-00011100: 7420 6665 6174 7572 6573 206f 7220 6174  t features or at
-00011110: 7472 6962 7574 6573 2066 6f72 2067 656e  tributes for gen
-00011120: 6520 706f 7369 7469 6f6e 2069 6e66 6f72  e position infor
-00011130: 6d61 7469 6f6e 2072 6574 7269 6576 616c  mation retrieval
-00011140: 2c20 7468 6520 6f70 7469 6f6e 2060 2d2d  , the option `--
-00011150: 6164 6469 7469 6f6e 616c 6766 6669 6e66  additionalgffinf
-00011160: 6f60 2063 616e 2062 6520 7365 7420 746f  o` can be set to
-00011170: 2070 726f 7669 6465 2074 6865 2061 6464   provide the add
-00011180: 6974 696f 6e61 6c20 696e 666f 726d 6174  itional informat
-00011190: 696f 6e2e 2054 6865 2072 656d 6169 6e69  ion. The remaini
-000111a0: 6e67 2070 6172 616d 6574 6572 2073 6574  ng parameter set
-000111b0: 7469 6e67 2069 7320 7468 6520 7361 6d65  ting is the same
-000111c0: 2061 7320 7468 6520 696e 7472 612d 7370   as the intra-sp
-000111d0: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
-000111e0: 6e66 6572 656e 6365 2e0a 0a2a 2a41 2073  nference...**A s
-000111f0: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
-00011200: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
-00011210: 6d6d 616e 6420 7369 6d70 6c79 2061 7320  mmand simply as 
-00011220: 6265 6c6f 772a 2a0a 0a60 6060 0a77 6764  below**..```.wgd
-00011230: 2073 796e 2066 616d 696c 6965 7320 6766   syn families gf
-00011240: 6631 2067 6666 320a 6060 600a 0a23 2323  f1 gff2.```..###
-00011250: 2077 6764 2076 697a 0a0a 2a2a 5468 6520   wgd viz..**The 
-00011260: 7669 7375 616c 697a 6174 696f 6e20 6f66  visualization of
-00011270: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00011280: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-00011290: 6e20 616e 6420 454c 4d4d 2061 6e61 6c79  n and ELMM analy
-000112a0: 7369 732a 2a0a 6060 600a 7767 6420 7669  sis**.```.wgd vi
-000112b0: 7a20 2d64 206b 7364 6174 610a 6060 600a  z -d ksdata.```.
-000112c0: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-000112d0: 6420 7669 7a60 2069 7320 6d61 696e 6c79  d viz` is mainly
-000112e0: 2066 6f72 2074 6865 2070 7572 706f 7365   for the purpose
-000112f0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
-00011300: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
-00011310: 2061 6e64 2073 796e 7465 6e79 2076 6973   and synteny vis
-00011320: 7561 6c69 7a61 7469 6f6e 2c20 7769 7468  ualization, with
-00011330: 2073 6f6d 6520 6f70 7469 6f6e 616c 206d   some optional m
-00011340: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00011350: 616e 616c 7973 6973 2e20 5468 6520 6261  analysis. The ba
-00011360: 7369 6320 6675 6e63 7469 6f6e 2069 7320  sic function is 
-00011370: 6a75 7374 2074 6f20 706c 6f74 2074 6865  just to plot the
-00011380: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00011390: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
-000113a0: 6420 636f 6e64 7563 7420 616e 2045 4c4d  d conduct an ELM
-000113b0: 4d20 616e 616c 7973 6973 2069 6e20 7365  M analysis in se
-000113c0: 6172 6368 206f 6620 706f 7465 6e74 6961  arch of potentia
-000113d0: 6c20 5747 4420 636f 6d70 6f6e 656e 7473  l WGD components
-000113e0: 2e20 536f 6d65 206b 6579 2070 6172 616d  . Some key param
-000113f0: 6574 6572 7320 6166 6665 6374 696e 6720  eters affecting 
-00011400: 7468 6520 454c 4d4d 2072 6573 756c 7420  the ELMM result 
-00011410: 696e 636c 7564 6520 602d 2d70 726f 6d69  include `--promi
-00011420: 6e65 6e63 655f 6375 746f 6666 602c 2060  nence_cutoff`, `
-00011430: 2d2d 7265 6c5f 6865 6967 6874 602c 2060  --rel_height`, `
-00011440: 2d2d 656d 5f69 7465 7261 7469 6f6e 7360  --em_iterations`
-00011450: 2061 6e64 2060 2d2d 656d 5f69 6e69 7469   and `--em_initi
-00011460: 616c 697a 6174 696f 6e73 602c 2061 6c6c  alizations`, all
-00011470: 206f 6620 7768 6963 6820 6861 7665 2062   of which have b
-00011480: 6565 6e20 6578 706c 6169 6e65 6420 6168  een explained ah
-00011490: 6561 642e 0a0a 2a2a 4120 7375 6767 6573  ead...**A sugges
-000114a0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
-000114b0: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
-000114c0: 2073 696d 706c 7920 6173 2062 656c 6f77   simply as below
-000114d0: 2a2a 0a0a 6060 600a 7767 6420 7669 7a20  **..```.wgd viz 
-000114e0: 2d64 206b 7364 6174 610a 6060 600a 0a2a  -d ksdata.```..*
-000114f0: 2a54 6865 2076 6973 7561 6c69 7a61 7469  *The visualizati
-00011500: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
-00011510: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00011520: 6275 7469 6f6e 2077 6974 6820 7261 7465  bution with rate
-00011530: 2063 6f72 7265 6374 696f 6e2a 2a0a 6060   correction**.``
-00011540: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
-00011550: 6174 6120 2d73 7020 7370 6461 7461 202d  ata -sp spdata -
-00011560: 2d66 6f63 7573 3261 6c6c 2066 6f63 616c  -focus2all focal
-00011570: 5f73 7065 6369 6573 202d 2d65 7874 7261  _species --extra
-00011580: 7061 7261 6e6f 6d65 6b73 206b 7364 6174  paranomeks ksdat
-00011590: 6120 282d 2d61 6e63 686f 7270 6f69 6e74  a (--anchorpoint
-000115a0: 7320 6170 6461 7461 202d 2d70 6c6f 7461  s apdata --plota
-000115b0: 7067 6d6d 202d 2d70 6c6f 7465 6c6d 6d29  pgmm --plotelmm)
-000115c0: 0a60 6060 0a0a 4265 7369 6465 7320 7468  .```..Besides th
-000115d0: 6520 6261 7369 6320 2a4b 2a3c 7375 623e  e basic *K*<sub>
-000115e0: 533c 2f73 7562 3e20 706c 6f74 2c20 7375  S</sub> plot, su
-000115f0: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
-00011600: 636f 7272 6563 7469 6f6e 2063 616e 2061  correction can a
-00011610: 6c73 6f20 6265 2061 6368 6965 7665 6420  lso be achieved 
-00011620: 6769 7665 6e20 6174 206c 6561 7374 2061  given at least a
-00011630: 2073 7065 6369 6573 2074 7265 6520 2876   species tree (v
-00011640: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00011650: 2d73 7065 6369 6573 7472 6565 6029 2061  -speciestree`) a
-00011660: 6e64 2061 2066 6f63 616c 2073 7065 6369  nd a focal speci
-00011670: 6573 2028 6569 7468 6572 2076 6961 2074  es (either via t
-00011680: 6865 206f 7074 696f 6e20 602d 2d66 6f63  he option `--foc
-00011690: 7573 3261 6c6c 6020 6f72 2076 6961 2074  us2all` or via t
-000116a0: 6865 206f 7074 696f 6e20 602d 2d73 7061  he option `--spa
-000116b0: 6972 6020 696e 2074 6865 2066 6f72 6d20  ir` in the form 
-000116c0: 6f66 2022 2466 6f63 616c 5f73 7065 6369  of "$focal_speci
-000116d0: 6573 3b24 666f 6361 6c5f 7370 6563 6965  es;$focal_specie
-000116e0: 7322 292e 2049 7427 7320 7375 6767 6573  s"). It's sugges
-000116f0: 7465 6420 7468 6174 2074 6865 206f 7274  ted that the ort
-00011700: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00011710: 3e53 3c2f 7375 623e 2064 6174 6120 6973  >S</sub> data is
-00011720: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
-00011730: 2060 2d2d 6461 7461 6669 6c65 6020 6f70   `--datafile` op
-00011740: 7469 6f6e 2061 6e64 2074 6865 2070 6172  tion and the par
-00011750: 616c 6f67 6f75 7320 2a4b 2a3c 7375 623e  alogous *K*<sub>
-00011760: 533c 2f73 7562 3e20 6461 7461 2069 7320  S</sub> data is 
-00011770: 7072 6f76 6964 6564 2062 7920 7468 6520  provided by the 
-00011780: 602d 2d65 7874 7261 7061 7261 6e6f 6d65  `--extraparanome
-00011790: 6b73 6020 6f70 7469 6f6e 2c20 616c 7468  ks` option, alth
-000117a0: 6f75 6768 2069 7427 7320 616c 6c6f 7765  ough it's allowe
-000117b0: 6420 746f 206f 6e6c 7920 7072 6f76 6964  d to only provid
-000117c0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000117d0: 3e20 6461 7461 2076 6961 2074 6865 2060  > data via the `
-000117e0: 2d2d 6461 7461 6669 6c65 6020 6f70 7469  --datafile` opti
-000117f0: 6f6e 2061 6e64 2064 6570 6f73 6974 2062  on and deposit b
-00011800: 6f74 6820 6f72 7468 6f6c 6f67 6f75 7326  oth orthologous&
-00011810: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-00011820: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
-00011830: 7468 6572 656f 6e2e 2054 6865 7265 2061  thereon. There a
-00011840: 7265 2074 776f 2074 7970 6573 206f 6620  re two types of 
-00011850: 6d69 7865 6420 706c 6f74 7320 2874 6865  mixed plots (the
-00011860: 2022 6d69 7865 6422 2068 6572 6520 7265   "mixed" here re
-00011870: 6665 7273 2074 6f20 6d69 7865 6420 6f72  fers to mixed or
-00011880: 7468 6f6c 6f67 6f75 7320 616e 6420 7061  thologous and pa
-00011890: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-000118a0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-000118b0: 7574 696f 6e73 292c 206f 6e65 206f 6620  utions), one of 
-000118c0: 7768 6963 6820 6973 2073 696d 696c 6172  which is similar
-000118d0: 2074 6f20 7768 6174 2060 6b73 7261 7465   to what `ksrate
-000118e0: 7360 2070 6c6f 7473 2c20 7768 696c 6520  s` plots, while 
-000118f0: 7468 6520 6f74 6865 7220 6f66 2077 6869  the other of whi
-00011900: 6368 2069 7320 6c69 6b65 2074 6865 2063  ch is like the c
-00011910: 6f6e 7665 6e74 696f 6e61 6c20 2a4b 2a3c  onventional *K*<
-00011920: 7375 623e 533c 2f73 7562 3e20 706c 6f74  sub>S</sub> plot
-00011930: 2074 6861 7420 626f 7468 2074 6865 206f   that both the o
-00011940: 7269 6769 6e61 6c20 6f72 7468 6f6c 6f67  riginal ortholog
-00011950: 6f75 7320 616e 6420 7061 7261 6c6f 676f  ous and paralogo
-00011960: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-00011970: 623e 2064 6973 7472 6962 7574 696f 6e73  b> distributions
-00011980: 2061 7265 2074 7275 7468 6675 6c6c 7920   are truthfully 
-00011990: 706c 6f74 7465 6420 696e 7374 6561 6420  plotted instead 
-000119a0: 6f66 206a 7573 7420 6265 696e 6720 7265  of just being re
-000119b0: 7072 6573 656e 7465 6420 6279 2073 6f6d  presented by som
-000119c0: 6520 7665 7274 6963 616c 206c 696e 6573  e vertical lines
-000119d0: 2e20 5765 2073 7567 6765 7374 2075 7365  . We suggest use
-000119e0: 7273 2074 6f20 6164 6f70 7420 7468 6520  rs to adopt the 
-000119f0: 606b 7372 6174 6573 602d 6c69 6b65 2070  `ksrates`-like p
-00011a00: 6c6f 7473 2c20 7768 6963 6820 6973 2074  lots, which is t
-00011a10: 6865 2064 6566 6175 6c74 2e20 4f74 6865  he default. Othe
-00011a20: 7277 6973 6520 7468 6520 6f70 7469 6f6e  rwise the option
-00011a30: 2060 2d2d 636c 6173 7369 6360 2077 696c   `--classic` wil
-00011a40: 6c20 6265 206e 6565 6465 6420 746f 2073  l be needed to s
-00011a50: 6574 2e20 4578 7472 6120 6d69 7874 7572  et. Extra mixtur
-00011a60: 6520 6d6f 6465 6c69 6e67 2061 6e61 6c79  e modeling analy
-00011a70: 7369 7320 6361 6e20 6265 2069 6e69 7469  sis can be initi
-00011a80: 6174 6564 2076 6961 2074 6865 206f 7074  ated via the opt
-00011a90: 696f 6e20 602d 2d70 6c6f 7465 6c6d 6d60  ion `--plotelmm`
-00011aa0: 2061 6e64 2060 2d2d 706c 6f74 6170 676d   and `--plotapgm
-00011ab0: 6d60 2077 6974 6820 7468 6520 616e 6368  m` with the anch
-00011ac0: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
-00011ad0: 6c65 2070 726f 7669 6465 6420 6279 2074  le provided by t
-00011ae0: 6865 2060 2d2d 616e 6368 6f72 706f 696e  he `--anchorpoin
-00011af0: 7473 6020 6f70 7469 6f6e 2e0a 0a2a 2a41  ts` option...**A
-00011b00: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
-00011b10: 696e 6720 7275 6e20 6361 6e20 7573 6520  ing run can use 
-00011b20: 636f 6d6d 616e 6420 7369 6d70 6c79 2061  command simply a
-00011b30: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-00011b40: 6764 2076 697a 202d 6420 6b73 6461 7461  gd viz -d ksdata
-00011b50: 202d 7370 2073 7064 6174 6120 2d2d 666f   -sp spdata --fo
-00011b60: 6375 7332 616c 6c20 666f 6361 6c5f 7370  cus2all focal_sp
-00011b70: 6563 6965 7320 2d2d 6578 7472 6170 6172  ecies --extrapar
-00011b80: 616e 6f6d 656b 7320 6b73 6461 7461 0a60  anomeks ksdata.`
-00011b90: 6060 0a0a 2a2a 5468 6520 7669 7375 616c  ``..**The visual
-00011ba0: 697a 6174 696f 6e20 6f66 2073 796e 7465  ization of synte
-00011bb0: 6e79 2a2a 0a60 6060 0a77 6764 2076 697a  ny**.```.wgd viz
-00011bc0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
-00011bd0: 736d 6461 7461 202d 6d74 206d 7464 6174  smdata -mt mtdat
-00011be0: 6120 2d67 7420 6774 6461 7461 202d 2d70  a -gt gtdata --p
-00011bf0: 6c6f 7473 796e 2028 2d2d 6d69 6e6c 656e  lotsyn (--minlen
-00011c00: 202d 3120 2d2d 6d69 6e73 6567 6c65 6e20   -1 --minseglen 
-00011c10: 3130 3030 3020 2d2d 6d69 6e67 656e 656e  10000 --mingenen
-00011c20: 756d 2033 3029 0a60 6060 0a0a 436f 6d70  um 30).```..Comp
-00011c30: 6172 6564 2074 6f20 7468 6520 6f72 6967  ared to the orig
-00011c40: 696e 616c 2060 7767 6460 2c20 7468 6520  inal `wgd`, the 
-00011c50: 6077 6764 2076 697a 6020 7072 6f67 7261  `wgd viz` progra
-00011c60: 6d20 6164 6465 6420 7379 6e74 656e 7920  m added synteny 
-00011c70: 7669 7375 616c 697a 6174 696f 6e20 7069  visualization pi
-00011c80: 7065 6c69 6e65 2e20 5573 6572 7320 6e65  peline. Users ne
-00011c90: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
-00011ca0: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
-00011cb0: 2d70 6c6f 7473 796e 6020 746f 2069 6e69  -plotsyn` to ini
-00011cc0: 7469 6174 6520 7468 6973 2070 6172 7420  tiate this part 
-00011cd0: 6f66 2070 6970 656c 696e 652e 2054 6869  of pipeline. Thi
-00011ce0: 7320 7374 6570 2061 7373 756d 6573 2074  s step assumes t
-00011cf0: 6861 7420 7573 6572 7320 6861 7665 206f  hat users have o
-00011d00: 6274 6169 6e65 6420 616c 7265 6164 7920  btained already 
-00011d10: 7468 6520 7379 6e74 656e 6963 2072 6573  the syntenic res
-00011d20: 756c 7473 2066 726f 6d20 6069 2d61 6468  ults from `i-adh
-00011d30: 6f72 6560 2061 6e64 2075 7365 7320 7468  ore` and uses th
-00011d40: 6f73 6520 7265 7375 6c74 2066 696c 6573  ose result files
-00011d50: 2074 6f20 7265 616c 697a 6520 7468 6520   to realize the 
-00011d60: 7379 6e74 656e 7920 7669 7375 616c 697a  synteny visualiz
-00011d70: 6174 696f 6e2e 2053 696d 6c69 6172 2074  ation. Simliar t
-00011d80: 6f20 6077 6764 2073 796e 602c 2074 6865  o `wgd syn`, the
-00011d90: 2065 7874 7261 202a 4b2a 3c73 7562 3e53   extra *K*<sub>S
-00011da0: 3c2f 7375 623e 2064 6174 6120 6669 6c65  </sub> data file
-00011db0: 2063 616e 2062 6520 7472 616e 736d 6974   can be transmit
-00011dc0: 7465 6420 7669 6120 7468 6520 6f70 7469  ted via the opti
-00011dd0: 6f6e 2060 2d2d 6461 7461 6669 6c65 6020  on `--datafile` 
-00011de0: 2869 6e73 7465 6164 206f 6620 602d 2d65  (instead of `--e
-00011df0: 7874 7261 7061 7261 6e6f 6d65 6b73 6020  xtraparanomeks` 
-00011e00: 6f70 7469 6f6e 292e 2042 6173 6963 616c  option). Basical
-00011e10: 6c79 2c20 7468 6520 7379 6e74 656e 6963  ly, the syntenic
-00011e20: 2072 6573 756c 7420 6669 6c65 7320 7265   result files re
-00011e30: 7175 6972 6564 2061 7265 2074 6865 2061  quired are the a
-00011e40: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
-00011e50: 6166 696c 652c 206d 756c 7469 706c 6963  afile, multiplic
-00011e60: 6f6e 7320 6461 7461 6669 6c65 2c20 6765  ons datafile, ge
-00011e70: 6e65 2074 6162 6c65 2064 6174 6166 696c  ne table datafil
-00011e80: 6520 2861 7574 6f6d 6174 656c 7920 7072  e (automately pr
-00011e90: 6f64 7563 6564 2062 7920 6077 6764 2073  oduced by `wgd s
-00011ea0: 796e 6029 2c20 616e 6420 7365 676d 656e  yn`), and segmen
-00011eb0: 7473 2064 6174 6166 696c 652e 0a0a 2a2a  ts datafile...**
-00011ec0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-00011ed0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-00011ee0: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
-00011ef0: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
-00011f00: 7767 6420 7669 7a20 2d61 7020 6170 6461  wgd viz -ap apda
-00011f10: 7461 202d 736d 2073 6d64 6174 6120 2d6d  ta -sm smdata -m
-00011f20: 7420 6d74 6461 7461 202d 6774 2067 7464  t mtdata -gt gtd
-00011f30: 6174 6120 2d2d 706c 6f74 7379 6e0a 6060  ata --plotsyn.``
-00011f40: 600a 0a23 2320 496c 6c75 7374 7261 7469  `..## Illustrati
-00011f50: 6f6e 0a0a 5765 2069 6c6c 7573 7472 6174  on..We illustrat
-00011f60: 6520 6f75 7220 7072 6f67 7261 6d20 6f6e  e our program on
-00011f70: 2061 6e20 6578 656d 706c 6172 7920 5747   an exemplary WG
-00011f80: 4420 696e 6665 7265 6e63 6520 616e 6420  D inference and 
-00011f90: 6461 7469 6e67 2075 706f 6e20 7370 6563  dating upon spec
-00011fa0: 6965 7320 2a41 7175 696c 6567 6961 2063  ies *Aquilegia c
-00011fb0: 6f65 7275 6c65 612a 2e0a 0a54 6865 202a  oerulea*...The *
-00011fc0: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00011fd0: 6561 2a20 7761 7320 7265 706f 7274 6564  ea* was reported
-00011fe0: 2074 6f20 6578 7065 7269 656e 6365 2061   to experience a
-00011ff0: 6e20 7061 6c65 6f2d 706f 6c79 706c 6f69  n paleo-polyploi
-00012000: 6469 7a61 7469 6f6e 2065 7665 6e74 2061  dization event a
-00012010: 6674 6572 2074 6865 2064 6976 6572 6765  fter the diverge
-00012020: 6e63 6520 6f66 2063 6f72 6520 6575 6469  nce of core eudi
-00012030: 636f 7473 2c20 7768 6963 6820 6973 206c  cots, which is l
-00012040: 696b 656c 7920 7368 6172 6564 2062 7920  ikely shared by 
-00012050: 616c 6c20 5261 6e75 6e63 756c 616c 6573  all Ranunculales
-00012060: 2e0a 0a46 6972 7374 2061 626f 7665 2061  ...First above a
-00012070: 6c6c 2c20 6c65 7427 7320 6465 6c69 6e65  ll, let's deline
-00012080: 6174 6520 7468 6520 7768 6f6c 6520 7061  ate the whole pa
-00012090: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-000120a0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-000120b0: 6962 7574 696f 6e20 616e 6420 6861 7665  ibution and have
-000120c0: 2061 2062 6173 6963 206f 6273 6572 7661   a basic observa
-000120d0: 7469 6f6e 2066 6f72 2070 6f74 656e 7469  tion for potenti
-000120e0: 616c 6c79 2063 6f6e 6365 6976 6162 6c65  ally conceivable
-000120f0: 2057 4744 732c 2075 7369 6e67 2074 6865   WGDs, using the
-00012100: 2063 6f6d 6d61 6e64 206c 696e 6520 6265   command line be
-00012110: 6c6f 772e 0a0a 6060 600a 7767 6420 646d  low...```.wgd dm
-00012120: 6420 4171 7569 6c65 6769 615f 636f 6572  d Aquilegia_coer
-00012130: 756c 6561 0a77 6764 206b 7364 2077 6764  ulea.wgd ksd wgd
-00012140: 5f64 6d64 2f41 7175 696c 6567 6961 5f63  _dmd/Aquilegia_c
-00012150: 6f65 7275 6c65 612e 7473 7620 4171 7569  oerulea.tsv Aqui
-00012160: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
-00012170: 6060 0a0a 5468 6520 636f 6e73 7472 7563  ``..The construc
-00012180: 7465 6420 7768 6f6c 6520 7061 7261 6e6f  ted whole parano
-00012190: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-000121a0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000121b0: 696f 6e20 6f66 202a 4171 7569 6c65 6769  ion of *Aquilegi
-000121c0: 6120 636f 6572 756c 6561 2a20 6973 2061  a coerulea* is a
-000121d0: 7320 6265 6c6f 772c 2077 6520 6361 6e20  s below, we can 
-000121e0: 7365 6520 7468 6174 2074 6865 7265 2073  see that there s
-000121f0: 6565 6d73 2074 6f20 6265 2061 2068 756d  eems to be a hum
-00012200: 7020 6174 202a 4b2a 3c73 7562 3e53 3c2f  p at *K*<sub>S</
-00012210: 7375 623e 2031 2062 7574 206e 6f74 2063  sub> 1 but not c
-00012220: 6c65 6172 2e0a 0a21 5b5d 2864 6174 612f  lear...![](data/
-00012230: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012240: 6561 2e74 7376 2e6b 7364 5f77 702e 7376  ea.tsv.ksd_wp.sv
-00012250: 6729 0a0a 5765 2074 6865 6e20 636f 6e73  g)..We then cons
-00012260: 7472 7563 7420 7468 6520 616e 6368 6f72  truct the anchor
-00012270: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012280: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-00012290: 6e20 7573 696e 6720 7468 6520 636f 6d6d  n using the comm
-000122a0: 616e 6420 6c69 6e65 2062 656c 6f77 2e0a  and line below..
-000122b0: 0a60 6060 0a77 6764 2073 796e 202d 6620  .```.wgd syn -f 
-000122c0: 6d52 4e41 202d 6120 4e61 6d65 2077 6764  mRNA -a Name wgd
-000122d0: 5f64 6d64 2f41 7175 696c 6567 6961 5f63  _dmd/Aquilegia_c
-000122e0: 6f65 7275 6c65 612e 7473 7620 4171 7569  oerulea.tsv Aqui
-000122f0: 6c65 6769 615f 636f 6572 756c 6561 2e67  legia_coerulea.g
-00012300: 6666 3320 2d6b 7320 7767 645f 6b73 642f  ff3 -ks wgd_ksd/
-00012310: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012320: 6561 2e74 7376 2e6b 732e 7473 760a 6060  ea.tsv.ks.tsv.``
-00012330: 600a 0a41 7320 7368 6f77 6e20 6265 6c6f  `..As shown belo
-00012340: 772c 2074 6865 7265 2061 7265 2073 6f6d  w, there are som
-00012350: 6520 7265 7461 696e 6564 2061 6e63 686f  e retained ancho
-00012360: 7220 7061 6972 7320 7769 7468 202a 4b2a  r pairs with *K*
-00012370: 3c73 7562 3e53 3c2f 7375 623e 2062 6574  <sub>S</sub> bet
-00012380: 7765 656e 2031 2061 6e64 2032 2c20 7768  ween 1 and 2, wh
-00012390: 6963 6820 7365 656d 7320 746f 2073 7567  ich seems to sug
-000123a0: 6765 7374 2061 2057 4744 2065 7665 6e74  gest a WGD event
-000123b0: 2e0a 0a21 5b5d 2864 6174 612f 4171 7569  ...![](data/Aqui
-000123c0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-000123d0: 7376 2e6b 7364 5f77 705f 6170 2e73 7667  sv.ksd_wp_ap.svg
-000123e0: 290a 0a54 6865 2061 7373 6f63 6961 7465  )..The associate
-000123f0: 6420 6064 7570 5374 6163 6b60 2070 6c6f  d `dupStack` plo
-00012400: 7420 7368 6f77 7320 7468 6174 2074 6865  t shows that the
-00012410: 7265 2061 7265 206e 756d 6572 6f75 7320  re are numerous 
-00012420: 6475 706c 6963 6174 6564 2073 6567 6d65  duplicated segme
-00012430: 6e74 7320 6163 726f 7373 206d 6f73 7420  nts across most 
-00012440: 6f66 2074 6865 2063 6872 6f6d 6f73 6f6d  of the chromosom
-00012450: 6573 2e0a 0a21 5b5d 2864 6174 612f 4171  es...![](data/Aq
-00012460: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012470: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
-00012480: 6c65 615f 6d75 6c74 6970 6c69 636f 6e73  lea_multiplicons
-00012490: 5f6c 6576 656c 2e73 7667 290a 0a57 6520  _level.svg)..We 
-000124a0: 696d 706c 656d 656e 7465 6420 7477 6f20  implemented two 
-000124b0: 7479 7065 7320 6f66 2064 6f74 2070 6c6f  types of dot plo
-000124c0: 7473 2069 6e20 6f78 666f 7264 2067 7269  ts in oxford gri
-000124d0: 643a 206f 6e65 2069 6e20 7468 6520 756e  d: one in the un
-000124e0: 6974 206f 6620 6261 7365 7320 616e 6420  it of bases and 
-000124f0: 7468 6520 6f74 6865 7220 696e 2074 6865  the other in the
-00012500: 2075 6e69 7420 6f66 2067 656e 6573 2c20   unit of genes, 
-00012510: 7768 6963 6820 6361 6e20 6265 2063 6f6c  which can be col
-00012520: 6f72 6564 2062 7920 2a4b 2a3c 7375 623e  ored by *K*<sub>
-00012530: 533c 2f73 7562 3e20 7661 6c75 6573 2067  S</sub> values g
-00012540: 6976 656e 202a 4b2a 3c73 7562 3e53 3c2f  iven *K*<sub>S</
-00012550: 7375 623e 2064 6174 612e 0a0a 215b 5d28  sub> data...![](
-00012560: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
-00012570: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
-00012580: 6567 6961 5f63 6f65 7275 6c65 615f 4b73  egia_coerulea_Ks
-00012590: 2e64 6f74 5f75 6e69 745f 6765 6e65 2e70  .dot_unit_gene.p
-000125a0: 6e67 290a 0a41 7320 7368 6f77 6e20 6162  ng)..As shown ab
-000125b0: 6f76 652c 2074 6865 2064 6f74 2070 6c6f  ove, the dot plo
-000125c0: 7420 696e 2074 6865 2075 6e69 7420 6f66  t in the unit of
-000125d0: 2067 656e 6573 2070 7265 7365 6e74 7320   genes presents 
-000125e0: 6e75 6d65 726f 7573 2064 656e 7365 6c79  numerous densely
-000125f0: 2061 6767 7265 6761 7465 6420 286c 696e   aggregated (lin
-00012600: 652d 6c69 6b65 2920 616e 6368 6f72 2070  e-like) anchor p
-00012610: 6f69 6e74 7320 6174 206d 6f73 7420 6f66  oints at most of
-00012620: 2074 6865 2063 6872 6f6d 6f73 6f6d 6573   the chromosomes
-00012630: 2077 6974 6820 636f 6e73 6973 7465 6e74   with consistent
-00012640: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012650: 2061 6765 2062 6574 7765 656e 2031 2061   age between 1 a
-00012660: 6e64 2032 2e20 5468 6520 646f 7420 706c  nd 2. The dot pl
-00012670: 6f74 2069 6e20 7468 6520 756e 6974 206f  ot in the unit o
-00012680: 6620 6261 7365 7320 7368 6f77 7320 7468  f bases shows th
-00012690: 6520 7361 6d65 2070 6174 7465 726e 2c20  e same pattern, 
-000126a0: 6173 206d 616e 6966 6573 7465 6420 6265  as manifested be
-000126b0: 6c6f 772e 0a0a 215b 5d28 6461 7461 2f41  low...![](data/A
-000126c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000126d0: 612d 7673 2d41 7175 696c 6567 6961 5f63  a-vs-Aquilegia_c
-000126e0: 6f65 7275 6c65 615f 4b73 2e64 6f74 2e70  oerulea_Ks.dot.p
-000126f0: 6e67 290a 0a54 6865 2064 6f74 2070 6c6f  ng)..The dot plo
-00012700: 7473 2077 6974 686f 7574 202a 4b2a 3c73  ts without *K*<s
-00012710: 7562 3e53 3c2f 7375 623e 2061 6e6e 6f74  ub>S</sub> annot
-00012720: 6174 696f 6e20 7769 6c6c 2061 6c73 6f20  ation will also 
-00012730: 6265 2061 7574 6f6d 6174 656c 7920 7072  be automately pr
-00012740: 6f64 7563 6564 2c20 6173 2073 686f 776e  oduced, as shown
-00012750: 2062 656c 6f77 2e0a 0a21 5b5d 2864 6174   below...![](dat
-00012760: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00012770: 756c 6561 2d76 732d 4171 7569 6c65 6769  ulea-vs-Aquilegi
-00012780: 615f 636f 6572 756c 6561 2e64 6f74 5f75  a_coerulea.dot_u
-00012790: 6e69 745f 6765 6e65 2e70 6e67 290a 0a21  nit_gene.png)..!
-000127a0: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-000127b0: 615f 636f 6572 756c 6561 2d76 732d 4171  a_coerulea-vs-Aq
-000127c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000127d0: 2e64 6f74 2e70 6e67 290a 0a4e 6f74 6520  .dot.png)..Note 
-000127e0: 7468 6174 2074 6865 206f 7061 6369 7479  that the opacity
-000127f0: 206f 6620 616e 6368 6f72 2064 6f74 7320   of anchor dots 
-00012800: 616e 6420 616c 6c20 686f 6d6f 6c6f 6720  and all homolog 
-00012810: 646f 7473 2063 616e 2062 6520 7365 7420  dots can be set 
-00012820: 6279 2074 6865 206f 7074 696f 6e20 6061  by the option `a
-00012830: 7061 6c70 6861 6020 616e 6420 6068 6f61  palpha` and `hoa
-00012840: 6c70 6861 6020 7365 7061 7261 7465 6c79  lpha` separately
-00012850: 2e20 4966 206f 6e65 206a 7573 7420 7761  . If one just wa
-00012860: 6e74 7320 746f 2073 6565 2074 6865 2061  nts to see the a
-00012870: 6e63 686f 7220 646f 7473 2c20 7365 7474  nchor dots, sett
-00012880: 696e 6720 7468 6520 6068 6f61 6c70 6861  ing the `hoalpha
-00012890: 6020 6173 2030 2028 6f72 206f 7468 6572  ` as 0 (or other
-000128a0: 206d 696e 7573 6375 6c65 2076 616c 7565   minuscule value
-000128b0: 7329 2077 696c 6c20 646f 2e20 4966 206f  s) will do. If o
-000128c0: 6e65 2077 616e 7473 2074 6f20 7365 6520  ne wants to see 
-000128d0: 7468 6520 6469 7374 7269 6275 7469 6f6e  the distribution
-000128e0: 206f 6620 7768 6f6c 6520 646f 7473 2062   of whole dots b
-000128f0: 6574 7465 722c 2073 6574 7469 6e67 2074  etter, setting t
-00012900: 6865 2060 686f 616c 7068 6160 2068 6967  he `hoalpha` hig
-00012910: 6865 7220 2861 6e64 2060 6170 616c 7068  her (and `apalph
-00012920: 6160 206c 6f77 6572 2920 7769 6c6c 2064  a` lower) will d
-00012930: 6f2e 2054 6865 2060 646f 7473 697a 6560  o. The `dotsize`
-00012940: 206f 7074 696f 6e20 6361 6e20 6265 2063   option can be c
-00012950: 616c 6c65 6420 746f 2061 646a 7573 7420  alled to adjust 
-00012960: 7468 6520 7369 7a65 206f 6620 646f 7473  the size of dots
-00012970: 2e0a 0a41 2066 7572 7468 6572 2061 7373  ...A further ass
-00012980: 6f63 6961 7465 6420 5379 6e64 6570 7468  ociated Syndepth
-00012990: 2070 6c6f 7420 7368 6f77 7320 7468 6174   plot shows that
-000129a0: 2074 6865 7265 2061 7265 206d 6f72 6520   there are more 
-000129b0: 7468 616e 2038 3020 6475 706c 6963 6174  than 80 duplicat
-000129c0: 6564 2073 6567 6d65 6e74 732c 2077 6869  ed segments, whi
-000129d0: 6368 2064 6f6d 696e 6174 6573 2074 6865  ch dominates the
-000129e0: 2077 686f 6c65 2063 6f6c 6c69 6e65 6172   whole collinear
-000129f0: 2072 6174 696f 2063 6174 6567 6f72 792e   ratio category.
-00012a00: 0a0a 215b 5d28 6461 7461 2f53 796e 6465  ..![](data/Synde
-00012a10: 7074 682e 7376 6729 0a0a 5765 2063 616e  pth.svg)..We can
-00012a20: 2066 6974 2061 6e20 454c 4d4d 206d 6978   fit an ELMM mix
-00012a30: 7475 7265 206d 6f64 656c 2075 706f 6e20  ture model upon 
-00012a40: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
-00012a50: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-00012a60: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00012a70: 696f 6e20 746f 2073 6565 206d 6f72 6520  ion to see more 
-00012a80: 6163 6375 7261 7465 6c79 2074 6865 2073  accurately the s
-00012a90: 6967 6e69 6669 6361 6e63 6520 616e 6420  ignificance and 
-00012aa0: 6c6f 6361 7469 6f6e 206f 6620 706f 7465  location of pote
-00012ab0: 6e74 6961 6c20 5747 4473 2c20 7573 696e  ntial WGDs, usin
-00012ac0: 6720 7468 6520 636f 6d6d 616e 6420 6c69  g the command li
-00012ad0: 6e65 2062 656c 6f77 2e0a 0a60 6060 0a77  ne below...```.w
-00012ae0: 6764 2076 697a 202d 6420 7767 645f 6b73  gd viz -d wgd_ks
-00012af0: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00012b00: 756c 6561 2e74 7376 2e6b 732e 7473 760a  ulea.tsv.ks.tsv.
-00012b10: 6060 600a 0a54 6865 2072 6573 756c 7420  ```..The result 
-00012b20: 6f66 2045 4c4d 4d20 6d69 7874 7572 6520  of ELMM mixture 
-00012b30: 6d6f 6465 6c20 636c 7573 7465 7269 6e67  model clustering
-00012b40: 2073 686f 7773 2074 6861 7420 7468 6572   shows that ther
-00012b50: 6520 6973 2061 206c 696b 656c 7920 5747  e is a likely WG
-00012b60: 4420 636f 6d70 6f6e 656e 7420 6174 202a  D component at *
-00012b70: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2031  K*<sub>S</sub> 1
-00012b80: 2e31 392e 0a0a 215b 5d28 6461 7461 2f65  .19...![](data/e
-00012b90: 6c6d 6d5f 4171 7569 6c65 6769 615f 636f  lmm_Aquilegia_co
-00012ba0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00012bb0: 765f 6265 7374 5f6d 6f64 656c 735f 7765  v_best_models_we
-00012bc0: 6967 6874 6564 2e73 7667 290a 0a4c 6574  ighted.svg)..Let
-00012bd0: 2773 2064 6f20 6120 6d69 7874 7572 6520  's do a mixture 
-00012be0: 6d6f 6465 6c20 636c 7573 7465 7269 6e67  model clustering
-00012bf0: 2066 6f72 2061 6e63 686f 7220 2a4b 2a3c   for anchor *K*<
-00012c00: 7375 623e 533c 2f73 7562 3e20 746f 6f2c  sub>S</sub> too,
-00012c10: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00012c20: 6e64 206c 696e 6520 6265 6c6f 772e 0a0a  nd line below...
-00012c30: 6060 600a 7767 6420 7065 616b 2077 6764  ```.wgd peak wgd
-00012c40: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
-00012c50: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
-00012c60: 7376 202d 2d61 6e63 686f 7270 6f69 6e74  sv --anchorpoint
-00012c70: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
-00012c80: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00012c90: 7473 2e74 7874 202d 2d73 6567 6d65 6e74  ts.txt --segment
-00012ca0: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
-00012cb0: 652d 6f75 742f 7365 676d 656e 7473 2e74  e-out/segments.t
-00012cc0: 7874 202d 2d6c 6973 7465 6c65 6d65 6e74  xt --listelement
-00012cd0: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
-00012ce0: 652d 6f75 742f 6c69 7374 5f65 6c65 6d65  e-out/list_eleme
-00012cf0: 6e74 732e 7478 7420 2d2d 6d75 6c74 6970  nts.txt --multip
-00012d00: 6c69 636f 6e70 6169 7273 2077 6764 5f73  liconpairs wgd_s
-00012d10: 796e 2f69 6164 686f 7265 2d6f 7574 2f6d  yn/iadhore-out/m
-00012d20: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
-00012d30: 2e74 7874 202d 2d77 6569 6768 7465 640a  .txt --weighted.
-00012d40: 6060 600a 0a54 6865 2061 6e63 686f 7220  ```..The anchor 
-00012d50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00012d60: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-00012d70: 2061 6c73 6f20 6861 7320 6120 6c69 6b65   also has a like
-00012d80: 6c79 2057 4744 2063 6f6d 706f 6e65 6e74  ly WGD component
-00012d90: 2077 6974 6820 6d6f 6465 2031 2e32 382e   with mode 1.28.
-00012da0: 0a0a 215b 5d28 6461 7461 2f4f 7269 6769  ..![](data/Origi
-00012db0: 6e61 6c5f 416e 6368 6f72 4b73 5f47 4d4d  nal_AnchorKs_GMM
-00012dc0: 5f43 6f6d 706f 6e65 6e74 335f 6e6f 6465  _Component3_node
-00012dd0: 5f77 6569 6768 7465 645f 4c6f 676e 6f72  _weighted_Lognor
-00012de0: 6d61 6c2e 7376 6729 0a0a 4e6f 7720 7468  mal.svg)..Now th
-00012df0: 6174 2077 6520 6861 7665 2073 6565 6e20  at we have seen 
-00012e00: 7468 6520 6576 6964 656e 6365 206f 6620  the evidence of 
-00012e10: 6e75 6d65 726f 7573 2064 7570 6c69 6361  numerous duplica
-00012e20: 7465 6420 7365 676d 656e 7473 2061 6e64  ted segments and
-00012e30: 2074 6865 2061 6767 7265 6761 7469 6f6e   the aggregation
-00012e40: 206f 6620 6475 706c 6963 6174 6573 2061   of duplicates a
-00012e50: 6765 2069 6e20 2a4b 2a3c 7375 623e 533c  ge in *K*<sub>S<
-00012e60: 2f73 7562 3e20 6172 6f75 6e64 2031 2e32  /sub> around 1.2
-00012e70: 2066 6f72 2061 6e63 686f 7220 7061 6972   for anchor pair
-00012e80: 7320 616e 6420 6e6f 6e2d 616e 6368 6f72  s and non-anchor
-00012e90: 2070 6169 7273 2074 6872 6f75 6768 6f75   pairs throughou
-00012ea0: 7420 7468 6520 7768 6f6c 6520 6765 6e6f  t the whole geno
-00012eb0: 6d65 2e20 5765 2063 616e 2063 6c61 696d  me. We can claim
-00012ec0: 2077 6974 6820 736f 6d65 2063 6f6e 6669   with some confi
-00012ed0: 6465 6e63 6520 7468 6174 202a 4171 7569  dence that *Aqui
-00012ee0: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00012ef0: 6d69 6768 7420 6861 7665 2065 7870 6572  might have exper
-00012f00: 6965 6e63 6564 2061 2070 616c 656f 2d70  ienced a paleo-p
-00012f10: 6f6c 7970 6c6f 6964 697a 6174 696f 6e20  olyploidization 
-00012f20: 6576 656e 742e 204e 6578 742c 204c 6574  event. Next, Let
-00012f30: 2773 2068 6176 6520 6120 6675 7274 6865  's have a furthe
-00012f40: 7220 6c6f 6f6b 2061 626f 7574 2069 7473  r look about its
-00012f50: 2070 6879 6c6f 6765 6e65 7469 6320 6c6f   phylogenetic lo
-00012f60: 6361 7469 6f6e 2e20 5765 206b 6e6f 7720  cation. We know 
-00012f70: 7468 6174 2074 6865 7265 2061 7265 2075  that there are u
-00012f80: 6e63 6572 7461 696e 7469 6573 2061 626f  ncertainties abo
-00012f90: 7574 2077 6865 7468 6572 2074 6869 7320  ut whether this 
-00012fa0: 7075 7461 7469 7665 2070 616c 656f 2d70  putative paleo-p
-00012fb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e20  olyploidization 
-00012fc0: 6576 656e 7420 6973 2073 6861 7265 6420  event is shared 
-00012fd0: 7769 7468 2061 6c6c 2065 7564 6963 6f74  with all eudicot
-00012fe0: 7320 6f72 206e 6f74 2e20 5765 2063 616e  s or not. We can
-00012ff0: 2063 686f 6f73 6520 736f 6d65 206f 7468   choose some oth
-00013000: 6572 2065 7564 6963 6f74 2067 656e 6f6d  er eudicot genom
-00013010: 6573 2074 6f20 7365 6520 7468 6520 6f72  es to see the or
-00013020: 6465 7269 6e67 206f 6620 7370 6563 6961  dering of specia
-00013030: 7469 6f6e 2061 6e64 2070 6f6c 7970 6c6f  tion and polyplo
-00013040: 6964 697a 6174 696f 6e20 6576 656e 7473  idization events
-00013050: 2e20 4865 7265 2077 6520 6368 6f6f 7365  . Here we choose
-00013060: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
-00013070: 2a2c 202a 5072 6f74 6561 2063 796e 6172  *, *Protea cynar
-00013080: 6f69 6465 732a 2061 6e64 202a 4163 6f72  oides* and *Acor
-00013090: 7573 2061 6d65 7269 6361 6e75 732a 2069  us americanus* i
-000130a0: 6e20 7468 6520 666f 6c6c 6f77 696e 6720  n the following 
-000130b0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000130c0: 616e 616c 7973 6973 2e20 4669 7273 742c  analysis. First,
-000130d0: 2077 6520 6275 696c 7420 6120 676c 6f62   we built a glob
-000130e0: 616c 204d 5242 4820 6661 6d69 6c79 2075  al MRBH family u
-000130f0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00013100: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00013110: 2064 6d64 202d 2d67 6c6f 6261 6c6d 7262   dmd --globalmrb
-00013120: 6820 4171 7569 6c65 6769 615f 636f 6572  h Aquilegia_coer
-00013130: 756c 6561 2050 726f 7465 615f 6379 6e61  ulea Protea_cyna
-00013140: 726f 6964 6573 2041 636f 7275 735f 616d  roides Acorus_am
-00013150: 6572 6963 616e 7573 2056 6974 6973 5f76  ericanus Vitis_v
-00013160: 696e 6966 6572 6120 2d6f 2077 6764 5f67  inifera -o wgd_g
-00013170: 6c6f 6261 6c6d 7262 680a 6060 600a 0a49  lobalmrbh.```..I
-00013180: 6e20 7468 6520 676c 6f62 616c 204d 5242  n the global MRB
-00013190: 4820 6661 6d69 6c79 2c20 6576 6572 7920  H family, every 
-000131a0: 7061 6972 206f 6620 6f72 7468 6f6c 6f67  pair of ortholog
-000131b0: 6f75 7320 6765 6e65 7320 6973 2074 6865  ous genes is the
-000131c0: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-000131d0: 2068 6974 2c20 7375 6767 6573 7469 6e67   hit, suggesting
-000131e0: 2074 7275 6520 6f72 7468 6f6c 6f67 6f75   true orthologou
-000131f0: 7320 7265 6c61 7469 6f6e 7368 6970 732e  s relationships.
-00013200: 2057 6520 776f 756c 6420 7573 6520 7468   We would use th
-00013210: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00013220: 3e20 7661 6c75 6573 2061 7373 6f63 6961  > values associa
-00013230: 7465 6420 7769 7468 2074 6865 7365 206f  ted with these o
-00013240: 7274 686f 6c6f 676f 7573 2070 6169 7273  rthologous pairs
-00013250: 2074 6f20 6465 6c69 6d69 7420 7468 6520   to delimit the 
-00013260: 6469 7665 7267 656e 6365 202a 4b2a 3c73  divergence *K*<s
-00013270: 7562 3e53 3c2f 7375 623e 2070 6561 6b2e  ub>S</sub> peak.
-00013280: 2054 6f67 6574 6865 7220 7769 7468 2074   Together with t
-00013290: 6865 2077 686f 6c65 2070 6172 616e 6f6d  he whole paranom
-000132a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000132b0: 3e20 6469 7374 7269 6275 7469 6f6e 2c20  > distribution, 
-000132c0: 7765 2063 6f6e 6475 6374 2074 6865 2072  we conduct the r
-000132d0: 6174 6520 636f 7272 6563 7469 6f6e 2075  ate correction u
-000132e0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-000132f0: 2062 656c 6f77 2e0a 0a21 2153 696e 6365   below...!!Since
-00013300: 2060 7767 6460 2076 6572 7369 6f6e 2032   `wgd` version 2
-00013310: 2e30 2e32 342c 2077 6520 7265 7772 6f74  .0.24, we rewrot
-00013320: 6520 6120 636c 6561 6e65 7220 616e 6420  e a cleaner and 
-00013330: 7175 6963 6b65 7220 7761 7920 6f66 2064  quicker way of d
-00013340: 6f69 6e67 2073 7562 7374 6974 7574 696f  oing substitutio
-00013350: 6e20 7261 7465 2063 6f72 7265 6374 696f  n rate correctio
-00013360: 6e2e 2049 7427 7320 6e6f 7420 7265 7175  n. It's not requ
-00013370: 6972 6564 2074 6f20 7479 7065 2069 6e20  ired to type in 
-00013380: 616e 7920 7370 6569 6365 7320 7061 6972  any speices pair
-00013390: 2061 6e64 2061 2073 6572 6965 7320 6f66   and a series of
-000133a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000133b0: 2070 6c6f 7473 2077 696c 6c20 6265 2070   plots will be p
-000133c0: 726f 6475 6365 642e 2054 6865 2072 6571  roduced. The req
-000133d0: 7569 7265 6420 6669 6c65 7320 6172 6520  uired files are 
-000133e0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-000133f0: 7375 623e 533c 2f73 7562 3e20 7461 626c  sub>S</sub> tabl
-00013400: 652c 2070 6172 616c 6f67 6f75 7320 2a4b  e, paralogous *K
-00013410: 2a3c 7375 623e 533c 2f73 7562 3e20 7461  *<sub>S</sub> ta
-00013420: 626c 652c 2061 2073 7065 6369 6573 2074  ble, a species t
-00013430: 7265 6520 616e 6420 6120 666f 6375 7365  ree and a focuse
-00013440: 6420 7370 6563 6965 7320 2874 6865 206f  d species (the o
-00013450: 6e65 2069 6e70 7574 7465 6420 7769 7468  ne inputted with
-00013460: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
-00013470: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-00013480: 292e 2055 7365 7273 2063 616e 2063 686f  ). Users can cho
-00013490: 6f73 6520 746f 2061 6464 206f 6e65 206d  ose to add one m
-000134a0: 6f72 6520 6c61 7965 7220 6f66 2065 6c6d  ore layer of elm
-000134b0: 6d20 6d6f 6465 6c69 6e67 206f 6e20 7061  m modeling on pa
-000134c0: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-000134d0: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-000134e0: 616e 642f 6f72 2067 6d6d 206d 6f64 656c  and/or gmm model
-000134f0: 696e 6720 6f6e 2061 6e63 686f 7220 2a4b  ing on anchor *K
-00013500: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00013510: 6c75 6573 2e20 5468 6520 6f72 7468 6f6c  lues. The orthol
-00013520: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00013530: 2f73 7562 3e20 7661 6c75 6573 2063 616e  /sub> values can
-00013540: 2062 6520 6361 6c63 756c 6174 6564 2075   be calculated u
-00013550: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00013560: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00013570: 206b 7364 2077 6764 5f67 6c6f 6261 6c6d   ksd wgd_globalm
-00013580: 7262 682f 676c 6f62 616c 5f4d 5242 482e  rbh/global_MRBH.
-00013590: 7473 7620 7365 7173 2a20 2d6f 2077 6764  tsv seqs* -o wgd
-000135a0: 5f67 6c6f 6261 6c6d 7262 685f 6b73 0a60  _globalmrbh_ks.`
-000135b0: 6060 0a0a 5769 7468 2074 6865 2063 616c  ``..With the cal
-000135c0: 6375 6c61 7465 6420 6f72 7468 6f6c 6f67  culated ortholog
-000135d0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
-000135e0: 7562 3e20 7461 626c 652c 2077 6520 6361  ub> table, we ca
-000135f0: 6e20 7573 6520 7468 6520 636f 6d6d 616e  n use the comman
-00013600: 6420 6265 6c6f 7720 746f 2063 6f6e 6475  d below to condu
-00013610: 6374 2074 6865 2072 6174 6520 636f 7272  ct the rate corr
-00013620: 6563 7469 6f6e 2061 6e64 2f6f 7220 6d69  ection and/or mi
-00013630: 7874 7572 6520 6d6f 6465 6c69 6e67 2061  xture modeling a
-00013640: 6e61 6c79 7369 732e 0a0a 6060 600a 7767  nalysis...```.wg
-00013650: 6420 7669 7a20 2d64 2077 6764 5f67 6c6f  d viz -d wgd_glo
-00013660: 6261 6c6d 7262 685f 6b73 2f67 6c6f 6261  balmrbh_ks/globa
-00013670: 6c5f 4d52 4248 2e74 7376 2e6b 732e 7473  l_MRBH.tsv.ks.ts
-00013680: 7620 2d66 6120 4171 7569 6c65 6769 615f  v -fa Aquilegia_
-00013690: 636f 6572 756c 6561 202d 6570 6b20 4171  coerulea -epk Aq
-000136a0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000136b0: 2e6b 732e 7473 7620 2d61 7020 616e 6368  .ks.tsv -ap anch
-000136c0: 6f72 706f 696e 7473 2e74 7874 202d 7370  orpoints.txt -sp
-000136d0: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
-000136e0: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
-000136f0: 5f4b 7320 2d2d 706c 6f74 656c 6d6d 202d  _Ks --plotelmm -
-00013700: 2d70 6c6f 7461 7067 6d6d 0a60 6060 0a0a  -plotapgmm.```..
-00013710: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
-00013720: 676c 6f62 616c 6d72 6268 2f67 6c6f 6261  globalmrbh/globa
-00013730: 6c5f 4d52 4248 2e74 7376 2073 6571 732a  l_MRBH.tsv seqs*
-00013740: 202d 2d65 7874 7261 7061 7261 6e6f 6d65   --extraparanome
-00013750: 6b73 2077 6764 5f6b 7364 2f41 7175 696c  ks wgd_ksd/Aquil
-00013760: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00013770: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00013780: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00013790: 7765 6967 6874 202d 6f20 7767 645f 676c  weight -o wgd_gl
-000137a0: 6f62 616c 6d72 6268 5f6b 7320 2d2d 7370  obalmrbh_ks --sp
-000137b0: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
-000137c0: 6f65 7275 6c65 613b 5072 6f74 6561 5f63  oerulea;Protea_c
-000137d0: 796e 6172 6f69 6465 7322 202d 2d73 7061  ynaroides" --spa
-000137e0: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-000137f0: 6572 756c 6561 3b56 6974 6973 5f76 696e  erulea;Vitis_vin
-00013800: 6966 6572 6122 202d 2d73 7061 6972 2022  ifera" --spair "
-00013810: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013820: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
-00013830: 616e 7573 2220 2d2d 7370 6169 7220 2241  anus" --spair "A
-00013840: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00013850: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00013860: 756c 6561 2220 2d2d 706c 6f74 6b64 6520  ulea" --plotkde 
-00013870: 282d 6170 2077 6764 5f73 796e 2f69 6164  (-ap wgd_syn/iad
-00013880: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00013890: 6f69 6e74 732e 7478 7429 0a60 6060 0a0a  oints.txt).```..
-000138a0: 5468 6520 6669 6c65 2060 7370 6563 6965  The file `specie
-000138b0: 7374 7265 652e 6e77 6020 6973 2074 6865  stree.nw` is the
-000138c0: 2074 6578 7420 6669 6c65 206f 6620 7370   text file of sp
-000138d0: 6563 6965 7320 7472 6565 2069 6e20 6e65  ecies tree in ne
-000138e0: 7769 636b 2074 6861 7420 7261 7465 2063  wick that rate c
-000138f0: 6f72 7265 6374 696f 6e20 776f 756c 6420  orrection would 
-00013900: 6265 2063 6f6e 6475 6374 6564 206f 6e2e  be conducted on.
-00013910: 2049 7473 2063 6f6e 7465 6e74 2069 7320   Its content is 
-00013920: 6173 2062 656c 6f77 2e20 5573 6572 7320  as below. Users 
-00013930: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-00013940: 7468 6520 7370 6563 6965 7320 7061 6972  the species pair
-00013950: 7320 746f 2062 6520 706c 6f74 7465 642e  s to be plotted.
-00013960: 2057 6520 7375 6767 6573 7420 6164 6469   We suggest addi
-00013970: 6e67 2074 6865 206f 7074 696f 6e20 602d  ng the option `-
-00013980: 2d72 6577 6569 6768 7460 2074 6f20 7265  -reweight` to re
-00013990: 6361 6c63 756c 6174 6520 7468 6520 7765  calculate the we
-000139a0: 6967 6874 2070 6572 2073 7065 6369 6573  ight per species
-000139b0: 2070 6169 7220 7375 6368 2074 6861 7420   pair such that 
-000139c0: 7468 6520 7765 6967 6874 206f 6620 6f72  the weight of or
-000139d0: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-000139e0: 6169 7273 2077 696c 6c20 6265 636f 6d65  airs will become
-000139f0: 2031 2061 7320 7468 6520 7061 7261 6c6f   1 as the paralo
-00013a00: 676f 7573 2067 656e 6520 7061 6972 732e  gous gene pairs.
-00013a10: 2054 6865 2066 6c61 6720 602d 2d70 6c6f   The flag `--plo
-00013a20: 746b 6465 6020 6361 6e20 6265 2061 6464  tkde` can be add
-00013a30: 6564 2077 6865 6e20 7468 6520 6b64 6520  ed when the kde 
-00013a40: 6375 7276 6520 6f66 206f 7274 686f 6c6f  curve of ortholo
-00013a50: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00013a60: 7375 623e 2069 7320 6465 7369 7265 642e  sub> is desired.
-00013a70: 2045 7874 7261 2063 6f6c 6c69 6e65 6172   Extra collinear
-00013a80: 2064 6174 6120 6361 6e20 6265 2061 6464   data can be add
-00013a90: 6564 2062 7920 7468 6520 6f70 7469 6f6e  ed by the option
-00013aa0: 2060 2d61 7060 2e0a 0a60 6060 0a28 2828   `-ap`...```.(((
-00013ab0: 5669 7469 735f 7669 6e69 6665 7261 2c50  Vitis_vinifera,P
-00013ac0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-00013ad0: 292c 4171 7569 6c65 6769 615f 636f 6572  ),Aquilegia_coer
-00013ae0: 756c 6561 292c 4163 6f72 7573 5f61 6d65  ulea),Acorus_ame
-00013af0: 7269 6361 6e75 7329 3b0a 6060 600a 0a21  ricanus);.```..!
-00013b00: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-00013b10: 615f 636f 6572 756c 6561 5f47 6c6f 6261  a_coerulea_Globa
-00013b20: 6c6d 7262 684b 735f 436f 7272 6563 7465  lmrbhKs_Correcte
-00013b30: 642e 6b73 642e 7376 6729 0a0a 4173 2073  d.ksd.svg)..As s
-00013b40: 686f 776e 2061 626f 7665 2c20 6265 6361  hown above, beca
-00013b50: 7573 6520 6f66 2074 6865 2068 6967 6865  use of the highe
-00013b60: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
-00013b70: 6174 6520 6f66 202a 4171 7569 6c65 6769  ate of *Aquilegi
-00013b80: 6120 636f 6572 756c 6561 2a2c 2074 6865  a coerulea*, the
-00013b90: 206f 7269 6769 6e61 6c20 6f72 7468 6f6c   original orthol
-00013ba0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00013bb0: 2f73 7562 3e20 7661 6c75 6573 2077 6572  /sub> values wer
-00013bc0: 6520 6163 7475 616c 6c79 2075 6e64 6572  e actually under
-00013bd0: 6573 7469 6d61 7465 6420 696e 2074 6865  estimated in the
-00013be0: 2074 696d 652d 6672 616d 6520 6f66 202a   time-frame of *
-00013bf0: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00013c00: 6561 2a2e 2057 6865 6e20 7765 2072 6563  ea*. When we rec
-00013c10: 6f76 6572 6564 2074 6865 2064 6976 6572  overed the diver
-00013c20: 6765 6e63 6520 7375 6273 7469 7475 7469  gence substituti
-00013c30: 6f6e 2064 6973 7461 6e63 6520 696e 2074  on distance in t
-00013c40: 6572 6d73 206f 6620 7477 6f20 7469 6d65  erms of two time
-00013c50: 7320 6f66 2074 6865 2062 7261 6e63 682d  s of the branch-
-00013c60: 7370 6563 6966 6963 2063 6f6e 7472 6962  specific contrib
-00013c70: 7574 696f 6e20 6f66 202a 4171 7569 6c65  ution of *Aquile
-00013c80: 6769 6120 636f 6572 756c 6561 2a20 7369  gia coerulea* si
-00013c90: 6e63 6520 6974 7320 6469 7665 7267 656e  nce its divergen
-00013ca0: 6365 2077 6974 6820 7468 6520 7369 7374  ce with the sist
-00013cb0: 6572 2073 7065 6369 6573 2070 6c75 7320  er species plus 
-00013cc0: 7468 6520 7368 6172 6564 2073 7562 7374  the shared subst
-00013cd0: 6974 7574 696f 6e20 6469 7374 616e 6365  itution distance
-00013ce0: 2062 6566 6f72 6520 6469 7665 7267 656e   before divergen
-00013cf0: 6365 2028 696e 2072 656c 6174 6976 6520  ce (in relative 
-00013d00: 746f 2074 6865 206f 7574 6772 6f75 7029  to the outgroup)
-00013d10: 2c20 7468 6520 636f 7272 6563 7465 6420  , the corrected 
-00013d20: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013d30: 6d6f 6465 2062 6563 616d 6520 6c61 7267  mode became larg
-00013d40: 6572 2e0a 0a49 6620 6f6e 6520 6861 6420  er...If one had 
-00013d50: 7468 6520 6f72 7468 6f6c 6f67 6f75 7320  the orthologous 
-00013d60: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013d70: 6461 7461 2061 6c72 6561 6479 2c20 6f6e  data already, on
-00013d80: 6520 636f 756c 6420 616c 736f 2061 7070  e could also app
-00013d90: 6c79 2074 6865 2070 726f 6772 616d 2060  ly the program `
-00013da0: 7767 6420 7669 7a60 2074 6f20 636f 6e64  wgd viz` to cond
-00013db0: 7563 7420 7468 6520 7261 7465 2063 6f72  uct the rate cor
-00013dc0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-00013dd0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013de0: 6e64 2062 656c 6f77 2e20 4e6f 7465 2074  nd below. Note t
-00013df0: 6861 7420 7468 6520 6f72 6465 7220 6f66  hat the order of
-00013e00: 2067 6976 656e 2060 7370 6169 7260 206f   given `spair` o
-00013e10: 7074 696f 6e73 2064 6563 6964 6573 2074  ptions decides t
-00013e20: 6865 2063 6f6c 6f72 206f 6620 7468 6520  he color of the 
-00013e30: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013e40: 6469 7374 7269 6275 7469 6f6e 206f 6620  distribution of 
-00013e50: 6561 6368 2073 7065 6369 6573 2070 6169  each species pai
-00013e60: 722e 2054 6865 2061 6464 6974 696f 6e61  r. The additiona
-00013e70: 6c20 6f70 7469 6f6e 2060 666f 6375 7332  l option `focus2
-00013e80: 616c 6c60 2069 6e20 6077 6764 206b 7364  all` in `wgd ksd
-00013e90: 6020 616e 6420 6077 6764 2076 697a 6020  ` and `wgd viz` 
-00013ea0: 6361 6e20 6265 2075 7365 6420 746f 2074  can be used to t
-00013eb0: 656c 6c20 7468 6520 7072 6f67 7261 6d20  ell the program 
-00013ec0: 7468 6174 2074 6865 2073 7065 6369 6573  that the species
-00013ed0: 2070 6169 7273 2061 7265 2062 6574 7765   pairs are betwe
-00013ee0: 656e 2074 6865 2066 6f63 7573 2073 7065  en the focus spe
-00013ef0: 6369 6573 2061 6e64 2061 6c6c 2074 6865  cies and all the
-00013f00: 206f 7468 6572 2073 7065 6369 6573 2073   other species s
-00013f10: 7563 6820 7468 6174 2075 7365 7273 2064  uch that users d
-00013f20: 6f6e 2774 206e 6565 6420 746f 2074 7970  on't need to typ
-00013f30: 6520 696e 2065 6163 6820 7061 6972 2069  e in each pair i
-00013f40: 6e64 6976 6964 7561 6c6c 792e 0a0a 6060  ndividually...``
-00013f50: 600a 7767 6420 7669 7a20 2d64 2077 6764  `.wgd viz -d wgd
-00013f60: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
-00013f70: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
-00013f80: 732e 7473 7620 2d2d 6578 7472 6170 6172  s.tsv --extrapar
-00013f90: 616e 6f6d 656b 7320 7767 645f 6b73 642f  anomeks wgd_ksd/
-00013fa0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013fb0: 6561 2e74 7376 2e6b 732e 7473 7620 2d73  ea.tsv.ks.tsv -s
-00013fc0: 7020 7370 6563 6965 7374 7265 652e 6e77  p speciestree.nw
-00013fd0: 202d 2d72 6577 6569 6768 7420 2d61 7020   --reweight -ap 
-00013fe0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00013ff0: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-00014000: 2e74 7874 202d 6f20 7767 645f 7669 7a5f  .txt -o wgd_viz_
-00014010: 6d69 7865 645f 4b73 202d 2d73 7061 6972  mixed_Ks --spair
-00014020: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014030: 756c 6561 3b50 726f 7465 615f 6379 6e61  ulea;Protea_cyna
-00014040: 726f 6964 6573 2220 2d2d 7370 6169 7220  roides" --spair 
-00014050: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00014060: 6c65 613b 5669 7469 735f 7669 6e69 6665  lea;Vitis_vinife
-00014070: 7261 2220 2d2d 7370 6169 7220 2241 7175  ra" --spair "Aqu
-00014080: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00014090: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-000140a0: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-000140b0: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
-000140c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000140d0: 6122 202d 2d70 6c6f 746b 6465 2028 2d2d  a" --plotkde (--
-000140e0: 6773 6d61 7020 6765 6e65 5f73 7065 6369  gsmap gene_speci
-000140f0: 6573 2e6d 6170 290a 6060 600a 0a57 6974  es.map).```..Wit
-00014100: 6820 7468 6520 6f70 7469 6f6e 2060 666f  h the option `fo
-00014110: 6375 7332 616c 6c60 2c20 7468 6520 636f  cus2all`, the co
-00014120: 6d6d 616e 6420 6361 6e20 6265 2061 6c73  mmand can be als
-00014130: 6f20 6c69 6b65 2074 6869 732c 2073 616d  o like this, sam
-00014140: 6520 7769 7468 2060 7767 6420 6b73 6460  e with `wgd ksd`
-00014150: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
-00014160: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
-00014170: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
-00014180: 7473 762e 6b73 2e74 7376 202d 2d65 7874  tsv.ks.tsv --ext
-00014190: 7261 7061 7261 6e6f 6d65 6b73 2077 6764  raparanomeks wgd
-000141a0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
-000141b0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
-000141c0: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
-000141d0: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
-000141e0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-000141f0: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00014200: 6f69 6e74 732e 7478 7420 2d6f 2077 6764  oints.txt -o wgd
-00014210: 5f76 697a 5f6d 6978 6564 5f4b 7320 2d2d  _viz_mixed_Ks --
-00014220: 666f 6375 7332 616c 6c20 4171 7569 6c65  focus2all Aquile
-00014230: 6769 615f 636f 6572 756c 6561 202d 2d70  gia_coerulea --p
-00014240: 6c6f 746b 6465 0a60 6060 0a0a 4e6f 7465  lotkde.```..Note
-00014250: 2074 6861 7420 7765 2063 616e 2065 6173   that we can eas
-00014260: 696c 7920 7368 6f77 2074 6861 7420 2a41  ily show that *A
-00014270: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
-00014280: 612a 2068 6173 2068 6967 6865 7220 7375  a* has higher su
-00014290: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
-000142a0: 7468 616e 202a 5072 6f74 6561 2063 796e  than *Protea cyn
-000142b0: 6172 6f69 6465 732a 2061 6e64 202a 5669  aroides* and *Vi
-000142c0: 7469 7320 7669 6e69 6665 7261 2a20 6279  tis vinifera* by
-000142d0: 2063 6f6d 7061 7269 6e67 2074 6865 6972   comparing their
-000142e0: 2073 7562 7374 6974 7574 696f 6e20 6469   substitution di
-000142f0: 7374 616e 6365 2069 6e20 7265 6761 7264  stance in regard
-00014300: 2074 6f20 7468 6520 7361 6d65 2064 6976   to the same div
-00014310: 6572 6765 6e63 6520 6576 656e 7420 7769  ergence event wi
-00014320: 7468 206f 7574 6772 6f75 7020 7370 6563  th outgroup spec
-00014330: 6965 7320 2a41 636f 7275 735f 616d 6572  ies *Acorus_amer
-00014340: 6963 616e 7573 2a2c 2075 7369 6e67 2063  icanus*, using c
-00014350: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a60  ommand below...`
-00014360: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
-00014370: 645f 676c 6f62 616c 6d72 6268 5f6b 732f  d_globalmrbh_ks/
-00014380: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
-00014390: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-000143a0: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-000143b0: 6967 6874 202d 6f20 7767 645f 7669 7a5f  ight -o wgd_viz_
-000143c0: 436f 6d70 6172 655f 7261 7465 202d 2d73  Compare_rate --s
-000143d0: 7061 6972 2022 4163 6f72 7573 5f61 6d65  pair "Acorus_ame
-000143e0: 7269 6361 6e75 733b 5072 6f74 6561 5f63  ricanus;Protea_c
-000143f0: 796e 6172 6f69 6465 7322 202d 2d73 7061  ynaroides" --spa
-00014400: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00014410: 6572 756c 6561 3b41 636f 7275 735f 616d  erulea;Acorus_am
-00014420: 6572 6963 616e 7573 2220 2d2d 7370 6169  ericanus" --spai
-00014430: 7220 2256 6974 6973 5f76 696e 6966 6572  r "Vitis_vinifer
-00014440: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-00014450: 6e75 7322 202d 2d67 736d 6170 2067 656e  nus" --gsmap gen
-00014460: 655f 7370 6563 6965 732e 6d61 7020 2d2d  e_species.map --
-00014470: 706c 6f74 6b64 650a 6060 600a 0a21 5b5d  plotkde.```..![]
-00014480: 2864 6174 612f 5261 775f 4f72 7468 6f6c  (data/Raw_Orthol
-00014490: 6f67 7565 735f 436f 6d70 6172 655f 7261  ogues_Compare_ra
-000144a0: 7465 2e6b 7364 2e73 7667 290a 0a41 7320  te.ksd.svg)..As 
-000144b0: 6469 7370 6c61 7965 6420 6162 6f76 652c  displayed above,
-000144c0: 2074 6865 206f 7274 686f 6c6f 676f 7573   the orthologous
-000144d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000144e0: 2076 616c 7565 7320 6265 7774 6565 6e20   values bewteen 
-000144f0: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00014500: 6c65 612a 2061 6e64 202a 4163 6f72 7573  lea* and *Acorus
-00014510: 2061 6d65 7269 6361 6e75 732a 2068 6173   americanus* has
-00014520: 2074 6865 2068 6967 6865 7374 206d 6f64   the highest mod
-00014530: 652c 2069 6e64 6963 6174 696e 6774 6865  e, indicatingthe
-00014540: 2066 6173 7465 7220 7375 6273 7469 7475   faster substitu
-00014550: 7469 6f6e 2072 6174 6520 6f66 202a 4171  tion rate of *Aq
-00014560: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-00014570: 2a20 636f 6d70 6172 6564 2074 6f20 2a50  * compared to *P
-00014580: 726f 7465 6120 6379 6e61 726f 6964 6573  rotea cynaroides
-00014590: 2a20 616e 6420 2a56 6974 6973 2076 696e  * and *Vitis vin
-000145a0: 6966 6572 612a 2e0a 0a42 6566 6f72 6520  ifera*...Before 
-000145b0: 7632 2e30 2e32 312c 2074 6865 2067 656e  v2.0.21, the gen
-000145c0: 652d 7370 6563 6965 7320 6d61 7020 6669  e-species map fi
-000145d0: 6c65 2069 7320 6e65 6363 6573 7361 7269  le is neccessari
-000145e0: 6c79 206e 6565 6465 6420 666f 7220 6974  ly needed for it
-000145f0: 7320 696d 706c 656d 656e 7461 7469 6f6e  s implementation
-00014600: 2069 6e20 6077 6764 2076 697a 602c 2077   in `wgd viz`, w
-00014610: 6869 6368 2073 686f 756c 6420 6265 2061  hich should be a
-00014620: 7574 6f6d 6174 656c 7920 7072 6f64 7563  utomately produc
-00014630: 6564 2062 7920 7468 6520 6c61 7374 2060  ed by the last `
-00014640: 7767 6420 6b73 6460 2073 7465 7020 6769  wgd ksd` step gi
-00014650: 7665 6e20 7468 6520 6073 7061 6972 6020  ven the `spair` 
-00014660: 616e 6420 6073 7065 6369 6573 7472 6565  and `speciestree
-00014670: 6020 7061 7261 6d65 7465 7273 2e20 5468  ` parameters. Th
-00014680: 6520 6067 656e 655f 7370 6563 6965 732e  e `gene_species.
-00014690: 6d61 7060 2068 6173 2063 6f6e 7465 6e74  map` has content
-000146a0: 7320 6173 2062 656c 6f77 2069 6e20 7768  s as below in wh
-000146b0: 6963 6820 6561 6368 206c 696e 6520 6973  ich each line is
-000146c0: 2074 6865 206a 6f69 6e65 6420 7374 7269   the joined stri
-000146d0: 6e67 206f 6620 6765 6e65 206e 616d 6520  ng of gene name 
-000146e0: 616e 6420 7370 6563 6965 7320 6e61 6d65  and species name
-000146f0: 2062 7920 7370 6163 652e 2041 6674 6572   by space. After
-00014700: 2076 322e 302e 3231 2028 696e 636c 7564   v2.0.21 (includ
-00014710: 6564 292c 2074 6865 2067 656e 652d 7370  ed), the gene-sp
-00014720: 6563 6965 7320 6d61 7020 6669 6c65 2069  ecies map file i
-00014730: 7320 6e6f 7420 6e65 6363 6573 7361 7269  s not neccessari
-00014740: 6c79 206e 6565 6465 6420 616e 796d 6f72  ly needed anymor
-00014750: 652e 0a0a 6060 600a 4171 636f 6536 4730  e...```.Aqcoe6G0
-00014760: 3537 3830 302e 3120 4171 7569 6c65 6769  57800.1 Aquilegi
-00014770: 615f 636f 6572 756c 6561 0a56 7669 5f56  a_coerulea.Vvi_V
-00014780: 4954 5f32 3031 7330 3031 3167 3031 3533  IT_201s0011g0153
-00014790: 302e 3120 5669 7469 735f 7669 6e69 6665  0.1 Vitis_vinife
-000147a0: 7261 0a50 6379 5f50 726f 6379 3031 6730  ra.Pcy_Procy01g0
-000147b0: 3835 3130 2050 726f 7465 615f 6379 6e61  8510 Protea_cyna
-000147c0: 726f 6964 6573 0a41 616d 5f41 636f 7261  roides.Aam_Acora
-000147d0: 2e30 3447 3134 3239 3030 2e31 2041 636f  .04G142900.1 Aco
-000147e0: 7275 735f 616d 6572 6963 616e 7573 0a60  rus_americanus.`
-000147f0: 6060 0a0a 4120 6d6f 7265 2063 6f6d 706c  ``..A more compl
-00014800: 6578 2070 6c6f 7420 6361 6e20 6265 206d  ex plot can be m
-00014810: 6164 6520 6279 2061 6464 2074 6865 2066  ade by add the f
-00014820: 6c61 6720 602d 2d70 6c6f 7465 6c6d 6d60  lag `--plotelmm`
-00014830: 2073 7563 6820 7468 6174 2074 6865 2045   such that the E
-00014840: 4c4d 4d20 6d69 7874 7572 6520 6d6f 6465  LMM mixture mode
-00014850: 6c69 6e67 206f 6620 7072 6f76 6964 6564  ling of provided
-00014860: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00014870: 623e 533c 2f73 7562 3e20 6361 6e20 6265  b>S</sub> can be
-00014880: 2073 7570 6572 696d 706f 7365 642c 2075   superimposed, u
-00014890: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-000148a0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-000148b0: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
-000148c0: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
-000148d0: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
-000148e0: 202d 2d65 7874 7261 7061 7261 6e6f 6d65   --extraparanome
-000148f0: 6b73 2077 6764 5f6b 7364 2f41 7175 696c  ks wgd_ksd/Aquil
-00014900: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00014910: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00014920: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00014930: 7765 6967 6874 202d 6170 2077 6764 5f73  weight -ap wgd_s
-00014940: 796e 2f69 6164 686f 7265 2d6f 7574 2f61  yn/iadhore-out/a
-00014950: 6e63 686f 7270 6f69 6e74 732e 7478 7420  nchorpoints.txt 
-00014960: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
-00014970: 5f4b 735f 656c 6d6d 202d 2d73 7061 6972  _Ks_elmm --spair
-00014980: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014990: 756c 6561 3b50 726f 7465 615f 6379 6e61  ulea;Protea_cyna
-000149a0: 726f 6964 6573 2220 2d2d 7370 6169 7220  roides" --spair 
-000149b0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-000149c0: 6c65 613b 5669 7469 735f 7669 6e69 6665  lea;Vitis_vinife
-000149d0: 7261 2220 2d2d 7370 6169 7220 2241 7175  ra" --spair "Aqu
-000149e0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-000149f0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00014a00: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-00014a10: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
-00014a20: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014a30: 6122 202d 2d67 736d 6170 2067 656e 655f  a" --gsmap gene_
-00014a40: 7370 6563 6965 732e 6d61 7020 2d2d 706c  species.map --pl
-00014a50: 6f74 6b64 6520 2d2d 706c 6f74 656c 6d6d  otkde --plotelmm
-00014a60: 0a60 6060 0a0a 215b 5d28 6461 7461 2f41  .```..![](data/A
-00014a70: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014a80: 615f 476c 6f62 616c 6d72 6268 4b73 5f45  a_GlobalmrbhKs_E
-00014a90: 6c6d 6d5f 436f 7272 6563 7465 642e 6b73  lmm_Corrected.ks
-00014aa0: 642e 7376 6729 0a0a 4672 6f6d 2074 6865  d.svg)..From the
-00014ab0: 206d 6978 6564 202a 4b2a 3c73 7562 3e53   mixed *K*<sub>S
-00014ac0: 3c2f 7375 623e 2070 6c6f 7420 6162 6f76  </sub> plot abov
-00014ad0: 652c 2077 6520 6361 6e20 7365 6520 7468  e, we can see th
-00014ae0: 6174 2074 6865 206f 7074 696d 697a 6564  at the optimized
-00014af0: 206c 6f67 6e6f 726d 616c 2063 6f6d 706f   lognormal compo
-00014b00: 6e65 6e74 2062 2077 6974 6820 6d6f 6465  nent b with mode
-00014b10: 2031 2e32 2069 7320 796f 756e 6765 7220   1.2 is younger 
-00014b20: 7468 616e 2074 6865 2063 6f72 7265 6374  than the correct
-00014b30: 6564 206f 7274 686f 6c6f 676f 7573 202a  ed orthologous *
-00014b40: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
-00014b50: 6f64 6520 7769 7468 202a 5072 6f74 6561  ode with *Protea
-00014b60: 2063 796e 6172 6f69 6465 732a 2061 6e64   cynaroides* and
-00014b70: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
-00014b80: 2a20 2831 2e33 3920 616e 6420 312e 3437  * (1.39 and 1.47
-00014b90: 2c20 7265 7370 6563 7469 7665 6c79 292e  , respectively).
-00014ba0: 0a0a 4265 7369 6465 732c 2077 6520 6361  ..Besides, we ca
-00014bb0: 6e20 616c 736f 2061 6464 2074 6865 2047  n also add the G
-00014bc0: 4d4d 206d 6978 7475 7265 206d 6f64 656c  MM mixture model
-00014bd0: 696e 6720 6f66 2061 6e63 686f 7220 2a4b  ing of anchor *K
-00014be0: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00014bf0: 6c75 6573 2077 6974 6820 7468 6520 666c  lues with the fl
-00014c00: 6167 2060 2d2d 706c 6f74 6170 676d 6d60  ag `--plotapgmm`
-00014c10: 2c20 7573 696e 6720 7468 6520 636f 6d6d  , using the comm
-00014c20: 616e 6420 6265 6c6f 772e 0a0a 6060 600a  and below...```.
-00014c30: 7767 6420 7669 7a20 2d64 2077 6764 5f67  wgd viz -d wgd_g
-00014c40: 6c6f 6261 6c6d 7262 685f 6b73 2f67 6c6f  lobalmrbh_ks/glo
-00014c50: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
-00014c60: 7473 7620 2d2d 6578 7472 6170 6172 616e  tsv --extraparan
-00014c70: 6f6d 656b 7320 7767 645f 6b73 642f 4171  omeks wgd_ksd/Aq
-00014c80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014c90: 2e74 7376 2e6b 732e 7473 7620 2d73 7020  .tsv.ks.tsv -sp 
-00014ca0: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
-00014cb0: 2d72 6577 6569 6768 7420 2d61 7020 7767  -reweight -ap wg
-00014cc0: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00014cd0: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
-00014ce0: 7874 202d 6f20 7767 645f 7669 7a5f 6d69  xt -o wgd_viz_mi
-00014cf0: 7865 645f 4b73 5f65 6c6d 6d20 2d2d 7370  xed_Ks_elmm --sp
-00014d00: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
-00014d10: 6f65 7275 6c65 613b 5072 6f74 6561 5f63  oerulea;Protea_c
-00014d20: 796e 6172 6f69 6465 7322 202d 2d73 7061  ynaroides" --spa
-00014d30: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00014d40: 6572 756c 6561 3b56 6974 6973 5f76 696e  erulea;Vitis_vin
-00014d50: 6966 6572 6122 202d 2d73 7061 6972 2022  ifera" --spair "
-00014d60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00014d70: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
-00014d80: 616e 7573 2220 2d2d 7370 6169 7220 2241  anus" --spair "A
-00014d90: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014da0: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00014db0: 756c 6561 2220 2d2d 6773 6d61 7020 6765  ulea" --gsmap ge
-00014dc0: 6e65 5f73 7065 6369 6573 2e6d 6170 202d  ne_species.map -
-00014dd0: 2d70 6c6f 746b 6465 202d 2d70 6c6f 7461  -plotkde --plota
-00014de0: 7067 6d6d 6d0a 6060 600a 0a21 5b5d 2864  pgmmm.```..![](d
-00014df0: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00014e00: 6572 756c 6561 5f47 6c6f 6261 6c6d 7262  erulea_Globalmrb
-00014e10: 684b 735f 4170 676d 6d5f 436f 7272 6563  hKs_Apgmm_Correc
-00014e20: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
-00014e30: 206d 616e 6966 6573 7465 6420 6162 6f76   manifested abov
-00014e40: 652c 2074 6865 2061 6e63 686f 7220 2a4b  e, the anchor *K
-00014e50: 2a3c 7375 623e 533c 2f73 7562 3e20 636f  *<sub>S</sub> co
-00014e60: 6d70 6f6e 656e 7420 3220 7769 7468 206d  mponent 2 with m
-00014e70: 6f64 6520 312e 3238 2069 7320 616c 736f  ode 1.28 is also
-00014e80: 2079 6f75 6e67 6572 2074 6861 6e20 7468   younger than th
-00014e90: 6520 636f 7272 6563 7465 6420 6f72 7468  e corrected orth
-00014ea0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00014eb0: 533c 2f73 7562 3e20 6d6f 6465 2077 6974  S</sub> mode wit
-00014ec0: 6820 2a50 726f 7465 6120 6379 6e61 726f  h *Protea cynaro
-00014ed0: 6964 6573 2a20 616e 6420 2a56 6974 6973  ides* and *Vitis
-00014ee0: 2076 696e 6966 6572 612a 2e20 4275 7420   vinifera*. But 
-00014ef0: 7765 206e 6565 6420 746f 2062 6520 6f66  we need to be of
-00014f00: 2063 6f75 7273 6520 6361 7574 696f 7573   course cautious
-00014f10: 2074 6861 7420 7375 6368 2064 6973 7469   that such disti
-00014f20: 6e63 7469 6f6e 2063 6f6d 6573 2077 6974  nction comes wit
-00014f30: 6820 7468 6520 756e 6365 7274 6169 6e74  h the uncertaint
-00014f40: 6965 7320 696e 7472 6f64 7563 6564 2066  ies introduced f
-00014f50: 726f 6d20 7468 6520 6170 706c 6965 6420  rom the applied 
-00014f60: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00014f70: 206d 6574 686f 646f 6c6f 6779 2069 6e20   methodology in 
-00014f80: 7465 726d 7320 6f66 2066 6f72 2069 6e73  terms of for ins
-00014f90: 7461 6e63 6520 6469 6666 6572 656e 7420  tance different 
-00014fa0: 696e 6974 6961 6c69 7a61 7469 6f6e 2070  initialization p
-00014fb0: 6f69 6e74 7320 616e 6420 7468 6520 6973  oints and the is
-00014fc0: 7375 6520 6f66 206f 7665 7266 6974 7469  sue of overfitti
-00014fd0: 6e67 2061 6e64 2074 6865 2073 6973 7465  ng and the siste
-00014fe0: 7220 7370 6563 6965 7373 2061 646f 7074  r speciess adopt
-00014ff0: 6564 2069 6e20 7468 6174 2074 6865 7265  ed in that there
-00015000: 206d 6967 6874 2062 6520 7370 6563 6965   might be specie
-00015010: 7320 7769 7468 206d 6f72 6520 6469 7370  s with more disp
-00015020: 6172 6174 6520 7375 6273 7469 7475 7469  arate substituti
-00015030: 6f6e 2072 6174 6573 2074 6861 6e20 7468  on rates than th
-00015040: 6520 6f6e 6520 7765 2063 686f 7365 2e0a  e one we chose..
-00015050: 0a41 6464 696e 6720 626f 7468 2045 4c4d  .Adding both ELM
-00015060: 4d20 7265 7375 6c74 2066 6f72 2070 6172  M result for par
-00015070: 616e 6f6d 6520 616e 6420 474d 4d20 7265  anome and GMM re
-00015080: 7375 6c74 2066 6f72 2061 6e63 686f 7220  sult for anchor 
-00015090: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000150a0: 6361 6e20 6265 2061 6368 6965 7665 6420  can be achieved 
-000150b0: 6a75 7374 2062 7920 6164 6420 7468 6520  just by add the 
-000150c0: 7477 6f20 666c 6167 7320 6d65 6e74 696f  two flags mentio
-000150d0: 6e65 6420 6162 6f76 652c 2075 7369 6e67  ned above, using
-000150e0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-000150f0: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
-00015100: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
-00015110: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
-00015120: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
-00015130: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-00015140: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00015150: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00015160: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-00015170: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-00015180: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
-00015190: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-000151a0: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
-000151b0: 6764 5f76 697a 5f6d 6978 6564 5f4b 735f  gd_viz_mixed_Ks_
-000151c0: 656c 6d6d 202d 2d73 7061 6972 2022 4171  elmm --spair "Aq
-000151d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000151e0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
-000151f0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
-00015200: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015210: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
-00015220: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00015230: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
-00015240: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015250: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015260: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
-00015270: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
-00015280: 2d67 736d 6170 2067 656e 655f 7370 6563  -gsmap gene_spec
-00015290: 6965 732e 6d61 7020 2d2d 706c 6f74 6b64  ies.map --plotkd
-000152a0: 6520 2d2d 706c 6f74 656c 6d6d 202d 2d70  e --plotelmm --p
-000152b0: 6c6f 7461 7067 6d6d 6d0a 6060 600a 0a21  lotapgmmm.```..!
-000152c0: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-000152d0: 615f 636f 6572 756c 6561 5f47 6c6f 6261  a_coerulea_Globa
-000152e0: 6c6d 7262 684b 735f 456c 6d6d 5f41 7067  lmrbhKs_Elmm_Apg
-000152f0: 6d6d 5f43 6f72 7265 6374 6564 2e6b 7364  mm_Corrected.ksd
-00015300: 2e73 7667 290a 0a41 6e20 616c 7465 726e  .svg)..An altern
-00015310: 6174 6976 6520 7761 7920 746f 2063 616c  ative way to cal
-00015320: 6375 6c61 7465 2074 6865 206f 7274 686f  culate the ortho
-00015330: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00015340: 3c2f 7375 623e 2069 7320 746f 2064 6972  </sub> is to dir
-00015350: 6563 746c 7920 7573 6520 7468 6520 6f72  ectly use the or
-00015360: 7468 6f67 726f 7570 7320 696e 7374 6561  thogroups instea
-00015370: 6420 6f66 2067 6c6f 6261 6c20 4d52 4248  d of global MRBH
-00015380: 2066 616d 696c 792e 2054 6861 7420 7761   family. That wa
-00015390: 7920 7765 2064 6f6e 2774 2075 7365 2074  y we don't use t
-000153a0: 6865 2070 7265 2d69 6e66 6572 7265 6420  he pre-inferred 
-000153b0: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-000153c0: 3e53 3c2f 7375 623e 2062 7574 2074 6865  >S</sub> but the
-000153d0: 2070 6172 616c 6f67 6f75 7320 6765 6e65   paralogous gene
-000153e0: 2070 6169 7273 2069 6e73 6964 6520 6561   pairs inside ea
-000153f0: 6368 206f 7274 686f 6772 6f75 7020 696e  ch orthogroup in
-00015400: 7374 6561 642e 2054 6f20 6163 6869 6576  stead. To achiev
-00015410: 6520 7468 6174 2c20 7765 2066 6972 7374  e that, we first
-00015420: 206e 6565 6420 746f 2069 6e66 6572 206f   need to infer o
-00015430: 7274 686f 6772 6f75 7073 2075 7369 6e67  rthogroups using
-00015440: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00015450: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-00015460: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00015470: 6c65 6120 5072 6f74 6561 5f63 796e 6172  lea Protea_cynar
-00015480: 6f69 6465 7320 4163 6f72 7573 5f61 6d65  oides Acorus_ame
-00015490: 7269 6361 6e75 7320 5669 7469 735f 7669  ricanus Vitis_vi
-000154a0: 6e69 6665 7261 202d 2d6f 7274 686f 696e  nifera --orthoin
-000154b0: 6665 7220 2d6f 2077 6764 5f6f 7274 686f  fer -o wgd_ortho
-000154c0: 2028 2d2d 6f6e 6c79 6f72 7468 6f29 200a   (--onlyortho) .
-000154d0: 6060 600a 0a55 7365 7273 2063 616e 2064  ```..Users can d
-000154e0: 6563 6964 6520 746f 206f 6e6c 7920 636f  ecide to only co
-000154f0: 6e64 7563 7420 7468 6520 6f72 7468 6f67  nduct the orthog
-00015500: 726f 7570 2061 6e61 6c79 7369 7320 7768  roup analysis wh
-00015510: 696c 6520 736b 6970 7069 6e67 206f 7468  ile skipping oth
-00015520: 6572 2061 6e61 6c79 7369 7320 6279 2061  er analysis by a
-00015530: 6464 696e 6720 7468 6520 666c 6167 2060  dding the flag `
-00015540: 2d2d 6f6e 6c79 6f72 7468 6f60 2e20 4e65  --onlyortho`. Ne
-00015550: 7874 2073 7465 7020 6973 2074 6865 2073  xt step is the s
-00015560: 616d 6520 7769 7468 2067 6c6f 6261 6c20  ame with global 
-00015570: 4d52 4248 2066 616d 696c 7920 6578 6365  MRBH family exce
-00015580: 7074 2074 6861 7420 7765 2064 6f6e 2774  pt that we don't
-00015590: 2075 7365 2074 6865 2065 7874 7261 2070   use the extra p
-000155a0: 7265 2d69 6e66 6572 7265 6420 7061 7261  re-inferred para
-000155b0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-000155c0: 7375 623e 2061 6e79 6d6f 7265 2e20 5765  sub> anymore. We
-000155d0: 2069 6e66 6572 2074 6865 202a 4b2a 3c73   infer the *K*<s
-000155e0: 7562 3e53 3c2f 7375 623e 2075 7369 6e67  ub>S</sub> using
-000155f0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00015600: 6f77 2e20 4e6f 7465 2074 6861 7420 7468  ow. Note that th
-00015610: 6520 7072 6f67 7261 6d20 6077 6764 2076  e program `wgd v
-00015620: 697a 6020 6361 6e20 706c 6f74 2074 6865  iz` can plot the
-00015630: 2073 616d 6520 6a75 7374 2061 7320 7368   same just as sh
-00015640: 6f77 6e20 6162 6f76 652e 0a0a 6060 600a  own above...```.
-00015650: 7767 6420 6b73 6420 7767 645f 6f72 7468  wgd ksd wgd_orth
-00015660: 6f2f 4f72 7468 6f67 726f 7570 732e 7370  o/Orthogroups.sp
-00015670: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-00015680: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-00015690: 6874 202d 6f20 7767 645f 6f72 7468 6f5f  ht -o wgd_ortho_
-000156a0: 6b73 202d 2d73 7061 6972 2022 4171 7569  ks --spair "Aqui
-000156b0: 6c65 6769 615f 636f 6572 756c 6561 3b50  legia_coerulea;P
-000156c0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-000156d0: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
-000156e0: 6567 6961 5f63 6f65 7275 6c65 613b 5669  egia_coerulea;Vi
-000156f0: 7469 735f 7669 6e69 6665 7261 2220 2d2d  tis_vinifera" --
-00015700: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00015710: 5f63 6f65 7275 6c65 613b 4163 6f72 7573  _coerulea;Acorus
-00015720: 5f61 6d65 7269 6361 6e75 7322 202d 2d73  _americanus" --s
-00015730: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00015740: 636f 6572 756c 6561 3b41 7175 696c 6567  coerulea;Aquileg
-00015750: 6961 5f63 6f65 7275 6c65 6122 202d 2d70  ia_coerulea" --p
-00015760: 6c6f 746b 6465 2028 2d61 7020 7767 645f  lotkde (-ap wgd_
-00015770: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
-00015780: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
-00015790: 290a 6060 600a 0a21 5b5d 2864 6174 612f  ).```..![](data/
-000157a0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000157b0: 6561 5f4f 7274 686f 4b73 5f43 6f72 7265  ea_OrthoKs_Corre
-000157c0: 6374 6564 2e6b 7364 2e73 7667 290a 0a41  cted.ksd.svg)..A
-000157d0: 7320 7368 6f77 6e20 6162 6f76 652c 2074  s shown above, t
-000157e0: 6865 206e 756d 6265 7220 6f66 2062 6f74  he number of bot
-000157f0: 6820 7061 7261 6c6f 676f 7573 2067 656e  h paralogous gen
-00015800: 6520 7061 6972 7320 616e 6420 6f72 7468  e pairs and orth
-00015810: 6f6c 6f67 6f75 7320 6765 6e65 2070 6169  ologous gene pai
-00015820: 7273 2069 7320 6469 6666 6572 656e 7420  rs is different 
-00015830: 7468 616e 2074 6865 206f 6e65 2066 726f  than the one fro
-00015840: 6d20 676c 6f62 616c 204d 5242 4820 6661  m global MRBH fa
-00015850: 6d69 6c79 2069 6e20 7468 6174 2068 6572  mily in that her
-00015860: 6520 7765 2070 6c6f 7474 6564 2061 6c6c  e we plotted all
-00015870: 206f 7274 686f 6c6f 676f 7573 2067 656e   orthologous gen
-00015880: 6520 7061 6972 7320 696e 7374 6561 6420  e pairs instead 
-00015890: 6f66 206f 6e6c 7920 676c 6f62 616c 204d  of only global M
-000158a0: 5242 4820 616e 6420 706f 7465 6e74 6961  RBH and potentia
-000158b0: 6c6c 7920 6e65 7720 7061 7261 6c6f 676f  lly new paralogo
-000158c0: 7573 2067 656e 6520 7061 6972 7320 6d69  us gene pairs mi
-000158d0: 6768 7420 6265 2070 726f 6475 6365 6420  ght be produced 
-000158e0: 696e 2074 6865 206f 7274 686f 6772 6f75  in the orthogrou
-000158f0: 7020 696e 6665 7265 6e63 6520 7374 6570  p inference step
-00015900: 2c20 746f 6765 7468 6572 2077 6974 6820  , together with 
-00015910: 6469 6666 6572 656e 7420 7265 6361 6c63  different recalc
-00015920: 756c 6174 6564 2077 6569 6768 7473 2e0a  ulated weights..
-00015930: 0a49 6620 6f6e 6520 6f6e 6c79 2077 616e  .If one only wan
-00015940: 7465 6420 746f 2070 6c6f 7420 7468 6520  ted to plot the 
-00015950: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-00015960: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00015970: 7269 6275 7469 6f6e 2c20 6173 2077 6520  ribution, as we 
-00015980: 6861 7665 2061 6c72 6561 6479 2073 686f  have already sho
-00015990: 776e 2061 626f 7665 2c20 6974 2063 616e  wn above, it can
-000159a0: 2062 6520 6561 7369 6c79 2061 6368 6965   be easily achie
-000159b0: 7665 6420 6279 2072 656d 6f76 696e 6720  ved by removing 
-000159c0: 7468 6520 7061 7261 6c6f 676f 7573 2073  the paralogous s
-000159d0: 7065 6369 6573 2070 6169 7220 6041 7175  pecies pair `Aqu
-000159e0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-000159f0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015a00: 6561 602c 2075 7369 6e67 2074 6865 2063  ea`, using the c
-00015a10: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a60  ommand below...`
-00015a20: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
-00015a30: 645f 676c 6f62 616c 6d72 6268 5f6b 732f  d_globalmrbh_ks/
-00015a40: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
-00015a50: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-00015a60: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015a70: 6967 6874 202d 6f20 7767 645f 676c 6f62  ight -o wgd_glob
-00015a80: 616c 6d72 6268 5f6f 6e6c 796f 7274 686f  almrbh_onlyortho
-00015a90: 5f6b 7320 2d2d 7370 6169 7220 2241 7175  _ks --spair "Aqu
-00015aa0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015ab0: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
-00015ac0: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-00015ad0: 6c65 6769 615f 636f 6572 756c 6561 3b56  legia_coerulea;V
-00015ae0: 6974 6973 5f76 696e 6966 6572 6122 202d  itis_vinifera" -
-00015af0: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015b00: 615f 636f 6572 756c 6561 3b41 636f 7275  a_coerulea;Acoru
-00015b10: 735f 616d 6572 6963 616e 7573 2220 2d2d  s_americanus" --
-00015b20: 6773 6d61 7020 6765 6e65 5f73 7065 6369  gsmap gene_speci
-00015b30: 6573 2e6d 6170 0a60 6060 0a0a 5765 2063  es.map.```..We c
-00015b40: 616e 2063 6c65 6172 6c79 2073 6565 2074  an clearly see t
-00015b50: 6861 7420 2a56 6974 6973 2076 696e 6966  hat *Vitis vinif
-00015b60: 6572 612a 2068 6173 2068 6967 6865 7220  era* has higher 
-00015b70: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00015b80: 6520 7468 616e 202a 5072 6f74 6561 2063  e than *Protea c
-00015b90: 796e 6172 6f69 6465 732a 2069 6e20 7468  ynaroides* in th
-00015ba0: 6174 2074 6865 6972 206f 7274 686f 6c6f  at their ortholo
-00015bb0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00015bc0: 7375 623e 2070 6561 6b73 2077 6974 6820  sub> peaks with 
-00015bd0: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00015be0: 6c65 612a 2c20 616c 7468 6f75 6768 2072  lea*, although r
-00015bf0: 6570 7265 7365 6e74 696e 6720 7468 6520  epresenting the 
-00015c00: 7361 6d65 2064 6976 6572 6765 6e63 6520  same divergence 
-00015c10: 6576 656e 742c 2064 6966 6665 7265 6420  event, differed 
-00015c20: 696e 2073 7562 7374 6974 7574 696f 6e20  in substitution 
-00015c30: 6469 7374 616e 6365 2e0a 215b 5d28 6461  distance..![](da
-00015c40: 7461 2f52 6177 5f4f 7274 686f 6c6f 6775  ta/Raw_Orthologu
-00015c50: 6573 2e6b 7364 2e73 7667 290a 0a41 6674  es.ksd.svg)..Aft
-00015c60: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
-00015c70: 7469 6320 7469 6d69 6e67 206f 6620 7468  tic timing of th
-00015c80: 6520 5261 6e75 6e63 756c 616c 6573 2057  e Ranunculales W
-00015c90: 4744 2c20 7765 2063 616e 2066 7572 7468  GD, we can furth
-00015ca0: 6572 2069 6e66 6572 2069 7473 2061 6273  er infer its abs
-00015cb0: 6f6c 7574 6520 6167 652e 2046 6972 7374  olute age. First
-00015cc0: 2077 6520 696e 6665 7220 7468 6520 6372   we infer the cr
-00015cd0: 6564 6962 6c65 2072 616e 6765 206f 6620  edible range of 
-00015ce0: 616e 6368 6f72 2070 6169 7273 2062 7920  anchor pairs by 
-00015cf0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00015d00: 6865 7572 6973 7469 6361 6c6c 7920 7573  heuristically us
-00015d10: 696e 6720 7468 6520 7072 6f67 7261 6d20  ing the program 
-00015d20: 6077 6764 2070 6561 6b60 2e0a 0a60 6060  `wgd peak`...```
-00015d30: 0a77 6764 2070 6561 6b20 2d2d 6865 7572  .wgd peak --heur
-00015d40: 6973 7469 6320 7767 645f 6b73 642f 4171  istic wgd_ksd/Aq
-00015d50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015d60: 2e74 7376 2e6b 732e 7473 7620 2d61 7020  .tsv.ks.tsv -ap 
-00015d70: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00015d80: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-00015d90: 2e74 7874 202d 736d 2077 6764 5f73 796e  .txt -sm wgd_syn
-00015da0: 2f69 6164 686f 7265 2d6f 7574 2f73 6567  /iadhore-out/seg
-00015db0: 6d65 6e74 732e 7478 7420 2d6c 6520 7767  ments.txt -le wg
-00015dc0: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00015dd0: 742f 6c69 7374 5f65 6c65 6d65 6e74 732e  t/list_elements.
-00015de0: 7478 7420 2d6d 7020 7767 645f 7379 6e2f  txt -mp wgd_syn/
-00015df0: 6961 6468 6f72 652d 6f75 742f 6d75 6c74  iadhore-out/mult
-00015e00: 6970 6c69 636f 6e5f 7061 6972 732e 7478  iplicon_pairs.tx
-00015e10: 7420 2d6f 2077 6764 5f70 6561 6b0a 6060  t -o wgd_peak.``
-00015e20: 600a 0a21 5b5d 2864 6174 612f 416e 6368  `..![](data/Anch
-00015e30: 6f72 4b73 5f50 6561 6b43 495f 4171 7569  orKs_PeakCI_Aqui
-00015e40: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-00015e50: 7376 2e6b 732e 7473 765f 6e6f 6465 5f77  sv.ks.tsv_node_w
-00015e60: 6569 6768 7465 642e 7376 6729 0a0a 4173  eighted.svg)..As
-00015e70: 2073 686f 776e 2061 626f 7665 2c20 7765   shown above, we
-00015e80: 2061 7373 756d 6564 2061 206c 6f67 6e6f   assumed a logno
-00015e90: 726d 616c 2064 6973 7472 6962 7574 696f  rmal distributio
-00015ea0: 6e20 6174 2074 6865 2070 6561 6b20 6c6f  n at the peak lo
-00015eb0: 6361 7469 6f6e 2064 6574 6563 7465 6420  cation detected 
-00015ec0: 6279 2074 6865 2060 7369 676e 616c 6020  by the `signal` 
-00015ed0: 6d6f 6475 6c65 206f 6620 6073 6369 7079  module of `scipy
-00015ee0: 6020 6c69 6272 6172 792e 2054 6865 2039  ` library. The 9
-00015ef0: 3525 2063 6f6e 6669 6465 6e63 6520 6c65  5% confidence le
-00015f00: 7665 6c20 6f66 2074 6865 206c 6f67 6e6f  vel of the logno
-00015f10: 726d 616c 2064 6973 7472 6962 7574 696f  rmal distributio
-00015f20: 6e20 7761 7320 6170 706c 6965 642c 2069  n was applied, i
-00015f30: 2e65 2e2c 2030 2e36 382d 322e 3734 2c20  .e., 0.68-2.74, 
-00015f40: 696e 2066 7572 7468 6572 206d 6f6c 6563  in further molec
-00015f50: 756c 6172 2064 6174 696e 672e 2054 6865  ular dating. The
-00015f60: 2066 696c 6520 6041 7175 696c 6567 6961   file `Aquilegia
-00015f70: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00015f80: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
-00015f90: 725f 6461 7469 6e67 5f77 6569 6768 7465  r_dating_weighte
-00015fa0: 645f 666f 726d 6174 2e74 7376 6020 6973  d_format.tsv` is
-00015fb0: 2077 6861 7420 7765 206e 6565 6420 666f   what we need fo
-00015fc0: 7220 6e65 7874 2073 7465 702e 2054 6f20  r next step. To 
-00015fd0: 6275 696c 6420 7468 6520 6f72 7468 6f67  build the orthog
-00015fe0: 726f 7570 7320 7573 6564 2069 6e20 7068  roups used in ph
-00015ff0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-00016000: 672c 2077 6520 6e65 6564 2074 6f20 7365  g, we need to se
-00016010: 6c65 6374 2073 6f6d 6520 7370 6563 6965  lect some specie
-00016020: 7320 616e 6420 666f 726d 2061 2073 7461  s and form a sta
-00016030: 7274 696e 6720 7472 6565 2077 6974 6820  rting tree with 
-00016040: 7072 6f70 6572 2066 6f73 7369 6c20 6361  proper fossil ca
-00016050: 6c69 6272 6174 696f 6e73 2e20 5765 2070  librations. We p
-00016060: 726f 7669 6465 206f 6e65 2069 6e20 6d63  rovide one in mc
-00016070: 6d63 7472 6565 2066 6f72 6d61 7420 6173  mctree format as
-00016080: 2062 656c 6f77 2e0a 0a60 6060 0a31 3720   below...```.17 
-00016090: 310a 2828 2828 506f 7461 6d6f 6765 746f  1.((((Potamogeto
-000160a0: 6e5f 6163 7574 6966 6f6c 6975 732c 2853  n_acutifolius,(S
-000160b0: 7069 726f 6465 6c61 5f69 6e74 6572 6d65  pirodela_interme
-000160c0: 6469 612c 416d 6f72 7068 6f70 6861 6c6c  dia,Amorphophall
-000160d0: 7573 5f6b 6f6e 6a61 6329 292c 2841 6361  us_konjac)),(Aca
-000160e0: 6e74 686f 6368 6c61 6d79 735f 6272 6163  nthochlamys_brac
-000160f0: 7465 6174 612c 2844 696f 7363 6f72 6561  teata,(Dioscorea
-00016100: 5f61 6c61 7461 2c44 696f 7363 6f72 6561  _alata,Dioscorea
-00016110: 5f72 6f74 756e 6461 7461 2929 273e 302e  _rotundata))'>0.
-00016120: 3536 3030 3c31 2e32 3836 3327 2927 3e30  5600<1.2863')'>0
-00016130: 2e38 3336 303c 312e 3238 3633 272c 2841  .8360<1.2863',(A
-00016140: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00016150: 2c41 636f 7275 735f 7461 7461 7269 6e6f  ,Acorus_tatarino
-00016160: 7769 6929 2927 3e30 2e38 3336 303c 312e  wii))'>0.8360<1.
-00016170: 3238 3633 272c 2828 2828 5465 7472 6163  2863',((((Tetrac
-00016180: 656e 7472 6f6e 5f73 696e 656e 7365 2c54  entron_sinense,T
-00016190: 726f 6368 6f64 656e 6472 6f6e 5f61 7261  rochodendron_ara
-000161a0: 6c69 6f69 6465 7329 2c28 4275 7875 735f  lioides),(Buxus_
-000161b0: 6175 7374 726f 7975 6e6e 616e 656e 7369  austroyunnanensi
-000161c0: 732c 4275 7875 735f 7369 6e69 6361 2929  s,Buxus_sinica))
-000161d0: 273e 312e 3130 3830 3c31 2e32 3836 3327  '>1.1080<1.2863'
-000161e0: 2c28 4e65 6c75 6d62 6f5f 6e75 6369 6665  ,(Nelumbo_nucife
-000161f0: 7261 2c28 5465 6c6f 7065 615f 7370 6563  ra,(Telopea_spec
-00016200: 696f 7369 7373 696d 612c 5072 6f74 6561  iosissima,Protea
-00016210: 5f63 796e 6172 6f69 6465 7329 2929 273e  _cynaroides)))'>
-00016220: 312e 3130 3830 3c31 2e32 3836 3327 2c28  1.1080<1.2863',(
-00016230: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016240: 6561 5f61 7031 2c41 7175 696c 6567 6961  ea_ap1,Aquilegia
-00016250: 5f63 6f65 7275 6c65 615f 6170 3229 2927  _coerulea_ap2))'
-00016260: 3e31 2e31 3038 303c 312e 3238 3633 2729  >1.1080<1.2863')
-00016270: 273e 312e 3237 3230 3c32 2e34 3732 3027  '>1.2720<2.4720'
-00016280: 3b0a 6060 600a 0a41 7320 7072 6573 656e  ;.```..As presen
-00016290: 7465 6420 6162 6f76 652c 2074 6865 2066  ted above, the f
-000162a0: 6f63 7573 2073 7065 6369 6573 2074 6861  ocus species tha
-000162b0: 7420 6973 2061 626f 7574 2074 6f20 6265  t is about to be
-000162c0: 2064 6174 6564 206e 6565 6473 2074 6f20   dated needs to 
-000162d0: 6265 2072 6570 6c61 6365 6420 7769 7468  be replaced with
-000162e0: 2060 2841 7175 696c 6567 6961 5f63 6f65   `(Aquilegia_coe
-000162f0: 7275 6c65 615f 6170 312c 4171 7569 6c65  rulea_ap1,Aquile
-00016300: 6769 615f 636f 6572 756c 6561 5f61 7032  gia_coerulea_ap2
-00016310: 2960 2e20 5769 7468 2074 6869 7320 7374  )`. With this st
-00016320: 6172 7469 6e67 2074 7265 6520 616e 6420  arting tree and 
-00016330: 7072 6564 6f77 6e6c 6f61 6465 6420 6364  predownloaded cd
-00016340: 7320 6669 6c65 7320 6f66 2061 6c6c 2074  s files of all t
-00016350: 6865 2073 7065 6369 6573 2c20 7765 2063  he species, we c
-00016360: 616e 2062 7569 6c64 2074 6865 206f 7274  an build the ort
-00016370: 686f 6772 6f75 7020 7573 6564 2069 6e20  hogroup used in 
-00016380: 7468 6520 6669 6e61 6c20 6d6f 6c65 6375  the final molecu
-00016390: 6c61 7220 6461 7469 6e67 2075 7369 6e67  lar dating using
-000163a0: 2074 6865 2063 6f6d 6d61 6e64 2061 7320   the command as 
-000163b0: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
-000163c0: 646d 6420 2d66 2041 7175 696c 6567 6961  dmd -f Aquilegia
-000163d0: 5f63 6f65 7275 6c65 6120 2d61 7020 7767  _coerulea -ap wg
-000163e0: 645f 7065 616b 2f41 7175 696c 6567 6961  d_peak/Aquilegia
-000163f0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00016400: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
-00016410: 725f 6461 7469 6e67 5f77 6569 6768 7465  r_dating_weighte
-00016420: 645f 666f 726d 6174 2e74 7376 202d 6f20  d_format.tsv -o 
-00016430: 7767 645f 646d 645f 6f72 7468 6f20 506f  wgd_dmd_ortho Po
-00016440: 7461 6d6f 6765 746f 6e5f 6163 7574 6966  tamogeton_acutif
-00016450: 6f6c 6975 7320 5370 6972 6f64 656c 615f  olius Spirodela_
-00016460: 696e 7465 726d 6564 6961 2041 6d6f 7270  intermedia Amorp
-00016470: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
-00016480: 2041 6361 6e74 686f 6368 6c61 6d79 735f   Acanthochlamys_
-00016490: 6272 6163 7465 6174 6120 4469 6f73 636f  bracteata Diosco
-000164a0: 7265 615f 616c 6174 6120 4469 6f73 636f  rea_alata Diosco
-000164b0: 7265 615f 726f 7475 6e64 6174 6120 4163  rea_rotundata Ac
-000164c0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
-000164d0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
-000164e0: 6969 2054 6574 7261 6365 6e74 726f 6e5f  ii Tetracentron_
-000164f0: 7369 6e65 6e73 6520 5472 6f63 686f 6465  sinense Trochode
-00016500: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
-00016510: 2042 7578 7573 5f61 7573 7472 6f79 756e   Buxus_austroyun
-00016520: 6e61 6e65 6e73 6973 2042 7578 7573 5f73  nanensis Buxus_s
-00016530: 696e 6963 6120 4e65 6c75 6d62 6f5f 6e75  inica Nelumbo_nu
-00016540: 6369 6665 7261 2054 656c 6f70 6561 5f73  cifera Telopea_s
-00016550: 7065 6369 6f73 6973 7369 6d61 2050 726f  peciosissima Pro
-00016560: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
-00016570: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00016580: 610a 6060 600a 0a54 6865 2072 6573 756c  a.```..The resul
-00016590: 7420 6669 6c65 2060 6d65 7267 655f 666f  t file `merge_fo
-000165a0: 6375 735f 6170 2e74 7376 6020 6973 2077  cus_ap.tsv` is w
-000165b0: 6861 7420 7765 206e 6565 6420 666f 7220  hat we need for 
-000165c0: 7468 6520 6669 6e61 6c20 7374 6570 206f  the final step o
-000165d0: 6620 6d6f 6c65 6375 6c61 7220 6461 7469  f molecular dati
-000165e0: 6e67 2069 6e20 7072 6f67 7261 6d20 6077  ng in program `w
-000165f0: 6764 2066 6f63 7573 602e 0a0a 6060 600a  gd focus`...```.
-00016600: 7767 6420 666f 6375 7320 2d2d 7072 6f74  wgd focus --prot
-00016610: 6461 7469 6e67 202d 2d61 616d 6f64 656c  dating --aamodel
-00016620: 206c 6720 7767 645f 646d 645f 6f72 7468   lg wgd_dmd_orth
-00016630: 6f2f 6d65 7267 655f 666f 6375 735f 6170  o/merge_focus_ap
-00016640: 2e74 7376 202d 7370 2064 6174 696e 675f  .tsv -sp dating_
-00016650: 7472 6565 2e6e 7720 2d6f 2077 6764 5f64  tree.nw -o wgd_d
-00016660: 6174 696e 6720 2d64 206d 636d 6374 7265  ating -d mcmctre
-00016670: 6520 2d64 7320 2762 7572 6e69 6e20 3d20  e -ds 'burnin = 
-00016680: 3230 3030 2720 2d64 7320 2773 616d 7066  2000' -ds 'sampf
-00016690: 7265 7120 3d20 3130 3030 2720 2d64 7320  req = 1000' -ds 
-000166a0: 276e 7361 6d70 6c65 203d 2032 3030 3030  'nsample = 20000
-000166b0: 2720 506f 7461 6d6f 6765 746f 6e5f 6163  ' Potamogeton_ac
-000166c0: 7574 6966 6f6c 6975 7320 5370 6972 6f64  utifolius Spirod
-000166d0: 656c 615f 696e 7465 726d 6564 6961 2041  ela_intermedia A
-000166e0: 6d6f 7270 686f 7068 616c 6c75 735f 6b6f  morphophallus_ko
-000166f0: 6e6a 6163 2041 6361 6e74 686f 6368 6c61  njac Acanthochla
-00016700: 6d79 735f 6272 6163 7465 6174 6120 4469  mys_bracteata Di
-00016710: 6f73 636f 7265 615f 616c 6174 6120 4469  oscorea_alata Di
-00016720: 6f73 636f 7265 615f 726f 7475 6e64 6174  oscorea_rotundat
-00016730: 6120 4163 6f72 7573 5f61 6d65 7269 6361  a Acorus_america
-00016740: 6e75 7320 4163 6f72 7573 5f74 6174 6172  nus Acorus_tatar
-00016750: 696e 6f77 6969 2054 6574 7261 6365 6e74  inowii Tetracent
-00016760: 726f 6e5f 7369 6e65 6e73 6520 5472 6f63  ron_sinense Troc
-00016770: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
-00016780: 6964 6573 2042 7578 7573 5f61 7573 7472  ides Buxus_austr
-00016790: 6f79 756e 6e61 6e65 6e73 6973 2042 7578  oyunnanensis Bux
-000167a0: 7573 5f73 696e 6963 6120 4e65 6c75 6d62  us_sinica Nelumb
-000167b0: 6f5f 6e75 6369 6665 7261 2054 656c 6f70  o_nucifera Telop
-000167c0: 6561 5f73 7065 6369 6f73 6973 7369 6d61  ea_speciosissima
-000167d0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
-000167e0: 6573 2041 7175 696c 6567 6961 5f63 6f65  es Aquilegia_coe
-000167f0: 7275 6c65 610a 6060 600a 0a48 6572 6520  rulea.```..Here 
-00016800: 7765 206f 6e6c 7920 696d 706c 656d 656e  we only implemen
-00016810: 7465 6420 7468 6520 636f 6e63 6174 656e  ted the concaten
-00016820: 6174 696f 6e20 616e 616c 7973 6973 2075  ation analysis u
-00016830: 7369 6e67 2070 726f 7465 696e 2073 6571  sing protein seq
-00016840: 7565 6e63 6520 6279 2061 6464 696e 6720  uence by adding 
-00016850: 7468 6520 666c 6167 2060 2d2d 7072 6f74  the flag `--prot
-00016860: 6461 7469 6e67 6020 616e 6420 7765 2073  dating` and we s
-00016870: 6574 2074 6865 2070 6172 616d 6574 6572  et the parameter
-00016880: 2066 6f72 2060 6d63 6d63 7472 6565 6020   for `mcmctree` 
-00016890: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-000168a0: 2d64 7360 2e20 4e6f 7465 2074 6861 7420  -ds`. Note that 
-000168b0: 6f74 6865 7220 6461 7469 6e67 2070 726f  other dating pro
-000168c0: 6772 616d 2073 7563 6820 6173 2060 7238  gram such as `r8
-000168d0: 7360 2061 6e64 2060 6265 6173 7460 2061  s` and `beast` a
-000168e0: 7265 2061 6c73 6f20 6176 6169 6c61 626c  re also availabl
-000168f0: 6520 6769 7665 6e20 736f 6d65 206d 616e  e given some man
-00016900: 6461 746f 7279 2070 6172 616d 6574 6572  datory parameter
-00016910: 732e 2054 6865 2066 696e 616c 206c 6f67  s. The final log
-00016920: 206f 6620 7468 6520 7375 6363 6573 7366   of the successf
-00016930: 756c 2072 756e 2069 7320 6173 2062 656c  ul run is as bel
-00016940: 6f77 2e0a 0a60 6060 0a31 363a 3034 3a32  ow...```.16:04:2
-00016950: 3520 494e 464f 2020 2020 2052 756e 6e69  5 INFO     Runni
-00016960: 6e67 206d 636d 6374 7265 6520 7573 696e  ng mcmctree usin
-00016970: 6720 4865 7373 6961 6e20 6d61 7472 6978  g Hessian matrix
-00016980: 206f 6620 4c47 2b47 616d 6d61 2020 636f   of LG+Gamma  co
-00016990: 7265 2e70 793a 3936 370a 2020 2020 2020  re.py:967.      
-000169a0: 2020 2020 2020 2020 2020 2020 666f 7220              for 
-000169b0: 7072 6f74 6569 6e20 6d6f 6465 6c0a 3233  protein model.23
-000169c0: 3a34 393a 3337 2049 4e46 4f20 2020 2020  :49:37 INFO     
-000169d0: 506f 7374 6572 696f 7220 6d65 616e 2066  Posterior mean f
-000169e0: 6f72 2074 6865 2061 6765 7320 6f66 2077  or the ages of w
-000169f0: 6764 2069 7320 3131 322e 3839 3435 206d  gd is 112.8945 m
-00016a00: 636d 6374 7265 652e 7079 3a32 3936 0a20  cmctree.py:296. 
-00016a10: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016a20: 206d 696c 6c69 6f6e 2079 6561 7273 2066   million years f
-00016a30: 726f 6d20 436f 6e63 6174 656e 6174 6564  rom Concatenated
-00016a40: 2070 6570 7469 6465 0a20 2020 2020 2020   peptide.       
-00016a50: 2020 2020 2020 2020 2020 2061 6c69 676e             align
-00016a60: 6d65 6e74 2061 6e64 2039 3525 2063 7265  ment and 95% cre
-00016a70: 6469 6269 6c69 7479 2069 6e74 6572 7661  dibility interva
-00016a80: 6c73 2028 4349 290a 2020 2020 2020 2020  ls (CI).        
-00016a90: 2020 2020 2020 2020 2020 6973 2031 3031            is 101
-00016aa0: 2e32 3234 2d31 3233 2e31 3231 206d 696c  .224-123.121 mil
-00016ab0: 6c69 6f6e 2079 6561 7273 0a20 2020 2020  lion years.     
-00016ac0: 2020 2020 494e 464f 2020 2020 2054 6f74      INFO     Tot
-00016ad0: 616c 2072 756e 2074 696d 653a 2032 3931  al run time: 291
-00016ae0: 3735 7320 2020 2020 2020 2020 2020 2020  75s             
-00016af0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016b00: 2063 6c69 2e70 793a 3234 310a 2020 2020   cli.py:241.    
-00016b10: 2020 2020 2049 4e46 4f20 2020 2020 446f       INFO     Do
-00016b20: 6e65 2020 2020 2020 2020 2020 2020 2020  ne              
-00016b30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016b40: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016b50: 2020 636c 692e 7079 3a32 3432 0a60 6060    cli.py:242.```
-00016b60: 0a0a 546f 2076 6973 7561 6c69 7a65 2074  ..To visualize t
-00016b70: 6865 2064 6174 652c 2077 6520 616c 736f  he date, we also
-00016b80: 2070 726f 7669 6465 6420 6120 7079 7468   provided a pyth
-00016b90: 6f6e 2073 6372 6970 7420 746f 2070 6c6f  on script to plo
-00016ba0: 7420 7468 6520 5747 4420 6461 7465 7320  t the WGD dates 
-00016bb0: 696e 2074 6865 2060 7767 6460 2066 6f6c  in the `wgd` fol
-00016bc0: 6465 722e 2055 7365 7273 206e 6565 6420  der. Users need 
-00016bd0: 746f 2065 7874 7261 6374 2074 6865 2072  to extract the r
-00016be0: 6177 2064 6174 6573 2066 726f 6d20 7468  aw dates from th
-00016bf0: 6520 606d 636d 632e 7478 7460 2066 6f72  e `mcmc.txt` for
-00016c00: 2074 6865 2057 4744 206e 6f64 6520 6669   the WGD node fi
-00016c10: 7273 7420 616e 6420 7361 7665 2069 7420  rst and save it 
-00016c20: 6173 2066 696c 6520 6064 6174 6573 2e74  as file `dates.t
-00016c30: 7874 6020 286f 7220 7768 6174 6576 6572  xt` (or whatever
-00016c40: 2070 7265 6665 7272 6564 206e 616d 6529   preferred name)
-00016c50: 2e20 416e 2065 7861 6d70 6c65 2063 6f6d  . An example com
-00016c60: 6d61 6e64 2069 7320 6173 2062 656c 6f77  mand is as below
-00016c70: 2e0a 0a60 6060 0a70 7974 686f 6e20 2450  ...```.python $P
-00016c80: 4154 482f 706f 7374 706c 6f74 2e70 7920  ATH/postplot.py 
-00016c90: 706f 7374 6469 7320 6461 7465 732e 7478  postdis dates.tx
-00016ca0: 7420 2d2d 7065 7263 656e 7469 6c65 2039  t --percentile 9
-00016cb0: 3020 2d2d 7469 746c 6520 2257 4744 2064  0 --title "WGD d
-00016cc0: 6174 6522 202d 2d68 7064 202d 6f20 2252  ate" --hpd -o "R
-00016cd0: 616e 756e 6375 6c61 6c65 735f 5747 445f  anunculales_WGD_
-00016ce0: 6461 7465 2e73 7667 220a 6060 600a 0a21  date.svg".```..!
-00016cf0: 5b5d 2864 6174 612f 5261 6e75 6e63 756c  [](data/Ranuncul
-00016d00: 616c 6573 5f57 4744 5f64 6174 652e 7376  ales_WGD_date.sv
-00016d10: 6729 0a0a 5468 6520 706f 7374 6572 696f  g)..The posterio
-00016d20: 7220 6d65 616e 2c20 6d65 6469 616e 2061  r mean, median a
-00016d30: 6e64 206d 6f64 6520 6f66 2074 6865 2052  nd mode of the R
-00016d40: 616e 756e 6375 6c61 6c65 7320 5747 4420  anunculales WGD 
-00016d50: 6167 6520 6973 2031 3132 2e39 322c 2031  age is 112.92, 1
-00016d60: 3133 2e34 3420 616e 6420 3131 322e 3534  13.44 and 112.54
-00016d70: 206d 7961 2c20 7769 7468 2039 3025 2048   mya, with 90% H
-00016d80: 5044 2031 3035 2e30 3720 2d20 3132 322e  PD 105.07 - 122.
-00016d90: 3332 206d 7961 2061 7320 6d61 6e69 6665  32 mya as manife
-00016da0: 7374 6564 2061 626f 7665 2e0a 0a23 2323  sted above...###
-00016db0: 204b 7374 7265 650a 0a49 6e20 6164 6469   Kstree..In addi
-00016dc0: 7469 6f6e 2074 6f20 7061 6972 7769 7365  tion to pairwise
-00016dd0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00016de0: 2065 7374 696d 6174 696f 6e2c 2061 202a   estimation, a *
-00016df0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2074  K*<sub>S</sub> t
-00016e00: 7265 6520 7769 7468 2062 7261 6e63 6820  ree with branch 
-00016e10: 6c65 6e67 7468 2069 6e20 2a4b 2a3c 7375  length in *K*<su
-00016e20: 623e 533c 2f73 7562 3e20 756e 6974 2063  b>S</sub> unit c
-00016e30: 616e 2061 6c73 6f20 6265 2064 6572 6976  an also be deriv
-00016e40: 6564 2066 726f 6d20 7468 6520 7072 6f67  ed from the prog
-00016e50: 7261 6d20 6077 6764 206b 7364 6020 6769  ram `wgd ksd` gi
-00016e60: 7665 6e20 7468 6520 6f70 7469 6f6e 2060  ven the option `
-00016e70: 2d2d 6b73 7472 6565 6020 616e 6420 602d  --kstree` and `-
-00016e80: 2d73 7065 6369 6573 7472 6565 602e 204e  -speciestree`. N
-00016e90: 6f74 6520 7468 6174 2074 6865 2061 6464  ote that the add
-00016ea0: 6974 696f 6e61 6c20 6f70 7469 6f6e 2060  itional option `
-00016eb0: 2d2d 6f6e 6c79 636f 6e63 6174 6b73 7472  --onlyconcatkstr
-00016ec0: 6565 6020 7769 6c6c 206f 6e6c 7920 6361  ee` will only ca
-00016ed0: 6c6c 2074 6865 202a 4b2a 3c73 7562 3e53  ll the *K*<sub>S
-00016ee0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
-00016ef0: 6e20 666f 7220 7468 6520 636f 6e63 6174  n for the concat
-00016f00: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
-00016f10: 2072 6174 6865 7220 7468 616e 2061 6c6c   rather than all
-00016f20: 2074 6865 2061 6c69 676e 6d65 6e74 732e   the alignments.
-00016f30: 2055 7365 7273 206e 6565 6420 746f 2070   Users need to p
-00016f40: 726f 7669 6465 2061 2070 7265 7365 7420  rovide a preset 
-00016f50: 7370 6563 6965 7320 7472 6565 2066 6f72  species tree for
-00016f60: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00016f70: 7375 623e 2074 7265 6520 696e 6665 7265  sub> tree infere
-00016f80: 6e63 6520 6f66 2074 6865 2063 6f6e 6361  nce of the conca
-00016f90: 7465 6e61 7465 6420 616c 6967 6e6d 656e  tenated alignmen
-00016fa0: 7420 7768 696c 6520 7468 6520 7265 6d61  t while the rema
-00016fb0: 696e 696e 6720 616c 6967 6e6d 656e 7473  ining alignments
-00016fc0: 2077 696c 6c20 6265 2061 6761 696e 7374   will be against
-00016fd0: 2061 6e20 6175 746f 6d61 7465 6c79 2069   an automately i
-00016fe0: 6e66 6572 7265 6420 7472 6565 2066 726f  nferred tree fro
-00016ff0: 6d20 6066 6173 7474 7265 6560 206f 7220  m `fasttree` or 
-00017000: 6069 7174 7265 6560 2e20 496e 2074 6865  `iqtree`. In the
-00017010: 2065 6e64 2c20 7573 6572 7320 7769 6c6c   end, users will
-00017020: 2067 6574 2061 202a 4b2a 3c73 7562 3e53   get a *K*<sub>S
-00017030: 3c2f 7375 623e 2074 7265 652c 2061 202a  </sub> tree, a *
-00017040: 4b2a 3c73 7562 3e41 3c2f 7375 623e 2074  K*<sub>A</sub> t
-00017050: 7265 6520 616e 6420 6120 cf89 2074 7265  ree and a .. tre
-00017060: 6520 7065 7220 6661 6d20 616e 6420 666f  e per fam and fo
-00017070: 7220 7468 6520 636f 6e63 6174 656e 6174  r the concatenat
-00017080: 6564 2061 6c69 676e 6d65 6e74 2e0a 0a60  ed alignment...`
-00017090: 6060 0a77 6764 206b 7364 2064 6174 612f  ``.wgd ksd data/
-000170a0: 6b73 7472 6565 5f64 6174 612f 6661 6d2e  kstree_data/fam.
-000170b0: 7473 7620 6461 7461 2f6b 7374 7265 655f  tsv data/kstree_
-000170c0: 6461 7461 2f41 636f 7275 735f 7461 7461  data/Acorus_tata
-000170d0: 7269 6e6f 7769 6920 6461 7461 2f6b 7374  rinowii data/kst
-000170e0: 7265 655f 6461 7461 2f41 6d62 6f72 656c  ree_data/Amborel
-000170f0: 6c61 5f74 7269 6368 6f70 6f64 6120 6461  la_trichopoda da
-00017100: 7461 2f6b 7374 7265 655f 6461 7461 2f41  ta/kstree_data/A
-00017110: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00017120: 6120 6461 7461 2f6b 7374 7265 655f 6461  a data/kstree_da
-00017130: 7461 2f41 7269 7374 6f6c 6f63 6869 615f  ta/Aristolochia_
-00017140: 6669 6d62 7269 6174 6120 6461 7461 2f6b  fimbriata data/k
-00017150: 7374 7265 655f 6461 7461 2f43 7963 6173  stree_data/Cycas
-00017160: 5f70 616e 7a68 6968 7561 656e 7369 202d  _panzhihuaensi -
-00017170: 2d6b 7374 7265 6520 2d2d 7370 6563 6965  -kstree --specie
-00017180: 7374 7265 6520 6461 7461 2f6b 7374 7265  stree data/kstre
-00017190: 655f 6461 7461 2f73 7065 6369 6573 5f74  e_data/species_t
-000171a0: 7265 6531 2e6e 7720 2d2d 6f6e 6c79 636f  ree1.nw --onlyco
-000171b0: 6e63 6174 6b73 7472 6565 202d 6f20 7767  ncatkstree -o wg
-000171c0: 645f 6b73 7472 6565 5f74 6f70 6f6c 6f67  d_kstree_topolog
-000171d0: 7931 0a60 6060 0a0a 215b 5d28 6461 7461  y1.```..![](data
-000171e0: 2f6b 7374 7265 655f 7265 7375 6c74 732f  /kstree_results/
-000171f0: 6b73 7472 6565 2e73 7667 290a 0a41 626f  kstree.svg)..Abo
-00017200: 7665 2077 6520 7573 6564 2074 6872 6565  ve we used three
-00017210: 2061 6c74 6572 6e61 7469 7665 2074 6f70   alternative top
-00017220: 6f6c 6f67 6965 7320 746f 2069 6e66 6572  ologies to infer
-00017230: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00017240: 7375 623e 2074 7265 6520 7768 6963 6820  sub> tree which 
-00017250: 6c65 6420 746f 2064 6966 6665 7265 6e74  led to different
-00017260: 2062 7261 6e63 6820 6c65 6e67 7468 2065   branch length e
-00017270: 7374 696d 6174 696f 6e2e 204e 6f74 6520  stimation. Note 
-00017280: 7468 6174 2074 6865 2066 616d 696c 6965  that the familie
-00017290: 7320 7765 2075 7365 6420 7765 7265 206f  s we used were o
-000172a0: 6e6c 7920 7477 6f20 676c 6f62 616c 204d  nly two global M
-000172b0: 5242 4820 6661 6d69 6c69 6573 2066 6f72  RBH families for
-000172c0: 2074 6865 2070 7572 706f 7365 206f 6620   the purpose of 
-000172d0: 696c 6c75 7374 7261 7469 6f6e 2e20 546f  illustration. To
-000172e0: 2061 6371 7569 7265 2061 6e20 6163 6375   acquire an accu
-000172f0: 7261 7465 2070 726f 6669 6c65 206f 6620  rate profile of 
-00017300: 7468 6520 7375 6273 7469 7475 7469 6f6e  the substitution
-00017310: 2072 6174 6520 7661 7269 6174 696f 6e2c   rate variation,
-00017320: 206f 7274 686f 6c6f 6775 6573 2061 7420   orthologues at 
-00017330: 7468 6520 7768 6f6c 6520 6765 6e6f 6d65  the whole genome
-00017340: 2073 6361 6c65 2073 686f 756c 6420 6265   scale should be
-00017350: 2075 7365 642e 0a0a 2323 2043 6974 6174   used...## Citat
-00017360: 696f 6e0a 200a 506c 6561 7365 2063 6974  ion. .Please cit
-00017370: 6520 7573 2061 7420 6874 7470 733a 2f2f  e us at https://
-00017380: 646f 692e 6f72 672f 3130 2e31 3030 372f  doi.org/10.1007/
-00017390: 3937 382d 312d 3037 3136 2d32 3536 312d  978-1-0716-2561-
-000173a0: 335f 3120 616e 6420 6874 7470 733a 2f2f  3_1 and https://
-000173b0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
-000173c0: 6269 6f69 6e66 6f72 6d61 7469 6373 2f62  bioinformatics/b
-000173d0: 7461 6532 3732 2e0a 0a60 6060 0a48 656e  tae272...```.Hen
-000173e0: 6763 6869 2043 6865 6e2c 2041 7274 6875  gchi Chen, Arthu
-000173f0: 7220 5a77 6165 6e65 706f 656c 2028 3230  r Zwaenepoel (20
-00017400: 3233 292e 2049 6e66 6572 656e 6365 206f  23). Inference o
-00017410: 6620 416e 6369 656e 7420 506f 6c79 706c  f Ancient Polypl
-00017420: 6f69 6479 2066 726f 6d20 4765 6e6f 6d69  oidy from Genomi
-00017430: 6320 4461 7461 2e20 496e 3a20 5661 6e20  c Data. In: Van 
-00017440: 6465 2050 6565 722c 2059 2e20 2865 6473  de Peer, Y. (eds
-00017450: 2920 506f 6c79 706c 6f69 6479 2e20 4d65  ) Polyploidy. Me
-00017460: 7468 6f64 7320 696e 204d 6f6c 6563 756c  thods in Molecul
-00017470: 6172 2042 696f 6c6f 6779 2c20 766f 6c20  ar Biology, vol 
-00017480: 3235 3435 2e20 4875 6d61 6e61 2c20 4e65  2545. Humana, Ne
-00017490: 7720 596f 726b 2c20 4e59 2e20 6874 7470  w York, NY. http
-000174a0: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
-000174b0: 3030 372f 3937 382d 312d 3037 3136 2d32  007/978-1-0716-2
-000174c0: 3536 312d 335f 310a 4865 6e67 6368 6920  561-3_1.Hengchi 
-000174d0: 4368 656e 2c20 4172 7468 7572 205a 7761  Chen, Arthur Zwa
-000174e0: 656e 6570 6f65 6c2c 2059 7665 7320 5661  enepoel, Yves Va
-000174f0: 6e20 6465 2050 6565 7220 2832 3032 3429  n de Peer (2024)
-00017500: 2e20 7767 6420 7632 3a20 6120 7375 6974  . wgd v2: a suit
-00017510: 6520 6f66 2074 6f6f 6c73 2074 6f20 756e  e of tools to un
-00017520: 636f 7665 7220 616e 6420 6461 7465 2061  cover and date a
-00017530: 6e63 6965 6e74 2070 6f6c 7970 6c6f 6964  ncient polyploid
-00017540: 7920 616e 6420 7768 6f6c 652d 6765 6e6f  y and whole-geno
-00017550: 6d65 2064 7570 6c69 6361 7469 6f6e 2e20  me duplication. 
-00017560: 4269 6f69 6e66 6f72 6d61 7469 6373 2c20  Bioinformatics, 
-00017570: 6874 7470 733a 2f2f 646f 692e 6f72 672f  https://doi.org/
-00017580: 3130 2e31 3039 332f 6269 6f69 6e66 6f72  10.1093/bioinfor
-00017590: 6d61 7469 6373 2f62 7461 6532 3732 0a60  matics/btae272.`
-000175a0: 6060 0a0a 466f 7220 6369 7461 7469 6f6e  ``..For citation
-000175b0: 206f 6620 7468 6520 746f 6f6c 7320 7573   of the tools us
-000175c0: 6564 2069 6e20 7767 642c 2070 6c65 6173  ed in wgd, pleas
-000175d0: 6520 636f 6e73 756c 7420 7468 6520 646f  e consult the do
-000175e0: 6375 6d65 6e74 6174 696f 6e20 6174 0a68  cumentation at.h
-000175f0: 7474 7073 3a2f 2f77 6764 7632 2e72 6561  ttps://wgdv2.rea
-00017600: 6474 6865 646f 6373 2e69 6f2f 656e 2f6c  dthedocs.io/en/l
-00017610: 6174 6573 742f 6369 7461 7469 6f6e 2e68  atest/citation.h
-00017620: 746d 6c2e 0a0a                           tml...
+00005310: 2075 7369 6e67 2074 6865 206f 7574 6772   using the outgr
+00005320: 6f75 7020 6174 2072 6f6f 7420 6173 206f  oup at root as o
+00005330: 7574 6772 6f75 702c 2069 6620 7468 6520  utgroup, if the 
+00005340: 666c 6167 2077 6173 2073 6574 2c20 6f6e  flag was set, on
+00005350: 6c79 2074 6865 206f 7574 6772 6f75 7020  ly the outgroup 
+00005360: 6174 2072 6f6f 7420 776f 756c 6420 6265  at root would be
+00005370: 2075 7365 6420 6173 206f 7574 6772 6f75   used as outgrou
+00005380: 700a 2d65 706b 2c20 2d2d 6578 7472 6170  p.-epk, --extrap
+00005390: 6172 616e 6f6d 656b 732c 2065 7874 7261  aranomeks, extra
+000053a0: 2070 6172 616e 6f6d 6520 4b73 2064 6174   paranome Ks dat
+000053b0: 6120 746f 2070 6c6f 7420 696e 2074 6865  a to plot in the
+000053c0: 206d 6978 6564 204b 7320 6469 7374 7269   mixed Ks distri
+000053d0: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
+000053e0: 4e6f 6e65 0a2d 6170 2c20 2d2d 616e 6368  None.-ap, --anch
+000053f0: 6f72 706f 696e 7473 2c20 616e 6368 6f72  orpoints, anchor
+00005400: 706f 696e 7473 2e74 7874 2066 696c 6520  points.txt file 
+00005410: 746f 2070 6c6f 7420 616e 6368 6f72 204b  to plot anchor K
+00005420: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
+00005430: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00005440: 6465 6661 756c 7420 4e6f 6e65 0a2d 706b  default None.-pk
+00005450: 2c20 2d2d 706c 6f74 6b64 652c 2066 6c61  , --plotkde, fla
+00005460: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00005470: 7220 746f 2070 6c6f 7420 6b64 6520 6375  r to plot kde cu
+00005480: 7276 6520 6f66 206f 7274 686f 6c6f 676f  rve of orthologo
+00005490: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
+000054a0: 6f6e 206f 7665 7220 6869 7374 6f67 7261  on over histogra
+000054b0: 6d20 696e 2074 6865 206d 6978 6564 204b  m in the mixed K
+000054c0: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+000054d0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+000054e0: 7365 742c 2074 6865 206b 6465 2063 7572  set, the kde cur
+000054f0: 7665 2077 6f75 6c64 2062 6520 706c 6f74  ve would be plot
+00005500: 7465 640a 2d70 6167 2c20 2d2d 706c 6f74  ted.-pag, --plot
+00005510: 6170 676d 6d2c 2066 6c61 6720 6f70 7469  apgmm, flag opti
+00005520: 6f6e 2c20 7768 6574 6865 7220 746f 2070  on, whether to p
+00005530: 6572 666f 726d 2061 6e64 2070 6c6f 7420  erform and plot 
+00005540: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+00005550: 206f 6620 616e 6368 6f72 204b 7320 696e   of anchor Ks in
+00005560: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
+00005570: 7374 7269 6275 7469 6f6e 2c20 6966 2074  stribution, if t
+00005580: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00005590: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
+000055a0: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
+000055b0: 4b73 2077 6f75 6c64 2062 6520 706c 6f74  Ks would be plot
+000055c0: 7465 640a 2d70 656d 2c20 2d2d 706c 6f74  ted.-pem, --plot
+000055d0: 656c 6d6d 2c20 666c 6167 206f 7074 696f  elmm, flag optio
+000055e0: 6e2c 2077 6865 7468 6572 2074 6f20 7065  n, whether to pe
+000055f0: 7266 6f72 6d20 616e 6420 706c 6f74 2065  rform and plot e
+00005600: 6c6d 6d20 6d69 7874 7572 6520 6d6f 6465  lmm mixture mode
+00005610: 6c69 6e67 206f 6620 7061 7261 6e6f 6d65  ling of paranome
+00005620: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
+00005630: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00005640: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00005650: 7320 7365 742c 2074 6865 2065 6c6d 6d20  s set, the elmm 
+00005660: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+00005670: 206f 6620 7061 7261 6e6f 6d65 204b 7320   of paranome Ks 
+00005680: 776f 756c 6420 6265 2070 6c6f 7474 6564  would be plotted
+00005690: 0a2d 632c 202d 2d63 6f6d 706f 6e65 6e74  .-c, --component
+000056a0: 732c 2074 6865 2072 616e 6765 206f 6620  s, the range of 
+000056b0: 7468 6520 6e75 6d62 6572 206f 6620 636f  the number of co
+000056c0: 6d70 6f6e 656e 7473 2074 6f20 6669 7420  mponents to fit 
+000056d0: 696e 2061 6e63 686f 7220 4b73 206d 6978  in anchor Ks mix
+000056e0: 7475 7265 206d 6f64 656c 696e 672c 2064  ture modeling, d
+000056f0: 6566 6175 6c74 2028 312c 3429 0a2d 786c  efault (1,4).-xl
+00005700: 2c20 2d2d 786c 696d 2c20 7468 6520 7820  , --xlim, the x 
+00005710: 6178 6973 206c 696d 6974 206f 6620 4b73  axis limit of Ks
+00005720: 2064 6973 7472 6962 7574 696f 6e0a 2d79   distribution.-y
+00005730: 6c2c 202d 2d79 6c69 6d2c 2074 6865 2079  l, --ylim, the y
+00005740: 2061 7869 7320 6c69 6d69 7420 6f66 204b   axis limit of K
+00005750: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
+00005760: 6164 6f2c 202d 2d61 646a 7573 746f 7274  ado, --adjustort
+00005770: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
+00005780: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
+00005790: 7374 2074 6865 2068 6973 746f 6772 616d  st the histogram
+000057a0: 2068 6569 6768 7420 6f66 206f 7274 686f   height of ortho
+000057b0: 6c6f 676f 7573 204b 7320 6173 2074 6f20  logous Ks as to 
+000057c0: 6d61 7463 6820 7468 6520 6865 6967 6874  match the height
+000057d0: 206f 6620 7061 7261 6c6f 676f 7573 204b   of paralogous K
+000057e0: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+000057f0: 6173 2073 6574 2c20 7468 6520 6164 6a75  as set, the adju
+00005800: 7374 6d65 6e74 2077 6f75 6c64 2062 6520  stment would be 
+00005810: 636f 6e64 7563 7465 640a 2d61 6466 2c20  conducted.-adf, 
+00005820: 2d2d 6164 6a75 7374 6661 6374 6f72 2c20  --adjustfactor, 
+00005830: 7468 6520 6164 6a75 7374 6d65 6e74 2066  the adjustment f
+00005840: 6163 746f 7220 6f66 206f 7274 686f 6c6f  actor of ortholo
+00005850: 676f 7573 204b 732c 2064 6566 6175 6c74  gous Ks, default
+00005860: 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c   0.5.-oa, --okal
+00005870: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+00005880: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+00005890: 4b73 2064 6973 7472 6962 7574 696f 6e20  Ks distribution 
+000058a0: 696e 206d 6978 6564 2070 6c6f 742c 2064  in mixed plot, d
+000058b0: 6566 6175 6c74 2030 2e35 0a2d 6661 2c20  efault 0.5.-fa, 
+000058c0: 2d2d 666f 6375 7332 616c 6c2c 2073 6574  --focus2all, set
+000058d0: 2066 6f63 616c 2073 7065 6369 6573 2061   focal species a
+000058e0: 6e64 206c 6574 2073 7065 6369 6573 2070  nd let species p
+000058f0: 6169 7220 746f 2062 6520 6265 7477 6565  air to be betwee
+00005900: 6e20 666f 6361 6c20 616e 6420 616c 6c20  n focal and all 
+00005910: 7468 6520 7265 6d61 696e 696e 6720 7370  the remaining sp
+00005920: 6563 6965 732c 2064 6566 6175 6c74 204e  ecies, default N
+00005930: 6f6e 650a 2d6b 732c 202d 2d6b 7374 7265  one.-ks, --kstre
+00005940: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
+00005950: 7768 6574 6865 7220 746f 2069 6e66 6572  whether to infer
+00005960: 204b 7320 7472 6565 2c20 6966 2074 6865   Ks tree, if the
+00005970: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00005980: 6865 204b 7320 7472 6565 2069 6e66 6572  he Ks tree infer
+00005990: 656e 6365 2061 6e61 6c79 7369 7320 776f  ence analysis wo
+000059a0: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
+000059b0: 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e  .-ock, --onlycon
+000059c0: 6361 746b 7374 7265 652c 2066 6c61 6720  catkstree, flag 
+000059d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000059e0: 746f 206f 6e6c 7920 696e 6665 7220 4b73  to only infer Ks
+000059f0: 2074 7265 6520 756e 6465 7220 636f 6e63   tree under conc
+00005a00: 6174 656e 6174 6564 2061 6c69 676e 6d65  atenated alignme
+00005a10: 6e74 2c20 6966 2074 6865 2066 6c61 6720  nt, if the flag 
+00005a20: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
+00005a30: 6520 4b73 2074 7265 6520 756e 6465 7220  e Ks tree under 
+00005a40: 636f 6e63 6174 656e 6174 6564 2061 6c69  concatenated ali
+00005a50: 676e 6d65 6e74 2077 6f75 6c64 2062 6520  gnment would be 
+00005a60: 6361 6c63 756c 6174 6564 0a2d 6373 2c20  calculated.-cs, 
+00005a70: 2d2d 636c 6173 7369 632c 2066 6c61 6720  --classic, flag 
+00005a80: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00005a90: 746f 2064 7261 7720 6d69 7865 6420 4b73  to draw mixed Ks
+00005aa0: 2070 6c6f 7420 696e 2061 2063 6c61 7373   plot in a class
+00005ab0: 6963 206d 616e 6e65 7220 7768 6572 6520  ic manner where 
+00005ac0: 7468 6520 6675 6c6c 206f 7274 686f 6c6f  the full ortholo
+00005ad0: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+00005ae0: 7469 6f6e 2069 7320 6472 6177 6564 2c20  tion is drawed, 
+00005af0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00005b00: 7365 742c 2074 6865 2063 6c61 7373 6963  set, the classic
+00005b10: 206d 6978 6564 204b 7320 706c 6f74 2077   mixed Ks plot w
+00005b20: 6f75 6c64 2062 6520 6472 6177 6e0a 2d74  ould be drawn.-t
+00005b30: 612c 202d 2d74 6f70 6172 726f 772c 2066  a, --toparrow, f
+00005b40: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00005b50: 6865 7220 746f 2061 646a 7573 7420 7468  her to adjust th
+00005b60: 6520 6172 726f 7720 746f 2062 6520 6174  e arrow to be at
+00005b70: 2074 6865 2074 6f70 206f 6620 7468 6520   the top of the 
+00005b80: 706c 6f74 2c20 696e 7374 6561 6420 6f66  plot, instead of
+00005b90: 2062 6569 6e67 2063 6f6f 7264 696e 6174   being coordinat
+00005ba0: 6564 2061 7320 7468 6520 4b44 4520 6f66  ed as the KDE of
+00005bb0: 2074 6865 206f 7274 686f 6c6f 676f 7573   the orthologous
+00005bc0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00005bd0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00005be0: 7320 7365 742c 2074 6865 2061 7272 6f77  s set, the arrow
+00005bf0: 2077 6f75 6c64 2062 6520 7365 7420 6174   would be set at
+00005c00: 2074 6865 2074 6f70 0a2d 6273 2c20 2d2d   the top.-bs, --
+00005c10: 626f 6f74 7374 7261 702c 2074 6865 206e  bootstrap, the n
+00005c20: 756d 6265 7220 6f66 2062 6f6f 7473 7472  umber of bootstr
+00005c30: 6170 2072 6570 6c69 6361 7465 7320 6f66  ap replicates of
+00005c40: 206f 7274 686f 6c6f 6720 4b73 2064 6973   ortholog Ks dis
+00005c50: 7472 6962 7574 696f 6e20 696e 206d 6978  tribution in mix
+00005c60: 6564 2070 6c6f 740a 6060 600a 0a54 6865  ed plot.```..The
+00005c70: 2070 726f 6772 616d 2060 7767 6420 6d69   program `wgd mi
+00005c80: 7860 2063 616e 2072 6561 6c69 7a65 2074  x` can realize t
+00005c90: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
+00005ca0: 2063 6c75 7374 6572 696e 6720 616e 616c   clustering anal
+00005cb0: 7973 6973 206f 6620 2a4b 2a3c 7375 623e  ysis of *K*<sub>
+00005cc0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00005cd0: 7269 6275 7469 6f6e 2e0a 6060 600a 7767  ribution..```.wg
+00005ce0: 6420 6d69 7820 6b73 5f64 6174 6166 696c  d mix ks_datafil
+00005cf0: 6520 286f 7074 696f 6e29 0a2d 2d2d 2d2d  e (option).-----
+00005d00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005d10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005d20: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005d30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005d40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20  -----------.-f, 
+00005d50: 2d2d 6669 6c74 6572 732c 2074 6865 2063  --filters, the c
+00005d60: 7574 6f66 6620 616c 6967 6e6d 656e 7420  utoff alignment 
+00005d70: 6c65 6e67 7468 2c20 6465 6661 756c 7420  length, default 
+00005d80: 3330 300a 2d72 2c20 2d2d 6b73 5f72 616e  300.-r, --ks_ran
+00005d90: 6765 2c20 7468 6520 4b73 2072 616e 6765  ge, the Ks range
+00005da0: 2074 6f20 6265 2063 6f6e 7369 6465 7265   to be considere
+00005db0: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
+00005dc0: 290a 2d62 2c20 2d2d 6269 6e73 2c20 7468  ).-b, --bins, th
+00005dd0: 6520 6e75 6d62 6572 206f 6620 6269 6e73  e number of bins
+00005de0: 2069 6e20 4b73 2064 6973 7472 6962 7574   in Ks distribut
+00005df0: 696f 6e2c 2064 6566 6175 6c74 2035 300a  ion, default 50.
+00005e00: 2d6f 2c20 2d2d 6f75 7464 6972 2c20 7468  -o, --outdir, th
+00005e10: 6520 6f75 7470 7574 2064 6972 6563 746f  e output directo
+00005e20: 7279 2c20 6465 6661 756c 7420 7767 645f  ry, default wgd_
+00005e30: 6d69 780a 2d2d 6d65 7468 6f64 2c20 7768  mix.--method, wh
+00005e40: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
+00005e50: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
+00005e60: 7420 676d 6d0a 2d6e 2c20 2d2d 636f 6d70  t gmm.-n, --comp
+00005e70: 6f6e 656e 7473 2c20 7468 6520 7261 6e67  onents, the rang
+00005e80: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
+00005e90: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
+00005ea0: 2066 6974 2c20 6465 6661 756c 7420 2831   fit, default (1
+00005eb0: 2c20 3429 0a2d 672c 202d 2d67 616d 6d61  , 4).-g, --gamma
+00005ec0: 2c20 7468 6520 6761 6d6d 6120 7061 7261  , the gamma para
+00005ed0: 6d65 7465 7220 666f 7220 6267 6d6d 206d  meter for bgmm m
+00005ee0: 6f64 656c 732c 2064 6566 6175 6c74 2030  odels, default 0
+00005ef0: 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f 696e  .001.-ni, --n_in
+00005f00: 6974 2c20 7468 6520 6e75 6d62 6572 206f  it, the number o
+00005f10: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
+00005f20: 6c69 7a61 7469 6f6e 732c 2064 6566 6175  lizations, defau
+00005f30: 6c74 2032 3030 0a2d 6d69 2c20 2d2d 6d61  lt 200.-mi, --ma
+00005f40: 785f 6974 6572 2c20 7468 6520 6d61 7869  x_iter, the maxi
+00005f50: 6d75 6d20 6e75 6d62 6572 206f 6620 6974  mum number of it
+00005f60: 6572 6174 696f 6e73 2c20 6465 6661 756c  erations, defaul
+00005f70: 7420 3230 300a 6060 600a 0a54 6865 2070  t 200.```..The p
+00005f80: 726f 6772 616d 2060 7767 6420 7065 616b  rogram `wgd peak
+00005f90: 6020 6361 6e20 7265 616c 697a 6520 7468  ` can realize th
+00005fa0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
+00005fb0: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
+00005fc0: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
+00005fd0: 2069 6e20 5747 4420 6461 7469 6e67 2e0a   in WGD dating..
+00005fe0: 6060 600a 7767 6420 7065 616b 206b 735f  ```.wgd peak ks_
+00005ff0: 6461 7461 6669 6c65 2028 6f70 7469 6f6e  datafile (option
+00006000: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
+00006010: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006020: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006030: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006040: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006050: 2d2d 0a2d 6170 2c20 2d2d 616e 6368 6f72  --.-ap, --anchor
+00006060: 706f 696e 7473 2c20 7468 6520 616e 6368  points, the anch
+00006070: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
+00006080: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00006090: 0a2d 736d 2c20 2d2d 7365 676d 656e 7473  .-sm, --segments
+000060a0: 2c20 7468 6520 7365 676d 656e 7473 2064  , the segments d
+000060b0: 6174 6166 696c 652c 2064 6566 6175 6c74  atafile, default
+000060c0: 204e 6f6e 650a 2d6c 652c 202d 2d6c 6973   None.-le, --lis
+000060d0: 7465 6c65 6d65 6e74 732c 2074 6865 206c  telements, the l
+000060e0: 6973 7473 6567 6d65 6e74 7320 6461 7461  istsegments data
+000060f0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00006100: 6e65 200a 2d6d 702c 202d 2d6d 756c 7469  ne .-mp, --multi
+00006110: 706c 6963 6f6e 7061 6972 732c 2074 6865  pliconpairs, the
+00006120: 206d 756c 7469 706c 6963 6f6e 7061 6972   multipliconpair
+00006130: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
+00006140: 756c 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f  ult None.-o, --o
+00006150: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
+00006160: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
+00006170: 6175 6c74 2077 6764 5f70 6561 6b0a 2d61  ault wgd_peak.-a
+00006180: 662c 202d 2d61 6c69 676e 6669 6c74 6572  f, --alignfilter
+00006190: 2c20 6375 746f 6666 2066 6f72 2061 6c69  , cutoff for ali
+000061a0: 676e 6d65 6e74 2069 6465 6e74 6974 792c  gnment identity,
+000061b0: 206c 656e 6774 6820 616e 6420 636f 7665   length and cove
+000061c0: 7261 6765 2c20 6465 6661 756c 7420 302e  rage, default 0.
+000061d0: 302c 2030 2c20 302e 300a 2d72 2c20 2d2d  0, 0, 0.0.-r, --
+000061e0: 6b73 7261 6e67 652c 2072 616e 6765 206f  ksrange, range o
+000061f0: 6620 4b73 2074 6f20 6265 2061 6e61 6c79  f Ks to be analy
+00006200: 7a65 642c 2064 6566 6175 6c74 2028 302c  zed, default (0,
+00006210: 2035 290a 2d62 772c 202d 2d62 696e 5f77   5).-bw, --bin_w
+00006220: 6964 7468 2c20 6261 6e64 7769 6474 6820  idth, bandwidth 
+00006230: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+00006240: 6f6e 2c20 6465 6661 756c 7420 302e 310a  on, default 0.1.
+00006250: 2d69 632c 202d 2d77 6569 6768 7473 5f6f  -ic, --weights_o
+00006260: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
+00006270: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00006280: 6865 7468 6572 2074 6f20 696e 636c 7564  hether to includ
+00006290: 6520 4b73 206f 7574 6c69 6572 732c 2069  e Ks outliers, i
+000062a0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+000062b0: 6574 2c20 4b73 206f 7574 6c69 6572 7320  et, Ks outliers 
+000062c0: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
+000062d0: 6420 696e 2074 6865 2061 6e61 6c79 7369  d in the analysi
+000062e0: 730a 2d6d 2c20 2d2d 6d65 7468 6f64 2c20  s.-m, --method, 
+000062f0: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
+00006300: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
+00006310: 756c 7420 676d 6d0a 2d2d 7365 6564 2c20  ult gmm.--seed, 
+00006320: 7261 6e64 6f6d 2073 6565 6420 6769 7665  random seed give
+00006330: 6e20 746f 2069 6e69 7469 616c 697a 6174  n to initializat
+00006340: 696f 6e2c 2064 6566 6175 6c74 2032 3335  ion, default 235
+00006350: 3238 3930 0a2d 6569 2c20 2d2d 656d 5f69  2890.-ei, --em_i
+00006360: 7465 722c 2074 6865 206e 756d 6265 7220  ter, the number 
+00006370: 6f66 2045 4d20 6974 6572 6174 696f 6e73  of EM iterations
+00006380: 2074 6f20 7065 7266 6f72 6d2c 2064 6566   to perform, def
+00006390: 6175 6c74 2032 3030 0a2d 6e69 2c20 2d2d  ault 200.-ni, --
+000063a0: 6e5f 696e 6974 2c20 7468 6520 6e75 6d62  n_init, the numb
+000063b0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
+000063c0: 6974 6961 6c69 7a61 7469 6f6e 732c 2064  itializations, d
+000063d0: 6566 6175 6c74 2032 3030 0a2d 6e2c 202d  efault 200.-n, -
+000063e0: 2d63 6f6d 706f 6e65 6e74 732c 2074 6865  -components, the
+000063f0: 2072 616e 6765 206f 6620 7468 6520 6e75   range of the nu
+00006400: 6d62 6572 206f 6620 636f 6d70 6f6e 656e  mber of componen
+00006410: 7473 2074 6f20 6669 742c 2064 6566 6175  ts to fit, defau
+00006420: 6c74 2028 312c 2034 290a 2d67 2c20 2d2d  lt (1, 4).-g, --
+00006430: 6761 6d6d 612c 2074 6865 2067 616d 6d61  gamma, the gamma
+00006440: 2070 6172 616d 6574 6572 2066 6f72 2062   parameter for b
+00006450: 676d 6d20 6d6f 6465 6c73 2c20 6465 6661  gmm models, defa
+00006460: 756c 7420 3165 2d33 0a2d 2d62 6f6f 7473  ult 1e-3.--boots
+00006470: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00006480: 626f 6f74 7374 7261 7020 7265 706c 6963  bootstrap replic
+00006490: 6174 6573 206f 6620 6b64 652c 2064 6566  ates of kde, def
+000064a0: 6175 6c74 2032 3030 0a2d 2d77 6569 6768  ault 200.--weigh
+000064b0: 7465 642c 2066 6c61 6720 6f70 7469 6f6e  ted, flag option
+000064c0: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
+000064d0: 206e 6f64 652d 7765 6967 6874 6564 206d   node-weighted m
+000064e0: 6574 686f 6420 6f66 2064 652d 7265 6475  ethod of de-redu
+000064f0: 6e64 616e 6379 2c20 6966 2074 6865 2066  ndancy, if the f
+00006500: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00006510: 206e 6f64 652d 7765 6967 6874 6564 206d   node-weighted m
+00006520: 6574 686f 6420 776f 756c 6420 6265 2075  ethod would be u
+00006530: 7365 640a 2d70 2c20 2d2d 706c 6f74 2c20  sed.-p, --plot, 
+00006540: 7468 6520 706c 6f74 7469 6e67 206d 6574  the plotting met
+00006550: 686f 6420 746f 2062 6520 7573 6564 2c20  hod to be used, 
+00006560: 6465 6661 756c 7420 6964 656e 7469 6361  default identica
+00006570: 6c0a 2d62 6d2c 202d 2d62 775f 6d65 7468  l.-bm, --bw_meth
+00006580: 6f64 2c20 7468 6520 6261 6e64 7769 6474  od, the bandwidt
+00006590: 6820 6d65 7468 6f64 2074 6f20 6265 2075  h method to be u
+000065a0: 7365 6420 696e 2061 6e61 6c79 7a69 6e67  sed in analyzing
+000065b0: 2074 6865 2070 6561 6b20 6f66 2057 4744   the peak of WGD
+000065c0: 2064 6174 6573 2c20 6465 6661 756c 7420   dates, default 
+000065d0: 7369 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65  silverman.--n_me
+000065e0: 646f 6964 732c 2074 6865 206e 756d 6265  doids, the numbe
+000065f0: 7220 6f66 206d 6564 6f69 6473 2074 6f20  r of medoids to 
+00006600: 6669 742c 2064 6566 6175 6c74 2032 0a2d  fit, default 2.-
+00006610: 6b6d 2c20 2d2d 6b64 656d 6574 686f 642c  km, --kdemethod,
+00006620: 2074 6865 206b 6465 206d 6574 686f 6420   the kde method 
+00006630: 746f 2062 6520 7573 6564 2069 6e20 616e  to be used in an
+00006640: 616c 797a 696e 6720 7468 6520 7065 616b  alyzing the peak
+00006650: 206f 6620 5747 4420 6461 7465 732c 206b   of WGD dates, k
+00006660: 6d65 646f 6964 7320 616e 616c 7973 6973  medoids analysis
+00006670: 206f 7220 7468 6520 6261 7369 6320 4b73   or the basic Ks
+00006680: 2070 6c6f 7474 696e 672c 2064 6566 6175   plotting, defau
+00006690: 6c74 2073 6369 7079 0a2d 2d6e 5f63 6c75  lt scipy.--n_clu
+000066a0: 7374 6572 732c 2074 6865 206e 756d 6265  sters, the numbe
+000066b0: 7220 6f66 2063 6c75 7374 6572 7320 746f  r of clusters to
+000066c0: 2070 6c6f 7420 456c 626f 7720 6c6f 7373   plot Elbow loss
+000066d0: 2066 756e 6374 696f 6e2c 2064 6566 6175   function, defau
+000066e0: 6c74 2035 0a2d 6764 2c20 2d2d 6775 6964  lt 5.-gd, --guid
+000066f0: 652c 2074 6865 2072 6567 696d 6520 7265  e, the regime re
+00006700: 7369 6469 6e67 2061 6e63 686f 7273 2c20  siding anchors, 
+00006710: 6465 6661 756c 7420 5365 676d 656e 740a  default Segment.
+00006720: 2d70 7263 742c 202d 2d70 726f 6d69 6e65  -prct, --promine
+00006730: 6e63 655f 6375 746f 6666 2c20 7468 6520  nce_cutoff, the 
+00006740: 7072 6f6d 696e 656e 6365 2063 7574 6f66  prominence cutof
+00006750: 6620 6f66 2061 6363 6570 7461 626c 6520  f of acceptable 
+00006760: 7065 616b 7320 696e 2070 6561 6b20 6669  peaks in peak fi
+00006770: 6e64 696e 6720 7374 6570 732c 2064 6566  nding steps, def
+00006780: 6175 6c74 2030 2e31 0a2d 7268 2c20 2d2d  ault 0.1.-rh, --
+00006790: 7265 6c5f 6865 6967 6874 2c20 7468 6520  rel_height, the 
+000067a0: 7265 6c61 7469 7665 2068 6569 6768 7420  relative height 
+000067b0: 6174 2077 6869 6368 2074 6865 2070 6561  at which the pea
+000067c0: 6b20 7769 6474 6820 6973 206d 6561 7375  k width is measu
+000067d0: 7265 642c 2064 6566 6175 6c74 2030 2e34  red, default 0.4
+000067e0: 0a2d 6b64 2c20 2d2d 6b73 746f 6461 7465  .-kd, --kstodate
+000067f0: 2c20 7468 6520 7261 6e67 6520 6f66 204b  , the range of K
+00006800: 7320 746f 2062 6520 6461 7465 6420 696e  s to be dated in
+00006810: 2068 6575 7269 7374 6963 2073 6561 7263   heuristic searc
+00006820: 682c 2064 6566 6175 6c74 2028 302e 352c  h, default (0.5,
+00006830: 2031 2e35 290a 2d78 6c2c 202d 2d78 6c69   1.5).-xl, --xli
+00006840: 6d2c 2074 6865 2078 2061 7869 7320 6c69  m, the x axis li
+00006850: 6d69 7420 6f66 2047 4d4d 204b 7320 6469  mit of GMM Ks di
+00006860: 7374 7269 6275 7469 6f6e 0a2d 796c 2c20  stribution.-yl, 
+00006870: 2d2d 796c 696d 2c20 7468 6520 7920 6178  --ylim, the y ax
+00006880: 6973 206c 696d 6974 206f 6620 474d 4d20  is limit of GMM 
+00006890: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
+000068a0: 2d2d 6d61 6e75 616c 7365 742c 2066 6c61  --manualset, fla
+000068b0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000068c0: 7220 746f 206f 7574 7075 7420 616e 6368  r to output anch
+000068d0: 6f72 2070 6169 7273 2077 6974 6820 6d61  or pairs with ma
+000068e0: 6e75 616c 6c79 2073 6574 204b 7320 7261  nually set Ks ra
+000068f0: 6e67 652c 2069 6620 7468 6520 666c 6167  nge, if the flag
+00006900: 2077 6173 2073 6574 2c20 6d61 6e75 616c   was set, manual
+00006910: 6c79 2073 6574 204b 7320 7261 6e67 6520  ly set Ks range 
+00006920: 776f 756c 6420 6265 206f 7574 7075 7474  would be outputt
+00006930: 6564 0a2d 2d63 692c 2074 6865 2063 6f6e  ed.--ci, the con
+00006940: 6669 6465 6e63 6520 6c65 7665 6c20 6f66  fidence level of
+00006950: 206c 6f67 2d6e 6f72 6d61 6c20 6469 7374   log-normal dist
+00006960: 7269 6275 7469 6f6e 2074 6f20 6461 7465  ribution to date
+00006970: 2c20 6465 6661 756c 7420 3935 0a2d 2d68  , default 95.--h
+00006980: 6472 2c20 7468 6520 6869 6768 6573 7420  dr, the highest 
+00006990: 6465 6e73 6974 7920 7265 6769 6f6e 2028  density region (
+000069a0: 4844 5229 2061 7070 6c69 6564 2069 6e20  HDR) applied in 
+000069b0: 7468 6520 7365 676d 656e 742d 6775 6964  the segment-guid
+000069c0: 6564 2061 6e63 686f 7220 7061 6972 204b  ed anchor pair K
+000069d0: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+000069e0: 6465 6661 756c 7420 3935 0a2d 2d68 6575  default 95.--heu
+000069f0: 7269 7374 6963 2c20 666c 6167 206f 7074  ristic, flag opt
+00006a00: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00006a10: 696e 6974 6961 7465 2068 6575 7269 7374  initiate heurist
+00006a20: 6963 206d 6574 686f 6420 6f66 2064 6566  ic method of def
+00006a30: 696e 696e 6720 4349 2066 6f72 2064 6174  ining CI for dat
+00006a40: 696e 672c 2069 6620 7468 6520 666c 6167  ing, if the flag
+00006a50: 2077 6173 2073 6574 2c20 7468 6520 6865   was set, the he
+00006a60: 7572 6973 7469 6320 6d65 7468 6f64 2077  uristic method w
+00006a70: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00006a80: 640a 2d6b 632c 202d 2d6b 7363 7574 6f66  d.-kc, --kscutof
+00006a90: 662c 2074 6865 204b 7320 7361 7475 7261  f, the Ks satura
+00006aa0: 7469 6f6e 2063 7574 6f66 6620 696e 2064  tion cutoff in d
+00006ab0: 6174 696e 672c 2064 6566 6175 6c74 2035  ating, default 5
+00006ac0: 0a2d 2d6b 6565 7074 6d70 6669 672c 2066  .--keeptmpfig, f
+00006ad0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00006ae0: 6865 7220 746f 206b 6565 7020 7465 6d70  her to keep temp
+00006af0: 6f72 6172 7920 6669 6775 7265 7320 696e  orary figures in
+00006b00: 2070 6561 6b20 6669 6e64 696e 6720 7072   peak finding pr
+00006b10: 6f63 6573 732c 2069 6620 7468 6520 666c  ocess, if the fl
+00006b20: 6167 2077 6173 2073 6574 2c20 7468 6f73  ag was set, thos
+00006b30: 6520 6669 6775 7265 7320 776f 756c 6420  e figures would 
+00006b40: 6265 206b 6570 740a 6060 600a 0a54 6865  be kept.```..The
+00006b50: 2070 726f 6772 616d 2060 7767 6420 7379   program `wgd sy
+00006b60: 6e60 2063 616e 2072 6561 6c69 7a65 2074  n` can realize t
+00006b70: 6865 2069 6e74 7261 2d20 616e 6420 696e  he intra- and in
+00006b80: 7465 722d 7370 6563 6966 6963 2073 796e  ter-specific syn
+00006b90: 7465 6e79 2069 6e66 6572 656e 6365 2e0a  teny inference..
+00006ba0: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+00006bb0: 6c69 6573 2067 6666 7320 286f 7074 696f  lies gffs (optio
+00006bc0: 6e29 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  n).-------------
+00006bd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006be0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006bf0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006c00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006c10: 2d2d 2d0a 2d6b 732c 202d 2d6b 735f 6469  ---.-ks, --ks_di
+00006c20: 7374 7269 6275 7469 6f6e 2c20 6b73 2064  stribution, ks d
+00006c30: 6973 7472 6962 7574 696f 6e20 6461 7461  istribution data
+00006c40: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00006c50: 6e65 0a2d 6f2c 202d 2d6f 7574 6469 722c  ne.-o, --outdir,
+00006c60: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
+00006c70: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
+00006c80: 6764 5f73 796e 0a2d 662c 202d 2d66 6561  gd_syn.-f, --fea
+00006c90: 7475 7265 2c20 7468 6520 6665 6174 7572  ture, the featur
+00006ca0: 6520 666f 7220 7061 7273 696e 6720 6765  e for parsing ge
+00006cb0: 6e65 2049 4473 2066 726f 6d20 4746 4620  ne IDs from GFF 
+00006cc0: 6669 6c65 732c 2064 6566 6175 6c74 2067  files, default g
+00006cd0: 656e 650a 2d61 2c20 2d2d 6174 7472 6962  ene.-a, --attrib
+00006ce0: 7574 652c 2074 6865 2061 7474 7269 6275  ute, the attribu
+00006cf0: 7465 2066 6f72 2070 6172 7369 6e67 2074  te for parsing t
+00006d00: 6865 2067 656e 6520 4944 7320 6672 6f6d  he gene IDs from
+00006d10: 2074 6865 2047 4646 2066 696c 6573 2c20   the GFF files, 
+00006d20: 6465 6661 756c 7420 4944 0a2d 6174 672c  default ID.-atg,
+00006d30: 202d 2d61 6464 6974 696f 6e61 6c67 6666   --additionalgff
+00006d40: 696e 666f 2c20 7468 6520 6665 6174 7572  info, the featur
+00006d50: 6520 616e 6420 6174 7472 6962 7574 6520  e and attribute 
+00006d60: 696e 666f 726d 6174 696f 6e20 6f66 2061  information of a
+00006d70: 6464 6974 696f 6e61 6c20 6766 6633 2066  dditional gff3 f
+00006d80: 696c 6573 2069 6620 6469 6666 6572 656e  iles if differen
+00006d90: 7420 696e 2074 6865 2066 6f72 6d61 7420  t in the format 
+00006da0: 6f66 2028 6665 6174 7572 653b 6174 7472  of (feature;attr
+00006db0: 6962 7574 6529 272c 2064 6566 6175 6c74  ibute)', default
+00006dc0: 204e 6f6e 650a 2d6d 6c2c 202d 2d6d 696e   None.-ml, --min
+00006dd0: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
+00006de0: 206c 656e 6774 6820 6f66 2061 2073 6361   length of a sca
+00006df0: 6666 6f6c 6420 746f 2062 6520 696e 636c  ffold to be incl
+00006e00: 7564 6564 2069 6e20 646f 7470 6c6f 742c  uded in dotplot,
+00006e10: 2064 6566 6175 6c74 202d 312c 2069 6620   default -1, if 
+00006e20: 2d31 2077 6173 2073 6574 2c20 7468 6520  -1 was set, the 
+00006e30: 3130 2520 6f66 2074 6865 206c 6f6e 6765  10% of the longe
+00006e40: 7374 2073 6361 6666 6f6c 6420 776f 756c  st scaffold woul
+00006e50: 6420 6265 2073 6574 0a2d 6d73 2c20 2d2d  d be set.-ms, --
+00006e60: 6d61 7873 697a 652c 2074 6865 206d 6178  maxsize, the max
+00006e70: 696d 756d 2066 616d 696c 7920 7369 7a65  imum family size
+00006e80: 2074 6f20 6265 2069 6e63 6c75 6465 642c   to be included,
+00006e90: 2064 6566 6175 6c74 2032 3030 0a2d 722c   default 200.-r,
+00006ea0: 202d 2d6b 735f 7261 6e67 652c 2074 6865   --ks_range, the
+00006eb0: 204b 7320 7261 6e67 6520 696e 2063 6f6c   Ks range in col
+00006ec0: 6f72 6564 2064 6f74 706c 6f74 2c20 6465  ored dotplot, de
+00006ed0: 6661 756c 7420 2830 2c20 3529 0a2d 2d69  fault (0, 5).--i
+00006ee0: 6164 686f 7265 5f6f 7074 696f 6e73 2c20  adhore_options, 
+00006ef0: 7468 6520 7061 7261 6d65 7465 7220 7365  the parameter se
+00006f00: 7474 696e 6720 696e 2069 6164 686f 7265  tting in iadhore
+00006f10: 2c20 6465 6661 756c 7420 6173 2061 2073  , default as a s
+00006f20: 7472 696e 6720 6f66 206c 656e 6774 6820  tring of length 
+00006f30: 7a65 726f 0a2d 6d67 2c20 2d2d 6d69 6e73  zero.-mg, --mins
+00006f40: 6567 6c65 6e2c 2074 6865 206d 696e 696d  eglen, the minim
+00006f50: 756d 206c 656e 6774 6820 6f66 2073 6567  um length of seg
+00006f60: 6d65 6e74 7320 746f 2069 6e63 6c75 6465  ments to include
+00006f70: 2069 6e20 7261 7469 6f20 6966 203c 3d20   in ratio if <= 
+00006f80: 312c 2064 6566 6175 6c74 2031 3030 3030  1, default 10000
+00006f90: 0a2d 6b72 2c20 2d2d 6b65 6570 7265 6475  .-kr, --keepredu
+00006fa0: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+00006fb0: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
+00006fc0: 7265 6475 6e64 616e 7420 6d75 6c74 6970  redundant multip
+00006fd0: 6c69 636f 6e73 2c20 6966 2074 6865 2066  licons, if the f
+00006fe0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00006ff0: 2072 6564 756e 6461 6e74 206d 756c 7469   redundant multi
+00007000: 706c 6963 6f6e 7320 776f 756c 6420 6265  plicons would be
+00007010: 206b 6570 740a 2d6d 676e 2c20 2d2d 6d69   kept.-mgn, --mi
+00007020: 6e67 656e 656e 756d 2c20 7468 6520 6d69  ngenenum, the mi
+00007030: 6e69 6d75 6d20 6e75 6d62 6572 206f 6620  nimum number of 
+00007040: 6765 6e65 7320 666f 7220 6120 7365 676d  genes for a segm
+00007050: 656e 7420 746f 2062 6520 636f 6e73 6964  ent to be consid
+00007060: 6572 6564 2c20 6465 6661 756c 7420 3330  ered, default 30
+00007070: 0a2d 6473 2c20 2d2d 646f 7473 697a 652c  .-ds, --dotsize,
+00007080: 2074 6865 2064 6f74 2073 697a 6520 696e   the dot size in
+00007090: 2064 6f74 2070 6c6f 742c 2064 6566 6175   dot plot, defau
+000070a0: 6c74 2030 2e33 0a2d 6161 2c20 2d2d 6170  lt 0.3.-aa, --ap
+000070b0: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
+000070c0: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
+000070d0: 732c 2064 6566 6175 6c74 2031 0a2d 6861  s, default 1.-ha
+000070e0: 2c20 2d2d 686f 616c 7068 612c 2074 6865  , --hoalpha, the
+000070f0: 206f 7061 6369 7479 206f 6620 686f 6d6f   opacity of homo
+00007100: 6c6f 6720 646f 7473 2c20 6465 6661 756c  log dots, defaul
+00007110: 7420 300a 2d73 7274 2c20 2d2d 7368 6f77  t 0.-srt, --show
+00007120: 7265 616c 7469 636b 2c20 666c 6167 206f  realtick, flag o
+00007130: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00007140: 6f20 7368 6f77 2074 6865 2072 6561 6c20  o show the real 
+00007150: 7469 636b 2069 6e20 6765 6e65 7320 6f72  tick in genes or
+00007160: 2062 6173 6573 2c20 6966 2074 6865 2066   bases, if the f
+00007170: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00007180: 2072 6561 6c20 7469 636b 2077 6f75 6c64   real tick would
+00007190: 2062 6520 7368 6f77 6564 0a2d 746c 732c   be showed.-tls,
+000071a0: 202d 2d74 6963 6b6c 6162 656c 7369 7a65   --ticklabelsize
+000071b0: 2c20 7468 6520 6c61 6265 6c20 7369 7a65  , the label size
+000071c0: 206f 6620 7469 636b 2c20 6465 6661 756c   of tick, defaul
+000071d0: 7420 350a 2d67 722c 202d 2d67 6973 7472  t 5.-gr, --gistr
+000071e0: 622c 2066 6c61 6720 6f70 7469 6f6e 2c20  b, flag option, 
+000071f0: 7768 6574 6865 7220 746f 2075 7365 2067  whether to use g
+00007200: 6973 745f 7261 696e 626f 7720 6173 2063  ist_rainbow as c
+00007210: 6f6c 6f72 206d 6170 206f 6620 646f 7470  olor map of dotp
+00007220: 6c6f 740a 2d6e 2c20 2d2d 6e74 6872 6561  lot.-n, --nthrea
+00007230: 6473 2c20 7468 6520 6e75 6d62 6572 206f  ds, the number o
+00007240: 6620 7468 7265 6164 7320 746f 2075 7365  f threads to use
+00007250: 2069 6e20 7379 6e74 656e 7920 696e 6665   in synteny infe
+00007260: 7265 6e63 652c 2064 6566 6175 6c74 2034  rence, default 4
+00007270: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+00007280: 6d20 6077 6764 2076 697a 6020 6361 6e20  m `wgd viz` can 
+00007290: 7265 616c 697a 6520 7468 6520 7669 7375  realize the visu
+000072a0: 616c 697a 6174 696f 6e20 6f66 202a 4b2a  alization of *K*
+000072b0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+000072c0: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+000072d0: 6420 7379 6e74 656e 792e 0a60 6060 0a77  d synteny..```.w
+000072e0: 6764 2076 697a 2028 6f70 7469 6f6e 290a  gd viz (option).
+000072f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007300: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007310: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007320: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007330: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007340: 0a2d 642c 202d 2d64 6174 6166 696c 652c  .-d, --datafile,
+00007350: 2074 6865 204b 7320 6461 7461 6669 6c65   the Ks datafile
+00007360: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00007370: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
+00007380: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
+00007390: 792c 2064 6566 6175 6c74 2077 6764 5f76  y, default wgd_v
+000073a0: 697a 0a2d 7372 2c20 2d2d 7370 6169 722c  iz.-sr, --spair,
+000073b0: 2074 6865 2073 7065 6369 6573 2070 6169   the species pai
+000073c0: 7220 746f 2062 6520 706c 6f74 7465 642c  r to be plotted,
+000073d0: 2064 6566 6175 6c74 204e 6f6e 652c 2074   default None, t
+000073e0: 6869 7320 6f70 7469 6f6e 2063 616e 2062  his option can b
+000073f0: 6520 7072 6f76 6964 6564 206d 756c 7469  e provided multi
+00007400: 706c 6520 7469 6d65 730a 2d66 612c 202d  ple times.-fa, -
+00007410: 2d66 6f63 7573 3261 6c6c 2c20 7365 7420  -focus2all, set 
+00007420: 666f 6361 6c20 7370 6563 6965 7320 616e  focal species an
+00007430: 6420 6c65 7420 7370 6563 6965 7320 7061  d let species pa
+00007440: 6972 2074 6f20 6265 2062 6574 7765 656e  ir to be between
+00007450: 2066 6f63 616c 2061 6e64 2061 6c6c 2074   focal and all t
+00007460: 6865 2072 656d 6169 6e69 6e67 2073 7065  he remaining spe
+00007470: 6369 6573 2c20 6465 6661 756c 7420 4e6f  cies, default No
+00007480: 6e65 0a2d 6773 2c20 2d2d 6773 6d61 702c  ne.-gs, --gsmap,
+00007490: 2074 6865 2067 656e 6520 6e61 6d65 2d73   the gene name-s
+000074a0: 7065 6369 6573 206e 616d 6520 6d61 702c  pecies name map,
+000074b0: 2064 6566 6175 6c74 204e 6f6e 650a 2d73   default None.-s
+000074c0: 702c 202d 2d73 7065 6369 6573 7472 6565  p, --speciestree
+000074d0: 2c20 7468 6520 7370 6563 6965 7320 7472  , the species tr
+000074e0: 6565 2074 6f20 7065 7266 6f72 6d20 7261  ee to perform ra
+000074f0: 7465 2063 6f72 7265 6374 696f 6e2c 2064  te correction, d
+00007500: 6566 6175 6c74 204e 6f6e 652c 2069 6620  efault None, if 
+00007510: 4e6f 6e65 2077 6173 2067 6976 656e 2c20  None was given, 
+00007520: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+00007530: 696f 6e20 616e 616c 7973 6973 2077 6f75  ion analysis wou
+00007540: 6c64 2062 6520 6361 6c6c 6564 206f 6666  ld be called off
+00007550: 0a2d 706b 2c20 2d2d 706c 6f74 6b64 652c  .-pk, --plotkde,
+00007560: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00007570: 6574 6865 7220 746f 2070 6c6f 7420 6b64  ether to plot kd
+00007580: 6520 6375 7276 6520 7570 6f6e 2068 6973  e curve upon his
+00007590: 746f 6772 616d 2c20 6966 2074 6865 2066  togram, if the f
+000075a0: 6c61 6720 7761 7320 7365 742c 206b 6465  lag was set, kde
+000075b0: 2063 7572 7665 2077 6f75 6c64 2062 6520   curve would be 
+000075c0: 6164 6465 640a 2d72 772c 202d 2d72 6577  added.-rw, --rew
+000075d0: 6569 6768 742c 2066 6c61 6720 6f70 7469  eight, flag opti
+000075e0: 6f6e 2c20 7768 6574 6865 7220 746f 2072  on, whether to r
+000075f0: 6563 616c 6375 6c61 7465 2074 6865 2077  ecalculate the w
+00007600: 6569 6768 7420 7065 7220 7370 6563 6965  eight per specie
+00007610: 7320 7061 6972 2c20 6966 2074 6865 2066  s pair, if the f
+00007620: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00007630: 2077 6569 6768 7420 776f 756c 6420 6265   weight would be
+00007640: 2072 6563 616c 6375 6c61 7465 640a 2d6f   recalculated.-o
+00007650: 722c 202d 2d6f 6e6c 7972 6f6f 746f 7574  r, --onlyrootout
+00007660: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00007670: 6865 7468 6572 2074 6f20 6f6e 6c79 2063  hether to only c
+00007680: 6f6e 6475 6374 2072 6174 6520 636f 7272  onduct rate corr
+00007690: 6563 7469 6f6e 2075 7369 6e67 2074 6865  ection using the
+000076a0: 206f 7574 6772 6f75 7020 6174 2072 6f6f   outgroup at roo
+000076b0: 7420 6173 206f 7574 6772 6f75 702c 2069  t as outgroup, i
+000076c0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+000076d0: 6574 2c20 6f6e 6c79 2074 6865 206f 7574  et, only the out
+000076e0: 6772 6f75 7020 6174 2072 6f6f 7420 776f  group at root wo
+000076f0: 756c 6420 6265 2075 7365 6420 6173 206f  uld be used as o
+00007700: 7574 6772 6f75 700a 2d69 7465 722c 202d  utgroup.-iter, -
+00007710: 2d65 6d5f 6974 6572 6174 696f 6e73 2c20  -em_iterations, 
+00007720: 7468 6520 6d61 7869 6d75 6d20 454d 2069  the maximum EM i
+00007730: 7465 7261 7469 6f6e 732c 2064 6566 6175  terations, defau
+00007740: 6c74 2032 3030 0a2d 696e 6974 2c20 2d2d  lt 200.-init, --
+00007750: 656d 5f69 6e69 7469 616c 697a 6174 696f  em_initializatio
+00007760: 6e73 2c20 7468 6520 6d61 7869 6d75 6d20  ns, the maximum 
+00007770: 454d 2069 6e69 7469 616c 697a 6174 696f  EM initializatio
+00007780: 6e73 2c20 6465 6661 756c 7420 3230 300a  ns, default 200.
+00007790: 2d70 7263 742c 202d 2d70 726f 6d69 6e65  -prct, --promine
+000077a0: 6e63 655f 6375 746f 6666 2c20 7468 6520  nce_cutoff, the 
+000077b0: 7072 6f6d 696e 656e 6365 2063 7574 6f66  prominence cutof
+000077c0: 6620 6f66 2061 6363 6570 7461 626c 6520  f of acceptable 
+000077d0: 7065 616b 732c 2064 6566 6175 6c74 2030  peaks, default 0
+000077e0: 2e31 0a2d 7268 2c20 2d2d 7265 6c5f 6865  .1.-rh, --rel_he
+000077f0: 6967 6874 2c20 7468 6520 7265 6c61 7469  ight, the relati
+00007800: 7665 2068 6569 6768 7420 6174 2077 6869  ve height at whi
+00007810: 6368 2074 6865 2070 6561 6b20 7769 6474  ch the peak widt
+00007820: 6820 6973 206d 6561 7375 7265 642c 2064  h is measured, d
+00007830: 6566 6175 6c74 2030 2e34 0a2d 736d 2c20  efault 0.4.-sm, 
+00007840: 2d2d 7365 676d 656e 7473 2c20 7468 6520  --segments, the 
+00007850: 7365 676d 656e 7473 2064 6174 6166 696c  segments datafil
+00007860: 652c 2064 6566 6175 6c74 204e 6f6e 650a  e, default None.
+00007870: 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c 2074  -ml, --minlen, t
+00007880: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
+00007890: 6820 6f66 2061 2073 6361 6666 6f6c 6420  h of a scaffold 
+000078a0: 746f 2062 6520 696e 636c 7564 6564 2069  to be included i
+000078b0: 6e20 646f 7470 6c6f 742c 2064 6566 6175  n dotplot, defau
+000078c0: 6c74 202d 312c 2069 6620 2d31 2077 6173  lt -1, if -1 was
+000078d0: 2073 6574 2c20 7468 6520 3130 2520 6f66   set, the 10% of
+000078e0: 2074 6865 206c 6f6e 6765 7374 2073 6361   the longest sca
+000078f0: 6666 6f6c 6473 2077 696c 6c20 6265 2073  ffolds will be s
+00007900: 6574 0a2d 6d73 2c20 2d2d 6d61 7873 697a  et.-ms, --maxsiz
+00007910: 652c 2074 6865 206d 6178 696d 756d 2066  e, the maximum f
+00007920: 616d 696c 7920 7369 7a65 2074 6f20 6265  amily size to be
+00007930: 2069 6e63 6c75 6465 642c 2064 6566 6175   included, defau
+00007940: 6c74 2032 3030 0a2d 6170 2c20 2d2d 616e  lt 200.-ap, --an
+00007950: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
+00007960: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
+00007970: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
+00007980: 4e6f 6e65 0a2d 6d74 2c20 2d2d 6d75 6c74  None.-mt, --mult
+00007990: 6970 6c69 636f 6e2c 2074 6865 206d 756c  iplicon, the mul
+000079a0: 7469 706c 6963 6f6e 7320 6461 7461 6669  tiplicons datafi
+000079b0: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+000079c0: 0a2d 6774 2c20 2d2d 6765 6e65 7461 626c  .-gt, --genetabl
+000079d0: 652c 2074 6865 2067 656e 6520 7461 626c  e, the gene tabl
+000079e0: 6520 6461 7461 6669 6c65 2c20 6465 6661  e datafile, defa
+000079f0: 756c 7420 4e6f 6e65 0a2d 6d67 2c20 2d2d  ult None.-mg, --
+00007a00: 6d69 6e73 6567 6c65 6e2c 2074 6865 206d  minseglen, the m
+00007a10: 696e 696d 756d 206c 656e 6774 6820 6f66  inimum length of
+00007a20: 2073 6567 6d65 6e74 7320 746f 2069 6e63   segments to inc
+00007a30: 6c75 6465 2c20 696e 2072 6174 696f 2069  lude, in ratio i
+00007a40: 6620 3c3d 2031 2c20 6465 6661 756c 7420  f <= 1, default 
+00007a50: 3130 3030 300a 2d6d 676e 2c20 2d2d 6d69  10000.-mgn, --mi
+00007a60: 6e67 656e 656e 756d 2c20 7468 6520 6d69  ngenenum, the mi
+00007a70: 6e69 6d75 6d20 6e75 6d62 6572 206f 6620  nimum number of 
+00007a80: 6765 6e65 7320 666f 7220 6120 7365 676d  genes for a segm
+00007a90: 656e 7420 746f 2062 6520 636f 6e73 6964  ent to be consid
+00007aa0: 6572 6564 2c20 6465 6661 756c 7420 3330  ered, default 30
+00007ab0: 0a2d 6b72 2c20 2d2d 6b65 6570 7265 6475  .-kr, --keepredu
+00007ac0: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+00007ad0: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
+00007ae0: 7265 6475 6e64 616e 7420 6d75 6c74 6970  redundant multip
+00007af0: 6c69 636f 6e73 2c20 6966 2074 6865 2066  licons, if the f
+00007b00: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00007b10: 2072 6564 756e 6461 6e74 206d 756c 7469   redundant multi
+00007b20: 706c 6963 6f6e 7320 776f 756c 6420 6265  plicons would be
+00007b30: 206b 6570 740a 2d65 706b 2c20 2d2d 6578   kept.-epk, --ex
+00007b40: 7472 6170 6172 616e 6f6d 656b 732c 2065  traparanomeks, e
+00007b50: 7874 7261 2070 6172 616e 6f6d 6520 4b73  xtra paranome Ks
+00007b60: 2064 6174 6120 746f 2070 6c6f 7420 696e   data to plot in
+00007b70: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
+00007b80: 7374 7269 6275 7469 6f6e 2c20 6465 6661  stribution, defa
+00007b90: 756c 7420 4e6f 6e65 0a2d 7061 672c 202d  ult None.-pag, -
+00007ba0: 2d70 6c6f 7461 7067 6d6d 2c20 666c 6167  -plotapgmm, flag
+00007bb0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+00007bc0: 2074 6f20 636f 6e64 7563 7420 616e 6420   to conduct and 
+00007bd0: 706c 6f74 206d 6978 7475 7265 206d 6f64  plot mixture mod
+00007be0: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
+00007bf0: 4b73 2069 6e20 7468 6520 6d69 7865 6420  Ks in the mixed 
+00007c00: 4b73 2064 6973 7472 6962 7574 696f 6e2c  Ks distribution,
+00007c10: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00007c20: 2073 6574 2c20 7468 6520 6d69 7874 7572   set, the mixtur
+00007c30: 6520 6d6f 6465 6c69 6e67 206f 6620 616e  e modeling of an
+00007c40: 6368 6f72 204b 7320 776f 756c 6420 6265  chor Ks would be
+00007c50: 2063 6f6e 6475 6374 6564 2061 6e64 2070   conducted and p
+00007c60: 6c6f 7474 6564 0a2d 7065 6d2c 202d 2d70  lotted.-pem, --p
+00007c70: 6c6f 7465 6c6d 6d2c 2066 6c61 6720 6f70  lotelmm, flag op
+00007c80: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007c90: 2063 6f6e 6475 6374 2061 6e64 2070 6c6f   conduct and plo
+00007ca0: 7420 656c 6d6d 206d 6978 7475 7265 206d  t elmm mixture m
+00007cb0: 6f64 656c 696e 6720 6f66 2070 6172 616e  odeling of paran
+00007cc0: 6f6d 6520 4b73 2069 6e20 7468 6520 6d69  ome Ks in the mi
+00007cd0: 7865 6420 4b73 2064 6973 7472 6962 7574  xed Ks distribut
+00007ce0: 696f 6e2c 2069 6620 7468 6520 666c 6167  ion, if the flag
+00007cf0: 2077 6173 2073 6574 2c20 7468 6520 656c   was set, the el
+00007d00: 6d6d 206d 6978 7475 7265 206d 6f64 656c  mm mixture model
+00007d10: 696e 6720 6f66 2070 6172 616e 6f6d 6520  ing of paranome 
+00007d20: 4b73 2077 6f75 6c64 2062 6520 636f 6e64  Ks would be cond
+00007d30: 7563 7465 6420 616e 6420 706c 6f74 7465  ucted and plotte
+00007d40: 640a 2d63 2c20 2d2d 636f 6d70 6f6e 656e  d.-c, --componen
+00007d50: 7473 2c20 7468 6520 7261 6e67 6520 6f66  ts, the range of
+00007d60: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
+00007d70: 6f6d 706f 6e65 6e74 7320 746f 2066 6974  omponents to fit
+00007d80: 2069 6e20 616e 6368 6f72 204b 7320 6d69   in anchor Ks mi
+00007d90: 7874 7572 6520 6d6f 6465 6c69 6e67 2c20  xture modeling, 
+00007da0: 6465 6661 756c 7420 2831 2c34 290a 2d70  default (1,4).-p
+00007db0: 7379 2c20 2d2d 706c 6f74 7379 6e2c 2066  sy, --plotsyn, f
+00007dc0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00007dd0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00007de0: 7468 6520 7379 6e74 656e 7920 706c 6f74  the synteny plot
+00007df0: 2c20 6f6e 6c79 2077 6865 6e20 7468 6520  , only when the 
+00007e00: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00007e10: 6520 7379 6e74 656e 7920 706c 6f74 2077  e synteny plot w
+00007e20: 6f75 6c64 2062 6520 7072 6f64 7563 6564  ould be produced
+00007e30: 0a2d 6473 2c20 2d2d 646f 7473 697a 652c  .-ds, --dotsize,
+00007e40: 2074 6865 2064 6f74 2073 697a 6520 696e   the dot size in
+00007e50: 2064 6f74 2070 6c6f 742c 2064 6566 6175   dot plot, defau
+00007e60: 6c74 2030 2e33 0a2d 6161 2c20 2d2d 6170  lt 0.3.-aa, --ap
+00007e70: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
+00007e80: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
+00007e90: 732c 2064 6566 6175 6c74 2031 0a2d 6861  s, default 1.-ha
+00007ea0: 2c20 2d2d 686f 616c 7068 612c 2074 6865  , --hoalpha, the
+00007eb0: 206f 7061 6369 7479 206f 6620 686f 6d6f   opacity of homo
+00007ec0: 6c6f 6720 646f 7473 2c20 6465 6661 756c  log dots, defaul
+00007ed0: 7420 300a 2d73 7274 2c20 2d2d 7368 6f77  t 0.-srt, --show
+00007ee0: 7265 616c 7469 636b 2c20 666c 6167 206f  realtick, flag o
+00007ef0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00007f00: 6f20 7368 6f77 2074 6865 2072 6561 6c20  o show the real 
+00007f10: 7469 636b 2069 6e20 6765 6e65 7320 6f72  tick in genes or
+00007f20: 2062 6173 6573 2c20 6966 2074 6865 2066   bases, if the f
+00007f30: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00007f40: 2072 6561 6c20 7469 636b 2077 6f75 6c64   real tick would
+00007f50: 2062 6520 7368 6f77 6564 0a2d 746c 732c   be showed.-tls,
+00007f60: 202d 2d74 6963 6b6c 6162 656c 7369 7a65   --ticklabelsize
+00007f70: 2c20 7468 6520 6c61 6265 6c20 7369 7a65  , the label size
+00007f80: 206f 6620 7469 636b 2c20 6465 6661 756c   of tick, defaul
+00007f90: 7420 350a 2d78 6c2c 202d 2d78 6c69 6d2c  t 5.-xl, --xlim,
+00007fa0: 2074 6865 2078 2061 7869 7320 6c69 6d69   the x axis limi
+00007fb0: 7420 6f66 204b 7320 6469 7374 7269 6275  t of Ks distribu
+00007fc0: 7469 6f6e 0a2d 796c 2c20 2d2d 796c 696d  tion.-yl, --ylim
+00007fd0: 2c20 7468 6520 7920 6178 6973 206c 696d  , the y axis lim
+00007fe0: 6974 206f 6620 4b73 2064 6973 7472 6962  it of Ks distrib
+00007ff0: 7574 696f 6e0a 2d61 646f 2c20 2d2d 6164  ution.-ado, --ad
+00008000: 6a75 7374 6f72 7468 6f2c 2066 6c61 6720  justortho, flag 
+00008010: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00008020: 746f 2061 646a 7573 7420 7468 6520 6869  to adjust the hi
+00008030: 7374 6f67 7261 6d20 6865 6967 6874 206f  stogram height o
+00008040: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
+00008050: 2061 7320 746f 206d 6174 6368 2074 6865   as to match the
+00008060: 2068 6569 6768 7420 6f66 2070 6172 616c   height of paral
+00008070: 6f67 6f75 7320 4b73 2c20 6966 2074 6865  ogous Ks, if the
+00008080: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00008090: 6865 2061 646a 7573 746d 656e 7420 776f  he adjustment wo
+000080a0: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
+000080b0: 0a2d 6164 662c 202d 2d61 646a 7573 7466  .-adf, --adjustf
+000080c0: 6163 746f 722c 2074 6865 2061 646a 7573  actor, the adjus
+000080d0: 746d 656e 7420 6661 6374 6f72 206f 6620  tment factor of 
+000080e0: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2c20  orthologous Ks, 
+000080f0: 6465 6661 756c 7420 302e 350a 2d6f 612c  default 0.5.-oa,
+00008100: 202d 2d6f 6b61 6c70 6861 2c20 7468 6520   --okalpha, the 
+00008110: 6f70 6163 6974 7920 6f66 206f 7274 686f  opacity of ortho
+00008120: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
+00008130: 6275 7469 6f6e 2069 6e20 6d69 7865 6420  bution in mixed 
+00008140: 706c 6f74 2c20 6465 6661 756c 7420 302e  plot, default 0.
+00008150: 350a 2d63 732c 202d 2d63 6c61 7373 6963  5.-cs, --classic
+00008160: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00008170: 6865 7468 6572 2074 6f20 6472 6177 206d  hether to draw m
+00008180: 6978 6564 204b 7320 706c 6f74 2069 6e20  ixed Ks plot in 
+00008190: 6120 636c 6173 7369 6320 6d61 6e6e 6572  a classic manner
+000081a0: 2077 6865 7265 2074 6865 2066 756c 6c20   where the full 
+000081b0: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2064  orthologous Ks d
+000081c0: 6973 7472 6962 7574 696f 6e20 6973 2064  istribution is d
+000081d0: 7261 7765 642c 2069 6620 7468 6520 666c  rawed, if the fl
+000081e0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+000081f0: 636c 6173 7369 6320 6d69 7865 6420 4b73  classic mixed Ks
+00008200: 2070 6c6f 7420 776f 756c 6420 6265 2064   plot would be d
+00008210: 7261 776e 0a2d 7461 2c20 2d2d 746f 7061  rawn.-ta, --topa
+00008220: 7272 6f77 2c20 666c 6167 206f 7074 696f  rrow, flag optio
+00008230: 6e2c 2077 6865 7468 6572 2074 6f20 6164  n, whether to ad
+00008240: 6a75 7374 2074 6865 2061 7272 6f77 2074  just the arrow t
+00008250: 6f20 6265 2061 7420 7468 6520 746f 7020  o be at the top 
+00008260: 6f66 2074 6865 2070 6c6f 742c 2069 6e73  of the plot, ins
+00008270: 7465 6164 206f 6620 6265 696e 6720 636f  tead of being co
+00008280: 6f72 6469 6e61 7465 6420 6173 2074 6865  ordinated as the
+00008290: 204b 4445 206f 6620 7468 6520 6f72 7468   KDE of the orth
+000082a0: 6f6c 6f67 6f75 7320 4b73 2064 6973 7472  ologous Ks distr
+000082b0: 6962 7574 696f 6e2c 2069 6620 7468 6520  ibution, if the 
+000082c0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+000082d0: 6520 6172 726f 7720 776f 756c 6420 6265  e arrow would be
+000082e0: 2073 6574 2061 7420 7468 6520 746f 700a   set at the top.
+000082f0: 2d6e 612c 202d 2d6e 6f64 6561 7665 7261  -na, --nodeavera
+00008300: 6765 642c 2066 6c61 6720 6f70 7469 6f6e  ged, flag option
+00008310: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
+00008320: 206e 6f64 652d 6176 6572 6167 6564 206d   node-averaged m
+00008330: 6574 686f 6420 666f 7220 6465 2d72 6564  ethod for de-red
+00008340: 756e 6461 6e63 792c 2069 6620 7468 6520  undancy, if the 
+00008350: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00008360: 6520 6e6f 6465 2d61 7665 7261 6765 6420  e node-averaged 
+00008370: 6d65 7468 6f64 2077 6f75 6c64 2062 6520  method would be 
+00008380: 696e 6974 6961 7465 640a 2d62 732c 202d  initiated.-bs, -
+00008390: 2d62 6f6f 7473 7472 6170 2c20 7468 6520  -bootstrap, the 
+000083a0: 6e75 6d62 6572 206f 6620 626f 6f74 7374  number of bootst
+000083b0: 7261 7020 7265 706c 6963 6174 6573 206f  rap replicates o
+000083c0: 6620 6f72 7468 6f6c 6f67 204b 7320 6469  f ortholog Ks di
+000083d0: 7374 7269 6275 7469 6f6e 2069 6e20 6d69  stribution in mi
+000083e0: 7865 6420 706c 6f74 0a2d 6772 2c20 2d2d  xed plot.-gr, --
+000083f0: 6769 7374 7262 2c20 666c 6167 206f 7074  gistrb, flag opt
+00008400: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00008410: 7573 6520 6769 7374 5f72 6169 6e62 6f77  use gist_rainbow
+00008420: 2061 7320 636f 6c6f 7220 6d61 7020 6f66   as color map of
+00008430: 2064 6f74 706c 6f74 0a2d 6e2c 202d 2d6e   dotplot.-n, --n
+00008440: 7468 7265 6164 732c 2074 6865 206e 756d  threads, the num
+00008450: 6265 7220 6f66 2074 6872 6561 6473 2074  ber of threads t
+00008460: 6f20 7573 6520 696e 2062 6f6f 7473 7472  o use in bootstr
+00008470: 6170 2073 616d 706c 696e 672c 2064 6566  ap sampling, def
+00008480: 6175 6c74 2031 0a60 6060 0a0a 2323 2055  ault 1.```..## U
+00008490: 7361 6765 0a0a 4865 7265 2077 6520 7072  sage..Here we pr
+000084a0: 6f76 6964 6564 2074 6865 2062 6173 6963  ovided the basic
+000084b0: 2075 7361 6765 2066 6f72 2065 6163 6820   usage for each 
+000084c0: 7072 6f67 7261 6d20 616e 6420 7468 6520  program and the 
+000084d0: 7265 6c65 7661 6e74 2070 6172 616d 6574  relevant paramet
+000084e0: 6572 7320 616e 6420 7375 6767 6573 7469  ers and suggesti
+000084f0: 6f6e 7320 6f6e 2070 6172 616d 6574 6572  ons on parameter
+00008500: 697a 6174 696f 6e2e 2041 2072 656d 696e  ization. A remin
+00008510: 6465 7220 7468 6174 2074 6865 2067 6976  der that the giv
+00008520: 656e 2063 6f72 6573 2061 6e64 2074 6872  en cores and thr
+00008530: 6561 6473 2063 616e 2073 6967 6e69 6669  eads can signifi
+00008540: 6361 6e74 6c79 2069 6d70 6163 7420 7468  cantly impact th
+00008550: 6520 7275 6e20 7469 6d65 2061 6e64 2074  e run time and t
+00008560: 6875 7320 7765 2061 6464 6564 2073 6f6d  hus we added som
+00008570: 6520 7265 706f 7274 2069 6e66 6f72 6d61  e report informa
+00008580: 7469 6f6e 2070 6572 7461 696e 696e 6720  tion pertaining 
+00008590: 746f 2074 6865 2073 7973 7465 6d20 6f66  to the system of
+000085a0: 2075 7365 7273 2074 6f20 6661 6369 6c69   users to facili
+000085b0: 7461 7465 2074 6865 2065 6666 6963 6965  tate the efficie
+000085c0: 6e74 2073 6574 7469 6e67 206f 6620 7468  nt setting of th
+000085d0: 7265 6164 7320 616e 6420 6d65 6d6f 7279  reads and memory
+000085e0: 2e20 5468 6520 6c6f 6769 6361 6c20 4350  . The logical CP
+000085f0: 5573 2072 6570 6f72 7465 6420 7265 7072  Us reported repr
+00008600: 6573 656e 7473 2074 6865 206e 756d 6265  esents the numbe
+00008610: 7220 6f66 2070 6879 7369 6361 6c20 636f  r of physical co
+00008620: 7265 7320 6d75 6c74 6970 6c69 6564 2062  res multiplied b
+00008630: 7920 7468 6520 6e75 6d62 6572 206f 6620  y the number of 
+00008640: 7468 7265 6164 7320 7468 6174 2063 616e  threads that can
+00008650: 2072 756e 206f 6e20 6561 6368 2063 6f72   run on each cor
+00008660: 652c 2061 6c73 6f20 6b6e 6f77 6e20 6173  e, also known as
+00008670: 2048 7970 6572 2054 6872 6561 6469 6e67   Hyper Threading
+00008680: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
+00008690: 6c6f 6769 6361 6c20 4350 5573 206d 6179  logical CPUs may
+000086a0: 206e 6f74 206e 6563 6573 7361 7269 6c79   not necessarily
+000086b0: 2062 6520 6571 7569 7661 6c65 6e74 2074   be equivalent t
+000086c0: 6f20 7468 6520 6163 7475 616c 206e 756d  o the actual num
+000086d0: 6265 7220 6f66 2043 5055 7320 7468 6520  ber of CPUs the 
+000086e0: 6375 7272 656e 7420 7072 6f63 6573 7320  current process 
+000086f0: 6361 6e20 7573 652e 2054 6865 2061 7661  can use. The ava
+00008700: 696c 6162 6c65 206d 656d 6f72 7920 7265  ilable memory re
+00008710: 6665 7273 2074 6f20 7468 6520 6d65 6d6f  fers to the memo
+00008720: 7279 2074 6861 7420 6361 6e20 6265 2067  ry that can be g
+00008730: 6976 656e 2069 6e73 7461 6e74 6c79 2074  iven instantly t
+00008740: 6f20 7072 6f63 6573 7365 7320 7769 7468  o processes with
+00008750: 6f75 7420 7468 6520 7379 7374 656d 2067  out the system g
+00008760: 6f69 6e67 2069 6e74 6f20 7377 6170 2061  oing into swap a
+00008770: 6e64 2072 6566 6c65 6374 7320 7468 6520  nd reflects the 
+00008780: 6163 7475 616c 206d 656d 6f72 7920 6176  actual memory av
+00008790: 6169 6c61 626c 652e 2054 6865 2066 7265  ailable. The fre
+000087a0: 6520 6d65 6d6f 7279 2072 6566 6572 7320  e memory refers 
+000087b0: 746f 2074 6865 206d 656d 6f72 7920 6e6f  to the memory no
+000087c0: 7420 6265 696e 6720 7573 6564 2061 7420  t being used at 
+000087d0: 616c 6c20 287a 6572 6f65 6429 2074 6861  all (zeroed) tha
+000087e0: 7420 6973 2072 6561 6469 6c79 2061 7661  t is readily ava
+000087f0: 696c 6162 6c65 2e20 5468 6520 6465 7363  ilable. The desc
+00008800: 7269 7074 696f 6e20 6162 6f76 6520 7265  ription above re
+00008810: 6665 7273 2074 6f20 7468 6520 646f 6375  fers to the docu
+00008820: 6d65 6e74 6174 696f 6e20 6f66 2060 7073  mentation of `ps
+00008830: 7574 696c 602e 0a0a 2323 2320 7767 6420  util`...### wgd 
+00008840: 646d 640a 0a2a 2a54 6865 2064 656c 696e  dmd..**The delin
+00008850: 6561 7469 6f6e 206f 6620 7768 6f6c 6520  eation of whole 
+00008860: 7061 7261 6e6f 6d65 2a2a 0a60 6060 0a77  paranome**.```.w
+00008870: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
+00008880: 5f63 6f65 7275 6c65 6120 2d49 2032 202d  _coerulea -I 2 -
+00008890: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
+000088a0: 2d6e 7020 3520 282d 6e6e 2920 282d 2d74  -np 5 (-nn) (--t
+000088b0: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
+000088c0: 282d 6e20 3429 2028 2d6f 2077 6764 5f64  (-n 4) (-o wgd_d
+000088d0: 6d64 2920 282d 7420 776f 726b 696e 675f  md) (-t working_
+000088e0: 746d 7029 0a60 6060 200a 0a4e 6f74 6520  tmp).``` ..Note 
+000088f0: 7468 6174 2077 6520 646f 6e27 7420 7072  that we don't pr
+00008900: 6f76 6964 6520 7468 6520 6461 7461 206f  ovide the data o
+00008910: 6620 7468 6973 2063 6f64 696e 6720 7365  f this coding se
+00008920: 7175 656e 6365 2028 6364 7329 2066 696c  quence (cds) fil
+00008930: 6520 6041 7175 696c 6567 6961 5f63 6f65  e `Aquilegia_coe
+00008940: 7275 6c65 6160 2062 7574 2069 7420 6361  rulea` but it ca
+00008950: 6e20 6265 2065 6173 696c 7920 646f 776e  n be easily down
+00008960: 6c6f 6164 6564 2061 7420 5b50 6879 746f  loaded at [Phyto
+00008970: 7a6f 6d65 5d28 6874 7470 733a 2f2f 7068  zome](https://ph
+00008980: 7974 6f7a 6f6d 652d 6e65 7874 2e6a 6769  ytozome-next.jgi
+00008990: 2e64 6f65 2e67 6f76 2f69 6e66 6f2f 4163  .doe.gov/info/Ac
+000089a0: 6f65 7275 6c65 615f 7633 5f31 2920 2873  oerulea_v3_1) (s
+000089b0: 616d 6520 666f 7220 6f74 6865 7220 5573  ame for other Us
+000089c0: 6167 6520 646f 6329 2e20 4e6f 7465 2074  age doc). Note t
+000089d0: 6861 7420 696e 2074 6865 2069 7373 7565  hat in the issue
+000089e0: 7320 736f 6d65 2075 7365 7273 2064 6964  s some users did
+000089f0: 6e27 7420 646f 776e 6c6f 6164 2074 6865  n't download the
+00008a00: 2060 4163 6f65 7275 6c65 615f 3332 325f   `Acoerulea_322_
+00008a10: 7633 2e31 2e63 6473 5f70 7269 6d61 7279  v3.1.cds_primary
+00008a20: 5472 616e 7363 7269 7074 4f6e 6c79 2e66  TranscriptOnly.f
+00008a30: 612e 677a 6020 6275 7420 696e 7374 6561  a.gz` but instea
+00008a40: 6420 7468 6520 6041 636f 6572 756c 6561  d the `Acoerulea
+00008a50: 5f33 3232 5f76 332e 312e 6364 732e 6661  _322_v3.1.cds.fa
+00008a60: 2e67 7a60 2066 696c 652e 2046 6f72 2074  .gz` file. For t
+00008a70: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+00008a80: 6f66 2077 686f 6c65 2070 6172 616e 6f6d  of whole paranom
+00008a90: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00008aa0: 3e20 6469 7374 7269 6275 7469 6f6e 2c20  > distribution, 
+00008ab0: 2a2a 6f6e 6c79 206f 6e65 2074 7261 6e73  **only one trans
+00008ac0: 6372 6970 7420 2874 6865 2070 7269 6d61  cript (the prima
+00008ad0: 7279 206f 6e65 2920 7065 7220 6765 6e65  ry one) per gene
+00008ae0: 2073 686f 756c 6420 6265 2069 6e63 6c75   should be inclu
+00008af0: 6465 642a 2a20 7375 6368 2074 6861 7420  ded** such that 
+00008b00: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+00008b10: 7562 3e20 6973 2072 6561 6c6c 7920 696e  ub> is really in
+00008b20: 6469 6361 7469 6e67 2074 6865 2061 6765  dicating the age
+00008b30: 206f 6620 6765 6e65 2064 7570 6c69 6361   of gene duplica
+00008b40: 7469 6f6e 2065 7665 6e74 2c20 696e 7374  tion event, inst
+00008b50: 6561 6420 6f66 2061 6c74 6572 6e61 7469  ead of alternati
+00008b60: 7665 2073 706c 6963 696e 672e 2054 7261  ve splicing. Tra
+00008b70: 6e73 6372 6970 746f 6d65 2064 6174 6120  nscriptome data 
+00008b80: 7368 6f75 6c64 2062 6520 6361 7265 6675  should be carefu
+00008b90: 6c6c 7920 7472 6561 7465 6420 7769 7468  lly treated with
+00008ba0: 2064 652d 7265 6475 6e64 616e 6379 2073   de-redundancy s
+00008bb0: 6f20 6173 2074 6f20 7265 6475 6365 2074  o as to reduce t
+00008bc0: 6865 2066 616c 7365 2070 6f73 6974 6976  he false positiv
+00008bd0: 6520 6475 706c 6963 6174 696f 6e20 6275  e duplication bu
+00008be0: 6d70 2063 6175 7365 6420 6279 2070 6572  mp caused by per
+00008bf0: 7661 7369 7665 2061 6c74 6572 6e61 7469  vasive alternati
+00008c00: 7665 2073 706c 6963 696e 672e 2048 6572  ve splicing. Her
+00008c10: 6520 7468 6520 696e 666c 6174 696f 6e20  e the inflation 
+00008c20: 6661 6374 6f72 2070 6172 616d 6574 6572  factor parameter
+00008c30: 2c20 6769 7665 6e20 6279 2060 2d49 6020  , given by `-I` 
+00008c40: 6f72 2060 2d2d 696e 666c 6174 696f 6e60  or `--inflation`
+00008c50: 2c20 6166 6665 6374 7320 7468 6520 6772  , affects the gr
+00008c60: 616e 756c 6172 6974 7920 6f72 2072 6573  anularity or res
+00008c70: 6f6c 7574 696f 6e20 6f66 2074 6865 2063  olution of the c
+00008c80: 6c75 7374 6572 696e 6720 6f75 7463 6f6d  lustering outcom
+00008c90: 6520 616e 6420 696d 706c 6963 6974 6c79  e and implicitly
+00008ca0: 2063 6f6e 7472 6f6c 6c73 2074 6865 206e   controlls the n
+00008cb0: 756d 6265 7220 6f66 2063 6c75 7374 6572  umber of cluster
+00008cc0: 732c 2077 6974 6820 6c6f 7720 7661 6c75  s, with low valu
+00008cd0: 6573 2073 7563 6820 6173 2031 2e33 206f  es such as 1.3 o
+00008ce0: 7220 312e 3420 6c65 6164 696e 6720 746f  r 1.4 leading to
+00008cf0: 2066 6577 6572 2062 7574 206c 6172 6765   fewer but large
+00008d00: 7220 636c 7573 7465 7273 2061 6e64 2068  r clusters and h
+00008d10: 6967 6820 7661 6c75 6573 2073 7563 6820  igh values such 
+00008d20: 6173 2035 206f 7220 3620 6c65 6164 696e  as 5 or 6 leadin
+00008d30: 6720 746f 206d 6f72 6520 6275 7420 736d  g to more but sm
+00008d40: 616c 6c65 7220 636c 7573 7465 7273 2e20  aller clusters. 
+00008d50: 5765 2073 6574 2074 6865 2064 6566 6175  We set the defau
+00008d60: 6c74 2076 616c 7565 2061 7320 3220 6173  lt value as 2 as
+00008d70: 2073 7567 6765 7374 6564 2062 7920 5b4d   suggested by [M
+00008d80: 434c 5d28 6874 7470 733a 2f2f 6d69 6361  CL](https://mica
+00008d90: 6e73 2e6f 7267 2f6d 636c 2f29 2e20 5468  ns.org/mcl/). Th
+00008da0: 6520 652d 7661 6c75 6520 6375 742d 6f66  e e-value cut-of
+00008db0: 6620 666f 7220 7365 7175 656e 6365 2073  f for sequence s
+00008dc0: 696d 696c 6172 6974 792c 2067 6976 656e  imilarity, given
+00008dd0: 2062 7920 602d 6560 206f 7220 602d 2d65   by `-e` or `--e
+00008de0: 7661 6c60 2c20 7768 6963 6820 6465 6e6f  val`, which deno
+00008df0: 7465 7320 7468 6520 6578 7065 6374 6564  tes the expected
+00008e00: 2076 616c 7565 206f 6620 7468 6520 6869   value of the hi
+00008e10: 7420 7175 616e 7469 6669 6573 2074 6865  t quantifies the
+00008e20: 206e 756d 6265 7220 6f66 2061 6c69 676e   number of align
+00008e30: 6d65 6e74 7320 6f66 2073 696d 696c 6172  ments of similar
+00008e40: 206f 7220 6265 7474 6572 2071 7561 6c69   or better quali
+00008e50: 7479 2074 6861 7420 796f 7520 6578 7065  ty that you expe
+00008e60: 6374 2074 6f20 6669 6e64 2073 6561 7263  ct to find searc
+00008e70: 6869 6e67 2074 6869 7320 7175 6572 7920  hing this query 
+00008e80: 6167 6169 6e73 7420 6120 6461 7461 6261  against a databa
+00008e90: 7365 206f 6620 7261 6e64 6f6d 2073 6571  se of random seq
+00008ea0: 7565 6e63 6573 2074 6865 2073 616d 6520  uences the same 
+00008eb0: 7369 7a65 2061 7320 7468 6520 6163 7475  size as the actu
+00008ec0: 616c 2074 6172 6765 7420 6461 7461 6261  al target databa
+00008ed0: 7365 2c20 6973 2074 6865 206b 6579 2070  se, is the key p
+00008ee0: 6172 616d 6574 6572 206d 6561 7375 7269  arameter measuri
+00008ef0: 6e67 2074 6865 2073 6967 6e69 6669 6361  ng the significa
+00008f00: 6e63 6520 6f66 2061 2068 6974 2c20 7768  nce of a hit, wh
+00008f10: 6963 6820 6973 2073 6574 2068 6572 6520  ich is set here 
+00008f20: 6173 2064 6566 6175 6c74 2031 652d 3130  as default 1e-10
+00008f30: 2e20 4e6f 7465 2074 6861 7420 5b44 4941  . Note that [DIA
+00008f40: 4d4f 4e44 5d28 6874 7470 733a 2f2f 6769  MOND](https://gi
+00008f50: 7468 7562 2e63 6f6d 2f62 6275 6368 6669  thub.com/bbuchfi
+00008f60: 6e6b 2f64 6961 6d6f 6e64 2f77 696b 692f  nk/diamond/wiki/
+00008f70: 312e 2d54 7574 6f72 6961 6c29 2069 7473  1.-Tutorial) its
+00008f80: 656c 6620 6279 2064 6566 6175 6c74 206f  elf by default o
+00008f90: 6e6c 7920 7265 706f 7274 7320 616c 6c20  nly reports all 
+00008fa0: 616c 6967 6e6d 656e 7473 2077 6974 6820  alignments with 
+00008fb0: 652d 7661 6c75 6520 3c20 302e 3030 312e  e-value < 0.001.
+00008fc0: 2054 6865 2070 6572 6365 6e74 6167 6520   The percentage 
+00008fd0: 6f66 2075 7070 6572 2068 6974 7320 7573  of upper hits us
+00008fe0: 6564 2066 6f72 2067 656e 6520 6c65 6e67  ed for gene leng
+00008ff0: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
+00009000: 2c20 6769 7665 6e20 6279 2060 2d6e 7060  , given by `-np`
+00009010: 206f 7220 602d 2d6e 6f72 6d61 6c69 7a65   or `--normalize
+00009020: 6470 6572 6365 6e74 602c 2077 6869 6368  dpercent`, which
+00009030: 2064 6574 6572 6d69 6e65 7320 7468 6520   determines the 
+00009040: 7570 7065 7220 7065 7263 656e 7469 6c65  upper percentile
+00009050: 206f 6620 6869 7473 2070 6572 2062 696e   of hits per bin
+00009060: 2028 6361 7465 676f 7269 7a65 6420 6279   (categorized by
+00009070: 2067 656e 6520 6c65 6e67 7468 2920 7573   gene length) us
+00009080: 6564 2069 6e20 7468 6520 6669 7420 6f66  ed in the fit of
+00009090: 206c 696e 6561 7220 7265 6772 6573 7369   linear regressi
+000090a0: 6f6e 2c20 636f 6e73 6964 6572 696e 6720  on, considering 
+000090b0: 7468 6174 206e 6f74 2061 6c6c 2068 6974  that not all hit
+000090c0: 7320 7065 7220 6269 6e20 7368 6f77 2061  s per bin show a
+000090d0: 7070 6172 656e 7420 6c69 6e65 6172 2072  pparent linear r
+000090e0: 656c 6174 696f 6e73 6869 702c 2069 7320  elationship, is 
+000090f0: 7365 7420 6173 2064 6566 6175 6c74 2035  set as default 5
+00009100: 2c20 696e 6469 6361 7469 6e67 2074 6865  , indicating the
+00009110: 2075 7361 6765 206f 6620 746f 7020 3525   usage of top 5%
+00009120: 2068 6974 7320 7065 7220 6269 6e2e 2054   hits per bin. T
+00009130: 6865 206e 756d 6265 7220 6f66 2062 696e  he number of bin
+00009140: 7320 6469 7669 6465 6420 696e 2067 656e  s divided in gen
+00009150: 6520 6c65 6e67 7468 206e 6f72 6d61 6c69  e length normali
+00009160: 7a61 7469 6f6e 2c20 6769 7665 6e20 6279  zation, given by
+00009170: 2060 2d62 7360 206f 7220 602d 2d62 696e   `-bs` or `--bin
+00009180: 7360 2c20 6465 7465 726d 696e 6573 2074  s`, determines t
+00009190: 6865 206e 756d 6265 7220 6f66 2062 696e  he number of bin
+000091a0: 7320 746f 2063 6174 6567 6f72 697a 6520  s to categorize 
+000091b0: 7468 6520 6765 6e65 206c 656e 6774 682c  the gene length,
+000091c0: 2069 7320 7365 7420 6173 2064 6566 6175   is set as defau
+000091d0: 6c74 2031 3030 2e20 5468 6520 7061 7261  lt 100. The para
+000091e0: 6d65 7465 7220 602d 6e6e 6020 6f72 2060  meter `-nn` or `
+000091f0: 2d2d 6e6f 6e6f 726d 616c 697a 6174 696f  --nonormalizatio
+00009200: 6e60 2063 616e 2062 6520 7365 7420 746f  n` can be set to
+00009210: 2063 616c 6c20 6f66 6620 7468 6520 6e6f   call off the no
+00009220: 726d 616c 697a 6174 696f 6e20 7072 6f63  rmalization proc
+00009230: 6573 732c 2061 6c74 686f 7567 6820 6974  ess, although it
+00009240: 2773 2073 7567 6765 7374 6564 2074 6f20  's suggested to 
+00009250: 636f 6e64 7563 7420 7468 6520 6e6f 726d  conduct the norm
+00009260: 616c 697a 6174 696f 6e20 746f 2061 6371  alization to acq
+00009270: 7569 7265 206d 6f72 6520 6163 6375 7261  uire more accura
+00009280: 7465 2067 656e 6520 6661 6d69 6c79 2063  te gene family c
+00009290: 6c75 7374 6572 696e 6720 7265 7375 6c74  lustering result
+000092a0: 2e20 5468 6520 7061 7261 6d65 7465 7273  . The parameters
+000092b0: 2060 2d2d 746f 5f73 746f 7060 2061 6e64   `--to_stop` and
+000092c0: 2060 2d2d 6364 7360 2063 6f6e 7472 6f6c   `--cds` control
+000092d0: 2074 6865 2062 6568 6176 696f 7572 206f   the behaviour o
+000092e0: 6620 7472 616e 736c 6174 696e 6720 636f  f translating co
+000092f0: 6469 6e67 2073 6571 7565 6e63 6520 696e  ding sequence in
+00009300: 746f 2061 6d69 6e6f 2061 6369 6420 7365  to amino acid se
+00009310: 7175 656e 6365 2e20 4966 2074 6865 2060  quence. If the `
+00009320: 2d2d 746f 5f73 746f 7060 2077 6173 2073  --to_stop` was s
+00009330: 6574 2c20 7468 6520 7472 616e 736c 6174  et, the translat
+00009340: 696f 6e20 776f 756c 6420 6265 2074 6572  ion would be ter
+00009350: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
+00009360: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
+00009370: 6f70 2063 6f64 6f6e 2c20 6f74 6865 7277  op codon, otherw
+00009380: 6973 6520 7468 6520 7472 616e 736c 6174  ise the translat
+00009390: 696f 6e20 776f 756c 6420 7369 6d70 6c79  ion would simply
+000093a0: 2073 6b69 7020 616e 7920 7374 6f70 2063   skip any stop c
+000093b0: 6f64 6f6e 732e 2049 6620 7468 6520 602d  odons. If the `-
+000093c0: 2d63 6473 6020 7761 7320 7365 742c 2073  -cds` was set, s
+000093d0: 6571 7565 6e63 6573 2074 6861 7420 646f  equences that do
+000093e0: 6573 6e27 7420 7374 6172 7420 7769 7468  esn't start with
+000093f0: 2061 2076 616c 6964 2073 7461 7274 2063   a valid start c
+00009400: 6f64 6f6e 2c20 6f72 2063 6f6e 7461 696e  odon, or contain
+00009410: 7320 6d6f 7265 2074 6861 6e20 6f6e 6520  s more than one 
+00009420: 696e 2d66 7261 6d65 2073 746f 7020 636f  in-frame stop co
+00009430: 646f 6e2c 206f 7220 6973 206e 6f74 2064  don, or is not d
+00009440: 6976 6964 6162 6c65 2062 7920 332c 2077  ividable by 3, w
+00009450: 6f75 6c64 2062 6520 7369 6d70 6c79 2064  ould be simply d
+00009460: 726f 7070 6564 2c20 7375 6368 2074 6861  ropped, such tha
+00009470: 7420 6f6e 6c79 2073 7472 6963 7420 636f  t only strict co
+00009480: 6d70 6c65 7465 2063 6f64 696e 6720 7365  mplete coding se
+00009490: 7175 656e 6365 7320 776f 756c 6420 6265  quences would be
+000094a0: 2069 6e63 6c75 6465 6420 696e 2074 6865   included in the
+000094b0: 2073 7562 7365 7175 656e 7420 616e 616c   subsequent anal
+000094c0: 7973 6973 2e20 5468 6520 6e75 6d62 6572  ysis. The number
+000094d0: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
+000094e0: 6561 6473 2062 7920 7468 6520 6f70 7469  eads by the opti
+000094f0: 6f6e 2060 2d6e 6020 6f72 2060 2d2d 6e74  on `-n` or `--nt
+00009500: 6872 6561 6473 6020 6361 6e20 6265 2073  hreads` can be s
+00009510: 6574 2074 6f20 6163 6365 6c65 7261 7465  et to accelerate
+00009520: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
+00009530: 2077 6974 6869 6e20 6064 6961 6d6f 6e64   within `diamond
+00009540: 602e 2054 6865 2064 6972 6563 746f 7279  `. The directory
+00009550: 206f 6620 6f75 7470 7574 206f 7220 696e   of output or in
+00009560: 7465 726d 6564 6961 7465 2066 696c 6573  termediate files
+00009570: 2069 7320 6465 7465 726d 696e 6564 2062   is determined b
+00009580: 7920 7468 6520 7061 7261 6d65 7465 7220  y the parameter 
+00009590: 602d 6f60 206f 7220 602d 2d6f 7574 6469  `-o` or `--outdi
+000095a0: 7260 2c20 616e 6420 602d 7460 206f 7220  r`, and `-t` or 
+000095b0: 602d 2d74 6d70 6469 7260 2c20 7768 6963  `--tmpdir`, whic
+000095c0: 6820 7769 6c6c 2062 6520 6372 6561 7465  h will be create
+000095d0: 6420 6279 2074 6865 2070 726f 6772 616d  d by the program
+000095e0: 2069 7473 656c 6620 616e 6420 6265 206f   itself and be o
+000095f0: 7665 7277 7269 7474 656e 2069 6620 7468  verwritten if th
+00009600: 6520 666f 6c64 6572 2068 6173 2061 6c72  e folder has alr
+00009610: 6561 6479 2062 6565 6e20 6372 6561 7465  eady been create
+00009620: 642e 204e 6f74 6520 7468 6174 2074 6865  d. Note that the
+00009630: 2073 6f66 7477 6172 6520 6064 6961 6d6f   software `diamo
+00009640: 6e64 6020 7368 6f75 6c64 2062 6520 7072  nd` should be pr
+00009650: 652d 696e 7374 616c 6c65 6420 616e 6420  e-installed and 
+00009660: 7365 7420 696e 2074 6865 2065 6e76 6972  set in the envir
+00009670: 6f6e 6d65 6e74 2070 6174 6820 696e 2061  onment path in a
+00009680: 6c6c 2074 6865 2061 6e61 6c79 7369 7320  ll the analysis 
+00009690: 7065 7266 6f72 6d65 6420 6279 2060 7767  performed by `wg
+000096a0: 6420 646d 6460 2065 7863 6570 7420 666f  d dmd` except fo
+000096b0: 7220 7468 6520 636f 6c6c 696e 6561 7220  r the collinear 
+000096c0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+000096d0: 7973 6973 2e0a 0a2a 2a57 6520 7375 6767  ysis...**We sugg
+000096e0: 6573 7420 7468 6174 2074 6865 2064 6566  est that the def
+000096f0: 6175 6c74 2073 6574 7469 6e67 2069 6e20  ault setting in 
+00009700: 7768 6963 6820 7468 6520 696e 666c 6174  which the inflat
+00009710: 696f 6e20 6661 6374 6f72 2069 7320 7365  ion factor is se
+00009720: 7420 6173 2032 2c20 652d 7661 6c75 6520  t as 2, e-value 
+00009730: 6375 742d 6f66 6620 6173 2031 652d 3130  cut-off as 1e-10
+00009740: 2061 6e64 206f 7468 6572 2070 6172 616d   and other param
+00009750: 6574 6572 7320 696e 2064 6566 6175 6c74  eters in default
+00009760: 2069 7320 6120 676f 6f64 2073 7461 7274   is a good start
+00009770: 696e 6720 706f 696e 742c 2075 6e6c 6573  ing point, unles
+00009780: 7320 796f 7520 7370 6563 6966 6963 616c  s you specifical
+00009790: 6c79 2077 616e 7420 746f 2065 7870 6c6f  ly want to explo
+000097a0: 7265 2074 6865 2065 6666 6563 7473 206f  re the effects o
+000097b0: 6620 6469 6666 6572 656e 7420 7061 7261  f different para
+000097c0: 6d65 7465 7273 2e2a 2a20 5375 6368 2074  meters.** Such t
+000097d0: 6861 7420 7468 6520 636f 6d6d 616e 6420  hat the command 
+000097e0: 666f 7220 7468 6520 6465 6c69 6e65 6174  for the delineat
+000097f0: 696f 6e20 6f66 2077 686f 6c65 2070 6172  ion of whole par
+00009800: 616e 6f6d 6520 6973 2073 696d 706c 7920  anome is simply 
+00009810: 6173 2062 656c 6f77 2e0a 0a60 6060 0a77  as below...```.w
+00009820: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
+00009830: 5f63 6f65 7275 6c65 610a 6060 600a 0a2a  _coerulea.```..*
+00009840: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
+00009850: 206f 6620 5242 4873 2a2a 0a60 6060 0a77   of RBHs**.```.w
+00009860: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
+00009870: 2073 6571 7565 6e63 6532 202d 6520 3165   sequence2 -e 1e
+00009880: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
+00009890: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
+000098a0: 2d63 2030 2e39 2920 282d 2d6f 6766 6f72  -c 0.9) (--ogfor
+000098b0: 6d61 7429 2028 2d2d 746f 5f73 746f 7029  mat) (--to_stop)
+000098c0: 2028 2d2d 6364 7329 2028 2d6f 2077 6764   (--cds) (-o wgd
+000098d0: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
+000098e0: 675f 746d 7029 0a60 6060 0a0a 546f 2064  g_tmp).```..To d
+000098f0: 656c 696e 6561 7465 2052 4248 7320 6265  elineate RBHs be
+00009900: 7477 6565 6e20 7477 6f20 6364 7320 7365  tween two cds se
+00009910: 7175 656e 6365 2066 696c 6573 2c20 7468  quence files, th
+00009920: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
+00009930: 6574 6572 2069 7320 6d6f 7374 6c79 2074  eter is mostly t
+00009940: 6865 2073 616d 6520 6173 2077 686f 6c65  he same as whole
+00009950: 2070 6172 616e 6f6d 6520 696e 6665 7265   paranome infere
+00009960: 6e63 652c 2065 7863 6570 7420 666f 7220  nce, except for 
+00009970: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+00009980: 6360 206f 7220 602d 2d63 7363 6f72 6560  c` or `--cscore`
+00009990: 2c20 7768 6963 6820 7261 6e67 6573 2062  , which ranges b
+000099a0: 6574 7765 656e 2030 2061 6e64 2031 2061  etween 0 and 1 a
+000099b0: 6e64 2069 7320 7573 6564 2074 6f20 7265  nd is used to re
+000099c0: 6c61 7820 7468 6520 7369 6d69 6c61 7269  lax the similari
+000099d0: 7479 2063 7574 6f66 6620 6672 6f6d 2074  ty cutoff from t
+000099e0: 6865 2065 7863 6c75 7369 7665 2072 6563  he exclusive rec
+000099f0: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
+00009a00: 7320 746f 2061 2063 6572 7461 696e 2072  s to a certain r
+00009a10: 6174 696f 2061 7320 746f 2074 6865 2062  atio as to the b
+00009a20: 6573 7420 6869 7473 2e20 466f 7220 696e  est hits. For in
+00009a30: 7374 616e 6365 2c20 6966 2074 6865 2067  stance, if the g
+00009a40: 656e 6520 6231 2066 726f 6d20 6765 6e6f  ene b1 from geno
+00009a50: 6d65 2042 2068 6173 2074 6865 2062 6573  me B has the bes
+00009a60: 7420 6869 7420 6765 6e65 2061 3120 6672  t hit gene a1 fr
+00009a70: 6f6d 2067 656e 6f6d 6520 4120 7769 7468  om genome A with
+00009a80: 2074 6865 2062 6974 2073 636f 7265 2061   the bit score a
+00009a90: 7320 3130 302c 2077 6869 6368 2069 7320  s 100, which is 
+00009aa0: 6120 7363 6f72 696e 6720 6d61 7472 6978  a scoring matrix
+00009ab0: 2069 6e64 6570 656e 6465 6e74 206d 6561   independent mea
+00009ac0: 7375 7265 206f 6620 7468 6520 286c 6f63  sure of the (loc
+00009ad0: 616c 2920 7369 6d69 6c61 7269 7479 206f  al) similarity o
+00009ae0: 6620 7468 6520 7477 6f20 616c 6967 6e65  f the two aligne
+00009af0: 6420 7365 7175 656e 6365 732c 2077 6974  d sequences, wit
+00009b00: 6820 6c61 7267 6572 2076 616c 7565 7320  h larger values 
+00009b10: 696e 6469 6361 7469 6e67 2068 6967 6865  indicating highe
+00009b20: 7220 7369 6d69 6c61 7269 7469 6573 2c20  r similarities, 
+00009b30: 6769 7665 6e20 7468 6520 602d 6320 302e  given the `-c 0.
+00009b40: 3960 2c20 6765 6e65 7320 6672 6f6d 2067  9`, genes from g
+00009b50: 656e 6f6d 6520 4120 7768 6963 6820 6861  enome A which ha
+00009b60: 7320 7468 6520 6269 7420 7363 6f72 6520  s the bit score 
+00009b70: 7769 7468 2067 656e 6520 6231 2068 6967  with gene b1 hig
+00009b80: 6865 7220 7468 616e 2030 2e39 7831 3030  her than 0.9x100
+00009b90: 2077 696c 6c20 616c 736f 2062 6520 7772   will also be wr
+00009ba0: 6974 7465 6e20 696e 2074 6865 2072 6573  itten in the res
+00009bb0: 756c 7420 6669 6c65 2c20 7768 6963 6820  ult file, which 
+00009bc0: 696e 2061 2073 656e 7365 2061 7265 206e  in a sense are n
+00009bd0: 6f74 2052 4248 7320 616e 796d 6f72 6520  ot RBHs anymore 
+00009be0: 6f66 2063 6f75 7273 652c 2062 7574 2074  of course, but t
+00009bf0: 6865 2068 6967 686c 7920 7369 6d69 6c61  he highly simila
+00009c00: 7220 686f 6d6f 6c6f 6775 6520 7061 6972  r homologue pair
+00009c10: 732e 2049 6620 6d6f 7265 2074 6861 6e20  s. If more than 
+00009c20: 3220 7365 7175 656e 6365 2066 696c 6573  2 sequence files
+00009c30: 2077 6572 6520 7072 6f76 6964 6564 2c20   were provided, 
+00009c40: 6576 6572 7920 7061 6972 2d77 6973 6520  every pair-wise 
+00009c50: 5242 4873 2077 6f75 6c64 2062 6520 6361  RBHs would be ca
+00009c60: 6c63 756c 6174 6564 2065 7863 6570 7420  lculated except 
+00009c70: 666f 7220 7175 6572 7969 6e67 2074 6865  for querying the
+00009c80: 2073 616d 6520 7365 7175 656e 6365 2069   same sequence i
+00009c90: 7473 656c 662e 2054 6865 206e 756d 6265  tself. The numbe
+00009ca0: 7220 6f66 2070 6172 616c 6c65 6c20 7468  r of parallel th
+00009cb0: 7265 6164 7320 746f 2062 6f6f 7374 6572  reads to booster
+00009cc0: 2074 6865 2072 756e 6e69 6e67 2073 7065   the running spe
+00009cd0: 6564 2063 616e 2062 6520 7365 7420 6279  ed can be set by
+00009ce0: 2074 6865 206f 7074 696f 6e20 602d 6e60   the option `-n`
+00009cf0: 206f 7220 602d 2d6e 7468 7265 6164 7360   or `--nthreads`
+00009d00: 2077 6869 6368 2069 7320 7375 6767 6573   which is sugges
+00009d10: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
+00009d20: 2028 4e2d 3129 4e2f 3220 7768 6572 6520   (N-1)N/2 where 
+00009d30: 4e20 6973 2074 6865 206e 756d 6265 7220  N is the number 
+00009d40: 6f66 2063 6473 2066 696c 6573 2074 6f20  of cds files to 
+00009d50: 6163 6869 6576 6520 7468 6520 6869 6768  achieve the high
+00009d60: 6573 7420 6566 6669 6369 656e 6379 2e20  est efficiency. 
+00009d70: 5468 6520 6f70 7469 6f6e 2060 2d2d 6f67  The option `--og
+00009d80: 666f 726d 6174 6020 6361 6e20 6265 2073  format` can be s
+00009d90: 6574 2074 6f20 6164 6420 696e 6465 7820  et to add index 
+00009da0: 2866 6f72 2069 6e73 7461 6e63 6520 4746  (for instance GF
+00009db0: 3030 3030 3030 3031 2920 746f 2074 6865  00000001) to the
+00009dc0: 206f 7574 7075 7420 5242 4820 6765 6e65   output RBH gene
+00009dd0: 2066 616d 696c 6965 7320 7768 6963 6820   families which 
+00009de0: 6361 6e20 6265 2066 7572 7468 6572 2075  can be further u
+00009df0: 7365 6420 696e 2074 6865 202a 4b2a 3c73  sed in the *K*<s
+00009e00: 7562 3e53 3c2f 7375 623e 2063 616c 6375  ub>S</sub> calcu
+00009e10: 6c61 7469 6f6e 2062 7920 6077 6764 206b  lation by `wgd k
+00009e20: 7364 602e 0a0a 2a2a 5468 6520 7375 6767  sd`...**The sugg
+00009e30: 6573 7465 6420 636f 6d6d 616e 6420 746f  ested command to
+00009e40: 2073 7461 7274 2077 6974 6820 6973 2061   start with is a
+00009e50: 6c73 6f20 756e 6465 7220 7468 6520 6465  lso under the de
+00009e60: 6661 756c 7420 7365 7474 696e 6720 7769  fault setting wi
+00009e70: 7468 2074 6865 2063 6f6d 6d61 6e64 2073  th the command s
+00009e80: 686f 776e 2062 656c 6f77 2a2a 0a0a 6060  hown below**..``
+00009e90: 600a 7767 6420 646d 6420 7365 7175 656e  `.wgd dmd sequen
+00009ea0: 6365 3120 7365 7175 656e 6365 320a 6060  ce1 sequence2.``
+00009eb0: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
+00009ec0: 7469 6f6e 206f 6620 6c6f 6361 6c20 4d52  tion of local MR
+00009ed0: 4248 732a 2a0a 6060 600a 7767 6420 646d  BHs**.```.wgd dm
+00009ee0: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+00009ef0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+00009f00: 2d66 2073 6571 7565 6e63 6531 202d 6520  -f sequence1 -e 
+00009f10: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+00009f20: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
+00009f30: 2028 2d6b 6629 2028 2d6b 6429 2028 2d63   (-kf) (-kd) (-c
+00009f40: 2030 2e39 2920 282d 2d74 6f5f 7374 6f70   0.9) (--to_stop
+00009f50: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+00009f60: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
+00009f70: 6e67 5f74 6d70 290a 6060 600a 0a2a 2a54  ng_tmp).```..**T
+00009f80: 6865 2064 6973 7469 6e63 7469 6f6e 2062  he distinction b
+00009f90: 6574 7765 656e 206c 6f63 616c 2061 6e64  etween local and
+00009fa0: 2067 6c6f 6261 6c20 2868 6572 6575 6e64   global (hereund
+00009fb0: 6572 2920 4d52 4248 7320 6973 2074 6861  er) MRBHs is tha
+00009fc0: 7420 6c6f 6361 6c20 4d52 4248 7320 6172  t local MRBHs ar
+00009fd0: 6520 7468 6520 7265 7375 6c74 7320 6f66  e the results of
+00009fe0: 206d 6572 6765 6420 5242 4873 206f 6e20   merged RBHs on 
+00009ff0: 6120 6a6f 696e 7420 666f 6361 6c20 7370  a joint focal sp
+0000a000: 6563 6965 732c 2066 6f72 2069 6e73 7461  ecies, for insta
+0000a010: 6e63 6520 696e 2061 2074 6872 6565 2073  nce in a three s
+0000a020: 7065 6369 6573 2073 7973 7465 6d20 2841  pecies system (A
+0000a030: 2c28 422c 4329 292c 2074 6865 206c 6f63  ,(B,C)), the loc
+0000a040: 616c 204d 5242 4873 206f 6620 4120 6f6e  al MRBHs of A on
+0000a050: 6c79 2072 6571 7569 7265 2074 6865 2063  ly require the c
+0000a060: 616c 6375 6c61 7469 6f6e 206f 6620 5242  alculation of RB
+0000a070: 4873 2062 6574 7765 656e 2041 2061 6e64  Hs between A and
+0000a080: 2043 2028 6465 6e6f 7465 6420 6173 2041   C (denoted as A
+0000a090: 4329 2061 6e64 2041 4220 616e 6420 7468  C) and AB and th
+0000a0a0: 656e 2074 6865 206d 6572 6769 6e67 206f  en the merging o
+0000a0b0: 6620 4142 2061 6e64 2041 4320 6174 2074  f AB and AC at t
+0000a0c0: 6865 2061 7869 7320 6f66 2041 2c20 7768  he axis of A, wh
+0000a0d0: 696c 6520 7468 6520 676c 6f61 626c 204d  ile the gloabl M
+0000a0e0: 5242 4873 2061 7265 2069 6e64 6570 656e  RBHs are indepen
+0000a0f0: 6465 6e74 206f 6620 666f 6361 6c20 7370  dent of focal sp
+0000a100: 6563 6965 7320 696e 2074 6861 7420 6974  ecies in that it
+0000a110: 206a 7573 7420 6361 6c6c 7320 7468 6520   just calls the 
+0000a120: 6361 6c63 756c 6174 696f 6e73 206f 6620  calculations of 
+0000a130: 616c 6c20 706f 7373 6962 6c65 2073 7065  all possible spe
+0000a140: 6369 6573 2070 6169 7220 286e 6f74 2073  cies pair (not s
+0000a150: 656c 6620 746f 2073 656c 6620 7370 6563  elf to self spec
+0000a160: 6965 7320 7061 6972 292c 2073 7563 6820  ies pair), such 
+0000a170: 7468 6174 2041 422c 2041 432c 2061 6e64  that AB, AC, and
+0000a180: 2042 4320 6172 6520 616c 6c20 746f 2062   BC are all to b
+0000a190: 6520 6361 6c63 756c 6174 6564 2061 6e64  e calculated and
+0000a1a0: 206d 6572 6765 642e 2a2a 0a0a 5477 6f20   merged.**..Two 
+0000a1b0: 7479 7065 7320 6f66 204d 5242 4873 2061  types of MRBHs a
+0000a1c0: 7320 696e 7465 7072 6574 6174 6564 2061  s intepretated a
+0000a1d0: 626f 7665 2063 616e 2062 6520 6465 6c69  bove can be deli
+0000a1e0: 6e65 6174 6564 2062 7920 6077 6764 2064  neated by `wgd d
+0000a1f0: 6d64 602c 2074 6865 206c 6f63 616c 204d  md`, the local M
+0000a200: 5242 4873 2061 6e64 2074 6865 2067 6c6f  RBHs and the glo
+0000a210: 6261 6c20 4d52 4248 732e 2054 6865 206c  bal MRBHs. The l
+0000a220: 6f63 616c 204d 5242 4873 2061 7265 2063  ocal MRBHs are c
+0000a230: 6f6e 7374 7275 6374 6564 2062 7920 6d65  onstructed by me
+0000a240: 7267 696e 6720 616c 6c20 7468 6520 7265  rging all the re
+0000a250: 6c65 7661 6e74 2052 4248 7320 6f6e 6c79  levant RBHs only
+0000a260: 2077 6974 6820 7468 6520 666f 6361 6c20   with the focal 
+0000a270: 7370 6563 6965 732c 2077 6869 6368 2069  species, which i
+0000a280: 7320 7365 7420 6279 2060 2d66 6020 6f72  s set by `-f` or
+0000a290: 2060 2d2d 666f 6375 7360 2e20 5468 6520   `--focus`. The 
+0000a2a0: 7061 7261 6d65 7465 7220 602d 6b66 6020  parameter `-kf` 
+0000a2b0: 6f72 2060 2d2d 6b65 6570 6661 7374 6160  or `--keepfasta`
+0000a2c0: 2063 616e 2062 6520 7365 7420 746f 2072   can be set to r
+0000a2d0: 6574 6169 6e20 7468 6520 7365 7175 656e  etain the sequen
+0000a2e0: 6365 2069 6e66 6f72 6d61 7469 6f6e 206f  ce information o
+0000a2f0: 6620 6561 6368 204d 5242 482e 2054 6865  f each MRBH. The
+0000a300: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
+0000a310: 206f 7220 602d 2d6b 6565 7064 7570 6c69   or `--keepdupli
+0000a320: 6361 7465 7360 2064 6574 6572 6d69 6e65  cates` determine
+0000a330: 7320 7768 6574 6865 7220 7468 6520 7361  s whether the sa
+0000a340: 6d65 2067 656e 6573 2063 616e 2061 7070  me genes can app
+0000a350: 6561 7220 696e 2064 6966 6665 7265 6e74  ear in different
+0000a360: 206c 6f63 616c 204d 5242 4873 2e20 4e6f   local MRBHs. No
+0000a370: 726d 616c 6c79 2074 6865 7265 2077 696c  rmally there wil
+0000a380: 6c20 6265 206e 6f20 6475 706c 6963 6174  l be no duplicat
+0000a390: 6573 2069 6e20 7468 6520 6c6f 6361 6c20  es in the local 
+0000a3a0: 4d52 4248 7320 6275 7420 6966 2075 7365  MRBHs but if use
+0000a3b0: 7273 2073 6574 2074 6865 2060 2d63 6020  rs set the `-c` 
+0000a3c0: 6173 2030 2e39 2028 6f72 2061 6e79 2076  as 0.9 (or any v
+0000a3d0: 616c 7565 2073 6d61 6c6c 6572 2074 6861  alue smaller tha
+0000a3e0: 6e20 3129 2c20 6974 2773 206c 696b 656c  n 1), it's likel
+0000a3f0: 7920 7468 6174 2074 6865 2073 616d 6520  y that the same 
+0000a400: 6765 6e65 2077 696c 6c20 6861 7665 2063  gene will have c
+0000a410: 6861 6e63 6520 746f 2061 7070 6561 7220  hance to appear 
+0000a420: 6d75 6c74 6970 6c65 2074 696d 6573 2069  multiple times i
+0000a430: 6e20 6469 6666 6572 656e 7420 6c6f 6361  n different loca
+0000a440: 6c20 4d52 4248 732e 2054 6861 7420 6973  l MRBHs. That is
+0000a450: 2074 6f20 7361 792c 2074 6865 2070 6172   to say, the par
+0000a460: 616d 6574 6572 2060 2d6b 6460 2069 7320  ameter `-kd` is 
+0000a470: 6d65 616e 696e 6766 756c 206f 6e6c 7920  meaningful only 
+0000a480: 7768 656e 2069 7427 7320 7365 7420 746f  when it's set to
+0000a490: 6765 7468 6572 2077 6974 6820 7468 6520  gether with the 
+0000a4a0: 7061 7261 6d65 7465 7220 602d 6360 2e20  parameter `-c`. 
+0000a4b0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
+0000a4c0: 7261 6c6c 656c 2074 6872 6561 6473 2069  rallel threads i
+0000a4d0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000a4e0: 6520 7365 7420 6173 2074 6865 206e 756d  e set as the num
+0000a4f0: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
+0000a500: 206d 696e 7573 2031 2e0a 0a2a 2a41 2073   minus 1...**A s
+0000a510: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+0000a520: 6720 7275 6e20 6973 2075 6e64 6572 2074  g run is under t
+0000a530: 6865 2064 6566 6175 6c74 2070 6172 616d  he default param
+0000a540: 6574 6572 2077 6974 6820 7468 6520 636f  eter with the co
+0000a550: 6d6d 616e 6420 7368 6f77 6e20 6173 2062  mmand shown as b
+0000a560: 656c 6f77 2e2a 2a0a 0a60 6060 0a77 6764  elow.**..```.wgd
+0000a570: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000a580: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000a590: 6533 202d 6620 7365 7175 656e 6365 310a  e3 -f sequence1.
+0000a5a0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
+0000a5b0: 6561 7469 6f6e 206f 6620 676c 6f62 616c  eation of global
+0000a5c0: 204d 5242 4873 2a2a 0a60 6060 0a77 6764   MRBHs**.```.wgd
+0000a5d0: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000a5e0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000a5f0: 6533 202d 676d 202d 6520 3165 2d31 3020  e3 -gm -e 1e-10 
+0000a600: 2d62 7320 3130 3020 2d6e 7020 3520 282d  -bs 100 -np 5 (-
+0000a610: 6e6e 2920 282d 6e20 3429 2028 2d6b 6629  nn) (-n 4) (-kf)
+0000a620: 2028 2d6b 6429 2028 2d63 2030 2e39 2920   (-kd) (-c 0.9) 
+0000a630: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+0000a640: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
+0000a650: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+0000a660: 290a 6060 600a 0a54 6865 2067 6c6f 6261  ).```..The globa
+0000a670: 6c20 4d52 4248 7320 6973 2063 6f6e 7374  l MRBHs is const
+0000a680: 7275 6374 6564 2062 7920 6578 6861 7573  ructed by exhaus
+0000a690: 7469 7665 6c79 206d 6572 6769 6e67 2061  tively merging a
+0000a6a0: 6c6c 2074 6865 2070 6f73 7369 626c 6520  ll the possible 
+0000a6b0: 7061 6972 2d77 6973 6520 5242 4873 2065  pair-wise RBHs e
+0000a6c0: 7863 6570 7420 666f 7220 7175 6572 7969  xcept for queryi
+0000a6d0: 6e67 2074 6865 2073 6571 7565 6e63 6520  ng the sequence 
+0000a6e0: 6974 7365 6c66 2c20 7768 6963 6820 6361  itself, which ca
+0000a6f0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
+0000a700: 7920 6164 6420 7468 6520 666c 6167 2060  y add the flag `
+0000a710: 2d67 6d60 206f 7220 602d 2d67 6c6f 6261  -gm` or `--globa
+0000a720: 6c6d 7262 6860 2e20 5468 6520 7265 7374  lmrbh`. The rest
+0000a730: 206f 6620 7265 6c65 7661 6e74 2070 6172   of relevant par
+0000a740: 616d 6574 6572 7320 7374 6179 7320 7468  ameters stays th
+0000a750: 6520 7361 6d65 2061 7320 7468 6520 6c6f  e same as the lo
+0000a760: 6361 6c20 4d52 4248 732e 2054 6865 206e  cal MRBHs. The n
+0000a770: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
+0000a780: 6c20 7468 7265 6164 7320 6973 2073 7567  l threads is sug
+0000a790: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
+0000a7a0: 2061 7320 284e 2d31 294e 2f32 2074 6f6f   as (N-1)N/2 too
+0000a7b0: 2077 6865 7265 204e 2069 7320 7468 6520   where N is the 
+0000a7c0: 6e75 6d62 6572 206f 6620 6364 7320 6669  number of cds fi
+0000a7d0: 6c65 7320 746f 2061 6368 6965 7665 2074  les to achieve t
+0000a7e0: 6865 2068 6967 6865 7374 2065 6666 6963  he highest effic
+0000a7f0: 6965 6e63 792e 0a0a 2a2a 4120 7375 6767  iency...**A sugg
+0000a800: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000a810: 756e 2069 7320 756e 6465 7220 7468 6520  un is under the 
+0000a820: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
+0000a830: 7220 7769 7468 2074 6865 2063 6f6d 6d61  r with the comma
+0000a840: 6e64 2073 686f 776e 2061 7320 6265 6c6f  nd shown as belo
+0000a850: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
+0000a860: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+0000a870: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000a880: 2d67 6d0a 6060 600a 0a2a 2a54 6865 2064  -gm.```..**The d
+0000a890: 656c 696e 6561 7469 6f6e 206f 6620 6f72  elineation of or
+0000a8a0: 7468 6f67 726f 7570 732a 2a0a 6060 600a  thogroups**.```.
+0000a8b0: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+0000a8c0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000a8d0: 656e 6365 3320 2d6f 6920 2d6f 6f20 2d65  ence3 -oi -oo -e
+0000a8e0: 2031 652d 3130 202d 6273 2031 3030 202d   1e-10 -bs 100 -
+0000a8f0: 6e70 2035 2028 2d6e 6e29 2028 2d63 6329  np 5 (-nn) (-cc)
+0000a900: 2028 2d74 6529 2028 2d6d 6320 302e 3829   (-te) (-mc 0.8)
+0000a910: 2028 2d67 6e29 2028 2d74 7265 6520 2766   (-gn) (-tree 'f
+0000a920: 6173 7474 7265 6527 2920 282d 7473 2027  asttree') (-ts '
+0000a930: 2d66 6173 7465 7374 2729 2028 2d6e 2034  -fastest') (-n 4
+0000a940: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
+0000a950: 2d63 6473 2920 282d 6f20 7767 645f 646d  -cds) (-o wgd_dm
+0000a960: 6429 2028 2d74 2077 6f72 6b69 6e67 5f74  d) (-t working_t
+0000a970: 6d70 290a 6060 600a 0a49 6e20 6077 6764  mp).```..In `wgd
+0000a980: 2076 3260 2c20 7765 2061 6c73 6f20 696d   v2`, we also im
+0000a990: 706c 656d 656e 7465 6420 616e 2061 6c67  plemented an alg
+0000a9a0: 6f72 6974 686d 206f 6620 6465 6c69 6e65  orithm of deline
+0000a9b0: 6174 696e 6720 6f72 7468 6f67 726f 7570  ating orthogroup
+0000a9c0: 732c 2077 6869 6368 2063 616e 2062 6520  s, which can be 
+0000a9d0: 696e 6974 6961 7465 6420 7769 7468 2074  initiated with t
+0000a9e0: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
+0000a9f0: 6960 206f 7220 602d 2d6f 7274 686f 696e  i` or `--orthoin
+0000aa00: 6665 7260 2e20 5477 6f20 7761 7973 206f  fer`. Two ways o
+0000aa10: 6620 6465 6c69 6e65 6174 696f 6e20 6361  f delineation ca
+0000aa20: 6e20 6265 2063 686f 7365 6e2c 2074 6865  n be chosen, the
+0000aa30: 2063 6f6e 6361 7465 6e61 7469 6f6e 2077   concatenation w
+0000aa40: 6179 2028 7365 7420 6279 2074 6865 2070  ay (set by the p
+0000aa50: 6172 616d 6574 6572 2060 2d63 6360 206f  arameter `-cc` o
+0000aa60: 7220 602d 2d63 6f6e 6361 7460 2920 6f72  r `--concat`) or
+0000aa70: 2074 6865 206e 6f6e 2d63 6f6e 6361 7465   the non-concate
+0000aa80: 6e61 7469 6f6e 2028 6465 6661 756c 7429  nation (default)
+0000aa90: 2077 6179 2e20 496e 2062 7269 6566 2c20   way. In brief, 
+0000aaa0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+0000aab0: 6e20 7761 7920 6f66 2064 656c 696e 6561  n way of delinea
+0000aac0: 7469 6e67 206f 7274 686f 6772 6f75 7073  ting orthogroups
+0000aad0: 2073 7461 7274 7320 7769 7468 2063 6f6e   starts with con
+0000aae0: 6361 7465 6e61 7469 6e67 2061 6c6c 2074  catenating all t
+0000aaf0: 6865 2073 6571 7565 6e63 6573 2069 6e74  he sequences int
+0000ab00: 6f20 6120 7369 6e67 6c65 2073 6571 7565  o a single seque
+0000ab10: 6e63 6520 6669 6c65 2061 6e64 2074 6865  nce file and the
+0000ab20: 6e20 696e 6665 7272 696e 6720 7468 6520  n inferring the 
+0000ab30: 7768 6f6c 6520 7061 7261 6e6f 6d65 206f  whole paranome o
+0000ab40: 6620 7468 6973 2073 696e 676c 6520 7365  f this single se
+0000ab50: 7175 656e 6365 2066 696c 6520 7769 7468  quence file with
+0000ab60: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
+0000ab70: 7265 7375 6c74 7320 6d61 7070 6564 2062  results mapped b
+0000ab80: 6163 6b20 746f 2074 6865 2062 656c 6f6e  ack to the belon
+0000ab90: 6769 6e67 2073 7065 6369 6573 2e20 5768  ging species. Wh
+0000aba0: 696c 6520 7468 6520 6e6f 6e2d 636f 6e63  ile the non-conc
+0000abb0: 6174 656e 6174 696f 6e20 7761 7920 7374  atenation way st
+0000abc0: 6172 7473 2077 6974 6820 7265 7370 6563  arts with respec
+0000abd0: 7469 7665 2070 6169 722d 7769 7365 2064  tive pair-wise d
+0000abe0: 6961 6d6f 6e64 2073 6561 7263 6820 2869  iamond search (i
+0000abf0: 6e63 6c75 6469 6e67 2071 7565 7279 696e  ncluding queryin
+0000ac00: 6720 7468 6520 7361 6d65 2073 6571 7565  g the same seque
+0000ac10: 6e63 6520 6974 7365 6c66 2920 616e 6420  nce itself) and 
+0000ac20: 7468 656e 2061 6c6c 2074 6865 2073 6571  then all the seq
+0000ac30: 7565 6e63 6520 7369 6d69 6c61 7269 7479  uence similarity
+0000ac40: 2074 6162 6c65 7320 7769 6c6c 2062 6520   tables will be 
+0000ac50: 636f 6e63 6174 656e 6174 6564 2061 6e64  concatenated and
+0000ac60: 2063 6c75 7374 6572 6564 2069 6e74 6f20   clustered into 
+0000ac70: 6f72 7468 6f67 726f 7570 732e 2053 6f6d  orthogroups. Som
+0000ac80: 6520 6f74 6865 7220 706f 7373 6962 6c79  e other possibly
+0000ac90: 2075 7365 6675 6c20 706f 7374 2d63 6c75   useful post-clu
+0000aca0: 7374 6572 696e 6720 6675 6e63 7469 6f6e  stering function
+0000acb0: 7320 6361 6e20 6265 2069 6e69 7469 6174  s can be initiat
+0000acc0: 6564 2c20 696e 636c 7564 696e 6720 7468  ed, including th
+0000acd0: 6520 7061 7261 6d65 7465 7220 602d 7465  e parameter `-te
+0000ace0: 6020 6f72 2060 2d2d 7465 7374 736f 6760  ` or `--testsog`
+0000acf0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
+0000ad00: 6574 2074 6f20 7374 6172 7420 7468 6520  et to start the 
+0000ad10: 756e 6269 6173 6564 2074 6573 7420 6f66  unbiased test of
+0000ad20: 2073 696e 676c 652d 636f 7079 2067 656e   single-copy gen
+0000ad30: 6520 6661 6d69 6c69 6573 2028 6e6f 7465  e families (note
+0000ad40: 2074 6861 7420 7468 6973 2066 756e 6374   that this funct
+0000ad50: 696f 6e20 6e65 6564 7320 6068 6d6d 6572  ion needs `hmmer
+0000ad60: 6020 2876 332e 3162 3229 2074 6f20 6265  ` (v3.1b2) to be
+0000ad70: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
+0000ad80: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
+0000ad90: 7468 292c 2074 6865 2070 6172 616d 6574  th), the paramet
+0000ada0: 6572 2060 2d6d 6360 206f 7220 602d 2d6d  er `-mc` or `--m
+0000adb0: 736f 6763 7574 602c 2072 616e 6769 6e67  sogcut`, ranging
+0000adc0: 2062 6574 7765 656e 2030 2074 6f20 312c   between 0 to 1,
+0000add0: 2077 6869 6368 2063 616e 2062 6520 7365   which can be se
+0000ade0: 7420 746f 2073 6561 7263 6820 7468 6520  t to search the 
+0000adf0: 736f 2d63 616c 6c65 6420 6d6f 7374 6c79  so-called mostly
+0000ae00: 2073 696e 676c 652d 636f 7079 2066 616d   single-copy fam
+0000ae10: 696c 7920 7768 6963 6820 6861 7320 6869  ily which has hi
+0000ae20: 6768 6572 2074 6861 6e20 6365 7274 6169  gher than certai
+0000ae30: 6e20 6375 742d 6f66 6620 7065 7263 656e  n cut-off percen
+0000ae40: 7461 6765 206f 6620 7370 6563 6965 7320  tage of species 
+0000ae50: 636f 7665 7261 6765 2c20 7468 6520 7061  coverage, the pa
+0000ae60: 7261 6d65 7465 7220 602d 676e 6020 6f72  rameter `-gn` or
+0000ae70: 2060 2d2d 6765 746e 736f 6760 2c20 7768   `--getnsog`, wh
+0000ae80: 6963 6820 6361 6e20 6265 2073 6574 2074  ich can be set t
+0000ae90: 6f20 7365 6172 6368 2066 6f72 206e 6573  o search for nes
+0000aea0: 7465 6420 7369 6e67 6c65 2d63 6f70 7920  ted single-copy 
+0000aeb0: 6765 6e65 2066 616d 696c 6965 7320 284e  gene families (N
+0000aec0: 534f 4773 2920 7768 6963 6820 6973 206f  SOGs) which is o
+0000aed0: 7269 6769 6e61 6c6c 7920 6d75 6c74 6979  riginally multiy
+0000aee0: 2d63 6f70 7920 6275 7420 6861 7320 6120  -copy but has a 
+0000aef0: 286d 6f73 746c 7929 2073 696e 676c 652d  (mostly) single-
+0000af00: 636f 7079 2062 7261 6e63 6820 2877 6869  copy branch (whi
+0000af10: 6368 2072 6571 7569 7265 7320 7468 6520  ch requires the 
+0000af20: 6368 6f73 656e 2074 7265 652d 696e 6665  chosen tree-infe
+0000af30: 7265 6e63 6520 7072 6f67 7261 6d20 7365  rence program se
+0000af40: 7420 6279 2060 2d74 7265 6560 206f 7220  t by `-tree` or 
+0000af50: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
+0000af60: 746f 2062 6520 7072 652d 696e 7374 616c  to be pre-instal
+0000af70: 6c65 6420 696e 2074 6865 2065 6e76 6972  led in the envir
+0000af80: 6f6e 6d65 6e74 2070 6174 6820 7769 7468  onment path with
+0000af90: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
+0000afa0: 7365 7474 696e 6720 666f 7220 6765 6e65  setting for gene
+0000afb0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000afc0: 636f 6e74 726f 6c6c 6564 2062 7920 602d  controlled by `-
+0000afd0: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+0000afe0: 7460 292e 2054 6865 2070 726f 6772 616d  t`). The program
+0000aff0: 2060 7767 6420 646d 6460 2077 6f75 6c64   `wgd dmd` would
+0000b000: 2073 7469 6c6c 2063 6f6e 6475 6374 2074   still conduct t
+0000b010: 6865 2052 4248 7320 6361 6c63 756c 6174  he RBHs calculat
+0000b020: 696f 6e20 756e 6c65 7373 2074 6865 2070  ion unless the p
+0000b030: 6172 616d 6574 6572 2060 2d6f 6f60 206f  arameter `-oo` o
+0000b040: 7220 602d 2d6f 6e6c 796f 7274 686f 6020  r `--onlyortho` 
+0000b050: 7761 7320 7365 742e 2049 6620 6f6e 6520  was set. If one 
+0000b060: 6f6e 6c79 2077 616e 7473 2074 6f20 696e  only wants to in
+0000b070: 6665 7220 7468 6520 6f72 7468 6f67 726f  fer the orthogro
+0000b080: 7570 732c 2069 7427 7320 7375 6767 6573  ups, it's sugges
+0000b090: 7465 6420 746f 2061 6464 2074 6865 2066  ted to add the f
+0000b0a0: 6c61 6720 602d 6f6f 6020 746f 206a 7573  lag `-oo` to jus
+0000b0b0: 7420 696d 706c 656d 656e 7420 7468 6520  t implement the 
+0000b0c0: 6f72 7468 6f67 726f 7570 7320 6465 6c69  orthogroups deli
+0000b0d0: 6e65 6174 696f 6e20 616e 616c 7973 6973  neation analysis
+0000b0e0: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
+0000b0f0: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
+0000b100: 2069 7320 7375 6767 6573 7465 6420 746f   is suggested to
+0000b110: 2062 6520 7365 7420 6173 2028 4e2b 3129   be set as (N+1)
+0000b120: 4e2f 3220 7768 6572 6520 4e20 6973 2074  N/2 where N is t
+0000b130: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
+0000b140: 2066 696c 6573 2074 6f20 6163 6869 6576   files to achiev
+0000b150: 6520 7468 6520 6869 6768 6573 7420 6566  e the highest ef
+0000b160: 6669 6369 656e 6379 2073 696e 6365 2074  ficiency since t
+0000b170: 6865 2073 656c 662d 636f 6d70 6172 6973  he self-comparis
+0000b180: 6f6e 2069 7320 616c 736f 2069 6e63 6c75  on is also inclu
+0000b190: 6465 642e 0a0a 2a2a 5468 6520 6465 6661  ded...**The defa
+0000b1a0: 756c 7420 7365 7474 696e 6720 6f66 2070  ult setting of p
+0000b1b0: 6172 616d 6574 6572 7320 6973 2061 2072  arameters is a r
+0000b1c0: 6561 736f 6e61 626c 6520 7374 6172 7469  easonable starti
+0000b1d0: 6e67 2070 6f69 6e74 2077 6974 6820 7468  ng point with th
+0000b1e0: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+0000b1f0: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2064  ow.**..```.wgd d
+0000b200: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000b210: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000b220: 202d 6f69 202d 6f6f 0a60 6060 0a0a 2a2a   -oi -oo.```..**
+0000b230: 5468 6520 636f 6c6c 696e 6561 7220 636f  The collinear co
+0000b240: 616c 6573 6365 6e63 6520 696e 6665 7265  alescence infere
+0000b250: 6e63 6520 6f66 2070 6879 6c6f 6765 6e79  nce of phylogeny
+0000b260: 2a2a 0a60 6060 0a77 6764 2064 6d64 2073  **.```.wgd dmd s
+0000b270: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+0000b280: 6532 2073 6571 7565 6e63 6533 202d 6170  e2 sequence3 -ap
+0000b290: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
+0000b2a0: 7461 202d 6c65 206c 6564 6174 6120 2d67  ta -le ledata -g
+0000b2b0: 7420 6774 6461 7461 202d 636f 6320 282d  t gtdata -coc (-
+0000b2c0: 7472 6565 2027 6661 7374 7472 6565 2729  tree 'fasttree')
+0000b2d0: 2028 2d74 7320 272d 6661 7374 6573 7427   (-ts '-fastest'
+0000b2e0: 2920 282d 6e20 3429 2028 2d2d 746f 5f73  ) (-n 4) (--to_s
+0000b2f0: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
+0000b300: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
+0000b310: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
+0000b320: 4120 6e6f 7665 6c20 7068 796c 6f67 656e  A novel phylogen
+0000b330: 6574 6963 2069 6e66 6572 656e 6365 206d  etic inference m
+0000b340: 6574 686f 6420 6e61 6d65 6420 2263 6f6c  ethod named "col
+0000b350: 6c69 6e65 6172 2063 6f61 6c65 7363 656e  linear coalescen
+0000b360: 6365 2069 6e66 6572 656e 6365 2220 6973  ce inference" is
+0000b370: 2061 6c73 6f20 696d 706c 656d 656e 7465   also implemente
+0000b380: 6420 696e 2060 7767 6420 7632 602e 2046  d in `wgd v2`. F
+0000b390: 6f72 2074 6869 7320 616e 616c 7973 6973  or this analysis
+0000b3a0: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000b3b0: 7072 6f76 6964 6520 7468 6520 616e 6368  provide the anch
+0000b3c0: 6f72 2070 6f69 6e74 7320 6669 6c65 2062  or points file b
+0000b3d0: 7920 602d 6170 6020 6f72 2060 2d2d 616e  y `-ap` or `--an
+0000b3e0: 6368 6f72 706f 696e 7473 602c 2074 6865  chorpoints`, the
+0000b3f0: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
+0000b400: 6e74 7320 6669 6c65 2062 7920 602d 736d  nts file by `-sm
+0000b410: 6020 6f72 2060 2d2d 7365 676d 656e 7473  ` or `--segments
+0000b420: 602c 2074 6865 206c 6973 7473 6567 6d65  `, the listsegme
+0000b430: 6e74 7320 6669 6c65 2062 7920 602d 6c65  nts file by `-le
+0000b440: 6020 6f72 2060 2d2d 6c69 7374 656c 656d  ` or `--listelem
+0000b450: 656e 7473 602c 2061 6e64 2074 6865 2067  ents`, and the g
+0000b460: 656e 6520 7461 626c 6520 6669 6c65 2062  ene table file b
+0000b470: 7920 602d 6774 6020 6f72 2060 2d2d 6765  y `-gt` or `--ge
+0000b480: 6e65 7461 626c 6560 2c20 616c 6c20 6f66  netable`, all of
+0000b490: 2077 6869 6368 2063 616e 2062 6520 7072   which can be pr
+0000b4a0: 6f64 7563 6564 2069 6e20 7468 6520 7072  oduced in the pr
+0000b4b0: 6f67 7261 6d20 6077 6764 2073 796e 602e  ogram `wgd syn`.
+0000b4c0: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
+0000b4d0: 2d63 6f63 6020 6f72 2060 2d2d 636f 6c6c  -coc` or `--coll
+0000b4e0: 696e 6561 7263 6f61 6c65 7363 656e 6365  inearcoalescence
+0000b4f0: 6020 6e65 6564 7320 746f 2062 6520 7365  ` needs to be se
+0000b500: 7420 746f 2073 7461 7274 2074 6869 7320  t to start this 
+0000b510: 616e 616c 7973 6973 2e20 5468 6520 7472  analysis. The tr
+0000b520: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+0000b530: 6772 616d 2061 6e64 2074 6865 2061 7373  gram and the ass
+0000b540: 6f63 6961 7465 6420 7061 7261 6d65 7465  ociated paramete
+0000b550: 7273 2063 616e 2062 6520 7365 7420 6a75  rs can be set ju
+0000b560: 7374 2061 7320 6162 6f76 6520 6279 2060  st as above by `
+0000b570: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
+0000b580: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
+0000b590: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+0000b5a0: 7460 2e20 506c 6561 7365 2061 6c73 6f20  t`. Please also 
+0000b5b0: 6d61 6b65 2073 7572 6520 7468 6520 6368  make sure the ch
+0000b5c0: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+0000b5d0: 6e63 6520 7072 6f67 7261 6d20 6973 2069  nce program is i
+0000b5e0: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+0000b5f0: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+0000b600: 2e20 5468 6520 7072 6f67 7261 6d20 6061  . The program `a
+0000b610: 7374 7261 6c2d 7072 6f60 2069 7320 7265  stral-pro` is re
+0000b620: 7175 6972 6564 2074 6f20 6265 2069 6e73  quired to be ins
+0000b630: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+0000b640: 7669 726f 6e6d 656e 7420 7061 7468 2074  vironment path t
+0000b650: 6f6f 2e20 4e6f 7465 2074 6861 7420 7468  oo. Note that th
+0000b660: 6572 6520 7368 6f75 6c64 2062 6520 6e6f  ere should be no
+0000b670: 2064 7570 6c69 6361 7465 6420 6765 6e65   duplicated gene
+0000b680: 2049 4473 2069 6e20 7468 6520 7365 7175   IDs in the sequ
+0000b690: 656e 6365 2066 696c 652e 2054 6865 2070  ence file. The p
+0000b6a0: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
+0000b6b0: 6865 7265 2061 7265 2074 6f20 6163 6365  here are to acce
+0000b6c0: 6c65 7261 7465 2074 6865 2073 6571 7565  lerate the seque
+0000b6d0: 6e63 6520 616c 6967 6e6d 656e 7420 616e  nce alignment an
+0000b6e0: 6420 6765 6e65 2074 7265 6520 696e 6665  d gene tree infe
+0000b6f0: 7265 6e63 6520 666f 7220 6561 6368 2067  rence for each g
+0000b700: 656e 6520 6661 6d69 6c79 2061 6e64 2074  ene family and t
+0000b710: 6875 7320 7375 6767 6573 7465 6420 746f  hus suggested to
+0000b720: 2062 6520 7365 7420 6173 206d 7563 6820   be set as much 
+0000b730: 6173 2074 6865 206e 756d 6265 7220 6f66  as the number of
+0000b740: 2067 656e 6520 6661 6d69 6c69 6573 2e0a   gene families..
+0000b750: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+0000b760: 7461 7274 696e 6720 7275 6e20 6f66 2074  tarting run of t
+0000b770: 6869 7320 616e 616c 7973 6973 2069 7320  his analysis is 
+0000b780: 7769 7468 2074 6865 2073 696d 706c 6520  with the simple 
+0000b790: 636f 6d6d 616e 6420 6265 6c6f 772e 2a2a  command below.**
+0000b7a0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
+0000b7b0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000b7c0: 3220 7365 7175 656e 6365 3320 2d61 7020  2 sequence3 -ap 
+0000b7d0: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
+0000b7e0: 6120 2d6c 6520 6c65 6461 7461 202d 6774  a -le ledata -gt
+0000b7f0: 2067 7464 6174 6120 2d63 6f63 0a60 6060   gtdata -coc.```
+0000b800: 0a0a 2323 2320 7767 6420 666f 6375 730a  ..### wgd focus.
+0000b810: 0a2a 2a54 6865 2063 6f6e 6361 7465 6e61  .**The concatena
+0000b820: 7469 6f6e 2d62 6173 6564 2f63 6f61 6c65  tion-based/coale
+0000b830: 7363 656e 6365 2d62 6173 6564 2070 6879  scence-based phy
+0000b840: 6c6f 6765 6e65 7469 6320 696e 6665 7265  logenetic infere
+0000b850: 6e63 652a 2a0a 6060 600a 7767 6420 666f  nce**.```.wgd fo
+0000b860: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000b870: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000b880: 2073 6571 7565 6e63 6533 2028 2d2d 636f   sequence3 (--co
+0000b890: 6e63 6174 656e 6174 696f 6e29 2028 2d2d  ncatenation) (--
+0000b8a0: 636f 616c 6573 6365 6e63 6529 2028 2d74  coalescence) (-t
+0000b8b0: 7265 6520 2766 6173 7474 7265 6527 2920  ree 'fasttree') 
+0000b8c0: 282d 7473 2027 2d66 6173 7465 7374 2729  (-ts '-fastest')
+0000b8d0: 2028 2d6e 2034 2920 282d 2d74 6f5f 7374   (-n 4) (--to_st
+0000b8e0: 6f70 2920 282d 2d63 6473 2920 282d 6f20  op) (--cds) (-o 
+0000b8f0: 7767 645f 666f 6375 7329 2028 2d74 2077  wgd_focus) (-t w
+0000b900: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
+0000b910: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
+0000b920: 6420 666f 6375 7360 2069 6d70 6c65 6d65  d focus` impleme
+0000b930: 6e74 6564 2074 776f 2062 6173 6963 2070  nted two basic p
+0000b940: 6879 6c6f 6765 6e65 7469 6320 696e 6665  hylogenetic infe
+0000b950: 7265 6e63 6520 6d65 7468 6f64 732c 2069  rence methods, i
+0000b960: 2e65 2e2c 2063 6f6e 6361 7465 6e61 7469  .e., concatenati
+0000b970: 6f6e 2d62 6173 6564 2061 6e64 2063 6f61  on-based and coa
+0000b980: 6c65 7363 656e 6365 2d62 6173 6564 206d  lescence-based m
+0000b990: 6574 686f 6473 2e20 546f 2069 6e69 7469  ethods. To initi
+0000b9a0: 6174 6520 7468 6573 6520 616e 616c 7973  ate these analys
+0000b9b0: 6973 2c20 7573 6572 7320 6e65 6564 2074  is, users need t
+0000b9c0: 6f20 7365 7420 7468 6520 666c 6167 206f  o set the flag o
+0000b9d0: 7074 696f 6e20 602d 2d63 6f6e 6361 7465  ption `--concate
+0000b9e0: 6e61 7469 6f6e 6020 6f72 2060 2d2d 636f  nation` or `--co
+0000b9f0: 616c 6573 6365 6e63 6560 2e20 5468 6520  alescence`. The 
+0000ba00: 636f 6e63 6174 656e 6174 696f 6e2d 6261  concatenation-ba
+0000ba10: 7365 6420 6d65 7468 6f64 2069 6e63 6c75  sed method inclu
+0000ba20: 6465 7320 6120 6665 7720 6d61 6a6f 7220  des a few major 
+0000ba30: 7374 6570 732c 2069 2e65 2e2c 2074 6865  steps, i.e., the
+0000ba40: 206d 756c 7469 706c 6520 7365 7175 656e   multiple sequen
+0000ba50: 6365 2061 6c69 676e 6d65 6e74 2028 4d53  ce alignment (MS
+0000ba60: 4129 206f 6620 6561 6368 2067 656e 6520  A) of each gene 
+0000ba70: 6661 6d69 6c79 2c20 7468 6520 636f 6e63  family, the conc
+0000ba80: 6174 656e 6174 696f 6e20 6f66 2061 6c6c  atenation of all
+0000ba90: 2067 656e 6520 6661 6d69 6c69 6573 2061   gene families a
+0000baa0: 6e64 2074 6865 6e20 7468 6520 6765 6e65  nd then the gene
+0000bab0: 2074 7265 6520 2861 6c73 6f20 7370 6563   tree (also spec
+0000bac0: 6965 7320 7472 6565 2069 6e20 7468 6973  ies tree in this
+0000bad0: 2063 6173 6529 2069 6e66 6572 656e 6365   case) inference
+0000bae0: 2e20 5468 6520 636f 616c 6573 6365 6e63  . The coalescenc
+0000baf0: 652d 6261 7365 6420 6d65 7468 6f64 2077  e-based method w
+0000bb00: 696c 6c20 696e 7374 6561 6420 7065 7266  ill instead perf
+0000bb10: 6f72 6d20 7468 6520 4d53 4120 6f66 2065  orm the MSA of e
+0000bb20: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
+0000bb30: 616e 6420 7468 6520 6765 6e65 2074 7265  and the gene tre
+0000bb40: 6520 696e 6665 7265 6e63 6520 6261 7365  e inference base
+0000bb50: 6420 6f6e 2065 6163 6820 4d53 412c 2061  d on each MSA, a
+0000bb60: 6e64 2074 6865 6e20 696e 6665 7220 7468  nd then infer th
+0000bb70: 6520 7370 6563 6965 7320 7472 6565 2062  e species tree b
+0000bb80: 6173 6564 206f 6e20 7468 6573 6520 696e  ased on these in
+0000bb90: 6469 7669 6475 616c 2067 656e 6520 7472  dividual gene tr
+0000bba0: 6565 732e 2054 6865 2074 7265 652d 696e  ees. The tree-in
+0000bbb0: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+0000bbc0: 616e 6420 7468 6520 6173 736f 6369 6174  and the associat
+0000bbd0: 6564 2070 6172 616d 6574 6572 7320 6361  ed parameters ca
+0000bbe0: 6e20 6265 2073 6574 206a 7573 7420 6173  n be set just as
+0000bbf0: 2061 626f 7665 2062 7920 602d 7472 6565   above by `-tree
+0000bc00: 6020 6f72 2060 2d2d 7472 6565 5f6d 6574  ` or `--tree_met
+0000bc10: 686f 6460 2061 6e64 2060 2d74 7360 206f  hod` and `-ts` o
+0000bc20: 7220 602d 2d74 7265 6573 6574 602e 2050  r `--treeset`. P
+0000bc30: 6c65 6173 6520 616c 736f 206d 616b 6520  lease also make 
+0000bc40: 7375 7265 2074 6865 2063 686f 7365 6e20  sure the chosen 
+0000bc50: 7472 6565 2d69 6e66 6572 656e 6365 2070  tree-inference p
+0000bc60: 726f 6772 616d 2069 7320 696e 7374 616c  rogram is instal
+0000bc70: 6c65 6420 696e 2074 6865 2065 6e76 6972  led in the envir
+0000bc80: 6f6e 6d65 6e74 2070 6174 682e 2054 6865  onment path. The
+0000bc90: 2070 726f 6772 616d 2060 6173 7472 616c   program `astral
+0000bca0: 2d70 726f 6020 6973 2072 6571 7569 7265  -pro` is require
+0000bcb0: 6420 746f 2062 6520 696e 7374 616c 6c65  d to be installe
+0000bcc0: 6420 696e 2074 6865 2065 6e76 6972 6f6e  d in the environ
+0000bcd0: 6d65 6e74 2070 6174 6820 6966 2074 6865  ment path if the
+0000bce0: 2063 6f61 6c65 7363 656e 6365 2d62 6173   coalescence-bas
+0000bcf0: 6564 206d 6574 686f 6420 6973 2063 686f  ed method is cho
+0000bd00: 7365 6e2e 2054 6865 2070 6172 616c 6c65  sen. The paralle
+0000bd10: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
+0000bd20: 7265 2074 6f20 6163 6365 6c65 7261 7465  re to accelerate
+0000bd30: 2074 6865 2073 6571 7565 6e63 6520 616c   the sequence al
+0000bd40: 6967 6e6d 656e 7420 616e 6420 6765 6e65  ignment and gene
+0000bd50: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000bd60: 666f 7220 6561 6368 2067 656e 6520 6661  for each gene fa
+0000bd70: 6d69 6c79 2074 6f6f 2061 6e64 2074 6875  mily too and thu
+0000bd80: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000bd90: 6520 7365 7420 6173 206d 7563 6820 6173  e set as much as
+0000bda0: 2074 6865 206e 756d 6265 7220 6f66 2067   the number of g
+0000bdb0: 656e 6520 6661 6d69 6c69 6573 2e0a 0a2a  ene families...*
+0000bdc0: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+0000bdd0: 7274 696e 6720 7275 6e20 6f66 2074 6869  rting run of thi
+0000bde0: 7320 616e 616c 7973 6973 2069 7320 7769  s analysis is wi
+0000bdf0: 7468 2074 6865 2073 696d 706c 6520 636f  th the simple co
+0000be00: 6d6d 616e 6420 6265 6c6f 772e 2a2a 0a0a  mmand below.**..
+0000be10: 6060 600a 7767 6420 666f 6375 7320 6661  ```.wgd focus fa
+0000be20: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
+0000be30: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+0000be40: 6e63 6533 202d 2d63 6f6e 6361 7465 6e61  nce3 --concatena
+0000be50: 7469 6f6e 0a77 6764 2066 6f63 7573 2066  tion.wgd focus f
+0000be60: 616d 696c 6965 7320 7365 7175 656e 6365  amilies sequence
+0000be70: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000be80: 656e 6365 3320 2d2d 636f 616c 6573 6365  ence3 --coalesce
+0000be90: 6e63 650a 6060 600a 0a2a 2a54 6865 2066  nce.```..**The f
+0000bea0: 756e 6374 696f 6e61 6c20 616e 6e6f 7461  unctional annota
+0000beb0: 7469 6f6e 206f 6620 6765 6e65 2066 616d  tion of gene fam
+0000bec0: 696c 6965 732a 2a0a 6060 600a 7767 6420  ilies**.```.wgd 
+0000bed0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
+0000bee0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+0000bef0: 6532 2073 6571 7565 6e63 6533 202d 2d61  e2 sequence3 --a
+0000bf00: 6e6e 6f74 6174 696f 6e20 6567 676e 6f67  nnotation eggnog
+0000bf10: 202d 6564 2065 6464 6174 6120 2d2d 646d   -ed eddata --dm
+0000bf20: 6e62 2064 6264 6174 610a 6060 600a 0a54  nb dbdata.```..T
+0000bf30: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+0000bf40: 666f 6375 7360 2061 6c73 6f20 6164 6465  focus` also adde
+0000bf50: 6420 736f 6d65 2077 7261 7070 696e 6720  d some wrapping 
+0000bf60: 6675 6e63 7469 6f6e 7320 666f 7220 6675  functions for fu
+0000bf70: 6e63 7469 6f6e 616c 2061 6e6e 6f74 6174  nctional annotat
+0000bf80: 696f 6e20 6f66 2067 656e 6520 6661 6d69  ion of gene fami
+0000bf90: 6c69 6573 206f 6e20 7468 6520 686f 6f64  lies on the hood
+0000bfa0: 206f 6620 6461 7461 6261 7365 7320 616e   of databases an
+0000bfb0: 6420 736f 6674 7761 7265 7320 666f 7220  d softwares for 
+0000bfc0: 696e 7374 616e 6365 2060 4567 674e 4f47  instance `EggNOG
+0000bfd0: 6020 616e 6420 6045 6767 4e4f 472d 6d61  ` and `EggNOG-ma
+0000bfe0: 7070 6572 602e 2046 6f72 2074 6865 2061  pper`. For the a
+0000bff0: 6e6e 6f74 6174 696f 6e20 7573 696e 6720  nnotation using 
+0000c000: 6045 6767 4e4f 472d 6d61 7070 6572 602c  `EggNOG-mapper`,
+0000c010: 2075 7365 7273 206e 6565 6420 746f 2070   users need to p
+0000c020: 726f 7669 6465 2074 6865 2070 6174 6820  rovide the path 
+0000c030: 746f 2074 6865 2065 6767 4e4f 4720 616e  to the eggNOG an
+0000c040: 6e6f 7461 7469 6f6e 2064 6174 6162 6173  notation databas
+0000c050: 6520 7669 6120 7468 6520 6f70 7469 6f6e  e via the option
+0000c060: 2060 2d65 6460 2c20 7468 6520 7061 7468   `-ed`, the path
+0000c070: 2074 6f20 7468 6520 6469 616d 6f6e 642d   to the diamond-
+0000c080: 636f 6d70 6174 6962 6c65 2064 6174 6162  compatible datab
+0000c090: 6173 6520 7669 6120 7468 6520 6f70 7469  ase via the opti
+0000c0a0: 6f6e 2060 2d2d 646d 6e64 5f64 6260 2061  on `--dmnd_db` a
+0000c0b0: 6e64 2074 6865 206f 7074 696f 6e20 602d  nd the option `-
+0000c0c0: 2d61 6e6e 6f74 6174 696f 6e60 2073 6574  -annotation` set
+0000c0d0: 2061 7320 2265 6767 6e6f 6722 2e20 5468   as "eggnog". Th
+0000c0e0: 6520 6d61 6e6e 6572 2068 6f77 2050 4641  e manner how PFA
+0000c0f0: 4d20 616e 6e6f 7461 7469 6f6e 2077 696c  M annotation wil
+0000c100: 6c20 6265 2070 6572 666f 726d 6564 2063  l be performed c
+0000c110: 616e 2062 6520 636f 6e74 726f 6c6c 6564  an be controlled
+0000c120: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
+0000c130: 6070 6661 6d60 2c20 6569 7468 6572 2022  `pfam`, either "
+0000c140: 6e6f 6e65 222c 2022 7265 616c 6967 6e22  none", "realign"
+0000c150: 206f 7220 2264 656e 6f76 6f22 2c20 7468   or "denovo", th
+0000c160: 6520 6465 7461 696c 6564 2065 7870 6c61  e detailed expla
+0000c170: 6e61 7469 6f6e 2063 616e 2062 6520 666f  nation can be fo
+0000c180: 756e 6420 6174 2074 6865 2077 696b 6920  und at the wiki 
+0000c190: 6f66 205b 4567 674e 4f47 2d6d 6170 7065  of [EggNOG-mappe
+0000c1a0: 725d 2868 7474 7073 3a2f 2f67 6974 6875  r](https://githu
+0000c1b0: 622e 636f 6d2f 6567 676e 6f67 6462 2f65  b.com/eggnogdb/e
+0000c1c0: 6767 6e6f 672d 6d61 7070 6572 2f77 696b  ggnog-mapper/wik
+0000c1d0: 6929 2e20 506c 6561 7365 2070 7265 2d69  i). Please pre-i
+0000c1e0: 6e73 7461 6c6c 2074 6865 2060 4567 674e  nstall the `EggN
+0000c1f0: 4f47 2d6d 6170 7065 7260 2070 7974 686f  OG-mapper` pytho
+0000c200: 6e20 7061 636b 6167 6520 6966 2075 7369  n package if usi
+0000c210: 6e67 2074 6869 7320 6675 6e63 7469 6f6e  ng this function
+0000c220: 2e20 466f 7220 7468 6520 616e 6e6f 7461  . For the annota
+0000c230: 7469 6f6e 2075 7369 6e67 2060 686d 6d73  tion using `hmms
+0000c240: 6361 6e60 2c20 7768 6174 2069 7320 696d  can`, what is im
+0000c250: 706c 656d 656e 7465 6420 696e 2060 7767  plemented in `wg
+0000c260: 6420 7632 6020 6973 2061 2073 696d 706c  d v2` is a simpl
+0000c270: 6520 6275 6e64 6c65 2066 756e 6374 696f  e bundle functio
+0000c280: 6e20 746f 2070 6572 666f 726d 2060 686d  n to perform `hm
+0000c290: 6d73 6361 6e60 2061 6e61 6c79 7369 7320  mscan` analysis 
+0000c2a0: 666f 7220 6120 6769 7665 6e20 686d 6d20  for a given hmm 
+0000c2b0: 7072 6f66 696c 6520 616e 6420 7365 7420  profile and set 
+0000c2c0: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
+0000c2d0: 2073 7563 6820 7468 6174 2075 7365 7273   such that users
+0000c2e0: 206e 6565 6420 746f 2073 6574 2074 6865   need to set the
+0000c2f0: 206f 7074 696f 6e20 602d 2d61 6e6e 6f74   option `--annot
+0000c300: 6174 696f 6e60 2061 7320 2268 6d6d 7066  ation` as "hmmpf
+0000c310: 616d 2220 616e 6420 7072 6f76 6964 6520  am" and provide 
+0000c320: 7468 6520 686d 6d70 726f 6669 6c65 2076  the hmmprofile v
+0000c330: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+0000c340: 2d68 6d6d 602e 2046 6f72 2074 6865 2061  -hmm`. For the a
+0000c350: 6e6e 6f74 6174 696f 6e20 7573 696e 6720  nnotation using 
+0000c360: 6069 6e74 6572 7072 6f73 6361 6e60 2c20  `interproscan`, 
+0000c370: 7573 6572 7320 6e65 6564 2074 6f20 7072  users need to pr
+0000c380: 6f76 6964 6520 7468 6520 7061 7468 2074  ovide the path t
+0000c390: 6f20 7468 6520 696e 7465 7270 726f 7363  o the interprosc
+0000c3a0: 616e 2069 6e73 7461 6c6c 6174 696f 6e20  an installation 
+0000c3b0: 666f 6c64 6572 2077 6865 7265 2074 6865  folder where the
+0000c3c0: 7265 2069 7320 6120 6069 6e74 6572 7072  re is a `interpr
+0000c3d0: 6f73 6361 6e2e 7368 6020 6669 6c65 2076  oscan.sh` file v
+0000c3e0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+0000c3f0: 2d65 7865 7061 7468 6020 616e 6420 7365  -exepath` and se
+0000c400: 7420 7468 6520 7468 6520 6f70 7469 6f6e  t the the option
+0000c410: 2060 2d2d 616e 6e6f 7461 7469 6f6e 6020   `--annotation` 
+0000c420: 6173 2022 696e 7465 7270 726f 7363 616e  as "interproscan
+0000c430: 222e 2054 6865 2070 6172 616c 6c65 6c20  ". The parallel 
+0000c440: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
+0000c450: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
+0000c460: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
+0000c470: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
+0000c480: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
+0000c490: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
+0000c4a0: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
+0000c4b0: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
+0000c4c0: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
+0000c4d0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000c4e0: 756e 206f 6620 7468 6973 2061 6e61 6c79  un of this analy
+0000c4f0: 7369 7320 6361 6e20 6265 2077 6974 6820  sis can be with 
+0000c500: 7468 6520 636f 6d6d 616e 6420 6265 6c6f  the command belo
+0000c510: 772e 2a2a 0a0a 6060 600a 7767 6420 666f  w.**..```.wgd fo
+0000c520: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000c530: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000c540: 2073 6571 7565 6e63 6533 202d 2d61 6e6e   sequence3 --ann
+0000c550: 6f74 6174 696f 6e20 6567 676e 6f67 202d  otation eggnog -
+0000c560: 6564 2065 6464 6174 6120 2d2d 646d 6e62  ed eddata --dmnb
+0000c570: 2064 6264 6174 610a 7767 6420 666f 6375   dbdata.wgd focu
+0000c580: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000c590: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000c5a0: 6571 7565 6e63 6533 202d 2d61 6e6e 6f74  equence3 --annot
+0000c5b0: 6174 696f 6e20 686d 6d70 6661 6d20 2d2d  ation hmmpfam --
+0000c5c0: 686d 6d20 686d 6d64 6174 610a 7767 6420  hmm hmmdata.wgd 
+0000c5d0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
+0000c5e0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+0000c5f0: 6532 2073 6571 7565 6e63 6533 202d 2d61  e2 sequence3 --a
+0000c600: 6e6e 6f74 6174 696f 6e20 696e 7465 7270  nnotation interp
+0000c610: 726f 7363 616e 202d 2d65 7865 7061 7468  roscan --exepath
+0000c620: 2024 5041 5448 0a60 6060 0a0a 2a2a 5468   $PATH.```..**Th
+0000c630: 6520 7068 796c 6f67 656e 6574 6963 2064  e phylogenetic d
+0000c640: 6174 696e 6720 6f66 2057 4744 732a 2a0a  ating of WGDs**.
+0000c650: 6060 600a 7767 6420 666f 6375 7320 6661  ```.wgd focus fa
+0000c660: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
+0000c670: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+0000c680: 6e63 6533 202d 6420 6d63 6d63 7472 6565  nce3 -d mcmctree
+0000c690: 202d 7370 2073 7064 6174 6120 282d 2d70   -sp spdata (--p
+0000c6a0: 726f 7463 6f63 6461 7469 6e67 2920 282d  rotcocdating) (-
+0000c6b0: 2d70 6172 7469 7469 6f6e 2920 282d 2d61  -partition) (--a
+0000c6c0: 616d 6f64 656c 206c 6729 2028 2d64 7320  amodel lg) (-ds 
+0000c6d0: 2762 7572 6e69 6e20 3d20 3230 3030 2729  'burnin = 2000')
+0000c6e0: 2028 2d64 7320 2773 616d 7066 7265 7120   (-ds 'sampfreq 
+0000c6f0: 3d20 3130 3030 2729 2028 2d64 7320 276e  = 1000') (-ds 'n
+0000c700: 7361 6d70 6c65 203d 2032 3030 3030 2729  sample = 20000')
+0000c710: 2020 282d 6e20 3429 2028 2d2d 746f 5f73    (-n 4) (--to_s
+0000c720: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
+0000c730: 2077 6764 5f66 6f63 7573 2920 282d 7420   wgd_focus) (-t 
+0000c740: 776f 726b 696e 675f 746d 7029 0a60 6060  working_tmp).```
+0000c750: 0a0a 5468 6520 6162 736f 6c75 7465 2064  ..The absolute d
+0000c760: 6174 696e 6720 6f66 2057 4744 7320 6973  ating of WGDs is
+0000c770: 2061 2073 7065 6369 6669 6320 7069 7065   a specific pipe
+0000c780: 6c69 6e65 2069 6d70 6c65 6d65 6e74 6564  line implemented
+0000c790: 2069 6e20 6077 6764 2076 3260 2075 7369   in `wgd v2` usi
+0000c7a0: 6e67 2074 6865 206d 6574 686f 6420 6f66  ng the method of
+0000c7b0: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+0000c7c0: 7469 6e67 2e20 5468 6520 6661 6d69 6c69  ting. The famili
+0000c7d0: 6573 2075 7365 6420 696e 2074 6869 7320  es used in this 
+0000c7e0: 7374 6570 2063 616e 2062 6520 7072 6f64  step can be prod
+0000c7f0: 7563 6564 2066 726f 6d20 6077 6764 2064  uced from `wgd d
+0000c800: 6d64 6020 616e 6420 6077 6764 2070 6561  md` and `wgd pea
+0000c810: 6b60 2e20 4e6f 7465 2074 6861 7420 6865  k`. Note that he
+0000c820: 7265 2077 6520 6f6e 6c79 2064 6973 6375  re we only discu
+0000c830: 7373 2068 6f77 2074 6f20 6461 7465 2057  ss how to date W
+0000c840: 4744 7320 7769 7468 2067 656e 6f6d 6520  GDs with genome 
+0000c850: 6173 7365 6d62 6c79 2c20 696e 7374 6561  assembly, instea
+0000c860: 6420 6f66 2074 7261 6e73 6372 6970 746f  d of transcripto
+0000c870: 6d65 2061 7373 656d 626c 7920 2877 6869  me assembly (whi
+0000c880: 6368 2077 696c 6c20 6265 2064 6973 6375  ch will be discu
+0000c890: 7373 6564 2069 6e20 6120 7365 7061 7261  ssed in a separa
+0000c8a0: 7465 2073 6563 7469 6f6e 2068 6572 6575  te section hereu
+0000c8b0: 6e64 6572 292e 2054 6865 2061 7373 756d  nder). The assum
+0000c8c0: 7074 696f 6e20 7765 206d 6164 6520 6865  ption we made he
+0000c8d0: 7265 2069 7320 7468 6174 206e 6f74 2061  re is that not a
+0000c8e0: 6c6c 2061 6e63 686f 7220 7061 6972 7320  ll anchor pairs 
+0000c8f0: 2863 6f6c 6c69 6e65 6172 2064 7570 6c69  (collinear dupli
+0000c900: 6361 7465 7329 2061 7265 2073 7569 7461  cates) are suita
+0000c910: 626c 6520 666f 7220 7068 796c 6f67 656e  ble for phylogen
+0000c920: 6574 6963 2064 6174 696e 672c 2066 6f72  etic dating, for
+0000c930: 2069 6e73 7461 6e63 6520 7468 6f73 6520   instance those 
+0000c940: 6661 7374 6c79 206f 7220 736c 6f77 6c79  fastly or slowly
+0000c950: 2065 766f 6c76 696e 6720 6765 6e65 2064   evolving gene d
+0000c960: 7570 6c69 6361 7465 732c 2062 6563 6175  uplicates, becau
+0000c970: 7365 2074 6865 7927 7265 2070 726f 6e65  se they're prone
+0000c980: 2074 6f20 6769 7665 2062 6961 7365 6420   to give biased 
+0000c990: 6461 7469 6e67 2065 7374 696d 6174 696f  dating estimatio
+0000c9a0: 6e2e 2053 6f20 6173 2074 6f20 7265 7472  n. So as to retr
+0000c9b0: 6965 7665 2074 6865 2022 7265 6c69 6162  ieve the "reliab
+0000c9c0: 6c65 2220 616e 6368 6f72 2070 6169 7273  le" anchor pairs
+0000c9d0: 2c20 7765 2069 6d70 6c65 6d65 6e74 6564  , we implemented
+0000c9e0: 2073 6f6d 6520 6d65 7468 6f64 7320 6f66   some methods of
+0000c9f0: 2069 6465 6e74 6966 7969 6e67 2063 7265   identifying cre
+0000ca00: 6469 6162 6c65 2061 6e63 686f 7220 7061  diable anchor pa
+0000ca10: 6972 7320 6261 7365 6420 6f6e 2074 6865  irs based on the
+0000ca20: 6972 204b 7320 7661 6c75 6573 2061 6e64  ir Ks values and
+0000ca30: 2f6f 7220 7265 7369 6469 6e67 2063 6f6c  /or residing col
+0000ca40: 6c69 6e65 6172 2073 6567 6d65 6e74 732e  linear segments.
+0000ca50: 2046 6f72 2061 2067 656e 6f6d 6520 7769   For a genome wi
+0000ca60: 7468 2063 6c65 6172 2073 6967 6e61 6c73  th clear signals
+0000ca70: 206f 6620 7075 7461 7469 7665 2057 4744   of putative WGD
+0000ca80: 732c 2073 7563 6820 6173 2074 6865 202a  s, such as the *
+0000ca90: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
+0000caa0: 6561 2a20 696e 2074 6865 2065 7861 6d70  ea* in the examp
+0000cab0: 6c65 2c20 6120 6865 7572 6973 7469 6320  le, a heuristic 
+0000cac0: 6d65 7468 6f64 2074 6861 7420 6170 706c  method that appl
+0000cad0: 6965 7320 7468 6520 7072 696e 6369 706c  ies the principl
+0000cae0: 6520 6f66 2068 6f77 205b 6b73 7261 7465  e of how [ksrate
+0000caf0: 735d 2868 7474 7073 3a2f 2f67 6974 6875  s](https://githu
+0000cb00: 622e 636f 6d2f 5649 422d 5053 422f 6b73  b.com/VIB-PSB/ks
+0000cb10: 7261 7465 7329 2066 696e 6420 7468 6520  rates) find the 
+0000cb20: 696e 6974 6961 6c20 7065 616b 7320 616e  initial peaks an
+0000cb30: 6420 7468 6569 7220 7061 7261 6d65 7465  d their paramete
+0000cb40: 7273 2077 6173 2069 6d70 6c65 6d65 6e74  rs was implement
+0000cb50: 6564 2074 6f20 6669 6e64 2074 6865 2039  ed to find the 9
+0000cb60: 3525 2063 6f6e 6669 6465 6e63 6520 6c65  5% confidence le
+0000cb70: 7665 6c20 6f66 2074 6865 2061 7373 756d  vel of the assum
+0000cb80: 6564 206c 6f67 6e6f 726d 616c 2064 6973  ed lognormal dis
+0000cb90: 7472 6962 7574 696f 6e20 6f66 2074 6865  tribution of the
+0000cba0: 2061 6e63 686f 7220 7061 6972 202a 4b2a   anchor pair *K*
+0000cbb0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000cbc0: 2064 6973 7472 6962 7574 696f 6e20 746f   distribution to
+0000cbd0: 2066 696c 7465 7220 616e 6368 6f72 2070   filter anchor p
+0000cbe0: 6169 7273 2077 6974 6820 746f 6f20 6869  airs with too hi
+0000cbf0: 6768 206f 7220 6c6f 7720 2a4b 2a3c 7375  gh or low *K*<su
+0000cc00: 623e 533c 2f73 7562 3e20 6167 6573 2c20  b>S</sub> ages, 
+0000cc10: 7468 6520 6578 616d 706c 6172 2063 6f6d  the examplar com
+0000cc20: 6d61 6e64 206f 6620 7768 6963 6820 6973  mand of which is
+0000cc30: 2073 686f 7765 6420 696e 2074 6865 2060   showed in the `
+0000cc40: 7767 6420 7065 616b 6020 7365 6374 696f  wgd peak` sectio
+0000cc50: 6e20 6f66 205b 496c 6c75 7374 7261 7469  n of [Illustrati
+0000cc60: 6f6e 5d28 6874 7470 733a 2f2f 6769 7468  on](https://gith
+0000cc70: 7562 2e63 6f6d 2f68 6563 6865 2d70 7362  ub.com/heche-psb
+0000cc80: 2f77 6764 3f74 6162 3d72 6561 646d 652d  /wgd?tab=readme-
+0000cc90: 6f76 2d66 696c 6523 696c 6c75 7374 7261  ov-file#illustra
+0000cca0: 7469 6f6e 292e 2049 6620 7468 6973 2068  tion). If this h
+0000ccb0: 6575 7269 7374 6963 206d 6574 686f 6420  euristic method 
+0000ccc0: 6661 696c 6564 2074 6f20 6769 7665 2072  failed to give r
+0000ccd0: 6561 736f 6e61 626c 6520 7265 7375 6c74  easonable result
+0000cce0: 732c 2077 6869 6368 2069 7320 7573 7561  s, which is usua
+0000ccf0: 6c6c 7920 6475 6520 746f 2074 6865 2065  lly due to the e
+0000cd00: 6666 6563 7420 6f66 206d 756c 7469 706c  ffect of multipl
+0000cd10: 6520 6164 6a61 6365 6e74 2057 4744 7320  e adjacent WGDs 
+0000cd20: 7468 6174 2062 6c75 7273 2074 6865 2070  that blurs the p
+0000cd30: 6561 6b2d 6669 6e64 696e 6720 7072 6f63  eak-finding proc
+0000cd40: 6573 732c 2075 7365 7273 2063 616e 2074  ess, users can t
+0000cd50: 7572 6e20 746f 2074 6865 2063 6f6c 6c69  urn to the colli
+0000cd60: 6e65 6172 2073 6567 6d65 6e74 732d 6775  near segments-gu
+0000cd70: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
+0000cd80: 2063 6c75 7374 6572 696e 6720 696d 706c   clustering impl
+0000cd90: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+0000cda0: 7632 602c 2069 6e20 7768 6963 6820 6120  v2`, in which a 
+0000cdb0: 636f 6c6c 696e 6561 7220 7365 676d 656e  collinear segmen
+0000cdc0: 742d 7769 7365 2047 4d4d 2063 6c75 7374  t-wise GMM clust
+0000cdd0: 6572 696e 6720 7769 6c6c 2062 6520 6669  ering will be fi
+0000cde0: 7273 7420 636f 6e64 7563 7465 6420 6261  rst conducted ba
+0000cdf0: 7365 6420 6f6e 2074 6865 2022 736f 2d63  sed on the "so-c
+0000ce00: 616c 6c65 6422 2073 6567 6d65 6e74 202a  alled" segment *
+0000ce10: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+0000ce20: 6765 2072 6570 7265 7365 6e74 6564 2062  ge represented b
+0000ce30: 7920 7468 6520 6d65 6469 616e 202a 4b2a  y the median *K*
+0000ce40: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000ce50: 206f 6620 616c 6c20 7468 6520 7265 7369   of all the resi
+0000ce60: 6469 6e67 2073 796e 7465 6c6f 6773 2028  ding syntelogs (
+0000ce70: 6e6f 7465 2074 6861 7420 7468 6520 6765  note that the ge
+0000ce80: 6e65 2064 7570 6c69 6361 7465 2070 6169  ne duplicate pai
+0000ce90: 7273 2061 646f 7074 6564 2069 6e20 7468  rs adopted in th
+0000cea0: 6973 2073 7465 7020 6973 2066 726f 6d20  is step is from 
+0000ceb0: 7468 6520 6669 6c65 2060 6d75 6c74 6970  the file `multip
+0000cec0: 6c69 636f 6e5f 7061 6972 732e 7478 7460  licon_pairs.txt`
+0000ced0: 2077 6869 6368 2063 6f6e 7461 696e 7320   which contains 
+0000cee0: 7468 6520 6675 6c6c 2073 6574 206f 6620  the full set of 
+0000cef0: 7379 6e74 656c 6f67 7329 2c20 696e 7374  syntelogs), inst
+0000cf00: 6561 6420 6f66 2074 6865 2073 6d61 6c6c  ead of the small
+0000cf10: 6572 2067 656e 6520 7365 7420 6f66 2061  er gene set of a
+0000cf20: 6e63 686f 7220 7061 6972 732e 2054 6865  nchor pairs. The
+0000cf30: 2064 6973 7469 6e63 7469 6f6e 2062 6574   distinction bet
+0000cf40: 7765 656e 2061 6e63 686f 7220 7061 6972  ween anchor pair
+0000cf50: 7320 616e 6420 7379 6e74 656c 6f67 7320  s and syntelogs 
+0000cf60: 6973 2074 6861 7420 7468 6520 6c61 7474  is that the latt
+0000cf70: 6572 2072 6566 6572 7320 746f 206d 756c  er refers to mul
+0000cf80: 7469 706c 6520 7365 7473 206f 6620 6765  tiple sets of ge
+0000cf90: 6e65 7320 6465 7269 7665 6420 6672 6f6d  nes derived from
+0000cfa0: 2074 6865 2073 616d 6520 616e 6365 7374   the same ancest
+0000cfb0: 7261 6c20 6765 6e6f 6d69 6320 7265 6769  ral genomic regi
+0000cfc0: 6f6e 2077 6869 6c65 2074 6865 2066 6f72  on while the for
+0000cfd0: 6d65 7220 696d 706c 6965 7320 7468 6520  mer implies the 
+0000cfe0: 6c61 7474 6572 2062 7574 2072 6571 7569  latter but requi
+0000cff0: 7265 7320 6164 6469 7469 6f6e 616c 6c79  res additionally
+0000d000: 2074 6865 2063 6f6e 7365 7276 6564 2067   the conserved g
+0000d010: 656e 6520 6f72 6465 722c 2062 6f74 6820  ene order, both 
+0000d020: 6f66 2077 6869 6368 2061 7265 2c20 7769  of which are, wi
+0000d030: 7468 696e 2061 2067 656e 6f6d 652c 2061  thin a genome, a
+0000d040: 7373 756d 6564 2074 6f20 6265 206f 7269  ssumed to be ori
+0000d050: 6769 6e61 7465 6420 6672 6f6d 2074 6865  ginated from the
+0000d060: 2064 7570 6c69 6361 7469 6f6e 206f 6620   duplication of 
+0000d070: 6120 636f 6d6d 6f6e 2061 6e63 6573 7472  a common ancestr
+0000d080: 616c 2067 656e 6f6d 6963 2072 6567 696f  al genomic regio
+0000d090: 6e20 616e 6420 6173 2073 7563 6820 6465  n and as such de
+0000d0a0: 656d 6564 2065 7669 6465 6e63 6520 666f  emed evidence fo
+0000d0b0: 7220 5747 442e 2054 6865 6e20 7468 6520  r WGD. Then the 
+0000d0c0: 7379 6e74 656c 6f67 7320 7769 6c6c 2062  syntelogs will b
+0000d0d0: 6520 6d61 7070 6564 2062 6163 6b20 6163  e mapped back ac
+0000d0e0: 636f 7264 696e 6720 746f 2074 6865 2063  cording to the c
+0000d0f0: 6c75 7374 6572 696e 6720 7265 7375 6c74  lustering result
+0000d100: 7320 6f66 2074 6865 6972 2061 6666 696c  s of their affil
+0000d110: 6961 7465 6420 7365 676d 656e 7473 2e20  iated segments. 
+0000d120: 5369 6e63 6520 7468 6520 4761 7573 7369  Since the Gaussi
+0000d130: 616e 2073 6861 7065 206f 6620 7468 6520  an shape of the 
+0000d140: 7365 676d 656e 7420 636c 7573 7465 7220  segment cluster 
+0000d150: 646f 6573 6e27 7420 6e65 6365 7373 6172  doesn't necessar
+0000d160: 696c 7920 696d 6974 6174 6520 7468 6520  ily imitate the 
+0000d170: 7368 6170 6520 6f66 2074 6865 2072 6573  shape of the res
+0000d180: 6964 696e 6720 7379 6e74 656c 6f67 732c  iding syntelogs,
+0000d190: 2073 6f20 6173 2074 6f20 7265 7472 6965   so as to retrie
+0000d1a0: 7665 2074 6865 2022 7265 6c69 6162 6c65  ve the "reliable
+0000d1b0: 2220 6765 6e65 2073 6574 7320 666f 7220  " gene sets for 
+0000d1c0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+0000d1d0: 696e 672c 2077 6520 6164 706f 7465 6420  ing, we adpoted 
+0000d1e0: 7468 6520 2839 3525 2920 6869 6768 6573  the (95%) highes
+0000d1f0: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
+0000d200: 2028 4844 5229 206f 6620 7468 6520 7379   (HDR) of the sy
+0000d210: 6e74 656c 6f67 202a 4b2a 3c73 7562 3e53  ntelog *K*<sub>S
+0000d220: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
+0000d230: 6962 7574 696f 6e20 666f 7220 7468 6520  ibution for the 
+0000d240: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+0000d250: 696e 672c 2074 6865 2063 616c 6375 6c61  ing, the calcula
+0000d260: 7469 6f6e 206f 6620 7468 6520 2839 3525  tion of the (95%
+0000d270: 2920 4844 5220 696e 2074 6865 2066 756e  ) HDR in the fun
+0000d280: 6374 696f 6e20 5b63 616c 6375 6c61 7465  ction [calculate
+0000d290: 4850 445d 2868 7474 7073 3a2f 2f67 6974  HPD](https://git
+0000d2a0: 6875 622e 636f 6d2f 6865 6368 652d 7073  hub.com/heche-ps
+0000d2b0: 622f 7767 642f 626c 6f62 2f70 6879 6c6f  b/wgd/blob/phylo
+0000d2c0: 6461 7469 6e67 2f77 6764 2f70 6561 6b2e  dating/wgd/peak.
+0000d2d0: 7079 234c 3137 3537 2920 7365 656b 7320  py#L1757) seeks 
+0000d2e0: 7468 6520 7368 6f72 7465 7374 202a 4b2a  the shortest *K*
+0000d2f0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
+0000d300: 6765 2028 612c 6229 2077 6869 6368 2073  ge (a,b) which s
+0000d310: 6174 6973 6669 6573 2074 6865 2072 6571  atisfies the req
+0000d320: 7569 7265 6d65 6e74 206f 6620 7370 616e  uirement of span
+0000d330: 6e69 6e67 206d 6f72 6520 7468 616e 2028  ning more than (
+0000d340: 3935 2529 206f 6620 616c 6c20 7468 6520  95%) of all the 
+0000d350: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000d360: 7661 6c75 6573 2e20 4f6e 2074 6865 2070  values. On the p
+0000d370: 7265 6d69 7365 206f 6620 6964 656e 7469  remise of identi
+0000d380: 6669 6564 2061 6e63 686f 7220 7061 6972  fied anchor pair
+0000d390: 7320 286e 6f74 6520 7468 6174 2074 6865  s (note that the
+0000d3a0: 2073 796e 7465 6c6f 6773 2061 7265 2061   syntelogs are a
+0000d3b0: 6c73 6f20 7265 6665 7272 6564 2061 7320  lso referred as 
+0000d3c0: 616e 6368 6f72 2070 6169 7273 2068 6572  anchor pairs her
+0000d3d0: 6561 6674 6572 292c 2077 6520 696d 706c  eafter), we impl
+0000d3e0: 656d 656e 7465 6420 696e 2074 6865 2070  emented in the p
+0000d3f0: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
+0000d400: 2074 6865 2073 6f2d 6361 6c6c 6564 2061   the so-called a
+0000d410: 6e63 686f 722d 6177 6172 6520 6c6f 6361  nchor-aware loca
+0000d420: 6c20 4d52 4248 7320 6f72 206f 7274 686f  l MRBHs or ortho
+0000d430: 6772 6f75 7073 2c20 696e 2077 6869 6368  groups, in which
+0000d440: 2074 6865 206f 7269 6769 6e61 6c20 6c6f   the original lo
+0000d450: 6361 6c20 4d52 4248 7320 6172 6520 6675  cal MRBHs are fu
+0000d460: 7274 6865 7220 6d65 7267 6564 2077 6974  rther merged wit
+0000d470: 6820 616e 6368 6f72 2070 6169 7273 2073  h anchor pairs s
+0000d480: 7563 6820 7468 6174 2065 6163 6820 6f72  uch that each or
+0000d490: 7468 6f67 726f 7570 2063 6f6e 7461 696e  thogroup contain
+0000d4a0: 7320 7468 6520 616e 6368 6f72 2070 6169  s the anchor pai
+0000d4b0: 7220 616e 6420 7468 6520 6f72 7468 6f6c  r and the orthol
+0000d4c0: 6f67 7565 732e 2057 6974 6820 7468 6973  ogues. With this
+0000d4d0: 206f 7274 686f 6772 6f75 702c 2075 7365   orthogroup, use
+0000d4e0: 7273 2074 6865 6e20 6e65 6564 206f 6e65  rs then need one
+0000d4f0: 2073 7461 7274 696e 6720 7472 6565 2066   starting tree f
+0000d500: 696c 6520 2861 7320 7368 6f77 6e20 696e  ile (as shown in
+0000d510: 2074 6865 205b 496c 6c75 7374 7261 7469   the [Illustrati
+0000d520: 6f6e 5d28 6874 7470 733a 2f2f 6769 7468  on](https://gith
+0000d530: 7562 2e63 6f6d 2f68 6563 6865 2d70 7362  ub.com/heche-psb
+0000d540: 2f77 6764 2f74 7265 652f 7068 796c 6f64  /wgd/tree/phylod
+0000d550: 6174 696e 673f 7461 623d 7265 6164 6d65  ating?tab=readme
+0000d560: 2d6f 762d 6669 6c65 2369 6c6c 7573 7472  -ov-file#illustr
+0000d570: 6174 696f 6e29 2070 6172 7429 2069 6e64  ation) part) ind
+0000d580: 6963 6174 696e 6720 7468 6520 7472 6565  icating the tree
+0000d590: 2074 6f70 6f6c 6f67 7920 616e 6420 666f   topology and fo
+0000d5a0: 7373 696c 2063 616c 6962 7261 7469 6f6e  ssil calibration
+0000d5b0: 2069 6e66 6f72 6d61 7469 6f6e 2066 6f72   information for
+0000d5c0: 2074 6865 2066 696e 616c 2070 6879 6c6f   the final phylo
+0000d5d0: 6765 6e65 7469 6320 6461 7469 6e67 2075  genetic dating u
+0000d5e0: 7369 6e67 2074 6865 2070 726f 6772 616d  sing the program
+0000d5f0: 2060 7767 6420 666f 6375 7360 2e20 5573   `wgd focus`. Us
+0000d600: 6572 7320 6361 6e20 7365 7420 7468 6520  ers can set the 
+0000d610: 666c 6167 206f 7074 696f 6e20 602d 2d70  flag option `--p
+0000d620: 726f 7463 6f63 6461 7469 6e67 6020 746f  rotcocdating` to
+0000d630: 206f 6e6c 7920 636f 6e64 7563 7420 7468   only conduct th
+0000d640: 6520 6461 7469 6e67 206f 6620 7468 6520  e dating of the 
+0000d650: 636f 6e63 6174 656e 6174 696f 6e20 7072  concatenation pr
+0000d660: 6f74 6569 6e20 4d53 4120 6f72 2074 6865  otein MSA or the
+0000d670: 2066 6c61 6720 6f70 7469 6f6e 2060 2d2d   flag option `--
+0000d680: 7072 6f74 6461 7469 6e67 6020 746f 206f  protdating` to o
+0000d690: 6e6c 7920 636f 6e64 7563 7420 7468 6520  nly conduct the 
+0000d6a0: 6461 7469 6e67 206f 6620 7468 6520 7072  dating of the pr
+0000d6b0: 6f74 6569 6e20 4d53 4173 2c20 6e6f 7465  otein MSAs, note
+0000d6c0: 6420 7468 6174 2074 6865 7365 2074 776f  d that these two
+0000d6d0: 206f 7074 696f 6e73 206f 6e6c 7920 776f   options only wo
+0000d6e0: 726b 2066 6f72 2074 6865 2060 6d63 6d63  rk for the `mcmc
+0000d6f0: 7472 6565 6020 6f70 7469 6f6e 2073 6f20  tree` option so 
+0000d700: 6661 722e 2054 6865 2066 6c61 6720 6f70  far. The flag op
+0000d710: 7469 6f6e 2060 2d2d 7061 7274 6974 696f  tion `--partitio
+0000d720: 6e60 2063 616e 2062 6520 7365 7420 746f  n` can be set to
+0000d730: 2070 6572 666f 726d 2060 6d63 6d63 7472   perform `mcmctr
+0000d740: 6565 6020 616e 616c 7973 6973 2075 7369  ee` analysis usi
+0000d750: 6e67 2074 6865 2070 6172 7469 7469 6f6e  ng the partition
+0000d760: 6564 2064 6174 6120 2869 2e65 2e2c 2031  ed data (i.e., 1
+0000d770: 7374 2c20 326e 6420 616e 6420 3372 6420  st, 2nd and 3rd 
+0000d780: 706f 7369 7469 6f6e 206f 6620 636f 646f  position of codo
+0000d790: 6e29 2069 6e73 7465 6164 206f 6620 7573  n) instead of us
+0000d7a0: 696e 6720 7468 6520 636f 646f 6e20 6173  ing the codon as
+0000d7b0: 2061 2077 686f 6c65 2e20 5468 6520 6f70   a whole. The op
+0000d7c0: 7469 6f6e 2060 2d2d 6161 6d6f 6465 6c60  tion `--aamodel`
+0000d7d0: 2063 616e 2062 6520 7365 7420 746f 2064   can be set to d
+0000d7e0: 6574 6572 6d69 6e65 2074 6865 2061 6d69  etermine the ami
+0000d7f0: 6e6f 2061 6369 6420 6d6f 6465 6c20 6170  no acid model ap
+0000d800: 706c 6965 6420 696e 2060 6d63 6d63 7472  plied in `mcmctr
+0000d810: 6565 6020 616e 616c 7973 6973 2e20 5468  ee` analysis. Th
+0000d820: 6520 6f70 7469 6f6e 2060 2d64 7360 2063  e option `-ds` c
+0000d830: 616e 2062 6520 7573 6564 2074 6f20 7365  an be used to se
+0000d840: 7420 7061 7261 6d65 7465 7273 2066 6f72  t parameters for
+0000d850: 2074 6865 206d 6f6c 6563 756c 6172 2064   the molecular d
+0000d860: 6174 696e 6720 7072 6f67 7261 6d2e 2054  ating program. T
+0000d870: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
+0000d880: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
+0000d890: 7061 7261 6c6c 656c 697a 6520 7468 6520  parallelize the 
+0000d8a0: 616e 616c 7973 6973 2066 6f72 2065 6163  analysis for eac
+0000d8b0: 6820 6765 6e65 2066 616d 696c 7920 616e  h gene family an
+0000d8c0: 6420 7468 7573 2073 7567 6765 7374 6564  d thus suggested
+0000d8d0: 2074 6f20 6265 2073 6574 2061 7320 6d75   to be set as mu
+0000d8e0: 6368 2061 7320 7468 6520 6e75 6d62 6572  ch as the number
+0000d8f0: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
+0000d900: 732e 0a0a 2a2a 4120 7375 6767 6573 7465  s...**A suggeste
+0000d910: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
+0000d920: 616e 2062 6520 7769 7468 2074 6865 2063  an be with the c
+0000d930: 6f6d 6d61 6e64 2062 656c 6f77 2e2a 2a0a  ommand below.**.
+0000d940: 0a60 6060 0a77 6764 2066 6f63 7573 2066  .```.wgd focus f
+0000d950: 616d 696c 6965 7320 7365 7175 656e 6365  amilies sequence
+0000d960: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000d970: 656e 6365 3320 2d64 206d 636d 6374 7265  ence3 -d mcmctre
+0000d980: 6520 2d73 7020 7370 6461 7461 0a77 6764  e -sp spdata.wgd
+0000d990: 2066 6f63 7573 2066 616d 696c 6965 7320   focus families 
+0000d9a0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000d9b0: 6365 3220 7365 7175 656e 6365 3320 2d64  ce2 sequence3 -d
+0000d9c0: 2072 3873 202d 7370 2073 7064 6174 6120   r8s -sp spdata 
+0000d9d0: 2d2d 6e73 6974 6573 206e 7369 7465 696e  --nsites nsitein
+0000d9e0: 666f 0a77 6764 2066 6f63 7573 2066 616d  fo.wgd focus fam
+0000d9f0: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000da00: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000da10: 6365 3320 2d64 2062 6561 7374 202d 2d66  ce3 -d beast --f
+0000da20: 6f73 7369 6c20 666f 7373 696c 696e 666f  ossil fossilinfo
+0000da30: 202d 2d72 6f6f 7468 6569 6768 7420 726f   --rootheight ro
+0000da40: 6f74 6865 6967 6874 696e 666f 202d 2d63  otheightinfo --c
+0000da50: 6861 696e 7365 7420 6368 6169 6e73 6574  hainset chainset
+0000da60: 696e 666f 202d 2d62 6561 7374 6c67 6a61  info --beastlgja
+0000da70: 7220 2450 4154 480a 6060 600a 0a23 2323  r $PATH.```..###
+0000da80: 2077 6764 206b 7364 0a0a 2a2a 5468 6520   wgd ksd..**The 
+0000da90: 636f 6e73 7472 7563 7469 6f6e 206f 6620  construction of 
+0000daa0: 7768 6f6c 6520 7061 7261 6e6f 6d65 202a  whole paranome *
+0000dab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+0000dac0: 6765 2064 6973 7472 6962 7574 696f 6e2a  ge distribution*
+0000dad0: 2a0a 6060 600a 7767 6420 6b73 6420 6661  *.```.wgd ksd fa
+0000dae0: 6d69 6c69 6573 2073 6571 7565 6e63 6520  milies sequence 
+0000daf0: 282d 6f20 7767 645f 6b73 6420 2d74 2077  (-o wgd_ksd -t w
+0000db00: 6764 5f6b 7364 5f74 6d70 202d 2d6e 7468  gd_ksd_tmp --nth
+0000db10: 7265 6164 7320 3420 2d2d 746f 5f73 746f  reads 4 --to_sto
+0000db20: 7020 2d2d 6364 7320 2d2d 7061 6972 7769  p --cds --pairwi
+0000db30: 7365 202d 2d73 7472 6970 5f67 6170 7320  se --strip_gaps 
+0000db40: 2d2d 616c 6967 6e65 7220 6d61 6666 7420  --aligner mafft 
+0000db50: 2d2d 7472 6565 5f6d 6574 686f 6420 6661  --tree_method fa
+0000db60: 7374 7472 6565 202d 2d6e 6f64 655f 6176  sttree --node_av
+0000db70: 6572 6167 6529 0a60 6060 0a0a 5468 6520  erage).```..The 
+0000db80: 7072 6f67 7261 6d20 6077 6764 206b 7364  program `wgd ksd
+0000db90: 602c 2061 7320 696d 7069 6564 2062 7920  `, as impied by 
+0000dba0: 6974 7320 6e61 6d65 2c20 6973 2066 6f72  its name, is for
+0000dbb0: 2074 6865 2063 6f6e 7374 7275 6374 696f   the constructio
+0000dbc0: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
+0000dbd0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000dbe0: 7574 696f 6e2e 2045 7863 6570 7420 666f  ution. Except fo
+0000dbf0: 7220 7468 6520 6166 6f72 656d 656e 7469  r the aforementi
+0000dc00: 6f6e 6564 2070 6172 616d 6574 6572 7320  oned parameters 
+0000dc10: 7375 6368 2061 7320 602d 2d74 6f5f 7374  such as `--to_st
+0000dc20: 6f70 6020 616e 6420 602d 2d63 6473 602c  op` and `--cds`,
+0000dc30: 2074 6865 7265 2061 7265 2073 6f6d 6520   there are some 
+0000dc40: 696d 706f 7274 616e 7420 7061 7261 6d65  important parame
+0000dc50: 7465 7273 2074 6861 7420 6861 7665 2063  ters that have c
+0000dc60: 7275 6369 616c 2069 6d70 6163 7420 6f6e  rucial impact on
+0000dc70: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0000dc80: 7375 623e 2065 7374 696d 6174 696f 6e2e  sub> estimation.
+0000dc90: 2054 6865 206f 7074 696f 6e20 602d 2d70   The option `--p
+0000dca0: 6169 7277 6973 6560 2069 7320 6120 7665  airwise` is a ve
+0000dcb0: 7279 2069 6d70 6f72 7461 6e74 2070 6172  ry important par
+0000dcc0: 616d 6574 6572 2066 6f72 2074 6865 202a  ameter for the *
+0000dcd0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+0000dce0: 7374 696d 6174 696f 6e2c 2077 6974 6820  stimation, with 
+0000dcf0: 7768 6963 6820 7468 6520 6043 4f44 454d  which the `CODEM
+0000dd00: 4c60 2077 696c 6c20 6361 6c63 756c 6174  L` will calculat
+0000dd10: 6520 7468 6520 2a4b 2a3c 7375 623e 533c  e the *K*<sub>S<
+0000dd20: 2f73 7562 3e20 666f 7220 6561 6368 2067  /sub> for each g
+0000dd30: 656e 6520 7061 6972 2073 6570 6172 6174  ene pair separat
+0000dd40: 656c 7920 6261 7365 6420 6f6e 2074 6865  ely based on the
+0000dd50: 2061 6c69 676e 6d65 6e74 206f 6620 6f6e   alignment of on
+0000dd60: 6c79 2074 6865 7365 2074 776f 2067 656e  ly these two gen
+0000dd70: 6573 2069 6e73 7465 6164 206f 6620 7468  es instead of th
+0000dd80: 6520 7768 6f6c 6520 616c 6967 6e6d 656e  e whole alignmen
+0000dd90: 7420 6f66 2074 6865 2066 616d 696c 792c  t of the family,
+0000dda0: 2073 7563 6820 7468 6174 206c 6573 7320   such that less 
+0000ddb0: 6761 7073 2061 7265 2065 7870 6563 7465  gaps are expecte
+0000ddc0: 6420 616e 6420 7468 7573 2074 6865 2061  d and thus the a
+0000ddd0: 6c69 676e 6d65 6e74 2069 6e20 7468 6520  lignment in the 
+0000dde0: 636f 6e73 6964 6572 6174 696f 6e20 6f66  consideration of
+0000ddf0: 2060 434f 4445 4d4c 6020 7769 6c6c 2062   `CODEML` will b
+0000de00: 6520 6c6f 6e67 6572 2028 6265 6361 7573  e longer (becaus
+0000de10: 6520 6043 4f44 454d 4c60 2077 696c 6c20  e `CODEML` will 
+0000de20: 6175 746f 6d61 7465 6c79 2073 6b69 7020  automately skip 
+0000de30: 6576 6572 7920 636f 6c75 6d6e 2077 6974  every column wit
+0000de40: 6820 6761 702c 2072 6567 6172 646c 6573  h gap, regardles
+0000de50: 7320 6f66 2077 6865 7468 6572 2069 7427  s of whether it'
+0000de60: 7320 616e 206f 7665 7261 6c6c 206f 7220  s an overall or 
+0000de70: 7061 7274 6961 6c20 6761 7029 2c20 7769  partial gap), wi
+0000de80: 7468 6f75 7420 7768 6963 6820 7468 6520  thout which the 
+0000de90: 6043 4f44 454d 4c60 2077 696c 6c20 6361  `CODEML` will ca
+0000dea0: 6c63 756c 6174 6520 7468 6520 2a4b 2a3c  lculate the *K*<
+0000deb0: 7375 623e 533c 2f73 7562 3e20 6261 7365  sub>S</sub> base
+0000dec0: 6420 6f6e 2074 6865 2077 686f 6c65 2061  d on the whole a
+0000ded0: 6c69 676e 6d65 6e74 206f 6620 7468 6520  lignment of the 
+0000dee0: 6661 6d69 6c79 2c20 7768 6963 6820 6d69  family, which mi
+0000def0: 6768 7420 6861 7665 206e 6f20 2a4b 2a3c  ght have no *K*<
+0000df00: 7375 623e 533c 2f73 7562 3e20 7265 7375  sub>S</sub> resu
+0000df10: 6c74 2061 7420 616c 6c20 6966 2074 6865  lt at all if the
+0000df20: 2073 7472 6970 7065 6420 616c 6967 6e6d   stripped alignm
+0000df30: 656e 7420 6c65 6e67 7468 2028 7265 6d6f  ent length (remo
+0000df40: 7669 6e67 2061 6c6c 2067 6170 2d63 6f6e  ving all gap-con
+0000df50: 7461 696e 696e 6720 636f 6c75 6d6e 7329  taining columns)
+0000df60: 2077 6173 207a 6572 6f2c 2061 2063 6175   was zero, a cau
+0000df70: 7365 206f 6620 6469 6666 6572 656e 7420  se of different 
+0000df80: 6e75 6d62 6572 206f 6620 2a4b 2a3c 7375  number of *K*<su
+0000df90: 623e 533c 2f73 7562 3e20 6573 7469 6d61  b>S</sub> estima
+0000dfa0: 7465 7320 6265 7477 6565 6e20 2270 6169  tes between "pai
+0000dfb0: 7277 6973 6520 6d6f 6465 2220 616e 6420  rwise mode" and 
+0000dfc0: 226e 6f6e 2d70 6169 7277 6973 6520 6d6f  "non-pairwise mo
+0000dfd0: 6465 222e 2049 7427 7320 6469 6666 6963  de". It's diffic
+0000dfe0: 756c 7420 746f 2073 6179 2077 6869 6368  ult to say which
+0000dff0: 206d 6f64 6520 6973 206d 6f72 6520 6964   mode is more id
+0000e000: 6561 6c2c 2061 6c74 686f 7567 6820 7468  eal, although th
+0000e010: 6520 226e 6f6e 2d70 6169 7277 6973 6520  e "non-pairwise 
+0000e020: 6d6f 6465 2220 2864 6566 6175 6c74 2073  mode" (default s
+0000e030: 6574 7469 6e67 2920 7768 6963 6820 7275  etting) which ru
+0000e040: 6e73 206f 6e20 7468 6520 7768 6f6c 6520  ns on the whole 
+0000e050: 616c 6967 6e6d 656e 7420 696e 7374 6561  alignment instea
+0000e060: 6420 6f66 2061 206c 6f63 616c 2061 6c69  d of a local ali
+0000e070: 676e 6d65 6e74 2c20 6d69 6768 7420 6265  gnment, might be
+0000e080: 206d 6f72 6520 6269 6f6c 6f67 6963 616c   more biological
+0000e090: 6c79 2063 6f6e 7365 7276 6564 2069 6e20  ly conserved in 
+0000e0a0: 7468 6174 2069 7420 6173 7375 7265 7320  that it assures 
+0000e0b0: 7468 6520 6576 6f6c 7574 696f 6e20 6f66  the evolution of
+0000e0c0: 2065 6163 6820 636f 6c75 6d6e 2074 6f20   each column to 
+0000e0d0: 6265 2073 7461 7274 6564 2066 726f 6d20  be started from 
+0000e0e0: 7468 6520 726f 6f74 206f 6620 7468 6520  the root of the 
+0000e0f0: 6661 6d69 6c79 2061 6e64 2061 6c6c 2074  family and all t
+0000e100: 6865 2067 656e 6520 6475 706c 6963 6174  he gene duplicat
+0000e110: 6573 2061 7265 2074 616b 656e 2069 6e74  es are taken int
+0000e120: 6f20 6163 636f 756e 7420 696e 2074 6865  o account in the
+0000e130: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000e140: 2065 7374 696d 6174 696f 6e20 7072 6f63   estimation proc
+0000e150: 6573 732e 2054 6865 206f 7074 696f 6e20  ess. The option 
+0000e160: 602d 2d73 7472 6970 5f67 6170 7360 2063  `--strip_gaps` c
+0000e170: 616e 2072 656d 6f76 6520 616c 6c20 7468  an remove all th
+0000e180: 6520 6761 702d 636f 6e74 6169 6e69 6e67  e gap-containing
+0000e190: 2063 6f6c 756d 6e73 2c20 7769 7468 206f   columns, with o
+0000e1a0: 7220 7769 7468 6f75 7420 7768 6963 6820  r without which 
+0000e1b0: 7468 6520 7265 7375 6c74 206f 6620 226e  the result of "n
+0000e1c0: 6f6e 2d70 6169 7277 6973 6520 6d6f 6465  on-pairwise mode
+0000e1d0: 2220 776f 6e27 7420 6265 2061 6666 6563  " won't be affec
+0000e1e0: 7465 642c 2077 6869 6c65 2077 6974 6820  ted, while with 
+0000e1f0: 7768 6963 6820 7468 6520 7265 7375 6c74  which the result
+0000e200: 206f 6620 2270 6169 7277 6973 6520 6d6f   of "pairwise mo
+0000e210: 6465 2220 7769 6c6c 2062 6520 616c 7465  de" will be alte
+0000e220: 7265 642e 2054 6865 206f 7074 696f 6e20  red. The option 
+0000e230: 602d 2d61 6c69 676e 6572 6020 616e 6420  `--aligner` and 
+0000e240: 602d 2d61 6c6e 5f6f 7074 696f 6e73 6020  `--aln_options` 
+0000e250: 7768 6963 6820 6465 6369 6465 2077 6869  which decide whi
+0000e260: 6368 2061 6c69 676e 6d65 6e74 2070 726f  ch alignment pro
+0000e270: 6772 616d 2074 6f20 6265 2075 7365 6420  gram to be used 
+0000e280: 616e 6420 7768 6963 6820 7061 7261 6d65  and which parame
+0000e290: 7465 7220 746f 2062 6520 7365 742c 2077  ter to be set, w
+0000e2a0: 696c 6c20 6861 7665 2069 6d70 6163 7420  ill have impact 
+0000e2b0: 6f6e 2074 6865 202a 4b2a 3c73 7562 3e53  on the *K*<sub>S
+0000e2c0: 3c2f 7375 623e 2072 6573 756c 7473 2c20  </sub> results, 
+0000e2d0: 6e6f 7465 6420 7468 6174 2074 6865 2064  noted that the d
+0000e2e0: 6566 6175 6c74 2070 726f 6772 616d 2069  efault program i
+0000e2f0: 7320 606d 6166 6674 6020 616e 6420 7468  s `mafft` and th
+0000e300: 6520 7061 7261 6d65 7465 7220 6973 2060  e parameter is `
+0000e310: 2d2d 6175 746f 602e 2054 6865 206f 7074  --auto`. The opt
+0000e320: 696f 6e20 602d 2d74 7265 655f 6d65 7468  ion `--tree_meth
+0000e330: 6f64 6020 616e 6420 602d 2d74 7265 655f  od` and `--tree_
+0000e340: 6f70 7469 6f6e 7360 2064 6563 6964 6520  options` decide 
+0000e350: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
+0000e360: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+0000e370: 6d20 746f 2062 6520 7573 6564 2061 6e64  m to be used and
+0000e380: 2077 6869 6368 2070 6172 616d 6574 6572   which parameter
+0000e390: 2074 6f20 6265 2073 6574 2c20 776f 6e27   to be set, won'
+0000e3a0: 7420 6166 6665 6374 2074 6865 202a 4b2a  t affect the *K*
+0000e3b0: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+0000e3c0: 696d 6174 696f 6e20 6974 7365 6c66 2062  imation itself b
+0000e3d0: 7574 2074 6865 2072 6573 756c 7420 6f66  ut the result of
+0000e3e0: 2064 652d 7265 6475 6e64 616e 6379 2c20   de-redundancy, 
+0000e3f0: 6e6f 7465 6420 7468 6174 2077 6520 696d  noted that we im
+0000e400: 706c 656d 656e 7465 6420 6120 6275 696c  plemented a buil
+0000e410: 742d 696e 2067 656e 6520 7472 6565 2069  t-in gene tree i
+0000e420: 6e66 6572 656e 6365 206d 6574 686f 6420  nference method 
+0000e430: 6261 7365 6420 6f6e 2074 6865 2041 7665  based on the Ave
+0000e440: 7261 6765 204c 696e 6b61 6765 2043 6c75  rage Linkage Clu
+0000e450: 7374 6572 696e 6720 2841 4c43 2920 2874  stering (ALC) (t
+0000e460: 6875 7320 6120 6469 7374 616e 6365 2d62  hus a distance-b
+0000e470: 6173 6564 2074 7265 6529 2077 6974 6820  ased tree) with 
+0000e480: 7468 6520 602d 2d74 7265 655f 6d65 7468  the `--tree_meth
+0000e490: 6f64 6020 7365 7420 6173 2022 636c 7573  od` set as "clus
+0000e4a0: 7465 7222 2e20 5477 6f20 6d65 7468 6f64  ter". Two method
+0000e4b0: 7320 6f66 2064 652d 7265 6475 6e64 616e  s of de-redundan
+0000e4c0: 6379 2061 7265 2069 6d70 6c65 6d65 6e74  cy are implement
+0000e4d0: 6564 2069 6e20 6077 6764 2076 3260 2c20  ed in `wgd v2`, 
+0000e4e0: 6e61 6d65 6c79 206e 6f64 652d 7765 6967  namely node-weig
+0000e4f0: 6874 6564 2061 6e64 206e 6f64 652d 6176  hted and node-av
+0000e500: 6572 6167 6564 206d 6574 686f 6473 2e20  eraged methods. 
+0000e510: 5468 6520 6e6f 6465 2d77 6569 6768 7465  The node-weighte
+0000e520: 6420 6d65 7468 6f64 2061 6368 6965 7665  d method achieve
+0000e530: 7320 7468 6520 6465 2d72 6564 756e 6461  s the de-redunda
+0000e540: 6e63 7920 7669 6120 7765 6967 6869 6e67  ncy via weighing
+0000e550: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0000e560: 7375 623e 2076 616c 7565 2061 7373 6f63  sub> value assoc
+0000e570: 6961 7465 6420 7769 7468 2065 6163 6820  iated with each 
+0000e580: 6765 6e65 2070 6169 7220 7375 6368 2074  gene pair such t
+0000e590: 6861 7420 7468 6520 7765 6967 6874 206f  hat the weight o
+0000e5a0: 6620 6120 7369 6e67 6c65 2064 7570 6c69  f a single dupli
+0000e5b0: 6361 7469 6f6e 2065 7665 6e74 2073 756d  cation event sum
+0000e5c0: 7320 7570 2074 6f20 3120 286e 6f74 6564  s up to 1 (noted
+0000e5d0: 2074 6861 7420 7468 6520 6e75 6d62 6572   that the number
+0000e5e0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
+0000e5f0: 7562 3e20 6573 7469 6d61 7469 6f6e 2072  ub> estimation r
+0000e600: 656d 6169 6e73 2074 6865 2073 616d 6529  emains the same)
+0000e610: 2077 6869 6c65 2074 6865 206e 6f64 652d   while the node-
+0000e620: 6176 6572 6167 6564 206d 6574 686f 6420  averaged method 
+0000e630: 7265 616c 697a 6573 2074 6865 2064 652d  realizes the de-
+0000e640: 7265 6475 6e64 616e 6379 2076 6961 2063  redundancy via c
+0000e650: 616c 6375 6c61 7469 6e67 2070 6572 2067  alculating per g
+0000e660: 656e 6520 7472 6565 206e 6f64 6520 6f6e  ene tree node on
+0000e670: 6520 6176 6572 6167 6564 202a 4b2a 3c73  e averaged *K*<s
+0000e680: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
+0000e690: 2074 6f20 7265 7072 6573 656e 7420 7468   to represent th
+0000e6a0: 6520 6167 6520 6f66 2065 6163 6820 6475  e age of each du
+0000e6b0: 706c 6963 6174 696f 6e20 6576 656e 742e  plication event.
+0000e6c0: 2054 6865 206f 7074 696f 6e20 602d 2d6e   The option `--n
+0000e6d0: 6f64 655f 6176 6572 6167 6560 2063 616e  ode_average` can
+0000e6e0: 2062 6520 7365 7420 746f 2063 686f 6f73   be set to choos
+0000e6f0: 6520 7468 6520 6e6f 6465 2d61 7665 7261  e the node-avera
+0000e700: 6765 6420 7761 7920 6f66 2064 652d 7265  ged way of de-re
+0000e710: 6475 6e64 616e 6379 2e20 4469 6666 6572  dundancy. Differ
+0000e720: 656e 7420 6d65 7468 6f64 7320 6f66 2064  ent methods of d
+0000e730: 652d 7265 6475 6e64 616e 6379 2077 696c  e-redundancy wil
+0000e740: 6c20 6861 7665 2069 6d70 6163 7420 6f6e  l have impact on
+0000e750: 2074 6865 2064 6574 6563 7469 6f6e 206f   the detection o
+0000e760: 6620 5747 4420 7369 676e 616c 732c 2077  f WGD signals, w
+0000e770: 6869 6368 2068 6173 2062 6565 6e20 696e  hich has been in
+0000e780: 7665 7374 6967 6174 6564 2069 6e20 7468  vestigated in th
+0000e790: 6973 205b 6c69 7465 7261 7475 7265 5d28  is [literature](
+0000e7a0: 6874 7470 733a 2f2f 646f 692e 6f72 672f  https://doi.org/
+0000e7b0: 3130 2e31 3039 332f 6762 652f 6576 7932  10.1093/gbe/evy2
+0000e7c0: 3030 292e 2054 6865 2070 6172 616c 6c65  00). The paralle
+0000e7d0: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
+0000e7e0: 7265 2074 6f20 7061 7261 6c6c 656c 697a  re to paralleliz
+0000e7f0: 6520 7468 6520 616e 616c 7973 6973 2066  e the analysis f
+0000e800: 6f72 2065 6163 6820 6765 6e65 2066 616d  or each gene fam
+0000e810: 696c 7920 616e 6420 7468 7573 2073 7567  ily and thus sug
+0000e820: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
+0000e830: 2061 7320 6d75 6368 2061 7320 7468 6520   as much as the 
+0000e840: 6e75 6d62 6572 206f 6620 6765 6e65 2066  number of gene f
+0000e850: 616d 696c 6965 732e 0a0a 2a2a 4120 7375  amilies...**A su
+0000e860: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+0000e870: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+0000e880: 6d61 6e64 2061 7320 6265 6c6f 772a 2a0a  mand as below**.
+0000e890: 0a60 6060 0a77 6764 206b 7364 2066 616d  .```.wgd ksd fam
+0000e8a0: 696c 6965 7320 7365 7175 656e 6365 0a60  ilies sequence.`
+0000e8b0: 6060 0a0a 2a2a 5468 6520 636f 6e73 7472  ``..**The constr
+0000e8c0: 7563 7469 6f6e 206f 6620 6f72 7468 6f6c  uction of orthol
+0000e8d0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+0000e8e0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+0000e8f0: 6275 7469 6f6e 2a2a 0a60 6060 0a77 6764  bution**.```.wgd
+0000e900: 206b 7364 2066 616d 696c 6965 7320 7365   ksd families se
+0000e910: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000e920: 3220 282d 2d72 6577 6569 6768 7429 0a60  2 (--reweight).`
+0000e930: 6060 0a0a 4672 6f6d 2070 6172 616c 6f67  ``..From paralog
+0000e940: 6f75 7320 746f 206f 7274 686f 6c6f 676f  ous to orthologo
+0000e950: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+0000e960: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000e970: 696f 6e2c 2075 7365 7273 206f 6e6c 7920  ion, users only 
+0000e980: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
+0000e990: 6d6f 7265 2063 6473 2066 696c 6573 2e20  more cds files. 
+0000e9a0: 4e6f 7465 2074 6861 7420 7769 7468 206f  Note that with o
+0000e9b0: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
+0000e9c0: 6661 6d69 6c69 6573 2074 6865 2077 6569  families the wei
+0000e9d0: 6768 7469 6e67 206d 6574 686f 6420 6361  ghting method ca
+0000e9e0: 6e20 6265 2073 6574 2074 6f20 6265 2063  n be set to be c
+0000e9f0: 616c 6375 6c61 7465 6420 7065 7220 7370  alculated per sp
+0000ea00: 6563 6965 7320 7061 6972 2069 6e73 7465  ecies pair inste
+0000ea10: 6164 206f 6620 636f 6e73 6964 6572 696e  ad of considerin
+0000ea20: 6720 7468 6520 7768 6f6c 6520 6661 6d69  g the whole fami
+0000ea30: 6c79 2062 6563 6175 7365 2077 6865 6e20  ly because when 
+0000ea40: 706c 6f74 7469 6e67 206f 7274 686f 6c6f  plotting ortholo
+0000ea50: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+0000ea60: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000ea70: 7574 696f 6e20 6265 7477 6565 6e20 7477  ution between tw
+0000ea80: 6f20 7370 6563 6965 7320 7468 6520 7765  o species the we
+0000ea90: 6967 6874 2063 616c 6375 6c61 7465 6420  ight calculated 
+0000eaa0: 6672 6f6d 2074 6869 7320 7370 6563 6966  from this specif
+0000eab0: 6963 2073 7065 6369 6573 2070 6169 7220  ic species pair 
+0000eac0: 7368 6f75 6c64 2062 6520 636f 6e73 6572  should be conser
+0000ead0: 7665 6420 7768 696c 6520 7468 6520 6f6e  ved while the on
+0000eae0: 6520 6361 6c63 756c 6174 6564 2066 726f  e calculated fro
+0000eaf0: 6d20 7468 6520 7768 6f6c 6520 6661 6d69  m the whole fami
+0000eb00: 6c79 2077 696c 6c20 7661 7279 2077 6974  ly will vary wit
+0000eb10: 6820 7468 6520 6e75 6d62 6572 206f 6620  h the number of 
+0000eb20: 7370 6563 6965 732e 2054 6f20 696e 6974  species. To init
+0000eb30: 6961 7465 2074 6865 2077 6569 6768 7469  iate the weighti
+0000eb40: 6e67 2070 6572 2073 7065 6369 6573 2070  ng per species p
+0000eb50: 6169 722c 2074 6865 206f 7074 696f 6e20  air, the option 
+0000eb60: 602d 2d72 6577 6569 6768 7460 2063 616e  `--reweight` can
+0000eb70: 2062 6520 7365 742e 0a0a 2a2a 4120 7375   be set...**A su
+0000eb80: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+0000eb90: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+0000eba0: 6d61 6e64 2061 7320 6265 6c6f 772a 2a0a  mand as below**.
+0000ebb0: 0a60 6060 0a77 6764 206b 7364 2066 616d  .```.wgd ksd fam
+0000ebc0: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000ebd0: 7365 7175 656e 6365 320a 6060 600a 0a2a  sequence2.```..*
+0000ebe0: 2a54 6865 2063 6f6e 7374 7275 6374 696f  *The constructio
+0000ebf0: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
+0000ec00: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000ec10: 7574 696f 6e20 7769 7468 2072 6174 6520  ution with rate 
+0000ec20: 636f 7272 6563 7469 6f6e 2a2a 0a60 6060  correction**.```
+0000ec30: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
+0000ec40: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+0000ec50: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000ec60: 2d2d 666f 6375 7332 616c 6c20 7365 7175  --focus2all sequ
+0000ec70: 656e 6365 3120 2d73 7020 7370 6461 7461  ence1 -sp spdata
+0000ec80: 0a60 6060 0a0a 496e 7370 6972 6564 2062  .```..Inspired b
+0000ec90: 7920 7468 6520 7261 7465 2063 6f72 7265  y the rate corre
+0000eca0: 6374 696f 6e20 616c 676f 7269 7468 6d20  ction algorithm 
+0000ecb0: 696e 205b 6b73 7261 7465 735d 2868 7474  in [ksrates](htt
+0000ecc0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+0000ecd0: 5649 422d 5053 422f 6b73 7261 7465 7329  VIB-PSB/ksrates)
+0000ece0: 2c20 7765 2069 6d70 6c65 6d65 6e74 6564  , we implemented
+0000ecf0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+0000ed00: 7469 6f6e 2061 6e61 6c79 7369 7320 616c  tion analysis al
+0000ed10: 736f 2069 6e20 6077 6764 2076 3260 2c20  so in `wgd v2`, 
+0000ed20: 7768 6963 6820 6973 206d 6f73 746c 7920  which is mostly 
+0000ed30: 7468 6520 7361 6d65 2061 7320 5b6b 7372  the same as [ksr
+0000ed40: 6174 6573 5d28 6874 7470 733a 2f2f 6769  ates](https://gi
+0000ed50: 7468 7562 2e63 6f6d 2f56 4942 2d50 5342  thub.com/VIB-PSB
+0000ed60: 2f6b 7372 6174 6573 2920 6275 7420 6469  /ksrates) but di
+0000ed70: 6666 6572 7320 696e 2074 6865 2063 616c  ffers in the cal
+0000ed80: 6375 6c61 7469 6f6e 206f 6620 7468 6520  culation of the 
+0000ed90: 7374 616e 6461 7264 2064 6576 6961 7469  standard deviati
+0000eda0: 6f6e 206f 6620 7265 7363 616c 6564 202a  on of rescaled *
+0000edb0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+0000edc0: 6765 732e 2054 6f20 7065 7266 6f72 6d20  ges. To perform 
+0000edd0: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+0000ede0: 696f 6e20 616e 616c 7973 6973 2c20 7573  ion analysis, us
+0000edf0: 6572 7320 6361 6e20 7573 6520 626f 7468  ers can use both
+0000ee00: 2074 6865 2060 7767 6420 6b73 6460 2061   the `wgd ksd` a
+0000ee10: 6e64 2060 7767 6420 7669 7a60 2070 726f  nd `wgd viz` pro
+0000ee20: 6772 616d 2e20 466f 7220 7468 6520 6077  gram. For the `w
+0000ee30: 6764 206b 7364 6020 7072 6f67 7261 6d2c  gd ksd` program,
+0000ee40: 2075 7365 7273 206e 6565 6420 746f 2070   users need to p
+0000ee50: 726f 7669 6465 2061 2073 7065 6369 6573  rovide a species
+0000ee60: 2074 7265 6520 7669 6120 7468 6520 6f70   tree via the op
+0000ee70: 7469 6f6e 2060 2d2d 7370 6563 6965 7374  tion `--speciest
+0000ee80: 7265 6560 206f 6e20 7768 6963 6820 7261  ree` on which ra
+0000ee90: 7465 2063 6f72 7265 6374 696f 6e20 6361  te correction ca
+0000eea0: 6e20 6265 2063 6f6e 6475 6374 6564 2e20  n be conducted. 
+0000eeb0: 4e6f 7465 2074 6861 7420 756e 6e65 6365  Note that unnece
+0000eec0: 7373 6172 7920 6272 6163 6b65 7473 206d  ssary brackets m
+0000eed0: 6967 6874 206c 6561 6420 746f 2075 6e65  ight lead to une
+0000eee0: 7870 6563 7465 6420 6572 726f 7273 2c20  xpected errors, 
+0000eef0: 666f 7220 696e 7374 616e 6365 2061 2074  for instance a t
+0000ef00: 7265 6520 6028 412c 2842 2c43 2929 3b60  ree `(A,(B,C));`
+0000ef10: 2073 686f 756c 6420 6e6f 7420 6265 2072   should not be r
+0000ef20: 6570 7265 7365 6e74 6564 2061 7320 6028  epresented as `(
+0000ef30: 412c 2828 422c 4329 2929 3b60 2e20 5468  A,((B,C)));`. Th
+0000ef40: 6520 7365 7420 6f66 2073 7065 6369 6573  e set of species
+0000ef50: 2070 6169 7273 2074 6f20 6265 2073 686f   pairs to be sho
+0000ef60: 776e 2069 7320 666c 6578 6962 6c65 2074  wn is flexible t
+0000ef70: 6861 7420 7468 6520 6d6f 7374 2063 6f6e  hat the most con
+0000ef80: 7665 6e69 656e 7420 6f70 7469 6f6e 2069  venient option i
+0000ef90: 7320 602d 2d66 6f63 7573 3261 6c6c 6020  s `--focus2all` 
+0000efa0: 7768 6963 6820 7369 6d70 6c79 2073 686f  which simply sho
+0000efb0: 7773 2061 6c6c 2074 6865 2070 6f73 7369  ws all the possi
+0000efc0: 626c 6520 666f 6361 6c2d 7369 7374 6572  ble focal-sister
+0000efd0: 2073 7065 6369 6573 2070 6169 7273 2c20   species pairs, 
+0000efe0: 6f72 2075 7365 7273 2063 616e 206d 616e  or users can man
+0000eff0: 7561 6c6c 7920 7365 7420 7468 6520 7370  ually set the sp
+0000f000: 6563 6965 7320 7061 6972 7320 7669 6120  ecies pairs via 
+0000f010: 7468 6520 6f70 7469 6f6e 2060 2d2d 7370  the option `--sp
+0000f020: 6169 7260 2e20 4e6f 7465 2074 6861 7420  air`. Note that 
+0000f030: 6966 2074 6865 2073 7065 6369 6573 2070  if the species p
+0000f040: 6169 7273 2077 6572 6520 6d61 6e75 616c  airs were manual
+0000f050: 6c79 2073 6574 2c20 6974 2077 6f75 6c64  ly set, it would
+0000f060: 2062 6520 6e65 6564 6564 2074 6f20 7365   be needed to se
+0000f070: 7420 7468 6520 6f70 7469 6f6e 2060 2d2d  t the option `--
+0000f080: 636c 6173 7369 6360 2e20 5468 6520 6f70  classic`. The op
+0000f090: 7469 6f6e 2060 6f6e 6c79 726f 6f74 6f75  tion `onlyrootou
+0000f0a0: 7460 2063 616e 2062 6520 7365 7420 746f  t` can be set to
+0000f0b0: 206f 6e6c 7920 636f 6e73 6964 6572 206f   only consider o
+0000f0c0: 7574 6772 6f75 7020 6174 2074 6865 2072  utgroup at the r
+0000f0d0: 6f6f 742c 2069 6e73 7465 6164 206f 6620  oot, instead of 
+0000f0e0: 616c 6c20 7468 6520 706f 7373 6962 6c65  all the possible
+0000f0f0: 206f 7574 6772 6f75 7073 2070 6572 2066   outgroups per f
+0000f100: 6f63 616c 2d73 6973 7465 7220 7370 6563  ocal-sister spec
+0000f110: 6965 7320 7061 6972 2c20 7768 6963 6820  ies pair, which 
+0000f120: 6861 7320 696d 7061 6374 206f 6e20 7468  has impact on th
+0000f130: 6520 6669 6e61 6c20 636f 7272 6563 7465  e final correcte
+0000f140: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
+0000f150: 3e20 6167 6573 2e20 536f 6d65 206f 7468  > ages. Some oth
+0000f160: 6572 206f 7074 696f 6e73 2077 6869 6368  er options which
+0000f170: 2068 6176 6520 6e6f 2069 6d70 6163 7420   have no impact 
+0000f180: 6f6e 2074 6865 2072 6174 6520 636f 7272  on the rate corr
+0000f190: 6563 7469 6f6e 2062 7574 2061 6464 206d  ection but add m
+0000f1a0: 6f72 6520 6c61 7965 7273 206f 7220 6368  ore layers or ch
+0000f1b0: 616e 6765 2074 6865 2061 7070 6561 7261  ange the appeara
+0000f1c0: 6e63 6520 6f6e 2074 6865 206d 6978 6564  nce on the mixed
+0000f1d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000f1e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+0000f1f0: 6e2c 2069 6e63 6c75 6465 2060 706c 6f74  n, include `plot
+0000f200: 6170 676d 6d60 2061 6e64 2060 706c 6f74  apgmm` and `plot
+0000f210: 656c 6d6d 6020 6574 632e 2054 6865 206f  elmm` etc. The o
+0000f220: 7074 696f 6e20 6070 6c6f 7461 7067 6d6d  ption `plotapgmm
+0000f230: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
+0000f240: 6361 6c6c 2074 6865 2047 4d4d 2061 6e61  call the GMM ana
+0000f250: 6c79 7369 7320 7570 6f6e 2074 6865 2061  lysis upon the a
+0000f260: 6e63 686f 7220 7061 6972 202a 4b2a 3c73  nchor pair *K*<s
+0000f270: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000f280: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+0000f290: 706c 6f74 2074 6865 2063 6c75 7374 6572  plot the cluster
+0000f2a0: 696e 6720 7265 7375 6c74 2075 706f 6e20  ing result upon 
+0000f2b0: 7468 6520 6d69 7865 6420 2a4b 2a3c 7375  the mixed *K*<su
+0000f2c0: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+0000f2d0: 7374 7269 6275 7469 6f6e 2c20 7768 6963  stribution, whic
+0000f2e0: 6820 6861 7320 746f 2062 6520 636f 2d73  h has to be co-s
+0000f2f0: 6574 2077 6974 6820 7468 6520 6f70 7469  et with the opti
+0000f300: 6f6e 2060 2d2d 616e 6368 6f72 706f 696e  on `--anchorpoin
+0000f310: 7473 6020 7072 6f76 6964 696e 6720 7468  ts` providing th
+0000f320: 6520 616e 6368 6f72 2070 6169 7273 2069  e anchor pairs i
+0000f330: 6e66 6f72 6d61 7469 6f6e 2e20 5468 6520  nformation. The 
+0000f340: 6f70 7469 6f6e 2060 2d2d 706c 6f74 656c  option `--plotel
+0000f350: 6d6d 6020 6361 6e20 6265 2073 6574 2074  mm` can be set t
+0000f360: 6f20 6361 6c6c 2074 6865 2045 4c4d 4d20  o call the ELMM 
+0000f370: 616e 616c 7973 6973 2075 706f 6e20 7468  analysis upon th
+0000f380: 6520 7061 7261 6c6f 676f 7573 202a 4b2a  e paralogous *K*
+0000f390: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000f3a0: 2064 6973 7472 6962 7574 696f 6e2e 0a0a   distribution...
+0000f3b0: 5468 6572 6520 6172 6520 3231 2063 6f6c  There are 21 col
+0000f3c0: 756d 6e73 2069 6e20 7468 6520 7265 7375  umns in the resu
+0000f3d0: 6c74 2060 2a6b 732e 7473 7660 2066 696c  lt `*ks.tsv` fil
+0000f3e0: 6520 6265 7369 6465 7320 7468 6520 696e  e besides the in
+0000f3f0: 6465 7820 636f 6c75 6d6e 7320 6070 6169  dex columns `pai
+0000f400: 7260 2061 7320 7468 6520 756e 6971 7565  r` as the unique
+0000f410: 2069 6465 6e74 6966 6965 7220 666f 7220   identifier for 
+0000f420: 6561 6368 2067 656e 6520 7061 6972 2e20  each gene pair. 
+0000f430: 5468 6520 604e 602c 2060 5360 2c20 6064  The `N`, `S`, `d
+0000f440: 4e60 2c20 6064 4e2f 6453 602c 2060 6453  N`, `dN/dS`, `dS
+0000f450: 602c 2060 6c60 2061 6e64 2060 7460 2061  `, `l` and `t` a
+0000f460: 7265 2066 726f 6d20 7468 6520 636f 6465  re from the code
+0000f470: 6d6c 2072 6573 756c 7473 2c20 7265 7072  ml results, repr
+0000f480: 6573 656e 7469 6e67 2074 6865 204e 2065  esenting the N e
+0000f490: 7374 696d 6174 652c 2074 6865 2053 2065  stimate, the S e
+0000f4a0: 7374 696d 6174 652c 2074 6865 2064 4e20  stimate, the dN 
+0000f4b0: 6573 7469 6d61 7465 2c20 7468 6520 644e  estimate, the dN
+0000f4c0: 2f64 5320 286f 6d65 6761 2920 6573 7469  /dS (omega) esti
+0000f4d0: 6d61 7465 2c20 7468 6520 6453 2065 7374  mate, the dS est
+0000f4e0: 696d 6174 652c 2074 6865 206c 6f67 2d6c  imate, the log-l
+0000f4f0: 696b 656c 6968 6f6f 6420 616e 6420 7468  ikelihood and th
+0000f500: 6520 7420 6573 7469 6d61 7465 2c20 7265  e t estimate, re
+0000f510: 7370 6563 7469 7665 6c79 2e20 5468 6520  spectively. The 
+0000f520: 6061 6c69 676e 6d65 6e74 636f 7665 7261  `alignmentcovera
+0000f530: 6765 602c 2060 616c 6967 6e6d 656e 7469  ge`, `alignmenti
+0000f540: 6465 6e74 6974 7960 2061 6e64 2060 616c  dentity` and `al
+0000f550: 6967 6e6d 656e 746c 656e 6774 6860 2061  ignmentlength` a
+0000f560: 7265 2074 6865 2069 6e66 6f72 6d61 7469  re the informati
+0000f570: 6f6e 2070 6572 7461 696e 696e 6720 746f  on pertaining to
+0000f580: 2074 6865 2061 6c69 676e 6d65 6e74 2066   the alignment f
+0000f590: 6f72 2065 6163 6820 6661 6d69 6c79 2c20  or each family, 
+0000f5a0: 7265 7072 6573 656e 7469 6e67 2074 6865  representing the
+0000f5b0: 2072 6174 696f 206f 6620 7468 6520 7374   ratio of the st
+0000f5c0: 7269 7070 6564 2061 6c69 676e 6d65 6e74  ripped alignment
+0000f5d0: 206c 656e 6774 6820 636f 6d70 6172 6564   length compared
+0000f5e0: 2074 6f20 7468 6520 6675 6c6c 2061 6c69   to the full ali
+0000f5f0: 676e 6d65 6e74 206c 656e 6774 682c 2074  gnment length, t
+0000f600: 6865 2072 6174 696f 206f 6620 636f 6c75  he ratio of colu
+0000f610: 6d6e 7320 7769 7468 2069 6465 6e74 6963  mns with identic
+0000f620: 616c 206e 7563 6c65 6f74 6964 6573 2063  al nucleotides c
+0000f630: 6f6d 7061 7265 6420 746f 2074 6865 206f  ompared to the o
+0000f640: 7665 7261 6c6c 2063 6f6c 756d 6e73 206f  verall columns o
+0000f650: 6620 7468 6520 7374 7269 7070 6564 2061  f the stripped a
+0000f660: 6c69 676e 6d65 6e74 2c20 616e 6420 7468  lignment, and th
+0000f670: 6520 6c65 6e67 7468 206f 6620 7468 6520  e length of the 
+0000f680: 6675 6c6c 2061 6c69 676e 6d65 6e74 2c20  full alignment, 
+0000f690: 7265 7370 6563 7469 7665 6c79 2e0a 2323  respectively..##
+0000f6a0: 2320 7767 6420 6d69 780a 0a2a 2a54 6865  # wgd mix..**The
+0000f6b0: 206d 6978 7475 7265 206d 6f64 656c 2063   mixture model c
+0000f6c0: 6c75 7374 6572 696e 6720 616e 616c 7973  lustering analys
+0000f6d0: 6973 206f 6620 2a4b 2a3c 7375 623e 533c  is of *K*<sub>S<
+0000f6e0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+0000f6f0: 6275 7469 6f6e 2a2a 0a60 6060 0a77 6764  bution**.```.wgd
+0000f700: 206d 6978 206b 7364 6174 6120 282d 2d6e   mix ksdata (--n
+0000f710: 5f69 6e69 7420 3230 3020 2d2d 6d61 785f  _init 200 --max_
+0000f720: 6974 6572 2032 3030 202d 2d6b 735f 7261  iter 200 --ks_ra
+0000f730: 6e67 6520 3020 3520 2d2d 6669 6c74 6572  nge 0 5 --filter
+0000f740: 7320 3330 3020 2d2d 6269 6e73 2035 3020  s 300 --bins 50 
+0000f750: 2d2d 636f 6d70 6f6e 656e 7473 2031 2034  --components 1 4
+0000f760: 202d 2d67 616d 6d61 2030 2e30 3031 290a   --gamma 0.001).
+0000f770: 6060 600a 0a54 6869 7320 7061 7274 206f  ```..This part o
+0000f780: 6620 4761 7573 7369 616e 206d 6978 7475  f Gaussian mixtu
+0000f790: 7265 206d 6f64 656c 696e 6720 2847 4d4d  re modeling (GMM
+0000f7a0: 2920 616e 616c 7973 6973 2069 7320 696e  ) analysis is in
+0000f7b0: 6865 7269 7465 6420 6672 6f6d 2074 6865  herited from the
+0000f7c0: 206f 7269 6769 6e61 6c20 6077 6764 6020   original `wgd` 
+0000f7d0: 7072 6f67 7261 6d2c 2062 7574 2077 7269  program, but wri
+0000f7e0: 7465 7320 6164 6469 7469 6f6e 616c 6c79  tes additionally
+0000f7f0: 2074 6865 2070 726f 6261 6269 6c69 7479   the probability
+0000f800: 206f 6620 6561 6368 202a 4b2a 3c73 7562   of each *K*<sub
+0000f810: 3e53 3c2f 7375 623e 2076 616c 7565 2069  >S</sub> value i
+0000f820: 6e74 6f20 7468 6520 6669 6e61 6c20 6461  nto the final da
+0000f830: 7461 6672 616d 652e 2042 6173 6963 616c  taframe. Basical
+0000f840: 6c79 2c20 7573 6572 7320 6e65 6564 2074  ly, users need t
+0000f850: 6f20 7072 6f76 6964 6520 7769 7468 2061  o provide with a
+0000f860: 2028 6e6f 726d 616c 6c79 2066 726f 6d20   (normally from 
+0000f870: 7768 6f6c 652d 7061 7261 6e6f 6d65 206f  whole-paranome o
+0000f880: 7220 616e 6368 6f72 2d70 6169 7273 2920  r anchor-pairs) 
+0000f890: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000f8a0: 6461 7461 6669 6c65 2061 6e64 2074 6865  datafile and the
+0000f8b0: 2047 4d4d 2061 6e61 6c79 7369 7320 7769   GMM analysis wi
+0000f8c0: 6c6c 2062 6520 636f 6e64 7563 7465 6420  ll be conducted 
+0000f8d0: 7570 6f6e 2074 6865 2064 6174 6166 696c  upon the datafil
+0000f8e0: 652e 2053 6f6d 6520 7061 7261 6d65 7465  e. Some paramete
+0000f8f0: 7273 2063 616e 2061 6666 6563 7420 7468  rs can affect th
+0000f900: 6520 7265 7375 6c74 732c 2069 6e63 6c75  e results, inclu
+0000f910: 6469 6e67 2060 2d2d 6e5f 696e 6974 602c  ding `--n_init`,
+0000f920: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+0000f930: 6e75 6d62 6572 206f 6620 6b2d 6d65 616e  number of k-mean
+0000f940: 7320 696e 6974 6961 6c69 7a61 7469 6f6e  s initialization
+0000f950: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
+0000f960: 2060 2d2d 6d61 785f 6974 6572 602c 2077   `--max_iter`, w
+0000f970: 6869 6368 2073 6574 7320 7468 6520 6d61  hich sets the ma
+0000f980: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
+0000f990: 6974 6572 6174 696f 6e73 2028 6465 6661  iterations (defa
+0000f9a0: 756c 7420 3230 3029 2c20 602d 2d6d 6574  ult 200), `--met
+0000f9b0: 686f 6460 2c20 7768 6963 6820 6465 7465  hod`, which dete
+0000f9c0: 726d 696e 6573 2077 6869 6368 2063 6c75  rmines which clu
+0000f9d0: 7374 6572 696e 6720 6d65 7468 6f64 2074  stering method t
+0000f9e0: 6f20 7573 6520 2864 6566 6175 6c74 2067  o use (default g
+0000f9f0: 6d6d 292c 2060 2d2d 6761 6d6d 6160 2c20  mm), `--gamma`, 
+0000fa00: 7768 6963 6820 7365 7473 2074 6865 2067  which sets the g
+0000fa10: 616d 6d61 2070 6172 616d 6574 6572 2066  amma parameter f
+0000fa20: 6f72 2074 6865 2062 676d 6d20 6d6f 6465  or the bgmm mode
+0000fa30: 6c20 2864 6566 6175 6c74 2030 2e30 3031  l (default 0.001
+0000fa40: 292c 2060 2d2d 636f 6d70 6f6e 656e 7473  ), `--components
+0000fa50: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+0000fa60: 6520 7261 6e67 6520 6f66 2074 6865 206e  e range of the n
+0000fa70: 756d 6265 7220 6f66 2063 6f6d 706f 6e65  umber of compone
+0000fa80: 6e74 7320 746f 2066 6974 2028 6465 6661  nts to fit (defa
+0000fa90: 756c 7420 3120 3429 2c20 616e 6420 7468  ult 1 4), and th
+0000faa0: 6520 6461 7461 2066 696c 7465 7269 6e67  e data filtering
+0000fab0: 2070 6172 616d 6574 6572 7320 602d 2d66   parameters `--f
+0000fac0: 696c 7465 7273 6020 7768 6963 6820 6669  ilters` which fi
+0000fad0: 6c74 6572 7320 6461 7461 2062 6173 6564  lters data based
+0000fae0: 206f 6e20 616c 6967 6e6d 656e 7420 6c65   on alignment le
+0000faf0: 6e67 7468 2c20 602d 2d6b 735f 7261 6e67  ngth, `--ks_rang
+0000fb00: 6560 2077 6869 6368 2066 696c 7465 7273  e` which filters
+0000fb10: 2064 6174 6120 6261 7365 6420 6f6e 202a   data based on *
+0000fb20: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
+0000fb30: 616c 7565 7320 616e 6420 7468 6520 7061  alues and the pa
+0000fb40: 7261 6d65 7465 7220 602d 2d62 696e 7360  rameter `--bins`
+0000fb50: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+0000fb60: 6e75 6d62 6572 206f 6620 6269 6e73 2069  number of bins i
+0000fb70: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+0000fb80: 3e20 6469 7374 7269 6275 7469 6f6e 2028  > distribution (
+0000fb90: 6465 6661 756c 7420 3530 292e 0a0a 2a2a  default 50)...**
+0000fba0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+0000fbb0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+0000fbc0: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
+0000fbd0: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
+0000fbe0: 7767 6420 6d69 7820 6b73 6461 7461 0a60  wgd mix ksdata.`
+0000fbf0: 6060 0a0a 2323 2320 7767 6420 7065 616b  ``..### wgd peak
+0000fc00: 0a0a 2a2a 5468 6520 7365 6172 6368 206f  ..**The search o
+0000fc10: 6620 6372 6564 6961 626c 6520 2a4b 2a3c  f crediable *K*<
+0000fc20: 7375 623e 533c 2f73 7562 3e20 7261 6e67  sub>S</sub> rang
+0000fc30: 6520 7573 6564 2069 6e20 5747 4420 6461  e used in WGD da
+0000fc40: 7469 6e67 2a2a 0a60 6060 0a77 6764 2070  ting**.```.wgd p
+0000fc50: 6561 6b20 6b73 6461 7461 202d 6170 2061  eak ksdata -ap a
+0000fc60: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
+0000fc70: 202d 6c65 206c 6564 6174 6120 2d6d 7020   -le ledata -mp 
+0000fc80: 6d70 6461 7461 202d 2d68 6575 7269 7374  mpdata --heurist
+0000fc90: 6963 2028 2d2d 616c 6967 6e66 696c 7465  ic (--alignfilte
+0000fca0: 7220 302e 3020 3020 302e 3020 2d2d 6b73  r 0.0 0 0.0 --ks
+0000fcb0: 7261 6e67 6520 3020 3520 2d2d 6269 6e5f  range 0 5 --bin_
+0000fcc0: 7769 6474 6820 302e 3120 2d2d 6775 6964  width 0.1 --guid
+0000fcd0: 6520 7365 676d 656e 7420 2d2d 7072 6f6d  e segment --prom
+0000fce0: 696e 656e 6365 5f63 7574 6f66 6620 302e  inence_cutoff 0.
+0000fcf0: 3120 2d2d 7265 6c5f 6865 6967 6874 2030  1 --rel_height 0
+0000fd00: 2e34 202d 2d63 6920 3935 202d 2d68 6472  .4 --ci 95 --hdr
+0000fd10: 2039 3520 2d2d 6b73 6375 746f 6666 2035   95 --kscutoff 5
+0000fd20: 290a 6060 600a 0a41 7320 6d65 6e74 696f  ).```..As mentio
+0000fd30: 6e65 6420 7072 6576 696f 7573 6c79 2c20  ned previously, 
+0000fd40: 6120 6865 7572 6973 7469 6320 6d65 7468  a heuristic meth
+0000fd50: 6f64 2061 6e64 2061 2063 6f6c 6c69 6e65  od and a colline
+0000fd60: 6172 2073 6567 6d65 6e74 732d 6775 6964  ar segments-guid
+0000fd70: 6564 2061 6e63 686f 7220 7061 6972 2063  ed anchor pair c
+0000fd80: 6c75 7374 6572 696e 6720 666f 7220 7468  lustering for th
+0000fd90: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
+0000fda0: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
+0000fdb0: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
+0000fdc0: 2069 6e20 5747 4420 6461 7469 6e67 2061   in WGD dating a
+0000fdd0: 7265 2069 6d70 6c65 6d65 6e74 6564 2069  re implemented i
+0000fde0: 6e20 6077 6764 2076 3260 2e20 5573 6572  n `wgd v2`. User
+0000fdf0: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
+0000fe00: 6520 7468 6520 616e 6368 6f72 2070 6f69  e the anchor poi
+0000fe10: 6e74 732c 2073 6567 6d65 6e74 732c 206c  nts, segments, l
+0000fe20: 6973 7473 6567 6d65 6e74 732c 206d 756c  istsegments, mul
+0000fe30: 7469 706c 6963 6f6e 7061 6972 7320 6461  tipliconpairs da
+0000fe40: 7461 6669 6c65 2066 726f 6d20 6069 2d61  tafile from `i-a
+0000fe50: 6468 6f72 6560 2074 6f20 6163 6869 6576  dhore` to achiev
+0000fe60: 6520 7468 6520 636c 7573 7465 7269 6e67  e the clustering
+0000fe70: 2066 756e 6374 696f 6e2e 2053 6f6d 6520   function. Some 
+0000fe80: 7061 7261 6d65 7465 7273 2074 6861 7420  parameters that 
+0000fe90: 6361 6e20 696d 7061 6374 2074 6865 2072  can impact the r
+0000fea0: 6573 756c 7473 2069 6e63 6c75 6465 2060  esults include `
+0000feb0: 2d2d 616c 6967 6e66 696c 7465 7260 2c20  --alignfilter`, 
+0000fec0: 7768 6963 6820 6669 6c74 6572 7320 7468  which filters th
+0000fed0: 6520 6461 7461 2062 6173 6564 206f 6e20  e data based on 
+0000fee0: 616c 6967 6e6d 656e 7420 6964 656e 7469  alignment identi
+0000fef0: 7479 2c20 6c65 6e67 7468 2061 6e64 2063  ty, length and c
+0000ff00: 6f76 6572 6167 652c 2060 2d2d 6b73 7261  overage, `--ksra
+0000ff10: 6e67 6560 2c20 7768 6963 6820 7365 7473  nge`, which sets
+0000ff20: 2074 6865 2072 616e 6765 206f 6620 4b73   the range of Ks
+0000ff30: 2074 6f20 6265 2061 6e61 6c79 7a65 642c   to be analyzed,
+0000ff40: 2060 2d2d 6269 6e5f 7769 6474 6860 2c20   `--bin_width`, 
+0000ff50: 7768 6963 6820 7365 7473 2074 6865 2062  which sets the b
+0000ff60: 616e 6477 6964 7468 206f 6620 2a4b 2a3c  andwidth of *K*<
+0000ff70: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+0000ff80: 7269 6275 7469 6f6e 2c20 602d 2d77 6569  ribution, `--wei
+0000ff90: 6768 7473 5f6f 7574 6c69 6572 735f 696e  ghts_outliers_in
+0000ffa0: 636c 7564 6564 6020 7768 6963 6820 6465  cluded` which de
+0000ffb0: 7465 726d 696e 6573 2077 6865 7468 6572  termines whether
+0000ffc0: 2074 6f20 696e 636c 7564 6520 2a4b 2a3c   to include *K*<
+0000ffd0: 7375 623e 533c 2f73 7562 3e20 6f75 746c  sub>S</sub> outl
+0000ffe0: 6965 7273 2028 7768 6f73 6520 7661 6c75  iers (whose valu
+0000fff0: 6520 6973 206f 7665 7220 3529 2069 6e20  e is over 5) in 
+00010000: 616e 616c 7973 6973 2c20 602d 2d6d 6574  analysis, `--met
+00010010: 686f 6460 2077 6869 6368 2064 6574 6572  hod` which deter
+00010020: 6d69 6e65 7320 7768 6963 6820 636c 7573  mines which clus
+00010030: 7465 7269 6e67 206d 6574 686f 6420 746f  tering method to
+00010040: 2075 7365 2028 6465 6661 756c 7420 676d   use (default gm
+00010050: 6d29 2c20 602d 2d73 6565 6460 2077 6869  m), `--seed` whi
+00010060: 6368 2073 6574 7320 7468 6520 7261 6e64  ch sets the rand
+00010070: 6f6d 2073 6565 6420 6769 7665 6e20 746f  om seed given to
+00010080: 2069 6e69 7469 616c 697a 6174 696f 6e20   initialization 
+00010090: 2864 6566 6175 6c74 2032 3335 3238 3930  (default 2352890
+000100a0: 292c 2060 2d2d 6e5f 696e 6974 602c 2077  ), `--n_init`, w
+000100b0: 6869 6368 2073 6574 7320 7468 6520 6e75  hich sets the nu
+000100c0: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
+000100d0: 696e 6974 6961 6c69 7a61 7469 6f6e 7320  initializations 
+000100e0: 2864 6566 6175 6c74 2032 3030 292c 2060  (default 200), `
+000100f0: 2d2d 656d 5f69 7465 7260 2c20 7768 6963  --em_iter`, whic
+00010100: 6820 7365 7473 2074 6865 206d 6178 696d  h sets the maxim
+00010110: 756d 206e 756d 6265 7220 6f66 2069 7465  um number of ite
+00010120: 7261 7469 6f6e 7320 2864 6566 6175 6c74  rations (default
+00010130: 2032 3030 292c 2060 2d2d 6761 6d6d 6160   200), `--gamma`
+00010140: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
+00010150: 2067 616d 6d61 2070 6172 616d 6574 6572   gamma parameter
+00010160: 2066 6f72 2074 6865 2062 676d 6d20 6d6f   for the bgmm mo
+00010170: 6465 6c20 2864 6566 6175 6c74 2030 2e30  del (default 0.0
+00010180: 3031 292c 2060 2d2d 636f 6d70 6f6e 656e  01), `--componen
+00010190: 7473 602c 2077 6869 6368 2073 6574 7320  ts`, which sets 
+000101a0: 7468 6520 7261 6e67 6520 6f66 2074 6865  the range of the
+000101b0: 206e 756d 6265 7220 6f66 2063 6f6d 706f   number of compo
+000101c0: 6e65 6e74 7320 746f 2066 6974 2028 6465  nents to fit (de
+000101d0: 6661 756c 7420 3120 3429 2c20 602d 2d77  fault 1 4), `--w
+000101e0: 6569 6768 7465 6460 2077 6869 6368 2064  eighted` which d
+000101f0: 6574 6572 6d69 6e65 7320 7768 6574 6865  etermines whethe
+00010200: 7220 746f 2075 7365 206e 6f64 652d 7765  r to use node-we
+00010210: 6967 6874 6564 206d 6574 686f 6420 666f  ighted method fo
+00010220: 7220 6465 2d72 6564 756e 6461 6e63 792c  r de-redundancy,
+00010230: 2060 2d2d 6775 6964 6560 2077 6869 6368   `--guide` which
+00010240: 2064 6574 6572 6d69 6e65 7320 7768 6963   determines whic
+00010250: 6820 7265 6769 6d65 2072 6573 6964 696e  h regime residin
+00010260: 6720 616e 6368 6f72 7320 746f 2062 6520  g anchors to be 
+00010270: 7573 6564 2028 6465 6661 756c 7420 5365  used (default Se
+00010280: 676d 656e 7429 2c20 602d 2d70 726f 6d69  gment), `--promi
+00010290: 6e65 6e63 655f 6375 746f 6666 6020 7768  nence_cutoff` wh
+000102a0: 6963 6820 7365 7473 2074 6865 2070 726f  ich sets the pro
+000102b0: 6d69 6e65 6e63 6520 6375 746f 6666 206f  minence cutoff o
+000102c0: 6620 6163 6365 7074 6162 6c65 2070 6561  f acceptable pea
+000102d0: 6b73 2069 6e20 7065 616b 2066 696e 6469  ks in peak findi
+000102e0: 6e67 2070 726f 6365 7373 2c20 602d 2d72  ng process, `--r
+000102f0: 656c 5f68 6569 6768 7460 2077 6869 6368  el_height` which
+00010300: 2073 6574 7320 7468 6520 7265 6c61 7469   sets the relati
+00010310: 7665 2068 6569 6768 7420 6174 2077 6869  ve height at whi
+00010320: 6368 2074 6865 2070 6561 6b20 7769 6474  ch the peak widt
+00010330: 6820 6973 206d 6561 7375 7265 642c 2060  h is measured, `
+00010340: 2d2d 6b73 746f 6461 7465 6020 7768 6963  --kstodate` whic
+00010350: 6820 6d61 6e75 616c 6c79 2073 6574 7320  h manually sets 
+00010360: 7468 6520 7261 6e67 6520 6f66 202a 4b2a  the range of *K*
+00010370: 3c73 7562 3e53 3c2f 7375 623e 2074 6f20  <sub>S</sub> to 
+00010380: 6265 2064 6174 6564 2069 6e20 6865 7572  be dated in heur
+00010390: 6973 7469 6320 7365 6172 6368 2061 6e64  istic search and
+000103a0: 206e 6565 6473 2074 6f20 6265 2063 6f2d   needs to be co-
+000103b0: 7365 7420 7769 7468 206f 7074 696f 6e20  set with option 
+000103c0: 602d 2d6d 616e 7561 6c73 6574 602c 2060  `--manualset`, `
+000103d0: 2d2d 786c 696d 6020 616e 6420 602d 2d79  --xlim` and `--y
+000103e0: 6c69 6d60 2064 6574 6572 6d69 6e69 6e67  lim` determining
+000103f0: 2074 6865 2078 2061 6e64 2079 2061 7869   the x and y axi
+00010400: 7320 6c69 6d69 7420 6f66 2047 4d4d 202a  s limit of GMM *
+00010410: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00010420: 6973 7472 6962 7574 696f 6e2c 2060 2d2d  istribution, `--
+00010430: 6369 6020 7365 7474 696e 6720 7468 6520  ci` setting the 
+00010440: 636f 6e66 6964 656e 6365 206c 6576 656c  confidence level
+00010450: 206f 6620 6c6f 672d 6e6f 726d 616c 2064   of log-normal d
+00010460: 6973 7472 6962 7574 696f 6e20 746f 2064  istribution to d
+00010470: 6174 6520 2864 6566 6175 6c74 2039 3529  ate (default 95)
+00010480: 2c20 602d 2d68 6472 6020 7365 7474 696e  , `--hdr` settin
+00010490: 6720 7468 6520 6869 6768 6573 7420 6465  g the highest de
+000104a0: 6e73 6974 7920 7265 6769 6f6e 2028 4844  nsity region (HD
+000104b0: 5229 2061 7070 6c69 6564 2069 6e20 7468  R) applied in th
+000104c0: 6520 7365 676d 656e 742d 6775 6964 6564  e segment-guided
+000104d0: 2061 6e63 686f 7220 7061 6972 202a 4b2a   anchor pair *K*
+000104e0: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+000104f0: 7472 6962 7574 696f 6e20 2864 6566 6175  tribution (defau
+00010500: 6c74 2039 3529 2c20 602d 2d68 6575 7269  lt 95), `--heuri
+00010510: 7374 6963 6020 6465 7465 726d 696e 696e  stic` determinin
+00010520: 6720 7768 6574 6865 7220 746f 2069 6e69  g whether to ini
+00010530: 7469 6174 6520 6865 7572 6973 7469 6320  tiate heuristic 
+00010540: 6d65 7468 6f64 206f 6620 6465 6669 6e69  method of defini
+00010550: 6e67 2043 4920 666f 7220 6461 7469 6e67  ng CI for dating
+00010560: 2c20 602d 2d6b 7363 7574 6f66 6660 2073  , `--kscutoff` s
+00010570: 6574 7469 6e67 2074 6865 202a 4b2a 3c73  etting the *K*<s
+00010580: 7562 3e53 3c2f 7375 623e 2073 6174 7572  ub>S</sub> satur
+00010590: 6174 696f 6e20 6375 746f 6666 2069 6e20  ation cutoff in 
+000105a0: 6461 7469 6e67 2028 6465 6661 756c 7420  dating (default 
+000105b0: 3529 2e0a 0a46 6f75 7220 7265 7375 6c74  5)...Four result
+000105c0: 2073 7562 666c 6f64 6572 7320 7769 6c6c   subfloders will
+000105d0: 2062 6520 7072 6f64 7563 6564 2c20 6e61   be produced, na
+000105e0: 6d65 6c79 2060 416e 6368 6f72 4b73 5f46  mely `AnchorKs_F
+000105f0: 696e 6450 6561 6b60 2c20 6041 6e63 686f  indPeak`, `Ancho
+00010600: 724b 735f 474d 4d60 2c20 6053 6567 6d65  rKs_GMM`, `Segme
+00010610: 6e74 4775 6964 654b 735f 474d 4d60 2061  ntGuideKs_GMM` a
+00010620: 6e64 2060 5365 676d 656e 744b 735f 4669  nd `SegmentKs_Fi
+00010630: 6e64 5065 616b 602e 2054 6865 2060 416e  ndPeak`. The `An
+00010640: 6368 6f72 4b73 5f46 696e 6450 6561 6b60  chorKs_FindPeak`
+00010650: 2073 7562 666c 6f64 6572 2063 6f6e 7461   subfloder conta
+00010660: 696e 7320 7265 7375 6c74 7320 6f66 2074  ins results of t
+00010670: 6865 2064 6574 6563 7465 6420 7065 616b  he detected peak
+00010680: 7320 6279 2074 6865 2060 7369 676e 616c  s by the `signal
+00010690: 6020 6d6f 6475 6c65 206f 6620 6053 6369  ` module of `Sci
+000106a0: 5079 6020 6c69 6272 6172 7920 616e 6420  Py` library and 
+000106b0: 7468 6520 6173 7375 6d65 6420 6869 6768  the assumed high
+000106c0: 6573 7420 6d61 7373 2070 6172 7420 2872  est mass part (r
+000106d0: 6566 6572 7265 6420 746f 2061 7320 4869  eferred to as Hi
+000106e0: 6768 4d61 7373 2068 6572 6561 6674 6572  ghMass hereafter
+000106f0: 2920 6f66 2065 6163 6820 7065 616b 2c20  ) of each peak, 
+00010700: 7768 6963 6820 6361 6e20 6265 2075 7365  which can be use
+00010710: 6420 666f 7220 6675 7274 6865 7220 5747  d for further WG
+00010720: 4420 6461 7469 6e67 2e20 5468 6520 6041  D dating. The `A
+00010730: 6e63 686f 724b 735f 474d 4d60 2073 686f  nchorKs_GMM` sho
+00010740: 7773 2074 6865 2047 4d4d 2072 6573 756c  ws the GMM resul
+00010750: 7473 2075 706f 6e20 7468 6520 6f72 6967  ts upon the orig
+00010760: 696e 616c 2061 6e63 686f 7220 2a4b 2a3c  inal anchor *K*<
+00010770: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00010780: 7269 6275 7469 6f6e 2062 7920 7468 6520  ribution by the 
+00010790: 606d 6978 7475 7265 6020 6d6f 6475 6c65  `mixture` module
+000107a0: 206f 6620 6073 6369 6b69 742d 6c65 6172   of `scikit-lear
+000107b0: 6e60 206c 6962 7261 7279 2061 6e64 2074  n` library and t
+000107c0: 776f 2073 7562 666c 6f64 6572 732c 2060  wo subfloders, `
+000107d0: 4c6f 6747 4d4d 5f43 4960 2063 6f6e 7461  LogGMM_CI` conta
+000107e0: 696e 696e 6720 7468 6520 7265 7375 6c74  ining the result
+000107f0: 7320 6f66 2039 3525 2043 4920 6f66 2065  s of 95% CI of e
+00010800: 6163 6820 636f 6d70 6f6e 656e 742c 2060  ach component, `
+00010810: 4869 6768 4d61 7373 5f43 4960 2063 6f6e  HighMass_CI` con
+00010820: 7461 696e 696e 6720 7468 6520 4869 6768  taining the High
+00010830: 4d61 7373 206f 6620 6561 6368 2063 6f6d  Mass of each com
+00010840: 706f 6e65 6e74 2c20 7768 6963 6820 6361  ponent, which ca
+00010850: 6e20 6265 2075 7365 6420 666f 7220 6675  n be used for fu
+00010860: 7274 6865 7220 5747 4420 6461 7469 6e67  rther WGD dating
+00010870: 2e20 5468 6520 6053 6567 6d65 6e74 4775  . The `SegmentGu
+00010880: 6964 654b 735f 474d 4d60 2073 7562 666f  ideKs_GMM` subfo
+00010890: 6c64 6572 2070 7265 7365 6e74 7320 7265  lder presents re
+000108a0: 7375 6c74 7320 6f66 2073 6567 6d65 6e74  sults of segment
+000108b0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000108c0: 2047 4d4d 2077 6869 6368 2061 7265 206d   GMM which are m
+000108d0: 6170 7065 6420 6261 636b 2074 6f20 7468  apped back to th
+000108e0: 6520 7265 7369 6469 6e67 2061 6e63 686f  e residing ancho
+000108f0: 7220 7061 6972 7320 616e 6420 7468 6520  r pairs and the 
+00010900: 6173 736f 6369 6174 6564 2039 3525 2048  associated 95% H
+00010910: 4452 2061 6e64 2048 6967 684d 6173 7320  DR and HighMass 
+00010920: 6f66 2065 6163 6820 7365 676d 656e 7420  of each segment 
+00010930: 636c 7573 7465 7220 696e 2073 7562 666c  cluster in subfl
+00010940: 6f64 6572 7320 6f66 2060 4844 525f 4349  oders of `HDR_CI
+00010950: 6020 616e 6420 6048 6967 684d 6173 735f  ` and `HighMass_
+00010960: 4349 602e 2054 6865 2060 5365 676d 656e  CI`. The `Segmen
+00010970: 744b 735f 4669 6e64 5065 616b 6020 7375  tKs_FindPeak` su
+00010980: 6266 6f6c 6465 7220 6973 2073 696d 696c  bfolder is simil
+00010990: 6172 2074 6f20 6041 6e63 686f 724b 735f  ar to `AnchorKs_
+000109a0: 4669 6e64 5065 616b 6020 6275 7420 7769  FindPeak` but wi
+000109b0: 7468 2073 6567 6d65 6e74 202a 4b2a 3c73  th segment *K*<s
+000109c0: 7562 3e53 3c2f 7375 623e 2069 6e73 7465  ub>S</sub> inste
+000109d0: 6164 2e20 5468 6520 2a4b 2a3c 7375 623e  ad. The *K*<sub>
+000109e0: 533c 2f73 7562 3e20 696e 2060 4d75 6c74  S</sub> in `Mult
+000109f0: 6970 6c69 636f 6e60 2063 616e 2061 6c73  iplicon` can als
+00010a00: 6f20 6265 2063 616c 6375 6c61 7465 6420  o be calculated 
+00010a10: 696e 2070 6c61 6365 206f 6620 6053 6567  in place of `Seg
+00010a20: 6d65 6e74 6020 7573 696e 6720 7468 6520  ment` using the 
+00010a30: 6f70 7469 6f6e 2060 2d2d 6775 6964 6560  option `--guide`
+00010a40: 2061 7320 7375 6368 2074 6865 2072 6573   as such the res
+00010a50: 756c 7420 7469 746c 652c 206c 6162 656c  ult title, label
+00010a60: 2c20 6669 6c65 2061 6e64 2066 6f6c 6465  , file and folde
+00010a70: 7220 6e61 6d65 7320 7769 6c6c 2062 6520  r names will be 
+00010a80: 6368 616e 6765 6420 6163 636f 7264 696e  changed accordin
+00010a90: 676c 792e 0a0a 2a2a 4120 7375 6767 6573  gly...**A sugges
+00010aa0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
+00010ab0: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
+00010ac0: 2073 696d 706c 7920 6173 2062 656c 6f77   simply as below
+00010ad0: 2a2a 0a0a 6060 600a 7767 6420 7065 616b  **..```.wgd peak
+00010ae0: 206b 7364 6174 6120 2d61 7020 6170 6461   ksdata -ap apda
+00010af0: 7461 202d 736d 2073 6d64 6174 6120 2d6c  ta -sm smdata -l
+00010b00: 6520 6c65 6461 7461 202d 6d70 206d 7064  e ledata -mp mpd
+00010b10: 6174 6120 2d2d 6865 7572 6973 7469 630a  ata --heuristic.
+00010b20: 6060 600a 0a23 2323 2077 6764 2073 796e  ```..### wgd syn
+00010b30: 0a0a 2a2a 5468 6520 696e 7472 612d 7370  ..**The intra-sp
+00010b40: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
+00010b50: 6e66 6572 656e 6365 2a2a 0a60 6060 0a77  nference**.```.w
+00010b60: 6764 2073 796e 2066 616d 696c 6965 7320  gd syn families 
+00010b70: 6766 6620 282d 2d6b 735f 6469 7374 7269  gff (--ks_distri
+00010b80: 6275 7469 6f6e 206b 7364 6174 6120 2d66  bution ksdata -f
+00010b90: 2067 656e 6520 2d61 2049 4420 2d2d 6d69   gene -a ID --mi
+00010ba0: 6e6c 656e 202d 3120 2d2d 6d69 6e73 6567  nlen -1 --minseg
+00010bb0: 6c65 6e20 3130 3030 3020 2d2d 6d69 6e67  len 10000 --ming
+00010bc0: 656e 656e 756d 2033 3029 0a60 6060 0a0a  enenum 30).```..
+00010bd0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
+00010be0: 2073 796e 6020 6973 206d 6169 6e6c 7920   syn` is mainly 
+00010bf0: 6465 616c 696e 6720 7769 7468 2063 6f6c  dealing with col
+00010c00: 6c69 6e65 6172 6974 7920 6f72 2073 796e  linearity or syn
+00010c10: 7465 6e79 2028 626f 7468 2072 6566 6572  teny (both refer
+00010c20: 7265 6420 746f 2061 7320 7379 6e74 656e  red to as synten
+00010c30: 7920 6865 7265 6166 7465 7229 2061 6e61  y hereafter) ana
+00010c40: 6c79 7369 732e 2054 776f 2069 6e70 7574  lysis. Two input
+00010c50: 2066 696c 6573 2061 7265 2065 7373 656e   files are essen
+00010c60: 7469 616c 2c20 7468 6520 6765 6e65 2066  tial, the gene f
+00010c70: 616d 696c 7920 6669 6c65 2061 6e64 2074  amily file and t
+00010c80: 6865 2067 6666 3320 6669 6c65 2e20 5468  he gff3 file. Th
+00010c90: 6520 6765 6e65 2066 616d 696c 7920 6669  e gene family fi
+00010ca0: 6c65 2069 7320 696e 2074 6865 2066 6f72  le is in the for
+00010cb0: 6d61 7420 6173 2060 4f72 7468 6f46 696e  mat as `OrthoFin
+00010cc0: 6465 7260 2e20 5468 6520 736f 6674 7761  der`. The softwa
+00010cd0: 7265 2060 692d 6164 686f 7265 6020 6973  re `i-adhore` is
+00010ce0: 2061 2070 7265 7265 7175 6973 6974 652e   a prerequisite.
+00010cf0: 2057 6974 6820 6465 6661 756c 7420 7061   With default pa
+00010d00: 7261 6d65 7465 7273 2c20 7468 6520 7072  rameters, the pr
+00010d10: 6f67 7261 6d20 6261 7369 6361 6c6c 7920  ogram basically 
+00010d20: 636f 6e64 7563 7473 2031 2920 6669 6c74  conducts 1) filt
+00010d30: 6572 696e 6720 6765 6e65 2066 616d 696c  ering gene famil
+00010d40: 6965 7320 6261 7365 6420 6f6e 206d 6178  ies based on max
+00010d50: 696d 756d 2066 616d 696c 7920 7369 7a65  imum family size
+00010d60: 2032 2920 7265 7472 6965 7669 6e67 2067   2) retrieving g
+00010d70: 656e 6520 706f 7369 7469 6f6e 2061 6e64  ene position and
+00010d80: 2073 6361 6666 6f6c 6420 696e 666f 726d   scaffold inform
+00010d90: 6174 696f 6e20 6672 6f6d 2067 6666 3320  ation from gff3 
+00010da0: 6669 6c65 2033 2920 7072 6f64 7563 696e  file 3) producin
+00010db0: 6720 7468 6520 636f 6e66 6967 7572 6174  g the configurat
+00010dc0: 696f 6e20 6669 6c65 2061 6e64 2061 7373  ion file and ass
+00010dd0: 6f63 6961 7465 6420 6461 7461 6669 6c65  ociated datafile
+00010de0: 7320 666f 7220 6069 2d61 6468 6f72 6560  s for `i-adhore`
+00010df0: 2034 2920 6361 6c6c 696e 6720 6069 2d61   4) calling `i-a
+00010e00: 6468 6f72 6560 2067 6976 656e 2074 6865  dhore` given the
+00010e10: 2070 6172 616d 6574 6572 7320 7365 7420   parameters set 
+00010e20: 746f 2069 6e66 6572 2073 796e 7465 6e79  to infer synteny
+00010e30: 2035 2920 7669 7375 616c 697a 696e 6720   5) visualizing 
+00010e40: 7468 6520 7379 6e74 656e 7920 696e 2022  the synteny in "
+00010e50: 646f 7470 6c6f 7422 2069 6e20 7468 6520  dotplot" in the 
+00010e60: 756e 6974 206f 6620 6765 6e65 7320 616e  unit of genes an
+00010e70: 6420 6261 7365 732c 2069 6e20 2253 796e  d bases, in "Syn
+00010e80: 6465 7074 6822 2070 6c6f 7420 7368 6f77  depth" plot show
+00010e90: 696e 6720 7468 6520 6469 7374 7269 6275  ing the distribu
+00010ea0: 7469 6f6e 206f 6620 6469 6666 6572 656e  tion of differen
+00010eb0: 7420 6361 7465 676f 7269 6573 206f 6620  t categories of 
+00010ec0: 636f 6c6c 696e 6561 7269 7479 2072 6174  collinearity rat
+00010ed0: 696f 7320 7769 7468 696e 2061 6e64 2062  ios within and b
+00010ee0: 6574 7765 656e 2073 7065 6369 6573 2c20  etween species, 
+00010ef0: 696e 2022 6475 7053 7461 636b 2220 706c  in "dupStack" pl
+00010f00: 6f74 2073 686f 7769 6e67 206d 756c 7469  ot showing multi
+00010f10: 706c 6963 6f6e 7320 7769 7468 2064 6966  plicons with dif
+00010f20: 6665 7265 6e74 206d 756c 7469 706c 6963  ferent multiplic
+00010f30: 6174 696f 6e20 6c65 7665 6c73 2e20 3629  ation levels. 6)
+00010f40: 2069 6620 7769 7468 202a 4b2a 3c73 7562   if with *K*<sub
+00010f50: 3e53 3c2f 7375 623e 2064 6174 612c 2061  >S</sub> data, a
+00010f60: 2022 2a4b 2a3c 7375 623e 533c 2f73 7562   "*K*<sub>S</sub
+00010f70: 3e20 646f 7470 6c6f 7422 2077 6974 6820  > dotplot" with 
+00010f80: 646f 7473 2061 6e6e 6f74 6174 6564 2069  dots annotated i
+00010f90: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+00010fa0: 3e20 7661 6c75 6573 2061 6e64 2061 202a  > values and a *
+00010fb0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00010fc0: 6973 7472 6962 7574 696f 6e20 7769 7468  istribution with
+00010fd0: 2061 6e63 686f 7220 7061 6972 7320 6465   anchor pairs de
+00010fe0: 6e6f 7465 6420 7769 6c6c 2062 6520 7072  noted will be pr
+00010ff0: 6f64 7563 6564 2e20 5468 6520 6765 6e65  oduced. The gene
+00011000: 2069 6e66 6f72 6d61 7469 6f6e 2069 6e20   information in 
+00011010: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
+00011020: 6669 6c65 2061 6e64 2067 6666 3320 6669  file and gff3 fi
+00011030: 6c65 2073 686f 756c 6420 6265 206d 6174  le should be mat
+00011040: 6368 6564 2077 6869 6368 2072 6571 7569  ched which requi
+00011050: 7265 7320 7573 6572 7320 746f 2073 6574  res users to set
+00011060: 2070 726f 7065 7220 602d 2d66 6561 7475   proper `--featu
+00011070: 7265 6020 616e 6420 602d 2d61 7474 7269  re` and `--attri
+00011080: 6275 7465 602e 2054 6865 206d 6178 696d  bute`. The maxim
+00011090: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
+000110a0: 6f20 6265 2069 6e63 6c75 6465 6420 6361  o be included ca
+000110b0: 6e20 6265 2073 6574 2076 6961 2074 6865  n be set via the
+000110c0: 206f 7074 696f 6e20 602d 2d6d 6178 7369   option `--maxsi
+000110d0: 7a65 602c 206e 6f74 6564 2074 6861 7420  ze`, noted that 
+000110e0: 7468 6973 2066 696c 7465 7269 6e67 2069  this filtering i
+000110f0: 7320 6d61 696e 6c79 2074 6f20 6472 6f70  s mainly to drop
+00011100: 2074 686f 7365 2068 7567 6520 7461 6e64   those huge tand
+00011110: 656d 2064 7570 6c69 6361 7465 7320 6661  em duplicates fa
+00011120: 6d69 6c79 2061 6e64 2074 7261 6e73 706f  mily and transpo
+00011130: 7361 626c 6520 656c 656d 656e 7473 2028  sable elements (
+00011140: 5445 7329 2066 616d 696c 792c 2061 6e64  TEs) family, and
+00011150: 206e 6f74 206d 616e 6461 746f 7279 2e20   not mandatory. 
+00011160: 5573 6572 7320 6361 6e20 6669 6c74 6572  Users can filter
+00011170: 2074 686f 7365 2066 7261 676d 656e 7461   those fragmenta
+00011180: 7279 2073 6361 6666 6f6c 6473 2076 6961  ry scaffolds via
+00011190: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
+000111a0: 696e 6c65 6e60 2e20 5468 6520 6d69 6e69  inlen`. The mini
+000111b0: 6d75 6d20 6c65 6e67 7468 2061 6e64 206e  mum length and n
+000111c0: 756d 6265 7220 6f66 2067 656e 6573 2066  umber of genes f
+000111d0: 6f72 2061 2073 6567 6d65 6e74 2074 6f20  or a segment to 
+000111e0: 6265 2063 6f6e 7369 6465 7265 6420 6361  be considered ca
+000111f0: 6e20 6265 2073 6574 2076 6961 2074 6865  n be set via the
+00011200: 206f 7074 696f 6e20 602d 2d6d 696e 7365   option `--minse
+00011210: 676c 656e 6020 616e 6420 602d 2d6d 696e  glen` and `--min
+00011220: 6765 6e65 6e75 6d60 2e20 5265 6475 6e64  genenum`. Redund
+00011230: 616e 7420 6d75 6c74 6970 6c69 636f 6e73  ant multiplicons
+00011240: 2063 616e 2062 6520 6b65 7074 2062 7920   can be kept by 
+00011250: 7365 7420 7468 6520 666c 6167 206f 7074  set the flag opt
+00011260: 696f 6e20 602d 2d6b 6565 7072 6564 756e  ion `--keepredun
+00011270: 602e 0a0a 2a2a 4120 7375 6767 6573 7465  `...**A suggeste
+00011280: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
+00011290: 616e 2075 7365 2063 6f6d 6d61 6e64 2073  an use command s
+000112a0: 696d 706c 7920 6173 2062 656c 6f77 2a2a  imply as below**
+000112b0: 0a0a 6060 600a 7767 6420 7379 6e20 6661  ..```.wgd syn fa
+000112c0: 6d69 6c69 6573 2067 6666 0a60 6060 0a0a  milies gff.```..
+000112d0: 2a2a 5468 6520 696e 7465 722d 7370 6563  **The inter-spec
+000112e0: 6966 6963 2073 796e 7465 6e79 2069 6e66  ific synteny inf
+000112f0: 6572 656e 6365 2a2a 0a60 6060 0a77 6764  erence**.```.wgd
+00011300: 2073 796e 2066 616d 696c 6965 7320 6766   syn families gf
+00011310: 6631 2067 6666 3220 282d 2d61 6464 6974  f1 gff2 (--addit
+00011320: 696f 6e61 6c67 6666 696e 666f 2022 6d52  ionalgffinfo "mR
+00011330: 4e41 3b4e 616d 6522 202d 2d61 6464 6974  NA;Name" --addit
+00011340: 696f 6e61 6c67 6666 696e 666f 2022 6765  ionalgffinfo "ge
+00011350: 6e65 3b49 4422 290a 6060 600a 0a46 6f72  ne;ID").```..For
+00011360: 206d 756c 7469 2d73 7065 6369 6573 2073   multi-species s
+00011370: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
+00011380: 2c20 6966 2075 7365 7273 2068 6176 6520  , if users have 
+00011390: 6766 6633 2066 696c 6573 2077 6869 6368  gff3 files which
+000113a0: 2068 6176 6520 6469 6666 6572 656e 7420   have different 
+000113b0: 6665 6174 7572 6573 206f 7220 6174 7472  features or attr
+000113c0: 6962 7574 6573 2066 6f72 2067 656e 6520  ibutes for gene 
+000113d0: 706f 7369 7469 6f6e 2069 6e66 6f72 6d61  position informa
+000113e0: 7469 6f6e 2072 6574 7269 6576 616c 2c20  tion retrieval, 
+000113f0: 7468 6520 6f70 7469 6f6e 2060 2d2d 6164  the option `--ad
+00011400: 6469 7469 6f6e 616c 6766 6669 6e66 6f60  ditionalgffinfo`
+00011410: 2063 616e 2062 6520 7365 7420 746f 2070   can be set to p
+00011420: 726f 7669 6465 2074 6865 2061 6464 6974  rovide the addit
+00011430: 696f 6e61 6c20 696e 666f 726d 6174 696f  ional informatio
+00011440: 6e2e 2054 6865 2072 656d 6169 6e69 6e67  n. The remaining
+00011450: 2070 6172 616d 6574 6572 2073 6574 7469   parameter setti
+00011460: 6e67 2069 7320 7468 6520 7361 6d65 2061  ng is the same a
+00011470: 7320 7468 6520 696e 7472 612d 7370 6563  s the intra-spec
+00011480: 6966 6963 2073 796e 7465 6e79 2069 6e66  ific synteny inf
+00011490: 6572 656e 6365 2e0a 0a2a 2a41 2073 7567  erence...**A sug
+000114a0: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+000114b0: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+000114c0: 616e 6420 7369 6d70 6c79 2061 7320 6265  and simply as be
+000114d0: 6c6f 772a 2a0a 0a60 6060 0a77 6764 2073  low**..```.wgd s
+000114e0: 796e 2066 616d 696c 6965 7320 6766 6631  yn families gff1
+000114f0: 2067 6666 320a 6060 600a 0a23 2323 2077   gff2.```..### w
+00011500: 6764 2076 697a 0a0a 2a2a 5468 6520 7669  gd viz..**The vi
+00011510: 7375 616c 697a 6174 696f 6e20 6f66 202a  sualization of *
+00011520: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00011530: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00011540: 616e 6420 454c 4d4d 2061 6e61 6c79 7369  and ELMM analysi
+00011550: 732a 2a0a 6060 600a 7767 6420 7669 7a20  s**.```.wgd viz 
+00011560: 2d64 206b 7364 6174 610a 6060 600a 0a54  -d ksdata.```..T
+00011570: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+00011580: 7669 7a60 2069 7320 6d61 696e 6c79 2066  viz` is mainly f
+00011590: 6f72 2074 6865 2070 7572 706f 7365 206f  or the purpose o
+000115a0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+000115b0: 3e20 6469 7374 7269 6275 7469 6f6e 2061  > distribution a
+000115c0: 6e64 2073 796e 7465 6e79 2076 6973 7561  nd synteny visua
+000115d0: 6c69 7a61 7469 6f6e 2c20 7769 7468 2073  lization, with s
+000115e0: 6f6d 6520 6f70 7469 6f6e 616c 206d 6978  ome optional mix
+000115f0: 7475 7265 206d 6f64 656c 696e 6720 616e  ture modeling an
+00011600: 616c 7973 6973 2e20 5468 6520 6261 7369  alysis. The basi
+00011610: 6320 6675 6e63 7469 6f6e 2069 7320 6a75  c function is ju
+00011620: 7374 2074 6f20 706c 6f74 2074 6865 202a  st to plot the *
+00011630: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00011640: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+00011650: 636f 6e64 7563 7420 616e 2045 4c4d 4d20  conduct an ELMM 
+00011660: 616e 616c 7973 6973 2069 6e20 7365 6172  analysis in sear
+00011670: 6368 206f 6620 706f 7465 6e74 6961 6c20  ch of potential 
+00011680: 5747 4420 636f 6d70 6f6e 656e 7473 2e20  WGD components. 
+00011690: 536f 6d65 206b 6579 2070 6172 616d 6574  Some key paramet
+000116a0: 6572 7320 6166 6665 6374 696e 6720 7468  ers affecting th
+000116b0: 6520 454c 4d4d 2072 6573 756c 7420 696e  e ELMM result in
+000116c0: 636c 7564 6520 602d 2d70 726f 6d69 6e65  clude `--promine
+000116d0: 6e63 655f 6375 746f 6666 602c 2060 2d2d  nce_cutoff`, `--
+000116e0: 7265 6c5f 6865 6967 6874 602c 2060 2d2d  rel_height`, `--
+000116f0: 656d 5f69 7465 7261 7469 6f6e 7360 2061  em_iterations` a
+00011700: 6e64 2060 2d2d 656d 5f69 6e69 7469 616c  nd `--em_initial
+00011710: 697a 6174 696f 6e73 602c 2061 6c6c 206f  izations`, all o
+00011720: 6620 7768 6963 6820 6861 7665 2062 6565  f which have bee
+00011730: 6e20 6578 706c 6169 6e65 6420 6168 6561  n explained ahea
+00011740: 642e 0a0a 2a2a 4120 7375 6767 6573 7465  d...**A suggeste
+00011750: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
+00011760: 616e 2075 7365 2063 6f6d 6d61 6e64 2073  an use command s
+00011770: 696d 706c 7920 6173 2062 656c 6f77 2a2a  imply as below**
+00011780: 0a0a 6060 600a 7767 6420 7669 7a20 2d64  ..```.wgd viz -d
+00011790: 206b 7364 6174 610a 6060 600a 0a2a 2a54   ksdata.```..**T
+000117a0: 6865 2076 6973 7561 6c69 7a61 7469 6f6e  he visualization
+000117b0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
+000117c0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+000117d0: 7469 6f6e 2077 6974 6820 7261 7465 2063  tion with rate c
+000117e0: 6f72 7265 6374 696f 6e2a 2a0a 6060 600a  orrection**.```.
+000117f0: 7767 6420 7669 7a20 2d64 206b 7364 6174  wgd viz -d ksdat
+00011800: 6120 2d73 7020 7370 6461 7461 202d 2d66  a -sp spdata --f
+00011810: 6f63 7573 3261 6c6c 2066 6f63 616c 5f73  ocus2all focal_s
+00011820: 7065 6369 6573 202d 2d65 7874 7261 7061  pecies --extrapa
+00011830: 7261 6e6f 6d65 6b73 206b 7364 6174 6120  ranomeks ksdata 
+00011840: 282d 2d61 6e63 686f 7270 6f69 6e74 7320  (--anchorpoints 
+00011850: 6170 6461 7461 202d 2d70 6c6f 7461 7067  apdata --plotapg
+00011860: 6d6d 202d 2d70 6c6f 7465 6c6d 6d29 0a60  mm --plotelmm).`
+00011870: 6060 0a0a 4265 7369 6465 7320 7468 6520  ``..Besides the 
+00011880: 6261 7369 6320 2a4b 2a3c 7375 623e 533c  basic *K*<sub>S<
+00011890: 2f73 7562 3e20 706c 6f74 2c20 7375 6273  /sub> plot, subs
+000118a0: 7469 7475 7469 6f6e 2072 6174 6520 636f  titution rate co
+000118b0: 7272 6563 7469 6f6e 2063 616e 2061 6c73  rrection can als
+000118c0: 6f20 6265 2061 6368 6965 7665 6420 6769  o be achieved gi
+000118d0: 7665 6e20 6174 206c 6561 7374 2061 2073  ven at least a s
+000118e0: 7065 6369 6573 2074 7265 6520 2876 6961  pecies tree (via
+000118f0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
+00011900: 7065 6369 6573 7472 6565 6029 2061 6e64  peciestree`) and
+00011910: 2061 2066 6f63 616c 2073 7065 6369 6573   a focal species
+00011920: 2028 6569 7468 6572 2076 6961 2074 6865   (either via the
+00011930: 206f 7074 696f 6e20 602d 2d66 6f63 7573   option `--focus
+00011940: 3261 6c6c 6020 6f72 2076 6961 2074 6865  2all` or via the
+00011950: 206f 7074 696f 6e20 602d 2d73 7061 6972   option `--spair
+00011960: 6020 696e 2074 6865 2066 6f72 6d20 6f66  ` in the form of
+00011970: 2022 2466 6f63 616c 5f73 7065 6369 6573   "$focal_species
+00011980: 3b24 666f 6361 6c5f 7370 6563 6965 7322  ;$focal_species"
+00011990: 292e 2049 7427 7320 7375 6767 6573 7465  ). It's suggeste
+000119a0: 6420 7468 6174 2074 6865 206f 7274 686f  d that the ortho
+000119b0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+000119c0: 3c2f 7375 623e 2064 6174 6120 6973 2070  </sub> data is p
+000119d0: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
+000119e0: 2d2d 6461 7461 6669 6c65 6020 6f70 7469  --datafile` opti
+000119f0: 6f6e 2061 6e64 2074 6865 2070 6172 616c  on and the paral
+00011a00: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00011a10: 2f73 7562 3e20 6461 7461 2069 7320 7072  /sub> data is pr
+00011a20: 6f76 6964 6564 2062 7920 7468 6520 602d  ovided by the `-
+00011a30: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
+00011a40: 6020 6f70 7469 6f6e 2c20 616c 7468 6f75  ` option, althou
+00011a50: 6768 2069 7427 7320 616c 6c6f 7765 6420  gh it's allowed 
+00011a60: 746f 206f 6e6c 7920 7072 6f76 6964 6520  to only provide 
+00011a70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00011a80: 6461 7461 2076 6961 2074 6865 2060 2d2d  data via the `--
+00011a90: 6461 7461 6669 6c65 6020 6f70 7469 6f6e  datafile` option
+00011aa0: 2061 6e64 2064 6570 6f73 6974 2062 6f74   and deposit bot
+00011ab0: 6820 6f72 7468 6f6c 6f67 6f75 7326 7061  h orthologous&pa
+00011ac0: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
+00011ad0: 3e53 3c2f 7375 623e 2064 6174 6120 7468  >S</sub> data th
+00011ae0: 6572 656f 6e2e 2054 6865 7265 2061 7265  ereon. There are
+00011af0: 2074 776f 2074 7970 6573 206f 6620 6d69   two types of mi
+00011b00: 7865 6420 706c 6f74 7320 2874 6865 2022  xed plots (the "
+00011b10: 6d69 7865 6422 2068 6572 6520 7265 6665  mixed" here refe
+00011b20: 7273 2074 6f20 6d69 7865 6420 6f72 7468  rs to mixed orth
+00011b30: 6f6c 6f67 6f75 7320 616e 6420 7061 7261  ologous and para
+00011b40: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00011b50: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00011b60: 696f 6e73 292c 206f 6e65 206f 6620 7768  ions), one of wh
+00011b70: 6963 6820 6973 2073 696d 696c 6172 2074  ich is similar t
+00011b80: 6f20 7768 6174 2060 6b73 7261 7465 7360  o what `ksrates`
+00011b90: 2070 6c6f 7473 2c20 7768 696c 6520 7468   plots, while th
+00011ba0: 6520 6f74 6865 7220 6f66 2077 6869 6368  e other of which
+00011bb0: 2069 7320 6c69 6b65 2074 6865 2063 6f6e   is like the con
+00011bc0: 7665 6e74 696f 6e61 6c20 2a4b 2a3c 7375  ventional *K*<su
+00011bd0: 623e 533c 2f73 7562 3e20 706c 6f74 2074  b>S</sub> plot t
+00011be0: 6861 7420 626f 7468 2074 6865 206f 7269  hat both the ori
+00011bf0: 6769 6e61 6c20 6f72 7468 6f6c 6f67 6f75  ginal orthologou
+00011c00: 7320 616e 6420 7061 7261 6c6f 676f 7573  s and paralogous
+00011c10: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00011c20: 2064 6973 7472 6962 7574 696f 6e73 2061   distributions a
+00011c30: 7265 2074 7275 7468 6675 6c6c 7920 706c  re truthfully pl
+00011c40: 6f74 7465 6420 696e 7374 6561 6420 6f66  otted instead of
+00011c50: 206a 7573 7420 6265 696e 6720 7265 7072   just being repr
+00011c60: 6573 656e 7465 6420 6279 2073 6f6d 6520  esented by some 
+00011c70: 7665 7274 6963 616c 206c 696e 6573 2e20  vertical lines. 
+00011c80: 5765 2073 7567 6765 7374 2075 7365 7273  We suggest users
+00011c90: 2074 6f20 6164 6f70 7420 7468 6520 606b   to adopt the `k
+00011ca0: 7372 6174 6573 602d 6c69 6b65 2070 6c6f  srates`-like plo
+00011cb0: 7473 2c20 7768 6963 6820 6973 2074 6865  ts, which is the
+00011cc0: 2064 6566 6175 6c74 2e20 4f74 6865 7277   default. Otherw
+00011cd0: 6973 6520 7468 6520 6f70 7469 6f6e 2060  ise the option `
+00011ce0: 2d2d 636c 6173 7369 6360 2077 696c 6c20  --classic` will 
+00011cf0: 6265 206e 6565 6465 6420 746f 2073 6574  be needed to set
+00011d00: 2e20 4578 7472 6120 6d69 7874 7572 6520  . Extra mixture 
+00011d10: 6d6f 6465 6c69 6e67 2061 6e61 6c79 7369  modeling analysi
+00011d20: 7320 6361 6e20 6265 2069 6e69 7469 6174  s can be initiat
+00011d30: 6564 2076 6961 2074 6865 206f 7074 696f  ed via the optio
+00011d40: 6e20 602d 2d70 6c6f 7465 6c6d 6d60 2061  n `--plotelmm` a
+00011d50: 6e64 2060 2d2d 706c 6f74 6170 676d 6d60  nd `--plotapgmm`
+00011d60: 2077 6974 6820 7468 6520 616e 6368 6f72   with the anchor
+00011d70: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
+00011d80: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
+00011d90: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
+00011da0: 6020 6f70 7469 6f6e 2e0a 0a2a 2a41 2073  ` option...**A s
+00011db0: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+00011dc0: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
+00011dd0: 6d6d 616e 6420 7369 6d70 6c79 2061 7320  mmand simply as 
+00011de0: 6265 6c6f 772a 2a0a 0a60 6060 0a77 6764  below**..```.wgd
+00011df0: 2076 697a 202d 6420 6b73 6461 7461 202d   viz -d ksdata -
+00011e00: 7370 2073 7064 6174 6120 2d2d 666f 6375  sp spdata --focu
+00011e10: 7332 616c 6c20 666f 6361 6c5f 7370 6563  s2all focal_spec
+00011e20: 6965 7320 2d2d 6578 7472 6170 6172 616e  ies --extraparan
+00011e30: 6f6d 656b 7320 6b73 6461 7461 0a60 6060  omeks ksdata.```
+00011e40: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
+00011e50: 6174 696f 6e20 6f66 2073 796e 7465 6e79  ation of synteny
+00011e60: 2a2a 0a60 6060 0a77 6764 2076 697a 202d  **.```.wgd viz -
+00011e70: 6170 2061 7064 6174 6120 2d73 6d20 736d  ap apdata -sm sm
+00011e80: 6461 7461 202d 6d74 206d 7464 6174 6120  data -mt mtdata 
+00011e90: 2d67 7420 6774 6461 7461 202d 2d70 6c6f  -gt gtdata --plo
+00011ea0: 7473 796e 2028 2d2d 6d69 6e6c 656e 202d  tsyn (--minlen -
+00011eb0: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
+00011ec0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
+00011ed0: 2033 3029 0a60 6060 0a0a 436f 6d70 6172   30).```..Compar
+00011ee0: 6564 2074 6f20 7468 6520 6f72 6967 696e  ed to the origin
+00011ef0: 616c 2060 7767 6460 2c20 7468 6520 6077  al `wgd`, the `w
+00011f00: 6764 2076 697a 6020 7072 6f67 7261 6d20  gd viz` program 
+00011f10: 6164 6465 6420 7379 6e74 656e 7920 7669  added synteny vi
+00011f20: 7375 616c 697a 6174 696f 6e20 7069 7065  sualization pipe
+00011f30: 6c69 6e65 2e20 5573 6572 7320 6e65 6564  line. Users need
+00011f40: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00011f50: 666c 6167 206f 7074 696f 6e20 602d 2d70  flag option `--p
+00011f60: 6c6f 7473 796e 6020 746f 2069 6e69 7469  lotsyn` to initi
+00011f70: 6174 6520 7468 6973 2070 6172 7420 6f66  ate this part of
+00011f80: 2070 6970 656c 696e 652e 2054 6869 7320   pipeline. This 
+00011f90: 7374 6570 2061 7373 756d 6573 2074 6861  step assumes tha
+00011fa0: 7420 7573 6572 7320 6861 7665 206f 6274  t users have obt
+00011fb0: 6169 6e65 6420 616c 7265 6164 7920 7468  ained already th
+00011fc0: 6520 7379 6e74 656e 6963 2072 6573 756c  e syntenic resul
+00011fd0: 7473 2066 726f 6d20 6069 2d61 6468 6f72  ts from `i-adhor
+00011fe0: 6560 2061 6e64 2075 7365 7320 7468 6f73  e` and uses thos
+00011ff0: 6520 7265 7375 6c74 2066 696c 6573 2074  e result files t
+00012000: 6f20 7265 616c 697a 6520 7468 6520 7379  o realize the sy
+00012010: 6e74 656e 7920 7669 7375 616c 697a 6174  nteny visualizat
+00012020: 696f 6e2e 2053 696d 6c69 6172 2074 6f20  ion. Simliar to 
+00012030: 6077 6764 2073 796e 602c 2074 6865 2065  `wgd syn`, the e
+00012040: 7874 7261 202a 4b2a 3c73 7562 3e53 3c2f  xtra *K*<sub>S</
+00012050: 7375 623e 2064 6174 6120 6669 6c65 2063  sub> data file c
+00012060: 616e 2062 6520 7472 616e 736d 6974 7465  an be transmitte
+00012070: 6420 7669 6120 7468 6520 6f70 7469 6f6e  d via the option
+00012080: 2060 2d2d 6461 7461 6669 6c65 6020 2869   `--datafile` (i
+00012090: 6e73 7465 6164 206f 6620 602d 2d65 7874  nstead of `--ext
+000120a0: 7261 7061 7261 6e6f 6d65 6b73 6020 6f70  raparanomeks` op
+000120b0: 7469 6f6e 292e 2042 6173 6963 616c 6c79  tion). Basically
+000120c0: 2c20 7468 6520 7379 6e74 656e 6963 2072  , the syntenic r
+000120d0: 6573 756c 7420 6669 6c65 7320 7265 7175  esult files requ
+000120e0: 6972 6564 2061 7265 2074 6865 2061 6e63  ired are the anc
+000120f0: 686f 7220 706f 696e 7473 2064 6174 6166  hor points dataf
+00012100: 696c 652c 206d 756c 7469 706c 6963 6f6e  ile, multiplicon
+00012110: 7320 6461 7461 6669 6c65 2c20 6765 6e65  s datafile, gene
+00012120: 2074 6162 6c65 2064 6174 6166 696c 6520   table datafile 
+00012130: 2861 7574 6f6d 6174 656c 7920 7072 6f64  (automately prod
+00012140: 7563 6564 2062 7920 6077 6764 2073 796e  uced by `wgd syn
+00012150: 6029 2c20 616e 6420 7365 676d 656e 7473  `), and segments
+00012160: 2064 6174 6166 696c 652e 0a0a 2a2a 4120   datafile...**A 
+00012170: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+00012180: 6e67 2072 756e 2063 616e 2075 7365 2063  ng run can use c
+00012190: 6f6d 6d61 6e64 2073 696d 706c 7920 6173  ommand simply as
+000121a0: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
+000121b0: 6420 7669 7a20 2d61 7020 6170 6461 7461  d viz -ap apdata
+000121c0: 202d 736d 2073 6d64 6174 6120 2d6d 7420   -sm smdata -mt 
+000121d0: 6d74 6461 7461 202d 6774 2067 7464 6174  mtdata -gt gtdat
+000121e0: 6120 2d2d 706c 6f74 7379 6e0a 6060 600a  a --plotsyn.```.
+000121f0: 0a23 2320 496c 6c75 7374 7261 7469 6f6e  .## Illustration
+00012200: 0a0a 5765 2069 6c6c 7573 7472 6174 6520  ..We illustrate 
+00012210: 6f75 7220 7072 6f67 7261 6d20 6f6e 2061  our program on a
+00012220: 6e20 6578 656d 706c 6172 7920 5747 4420  n exemplary WGD 
+00012230: 696e 6665 7265 6e63 6520 616e 6420 6461  inference and da
+00012240: 7469 6e67 2075 706f 6e20 7370 6563 6965  ting upon specie
+00012250: 7320 2a41 7175 696c 6567 6961 2063 6f65  s *Aquilegia coe
+00012260: 7275 6c65 612a 2e0a 0a54 6865 202a 4171  rulea*...The *Aq
+00012270: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
+00012280: 2a20 7761 7320 7265 706f 7274 6564 2074  * was reported t
+00012290: 6f20 6578 7065 7269 656e 6365 2061 6e20  o experience an 
+000122a0: 7061 6c65 6f2d 706f 6c79 706c 6f69 6469  paleo-polyploidi
+000122b0: 7a61 7469 6f6e 2065 7665 6e74 2061 6674  zation event aft
+000122c0: 6572 2074 6865 2064 6976 6572 6765 6e63  er the divergenc
+000122d0: 6520 6f66 2063 6f72 6520 6575 6469 636f  e of core eudico
+000122e0: 7473 2c20 7768 6963 6820 6973 206c 696b  ts, which is lik
+000122f0: 656c 7920 7368 6172 6564 2062 7920 616c  ely shared by al
+00012300: 6c20 5261 6e75 6e63 756c 616c 6573 2e0a  l Ranunculales..
+00012310: 0a46 6972 7374 2061 626f 7665 2061 6c6c  .First above all
+00012320: 2c20 6c65 7427 7320 6465 6c69 6e65 6174  , let's delineat
+00012330: 6520 7468 6520 7768 6f6c 6520 7061 7261  e the whole para
+00012340: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00012350: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+00012360: 7574 696f 6e20 616e 6420 6861 7665 2061  ution and have a
+00012370: 2062 6173 6963 206f 6273 6572 7661 7469   basic observati
+00012380: 6f6e 2066 6f72 2070 6f74 656e 7469 616c  on for potential
+00012390: 6c79 2063 6f6e 6365 6976 6162 6c65 2057  ly conceivable W
+000123a0: 4744 732c 2075 7369 6e67 2074 6865 2063  GDs, using the c
+000123b0: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
+000123c0: 772e 0a0a 6060 600a 7767 6420 646d 6420  w...```.wgd dmd 
+000123d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000123e0: 6561 0a77 6764 206b 7364 2077 6764 5f64  ea.wgd ksd wgd_d
+000123f0: 6d64 2f41 7175 696c 6567 6961 5f63 6f65  md/Aquilegia_coe
+00012400: 7275 6c65 612e 7473 7620 4171 7569 6c65  rulea.tsv Aquile
+00012410: 6769 615f 636f 6572 756c 6561 0a60 6060  gia_coerulea.```
+00012420: 0a0a 5468 6520 636f 6e73 7472 7563 7465  ..The constructe
+00012430: 6420 7768 6f6c 6520 7061 7261 6e6f 6d65  d whole paranome
+00012440: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00012450: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+00012460: 6e20 6f66 202a 4171 7569 6c65 6769 6120  n of *Aquilegia 
+00012470: 636f 6572 756c 6561 2a20 6973 2061 7320  coerulea* is as 
+00012480: 6265 6c6f 772c 2077 6520 6361 6e20 7365  below, we can se
+00012490: 6520 7468 6174 2074 6865 7265 2073 6565  e that there see
+000124a0: 6d73 2074 6f20 6265 2061 2068 756d 7020  ms to be a hump 
+000124b0: 6174 202a 4b2a 3c73 7562 3e53 3c2f 7375  at *K*<sub>S</su
+000124c0: 623e 2031 2062 7574 206e 6f74 2063 6c65  b> 1 but not cle
+000124d0: 6172 2e0a 0a21 5b5d 2864 6174 612f 4171  ar...![](data/Aq
+000124e0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000124f0: 2e74 7376 2e6b 7364 5f77 702e 7376 6729  .tsv.ksd_wp.svg)
+00012500: 0a0a 5765 2074 6865 6e20 636f 6e73 7472  ..We then constr
+00012510: 7563 7420 7468 6520 616e 6368 6f72 202a  uct the anchor *
+00012520: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00012530: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00012540: 7573 696e 6720 7468 6520 636f 6d6d 616e  using the comman
+00012550: 6420 6c69 6e65 2062 656c 6f77 2e0a 0a60  d line below...`
+00012560: 6060 0a77 6764 2073 796e 202d 6620 6d52  ``.wgd syn -f mR
+00012570: 4e41 202d 6120 4e61 6d65 2077 6764 5f64  NA -a Name wgd_d
+00012580: 6d64 2f41 7175 696c 6567 6961 5f63 6f65  md/Aquilegia_coe
+00012590: 7275 6c65 612e 7473 7620 4171 7569 6c65  rulea.tsv Aquile
+000125a0: 6769 615f 636f 6572 756c 6561 2e67 6666  gia_coerulea.gff
+000125b0: 3320 2d6b 7320 7767 645f 6b73 642f 4171  3 -ks wgd_ksd/Aq
+000125c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000125d0: 2e74 7376 2e6b 732e 7473 760a 6060 600a  .tsv.ks.tsv.```.
+000125e0: 0a41 7320 7368 6f77 6e20 6265 6c6f 772c  .As shown below,
+000125f0: 2074 6865 7265 2061 7265 2073 6f6d 6520   there are some 
+00012600: 7265 7461 696e 6564 2061 6e63 686f 7220  retained anchor 
+00012610: 7061 6972 7320 7769 7468 202a 4b2a 3c73  pairs with *K*<s
+00012620: 7562 3e53 3c2f 7375 623e 2062 6574 7765  ub>S</sub> betwe
+00012630: 656e 2031 2061 6e64 2032 2c20 7768 6963  en 1 and 2, whic
+00012640: 6820 7365 656d 7320 746f 2073 7567 6765  h seems to sugge
+00012650: 7374 2061 2057 4744 2065 7665 6e74 2e0a  st a WGD event..
+00012660: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
+00012670: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00012680: 2e6b 7364 5f77 705f 6170 2e73 7667 290a  .ksd_wp_ap.svg).
+00012690: 0a54 6865 2061 7373 6f63 6961 7465 6420  .The associated 
+000126a0: 6064 7570 5374 6163 6b60 2070 6c6f 7420  `dupStack` plot 
+000126b0: 7368 6f77 7320 7468 6174 2074 6865 7265  shows that there
+000126c0: 2061 7265 206e 756d 6572 6f75 7320 6475   are numerous du
+000126d0: 706c 6963 6174 6564 2073 6567 6d65 6e74  plicated segment
+000126e0: 7320 6163 726f 7373 206d 6f73 7420 6f66  s across most of
+000126f0: 2074 6865 2063 6872 6f6d 6f73 6f6d 6573   the chromosomes
+00012700: 2e0a 0a21 5b5d 2864 6174 612f 4171 7569  ...![](data/Aqui
+00012710: 6c65 6769 615f 636f 6572 756c 6561 5f41  legia_coerulea_A
+00012720: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012730: 615f 6d75 6c74 6970 6c69 636f 6e73 5f6c  a_multiplicons_l
+00012740: 6576 656c 2e73 7667 290a 0a57 6520 696d  evel.svg)..We im
+00012750: 706c 656d 656e 7465 6420 7477 6f20 7479  plemented two ty
+00012760: 7065 7320 6f66 2064 6f74 2070 6c6f 7473  pes of dot plots
+00012770: 2069 6e20 6f78 666f 7264 2067 7269 643a   in oxford grid:
+00012780: 206f 6e65 2069 6e20 7468 6520 756e 6974   one in the unit
+00012790: 206f 6620 6261 7365 7320 616e 6420 7468   of bases and th
+000127a0: 6520 6f74 6865 7220 696e 2074 6865 2075  e other in the u
+000127b0: 6e69 7420 6f66 2067 656e 6573 2c20 7768  nit of genes, wh
+000127c0: 6963 6820 6361 6e20 6265 2063 6f6c 6f72  ich can be color
+000127d0: 6564 2062 7920 2a4b 2a3c 7375 623e 533c  ed by *K*<sub>S<
+000127e0: 2f73 7562 3e20 7661 6c75 6573 2067 6976  /sub> values giv
+000127f0: 656e 202a 4b2a 3c73 7562 3e53 3c2f 7375  en *K*<sub>S</su
+00012800: 623e 2064 6174 612e 0a0a 215b 5d28 6461  b> data...![](da
+00012810: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
+00012820: 7275 6c65 612d 7673 2d41 7175 696c 6567  rulea-vs-Aquileg
+00012830: 6961 5f63 6f65 7275 6c65 615f 4b73 2e64  ia_coerulea_Ks.d
+00012840: 6f74 5f75 6e69 745f 6765 6e65 2e70 6e67  ot_unit_gene.png
+00012850: 290a 0a41 7320 7368 6f77 6e20 6162 6f76  )..As shown abov
+00012860: 652c 2074 6865 2064 6f74 2070 6c6f 7420  e, the dot plot 
+00012870: 696e 2074 6865 2075 6e69 7420 6f66 2067  in the unit of g
+00012880: 656e 6573 2070 7265 7365 6e74 7320 6e75  enes presents nu
+00012890: 6d65 726f 7573 2064 656e 7365 6c79 2061  merous densely a
+000128a0: 6767 7265 6761 7465 6420 286c 696e 652d  ggregated (line-
+000128b0: 6c69 6b65 2920 616e 6368 6f72 2070 6f69  like) anchor poi
+000128c0: 6e74 7320 6174 206d 6f73 7420 6f66 2074  nts at most of t
+000128d0: 6865 2063 6872 6f6d 6f73 6f6d 6573 2077  he chromosomes w
+000128e0: 6974 6820 636f 6e73 6973 7465 6e74 202a  ith consistent *
+000128f0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00012900: 6765 2062 6574 7765 656e 2031 2061 6e64  ge between 1 and
+00012910: 2032 2e20 5468 6520 646f 7420 706c 6f74   2. The dot plot
+00012920: 2069 6e20 7468 6520 756e 6974 206f 6620   in the unit of 
+00012930: 6261 7365 7320 7368 6f77 7320 7468 6520  bases shows the 
+00012940: 7361 6d65 2070 6174 7465 726e 2c20 6173  same pattern, as
+00012950: 206d 616e 6966 6573 7465 6420 6265 6c6f   manifested belo
+00012960: 772e 0a0a 215b 5d28 6461 7461 2f41 7175  w...![](data/Aqu
+00012970: 696c 6567 6961 5f63 6f65 7275 6c65 612d  ilegia_coerulea-
+00012980: 7673 2d41 7175 696c 6567 6961 5f63 6f65  vs-Aquilegia_coe
+00012990: 7275 6c65 615f 4b73 2e64 6f74 2e70 6e67  rulea_Ks.dot.png
+000129a0: 290a 0a54 6865 2064 6f74 2070 6c6f 7473  )..The dot plots
+000129b0: 2077 6974 686f 7574 202a 4b2a 3c73 7562   without *K*<sub
+000129c0: 3e53 3c2f 7375 623e 2061 6e6e 6f74 6174  >S</sub> annotat
+000129d0: 696f 6e20 7769 6c6c 2061 6c73 6f20 6265  ion will also be
+000129e0: 2061 7574 6f6d 6174 656c 7920 7072 6f64   automately prod
+000129f0: 7563 6564 2c20 6173 2073 686f 776e 2062  uced, as shown b
+00012a00: 656c 6f77 2e0a 0a21 5b5d 2864 6174 612f  elow...![](data/
+00012a10: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00012a20: 6561 2d76 732d 4171 7569 6c65 6769 615f  ea-vs-Aquilegia_
+00012a30: 636f 6572 756c 6561 2e64 6f74 5f75 6e69  coerulea.dot_uni
+00012a40: 745f 6765 6e65 2e70 6e67 290a 0a21 5b5d  t_gene.png)..![]
+00012a50: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
+00012a60: 636f 6572 756c 6561 2d76 732d 4171 7569  coerulea-vs-Aqui
+00012a70: 6c65 6769 615f 636f 6572 756c 6561 2e64  legia_coerulea.d
+00012a80: 6f74 2e70 6e67 290a 0a4e 6f74 6520 7468  ot.png)..Note th
+00012a90: 6174 2074 6865 206f 7061 6369 7479 206f  at the opacity o
+00012aa0: 6620 616e 6368 6f72 2064 6f74 7320 616e  f anchor dots an
+00012ab0: 6420 616c 6c20 686f 6d6f 6c6f 6720 646f  d all homolog do
+00012ac0: 7473 2063 616e 2062 6520 7365 7420 6279  ts can be set by
+00012ad0: 2074 6865 206f 7074 696f 6e20 6061 7061   the option `apa
+00012ae0: 6c70 6861 6020 616e 6420 6068 6f61 6c70  lpha` and `hoalp
+00012af0: 6861 6020 7365 7061 7261 7465 6c79 2e20  ha` separately. 
+00012b00: 4966 206f 6e65 206a 7573 7420 7761 6e74  If one just want
+00012b10: 7320 746f 2073 6565 2074 6865 2061 6e63  s to see the anc
+00012b20: 686f 7220 646f 7473 2c20 7365 7474 696e  hor dots, settin
+00012b30: 6720 7468 6520 6068 6f61 6c70 6861 6020  g the `hoalpha` 
+00012b40: 6173 2030 2028 6f72 206f 7468 6572 206d  as 0 (or other m
+00012b50: 696e 7573 6375 6c65 2076 616c 7565 7329  inuscule values)
+00012b60: 2077 696c 6c20 646f 2e20 4966 206f 6e65   will do. If one
+00012b70: 2077 616e 7473 2074 6f20 7365 6520 7468   wants to see th
+00012b80: 6520 6469 7374 7269 6275 7469 6f6e 206f  e distribution o
+00012b90: 6620 7768 6f6c 6520 646f 7473 2062 6574  f whole dots bet
+00012ba0: 7465 722c 2073 6574 7469 6e67 2074 6865  ter, setting the
+00012bb0: 2060 686f 616c 7068 6160 2068 6967 6865   `hoalpha` highe
+00012bc0: 7220 2861 6e64 2060 6170 616c 7068 6160  r (and `apalpha`
+00012bd0: 206c 6f77 6572 2920 7769 6c6c 2064 6f2e   lower) will do.
+00012be0: 2054 6865 2060 646f 7473 697a 6560 206f   The `dotsize` o
+00012bf0: 7074 696f 6e20 6361 6e20 6265 2063 616c  ption can be cal
+00012c00: 6c65 6420 746f 2061 646a 7573 7420 7468  led to adjust th
+00012c10: 6520 7369 7a65 206f 6620 646f 7473 2e0a  e size of dots..
+00012c20: 0a41 2066 7572 7468 6572 2061 7373 6f63  .A further assoc
+00012c30: 6961 7465 6420 5379 6e64 6570 7468 2070  iated Syndepth p
+00012c40: 6c6f 7420 7368 6f77 7320 7468 6174 2074  lot shows that t
+00012c50: 6865 7265 2061 7265 206d 6f72 6520 7468  here are more th
+00012c60: 616e 2038 3020 6475 706c 6963 6174 6564  an 80 duplicated
+00012c70: 2073 6567 6d65 6e74 732c 2077 6869 6368   segments, which
+00012c80: 2064 6f6d 696e 6174 6573 2074 6865 2077   dominates the w
+00012c90: 686f 6c65 2063 6f6c 6c69 6e65 6172 2072  hole collinear r
+00012ca0: 6174 696f 2063 6174 6567 6f72 792e 0a0a  atio category...
+00012cb0: 215b 5d28 6461 7461 2f53 796e 6465 7074  ![](data/Syndept
+00012cc0: 682e 7376 6729 0a0a 5765 2063 616e 2066  h.svg)..We can f
+00012cd0: 6974 2061 6e20 454c 4d4d 206d 6978 7475  it an ELMM mixtu
+00012ce0: 7265 206d 6f64 656c 2075 706f 6e20 7468  re model upon th
+00012cf0: 6520 7768 6f6c 6520 7061 7261 6e6f 6d65  e whole paranome
+00012d00: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00012d10: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+00012d20: 6e20 746f 2073 6565 206d 6f72 6520 6163  n to see more ac
+00012d30: 6375 7261 7465 6c79 2074 6865 2073 6967  curately the sig
+00012d40: 6e69 6669 6361 6e63 6520 616e 6420 6c6f  nificance and lo
+00012d50: 6361 7469 6f6e 206f 6620 706f 7465 6e74  cation of potent
+00012d60: 6961 6c20 5747 4473 2c20 7573 696e 6720  ial WGDs, using 
+00012d70: 7468 6520 636f 6d6d 616e 6420 6c69 6e65  the command line
+00012d80: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+00012d90: 2076 697a 202d 6420 7767 645f 6b73 642f   viz -d wgd_ksd/
+00012da0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00012db0: 6561 2e74 7376 2e6b 732e 7473 760a 6060  ea.tsv.ks.tsv.``
+00012dc0: 600a 0a54 6865 2072 6573 756c 7420 6f66  `..The result of
+00012dd0: 2045 4c4d 4d20 6d69 7874 7572 6520 6d6f   ELMM mixture mo
+00012de0: 6465 6c20 636c 7573 7465 7269 6e67 2073  del clustering s
+00012df0: 686f 7773 2074 6861 7420 7468 6572 6520  hows that there 
+00012e00: 6973 2061 206c 696b 656c 7920 5747 4420  is a likely WGD 
+00012e10: 636f 6d70 6f6e 656e 7420 6174 202a 4b2a  component at *K*
+00012e20: 3c73 7562 3e53 3c2f 7375 623e 2031 2e31  <sub>S</sub> 1.1
+00012e30: 392e 0a0a 215b 5d28 6461 7461 2f65 6c6d  9...![](data/elm
+00012e40: 6d5f 4171 7569 6c65 6769 615f 636f 6572  m_Aquilegia_coer
+00012e50: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00012e60: 6265 7374 5f6d 6f64 656c 735f 7765 6967  best_models_weig
+00012e70: 6874 6564 2e73 7667 290a 0a4c 6574 2773  hted.svg)..Let's
+00012e80: 2064 6f20 6120 6d69 7874 7572 6520 6d6f   do a mixture mo
+00012e90: 6465 6c20 636c 7573 7465 7269 6e67 2066  del clustering f
+00012ea0: 6f72 2061 6e63 686f 7220 2a4b 2a3c 7375  or anchor *K*<su
+00012eb0: 623e 533c 2f73 7562 3e20 746f 6f2c 2075  b>S</sub> too, u
+00012ec0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
+00012ed0: 206c 696e 6520 6265 6c6f 772e 0a0a 6060   line below...``
+00012ee0: 600a 7767 6420 7065 616b 2077 6764 5f6b  `.wgd peak wgd_k
+00012ef0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00012f00: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00012f10: 202d 2d61 6e63 686f 7270 6f69 6e74 7320   --anchorpoints 
+00012f20: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
+00012f30: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
+00012f40: 2e74 7874 202d 2d73 6567 6d65 6e74 7320  .txt --segments 
+00012f50: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
+00012f60: 6f75 742f 7365 676d 656e 7473 2e74 7874  out/segments.txt
+00012f70: 202d 2d6c 6973 7465 6c65 6d65 6e74 7320   --listelements 
+00012f80: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
+00012f90: 6f75 742f 6c69 7374 5f65 6c65 6d65 6e74  out/list_element
+00012fa0: 732e 7478 7420 2d2d 6d75 6c74 6970 6c69  s.txt --multipli
+00012fb0: 636f 6e70 6169 7273 2077 6764 5f73 796e  conpairs wgd_syn
+00012fc0: 2f69 6164 686f 7265 2d6f 7574 2f6d 756c  /iadhore-out/mul
+00012fd0: 7469 706c 6963 6f6e 5f70 6169 7273 2e74  tiplicon_pairs.t
+00012fe0: 7874 202d 2d77 6569 6768 7465 640a 6060  xt --weighted.``
+00012ff0: 600a 0a54 6865 2061 6e63 686f 7220 2a4b  `..The anchor *K
+00013000: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+00013010: 6520 6469 7374 7269 6275 7469 6f6e 2061  e distribution a
+00013020: 6c73 6f20 6861 7320 6120 6c69 6b65 6c79  lso has a likely
+00013030: 2057 4744 2063 6f6d 706f 6e65 6e74 2077   WGD component w
+00013040: 6974 6820 6d6f 6465 2031 2e32 382e 0a0a  ith mode 1.28...
+00013050: 215b 5d28 6461 7461 2f4f 7269 6769 6e61  ![](data/Origina
+00013060: 6c5f 416e 6368 6f72 4b73 5f47 4d4d 5f43  l_AnchorKs_GMM_C
+00013070: 6f6d 706f 6e65 6e74 335f 6e6f 6465 5f77  omponent3_node_w
+00013080: 6569 6768 7465 645f 4c6f 676e 6f72 6d61  eighted_Lognorma
+00013090: 6c2e 7376 6729 0a0a 4e6f 7720 7468 6174  l.svg)..Now that
+000130a0: 2077 6520 6861 7665 2073 6565 6e20 7468   we have seen th
+000130b0: 6520 6576 6964 656e 6365 206f 6620 6e75  e evidence of nu
+000130c0: 6d65 726f 7573 2064 7570 6c69 6361 7465  merous duplicate
+000130d0: 6420 7365 676d 656e 7473 2061 6e64 2074  d segments and t
+000130e0: 6865 2061 6767 7265 6761 7469 6f6e 206f  he aggregation o
+000130f0: 6620 6475 706c 6963 6174 6573 2061 6765  f duplicates age
+00013100: 2069 6e20 2a4b 2a3c 7375 623e 533c 2f73   in *K*<sub>S</s
+00013110: 7562 3e20 6172 6f75 6e64 2031 2e32 2066  ub> around 1.2 f
+00013120: 6f72 2061 6e63 686f 7220 7061 6972 7320  or anchor pairs 
+00013130: 616e 6420 6e6f 6e2d 616e 6368 6f72 2070  and non-anchor p
+00013140: 6169 7273 2074 6872 6f75 6768 6f75 7420  airs throughout 
+00013150: 7468 6520 7768 6f6c 6520 6765 6e6f 6d65  the whole genome
+00013160: 2e20 5765 2063 616e 2063 6c61 696d 2077  . We can claim w
+00013170: 6974 6820 736f 6d65 2063 6f6e 6669 6465  ith some confide
+00013180: 6e63 6520 7468 6174 202a 4171 7569 6c65  nce that *Aquile
+00013190: 6769 6120 636f 6572 756c 6561 2a20 6d69  gia coerulea* mi
+000131a0: 6768 7420 6861 7665 2065 7870 6572 6965  ght have experie
+000131b0: 6e63 6564 2061 2070 616c 656f 2d70 6f6c  nced a paleo-pol
+000131c0: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
+000131d0: 656e 742e 204e 6578 742c 204c 6574 2773  ent. Next, Let's
+000131e0: 2068 6176 6520 6120 6675 7274 6865 7220   have a further 
+000131f0: 6c6f 6f6b 2061 626f 7574 2069 7473 2070  look about its p
+00013200: 6879 6c6f 6765 6e65 7469 6320 6c6f 6361  hylogenetic loca
+00013210: 7469 6f6e 2e20 5765 206b 6e6f 7720 7468  tion. We know th
+00013220: 6174 2074 6865 7265 2061 7265 2075 6e63  at there are unc
+00013230: 6572 7461 696e 7469 6573 2061 626f 7574  ertainties about
+00013240: 2077 6865 7468 6572 2074 6869 7320 7075   whether this pu
+00013250: 7461 7469 7665 2070 616c 656f 2d70 6f6c  tative paleo-pol
+00013260: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
+00013270: 656e 7420 6973 2073 6861 7265 6420 7769  ent is shared wi
+00013280: 7468 2061 6c6c 2065 7564 6963 6f74 7320  th all eudicots 
+00013290: 6f72 206e 6f74 2e20 5765 2063 616e 2063  or not. We can c
+000132a0: 686f 6f73 6520 736f 6d65 206f 7468 6572  hoose some other
+000132b0: 2065 7564 6963 6f74 2067 656e 6f6d 6573   eudicot genomes
+000132c0: 2074 6f20 7365 6520 7468 6520 6f72 6465   to see the orde
+000132d0: 7269 6e67 206f 6620 7370 6563 6961 7469  ring of speciati
+000132e0: 6f6e 2061 6e64 2070 6f6c 7970 6c6f 6964  on and polyploid
+000132f0: 697a 6174 696f 6e20 6576 656e 7473 2e20  ization events. 
+00013300: 4865 7265 2077 6520 6368 6f6f 7365 202a  Here we choose *
+00013310: 5669 7469 7320 7669 6e69 6665 7261 2a2c  Vitis vinifera*,
+00013320: 202a 5072 6f74 6561 2063 796e 6172 6f69   *Protea cynaroi
+00013330: 6465 732a 2061 6e64 202a 4163 6f72 7573  des* and *Acorus
+00013340: 2061 6d65 7269 6361 6e75 732a 2069 6e20   americanus* in 
+00013350: 7468 6520 666f 6c6c 6f77 696e 6720 2a4b  the following *K
+00013360: 2a3c 7375 623e 533c 2f73 7562 3e20 616e  *<sub>S</sub> an
+00013370: 616c 7973 6973 2e20 4669 7273 742c 2077  alysis. First, w
+00013380: 6520 6275 696c 7420 6120 676c 6f62 616c  e built a global
+00013390: 204d 5242 4820 6661 6d69 6c79 2075 7369   MRBH family usi
+000133a0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+000133b0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
+000133c0: 6d64 202d 2d67 6c6f 6261 6c6d 7262 6820  md --globalmrbh 
+000133d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000133e0: 6561 2050 726f 7465 615f 6379 6e61 726f  ea Protea_cynaro
+000133f0: 6964 6573 2041 636f 7275 735f 616d 6572  ides Acorus_amer
+00013400: 6963 616e 7573 2056 6974 6973 5f76 696e  icanus Vitis_vin
+00013410: 6966 6572 6120 2d6f 2077 6764 5f67 6c6f  ifera -o wgd_glo
+00013420: 6261 6c6d 7262 680a 6060 600a 0a49 6e20  balmrbh.```..In 
+00013430: 7468 6520 676c 6f62 616c 204d 5242 4820  the global MRBH 
+00013440: 6661 6d69 6c79 2c20 6576 6572 7920 7061  family, every pa
+00013450: 6972 206f 6620 6f72 7468 6f6c 6f67 6f75  ir of orthologou
+00013460: 7320 6765 6e65 7320 6973 2074 6865 2072  s genes is the r
+00013470: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
+00013480: 6974 2c20 7375 6767 6573 7469 6e67 2074  it, suggesting t
+00013490: 7275 6520 6f72 7468 6f6c 6f67 6f75 7320  rue orthologous 
+000134a0: 7265 6c61 7469 6f6e 7368 6970 732e 2057  relationships. W
+000134b0: 6520 776f 756c 6420 7573 6520 7468 6520  e would use the 
+000134c0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000134d0: 7661 6c75 6573 2061 7373 6f63 6961 7465  values associate
+000134e0: 6420 7769 7468 2074 6865 7365 206f 7274  d with these ort
+000134f0: 686f 6c6f 676f 7573 2070 6169 7273 2074  hologous pairs t
+00013500: 6f20 6465 6c69 6d69 7420 7468 6520 6469  o delimit the di
+00013510: 7665 7267 656e 6365 202a 4b2a 3c73 7562  vergence *K*<sub
+00013520: 3e53 3c2f 7375 623e 2070 6561 6b2e 2054  >S</sub> peak. T
+00013530: 6f67 6574 6865 7220 7769 7468 2074 6865  ogether with the
+00013540: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
+00013550: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013560: 6469 7374 7269 6275 7469 6f6e 2c20 7765  distribution, we
+00013570: 2063 6f6e 6475 6374 2074 6865 2072 6174   conduct the rat
+00013580: 6520 636f 7272 6563 7469 6f6e 2075 7369  e correction usi
+00013590: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+000135a0: 656c 6f77 2e0a 0a21 2153 696e 6365 2060  elow...!!Since `
+000135b0: 7767 6460 2076 6572 7369 6f6e 2032 2e30  wgd` version 2.0
+000135c0: 2e32 342c 2077 6520 7265 7772 6f74 6520  .24, we rewrote 
+000135d0: 6120 636c 6561 6e65 7220 616e 6420 7175  a cleaner and qu
+000135e0: 6963 6b65 7220 7761 7920 6f66 2064 6f69  icker way of doi
+000135f0: 6e67 2073 7562 7374 6974 7574 696f 6e20  ng substitution 
+00013600: 7261 7465 2063 6f72 7265 6374 696f 6e2e  rate correction.
+00013610: 2049 7427 7320 6e6f 7420 7265 7175 6972   It's not requir
+00013620: 6564 2074 6f20 7479 7065 2069 6e20 616e  ed to type in an
+00013630: 7920 7370 6569 6365 7320 7061 6972 2061  y speices pair a
+00013640: 6e64 2061 2073 6572 6965 7320 6f66 202a  nd a series of *
+00013650: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00013660: 6c6f 7473 2077 696c 6c20 6265 2070 726f  lots will be pro
+00013670: 6475 6365 642e 2054 6865 2072 6571 7569  duced. The requi
+00013680: 7265 6420 6669 6c65 7320 6172 6520 6f72  red files are or
+00013690: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+000136a0: 623e 533c 2f73 7562 3e20 7461 626c 652c  b>S</sub> table,
+000136b0: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
+000136c0: 7375 623e 533c 2f73 7562 3e20 7461 626c  sub>S</sub> tabl
+000136d0: 652c 2061 2073 7065 6369 6573 2074 7265  e, a species tre
+000136e0: 6520 616e 6420 6120 666f 6375 7365 6420  e and a focused 
+000136f0: 7370 6563 6965 7320 2874 6865 206f 6e65  species (the one
+00013700: 2069 6e70 7574 7465 6420 7769 7468 2070   inputted with p
+00013710: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
+00013720: 623e 533c 2f73 7562 3e20 6461 7461 292e  b>S</sub> data).
+00013730: 2055 7365 7273 2063 616e 2063 686f 6f73   Users can choos
+00013740: 6520 746f 2061 6464 206f 6e65 206d 6f72  e to add one mor
+00013750: 6520 6c61 7965 7220 6f66 2065 6c6d 6d20  e layer of elmm 
+00013760: 6d6f 6465 6c69 6e67 206f 6e20 7061 7261  modeling on para
+00013770: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00013780: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
+00013790: 642f 6f72 2067 6d6d 206d 6f64 656c 696e  d/or gmm modelin
+000137a0: 6720 6f6e 2061 6e63 686f 7220 2a4b 2a3c  g on anchor *K*<
+000137b0: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+000137c0: 6573 2e20 5468 6520 6f72 7468 6f6c 6f67  es. The ortholog
+000137d0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+000137e0: 7562 3e20 7661 6c75 6573 2063 616e 2062  ub> values can b
+000137f0: 6520 6361 6c63 756c 6174 6564 2075 7369  e calculated usi
+00013800: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00013810: 656c 6f77 2e0a 0a60 6060 0a77 6764 206b  elow...```.wgd k
+00013820: 7364 2077 6764 5f67 6c6f 6261 6c6d 7262  sd wgd_globalmrb
+00013830: 682f 676c 6f62 616c 5f4d 5242 482e 7473  h/global_MRBH.ts
+00013840: 7620 7365 7173 2a20 2d6f 2077 6764 5f67  v seqs* -o wgd_g
+00013850: 6c6f 6261 6c6d 7262 685f 6b73 0a60 6060  lobalmrbh_ks.```
+00013860: 0a0a 5769 7468 2074 6865 2063 616c 6375  ..With the calcu
+00013870: 6c61 7465 6420 6f72 7468 6f6c 6f67 6f75  lated orthologou
+00013880: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+00013890: 3e20 7461 626c 652c 2077 6520 6361 6e20  > table, we can 
+000138a0: 7573 6520 7468 6520 636f 6d6d 616e 6420  use the command 
+000138b0: 6265 6c6f 7720 746f 2063 6f6e 6475 6374  below to conduct
+000138c0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+000138d0: 7469 6f6e 2061 6e64 2f6f 7220 6d69 7874  tion and/or mixt
+000138e0: 7572 6520 6d6f 6465 6c69 6e67 2061 6e61  ure modeling ana
+000138f0: 6c79 7369 732e 0a0a 6060 600a 7767 6420  lysis...```.wgd 
+00013900: 7669 7a20 2d64 2077 6764 5f67 6c6f 6261  viz -d wgd_globa
+00013910: 6c6d 7262 685f 6b73 2f67 6c6f 6261 6c5f  lmrbh_ks/global_
+00013920: 4d52 4248 2e74 7376 2e6b 732e 7473 7620  MRBH.tsv.ks.tsv 
+00013930: 2d66 6120 4171 7569 6c65 6769 615f 636f  -fa Aquilegia_co
+00013940: 6572 756c 6561 202d 6570 6b20 4171 7569  erulea -epk Aqui
+00013950: 6c65 6769 615f 636f 6572 756c 6561 2e6b  legia_coerulea.k
+00013960: 732e 7473 7620 2d61 7020 616e 6368 6f72  s.tsv -ap anchor
+00013970: 706f 696e 7473 2e74 7874 202d 7370 2073  points.txt -sp s
+00013980: 7065 6369 6573 7472 6565 2e6e 7720 2d6f  peciestree.nw -o
+00013990: 2077 6764 5f76 697a 5f6d 6978 6564 5f4b   wgd_viz_mixed_K
+000139a0: 7320 2d2d 706c 6f74 656c 6d6d 202d 2d70  s --plotelmm --p
+000139b0: 6c6f 7461 7067 6d6d 0a60 6060 0a0a 6060  lotapgmm.```..``
+000139c0: 600a 7767 6420 6b73 6420 7767 645f 676c  `.wgd ksd wgd_gl
+000139d0: 6f62 616c 6d72 6268 2f67 6c6f 6261 6c5f  obalmrbh/global_
+000139e0: 4d52 4248 2e74 7376 2073 6571 732a 202d  MRBH.tsv seqs* -
+000139f0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
+00013a00: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+00013a10: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00013a20: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
+00013a30: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
+00013a40: 6967 6874 202d 6f20 7767 645f 676c 6f62  ight -o wgd_glob
+00013a50: 616c 6d72 6268 5f6b 7320 2d2d 7370 6169  almrbh_ks --spai
+00013a60: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
+00013a70: 7275 6c65 613b 5072 6f74 6561 5f63 796e  rulea;Protea_cyn
+00013a80: 6172 6f69 6465 7322 202d 2d73 7061 6972  aroides" --spair
+00013a90: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
+00013aa0: 756c 6561 3b56 6974 6973 5f76 696e 6966  ulea;Vitis_vinif
+00013ab0: 6572 6122 202d 2d73 7061 6972 2022 4171  era" --spair "Aq
+00013ac0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00013ad0: 3b41 636f 7275 735f 616d 6572 6963 616e  ;Acorus_american
+00013ae0: 7573 2220 2d2d 7370 6169 7220 2241 7175  us" --spair "Aqu
+00013af0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
+00013b00: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013b10: 6561 2220 2d2d 706c 6f74 6b64 6520 282d  ea" --plotkde (-
+00013b20: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+00013b30: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+00013b40: 6e74 732e 7478 7429 0a60 6060 0a0a 5468  nts.txt).```..Th
+00013b50: 6520 6669 6c65 2060 7370 6563 6965 7374  e file `speciest
+00013b60: 7265 652e 6e77 6020 6973 2074 6865 2074  ree.nw` is the t
+00013b70: 6578 7420 6669 6c65 206f 6620 7370 6563  ext file of spec
+00013b80: 6965 7320 7472 6565 2069 6e20 6e65 7769  ies tree in newi
+00013b90: 636b 2074 6861 7420 7261 7465 2063 6f72  ck that rate cor
+00013ba0: 7265 6374 696f 6e20 776f 756c 6420 6265  rection would be
+00013bb0: 2063 6f6e 6475 6374 6564 206f 6e2e 2049   conducted on. I
+00013bc0: 7473 2063 6f6e 7465 6e74 2069 7320 6173  ts content is as
+00013bd0: 2062 656c 6f77 2e20 5573 6572 7320 6e65   below. Users ne
+00013be0: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
+00013bf0: 6520 7370 6563 6965 7320 7061 6972 7320  e species pairs 
+00013c00: 746f 2062 6520 706c 6f74 7465 642e 2057  to be plotted. W
+00013c10: 6520 7375 6767 6573 7420 6164 6469 6e67  e suggest adding
+00013c20: 2074 6865 206f 7074 696f 6e20 602d 2d72   the option `--r
+00013c30: 6577 6569 6768 7460 2074 6f20 7265 6361  eweight` to reca
+00013c40: 6c63 756c 6174 6520 7468 6520 7765 6967  lculate the weig
+00013c50: 6874 2070 6572 2073 7065 6369 6573 2070  ht per species p
+00013c60: 6169 7220 7375 6368 2074 6861 7420 7468  air such that th
+00013c70: 6520 7765 6967 6874 206f 6620 6f72 7468  e weight of orth
+00013c80: 6f6c 6f67 6f75 7320 6765 6e65 2070 6169  ologous gene pai
+00013c90: 7273 2077 696c 6c20 6265 636f 6d65 2031  rs will become 1
+00013ca0: 2061 7320 7468 6520 7061 7261 6c6f 676f   as the paralogo
+00013cb0: 7573 2067 656e 6520 7061 6972 732e 2054  us gene pairs. T
+00013cc0: 6865 2066 6c61 6720 602d 2d70 6c6f 746b  he flag `--plotk
+00013cd0: 6465 6020 6361 6e20 6265 2061 6464 6564  de` can be added
+00013ce0: 2077 6865 6e20 7468 6520 6b64 6520 6375   when the kde cu
+00013cf0: 7276 6520 6f66 206f 7274 686f 6c6f 676f  rve of orthologo
+00013d00: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+00013d10: 623e 2069 7320 6465 7369 7265 642e 2045  b> is desired. E
+00013d20: 7874 7261 2063 6f6c 6c69 6e65 6172 2064  xtra collinear d
+00013d30: 6174 6120 6361 6e20 6265 2061 6464 6564  ata can be added
+00013d40: 2062 7920 7468 6520 6f70 7469 6f6e 2060   by the option `
+00013d50: 2d61 7060 2e0a 0a60 6060 0a28 2828 5669  -ap`...```.(((Vi
+00013d60: 7469 735f 7669 6e69 6665 7261 2c50 726f  tis_vinifera,Pro
+00013d70: 7465 615f 6379 6e61 726f 6964 6573 292c  tea_cynaroides),
+00013d80: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013d90: 6561 292c 4163 6f72 7573 5f61 6d65 7269  ea),Acorus_ameri
+00013da0: 6361 6e75 7329 3b0a 6060 600a 0a21 5b5d  canus);.```..![]
+00013db0: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
+00013dc0: 636f 6572 756c 6561 5f47 6c6f 6261 6c6d  coerulea_Globalm
+00013dd0: 7262 684b 735f 436f 7272 6563 7465 642e  rbhKs_Corrected.
+00013de0: 6b73 642e 7376 6729 0a0a 4173 2073 686f  ksd.svg)..As sho
+00013df0: 776e 2061 626f 7665 2c20 6265 6361 7573  wn above, becaus
+00013e00: 6520 6f66 2074 6865 2068 6967 6865 7220  e of the higher 
+00013e10: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+00013e20: 6520 6f66 202a 4171 7569 6c65 6769 6120  e of *Aquilegia 
+00013e30: 636f 6572 756c 6561 2a2c 2074 6865 206f  coerulea*, the o
+00013e40: 7269 6769 6e61 6c20 6f72 7468 6f6c 6f67  riginal ortholog
+00013e50: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00013e60: 7562 3e20 7661 6c75 6573 2077 6572 6520  ub> values were 
+00013e70: 6163 7475 616c 6c79 2075 6e64 6572 6573  actually underes
+00013e80: 7469 6d61 7465 6420 696e 2074 6865 2074  timated in the t
+00013e90: 696d 652d 6672 616d 6520 6f66 202a 4171  ime-frame of *Aq
+00013ea0: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
+00013eb0: 2a2e 2057 6865 6e20 7765 2072 6563 6f76  *. When we recov
+00013ec0: 6572 6564 2074 6865 2064 6976 6572 6765  ered the diverge
+00013ed0: 6e63 6520 7375 6273 7469 7475 7469 6f6e  nce substitution
+00013ee0: 2064 6973 7461 6e63 6520 696e 2074 6572   distance in ter
+00013ef0: 6d73 206f 6620 7477 6f20 7469 6d65 7320  ms of two times 
+00013f00: 6f66 2074 6865 2062 7261 6e63 682d 7370  of the branch-sp
+00013f10: 6563 6966 6963 2063 6f6e 7472 6962 7574  ecific contribut
+00013f20: 696f 6e20 6f66 202a 4171 7569 6c65 6769  ion of *Aquilegi
+00013f30: 6120 636f 6572 756c 6561 2a20 7369 6e63  a coerulea* sinc
+00013f40: 6520 6974 7320 6469 7665 7267 656e 6365  e its divergence
+00013f50: 2077 6974 6820 7468 6520 7369 7374 6572   with the sister
+00013f60: 2073 7065 6369 6573 2070 6c75 7320 7468   species plus th
+00013f70: 6520 7368 6172 6564 2073 7562 7374 6974  e shared substit
+00013f80: 7574 696f 6e20 6469 7374 616e 6365 2062  ution distance b
+00013f90: 6566 6f72 6520 6469 7665 7267 656e 6365  efore divergence
+00013fa0: 2028 696e 2072 656c 6174 6976 6520 746f   (in relative to
+00013fb0: 2074 6865 206f 7574 6772 6f75 7029 2c20   the outgroup), 
+00013fc0: 7468 6520 636f 7272 6563 7465 6420 2a4b  the corrected *K
+00013fd0: 2a3c 7375 623e 533c 2f73 7562 3e20 6d6f  *<sub>S</sub> mo
+00013fe0: 6465 2062 6563 616d 6520 6c61 7267 6572  de became larger
+00013ff0: 2e0a 0a49 6620 6f6e 6520 6861 6420 7468  ...If one had th
+00014000: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
+00014010: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+00014020: 7461 2061 6c72 6561 6479 2c20 6f6e 6520  ta already, one 
+00014030: 636f 756c 6420 616c 736f 2061 7070 6c79  could also apply
+00014040: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
+00014050: 6420 7669 7a60 2074 6f20 636f 6e64 7563  d viz` to conduc
+00014060: 7420 7468 6520 7261 7465 2063 6f72 7265  t the rate corre
+00014070: 6374 696f 6e20 616e 616c 7973 6973 2075  ction analysis u
+00014080: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
+00014090: 2062 656c 6f77 2e20 4e6f 7465 2074 6861   below. Note tha
+000140a0: 7420 7468 6520 6f72 6465 7220 6f66 2067  t the order of g
+000140b0: 6976 656e 2060 7370 6169 7260 206f 7074  iven `spair` opt
+000140c0: 696f 6e73 2064 6563 6964 6573 2074 6865  ions decides the
+000140d0: 2063 6f6c 6f72 206f 6620 7468 6520 2a4b   color of the *K
+000140e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+000140f0: 7374 7269 6275 7469 6f6e 206f 6620 6561  stribution of ea
+00014100: 6368 2073 7065 6369 6573 2070 6169 722e  ch species pair.
+00014110: 2054 6865 2061 6464 6974 696f 6e61 6c20   The additional 
+00014120: 6f70 7469 6f6e 2060 666f 6375 7332 616c  option `focus2al
+00014130: 6c60 2069 6e20 6077 6764 206b 7364 6020  l` in `wgd ksd` 
+00014140: 616e 6420 6077 6764 2076 697a 6020 6361  and `wgd viz` ca
+00014150: 6e20 6265 2075 7365 6420 746f 2074 656c  n be used to tel
+00014160: 6c20 7468 6520 7072 6f67 7261 6d20 7468  l the program th
+00014170: 6174 2074 6865 2073 7065 6369 6573 2070  at the species p
+00014180: 6169 7273 2061 7265 2062 6574 7765 656e  airs are between
+00014190: 2074 6865 2066 6f63 7573 2073 7065 6369   the focus speci
+000141a0: 6573 2061 6e64 2061 6c6c 2074 6865 206f  es and all the o
+000141b0: 7468 6572 2073 7065 6369 6573 2073 7563  ther species suc
+000141c0: 6820 7468 6174 2075 7365 7273 2064 6f6e  h that users don
+000141d0: 2774 206e 6565 6420 746f 2074 7970 6520  't need to type 
+000141e0: 696e 2065 6163 6820 7061 6972 2069 6e64  in each pair ind
+000141f0: 6976 6964 7561 6c6c 792e 0a0a 6060 600a  ividually...```.
+00014200: 7767 6420 7669 7a20 2d64 2077 6764 5f67  wgd viz -d wgd_g
+00014210: 6c6f 6261 6c6d 7262 685f 6b73 2f67 6c6f  lobalmrbh_ks/glo
+00014220: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
+00014230: 7473 7620 2d2d 6578 7472 6170 6172 616e  tsv --extraparan
+00014240: 6f6d 656b 7320 7767 645f 6b73 642f 4171  omeks wgd_ksd/Aq
+00014250: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014260: 2e74 7376 2e6b 732e 7473 7620 2d73 7020  .tsv.ks.tsv -sp 
+00014270: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
+00014280: 2d72 6577 6569 6768 7420 2d61 7020 7767  -reweight -ap wg
+00014290: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+000142a0: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
+000142b0: 7874 202d 6f20 7767 645f 7669 7a5f 6d69  xt -o wgd_viz_mi
+000142c0: 7865 645f 4b73 202d 2d73 7061 6972 2022  xed_Ks --spair "
+000142d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000142e0: 6561 3b50 726f 7465 615f 6379 6e61 726f  ea;Protea_cynaro
+000142f0: 6964 6573 2220 2d2d 7370 6169 7220 2241  ides" --spair "A
+00014300: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00014310: 613b 5669 7469 735f 7669 6e69 6665 7261  a;Vitis_vinifera
+00014320: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+00014330: 6567 6961 5f63 6f65 7275 6c65 613b 4163  egia_coerulea;Ac
+00014340: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
+00014350: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00014360: 6769 615f 636f 6572 756c 6561 3b41 7175  gia_coerulea;Aqu
+00014370: 696c 6567 6961 5f63 6f65 7275 6c65 6122  ilegia_coerulea"
+00014380: 202d 2d70 6c6f 746b 6465 2028 2d2d 6773   --plotkde (--gs
+00014390: 6d61 7020 6765 6e65 5f73 7065 6369 6573  map gene_species
+000143a0: 2e6d 6170 290a 6060 600a 0a57 6974 6820  .map).```..With 
+000143b0: 7468 6520 6f70 7469 6f6e 2060 666f 6375  the option `focu
+000143c0: 7332 616c 6c60 2c20 7468 6520 636f 6d6d  s2all`, the comm
+000143d0: 616e 6420 6361 6e20 6265 2061 6c73 6f20  and can be also 
+000143e0: 6c69 6b65 2074 6869 732c 2073 616d 6520  like this, same 
+000143f0: 7769 7468 2060 7767 6420 6b73 6460 2e0a  with `wgd ksd`..
+00014400: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
+00014410: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+00014420: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
+00014430: 762e 6b73 2e74 7376 202d 2d65 7874 7261  v.ks.tsv --extra
+00014440: 7061 7261 6e6f 6d65 6b73 2077 6764 5f6b  paranomeks wgd_k
+00014450: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00014460: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00014470: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00014480: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00014490: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+000144a0: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+000144b0: 6e74 732e 7478 7420 2d6f 2077 6764 5f76  nts.txt -o wgd_v
+000144c0: 697a 5f6d 6978 6564 5f4b 7320 2d2d 666f  iz_mixed_Ks --fo
+000144d0: 6375 7332 616c 6c20 4171 7569 6c65 6769  cus2all Aquilegi
+000144e0: 615f 636f 6572 756c 6561 202d 2d70 6c6f  a_coerulea --plo
+000144f0: 746b 6465 0a60 6060 0a0a 4e6f 7465 2074  tkde.```..Note t
+00014500: 6861 7420 7765 2063 616e 2065 6173 696c  hat we can easil
+00014510: 7920 7368 6f77 2074 6861 7420 2a41 7175  y show that *Aqu
+00014520: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+00014530: 2068 6173 2068 6967 6865 7220 7375 6273   has higher subs
+00014540: 7469 7475 7469 6f6e 2072 6174 6520 7468  titution rate th
+00014550: 616e 202a 5072 6f74 6561 2063 796e 6172  an *Protea cynar
+00014560: 6f69 6465 732a 2061 6e64 202a 5669 7469  oides* and *Viti
+00014570: 7320 7669 6e69 6665 7261 2a20 6279 2063  s vinifera* by c
+00014580: 6f6d 7061 7269 6e67 2074 6865 6972 2073  omparing their s
+00014590: 7562 7374 6974 7574 696f 6e20 6469 7374  ubstitution dist
+000145a0: 616e 6365 2069 6e20 7265 6761 7264 2074  ance in regard t
+000145b0: 6f20 7468 6520 7361 6d65 2064 6976 6572  o the same diver
+000145c0: 6765 6e63 6520 6576 656e 7420 7769 7468  gence event with
+000145d0: 206f 7574 6772 6f75 7020 7370 6563 6965   outgroup specie
+000145e0: 7320 2a41 636f 7275 735f 616d 6572 6963  s *Acorus_americ
+000145f0: 616e 7573 2a2c 2075 7369 6e67 2063 6f6d  anus*, using com
+00014600: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
+00014610: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+00014620: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
+00014630: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
+00014640: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00014650: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00014660: 6874 202d 6f20 7767 645f 7669 7a5f 436f  ht -o wgd_viz_Co
+00014670: 6d70 6172 655f 7261 7465 202d 2d73 7061  mpare_rate --spa
+00014680: 6972 2022 4163 6f72 7573 5f61 6d65 7269  ir "Acorus_ameri
+00014690: 6361 6e75 733b 5072 6f74 6561 5f63 796e  canus;Protea_cyn
+000146a0: 6172 6f69 6465 7322 202d 2d73 7061 6972  aroides" --spair
+000146b0: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
+000146c0: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
+000146d0: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
+000146e0: 2256 6974 6973 5f76 696e 6966 6572 613b  "Vitis_vinifera;
+000146f0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00014700: 7322 202d 2d67 736d 6170 2067 656e 655f  s" --gsmap gene_
+00014710: 7370 6563 6965 732e 6d61 7020 2d2d 706c  species.map --pl
+00014720: 6f74 6b64 650a 6060 600a 0a21 5b5d 2864  otkde.```..![](d
+00014730: 6174 612f 5261 775f 4f72 7468 6f6c 6f67  ata/Raw_Ortholog
+00014740: 7565 735f 436f 6d70 6172 655f 7261 7465  ues_Compare_rate
+00014750: 2e6b 7364 2e73 7667 290a 0a41 7320 6469  .ksd.svg)..As di
+00014760: 7370 6c61 7965 6420 6162 6f76 652c 2074  splayed above, t
+00014770: 6865 206f 7274 686f 6c6f 676f 7573 202a  he orthologous *
+00014780: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
+00014790: 616c 7565 7320 6265 7774 6565 6e20 2a41  alues bewteen *A
+000147a0: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
+000147b0: 612a 2061 6e64 202a 4163 6f72 7573 2061  a* and *Acorus a
+000147c0: 6d65 7269 6361 6e75 732a 2068 6173 2074  mericanus* has t
+000147d0: 6865 2068 6967 6865 7374 206d 6f64 652c  he highest mode,
+000147e0: 2069 6e64 6963 6174 696e 6774 6865 2066   indicatingthe f
+000147f0: 6173 7465 7220 7375 6273 7469 7475 7469  aster substituti
+00014800: 6f6e 2072 6174 6520 6f66 202a 4171 7569  on rate of *Aqui
+00014810: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
+00014820: 636f 6d70 6172 6564 2074 6f20 2a50 726f  compared to *Pro
+00014830: 7465 6120 6379 6e61 726f 6964 6573 2a20  tea cynaroides* 
+00014840: 616e 6420 2a56 6974 6973 2076 696e 6966  and *Vitis vinif
+00014850: 6572 612a 2e0a 0a42 6566 6f72 6520 7632  era*...Before v2
+00014860: 2e30 2e32 312c 2074 6865 2067 656e 652d  .0.21, the gene-
+00014870: 7370 6563 6965 7320 6d61 7020 6669 6c65  species map file
+00014880: 2069 7320 6e65 6363 6573 7361 7269 6c79   is neccessarily
+00014890: 206e 6565 6465 6420 666f 7220 6974 7320   needed for its 
+000148a0: 696d 706c 656d 656e 7461 7469 6f6e 2069  implementation i
+000148b0: 6e20 6077 6764 2076 697a 602c 2077 6869  n `wgd viz`, whi
+000148c0: 6368 2073 686f 756c 6420 6265 2061 7574  ch should be aut
+000148d0: 6f6d 6174 656c 7920 7072 6f64 7563 6564  omately produced
+000148e0: 2062 7920 7468 6520 6c61 7374 2060 7767   by the last `wg
+000148f0: 6420 6b73 6460 2073 7465 7020 6769 7665  d ksd` step give
+00014900: 6e20 7468 6520 6073 7061 6972 6020 616e  n the `spair` an
+00014910: 6420 6073 7065 6369 6573 7472 6565 6020  d `speciestree` 
+00014920: 7061 7261 6d65 7465 7273 2e20 5468 6520  parameters. The 
+00014930: 6067 656e 655f 7370 6563 6965 732e 6d61  `gene_species.ma
+00014940: 7060 2068 6173 2063 6f6e 7465 6e74 7320  p` has contents 
+00014950: 6173 2062 656c 6f77 2069 6e20 7768 6963  as below in whic
+00014960: 6820 6561 6368 206c 696e 6520 6973 2074  h each line is t
+00014970: 6865 206a 6f69 6e65 6420 7374 7269 6e67  he joined string
+00014980: 206f 6620 6765 6e65 206e 616d 6520 616e   of gene name an
+00014990: 6420 7370 6563 6965 7320 6e61 6d65 2062  d species name b
+000149a0: 7920 7370 6163 652e 2041 6674 6572 2076  y space. After v
+000149b0: 322e 302e 3231 2028 696e 636c 7564 6564  2.0.21 (included
+000149c0: 292c 2074 6865 2067 656e 652d 7370 6563  ), the gene-spec
+000149d0: 6965 7320 6d61 7020 6669 6c65 2069 7320  ies map file is 
+000149e0: 6e6f 7420 6e65 6363 6573 7361 7269 6c79  not neccessarily
+000149f0: 206e 6565 6465 6420 616e 796d 6f72 652e   needed anymore.
+00014a00: 0a0a 6060 600a 4171 636f 6536 4730 3537  ..```.Aqcoe6G057
+00014a10: 3830 302e 3120 4171 7569 6c65 6769 615f  800.1 Aquilegia_
+00014a20: 636f 6572 756c 6561 0a56 7669 5f56 4954  coerulea.Vvi_VIT
+00014a30: 5f32 3031 7330 3031 3167 3031 3533 302e  _201s0011g01530.
+00014a40: 3120 5669 7469 735f 7669 6e69 6665 7261  1 Vitis_vinifera
+00014a50: 0a50 6379 5f50 726f 6379 3031 6730 3835  .Pcy_Procy01g085
+00014a60: 3130 2050 726f 7465 615f 6379 6e61 726f  10 Protea_cynaro
+00014a70: 6964 6573 0a41 616d 5f41 636f 7261 2e30  ides.Aam_Acora.0
+00014a80: 3447 3134 3239 3030 2e31 2041 636f 7275  4G142900.1 Acoru
+00014a90: 735f 616d 6572 6963 616e 7573 0a60 6060  s_americanus.```
+00014aa0: 0a0a 4120 6d6f 7265 2063 6f6d 706c 6578  ..A more complex
+00014ab0: 2070 6c6f 7420 6361 6e20 6265 206d 6164   plot can be mad
+00014ac0: 6520 6279 2061 6464 2074 6865 2066 6c61  e by add the fla
+00014ad0: 6720 602d 2d70 6c6f 7465 6c6d 6d60 2073  g `--plotelmm` s
+00014ae0: 7563 6820 7468 6174 2074 6865 2045 4c4d  uch that the ELM
+00014af0: 4d20 6d69 7874 7572 6520 6d6f 6465 6c69  M mixture modeli
+00014b00: 6e67 206f 6620 7072 6f76 6964 6564 2070  ng of provided p
+00014b10: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
+00014b20: 533c 2f73 7562 3e20 6361 6e20 6265 2073  S</sub> can be s
+00014b30: 7570 6572 696d 706f 7365 642c 2075 7369  uperimposed, usi
+00014b40: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00014b50: 656c 6f77 2e0a 0a60 6060 0a77 6764 2076  elow...```.wgd v
+00014b60: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
+00014b70: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
+00014b80: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
+00014b90: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
+00014ba0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+00014bb0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00014bc0: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
+00014bd0: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
+00014be0: 6967 6874 202d 6170 2077 6764 5f73 796e  ight -ap wgd_syn
+00014bf0: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
+00014c00: 686f 7270 6f69 6e74 732e 7478 7420 2d6f  horpoints.txt -o
+00014c10: 2077 6764 5f76 697a 5f6d 6978 6564 5f4b   wgd_viz_mixed_K
+00014c20: 735f 656c 6d6d 202d 2d73 7061 6972 2022  s_elmm --spair "
+00014c30: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00014c40: 6561 3b50 726f 7465 615f 6379 6e61 726f  ea;Protea_cynaro
+00014c50: 6964 6573 2220 2d2d 7370 6169 7220 2241  ides" --spair "A
+00014c60: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00014c70: 613b 5669 7469 735f 7669 6e69 6665 7261  a;Vitis_vinifera
+00014c80: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+00014c90: 6567 6961 5f63 6f65 7275 6c65 613b 4163  egia_coerulea;Ac
+00014ca0: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
+00014cb0: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00014cc0: 6769 615f 636f 6572 756c 6561 3b41 7175  gia_coerulea;Aqu
+00014cd0: 696c 6567 6961 5f63 6f65 7275 6c65 6122  ilegia_coerulea"
+00014ce0: 202d 2d67 736d 6170 2067 656e 655f 7370   --gsmap gene_sp
+00014cf0: 6563 6965 732e 6d61 7020 2d2d 706c 6f74  ecies.map --plot
+00014d00: 6b64 6520 2d2d 706c 6f74 656c 6d6d 0a60  kde --plotelmm.`
+00014d10: 6060 0a0a 215b 5d28 6461 7461 2f41 7175  ``..![](data/Aqu
+00014d20: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00014d30: 476c 6f62 616c 6d72 6268 4b73 5f45 6c6d  GlobalmrbhKs_Elm
+00014d40: 6d5f 436f 7272 6563 7465 642e 6b73 642e  m_Corrected.ksd.
+00014d50: 7376 6729 0a0a 4672 6f6d 2074 6865 206d  svg)..From the m
+00014d60: 6978 6564 202a 4b2a 3c73 7562 3e53 3c2f  ixed *K*<sub>S</
+00014d70: 7375 623e 2070 6c6f 7420 6162 6f76 652c  sub> plot above,
+00014d80: 2077 6520 6361 6e20 7365 6520 7468 6174   we can see that
+00014d90: 2074 6865 206f 7074 696d 697a 6564 206c   the optimized l
+00014da0: 6f67 6e6f 726d 616c 2063 6f6d 706f 6e65  ognormal compone
+00014db0: 6e74 2062 2077 6974 6820 6d6f 6465 2031  nt b with mode 1
+00014dc0: 2e32 2069 7320 796f 756e 6765 7220 7468  .2 is younger th
+00014dd0: 616e 2074 6865 2063 6f72 7265 6374 6564  an the corrected
+00014de0: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
+00014df0: 3c73 7562 3e53 3c2f 7375 623e 206d 6f64  <sub>S</sub> mod
+00014e00: 6520 7769 7468 202a 5072 6f74 6561 2063  e with *Protea c
+00014e10: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
+00014e20: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
+00014e30: 2831 2e33 3920 616e 6420 312e 3437 2c20  (1.39 and 1.47, 
+00014e40: 7265 7370 6563 7469 7665 6c79 292e 0a0a  respectively)...
+00014e50: 4265 7369 6465 732c 2077 6520 6361 6e20  Besides, we can 
+00014e60: 616c 736f 2061 6464 2074 6865 2047 4d4d  also add the GMM
+00014e70: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+00014e80: 6720 6f66 2061 6e63 686f 7220 2a4b 2a3c  g of anchor *K*<
+00014e90: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00014ea0: 6573 2077 6974 6820 7468 6520 666c 6167  es with the flag
+00014eb0: 2060 2d2d 706c 6f74 6170 676d 6d60 2c20   `--plotapgmm`, 
+00014ec0: 7573 696e 6720 7468 6520 636f 6d6d 616e  using the comman
+00014ed0: 6420 6265 6c6f 772e 0a0a 6060 600a 7767  d below...```.wg
+00014ee0: 6420 7669 7a20 2d64 2077 6764 5f67 6c6f  d viz -d wgd_glo
+00014ef0: 6261 6c6d 7262 685f 6b73 2f67 6c6f 6261  balmrbh_ks/globa
+00014f00: 6c5f 4d52 4248 2e74 7376 2e6b 732e 7473  l_MRBH.tsv.ks.ts
+00014f10: 7620 2d2d 6578 7472 6170 6172 616e 6f6d  v --extraparanom
+00014f20: 656b 7320 7767 645f 6b73 642f 4171 7569  eks wgd_ksd/Aqui
+00014f30: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00014f40: 7376 2e6b 732e 7473 7620 2d73 7020 7370  sv.ks.tsv -sp sp
+00014f50: 6563 6965 7374 7265 652e 6e77 202d 2d72  eciestree.nw --r
+00014f60: 6577 6569 6768 7420 2d61 7020 7767 645f  eweight -ap wgd_
+00014f70: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00014f80: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
+00014f90: 202d 6f20 7767 645f 7669 7a5f 6d69 7865   -o wgd_viz_mixe
+00014fa0: 645f 4b73 5f65 6c6d 6d20 2d2d 7370 6169  d_Ks_elmm --spai
+00014fb0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
+00014fc0: 7275 6c65 613b 5072 6f74 6561 5f63 796e  rulea;Protea_cyn
+00014fd0: 6172 6f69 6465 7322 202d 2d73 7061 6972  aroides" --spair
+00014fe0: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
+00014ff0: 756c 6561 3b56 6974 6973 5f76 696e 6966  ulea;Vitis_vinif
+00015000: 6572 6122 202d 2d73 7061 6972 2022 4171  era" --spair "Aq
+00015010: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015020: 3b41 636f 7275 735f 616d 6572 6963 616e  ;Acorus_american
+00015030: 7573 2220 2d2d 7370 6169 7220 2241 7175  us" --spair "Aqu
+00015040: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
+00015050: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00015060: 6561 2220 2d2d 6773 6d61 7020 6765 6e65  ea" --gsmap gene
+00015070: 5f73 7065 6369 6573 2e6d 6170 202d 2d70  _species.map --p
+00015080: 6c6f 746b 6465 202d 2d70 6c6f 7461 7067  lotkde --plotapg
+00015090: 6d6d 6d0a 6060 600a 0a21 5b5d 2864 6174  mmm.```..![](dat
+000150a0: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
+000150b0: 756c 6561 5f47 6c6f 6261 6c6d 7262 684b  ulea_GlobalmrbhK
+000150c0: 735f 4170 676d 6d5f 436f 7272 6563 7465  s_Apgmm_Correcte
+000150d0: 642e 6b73 642e 7376 6729 0a0a 4173 206d  d.ksd.svg)..As m
+000150e0: 616e 6966 6573 7465 6420 6162 6f76 652c  anifested above,
+000150f0: 2074 6865 2061 6e63 686f 7220 2a4b 2a3c   the anchor *K*<
+00015100: 7375 623e 533c 2f73 7562 3e20 636f 6d70  sub>S</sub> comp
+00015110: 6f6e 656e 7420 3220 7769 7468 206d 6f64  onent 2 with mod
+00015120: 6520 312e 3238 2069 7320 616c 736f 2079  e 1.28 is also y
+00015130: 6f75 6e67 6572 2074 6861 6e20 7468 6520  ounger than the 
+00015140: 636f 7272 6563 7465 6420 6f72 7468 6f6c  corrected orthol
+00015150: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00015160: 2f73 7562 3e20 6d6f 6465 2077 6974 6820  /sub> mode with 
+00015170: 2a50 726f 7465 6120 6379 6e61 726f 6964  *Protea cynaroid
+00015180: 6573 2a20 616e 6420 2a56 6974 6973 2076  es* and *Vitis v
+00015190: 696e 6966 6572 612a 2e20 4275 7420 7765  inifera*. But we
+000151a0: 206e 6565 6420 746f 2062 6520 6f66 2063   need to be of c
+000151b0: 6f75 7273 6520 6361 7574 696f 7573 2074  ourse cautious t
+000151c0: 6861 7420 7375 6368 2064 6973 7469 6e63  hat such distinc
+000151d0: 7469 6f6e 2063 6f6d 6573 2077 6974 6820  tion comes with 
+000151e0: 7468 6520 756e 6365 7274 6169 6e74 6965  the uncertaintie
+000151f0: 7320 696e 7472 6f64 7563 6564 2066 726f  s introduced fro
+00015200: 6d20 7468 6520 6170 706c 6965 6420 6d69  m the applied mi
+00015210: 7874 7572 6520 6d6f 6465 6c69 6e67 206d  xture modeling m
+00015220: 6574 686f 646f 6c6f 6779 2069 6e20 7465  ethodology in te
+00015230: 726d 7320 6f66 2066 6f72 2069 6e73 7461  rms of for insta
+00015240: 6e63 6520 6469 6666 6572 656e 7420 696e  nce different in
+00015250: 6974 6961 6c69 7a61 7469 6f6e 2070 6f69  itialization poi
+00015260: 6e74 7320 616e 6420 7468 6520 6973 7375  nts and the issu
+00015270: 6520 6f66 206f 7665 7266 6974 7469 6e67  e of overfitting
+00015280: 2061 6e64 2074 6865 2073 6973 7465 7220   and the sister 
+00015290: 7370 6563 6965 7373 2061 646f 7074 6564  speciess adopted
+000152a0: 2069 6e20 7468 6174 2074 6865 7265 206d   in that there m
+000152b0: 6967 6874 2062 6520 7370 6563 6965 7320  ight be species 
+000152c0: 7769 7468 206d 6f72 6520 6469 7370 6172  with more dispar
+000152d0: 6174 6520 7375 6273 7469 7475 7469 6f6e  ate substitution
+000152e0: 2072 6174 6573 2074 6861 6e20 7468 6520   rates than the 
+000152f0: 6f6e 6520 7765 2063 686f 7365 2e0a 0a41  one we chose...A
+00015300: 6464 696e 6720 626f 7468 2045 4c4d 4d20  dding both ELMM 
+00015310: 7265 7375 6c74 2066 6f72 2070 6172 616e  result for paran
+00015320: 6f6d 6520 616e 6420 474d 4d20 7265 7375  ome and GMM resu
+00015330: 6c74 2066 6f72 2061 6e63 686f 7220 2a4b  lt for anchor *K
+00015340: 2a3c 7375 623e 533c 2f73 7562 3e20 6361  *<sub>S</sub> ca
+00015350: 6e20 6265 2061 6368 6965 7665 6420 6a75  n be achieved ju
+00015360: 7374 2062 7920 6164 6420 7468 6520 7477  st by add the tw
+00015370: 6f20 666c 6167 7320 6d65 6e74 696f 6e65  o flags mentione
+00015380: 6420 6162 6f76 652c 2075 7369 6e67 2074  d above, using t
+00015390: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000153a0: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
+000153b0: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
+000153c0: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
+000153d0: 7473 762e 6b73 2e74 7376 202d 2d65 7874  tsv.ks.tsv --ext
+000153e0: 7261 7061 7261 6e6f 6d65 6b73 2077 6764  raparanomeks wgd
+000153f0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
+00015400: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00015410: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00015420: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00015430: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
+00015440: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
+00015450: 6f69 6e74 732e 7478 7420 2d6f 2077 6764  oints.txt -o wgd
+00015460: 5f76 697a 5f6d 6978 6564 5f4b 735f 656c  _viz_mixed_Ks_el
+00015470: 6d6d 202d 2d73 7061 6972 2022 4171 7569  mm --spair "Aqui
+00015480: 6c65 6769 615f 636f 6572 756c 6561 3b50  legia_coerulea;P
+00015490: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+000154a0: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+000154b0: 6567 6961 5f63 6f65 7275 6c65 613b 5669  egia_coerulea;Vi
+000154c0: 7469 735f 7669 6e69 6665 7261 2220 2d2d  tis_vinifera" --
+000154d0: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
+000154e0: 5f63 6f65 7275 6c65 613b 4163 6f72 7573  _coerulea;Acorus
+000154f0: 5f61 6d65 7269 6361 6e75 7322 202d 2d73  _americanus" --s
+00015500: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
+00015510: 636f 6572 756c 6561 3b41 7175 696c 6567  coerulea;Aquileg
+00015520: 6961 5f63 6f65 7275 6c65 6122 202d 2d67  ia_coerulea" --g
+00015530: 736d 6170 2067 656e 655f 7370 6563 6965  smap gene_specie
+00015540: 732e 6d61 7020 2d2d 706c 6f74 6b64 6520  s.map --plotkde 
+00015550: 2d2d 706c 6f74 656c 6d6d 202d 2d70 6c6f  --plotelmm --plo
+00015560: 7461 7067 6d6d 6d0a 6060 600a 0a21 5b5d  tapgmmm.```..![]
+00015570: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
+00015580: 636f 6572 756c 6561 5f47 6c6f 6261 6c6d  coerulea_Globalm
+00015590: 7262 684b 735f 456c 6d6d 5f41 7067 6d6d  rbhKs_Elmm_Apgmm
+000155a0: 5f43 6f72 7265 6374 6564 2e6b 7364 2e73  _Corrected.ksd.s
+000155b0: 7667 290a 0a41 6e20 616c 7465 726e 6174  vg)..An alternat
+000155c0: 6976 6520 7761 7920 746f 2063 616c 6375  ive way to calcu
+000155d0: 6c61 7465 2074 6865 206f 7274 686f 6c6f  late the ortholo
+000155e0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000155f0: 7375 623e 2069 7320 746f 2064 6972 6563  sub> is to direc
+00015600: 746c 7920 7573 6520 7468 6520 6f72 7468  tly use the orth
+00015610: 6f67 726f 7570 7320 696e 7374 6561 6420  ogroups instead 
+00015620: 6f66 2067 6c6f 6261 6c20 4d52 4248 2066  of global MRBH f
+00015630: 616d 696c 792e 2054 6861 7420 7761 7920  amily. That way 
+00015640: 7765 2064 6f6e 2774 2075 7365 2074 6865  we don't use the
+00015650: 2070 7265 2d69 6e66 6572 7265 6420 7061   pre-inferred pa
+00015660: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
+00015670: 3c2f 7375 623e 2062 7574 2074 6865 2070  </sub> but the p
+00015680: 6172 616c 6f67 6f75 7320 6765 6e65 2070  aralogous gene p
+00015690: 6169 7273 2069 6e73 6964 6520 6561 6368  airs inside each
+000156a0: 206f 7274 686f 6772 6f75 7020 696e 7374   orthogroup inst
+000156b0: 6561 642e 2054 6f20 6163 6869 6576 6520  ead. To achieve 
+000156c0: 7468 6174 2c20 7765 2066 6972 7374 206e  that, we first n
+000156d0: 6565 6420 746f 2069 6e66 6572 206f 7274  eed to infer ort
+000156e0: 686f 6772 6f75 7073 2075 7369 6e67 2074  hogroups using t
+000156f0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+00015700: 2e0a 0a60 6060 0a77 6764 2064 6d64 2041  ...```.wgd dmd A
+00015710: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00015720: 6120 5072 6f74 6561 5f63 796e 6172 6f69  a Protea_cynaroi
+00015730: 6465 7320 4163 6f72 7573 5f61 6d65 7269  des Acorus_ameri
+00015740: 6361 6e75 7320 5669 7469 735f 7669 6e69  canus Vitis_vini
+00015750: 6665 7261 202d 2d6f 7274 686f 696e 6665  fera --orthoinfe
+00015760: 7220 2d6f 2077 6764 5f6f 7274 686f 2028  r -o wgd_ortho (
+00015770: 2d2d 6f6e 6c79 6f72 7468 6f29 200a 6060  --onlyortho) .``
+00015780: 600a 0a55 7365 7273 2063 616e 2064 6563  `..Users can dec
+00015790: 6964 6520 746f 206f 6e6c 7920 636f 6e64  ide to only cond
+000157a0: 7563 7420 7468 6520 6f72 7468 6f67 726f  uct the orthogro
+000157b0: 7570 2061 6e61 6c79 7369 7320 7768 696c  up analysis whil
+000157c0: 6520 736b 6970 7069 6e67 206f 7468 6572  e skipping other
+000157d0: 2061 6e61 6c79 7369 7320 6279 2061 6464   analysis by add
+000157e0: 696e 6720 7468 6520 666c 6167 2060 2d2d  ing the flag `--
+000157f0: 6f6e 6c79 6f72 7468 6f60 2e20 4e65 7874  onlyortho`. Next
+00015800: 2073 7465 7020 6973 2074 6865 2073 616d   step is the sam
+00015810: 6520 7769 7468 2067 6c6f 6261 6c20 4d52  e with global MR
+00015820: 4248 2066 616d 696c 7920 6578 6365 7074  BH family except
+00015830: 2074 6861 7420 7765 2064 6f6e 2774 2075   that we don't u
+00015840: 7365 2074 6865 2065 7874 7261 2070 7265  se the extra pre
+00015850: 2d69 6e66 6572 7265 6420 7061 7261 6e6f  -inferred parano
+00015860: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+00015870: 623e 2061 6e79 6d6f 7265 2e20 5765 2069  b> anymore. We i
+00015880: 6e66 6572 2074 6865 202a 4b2a 3c73 7562  nfer the *K*<sub
+00015890: 3e53 3c2f 7375 623e 2075 7369 6e67 2074  >S</sub> using t
+000158a0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000158b0: 2e20 4e6f 7465 2074 6861 7420 7468 6520  . Note that the 
+000158c0: 7072 6f67 7261 6d20 6077 6764 2076 697a  program `wgd viz
+000158d0: 6020 6361 6e20 706c 6f74 2074 6865 2073  ` can plot the s
+000158e0: 616d 6520 6a75 7374 2061 7320 7368 6f77  ame just as show
+000158f0: 6e20 6162 6f76 652e 0a0a 6060 600a 7767  n above...```.wg
+00015900: 6420 6b73 6420 7767 645f 6f72 7468 6f2f  d ksd wgd_ortho/
+00015910: 4f72 7468 6f67 726f 7570 732e 7370 2e74  Orthogroups.sp.t
+00015920: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00015930: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00015940: 202d 6f20 7767 645f 6f72 7468 6f5f 6b73   -o wgd_ortho_ks
+00015950: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00015960: 6769 615f 636f 6572 756c 6561 3b50 726f  gia_coerulea;Pro
+00015970: 7465 615f 6379 6e61 726f 6964 6573 2220  tea_cynaroides" 
+00015980: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00015990: 6961 5f63 6f65 7275 6c65 613b 5669 7469  ia_coerulea;Viti
+000159a0: 735f 7669 6e69 6665 7261 2220 2d2d 7370  s_vinifera" --sp
+000159b0: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
+000159c0: 6f65 7275 6c65 613b 4163 6f72 7573 5f61  oerulea;Acorus_a
+000159d0: 6d65 7269 6361 6e75 7322 202d 2d73 7061  mericanus" --spa
+000159e0: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
+000159f0: 6572 756c 6561 3b41 7175 696c 6567 6961  erulea;Aquilegia
+00015a00: 5f63 6f65 7275 6c65 6122 202d 2d70 6c6f  _coerulea" --plo
+00015a10: 746b 6465 2028 2d61 7020 7767 645f 7379  tkde (-ap wgd_sy
+00015a20: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+00015a30: 6368 6f72 706f 696e 7473 2e74 7874 290a  chorpoints.txt).
+00015a40: 6060 600a 0a21 5b5d 2864 6174 612f 4171  ```..![](data/Aq
+00015a50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015a60: 5f4f 7274 686f 4b73 5f43 6f72 7265 6374  _OrthoKs_Correct
+00015a70: 6564 2e6b 7364 2e73 7667 290a 0a41 7320  ed.ksd.svg)..As 
+00015a80: 7368 6f77 6e20 6162 6f76 652c 2074 6865  shown above, the
+00015a90: 206e 756d 6265 7220 6f66 2062 6f74 6820   number of both 
+00015aa0: 7061 7261 6c6f 676f 7573 2067 656e 6520  paralogous gene 
+00015ab0: 7061 6972 7320 616e 6420 6f72 7468 6f6c  pairs and orthol
+00015ac0: 6f67 6f75 7320 6765 6e65 2070 6169 7273  ogous gene pairs
+00015ad0: 2069 7320 6469 6666 6572 656e 7420 7468   is different th
+00015ae0: 616e 2074 6865 206f 6e65 2066 726f 6d20  an the one from 
+00015af0: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
+00015b00: 6c79 2069 6e20 7468 6174 2068 6572 6520  ly in that here 
+00015b10: 7765 2070 6c6f 7474 6564 2061 6c6c 206f  we plotted all o
+00015b20: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
+00015b30: 7061 6972 7320 696e 7374 6561 6420 6f66  pairs instead of
+00015b40: 206f 6e6c 7920 676c 6f62 616c 204d 5242   only global MRB
+00015b50: 4820 616e 6420 706f 7465 6e74 6961 6c6c  H and potentiall
+00015b60: 7920 6e65 7720 7061 7261 6c6f 676f 7573  y new paralogous
+00015b70: 2067 656e 6520 7061 6972 7320 6d69 6768   gene pairs migh
+00015b80: 7420 6265 2070 726f 6475 6365 6420 696e  t be produced in
+00015b90: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
+00015ba0: 696e 6665 7265 6e63 6520 7374 6570 2c20  inference step, 
+00015bb0: 746f 6765 7468 6572 2077 6974 6820 6469  together with di
+00015bc0: 6666 6572 656e 7420 7265 6361 6c63 756c  fferent recalcul
+00015bd0: 6174 6564 2077 6569 6768 7473 2e0a 0a49  ated weights...I
+00015be0: 6620 6f6e 6520 6f6e 6c79 2077 616e 7465  f one only wante
+00015bf0: 6420 746f 2070 6c6f 7420 7468 6520 6f72  d to plot the or
+00015c00: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00015c10: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00015c20: 6275 7469 6f6e 2c20 6173 2077 6520 6861  bution, as we ha
+00015c30: 7665 2061 6c72 6561 6479 2073 686f 776e  ve already shown
+00015c40: 2061 626f 7665 2c20 6974 2063 616e 2062   above, it can b
+00015c50: 6520 6561 7369 6c79 2061 6368 6965 7665  e easily achieve
+00015c60: 6420 6279 2072 656d 6f76 696e 6720 7468  d by removing th
+00015c70: 6520 7061 7261 6c6f 676f 7573 2073 7065  e paralogous spe
+00015c80: 6369 6573 2070 6169 7220 6041 7175 696c  cies pair `Aquil
+00015c90: 6567 6961 5f63 6f65 7275 6c65 613b 4171  egia_coerulea;Aq
+00015ca0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015cb0: 602c 2075 7369 6e67 2074 6865 2063 6f6d  `, using the com
+00015cc0: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
+00015cd0: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+00015ce0: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
+00015cf0: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
+00015d00: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00015d10: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00015d20: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
+00015d30: 6d72 6268 5f6f 6e6c 796f 7274 686f 5f6b  mrbh_onlyortho_k
+00015d40: 7320 2d2d 7370 6169 7220 2241 7175 696c  s --spair "Aquil
+00015d50: 6567 6961 5f63 6f65 7275 6c65 613b 5072  egia_coerulea;Pr
+00015d60: 6f74 6561 5f63 796e 6172 6f69 6465 7322  otea_cynaroides"
+00015d70: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00015d80: 6769 615f 636f 6572 756c 6561 3b56 6974  gia_coerulea;Vit
+00015d90: 6973 5f76 696e 6966 6572 6122 202d 2d73  is_vinifera" --s
+00015da0: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
+00015db0: 636f 6572 756c 6561 3b41 636f 7275 735f  coerulea;Acorus_
+00015dc0: 616d 6572 6963 616e 7573 2220 2d2d 6773  americanus" --gs
+00015dd0: 6d61 7020 6765 6e65 5f73 7065 6369 6573  map gene_species
+00015de0: 2e6d 6170 0a60 6060 0a0a 5765 2063 616e  .map.```..We can
+00015df0: 2063 6c65 6172 6c79 2073 6565 2074 6861   clearly see tha
+00015e00: 7420 2a56 6974 6973 2076 696e 6966 6572  t *Vitis vinifer
+00015e10: 612a 2068 6173 2068 6967 6865 7220 7375  a* has higher su
+00015e20: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
+00015e30: 7468 616e 202a 5072 6f74 6561 2063 796e  than *Protea cyn
+00015e40: 6172 6f69 6465 732a 2069 6e20 7468 6174  aroides* in that
+00015e50: 2074 6865 6972 206f 7274 686f 6c6f 676f   their orthologo
+00015e60: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+00015e70: 623e 2070 6561 6b73 2077 6974 6820 2a41  b> peaks with *A
+00015e80: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
+00015e90: 612a 2c20 616c 7468 6f75 6768 2072 6570  a*, although rep
+00015ea0: 7265 7365 6e74 696e 6720 7468 6520 7361  resenting the sa
+00015eb0: 6d65 2064 6976 6572 6765 6e63 6520 6576  me divergence ev
+00015ec0: 656e 742c 2064 6966 6665 7265 6420 696e  ent, differed in
+00015ed0: 2073 7562 7374 6974 7574 696f 6e20 6469   substitution di
+00015ee0: 7374 616e 6365 2e0a 215b 5d28 6461 7461  stance..![](data
+00015ef0: 2f52 6177 5f4f 7274 686f 6c6f 6775 6573  /Raw_Orthologues
+00015f00: 2e6b 7364 2e73 7667 290a 0a41 6674 6572  .ksd.svg)..After
+00015f10: 2074 6865 2070 6879 6c6f 6765 6e65 7469   the phylogeneti
+00015f20: 6320 7469 6d69 6e67 206f 6620 7468 6520  c timing of the 
+00015f30: 5261 6e75 6e63 756c 616c 6573 2057 4744  Ranunculales WGD
+00015f40: 2c20 7765 2063 616e 2066 7572 7468 6572  , we can further
+00015f50: 2069 6e66 6572 2069 7473 2061 6273 6f6c   infer its absol
+00015f60: 7574 6520 6167 652e 2046 6972 7374 2077  ute age. First w
+00015f70: 6520 696e 6665 7220 7468 6520 6372 6564  e infer the cred
+00015f80: 6962 6c65 2072 616e 6765 206f 6620 616e  ible range of an
+00015f90: 6368 6f72 2070 6169 7273 2062 7920 2a4b  chor pairs by *K
+00015fa0: 2a3c 7375 623e 533c 2f73 7562 3e20 6865  *<sub>S</sub> he
+00015fb0: 7572 6973 7469 6361 6c6c 7920 7573 696e  uristically usin
+00015fc0: 6720 7468 6520 7072 6f67 7261 6d20 6077  g the program `w
+00015fd0: 6764 2070 6561 6b60 2e0a 0a60 6060 0a77  gd peak`...```.w
+00015fe0: 6764 2070 6561 6b20 2d2d 6865 7572 6973  gd peak --heuris
+00015ff0: 7469 6320 7767 645f 6b73 642f 4171 7569  tic wgd_ksd/Aqui
+00016000: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00016010: 7376 2e6b 732e 7473 7620 2d61 7020 7767  sv.ks.tsv -ap wg
+00016020: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00016030: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
+00016040: 7874 202d 736d 2077 6764 5f73 796e 2f69  xt -sm wgd_syn/i
+00016050: 6164 686f 7265 2d6f 7574 2f73 6567 6d65  adhore-out/segme
+00016060: 6e74 732e 7478 7420 2d6c 6520 7767 645f  nts.txt -le wgd_
+00016070: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00016080: 6c69 7374 5f65 6c65 6d65 6e74 732e 7478  list_elements.tx
+00016090: 7420 2d6d 7020 7767 645f 7379 6e2f 6961  t -mp wgd_syn/ia
+000160a0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+000160b0: 6c69 636f 6e5f 7061 6972 732e 7478 7420  licon_pairs.txt 
+000160c0: 2d6f 2077 6764 5f70 6561 6b0a 6060 600a  -o wgd_peak.```.
+000160d0: 0a21 5b5d 2864 6174 612f 416e 6368 6f72  .![](data/Anchor
+000160e0: 4b73 5f50 6561 6b43 495f 4171 7569 6c65  Ks_PeakCI_Aquile
+000160f0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00016100: 2e6b 732e 7473 765f 6e6f 6465 5f77 6569  .ks.tsv_node_wei
+00016110: 6768 7465 642e 7376 6729 0a0a 4173 2073  ghted.svg)..As s
+00016120: 686f 776e 2061 626f 7665 2c20 7765 2061  hown above, we a
+00016130: 7373 756d 6564 2061 206c 6f67 6e6f 726d  ssumed a lognorm
+00016140: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
+00016150: 6174 2074 6865 2070 6561 6b20 6c6f 6361  at the peak loca
+00016160: 7469 6f6e 2064 6574 6563 7465 6420 6279  tion detected by
+00016170: 2074 6865 2060 7369 676e 616c 6020 6d6f   the `signal` mo
+00016180: 6475 6c65 206f 6620 6073 6369 7079 6020  dule of `scipy` 
+00016190: 6c69 6272 6172 792e 2054 6865 2039 3525  library. The 95%
+000161a0: 2063 6f6e 6669 6465 6e63 6520 6c65 7665   confidence leve
+000161b0: 6c20 6f66 2074 6865 206c 6f67 6e6f 726d  l of the lognorm
+000161c0: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
+000161d0: 7761 7320 6170 706c 6965 642c 2069 2e65  was applied, i.e
+000161e0: 2e2c 2030 2e36 382d 322e 3734 2c20 696e  ., 0.68-2.74, in
+000161f0: 2066 7572 7468 6572 206d 6f6c 6563 756c   further molecul
+00016200: 6172 2064 6174 696e 672e 2054 6865 2066  ar dating. The f
+00016210: 696c 6520 6041 7175 696c 6567 6961 5f63  ile `Aquilegia_c
+00016220: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00016230: 7376 5f39 3525 4349 5f41 505f 666f 725f  sv_95%CI_AP_for_
+00016240: 6461 7469 6e67 5f77 6569 6768 7465 645f  dating_weighted_
+00016250: 666f 726d 6174 2e74 7376 6020 6973 2077  format.tsv` is w
+00016260: 6861 7420 7765 206e 6565 6420 666f 7220  hat we need for 
+00016270: 6e65 7874 2073 7465 702e 2054 6f20 6275  next step. To bu
+00016280: 696c 6420 7468 6520 6f72 7468 6f67 726f  ild the orthogro
+00016290: 7570 7320 7573 6564 2069 6e20 7068 796c  ups used in phyl
+000162a0: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
+000162b0: 2077 6520 6e65 6564 2074 6f20 7365 6c65   we need to sele
+000162c0: 6374 2073 6f6d 6520 7370 6563 6965 7320  ct some species 
+000162d0: 616e 6420 666f 726d 2061 2073 7461 7274  and form a start
+000162e0: 696e 6720 7472 6565 2077 6974 6820 7072  ing tree with pr
+000162f0: 6f70 6572 2066 6f73 7369 6c20 6361 6c69  oper fossil cali
+00016300: 6272 6174 696f 6e73 2e20 5765 2070 726f  brations. We pro
+00016310: 7669 6465 206f 6e65 2069 6e20 6d63 6d63  vide one in mcmc
+00016320: 7472 6565 2066 6f72 6d61 7420 6173 2062  tree format as b
+00016330: 656c 6f77 2e0a 0a60 6060 0a31 3720 310a  elow...```.17 1.
+00016340: 2828 2828 506f 7461 6d6f 6765 746f 6e5f  ((((Potamogeton_
+00016350: 6163 7574 6966 6f6c 6975 732c 2853 7069  acutifolius,(Spi
+00016360: 726f 6465 6c61 5f69 6e74 6572 6d65 6469  rodela_intermedi
+00016370: 612c 416d 6f72 7068 6f70 6861 6c6c 7573  a,Amorphophallus
+00016380: 5f6b 6f6e 6a61 6329 292c 2841 6361 6e74  _konjac)),(Acant
+00016390: 686f 6368 6c61 6d79 735f 6272 6163 7465  hochlamys_bracte
+000163a0: 6174 612c 2844 696f 7363 6f72 6561 5f61  ata,(Dioscorea_a
+000163b0: 6c61 7461 2c44 696f 7363 6f72 6561 5f72  lata,Dioscorea_r
+000163c0: 6f74 756e 6461 7461 2929 273e 302e 3536  otundata))'>0.56
+000163d0: 3030 3c31 2e32 3836 3327 2927 3e30 2e38  00<1.2863')'>0.8
+000163e0: 3336 303c 312e 3238 3633 272c 2841 636f  360<1.2863',(Aco
+000163f0: 7275 735f 616d 6572 6963 616e 7573 2c41  rus_americanus,A
+00016400: 636f 7275 735f 7461 7461 7269 6e6f 7769  corus_tatarinowi
+00016410: 6929 2927 3e30 2e38 3336 303c 312e 3238  i))'>0.8360<1.28
+00016420: 3633 272c 2828 2828 5465 7472 6163 656e  63',((((Tetracen
+00016430: 7472 6f6e 5f73 696e 656e 7365 2c54 726f  tron_sinense,Tro
+00016440: 6368 6f64 656e 6472 6f6e 5f61 7261 6c69  chodendron_arali
+00016450: 6f69 6465 7329 2c28 4275 7875 735f 6175  oides),(Buxus_au
+00016460: 7374 726f 7975 6e6e 616e 656e 7369 732c  stroyunnanensis,
+00016470: 4275 7875 735f 7369 6e69 6361 2929 273e  Buxus_sinica))'>
+00016480: 312e 3130 3830 3c31 2e32 3836 3327 2c28  1.1080<1.2863',(
+00016490: 4e65 6c75 6d62 6f5f 6e75 6369 6665 7261  Nelumbo_nucifera
+000164a0: 2c28 5465 6c6f 7065 615f 7370 6563 696f  ,(Telopea_specio
+000164b0: 7369 7373 696d 612c 5072 6f74 6561 5f63  sissima,Protea_c
+000164c0: 796e 6172 6f69 6465 7329 2929 273e 312e  ynaroides)))'>1.
+000164d0: 3130 3830 3c31 2e32 3836 3327 2c28 4171  1080<1.2863',(Aq
+000164e0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000164f0: 5f61 7031 2c41 7175 696c 6567 6961 5f63  _ap1,Aquilegia_c
+00016500: 6f65 7275 6c65 615f 6170 3229 2927 3e31  oerulea_ap2))'>1
+00016510: 2e31 3038 303c 312e 3238 3633 2729 273e  .1080<1.2863')'>
+00016520: 312e 3237 3230 3c32 2e34 3732 3027 3b0a  1.2720<2.4720';.
+00016530: 6060 600a 0a41 7320 7072 6573 656e 7465  ```..As presente
+00016540: 6420 6162 6f76 652c 2074 6865 2066 6f63  d above, the foc
+00016550: 7573 2073 7065 6369 6573 2074 6861 7420  us species that 
+00016560: 6973 2061 626f 7574 2074 6f20 6265 2064  is about to be d
+00016570: 6174 6564 206e 6565 6473 2074 6f20 6265  ated needs to be
+00016580: 2072 6570 6c61 6365 6420 7769 7468 2060   replaced with `
+00016590: 2841 7175 696c 6567 6961 5f63 6f65 7275  (Aquilegia_coeru
+000165a0: 6c65 615f 6170 312c 4171 7569 6c65 6769  lea_ap1,Aquilegi
+000165b0: 615f 636f 6572 756c 6561 5f61 7032 2960  a_coerulea_ap2)`
+000165c0: 2e20 5769 7468 2074 6869 7320 7374 6172  . With this star
+000165d0: 7469 6e67 2074 7265 6520 616e 6420 7072  ting tree and pr
+000165e0: 6564 6f77 6e6c 6f61 6465 6420 6364 7320  edownloaded cds 
+000165f0: 6669 6c65 7320 6f66 2061 6c6c 2074 6865  files of all the
+00016600: 2073 7065 6369 6573 2c20 7765 2063 616e   species, we can
+00016610: 2062 7569 6c64 2074 6865 206f 7274 686f   build the ortho
+00016620: 6772 6f75 7020 7573 6564 2069 6e20 7468  group used in th
+00016630: 6520 6669 6e61 6c20 6d6f 6c65 6375 6c61  e final molecula
+00016640: 7220 6461 7469 6e67 2075 7369 6e67 2074  r dating using t
+00016650: 6865 2063 6f6d 6d61 6e64 2061 7320 6265  he command as be
+00016660: 6c6f 772e 0a0a 6060 600a 7767 6420 646d  low...```.wgd dm
+00016670: 6420 2d66 2041 7175 696c 6567 6961 5f63  d -f Aquilegia_c
+00016680: 6f65 7275 6c65 6120 2d61 7020 7767 645f  oerulea -ap wgd_
+00016690: 7065 616b 2f41 7175 696c 6567 6961 5f63  peak/Aquilegia_c
+000166a0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+000166b0: 7376 5f39 3525 4349 5f41 505f 666f 725f  sv_95%CI_AP_for_
+000166c0: 6461 7469 6e67 5f77 6569 6768 7465 645f  dating_weighted_
+000166d0: 666f 726d 6174 2e74 7376 202d 6f20 7767  format.tsv -o wg
+000166e0: 645f 646d 645f 6f72 7468 6f20 506f 7461  d_dmd_ortho Pota
+000166f0: 6d6f 6765 746f 6e5f 6163 7574 6966 6f6c  mogeton_acutifol
+00016700: 6975 7320 5370 6972 6f64 656c 615f 696e  ius Spirodela_in
+00016710: 7465 726d 6564 6961 2041 6d6f 7270 686f  termedia Amorpho
+00016720: 7068 616c 6c75 735f 6b6f 6e6a 6163 2041  phallus_konjac A
+00016730: 6361 6e74 686f 6368 6c61 6d79 735f 6272  canthochlamys_br
+00016740: 6163 7465 6174 6120 4469 6f73 636f 7265  acteata Dioscore
+00016750: 615f 616c 6174 6120 4469 6f73 636f 7265  a_alata Dioscore
+00016760: 615f 726f 7475 6e64 6174 6120 4163 6f72  a_rotundata Acor
+00016770: 7573 5f61 6d65 7269 6361 6e75 7320 4163  us_americanus Ac
+00016780: 6f72 7573 5f74 6174 6172 696e 6f77 6969  orus_tatarinowii
+00016790: 2054 6574 7261 6365 6e74 726f 6e5f 7369   Tetracentron_si
+000167a0: 6e65 6e73 6520 5472 6f63 686f 6465 6e64  nense Trochodend
+000167b0: 726f 6e5f 6172 616c 696f 6964 6573 2042  ron_aralioides B
+000167c0: 7578 7573 5f61 7573 7472 6f79 756e 6e61  uxus_austroyunna
+000167d0: 6e65 6e73 6973 2042 7578 7573 5f73 696e  nensis Buxus_sin
+000167e0: 6963 6120 4e65 6c75 6d62 6f5f 6e75 6369  ica Nelumbo_nuci
+000167f0: 6665 7261 2054 656c 6f70 6561 5f73 7065  fera Telopea_spe
+00016800: 6369 6f73 6973 7369 6d61 2050 726f 7465  ciosissima Prote
+00016810: 615f 6379 6e61 726f 6964 6573 2041 7175  a_cynaroides Aqu
+00016820: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
+00016830: 6060 600a 0a54 6865 2072 6573 756c 7420  ```..The result 
+00016840: 6669 6c65 2060 6d65 7267 655f 666f 6375  file `merge_focu
+00016850: 735f 6170 2e74 7376 6020 6973 2077 6861  s_ap.tsv` is wha
+00016860: 7420 7765 206e 6565 6420 666f 7220 7468  t we need for th
+00016870: 6520 6669 6e61 6c20 7374 6570 206f 6620  e final step of 
+00016880: 6d6f 6c65 6375 6c61 7220 6461 7469 6e67  molecular dating
+00016890: 2069 6e20 7072 6f67 7261 6d20 6077 6764   in program `wgd
+000168a0: 2066 6f63 7573 602e 0a0a 6060 600a 7767   focus`...```.wg
+000168b0: 6420 666f 6375 7320 2d2d 7072 6f74 6461  d focus --protda
+000168c0: 7469 6e67 202d 2d61 616d 6f64 656c 206c  ting --aamodel l
+000168d0: 6720 7767 645f 646d 645f 6f72 7468 6f2f  g wgd_dmd_ortho/
+000168e0: 6d65 7267 655f 666f 6375 735f 6170 2e74  merge_focus_ap.t
+000168f0: 7376 202d 7370 2064 6174 696e 675f 7472  sv -sp dating_tr
+00016900: 6565 2e6e 7720 2d6f 2077 6764 5f64 6174  ee.nw -o wgd_dat
+00016910: 696e 6720 2d64 206d 636d 6374 7265 6520  ing -d mcmctree 
+00016920: 2d64 7320 2762 7572 6e69 6e20 3d20 3230  -ds 'burnin = 20
+00016930: 3030 2720 2d64 7320 2773 616d 7066 7265  00' -ds 'sampfre
+00016940: 7120 3d20 3130 3030 2720 2d64 7320 276e  q = 1000' -ds 'n
+00016950: 7361 6d70 6c65 203d 2032 3030 3030 2720  sample = 20000' 
+00016960: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
+00016970: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
+00016980: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
+00016990: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
+000169a0: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
+000169b0: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
+000169c0: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
+000169d0: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
+000169e0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+000169f0: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
+00016a00: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
+00016a10: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
+00016a20: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
+00016a30: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
+00016a40: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
+00016a50: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
+00016a60: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
+00016a70: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
+00016a80: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+00016a90: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00016aa0: 6c65 610a 6060 600a 0a48 6572 6520 7765  lea.```..Here we
+00016ab0: 206f 6e6c 7920 696d 706c 656d 656e 7465   only implemente
+00016ac0: 6420 7468 6520 636f 6e63 6174 656e 6174  d the concatenat
+00016ad0: 696f 6e20 616e 616c 7973 6973 2075 7369  ion analysis usi
+00016ae0: 6e67 2070 726f 7465 696e 2073 6571 7565  ng protein seque
+00016af0: 6e63 6520 6279 2061 6464 696e 6720 7468  nce by adding th
+00016b00: 6520 666c 6167 2060 2d2d 7072 6f74 6461  e flag `--protda
+00016b10: 7469 6e67 6020 616e 6420 7765 2073 6574  ting` and we set
+00016b20: 2074 6865 2070 6172 616d 6574 6572 2066   the parameter f
+00016b30: 6f72 2060 6d63 6d63 7472 6565 6020 7669  or `mcmctree` vi
+00016b40: 6120 7468 6520 6f70 7469 6f6e 2060 2d64  a the option `-d
+00016b50: 7360 2e20 4e6f 7465 2074 6861 7420 6f74  s`. Note that ot
+00016b60: 6865 7220 6461 7469 6e67 2070 726f 6772  her dating progr
+00016b70: 616d 2073 7563 6820 6173 2060 7238 7360  am such as `r8s`
+00016b80: 2061 6e64 2060 6265 6173 7460 2061 7265   and `beast` are
+00016b90: 2061 6c73 6f20 6176 6169 6c61 626c 6520   also available 
+00016ba0: 6769 7665 6e20 736f 6d65 206d 616e 6461  given some manda
+00016bb0: 746f 7279 2070 6172 616d 6574 6572 732e  tory parameters.
+00016bc0: 2054 6865 2066 696e 616c 206c 6f67 206f   The final log o
+00016bd0: 6620 7468 6520 7375 6363 6573 7366 756c  f the successful
+00016be0: 2072 756e 2069 7320 6173 2062 656c 6f77   run is as below
+00016bf0: 2e0a 0a60 6060 0a31 363a 3034 3a32 3520  ...```.16:04:25 
+00016c00: 494e 464f 2020 2020 2052 756e 6e69 6e67  INFO     Running
+00016c10: 206d 636d 6374 7265 6520 7573 696e 6720   mcmctree using 
+00016c20: 4865 7373 6961 6e20 6d61 7472 6978 206f  Hessian matrix o
+00016c30: 6620 4c47 2b47 616d 6d61 2020 636f 7265  f LG+Gamma  core
+00016c40: 2e70 793a 3936 370a 2020 2020 2020 2020  .py:967.        
+00016c50: 2020 2020 2020 2020 2020 666f 7220 7072            for pr
+00016c60: 6f74 6569 6e20 6d6f 6465 6c0a 3233 3a34  otein model.23:4
+00016c70: 393a 3337 2049 4e46 4f20 2020 2020 506f  9:37 INFO     Po
+00016c80: 7374 6572 696f 7220 6d65 616e 2066 6f72  sterior mean for
+00016c90: 2074 6865 2061 6765 7320 6f66 2077 6764   the ages of wgd
+00016ca0: 2069 7320 3131 322e 3839 3435 206d 636d   is 112.8945 mcm
+00016cb0: 6374 7265 652e 7079 3a32 3936 0a20 2020  ctree.py:296.   
+00016cc0: 2020 2020 2020 2020 2020 2020 2020 206d                 m
+00016cd0: 696c 6c69 6f6e 2079 6561 7273 2066 726f  illion years fro
+00016ce0: 6d20 436f 6e63 6174 656e 6174 6564 2070  m Concatenated p
+00016cf0: 6570 7469 6465 0a20 2020 2020 2020 2020  eptide.         
+00016d00: 2020 2020 2020 2020 2061 6c69 676e 6d65           alignme
+00016d10: 6e74 2061 6e64 2039 3525 2063 7265 6469  nt and 95% credi
+00016d20: 6269 6c69 7479 2069 6e74 6572 7661 6c73  bility intervals
+00016d30: 2028 4349 290a 2020 2020 2020 2020 2020   (CI).          
+00016d40: 2020 2020 2020 2020 6973 2031 3031 2e32          is 101.2
+00016d50: 3234 2d31 3233 2e31 3231 206d 696c 6c69  24-123.121 milli
+00016d60: 6f6e 2079 6561 7273 0a20 2020 2020 2020  on years.       
+00016d70: 2020 494e 464f 2020 2020 2054 6f74 616c    INFO     Total
+00016d80: 2072 756e 2074 696d 653a 2032 3931 3735   run time: 29175
+00016d90: 7320 2020 2020 2020 2020 2020 2020 2020  s               
+00016da0: 2020 2020 2020 2020 2020 2020 2020 2063                 c
+00016db0: 6c69 2e70 793a 3234 310a 2020 2020 2020  li.py:241.      
+00016dc0: 2020 2049 4e46 4f20 2020 2020 446f 6e65     INFO     Done
+00016dd0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016de0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016df0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016e00: 636c 692e 7079 3a32 3432 0a60 6060 0a0a  cli.py:242.```..
+00016e10: 546f 2076 6973 7561 6c69 7a65 2074 6865  To visualize the
+00016e20: 2064 6174 652c 2077 6520 616c 736f 2070   date, we also p
+00016e30: 726f 7669 6465 6420 6120 7079 7468 6f6e  rovided a python
+00016e40: 2073 6372 6970 7420 746f 2070 6c6f 7420   script to plot 
+00016e50: 7468 6520 5747 4420 6461 7465 7320 696e  the WGD dates in
+00016e60: 2074 6865 2060 7767 6460 2066 6f6c 6465   the `wgd` folde
+00016e70: 722e 2055 7365 7273 206e 6565 6420 746f  r. Users need to
+00016e80: 2065 7874 7261 6374 2074 6865 2072 6177   extract the raw
+00016e90: 2064 6174 6573 2066 726f 6d20 7468 6520   dates from the 
+00016ea0: 606d 636d 632e 7478 7460 2066 6f72 2074  `mcmc.txt` for t
+00016eb0: 6865 2057 4744 206e 6f64 6520 6669 7273  he WGD node firs
+00016ec0: 7420 616e 6420 7361 7665 2069 7420 6173  t and save it as
+00016ed0: 2066 696c 6520 6064 6174 6573 2e74 7874   file `dates.txt
+00016ee0: 6020 286f 7220 7768 6174 6576 6572 2070  ` (or whatever p
+00016ef0: 7265 6665 7272 6564 206e 616d 6529 2e20  referred name). 
+00016f00: 416e 2065 7861 6d70 6c65 2063 6f6d 6d61  An example comma
+00016f10: 6e64 2069 7320 6173 2062 656c 6f77 2e0a  nd is as below..
+00016f20: 0a60 6060 0a70 7974 686f 6e20 2450 4154  .```.python $PAT
+00016f30: 482f 706f 7374 706c 6f74 2e70 7920 706f  H/postplot.py po
+00016f40: 7374 6469 7320 6461 7465 732e 7478 7420  stdis dates.txt 
+00016f50: 2d2d 7065 7263 656e 7469 6c65 2039 3020  --percentile 90 
+00016f60: 2d2d 7469 746c 6520 2257 4744 2064 6174  --title "WGD dat
+00016f70: 6522 202d 2d68 7064 202d 6f20 2252 616e  e" --hpd -o "Ran
+00016f80: 756e 6375 6c61 6c65 735f 5747 445f 6461  unculales_WGD_da
+00016f90: 7465 2e73 7667 220a 6060 600a 0a21 5b5d  te.svg".```..![]
+00016fa0: 2864 6174 612f 5261 6e75 6e63 756c 616c  (data/Ranunculal
+00016fb0: 6573 5f57 4744 5f64 6174 652e 7376 6729  es_WGD_date.svg)
+00016fc0: 0a0a 5468 6520 706f 7374 6572 696f 7220  ..The posterior 
+00016fd0: 6d65 616e 2c20 6d65 6469 616e 2061 6e64  mean, median and
+00016fe0: 206d 6f64 6520 6f66 2074 6865 2052 616e   mode of the Ran
+00016ff0: 756e 6375 6c61 6c65 7320 5747 4420 6167  unculales WGD ag
+00017000: 6520 6973 2031 3132 2e39 322c 2031 3133  e is 112.92, 113
+00017010: 2e34 3420 616e 6420 3131 322e 3534 206d  .44 and 112.54 m
+00017020: 7961 2c20 7769 7468 2039 3025 2048 5044  ya, with 90% HPD
+00017030: 2031 3035 2e30 3720 2d20 3132 322e 3332   105.07 - 122.32
+00017040: 206d 7961 2061 7320 6d61 6e69 6665 7374   mya as manifest
+00017050: 6564 2061 626f 7665 2e0a 0a23 2323 204b  ed above...### K
+00017060: 7374 7265 650a 0a49 6e20 6164 6469 7469  stree..In additi
+00017070: 6f6e 2074 6f20 7061 6972 7769 7365 202a  on to pairwise *
+00017080: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+00017090: 7374 696d 6174 696f 6e2c 2061 202a 4b2a  stimation, a *K*
+000170a0: 3c73 7562 3e53 3c2f 7375 623e 2074 7265  <sub>S</sub> tre
+000170b0: 6520 7769 7468 2062 7261 6e63 6820 6c65  e with branch le
+000170c0: 6e67 7468 2069 6e20 2a4b 2a3c 7375 623e  ngth in *K*<sub>
+000170d0: 533c 2f73 7562 3e20 756e 6974 2063 616e  S</sub> unit can
+000170e0: 2061 6c73 6f20 6265 2064 6572 6976 6564   also be derived
+000170f0: 2066 726f 6d20 7468 6520 7072 6f67 7261   from the progra
+00017100: 6d20 6077 6764 206b 7364 6020 6769 7665  m `wgd ksd` give
+00017110: 6e20 7468 6520 6f70 7469 6f6e 2060 2d2d  n the option `--
+00017120: 6b73 7472 6565 6020 616e 6420 602d 2d73  kstree` and `--s
+00017130: 7065 6369 6573 7472 6565 602e 204e 6f74  peciestree`. Not
+00017140: 6520 7468 6174 2074 6865 2061 6464 6974  e that the addit
+00017150: 696f 6e61 6c20 6f70 7469 6f6e 2060 2d2d  ional option `--
+00017160: 6f6e 6c79 636f 6e63 6174 6b73 7472 6565  onlyconcatkstree
+00017170: 6020 7769 6c6c 206f 6e6c 7920 6361 6c6c  ` will only call
+00017180: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+00017190: 7375 623e 2065 7374 696d 6174 696f 6e20  sub> estimation 
+000171a0: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
+000171b0: 6174 6564 2061 6c69 676e 6d65 6e74 2072  ated alignment r
+000171c0: 6174 6865 7220 7468 616e 2061 6c6c 2074  ather than all t
+000171d0: 6865 2061 6c69 676e 6d65 6e74 732e 2055  he alignments. U
+000171e0: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+000171f0: 7669 6465 2061 2070 7265 7365 7420 7370  vide a preset sp
+00017200: 6563 6965 7320 7472 6565 2066 6f72 2074  ecies tree for t
+00017210: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+00017220: 623e 2074 7265 6520 696e 6665 7265 6e63  b> tree inferenc
+00017230: 6520 6f66 2074 6865 2063 6f6e 6361 7465  e of the concate
+00017240: 6e61 7465 6420 616c 6967 6e6d 656e 7420  nated alignment 
+00017250: 7768 696c 6520 7468 6520 7265 6d61 696e  while the remain
+00017260: 696e 6720 616c 6967 6e6d 656e 7473 2077  ing alignments w
+00017270: 696c 6c20 6265 2061 6761 696e 7374 2061  ill be against a
+00017280: 6e20 6175 746f 6d61 7465 6c79 2069 6e66  n automately inf
+00017290: 6572 7265 6420 7472 6565 2066 726f 6d20  erred tree from 
+000172a0: 6066 6173 7474 7265 6560 206f 7220 6069  `fasttree` or `i
+000172b0: 7174 7265 6560 2e20 496e 2074 6865 2065  qtree`. In the e
+000172c0: 6e64 2c20 7573 6572 7320 7769 6c6c 2067  nd, users will g
+000172d0: 6574 2061 202a 4b2a 3c73 7562 3e53 3c2f  et a *K*<sub>S</
+000172e0: 7375 623e 2074 7265 652c 2061 202a 4b2a  sub> tree, a *K*
+000172f0: 3c73 7562 3e41 3c2f 7375 623e 2074 7265  <sub>A</sub> tre
+00017300: 6520 616e 6420 6120 cf89 2074 7265 6520  e and a .. tree 
+00017310: 7065 7220 6661 6d20 616e 6420 666f 7220  per fam and for 
+00017320: 7468 6520 636f 6e63 6174 656e 6174 6564  the concatenated
+00017330: 2061 6c69 676e 6d65 6e74 2e0a 0a60 6060   alignment...```
+00017340: 0a77 6764 206b 7364 2064 6174 612f 6b73  .wgd ksd data/ks
+00017350: 7472 6565 5f64 6174 612f 6661 6d2e 7473  tree_data/fam.ts
+00017360: 7620 6461 7461 2f6b 7374 7265 655f 6461  v data/kstree_da
+00017370: 7461 2f41 636f 7275 735f 7461 7461 7269  ta/Acorus_tatari
+00017380: 6e6f 7769 6920 6461 7461 2f6b 7374 7265  nowii data/kstre
+00017390: 655f 6461 7461 2f41 6d62 6f72 656c 6c61  e_data/Amborella
+000173a0: 5f74 7269 6368 6f70 6f64 6120 6461 7461  _trichopoda data
+000173b0: 2f6b 7374 7265 655f 6461 7461 2f41 7175  /kstree_data/Aqu
+000173c0: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+000173d0: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
+000173e0: 2f41 7269 7374 6f6c 6f63 6869 615f 6669  /Aristolochia_fi
+000173f0: 6d62 7269 6174 6120 6461 7461 2f6b 7374  mbriata data/kst
+00017400: 7265 655f 6461 7461 2f43 7963 6173 5f70  ree_data/Cycas_p
+00017410: 616e 7a68 6968 7561 656e 7369 202d 2d6b  anzhihuaensi --k
+00017420: 7374 7265 6520 2d2d 7370 6563 6965 7374  stree --speciest
+00017430: 7265 6520 6461 7461 2f6b 7374 7265 655f  ree data/kstree_
+00017440: 6461 7461 2f73 7065 6369 6573 5f74 7265  data/species_tre
+00017450: 6531 2e6e 7720 2d2d 6f6e 6c79 636f 6e63  e1.nw --onlyconc
+00017460: 6174 6b73 7472 6565 202d 6f20 7767 645f  atkstree -o wgd_
+00017470: 6b73 7472 6565 5f74 6f70 6f6c 6f67 7931  kstree_topology1
+00017480: 0a60 6060 0a0a 215b 5d28 6461 7461 2f6b  .```..![](data/k
+00017490: 7374 7265 655f 7265 7375 6c74 732f 6b73  stree_results/ks
+000174a0: 7472 6565 2e73 7667 290a 0a41 626f 7665  tree.svg)..Above
+000174b0: 2077 6520 7573 6564 2074 6872 6565 2061   we used three a
+000174c0: 6c74 6572 6e61 7469 7665 2074 6f70 6f6c  lternative topol
+000174d0: 6f67 6965 7320 746f 2069 6e66 6572 2074  ogies to infer t
+000174e0: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+000174f0: 623e 2074 7265 6520 7768 6963 6820 6c65  b> tree which le
+00017500: 6420 746f 2064 6966 6665 7265 6e74 2062  d to different b
+00017510: 7261 6e63 6820 6c65 6e67 7468 2065 7374  ranch length est
+00017520: 696d 6174 696f 6e2e 204e 6f74 6520 7468  imation. Note th
+00017530: 6174 2074 6865 2066 616d 696c 6965 7320  at the families 
+00017540: 7765 2075 7365 6420 7765 7265 206f 6e6c  we used were onl
+00017550: 7920 7477 6f20 676c 6f62 616c 204d 5242  y two global MRB
+00017560: 4820 6661 6d69 6c69 6573 2066 6f72 2074  H families for t
+00017570: 6865 2070 7572 706f 7365 206f 6620 696c  he purpose of il
+00017580: 6c75 7374 7261 7469 6f6e 2e20 546f 2061  lustration. To a
+00017590: 6371 7569 7265 2061 6e20 6163 6375 7261  cquire an accura
+000175a0: 7465 2070 726f 6669 6c65 206f 6620 7468  te profile of th
+000175b0: 6520 7375 6273 7469 7475 7469 6f6e 2072  e substitution r
+000175c0: 6174 6520 7661 7269 6174 696f 6e2c 206f  ate variation, o
+000175d0: 7274 686f 6c6f 6775 6573 2061 7420 7468  rthologues at th
+000175e0: 6520 7768 6f6c 6520 6765 6e6f 6d65 2073  e whole genome s
+000175f0: 6361 6c65 2073 686f 756c 6420 6265 2075  cale should be u
+00017600: 7365 642e 0a0a 2323 2043 6974 6174 696f  sed...## Citatio
+00017610: 6e0a 200a 506c 6561 7365 2063 6974 6520  n. .Please cite 
+00017620: 7573 2061 7420 6874 7470 733a 2f2f 646f  us at https://do
+00017630: 692e 6f72 672f 3130 2e31 3030 372f 3937  i.org/10.1007/97
+00017640: 382d 312d 3037 3136 2d32 3536 312d 335f  8-1-0716-2561-3_
+00017650: 3120 616e 6420 6874 7470 733a 2f2f 646f  1 and https://do
+00017660: 692e 6f72 672f 3130 2e31 3039 332f 6269  i.org/10.1093/bi
+00017670: 6f69 6e66 6f72 6d61 7469 6373 2f62 7461  oinformatics/bta
+00017680: 6532 3732 2e0a 0a60 6060 0a48 656e 6763  e272...```.Hengc
+00017690: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
+000176a0: 5a77 6165 6e65 706f 656c 2028 3230 3233  Zwaenepoel (2023
+000176b0: 292e 2049 6e66 6572 656e 6365 206f 6620  ). Inference of 
+000176c0: 416e 6369 656e 7420 506f 6c79 706c 6f69  Ancient Polyploi
+000176d0: 6479 2066 726f 6d20 4765 6e6f 6d69 6320  dy from Genomic 
+000176e0: 4461 7461 2e20 496e 3a20 5661 6e20 6465  Data. In: Van de
+000176f0: 2050 6565 722c 2059 2e20 2865 6473 2920   Peer, Y. (eds) 
+00017700: 506f 6c79 706c 6f69 6479 2e20 4d65 7468  Polyploidy. Meth
+00017710: 6f64 7320 696e 204d 6f6c 6563 756c 6172  ods in Molecular
+00017720: 2042 696f 6c6f 6779 2c20 766f 6c20 3235   Biology, vol 25
+00017730: 3435 2e20 4875 6d61 6e61 2c20 4e65 7720  45. Humana, New 
+00017740: 596f 726b 2c20 4e59 2e20 6874 7470 733a  York, NY. https:
+00017750: 2f2f 646f 692e 6f72 672f 3130 2e31 3030  //doi.org/10.100
+00017760: 372f 3937 382d 312d 3037 3136 2d32 3536  7/978-1-0716-256
+00017770: 312d 335f 310a 4865 6e67 6368 6920 4368  1-3_1.Hengchi Ch
+00017780: 656e 2c20 4172 7468 7572 205a 7761 656e  en, Arthur Zwaen
+00017790: 6570 6f65 6c2c 2059 7665 7320 5661 6e20  epoel, Yves Van 
+000177a0: 6465 2050 6565 7220 2832 3032 3429 2e20  de Peer (2024). 
+000177b0: 7767 6420 7632 3a20 6120 7375 6974 6520  wgd v2: a suite 
+000177c0: 6f66 2074 6f6f 6c73 2074 6f20 756e 636f  of tools to unco
+000177d0: 7665 7220 616e 6420 6461 7465 2061 6e63  ver and date anc
+000177e0: 6965 6e74 2070 6f6c 7970 6c6f 6964 7920  ient polyploidy 
+000177f0: 616e 6420 7768 6f6c 652d 6765 6e6f 6d65  and whole-genome
+00017800: 2064 7570 6c69 6361 7469 6f6e 2e20 4269   duplication. Bi
+00017810: 6f69 6e66 6f72 6d61 7469 6373 2c20 6874  oinformatics, ht
+00017820: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
+00017830: 2e31 3039 332f 6269 6f69 6e66 6f72 6d61  .1093/bioinforma
+00017840: 7469 6373 2f62 7461 6532 3732 0a60 6060  tics/btae272.```
+00017850: 0a0a 466f 7220 6369 7461 7469 6f6e 206f  ..For citation o
+00017860: 6620 7468 6520 746f 6f6c 7320 7573 6564  f the tools used
+00017870: 2069 6e20 7767 642c 2070 6c65 6173 6520   in wgd, please 
+00017880: 636f 6e73 756c 7420 7468 6520 646f 6375  consult the docu
+00017890: 6d65 6e74 6174 696f 6e20 6174 0a68 7474  mentation at.htt
+000178a0: 7073 3a2f 2f77 6764 7632 2e72 6561 6474  ps://wgdv2.readt
+000178b0: 6865 646f 6373 2e69 6f2f 656e 2f6c 6174  hedocs.io/en/lat
+000178c0: 6573 742f 6369 7461 7469 6f6e 2e68 746d  est/citation.htm
+000178d0: 6c2e 0a0a                                l...
```

### Comparing `wgd-2.0.32/README.md` & `wgd-2.0.33/PKG-INFO`

 * *Files 1% similar despite different names*

```diff
@@ -1,5972 +1,6045 @@
-00000000: 3c64 6976 2061 6c69 676e 3d22 6365 6e74  <div align="cent
-00000010: 6572 223e 0a0a 2320 6077 6764 2076 3260  er">..# `wgd v2`
-00000020: 203a 2061 2073 7569 7465 2074 6f6f 6c20   : a suite tool 
-00000030: 6f66 2057 4744 2069 6e66 6572 656e 6365  of WGD inference
-00000040: 2061 6e64 2074 696d 696e 670a 0a5b 215b   and timing..[![
-00000050: 4275 696c 6420 5374 6174 7573 5d28 6874  Build Status](ht
-00000060: 7470 733a 2f2f 6170 702e 7472 6176 6973  tps://app.travis
-00000070: 2d63 692e 636f 6d2f 6865 6368 652d 7073  -ci.com/heche-ps
-00000080: 622f 7767 642e 7376 673f 6272 616e 6368  b/wgd.svg?branch
-00000090: 3d70 6879 6c6f 6461 7469 6e67 295d 2868  =phylodating)](h
-000000a0: 7474 7073 3a2f 2f61 7070 2e74 7261 7669  ttps://app.travi
-000000b0: 732d 6369 2e63 6f6d 2f68 6563 6865 2d70  s-ci.com/heche-p
-000000c0: 7362 2f77 6764 290a 5b21 5b44 6f63 756d  sb/wgd).[![Docum
-000000d0: 656e 7461 7469 6f6e 2053 7461 7475 735d  entation Status]
-000000e0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
-000000f0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
-00000100: 732f 7767 6476 322f 6261 6467 652f 3f76  s/wgdv2/badge/?v
-00000110: 6572 7369 6f6e 3d6c 6174 6573 7429 5d28  ersion=latest)](
-00000120: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
-00000130: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
-00000140: 6c61 7465 7374 2f3f 6261 6467 653d 6c61  latest/?badge=la
-00000150: 7465 7374 290a 5b21 5b6c 6963 656e 7365  test).[![license
-00000160: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
-00000170: 6965 6c64 732e 696f 2f70 7970 692f 6c2f  ields.io/pypi/l/
-00000180: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
-00000190: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
-000001a0: 672f 7079 7069 2f77 6764 290a 5b21 5b4c  g/pypi/wgd).[![L
-000001b0: 6174 6573 7420 5079 5049 2076 6572 7369  atest PyPI versi
-000001c0: 6f6e 5d28 6874 7470 733a 2f2f 696d 672e  on](https://img.
-000001d0: 7368 6965 6c64 732e 696f 2f70 7970 692f  shields.io/pypi/
-000001e0: 762f 7767 642e 7376 6729 5d28 6874 7470  v/wgd.svg)](http
-000001f0: 733a 2f2f 7079 7069 2e70 7974 686f 6e2e  s://pypi.python.
-00000200: 6f72 672f 7079 7069 2f77 6764 290a 5b21  org/pypi/wgd).[!
-00000210: 5b69 6e73 7461 6c6c 2077 6974 6820 6269  [install with bi
-00000220: 6f63 6f6e 6461 5d28 6874 7470 733a 2f2f  oconda](https://
-00000230: 696d 672e 7368 6965 6c64 732e 696f 2f62  img.shields.io/b
-00000240: 6164 6765 2f69 6e73 7461 6c6c 2532 3077  adge/install%20w
-00000250: 6974 682d 6269 6f63 6f6e 6461 2d62 7269  ith-bioconda-bri
-00000260: 6768 7467 7265 656e 2e73 7667 3f73 7479  ghtgreen.svg?sty
-00000270: 6c65 3d66 6c61 7429 5d28 6874 7470 3a2f  le=flat)](http:/
-00000280: 2f62 696f 636f 6e64 612e 6769 7468 7562  /bioconda.github
-00000290: 2e69 6f2f 7265 6369 7065 732f 7767 642f  .io/recipes/wgd/
-000002a0: 5245 4144 4d45 2e68 746d 6c29 0a5b 215b  README.html).[![
-000002b0: 446f 776e 6c6f 6164 735d 2868 7474 7073  Downloads](https
-000002c0: 3a2f 2f70 6570 792e 7465 6368 2f62 6164  ://pepy.tech/bad
-000002d0: 6765 2f77 6764 295d 2868 7474 7073 3a2f  ge/wgd)](https:/
-000002e0: 2f70 6570 792e 7465 6368 2f70 726f 6a65  /pepy.tech/proje
-000002f0: 6374 2f77 6764 290a 0a2a 2a48 656e 6763  ct/wgd)..**Hengc
-00000300: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
-00000310: 5a77 6165 6e65 706f 656c 2c20 5976 6573  Zwaenepoel, Yves
-00000320: 2056 616e 2064 6520 5065 6572 2a2a 0a0a   Van de Peer**..
-00000330: 5b2a 2a42 696f 696e 666f 726d 6174 6963  [**Bioinformatic
-00000340: 7320 2620 4576 6f6c 7574 696f 6e61 7279  s & Evolutionary
-00000350: 2047 656e 6f6d 6963 7320 4772 6f75 702a   Genomics Group*
-00000360: 2a5d 2868 7474 7073 3a2f 2f77 7777 2e76  *](https://www.v
-00000370: 616e 6465 7065 6572 6c61 622e 6f72 672f  andepeerlab.org/
-00000380: 292a 2a2c 2056 4942 2d55 4765 6e74 2043  )**, VIB-UGent C
-00000390: 656e 7465 7220 666f 7220 506c 616e 7420  enter for Plant 
-000003a0: 5379 7374 656d 7320 4269 6f6c 6f67 792a  Systems Biology*
-000003b0: 2a0a 0a5b 2a2a 496e 7472 6f64 7563 7469  *..[**Introducti
-000003c0: 6f6e 2a2a 5d28 2369 6e74 726f 6475 6374  on**](#introduct
-000003d0: 696f 6e29 207c 200a 5b2a 2a49 6e73 7461  ion) | .[**Insta
-000003e0: 6c6c 6174 696f 6e2a 2a5d 2823 696e 7374  llation**](#inst
-000003f0: 616c 6c61 7469 6f6e 2920 7c20 0a5b 2a2a  allation) | .[**
-00000400: 5061 7261 6d65 7465 7273 2a2a 5d28 2370  Parameters**](#p
-00000410: 6172 616d 6574 6572 7329 207c 200a 5b2a  arameters) | .[*
-00000420: 2a55 7361 6765 2a2a 5d28 2375 7361 6765  *Usage**](#usage
-00000430: 2920 7c20 0a5b 2a2a 496c 6c75 7374 7261  ) | .[**Illustra
-00000440: 7469 6f6e 2a2a 5d28 2369 6c6c 7573 7472  tion**](#illustr
-00000450: 6174 696f 6e29 207c 0a5b 2a2a 446f 6375  ation) |.[**Docu
-00000460: 6d65 6e74 6174 696f 6e2a 2a5d 2868 7474  mentation**](htt
-00000470: 7073 3a2f 2f77 6764 7632 2e72 6561 6474  ps://wgdv2.readt
-00000480: 6865 646f 6373 2e69 6f2f 656e 2f6c 6174  hedocs.io/en/lat
-00000490: 6573 742f 3f62 6164 6765 3d6c 6174 6573  est/?badge=lates
-000004a0: 7429 207c 0a5b 2a2a 4369 7461 7469 6f6e  t) |.[**Citation
-000004b0: 2a2a 5d28 2363 6974 6174 696f 6e29 0a0a  **](#citation)..
-000004c0: 3c2f 6469 763e 0a0a 6077 6764 2076 3260  </div>..`wgd v2`
-000004d0: 2069 7320 6120 7079 7468 6f6e 2070 6163   is a python pac
-000004e0: 6b61 6765 2075 7067 7261 6465 6420 6672  kage upgraded fr
-000004f0: 6f6d 2074 6865 206f 7269 6769 6e61 6c20  om the original 
-00000500: 6077 6764 6020 7061 636b 6167 6520 6169  `wgd` package ai
-00000510: 6d69 6e67 2066 6f72 2074 6865 2069 6e66  ming for the inf
-00000520: 6572 656e 6365 2061 6e64 2074 696d 696e  erence and timin
-00000530: 6720 6f66 2061 6e63 6965 6e74 2077 686f  g of ancient who
-00000540: 6c65 2d67 656e 6f6d 6520 6475 706c 6963  le-genome duplic
-00000550: 6174 696f 6e20 2857 4744 2920 6576 656e  ation (WGD) even
-00000560: 7473 2e20 466f 7220 7468 6520 7072 6f70  ts. For the prop
-00000570: 6f73 6520 6f66 2069 6c6c 7573 7472 6174  ose of illustrat
-00000580: 696e 6720 7468 6520 7072 696e 6369 706c  ing the principl
-00000590: 6520 616e 6420 7573 6167 6520 6f66 2060  e and usage of `
-000005a0: 7767 6420 7632 602c 2077 6520 636f 6d70  wgd v2`, we comp
-000005b0: 696c 6564 2074 6869 7320 646f 6375 6d65  iled this docume
-000005c0: 6e74 6174 696f 6e2e 2042 656c 6f77 2077  ntation. Below w
-000005d0: 6520 6669 7273 7420 6761 7665 2061 6e20  e first gave an 
-000005e0: 696e 7472 6f64 7563 7469 6f6e 206f 7665  introduction ove
-000005f0: 7220 7468 6520 7363 6f70 6520 616e 6420  r the scope and 
-00000600: 6d65 6368 616e 6973 6d20 6f66 2060 7767  mechanism of `wg
-00000610: 6420 7632 6020 616e 6420 7468 656e 2074  d v2` and then t
-00000620: 6865 2070 7261 6374 6963 616c 2069 6e66  he practical inf
-00000630: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
-00000640: 696e 6720 746f 2074 6865 2069 6e73 7461  ing to the insta
-00000650: 6c6c 6174 696f 6e20 616e 6420 7573 6167  llation and usag
-00000660: 652e 2041 6e20 6578 616d 706c 6172 2077  e. An examplar w
-00000670: 6f72 6b66 6c6f 7720 6973 2070 726f 7669  orkflow is provi
-00000680: 6465 6420 696e 2074 6865 2074 7574 6f72  ded in the tutor
-00000690: 6961 6c20 7365 6374 696f 6e20 6f6e 2068  ial section on h
-000006a0: 6f77 2074 6f20 7365 656b 2065 7669 6465  ow to seek evide
-000006b0: 6e63 6520 666f 7220 6120 7075 7461 7469  nce for a putati
-000006c0: 7665 2057 4744 2065 7665 6e74 2061 6e64  ve WGD event and
-000006d0: 2070 6572 666f 726d 2070 726f 7065 7220   perform proper 
-000006e0: 7469 6d69 6e67 2077 6974 6820 6120 6672  timing with a fr
-000006f0: 6573 686c 7920 6f62 7461 696e 6564 2067  eshly obtained g
-00000700: 656e 6f6d 6520 6173 7365 6d62 6c79 2069  enome assembly i
-00000710: 6e20 6861 6e64 2e20 466f 7220 7468 6f73  n hand. For thos
-00000720: 6520 7768 6f20 6172 6520 696e 7465 7265  e who are intere
-00000730: 7374 6564 2069 6e20 6d6f 7265 2074 6865  sted in more the
-00000740: 6f72 6574 6963 616c 2064 6574 6169 6c73  oretical details
-00000750: 2c20 7765 2072 6563 6f6d 6d65 6e64 2074  , we recommend t
-00000760: 7572 6e69 6e67 2074 6f20 6f75 7220 7061  urning to our pa
-00000770: 7065 7220 616e 6420 626f 6f6b 2063 6861  per and book cha
-00000780: 7074 6572 2066 6f72 206d 6f72 6520 6465  pter for more de
-00000790: 7461 696c 6564 2064 6573 6372 6970 7469  tailed descripti
-000007a0: 6f6e 2061 6e64 2069 6e73 6967 6874 6675  on and insightfu
-000007b0: 6c20 6469 7363 7573 7369 6f6e 732e 2054  l discussions. T
-000007c0: 6865 206b 6579 2069 6d70 726f 7665 6420  he key improved 
-000007d0: 6665 6174 7572 6573 206f 6620 6077 6764  features of `wgd
-000007e0: 2076 3260 2061 7265 2064 656d 6f6e 7374   v2` are demonst
-000007f0: 7261 7465 6420 696e 206f 7572 206c 6174  rated in our lat
-00000800: 6573 7420 6d61 6e75 7363 7269 7074 206f  est manuscript o
-00000810: 6e20 5b42 696f 696e 666f 726d 6174 6963  n [Bioinformatic
-00000820: 735d 2868 7474 7073 3a2f 2f64 6f69 2e6f  s](https://doi.o
-00000830: 7267 2f31 302e 3130 3933 2f62 696f 696e  rg/10.1093/bioin
-00000840: 666f 726d 6174 6963 732f 6274 6165 3237  formatics/btae27
-00000850: 3229 2e20 4966 2079 6f75 2075 7365 2060  2). If you use `
-00000860: 7767 6420 7632 6020 696e 2079 6f75 7220  wgd v2` in your 
-00000870: 7265 7365 6172 6368 2c20 706c 6561 7365  research, please
-00000880: 2063 6974 6520 7573 2061 7320 7375 6767   cite us as sugg
-00000890: 6573 7465 6420 696e 205b 4369 7461 7469  ested in [Citati
-000008a0: 6f6e 5d28 2363 6974 6174 696f 6e29 2073  on](#citation) s
-000008b0: 6563 7469 6f6e 2e0a 0a23 2320 496e 7472  ection...## Intr
-000008c0: 6f64 7563 7469 6f6e 0a0a 506f 6c79 706c  oduction..Polypl
-000008d0: 6f69 6469 7a61 7469 6f6e 732c 2074 6865  oidizations, the
-000008e0: 2065 766f 6c75 7469 6f6e 6172 7920 7072   evolutionary pr
-000008f0: 6f63 6573 7320 7468 6174 2074 6865 2065  ocess that the e
-00000900: 6e74 6972 6520 6765 6e6f 6d65 206f 6620  ntire genome of 
-00000910: 616e 206f 7267 616e 6973 6d20 6973 2064  an organism is d
-00000920: 7570 6c69 6361 7465 642c 2061 6c73 6f20  uplicated, also 
-00000930: 6e61 6d65 6420 6173 2077 686f 6c65 2d67  named as whole-g
-00000940: 656e 6f6d 6520 6475 706c 6963 6174 696f  enome duplicatio
-00000950: 6e73 2028 5747 4473 292c 206f 6363 7572  ns (WGDs), occur
-00000960: 2072 6563 7572 7265 6e74 6c79 2061 6372   recurrently acr
-00000970: 6f73 7320 7468 6520 7472 6565 206f 6620  oss the tree of 
-00000980: 6c69 6665 2e20 5468 6572 6520 6172 6520  life. There are 
-00000990: 7477 6f20 6d6f 6465 7320 6f66 2070 6f6c  two modes of pol
-000009a0: 7970 6c6f 6964 697a 6174 696f 6e73 2c20  yploidizations, 
-000009b0: 6175 746f 706f 6c79 706c 6f69 6469 7a61  autopolyploidiza
-000009c0: 7469 6f6e 7320 616e 6420 616c 6c6f 706f  tions and allopo
-000009d0: 6c79 706c 6f69 6469 7a61 7469 6f6e 732e  lyploidizations.
-000009e0: 2041 7574 6f70 6f6c 7970 6c6f 6964 697a   Autopolyploidiz
-000009f0: 6174 696f 6e73 2061 7265 2074 6865 2064  ations are the d
-00000a00: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
-00000a10: 6520 7361 6d65 2067 656e 6f6d 652c 2072  e same genome, r
-00000a20: 6573 756c 7469 6e67 2069 6e20 7477 6f20  esulting in two 
-00000a30: 6964 656e 7469 6361 6c20 7375 6267 656e  identical subgen
-00000a40: 6f6d 6573 2061 7420 7468 6520 7469 6d65  omes at the time
-00000a50: 2069 7420 656d 6572 6765 642e 2057 6869   it emerged. Whi
-00000a60: 6c65 2074 6865 2061 6c6c 6f70 6f6c 7970  le the allopolyp
-00000a70: 6c6f 6964 697a 6174 696f 6e73 2061 7265  loidizations are
-00000a80: 206e 6f72 6d61 6c6c 7920 6163 6869 6576   normally achiev
-00000a90: 6564 2069 6e20 7477 6f20 7374 6570 732c  ed in two steps,
-00000aa0: 2066 6972 7374 2074 6865 2068 7962 7269   first the hybri
-00000ab0: 6469 7a61 7469 6f6e 2062 6574 7765 656e  dization between
-00000ac0: 2074 776f 2064 6966 6665 7265 6e74 2073   two different s
-00000ad0: 7065 6369 6573 2c20 7265 7375 6c74 696e  pecies, resultin
-00000ae0: 6720 696e 2074 6865 2061 7269 7369 6e67  g in the arising
-00000af0: 206f 6620 7472 616e 7369 656e 7420 686f   of transient ho
-00000b00: 6d6f 706c 6f69 6479 2c73 6563 6f6e 6420  moploidy,second 
-00000b10: 7468 6520 6475 706c 6963 6174 696f 6e20  the duplication 
-00000b20: 6f66 2074 6865 2068 6f6d 6f70 6c6f 6964  of the homoploid
-00000b30: 792c 2072 6573 756c 7469 6e67 2069 6e20  y, resulting in 
-00000b40: 7468 6520 656d 6572 6765 6e63 6520 6f66  the emergence of
-00000b50: 2061 6c6c 6f70 6f6c 7970 6c6f 6964 792e   allopolyploidy.
-00000b60: 2044 7565 2074 6f20 7468 6520 756e 7374   Due to the unst
-00000b70: 6162 696c 6974 7920 616e 6420 756e 6261  ability and unba
-00000b80: 6c61 6e63 6564 2074 6574 7261 736f 6d69  lanced tetrasomi
-00000b90: 6320 696e 6865 7269 7461 6e63 652c 2066  c inheritance, f
-00000ba0: 6f72 2069 6e73 7461 6e63 6520 7468 6520  or instance the 
-00000bb0: 6e75 636c 6561 722d 6379 746f 706c 6173  nuclear-cytoplas
-00000bc0: 6d69 6320 696e 636f 6d70 6174 6962 696c  mic incompatibil
-00000bd0: 6974 792c 2074 6865 2070 6f6c 7970 6c6f  ity, the polyplo
-00000be0: 6964 7920 6765 6e6f 6d65 2077 696c 6c20  idy genome will 
-00000bf0: 7468 656e 2065 7870 6572 6965 6e63 6520  then experience 
-00000c00: 6120 7072 6f63 6573 7320 6361 6c6c 6564  a process called
-00000c10: 2064 6970 6c6f 6964 697a 6174 696f 6e2c   diploidization,
-00000c20: 2061 6c73 6f20 6e61 6d65 6420 6173 2066   also named as f
-00000c30: 7261 6374 696f 6e61 7469 6f6e 2c20 6475  ractionation, du
-00000c40: 7269 6e67 2077 6869 6368 2061 206c 6172  ring which a lar
-00000c50: 6765 2070 6f72 7469 6f6e 206f 6620 6765  ge portion of ge
-00000c60: 6e65 2064 7570 6c69 6361 7465 7320 7769  ne duplicates wi
-00000c70: 6c6c 2067 6574 206c 6f73 7420 616e 6420  ll get lost and 
-00000c80: 6f6e 6c79 2061 2066 7261 6374 696f 6e20  only a fraction 
-00000c90: 6361 6e20 6265 2072 6574 6169 6e65 642e  can be retained.
-00000ca0: 2054 6865 2074 7261 6365 7320 6f66 2070   The traces of p
-00000cb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
-00000cc0: 2063 616e 2062 6520 7468 7573 2075 6e65   can be thus une
-00000cd0: 6172 7468 6564 2066 726f 6d20 7468 6573  arthed from thes
-00000ce0: 6520 7265 7461 696e 6564 2067 656e 6520  e retained gene 
-00000cf0: 6475 706c 6963 6174 6573 2e20 5468 7265  duplicates. Thre
-00000d00: 6520 6170 7072 6f61 6368 6573 2062 6173  e approaches bas
-00000d10: 6564 206f 6e20 6765 6e65 2064 7570 6c69  ed on gene dupli
-00000d20: 6361 7465 732c 206e 616d 656c 792c 202a  cates, namely, *
-00000d30: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
-00000d40: 6574 686f 642c 2067 656e 6520 7472 6565  ethod, gene tree
-00000d50: 202d 2073 7065 6369 6573 2074 7265 6520   - species tree 
-00000d60: 7265 636f 6e63 696c 6961 7469 6f6e 206d  reconciliation m
-00000d70: 6574 686f 6420 616e 6420 7379 6e74 656e  ethod and synten
-00000d80: 7920 6d65 7468 6f64 2c20 6172 6520 636f  y method, are co
-00000d90: 6d6d 6f6e 6c79 2075 7365 6420 696e 2064  mmonly used in d
-00000da0: 6574 6563 7469 6e67 2065 7669 6465 6e63  etecting evidenc
-00000db0: 6520 666f 7220 5747 4473 2e20 5468 6520  e for WGDs. The 
-00000dc0: 6765 6e65 2074 7265 6520 2d20 7370 6563  gene tree - spec
-00000dd0: 6965 7320 7472 6565 2072 6563 6f6e 6369  ies tree reconci
-00000de0: 6c69 6174 696f 6e20 6d65 7468 6f64 2069  liation method i
-00000df0: 7320 6e6f 7420 7769 7468 696e 2074 6865  s not within the
-00000e00: 2073 636f 7065 206f 6620 6077 6764 2076   scope of `wgd v
-00000e10: 3260 2c20 6275 7420 7765 206b 696e 646c  2`, but we kindl
-00000e20: 7920 7265 6665 7220 7265 6164 6572 7320  y refer readers 
-00000e30: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000e40: 6564 2074 6f20 7468 6520 7068 796c 6f67  ed to the phylog
-00000e50: 656e 6f6d 6963 2070 726f 6772 616d 2064  enomic program d
-00000e60: 6576 656c 6f70 6564 2062 7920 4172 7468  eveloped by Arth
-00000e70: 7572 205a 7761 656e 6570 6f65 6c20 6e61  ur Zwaenepoel na
-00000e80: 6d65 6420 5b57 4841 4c45 5d28 6874 7470  med [WHALE](http
-00000e90: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f61  s://github.com/a
-00000ea0: 727a 7761 2f57 6861 6c65 2e6a 6c29 2061  rzwa/Whale.jl) a
-00000eb0: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
-00000ec0: 6420 5b70 6170 6572 5d28 6874 7470 733a  d [paper](https:
-00000ed0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-00000ee0: 332f 6d6f 6c62 6576 2f6d 737a 3038 3829  3/molbev/msz088)
-00000ef0: 2066 6f72 206d 6f72 6520 7465 6368 6e69   for more techni
-00000f00: 6361 6c20 616e 6420 7468 656f 7265 7469  cal and theoreti
-00000f10: 6361 6c20 6465 7461 696c 732e 0a0a 5468  cal details...Th
-00000f20: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00000f30: 3e20 6d65 7468 6f64 2069 7320 6573 7461  > method is esta
-00000f40: 626c 6973 6865 6420 6f6e 2061 206d 6f64  blished on a mod
-00000f50: 656c 206f 6620 6765 6e65 2066 616d 696c  el of gene famil
-00000f60: 7920 6576 6f6c 7574 696f 6e20 7468 6174  y evolution that
-00000f70: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
-00000f80: 7920 6973 2061 6c6c 6f77 6564 2074 6f20  y is allowed to 
-00000f90: 6576 6f6c 7665 2076 6961 2067 656e 6520  evolve via gene 
-00000fa0: 6475 706c 6963 6174 696f 6e20 616e 6420  duplication and 
-00000fb0: 6c6f 7373 2e20 4e6f 7465 2074 6861 7420  loss. Note that 
-00000fc0: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
-00000fd0: 6865 7265 2069 7320 6173 7375 6d65 6420  here is assumed 
-00000fe0: 746f 2062 6520 7468 6520 636c 7573 7465  to be the cluste
-00000ff0: 7220 6f66 2061 6c6c 2067 656e 6573 2064  r of all genes d
-00001000: 6573 6365 6e64 6564 2066 726f 6d20 616e  escended from an
-00001010: 2061 6e63 6573 7472 616c 2067 656e 6520   ancestral gene 
-00001020: 696e 2061 2073 696e 676c 6520 6765 6e6f  in a single geno
-00001030: 6d65 2e20 5265 636f 7665 7269 6e67 2074  me. Recovering t
-00001040: 6865 2067 656e 6520 7472 6565 206f 6620  he gene tree of 
-00001050: 7375 6368 2067 656e 6520 6661 6d69 6c79  such gene family
-00001060: 2069 6e66 6f72 6d73 2074 6865 2074 696d   informs the tim
-00001070: 696e 672c 2073 6369 6c69 6365 7420 7468  ing, scilicet th
-00001080: 6520 6167 652c 206f 6620 6765 6e65 2064  e age, of gene d
-00001090: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
-000010a0: 732e 2054 6865 2061 6765 2072 6566 6572  s. The age refer
-000010b0: 6564 2068 6572 652c 2069 7320 6e6f 7420  ed here, is not 
-000010c0: 696e 2072 6561 6c20 6765 6f6c 6f67 6963  in real geologic
-000010d0: 616c 2074 696d 652c 2062 7574 2069 6e20  al time, but in 
-000010e0: 7468 6520 756e 6974 206f 6620 6576 6f6c  the unit of evol
-000010f0: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001100: 652c 2069 2e65 2e2c 2074 6865 206e 756d  e, i.e., the num
-00001110: 6265 7220 6f66 2073 7562 7374 6974 7574  ber of substitut
-00001120: 696f 6e73 2070 6572 2073 6974 652e 2057  ions per site. W
-00001130: 6865 6e20 7468 6520 6576 6f6c 7574 696f  hen the evolutio
-00001140: 6e61 7279 2072 6174 6520 7265 6d61 696e  nary rate remain
-00001150: 7320 6170 7072 6f78 696d 6174 656c 7920  s approximately 
-00001160: 636f 6e73 7461 6e74 2c20 7468 6520 6576  constant, the ev
-00001170: 6f6c 7574 696f 6e61 7279 2064 6973 7461  olutionary dista
-00001180: 6e63 6520 6973 2074 6865 6e20 7375 7070  nce is then supp
-00001190: 6f73 6564 2074 6f20 6265 2070 726f 706f  osed to be propo
-000011a0: 7274 696f 6e61 6c20 746f 2074 6865 2072  rtional to the r
-000011b0: 6561 6c20 6576 6f6c 7574 696f 6e61 7279  eal evolutionary
-000011c0: 2074 696d 652e 2054 6865 2073 796e 6f6e   time. The synon
-000011d0: 796d 6f75 7320 6469 7374 616e 6365 202a  ymous distance *
-000011e0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2c20  K*<sub>S</sub>, 
-000011f0: 7468 6520 6e75 6d62 6572 206f 6620 7379  the number of sy
-00001200: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
-00001210: 7574 696f 6e73 2070 6572 2073 796e 6f6e  utions per synon
-00001220: 796d 6f75 7320 7369 7465 2c20 6973 2073  ymous site, is s
-00001230: 7563 6820 6361 6e64 6964 6174 6520 7468  uch candidate th
-00001240: 6174 2073 796e 6f6e 796d 6f75 7320 7375  at synonymous su
-00001250: 6273 7469 7475 7469 6f6e 7320 776f 756c  bstitutions woul
-00001260: 6420 6e6f 7420 696e 6375 7220 7468 6520  d not incur the 
-00001270: 6368 616e 6765 206f 6620 616d 696e 6f20  change of amino 
-00001280: 6163 6964 2061 6e64 2061 7265 2074 6875  acid and are thu
-00001290: 7320 7265 6761 7264 6564 2061 7320 6e65  s regarded as ne
-000012a0: 7574 7261 6c2c 2077 6869 6368 2061 6363  utral, which acc
-000012b0: 6f72 6469 6e67 2074 6f20 7468 6520 6e65  ording to the ne
-000012c0: 7574 7261 6c20 7468 656f 7279 2073 686f  utral theory sho
-000012d0: 756c 6420 6f63 6375 7220 696e 2063 6f6e  uld occur in con
-000012e0: 7374 616e 7420 7261 7465 2e20 4769 7665  stant rate. Give
-000012f0: 6e20 6120 6d6f 6465 6c20 6f66 2067 656e  n a model of gen
-00001300: 6520 6661 6d69 6c79 2074 6861 7420 616c  e family that al
-00001310: 6c6f 7773 2074 6865 2067 656e 6520 746f  lows the gene to
-00001320: 2064 7570 6c69 6361 7465 2061 6e64 2067   duplicate and g
-00001330: 6574 206c 6f73 7420 696e 2061 2066 6978  et lost in a fix
-00001340: 6564 2072 6174 652c 206f 6e65 2063 616e  ed rate, one can
-00001350: 2064 6572 6976 6520 7468 6174 2074 6865   derive that the
-00001360: 2070 726f 6261 6269 6c69 7479 2064 656e   probability den
-00001370: 7369 7479 2066 756e 6374 696f 6e20 6f66  sity function of
-00001380: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00001390: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000013a0: 7574 696f 6e20 6f66 2072 6574 6169 6e65  ution of retaine
-000013b0: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-000013c0: 7320 6973 2061 2071 7561 7369 2d65 7870  s is a quasi-exp
-000013d0: 6f6e 656e 7469 616c 2066 756e 6374 696f  onential functio
-000013e0: 6e20 7468 6174 206d 6f73 7420 7265 7461  n that most reta
-000013f0: 696e 6564 2067 656e 6520 6475 706c 6963  ined gene duplic
-00001400: 6174 6573 2061 7265 2072 6563 656e 746c  ates are recentl
-00001410: 7920 626f 726e 6564 2077 6974 6820 7e30  y borned with ~0
-00001420: 2061 6765 2077 6869 6c65 2061 7320 7468   age while as th
-00001430: 6520 6167 6520 676f 696e 6720 6f6c 6465  e age going olde
-00001440: 7220 7468 6520 6173 736f 6369 6174 6564  r the associated
-00001450: 206e 756d 6265 7220 6f66 2072 6574 6169   number of retai
-00001460: 6e65 6420 6765 6e65 2064 7570 6c69 6361  ned gene duplica
-00001470: 7465 7320 6465 6361 7920 7175 6173 692d  tes decay quasi-
-00001480: 6578 706f 6e65 6e74 6961 6c6c 792e 2054  exponentially. T
-00001490: 6865 7265 666f 7265 2c20 7468 6520 6f63  herefore, the oc
-000014a0: 6375 7261 6e63 6520 6f66 206c 6172 6765  curance of large
-000014b0: 2d73 6361 6c65 2067 656e 6520 6475 706c  -scale gene dupl
-000014c0: 6963 6174 696f 6e20 6576 656e 7473 2c20  ication events, 
-000014d0: 666f 7220 696e 7374 616e 6520 5747 4473  for instane WGDs
-000014e0: 2c20 7769 7468 2076 6172 6965 6420 7265  , with varied re
-000014f0: 7465 6e74 696f 6e20 7261 7465 2c20 7769  tention rate, wi
-00001500: 6c6c 206c 6561 7665 2061 6e20 6167 6520  ll leave an age 
-00001510: 7065 616b 2066 726f 6d20 7468 6520 6275  peak from the bu
-00001520: 7273 7420 6f66 2067 656e 6520 6475 706c  rst of gene dupl
-00001530: 6963 6174 6573 2069 6e20 6120 7368 6f72  icates in a shor
-00001540: 7420 7469 6d65 2d66 7261 6d65 2075 706f  t time-frame upo
-00001550: 6e20 7468 6520 696e 6974 6961 6c20 6167  n the initial ag
-00001560: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
-00001570: 616e 6420 6361 6e20 6265 2075 6e76 6569  and can be unvei
-00001580: 6c65 6420 6672 6f6d 206d 6978 7475 7265  led from mixture
-00001590: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
-000015a0: 6973 2e20 486f 7765 7665 722c 2057 4744  is. However, WGD
-000015b0: 7320 6964 656e 7469 6669 6564 2066 726f  s identified fro
-000015c0: 6d20 7468 6520 7061 7261 6c6f 676f 7573  m the paralogous
-000015d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000015e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-000015f0: 6e73 2063 616e 206f 6e6c 7920 696e 666f  ns can only info
-00001600: 726d 2074 6865 2057 4744 2074 696d 696e  rm the WGD timin
-00001610: 6720 696e 2074 6865 2074 696d 652d 7363  g in the time-sc
-00001620: 616c 6520 6f66 2074 6861 7420 7370 6563  ale of that spec
-00001630: 6966 6963 2073 7065 6369 6573 2c20 7768  ific species, wh
-00001640: 6963 6820 6973 206e 6f74 2063 6f6d 7061  ich is not compa
-00001650: 7261 626c 6520 696e 2074 6865 2070 6879  rable in the phy
-00001660: 6c6f 6765 6e65 7469 6320 636f 6e74 6578  logenetic contex
-00001670: 742e 204f 6e6c 7920 7769 7468 2074 6865  t. Only with the
-00001680: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
-00001690: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-000016a0: 2064 6973 7472 6962 7574 696f 6e73 2c20   distributions, 
-000016b0: 7768 6963 6820 636f 6e76 6572 7420 7468  which convert th
-000016c0: 6520 6573 7469 6d61 7465 6420 626f 6479  e estimated body
-000016d0: 2066 726f 6d20 7061 7261 6c6f 6775 6573   from paralogues
-000016e0: 2074 6f20 6f72 7468 6f6c 6f67 7565 7320   to orthologues 
-000016f0: 616e 6420 696e 666f 726d 2074 6865 2072  and inform the r
-00001700: 656c 6174 6976 6520 7469 6d69 6e67 206f  elative timing o
-00001710: 6620 7370 6563 6961 7469 6f6e 2065 7665  f speciation eve
-00001720: 6e74 732c 2063 616e 2077 6520 6465 6369  nts, can we deci
-00001730: 7068 6572 2074 6865 2070 6879 6c6f 6765  pher the phyloge
-00001740: 6e65 7469 6320 706c 6163 656d 656e 7420  netic placement 
-00001750: 6f66 2057 4744 7320 6166 7465 7220 7072  of WGDs after pr
-00001760: 6f70 6572 2072 6174 6520 636f 7272 6563  oper rate correc
-00001770: 7469 6f6e 2e20 6077 6764 2076 3260 2069  tion. `wgd v2` i
-00001780: 7320 7375 6368 2070 726f 6772 616d 2074  s such program t
-00001790: 6861 7420 6865 6c70 7320 7573 6572 7320  hat helps users 
-000017a0: 636f 6e73 7472 7563 7420 7061 7261 6c6f  construct paralo
-000017b0: 676f 7573 2061 6e64 206f 7274 686f 6c6f  gous and ortholo
-000017c0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-000017d0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000017e0: 7574 696f 6e73 2061 6e64 2072 6561 6c69  utions and reali
-000017f0: 7a65 2062 6f74 6820 7468 6520 6964 656e  ze both the iden
-00001800: 7469 6669 6361 7469 6f6e 2061 6e64 2070  tification and p
-00001810: 6c61 6365 6d65 6e74 206f 6620 5747 4473  lacement of WGDs
-00001820: 2e0a 0a49 6e20 7468 6520 7072 656d 6973  ...In the premis
-00001830: 6520 6f66 2070 6879 6c6f 6765 6e65 7469  e of phylogeneti
-00001840: 6361 6c6c 7920 6c6f 6361 7465 6420 5747  cally located WG
-00001850: 4473 2c20 7468 6520 6162 736f 6c75 7465  Ds, the absolute
-00001860: 2061 6765 2028 696e 2067 656f 6c6f 6769   age (in geologi
-00001870: 6361 6c20 7469 6d65 2920 6f66 2057 4744  cal time) of WGD
-00001880: 7320 6361 6e20 616c 736f 2062 6520 696e  s can also be in
-00001890: 6665 7272 6564 2066 726f 6d20 7468 6f73  ferred from thos
-000018a0: 6520 5747 442d 7265 7461 696e 6564 2067  e WGD-retained g
-000018b0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
-000018c0: 616c 7468 6f75 6768 2074 6865 7265 2068  although there h
-000018d0: 6173 2062 6565 6e20 6e6f 2065 6173 7920  as been no easy 
-000018e0: 6f72 2073 7472 6169 6768 7466 6f72 7761  or straightforwa
-000018f0: 7264 2070 6970 656c 696e 6520 666f 7220  rd pipeline for 
-00001900: 7468 6973 206a 6f62 2073 6f20 6661 722e  this job so far.
-00001910: 2049 6e20 6077 6764 2076 3260 2c20 7765   In `wgd v2`, we
-00001920: 2064 6576 656c 6f70 6564 2061 2066 6561   developed a fea
-00001930: 7369 626c 6520 696e 7465 6772 6174 6564  sible integrated
-00001940: 2070 6970 6c65 6c69 6e65 2066 6f72 2061   pipleline for a
-00001950: 6273 6f6c 7574 6520 6461 7469 6e67 206f  bsolute dating o
-00001960: 6620 5747 4473 2e20 5468 6520 7069 7065  f WGDs. The pipe
-00001970: 6c69 6e65 2063 616e 2062 6520 726f 7567  line can be roug
-00001980: 686c 7920 6469 7669 6465 6420 696e 746f  hly divided into
-00001990: 2074 6872 6565 206d 6169 6e20 7374 6570   three main step
-000019a0: 732e 2031 2920 5468 6520 636f 6e73 7472  s. 1) The constr
-000019b0: 7563 7469 6f6e 206f 6620 616e 6368 6f72  uction of anchor
-000019c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000019d0: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
-000019e0: 6420 7468 6520 6465 6c69 6e65 6174 696f  d the delineatio
-000019f0: 6e20 6f66 2063 7265 6469 6162 6c65 202a  n of crediable *
-00001a00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
-00001a10: 616e 6765 2061 646f 7074 6564 2066 6f72  ange adopted for
-00001a20: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
-00001a30: 7469 6e67 2c20 7573 696e 6720 6077 6764  ting, using `wgd
-00001a40: 2064 6d64 602c 2060 7767 6420 6b73 6460   dmd`, `wgd ksd`
-00001a50: 2c20 6077 6764 2073 796e 6020 616e 6420  , `wgd syn` and 
-00001a60: 6077 6764 2070 6561 6b60 2e20 4e6f 7465  `wgd peak`. Note
-00001a70: 2074 6861 7420 6865 7265 2077 6520 6f6e   that here we on
-00001a80: 6c79 2063 6f6e 7369 6465 7220 6765 6e6f  ly consider geno
-00001a90: 6d65 2061 7373 656d 626c 7920 6265 6361  me assembly beca
-00001aa0: 7573 6520 666f 7220 7472 616e 7363 7269  use for transcri
-00001ab0: 7074 6f6d 6520 6173 7365 6d62 6c79 2069  ptome assembly i
-00001ac0: 7427 7320 696d 706f 7373 6962 6c65 2074  t's impossible t
-00001ad0: 6f20 6469 7374 696e 6775 6973 6820 5747  o distinguish WG
-00001ae0: 442d 6465 7269 7665 6420 6475 706c 6963  D-derived duplic
-00001af0: 6174 6573 2066 726f 6d20 736d 616c 6c2d  ates from small-
-00001b00: 7363 616c 6520 6475 706c 6963 6174 696f  scale duplicatio
-00001b10: 6e2d 6465 7269 7665 6420 6475 706c 6963  n-derived duplic
-00001b20: 6174 6573 2c20 7768 6963 6820 6861 7070  ates, which happ
-00001b30: 656e 6564 2069 6e20 6120 636f 6e74 696e  ened in a contin
-00001b40: 756f 7573 2074 696d 652d 6672 616d 6520  uous time-frame 
-00001b50: 696e 7374 6561 6420 6f66 206f 6e6c 7920  instead of only 
-00001b60: 6120 7365 7061 7261 7465 2073 686f 7274  a separate short
-00001b70: 2074 696d 652d 6672 616d 6520 616e 6420   time-frame and 
-00001b80: 7468 7573 2072 6566 6c65 6374 7320 7468  thus reflects th
-00001b90: 6520 6475 7261 7469 6f6e 206f 6620 7468  e duration of th
-00001ba0: 6174 2062 7261 6e63 6820 7261 7468 6572  at branch rather
-00001bb0: 2074 6861 6e20 7468 6520 7469 6d65 2061   than the time a
-00001bc0: 7420 7768 6963 6820 5747 4420 6f63 6375  t which WGD occu
-00001bd0: 7272 6564 2e20 3229 2054 6865 2066 6f72  rred. 2) The for
-00001be0: 6d75 6c61 7469 6f6e 206f 6620 6120 7374  mulation of a st
-00001bf0: 6172 7469 6e67 2074 7265 6520 7573 6564  arting tree used
-00001c00: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
-00001c10: 6574 6963 2064 6174 696e 672c 2063 6f6d  etic dating, com
-00001c20: 706f 7365 6420 6f66 2061 2066 6577 2073  posed of a few s
-00001c30: 7065 6369 6573 2061 6e64 2061 6e6e 6f74  pecies and annot
-00001c40: 6174 6564 2077 6974 6820 666f 7373 696c  ated with fossil
-00001c50: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
-00001c60: 6f72 6d61 7469 6f6e 2e20 5468 6973 2073  ormation. This s
-00001c70: 7465 7020 6973 2065 7373 656e 7469 616c  tep is essential
-00001c80: 2066 6f72 2074 6865 2072 6573 756c 7420   for the result 
-00001c90: 6f66 2061 6273 6f6c 7574 6520 5747 4420  of absolute WGD 
-00001ca0: 6461 7469 6e67 2074 6861 7420 7765 2073  dating that we s
-00001cb0: 7567 6765 7374 2075 7365 7273 206f 6620  uggest users of 
-00001cc0: 7461 6b69 6e67 2067 7265 6174 2063 6175  taking great cau
-00001cd0: 7469 6f6e 2074 6f20 6173 7375 7265 2074  tion to assure t
-00001ce0: 6865 2074 7265 6520 746f 706f 6c6f 6779  he tree topology
-00001cf0: 2061 6e64 2070 726f 7065 7220 626f 756e   and proper boun
-00001d00: 6473 2066 6f72 2066 6f73 7369 6c20 6361  ds for fossil ca
-00001d10: 6c69 6272 6174 696f 6e73 2e20 3329 2054  librations. 3) T
-00001d20: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
-00001d30: 6f66 206f 7274 686f 6772 6f75 7073 2063  of orthogroups c
-00001d40: 6f6e 7369 7374 696e 6720 6f66 2063 6f6c  onsisting of col
-00001d50: 6c69 6e65 6172 2064 7570 6c69 6361 7465  linear duplicate
-00001d60: 7320 6f66 2074 6865 2066 6f63 616c 2073  s of the focal s
-00001d70: 7065 6369 6573 2061 6e64 2074 6865 6972  pecies and their
-00001d80: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-00001d90: 2068 6974 7320 2852 4248 7329 2061 6761   hits (RBHs) aga
-00001da0: 696e 7374 206f 7468 6572 2073 7065 6369  inst other speci
-00001db0: 6573 2069 6e20 7468 6520 7374 6172 7469  es in the starti
-00001dc0: 6e67 2074 7265 6520 616e 6420 7468 6520  ng tree and the 
-00001dd0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
-00001de0: 696e 6720 7573 696e 6720 6120 6d6f 6c65  ing using a mole
-00001df0: 6375 6c61 7220 6461 7469 6e67 2070 726f  cular dating pro
-00001e00: 6772 616d 2066 6f72 2069 6e73 7461 6e63  gram for instanc
-00001e10: 6520 606d 636d 6374 7265 6560 2c20 7669  e `mcmctree`, vi
-00001e20: 6120 7072 6f67 7261 6d20 6077 6764 2064  a program `wgd d
-00001e30: 6d64 6020 616e 6420 6077 6764 2066 6f63  md` and `wgd foc
-00001e40: 7573 602e 2057 6520 7265 636f 6d6d 656e  us`. We recommen
-00001e50: 6420 7468 6520 7573 6167 6520 6f66 2042  d the usage of B
-00001e60: 6179 6573 6961 6e20 6d6f 6c65 6375 6c61  ayesian molecula
-00001e70: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
-00001e80: 2060 6d63 6d63 7472 6565 602c 2077 6869   `mcmctree`, whi
-00001e90: 6368 2070 726f 7669 6465 7320 6120 7661  ch provides a va
-00001ea0: 7269 6574 7920 6f66 2064 6966 6665 7265  riety of differe
-00001eb0: 6e74 2073 7562 7374 6974 7574 696f 6e20  nt substitution 
-00001ec0: 616e 6420 7261 7465 206d 6f64 656c 732e  and rate models.
-00001ed0: 204e 6f6e 6574 6865 6c65 7373 2c20 7765   Nonetheless, we
-00001ee0: 2075 7267 6520 7573 6572 7320 746f 2073   urge users to s
-00001ef0: 6574 2070 7269 6f72 2064 6973 7472 6962  et prior distrib
-00001f00: 7574 696f 6e20 6f66 2064 6966 6665 7265  ution of differe
-00001f10: 6e74 2070 6172 616d 6574 6572 7320 7769  nt parameters wi
-00001f20: 7468 2063 6175 7469 6f6e 2061 6e64 2061  th caution and a
-00001f30: 7373 7572 6520 6164 6571 7561 7465 2073  ssure adequate s
-00001f40: 616d 706c 696e 6720 6f66 2064 6966 6665  ampling of diffe
-00001f50: 7265 6e74 2070 6172 616d 6574 6572 732e  rent parameters.
-00001f60: 0a0a 2323 2049 6e73 7461 6c6c 6174 696f  ..## Installatio
-00001f70: 6e0a 0a54 6865 2065 6173 6965 7374 2077  n..The easiest w
-00001f80: 6179 2074 6f20 696e 7374 616c 6c20 6077  ay to install `w
-00001f90: 6764 2076 3260 2069 7320 7573 696e 6720  gd v2` is using 
-00001fa0: 5059 5049 2e20 4e6f 7465 2074 6861 7420  PYPI. Note that 
-00001fb0: 6966 2079 6f75 2077 616e 7420 746f 2067  if you want to g
-00001fc0: 6574 2074 6865 206c 6174 6573 7420 7570  et the latest up
-00001fd0: 6461 7465 2c20 7765 2073 7567 6765 7374  date, we suggest
-00001fe0: 2069 6e73 7461 6c6c 696e 6720 6672 6f6d   installing from
-00001ff0: 2074 6865 2073 6f75 7263 652c 2073 696e   the source, sin
-00002000: 6365 2074 6865 2075 7064 6174 6520 6f6e  ce the update on
-00002010: 2050 5950 4920 7769 6c6c 2062 6520 6465   PYPI will be de
-00002020: 6c61 7965 6420 636f 6d70 6172 6564 2074  layed compared t
-00002030: 6f20 6865 7265 206f 6620 736f 7572 6365  o here of source
-00002040: 2e0a 0a60 6060 0a70 6970 2069 6e73 7461  ...```.pip insta
-00002050: 6c6c 2077 6764 0a60 6060 0a0a 546f 2069  ll wgd.```..To i
-00002060: 6e73 7461 6c6c 2060 7767 6420 7632 6020  nstall `wgd v2` 
-00002070: 696e 2061 2076 6972 7475 616c 2065 6e76  in a virtual env
-00002080: 6972 6f6e 6d65 6e74 2c20 7468 6520 666f  ironment, the fo
-00002090: 6c6c 6f77 696e 6720 636f 6d6d 616e 6420  llowing command 
-000020a0: 6c69 6e65 7320 636f 756c 6420 6265 2075  lines could be u
-000020b0: 7365 642e 0a0a 6060 600a 6769 7420 636c  sed...```.git cl
-000020c0: 6f6e 6520 3c77 6764 2072 6570 6f3e 0a63  one <wgd repo>.c
-000020d0: 6420 7767 640a 7669 7274 7561 6c65 6e76  d wgd.virtualenv
-000020e0: 202d 703d 7079 7468 6f6e 3320 454e 5620   -p=python3 ENV 
-000020f0: 286f 7220 7079 7468 6f6e 3320 2d6d 2076  (or python3 -m v
-00002100: 656e 7620 454e 5629 0a73 6f75 7263 6520  env ENV).source 
-00002110: 454e 562f 6269 6e2f 6163 7469 7661 7465  ENV/bin/activate
-00002120: 0a70 6970 2069 6e73 7461 6c6c 202d 7220  .pip install -r 
-00002130: 7265 7175 6972 656d 656e 7473 2e74 7874  requirements.txt
-00002140: 0a70 6970 2069 6e73 7461 6c6c 202e 0a60  .pip install ..`
-00002150: 6060 0a0a 5768 656e 206d 6574 2077 6974  ``..When met wit
-00002160: 6820 7065 726d 6973 7369 6f6e 2070 726f  h permission pro
-00002170: 626c 656d 2069 6e20 696e 7374 616c 6c61  blem in installa
-00002180: 7469 6f6e 2c20 706c 6561 7365 2074 7279  tion, please try
-00002190: 2074 6865 2066 6f6c 6c6f 7769 6e67 2063   the following c
-000021a0: 6f6d 6d61 6e64 206c 696e 652e 0a0a 6060  ommand line...``
-000021b0: 600a 7069 7020 696e 7374 616c 6c20 2d65  `.pip install -e
-000021c0: 202e 0a60 6060 0a0a 4966 206d 756c 7469   ..```..If multi
-000021d0: 706c 7920 7665 7273 696f 6e73 206f 6620  ply versions of 
-000021e0: 6077 6764 6020 7765 7265 2069 6e73 7461  `wgd` were insta
-000021f0: 6c6c 6564 2069 6e20 7468 6520 7379 7374  lled in the syst
-00002200: 656d 2c20 706c 6561 7365 2061 6464 2074  em, please add t
-00002210: 6865 2072 6967 6874 2070 6174 6820 6f66  he right path of
-00002220: 2069 6e74 6572 6573 7465 6420 7665 7273   interested vers
-00002230: 696f 6e20 696e 746f 2074 6865 2065 6e76  ion into the env
-00002240: 6972 6f6e 6d65 6e74 2076 6172 6961 626c  ironment variabl
-00002250: 6573 2c20 666f 7220 6578 616d 706c 650a  es, for example.
-00002260: 0a60 6060 0a65 7870 6f72 7420 5041 5448  .```.export PATH
-00002270: 3d22 2450 4154 483a 7e2f 2e6c 6f63 616c  ="$PATH:~/.local
-00002280: 2f62 696e 2f77 6764 220a 6060 600a 0a4e  /bin/wgd".```..N
-00002290: 6f74 6520 7468 6174 2074 6865 2076 6572  ote that the ver
-000022a0: 7369 6f6e 206f 6620 606e 756d 7079 6020  sion of `numpy` 
-000022b0: 6973 2069 6d70 6f72 7461 6e74 2028 666f  is important (fo
-000022c0: 7220 6d61 6e79 206f 7468 6572 2070 6163  r many other pac
-000022d0: 6b61 6765 7320 6172 6520 7468 6520 7361  kages are the sa
-000022e0: 6d65 206f 6620 636f 7572 7365 292c 2065  me of course), e
-000022f0: 7370 6563 6961 6c6c 7920 666f 7220 6066  specially for `f
-00002300: 6173 7463 6c75 7374 6572 6020 7061 636b  astcluster` pack
-00002310: 6167 652e 2049 6e20 6f75 7220 7465 7374  age. In our test
-00002320: 2c20 7468 6520 606e 756d 7079 6020 312e  , the `numpy` 1.
-00002330: 3139 2e30 2077 6f72 6b73 2066 696e 6520  19.0 works fine 
-00002340: 6f6e 2060 7079 7468 6f6e 332e 362f 3860  on `python3.6/8`
-00002350: 2e20 4966 2079 6f75 206d 6574 2073 6f6d  . If you met som
-00002360: 6520 6572 726f 7273 206f 7220 7761 726e  e errors or warn
-00002370: 696e 6773 2061 626f 7574 2060 6e75 6d70  ings about `nump
-00002380: 7960 2c20 6d61 7962 6520 636f 6e73 6964  y`, maybe consid
-00002390: 6572 696e 6720 7072 652d 696e 7374 616c  ering pre-instal
-000023a0: 6c20 606e 756d 7079 6020 6173 2031 2e31  l `numpy` as 1.1
-000023b0: 392e 3020 6f72 206f 7468 6572 2063 6c6f  9.0 or other clo
-000023c0: 7365 2d62 7920 7665 7273 696f 6e73 2062  se-by versions b
-000023d0: 6566 6f72 6520 796f 7520 696e 7374 616c  efore you instal
-000023e0: 6c20 6077 6764 602e 0a0a 2323 2050 6172  l `wgd`...## Par
-000023f0: 616d 6574 6572 730a 0a54 6865 7265 2061  ameters..There a
-00002400: 7265 2037 206d 6169 6e20 7072 6f67 7261  re 7 main progra
-00002410: 6d73 2069 6e20 6077 6764 2076 3260 3a20  ms in `wgd v2`: 
-00002420: 6064 6d64 602c 6066 6f63 7573 602c 606b  `dmd`,`focus`,`k
-00002430: 7364 602c 606d 6978 602c 6070 6561 6b60  sd`,`mix`,`peak`
-00002440: 2c60 7379 6e60 2c60 7669 7a60 2e20 4865  ,`syn`,`viz`. He
-00002450: 7265 6166 7465 7220 7765 2077 696c 6c20  reafter we will 
-00002460: 7072 6f76 6964 6520 6120 6465 7461 696c  provide a detail
-00002470: 6564 2065 6c75 6369 6461 7469 6f6e 206f  ed elucidation o
-00002480: 6e20 6561 6368 206f 6620 7468 6520 7072  n each of the pr
-00002490: 6f67 7261 6d20 616e 6420 6974 7320 6173  ogram and its as
-000024a0: 736f 6369 6174 6564 2070 6172 616d 6574  sociated paramet
-000024b0: 6572 732e 2050 6c65 6173 6520 7265 6665  ers. Please refe
-000024c0: 7220 746f 2074 6865 205b 5573 6167 655d  r to the [Usage]
-000024d0: 2823 7573 6167 6529 2066 6f72 2074 6865  (#usage) for the
-000024e0: 2073 6365 6e61 7269 6f73 2074 6f20 7768   scenarios to wh
-000024f0: 6963 6820 6561 6368 2070 6172 616d 6574  ich each paramet
-00002500: 6572 2061 7070 6c69 6573 2e0a 0a54 6865  er applies...The
-00002510: 2070 726f 6772 616d 2060 7767 6420 646d   program `wgd dm
-00002520: 6460 2063 616e 2072 6561 6c69 7a65 2074  d` can realize t
-00002530: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
-00002540: 6620 7768 6f6c 6520 7061 7261 6e6f 6d65  f whole paranome
-00002550: 2c20 5242 4873 2028 5265 6369 7072 6f63  , RBHs (Reciproc
-00002560: 616c 2042 6573 7420 4869 7473 292c 204d  al Best Hits), M
-00002570: 5242 4873 2028 4d75 6c74 6970 6c65 2052  RBHs (Multiple R
-00002580: 6563 6970 726f 6361 6c20 4265 7374 2048  eciprocal Best H
-00002590: 6974 7329 2c20 6f72 7468 6f67 726f 7570  its), orthogroup
-000025a0: 7320 616e 6420 736f 6d65 206f 7468 6572  s and some other
-000025b0: 206f 7274 686f 6772 6f75 702d 7265 6c61   orthogroup-rela
-000025c0: 7465 6420 6675 6e63 7469 6f6e 732c 2069  ted functions, i
-000025d0: 6e63 6c75 6469 6e67 2063 6972 6375 6d73  ncluding circums
-000025e0: 6372 6970 7469 6f6e 206f 6620 6e65 7374  cription of nest
-000025f0: 6564 2073 696e 676c 652d 636f 7079 206f  ed single-copy o
-00002600: 7274 686f 6772 6f75 7073 2028 4e53 4f47  rthogroups (NSOG
-00002610: 7329 2c20 756e 6269 6173 6564 2074 6573  s), unbiased tes
-00002620: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
-00002630: 206f 7274 686f 6772 6f75 7073 2028 534f   orthogroups (SO
-00002640: 4773 2920 6f76 6572 206d 6973 7369 6e67  Gs) over missing
-00002650: 2069 6e70 6172 616c 6f67 732c 2063 6f6e   inparalogs, con
-00002660: 7374 7275 6374 696f 6e20 6f66 2042 5553  struction of BUS
-00002670: 434f 2d67 7569 6465 6420 7369 6e67 6c65  CO-guided single
-00002680: 2d63 6f70 7920 6f72 7468 6f67 726f 7570  -copy orthogroup
-00002690: 7320 2853 4f47 7329 2c61 6e64 2074 6865  s (SOGs),and the
-000026a0: 2063 6f6c 6c69 6e65 6172 2063 6f61 6c65   collinear coale
-000026b0: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
-000026c0: 206f 6620 7068 796c 6f67 656e 792e 0a60   of phylogeny..`
-000026d0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-000026e0: 6e63 6573 2028 6f70 7469 6f6e 290a 2d2d  nces (option).--
-000026f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002700: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002710: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002720: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002730: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
-00002740: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
-00002750: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
-00002760: 792c 2064 6566 6175 6c74 2077 6764 5f64  y, default wgd_d
-00002770: 6d64 0a2d 742c 202d 2d74 6d70 6469 722c  md.-t, --tmpdir,
-00002780: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
-00002790: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
-000027a0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
-000027b0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
-000027c0: 6e2c 2074 6865 2074 6d70 6469 7220 776f  n, the tmpdir wo
-000027d0: 756c 6420 6265 2061 7373 6967 6e65 6420  uld be assigned 
-000027e0: 7261 6e64 6f6d 206e 616d 6573 2069 6e20  random names in 
-000027f0: 6375 7272 656e 7420 6469 7265 6374 6f72  current director
-00002800: 7920 616e 6420 6175 746f 6d61 7465 6c79  y and automately
-00002810: 2072 656d 6f76 6564 2061 7420 7468 6520   removed at the 
-00002820: 636f 6d70 6c65 7469 6f6e 206f 6620 7072  completion of pr
-00002830: 6f67 7261 6d2c 2065 6c73 6520 7468 6520  ogram, else the 
-00002840: 746d 7064 6972 2077 6f75 6c64 2062 6520  tmpdir would be 
-00002850: 6b65 7074 0a2d 702c 202d 2d70 726f 742c  kept.-p, --prot,
-00002860: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00002870: 6574 6865 7220 7573 696e 6720 7072 6f74  ether using prot
-00002880: 6569 6e20 6f72 206e 7563 6c65 6f74 6964  ein or nucleotid
-00002890: 6520 7365 7175 656e 6365 730a 2d63 2c20  e sequences.-c, 
-000028a0: 2d2d 6373 636f 7265 2c20 7468 6520 632d  --cscore, the c-
-000028b0: 7363 6f72 6520 746f 2072 6573 7472 6963  score to restric
-000028c0: 7420 7468 6520 686f 6d6f 6c6f 6720 7369  t the homolog si
-000028d0: 6d69 6c61 7269 7479 206f 6620 4d52 4248  milarity of MRBH
-000028e0: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
-000028f0: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
-00002900: 656e 2c20 7468 6520 632d 7363 6f72 6520  en, the c-score 
-00002910: 6675 6e63 696f 6e20 776f 756c 646e 2774  funcion wouldn't
-00002920: 2062 6520 6163 7469 7661 7465 642c 2065   be activated, e
-00002930: 6c73 6520 6578 7065 6374 696e 6720 6120  lse expecting a 
-00002940: 6465 6369 6d61 6c20 7769 7468 696e 2074  decimal within t
-00002950: 6865 2072 616e 6765 206f 6620 3020 616e  he range of 0 an
-00002960: 6420 310a 2d49 2c20 2d2d 696e 666c 6174  d 1.-I, --inflat
-00002970: 696f 6e2c 2074 6865 2069 6e66 6c61 7469  ion, the inflati
-00002980: 6f6e 2066 6163 746f 7220 666f 7220 4d43  on factor for MC
-00002990: 4c20 7072 6f67 7261 6d2c 2064 6566 6175  L program, defau
-000029a0: 6c74 2032 2e30 2c20 7769 7468 2068 6967  lt 2.0, with hig
-000029b0: 6865 7220 7661 6c75 6520 6c65 6164 696e  her value leadin
-000029c0: 6720 746f 206d 6f72 6520 6275 7420 736d  g to more but sm
-000029d0: 616c 6c65 7220 636c 7573 7465 7273 0a2d  aller clusters.-
-000029e0: 652c 202d 2d65 7661 6c2c 2074 6865 2065  e, --eval, the e
-000029f0: 2d76 616c 7565 2063 7574 2d6f 6666 2066  -value cut-off f
-00002a00: 6f72 2073 696d 696c 6172 6974 7920 696e  or similarity in
-00002a10: 2064 6961 6d6f 6e64 2061 6e64 2f6f 7220   diamond and/or 
-00002a20: 686d 6d65 722c 2064 6566 6175 6c74 2031  hmmer, default 1
-00002a30: 652d 3130 0a2d 2d74 6f5f 7374 6f70 2c20  e-10.--to_stop, 
-00002a40: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00002a50: 7468 6572 2074 6f20 7472 616e 736c 6174  ther to translat
-00002a60: 6520 7468 726f 7567 6820 5354 4f50 2063  e through STOP c
-00002a70: 6f64 6f6e 732c 2069 6620 7468 6520 666c  odons, if the fl
-00002a80: 6167 2077 6173 2073 6574 2c20 7472 616e  ag was set, tran
-00002a90: 736c 6174 696f 6e20 7769 6c6c 2062 6520  slation will be 
-00002aa0: 7465 726d 696e 6174 6564 2061 7420 7468  terminated at th
-00002ab0: 6520 6669 7273 7420 696e 2d66 7261 6d65  e first in-frame
-00002ac0: 2073 746f 7020 636f 646f 6e2c 2065 6c73   stop codon, els
-00002ad0: 6520 6120 6675 6c6c 2074 7261 6e73 6c61  e a full transla
-00002ae0: 7469 6f6e 2063 6f6e 7469 6e75 696e 6720  tion continuing 
-00002af0: 6f6e 2070 6173 7369 6e67 2061 6e79 2073  on passing any s
-00002b00: 746f 7020 636f 646f 6e73 2077 6f75 6c64  top codons would
-00002b10: 2062 6520 696e 6974 6961 7465 640a 2d2d   be initiated.--
-00002b20: 6364 732c 2066 6c61 6720 6f70 7469 6f6e  cds, flag option
-00002b30: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
-00002b40: 7920 7472 616e 736c 6174 6520 7468 6520  y translate the 
-00002b50: 636f 6d70 6c65 7465 2043 4453 2074 6861  complete CDS tha
-00002b60: 7420 7374 6172 7473 2077 6974 6820 6120  t starts with a 
-00002b70: 7661 6c69 6420 7374 6172 7420 636f 646f  valid start codo
-00002b80: 6e20 616e 6420 6f6e 6c79 2063 6f6e 7461  n and only conta
-00002b90: 696e 7320 6120 7369 6e67 6c65 2069 6e2d  ins a single in-
-00002ba0: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
-00002bb0: 2061 7420 7468 6520 656e 6420 616e 6420   at the end and 
-00002bc0: 6d75 7374 2062 6520 6469 7669 6461 626c  must be dividabl
-00002bd0: 6520 6279 2074 6872 6565 2c20 6966 2074  e by three, if t
-00002be0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00002bf0: 206f 6e6c 7920 7468 6520 636f 6d70 6c65   only the comple
-00002c00: 7465 2043 4453 2077 6f75 6c64 2062 6520  te CDS would be 
-00002c10: 7472 616e 736c 6174 6564 0a2d 662c 202d  translated.-f, -
-00002c20: 2d66 6f63 7573 2c20 7468 6520 7370 6563  -focus, the spec
-00002c30: 6965 7320 746f 2062 6520 6d65 7267 6564  ies to be merged
-00002c40: 206f 6e20 6c6f 6361 6c20 4d52 4248 732c   on local MRBHs,
-00002c50: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
-00002c60: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
-00002c70: 2c20 7468 6520 6c6f 6361 6c20 4d52 4248  , the local MRBH
-00002c80: 7320 776f 756c 646e 2774 2062 6520 696e  s wouldn't be in
-00002c90: 6665 7272 6564 0a2d 6170 2c20 2d2d 616e  ferred.-ap, --an
-00002ca0: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
-00002cb0: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
-00002cc0: 7461 2066 696c 6520 6672 6f6d 2069 2d61  ta file from i-a
-00002cd0: 6468 6f72 6520 666f 7220 636f 6e73 7472  dhore for constr
-00002ce0: 7563 7469 6e67 2074 6865 206f 7274 686f  ucting the ortho
-00002cf0: 6772 6f75 7073 2077 6974 6820 616e 6368  groups with anch
-00002d00: 6f72 2070 6169 7273 2c20 6465 6661 756c  or pairs, defaul
-00002d10: 7420 4e6f 6e65 0a2d 736d 2c20 2d2d 7365  t None.-sm, --se
-00002d20: 676d 656e 7473 2c20 7468 6520 7365 676d  gments, the segm
-00002d30: 656e 7473 2064 6174 6166 696c 6520 7573  ents datafile us
-00002d40: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
-00002d50: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-00002d60: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
-00002d70: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
-00002d80: 2d6c 652c 202d 2d6c 6973 7465 6c65 6d65  -le, --listeleme
-00002d90: 6e74 732c 2074 6865 206c 6973 7473 6567  nts, the listseg
-00002da0: 6d65 6e74 7320 6461 7461 2066 696c 6520  ments data file 
-00002db0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
-00002dc0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
-00002dd0: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
-00002de0: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00002df0: 650a 2d67 742c 202d 2d67 656e 6574 6162  e.-gt, --genetab
-00002e00: 6c65 2c20 7468 6520 6765 6e65 2074 6162  le, the gene tab
-00002e10: 6c65 2064 6174 6166 696c 6520 7573 6564  le datafile used
-00002e20: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
-00002e30: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
-00002e40: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
-00002e50: 2064 6566 6175 6c74 204e 6f6e 650a 2d63   default None.-c
-00002e60: 6f63 2c20 2d2d 636f 6c6c 696e 6561 7263  oc, --collinearc
-00002e70: 6f61 6c65 7363 656e 6365 2c20 666c 6167  oalescence, flag
-00002e80: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00002e90: 2074 6f20 696e 6974 6961 7465 2074 6865   to initiate the
-00002ea0: 2063 6f6c 6c69 6e65 6172 2063 6f61 6c65   collinear coale
-00002eb0: 7363 656e 6365 2061 6e61 6c79 7369 732c  scence analysis,
-00002ec0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00002ed0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00002ee0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00002ef0: 6961 7465 640a 2d6b 662c 202d 2d6b 6565  iated.-kf, --kee
-00002f00: 7066 6173 7461 2c20 666c 6167 206f 7074  pfasta, flag opt
-00002f10: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00002f20: 6f75 7470 7574 2074 6865 2073 6571 7565  output the seque
-00002f30: 6e63 6520 696e 666f 726d 6174 696f 6e20  nce information 
-00002f40: 6f66 204d 5242 4873 2c20 6966 2074 6865  of MRBHs, if the
-00002f50: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00002f60: 6865 2073 6571 7565 6e63 6573 206f 6620  he sequences of 
-00002f70: 4d52 4248 7320 776f 756c 6420 6265 2069  MRBHs would be i
-00002f80: 6e20 6f75 7470 7574 0a2d 6b64 2c20 2d2d  n output.-kd, --
-00002f90: 6b65 6570 6475 706c 6963 6174 6573 2c20  keepduplicates, 
-00002fa0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00002fb0: 7468 6572 2074 6f20 616c 6c6f 7720 7468  ther to allow th
-00002fc0: 6520 7361 6d65 2067 656e 6520 746f 206f  e same gene to o
-00002fd0: 6363 7572 2069 6e20 6469 6666 6572 656e  ccur in differen
-00002fe0: 7420 4d52 4248 7320 286f 6e6c 7920 6d65  t MRBHs (only me
-00002ff0: 616e 696e 6766 756c 2077 6865 6e20 7468  aningful when th
-00003000: 6520 6373 636f 7265 2077 6173 2075 7365  e cscore was use
-00003010: 6429 2c20 6966 2074 6865 2066 6c61 6720  d), if the flag 
-00003020: 7761 7320 7365 742c 2074 6865 2073 616d  was set, the sam
-00003030: 6520 6765 6e65 2063 6f75 6c64 2062 6520  e gene could be 
-00003040: 6173 7369 676e 6564 2074 6f20 6469 6666  assigned to diff
-00003050: 6572 656e 7420 4d52 4248 730a 2d67 6d2c  erent MRBHs.-gm,
-00003060: 202d 2d67 6c6f 6261 6c6d 7262 682c 2066   --globalmrbh, f
-00003070: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003080: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00003090: 676c 6f62 616c 204d 5242 4873 2063 6f6e  global MRBHs con
-000030a0: 7374 7275 6374 696f 6e2c 2069 6620 7468  struction, if th
-000030b0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000030c0: 7468 6520 2d2d 666f 6375 7320 6f70 7469  the --focus opti
-000030d0: 6f6e 2077 6f75 6c64 2062 6520 6967 6e6f  on would be igno
-000030e0: 7265 6420 616e 6420 6f6e 6c79 2067 6c6f  red and only glo
-000030f0: 6261 6c20 4d52 4248 7320 776f 756c 6420  bal MRBHs would 
-00003100: 6265 2062 7569 6c74 0a2d 6e2c 202d 2d6e  be built.-n, --n
-00003110: 7468 7265 6164 732c 2074 6865 206e 756d  threads, the num
-00003120: 6265 7220 6f66 2074 6872 6561 6473 2074  ber of threads t
-00003130: 6f20 7573 652c 2064 6566 6175 6c74 2034  o use, default 4
-00003140: 0a2d 6f69 2c20 2d2d 6f72 7468 6f69 6e66  .-oi, --orthoinf
-00003150: 6572 2c20 666c 6167 206f 7074 696f 6e2c  er, flag option,
-00003160: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003170: 6961 7465 206f 7274 686f 6772 6f75 7020  iate orthogroup 
-00003180: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
-00003190: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000031a0: 7468 6520 6f72 7468 6f67 726f 7570 2069  the orthogroup i
-000031b0: 6e66 6572 6e65 6365 2070 726f 6772 616d  nfernece program
-000031c0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-000031d0: 7465 640a 2d6f 6f2c 202d 2d6f 6e6c 796f  ted.-oo, --onlyo
-000031e0: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
-000031f0: 6e2c 2077 6865 7468 6572 2074 6f20 6f6e  n, whether to on
-00003200: 6c79 2063 6f6e 6475 6374 206f 7274 686f  ly conduct ortho
-00003210: 6772 6f75 7020 696e 6665 726e 6563 652c  group infernece,
-00003220: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003230: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
-00003240: 7274 686f 6772 6f75 7020 696e 6665 726e  rthogroup infern
-00003250: 6563 6520 7069 7065 6c69 6e65 2077 6f75  ece pipeline wou
-00003260: 6c64 2062 6520 7065 7266 6f72 6d65 6420  ld be performed 
-00003270: 7768 696c 6520 7468 6520 6f74 6865 7220  while the other 
-00003280: 616e 616c 7973 6973 2077 6f75 6c64 6e27  analysis wouldn'
-00003290: 7420 6265 2069 6e69 7469 6174 6564 0a2d  t be initiated.-
-000032a0: 676e 2c20 2d2d 6765 746e 736f 672c 2066  gn, --getnsog, f
-000032b0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000032c0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-000032d0: 7468 6520 7365 6172 6368 696e 6720 666f  the searching fo
-000032e0: 7220 6e65 7374 6564 2073 696e 676c 652d  r nested single-
-000032f0: 636f 7079 2067 656e 6520 6661 6d69 6c69  copy gene famili
-00003300: 6573 2028 4e53 4f47 7329 2028 6f6e 6c79  es (NSOGs) (only
-00003310: 206d 6561 6e69 6e67 6675 6c20 7768 656e   meaningful when
-00003320: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
-00003330: 696e 6665 726e 6563 6520 7069 7065 6c69  infernece pipeli
-00003340: 6e65 2077 6173 2061 6374 6976 6174 6564  ne was activated
-00003350: 292c 2069 6620 7468 6520 666c 6167 2077  ), if the flag w
-00003360: 6173 2073 6574 2c20 6164 6469 7469 6f6e  as set, addition
-00003370: 616c 204e 534f 4773 2061 6e61 6c79 7369  al NSOGs analysi
-00003380: 7320 776f 756c 6420 6265 2070 6572 666f  s would be perfo
-00003390: 726d 6564 2062 6573 6964 6573 2074 6865  rmed besides the
-000033a0: 2062 6173 6963 206f 7274 686f 6772 6f75   basic orthogrou
-000033b0: 7020 696e 6665 726e 6563 650a 2d74 7265  p infernece.-tre
-000033c0: 652c 202d 2d74 7265 655f 6d65 7468 6f64  e, --tree_method
-000033d0: 2c20 7768 6963 6820 6765 6e65 2074 7265  , which gene tre
-000033e0: 6520 696e 6665 7265 6e63 6520 7072 6f67  e inference prog
-000033f0: 7261 6d20 746f 2069 6e76 6f6b 6520 286f  ram to invoke (o
-00003400: 6e6c 7920 6d65 616e 696e 6766 756c 2077  nly meaningful w
-00003410: 6865 6e20 7468 6520 636f 6c6c 696e 6561  hen the collinea
-00003420: 7220 636f 616c 6573 6365 6e63 652c 2067  r coalescence, g
-00003430: 656e 652d 746f 2d66 616d 696c 7920 6173  ene-to-family as
-00003440: 7369 676e 6d65 6e74 206f 7220 4e53 4f47  signment or NSOG
-00003450: 7320 616e 616c 7973 6973 2077 6572 6520  s analysis were 
-00003460: 6163 7469 7661 7465 6429 2c20 6465 6661  activated), defa
-00003470: 756c 7420 6661 7374 7472 6565 0a2d 7473  ult fasttree.-ts
-00003480: 2c20 2d2d 7472 6565 7365 742c 2074 6865  , --treeset, the
-00003490: 2070 6172 616d 6574 6572 7320 7365 7474   parameters sett
-000034a0: 696e 6720 666f 7220 6765 6e65 2074 7265  ing for gene tre
-000034b0: 6520 696e 6665 7265 6e63 652c 2064 6566  e inference, def
-000034c0: 6175 6c74 204e 6f6e 652c 2074 6869 7320  ault None, this 
-000034d0: 6f70 7469 6f6e 2063 616e 2062 6520 7072  option can be pr
-000034e0: 6f76 6964 6564 206d 756c 7469 706c 6520  ovided multiple 
-000034f0: 7469 6d65 730a 2d6d 632c 202d 2d6d 736f  times.-mc, --mso
-00003500: 6763 7574 2c20 7468 6520 7261 7469 6f20  gcut, the ratio 
-00003510: 6375 746f 6666 2066 6f72 206d 6f73 746c  cutoff for mostl
-00003520: 7920 7369 6e67 6c65 2d63 6f70 7920 6661  y single-copy fa
-00003530: 6d69 6c79 2028 6d65 616e 696e 6766 756c  mily (meaningful
-00003540: 2077 6865 6e20 6163 7469 7661 7469 6e67   when activating
-00003550: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
-00003560: 696e 6665 726e 6563 6520 7069 7065 6c69  infernece pipeli
-00003570: 6e65 2920 616e 6420 7370 6563 6965 7320  ne) and species 
-00003580: 7265 7072 6573 656e 7461 7469 6f6e 2069  representation i
-00003590: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
-000035a0: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
-000035b0: 2c20 6465 6661 756c 7420 302e 382e 0a2d  , default 0.8..-
-000035c0: 6761 2c20 2d2d 6765 6e65 6173 7369 676e  ga, --geneassign
-000035d0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-000035e0: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-000035f0: 7465 2074 6865 2067 656e 652d 746f 2d66  te the gene-to-f
-00003600: 616d 696c 7920 6173 7369 676e 6d65 6e74  amily assignment
-00003610: 2061 6e61 6c79 7369 732c 2069 6620 7468   analysis, if th
-00003620: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00003630: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
-00003640: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
-00003650: 2d73 612c 202d 2d73 6571 3261 7373 6967  -sa, --seq2assig
-00003660: 6e2c 2074 6865 2071 7565 7269 6564 2073  n, the queried s
-00003670: 6571 7565 6e63 6573 2064 6174 6120 6669  equences data fi
-00003680: 6c65 2069 6e20 6765 6e65 2d74 6f2d 6661  le in gene-to-fa
-00003690: 6d69 6c79 2061 7373 6967 6e6d 656e 7420  mily assignment 
-000036a0: 616e 616c 7973 6973 2c20 6465 6661 756c  analysis, defaul
-000036b0: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
-000036c0: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
-000036d0: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
-000036e0: 6573 0a2d 6661 2c20 2d2d 6661 6d32 6173  es.-fa, --fam2as
-000036f0: 7369 676e 2c20 7468 6520 7175 6572 6965  sign, the querie
-00003700: 6420 6661 6d69 6c69 7920 6461 7461 2066  d familiy data f
-00003710: 696c 6520 696e 2067 656e 652d 746f 2d66  ile in gene-to-f
-00003720: 616d 696c 7920 6173 7369 676e 6d65 6e74  amily assignment
-00003730: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
-00003740: 6c74 204e 6f6e 650a 2d63 632c 202d 2d63  lt None.-cc, --c
-00003750: 6f6e 6361 742c 2066 6c61 6720 6f70 7469  oncat, flag opti
-00003760: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00003770: 6e69 7469 6174 6520 7468 6520 636f 6e63  nitiate the conc
-00003780: 6174 656e 6174 696f 6e20 7069 7065 6c69  atenation pipeli
-00003790: 6e65 2066 6f72 206f 7274 686f 6772 6f75  ne for orthogrou
-000037a0: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
-000037b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-000037c0: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
-000037d0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-000037e0: 640a 2d74 652c 202d 2d74 6573 7473 6f67  d.-te, --testsog
-000037f0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00003800: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00003810: 7465 2074 6865 2075 6e62 6961 7365 6420  te the unbiased 
-00003820: 7465 7374 206f 6620 7369 6e67 6c65 2d63  test of single-c
-00003830: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
-00003840: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00003850: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
-00003860: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
-00003870: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
-00003880: 696e 732c 2074 6865 206e 756d 6265 7220  ins, the number 
-00003890: 6f66 2062 696e 7320 6469 7669 6465 6420  of bins divided 
-000038a0: 696e 2074 6865 2067 656e 6520 6c65 6e67  in the gene leng
-000038b0: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
-000038c0: 2c20 6465 6661 756c 7420 3130 300a 2d6e  , default 100.-n
-000038d0: 702c 202d 2d6e 6f72 6d61 6c69 7a65 6470  p, --normalizedp
-000038e0: 6572 6365 6e74 2c20 7468 6520 7065 7263  ercent, the perc
-000038f0: 656e 7461 6765 206f 6620 7570 7065 7220  entage of upper 
-00003900: 6869 7473 2075 7365 6420 666f 7220 6765  hits used for ge
-00003910: 6e65 206c 656e 6774 6820 6e6f 726d 616c  ne length normal
-00003920: 697a 6174 696f 6e2c 2064 6566 6175 6c74  ization, default
-00003930: 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d   5.-nn, --nonorm
-00003940: 616c 697a 6174 696f 6e2c 2066 6c61 6720  alization, flag 
-00003950: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00003960: 746f 2063 616c 6c20 6f66 6620 7468 6520  to call off the 
-00003970: 6e6f 726d 616c 697a 6174 696f 6e2c 2069  normalization, i
-00003980: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003990: 6574 2c20 6e6f 206e 6f72 6d61 6c69 7a61  et, no normaliza
-000039a0: 7469 6f6e 2077 6f75 6c64 2062 6520 636f  tion would be co
-000039b0: 6e64 7563 7465 640a 2d62 736f 672c 202d  nducted.-bsog, -
-000039c0: 2d62 7573 636f 736f 672c 2066 6c61 6720  -buscosog, flag 
-000039d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000039e0: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
-000039f0: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
-00003a00: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-00003a10: 6d69 6c79 2061 6e61 6c79 7369 732c 2069  mily analysis, i
-00003a20: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003a30: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00003a40: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003a50: 7465 640a 2d62 686d 6d2c 202d 2d62 7573  ted.-bhmm, --bus
-00003a60: 636f 686d 6d2c 2074 6865 2048 4d4d 2070  cohmm, the HMM p
-00003a70: 726f 6669 6c65 2064 6174 6166 696c 6520  rofile datafile 
-00003a80: 696e 2074 6865 2062 7573 636f 2d67 7569  in the busco-gui
-00003a90: 6465 6420 7369 6e67 6c65 2d63 6f70 7920  ded single-copy 
-00003aa0: 6765 6e65 2066 616d 696c 7920 616e 616c  gene family anal
-00003ab0: 7973 6973 2c20 6465 6661 756c 7420 4e6f  ysis, default No
-00003ac0: 6e65 0a2d 6263 7466 2c20 2d2d 6275 7363  ne.-bctf, --busc
-00003ad0: 6f63 7574 6f66 662c 2074 6865 2048 4d4d  ocutoff, the HMM
-00003ae0: 2073 636f 7265 2063 7574 6f66 6620 6461   score cutoff da
-00003af0: 7461 6669 6c65 2069 6e20 7468 6520 6275  tafile in the bu
-00003b00: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
-00003b10: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-00003b20: 6c79 2061 6e61 6c79 7369 732c 2064 6566  ly analysis, def
-00003b30: 6175 6c74 204e 6f6e 650a 2d6f 6620 2c2d  ault None.-of ,-
-00003b40: 2d6f 6766 6f72 6d61 742c 2066 6c61 6720  -ogformat, flag 
-00003b50: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00003b60: 746f 2061 6464 2069 6e64 6578 2074 6f20  to add index to 
-00003b70: 7468 6520 5242 4820 6661 6d69 6c69 6573  the RBH families
-00003b80: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
-00003b90: 6d20 6077 6764 2066 6f63 7573 6020 6361  m `wgd focus` ca
-00003ba0: 6e20 7265 616c 697a 6520 7468 6520 636f  n realize the co
-00003bb0: 6e63 6174 656e 6174 696f 6e2d 6261 7365  ncatenation-base
-00003bc0: 6420 616e 6420 636f 616c 6573 6365 6e63  d and coalescenc
-00003bd0: 652d 6261 7365 6420 7068 796c 6f67 656e  e-based phylogen
-00003be0: 6574 6963 2069 6e66 6572 656e 6365 2061  etic inference a
-00003bf0: 6e64 2070 6879 6c6f 6765 6e65 7469 6320  nd phylogenetic 
-00003c00: 6461 7469 6e67 206f 6620 5747 4473 2065  dating of WGDs e
-00003c10: 7463 2e0a 6060 600a 7767 6420 666f 6375  tc..```.wgd focu
-00003c20: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-00003c30: 6e63 6573 2028 6f70 7469 6f6e 290a 2d2d  nces (option).--
-00003c40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003c50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003c60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003c70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00003c80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
-00003c90: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
-00003ca0: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
-00003cb0: 792c 2064 6566 6175 6c74 2077 6764 5f66  y, default wgd_f
-00003cc0: 6f63 7573 0a2d 742c 202d 2d74 6d70 6469  ocus.-t, --tmpdi
-00003cd0: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
-00003ce0: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
-00003cf0: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
-00003d00: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
-00003d10: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
-00003d20: 7769 6c6c 2062 6520 6173 7369 676e 6564  will be assigned
-00003d30: 2072 616e 646f 6d20 6e61 6d65 7320 696e   random names in
-00003d40: 2063 7572 7265 6e74 2064 6972 6563 746f   current directo
-00003d50: 7279 2061 6e64 2061 7574 6f6d 6174 656c  ry and automatel
-00003d60: 7920 7265 6d6f 7665 6420 6174 2074 6865  y removed at the
-00003d70: 2063 6f6d 706c 6574 696f 6e20 6f66 2070   completion of p
-00003d80: 726f 6772 616d 2c20 656c 7365 2074 6865  rogram, else the
-00003d90: 2074 6d70 6469 7220 776f 756c 6420 6265   tmpdir would be
-00003da0: 206b 6570 740a 2d6e 2c20 2d2d 6e74 6872   kept.-n, --nthr
-00003db0: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
-00003dc0: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
-00003dd0: 7365 2c20 6465 6661 756c 7420 340a 2d2d  se, default 4.--
-00003de0: 746f 5f73 746f 702c 2066 6c61 6720 6f70  to_stop, flag op
-00003df0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003e00: 2074 7261 6e73 6c61 7465 2074 6872 6f75   translate throu
-00003e10: 6768 2053 544f 5020 636f 646f 6e73 2c20  gh STOP codons, 
-00003e20: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00003e30: 7365 742c 2074 7261 6e73 6c61 7469 6f6e  set, translation
-00003e40: 2077 696c 6c20 6265 2074 6572 6d69 6e61   will be termina
-00003e50: 7465 6420 6174 2074 6865 2066 6972 7374  ted at the first
-00003e60: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
-00003e70: 6f64 6f6e 2c20 656c 7365 2061 2066 756c  odon, else a ful
-00003e80: 6c20 7472 616e 736c 6174 696f 6e20 636f  l translation co
-00003e90: 6e74 696e 7569 6e67 206f 6e20 7061 7374  ntinuing on past
-00003ea0: 2061 6e79 2073 746f 7020 636f 646f 6e73   any stop codons
-00003eb0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003ec0: 7465 640a 2d2d 6364 732c 2066 6c61 6720  ted.--cds, flag 
-00003ed0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00003ee0: 746f 206f 6e6c 7920 7472 616e 736c 6174  to only translat
-00003ef0: 6520 7468 6520 636f 6d70 6c65 7465 2043  e the complete C
-00003f00: 4453 2074 6861 7420 7374 6172 7473 2077  DS that starts w
-00003f10: 6974 6820 6120 7661 6c69 6420 7374 6172  ith a valid star
-00003f20: 7420 636f 646f 6e20 616e 6420 6f6e 6c79  t codon and only
-00003f30: 2063 6f6e 7461 696e 7320 6120 7369 6e67   contains a sing
-00003f40: 6c65 2069 6e2d 6672 616d 6520 7374 6f70  le in-frame stop
-00003f50: 2063 6f64 6f6e 2061 7420 7468 6520 656e   codon at the en
-00003f60: 6420 616e 6420 6d75 7374 2062 6520 6469  d and must be di
-00003f70: 7669 6461 626c 6520 6279 2074 6872 6565  vidable by three
-00003f80: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00003f90: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
-00003fa0: 636f 6d70 6c65 7465 2043 4453 2077 6f75  complete CDS wou
-00003fb0: 6c64 2062 6520 7472 616e 736c 6174 6564  ld be translated
-00003fc0: 0a2d 2d73 7472 6970 5f67 6170 732c 2066  .--strip_gaps, f
-00003fd0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003fe0: 6865 7220 746f 2064 726f 7020 616c 6c20  her to drop all 
-00003ff0: 6761 7073 2069 6e20 6d75 6c74 6970 6c65  gaps in multiple
-00004000: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
-00004010: 656e 742c 2069 6620 7468 6520 666c 6167  ent, if the flag
-00004020: 2077 6173 2073 6574 2c20 616c 6c20 6761   was set, all ga
-00004030: 7073 2077 6f75 6c64 2062 6520 6472 6f70  ps would be drop
-00004040: 7065 640a 2d61 2c20 2d2d 616c 6967 6e65  ped.-a, --aligne
-00004050: 722c 2077 6869 6368 2061 6c69 676e 6d65  r, which alignme
-00004060: 6e74 2070 726f 6772 616d 2074 6f20 7573  nt program to us
-00004070: 652c 2064 6566 6175 6c74 206d 6166 6674  e, default mafft
-00004080: 0a2d 7472 6565 2c20 2d2d 7472 6565 5f6d  .-tree, --tree_m
-00004090: 6574 686f 642c 2077 6869 6368 2067 656e  ethod, which gen
-000040a0: 6520 7472 6565 2069 6e66 6572 656e 6365  e tree inference
-000040b0: 2070 726f 6772 616d 2074 6f20 696e 766f   program to invo
-000040c0: 6b65 2c20 6465 6661 756c 7420 6661 7374  ke, default fast
-000040d0: 7472 6565 0a2d 7473 2c20 2d2d 7472 6565  tree.-ts, --tree
-000040e0: 7365 742c 2074 6865 2070 6172 616d 6574  set, the paramet
-000040f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
-00004100: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-00004110: 6e63 652c 2064 6566 6175 6c74 204e 6f6e  nce, default Non
-00004120: 652c 2074 6869 7320 6f70 7469 6f6e 2063  e, this option c
-00004130: 616e 2062 6520 7072 6f76 6964 6564 206d  an be provided m
-00004140: 756c 7469 706c 6520 7469 6d65 730a 2d2d  ultiple times.--
-00004150: 636f 6e63 6174 656e 6174 696f 6e2c 2066  concatenation, f
-00004160: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004170: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00004180: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
-00004190: 6e2d 6261 7365 6420 7370 6563 6965 7320  n-based species 
-000041a0: 7472 6565 2069 6e66 6572 656e 6365 2c20  tree inference, 
-000041b0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-000041c0: 7365 742c 2063 6f6e 6361 7465 6e61 7469  set, concatenati
-000041d0: 6f6e 2d62 6173 6564 2073 7065 6369 6573  on-based species
-000041e0: 2074 7265 6520 776f 756c 6420 6265 2069   tree would be i
-000041f0: 6e66 6572 6564 0a2d 2d63 6f61 6c65 7363  nfered.--coalesc
-00004200: 656e 6365 2c20 666c 6167 206f 7074 696f  ence, flag optio
-00004210: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00004220: 6974 6961 7465 2074 6865 2063 6f61 6c65  itiate the coale
-00004230: 7363 656e 6365 2d62 6173 6564 2073 7065  scence-based spe
-00004240: 6369 6573 2074 7265 6520 696e 6665 7265  cies tree infere
-00004250: 6e63 652c 2069 6620 7468 6520 666c 6167  nce, if the flag
-00004260: 2077 6173 2073 6574 2c20 636f 616c 6573   was set, coales
-00004270: 6365 6e63 652d 6261 7365 6420 7370 6563  cence-based spec
-00004280: 6965 7320 7472 6565 2077 6f75 6c64 2062  ies tree would b
-00004290: 6520 696e 6665 7265 640a 2d73 702c 202d  e infered.-sp, -
-000042a0: 2d73 7065 6369 6573 7472 6565 2c20 7370  -speciestree, sp
-000042b0: 6563 6965 7320 7472 6565 2064 6174 6166  ecies tree dataf
-000042c0: 696c 6520 666f 7220 6461 7469 6e67 2c20  ile for dating, 
-000042d0: 6465 6661 756c 7420 4e6f 6e65 0a2d 642c  default None.-d,
-000042e0: 202d 2d64 6174 696e 672c 2077 6869 6368   --dating, which
-000042f0: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
-00004300: 6720 7072 6f67 7261 6d20 746f 2075 7365  g program to use
-00004310: 2c20 6465 6661 756c 7420 6e6f 6e65 0a2d  , default none.-
-00004320: 6473 2c20 2d2d 6461 7469 6e67 7365 742c  ds, --datingset,
-00004330: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
-00004340: 7365 7474 696e 6720 666f 7220 6461 7469  setting for dati
-00004350: 6e67 2070 726f 6772 616d 2c20 6465 6661  ng program, defa
-00004360: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
-00004370: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
-00004380: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
-00004390: 696d 6573 0a2d 6e73 2c20 2d2d 6e73 6974  imes.-ns, --nsit
-000043a0: 6573 2c20 7468 6520 6e73 6974 6573 2069  es, the nsites i
-000043b0: 6e66 6f72 6d61 7469 6f6e 2066 6f72 2072  nformation for r
-000043c0: 3873 2064 6174 696e 672c 2064 6566 6175  8s dating, defau
-000043d0: 6c74 204e 6f6e 650a 2d6f 742c 202d 2d6f  lt None.-ot, --o
-000043e0: 7574 6772 6f75 702c 2074 6865 206f 7574  utgroup, the out
-000043f0: 6772 6f75 7020 696e 666f 726d 6174 696f  group informatio
-00004400: 6e20 666f 7220 7238 7320 6461 7469 6e67  n for r8s dating
-00004410: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004420: 7074 2c20 2d2d 7061 7274 6974 696f 6e2c  pt, --partition,
-00004430: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004440: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00004450: 6520 7061 7274 6974 696f 6e20 6461 7469  e partition dati
-00004460: 6e67 2061 6e61 6c79 7369 7320 666f 7220  ng analysis for 
-00004470: 636f 646f 6e2c 2069 6620 7468 6520 666c  codon, if the fl
-00004480: 6167 2077 6173 2073 6574 2c20 616e 2061  ag was set, an a
-00004490: 6464 6974 696f 6e61 6c20 7061 7274 6974  dditional partit
-000044a0: 696f 6e20 6461 7469 6e67 2061 6e61 6c79  ion dating analy
-000044b0: 7369 7320 776f 756c 6420 6265 2069 6e69  sis would be ini
-000044c0: 7469 6174 6564 0a2d 616d 2c20 2d2d 6161  tiated.-am, --aa
-000044d0: 6d6f 6465 6c2c 2077 6869 6368 2070 726f  model, which pro
-000044e0: 7465 696e 206d 6f64 656c 2074 6f20 6265  tein model to be
-000044f0: 2075 7365 6420 696e 206d 636d 6374 7265   used in mcmctre
-00004500: 652c 2064 6566 6175 6c74 2070 6f69 7373  e, default poiss
-00004510: 6f6e 0a2d 6b73 2c20 666c 6167 206f 7074  on.-ks, flag opt
-00004520: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00004530: 696e 6974 6961 7465 204b 7320 6361 6c63  initiate Ks calc
-00004540: 756c 6174 696f 6e20 666f 7220 686f 6d6f  ulation for homo
-00004550: 6c6f 6775 6573 2069 6e20 7468 6520 7072  logues in the pr
-00004560: 6f76 6964 6564 206f 7274 686f 6c6f 676f  ovided orthologo
-00004570: 7573 2067 656e 6520 6661 6d69 6c79 0a2d  us gene family.-
-00004580: 2d61 6e6e 6f74 6174 696f 6e2c 2077 6869  -annotation, whi
-00004590: 6368 2061 6e6e 6f74 6174 696f 6e20 7072  ch annotation pr
-000045a0: 6f67 7261 6d20 746f 2075 7365 2c20 6465  ogram to use, de
-000045b0: 6661 756c 7420 4e6f 6e65 0a2d 2d70 6169  fault None.--pai
-000045c0: 7277 6973 652c 2066 6c61 6720 6f70 7469  rwise, flag opti
-000045d0: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-000045e0: 6e69 7469 6174 6520 7061 6972 7769 7365  nitiate pairwise
-000045f0: 204b 7320 6573 7469 6d61 7469 6f6e 2c20   Ks estimation, 
-00004600: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00004610: 7365 742c 2070 6169 7277 6973 6520 4b73  set, pairwise Ks
-00004620: 2076 616c 7565 7320 776f 756c 6420 6265   values would be
-00004630: 2065 7374 696d 6174 6564 0a2d 6564 2c20   estimated.-ed, 
-00004640: 2d2d 6567 676e 6f67 6461 7461 2c20 7468  --eggnogdata, th
-00004650: 6520 6567 676e 6f67 2061 6e6e 6f74 6174  e eggnog annotat
-00004660: 696f 6e20 6461 7461 6669 6c65 2c20 6465  ion datafile, de
-00004670: 6661 756c 7420 4e6f 6e65 0a2d 2d70 6661  fault None.--pfa
-00004680: 6d2c 2077 6869 6368 206f 7074 696f 6e20  m, which option 
-00004690: 746f 2075 7365 2066 6f72 2070 6661 6d20  to use for pfam 
-000046a0: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
-000046b0: 756c 7420 4e6f 6e65 0a2d 2d64 6d6e 622c  ult None.--dmnb,
-000046c0: 2074 6865 2064 6961 6d6f 6e64 2064 6174   the diamond dat
-000046d0: 6162 6173 6520 666f 7220 616e 6e6f 7461  abase for annota
-000046e0: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
-000046f0: 6e65 0a2d 2d68 6d6d 2c20 7468 6520 484d  ne.--hmm, the HM
-00004700: 4d20 7072 6f66 696c 6520 666f 7220 616e  M profile for an
-00004710: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
-00004720: 7420 4e6f 6e65 0a2d 2d65 7661 6c75 652c  t None.--evalue,
-00004730: 2074 6865 2065 2d76 616c 7565 2063 7574   the e-value cut
-00004740: 2d6f 6666 2066 6f72 2061 6e6e 6f74 6174  -off for annotat
-00004750: 696f 6e2c 2064 6566 6175 6c74 2031 652d  ion, default 1e-
-00004760: 3130 0a2d 2d65 7865 7061 7468 2c20 7468  10.--exepath, th
-00004770: 6520 7061 7468 2074 6f20 7468 6520 696e  e path to the in
-00004780: 7465 7270 726f 7363 616e 2065 7865 6375  terproscan execu
-00004790: 7461 626c 652c 2064 6566 6175 6c74 204e  table, default N
-000047a0: 6f6e 650a 2d66 2c20 2d2d 666f 7373 696c  one.-f, --fossil
-000047b0: 2c20 7468 6520 666f 7373 696c 2063 616c  , the fossil cal
-000047c0: 6962 7261 7469 6f6e 2069 6e66 6f72 6d61  ibration informa
-000047d0: 7469 6f6e 2069 6e20 4265 6173 742c 2064  tion in Beast, d
-000047e0: 6566 6175 6c74 2028 2763 6c61 6465 313b  efault ('clade1;
-000047f0: 636c 6164 6532 272c 2027 7461 7861 312c  clade2', 'taxa1,
-00004800: 7461 7861 323b 7461 7861 332c 7461 7861  taxa2;taxa3,taxa
-00004810: 3427 2c20 2734 3b35 272c 2027 302e 353b  4', '4;5', '0.5;
-00004820: 302e 3627 2c20 2734 3030 3b35 3030 2729  0.6', '400;500')
-00004830: 0a2d 7268 2c20 2d2d 726f 6f74 6865 6967  .-rh, --rootheig
-00004840: 6874 2c20 7468 6520 726f 6f74 2068 6569  ht, the root hei
-00004850: 6768 7420 6361 6c69 6272 6174 696f 6e20  ght calibration 
-00004860: 696e 666f 2069 6e20 4265 6173 742c 2064  info in Beast, d
-00004870: 6566 6175 6c74 2028 342c 302e 352c 3430  efault (4,0.5,40
-00004880: 3029 0a2d 6373 2c20 2d2d 6368 6169 6e73  0).-cs, --chains
-00004890: 6574 2c20 7468 6520 7061 7261 6d65 7465  et, the paramete
-000048a0: 7273 206f 6620 4d43 4d43 2063 6861 696e  rs of MCMC chain
-000048b0: 2069 6e20 4265 6173 742c 2064 6566 6175   in Beast, defau
-000048c0: 6c74 2028 3130 3030 302c 3130 3029 0a2d  lt (10000,100).-
-000048d0: 2d62 6561 7374 6c67 6a61 722c 2074 6865  -beastlgjar, the
-000048e0: 2070 6174 6820 746f 2062 6561 7374 4c47   path to beastLG
-000048f0: 2e6a 6172 2c20 6465 6661 756c 7420 4e6f  .jar, default No
-00004900: 6e65 0a2d 2d62 6561 676c 652c 2066 6c61  ne.--beagle, fla
-00004910: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004920: 7220 746f 2075 7365 2062 6561 676c 6520  r to use beagle 
-00004930: 696e 2042 6561 7374 2c20 6966 2074 6865  in Beast, if the
-00004940: 2066 6c61 6720 7761 7320 7365 742c 2062   flag was set, b
-00004950: 6561 676c 6520 776f 756c 6420 6265 2075  eagle would be u
-00004960: 7365 640a 2d2d 7072 6f74 636f 6364 6174  sed.--protcocdat
-00004970: 696e 672c 2066 6c61 6720 6f70 7469 6f6e  ing, flag option
-00004980: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
-00004990: 7920 696e 6974 6961 7465 2074 6865 2070  y initiate the p
-000049a0: 726f 7465 696e 2d63 6f6e 6361 7465 6e61  rotein-concatena
-000049b0: 7469 6f6e 2d62 6173 6564 2064 6174 696e  tion-based datin
-000049c0: 6720 616e 616c 7973 6973 2c20 6966 2074  g analysis, if t
-000049d0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000049e0: 2074 6865 2061 6e61 6c79 7369 7320 776f   the analysis wo
-000049f0: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
-00004a00: 0a2d 2d70 726f 7464 6174 696e 672c 2066  .--protdating, f
-00004a10: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004a20: 6865 7220 746f 206f 6e6c 7920 696e 6974  her to only init
-00004a30: 6961 7465 2074 6865 2070 726f 7465 696e  iate the protein
-00004a40: 2d62 6173 6564 2064 6174 696e 6720 616e  -based dating an
-00004a50: 616c 7973 6973 2c20 6966 2074 6865 2066  alysis, if the f
-00004a60: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
-00004a70: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
-00004a80: 6265 2069 6e69 7469 6174 6564 0a60 6060  be initiated.```
-00004a90: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
-00004aa0: 6764 206b 7364 6020 6361 6e20 7265 616c  gd ksd` can real
-00004ab0: 697a 6520 7468 6520 636f 6e73 7472 7563  ize the construc
-00004ac0: 7469 6f6e 206f 6620 2a4b 2a3c 7375 623e  tion of *K*<sub>
-00004ad0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00004ae0: 7269 6275 7469 6f6e 2061 6e64 2072 6174  ribution and rat
-00004af0: 6520 636f 7272 6563 7469 6f6e 2e0a 6060  e correction..``
-00004b00: 600a 7767 6420 6b73 6420 6661 6d69 6c69  `.wgd ksd famili
-00004b10: 6573 2073 6571 7565 6e63 6573 2028 6f70  es sequences (op
-00004b20: 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d  tion).----------
-00004b30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004b40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004b50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004b60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004b70: 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574  ------.-o, --out
-00004b80: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
-00004b90: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
-00004ba0: 6c74 2077 6764 5f6b 7364 0a2d 742c 202d  lt wgd_ksd.-t, -
-00004bb0: 2d74 6d70 6469 722c 2074 6865 2074 656d  -tmpdir, the tem
-00004bc0: 706f 7261 7279 2077 6f72 6b69 6e67 2064  porary working d
-00004bd0: 6972 6563 746f 7279 2c20 6465 6661 756c  irectory, defaul
-00004be0: 7420 4e6f 6e65 2c20 6966 204e 6f6e 6520  t None, if None 
-00004bf0: 7761 7320 6769 7665 6e2c 2074 6865 2074  was given, the t
-00004c00: 6d70 6469 7220 7769 6c6c 2062 6520 6173  mpdir will be as
-00004c10: 7369 676e 6564 2072 616e 646f 6d20 6e61  signed random na
-00004c20: 6d65 7320 696e 2063 7572 7265 6e74 2064  mes in current d
-00004c30: 6972 6563 746f 7279 2061 6e64 2061 7574  irectory and aut
-00004c40: 6f6d 6174 656c 7920 7265 6d6f 7665 6420  omately removed 
-00004c50: 6174 2074 6865 2063 6f6d 706c 6574 696f  at the completio
-00004c60: 6e20 6f66 2070 726f 6772 616d 2c20 656c  n of program, el
-00004c70: 7365 2074 6865 2074 6d70 6469 7220 776f  se the tmpdir wo
-00004c80: 756c 6420 6265 206b 6570 740a 2d6e 2c20  uld be kept.-n, 
-00004c90: 2d2d 6e74 6872 6561 6473 2c20 7468 6520  --nthreads, the 
-00004ca0: 6e75 6d62 6572 206f 6620 7468 7265 6164  number of thread
-00004cb0: 7320 746f 2075 7365 2c20 6465 6661 756c  s to use, defaul
-00004cc0: 7420 340a 2d2d 746f 5f73 746f 702c 2066  t 4.--to_stop, f
-00004cd0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004ce0: 6865 7220 746f 2074 7261 6e73 6c61 7465  her to translate
-00004cf0: 2074 6872 6f75 6768 2053 544f 5020 636f   through STOP co
-00004d00: 646f 6e73 2c20 6966 2074 6865 2066 6c61  dons, if the fla
-00004d10: 6720 7761 7320 7365 742c 2074 7261 6e73  g was set, trans
-00004d20: 6c61 7469 6f6e 2077 696c 6c20 6265 2074  lation will be t
-00004d30: 6572 6d69 6e61 7465 6420 6174 2074 6865  erminated at the
-00004d40: 2066 6972 7374 2069 6e2d 6672 616d 6520   first in-frame 
-00004d50: 7374 6f70 2063 6f64 6f6e 2c20 656c 7365  stop codon, else
-00004d60: 2061 2066 756c 6c20 7472 616e 736c 6174   a full translat
-00004d70: 696f 6e20 636f 6e74 696e 7569 6e67 206f  ion continuing o
-00004d80: 6e20 7061 7374 2061 6e79 2073 746f 7020  n past any stop 
-00004d90: 636f 646f 6e73 2077 6f75 6c64 2062 6520  codons would be 
-00004da0: 696e 6974 6961 7465 640a 2d2d 6364 732c  initiated.--cds,
-00004db0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004dc0: 6574 6865 7220 746f 206f 6e6c 7920 7472  ether to only tr
-00004dd0: 616e 736c 6174 6520 7468 6520 636f 6d70  anslate the comp
-00004de0: 6c65 7465 2043 4453 2074 6861 7420 7374  lete CDS that st
-00004df0: 6172 7473 2077 6974 6820 6120 7661 6c69  arts with a vali
-00004e00: 6420 7374 6172 7420 636f 646f 6e20 616e  d start codon an
-00004e10: 6420 6f6e 6c79 2063 6f6e 7461 696e 7320  d only contains 
-00004e20: 6120 7369 6e67 6c65 2069 6e2d 6672 616d  a single in-fram
-00004e30: 6520 7374 6f70 2063 6f64 6f6e 2061 7420  e stop codon at 
-00004e40: 7468 6520 656e 6420 616e 6420 6d75 7374  the end and must
-00004e50: 2062 6520 6469 7669 6461 626c 6520 6279   be dividable by
-00004e60: 2074 6872 6565 2c20 6966 2074 6865 2066   three, if the f
-00004e70: 6c61 6720 7761 7320 7365 742c 206f 6e6c  lag was set, onl
-00004e80: 7920 7468 6520 636f 6d70 6c65 7465 2043  y the complete C
-00004e90: 4453 2077 6f75 6c64 2062 6520 7472 616e  DS would be tran
-00004ea0: 736c 6174 6564 0a2d 2d70 6169 7277 6973  slated.--pairwis
-00004eb0: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
-00004ec0: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00004ed0: 6174 6520 7061 6972 7769 7365 204b 7320  ate pairwise Ks 
-00004ee0: 6573 7469 6d61 7469 6f6e 2c20 6966 2074  estimation, if t
-00004ef0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00004f00: 2070 6169 7277 6973 6520 4b73 2076 616c   pairwise Ks val
-00004f10: 7565 7320 776f 756c 6420 6265 2065 7374  ues would be est
-00004f20: 696d 6174 6564 0a2d 2d73 7472 6970 5f67  imated.--strip_g
-00004f30: 6170 732c 2066 6c61 6720 6f70 7469 6f6e  aps, flag option
-00004f40: 2c20 7768 6574 6865 7220 746f 2064 726f  , whether to dro
-00004f50: 7020 616c 6c20 6761 7073 2069 6e20 6d75  p all gaps in mu
-00004f60: 6c74 6970 6c65 2073 6571 7565 6e63 6520  ltiple sequence 
-00004f70: 616c 6967 6e6d 656e 742c 2069 6620 7468  alignment, if th
-00004f80: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00004f90: 616c 6c20 6761 7073 2077 6f75 6c64 2062  all gaps would b
-00004fa0: 6520 6472 6f70 7065 640a 2d61 2c20 2d2d  e dropped.-a, --
-00004fb0: 616c 6967 6e65 722c 2077 6869 6368 2061  aligner, which a
-00004fc0: 6c69 676e 6d65 6e74 2070 726f 6772 616d  lignment program
-00004fd0: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
-00004fe0: 206d 6166 6674 200a 2d74 7265 652c 202d   mafft .-tree, -
-00004ff0: 2d74 7265 655f 6d65 7468 6f64 2c20 7768  -tree_method, wh
-00005000: 6963 6820 6765 6e65 2074 7265 6520 696e  ich gene tree in
-00005010: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
-00005020: 746f 2069 6e76 6f6b 652c 2064 6566 6175  to invoke, defau
-00005030: 6c74 2066 6173 7474 7265 650a 2d2d 7472  lt fasttree.--tr
-00005040: 6565 5f6f 7074 696f 6e73 2c20 6f70 7469  ee_options, opti
-00005050: 6f6e 7320 696e 2074 7265 6520 696e 6665  ons in tree infe
-00005060: 7265 6e63 6520 6173 2061 2063 6f6d 6d61  rence as a comma
-00005070: 2073 6570 6172 6174 6564 2073 7472 696e   separated strin
-00005080: 672c 2064 6566 6175 6c74 204e 6f6e 650a  g, default None.
-00005090: 2d2d 6e6f 6465 5f61 7665 7261 6765 2c20  --node_average, 
-000050a0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-000050b0: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
-000050c0: 206e 6f64 652d 6176 6572 6167 6520 7761   node-average wa
-000050d0: 7920 6f66 2064 652d 7265 6475 6e64 616e  y of de-redundan
-000050e0: 6379 2069 6e73 7465 6164 206f 6620 6e6f  cy instead of no
-000050f0: 6465 2d77 6569 6768 7465 642c 2069 6620  de-weighted, if 
-00005100: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00005110: 2c20 7468 6520 6e6f 6465 2d61 7665 7261  , the node-avera
-00005120: 6769 6e67 2064 652d 7265 6475 6e64 616e  ging de-redundan
-00005130: 6379 2077 6f75 6c64 2062 6520 696e 6974  cy would be init
-00005140: 6961 7465 640a 2d73 722c 202d 2d73 7061  iated.-sr, --spa
-00005150: 6972 2c20 7468 6520 7370 6563 6965 7320  ir, the species 
-00005160: 7061 6972 2074 6f20 6265 2070 6c6f 7474  pair to be plott
-00005170: 6564 2c20 6465 6661 756c 7420 4e6f 6e65  ed, default None
-00005180: 2c20 7468 6973 206f 7074 696f 6e20 6361  , this option ca
-00005190: 6e20 6265 2070 726f 7669 6465 6420 6d75  n be provided mu
-000051a0: 6c74 6970 6c65 2074 696d 6573 0a2d 7370  ltiple times.-sp
-000051b0: 2c20 2d2d 7370 6563 6965 7374 7265 652c  , --speciestree,
-000051c0: 2074 6865 2073 7065 6369 6573 2074 7265   the species tre
-000051d0: 6520 746f 2070 6572 666f 726d 2072 6174  e to perform rat
-000051e0: 6520 636f 7272 6563 7469 6f6e 2c20 6465  e correction, de
-000051f0: 6661 756c 7420 4e6f 6e65 2c20 6966 204e  fault None, if N
-00005200: 6f6e 6520 7761 7320 6769 7665 6e2c 2074  one was given, t
-00005210: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
-00005220: 6f6e 2061 6e61 6c79 7369 7320 776f 756c  on analysis woul
-00005230: 6420 6265 2063 616c 6c65 6420 6f66 660a  d be called off.
-00005240: 2d72 772c 202d 2d72 6577 6569 6768 742c  -rw, --reweight,
-00005250: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005260: 6574 6865 7220 746f 2072 6563 616c 6375  ether to recalcu
-00005270: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
-00005280: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
-00005290: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-000052a0: 7320 7365 742c 2074 6865 2077 6569 6768  s set, the weigh
-000052b0: 7420 776f 756c 6420 6265 2072 6563 616c  t would be recal
-000052c0: 6375 6c61 7465 640a 2d6f 722c 202d 2d6f  culated.-or, --o
-000052d0: 6e6c 7972 6f6f 746f 7574 2c20 666c 6167  nlyrootout, flag
-000052e0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000052f0: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
+00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
+00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
+00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 330a  Version: 2.0.33.
+00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
+00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
+00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
+00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
+00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
+00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
+00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
+000000a0: 650a 4c69 6365 6e73 653a 2047 504c 0a44  e.License: GPL.D
+000000b0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
+000000c0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
+000000d0: 726b 646f 776e 0a4c 6963 656e 7365 2d46  rkdown.License-F
+000000e0: 696c 653a 204c 4943 454e 5345 0a0a 3c64  ile: LICENSE..<d
+000000f0: 6976 2061 6c69 676e 3d22 6365 6e74 6572  iv align="center
+00000100: 223e 0a0a 2320 6077 6764 2076 3260 203a  ">..# `wgd v2` :
+00000110: 2061 2073 7569 7465 2074 6f6f 6c20 6f66   a suite tool of
+00000120: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
+00000130: 6e64 2074 696d 696e 670a 0a5b 215b 4275  nd timing..[![Bu
+00000140: 696c 6420 5374 6174 7573 5d28 6874 7470  ild Status](http
+00000150: 733a 2f2f 6170 702e 7472 6176 6973 2d63  s://app.travis-c
+00000160: 692e 636f 6d2f 6865 6368 652d 7073 622f  i.com/heche-psb/
+00000170: 7767 642e 7376 673f 6272 616e 6368 3d70  wgd.svg?branch=p
+00000180: 6879 6c6f 6461 7469 6e67 295d 2868 7474  hylodating)](htt
+00000190: 7073 3a2f 2f61 7070 2e74 7261 7669 732d  ps://app.travis-
+000001a0: 6369 2e63 6f6d 2f68 6563 6865 2d70 7362  ci.com/heche-psb
+000001b0: 2f77 6764 290a 5b21 5b44 6f63 756d 656e  /wgd).[![Documen
+000001c0: 7461 7469 6f6e 2053 7461 7475 735d 2868  tation Status](h
+000001d0: 7474 7073 3a2f 2f72 6561 6474 6865 646f  ttps://readthedo
+000001e0: 6373 2e6f 7267 2f70 726f 6a65 6374 732f  cs.org/projects/
+000001f0: 7767 6476 322f 6261 6467 652f 3f76 6572  wgdv2/badge/?ver
+00000200: 7369 6f6e 3d6c 6174 6573 7429 5d28 6874  sion=latest)](ht
+00000210: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
+00000220: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
+00000230: 7465 7374 2f3f 6261 6467 653d 6c61 7465  test/?badge=late
+00000240: 7374 290a 5b21 5b6c 6963 656e 7365 5d28  st).[![license](
+00000250: 6874 7470 733a 2f2f 696d 672e 7368 6965  https://img.shie
+00000260: 6c64 732e 696f 2f70 7970 692f 6c2f 7767  lds.io/pypi/l/wg
+00000270: 642e 7376 6729 5d28 6874 7470 733a 2f2f  d.svg)](https://
+00000280: 7079 7069 2e70 7974 686f 6e2e 6f72 672f  pypi.python.org/
+00000290: 7079 7069 2f77 6764 290a 5b21 5b4c 6174  pypi/wgd).[![Lat
+000002a0: 6573 7420 5079 5049 2076 6572 7369 6f6e  est PyPI version
+000002b0: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+000002c0: 6965 6c64 732e 696f 2f70 7970 692f 762f  ields.io/pypi/v/
+000002d0: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
+000002e0: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
+000002f0: 672f 7079 7069 2f77 6764 290a 5b21 5b69  g/pypi/wgd).[![i
+00000300: 6e73 7461 6c6c 2077 6974 6820 6269 6f63  nstall with bioc
+00000310: 6f6e 6461 5d28 6874 7470 733a 2f2f 696d  onda](https://im
+00000320: 672e 7368 6965 6c64 732e 696f 2f62 6164  g.shields.io/bad
+00000330: 6765 2f69 6e73 7461 6c6c 2532 3077 6974  ge/install%20wit
+00000340: 682d 6269 6f63 6f6e 6461 2d62 7269 6768  h-bioconda-brigh
+00000350: 7467 7265 656e 2e73 7667 3f73 7479 6c65  tgreen.svg?style
+00000360: 3d66 6c61 7429 5d28 6874 7470 3a2f 2f62  =flat)](http://b
+00000370: 696f 636f 6e64 612e 6769 7468 7562 2e69  ioconda.github.i
+00000380: 6f2f 7265 6369 7065 732f 7767 642f 5245  o/recipes/wgd/RE
+00000390: 4144 4d45 2e68 746d 6c29 0a5b 215b 446f  ADME.html).[![Do
+000003a0: 776e 6c6f 6164 735d 2868 7474 7073 3a2f  wnloads](https:/
+000003b0: 2f70 6570 792e 7465 6368 2f62 6164 6765  /pepy.tech/badge
+000003c0: 2f77 6764 295d 2868 7474 7073 3a2f 2f70  /wgd)](https://p
+000003d0: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
+000003e0: 2f77 6764 290a 0a2a 2a48 656e 6763 6869  /wgd)..**Hengchi
+000003f0: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
+00000400: 6165 6e65 706f 656c 2c20 5976 6573 2056  aenepoel, Yves V
+00000410: 616e 2064 6520 5065 6572 2a2a 0a0a 5b2a  an de Peer**..[*
+00000420: 2a42 696f 696e 666f 726d 6174 6963 7320  *Bioinformatics 
+00000430: 2620 4576 6f6c 7574 696f 6e61 7279 2047  & Evolutionary G
+00000440: 656e 6f6d 6963 7320 4772 6f75 702a 2a5d  enomics Group**]
+00000450: 2868 7474 7073 3a2f 2f77 7777 2e76 616e  (https://www.van
+00000460: 6465 7065 6572 6c61 622e 6f72 672f 292a  depeerlab.org/)*
+00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
+00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
+00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
+000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
+000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
+000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
+000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
+000004e0: 6c61 7469 6f6e 2920 7c20 0a5b 2a2a 5061  lation) | .[**Pa
+000004f0: 7261 6d65 7465 7273 2a2a 5d28 2370 6172  rameters**](#par
+00000500: 616d 6574 6572 7329 207c 200a 5b2a 2a55  ameters) | .[**U
+00000510: 7361 6765 2a2a 5d28 2375 7361 6765 2920  sage**](#usage) 
+00000520: 7c20 0a5b 2a2a 496c 6c75 7374 7261 7469  | .[**Illustrati
+00000530: 6f6e 2a2a 5d28 2369 6c6c 7573 7472 6174  on**](#illustrat
+00000540: 696f 6e29 207c 0a5b 2a2a 446f 6375 6d65  ion) |.[**Docume
+00000550: 6e74 6174 696f 6e2a 2a5d 2868 7474 7073  ntation**](https
+00000560: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
+00000570: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
+00000580: 742f 3f62 6164 6765 3d6c 6174 6573 7429  t/?badge=latest)
+00000590: 207c 0a5b 2a2a 4369 7461 7469 6f6e 2a2a   |.[**Citation**
+000005a0: 5d28 2363 6974 6174 696f 6e29 0a0a 3c2f  ](#citation)..</
+000005b0: 6469 763e 0a0a 6077 6764 2076 3260 2069  div>..`wgd v2` i
+000005c0: 7320 6120 7079 7468 6f6e 2070 6163 6b61  s a python packa
+000005d0: 6765 2075 7067 7261 6465 6420 6672 6f6d  ge upgraded from
+000005e0: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
+000005f0: 6764 6020 7061 636b 6167 6520 6169 6d69  gd` package aimi
+00000600: 6e67 2066 6f72 2074 6865 2069 6e66 6572  ng for the infer
+00000610: 656e 6365 2061 6e64 2074 696d 696e 6720  ence and timing 
+00000620: 6f66 2061 6e63 6965 6e74 2077 686f 6c65  of ancient whole
+00000630: 2d67 656e 6f6d 6520 6475 706c 6963 6174  -genome duplicat
+00000640: 696f 6e20 2857 4744 2920 6576 656e 7473  ion (WGD) events
+00000650: 2e20 466f 7220 7468 6520 7072 6f70 6f73  . For the propos
+00000660: 6520 6f66 2069 6c6c 7573 7472 6174 696e  e of illustratin
+00000670: 6720 7468 6520 7072 696e 6369 706c 6520  g the principle 
+00000680: 616e 6420 7573 6167 6520 6f66 2060 7767  and usage of `wg
+00000690: 6420 7632 602c 2077 6520 636f 6d70 696c  d v2`, we compil
+000006a0: 6564 2074 6869 7320 646f 6375 6d65 6e74  ed this document
+000006b0: 6174 696f 6e2e 2042 656c 6f77 2077 6520  ation. Below we 
+000006c0: 6669 7273 7420 6761 7665 2061 6e20 696e  first gave an in
+000006d0: 7472 6f64 7563 7469 6f6e 206f 7665 7220  troduction over 
+000006e0: 7468 6520 7363 6f70 6520 616e 6420 6d65  the scope and me
+000006f0: 6368 616e 6973 6d20 6f66 2060 7767 6420  chanism of `wgd 
+00000700: 7632 6020 616e 6420 7468 656e 2074 6865  v2` and then the
+00000710: 2070 7261 6374 6963 616c 2069 6e66 6f72   practical infor
+00000720: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
+00000730: 6720 746f 2074 6865 2069 6e73 7461 6c6c  g to the install
+00000740: 6174 696f 6e20 616e 6420 7573 6167 652e  ation and usage.
+00000750: 2041 6e20 6578 616d 706c 6172 2077 6f72   An examplar wor
+00000760: 6b66 6c6f 7720 6973 2070 726f 7669 6465  kflow is provide
+00000770: 6420 696e 2074 6865 2074 7574 6f72 6961  d in the tutoria
+00000780: 6c20 7365 6374 696f 6e20 6f6e 2068 6f77  l section on how
+00000790: 2074 6f20 7365 656b 2065 7669 6465 6e63   to seek evidenc
+000007a0: 6520 666f 7220 6120 7075 7461 7469 7665  e for a putative
+000007b0: 2057 4744 2065 7665 6e74 2061 6e64 2070   WGD event and p
+000007c0: 6572 666f 726d 2070 726f 7065 7220 7469  erform proper ti
+000007d0: 6d69 6e67 2077 6974 6820 6120 6672 6573  ming with a fres
+000007e0: 686c 7920 6f62 7461 696e 6564 2067 656e  hly obtained gen
+000007f0: 6f6d 6520 6173 7365 6d62 6c79 2069 6e20  ome assembly in 
+00000800: 6861 6e64 2e20 466f 7220 7468 6f73 6520  hand. For those 
+00000810: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
+00000820: 6564 2069 6e20 6d6f 7265 2074 6865 6f72  ed in more theor
+00000830: 6574 6963 616c 2064 6574 6169 6c73 2c20  etical details, 
+00000840: 7765 2072 6563 6f6d 6d65 6e64 2074 7572  we recommend tur
+00000850: 6e69 6e67 2074 6f20 6f75 7220 7061 7065  ning to our pape
+00000860: 7220 616e 6420 626f 6f6b 2063 6861 7074  r and book chapt
+00000870: 6572 2066 6f72 206d 6f72 6520 6465 7461  er for more deta
+00000880: 696c 6564 2064 6573 6372 6970 7469 6f6e  iled description
+00000890: 2061 6e64 2069 6e73 6967 6874 6675 6c20   and insightful 
+000008a0: 6469 7363 7573 7369 6f6e 732e 2054 6865  discussions. The
+000008b0: 206b 6579 2069 6d70 726f 7665 6420 6665   key improved fe
+000008c0: 6174 7572 6573 206f 6620 6077 6764 2076  atures of `wgd v
+000008d0: 3260 2061 7265 2064 656d 6f6e 7374 7261  2` are demonstra
+000008e0: 7465 6420 696e 206f 7572 206c 6174 6573  ted in our lates
+000008f0: 7420 6d61 6e75 7363 7269 7074 206f 6e20  t manuscript on 
+00000900: 5b42 696f 696e 666f 726d 6174 6963 735d  [Bioinformatics]
+00000910: 2868 7474 7073 3a2f 2f64 6f69 2e6f 7267  (https://doi.org
+00000920: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
+00000930: 726d 6174 6963 732f 6274 6165 3237 3229  rmatics/btae272)
+00000940: 2e20 4966 2079 6f75 2075 7365 2060 7767  . If you use `wg
+00000950: 6420 7632 6020 696e 2079 6f75 7220 7265  d v2` in your re
+00000960: 7365 6172 6368 2c20 706c 6561 7365 2063  search, please c
+00000970: 6974 6520 7573 2061 7320 7375 6767 6573  ite us as sugges
+00000980: 7465 6420 696e 205b 4369 7461 7469 6f6e  ted in [Citation
+00000990: 5d28 2363 6974 6174 696f 6e29 2073 6563  ](#citation) sec
+000009a0: 7469 6f6e 2e0a 0a23 2320 496e 7472 6f64  tion...## Introd
+000009b0: 7563 7469 6f6e 0a0a 506f 6c79 706c 6f69  uction..Polyploi
+000009c0: 6469 7a61 7469 6f6e 732c 2074 6865 2065  dizations, the e
+000009d0: 766f 6c75 7469 6f6e 6172 7920 7072 6f63  volutionary proc
+000009e0: 6573 7320 7468 6174 2074 6865 2065 6e74  ess that the ent
+000009f0: 6972 6520 6765 6e6f 6d65 206f 6620 616e  ire genome of an
+00000a00: 206f 7267 616e 6973 6d20 6973 2064 7570   organism is dup
+00000a10: 6c69 6361 7465 642c 2061 6c73 6f20 6e61  licated, also na
+00000a20: 6d65 6420 6173 2077 686f 6c65 2d67 656e  med as whole-gen
+00000a30: 6f6d 6520 6475 706c 6963 6174 696f 6e73  ome duplications
+00000a40: 2028 5747 4473 292c 206f 6363 7572 2072   (WGDs), occur r
+00000a50: 6563 7572 7265 6e74 6c79 2061 6372 6f73  ecurrently acros
+00000a60: 7320 7468 6520 7472 6565 206f 6620 6c69  s the tree of li
+00000a70: 6665 2e20 5468 6572 6520 6172 6520 7477  fe. There are tw
+00000a80: 6f20 6d6f 6465 7320 6f66 2070 6f6c 7970  o modes of polyp
+00000a90: 6c6f 6964 697a 6174 696f 6e73 2c20 6175  loidizations, au
+00000aa0: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
+00000ab0: 6f6e 7320 616e 6420 616c 6c6f 706f 6c79  ons and allopoly
+00000ac0: 706c 6f69 6469 7a61 7469 6f6e 732e 2041  ploidizations. A
+00000ad0: 7574 6f70 6f6c 7970 6c6f 6964 697a 6174  utopolyploidizat
+00000ae0: 696f 6e73 2061 7265 2074 6865 2064 7570  ions are the dup
+00000af0: 6c69 6361 7469 6f6e 206f 6620 7468 6520  lication of the 
+00000b00: 7361 6d65 2067 656e 6f6d 652c 2072 6573  same genome, res
+00000b10: 756c 7469 6e67 2069 6e20 7477 6f20 6964  ulting in two id
+00000b20: 656e 7469 6361 6c20 7375 6267 656e 6f6d  entical subgenom
+00000b30: 6573 2061 7420 7468 6520 7469 6d65 2069  es at the time i
+00000b40: 7420 656d 6572 6765 642e 2057 6869 6c65  t emerged. While
+00000b50: 2074 6865 2061 6c6c 6f70 6f6c 7970 6c6f   the allopolyplo
+00000b60: 6964 697a 6174 696f 6e73 2061 7265 206e  idizations are n
+00000b70: 6f72 6d61 6c6c 7920 6163 6869 6576 6564  ormally achieved
+00000b80: 2069 6e20 7477 6f20 7374 6570 732c 2066   in two steps, f
+00000b90: 6972 7374 2074 6865 2068 7962 7269 6469  irst the hybridi
+00000ba0: 7a61 7469 6f6e 2062 6574 7765 656e 2074  zation between t
+00000bb0: 776f 2064 6966 6665 7265 6e74 2073 7065  wo different spe
+00000bc0: 6369 6573 2c20 7265 7375 6c74 696e 6720  cies, resulting 
+00000bd0: 696e 2074 6865 2061 7269 7369 6e67 206f  in the arising o
+00000be0: 6620 7472 616e 7369 656e 7420 686f 6d6f  f transient homo
+00000bf0: 706c 6f69 6479 2c73 6563 6f6e 6420 7468  ploidy,second th
+00000c00: 6520 6475 706c 6963 6174 696f 6e20 6f66  e duplication of
+00000c10: 2074 6865 2068 6f6d 6f70 6c6f 6964 792c   the homoploidy,
+00000c20: 2072 6573 756c 7469 6e67 2069 6e20 7468   resulting in th
+00000c30: 6520 656d 6572 6765 6e63 6520 6f66 2061  e emergence of a
+00000c40: 6c6c 6f70 6f6c 7970 6c6f 6964 792e 2044  llopolyploidy. D
+00000c50: 7565 2074 6f20 7468 6520 756e 7374 6162  ue to the unstab
+00000c60: 696c 6974 7920 616e 6420 756e 6261 6c61  ility and unbala
+00000c70: 6e63 6564 2074 6574 7261 736f 6d69 6320  nced tetrasomic 
+00000c80: 696e 6865 7269 7461 6e63 652c 2066 6f72  inheritance, for
+00000c90: 2069 6e73 7461 6e63 6520 7468 6520 6e75   instance the nu
+00000ca0: 636c 6561 722d 6379 746f 706c 6173 6d69  clear-cytoplasmi
+00000cb0: 6320 696e 636f 6d70 6174 6962 696c 6974  c incompatibilit
+00000cc0: 792c 2074 6865 2070 6f6c 7970 6c6f 6964  y, the polyploid
+00000cd0: 7920 6765 6e6f 6d65 2077 696c 6c20 7468  y genome will th
+00000ce0: 656e 2065 7870 6572 6965 6e63 6520 6120  en experience a 
+00000cf0: 7072 6f63 6573 7320 6361 6c6c 6564 2064  process called d
+00000d00: 6970 6c6f 6964 697a 6174 696f 6e2c 2061  iploidization, a
+00000d10: 6c73 6f20 6e61 6d65 6420 6173 2066 7261  lso named as fra
+00000d20: 6374 696f 6e61 7469 6f6e 2c20 6475 7269  ctionation, duri
+00000d30: 6e67 2077 6869 6368 2061 206c 6172 6765  ng which a large
+00000d40: 2070 6f72 7469 6f6e 206f 6620 6765 6e65   portion of gene
+00000d50: 2064 7570 6c69 6361 7465 7320 7769 6c6c   duplicates will
+00000d60: 2067 6574 206c 6f73 7420 616e 6420 6f6e   get lost and on
+00000d70: 6c79 2061 2066 7261 6374 696f 6e20 6361  ly a fraction ca
+00000d80: 6e20 6265 2072 6574 6169 6e65 642e 2054  n be retained. T
+00000d90: 6865 2074 7261 6365 7320 6f66 2070 6f6c  he traces of pol
+00000da0: 7970 6c6f 6964 697a 6174 696f 6e73 2063  yploidizations c
+00000db0: 616e 2062 6520 7468 7573 2075 6e65 6172  an be thus unear
+00000dc0: 7468 6564 2066 726f 6d20 7468 6573 6520  thed from these 
+00000dd0: 7265 7461 696e 6564 2067 656e 6520 6475  retained gene du
+00000de0: 706c 6963 6174 6573 2e20 5468 7265 6520  plicates. Three 
+00000df0: 6170 7072 6f61 6368 6573 2062 6173 6564  approaches based
+00000e00: 206f 6e20 6765 6e65 2064 7570 6c69 6361   on gene duplica
+00000e10: 7465 732c 206e 616d 656c 792c 202a 4b2a  tes, namely, *K*
+00000e20: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
+00000e30: 686f 642c 2067 656e 6520 7472 6565 202d  hod, gene tree -
+00000e40: 2073 7065 6369 6573 2074 7265 6520 7265   species tree re
+00000e50: 636f 6e63 696c 6961 7469 6f6e 206d 6574  conciliation met
+00000e60: 686f 6420 616e 6420 7379 6e74 656e 7920  hod and synteny 
+00000e70: 6d65 7468 6f64 2c20 6172 6520 636f 6d6d  method, are comm
+00000e80: 6f6e 6c79 2075 7365 6420 696e 2064 6574  only used in det
+00000e90: 6563 7469 6e67 2065 7669 6465 6e63 6520  ecting evidence 
+00000ea0: 666f 7220 5747 4473 2e20 5468 6520 6765  for WGDs. The ge
+00000eb0: 6e65 2074 7265 6520 2d20 7370 6563 6965  ne tree - specie
+00000ec0: 7320 7472 6565 2072 6563 6f6e 6369 6c69  s tree reconcili
+00000ed0: 6174 696f 6e20 6d65 7468 6f64 2069 7320  ation method is 
+00000ee0: 6e6f 7420 7769 7468 696e 2074 6865 2073  not within the s
+00000ef0: 636f 7065 206f 6620 6077 6764 2076 3260  cope of `wgd v2`
+00000f00: 2c20 6275 7420 7765 206b 696e 646c 7920  , but we kindly 
+00000f10: 7265 6665 7220 7265 6164 6572 7320 7768  refer readers wh
+00000f20: 6f20 6172 6520 696e 7465 7265 7374 6564  o are interested
+00000f30: 2074 6f20 7468 6520 7068 796c 6f67 656e   to the phylogen
+00000f40: 6f6d 6963 2070 726f 6772 616d 2064 6576  omic program dev
+00000f50: 656c 6f70 6564 2062 7920 4172 7468 7572  eloped by Arthur
+00000f60: 205a 7761 656e 6570 6f65 6c20 6e61 6d65   Zwaenepoel name
+00000f70: 6420 5b57 4841 4c45 5d28 6874 7470 733a  d [WHALE](https:
+00000f80: 2f2f 6769 7468 7562 2e63 6f6d 2f61 727a  //github.com/arz
+00000f90: 7761 2f57 6861 6c65 2e6a 6c29 2061 6e64  wa/Whale.jl) and
+00000fa0: 2074 6865 2061 7373 6f63 6961 7465 6420   the associated 
+00000fb0: 5b70 6170 6572 5d28 6874 7470 733a 2f2f  [paper](https://
+00000fc0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
+00000fd0: 6d6f 6c62 6576 2f6d 737a 3038 3829 2066  molbev/msz088) f
+00000fe0: 6f72 206d 6f72 6520 7465 6368 6e69 6361  or more technica
+00000ff0: 6c20 616e 6420 7468 656f 7265 7469 6361  l and theoretica
+00001000: 6c20 6465 7461 696c 732e 0a0a 5468 6520  l details...The 
+00001010: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00001020: 6d65 7468 6f64 2069 7320 6573 7461 626c  method is establ
+00001030: 6973 6865 6420 6f6e 2061 206d 6f64 656c  ished on a model
+00001040: 206f 6620 6765 6e65 2066 616d 696c 7920   of gene family 
+00001050: 6576 6f6c 7574 696f 6e20 7468 6174 2065  evolution that e
+00001060: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
+00001070: 6973 2061 6c6c 6f77 6564 2074 6f20 6576  is allowed to ev
+00001080: 6f6c 7665 2076 6961 2067 656e 6520 6475  olve via gene du
+00001090: 706c 6963 6174 696f 6e20 616e 6420 6c6f  plication and lo
+000010a0: 7373 2e20 4e6f 7465 2074 6861 7420 7468  ss. Note that th
+000010b0: 6520 6765 6e65 2066 616d 696c 7920 6865  e gene family he
+000010c0: 7265 2069 7320 6173 7375 6d65 6420 746f  re is assumed to
+000010d0: 2062 6520 7468 6520 636c 7573 7465 7220   be the cluster 
+000010e0: 6f66 2061 6c6c 2067 656e 6573 2064 6573  of all genes des
+000010f0: 6365 6e64 6564 2066 726f 6d20 616e 2061  cended from an a
+00001100: 6e63 6573 7472 616c 2067 656e 6520 696e  ncestral gene in
+00001110: 2061 2073 696e 676c 6520 6765 6e6f 6d65   a single genome
+00001120: 2e20 5265 636f 7665 7269 6e67 2074 6865  . Recovering the
+00001130: 2067 656e 6520 7472 6565 206f 6620 7375   gene tree of su
+00001140: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
+00001150: 6e66 6f72 6d73 2074 6865 2074 696d 696e  nforms the timin
+00001160: 672c 2073 6369 6c69 6365 7420 7468 6520  g, scilicet the 
+00001170: 6167 652c 206f 6620 6765 6e65 2064 7570  age, of gene dup
+00001180: 6c69 6361 7469 6f6e 2065 7665 6e74 732e  lication events.
+00001190: 2054 6865 2061 6765 2072 6566 6572 6564   The age refered
+000011a0: 2068 6572 652c 2069 7320 6e6f 7420 696e   here, is not in
+000011b0: 2072 6561 6c20 6765 6f6c 6f67 6963 616c   real geological
+000011c0: 2074 696d 652c 2062 7574 2069 6e20 7468   time, but in th
+000011d0: 6520 756e 6974 206f 6620 6576 6f6c 7574  e unit of evolut
+000011e0: 696f 6e61 7279 2064 6973 7461 6e63 652c  ionary distance,
+000011f0: 2069 2e65 2e2c 2074 6865 206e 756d 6265   i.e., the numbe
+00001200: 7220 6f66 2073 7562 7374 6974 7574 696f  r of substitutio
+00001210: 6e73 2070 6572 2073 6974 652e 2057 6865  ns per site. Whe
+00001220: 6e20 7468 6520 6576 6f6c 7574 696f 6e61  n the evolutiona
+00001230: 7279 2072 6174 6520 7265 6d61 696e 7320  ry rate remains 
+00001240: 6170 7072 6f78 696d 6174 656c 7920 636f  approximately co
+00001250: 6e73 7461 6e74 2c20 7468 6520 6576 6f6c  nstant, the evol
+00001260: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
+00001270: 6520 6973 2074 6865 6e20 7375 7070 6f73  e is then suppos
+00001280: 6564 2074 6f20 6265 2070 726f 706f 7274  ed to be proport
+00001290: 696f 6e61 6c20 746f 2074 6865 2072 6561  ional to the rea
+000012a0: 6c20 6576 6f6c 7574 696f 6e61 7279 2074  l evolutionary t
+000012b0: 696d 652e 2054 6865 2073 796e 6f6e 796d  ime. The synonym
+000012c0: 6f75 7320 6469 7374 616e 6365 202a 4b2a  ous distance *K*
+000012d0: 3c73 7562 3e53 3c2f 7375 623e 2c20 7468  <sub>S</sub>, th
+000012e0: 6520 6e75 6d62 6572 206f 6620 7379 6e6f  e number of syno
+000012f0: 6e79 6d6f 7573 2073 7562 7374 6974 7574  nymous substitut
+00001300: 696f 6e73 2070 6572 2073 796e 6f6e 796d  ions per synonym
+00001310: 6f75 7320 7369 7465 2c20 6973 2073 7563  ous site, is suc
+00001320: 6820 6361 6e64 6964 6174 6520 7468 6174  h candidate that
+00001330: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
+00001340: 7469 7475 7469 6f6e 7320 776f 756c 6420  titutions would 
+00001350: 6e6f 7420 696e 6375 7220 7468 6520 6368  not incur the ch
+00001360: 616e 6765 206f 6620 616d 696e 6f20 6163  ange of amino ac
+00001370: 6964 2061 6e64 2061 7265 2074 6875 7320  id and are thus 
+00001380: 7265 6761 7264 6564 2061 7320 6e65 7574  regarded as neut
+00001390: 7261 6c2c 2077 6869 6368 2061 6363 6f72  ral, which accor
+000013a0: 6469 6e67 2074 6f20 7468 6520 6e65 7574  ding to the neut
+000013b0: 7261 6c20 7468 656f 7279 2073 686f 756c  ral theory shoul
+000013c0: 6420 6f63 6375 7220 696e 2063 6f6e 7374  d occur in const
+000013d0: 616e 7420 7261 7465 2e20 4769 7665 6e20  ant rate. Given 
+000013e0: 6120 6d6f 6465 6c20 6f66 2067 656e 6520  a model of gene 
+000013f0: 6661 6d69 6c79 2074 6861 7420 616c 6c6f  family that allo
+00001400: 7773 2074 6865 2067 656e 6520 746f 2064  ws the gene to d
+00001410: 7570 6c69 6361 7465 2061 6e64 2067 6574  uplicate and get
+00001420: 206c 6f73 7420 696e 2061 2066 6978 6564   lost in a fixed
+00001430: 2072 6174 652c 206f 6e65 2063 616e 2064   rate, one can d
+00001440: 6572 6976 6520 7468 6174 2074 6865 2070  erive that the p
+00001450: 726f 6261 6269 6c69 7479 2064 656e 7369  robability densi
+00001460: 7479 2066 756e 6374 696f 6e20 6f66 2074  ty function of t
+00001470: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+00001480: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+00001490: 696f 6e20 6f66 2072 6574 6169 6e65 6420  ion of retained 
+000014a0: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
+000014b0: 6973 2061 2071 7561 7369 2d65 7870 6f6e  is a quasi-expon
+000014c0: 656e 7469 616c 2066 756e 6374 696f 6e20  ential function 
+000014d0: 7468 6174 206d 6f73 7420 7265 7461 696e  that most retain
+000014e0: 6564 2067 656e 6520 6475 706c 6963 6174  ed gene duplicat
+000014f0: 6573 2061 7265 2072 6563 656e 746c 7920  es are recently 
+00001500: 626f 726e 6564 2077 6974 6820 7e30 2061  borned with ~0 a
+00001510: 6765 2077 6869 6c65 2061 7320 7468 6520  ge while as the 
+00001520: 6167 6520 676f 696e 6720 6f6c 6465 7220  age going older 
+00001530: 7468 6520 6173 736f 6369 6174 6564 206e  the associated n
+00001540: 756d 6265 7220 6f66 2072 6574 6169 6e65  umber of retaine
+00001550: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
+00001560: 7320 6465 6361 7920 7175 6173 692d 6578  s decay quasi-ex
+00001570: 706f 6e65 6e74 6961 6c6c 792e 2054 6865  ponentially. The
+00001580: 7265 666f 7265 2c20 7468 6520 6f63 6375  refore, the occu
+00001590: 7261 6e63 6520 6f66 206c 6172 6765 2d73  rance of large-s
+000015a0: 6361 6c65 2067 656e 6520 6475 706c 6963  cale gene duplic
+000015b0: 6174 696f 6e20 6576 656e 7473 2c20 666f  ation events, fo
+000015c0: 7220 696e 7374 616e 6520 5747 4473 2c20  r instane WGDs, 
+000015d0: 7769 7468 2076 6172 6965 6420 7265 7465  with varied rete
+000015e0: 6e74 696f 6e20 7261 7465 2c20 7769 6c6c  ntion rate, will
+000015f0: 206c 6561 7665 2061 6e20 6167 6520 7065   leave an age pe
+00001600: 616b 2066 726f 6d20 7468 6520 6275 7273  ak from the burs
+00001610: 7420 6f66 2067 656e 6520 6475 706c 6963  t of gene duplic
+00001620: 6174 6573 2069 6e20 6120 7368 6f72 7420  ates in a short 
+00001630: 7469 6d65 2d66 7261 6d65 2075 706f 6e20  time-frame upon 
+00001640: 7468 6520 696e 6974 6961 6c20 6167 6520  the initial age 
+00001650: 6469 7374 7269 6275 7469 6f6e 2c20 616e  distribution, an
+00001660: 6420 6361 6e20 6265 2075 6e76 6569 6c65  d can be unveile
+00001670: 6420 6672 6f6d 206d 6978 7475 7265 206d  d from mixture m
+00001680: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
+00001690: 2e20 486f 7765 7665 722c 2057 4744 7320  . However, WGDs 
+000016a0: 6964 656e 7469 6669 6564 2066 726f 6d20  identified from 
+000016b0: 7468 6520 7061 7261 6c6f 676f 7573 202a  the paralogous *
+000016c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+000016d0: 6765 2064 6973 7472 6962 7574 696f 6e73  ge distributions
+000016e0: 2063 616e 206f 6e6c 7920 696e 666f 726d   can only inform
+000016f0: 2074 6865 2057 4744 2074 696d 696e 6720   the WGD timing 
+00001700: 696e 2074 6865 2074 696d 652d 7363 616c  in the time-scal
+00001710: 6520 6f66 2074 6861 7420 7370 6563 6966  e of that specif
+00001720: 6963 2073 7065 6369 6573 2c20 7768 6963  ic species, whic
+00001730: 6820 6973 206e 6f74 2063 6f6d 7061 7261  h is not compara
+00001740: 626c 6520 696e 2074 6865 2070 6879 6c6f  ble in the phylo
+00001750: 6765 6e65 7469 6320 636f 6e74 6578 742e  genetic context.
+00001760: 204f 6e6c 7920 7769 7468 2074 6865 206f   Only with the o
+00001770: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00001780: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00001790: 6973 7472 6962 7574 696f 6e73 2c20 7768  istributions, wh
+000017a0: 6963 6820 636f 6e76 6572 7420 7468 6520  ich convert the 
+000017b0: 6573 7469 6d61 7465 6420 626f 6479 2066  estimated body f
+000017c0: 726f 6d20 7061 7261 6c6f 6775 6573 2074  rom paralogues t
+000017d0: 6f20 6f72 7468 6f6c 6f67 7565 7320 616e  o orthologues an
+000017e0: 6420 696e 666f 726d 2074 6865 2072 656c  d inform the rel
+000017f0: 6174 6976 6520 7469 6d69 6e67 206f 6620  ative timing of 
+00001800: 7370 6563 6961 7469 6f6e 2065 7665 6e74  speciation event
+00001810: 732c 2063 616e 2077 6520 6465 6369 7068  s, can we deciph
+00001820: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
+00001830: 7469 6320 706c 6163 656d 656e 7420 6f66  tic placement of
+00001840: 2057 4744 7320 6166 7465 7220 7072 6f70   WGDs after prop
+00001850: 6572 2072 6174 6520 636f 7272 6563 7469  er rate correcti
+00001860: 6f6e 2e20 6077 6764 2076 3260 2069 7320  on. `wgd v2` is 
+00001870: 7375 6368 2070 726f 6772 616d 2074 6861  such program tha
+00001880: 7420 6865 6c70 7320 7573 6572 7320 636f  t helps users co
+00001890: 6e73 7472 7563 7420 7061 7261 6c6f 676f  nstruct paralogo
+000018a0: 7573 2061 6e64 206f 7274 686f 6c6f 676f  us and orthologo
+000018b0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+000018c0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+000018d0: 696f 6e73 2061 6e64 2072 6561 6c69 7a65  ions and realize
+000018e0: 2062 6f74 6820 7468 6520 6964 656e 7469   both the identi
+000018f0: 6669 6361 7469 6f6e 2061 6e64 2070 6c61  fication and pla
+00001900: 6365 6d65 6e74 206f 6620 5747 4473 2e0a  cement of WGDs..
+00001910: 0a49 6e20 7468 6520 7072 656d 6973 6520  .In the premise 
+00001920: 6f66 2070 6879 6c6f 6765 6e65 7469 6361  of phylogenetica
+00001930: 6c6c 7920 6c6f 6361 7465 6420 5747 4473  lly located WGDs
+00001940: 2c20 7468 6520 6162 736f 6c75 7465 2061  , the absolute a
+00001950: 6765 2028 696e 2067 656f 6c6f 6769 6361  ge (in geologica
+00001960: 6c20 7469 6d65 2920 6f66 2057 4744 7320  l time) of WGDs 
+00001970: 6361 6e20 616c 736f 2062 6520 696e 6665  can also be infe
+00001980: 7272 6564 2066 726f 6d20 7468 6f73 6520  rred from those 
+00001990: 5747 442d 7265 7461 696e 6564 2067 656e  WGD-retained gen
+000019a0: 6520 6475 706c 6963 6174 6573 2c20 616c  e duplicates, al
+000019b0: 7468 6f75 6768 2074 6865 7265 2068 6173  though there has
+000019c0: 2062 6565 6e20 6e6f 2065 6173 7920 6f72   been no easy or
+000019d0: 2073 7472 6169 6768 7466 6f72 7761 7264   straightforward
+000019e0: 2070 6970 656c 696e 6520 666f 7220 7468   pipeline for th
+000019f0: 6973 206a 6f62 2073 6f20 6661 722e 2049  is job so far. I
+00001a00: 6e20 6077 6764 2076 3260 2c20 7765 2064  n `wgd v2`, we d
+00001a10: 6576 656c 6f70 6564 2061 2066 6561 7369  eveloped a feasi
+00001a20: 626c 6520 696e 7465 6772 6174 6564 2070  ble integrated p
+00001a30: 6970 6c65 6c69 6e65 2066 6f72 2061 6273  ipleline for abs
+00001a40: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
+00001a50: 5747 4473 2e20 5468 6520 7069 7065 6c69  WGDs. The pipeli
+00001a60: 6e65 2063 616e 2062 6520 726f 7567 686c  ne can be roughl
+00001a70: 7920 6469 7669 6465 6420 696e 746f 2074  y divided into t
+00001a80: 6872 6565 206d 6169 6e20 7374 6570 732e  hree main steps.
+00001a90: 2031 2920 5468 6520 636f 6e73 7472 7563   1) The construc
+00001aa0: 7469 6f6e 206f 6620 616e 6368 6f72 202a  tion of anchor *
+00001ab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00001ac0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+00001ad0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
+00001ae0: 6f66 2063 7265 6469 6162 6c65 202a 4b2a  of crediable *K*
+00001af0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
+00001b00: 6765 2061 646f 7074 6564 2066 6f72 2070  ge adopted for p
+00001b10: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
+00001b20: 6e67 2c20 7573 696e 6720 6077 6764 2064  ng, using `wgd d
+00001b30: 6d64 602c 2060 7767 6420 6b73 6460 2c20  md`, `wgd ksd`, 
+00001b40: 6077 6764 2073 796e 6020 616e 6420 6077  `wgd syn` and `w
+00001b50: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
+00001b60: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
+00001b70: 2063 6f6e 7369 6465 7220 6765 6e6f 6d65   consider genome
+00001b80: 2061 7373 656d 626c 7920 6265 6361 7573   assembly becaus
+00001b90: 6520 666f 7220 7472 616e 7363 7269 7074  e for transcript
+00001ba0: 6f6d 6520 6173 7365 6d62 6c79 2069 7427  ome assembly it'
+00001bb0: 7320 696d 706f 7373 6962 6c65 2074 6f20  s impossible to 
+00001bc0: 6469 7374 696e 6775 6973 6820 5747 442d  distinguish WGD-
+00001bd0: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
+00001be0: 6573 2066 726f 6d20 736d 616c 6c2d 7363  es from small-sc
+00001bf0: 616c 6520 6475 706c 6963 6174 696f 6e2d  ale duplication-
+00001c00: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
+00001c10: 6573 2c20 7768 6963 6820 6861 7070 656e  es, which happen
+00001c20: 6564 2069 6e20 6120 636f 6e74 696e 756f  ed in a continuo
+00001c30: 7573 2074 696d 652d 6672 616d 6520 696e  us time-frame in
+00001c40: 7374 6561 6420 6f66 206f 6e6c 7920 6120  stead of only a 
+00001c50: 7365 7061 7261 7465 2073 686f 7274 2074  separate short t
+00001c60: 696d 652d 6672 616d 6520 616e 6420 7468  ime-frame and th
+00001c70: 7573 2072 6566 6c65 6374 7320 7468 6520  us reflects the 
+00001c80: 6475 7261 7469 6f6e 206f 6620 7468 6174  duration of that
+00001c90: 2062 7261 6e63 6820 7261 7468 6572 2074   branch rather t
+00001ca0: 6861 6e20 7468 6520 7469 6d65 2061 7420  han the time at 
+00001cb0: 7768 6963 6820 5747 4420 6f63 6375 7272  which WGD occurr
+00001cc0: 6564 2e20 3229 2054 6865 2066 6f72 6d75  ed. 2) The formu
+00001cd0: 6c61 7469 6f6e 206f 6620 6120 7374 6172  lation of a star
+00001ce0: 7469 6e67 2074 7265 6520 7573 6564 2069  ting tree used i
+00001cf0: 6e20 7468 6520 7068 796c 6f67 656e 6574  n the phylogenet
+00001d00: 6963 2064 6174 696e 672c 2063 6f6d 706f  ic dating, compo
+00001d10: 7365 6420 6f66 2061 2066 6577 2073 7065  sed of a few spe
+00001d20: 6369 6573 2061 6e64 2061 6e6e 6f74 6174  cies and annotat
+00001d30: 6564 2077 6974 6820 666f 7373 696c 2063  ed with fossil c
+00001d40: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
+00001d50: 6d61 7469 6f6e 2e20 5468 6973 2073 7465  mation. This ste
+00001d60: 7020 6973 2065 7373 656e 7469 616c 2066  p is essential f
+00001d70: 6f72 2074 6865 2072 6573 756c 7420 6f66  or the result of
+00001d80: 2061 6273 6f6c 7574 6520 5747 4420 6461   absolute WGD da
+00001d90: 7469 6e67 2074 6861 7420 7765 2073 7567  ting that we sug
+00001da0: 6765 7374 2075 7365 7273 206f 6620 7461  gest users of ta
+00001db0: 6b69 6e67 2067 7265 6174 2063 6175 7469  king great cauti
+00001dc0: 6f6e 2074 6f20 6173 7375 7265 2074 6865  on to assure the
+00001dd0: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
+00001de0: 6e64 2070 726f 7065 7220 626f 756e 6473  nd proper bounds
+00001df0: 2066 6f72 2066 6f73 7369 6c20 6361 6c69   for fossil cali
+00001e00: 6272 6174 696f 6e73 2e20 3329 2054 6865  brations. 3) The
+00001e10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00001e20: 206f 7274 686f 6772 6f75 7073 2063 6f6e   orthogroups con
+00001e30: 7369 7374 696e 6720 6f66 2063 6f6c 6c69  sisting of colli
+00001e40: 6e65 6172 2064 7570 6c69 6361 7465 7320  near duplicates 
+00001e50: 6f66 2074 6865 2066 6f63 616c 2073 7065  of the focal spe
+00001e60: 6369 6573 2061 6e64 2074 6865 6972 2072  cies and their r
+00001e70: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
+00001e80: 6974 7320 2852 4248 7329 2061 6761 696e  its (RBHs) again
+00001e90: 7374 206f 7468 6572 2073 7065 6369 6573  st other species
+00001ea0: 2069 6e20 7468 6520 7374 6172 7469 6e67   in the starting
+00001eb0: 2074 7265 6520 616e 6420 7468 6520 7068   tree and the ph
+00001ec0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+00001ed0: 6720 7573 696e 6720 6120 6d6f 6c65 6375  g using a molecu
+00001ee0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
+00001ef0: 616d 2066 6f72 2069 6e73 7461 6e63 6520  am for instance 
+00001f00: 606d 636d 6374 7265 6560 2c20 7669 6120  `mcmctree`, via 
+00001f10: 7072 6f67 7261 6d20 6077 6764 2064 6d64  program `wgd dmd
+00001f20: 6020 616e 6420 6077 6764 2066 6f63 7573  ` and `wgd focus
+00001f30: 602e 2057 6520 7265 636f 6d6d 656e 6420  `. We recommend 
+00001f40: 7468 6520 7573 6167 6520 6f66 2042 6179  the usage of Bay
+00001f50: 6573 6961 6e20 6d6f 6c65 6375 6c61 7220  esian molecular 
+00001f60: 6461 7469 6e67 2070 726f 6772 616d 2060  dating program `
+00001f70: 6d63 6d63 7472 6565 602c 2077 6869 6368  mcmctree`, which
+00001f80: 2070 726f 7669 6465 7320 6120 7661 7269   provides a vari
+00001f90: 6574 7920 6f66 2064 6966 6665 7265 6e74  ety of different
+00001fa0: 2073 7562 7374 6974 7574 696f 6e20 616e   substitution an
+00001fb0: 6420 7261 7465 206d 6f64 656c 732e 204e  d rate models. N
+00001fc0: 6f6e 6574 6865 6c65 7373 2c20 7765 2075  onetheless, we u
+00001fd0: 7267 6520 7573 6572 7320 746f 2073 6574  rge users to set
+00001fe0: 2070 7269 6f72 2064 6973 7472 6962 7574   prior distribut
+00001ff0: 696f 6e20 6f66 2064 6966 6665 7265 6e74  ion of different
+00002000: 2070 6172 616d 6574 6572 7320 7769 7468   parameters with
+00002010: 2063 6175 7469 6f6e 2061 6e64 2061 7373   caution and ass
+00002020: 7572 6520 6164 6571 7561 7465 2073 616d  ure adequate sam
+00002030: 706c 696e 6720 6f66 2064 6966 6665 7265  pling of differe
+00002040: 6e74 2070 6172 616d 6574 6572 732e 0a0a  nt parameters...
+00002050: 2323 2049 6e73 7461 6c6c 6174 696f 6e0a  ## Installation.
+00002060: 0a54 6865 2065 6173 6965 7374 2077 6179  .The easiest way
+00002070: 2074 6f20 696e 7374 616c 6c20 6077 6764   to install `wgd
+00002080: 2076 3260 2069 7320 7573 696e 6720 5059   v2` is using PY
+00002090: 5049 2e20 4e6f 7465 2074 6861 7420 6966  PI. Note that if
+000020a0: 2079 6f75 2077 616e 7420 746f 2067 6574   you want to get
+000020b0: 2074 6865 206c 6174 6573 7420 7570 6461   the latest upda
+000020c0: 7465 2c20 7765 2073 7567 6765 7374 2069  te, we suggest i
+000020d0: 6e73 7461 6c6c 696e 6720 6672 6f6d 2074  nstalling from t
+000020e0: 6865 2073 6f75 7263 652c 2073 696e 6365  he source, since
+000020f0: 2074 6865 2075 7064 6174 6520 6f6e 2050   the update on P
+00002100: 5950 4920 7769 6c6c 2062 6520 6465 6c61  YPI will be dela
+00002110: 7965 6420 636f 6d70 6172 6564 2074 6f20  yed compared to 
+00002120: 6865 7265 206f 6620 736f 7572 6365 2e0a  here of source..
+00002130: 0a60 6060 0a70 6970 2069 6e73 7461 6c6c  .```.pip install
+00002140: 2077 6764 0a60 6060 0a0a 546f 2069 6e73   wgd.```..To ins
+00002150: 7461 6c6c 2060 7767 6420 7632 6020 696e  tall `wgd v2` in
+00002160: 2061 2076 6972 7475 616c 2065 6e76 6972   a virtual envir
+00002170: 6f6e 6d65 6e74 2c20 7468 6520 666f 6c6c  onment, the foll
+00002180: 6f77 696e 6720 636f 6d6d 616e 6420 6c69  owing command li
+00002190: 6e65 7320 636f 756c 6420 6265 2075 7365  nes could be use
+000021a0: 642e 0a0a 6060 600a 6769 7420 636c 6f6e  d...```.git clon
+000021b0: 6520 3c77 6764 2072 6570 6f3e 0a63 6420  e <wgd repo>.cd 
+000021c0: 7767 640a 7669 7274 7561 6c65 6e76 202d  wgd.virtualenv -
+000021d0: 703d 7079 7468 6f6e 3320 454e 5620 286f  p=python3 ENV (o
+000021e0: 7220 7079 7468 6f6e 3320 2d6d 2076 656e  r python3 -m ven
+000021f0: 7620 454e 5629 0a73 6f75 7263 6520 454e  v ENV).source EN
+00002200: 562f 6269 6e2f 6163 7469 7661 7465 0a70  V/bin/activate.p
+00002210: 6970 2069 6e73 7461 6c6c 202d 7220 7265  ip install -r re
+00002220: 7175 6972 656d 656e 7473 2e74 7874 0a70  quirements.txt.p
+00002230: 6970 2069 6e73 7461 6c6c 202e 0a60 6060  ip install ..```
+00002240: 0a0a 5768 656e 206d 6574 2077 6974 6820  ..When met with 
+00002250: 7065 726d 6973 7369 6f6e 2070 726f 626c  permission probl
+00002260: 656d 2069 6e20 696e 7374 616c 6c61 7469  em in installati
+00002270: 6f6e 2c20 706c 6561 7365 2074 7279 2074  on, please try t
+00002280: 6865 2066 6f6c 6c6f 7769 6e67 2063 6f6d  he following com
+00002290: 6d61 6e64 206c 696e 652e 0a0a 6060 600a  mand line...```.
+000022a0: 7069 7020 696e 7374 616c 6c20 2d65 202e  pip install -e .
+000022b0: 0a60 6060 0a0a 4966 206d 756c 7469 706c  .```..If multipl
+000022c0: 7920 7665 7273 696f 6e73 206f 6620 6077  y versions of `w
+000022d0: 6764 6020 7765 7265 2069 6e73 7461 6c6c  gd` were install
+000022e0: 6564 2069 6e20 7468 6520 7379 7374 656d  ed in the system
+000022f0: 2c20 706c 6561 7365 2061 6464 2074 6865  , please add the
+00002300: 2072 6967 6874 2070 6174 6820 6f66 2069   right path of i
+00002310: 6e74 6572 6573 7465 6420 7665 7273 696f  nterested versio
+00002320: 6e20 696e 746f 2074 6865 2065 6e76 6972  n into the envir
+00002330: 6f6e 6d65 6e74 2076 6172 6961 626c 6573  onment variables
+00002340: 2c20 666f 7220 6578 616d 706c 650a 0a60  , for example..`
+00002350: 6060 0a65 7870 6f72 7420 5041 5448 3d22  ``.export PATH="
+00002360: 2450 4154 483a 7e2f 2e6c 6f63 616c 2f62  $PATH:~/.local/b
+00002370: 696e 2f77 6764 220a 6060 600a 0a4e 6f74  in/wgd".```..Not
+00002380: 6520 7468 6174 2074 6865 2076 6572 7369  e that the versi
+00002390: 6f6e 206f 6620 606e 756d 7079 6020 6973  on of `numpy` is
+000023a0: 2069 6d70 6f72 7461 6e74 2028 666f 7220   important (for 
+000023b0: 6d61 6e79 206f 7468 6572 2070 6163 6b61  many other packa
+000023c0: 6765 7320 6172 6520 7468 6520 7361 6d65  ges are the same
+000023d0: 206f 6620 636f 7572 7365 292c 2065 7370   of course), esp
+000023e0: 6563 6961 6c6c 7920 666f 7220 6066 6173  ecially for `fas
+000023f0: 7463 6c75 7374 6572 6020 7061 636b 6167  tcluster` packag
+00002400: 652e 2049 6e20 6f75 7220 7465 7374 2c20  e. In our test, 
+00002410: 7468 6520 606e 756d 7079 6020 312e 3139  the `numpy` 1.19
+00002420: 2e30 2077 6f72 6b73 2066 696e 6520 6f6e  .0 works fine on
+00002430: 2060 7079 7468 6f6e 332e 362f 3860 2e20   `python3.6/8`. 
+00002440: 4966 2079 6f75 206d 6574 2073 6f6d 6520  If you met some 
+00002450: 6572 726f 7273 206f 7220 7761 726e 696e  errors or warnin
+00002460: 6773 2061 626f 7574 2060 6e75 6d70 7960  gs about `numpy`
+00002470: 2c20 6d61 7962 6520 636f 6e73 6964 6572  , maybe consider
+00002480: 696e 6720 7072 652d 696e 7374 616c 6c20  ing pre-install 
+00002490: 606e 756d 7079 6020 6173 2031 2e31 392e  `numpy` as 1.19.
+000024a0: 3020 6f72 206f 7468 6572 2063 6c6f 7365  0 or other close
+000024b0: 2d62 7920 7665 7273 696f 6e73 2062 6566  -by versions bef
+000024c0: 6f72 6520 796f 7520 696e 7374 616c 6c20  ore you install 
+000024d0: 6077 6764 602e 0a0a 2323 2050 6172 616d  `wgd`...## Param
+000024e0: 6574 6572 730a 0a54 6865 7265 2061 7265  eters..There are
+000024f0: 2037 206d 6169 6e20 7072 6f67 7261 6d73   7 main programs
+00002500: 2069 6e20 6077 6764 2076 3260 3a20 6064   in `wgd v2`: `d
+00002510: 6d64 602c 6066 6f63 7573 602c 606b 7364  md`,`focus`,`ksd
+00002520: 602c 606d 6978 602c 6070 6561 6b60 2c60  `,`mix`,`peak`,`
+00002530: 7379 6e60 2c60 7669 7a60 2e20 4865 7265  syn`,`viz`. Here
+00002540: 6166 7465 7220 7765 2077 696c 6c20 7072  after we will pr
+00002550: 6f76 6964 6520 6120 6465 7461 696c 6564  ovide a detailed
+00002560: 2065 6c75 6369 6461 7469 6f6e 206f 6e20   elucidation on 
+00002570: 6561 6368 206f 6620 7468 6520 7072 6f67  each of the prog
+00002580: 7261 6d20 616e 6420 6974 7320 6173 736f  ram and its asso
+00002590: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
+000025a0: 732e 2050 6c65 6173 6520 7265 6665 7220  s. Please refer 
+000025b0: 746f 2074 6865 205b 5573 6167 655d 2823  to the [Usage](#
+000025c0: 7573 6167 6529 2066 6f72 2074 6865 2073  usage) for the s
+000025d0: 6365 6e61 7269 6f73 2074 6f20 7768 6963  cenarios to whic
+000025e0: 6820 6561 6368 2070 6172 616d 6574 6572  h each parameter
+000025f0: 2061 7070 6c69 6573 2e0a 0a54 6865 2070   applies...The p
+00002600: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
+00002610: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+00002620: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+00002630: 7768 6f6c 6520 7061 7261 6e6f 6d65 2c20  whole paranome, 
+00002640: 5242 4873 2028 5265 6369 7072 6f63 616c  RBHs (Reciprocal
+00002650: 2042 6573 7420 4869 7473 292c 204d 5242   Best Hits), MRB
+00002660: 4873 2028 4d75 6c74 6970 6c65 2052 6563  Hs (Multiple Rec
+00002670: 6970 726f 6361 6c20 4265 7374 2048 6974  iprocal Best Hit
+00002680: 7329 2c20 6f72 7468 6f67 726f 7570 7320  s), orthogroups 
+00002690: 616e 6420 736f 6d65 206f 7468 6572 206f  and some other o
+000026a0: 7274 686f 6772 6f75 702d 7265 6c61 7465  rthogroup-relate
+000026b0: 6420 6675 6e63 7469 6f6e 732c 2069 6e63  d functions, inc
+000026c0: 6c75 6469 6e67 2063 6972 6375 6d73 6372  luding circumscr
+000026d0: 6970 7469 6f6e 206f 6620 6e65 7374 6564  iption of nested
+000026e0: 2073 696e 676c 652d 636f 7079 206f 7274   single-copy ort
+000026f0: 686f 6772 6f75 7073 2028 4e53 4f47 7329  hogroups (NSOGs)
+00002700: 2c20 756e 6269 6173 6564 2074 6573 7420  , unbiased test 
+00002710: 6f66 2073 696e 676c 652d 636f 7079 206f  of single-copy o
+00002720: 7274 686f 6772 6f75 7073 2028 534f 4773  rthogroups (SOGs
+00002730: 2920 6f76 6572 206d 6973 7369 6e67 2069  ) over missing i
+00002740: 6e70 6172 616c 6f67 732c 2063 6f6e 7374  nparalogs, const
+00002750: 7275 6374 696f 6e20 6f66 2042 5553 434f  ruction of BUSCO
+00002760: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
+00002770: 6f70 7920 6f72 7468 6f67 726f 7570 7320  opy orthogroups 
+00002780: 2853 4f47 7329 2c61 6e64 2074 6865 2063  (SOGs),and the c
+00002790: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+000027a0: 656e 6365 2069 6e66 6572 656e 6365 206f  ence inference o
+000027b0: 6620 7068 796c 6f67 656e 792e 0a60 6060  f phylogeny..```
+000027c0: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+000027d0: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
+000027e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000027f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002800: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002810: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002820: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
+00002830: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+00002840: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00002850: 2064 6566 6175 6c74 2077 6764 5f64 6d64   default wgd_dmd
+00002860: 0a2d 742c 202d 2d74 6d70 6469 722c 2074  .-t, --tmpdir, t
+00002870: 6865 2074 656d 706f 7261 7279 2077 6f72  he temporary wor
+00002880: 6b69 6e67 2064 6972 6563 746f 7279 2c20  king directory, 
+00002890: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
+000028a0: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
+000028b0: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+000028c0: 6420 6265 2061 7373 6967 6e65 6420 7261  d be assigned ra
+000028d0: 6e64 6f6d 206e 616d 6573 2069 6e20 6375  ndom names in cu
+000028e0: 7272 656e 7420 6469 7265 6374 6f72 7920  rrent directory 
+000028f0: 616e 6420 6175 746f 6d61 7465 6c79 2072  and automately r
+00002900: 656d 6f76 6564 2061 7420 7468 6520 636f  emoved at the co
+00002910: 6d70 6c65 7469 6f6e 206f 6620 7072 6f67  mpletion of prog
+00002920: 7261 6d2c 2065 6c73 6520 7468 6520 746d  ram, else the tm
+00002930: 7064 6972 2077 6f75 6c64 2062 6520 6b65  pdir would be ke
+00002940: 7074 0a2d 702c 202d 2d70 726f 742c 2066  pt.-p, --prot, f
+00002950: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00002960: 6865 7220 7573 696e 6720 7072 6f74 6569  her using protei
+00002970: 6e20 6f72 206e 7563 6c65 6f74 6964 6520  n or nucleotide 
+00002980: 7365 7175 656e 6365 730a 2d63 2c20 2d2d  sequences.-c, --
+00002990: 6373 636f 7265 2c20 7468 6520 632d 7363  cscore, the c-sc
+000029a0: 6f72 6520 746f 2072 6573 7472 6963 7420  ore to restrict 
+000029b0: 7468 6520 686f 6d6f 6c6f 6720 7369 6d69  the homolog simi
+000029c0: 6c61 7269 7479 206f 6620 4d52 4248 732c  larity of MRBHs,
+000029d0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
+000029e0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
+000029f0: 2c20 7468 6520 632d 7363 6f72 6520 6675  , the c-score fu
+00002a00: 6e63 696f 6e20 776f 756c 646e 2774 2062  ncion wouldn't b
+00002a10: 6520 6163 7469 7661 7465 642c 2065 6c73  e activated, els
+00002a20: 6520 6578 7065 6374 696e 6720 6120 6465  e expecting a de
+00002a30: 6369 6d61 6c20 7769 7468 696e 2074 6865  cimal within the
+00002a40: 2072 616e 6765 206f 6620 3020 616e 6420   range of 0 and 
+00002a50: 310a 2d49 2c20 2d2d 696e 666c 6174 696f  1.-I, --inflatio
+00002a60: 6e2c 2074 6865 2069 6e66 6c61 7469 6f6e  n, the inflation
+00002a70: 2066 6163 746f 7220 666f 7220 4d43 4c20   factor for MCL 
+00002a80: 7072 6f67 7261 6d2c 2064 6566 6175 6c74  program, default
+00002a90: 2032 2e30 2c20 7769 7468 2068 6967 6865   2.0, with highe
+00002aa0: 7220 7661 6c75 6520 6c65 6164 696e 6720  r value leading 
+00002ab0: 746f 206d 6f72 6520 6275 7420 736d 616c  to more but smal
+00002ac0: 6c65 7220 636c 7573 7465 7273 0a2d 652c  ler clusters.-e,
+00002ad0: 202d 2d65 7661 6c2c 2074 6865 2065 2d76   --eval, the e-v
+00002ae0: 616c 7565 2063 7574 2d6f 6666 2066 6f72  alue cut-off for
+00002af0: 2073 696d 696c 6172 6974 7920 696e 2064   similarity in d
+00002b00: 6961 6d6f 6e64 2061 6e64 2f6f 7220 686d  iamond and/or hm
+00002b10: 6d65 722c 2064 6566 6175 6c74 2031 652d  mer, default 1e-
+00002b20: 3130 0a2d 2d74 6f5f 7374 6f70 2c20 666c  10.--to_stop, fl
+00002b30: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00002b40: 6572 2074 6f20 7472 616e 736c 6174 6520  er to translate 
+00002b50: 7468 726f 7567 6820 5354 4f50 2063 6f64  through STOP cod
+00002b60: 6f6e 732c 2069 6620 7468 6520 666c 6167  ons, if the flag
+00002b70: 2077 6173 2073 6574 2c20 7472 616e 736c   was set, transl
+00002b80: 6174 696f 6e20 7769 6c6c 2062 6520 7465  ation will be te
+00002b90: 726d 696e 6174 6564 2061 7420 7468 6520  rminated at the 
+00002ba0: 6669 7273 7420 696e 2d66 7261 6d65 2073  first in-frame s
+00002bb0: 746f 7020 636f 646f 6e2c 2065 6c73 6520  top codon, else 
+00002bc0: 6120 6675 6c6c 2074 7261 6e73 6c61 7469  a full translati
+00002bd0: 6f6e 2063 6f6e 7469 6e75 696e 6720 6f6e  on continuing on
+00002be0: 2070 6173 7369 6e67 2061 6e79 2073 746f   passing any sto
+00002bf0: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
+00002c00: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
+00002c10: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00002c20: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+00002c30: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
+00002c40: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
+00002c50: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
+00002c60: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
+00002c70: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
+00002c80: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
+00002c90: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
+00002ca0: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
+00002cb0: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
+00002cc0: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
+00002cd0: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
+00002ce0: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
+00002cf0: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
+00002d00: 616e 736c 6174 6564 0a2d 662c 202d 2d66  anslated.-f, --f
+00002d10: 6f63 7573 2c20 7468 6520 7370 6563 6965  ocus, the specie
+00002d20: 7320 746f 2062 6520 6d65 7267 6564 206f  s to be merged o
+00002d30: 6e20 6c6f 6361 6c20 4d52 4248 732c 2064  n local MRBHs, d
+00002d40: 6566 6175 6c74 204e 6f6e 652c 2069 6620  efault None, if 
+00002d50: 4e6f 6e65 2077 6173 2067 6976 656e 2c20  None was given, 
+00002d60: 7468 6520 6c6f 6361 6c20 4d52 4248 7320  the local MRBHs 
+00002d70: 776f 756c 646e 2774 2062 6520 696e 6665  wouldn't be infe
+00002d80: 7272 6564 0a2d 6170 2c20 2d2d 616e 6368  rred.-ap, --anch
+00002d90: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+00002da0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+00002db0: 2066 696c 6520 6672 6f6d 2069 2d61 6468   file from i-adh
+00002dc0: 6f72 6520 666f 7220 636f 6e73 7472 7563  ore for construc
+00002dd0: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
+00002de0: 6f75 7073 2077 6974 6820 616e 6368 6f72  oups with anchor
+00002df0: 2070 6169 7273 2c20 6465 6661 756c 7420   pairs, default 
+00002e00: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
+00002e10: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
+00002e20: 7473 2064 6174 6166 696c 6520 7573 6564  ts datafile used
+00002e30: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
+00002e40: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+00002e50: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
+00002e60: 2064 6566 6175 6c74 204e 6f6e 650a 2d6c   default None.-l
+00002e70: 652c 202d 2d6c 6973 7465 6c65 6d65 6e74  e, --listelement
+00002e80: 732c 2074 6865 206c 6973 7473 6567 6d65  s, the listsegme
+00002e90: 6e74 7320 6461 7461 2066 696c 6520 7573  nts data file us
+00002ea0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
+00002eb0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+00002ec0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
+00002ed0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
+00002ee0: 2d67 742c 202d 2d67 656e 6574 6162 6c65  -gt, --genetable
+00002ef0: 2c20 7468 6520 6765 6e65 2074 6162 6c65  , the gene table
+00002f00: 2064 6174 6166 696c 6520 7573 6564 2069   datafile used i
+00002f10: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
+00002f20: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
+00002f30: 2069 6620 696e 6974 6961 7465 642c 2064   if initiated, d
+00002f40: 6566 6175 6c74 204e 6f6e 650a 2d63 6f63  efault None.-coc
+00002f50: 2c20 2d2d 636f 6c6c 696e 6561 7263 6f61  , --collinearcoa
+00002f60: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
+00002f70: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00002f80: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
+00002f90: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+00002fa0: 656e 6365 2061 6e61 6c79 7369 732c 2069  ence analysis, i
+00002fb0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00002fc0: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+00002fd0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00002fe0: 7465 640a 2d6b 662c 202d 2d6b 6565 7066  ted.-kf, --keepf
+00002ff0: 6173 7461 2c20 666c 6167 206f 7074 696f  asta, flag optio
+00003000: 6e2c 2077 6865 7468 6572 2074 6f20 6f75  n, whether to ou
+00003010: 7470 7574 2074 6865 2073 6571 7565 6e63  tput the sequenc
+00003020: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
+00003030: 204d 5242 4873 2c20 6966 2074 6865 2066   MRBHs, if the f
+00003040: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00003050: 2073 6571 7565 6e63 6573 206f 6620 4d52   sequences of MR
+00003060: 4248 7320 776f 756c 6420 6265 2069 6e20  BHs would be in 
+00003070: 6f75 7470 7574 0a2d 6b64 2c20 2d2d 6b65  output.-kd, --ke
+00003080: 6570 6475 706c 6963 6174 6573 2c20 666c  epduplicates, fl
+00003090: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+000030a0: 6572 2074 6f20 616c 6c6f 7720 7468 6520  er to allow the 
+000030b0: 7361 6d65 2067 656e 6520 746f 206f 6363  same gene to occ
+000030c0: 7572 2069 6e20 6469 6666 6572 656e 7420  ur in different 
+000030d0: 4d52 4248 7320 286f 6e6c 7920 6d65 616e  MRBHs (only mean
+000030e0: 696e 6766 756c 2077 6865 6e20 7468 6520  ingful when the 
+000030f0: 6373 636f 7265 2077 6173 2075 7365 6429  cscore was used)
+00003100: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00003110: 7320 7365 742c 2074 6865 2073 616d 6520  s set, the same 
+00003120: 6765 6e65 2063 6f75 6c64 2062 6520 6173  gene could be as
+00003130: 7369 676e 6564 2074 6f20 6469 6666 6572  signed to differ
+00003140: 656e 7420 4d52 4248 730a 2d67 6d2c 202d  ent MRBHs.-gm, -
+00003150: 2d67 6c6f 6261 6c6d 7262 682c 2066 6c61  -globalmrbh, fla
+00003160: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00003170: 7220 746f 2069 6e69 7469 6174 6520 676c  r to initiate gl
+00003180: 6f62 616c 204d 5242 4873 2063 6f6e 7374  obal MRBHs const
+00003190: 7275 6374 696f 6e2c 2069 6620 7468 6520  ruction, if the 
+000031a0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+000031b0: 6520 2d2d 666f 6375 7320 6f70 7469 6f6e  e --focus option
+000031c0: 2077 6f75 6c64 2062 6520 6967 6e6f 7265   would be ignore
+000031d0: 6420 616e 6420 6f6e 6c79 2067 6c6f 6261  d and only globa
+000031e0: 6c20 4d52 4248 7320 776f 756c 6420 6265  l MRBHs would be
+000031f0: 2062 7569 6c74 0a2d 6e2c 202d 2d6e 7468   built.-n, --nth
+00003200: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+00003210: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+00003220: 7573 652c 2064 6566 6175 6c74 2034 0a2d  use, default 4.-
+00003230: 6f69 2c20 2d2d 6f72 7468 6f69 6e66 6572  oi, --orthoinfer
+00003240: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00003250: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00003260: 7465 206f 7274 686f 6772 6f75 7020 696e  te orthogroup in
+00003270: 6665 726e 6563 652c 2069 6620 7468 6520  fernece, if the 
+00003280: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00003290: 6520 6f72 7468 6f67 726f 7570 2069 6e66  e orthogroup inf
+000032a0: 6572 6e65 6365 2070 726f 6772 616d 2077  ernece program w
+000032b0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+000032c0: 640a 2d6f 6f2c 202d 2d6f 6e6c 796f 7274  d.-oo, --onlyort
+000032d0: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
+000032e0: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
+000032f0: 2063 6f6e 6475 6374 206f 7274 686f 6772   conduct orthogr
+00003300: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
+00003310: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00003320: 6574 2c20 6f6e 6c79 2074 6865 206f 7274  et, only the ort
+00003330: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
+00003340: 6520 7069 7065 6c69 6e65 2077 6f75 6c64  e pipeline would
+00003350: 2062 6520 7065 7266 6f72 6d65 6420 7768   be performed wh
+00003360: 696c 6520 7468 6520 6f74 6865 7220 616e  ile the other an
+00003370: 616c 7973 6973 2077 6f75 6c64 6e27 7420  alysis wouldn't 
+00003380: 6265 2069 6e69 7469 6174 6564 0a2d 676e  be initiated.-gn
+00003390: 2c20 2d2d 6765 746e 736f 672c 2066 6c61  , --getnsog, fla
+000033a0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000033b0: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+000033c0: 6520 7365 6172 6368 696e 6720 666f 7220  e searching for 
+000033d0: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
+000033e0: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
+000033f0: 2028 4e53 4f47 7329 2028 6f6e 6c79 206d   (NSOGs) (only m
+00003400: 6561 6e69 6e67 6675 6c20 7768 656e 2074  eaningful when t
+00003410: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00003420: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+00003430: 2077 6173 2061 6374 6976 6174 6564 292c   was activated),
+00003440: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00003450: 2073 6574 2c20 6164 6469 7469 6f6e 616c   set, additional
+00003460: 204e 534f 4773 2061 6e61 6c79 7369 7320   NSOGs analysis 
+00003470: 776f 756c 6420 6265 2070 6572 666f 726d  would be perform
+00003480: 6564 2062 6573 6964 6573 2074 6865 2062  ed besides the b
+00003490: 6173 6963 206f 7274 686f 6772 6f75 7020  asic orthogroup 
+000034a0: 696e 6665 726e 6563 650a 2d74 7265 652c  infernece.-tree,
+000034b0: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
+000034c0: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
+000034d0: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+000034e0: 6d20 746f 2069 6e76 6f6b 6520 286f 6e6c  m to invoke (onl
+000034f0: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
+00003500: 6e20 7468 6520 636f 6c6c 696e 6561 7220  n the collinear 
+00003510: 636f 616c 6573 6365 6e63 652c 2067 656e  coalescence, gen
+00003520: 652d 746f 2d66 616d 696c 7920 6173 7369  e-to-family assi
+00003530: 676e 6d65 6e74 206f 7220 4e53 4f47 7320  gnment or NSOGs 
+00003540: 616e 616c 7973 6973 2077 6572 6520 6163  analysis were ac
+00003550: 7469 7661 7465 6429 2c20 6465 6661 756c  tivated), defaul
+00003560: 7420 6661 7374 7472 6565 0a2d 7473 2c20  t fasttree.-ts, 
+00003570: 2d2d 7472 6565 7365 742c 2074 6865 2070  --treeset, the p
+00003580: 6172 616d 6574 6572 7320 7365 7474 696e  arameters settin
+00003590: 6720 666f 7220 6765 6e65 2074 7265 6520  g for gene tree 
+000035a0: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
+000035b0: 6c74 204e 6f6e 652c 2074 6869 7320 6f70  lt None, this op
+000035c0: 7469 6f6e 2063 616e 2062 6520 7072 6f76  tion can be prov
+000035d0: 6964 6564 206d 756c 7469 706c 6520 7469  ided multiple ti
+000035e0: 6d65 730a 2d6d 632c 202d 2d6d 736f 6763  mes.-mc, --msogc
+000035f0: 7574 2c20 7468 6520 7261 7469 6f20 6375  ut, the ratio cu
+00003600: 746f 6666 2066 6f72 206d 6f73 746c 7920  toff for mostly 
+00003610: 7369 6e67 6c65 2d63 6f70 7920 6661 6d69  single-copy fami
+00003620: 6c79 2028 6d65 616e 696e 6766 756c 2077  ly (meaningful w
+00003630: 6865 6e20 6163 7469 7661 7469 6e67 2074  hen activating t
+00003640: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00003650: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+00003660: 2920 616e 6420 7370 6563 6965 7320 7265  ) and species re
+00003670: 7072 6573 656e 7461 7469 6f6e 2069 6e20  presentation in 
+00003680: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
+00003690: 6365 6e63 6520 616e 616c 7973 6973 2c20  cence analysis, 
+000036a0: 6465 6661 756c 7420 302e 382e 0a2d 6761  default 0.8..-ga
+000036b0: 2c20 2d2d 6765 6e65 6173 7369 676e 2c20  , --geneassign, 
+000036c0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+000036d0: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+000036e0: 2074 6865 2067 656e 652d 746f 2d66 616d   the gene-to-fam
+000036f0: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
+00003700: 6e61 6c79 7369 732c 2069 6620 7468 6520  nalysis, if the 
+00003710: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00003720: 6520 616e 616c 7973 6973 2077 6f75 6c64  e analysis would
+00003730: 2062 6520 696e 6974 6961 7465 640a 2d73   be initiated.-s
+00003740: 612c 202d 2d73 6571 3261 7373 6967 6e2c  a, --seq2assign,
+00003750: 2074 6865 2071 7565 7269 6564 2073 6571   the queried seq
+00003760: 7565 6e63 6573 2064 6174 6120 6669 6c65  uences data file
+00003770: 2069 6e20 6765 6e65 2d74 6f2d 6661 6d69   in gene-to-fami
+00003780: 6c79 2061 7373 6967 6e6d 656e 7420 616e  ly assignment an
+00003790: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
+000037a0: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
+000037b0: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
+000037c0: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
+000037d0: 0a2d 6661 2c20 2d2d 6661 6d32 6173 7369  .-fa, --fam2assi
+000037e0: 676e 2c20 7468 6520 7175 6572 6965 6420  gn, the queried 
+000037f0: 6661 6d69 6c69 7920 6461 7461 2066 696c  familiy data fil
+00003800: 6520 696e 2067 656e 652d 746f 2d66 616d  e in gene-to-fam
+00003810: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
+00003820: 6e61 6c79 7369 732c 2064 6566 6175 6c74  nalysis, default
+00003830: 204e 6f6e 650a 2d63 632c 202d 2d63 6f6e   None.-cc, --con
+00003840: 6361 742c 2066 6c61 6720 6f70 7469 6f6e  cat, flag option
+00003850: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+00003860: 7469 6174 6520 7468 6520 636f 6e63 6174  tiate the concat
+00003870: 656e 6174 696f 6e20 7069 7065 6c69 6e65  enation pipeline
+00003880: 2066 6f72 206f 7274 686f 6772 6f75 7020   for orthogroup 
+00003890: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
+000038a0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+000038b0: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
+000038c0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+000038d0: 2d74 652c 202d 2d74 6573 7473 6f67 2c20  -te, --testsog, 
+000038e0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+000038f0: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00003900: 2074 6865 2075 6e62 6961 7365 6420 7465   the unbiased te
+00003910: 7374 206f 6620 7369 6e67 6c65 2d63 6f70  st of single-cop
+00003920: 7920 6765 6e65 2066 616d 696c 6965 732c  y gene families,
+00003930: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00003940: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+00003950: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+00003960: 6961 7465 640a 2d62 732c 202d 2d62 696e  iated.-bs, --bin
+00003970: 732c 2074 6865 206e 756d 6265 7220 6f66  s, the number of
+00003980: 2062 696e 7320 6469 7669 6465 6420 696e   bins divided in
+00003990: 2074 6865 2067 656e 6520 6c65 6e67 7468   the gene length
+000039a0: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
+000039b0: 6465 6661 756c 7420 3130 300a 2d6e 702c  default 100.-np,
+000039c0: 202d 2d6e 6f72 6d61 6c69 7a65 6470 6572   --normalizedper
+000039d0: 6365 6e74 2c20 7468 6520 7065 7263 656e  cent, the percen
+000039e0: 7461 6765 206f 6620 7570 7065 7220 6869  tage of upper hi
+000039f0: 7473 2075 7365 6420 666f 7220 6765 6e65  ts used for gene
+00003a00: 206c 656e 6774 6820 6e6f 726d 616c 697a   length normaliz
+00003a10: 6174 696f 6e2c 2064 6566 6175 6c74 2035  ation, default 5
+00003a20: 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d 616c  .-nn, --nonormal
+00003a30: 697a 6174 696f 6e2c 2066 6c61 6720 6f70  ization, flag op
+00003a40: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00003a50: 2063 616c 6c20 6f66 6620 7468 6520 6e6f   call off the no
+00003a60: 726d 616c 697a 6174 696f 6e2c 2069 6620  rmalization, if 
+00003a70: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00003a80: 2c20 6e6f 206e 6f72 6d61 6c69 7a61 7469  , no normalizati
+00003a90: 6f6e 2077 6f75 6c64 2062 6520 636f 6e64  on would be cond
+00003aa0: 7563 7465 640a 2d62 736f 672c 202d 2d62  ucted.-bsog, --b
+00003ab0: 7573 636f 736f 672c 2066 6c61 6720 6f70  uscosog, flag op
+00003ac0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00003ad0: 2069 6e69 7469 6174 6520 7468 6520 6275   initiate the bu
+00003ae0: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
+00003af0: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
+00003b00: 6c79 2061 6e61 6c79 7369 732c 2069 6620  ly analysis, if 
+00003b10: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00003b20: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
+00003b30: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00003b40: 640a 2d62 686d 6d2c 202d 2d62 7573 636f  d.-bhmm, --busco
+00003b50: 686d 6d2c 2074 6865 2048 4d4d 2070 726f  hmm, the HMM pro
+00003b60: 6669 6c65 2064 6174 6166 696c 6520 696e  file datafile in
+00003b70: 2074 6865 2062 7573 636f 2d67 7569 6465   the busco-guide
+00003b80: 6420 7369 6e67 6c65 2d63 6f70 7920 6765  d single-copy ge
+00003b90: 6e65 2066 616d 696c 7920 616e 616c 7973  ne family analys
+00003ba0: 6973 2c20 6465 6661 756c 7420 4e6f 6e65  is, default None
+00003bb0: 0a2d 6263 7466 2c20 2d2d 6275 7363 6f63  .-bctf, --buscoc
+00003bc0: 7574 6f66 662c 2074 6865 2048 4d4d 2073  utoff, the HMM s
+00003bd0: 636f 7265 2063 7574 6f66 6620 6461 7461  core cutoff data
+00003be0: 6669 6c65 2069 6e20 7468 6520 6275 7363  file in the busc
+00003bf0: 6f2d 6775 6964 6564 2073 696e 676c 652d  o-guided single-
+00003c00: 636f 7079 2067 656e 6520 6661 6d69 6c79  copy gene family
+00003c10: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
+00003c20: 6c74 204e 6f6e 650a 2d6f 6620 2c2d 2d6f  lt None.-of ,--o
+00003c30: 6766 6f72 6d61 742c 2066 6c61 6720 6f70  gformat, flag op
+00003c40: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00003c50: 2061 6464 2069 6e64 6578 2074 6f20 7468   add index to th
+00003c60: 6520 5242 4820 6661 6d69 6c69 6573 0a60  e RBH families.`
+00003c70: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+00003c80: 6077 6764 2066 6f63 7573 6020 6361 6e20  `wgd focus` can 
+00003c90: 7265 616c 697a 6520 7468 6520 636f 6e63  realize the conc
+00003ca0: 6174 656e 6174 696f 6e2d 6261 7365 6420  atenation-based 
+00003cb0: 616e 6420 636f 616c 6573 6365 6e63 652d  and coalescence-
+00003cc0: 6261 7365 6420 7068 796c 6f67 656e 6574  based phylogenet
+00003cd0: 6963 2069 6e66 6572 656e 6365 2061 6e64  ic inference and
+00003ce0: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+00003cf0: 7469 6e67 206f 6620 5747 4473 2065 7463  ting of WGDs etc
+00003d00: 2e0a 6060 600a 7767 6420 666f 6375 7320  ..```.wgd focus 
+00003d10: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
+00003d20: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
+00003d30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003d40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003d50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003d60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00003d70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
+00003d80: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+00003d90: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00003da0: 2064 6566 6175 6c74 2077 6764 5f66 6f63   default wgd_foc
+00003db0: 7573 0a2d 742c 202d 2d74 6d70 6469 722c  us.-t, --tmpdir,
+00003dc0: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
+00003dd0: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
+00003de0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00003df0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
+00003e00: 6e2c 2074 6865 2074 6d70 6469 7220 7769  n, the tmpdir wi
+00003e10: 6c6c 2062 6520 6173 7369 676e 6564 2072  ll be assigned r
+00003e20: 616e 646f 6d20 6e61 6d65 7320 696e 2063  andom names in c
+00003e30: 7572 7265 6e74 2064 6972 6563 746f 7279  urrent directory
+00003e40: 2061 6e64 2061 7574 6f6d 6174 656c 7920   and automately 
+00003e50: 7265 6d6f 7665 6420 6174 2074 6865 2063  removed at the c
+00003e60: 6f6d 706c 6574 696f 6e20 6f66 2070 726f  ompletion of pro
+00003e70: 6772 616d 2c20 656c 7365 2074 6865 2074  gram, else the t
+00003e80: 6d70 6469 7220 776f 756c 6420 6265 206b  mpdir would be k
+00003e90: 6570 740a 2d6e 2c20 2d2d 6e74 6872 6561  ept.-n, --nthrea
+00003ea0: 6473 2c20 7468 6520 6e75 6d62 6572 206f  ds, the number o
+00003eb0: 6620 7468 7265 6164 7320 746f 2075 7365  f threads to use
+00003ec0: 2c20 6465 6661 756c 7420 340a 2d2d 746f  , default 4.--to
+00003ed0: 5f73 746f 702c 2066 6c61 6720 6f70 7469  _stop, flag opti
+00003ee0: 6f6e 2c20 7768 6574 6865 7220 746f 2074  on, whether to t
+00003ef0: 7261 6e73 6c61 7465 2074 6872 6f75 6768  ranslate through
+00003f00: 2053 544f 5020 636f 646f 6e73 2c20 6966   STOP codons, if
+00003f10: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00003f20: 742c 2074 7261 6e73 6c61 7469 6f6e 2077  t, translation w
+00003f30: 696c 6c20 6265 2074 6572 6d69 6e61 7465  ill be terminate
+00003f40: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
+00003f50: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
+00003f60: 6f6e 2c20 656c 7365 2061 2066 756c 6c20  on, else a full 
+00003f70: 7472 616e 736c 6174 696f 6e20 636f 6e74  translation cont
+00003f80: 696e 7569 6e67 206f 6e20 7061 7374 2061  inuing on past a
+00003f90: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
+00003fa0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00003fb0: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
+00003fc0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00003fd0: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
+00003fe0: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00003ff0: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
+00004000: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
+00004010: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
+00004020: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
+00004030: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
+00004040: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
+00004050: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
+00004060: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
+00004070: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00004080: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
+00004090: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
+000040a0: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
+000040b0: 2d73 7472 6970 5f67 6170 732c 2066 6c61  -strip_gaps, fla
+000040c0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000040d0: 7220 746f 2064 726f 7020 616c 6c20 6761  r to drop all ga
+000040e0: 7073 2069 6e20 6d75 6c74 6970 6c65 2073  ps in multiple s
+000040f0: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
+00004100: 742c 2069 6620 7468 6520 666c 6167 2077  t, if the flag w
+00004110: 6173 2073 6574 2c20 616c 6c20 6761 7073  as set, all gaps
+00004120: 2077 6f75 6c64 2062 6520 6472 6f70 7065   would be droppe
+00004130: 640a 2d61 2c20 2d2d 616c 6967 6e65 722c  d.-a, --aligner,
+00004140: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
+00004150: 2070 726f 6772 616d 2074 6f20 7573 652c   program to use,
+00004160: 2064 6566 6175 6c74 206d 6166 6674 0a2d   default mafft.-
+00004170: 7472 6565 2c20 2d2d 7472 6565 5f6d 6574  tree, --tree_met
+00004180: 686f 642c 2077 6869 6368 2067 656e 6520  hod, which gene 
+00004190: 7472 6565 2069 6e66 6572 656e 6365 2070  tree inference p
+000041a0: 726f 6772 616d 2074 6f20 696e 766f 6b65  rogram to invoke
+000041b0: 2c20 6465 6661 756c 7420 6661 7374 7472  , default fasttr
+000041c0: 6565 0a2d 7473 2c20 2d2d 7472 6565 7365  ee.-ts, --treese
+000041d0: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
+000041e0: 7320 7365 7474 696e 6720 666f 7220 6765  s setting for ge
+000041f0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
+00004200: 652c 2064 6566 6175 6c74 204e 6f6e 652c  e, default None,
+00004210: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
+00004220: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
+00004230: 7469 706c 6520 7469 6d65 730a 2d2d 636f  tiple times.--co
+00004240: 6e63 6174 656e 6174 696f 6e2c 2066 6c61  ncatenation, fla
+00004250: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00004260: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00004270: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
+00004280: 6261 7365 6420 7370 6563 6965 7320 7472  based species tr
+00004290: 6565 2069 6e66 6572 656e 6365 2c20 6966  ee inference, if
+000042a0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+000042b0: 742c 2063 6f6e 6361 7465 6e61 7469 6f6e  t, concatenation
+000042c0: 2d62 6173 6564 2073 7065 6369 6573 2074  -based species t
+000042d0: 7265 6520 776f 756c 6420 6265 2069 6e66  ree would be inf
+000042e0: 6572 6564 0a2d 2d63 6f61 6c65 7363 656e  ered.--coalescen
+000042f0: 6365 2c20 666c 6167 206f 7074 696f 6e2c  ce, flag option,
+00004300: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00004310: 6961 7465 2074 6865 2063 6f61 6c65 7363  iate the coalesc
+00004320: 656e 6365 2d62 6173 6564 2073 7065 6369  ence-based speci
+00004330: 6573 2074 7265 6520 696e 6665 7265 6e63  es tree inferenc
+00004340: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+00004350: 6173 2073 6574 2c20 636f 616c 6573 6365  as set, coalesce
+00004360: 6e63 652d 6261 7365 6420 7370 6563 6965  nce-based specie
+00004370: 7320 7472 6565 2077 6f75 6c64 2062 6520  s tree would be 
+00004380: 696e 6665 7265 640a 2d73 702c 202d 2d73  infered.-sp, --s
+00004390: 7065 6369 6573 7472 6565 2c20 7370 6563  peciestree, spec
+000043a0: 6965 7320 7472 6565 2064 6174 6166 696c  ies tree datafil
+000043b0: 6520 666f 7220 6461 7469 6e67 2c20 6465  e for dating, de
+000043c0: 6661 756c 7420 4e6f 6e65 0a2d 642c 202d  fault None.-d, -
+000043d0: 2d64 6174 696e 672c 2077 6869 6368 206d  -dating, which m
+000043e0: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
+000043f0: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
+00004400: 6465 6661 756c 7420 6e6f 6e65 0a2d 6473  default none.-ds
+00004410: 2c20 2d2d 6461 7469 6e67 7365 742c 2074  , --datingset, t
+00004420: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
+00004430: 7474 696e 6720 666f 7220 6461 7469 6e67  tting for dating
+00004440: 2070 726f 6772 616d 2c20 6465 6661 756c   program, defaul
+00004450: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
+00004460: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
+00004470: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
+00004480: 6573 0a2d 6e73 2c20 2d2d 6e73 6974 6573  es.-ns, --nsites
+00004490: 2c20 7468 6520 6e73 6974 6573 2069 6e66  , the nsites inf
+000044a0: 6f72 6d61 7469 6f6e 2066 6f72 2072 3873  ormation for r8s
+000044b0: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
+000044c0: 204e 6f6e 650a 2d6f 742c 202d 2d6f 7574   None.-ot, --out
+000044d0: 6772 6f75 702c 2074 6865 206f 7574 6772  group, the outgr
+000044e0: 6f75 7020 696e 666f 726d 6174 696f 6e20  oup information 
+000044f0: 666f 7220 7238 7320 6461 7469 6e67 2c20  for r8s dating, 
+00004500: 6465 6661 756c 7420 4e6f 6e65 0a2d 7074  default None.-pt
+00004510: 2c20 2d2d 7061 7274 6974 696f 6e2c 2066  , --partition, f
+00004520: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004530: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00004540: 7061 7274 6974 696f 6e20 6461 7469 6e67  partition dating
+00004550: 2061 6e61 6c79 7369 7320 666f 7220 636f   analysis for co
+00004560: 646f 6e2c 2069 6620 7468 6520 666c 6167  don, if the flag
+00004570: 2077 6173 2073 6574 2c20 616e 2061 6464   was set, an add
+00004580: 6974 696f 6e61 6c20 7061 7274 6974 696f  itional partitio
+00004590: 6e20 6461 7469 6e67 2061 6e61 6c79 7369  n dating analysi
+000045a0: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
+000045b0: 6174 6564 0a2d 616d 2c20 2d2d 6161 6d6f  ated.-am, --aamo
+000045c0: 6465 6c2c 2077 6869 6368 2070 726f 7465  del, which prote
+000045d0: 696e 206d 6f64 656c 2074 6f20 6265 2075  in model to be u
+000045e0: 7365 6420 696e 206d 636d 6374 7265 652c  sed in mcmctree,
+000045f0: 2064 6566 6175 6c74 2070 6f69 7373 6f6e   default poisson
+00004600: 0a2d 6b73 2c20 666c 6167 206f 7074 696f  .-ks, flag optio
+00004610: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00004620: 6974 6961 7465 204b 7320 6361 6c63 756c  itiate Ks calcul
+00004630: 6174 696f 6e20 666f 7220 686f 6d6f 6c6f  ation for homolo
+00004640: 6775 6573 2069 6e20 7468 6520 7072 6f76  gues in the prov
+00004650: 6964 6564 206f 7274 686f 6c6f 676f 7573  ided orthologous
+00004660: 2067 656e 6520 6661 6d69 6c79 0a2d 2d61   gene family.--a
+00004670: 6e6e 6f74 6174 696f 6e2c 2077 6869 6368  nnotation, which
+00004680: 2061 6e6e 6f74 6174 696f 6e20 7072 6f67   annotation prog
+00004690: 7261 6d20 746f 2075 7365 2c20 6465 6661  ram to use, defa
+000046a0: 756c 7420 4e6f 6e65 0a2d 2d70 6169 7277  ult None.--pairw
+000046b0: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
+000046c0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+000046d0: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
+000046e0: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
+000046f0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00004700: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
+00004710: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
+00004720: 7374 696d 6174 6564 0a2d 6564 2c20 2d2d  stimated.-ed, --
+00004730: 6567 676e 6f67 6461 7461 2c20 7468 6520  eggnogdata, the 
+00004740: 6567 676e 6f67 2061 6e6e 6f74 6174 696f  eggnog annotatio
+00004750: 6e20 6461 7461 6669 6c65 2c20 6465 6661  n datafile, defa
+00004760: 756c 7420 4e6f 6e65 0a2d 2d70 6661 6d2c  ult None.--pfam,
+00004770: 2077 6869 6368 206f 7074 696f 6e20 746f   which option to
+00004780: 2075 7365 2066 6f72 2070 6661 6d20 616e   use for pfam an
+00004790: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
+000047a0: 7420 4e6f 6e65 0a2d 2d64 6d6e 622c 2074  t None.--dmnb, t
+000047b0: 6865 2064 6961 6d6f 6e64 2064 6174 6162  he diamond datab
+000047c0: 6173 6520 666f 7220 616e 6e6f 7461 7469  ase for annotati
+000047d0: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
+000047e0: 0a2d 2d68 6d6d 2c20 7468 6520 484d 4d20  .--hmm, the HMM 
+000047f0: 7072 6f66 696c 6520 666f 7220 616e 6e6f  profile for anno
+00004800: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
+00004810: 4e6f 6e65 0a2d 2d65 7661 6c75 652c 2074  None.--evalue, t
+00004820: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
+00004830: 6666 2066 6f72 2061 6e6e 6f74 6174 696f  ff for annotatio
+00004840: 6e2c 2064 6566 6175 6c74 2031 652d 3130  n, default 1e-10
+00004850: 0a2d 2d65 7865 7061 7468 2c20 7468 6520  .--exepath, the 
+00004860: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
+00004870: 7270 726f 7363 616e 2065 7865 6375 7461  rproscan executa
+00004880: 626c 652c 2064 6566 6175 6c74 204e 6f6e  ble, default Non
+00004890: 650a 2d66 2c20 2d2d 666f 7373 696c 2c20  e.-f, --fossil, 
+000048a0: 7468 6520 666f 7373 696c 2063 616c 6962  the fossil calib
+000048b0: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
+000048c0: 6f6e 2069 6e20 4265 6173 742c 2064 6566  on in Beast, def
+000048d0: 6175 6c74 2028 2763 6c61 6465 313b 636c  ault ('clade1;cl
+000048e0: 6164 6532 272c 2027 7461 7861 312c 7461  ade2', 'taxa1,ta
+000048f0: 7861 323b 7461 7861 332c 7461 7861 3427  xa2;taxa3,taxa4'
+00004900: 2c20 2734 3b35 272c 2027 302e 353b 302e  , '4;5', '0.5;0.
+00004910: 3627 2c20 2734 3030 3b35 3030 2729 0a2d  6', '400;500').-
+00004920: 7268 2c20 2d2d 726f 6f74 6865 6967 6874  rh, --rootheight
+00004930: 2c20 7468 6520 726f 6f74 2068 6569 6768  , the root heigh
+00004940: 7420 6361 6c69 6272 6174 696f 6e20 696e  t calibration in
+00004950: 666f 2069 6e20 4265 6173 742c 2064 6566  fo in Beast, def
+00004960: 6175 6c74 2028 342c 302e 352c 3430 3029  ault (4,0.5,400)
+00004970: 0a2d 6373 2c20 2d2d 6368 6169 6e73 6574  .-cs, --chainset
+00004980: 2c20 7468 6520 7061 7261 6d65 7465 7273  , the parameters
+00004990: 206f 6620 4d43 4d43 2063 6861 696e 2069   of MCMC chain i
+000049a0: 6e20 4265 6173 742c 2064 6566 6175 6c74  n Beast, default
+000049b0: 2028 3130 3030 302c 3130 3029 0a2d 2d62   (10000,100).--b
+000049c0: 6561 7374 6c67 6a61 722c 2074 6865 2070  eastlgjar, the p
+000049d0: 6174 6820 746f 2062 6561 7374 4c47 2e6a  ath to beastLG.j
+000049e0: 6172 2c20 6465 6661 756c 7420 4e6f 6e65  ar, default None
+000049f0: 0a2d 2d62 6561 676c 652c 2066 6c61 6720  .--beagle, flag 
+00004a00: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00004a10: 746f 2075 7365 2062 6561 676c 6520 696e  to use beagle in
+00004a20: 2042 6561 7374 2c20 6966 2074 6865 2066   Beast, if the f
+00004a30: 6c61 6720 7761 7320 7365 742c 2062 6561  lag was set, bea
+00004a40: 676c 6520 776f 756c 6420 6265 2075 7365  gle would be use
+00004a50: 640a 2d2d 7072 6f74 636f 6364 6174 696e  d.--protcocdatin
+00004a60: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
+00004a70: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+00004a80: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
+00004a90: 7465 696e 2d63 6f6e 6361 7465 6e61 7469  tein-concatenati
+00004aa0: 6f6e 2d62 6173 6564 2064 6174 696e 6720  on-based dating 
+00004ab0: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
+00004ac0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00004ad0: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
+00004ae0: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
+00004af0: 2d70 726f 7464 6174 696e 672c 2066 6c61  -protdating, fla
+00004b00: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00004b10: 7220 746f 206f 6e6c 7920 696e 6974 6961  r to only initia
+00004b20: 7465 2074 6865 2070 726f 7465 696e 2d62  te the protein-b
+00004b30: 6173 6564 2064 6174 696e 6720 616e 616c  ased dating anal
+00004b40: 7973 6973 2c20 6966 2074 6865 2066 6c61  ysis, if the fla
+00004b50: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
+00004b60: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
+00004b70: 2069 6e69 7469 6174 6564 0a60 6060 0a0a   initiated.```..
+00004b80: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
+00004b90: 206b 7364 6020 6361 6e20 7265 616c 697a   ksd` can realiz
+00004ba0: 6520 7468 6520 636f 6e73 7472 7563 7469  e the constructi
+00004bb0: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
+00004bc0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00004bd0: 6275 7469 6f6e 2061 6e64 2072 6174 6520  bution and rate 
+00004be0: 636f 7272 6563 7469 6f6e 2e0a 6060 600a  correction..```.
+00004bf0: 7767 6420 6b73 6420 6661 6d69 6c69 6573  wgd ksd families
+00004c00: 2073 6571 7565 6e63 6573 2028 6f70 7469   sequences (opti
+00004c10: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
+00004c20: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004c30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004c40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004c50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004c60: 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574 6469  ----.-o, --outdi
+00004c70: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
+00004c80: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
+00004c90: 2077 6764 5f6b 7364 0a2d 742c 202d 2d74   wgd_ksd.-t, --t
+00004ca0: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
+00004cb0: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
+00004cc0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
+00004cd0: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
+00004ce0: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
+00004cf0: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
+00004d00: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
+00004d10: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
+00004d20: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
+00004d30: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
+00004d40: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
+00004d50: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
+00004d60: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+00004d70: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
+00004d80: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
+00004d90: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
+00004da0: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00004db0: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
+00004dc0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00004dd0: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
+00004de0: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
+00004df0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
+00004e00: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
+00004e10: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
+00004e20: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
+00004e30: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
+00004e40: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
+00004e50: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
+00004e60: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
+00004e70: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
+00004e80: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
+00004e90: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
+00004ea0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004eb0: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
+00004ec0: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
+00004ed0: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
+00004ee0: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
+00004ef0: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
+00004f00: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
+00004f10: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
+00004f20: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
+00004f30: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
+00004f40: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
+00004f50: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
+00004f60: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
+00004f70: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00004f80: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
+00004f90: 6174 6564 0a2d 2d70 6169 7277 6973 652c  ated.--pairwise,
+00004fa0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004fb0: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00004fc0: 6520 7061 6972 7769 7365 204b 7320 6573  e pairwise Ks es
+00004fd0: 7469 6d61 7469 6f6e 2c20 6966 2074 6865  timation, if the
+00004fe0: 2066 6c61 6720 7761 7320 7365 742c 2070   flag was set, p
+00004ff0: 6169 7277 6973 6520 4b73 2076 616c 7565  airwise Ks value
+00005000: 7320 776f 756c 6420 6265 2065 7374 696d  s would be estim
+00005010: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
+00005020: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00005030: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
+00005040: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
+00005050: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
+00005060: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
+00005070: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
+00005080: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
+00005090: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
+000050a0: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
+000050b0: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
+000050c0: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
+000050d0: 6166 6674 200a 2d74 7265 652c 202d 2d74  afft .-tree, --t
+000050e0: 7265 655f 6d65 7468 6f64 2c20 7768 6963  ree_method, whic
+000050f0: 6820 6765 6e65 2074 7265 6520 696e 6665  h gene tree infe
+00005100: 7265 6e63 6520 7072 6f67 7261 6d20 746f  rence program to
+00005110: 2069 6e76 6f6b 652c 2064 6566 6175 6c74   invoke, default
+00005120: 2066 6173 7474 7265 650a 2d2d 7472 6565   fasttree.--tree
+00005130: 5f6f 7074 696f 6e73 2c20 6f70 7469 6f6e  _options, option
+00005140: 7320 696e 2074 7265 6520 696e 6665 7265  s in tree infere
+00005150: 6e63 6520 6173 2061 2063 6f6d 6d61 2073  nce as a comma s
+00005160: 6570 6172 6174 6564 2073 7472 696e 672c  eparated string,
+00005170: 2064 6566 6175 6c74 204e 6f6e 650a 2d2d   default None.--
+00005180: 6e6f 6465 5f61 7665 7261 6765 2c20 666c  node_average, fl
+00005190: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+000051a0: 6572 2074 6f20 696e 6974 6961 7465 206e  er to initiate n
+000051b0: 6f64 652d 6176 6572 6167 6520 7761 7920  ode-average way 
+000051c0: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
+000051d0: 2069 6e73 7465 6164 206f 6620 6e6f 6465   instead of node
+000051e0: 2d77 6569 6768 7465 642c 2069 6620 7468  -weighted, if th
+000051f0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00005200: 7468 6520 6e6f 6465 2d61 7665 7261 6769  the node-averagi
+00005210: 6e67 2064 652d 7265 6475 6e64 616e 6379  ng de-redundancy
+00005220: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00005230: 7465 640a 2d73 722c 202d 2d73 7061 6972  ted.-sr, --spair
+00005240: 2c20 7468 6520 7370 6563 6965 7320 7061  , the species pa
+00005250: 6972 2074 6f20 6265 2070 6c6f 7474 6564  ir to be plotted
+00005260: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00005270: 7468 6973 206f 7074 696f 6e20 6361 6e20  this option can 
+00005280: 6265 2070 726f 7669 6465 6420 6d75 6c74  be provided mult
+00005290: 6970 6c65 2074 696d 6573 0a2d 7370 2c20  iple times.-sp, 
+000052a0: 2d2d 7370 6563 6965 7374 7265 652c 2074  --speciestree, t
+000052b0: 6865 2073 7065 6369 6573 2074 7265 6520  he species tree 
+000052c0: 746f 2070 6572 666f 726d 2072 6174 6520  to perform rate 
+000052d0: 636f 7272 6563 7469 6f6e 2c20 6465 6661  correction, defa
+000052e0: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
+000052f0: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
 00005300: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00005310: 2075 7369 6e67 2074 6865 206f 7574 6772   using the outgr
-00005320: 6f75 7020 6174 2072 6f6f 7420 6173 206f  oup at root as o
-00005330: 7574 6772 6f75 702c 2069 6620 7468 6520  utgroup, if the 
-00005340: 666c 6167 2077 6173 2073 6574 2c20 6f6e  flag was set, on
-00005350: 6c79 2074 6865 206f 7574 6772 6f75 7020  ly the outgroup 
-00005360: 6174 2072 6f6f 7420 776f 756c 6420 6265  at root would be
-00005370: 2075 7365 6420 6173 206f 7574 6772 6f75   used as outgrou
-00005380: 700a 2d65 706b 2c20 2d2d 6578 7472 6170  p.-epk, --extrap
-00005390: 6172 616e 6f6d 656b 732c 2065 7874 7261  aranomeks, extra
-000053a0: 2070 6172 616e 6f6d 6520 4b73 2064 6174   paranome Ks dat
-000053b0: 6120 746f 2070 6c6f 7420 696e 2074 6865  a to plot in the
-000053c0: 206d 6978 6564 204b 7320 6469 7374 7269   mixed Ks distri
-000053d0: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
-000053e0: 4e6f 6e65 0a2d 6170 2c20 2d2d 616e 6368  None.-ap, --anch
-000053f0: 6f72 706f 696e 7473 2c20 616e 6368 6f72  orpoints, anchor
-00005400: 706f 696e 7473 2e74 7874 2066 696c 6520  points.txt file 
-00005410: 746f 2070 6c6f 7420 616e 6368 6f72 204b  to plot anchor K
-00005420: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
-00005430: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00005440: 6465 6661 756c 7420 4e6f 6e65 0a2d 706b  default None.-pk
-00005450: 2c20 2d2d 706c 6f74 6b64 652c 2066 6c61  , --plotkde, fla
-00005460: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005470: 7220 746f 2070 6c6f 7420 6b64 6520 6375  r to plot kde cu
-00005480: 7276 6520 6f66 206f 7274 686f 6c6f 676f  rve of orthologo
-00005490: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
-000054a0: 6f6e 206f 7665 7220 6869 7374 6f67 7261  on over histogra
-000054b0: 6d20 696e 2074 6865 206d 6978 6564 204b  m in the mixed K
-000054c0: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-000054d0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-000054e0: 7365 742c 2074 6865 206b 6465 2063 7572  set, the kde cur
-000054f0: 7665 2077 6f75 6c64 2062 6520 706c 6f74  ve would be plot
-00005500: 7465 640a 2d70 6167 2c20 2d2d 706c 6f74  ted.-pag, --plot
-00005510: 6170 676d 6d2c 2066 6c61 6720 6f70 7469  apgmm, flag opti
-00005520: 6f6e 2c20 7768 6574 6865 7220 746f 2070  on, whether to p
-00005530: 6572 666f 726d 2061 6e64 2070 6c6f 7420  erform and plot 
-00005540: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00005550: 206f 6620 616e 6368 6f72 204b 7320 696e   of anchor Ks in
-00005560: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
-00005570: 7374 7269 6275 7469 6f6e 2c20 6966 2074  stribution, if t
-00005580: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00005590: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
-000055a0: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
-000055b0: 4b73 2077 6f75 6c64 2062 6520 706c 6f74  Ks would be plot
-000055c0: 7465 640a 2d70 656d 2c20 2d2d 706c 6f74  ted.-pem, --plot
-000055d0: 656c 6d6d 2c20 666c 6167 206f 7074 696f  elmm, flag optio
-000055e0: 6e2c 2077 6865 7468 6572 2074 6f20 7065  n, whether to pe
-000055f0: 7266 6f72 6d20 616e 6420 706c 6f74 2065  rform and plot e
-00005600: 6c6d 6d20 6d69 7874 7572 6520 6d6f 6465  lmm mixture mode
-00005610: 6c69 6e67 206f 6620 7061 7261 6e6f 6d65  ling of paranome
-00005620: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
-00005630: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005640: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005650: 7320 7365 742c 2074 6865 2065 6c6d 6d20  s set, the elmm 
-00005660: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00005670: 206f 6620 7061 7261 6e6f 6d65 204b 7320   of paranome Ks 
-00005680: 776f 756c 6420 6265 2070 6c6f 7474 6564  would be plotted
-00005690: 0a2d 632c 202d 2d63 6f6d 706f 6e65 6e74  .-c, --component
-000056a0: 732c 2074 6865 2072 616e 6765 206f 6620  s, the range of 
-000056b0: 7468 6520 6e75 6d62 6572 206f 6620 636f  the number of co
-000056c0: 6d70 6f6e 656e 7473 2074 6f20 6669 7420  mponents to fit 
-000056d0: 696e 2061 6e63 686f 7220 4b73 206d 6978  in anchor Ks mix
-000056e0: 7475 7265 206d 6f64 656c 696e 672c 2064  ture modeling, d
-000056f0: 6566 6175 6c74 2028 312c 3429 0a2d 786c  efault (1,4).-xl
-00005700: 2c20 2d2d 786c 696d 2c20 7468 6520 7820  , --xlim, the x 
-00005710: 6178 6973 206c 696d 6974 206f 6620 4b73  axis limit of Ks
-00005720: 2064 6973 7472 6962 7574 696f 6e0a 2d79   distribution.-y
-00005730: 6c2c 202d 2d79 6c69 6d2c 2074 6865 2079  l, --ylim, the y
-00005740: 2061 7869 7320 6c69 6d69 7420 6f66 204b   axis limit of K
-00005750: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-00005760: 6164 6f2c 202d 2d61 646a 7573 746f 7274  ado, --adjustort
-00005770: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
-00005780: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
-00005790: 7374 2074 6865 2068 6973 746f 6772 616d  st the histogram
-000057a0: 2068 6569 6768 7420 6f66 206f 7274 686f   height of ortho
-000057b0: 6c6f 676f 7573 204b 7320 6173 2074 6f20  logous Ks as to 
-000057c0: 6d61 7463 6820 7468 6520 6865 6967 6874  match the height
-000057d0: 206f 6620 7061 7261 6c6f 676f 7573 204b   of paralogous K
-000057e0: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-000057f0: 6173 2073 6574 2c20 7468 6520 6164 6a75  as set, the adju
-00005800: 7374 6d65 6e74 2077 6f75 6c64 2062 6520  stment would be 
-00005810: 636f 6e64 7563 7465 640a 2d61 6466 2c20  conducted.-adf, 
-00005820: 2d2d 6164 6a75 7374 6661 6374 6f72 2c20  --adjustfactor, 
-00005830: 7468 6520 6164 6a75 7374 6d65 6e74 2066  the adjustment f
-00005840: 6163 746f 7220 6f66 206f 7274 686f 6c6f  actor of ortholo
-00005850: 676f 7573 204b 732c 2064 6566 6175 6c74  gous Ks, default
-00005860: 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c   0.5.-oa, --okal
-00005870: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
-00005880: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-00005890: 4b73 2064 6973 7472 6962 7574 696f 6e20  Ks distribution 
-000058a0: 696e 206d 6978 6564 2070 6c6f 742c 2064  in mixed plot, d
-000058b0: 6566 6175 6c74 2030 2e35 0a2d 6661 2c20  efault 0.5.-fa, 
-000058c0: 2d2d 666f 6375 7332 616c 6c2c 2073 6574  --focus2all, set
-000058d0: 2066 6f63 616c 2073 7065 6369 6573 2061   focal species a
-000058e0: 6e64 206c 6574 2073 7065 6369 6573 2070  nd let species p
-000058f0: 6169 7220 746f 2062 6520 6265 7477 6565  air to be betwee
-00005900: 6e20 666f 6361 6c20 616e 6420 616c 6c20  n focal and all 
-00005910: 7468 6520 7265 6d61 696e 696e 6720 7370  the remaining sp
-00005920: 6563 6965 732c 2064 6566 6175 6c74 204e  ecies, default N
-00005930: 6f6e 650a 2d6b 732c 202d 2d6b 7374 7265  one.-ks, --kstre
-00005940: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
-00005950: 7768 6574 6865 7220 746f 2069 6e66 6572  whether to infer
-00005960: 204b 7320 7472 6565 2c20 6966 2074 6865   Ks tree, if the
-00005970: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00005980: 6865 204b 7320 7472 6565 2069 6e66 6572  he Ks tree infer
-00005990: 656e 6365 2061 6e61 6c79 7369 7320 776f  ence analysis wo
-000059a0: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
-000059b0: 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e  .-ock, --onlycon
-000059c0: 6361 746b 7374 7265 652c 2066 6c61 6720  catkstree, flag 
-000059d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000059e0: 746f 206f 6e6c 7920 696e 6665 7220 4b73  to only infer Ks
-000059f0: 2074 7265 6520 756e 6465 7220 636f 6e63   tree under conc
-00005a00: 6174 656e 6174 6564 2061 6c69 676e 6d65  atenated alignme
-00005a10: 6e74 2c20 6966 2074 6865 2066 6c61 6720  nt, if the flag 
-00005a20: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
-00005a30: 6520 4b73 2074 7265 6520 756e 6465 7220  e Ks tree under 
-00005a40: 636f 6e63 6174 656e 6174 6564 2061 6c69  concatenated ali
-00005a50: 676e 6d65 6e74 2077 6f75 6c64 2062 6520  gnment would be 
-00005a60: 6361 6c63 756c 6174 6564 0a2d 6373 2c20  calculated.-cs, 
-00005a70: 2d2d 636c 6173 7369 632c 2066 6c61 6720  --classic, flag 
-00005a80: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00005a90: 746f 2064 7261 7720 6d69 7865 6420 4b73  to draw mixed Ks
-00005aa0: 2070 6c6f 7420 696e 2061 2063 6c61 7373   plot in a class
-00005ab0: 6963 206d 616e 6e65 7220 7768 6572 6520  ic manner where 
-00005ac0: 7468 6520 6675 6c6c 206f 7274 686f 6c6f  the full ortholo
-00005ad0: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
-00005ae0: 7469 6f6e 2069 7320 6472 6177 6564 2c20  tion is drawed, 
-00005af0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00005b00: 7365 742c 2074 6865 2063 6c61 7373 6963  set, the classic
-00005b10: 206d 6978 6564 204b 7320 706c 6f74 2077   mixed Ks plot w
-00005b20: 6f75 6c64 2062 6520 6472 6177 6e0a 2d74  ould be drawn.-t
-00005b30: 612c 202d 2d74 6f70 6172 726f 772c 2066  a, --toparrow, f
-00005b40: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005b50: 6865 7220 746f 2061 646a 7573 7420 7468  her to adjust th
-00005b60: 6520 6172 726f 7720 746f 2062 6520 6174  e arrow to be at
-00005b70: 2074 6865 2074 6f70 206f 6620 7468 6520   the top of the 
-00005b80: 706c 6f74 2c20 696e 7374 6561 6420 6f66  plot, instead of
-00005b90: 2062 6569 6e67 2063 6f6f 7264 696e 6174   being coordinat
-00005ba0: 6564 2061 7320 7468 6520 4b44 4520 6f66  ed as the KDE of
-00005bb0: 2074 6865 206f 7274 686f 6c6f 676f 7573   the orthologous
-00005bc0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005bd0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005be0: 7320 7365 742c 2074 6865 2061 7272 6f77  s set, the arrow
-00005bf0: 2077 6f75 6c64 2062 6520 7365 7420 6174   would be set at
-00005c00: 2074 6865 2074 6f70 0a2d 6273 2c20 2d2d   the top.-bs, --
-00005c10: 626f 6f74 7374 7261 702c 2074 6865 206e  bootstrap, the n
-00005c20: 756d 6265 7220 6f66 2062 6f6f 7473 7472  umber of bootstr
-00005c30: 6170 2072 6570 6c69 6361 7465 7320 6f66  ap replicates of
-00005c40: 206f 7274 686f 6c6f 6720 4b73 2064 6973   ortholog Ks dis
-00005c50: 7472 6962 7574 696f 6e20 696e 206d 6978  tribution in mix
-00005c60: 6564 2070 6c6f 740a 6060 600a 0a54 6865  ed plot.```..The
-00005c70: 2070 726f 6772 616d 2060 7767 6420 6d69   program `wgd mi
-00005c80: 7860 2063 616e 2072 6561 6c69 7a65 2074  x` can realize t
-00005c90: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
-00005ca0: 2063 6c75 7374 6572 696e 6720 616e 616c   clustering anal
-00005cb0: 7973 6973 206f 6620 2a4b 2a3c 7375 623e  ysis of *K*<sub>
-00005cc0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00005cd0: 7269 6275 7469 6f6e 2e0a 6060 600a 7767  ribution..```.wg
-00005ce0: 6420 6d69 7820 6b73 5f64 6174 6166 696c  d mix ks_datafil
-00005cf0: 6520 286f 7074 696f 6e29 0a2d 2d2d 2d2d  e (option).-----
-00005d00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005d10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005d20: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005d30: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005d40: 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20  -----------.-f, 
-00005d50: 2d2d 6669 6c74 6572 732c 2074 6865 2063  --filters, the c
-00005d60: 7574 6f66 6620 616c 6967 6e6d 656e 7420  utoff alignment 
-00005d70: 6c65 6e67 7468 2c20 6465 6661 756c 7420  length, default 
-00005d80: 3330 300a 2d72 2c20 2d2d 6b73 5f72 616e  300.-r, --ks_ran
-00005d90: 6765 2c20 7468 6520 4b73 2072 616e 6765  ge, the Ks range
-00005da0: 2074 6f20 6265 2063 6f6e 7369 6465 7265   to be considere
-00005db0: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
-00005dc0: 290a 2d62 2c20 2d2d 6269 6e73 2c20 7468  ).-b, --bins, th
-00005dd0: 6520 6e75 6d62 6572 206f 6620 6269 6e73  e number of bins
-00005de0: 2069 6e20 4b73 2064 6973 7472 6962 7574   in Ks distribut
-00005df0: 696f 6e2c 2064 6566 6175 6c74 2035 300a  ion, default 50.
-00005e00: 2d6f 2c20 2d2d 6f75 7464 6972 2c20 7468  -o, --outdir, th
-00005e10: 6520 6f75 7470 7574 2064 6972 6563 746f  e output directo
-00005e20: 7279 2c20 6465 6661 756c 7420 7767 645f  ry, default wgd_
-00005e30: 6d69 780a 2d2d 6d65 7468 6f64 2c20 7768  mix.--method, wh
-00005e40: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
-00005e50: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
-00005e60: 7420 676d 6d0a 2d6e 2c20 2d2d 636f 6d70  t gmm.-n, --comp
-00005e70: 6f6e 656e 7473 2c20 7468 6520 7261 6e67  onents, the rang
-00005e80: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-00005e90: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-00005ea0: 2066 6974 2c20 6465 6661 756c 7420 2831   fit, default (1
-00005eb0: 2c20 3429 0a2d 672c 202d 2d67 616d 6d61  , 4).-g, --gamma
-00005ec0: 2c20 7468 6520 6761 6d6d 6120 7061 7261  , the gamma para
-00005ed0: 6d65 7465 7220 666f 7220 6267 6d6d 206d  meter for bgmm m
-00005ee0: 6f64 656c 732c 2064 6566 6175 6c74 2030  odels, default 0
-00005ef0: 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f 696e  .001.-ni, --n_in
-00005f00: 6974 2c20 7468 6520 6e75 6d62 6572 206f  it, the number o
-00005f10: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
-00005f20: 6c69 7a61 7469 6f6e 732c 2064 6566 6175  lizations, defau
-00005f30: 6c74 2032 3030 0a2d 6d69 2c20 2d2d 6d61  lt 200.-mi, --ma
-00005f40: 785f 6974 6572 2c20 7468 6520 6d61 7869  x_iter, the maxi
-00005f50: 6d75 6d20 6e75 6d62 6572 206f 6620 6974  mum number of it
-00005f60: 6572 6174 696f 6e73 2c20 6465 6661 756c  erations, defaul
-00005f70: 7420 3230 300a 6060 600a 0a54 6865 2070  t 200.```..The p
-00005f80: 726f 6772 616d 2060 7767 6420 7065 616b  rogram `wgd peak
-00005f90: 6020 6361 6e20 7265 616c 697a 6520 7468  ` can realize th
-00005fa0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
-00005fb0: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
-00005fc0: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
-00005fd0: 2069 6e20 5747 4420 6461 7469 6e67 2e0a   in WGD dating..
-00005fe0: 6060 600a 7767 6420 7065 616b 206b 735f  ```.wgd peak ks_
-00005ff0: 6461 7461 6669 6c65 2028 6f70 7469 6f6e  datafile (option
-00006000: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-00006010: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006020: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006030: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006040: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006050: 2d2d 0a2d 6170 2c20 2d2d 616e 6368 6f72  --.-ap, --anchor
-00006060: 706f 696e 7473 2c20 7468 6520 616e 6368  points, the anch
-00006070: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
-00006080: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
-00006090: 0a2d 736d 2c20 2d2d 7365 676d 656e 7473  .-sm, --segments
-000060a0: 2c20 7468 6520 7365 676d 656e 7473 2064  , the segments d
-000060b0: 6174 6166 696c 652c 2064 6566 6175 6c74  atafile, default
-000060c0: 204e 6f6e 650a 2d6c 652c 202d 2d6c 6973   None.-le, --lis
-000060d0: 7465 6c65 6d65 6e74 732c 2074 6865 206c  telements, the l
-000060e0: 6973 7473 6567 6d65 6e74 7320 6461 7461  istsegments data
-000060f0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00006100: 6e65 200a 2d6d 702c 202d 2d6d 756c 7469  ne .-mp, --multi
-00006110: 706c 6963 6f6e 7061 6972 732c 2074 6865  pliconpairs, the
-00006120: 206d 756c 7469 706c 6963 6f6e 7061 6972   multipliconpair
-00006130: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
-00006140: 756c 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f  ult None.-o, --o
-00006150: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
-00006160: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
-00006170: 6175 6c74 2077 6764 5f70 6561 6b0a 2d61  ault wgd_peak.-a
-00006180: 662c 202d 2d61 6c69 676e 6669 6c74 6572  f, --alignfilter
-00006190: 2c20 6375 746f 6666 2066 6f72 2061 6c69  , cutoff for ali
-000061a0: 676e 6d65 6e74 2069 6465 6e74 6974 792c  gnment identity,
-000061b0: 206c 656e 6774 6820 616e 6420 636f 7665   length and cove
-000061c0: 7261 6765 2c20 6465 6661 756c 7420 302e  rage, default 0.
-000061d0: 302c 2030 2c20 302e 300a 2d72 2c20 2d2d  0, 0, 0.0.-r, --
-000061e0: 6b73 7261 6e67 652c 2072 616e 6765 206f  ksrange, range o
-000061f0: 6620 4b73 2074 6f20 6265 2061 6e61 6c79  f Ks to be analy
-00006200: 7a65 642c 2064 6566 6175 6c74 2028 302c  zed, default (0,
-00006210: 2035 290a 2d62 772c 202d 2d62 696e 5f77   5).-bw, --bin_w
-00006220: 6964 7468 2c20 6261 6e64 7769 6474 6820  idth, bandwidth 
-00006230: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
-00006240: 6f6e 2c20 6465 6661 756c 7420 302e 310a  on, default 0.1.
-00006250: 2d69 632c 202d 2d77 6569 6768 7473 5f6f  -ic, --weights_o
-00006260: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
-00006270: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00006280: 6865 7468 6572 2074 6f20 696e 636c 7564  hether to includ
-00006290: 6520 4b73 206f 7574 6c69 6572 732c 2069  e Ks outliers, i
-000062a0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-000062b0: 6574 2c20 4b73 206f 7574 6c69 6572 7320  et, Ks outliers 
-000062c0: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
-000062d0: 6420 696e 2074 6865 2061 6e61 6c79 7369  d in the analysi
-000062e0: 730a 2d6d 2c20 2d2d 6d65 7468 6f64 2c20  s.-m, --method, 
-000062f0: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
-00006300: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
-00006310: 756c 7420 676d 6d0a 2d2d 7365 6564 2c20  ult gmm.--seed, 
-00006320: 7261 6e64 6f6d 2073 6565 6420 6769 7665  random seed give
-00006330: 6e20 746f 2069 6e69 7469 616c 697a 6174  n to initializat
-00006340: 696f 6e2c 2064 6566 6175 6c74 2032 3335  ion, default 235
-00006350: 3238 3930 0a2d 6569 2c20 2d2d 656d 5f69  2890.-ei, --em_i
-00006360: 7465 722c 2074 6865 206e 756d 6265 7220  ter, the number 
-00006370: 6f66 2045 4d20 6974 6572 6174 696f 6e73  of EM iterations
-00006380: 2074 6f20 7065 7266 6f72 6d2c 2064 6566   to perform, def
-00006390: 6175 6c74 2032 3030 0a2d 6e69 2c20 2d2d  ault 200.-ni, --
-000063a0: 6e5f 696e 6974 2c20 7468 6520 6e75 6d62  n_init, the numb
-000063b0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
-000063c0: 6974 6961 6c69 7a61 7469 6f6e 732c 2064  itializations, d
-000063d0: 6566 6175 6c74 2032 3030 0a2d 6e2c 202d  efault 200.-n, -
-000063e0: 2d63 6f6d 706f 6e65 6e74 732c 2074 6865  -components, the
-000063f0: 2072 616e 6765 206f 6620 7468 6520 6e75   range of the nu
-00006400: 6d62 6572 206f 6620 636f 6d70 6f6e 656e  mber of componen
-00006410: 7473 2074 6f20 6669 742c 2064 6566 6175  ts to fit, defau
-00006420: 6c74 2028 312c 2034 290a 2d67 2c20 2d2d  lt (1, 4).-g, --
-00006430: 6761 6d6d 612c 2074 6865 2067 616d 6d61  gamma, the gamma
-00006440: 2070 6172 616d 6574 6572 2066 6f72 2062   parameter for b
-00006450: 676d 6d20 6d6f 6465 6c73 2c20 6465 6661  gmm models, defa
-00006460: 756c 7420 3165 2d33 0a2d 2d62 6f6f 7473  ult 1e-3.--boots
-00006470: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
-00006480: 626f 6f74 7374 7261 7020 7265 706c 6963  bootstrap replic
-00006490: 6174 6573 206f 6620 6b64 652c 2064 6566  ates of kde, def
-000064a0: 6175 6c74 2032 3030 0a2d 2d77 6569 6768  ault 200.--weigh
-000064b0: 7465 642c 2066 6c61 6720 6f70 7469 6f6e  ted, flag option
-000064c0: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
-000064d0: 206e 6f64 652d 7765 6967 6874 6564 206d   node-weighted m
-000064e0: 6574 686f 6420 6f66 2064 652d 7265 6475  ethod of de-redu
-000064f0: 6e64 616e 6379 2c20 6966 2074 6865 2066  ndancy, if the f
-00006500: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
-00006510: 206e 6f64 652d 7765 6967 6874 6564 206d   node-weighted m
-00006520: 6574 686f 6420 776f 756c 6420 6265 2075  ethod would be u
-00006530: 7365 640a 2d70 2c20 2d2d 706c 6f74 2c20  sed.-p, --plot, 
-00006540: 7468 6520 706c 6f74 7469 6e67 206d 6574  the plotting met
-00006550: 686f 6420 746f 2062 6520 7573 6564 2c20  hod to be used, 
-00006560: 6465 6661 756c 7420 6964 656e 7469 6361  default identica
-00006570: 6c0a 2d62 6d2c 202d 2d62 775f 6d65 7468  l.-bm, --bw_meth
-00006580: 6f64 2c20 7468 6520 6261 6e64 7769 6474  od, the bandwidt
-00006590: 6820 6d65 7468 6f64 2074 6f20 6265 2075  h method to be u
-000065a0: 7365 6420 696e 2061 6e61 6c79 7a69 6e67  sed in analyzing
-000065b0: 2074 6865 2070 6561 6b20 6f66 2057 4744   the peak of WGD
-000065c0: 2064 6174 6573 2c20 6465 6661 756c 7420   dates, default 
-000065d0: 7369 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65  silverman.--n_me
-000065e0: 646f 6964 732c 2074 6865 206e 756d 6265  doids, the numbe
-000065f0: 7220 6f66 206d 6564 6f69 6473 2074 6f20  r of medoids to 
-00006600: 6669 742c 2064 6566 6175 6c74 2032 0a2d  fit, default 2.-
-00006610: 6b6d 2c20 2d2d 6b64 656d 6574 686f 642c  km, --kdemethod,
-00006620: 2074 6865 206b 6465 206d 6574 686f 6420   the kde method 
-00006630: 746f 2062 6520 7573 6564 2069 6e20 616e  to be used in an
-00006640: 616c 797a 696e 6720 7468 6520 7065 616b  alyzing the peak
-00006650: 206f 6620 5747 4420 6461 7465 732c 206b   of WGD dates, k
-00006660: 6d65 646f 6964 7320 616e 616c 7973 6973  medoids analysis
-00006670: 206f 7220 7468 6520 6261 7369 6320 4b73   or the basic Ks
-00006680: 2070 6c6f 7474 696e 672c 2064 6566 6175   plotting, defau
-00006690: 6c74 2073 6369 7079 0a2d 2d6e 5f63 6c75  lt scipy.--n_clu
-000066a0: 7374 6572 732c 2074 6865 206e 756d 6265  sters, the numbe
-000066b0: 7220 6f66 2063 6c75 7374 6572 7320 746f  r of clusters to
-000066c0: 2070 6c6f 7420 456c 626f 7720 6c6f 7373   plot Elbow loss
-000066d0: 2066 756e 6374 696f 6e2c 2064 6566 6175   function, defau
-000066e0: 6c74 2035 0a2d 2d6b 6d65 646f 6964 732c  lt 5.--kmedoids,
-000066f0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00006700: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00006710: 6520 4b2d 4d65 646f 6964 7320 636c 7573  e K-Medoids clus
-00006720: 7465 7269 6e67 2061 6e61 6c79 7369 732c  tering analysis,
-00006730: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00006740: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00006750: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00006760: 6961 7465 640a 2d67 642c 202d 2d67 7569  iated.-gd, --gui
-00006770: 6465 2c20 7468 6520 7265 6769 6d65 2072  de, the regime r
-00006780: 6573 6964 696e 6720 616e 6368 6f72 732c  esiding anchors,
-00006790: 2064 6566 6175 6c74 2073 6567 6d65 6e74   default segment
-000067a0: 0a2d 7072 6374 2c20 2d2d 7072 6f6d 696e  .-prct, --promin
-000067b0: 656e 6365 5f63 7574 6f66 662c 2074 6865  ence_cutoff, the
-000067c0: 2070 726f 6d69 6e65 6e63 6520 6375 746f   prominence cuto
-000067d0: 6666 206f 6620 6163 6365 7074 6162 6c65  ff of acceptable
-000067e0: 2070 6561 6b73 2069 6e20 7065 616b 2066   peaks in peak f
-000067f0: 696e 6469 6e67 2073 7465 7073 2c20 6465  inding steps, de
-00006800: 6661 756c 7420 302e 310a 2d72 682c 202d  fault 0.1.-rh, -
-00006810: 2d72 656c 5f68 6569 6768 742c 2074 6865  -rel_height, the
-00006820: 2072 656c 6174 6976 6520 6865 6967 6874   relative height
-00006830: 2061 7420 7768 6963 6820 7468 6520 7065   at which the pe
-00006840: 616b 2077 6964 7468 2069 7320 6d65 6173  ak width is meas
-00006850: 7572 6564 2c20 6465 6661 756c 7420 302e  ured, default 0.
-00006860: 340a 2d6b 642c 202d 2d6b 7374 6f64 6174  4.-kd, --kstodat
-00006870: 652c 2074 6865 2072 616e 6765 206f 6620  e, the range of 
-00006880: 4b73 2074 6f20 6265 2064 6174 6564 2069  Ks to be dated i
-00006890: 6e20 6865 7572 6973 7469 6320 7365 6172  n heuristic sear
-000068a0: 6368 2c20 6465 6661 756c 7420 2830 2e35  ch, default (0.5
-000068b0: 2c20 312e 3529 0a2d 786c 2c20 2d2d 786c  , 1.5).-xl, --xl
-000068c0: 696d 2c20 7468 6520 7820 6178 6973 206c  im, the x axis l
-000068d0: 696d 6974 206f 6620 474d 4d20 4b73 2064  imit of GMM Ks d
-000068e0: 6973 7472 6962 7574 696f 6e0a 2d79 6c2c  istribution.-yl,
-000068f0: 202d 2d79 6c69 6d2c 2074 6865 2079 2061   --ylim, the y a
-00006900: 7869 7320 6c69 6d69 7420 6f66 2047 4d4d  xis limit of GMM
-00006910: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00006920: 0a2d 2d6d 616e 7561 6c73 6574 2c20 666c  .--manualset, fl
-00006930: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00006940: 6572 2074 6f20 6f75 7470 7574 2061 6e63  er to output anc
-00006950: 686f 7220 7061 6972 7320 7769 7468 206d  hor pairs with m
-00006960: 616e 7561 6c6c 7920 7365 7420 4b73 2072  anually set Ks r
-00006970: 616e 6765 2c20 6966 2074 6865 2066 6c61  ange, if the fla
-00006980: 6720 7761 7320 7365 742c 206d 616e 7561  g was set, manua
-00006990: 6c6c 7920 7365 7420 4b73 2072 616e 6765  lly set Ks range
-000069a0: 2077 6f75 6c64 2062 6520 6f75 7470 7574   would be output
-000069b0: 7465 640a 2d2d 6369 2c20 7468 6520 636f  ted.--ci, the co
-000069c0: 6e66 6964 656e 6365 206c 6576 656c 206f  nfidence level o
-000069d0: 6620 6c6f 672d 6e6f 726d 616c 2064 6973  f log-normal dis
-000069e0: 7472 6962 7574 696f 6e20 746f 2064 6174  tribution to dat
-000069f0: 652c 2064 6566 6175 6c74 2039 350a 2d2d  e, default 95.--
-00006a00: 6864 722c 2074 6865 2068 6967 6865 7374  hdr, the highest
-00006a10: 2064 656e 7369 7479 2072 6567 696f 6e20   density region 
-00006a20: 2848 4452 2920 6170 706c 6965 6420 696e  (HDR) applied in
-00006a30: 2074 6865 2073 6567 6d65 6e74 2d67 7569   the segment-gui
-00006a40: 6465 6420 616e 6368 6f72 2070 6169 7220  ded anchor pair 
-00006a50: 4b73 2064 6973 7472 6962 7574 696f 6e2c  Ks distribution,
-00006a60: 2064 6566 6175 6c74 2039 350a 2d2d 6865   default 95.--he
-00006a70: 7572 6973 7469 632c 2066 6c61 6720 6f70  uristic, flag op
-00006a80: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00006a90: 2069 6e69 7469 6174 6520 6865 7572 6973   initiate heuris
-00006aa0: 7469 6320 6d65 7468 6f64 206f 6620 6465  tic method of de
-00006ab0: 6669 6e69 6e67 2043 4920 666f 7220 6461  fining CI for da
-00006ac0: 7469 6e67 2c20 6966 2074 6865 2066 6c61  ting, if the fla
-00006ad0: 6720 7761 7320 7365 742c 2074 6865 2068  g was set, the h
-00006ae0: 6575 7269 7374 6963 206d 6574 686f 6420  euristic method 
-00006af0: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00006b00: 6564 0a2d 6b63 2c20 2d2d 6b73 6375 746f  ed.-kc, --kscuto
-00006b10: 6666 2c20 7468 6520 4b73 2073 6174 7572  ff, the Ks satur
-00006b20: 6174 696f 6e20 6375 746f 6666 2069 6e20  ation cutoff in 
-00006b30: 6461 7469 6e67 2c20 6465 6661 756c 7420  dating, default 
-00006b40: 350a 2d2d 7368 6f77 6369 2c20 666c 6167  5.--showci, flag
-00006b50: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00006b60: 2074 6f20 7368 6f77 2043 4920 666f 7220   to show CI for 
-00006b70: 7468 6520 6f72 6967 696e 616c 2061 6e63  the original anc
-00006b80: 686f 7220 4b73 2047 4d4d 2061 6e61 6c79  hor Ks GMM analy
-00006b90: 7369 732c 2069 6620 7468 6520 666c 6167  sis, if the flag
-00006ba0: 2077 6173 2073 6574 2c20 6974 2077 6f75   was set, it wou
-00006bb0: 6c64 2062 6520 7368 6f77 6564 0a2d 2d6b  ld be showed.--k
-00006bc0: 6565 7074 6d70 6669 672c 2066 6c61 6720  eeptmpfig, flag 
-00006bd0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00006be0: 746f 206b 6565 7020 7465 6d70 6f72 6172  to keep temporar
-00006bf0: 7920 6669 6775 7265 7320 696e 2070 6561  y figures in pea
-00006c00: 6b20 6669 6e64 696e 6720 7072 6f63 6573  k finding proces
-00006c10: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00006c20: 6173 2073 6574 2c20 7468 6f73 6520 6669  as set, those fi
-00006c30: 6775 7265 7320 776f 756c 6420 6265 206b  gures would be k
-00006c40: 6570 740a 6060 600a 0a54 6865 2070 726f  ept.```..The pro
-00006c50: 6772 616d 2060 7767 6420 7379 6e60 2063  gram `wgd syn` c
-00006c60: 616e 2072 6561 6c69 7a65 2074 6865 2069  an realize the i
-00006c70: 6e74 7261 2d20 616e 6420 696e 7465 722d  ntra- and inter-
-00006c80: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-00006c90: 2069 6e66 6572 656e 6365 2e0a 6060 600a   inference..```.
-00006ca0: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
-00006cb0: 2067 6666 7320 286f 7074 696f 6e29 0a2d   gffs (option).-
+00005310: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
+00005320: 6265 2063 616c 6c65 6420 6f66 660a 2d72  be called off.-r
+00005330: 772c 202d 2d72 6577 6569 6768 742c 2066  w, --reweight, f
+00005340: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00005350: 6865 7220 746f 2072 6563 616c 6375 6c61  her to recalcula
+00005360: 7465 2074 6865 2077 6569 6768 7420 7065  te the weight pe
+00005370: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
+00005380: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00005390: 7365 742c 2074 6865 2077 6569 6768 7420  set, the weight 
+000053a0: 776f 756c 6420 6265 2072 6563 616c 6375  would be recalcu
+000053b0: 6c61 7465 640a 2d6f 722c 202d 2d6f 6e6c  lated.-or, --onl
+000053c0: 7972 6f6f 746f 7574 2c20 666c 6167 206f  yrootout, flag o
+000053d0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000053e0: 6f20 6f6e 6c79 2063 6f6e 6475 6374 2072  o only conduct r
+000053f0: 6174 6520 636f 7272 6563 7469 6f6e 2075  ate correction u
+00005400: 7369 6e67 2074 6865 206f 7574 6772 6f75  sing the outgrou
+00005410: 7020 6174 2072 6f6f 7420 6173 206f 7574  p at root as out
+00005420: 6772 6f75 702c 2069 6620 7468 6520 666c  group, if the fl
+00005430: 6167 2077 6173 2073 6574 2c20 6f6e 6c79  ag was set, only
+00005440: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
+00005450: 2072 6f6f 7420 776f 756c 6420 6265 2075   root would be u
+00005460: 7365 6420 6173 206f 7574 6772 6f75 700a  sed as outgroup.
+00005470: 2d65 706b 2c20 2d2d 6578 7472 6170 6172  -epk, --extrapar
+00005480: 616e 6f6d 656b 732c 2065 7874 7261 2070  anomeks, extra p
+00005490: 6172 616e 6f6d 6520 4b73 2064 6174 6120  aranome Ks data 
+000054a0: 746f 2070 6c6f 7420 696e 2074 6865 206d  to plot in the m
+000054b0: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
+000054c0: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
+000054d0: 6e65 0a2d 6170 2c20 2d2d 616e 6368 6f72  ne.-ap, --anchor
+000054e0: 706f 696e 7473 2c20 616e 6368 6f72 706f  points, anchorpo
+000054f0: 696e 7473 2e74 7874 2066 696c 6520 746f  ints.txt file to
+00005500: 2070 6c6f 7420 616e 6368 6f72 204b 7320   plot anchor Ks 
+00005510: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00005520: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+00005530: 6661 756c 7420 4e6f 6e65 0a2d 706b 2c20  fault None.-pk, 
+00005540: 2d2d 706c 6f74 6b64 652c 2066 6c61 6720  --plotkde, flag 
+00005550: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00005560: 746f 2070 6c6f 7420 6b64 6520 6375 7276  to plot kde curv
+00005570: 6520 6f66 206f 7274 686f 6c6f 676f 7573  e of orthologous
+00005580: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00005590: 206f 7665 7220 6869 7374 6f67 7261 6d20   over histogram 
+000055a0: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+000055b0: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
+000055c0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+000055d0: 742c 2074 6865 206b 6465 2063 7572 7665  t, the kde curve
+000055e0: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
+000055f0: 640a 2d70 6167 2c20 2d2d 706c 6f74 6170  d.-pag, --plotap
+00005600: 676d 6d2c 2066 6c61 6720 6f70 7469 6f6e  gmm, flag option
+00005610: 2c20 7768 6574 6865 7220 746f 2070 6572  , whether to per
+00005620: 666f 726d 2061 6e64 2070 6c6f 7420 6d69  form and plot mi
+00005630: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+00005640: 6620 616e 6368 6f72 204b 7320 696e 2074  f anchor Ks in t
+00005650: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+00005660: 7269 6275 7469 6f6e 2c20 6966 2074 6865  ribution, if the
+00005670: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00005680: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
+00005690: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+000056a0: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
+000056b0: 640a 2d70 656d 2c20 2d2d 706c 6f74 656c  d.-pem, --plotel
+000056c0: 6d6d 2c20 666c 6167 206f 7074 696f 6e2c  mm, flag option,
+000056d0: 2077 6865 7468 6572 2074 6f20 7065 7266   whether to perf
+000056e0: 6f72 6d20 616e 6420 706c 6f74 2065 6c6d  orm and plot elm
+000056f0: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
+00005700: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
+00005710: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
+00005720: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00005730: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00005740: 7365 742c 2074 6865 2065 6c6d 6d20 6d69  set, the elmm mi
+00005750: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+00005760: 6620 7061 7261 6e6f 6d65 204b 7320 776f  f paranome Ks wo
+00005770: 756c 6420 6265 2070 6c6f 7474 6564 0a2d  uld be plotted.-
+00005780: 632c 202d 2d63 6f6d 706f 6e65 6e74 732c  c, --components,
+00005790: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
+000057a0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
+000057b0: 6f6e 656e 7473 2074 6f20 6669 7420 696e  onents to fit in
+000057c0: 2061 6e63 686f 7220 4b73 206d 6978 7475   anchor Ks mixtu
+000057d0: 7265 206d 6f64 656c 696e 672c 2064 6566  re modeling, def
+000057e0: 6175 6c74 2028 312c 3429 0a2d 786c 2c20  ault (1,4).-xl, 
+000057f0: 2d2d 786c 696d 2c20 7468 6520 7820 6178  --xlim, the x ax
+00005800: 6973 206c 696d 6974 206f 6620 4b73 2064  is limit of Ks d
+00005810: 6973 7472 6962 7574 696f 6e0a 2d79 6c2c  istribution.-yl,
+00005820: 202d 2d79 6c69 6d2c 2074 6865 2079 2061   --ylim, the y a
+00005830: 7869 7320 6c69 6d69 7420 6f66 204b 7320  xis limit of Ks 
+00005840: 6469 7374 7269 6275 7469 6f6e 0a2d 6164  distribution.-ad
+00005850: 6f2c 202d 2d61 646a 7573 746f 7274 686f  o, --adjustortho
+00005860: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00005870: 6865 7468 6572 2074 6f20 6164 6a75 7374  hether to adjust
+00005880: 2074 6865 2068 6973 746f 6772 616d 2068   the histogram h
+00005890: 6569 6768 7420 6f66 206f 7274 686f 6c6f  eight of ortholo
+000058a0: 676f 7573 204b 7320 6173 2074 6f20 6d61  gous Ks as to ma
+000058b0: 7463 6820 7468 6520 6865 6967 6874 206f  tch the height o
+000058c0: 6620 7061 7261 6c6f 676f 7573 204b 732c  f paralogous Ks,
+000058d0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+000058e0: 2073 6574 2c20 7468 6520 6164 6a75 7374   set, the adjust
+000058f0: 6d65 6e74 2077 6f75 6c64 2062 6520 636f  ment would be co
+00005900: 6e64 7563 7465 640a 2d61 6466 2c20 2d2d  nducted.-adf, --
+00005910: 6164 6a75 7374 6661 6374 6f72 2c20 7468  adjustfactor, th
+00005920: 6520 6164 6a75 7374 6d65 6e74 2066 6163  e adjustment fac
+00005930: 746f 7220 6f66 206f 7274 686f 6c6f 676f  tor of orthologo
+00005940: 7573 204b 732c 2064 6566 6175 6c74 2030  us Ks, default 0
+00005950: 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c 7068  .5.-oa, --okalph
+00005960: 612c 2074 6865 206f 7061 6369 7479 206f  a, the opacity o
+00005970: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
+00005980: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
+00005990: 206d 6978 6564 2070 6c6f 742c 2064 6566   mixed plot, def
+000059a0: 6175 6c74 2030 2e35 0a2d 6661 2c20 2d2d  ault 0.5.-fa, --
+000059b0: 666f 6375 7332 616c 6c2c 2073 6574 2066  focus2all, set f
+000059c0: 6f63 616c 2073 7065 6369 6573 2061 6e64  ocal species and
+000059d0: 206c 6574 2073 7065 6369 6573 2070 6169   let species pai
+000059e0: 7220 746f 2062 6520 6265 7477 6565 6e20  r to be between 
+000059f0: 666f 6361 6c20 616e 6420 616c 6c20 7468  focal and all th
+00005a00: 6520 7265 6d61 696e 696e 6720 7370 6563  e remaining spec
+00005a10: 6965 732c 2064 6566 6175 6c74 204e 6f6e  ies, default Non
+00005a20: 650a 2d6b 732c 202d 2d6b 7374 7265 652c  e.-ks, --kstree,
+00005a30: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00005a40: 6574 6865 7220 746f 2069 6e66 6572 204b  ether to infer K
+00005a50: 7320 7472 6565 2c20 6966 2074 6865 2066  s tree, if the f
+00005a60: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00005a70: 204b 7320 7472 6565 2069 6e66 6572 656e   Ks tree inferen
+00005a80: 6365 2061 6e61 6c79 7369 7320 776f 756c  ce analysis woul
+00005a90: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
+00005aa0: 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e 6361  ock, --onlyconca
+00005ab0: 746b 7374 7265 652c 2066 6c61 6720 6f70  tkstree, flag op
+00005ac0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00005ad0: 206f 6e6c 7920 696e 6665 7220 4b73 2074   only infer Ks t
+00005ae0: 7265 6520 756e 6465 7220 636f 6e63 6174  ree under concat
+00005af0: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
+00005b00: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00005b10: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
+00005b20: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
+00005b30: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
+00005b40: 6d65 6e74 2077 6f75 6c64 2062 6520 6361  ment would be ca
+00005b50: 6c63 756c 6174 6564 0a2d 6373 2c20 2d2d  lculated.-cs, --
+00005b60: 636c 6173 7369 632c 2066 6c61 6720 6f70  classic, flag op
+00005b70: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00005b80: 2064 7261 7720 6d69 7865 6420 4b73 2070   draw mixed Ks p
+00005b90: 6c6f 7420 696e 2061 2063 6c61 7373 6963  lot in a classic
+00005ba0: 206d 616e 6e65 7220 7768 6572 6520 7468   manner where th
+00005bb0: 6520 6675 6c6c 206f 7274 686f 6c6f 676f  e full orthologo
+00005bc0: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
+00005bd0: 6f6e 2069 7320 6472 6177 6564 2c20 6966  on is drawed, if
+00005be0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00005bf0: 742c 2074 6865 2063 6c61 7373 6963 206d  t, the classic m
+00005c00: 6978 6564 204b 7320 706c 6f74 2077 6f75  ixed Ks plot wou
+00005c10: 6c64 2062 6520 6472 6177 6e0a 2d74 612c  ld be drawn.-ta,
+00005c20: 202d 2d74 6f70 6172 726f 772c 2066 6c61   --toparrow, fla
+00005c30: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00005c40: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
+00005c50: 6172 726f 7720 746f 2062 6520 6174 2074  arrow to be at t
+00005c60: 6865 2074 6f70 206f 6620 7468 6520 706c  he top of the pl
+00005c70: 6f74 2c20 696e 7374 6561 6420 6f66 2062  ot, instead of b
+00005c80: 6569 6e67 2063 6f6f 7264 696e 6174 6564  eing coordinated
+00005c90: 2061 7320 7468 6520 4b44 4520 6f66 2074   as the KDE of t
+00005ca0: 6865 206f 7274 686f 6c6f 676f 7573 204b  he orthologous K
+00005cb0: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00005cc0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00005cd0: 7365 742c 2074 6865 2061 7272 6f77 2077  set, the arrow w
+00005ce0: 6f75 6c64 2062 6520 7365 7420 6174 2074  ould be set at t
+00005cf0: 6865 2074 6f70 0a2d 6273 2c20 2d2d 626f  he top.-bs, --bo
+00005d00: 6f74 7374 7261 702c 2074 6865 206e 756d  otstrap, the num
+00005d10: 6265 7220 6f66 2062 6f6f 7473 7472 6170  ber of bootstrap
+00005d20: 2072 6570 6c69 6361 7465 7320 6f66 206f   replicates of o
+00005d30: 7274 686f 6c6f 6720 4b73 2064 6973 7472  rtholog Ks distr
+00005d40: 6962 7574 696f 6e20 696e 206d 6978 6564  ibution in mixed
+00005d50: 2070 6c6f 740a 6060 600a 0a54 6865 2070   plot.```..The p
+00005d60: 726f 6772 616d 2060 7767 6420 6d69 7860  rogram `wgd mix`
+00005d70: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+00005d80: 206d 6978 7475 7265 206d 6f64 656c 2063   mixture model c
+00005d90: 6c75 7374 6572 696e 6720 616e 616c 7973  lustering analys
+00005da0: 6973 206f 6620 2a4b 2a3c 7375 623e 533c  is of *K*<sub>S<
+00005db0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00005dc0: 6275 7469 6f6e 2e0a 6060 600a 7767 6420  bution..```.wgd 
+00005dd0: 6d69 7820 6b73 5f64 6174 6166 696c 6520  mix ks_datafile 
+00005de0: 286f 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d  (option).-------
+00005df0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005e00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005e10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005e20: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005e30: 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20 2d2d  ---------.-f, --
+00005e40: 6669 6c74 6572 732c 2074 6865 2063 7574  filters, the cut
+00005e50: 6f66 6620 616c 6967 6e6d 656e 7420 6c65  off alignment le
+00005e60: 6e67 7468 2c20 6465 6661 756c 7420 3330  ngth, default 30
+00005e70: 300a 2d72 2c20 2d2d 6b73 5f72 616e 6765  0.-r, --ks_range
+00005e80: 2c20 7468 6520 4b73 2072 616e 6765 2074  , the Ks range t
+00005e90: 6f20 6265 2063 6f6e 7369 6465 7265 642c  o be considered,
+00005ea0: 2064 6566 6175 6c74 2028 302c 2035 290a   default (0, 5).
+00005eb0: 2d62 2c20 2d2d 6269 6e73 2c20 7468 6520  -b, --bins, the 
+00005ec0: 6e75 6d62 6572 206f 6620 6269 6e73 2069  number of bins i
+00005ed0: 6e20 4b73 2064 6973 7472 6962 7574 696f  n Ks distributio
+00005ee0: 6e2c 2064 6566 6175 6c74 2035 300a 2d6f  n, default 50.-o
+00005ef0: 2c20 2d2d 6f75 7464 6972 2c20 7468 6520  , --outdir, the 
+00005f00: 6f75 7470 7574 2064 6972 6563 746f 7279  output directory
+00005f10: 2c20 6465 6661 756c 7420 7767 645f 6d69  , default wgd_mi
+00005f20: 780a 2d2d 6d65 7468 6f64 2c20 7768 6963  x.--method, whic
+00005f30: 6820 6d69 7874 7572 6520 6d6f 6465 6c20  h mixture model 
+00005f40: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00005f50: 676d 6d0a 2d6e 2c20 2d2d 636f 6d70 6f6e  gmm.-n, --compon
+00005f60: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
+00005f70: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
+00005f80: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
+00005f90: 6974 2c20 6465 6661 756c 7420 2831 2c20  it, default (1, 
+00005fa0: 3429 0a2d 672c 202d 2d67 616d 6d61 2c20  4).-g, --gamma, 
+00005fb0: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
+00005fc0: 7465 7220 666f 7220 6267 6d6d 206d 6f64  ter for bgmm mod
+00005fd0: 656c 732c 2064 6566 6175 6c74 2030 2e30  els, default 0.0
+00005fe0: 3031 0a2d 6e69 2c20 2d2d 6e5f 696e 6974  01.-ni, --n_init
+00005ff0: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00006000: 6b2d 6d65 616e 7320 696e 6974 6961 6c69  k-means initiali
+00006010: 7a61 7469 6f6e 732c 2064 6566 6175 6c74  zations, default
+00006020: 2032 3030 0a2d 6d69 2c20 2d2d 6d61 785f   200.-mi, --max_
+00006030: 6974 6572 2c20 7468 6520 6d61 7869 6d75  iter, the maximu
+00006040: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
+00006050: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
+00006060: 3230 300a 6060 600a 0a54 6865 2070 726f  200.```..The pro
+00006070: 6772 616d 2060 7767 6420 7065 616b 6020  gram `wgd peak` 
+00006080: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
+00006090: 7365 6172 6368 206f 6620 6372 6564 6961  search of credia
+000060a0: 626c 6520 2a4b 2a3c 7375 623e 533c 2f73  ble *K*<sub>S</s
+000060b0: 7562 3e20 7261 6e67 6520 7573 6564 2069  ub> range used i
+000060c0: 6e20 5747 4420 6461 7469 6e67 2e0a 6060  n WGD dating..``
+000060d0: 600a 7767 6420 7065 616b 206b 735f 6461  `.wgd peak ks_da
+000060e0: 7461 6669 6c65 2028 6f70 7469 6f6e 290a  tafile (option).
+000060f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006100: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006110: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006120: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006130: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006140: 0a2d 6170 2c20 2d2d 616e 6368 6f72 706f  .-ap, --anchorpo
+00006150: 696e 7473 2c20 7468 6520 616e 6368 6f72  ints, the anchor
+00006160: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
+00006170: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00006180: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
+00006190: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
+000061a0: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
+000061b0: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
+000061c0: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
+000061d0: 7473 6567 6d65 6e74 7320 6461 7461 6669  tsegments datafi
+000061e0: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+000061f0: 200a 2d6d 702c 202d 2d6d 756c 7469 706c   .-mp, --multipl
+00006200: 6963 6f6e 7061 6972 732c 2074 6865 206d  iconpairs, the m
+00006210: 756c 7469 706c 6963 6f6e 7061 6972 7320  ultipliconpairs 
+00006220: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00006230: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
+00006240: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
+00006250: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
+00006260: 6c74 2077 6764 5f70 6561 6b0a 2d61 662c  lt wgd_peak.-af,
+00006270: 202d 2d61 6c69 676e 6669 6c74 6572 2c20   --alignfilter, 
+00006280: 6375 746f 6666 2066 6f72 2061 6c69 676e  cutoff for align
+00006290: 6d65 6e74 2069 6465 6e74 6974 792c 206c  ment identity, l
+000062a0: 656e 6774 6820 616e 6420 636f 7665 7261  ength and covera
+000062b0: 6765 2c20 6465 6661 756c 7420 302e 302c  ge, default 0.0,
+000062c0: 2030 2c20 302e 300a 2d72 2c20 2d2d 6b73   0, 0.0.-r, --ks
+000062d0: 7261 6e67 652c 2072 616e 6765 206f 6620  range, range of 
+000062e0: 4b73 2074 6f20 6265 2061 6e61 6c79 7a65  Ks to be analyze
+000062f0: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
+00006300: 290a 2d62 772c 202d 2d62 696e 5f77 6964  ).-bw, --bin_wid
+00006310: 7468 2c20 6261 6e64 7769 6474 6820 6f66  th, bandwidth of
+00006320: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00006330: 2c20 6465 6661 756c 7420 302e 310a 2d69  , default 0.1.-i
+00006340: 632c 202d 2d77 6569 6768 7473 5f6f 7574  c, --weights_out
+00006350: 6c69 6572 735f 696e 636c 7564 6564 2c20  liers_included, 
+00006360: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00006370: 7468 6572 2074 6f20 696e 636c 7564 6520  ther to include 
+00006380: 4b73 206f 7574 6c69 6572 732c 2069 6620  Ks outliers, if 
+00006390: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000063a0: 2c20 4b73 206f 7574 6c69 6572 7320 776f  , Ks outliers wo
+000063b0: 756c 6420 6265 2069 6e63 6c75 6465 6420  uld be included 
+000063c0: 696e 2074 6865 2061 6e61 6c79 7369 730a  in the analysis.
+000063d0: 2d6d 2c20 2d2d 6d65 7468 6f64 2c20 7768  -m, --method, wh
+000063e0: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
+000063f0: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
+00006400: 7420 676d 6d0a 2d2d 7365 6564 2c20 7261  t gmm.--seed, ra
+00006410: 6e64 6f6d 2073 6565 6420 6769 7665 6e20  ndom seed given 
+00006420: 746f 2069 6e69 7469 616c 697a 6174 696f  to initializatio
+00006430: 6e2c 2064 6566 6175 6c74 2032 3335 3238  n, default 23528
+00006440: 3930 0a2d 6569 2c20 2d2d 656d 5f69 7465  90.-ei, --em_ite
+00006450: 722c 2074 6865 206e 756d 6265 7220 6f66  r, the number of
+00006460: 2045 4d20 6974 6572 6174 696f 6e73 2074   EM iterations t
+00006470: 6f20 7065 7266 6f72 6d2c 2064 6566 6175  o perform, defau
+00006480: 6c74 2032 3030 0a2d 6e69 2c20 2d2d 6e5f  lt 200.-ni, --n_
+00006490: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
+000064a0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+000064b0: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
+000064c0: 6175 6c74 2032 3030 0a2d 6e2c 202d 2d63  ault 200.-n, --c
+000064d0: 6f6d 706f 6e65 6e74 732c 2074 6865 2072  omponents, the r
+000064e0: 616e 6765 206f 6620 7468 6520 6e75 6d62  ange of the numb
+000064f0: 6572 206f 6620 636f 6d70 6f6e 656e 7473  er of components
+00006500: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
+00006510: 2028 312c 2034 290a 2d67 2c20 2d2d 6761   (1, 4).-g, --ga
+00006520: 6d6d 612c 2074 6865 2067 616d 6d61 2070  mma, the gamma p
+00006530: 6172 616d 6574 6572 2066 6f72 2062 676d  arameter for bgm
+00006540: 6d20 6d6f 6465 6c73 2c20 6465 6661 756c  m models, defaul
+00006550: 7420 3165 2d33 0a2d 2d62 6f6f 7473 2c20  t 1e-3.--boots, 
+00006560: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
+00006570: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
+00006580: 6573 206f 6620 6b64 652c 2064 6566 6175  es of kde, defau
+00006590: 6c74 2032 3030 0a2d 2d77 6569 6768 7465  lt 200.--weighte
+000065a0: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+000065b0: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+000065c0: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
+000065d0: 686f 6420 6f66 2064 652d 7265 6475 6e64  hod of de-redund
+000065e0: 616e 6379 2c20 6966 2074 6865 2066 6c61  ancy, if the fla
+000065f0: 6720 7761 7320 7365 742c 2074 6865 206e  g was set, the n
+00006600: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
+00006610: 686f 6420 776f 756c 6420 6265 2075 7365  hod would be use
+00006620: 640a 2d70 2c20 2d2d 706c 6f74 2c20 7468  d.-p, --plot, th
+00006630: 6520 706c 6f74 7469 6e67 206d 6574 686f  e plotting metho
+00006640: 6420 746f 2062 6520 7573 6564 2c20 6465  d to be used, de
+00006650: 6661 756c 7420 6964 656e 7469 6361 6c0a  fault identical.
+00006660: 2d62 6d2c 202d 2d62 775f 6d65 7468 6f64  -bm, --bw_method
+00006670: 2c20 7468 6520 6261 6e64 7769 6474 6820  , the bandwidth 
+00006680: 6d65 7468 6f64 2074 6f20 6265 2075 7365  method to be use
+00006690: 6420 696e 2061 6e61 6c79 7a69 6e67 2074  d in analyzing t
+000066a0: 6865 2070 6561 6b20 6f66 2057 4744 2064  he peak of WGD d
+000066b0: 6174 6573 2c20 6465 6661 756c 7420 7369  ates, default si
+000066c0: 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65 646f  lverman.--n_medo
+000066d0: 6964 732c 2074 6865 206e 756d 6265 7220  ids, the number 
+000066e0: 6f66 206d 6564 6f69 6473 2074 6f20 6669  of medoids to fi
+000066f0: 742c 2064 6566 6175 6c74 2032 0a2d 6b6d  t, default 2.-km
+00006700: 2c20 2d2d 6b64 656d 6574 686f 642c 2074  , --kdemethod, t
+00006710: 6865 206b 6465 206d 6574 686f 6420 746f  he kde method to
+00006720: 2062 6520 7573 6564 2069 6e20 616e 616c   be used in anal
+00006730: 797a 696e 6720 7468 6520 7065 616b 206f  yzing the peak o
+00006740: 6620 5747 4420 6461 7465 732c 206b 6d65  f WGD dates, kme
+00006750: 646f 6964 7320 616e 616c 7973 6973 206f  doids analysis o
+00006760: 7220 7468 6520 6261 7369 6320 4b73 2070  r the basic Ks p
+00006770: 6c6f 7474 696e 672c 2064 6566 6175 6c74  lotting, default
+00006780: 2073 6369 7079 0a2d 2d6e 5f63 6c75 7374   scipy.--n_clust
+00006790: 6572 732c 2074 6865 206e 756d 6265 7220  ers, the number 
+000067a0: 6f66 2063 6c75 7374 6572 7320 746f 2070  of clusters to p
+000067b0: 6c6f 7420 456c 626f 7720 6c6f 7373 2066  lot Elbow loss f
+000067c0: 756e 6374 696f 6e2c 2064 6566 6175 6c74  unction, default
+000067d0: 2035 0a2d 6764 2c20 2d2d 6775 6964 652c   5.-gd, --guide,
+000067e0: 2074 6865 2072 6567 696d 6520 7265 7369   the regime resi
+000067f0: 6469 6e67 2061 6e63 686f 7273 2c20 6465  ding anchors, de
+00006800: 6661 756c 7420 5365 676d 656e 740a 2d70  fault Segment.-p
+00006810: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+00006820: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+00006830: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+00006840: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+00006850: 616b 7320 696e 2070 6561 6b20 6669 6e64  aks in peak find
+00006860: 696e 6720 7374 6570 732c 2064 6566 6175  ing steps, defau
+00006870: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
+00006880: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
+00006890: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
+000068a0: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
+000068b0: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
+000068c0: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
+000068d0: 6b64 2c20 2d2d 6b73 746f 6461 7465 2c20  kd, --kstodate, 
+000068e0: 7468 6520 7261 6e67 6520 6f66 204b 7320  the range of Ks 
+000068f0: 746f 2062 6520 6461 7465 6420 696e 2068  to be dated in h
+00006900: 6575 7269 7374 6963 2073 6561 7263 682c  euristic search,
+00006910: 2064 6566 6175 6c74 2028 302e 352c 2031   default (0.5, 1
+00006920: 2e35 290a 2d78 6c2c 202d 2d78 6c69 6d2c  .5).-xl, --xlim,
+00006930: 2074 6865 2078 2061 7869 7320 6c69 6d69   the x axis limi
+00006940: 7420 6f66 2047 4d4d 204b 7320 6469 7374  t of GMM Ks dist
+00006950: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
+00006960: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
+00006970: 206c 696d 6974 206f 6620 474d 4d20 4b73   limit of GMM Ks
+00006980: 2064 6973 7472 6962 7574 696f 6e0a 2d2d   distribution.--
+00006990: 6d61 6e75 616c 7365 742c 2066 6c61 6720  manualset, flag 
+000069a0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000069b0: 746f 206f 7574 7075 7420 616e 6368 6f72  to output anchor
+000069c0: 2070 6169 7273 2077 6974 6820 6d61 6e75   pairs with manu
+000069d0: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
+000069e0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+000069f0: 6173 2073 6574 2c20 6d61 6e75 616c 6c79  as set, manually
+00006a00: 2073 6574 204b 7320 7261 6e67 6520 776f   set Ks range wo
+00006a10: 756c 6420 6265 206f 7574 7075 7474 6564  uld be outputted
+00006a20: 0a2d 2d63 692c 2074 6865 2063 6f6e 6669  .--ci, the confi
+00006a30: 6465 6e63 6520 6c65 7665 6c20 6f66 206c  dence level of l
+00006a40: 6f67 2d6e 6f72 6d61 6c20 6469 7374 7269  og-normal distri
+00006a50: 6275 7469 6f6e 2074 6f20 6461 7465 2c20  bution to date, 
+00006a60: 6465 6661 756c 7420 3935 0a2d 2d68 6472  default 95.--hdr
+00006a70: 2c20 7468 6520 6869 6768 6573 7420 6465  , the highest de
+00006a80: 6e73 6974 7920 7265 6769 6f6e 2028 4844  nsity region (HD
+00006a90: 5229 2061 7070 6c69 6564 2069 6e20 7468  R) applied in th
+00006aa0: 6520 7365 676d 656e 742d 6775 6964 6564  e segment-guided
+00006ab0: 2061 6e63 686f 7220 7061 6972 204b 7320   anchor pair Ks 
+00006ac0: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+00006ad0: 6661 756c 7420 3935 0a2d 2d68 6575 7269  fault 95.--heuri
+00006ae0: 7374 6963 2c20 666c 6167 206f 7074 696f  stic, flag optio
+00006af0: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00006b00: 6974 6961 7465 2068 6575 7269 7374 6963  itiate heuristic
+00006b10: 206d 6574 686f 6420 6f66 2064 6566 696e   method of defin
+00006b20: 696e 6720 4349 2066 6f72 2064 6174 696e  ing CI for datin
+00006b30: 672c 2069 6620 7468 6520 666c 6167 2077  g, if the flag w
+00006b40: 6173 2073 6574 2c20 7468 6520 6865 7572  as set, the heur
+00006b50: 6973 7469 6320 6d65 7468 6f64 2077 6f75  istic method wou
+00006b60: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+00006b70: 2d6b 632c 202d 2d6b 7363 7574 6f66 662c  -kc, --kscutoff,
+00006b80: 2074 6865 204b 7320 7361 7475 7261 7469   the Ks saturati
+00006b90: 6f6e 2063 7574 6f66 6620 696e 2064 6174  on cutoff in dat
+00006ba0: 696e 672c 2064 6566 6175 6c74 2035 0a2d  ing, default 5.-
+00006bb0: 2d6b 6565 7074 6d70 6669 672c 2066 6c61  -keeptmpfig, fla
+00006bc0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00006bd0: 7220 746f 206b 6565 7020 7465 6d70 6f72  r to keep tempor
+00006be0: 6172 7920 6669 6775 7265 7320 696e 2070  ary figures in p
+00006bf0: 6561 6b20 6669 6e64 696e 6720 7072 6f63  eak finding proc
+00006c00: 6573 732c 2069 6620 7468 6520 666c 6167  ess, if the flag
+00006c10: 2077 6173 2073 6574 2c20 7468 6f73 6520   was set, those 
+00006c20: 6669 6775 7265 7320 776f 756c 6420 6265  figures would be
+00006c30: 206b 6570 740a 6060 600a 0a54 6865 2070   kept.```..The p
+00006c40: 726f 6772 616d 2060 7767 6420 7379 6e60  rogram `wgd syn`
+00006c50: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+00006c60: 2069 6e74 7261 2d20 616e 6420 696e 7465   intra- and inte
+00006c70: 722d 7370 6563 6966 6963 2073 796e 7465  r-specific synte
+00006c80: 6e79 2069 6e66 6572 656e 6365 2e0a 6060  ny inference..``
+00006c90: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+00006ca0: 6573 2067 6666 7320 286f 7074 696f 6e29  es gffs (option)
+00006cb0: 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  .---------------
 00006cc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
 00006cd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
 00006ce0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
 00006cf0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006d00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a  ---------------.
-00006d10: 2d6b 732c 202d 2d6b 735f 6469 7374 7269  -ks, --ks_distri
-00006d20: 6275 7469 6f6e 2c20 6b73 2064 6973 7472  bution, ks distr
-00006d30: 6962 7574 696f 6e20 6461 7461 6669 6c65  ibution datafile
-00006d40: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00006d50: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
-00006d60: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
-00006d70: 792c 2064 6566 6175 6c74 2077 6764 5f73  y, default wgd_s
-00006d80: 796e 0a2d 662c 202d 2d66 6561 7475 7265  yn.-f, --feature
-00006d90: 2c20 7468 6520 6665 6174 7572 6520 666f  , the feature fo
-00006da0: 7220 7061 7273 696e 6720 6765 6e65 2049  r parsing gene I
-00006db0: 4473 2066 726f 6d20 4746 4620 6669 6c65  Ds from GFF file
-00006dc0: 732c 2064 6566 6175 6c74 2067 656e 650a  s, default gene.
-00006dd0: 2d61 2c20 2d2d 6174 7472 6962 7574 652c  -a, --attribute,
-00006de0: 2074 6865 2061 7474 7269 6275 7465 2066   the attribute f
-00006df0: 6f72 2070 6172 7369 6e67 2074 6865 2067  or parsing the g
-00006e00: 656e 6520 4944 7320 6672 6f6d 2074 6865  ene IDs from the
-00006e10: 2047 4646 2066 696c 6573 2c20 6465 6661   GFF files, defa
-00006e20: 756c 7420 4944 0a2d 6174 672c 202d 2d61  ult ID.-atg, --a
-00006e30: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-00006e40: 2c20 7468 6520 6665 6174 7572 6520 616e  , the feature an
-00006e50: 6420 6174 7472 6962 7574 6520 696e 666f  d attribute info
-00006e60: 726d 6174 696f 6e20 6f66 2061 6464 6974  rmation of addit
-00006e70: 696f 6e61 6c20 6766 6633 2066 696c 6573  ional gff3 files
-00006e80: 2069 6620 6469 6666 6572 656e 7420 696e   if different in
-00006e90: 2074 6865 2066 6f72 6d61 7420 6f66 2028   the format of (
-00006ea0: 6665 6174 7572 653b 6174 7472 6962 7574  feature;attribut
-00006eb0: 6529 272c 2064 6566 6175 6c74 204e 6f6e  e)', default Non
-00006ec0: 650a 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c  e.-ml, --minlen,
-00006ed0: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
-00006ee0: 6774 6820 6f66 2061 2073 6361 6666 6f6c  gth of a scaffol
-00006ef0: 6420 746f 2062 6520 696e 636c 7564 6564  d to be included
-00006f00: 2069 6e20 646f 7470 6c6f 742c 2064 6566   in dotplot, def
-00006f10: 6175 6c74 202d 312c 2069 6620 2d31 2077  ault -1, if -1 w
-00006f20: 6173 2073 6574 2c20 7468 6520 3130 2520  as set, the 10% 
-00006f30: 6f66 2074 6865 206c 6f6e 6765 7374 2073  of the longest s
-00006f40: 6361 6666 6f6c 6420 776f 756c 6420 6265  caffold would be
-00006f50: 2073 6574 0a2d 6d73 2c20 2d2d 6d61 7873   set.-ms, --maxs
-00006f60: 697a 652c 2074 6865 206d 6178 696d 756d  ize, the maximum
-00006f70: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
-00006f80: 6265 2069 6e63 6c75 6465 642c 2064 6566  be included, def
-00006f90: 6175 6c74 2032 3030 0a2d 722c 202d 2d6b  ault 200.-r, --k
-00006fa0: 735f 7261 6e67 652c 2074 6865 204b 7320  s_range, the Ks 
-00006fb0: 7261 6e67 6520 696e 2063 6f6c 6f72 6564  range in colored
-00006fc0: 2064 6f74 706c 6f74 2c20 6465 6661 756c   dotplot, defaul
-00006fd0: 7420 2830 2c20 3529 0a2d 2d69 6164 686f  t (0, 5).--iadho
-00006fe0: 7265 5f6f 7074 696f 6e73 2c20 7468 6520  re_options, the 
-00006ff0: 7061 7261 6d65 7465 7220 7365 7474 696e  parameter settin
-00007000: 6720 696e 2069 6164 686f 7265 2c20 6465  g in iadhore, de
-00007010: 6661 756c 7420 6173 2061 2073 7472 696e  fault as a strin
-00007020: 6720 6f66 206c 656e 6774 6820 7a65 726f  g of length zero
-00007030: 0a2d 6d67 2c20 2d2d 6d69 6e73 6567 6c65  .-mg, --minsegle
-00007040: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
-00007050: 656e 6774 6820 6f66 2073 6567 6d65 6e74  ength of segment
-00007060: 7320 746f 2069 6e63 6c75 6465 2069 6e20  s to include in 
-00007070: 7261 7469 6f20 6966 203c 3d20 312c 2064  ratio if <= 1, d
-00007080: 6566 6175 6c74 2031 3030 3030 0a2d 6b72  efault 10000.-kr
-00007090: 2c20 2d2d 6b65 6570 7265 6475 6e2c 2066  , --keepredun, f
-000070a0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000070b0: 6865 7220 746f 206b 6565 7020 7265 6475  her to keep redu
-000070c0: 6e64 616e 7420 6d75 6c74 6970 6c69 636f  ndant multiplico
-000070d0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
-000070e0: 7761 7320 7365 742c 2074 6865 2072 6564  was set, the red
-000070f0: 756e 6461 6e74 206d 756c 7469 706c 6963  undant multiplic
-00007100: 6f6e 7320 776f 756c 6420 6265 206b 6570  ons would be kep
-00007110: 740a 2d6d 676e 2c20 2d2d 6d69 6e67 656e  t.-mgn, --mingen
-00007120: 656e 756d 2c20 7468 6520 6d69 6e69 6d75  enum, the minimu
-00007130: 6d20 6e75 6d62 6572 206f 6620 6765 6e65  m number of gene
-00007140: 7320 666f 7220 6120 7365 676d 656e 7420  s for a segment 
-00007150: 746f 2062 6520 636f 6e73 6964 6572 6564  to be considered
-00007160: 2c20 6465 6661 756c 7420 3330 0a2d 6473  , default 30.-ds
-00007170: 2c20 2d2d 646f 7473 697a 652c 2074 6865  , --dotsize, the
-00007180: 2064 6f74 2073 697a 6520 696e 2064 6f74   dot size in dot
-00007190: 2070 6c6f 742c 2064 6566 6175 6c74 2030   plot, default 0
-000071a0: 2e33 0a2d 6161 2c20 2d2d 6170 616c 7068  .3.-aa, --apalph
-000071b0: 612c 2074 6865 206f 7061 6369 7479 206f  a, the opacity o
-000071c0: 6620 616e 6368 6f72 2064 6f74 732c 2064  f anchor dots, d
-000071d0: 6566 6175 6c74 2031 0a2d 6861 2c20 2d2d  efault 1.-ha, --
-000071e0: 686f 616c 7068 612c 2074 6865 206f 7061  hoalpha, the opa
-000071f0: 6369 7479 206f 6620 686f 6d6f 6c6f 6720  city of homolog 
-00007200: 646f 7473 2c20 6465 6661 756c 7420 300a  dots, default 0.
-00007210: 2d73 7274 2c20 2d2d 7368 6f77 7265 616c  -srt, --showreal
-00007220: 7469 636b 2c20 666c 6167 206f 7074 696f  tick, flag optio
-00007230: 6e2c 2077 6865 7468 6572 2074 6f20 7368  n, whether to sh
-00007240: 6f77 2074 6865 2072 6561 6c20 7469 636b  ow the real tick
-00007250: 2069 6e20 6765 6e65 7320 6f72 2062 6173   in genes or bas
-00007260: 6573 2c20 6966 2074 6865 2066 6c61 6720  es, if the flag 
-00007270: 7761 7320 7365 742c 2074 6865 2072 6561  was set, the rea
-00007280: 6c20 7469 636b 2077 6f75 6c64 2062 6520  l tick would be 
-00007290: 7368 6f77 6564 0a2d 746c 732c 202d 2d74  showed.-tls, --t
-000072a0: 6963 6b6c 6162 656c 7369 7a65 2c20 7468  icklabelsize, th
-000072b0: 6520 6c61 6265 6c20 7369 7a65 206f 6620  e label size of 
-000072c0: 7469 636b 2c20 6465 6661 756c 7420 350a  tick, default 5.
-000072d0: 2d67 722c 202d 2d67 6973 7472 622c 2066  -gr, --gistrb, f
-000072e0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000072f0: 6865 7220 746f 2075 7365 2067 6973 745f  her to use gist_
-00007300: 7261 696e 626f 7720 6173 2063 6f6c 6f72  rainbow as color
-00007310: 206d 6170 206f 6620 646f 7470 6c6f 740a   map of dotplot.
-00007320: 2d6e 2c20 2d2d 6e74 6872 6561 6473 2c20  -n, --nthreads, 
-00007330: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
-00007340: 7265 6164 7320 746f 2075 7365 2069 6e20  reads to use in 
-00007350: 7379 6e74 656e 7920 696e 6665 7265 6e63  synteny inferenc
-00007360: 652c 2064 6566 6175 6c74 2034 0a60 6060  e, default 4.```
-00007370: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
-00007380: 6764 2076 697a 6020 6361 6e20 7265 616c  gd viz` can real
-00007390: 697a 6520 7468 6520 7669 7375 616c 697a  ize the visualiz
-000073a0: 6174 696f 6e20 6f66 202a 4b2a 3c73 7562  ation of *K*<sub
-000073b0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-000073c0: 7472 6962 7574 696f 6e20 616e 6420 7379  tribution and sy
-000073d0: 6e74 656e 792e 0a60 6060 0a77 6764 2076  nteny..```.wgd v
-000073e0: 697a 2028 6f70 7469 6f6e 290a 2d2d 2d2d  iz (option).----
+00006d00: 2d0a 2d6b 732c 202d 2d6b 735f 6469 7374  -.-ks, --ks_dist
+00006d10: 7269 6275 7469 6f6e 2c20 6b73 2064 6973  ribution, ks dis
+00006d20: 7472 6962 7574 696f 6e20 6461 7461 6669  tribution datafi
+00006d30: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00006d40: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+00006d50: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+00006d60: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+00006d70: 5f73 796e 0a2d 662c 202d 2d66 6561 7475  _syn.-f, --featu
+00006d80: 7265 2c20 7468 6520 6665 6174 7572 6520  re, the feature 
+00006d90: 666f 7220 7061 7273 696e 6720 6765 6e65  for parsing gene
+00006da0: 2049 4473 2066 726f 6d20 4746 4620 6669   IDs from GFF fi
+00006db0: 6c65 732c 2064 6566 6175 6c74 2067 656e  les, default gen
+00006dc0: 650a 2d61 2c20 2d2d 6174 7472 6962 7574  e.-a, --attribut
+00006dd0: 652c 2074 6865 2061 7474 7269 6275 7465  e, the attribute
+00006de0: 2066 6f72 2070 6172 7369 6e67 2074 6865   for parsing the
+00006df0: 2067 656e 6520 4944 7320 6672 6f6d 2074   gene IDs from t
+00006e00: 6865 2047 4646 2066 696c 6573 2c20 6465  he GFF files, de
+00006e10: 6661 756c 7420 4944 0a2d 6174 672c 202d  fault ID.-atg, -
+00006e20: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+00006e30: 666f 2c20 7468 6520 6665 6174 7572 6520  fo, the feature 
+00006e40: 616e 6420 6174 7472 6962 7574 6520 696e  and attribute in
+00006e50: 666f 726d 6174 696f 6e20 6f66 2061 6464  formation of add
+00006e60: 6974 696f 6e61 6c20 6766 6633 2066 696c  itional gff3 fil
+00006e70: 6573 2069 6620 6469 6666 6572 656e 7420  es if different 
+00006e80: 696e 2074 6865 2066 6f72 6d61 7420 6f66  in the format of
+00006e90: 2028 6665 6174 7572 653b 6174 7472 6962   (feature;attrib
+00006ea0: 7574 6529 272c 2064 6566 6175 6c74 204e  ute)', default N
+00006eb0: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
+00006ec0: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
+00006ed0: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
+00006ee0: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
+00006ef0: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
+00006f00: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
+00006f10: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
+00006f20: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
+00006f30: 2073 6361 6666 6f6c 6420 776f 756c 6420   scaffold would 
+00006f40: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
+00006f50: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
+00006f60: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
+00006f70: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
+00006f80: 6566 6175 6c74 2032 3030 0a2d 722c 202d  efault 200.-r, -
+00006f90: 2d6b 735f 7261 6e67 652c 2074 6865 204b  -ks_range, the K
+00006fa0: 7320 7261 6e67 6520 696e 2063 6f6c 6f72  s range in color
+00006fb0: 6564 2064 6f74 706c 6f74 2c20 6465 6661  ed dotplot, defa
+00006fc0: 756c 7420 2830 2c20 3529 0a2d 2d69 6164  ult (0, 5).--iad
+00006fd0: 686f 7265 5f6f 7074 696f 6e73 2c20 7468  hore_options, th
+00006fe0: 6520 7061 7261 6d65 7465 7220 7365 7474  e parameter sett
+00006ff0: 696e 6720 696e 2069 6164 686f 7265 2c20  ing in iadhore, 
+00007000: 6465 6661 756c 7420 6173 2061 2073 7472  default as a str
+00007010: 696e 6720 6f66 206c 656e 6774 6820 7a65  ing of length ze
+00007020: 726f 0a2d 6d67 2c20 2d2d 6d69 6e73 6567  ro.-mg, --minseg
+00007030: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
+00007040: 206c 656e 6774 6820 6f66 2073 6567 6d65   length of segme
+00007050: 6e74 7320 746f 2069 6e63 6c75 6465 2069  nts to include i
+00007060: 6e20 7261 7469 6f20 6966 203c 3d20 312c  n ratio if <= 1,
+00007070: 2064 6566 6175 6c74 2031 3030 3030 0a2d   default 10000.-
+00007080: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+00007090: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000070a0: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+000070b0: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+000070c0: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+000070d0: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+000070e0: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+000070f0: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+00007100: 6570 740a 2d6d 676e 2c20 2d2d 6d69 6e67  ept.-mgn, --ming
+00007110: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+00007120: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+00007130: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+00007140: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+00007150: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+00007160: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+00007170: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+00007180: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+00007190: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+000071a0: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+000071b0: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+000071c0: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+000071d0: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+000071e0: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+000071f0: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+00007200: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+00007210: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+00007220: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00007230: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+00007240: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+00007250: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+00007260: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+00007270: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+00007280: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+00007290: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+000072a0: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+000072b0: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+000072c0: 350a 2d67 722c 202d 2d67 6973 7472 622c  5.-gr, --gistrb,
+000072d0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000072e0: 6574 6865 7220 746f 2075 7365 2067 6973  ether to use gis
+000072f0: 745f 7261 696e 626f 7720 6173 2063 6f6c  t_rainbow as col
+00007300: 6f72 206d 6170 206f 6620 646f 7470 6c6f  or map of dotplo
+00007310: 740a 2d6e 2c20 2d2d 6e74 6872 6561 6473  t.-n, --nthreads
+00007320: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00007330: 7468 7265 6164 7320 746f 2075 7365 2069  threads to use i
+00007340: 6e20 7379 6e74 656e 7920 696e 6665 7265  n synteny infere
+00007350: 6e63 652c 2064 6566 6175 6c74 2034 0a60  nce, default 4.`
+00007360: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+00007370: 6077 6764 2076 697a 6020 6361 6e20 7265  `wgd viz` can re
+00007380: 616c 697a 6520 7468 6520 7669 7375 616c  alize the visual
+00007390: 697a 6174 696f 6e20 6f66 202a 4b2a 3c73  ization of *K*<s
+000073a0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+000073b0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+000073c0: 7379 6e74 656e 792e 0a60 6060 0a77 6764  synteny..```.wgd
+000073d0: 2076 697a 2028 6f70 7469 6f6e 290a 2d2d   viz (option).--
+000073e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
 000073f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
 00007400: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
 00007410: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007420: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007430: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 642c  ------------.-d,
-00007440: 202d 2d64 6174 6166 696c 652c 2074 6865   --datafile, the
-00007450: 204b 7320 6461 7461 6669 6c65 2c20 6465   Ks datafile, de
-00007460: 6661 756c 7420 4e6f 6e65 0a2d 6f2c 202d  fault None.-o, -
-00007470: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
-00007480: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
-00007490: 6566 6175 6c74 2077 6764 5f76 697a 0a2d  efault wgd_viz.-
-000074a0: 7372 2c20 2d2d 7370 6169 722c 2074 6865  sr, --spair, the
-000074b0: 2073 7065 6369 6573 2070 6169 7220 746f   species pair to
-000074c0: 2062 6520 706c 6f74 7465 642c 2064 6566   be plotted, def
-000074d0: 6175 6c74 204e 6f6e 652c 2074 6869 7320  ault None, this 
-000074e0: 6f70 7469 6f6e 2063 616e 2062 6520 7072  option can be pr
-000074f0: 6f76 6964 6564 206d 756c 7469 706c 6520  ovided multiple 
-00007500: 7469 6d65 730a 2d66 612c 202d 2d66 6f63  times.-fa, --foc
-00007510: 7573 3261 6c6c 2c20 7365 7420 666f 6361  us2all, set foca
-00007520: 6c20 7370 6563 6965 7320 616e 6420 6c65  l species and le
-00007530: 7420 7370 6563 6965 7320 7061 6972 2074  t species pair t
-00007540: 6f20 6265 2062 6574 7765 656e 2066 6f63  o be between foc
-00007550: 616c 2061 6e64 2061 6c6c 2074 6865 2072  al and all the r
-00007560: 656d 6169 6e69 6e67 2073 7065 6369 6573  emaining species
-00007570: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00007580: 6773 2c20 2d2d 6773 6d61 702c 2074 6865  gs, --gsmap, the
-00007590: 2067 656e 6520 6e61 6d65 2d73 7065 6369   gene name-speci
-000075a0: 6573 206e 616d 6520 6d61 702c 2064 6566  es name map, def
-000075b0: 6175 6c74 204e 6f6e 650a 2d73 702c 202d  ault None.-sp, -
-000075c0: 2d73 7065 6369 6573 7472 6565 2c20 7468  -speciestree, th
-000075d0: 6520 7370 6563 6965 7320 7472 6565 2074  e species tree t
-000075e0: 6f20 7065 7266 6f72 6d20 7261 7465 2063  o perform rate c
-000075f0: 6f72 7265 6374 696f 6e2c 2064 6566 6175  orrection, defau
-00007600: 6c74 204e 6f6e 652c 2069 6620 4e6f 6e65  lt None, if None
-00007610: 2077 6173 2067 6976 656e 2c20 7468 6520   was given, the 
-00007620: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-00007630: 616e 616c 7973 6973 2077 6f75 6c64 2062  analysis would b
-00007640: 6520 6361 6c6c 6564 206f 6666 0a2d 706b  e called off.-pk
-00007650: 2c20 2d2d 706c 6f74 6b64 652c 2066 6c61  , --plotkde, fla
-00007660: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00007670: 7220 746f 2070 6c6f 7420 6b64 6520 6375  r to plot kde cu
-00007680: 7276 6520 7570 6f6e 2068 6973 746f 6772  rve upon histogr
-00007690: 616d 2c20 6966 2074 6865 2066 6c61 6720  am, if the flag 
-000076a0: 7761 7320 7365 742c 206b 6465 2063 7572  was set, kde cur
-000076b0: 7665 2077 6f75 6c64 2062 6520 6164 6465  ve would be adde
-000076c0: 640a 2d72 772c 202d 2d72 6577 6569 6768  d.-rw, --reweigh
-000076d0: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-000076e0: 7768 6574 6865 7220 746f 2072 6563 616c  whether to recal
-000076f0: 6375 6c61 7465 2074 6865 2077 6569 6768  culate the weigh
-00007700: 7420 7065 7220 7370 6563 6965 7320 7061  t per species pa
-00007710: 6972 2c20 6966 2074 6865 2066 6c61 6720  ir, if the flag 
-00007720: 7761 7320 7365 742c 2074 6865 2077 6569  was set, the wei
-00007730: 6768 7420 776f 756c 6420 6265 2072 6563  ght would be rec
-00007740: 616c 6375 6c61 7465 640a 2d6f 722c 202d  alculated.-or, -
-00007750: 2d6f 6e6c 7972 6f6f 746f 7574 2c20 666c  -onlyrootout, fl
-00007760: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00007770: 6572 2074 6f20 6f6e 6c79 2063 6f6e 6475  er to only condu
-00007780: 6374 2072 6174 6520 636f 7272 6563 7469  ct rate correcti
-00007790: 6f6e 2075 7369 6e67 2074 6865 206f 7574  on using the out
-000077a0: 6772 6f75 7020 6174 2072 6f6f 7420 6173  group at root as
-000077b0: 206f 7574 6772 6f75 702c 2069 6620 7468   outgroup, if th
-000077c0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000077d0: 6f6e 6c79 2074 6865 206f 7574 6772 6f75  only the outgrou
-000077e0: 7020 6174 2072 6f6f 7420 776f 756c 6420  p at root would 
-000077f0: 6265 2075 7365 6420 6173 206f 7574 6772  be used as outgr
-00007800: 6f75 700a 2d69 7465 722c 202d 2d65 6d5f  oup.-iter, --em_
-00007810: 6974 6572 6174 696f 6e73 2c20 7468 6520  iterations, the 
-00007820: 6d61 7869 6d75 6d20 454d 2069 7465 7261  maximum EM itera
-00007830: 7469 6f6e 732c 2064 6566 6175 6c74 2032  tions, default 2
-00007840: 3030 0a2d 696e 6974 2c20 2d2d 656d 5f69  00.-init, --em_i
-00007850: 6e69 7469 616c 697a 6174 696f 6e73 2c20  nitializations, 
-00007860: 7468 6520 6d61 7869 6d75 6d20 454d 2069  the maximum EM i
-00007870: 6e69 7469 616c 697a 6174 696f 6e73 2c20  nitializations, 
-00007880: 6465 6661 756c 7420 3230 300a 2d70 7263  default 200.-prc
-00007890: 742c 202d 2d70 726f 6d69 6e65 6e63 655f  t, --prominence_
-000078a0: 6375 746f 6666 2c20 7468 6520 7072 6f6d  cutoff, the prom
-000078b0: 696e 656e 6365 2063 7574 6f66 6620 6f66  inence cutoff of
-000078c0: 2061 6363 6570 7461 626c 6520 7065 616b   acceptable peak
-000078d0: 732c 2064 6566 6175 6c74 2030 2e31 0a2d  s, default 0.1.-
-000078e0: 7268 2c20 2d2d 7265 6c5f 6865 6967 6874  rh, --rel_height
-000078f0: 2c20 7468 6520 7265 6c61 7469 7665 2068  , the relative h
-00007900: 6569 6768 7420 6174 2077 6869 6368 2074  eight at which t
-00007910: 6865 2070 6561 6b20 7769 6474 6820 6973  he peak width is
-00007920: 206d 6561 7375 7265 642c 2064 6566 6175   measured, defau
-00007930: 6c74 2030 2e34 0a2d 736d 2c20 2d2d 7365  lt 0.4.-sm, --se
-00007940: 676d 656e 7473 2c20 7468 6520 7365 676d  gments, the segm
-00007950: 656e 7473 2064 6174 6166 696c 652c 2064  ents datafile, d
-00007960: 6566 6175 6c74 204e 6f6e 650a 2d6d 6c2c  efault None.-ml,
-00007970: 202d 2d6d 696e 6c65 6e2c 2074 6865 206d   --minlen, the m
-00007980: 696e 696d 756d 206c 656e 6774 6820 6f66  inimum length of
-00007990: 2061 2073 6361 6666 6f6c 6420 746f 2062   a scaffold to b
-000079a0: 6520 696e 636c 7564 6564 2069 6e20 646f  e included in do
-000079b0: 7470 6c6f 742c 2064 6566 6175 6c74 202d  tplot, default -
-000079c0: 312c 2069 6620 2d31 2077 6173 2073 6574  1, if -1 was set
-000079d0: 2c20 7468 6520 3130 2520 6f66 2074 6865  , the 10% of the
-000079e0: 206c 6f6e 6765 7374 2073 6361 6666 6f6c   longest scaffol
-000079f0: 6473 2077 696c 6c20 6265 2073 6574 0a2d  ds will be set.-
-00007a00: 6d73 2c20 2d2d 6d61 7873 697a 652c 2074  ms, --maxsize, t
-00007a10: 6865 206d 6178 696d 756d 2066 616d 696c  he maximum famil
-00007a20: 7920 7369 7a65 2074 6f20 6265 2069 6e63  y size to be inc
-00007a30: 6c75 6465 642c 2064 6566 6175 6c74 2032  luded, default 2
-00007a40: 3030 0a2d 6170 2c20 2d2d 616e 6368 6f72  00.-ap, --anchor
-00007a50: 706f 696e 7473 2c20 7468 6520 616e 6368  points, the anch
-00007a60: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
-00007a70: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
-00007a80: 0a2d 6d74 2c20 2d2d 6d75 6c74 6970 6c69  .-mt, --multipli
-00007a90: 636f 6e2c 2074 6865 206d 756c 7469 706c  con, the multipl
-00007aa0: 6963 6f6e 7320 6461 7461 6669 6c65 2c20  icons datafile, 
-00007ab0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6774  default None.-gt
-00007ac0: 2c20 2d2d 6765 6e65 7461 626c 652c 2074  , --genetable, t
-00007ad0: 6865 2067 656e 6520 7461 626c 6520 6461  he gene table da
-00007ae0: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-00007af0: 4e6f 6e65 0a2d 6d67 2c20 2d2d 6d69 6e73  None.-mg, --mins
-00007b00: 6567 6c65 6e2c 2074 6865 206d 696e 696d  eglen, the minim
-00007b10: 756d 206c 656e 6774 6820 6f66 2073 6567  um length of seg
-00007b20: 6d65 6e74 7320 746f 2069 6e63 6c75 6465  ments to include
-00007b30: 2c20 696e 2072 6174 696f 2069 6620 3c3d  , in ratio if <=
-00007b40: 2031 2c20 6465 6661 756c 7420 3130 3030   1, default 1000
-00007b50: 300a 2d6d 676e 2c20 2d2d 6d69 6e67 656e  0.-mgn, --mingen
-00007b60: 656e 756d 2c20 7468 6520 6d69 6e69 6d75  enum, the minimu
-00007b70: 6d20 6e75 6d62 6572 206f 6620 6765 6e65  m number of gene
-00007b80: 7320 666f 7220 6120 7365 676d 656e 7420  s for a segment 
-00007b90: 746f 2062 6520 636f 6e73 6964 6572 6564  to be considered
-00007ba0: 2c20 6465 6661 756c 7420 3330 0a2d 6b72  , default 30.-kr
-00007bb0: 2c20 2d2d 6b65 6570 7265 6475 6e2c 2066  , --keepredun, f
-00007bc0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00007bd0: 6865 7220 746f 206b 6565 7020 7265 6475  her to keep redu
-00007be0: 6e64 616e 7420 6d75 6c74 6970 6c69 636f  ndant multiplico
-00007bf0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
-00007c00: 7761 7320 7365 742c 2074 6865 2072 6564  was set, the red
-00007c10: 756e 6461 6e74 206d 756c 7469 706c 6963  undant multiplic
-00007c20: 6f6e 7320 776f 756c 6420 6265 206b 6570  ons would be kep
-00007c30: 740a 2d65 706b 2c20 2d2d 6578 7472 6170  t.-epk, --extrap
-00007c40: 6172 616e 6f6d 656b 732c 2065 7874 7261  aranomeks, extra
-00007c50: 2070 6172 616e 6f6d 6520 4b73 2064 6174   paranome Ks dat
-00007c60: 6120 746f 2070 6c6f 7420 696e 2074 6865  a to plot in the
-00007c70: 206d 6978 6564 204b 7320 6469 7374 7269   mixed Ks distri
-00007c80: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
-00007c90: 4e6f 6e65 0a2d 7061 672c 202d 2d70 6c6f  None.-pag, --plo
-00007ca0: 7461 7067 6d6d 2c20 666c 6167 206f 7074  tapgmm, flag opt
-00007cb0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00007cc0: 636f 6e64 7563 7420 616e 6420 706c 6f74  conduct and plot
-00007cd0: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00007ce0: 6720 6f66 2061 6e63 686f 7220 4b73 2069  g of anchor Ks i
-00007cf0: 6e20 7468 6520 6d69 7865 6420 4b73 2064  n the mixed Ks d
-00007d00: 6973 7472 6962 7574 696f 6e2c 2069 6620  istribution, if 
-00007d10: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00007d20: 2c20 7468 6520 6d69 7874 7572 6520 6d6f  , the mixture mo
-00007d30: 6465 6c69 6e67 206f 6620 616e 6368 6f72  deling of anchor
-00007d40: 204b 7320 776f 756c 6420 6265 2063 6f6e   Ks would be con
-00007d50: 6475 6374 6564 2061 6e64 2070 6c6f 7474  ducted and plott
-00007d60: 6564 0a2d 7065 6d2c 202d 2d70 6c6f 7465  ed.-pem, --plote
-00007d70: 6c6d 6d2c 2066 6c61 6720 6f70 7469 6f6e  lmm, flag option
-00007d80: 2c20 7768 6574 6865 7220 746f 2063 6f6e  , whether to con
-00007d90: 6475 6374 2061 6e64 2070 6c6f 7420 656c  duct and plot el
-00007da0: 6d6d 206d 6978 7475 7265 206d 6f64 656c  mm mixture model
-00007db0: 696e 6720 6f66 2070 6172 616e 6f6d 6520  ing of paranome 
-00007dc0: 4b73 2069 6e20 7468 6520 6d69 7865 6420  Ks in the mixed 
-00007dd0: 4b73 2064 6973 7472 6962 7574 696f 6e2c  Ks distribution,
-00007de0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00007df0: 2073 6574 2c20 7468 6520 656c 6d6d 206d   set, the elmm m
-00007e00: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00007e10: 6f66 2070 6172 616e 6f6d 6520 4b73 2077  of paranome Ks w
-00007e20: 6f75 6c64 2062 6520 636f 6e64 7563 7465  ould be conducte
-00007e30: 6420 616e 6420 706c 6f74 7465 640a 2d63  d and plotted.-c
-00007e40: 2c20 2d2d 636f 6d70 6f6e 656e 7473 2c20  , --components, 
-00007e50: 7468 6520 7261 6e67 6520 6f66 2074 6865  the range of the
-00007e60: 206e 756d 6265 7220 6f66 2063 6f6d 706f   number of compo
-00007e70: 6e65 6e74 7320 746f 2066 6974 2069 6e20  nents to fit in 
-00007e80: 616e 6368 6f72 204b 7320 6d69 7874 7572  anchor Ks mixtur
-00007e90: 6520 6d6f 6465 6c69 6e67 2c20 6465 6661  e modeling, defa
-00007ea0: 756c 7420 2831 2c34 290a 2d70 7379 2c20  ult (1,4).-psy, 
-00007eb0: 2d2d 706c 6f74 7379 6e2c 2066 6c61 6720  --plotsyn, flag 
-00007ec0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00007ed0: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
-00007ee0: 7379 6e74 656e 7920 706c 6f74 2c20 6f6e  synteny plot, on
-00007ef0: 6c79 2077 6865 6e20 7468 6520 666c 6167  ly when the flag
-00007f00: 2077 6173 2073 6574 2c20 7468 6520 7379   was set, the sy
-00007f10: 6e74 656e 7920 706c 6f74 2077 6f75 6c64  nteny plot would
-00007f20: 2062 6520 7072 6f64 7563 6564 0a2d 6473   be produced.-ds
-00007f30: 2c20 2d2d 646f 7473 697a 652c 2074 6865  , --dotsize, the
-00007f40: 2064 6f74 2073 697a 6520 696e 2064 6f74   dot size in dot
-00007f50: 2070 6c6f 742c 2064 6566 6175 6c74 2030   plot, default 0
-00007f60: 2e33 0a2d 6161 2c20 2d2d 6170 616c 7068  .3.-aa, --apalph
-00007f70: 612c 2074 6865 206f 7061 6369 7479 206f  a, the opacity o
-00007f80: 6620 616e 6368 6f72 2064 6f74 732c 2064  f anchor dots, d
-00007f90: 6566 6175 6c74 2031 0a2d 6861 2c20 2d2d  efault 1.-ha, --
-00007fa0: 686f 616c 7068 612c 2074 6865 206f 7061  hoalpha, the opa
-00007fb0: 6369 7479 206f 6620 686f 6d6f 6c6f 6720  city of homolog 
-00007fc0: 646f 7473 2c20 6465 6661 756c 7420 300a  dots, default 0.
-00007fd0: 2d73 7274 2c20 2d2d 7368 6f77 7265 616c  -srt, --showreal
-00007fe0: 7469 636b 2c20 666c 6167 206f 7074 696f  tick, flag optio
-00007ff0: 6e2c 2077 6865 7468 6572 2074 6f20 7368  n, whether to sh
-00008000: 6f77 2074 6865 2072 6561 6c20 7469 636b  ow the real tick
-00008010: 2069 6e20 6765 6e65 7320 6f72 2062 6173   in genes or bas
-00008020: 6573 2c20 6966 2074 6865 2066 6c61 6720  es, if the flag 
-00008030: 7761 7320 7365 742c 2074 6865 2072 6561  was set, the rea
-00008040: 6c20 7469 636b 2077 6f75 6c64 2062 6520  l tick would be 
-00008050: 7368 6f77 6564 0a2d 746c 732c 202d 2d74  showed.-tls, --t
-00008060: 6963 6b6c 6162 656c 7369 7a65 2c20 7468  icklabelsize, th
-00008070: 6520 6c61 6265 6c20 7369 7a65 206f 6620  e label size of 
-00008080: 7469 636b 2c20 6465 6661 756c 7420 350a  tick, default 5.
-00008090: 2d78 6c2c 202d 2d78 6c69 6d2c 2074 6865  -xl, --xlim, the
-000080a0: 2078 2061 7869 7320 6c69 6d69 7420 6f66   x axis limit of
-000080b0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-000080c0: 0a2d 796c 2c20 2d2d 796c 696d 2c20 7468  .-yl, --ylim, th
-000080d0: 6520 7920 6178 6973 206c 696d 6974 206f  e y axis limit o
-000080e0: 6620 4b73 2064 6973 7472 6962 7574 696f  f Ks distributio
-000080f0: 6e0a 2d61 646f 2c20 2d2d 6164 6a75 7374  n.-ado, --adjust
-00008100: 6f72 7468 6f2c 2066 6c61 6720 6f70 7469  ortho, flag opti
-00008110: 6f6e 2c20 7768 6574 6865 7220 746f 2061  on, whether to a
-00008120: 646a 7573 7420 7468 6520 6869 7374 6f67  djust the histog
-00008130: 7261 6d20 6865 6967 6874 206f 6620 6f72  ram height of or
-00008140: 7468 6f6c 6f67 6f75 7320 4b73 2061 7320  thologous Ks as 
-00008150: 746f 206d 6174 6368 2074 6865 2068 6569  to match the hei
-00008160: 6768 7420 6f66 2070 6172 616c 6f67 6f75  ght of paralogou
-00008170: 7320 4b73 2c20 6966 2074 6865 2066 6c61  s Ks, if the fla
-00008180: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
-00008190: 646a 7573 746d 656e 7420 776f 756c 6420  djustment would 
-000081a0: 6265 2063 6f6e 6475 6374 6564 0a2d 6164  be conducted.-ad
-000081b0: 662c 202d 2d61 646a 7573 7466 6163 746f  f, --adjustfacto
-000081c0: 722c 2074 6865 2061 646a 7573 746d 656e  r, the adjustmen
-000081d0: 7420 6661 6374 6f72 206f 6620 6f72 7468  t factor of orth
-000081e0: 6f6c 6f67 6f75 7320 4b73 2c20 6465 6661  ologous Ks, defa
-000081f0: 756c 7420 302e 350a 2d6f 612c 202d 2d6f  ult 0.5.-oa, --o
-00008200: 6b61 6c70 6861 2c20 7468 6520 6f70 6163  kalpha, the opac
-00008210: 6974 7920 6f66 206f 7274 686f 6c6f 676f  ity of orthologo
-00008220: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
-00008230: 6f6e 2069 6e20 6d69 7865 6420 706c 6f74  on in mixed plot
-00008240: 2c20 6465 6661 756c 7420 302e 350a 2d63  , default 0.5.-c
-00008250: 732c 202d 2d63 6c61 7373 6963 2c20 666c  s, --classic, fl
-00008260: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00008270: 6572 2074 6f20 6472 6177 206d 6978 6564  er to draw mixed
-00008280: 204b 7320 706c 6f74 2069 6e20 6120 636c   Ks plot in a cl
-00008290: 6173 7369 6320 6d61 6e6e 6572 2077 6865  assic manner whe
-000082a0: 7265 2074 6865 2066 756c 6c20 6f72 7468  re the full orth
-000082b0: 6f6c 6f67 6f75 7320 4b73 2064 6973 7472  ologous Ks distr
-000082c0: 6962 7574 696f 6e20 6973 2064 7261 7765  ibution is drawe
-000082d0: 642c 2069 6620 7468 6520 666c 6167 2077  d, if the flag w
-000082e0: 6173 2073 6574 2c20 7468 6520 636c 6173  as set, the clas
-000082f0: 7369 6320 6d69 7865 6420 4b73 2070 6c6f  sic mixed Ks plo
-00008300: 7420 776f 756c 6420 6265 2064 7261 776e  t would be drawn
-00008310: 0a2d 7461 2c20 2d2d 746f 7061 7272 6f77  .-ta, --toparrow
-00008320: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00008330: 6865 7468 6572 2074 6f20 6164 6a75 7374  hether to adjust
-00008340: 2074 6865 2061 7272 6f77 2074 6f20 6265   the arrow to be
-00008350: 2061 7420 7468 6520 746f 7020 6f66 2074   at the top of t
-00008360: 6865 2070 6c6f 742c 2069 6e73 7465 6164  he plot, instead
-00008370: 206f 6620 6265 696e 6720 636f 6f72 6469   of being coordi
-00008380: 6e61 7465 6420 6173 2074 6865 204b 4445  nated as the KDE
-00008390: 206f 6620 7468 6520 6f72 7468 6f6c 6f67   of the ortholog
-000083a0: 6f75 7320 4b73 2064 6973 7472 6962 7574  ous Ks distribut
-000083b0: 696f 6e2c 2069 6620 7468 6520 666c 6167  ion, if the flag
-000083c0: 2077 6173 2073 6574 2c20 7468 6520 6172   was set, the ar
-000083d0: 726f 7720 776f 756c 6420 6265 2073 6574  row would be set
-000083e0: 2061 7420 7468 6520 746f 700a 2d6e 612c   at the top.-na,
-000083f0: 202d 2d6e 6f64 6561 7665 7261 6765 642c   --nodeaveraged,
-00008400: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00008410: 6574 6865 7220 746f 2075 7365 206e 6f64  ether to use nod
-00008420: 652d 6176 6572 6167 6564 206d 6574 686f  e-averaged metho
-00008430: 6420 666f 7220 6465 2d72 6564 756e 6461  d for de-redunda
-00008440: 6e63 792c 2069 6620 7468 6520 666c 6167  ncy, if the flag
-00008450: 2077 6173 2073 6574 2c20 7468 6520 6e6f   was set, the no
-00008460: 6465 2d61 7665 7261 6765 6420 6d65 7468  de-averaged meth
-00008470: 6f64 2077 6f75 6c64 2062 6520 696e 6974  od would be init
-00008480: 6961 7465 640a 2d62 732c 202d 2d62 6f6f  iated.-bs, --boo
-00008490: 7473 7472 6170 2c20 7468 6520 6e75 6d62  tstrap, the numb
-000084a0: 6572 206f 6620 626f 6f74 7374 7261 7020  er of bootstrap 
-000084b0: 7265 706c 6963 6174 6573 206f 6620 6f72  replicates of or
-000084c0: 7468 6f6c 6f67 204b 7320 6469 7374 7269  tholog Ks distri
-000084d0: 6275 7469 6f6e 2069 6e20 6d69 7865 6420  bution in mixed 
-000084e0: 706c 6f74 0a2d 6772 2c20 2d2d 6769 7374  plot.-gr, --gist
-000084f0: 7262 2c20 666c 6167 206f 7074 696f 6e2c  rb, flag option,
-00008500: 2077 6865 7468 6572 2074 6f20 7573 6520   whether to use 
-00008510: 6769 7374 5f72 6169 6e62 6f77 2061 7320  gist_rainbow as 
-00008520: 636f 6c6f 7220 6d61 7020 6f66 2064 6f74  color map of dot
-00008530: 706c 6f74 0a2d 6e2c 202d 2d6e 7468 7265  plot.-n, --nthre
-00008540: 6164 732c 2074 6865 206e 756d 6265 7220  ads, the number 
-00008550: 6f66 2074 6872 6561 6473 2074 6f20 7573  of threads to us
-00008560: 6520 696e 2062 6f6f 7473 7472 6170 2073  e in bootstrap s
-00008570: 616d 706c 696e 672c 2064 6566 6175 6c74  ampling, default
-00008580: 2031 0a60 6060 0a0a 2323 2055 7361 6765   1.```..## Usage
-00008590: 0a0a 4865 7265 2077 6520 7072 6f76 6964  ..Here we provid
-000085a0: 6564 2074 6865 2062 6173 6963 2075 7361  ed the basic usa
-000085b0: 6765 2066 6f72 2065 6163 6820 7072 6f67  ge for each prog
-000085c0: 7261 6d20 616e 6420 7468 6520 7265 6c65  ram and the rele
-000085d0: 7661 6e74 2070 6172 616d 6574 6572 7320  vant parameters 
-000085e0: 616e 6420 7375 6767 6573 7469 6f6e 7320  and suggestions 
-000085f0: 6f6e 2070 6172 616d 6574 6572 697a 6174  on parameterizat
-00008600: 696f 6e2e 2041 2072 656d 696e 6465 7220  ion. A reminder 
-00008610: 7468 6174 2074 6865 2067 6976 656e 2063  that the given c
-00008620: 6f72 6573 2061 6e64 2074 6872 6561 6473  ores and threads
-00008630: 2063 616e 2073 6967 6e69 6669 6361 6e74   can significant
-00008640: 6c79 2069 6d70 6163 7420 7468 6520 7275  ly impact the ru
-00008650: 6e20 7469 6d65 2061 6e64 2074 6875 7320  n time and thus 
-00008660: 7765 2061 6464 6564 2073 6f6d 6520 7265  we added some re
-00008670: 706f 7274 2069 6e66 6f72 6d61 7469 6f6e  port information
-00008680: 2070 6572 7461 696e 696e 6720 746f 2074   pertaining to t
-00008690: 6865 2073 7973 7465 6d20 6f66 2075 7365  he system of use
-000086a0: 7273 2074 6f20 6661 6369 6c69 7461 7465  rs to facilitate
-000086b0: 2074 6865 2065 6666 6963 6965 6e74 2073   the efficient s
-000086c0: 6574 7469 6e67 206f 6620 7468 7265 6164  etting of thread
-000086d0: 7320 616e 6420 6d65 6d6f 7279 2e20 5468  s and memory. Th
-000086e0: 6520 6c6f 6769 6361 6c20 4350 5573 2072  e logical CPUs r
-000086f0: 6570 6f72 7465 6420 7265 7072 6573 656e  eported represen
-00008700: 7473 2074 6865 206e 756d 6265 7220 6f66  ts the number of
-00008710: 2070 6879 7369 6361 6c20 636f 7265 7320   physical cores 
-00008720: 6d75 6c74 6970 6c69 6564 2062 7920 7468  multiplied by th
-00008730: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
-00008740: 6164 7320 7468 6174 2063 616e 2072 756e  ads that can run
-00008750: 206f 6e20 6561 6368 2063 6f72 652c 2061   on each core, a
-00008760: 6c73 6f20 6b6e 6f77 6e20 6173 2048 7970  lso known as Hyp
-00008770: 6572 2054 6872 6561 6469 6e67 2e20 5468  er Threading. Th
-00008780: 6520 6e75 6d62 6572 206f 6620 6c6f 6769  e number of logi
-00008790: 6361 6c20 4350 5573 206d 6179 206e 6f74  cal CPUs may not
-000087a0: 206e 6563 6573 7361 7269 6c79 2062 6520   necessarily be 
-000087b0: 6571 7569 7661 6c65 6e74 2074 6f20 7468  equivalent to th
-000087c0: 6520 6163 7475 616c 206e 756d 6265 7220  e actual number 
-000087d0: 6f66 2043 5055 7320 7468 6520 6375 7272  of CPUs the curr
-000087e0: 656e 7420 7072 6f63 6573 7320 6361 6e20  ent process can 
-000087f0: 7573 652e 2054 6865 2061 7661 696c 6162  use. The availab
-00008800: 6c65 206d 656d 6f72 7920 7265 6665 7273  le memory refers
-00008810: 2074 6f20 7468 6520 6d65 6d6f 7279 2074   to the memory t
-00008820: 6861 7420 6361 6e20 6265 2067 6976 656e  hat can be given
-00008830: 2069 6e73 7461 6e74 6c79 2074 6f20 7072   instantly to pr
-00008840: 6f63 6573 7365 7320 7769 7468 6f75 7420  ocesses without 
-00008850: 7468 6520 7379 7374 656d 2067 6f69 6e67  the system going
-00008860: 2069 6e74 6f20 7377 6170 2061 6e64 2072   into swap and r
-00008870: 6566 6c65 6374 7320 7468 6520 6163 7475  eflects the actu
-00008880: 616c 206d 656d 6f72 7920 6176 6169 6c61  al memory availa
-00008890: 626c 652e 2054 6865 2066 7265 6520 6d65  ble. The free me
-000088a0: 6d6f 7279 2072 6566 6572 7320 746f 2074  mory refers to t
-000088b0: 6865 206d 656d 6f72 7920 6e6f 7420 6265  he memory not be
-000088c0: 696e 6720 7573 6564 2061 7420 616c 6c20  ing used at all 
-000088d0: 287a 6572 6f65 6429 2074 6861 7420 6973  (zeroed) that is
-000088e0: 2072 6561 6469 6c79 2061 7661 696c 6162   readily availab
-000088f0: 6c65 2e20 5468 6520 6465 7363 7269 7074  le. The descript
-00008900: 696f 6e20 6162 6f76 6520 7265 6665 7273  ion above refers
-00008910: 2074 6f20 7468 6520 646f 6375 6d65 6e74   to the document
-00008920: 6174 696f 6e20 6f66 2060 7073 7574 696c  ation of `psutil
-00008930: 602e 0a0a 2323 2320 7767 6420 646d 640a  `...### wgd dmd.
-00008940: 0a2a 2a54 6865 2064 656c 696e 6561 7469  .**The delineati
-00008950: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
-00008960: 6e6f 6d65 2a2a 0a60 6060 0a77 6764 2064  nome**.```.wgd d
-00008970: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00008980: 7275 6c65 6120 2d49 2032 202d 6520 3165  rulea -I 2 -e 1e
-00008990: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
-000089a0: 3520 282d 6e6e 2920 282d 2d74 6f5f 7374  5 (-nn) (--to_st
-000089b0: 6f70 2920 282d 2d63 6473 2920 282d 6e20  op) (--cds) (-n 
-000089c0: 3429 2028 2d6f 2077 6764 5f64 6d64 2920  4) (-o wgd_dmd) 
-000089d0: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-000089e0: 0a60 6060 200a 0a4e 6f74 6520 7468 6174  .``` ..Note that
-000089f0: 2077 6520 646f 6e27 7420 7072 6f76 6964   we don't provid
-00008a00: 6520 7468 6520 6461 7461 206f 6620 7468  e the data of th
-00008a10: 6973 2063 6f64 696e 6720 7365 7175 656e  is coding sequen
-00008a20: 6365 2028 6364 7329 2066 696c 6520 6041  ce (cds) file `A
-00008a30: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00008a40: 6160 2062 7574 2069 7420 6361 6e20 6265  a` but it can be
-00008a50: 2065 6173 696c 7920 646f 776e 6c6f 6164   easily download
-00008a60: 6564 2061 7420 5b50 6879 746f 7a6f 6d65  ed at [Phytozome
-00008a70: 5d28 6874 7470 733a 2f2f 7068 7974 6f7a  ](https://phytoz
-00008a80: 6f6d 652d 6e65 7874 2e6a 6769 2e64 6f65  ome-next.jgi.doe
-00008a90: 2e67 6f76 2f69 6e66 6f2f 4163 6f65 7275  .gov/info/Acoeru
-00008aa0: 6c65 615f 7633 5f31 2920 2873 616d 6520  lea_v3_1) (same 
-00008ab0: 666f 7220 6f74 6865 7220 5573 6167 6520  for other Usage 
-00008ac0: 646f 6329 2e20 4e6f 7465 2074 6861 7420  doc). Note that 
-00008ad0: 696e 2074 6865 2069 7373 7565 7320 736f  in the issues so
-00008ae0: 6d65 2075 7365 7273 2064 6964 6e27 7420  me users didn't 
-00008af0: 646f 776e 6c6f 6164 2074 6865 2060 4163  download the `Ac
-00008b00: 6f65 7275 6c65 615f 3332 325f 7633 2e31  oerulea_322_v3.1
-00008b10: 2e63 6473 5f70 7269 6d61 7279 5472 616e  .cds_primaryTran
-00008b20: 7363 7269 7074 4f6e 6c79 2e66 612e 677a  scriptOnly.fa.gz
-00008b30: 6020 6275 7420 696e 7374 6561 6420 7468  ` but instead th
-00008b40: 6520 6041 636f 6572 756c 6561 5f33 3232  e `Acoerulea_322
-00008b50: 5f76 332e 312e 6364 732e 6661 2e67 7a60  _v3.1.cds.fa.gz`
-00008b60: 2066 696c 652e 2046 6f72 2074 6865 2063   file. For the c
-00008b70: 6f6e 7374 7275 6374 696f 6e20 6f66 2077  onstruction of w
-00008b80: 686f 6c65 2070 6172 616e 6f6d 6520 2a4b  hole paranome *K
-00008b90: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00008ba0: 7374 7269 6275 7469 6f6e 2c20 2a2a 6f6e  stribution, **on
-00008bb0: 6c79 206f 6e65 2074 7261 6e73 6372 6970  ly one transcrip
-00008bc0: 7420 2874 6865 2070 7269 6d61 7279 206f  t (the primary o
-00008bd0: 6e65 2920 7065 7220 6765 6e65 2073 686f  ne) per gene sho
-00008be0: 756c 6420 6265 2069 6e63 6c75 6465 642a  uld be included*
-00008bf0: 2a20 7375 6368 2074 6861 7420 7468 6520  * such that the 
-00008c00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00008c10: 6973 2072 6561 6c6c 7920 696e 6469 6361  is really indica
-00008c20: 7469 6e67 2074 6865 2061 6765 206f 6620  ting the age of 
-00008c30: 6765 6e65 2064 7570 6c69 6361 7469 6f6e  gene duplication
-00008c40: 2065 7665 6e74 2c20 696e 7374 6561 6420   event, instead 
-00008c50: 6f66 2061 6c74 6572 6e61 7469 7665 2073  of alternative s
-00008c60: 706c 6963 696e 672e 2054 7261 6e73 6372  plicing. Transcr
-00008c70: 6970 746f 6d65 2064 6174 6120 7368 6f75  iptome data shou
-00008c80: 6c64 2062 6520 6361 7265 6675 6c6c 7920  ld be carefully 
-00008c90: 7472 6561 7465 6420 7769 7468 2064 652d  treated with de-
-00008ca0: 7265 6475 6e64 616e 6379 2073 6f20 6173  redundancy so as
-00008cb0: 2074 6f20 7265 6475 6365 2074 6865 2066   to reduce the f
-00008cc0: 616c 7365 2070 6f73 6974 6976 6520 6475  alse positive du
-00008cd0: 706c 6963 6174 696f 6e20 6275 6d70 2063  plication bump c
-00008ce0: 6175 7365 6420 6279 2070 6572 7661 7369  aused by pervasi
-00008cf0: 7665 2061 6c74 6572 6e61 7469 7665 2073  ve alternative s
-00008d00: 706c 6963 696e 672e 2048 6572 6520 7468  plicing. Here th
-00008d10: 6520 696e 666c 6174 696f 6e20 6661 6374  e inflation fact
-00008d20: 6f72 2070 6172 616d 6574 6572 2c20 6769  or parameter, gi
-00008d30: 7665 6e20 6279 2060 2d49 6020 6f72 2060  ven by `-I` or `
-00008d40: 2d2d 696e 666c 6174 696f 6e60 2c20 6166  --inflation`, af
-00008d50: 6665 6374 7320 7468 6520 6772 616e 756c  fects the granul
-00008d60: 6172 6974 7920 6f72 2072 6573 6f6c 7574  arity or resolut
-00008d70: 696f 6e20 6f66 2074 6865 2063 6c75 7374  ion of the clust
-00008d80: 6572 696e 6720 6f75 7463 6f6d 6520 616e  ering outcome an
-00008d90: 6420 696d 706c 6963 6974 6c79 2063 6f6e  d implicitly con
-00008da0: 7472 6f6c 6c73 2074 6865 206e 756d 6265  trolls the numbe
-00008db0: 7220 6f66 2063 6c75 7374 6572 732c 2077  r of clusters, w
-00008dc0: 6974 6820 6c6f 7720 7661 6c75 6573 2073  ith low values s
-00008dd0: 7563 6820 6173 2031 2e33 206f 7220 312e  uch as 1.3 or 1.
-00008de0: 3420 6c65 6164 696e 6720 746f 2066 6577  4 leading to few
-00008df0: 6572 2062 7574 206c 6172 6765 7220 636c  er but larger cl
-00008e00: 7573 7465 7273 2061 6e64 2068 6967 6820  usters and high 
-00008e10: 7661 6c75 6573 2073 7563 6820 6173 2035  values such as 5
-00008e20: 206f 7220 3620 6c65 6164 696e 6720 746f   or 6 leading to
-00008e30: 206d 6f72 6520 6275 7420 736d 616c 6c65   more but smalle
-00008e40: 7220 636c 7573 7465 7273 2e20 5765 2073  r clusters. We s
-00008e50: 6574 2074 6865 2064 6566 6175 6c74 2076  et the default v
-00008e60: 616c 7565 2061 7320 3220 6173 2073 7567  alue as 2 as sug
-00008e70: 6765 7374 6564 2062 7920 5b4d 434c 5d28  gested by [MCL](
-00008e80: 6874 7470 733a 2f2f 6d69 6361 6e73 2e6f  https://micans.o
-00008e90: 7267 2f6d 636c 2f29 2e20 5468 6520 652d  rg/mcl/). The e-
-00008ea0: 7661 6c75 6520 6375 742d 6f66 6620 666f  value cut-off fo
-00008eb0: 7220 7365 7175 656e 6365 2073 696d 696c  r sequence simil
-00008ec0: 6172 6974 792c 2067 6976 656e 2062 7920  arity, given by 
-00008ed0: 602d 6560 206f 7220 602d 2d65 7661 6c60  `-e` or `--eval`
-00008ee0: 2c20 7768 6963 6820 6465 6e6f 7465 7320  , which denotes 
-00008ef0: 7468 6520 6578 7065 6374 6564 2076 616c  the expected val
-00008f00: 7565 206f 6620 7468 6520 6869 7420 7175  ue of the hit qu
-00008f10: 616e 7469 6669 6573 2074 6865 206e 756d  antifies the num
-00008f20: 6265 7220 6f66 2061 6c69 676e 6d65 6e74  ber of alignment
-00008f30: 7320 6f66 2073 696d 696c 6172 206f 7220  s of similar or 
-00008f40: 6265 7474 6572 2071 7561 6c69 7479 2074  better quality t
-00008f50: 6861 7420 796f 7520 6578 7065 6374 2074  hat you expect t
-00008f60: 6f20 6669 6e64 2073 6561 7263 6869 6e67  o find searching
-00008f70: 2074 6869 7320 7175 6572 7920 6167 6169   this query agai
-00008f80: 6e73 7420 6120 6461 7461 6261 7365 206f  nst a database o
-00008f90: 6620 7261 6e64 6f6d 2073 6571 7565 6e63  f random sequenc
-00008fa0: 6573 2074 6865 2073 616d 6520 7369 7a65  es the same size
-00008fb0: 2061 7320 7468 6520 6163 7475 616c 2074   as the actual t
-00008fc0: 6172 6765 7420 6461 7461 6261 7365 2c20  arget database, 
-00008fd0: 6973 2074 6865 206b 6579 2070 6172 616d  is the key param
-00008fe0: 6574 6572 206d 6561 7375 7269 6e67 2074  eter measuring t
-00008ff0: 6865 2073 6967 6e69 6669 6361 6e63 6520  he significance 
-00009000: 6f66 2061 2068 6974 2c20 7768 6963 6820  of a hit, which 
-00009010: 6973 2073 6574 2068 6572 6520 6173 2064  is set here as d
-00009020: 6566 6175 6c74 2031 652d 3130 2e20 4e6f  efault 1e-10. No
-00009030: 7465 2074 6861 7420 5b44 4941 4d4f 4e44  te that [DIAMOND
-00009040: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-00009050: 2e63 6f6d 2f62 6275 6368 6669 6e6b 2f64  .com/bbuchfink/d
-00009060: 6961 6d6f 6e64 2f77 696b 692f 312e 2d54  iamond/wiki/1.-T
-00009070: 7574 6f72 6961 6c29 2069 7473 656c 6620  utorial) itself 
-00009080: 6279 2064 6566 6175 6c74 206f 6e6c 7920  by default only 
-00009090: 7265 706f 7274 7320 616c 6c20 616c 6967  reports all alig
-000090a0: 6e6d 656e 7473 2077 6974 6820 652d 7661  nments with e-va
-000090b0: 6c75 6520 3c20 302e 3030 312e 2054 6865  lue < 0.001. The
-000090c0: 2070 6572 6365 6e74 6167 6520 6f66 2075   percentage of u
-000090d0: 7070 6572 2068 6974 7320 7573 6564 2066  pper hits used f
-000090e0: 6f72 2067 656e 6520 6c65 6e67 7468 206e  or gene length n
-000090f0: 6f72 6d61 6c69 7a61 7469 6f6e 2c20 6769  ormalization, gi
-00009100: 7665 6e20 6279 2060 2d6e 7060 206f 7220  ven by `-np` or 
-00009110: 602d 2d6e 6f72 6d61 6c69 7a65 6470 6572  `--normalizedper
-00009120: 6365 6e74 602c 2077 6869 6368 2064 6574  cent`, which det
-00009130: 6572 6d69 6e65 7320 7468 6520 7570 7065  ermines the uppe
-00009140: 7220 7065 7263 656e 7469 6c65 206f 6620  r percentile of 
-00009150: 6869 7473 2070 6572 2062 696e 2028 6361  hits per bin (ca
-00009160: 7465 676f 7269 7a65 6420 6279 2067 656e  tegorized by gen
-00009170: 6520 6c65 6e67 7468 2920 7573 6564 2069  e length) used i
-00009180: 6e20 7468 6520 6669 7420 6f66 206c 696e  n the fit of lin
-00009190: 6561 7220 7265 6772 6573 7369 6f6e 2c20  ear regression, 
-000091a0: 636f 6e73 6964 6572 696e 6720 7468 6174  considering that
-000091b0: 206e 6f74 2061 6c6c 2068 6974 7320 7065   not all hits pe
-000091c0: 7220 6269 6e20 7368 6f77 2061 7070 6172  r bin show appar
-000091d0: 656e 7420 6c69 6e65 6172 2072 656c 6174  ent linear relat
-000091e0: 696f 6e73 6869 702c 2069 7320 7365 7420  ionship, is set 
-000091f0: 6173 2064 6566 6175 6c74 2035 2c20 696e  as default 5, in
-00009200: 6469 6361 7469 6e67 2074 6865 2075 7361  dicating the usa
-00009210: 6765 206f 6620 746f 7020 3525 2068 6974  ge of top 5% hit
-00009220: 7320 7065 7220 6269 6e2e 2054 6865 206e  s per bin. The n
-00009230: 756d 6265 7220 6f66 2062 696e 7320 6469  umber of bins di
-00009240: 7669 6465 6420 696e 2067 656e 6520 6c65  vided in gene le
-00009250: 6e67 7468 206e 6f72 6d61 6c69 7a61 7469  ngth normalizati
-00009260: 6f6e 2c20 6769 7665 6e20 6279 2060 2d62  on, given by `-b
-00009270: 7360 206f 7220 602d 2d62 696e 7360 2c20  s` or `--bins`, 
-00009280: 6465 7465 726d 696e 6573 2074 6865 206e  determines the n
-00009290: 756d 6265 7220 6f66 2062 696e 7320 746f  umber of bins to
-000092a0: 2063 6174 6567 6f72 697a 6520 7468 6520   categorize the 
-000092b0: 6765 6e65 206c 656e 6774 682c 2069 7320  gene length, is 
-000092c0: 7365 7420 6173 2064 6566 6175 6c74 2031  set as default 1
-000092d0: 3030 2e20 5468 6520 7061 7261 6d65 7465  00. The paramete
-000092e0: 7220 602d 6e6e 6020 6f72 2060 2d2d 6e6f  r `-nn` or `--no
-000092f0: 6e6f 726d 616c 697a 6174 696f 6e60 2063  normalization` c
-00009300: 616e 2062 6520 7365 7420 746f 2063 616c  an be set to cal
-00009310: 6c20 6f66 6620 7468 6520 6e6f 726d 616c  l off the normal
-00009320: 697a 6174 696f 6e20 7072 6f63 6573 732c  ization process,
-00009330: 2061 6c74 686f 7567 6820 6974 2773 2073   although it's s
-00009340: 7567 6765 7374 6564 2074 6f20 636f 6e64  uggested to cond
-00009350: 7563 7420 7468 6520 6e6f 726d 616c 697a  uct the normaliz
-00009360: 6174 696f 6e20 746f 2061 6371 7569 7265  ation to acquire
-00009370: 206d 6f72 6520 6163 6375 7261 7465 2067   more accurate g
-00009380: 656e 6520 6661 6d69 6c79 2063 6c75 7374  ene family clust
-00009390: 6572 696e 6720 7265 7375 6c74 2e20 5468  ering result. Th
-000093a0: 6520 7061 7261 6d65 7465 7273 2060 2d2d  e parameters `--
-000093b0: 746f 5f73 746f 7060 2061 6e64 2060 2d2d  to_stop` and `--
-000093c0: 6364 7360 2063 6f6e 7472 6f6c 2074 6865  cds` control the
-000093d0: 2062 6568 6176 696f 7572 206f 6620 7472   behaviour of tr
-000093e0: 616e 736c 6174 696e 6720 636f 6469 6e67  anslating coding
-000093f0: 2073 6571 7565 6e63 6520 696e 746f 2061   sequence into a
-00009400: 6d69 6e6f 2061 6369 6420 7365 7175 656e  mino acid sequen
-00009410: 6365 2e20 4966 2074 6865 2060 2d2d 746f  ce. If the `--to
-00009420: 5f73 746f 7060 2077 6173 2073 6574 2c20  _stop` was set, 
-00009430: 7468 6520 7472 616e 736c 6174 696f 6e20  the translation 
-00009440: 776f 756c 6420 6265 2074 6572 6d69 6e61  would be termina
-00009450: 7465 6420 6174 2074 6865 2066 6972 7374  ted at the first
-00009460: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
-00009470: 6f64 6f6e 2c20 6f74 6865 7277 6973 6520  odon, otherwise 
-00009480: 7468 6520 7472 616e 736c 6174 696f 6e20  the translation 
-00009490: 776f 756c 6420 7369 6d70 6c79 2073 6b69  would simply ski
-000094a0: 7020 616e 7920 7374 6f70 2063 6f64 6f6e  p any stop codon
-000094b0: 732e 2049 6620 7468 6520 602d 2d63 6473  s. If the `--cds
-000094c0: 6020 7761 7320 7365 742c 2073 6571 7565  ` was set, seque
-000094d0: 6e63 6573 2074 6861 7420 646f 6573 6e27  nces that doesn'
-000094e0: 7420 7374 6172 7420 7769 7468 2061 2076  t start with a v
-000094f0: 616c 6964 2073 7461 7274 2063 6f64 6f6e  alid start codon
-00009500: 2c20 6f72 2063 6f6e 7461 696e 7320 6d6f  , or contains mo
-00009510: 7265 2074 6861 6e20 6f6e 6520 696e 2d66  re than one in-f
-00009520: 7261 6d65 2073 746f 7020 636f 646f 6e2c  rame stop codon,
-00009530: 206f 7220 6973 206e 6f74 2064 6976 6964   or is not divid
-00009540: 6162 6c65 2062 7920 332c 2077 6f75 6c64  able by 3, would
-00009550: 2062 6520 7369 6d70 6c79 2064 726f 7070   be simply dropp
-00009560: 6564 2c20 7375 6368 2074 6861 7420 6f6e  ed, such that on
-00009570: 6c79 2073 7472 6963 7420 636f 6d70 6c65  ly strict comple
-00009580: 7465 2063 6f64 696e 6720 7365 7175 656e  te coding sequen
-00009590: 6365 7320 776f 756c 6420 6265 2069 6e63  ces would be inc
-000095a0: 6c75 6465 6420 696e 2074 6865 2073 7562  luded in the sub
-000095b0: 7365 7175 656e 7420 616e 616c 7973 6973  sequent analysis
-000095c0: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
-000095d0: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
-000095e0: 2062 7920 7468 6520 6f70 7469 6f6e 2060   by the option `
-000095f0: 2d6e 6020 6f72 2060 2d2d 6e74 6872 6561  -n` or `--nthrea
-00009600: 6473 6020 6361 6e20 6265 2073 6574 2074  ds` can be set t
-00009610: 6f20 6163 6365 6c65 7261 7465 2074 6865  o accelerate the
-00009620: 2063 616c 6375 6c61 7469 6f6e 2077 6974   calculation wit
-00009630: 6869 6e20 6064 6961 6d6f 6e64 602e 2054  hin `diamond`. T
-00009640: 6865 2064 6972 6563 746f 7279 206f 6620  he directory of 
-00009650: 6f75 7470 7574 206f 7220 696e 7465 726d  output or interm
-00009660: 6564 6961 7465 2066 696c 6573 2069 7320  ediate files is 
-00009670: 6465 7465 726d 696e 6564 2062 7920 7468  determined by th
-00009680: 6520 7061 7261 6d65 7465 7220 602d 6f60  e parameter `-o`
-00009690: 206f 7220 602d 2d6f 7574 6469 7260 2c20   or `--outdir`, 
-000096a0: 616e 6420 602d 7460 206f 7220 602d 2d74  and `-t` or `--t
-000096b0: 6d70 6469 7260 2c20 7768 6963 6820 7769  mpdir`, which wi
-000096c0: 6c6c 2062 6520 6372 6561 7465 6420 6279  ll be created by
-000096d0: 2074 6865 2070 726f 6772 616d 2069 7473   the program its
-000096e0: 656c 6620 616e 6420 6265 206f 7665 7277  elf and be overw
-000096f0: 7269 7474 656e 2069 6620 7468 6520 666f  ritten if the fo
-00009700: 6c64 6572 2068 6173 2061 6c72 6561 6479  lder has already
-00009710: 2062 6565 6e20 6372 6561 7465 642e 204e   been created. N
-00009720: 6f74 6520 7468 6174 2074 6865 2073 6f66  ote that the sof
-00009730: 7477 6172 6520 6064 6961 6d6f 6e64 6020  tware `diamond` 
-00009740: 7368 6f75 6c64 2062 6520 7072 652d 696e  should be pre-in
-00009750: 7374 616c 6c65 6420 616e 6420 7365 7420  stalled and set 
-00009760: 696e 2074 6865 2065 6e76 6972 6f6e 6d65  in the environme
-00009770: 6e74 2070 6174 6820 696e 2061 6c6c 2074  nt path in all t
-00009780: 6865 2061 6e61 6c79 7369 7320 7065 7266  he analysis perf
-00009790: 6f72 6d65 6420 6279 2060 7767 6420 646d  ormed by `wgd dm
-000097a0: 6460 2065 7863 6570 7420 666f 7220 7468  d` except for th
-000097b0: 6520 636f 6c6c 696e 6561 7220 636f 616c  e collinear coal
-000097c0: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
-000097d0: 2e0a 0a2a 2a57 6520 7375 6767 6573 7420  ...**We suggest 
-000097e0: 7468 6174 2074 6865 2064 6566 6175 6c74  that the default
-000097f0: 2073 6574 7469 6e67 2069 6e20 7768 6963   setting in whic
-00009800: 6820 7468 6520 696e 666c 6174 696f 6e20  h the inflation 
-00009810: 6661 6374 6f72 2069 7320 7365 7420 6173  factor is set as
-00009820: 2032 2c20 652d 7661 6c75 6520 6375 742d   2, e-value cut-
-00009830: 6f66 6620 6173 2031 652d 3130 2061 6e64  off as 1e-10 and
-00009840: 206f 7468 6572 2070 6172 616d 6574 6572   other parameter
-00009850: 7320 696e 2064 6566 6175 6c74 2069 7320  s in default is 
-00009860: 6120 676f 6f64 2073 7461 7274 696e 6720  a good starting 
-00009870: 706f 696e 742c 2075 6e6c 6573 7320 796f  point, unless yo
-00009880: 7520 7370 6563 6966 6963 616c 6c79 2077  u specifically w
-00009890: 616e 7420 746f 2065 7870 6c6f 7265 2074  ant to explore t
-000098a0: 6865 2065 6666 6563 7473 206f 6620 6469  he effects of di
-000098b0: 6666 6572 656e 7420 7061 7261 6d65 7465  fferent paramete
-000098c0: 7273 2e2a 2a20 5375 6368 2074 6861 7420  rs.** Such that 
-000098d0: 7468 6520 636f 6d6d 616e 6420 666f 7220  the command for 
-000098e0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
-000098f0: 6f66 2077 686f 6c65 2070 6172 616e 6f6d  of whole paranom
-00009900: 6520 6973 2073 696d 706c 7920 6173 2062  e is simply as b
-00009910: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00009920: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00009930: 7275 6c65 610a 6060 600a 0a2a 2a54 6865  rulea.```..**The
-00009940: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
-00009950: 5242 4873 2a2a 0a60 6060 0a77 6764 2064  RBHs**.```.wgd d
-00009960: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
-00009970: 7565 6e63 6532 202d 6520 3165 2d31 3020  uence2 -e 1e-10 
-00009980: 2d62 7320 3130 3020 2d6e 7020 3520 282d  -bs 100 -np 5 (-
-00009990: 6e6e 2920 282d 6e20 3429 2028 2d63 2030  nn) (-n 4) (-c 0
-000099a0: 2e39 2920 282d 2d6f 6766 6f72 6d61 7429  .9) (--ogformat)
-000099b0: 2028 2d2d 746f 5f73 746f 7029 2028 2d2d   (--to_stop) (--
-000099c0: 6364 7329 2028 2d6f 2077 6764 5f64 6d64  cds) (-o wgd_dmd
-000099d0: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
-000099e0: 7029 0a60 6060 0a0a 546f 2064 656c 696e  p).```..To delin
-000099f0: 6561 7465 2052 4248 7320 6265 7477 6565  eate RBHs betwee
-00009a00: 6e20 7477 6f20 6364 7320 7365 7175 656e  n two cds sequen
-00009a10: 6365 2066 696c 6573 2c20 7468 6520 7265  ce files, the re
-00009a20: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
-00009a30: 2069 7320 6d6f 7374 6c79 2074 6865 2073   is mostly the s
-00009a40: 616d 6520 6173 2077 686f 6c65 2070 6172  ame as whole par
-00009a50: 616e 6f6d 6520 696e 6665 7265 6e63 652c  anome inference,
-00009a60: 2065 7863 6570 7420 666f 7220 7468 6520   except for the 
-00009a70: 7061 7261 6d65 7465 7220 602d 6360 206f  parameter `-c` o
-00009a80: 7220 602d 2d63 7363 6f72 6560 2c20 7768  r `--cscore`, wh
-00009a90: 6963 6820 7261 6e67 6573 2062 6574 7765  ich ranges betwe
-00009aa0: 656e 2030 2061 6e64 2031 2061 6e64 2069  en 0 and 1 and i
-00009ab0: 7320 7573 6564 2074 6f20 7265 6c61 7820  s used to relax 
-00009ac0: 7468 6520 7369 6d69 6c61 7269 7479 2063  the similarity c
-00009ad0: 7574 6f66 6620 6672 6f6d 2074 6865 2065  utoff from the e
-00009ae0: 7863 6c75 7369 7665 2072 6563 6970 726f  xclusive recipro
-00009af0: 6361 6c20 6265 7374 2068 6974 7320 746f  cal best hits to
-00009b00: 2061 2063 6572 7461 696e 2072 6174 696f   a certain ratio
-00009b10: 2061 7320 746f 2074 6865 2062 6573 7420   as to the best 
-00009b20: 6869 7473 2e20 466f 7220 696e 7374 616e  hits. For instan
-00009b30: 6365 2c20 6966 2074 6865 2067 656e 6520  ce, if the gene 
-00009b40: 6231 2066 726f 6d20 6765 6e6f 6d65 2042  b1 from genome B
-00009b50: 2068 6173 2074 6865 2062 6573 7420 6869   has the best hi
-00009b60: 7420 6765 6e65 2061 3120 6672 6f6d 2067  t gene a1 from g
-00009b70: 656e 6f6d 6520 4120 7769 7468 2074 6865  enome A with the
-00009b80: 2062 6974 2073 636f 7265 2061 7320 3130   bit score as 10
-00009b90: 302c 2077 6869 6368 2069 7320 6120 7363  0, which is a sc
-00009ba0: 6f72 696e 6720 6d61 7472 6978 2069 6e64  oring matrix ind
-00009bb0: 6570 656e 6465 6e74 206d 6561 7375 7265  ependent measure
-00009bc0: 206f 6620 7468 6520 286c 6f63 616c 2920   of the (local) 
-00009bd0: 7369 6d69 6c61 7269 7479 206f 6620 7468  similarity of th
-00009be0: 6520 7477 6f20 616c 6967 6e65 6420 7365  e two aligned se
-00009bf0: 7175 656e 6365 732c 2077 6974 6820 6c61  quences, with la
-00009c00: 7267 6572 2076 616c 7565 7320 696e 6469  rger values indi
-00009c10: 6361 7469 6e67 2068 6967 6865 7220 7369  cating higher si
-00009c20: 6d69 6c61 7269 7469 6573 2c20 6769 7665  milarities, give
-00009c30: 6e20 7468 6520 602d 6320 302e 3960 2c20  n the `-c 0.9`, 
-00009c40: 6765 6e65 7320 6672 6f6d 2067 656e 6f6d  genes from genom
-00009c50: 6520 4120 7768 6963 6820 6861 7320 7468  e A which has th
-00009c60: 6520 6269 7420 7363 6f72 6520 7769 7468  e bit score with
-00009c70: 2067 656e 6520 6231 2068 6967 6865 7220   gene b1 higher 
-00009c80: 7468 616e 2030 2e39 7831 3030 2077 696c  than 0.9x100 wil
-00009c90: 6c20 616c 736f 2062 6520 7772 6974 7465  l also be writte
-00009ca0: 6e20 696e 2074 6865 2072 6573 756c 7420  n in the result 
-00009cb0: 6669 6c65 2c20 7768 6963 6820 696e 2061  file, which in a
-00009cc0: 2073 656e 7365 2061 7265 206e 6f74 2052   sense are not R
-00009cd0: 4248 7320 616e 796d 6f72 6520 6f66 2063  BHs anymore of c
-00009ce0: 6f75 7273 652c 2062 7574 2074 6865 2068  ourse, but the h
-00009cf0: 6967 686c 7920 7369 6d69 6c61 7220 686f  ighly similar ho
-00009d00: 6d6f 6c6f 6775 6520 7061 6972 732e 2049  mologue pairs. I
-00009d10: 6620 6d6f 7265 2074 6861 6e20 3220 7365  f more than 2 se
-00009d20: 7175 656e 6365 2066 696c 6573 2077 6572  quence files wer
-00009d30: 6520 7072 6f76 6964 6564 2c20 6576 6572  e provided, ever
-00009d40: 7920 7061 6972 2d77 6973 6520 5242 4873  y pair-wise RBHs
-00009d50: 2077 6f75 6c64 2062 6520 6361 6c63 756c   would be calcul
-00009d60: 6174 6564 2065 7863 6570 7420 666f 7220  ated except for 
-00009d70: 7175 6572 7969 6e67 2074 6865 2073 616d  querying the sam
-00009d80: 6520 7365 7175 656e 6365 2069 7473 656c  e sequence itsel
-00009d90: 662e 2054 6865 206e 756d 6265 7220 6f66  f. The number of
-00009da0: 2070 6172 616c 6c65 6c20 7468 7265 6164   parallel thread
-00009db0: 7320 746f 2062 6f6f 7374 6572 2074 6865  s to booster the
-00009dc0: 2072 756e 6e69 6e67 2073 7065 6564 2063   running speed c
-00009dd0: 616e 2062 6520 7365 7420 6279 2074 6865  an be set by the
-00009de0: 206f 7074 696f 6e20 602d 6e60 206f 7220   option `-n` or 
-00009df0: 602d 2d6e 7468 7265 6164 7360 2077 6869  `--nthreads` whi
-00009e00: 6368 2069 7320 7375 6767 6573 7465 6420  ch is suggested 
-00009e10: 746f 2062 6520 7365 7420 6173 2028 4e2d  to be set as (N-
-00009e20: 3129 4e2f 3220 7768 6572 6520 4e20 6973  1)N/2 where N is
-00009e30: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
-00009e40: 6473 2066 696c 6573 2074 6f20 6163 6869  ds files to achi
-00009e50: 6576 6520 7468 6520 6869 6768 6573 7420  eve the highest 
-00009e60: 6566 6669 6369 656e 6379 2e20 5468 6520  efficiency. The 
-00009e70: 6f70 7469 6f6e 2060 2d2d 6f67 666f 726d  option `--ogform
-00009e80: 6174 6020 6361 6e20 6265 2073 6574 2074  at` can be set t
-00009e90: 6f20 6164 6420 696e 6465 7820 2866 6f72  o add index (for
-00009ea0: 2069 6e73 7461 6e63 6520 4746 3030 3030   instance GF0000
-00009eb0: 3030 3031 2920 746f 2074 6865 206f 7574  0001) to the out
-00009ec0: 7075 7420 5242 4820 6765 6e65 2066 616d  put RBH gene fam
-00009ed0: 696c 6965 7320 7768 6963 6820 6361 6e20  ilies which can 
-00009ee0: 6265 2066 7572 7468 6572 2075 7365 6420  be further used 
-00009ef0: 696e 2074 6865 202a 4b2a 3c73 7562 3e53  in the *K*<sub>S
-00009f00: 3c2f 7375 623e 2063 616c 6375 6c61 7469  </sub> calculati
-00009f10: 6f6e 2062 7920 6077 6764 206b 7364 602e  on by `wgd ksd`.
-00009f20: 0a0a 2a2a 5468 6520 7375 6767 6573 7465  ..**The suggeste
-00009f30: 6420 636f 6d6d 616e 6420 746f 2073 7461  d command to sta
-00009f40: 7274 2077 6974 6820 6973 2061 6c73 6f20  rt with is also 
-00009f50: 756e 6465 7220 7468 6520 6465 6661 756c  under the defaul
-00009f60: 7420 7365 7474 696e 6720 7769 7468 2074  t setting with t
-00009f70: 6865 2063 6f6d 6d61 6e64 2073 686f 776e  he command shown
-00009f80: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
-00009f90: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-00009fa0: 7365 7175 656e 6365 320a 6060 600a 0a2a  sequence2.```..*
-00009fb0: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-00009fc0: 206f 6620 6c6f 6361 6c20 4d52 4248 732a   of local MRBHs*
-00009fd0: 2a0a 6060 600a 7767 6420 646d 6420 7365  *.```.wgd dmd se
-00009fe0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-00009ff0: 3220 7365 7175 656e 6365 3320 2d66 2073  2 sequence3 -f s
-0000a000: 6571 7565 6e63 6531 202d 6520 3165 2d31  equence1 -e 1e-1
-0000a010: 3020 2d62 7320 3130 3020 2d6e 7020 3520  0 -bs 100 -np 5 
-0000a020: 282d 6e6e 2920 282d 6e20 3429 2028 2d6b  (-nn) (-n 4) (-k
-0000a030: 6629 2028 2d6b 6429 2028 2d63 2030 2e39  f) (-kd) (-c 0.9
-0000a040: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
-0000a050: 2d63 6473 2920 282d 6f20 7767 645f 646d  -cds) (-o wgd_dm
-0000a060: 6429 2028 2d74 2077 6f72 6b69 6e67 5f74  d) (-t working_t
-0000a070: 6d70 290a 6060 600a 0a2a 2a54 6865 2064  mp).```..**The d
-0000a080: 6973 7469 6e63 7469 6f6e 2062 6574 7765  istinction betwe
-0000a090: 656e 206c 6f63 616c 2061 6e64 2067 6c6f  en local and glo
-0000a0a0: 6261 6c20 2868 6572 6575 6e64 6572 2920  bal (hereunder) 
-0000a0b0: 4d52 4248 7320 6973 2074 6861 7420 6c6f  MRBHs is that lo
-0000a0c0: 6361 6c20 4d52 4248 7320 6172 6520 7468  cal MRBHs are th
-0000a0d0: 6520 7265 7375 6c74 7320 6f66 206d 6572  e results of mer
-0000a0e0: 6765 6420 5242 4873 206f 6e20 6120 6a6f  ged RBHs on a jo
-0000a0f0: 696e 7420 666f 6361 6c20 7370 6563 6965  int focal specie
-0000a100: 732c 2066 6f72 2069 6e73 7461 6e63 6520  s, for instance 
-0000a110: 696e 2061 2074 6872 6565 2073 7065 6369  in a three speci
-0000a120: 6573 2073 7973 7465 6d20 2841 2c28 422c  es system (A,(B,
-0000a130: 4329 292c 2074 6865 206c 6f63 616c 204d  C)), the local M
-0000a140: 5242 4873 206f 6620 4120 6f6e 6c79 2072  RBHs of A only r
-0000a150: 6571 7569 7265 2074 6865 2063 616c 6375  equire the calcu
-0000a160: 6c61 7469 6f6e 206f 6620 5242 4873 2062  lation of RBHs b
-0000a170: 6574 7765 656e 2041 2061 6e64 2043 2028  etween A and C (
-0000a180: 6465 6e6f 7465 6420 6173 2041 4329 2061  denoted as AC) a
-0000a190: 6e64 2041 4220 616e 6420 7468 656e 2074  nd AB and then t
-0000a1a0: 6865 206d 6572 6769 6e67 206f 6620 4142  he merging of AB
-0000a1b0: 2061 6e64 2041 4320 6174 2074 6865 2061   and AC at the a
-0000a1c0: 7869 7320 6f66 2041 2c20 7768 696c 6520  xis of A, while 
-0000a1d0: 7468 6520 676c 6f61 626c 204d 5242 4873  the gloabl MRBHs
-0000a1e0: 2061 7265 2069 6e64 6570 656e 6465 6e74   are independent
-0000a1f0: 206f 6620 666f 6361 6c20 7370 6563 6965   of focal specie
-0000a200: 7320 696e 2074 6861 7420 6974 206a 7573  s in that it jus
-0000a210: 7420 6361 6c6c 7320 7468 6520 6361 6c63  t calls the calc
-0000a220: 756c 6174 696f 6e73 206f 6620 616c 6c20  ulations of all 
-0000a230: 706f 7373 6962 6c65 2073 7065 6369 6573  possible species
-0000a240: 2070 6169 7220 286e 6f74 2073 656c 6620   pair (not self 
-0000a250: 746f 2073 656c 6620 7370 6563 6965 7320  to self species 
-0000a260: 7061 6972 292c 2073 7563 6820 7468 6174  pair), such that
-0000a270: 2041 422c 2041 432c 2061 6e64 2042 4320   AB, AC, and BC 
-0000a280: 6172 6520 616c 6c20 746f 2062 6520 6361  are all to be ca
-0000a290: 6c63 756c 6174 6564 2061 6e64 206d 6572  lculated and mer
-0000a2a0: 6765 642e 2a2a 0a0a 5477 6f20 7479 7065  ged.**..Two type
-0000a2b0: 7320 6f66 204d 5242 4873 2061 7320 696e  s of MRBHs as in
-0000a2c0: 7465 7072 6574 6174 6564 2061 626f 7665  tepretated above
-0000a2d0: 2063 616e 2062 6520 6465 6c69 6e65 6174   can be delineat
-0000a2e0: 6564 2062 7920 6077 6764 2064 6d64 602c  ed by `wgd dmd`,
-0000a2f0: 2074 6865 206c 6f63 616c 204d 5242 4873   the local MRBHs
-0000a300: 2061 6e64 2074 6865 2067 6c6f 6261 6c20   and the global 
-0000a310: 4d52 4248 732e 2054 6865 206c 6f63 616c  MRBHs. The local
-0000a320: 204d 5242 4873 2061 7265 2063 6f6e 7374   MRBHs are const
-0000a330: 7275 6374 6564 2062 7920 6d65 7267 696e  ructed by mergin
-0000a340: 6720 616c 6c20 7468 6520 7265 6c65 7661  g all the releva
-0000a350: 6e74 2052 4248 7320 6f6e 6c79 2077 6974  nt RBHs only wit
-0000a360: 6820 7468 6520 666f 6361 6c20 7370 6563  h the focal spec
-0000a370: 6965 732c 2077 6869 6368 2069 7320 7365  ies, which is se
-0000a380: 7420 6279 2060 2d66 6020 6f72 2060 2d2d  t by `-f` or `--
-0000a390: 666f 6375 7360 2e20 5468 6520 7061 7261  focus`. The para
-0000a3a0: 6d65 7465 7220 602d 6b66 6020 6f72 2060  meter `-kf` or `
-0000a3b0: 2d2d 6b65 6570 6661 7374 6160 2063 616e  --keepfasta` can
-0000a3c0: 2062 6520 7365 7420 746f 2072 6574 6169   be set to retai
-0000a3d0: 6e20 7468 6520 7365 7175 656e 6365 2069  n the sequence i
-0000a3e0: 6e66 6f72 6d61 7469 6f6e 206f 6620 6561  nformation of ea
-0000a3f0: 6368 204d 5242 482e 2054 6865 2070 6172  ch MRBH. The par
-0000a400: 616d 6574 6572 2060 2d6b 6460 206f 7220  ameter `-kd` or 
-0000a410: 602d 2d6b 6565 7064 7570 6c69 6361 7465  `--keepduplicate
-0000a420: 7360 2064 6574 6572 6d69 6e65 7320 7768  s` determines wh
-0000a430: 6574 6865 7220 7468 6520 7361 6d65 2067  ether the same g
-0000a440: 656e 6573 2063 616e 2061 7070 6561 7220  enes can appear 
-0000a450: 696e 2064 6966 6665 7265 6e74 206c 6f63  in different loc
-0000a460: 616c 204d 5242 4873 2e20 4e6f 726d 616c  al MRBHs. Normal
-0000a470: 6c79 2074 6865 7265 2077 696c 6c20 6265  ly there will be
-0000a480: 206e 6f20 6475 706c 6963 6174 6573 2069   no duplicates i
-0000a490: 6e20 7468 6520 6c6f 6361 6c20 4d52 4248  n the local MRBH
-0000a4a0: 7320 6275 7420 6966 2075 7365 7273 2073  s but if users s
-0000a4b0: 6574 2074 6865 2060 2d63 6020 6173 2030  et the `-c` as 0
-0000a4c0: 2e39 2028 6f72 2061 6e79 2076 616c 7565  .9 (or any value
-0000a4d0: 2073 6d61 6c6c 6572 2074 6861 6e20 3129   smaller than 1)
-0000a4e0: 2c20 6974 2773 206c 696b 656c 7920 7468  , it's likely th
-0000a4f0: 6174 2074 6865 2073 616d 6520 6765 6e65  at the same gene
-0000a500: 2077 696c 6c20 6861 7665 2063 6861 6e63   will have chanc
-0000a510: 6520 746f 2061 7070 6561 7220 6d75 6c74  e to appear mult
-0000a520: 6970 6c65 2074 696d 6573 2069 6e20 6469  iple times in di
-0000a530: 6666 6572 656e 7420 6c6f 6361 6c20 4d52  fferent local MR
-0000a540: 4248 732e 2054 6861 7420 6973 2074 6f20  BHs. That is to 
-0000a550: 7361 792c 2074 6865 2070 6172 616d 6574  say, the paramet
-0000a560: 6572 2060 2d6b 6460 2069 7320 6d65 616e  er `-kd` is mean
-0000a570: 696e 6766 756c 206f 6e6c 7920 7768 656e  ingful only when
-0000a580: 2069 7427 7320 7365 7420 746f 6765 7468   it's set togeth
-0000a590: 6572 2077 6974 6820 7468 6520 7061 7261  er with the para
-0000a5a0: 6d65 7465 7220 602d 6360 2e20 5468 6520  meter `-c`. The 
-0000a5b0: 6e75 6d62 6572 206f 6620 7061 7261 6c6c  number of parall
-0000a5c0: 656c 2074 6872 6561 6473 2069 7320 7375  el threads is su
-0000a5d0: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000a5e0: 7420 6173 2074 6865 206e 756d 6265 7220  t as the number 
-0000a5f0: 6f66 2063 6473 2066 696c 6573 206d 696e  of cds files min
-0000a600: 7573 2031 2e0a 0a2a 2a41 2073 7567 6765  us 1...**A sugge
-0000a610: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
-0000a620: 6e20 6973 2075 6e64 6572 2074 6865 2064  n is under the d
-0000a630: 6566 6175 6c74 2070 6172 616d 6574 6572  efault parameter
-0000a640: 2077 6974 6820 7468 6520 636f 6d6d 616e   with the comman
-0000a650: 6420 7368 6f77 6e20 6173 2062 656c 6f77  d shown as below
-0000a660: 2e2a 2a0a 0a60 6060 0a77 6764 2064 6d64  .**..```.wgd dmd
-0000a670: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000a680: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000a690: 6620 7365 7175 656e 6365 310a 6060 600a  f sequence1.```.
-0000a6a0: 0a2a 2a54 6865 2064 656c 696e 6561 7469  .**The delineati
-0000a6b0: 6f6e 206f 6620 676c 6f62 616c 204d 5242  on of global MRB
-0000a6c0: 4873 2a2a 0a60 6060 0a77 6764 2064 6d64  Hs**.```.wgd dmd
-0000a6d0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000a6e0: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000a6f0: 676d 202d 6520 3165 2d31 3020 2d62 7320  gm -e 1e-10 -bs 
-0000a700: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
-0000a710: 282d 6e20 3429 2028 2d6b 6629 2028 2d6b  (-n 4) (-kf) (-k
-0000a720: 6429 2028 2d63 2030 2e39 2920 282d 2d74  d) (-c 0.9) (--t
-0000a730: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000a740: 282d 6f20 7767 645f 646d 6429 2028 2d74  (-o wgd_dmd) (-t
-0000a750: 2077 6f72 6b69 6e67 5f74 6d70 290a 6060   working_tmp).``
-0000a760: 600a 0a54 6865 2067 6c6f 6261 6c20 4d52  `..The global MR
-0000a770: 4248 7320 6973 2063 6f6e 7374 7275 6374  BHs is construct
-0000a780: 6564 2062 7920 6578 6861 7573 7469 7665  ed by exhaustive
-0000a790: 6c79 206d 6572 6769 6e67 2061 6c6c 2074  ly merging all t
-0000a7a0: 6865 2070 6f73 7369 626c 6520 7061 6972  he possible pair
-0000a7b0: 2d77 6973 6520 5242 4873 2065 7863 6570  -wise RBHs excep
-0000a7c0: 7420 666f 7220 7175 6572 7969 6e67 2074  t for querying t
-0000a7d0: 6865 2073 6571 7565 6e63 6520 6974 7365  he sequence itse
-0000a7e0: 6c66 2c20 7768 6963 6820 6361 6e20 6265  lf, which can be
-0000a7f0: 2069 6e69 7469 6174 6564 2062 7920 6164   initiated by ad
-0000a800: 6420 7468 6520 666c 6167 2060 2d67 6d60  d the flag `-gm`
-0000a810: 206f 7220 602d 2d67 6c6f 6261 6c6d 7262   or `--globalmrb
-0000a820: 6860 2e20 5468 6520 7265 7374 206f 6620  h`. The rest of 
-0000a830: 7265 6c65 7661 6e74 2070 6172 616d 6574  relevant paramet
-0000a840: 6572 7320 7374 6179 7320 7468 6520 7361  ers stays the sa
-0000a850: 6d65 2061 7320 7468 6520 6c6f 6361 6c20  me as the local 
-0000a860: 4d52 4248 732e 2054 6865 206e 756d 6265  MRBHs. The numbe
-0000a870: 7220 6f66 2070 6172 616c 6c65 6c20 7468  r of parallel th
-0000a880: 7265 6164 7320 6973 2073 7567 6765 7374  reads is suggest
-0000a890: 6564 2074 6f20 6265 2073 6574 2061 7320  ed to be set as 
-0000a8a0: 284e 2d31 294e 2f32 2074 6f6f 2077 6865  (N-1)N/2 too whe
-0000a8b0: 7265 204e 2069 7320 7468 6520 6e75 6d62  re N is the numb
-0000a8c0: 6572 206f 6620 6364 7320 6669 6c65 7320  er of cds files 
-0000a8d0: 746f 2061 6368 6965 7665 2074 6865 2068  to achieve the h
-0000a8e0: 6967 6865 7374 2065 6666 6963 6965 6e63  ighest efficienc
-0000a8f0: 792e 0a0a 2a2a 4120 7375 6767 6573 7465  y...**A suggeste
-0000a900: 6420 7374 6172 7469 6e67 2072 756e 2069  d starting run i
-0000a910: 7320 756e 6465 7220 7468 6520 6465 6661  s under the defa
-0000a920: 756c 7420 7061 7261 6d65 7465 7220 7769  ult parameter wi
-0000a930: 7468 2074 6865 2063 6f6d 6d61 6e64 2073  th the command s
-0000a940: 686f 776e 2061 7320 6265 6c6f 772e 2a2a  hown as below.**
-0000a950: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
-0000a960: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000a970: 3220 7365 7175 656e 6365 3320 2d67 6d0a  2 sequence3 -gm.
-0000a980: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
-0000a990: 6561 7469 6f6e 206f 6620 6f72 7468 6f67  eation of orthog
-0000a9a0: 726f 7570 732a 2a0a 6060 600a 7767 6420  roups**.```.wgd 
-0000a9b0: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
-0000a9c0: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000a9d0: 3320 2d6f 6920 2d6f 6f20 2d65 2031 652d  3 -oi -oo -e 1e-
-0000a9e0: 3130 202d 6273 2031 3030 202d 6e70 2035  10 -bs 100 -np 5
-0000a9f0: 2028 2d6e 6e29 2028 2d63 6329 2028 2d74   (-nn) (-cc) (-t
-0000aa00: 6529 2028 2d6d 6320 302e 3829 2028 2d67  e) (-mc 0.8) (-g
-0000aa10: 6e29 2028 2d74 7265 6520 2766 6173 7474  n) (-tree 'fastt
-0000aa20: 7265 6527 2920 282d 7473 2027 2d66 6173  ree') (-ts '-fas
-0000aa30: 7465 7374 2729 2028 2d6e 2034 2920 282d  test') (-n 4) (-
-0000aa40: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
-0000aa50: 2920 282d 6f20 7767 645f 646d 6429 2028  ) (-o wgd_dmd) (
-0000aa60: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
-0000aa70: 6060 600a 0a49 6e20 6077 6764 2076 3260  ```..In `wgd v2`
-0000aa80: 2c20 7765 2061 6c73 6f20 696d 706c 656d  , we also implem
-0000aa90: 656e 7465 6420 616e 2061 6c67 6f72 6974  ented an algorit
-0000aaa0: 686d 206f 6620 6465 6c69 6e65 6174 696e  hm of delineatin
-0000aab0: 6720 6f72 7468 6f67 726f 7570 732c 2077  g orthogroups, w
-0000aac0: 6869 6368 2063 616e 2062 6520 696e 6974  hich can be init
-0000aad0: 6961 7465 6420 7769 7468 2074 6865 2070  iated with the p
-0000aae0: 6172 616d 6574 6572 2060 2d6f 6960 206f  arameter `-oi` o
-0000aaf0: 7220 602d 2d6f 7274 686f 696e 6665 7260  r `--orthoinfer`
-0000ab00: 2e20 5477 6f20 7761 7973 206f 6620 6465  . Two ways of de
-0000ab10: 6c69 6e65 6174 696f 6e20 6361 6e20 6265  lineation can be
-0000ab20: 2063 686f 7365 6e2c 2074 6865 2063 6f6e   chosen, the con
-0000ab30: 6361 7465 6e61 7469 6f6e 2077 6179 2028  catenation way (
-0000ab40: 7365 7420 6279 2074 6865 2070 6172 616d  set by the param
-0000ab50: 6574 6572 2060 2d63 6360 206f 7220 602d  eter `-cc` or `-
-0000ab60: 2d63 6f6e 6361 7460 2920 6f72 2074 6865  -concat`) or the
-0000ab70: 206e 6f6e 2d63 6f6e 6361 7465 6e61 7469   non-concatenati
-0000ab80: 6f6e 2028 6465 6661 756c 7429 2077 6179  on (default) way
-0000ab90: 2e20 496e 2062 7269 6566 2c20 7468 6520  . In brief, the 
-0000aba0: 636f 6e63 6174 656e 6174 696f 6e20 7761  concatenation wa
-0000abb0: 7920 6f66 2064 656c 696e 6561 7469 6e67  y of delineating
-0000abc0: 206f 7274 686f 6772 6f75 7073 2073 7461   orthogroups sta
-0000abd0: 7274 7320 7769 7468 2063 6f6e 6361 7465  rts with concate
-0000abe0: 6e61 7469 6e67 2061 6c6c 2074 6865 2073  nating all the s
-0000abf0: 6571 7565 6e63 6573 2069 6e74 6f20 6120  equences into a 
-0000ac00: 7369 6e67 6c65 2073 6571 7565 6e63 6520  single sequence 
-0000ac10: 6669 6c65 2061 6e64 2074 6865 6e20 696e  file and then in
-0000ac20: 6665 7272 696e 6720 7468 6520 7768 6f6c  ferring the whol
-0000ac30: 6520 7061 7261 6e6f 6d65 206f 6620 7468  e paranome of th
-0000ac40: 6973 2073 696e 676c 6520 7365 7175 656e  is single sequen
-0000ac50: 6365 2066 696c 6520 7769 7468 2074 6865  ce file with the
-0000ac60: 2063 6c75 7374 6572 696e 6720 7265 7375   clustering resu
-0000ac70: 6c74 7320 6d61 7070 6564 2062 6163 6b20  lts mapped back 
-0000ac80: 746f 2074 6865 2062 656c 6f6e 6769 6e67  to the belonging
-0000ac90: 2073 7065 6369 6573 2e20 5768 696c 6520   species. While 
-0000aca0: 7468 6520 6e6f 6e2d 636f 6e63 6174 656e  the non-concaten
-0000acb0: 6174 696f 6e20 7761 7920 7374 6172 7473  ation way starts
-0000acc0: 2077 6974 6820 7265 7370 6563 7469 7665   with respective
-0000acd0: 2070 6169 722d 7769 7365 2064 6961 6d6f   pair-wise diamo
-0000ace0: 6e64 2073 6561 7263 6820 2869 6e63 6c75  nd search (inclu
-0000acf0: 6469 6e67 2071 7565 7279 696e 6720 7468  ding querying th
-0000ad00: 6520 7361 6d65 2073 6571 7565 6e63 6520  e same sequence 
-0000ad10: 6974 7365 6c66 2920 616e 6420 7468 656e  itself) and then
-0000ad20: 2061 6c6c 2074 6865 2073 6571 7565 6e63   all the sequenc
-0000ad30: 6520 7369 6d69 6c61 7269 7479 2074 6162  e similarity tab
-0000ad40: 6c65 7320 7769 6c6c 2062 6520 636f 6e63  les will be conc
-0000ad50: 6174 656e 6174 6564 2061 6e64 2063 6c75  atenated and clu
-0000ad60: 7374 6572 6564 2069 6e74 6f20 6f72 7468  stered into orth
-0000ad70: 6f67 726f 7570 732e 2053 6f6d 6520 6f74  ogroups. Some ot
-0000ad80: 6865 7220 706f 7373 6962 6c79 2075 7365  her possibly use
-0000ad90: 6675 6c20 706f 7374 2d63 6c75 7374 6572  ful post-cluster
-0000ada0: 696e 6720 6675 6e63 7469 6f6e 7320 6361  ing functions ca
-0000adb0: 6e20 6265 2069 6e69 7469 6174 6564 2c20  n be initiated, 
-0000adc0: 696e 636c 7564 696e 6720 7468 6520 7061  including the pa
-0000add0: 7261 6d65 7465 7220 602d 7465 6020 6f72  rameter `-te` or
-0000ade0: 2060 2d2d 7465 7374 736f 6760 2c20 7768   `--testsog`, wh
-0000adf0: 6963 6820 6361 6e20 6265 2073 6574 2074  ich can be set t
-0000ae00: 6f20 7374 6172 7420 7468 6520 756e 6269  o start the unbi
-0000ae10: 6173 6564 2074 6573 7420 6f66 2073 696e  ased test of sin
-0000ae20: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-0000ae30: 6d69 6c69 6573 2028 6e6f 7465 2074 6861  milies (note tha
-0000ae40: 7420 7468 6973 2066 756e 6374 696f 6e20  t this function 
-0000ae50: 6e65 6564 7320 6068 6d6d 6572 6020 2876  needs `hmmer` (v
-0000ae60: 332e 3162 3229 2074 6f20 6265 2069 6e73  3.1b2) to be ins
-0000ae70: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
-0000ae80: 7669 726f 6e6d 656e 7420 7061 7468 292c  vironment path),
-0000ae90: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
-0000aea0: 2d6d 6360 206f 7220 602d 2d6d 736f 6763  -mc` or `--msogc
-0000aeb0: 7574 602c 2072 616e 6769 6e67 2062 6574  ut`, ranging bet
-0000aec0: 7765 656e 2030 2074 6f20 312c 2077 6869  ween 0 to 1, whi
-0000aed0: 6368 2063 616e 2062 6520 7365 7420 746f  ch can be set to
-0000aee0: 2073 6561 7263 6820 7468 6520 736f 2d63   search the so-c
-0000aef0: 616c 6c65 6420 6d6f 7374 6c79 2073 696e  alled mostly sin
-0000af00: 676c 652d 636f 7079 2066 616d 696c 7920  gle-copy family 
-0000af10: 7768 6963 6820 6861 7320 6869 6768 6572  which has higher
-0000af20: 2074 6861 6e20 6365 7274 6169 6e20 6375   than certain cu
-0000af30: 742d 6f66 6620 7065 7263 656e 7461 6765  t-off percentage
-0000af40: 206f 6620 7370 6563 6965 7320 636f 7665   of species cove
-0000af50: 7261 6765 2c20 7468 6520 7061 7261 6d65  rage, the parame
-0000af60: 7465 7220 602d 676e 6020 6f72 2060 2d2d  ter `-gn` or `--
-0000af70: 6765 746e 736f 6760 2c20 7768 6963 6820  getnsog`, which 
-0000af80: 6361 6e20 6265 2073 6574 2074 6f20 7365  can be set to se
-0000af90: 6172 6368 2066 6f72 206e 6573 7465 6420  arch for nested 
-0000afa0: 7369 6e67 6c65 2d63 6f70 7920 6765 6e65  single-copy gene
-0000afb0: 2066 616d 696c 6965 7320 284e 534f 4773   families (NSOGs
-0000afc0: 2920 7768 6963 6820 6973 206f 7269 6769  ) which is origi
-0000afd0: 6e61 6c6c 7920 6d75 6c74 6979 2d63 6f70  nally multiy-cop
-0000afe0: 7920 6275 7420 6861 7320 6120 286d 6f73  y but has a (mos
-0000aff0: 746c 7929 2073 696e 676c 652d 636f 7079  tly) single-copy
-0000b000: 2062 7261 6e63 6820 2877 6869 6368 2072   branch (which r
-0000b010: 6571 7569 7265 7320 7468 6520 6368 6f73  equires the chos
-0000b020: 656e 2074 7265 652d 696e 6665 7265 6e63  en tree-inferenc
-0000b030: 6520 7072 6f67 7261 6d20 7365 7420 6279  e program set by
-0000b040: 2060 2d74 7265 6560 206f 7220 602d 2d74   `-tree` or `--t
-0000b050: 7265 655f 6d65 7468 6f64 6020 746f 2062  ree_method` to b
-0000b060: 6520 7072 652d 696e 7374 616c 6c65 6420  e pre-installed 
-0000b070: 696e 2074 6865 2065 6e76 6972 6f6e 6d65  in the environme
-0000b080: 6e74 2070 6174 6820 7769 7468 2074 6865  nt path with the
-0000b090: 2070 6172 616d 6574 6572 7320 7365 7474   parameters sett
-0000b0a0: 696e 6720 666f 7220 6765 6e65 2074 7265  ing for gene tre
-0000b0b0: 6520 696e 6665 7265 6e63 6520 636f 6e74  e inference cont
-0000b0c0: 726f 6c6c 6564 2062 7920 602d 7473 6020  rolled by `-ts` 
-0000b0d0: 6f72 2060 2d2d 7472 6565 7365 7460 292e  or `--treeset`).
-0000b0e0: 2054 6865 2070 726f 6772 616d 2060 7767   The program `wg
-0000b0f0: 6420 646d 6460 2077 6f75 6c64 2073 7469  d dmd` would sti
-0000b100: 6c6c 2063 6f6e 6475 6374 2074 6865 2052  ll conduct the R
-0000b110: 4248 7320 6361 6c63 756c 6174 696f 6e20  BHs calculation 
-0000b120: 756e 6c65 7373 2074 6865 2070 6172 616d  unless the param
-0000b130: 6574 6572 2060 2d6f 6f60 206f 7220 602d  eter `-oo` or `-
-0000b140: 2d6f 6e6c 796f 7274 686f 6020 7761 7320  -onlyortho` was 
-0000b150: 7365 742e 2049 6620 6f6e 6520 6f6e 6c79  set. If one only
-0000b160: 2077 616e 7473 2074 6f20 696e 6665 7220   wants to infer 
-0000b170: 7468 6520 6f72 7468 6f67 726f 7570 732c  the orthogroups,
-0000b180: 2069 7427 7320 7375 6767 6573 7465 6420   it's suggested 
-0000b190: 746f 2061 6464 2074 6865 2066 6c61 6720  to add the flag 
-0000b1a0: 602d 6f6f 6020 746f 206a 7573 7420 696d  `-oo` to just im
-0000b1b0: 706c 656d 656e 7420 7468 6520 6f72 7468  plement the orth
-0000b1c0: 6f67 726f 7570 7320 6465 6c69 6e65 6174  ogroups delineat
-0000b1d0: 696f 6e20 616e 616c 7973 6973 2e20 5468  ion analysis. Th
-0000b1e0: 6520 6e75 6d62 6572 206f 6620 7061 7261  e number of para
-0000b1f0: 6c6c 656c 2074 6872 6561 6473 2069 7320  llel threads is 
-0000b200: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
-0000b210: 7365 7420 6173 2028 4e2b 3129 4e2f 3220  set as (N+1)N/2 
-0000b220: 7768 6572 6520 4e20 6973 2074 6865 206e  where N is the n
-0000b230: 756d 6265 7220 6f66 2063 6473 2066 696c  umber of cds fil
-0000b240: 6573 2074 6f20 6163 6869 6576 6520 7468  es to achieve th
-0000b250: 6520 6869 6768 6573 7420 6566 6669 6369  e highest effici
-0000b260: 656e 6379 2073 696e 6365 2074 6865 2073  ency since the s
-0000b270: 656c 662d 636f 6d70 6172 6973 6f6e 2069  elf-comparison i
-0000b280: 7320 616c 736f 2069 6e63 6c75 6465 642e  s also included.
-0000b290: 0a0a 2a2a 5468 6520 6465 6661 756c 7420  ..**The default 
-0000b2a0: 7365 7474 696e 6720 6f66 2070 6172 616d  setting of param
-0000b2b0: 6574 6572 7320 6973 2061 2072 6561 736f  eters is a reaso
-0000b2c0: 6e61 626c 6520 7374 6172 7469 6e67 2070  nable starting p
-0000b2d0: 6f69 6e74 2077 6974 6820 7468 6520 636f  oint with the co
-0000b2e0: 6d6d 616e 6420 6173 2062 656c 6f77 2e2a  mmand as below.*
-0000b2f0: 2a0a 0a60 6060 0a77 6764 2064 6d64 2073  *..```.wgd dmd s
-0000b300: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000b310: 6532 2073 6571 7565 6e63 6533 202d 6f69  e2 sequence3 -oi
-0000b320: 202d 6f6f 0a60 6060 0a0a 2a2a 5468 6520   -oo.```..**The 
-0000b330: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
-0000b340: 6365 6e63 6520 696e 6665 7265 6e63 6520  cence inference 
-0000b350: 6f66 2070 6879 6c6f 6765 6e79 2a2a 0a60  of phylogeny**.`
-0000b360: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-0000b370: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000b380: 6571 7565 6e63 6533 202d 6170 2061 7064  equence3 -ap apd
-0000b390: 6174 6120 2d73 6d20 736d 6461 7461 202d  ata -sm smdata -
-0000b3a0: 6c65 206c 6564 6174 6120 2d67 7420 6774  le ledata -gt gt
-0000b3b0: 6461 7461 202d 636f 6320 282d 7472 6565  data -coc (-tree
-0000b3c0: 2027 6661 7374 7472 6565 2729 2028 2d74   'fasttree') (-t
-0000b3d0: 7320 272d 6661 7374 6573 7427 2920 282d  s '-fastest') (-
-0000b3e0: 6e20 3429 2028 2d2d 746f 5f73 746f 7029  n 4) (--to_stop)
-0000b3f0: 2028 2d2d 6364 7329 2028 2d6f 2077 6764   (--cds) (-o wgd
-0000b400: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
-0000b410: 675f 746d 7029 0a60 6060 0a0a 4120 6e6f  g_tmp).```..A no
-0000b420: 7665 6c20 7068 796c 6f67 656e 6574 6963  vel phylogenetic
-0000b430: 2069 6e66 6572 656e 6365 206d 6574 686f   inference metho
-0000b440: 6420 6e61 6d65 6420 2263 6f6c 6c69 6e65  d named "colline
-0000b450: 6172 2063 6f61 6c65 7363 656e 6365 2069  ar coalescence i
-0000b460: 6e66 6572 656e 6365 2220 6973 2061 6c73  nference" is als
-0000b470: 6f20 696d 706c 656d 656e 7465 6420 696e  o implemented in
-0000b480: 2060 7767 6420 7632 602e 2046 6f72 2074   `wgd v2`. For t
-0000b490: 6869 7320 616e 616c 7973 6973 2c20 7573  his analysis, us
-0000b4a0: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-0000b4b0: 6964 6520 7468 6520 616e 6368 6f72 2070  ide the anchor p
-0000b4c0: 6f69 6e74 7320 6669 6c65 2062 7920 602d  oints file by `-
-0000b4d0: 6170 6020 6f72 2060 2d2d 616e 6368 6f72  ap` or `--anchor
-0000b4e0: 706f 696e 7473 602c 2074 6865 2063 6f6c  points`, the col
-0000b4f0: 6c69 6e65 6172 2073 6567 6d65 6e74 7320  linear segments 
-0000b500: 6669 6c65 2062 7920 602d 736d 6020 6f72  file by `-sm` or
-0000b510: 2060 2d2d 7365 676d 656e 7473 602c 2074   `--segments`, t
-0000b520: 6865 206c 6973 7473 6567 6d65 6e74 7320  he listsegments 
-0000b530: 6669 6c65 2062 7920 602d 6c65 6020 6f72  file by `-le` or
-0000b540: 2060 2d2d 6c69 7374 656c 656d 656e 7473   `--listelements
-0000b550: 602c 2061 6e64 2074 6865 2067 656e 6520  `, and the gene 
-0000b560: 7461 626c 6520 6669 6c65 2062 7920 602d  table file by `-
-0000b570: 6774 6020 6f72 2060 2d2d 6765 6e65 7461  gt` or `--geneta
-0000b580: 626c 6560 2c20 616c 6c20 6f66 2077 6869  ble`, all of whi
-0000b590: 6368 2063 616e 2062 6520 7072 6f64 7563  ch can be produc
-0000b5a0: 6564 2069 6e20 7468 6520 7072 6f67 7261  ed in the progra
-0000b5b0: 6d20 6077 6764 2073 796e 602e 2054 6865  m `wgd syn`. The
-0000b5c0: 2070 6172 616d 6574 6572 2060 2d63 6f63   parameter `-coc
-0000b5d0: 6020 6f72 2060 2d2d 636f 6c6c 696e 6561  ` or `--collinea
-0000b5e0: 7263 6f61 6c65 7363 656e 6365 6020 6e65  rcoalescence` ne
-0000b5f0: 6564 7320 746f 2062 6520 7365 7420 746f  eds to be set to
-0000b600: 2073 7461 7274 2074 6869 7320 616e 616c   start this anal
-0000b610: 7973 6973 2e20 5468 6520 7472 6565 2d69  ysis. The tree-i
-0000b620: 6e66 6572 656e 6365 2070 726f 6772 616d  nference program
-0000b630: 2061 6e64 2074 6865 2061 7373 6f63 6961   and the associa
-0000b640: 7465 6420 7061 7261 6d65 7465 7273 2063  ted parameters c
-0000b650: 616e 2062 6520 7365 7420 6a75 7374 2061  an be set just a
-0000b660: 7320 6162 6f76 6520 6279 2060 2d74 7265  s above by `-tre
-0000b670: 6560 206f 7220 602d 2d74 7265 655f 6d65  e` or `--tree_me
-0000b680: 7468 6f64 6020 616e 6420 602d 7473 6020  thod` and `-ts` 
-0000b690: 6f72 2060 2d2d 7472 6565 7365 7460 2e20  or `--treeset`. 
-0000b6a0: 506c 6561 7365 2061 6c73 6f20 6d61 6b65  Please also make
-0000b6b0: 2073 7572 6520 7468 6520 6368 6f73 656e   sure the chosen
-0000b6c0: 2074 7265 652d 696e 6665 7265 6e63 6520   tree-inference 
-0000b6d0: 7072 6f67 7261 6d20 6973 2069 6e73 7461  program is insta
-0000b6e0: 6c6c 6564 2069 6e20 7468 6520 656e 7669  lled in the envi
-0000b6f0: 726f 6e6d 656e 7420 7061 7468 2e20 5468  ronment path. Th
-0000b700: 6520 7072 6f67 7261 6d20 6061 7374 7261  e program `astra
-0000b710: 6c2d 7072 6f60 2069 7320 7265 7175 6972  l-pro` is requir
-0000b720: 6564 2074 6f20 6265 2069 6e73 7461 6c6c  ed to be install
-0000b730: 6564 2069 6e20 7468 6520 656e 7669 726f  ed in the enviro
-0000b740: 6e6d 656e 7420 7061 7468 2074 6f6f 2e20  nment path too. 
-0000b750: 4e6f 7465 2074 6861 7420 7468 6572 6520  Note that there 
-0000b760: 7368 6f75 6c64 2062 6520 6e6f 2064 7570  should be no dup
-0000b770: 6c69 6361 7465 6420 6765 6e65 2049 4473  licated gene IDs
-0000b780: 2069 6e20 7468 6520 7365 7175 656e 6365   in the sequence
-0000b790: 2066 696c 652e 2054 6865 2070 6172 616c   file. The paral
-0000b7a0: 6c65 6c20 7468 7265 6164 7320 6865 7265  lel threads here
-0000b7b0: 2061 7265 2074 6f20 6163 6365 6c65 7261   are to accelera
-0000b7c0: 7465 2074 6865 2073 6571 7565 6e63 6520  te the sequence 
-0000b7d0: 616c 6967 6e6d 656e 7420 616e 6420 6765  alignment and ge
-0000b7e0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
-0000b7f0: 6520 666f 7220 6561 6368 2067 656e 6520  e for each gene 
-0000b800: 6661 6d69 6c79 2061 6e64 2074 6875 7320  family and thus 
-0000b810: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
-0000b820: 7365 7420 6173 206d 7563 6820 6173 2074  set as much as t
-0000b830: 6865 206e 756d 6265 7220 6f66 2067 656e  he number of gen
-0000b840: 6520 6661 6d69 6c69 6573 2e0a 0a2a 2a41  e families...**A
-0000b850: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
-0000b860: 696e 6720 7275 6e20 6f66 2074 6869 7320  ing run of this 
-0000b870: 616e 616c 7973 6973 2069 7320 7769 7468  analysis is with
-0000b880: 2074 6865 2073 696d 706c 6520 636f 6d6d   the simple comm
-0000b890: 616e 6420 6265 6c6f 772e 2a2a 0a0a 6060  and below.**..``
-0000b8a0: 600a 7767 6420 646d 6420 7365 7175 656e  `.wgd dmd sequen
-0000b8b0: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000b8c0: 7175 656e 6365 3320 2d61 7020 6170 6461  quence3 -ap apda
-0000b8d0: 7461 202d 736d 2073 6d64 6174 6120 2d6c  ta -sm smdata -l
-0000b8e0: 6520 6c65 6461 7461 202d 6774 2067 7464  e ledata -gt gtd
-0000b8f0: 6174 6120 2d63 6f63 0a60 6060 0a0a 2323  ata -coc.```..##
-0000b900: 2320 7767 6420 666f 6375 730a 0a2a 2a54  # wgd focus..**T
-0000b910: 6865 2063 6f6e 6361 7465 6e61 7469 6f6e  he concatenation
-0000b920: 2d62 6173 6564 2f63 6f61 6c65 7363 656e  -based/coalescen
-0000b930: 6365 2d62 6173 6564 2070 6879 6c6f 6765  ce-based phyloge
-0000b940: 6e65 7469 6320 696e 6665 7265 6e63 652a  netic inference*
-0000b950: 2a0a 6060 600a 7767 6420 666f 6375 7320  *.```.wgd focus 
-0000b960: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-0000b970: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000b980: 7565 6e63 6533 2028 2d2d 636f 6e63 6174  uence3 (--concat
-0000b990: 656e 6174 696f 6e29 2028 2d2d 636f 616c  enation) (--coal
-0000b9a0: 6573 6365 6e63 6529 2028 2d74 7265 6520  escence) (-tree 
-0000b9b0: 2766 6173 7474 7265 6527 2920 282d 7473  'fasttree') (-ts
-0000b9c0: 2027 2d66 6173 7465 7374 2729 2028 2d6e   '-fastest') (-n
-0000b9d0: 2034 2920 282d 2d74 6f5f 7374 6f70 2920   4) (--to_stop) 
-0000b9e0: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
-0000b9f0: 666f 6375 7329 2028 2d74 2077 6f72 6b69  focus) (-t worki
-0000ba00: 6e67 5f74 6d70 290a 6060 600a 0a54 6865  ng_tmp).```..The
-0000ba10: 2070 726f 6772 616d 2060 7767 6420 666f   program `wgd fo
-0000ba20: 6375 7360 2069 6d70 6c65 6d65 6e74 6564  cus` implemented
-0000ba30: 2074 776f 2062 6173 6963 2070 6879 6c6f   two basic phylo
-0000ba40: 6765 6e65 7469 6320 696e 6665 7265 6e63  genetic inferenc
-0000ba50: 6520 6d65 7468 6f64 732c 2069 2e65 2e2c  e methods, i.e.,
-0000ba60: 2063 6f6e 6361 7465 6e61 7469 6f6e 2d62   concatenation-b
-0000ba70: 6173 6564 2061 6e64 2063 6f61 6c65 7363  ased and coalesc
-0000ba80: 656e 6365 2d62 6173 6564 206d 6574 686f  ence-based metho
-0000ba90: 6473 2e20 546f 2069 6e69 7469 6174 6520  ds. To initiate 
-0000baa0: 7468 6573 6520 616e 616c 7973 6973 2c20  these analysis, 
-0000bab0: 7573 6572 7320 6e65 6564 2074 6f20 7365  users need to se
-0000bac0: 7420 7468 6520 666c 6167 206f 7074 696f  t the flag optio
-0000bad0: 6e20 602d 2d63 6f6e 6361 7465 6e61 7469  n `--concatenati
-0000bae0: 6f6e 6020 6f72 2060 2d2d 636f 616c 6573  on` or `--coales
-0000baf0: 6365 6e63 6560 2e20 5468 6520 636f 6e63  cence`. The conc
-0000bb00: 6174 656e 6174 696f 6e2d 6261 7365 6420  atenation-based 
-0000bb10: 6d65 7468 6f64 2069 6e63 6c75 6465 7320  method includes 
-0000bb20: 6120 6665 7720 6d61 6a6f 7220 7374 6570  a few major step
-0000bb30: 732c 2069 2e65 2e2c 2074 6865 206d 756c  s, i.e., the mul
-0000bb40: 7469 706c 6520 7365 7175 656e 6365 2061  tiple sequence a
-0000bb50: 6c69 676e 6d65 6e74 2028 4d53 4129 206f  lignment (MSA) o
-0000bb60: 6620 6561 6368 2067 656e 6520 6661 6d69  f each gene fami
-0000bb70: 6c79 2c20 7468 6520 636f 6e63 6174 656e  ly, the concaten
-0000bb80: 6174 696f 6e20 6f66 2061 6c6c 2067 656e  ation of all gen
-0000bb90: 6520 6661 6d69 6c69 6573 2061 6e64 2074  e families and t
-0000bba0: 6865 6e20 7468 6520 6765 6e65 2074 7265  hen the gene tre
-0000bbb0: 6520 2861 6c73 6f20 7370 6563 6965 7320  e (also species 
-0000bbc0: 7472 6565 2069 6e20 7468 6973 2063 6173  tree in this cas
-0000bbd0: 6529 2069 6e66 6572 656e 6365 2e20 5468  e) inference. Th
-0000bbe0: 6520 636f 616c 6573 6365 6e63 652d 6261  e coalescence-ba
-0000bbf0: 7365 6420 6d65 7468 6f64 2077 696c 6c20  sed method will 
-0000bc00: 696e 7374 6561 6420 7065 7266 6f72 6d20  instead perform 
-0000bc10: 7468 6520 4d53 4120 6f66 2065 6163 6820  the MSA of each 
-0000bc20: 6765 6e65 2066 616d 696c 7920 616e 6420  gene family and 
-0000bc30: 7468 6520 6765 6e65 2074 7265 6520 696e  the gene tree in
-0000bc40: 6665 7265 6e63 6520 6261 7365 6420 6f6e  ference based on
-0000bc50: 2065 6163 6820 4d53 412c 2061 6e64 2074   each MSA, and t
-0000bc60: 6865 6e20 696e 6665 7220 7468 6520 7370  hen infer the sp
-0000bc70: 6563 6965 7320 7472 6565 2062 6173 6564  ecies tree based
-0000bc80: 206f 6e20 7468 6573 6520 696e 6469 7669   on these indivi
-0000bc90: 6475 616c 2067 656e 6520 7472 6565 732e  dual gene trees.
-0000bca0: 2054 6865 2074 7265 652d 696e 6665 7265   The tree-infere
-0000bcb0: 6e63 6520 7072 6f67 7261 6d20 616e 6420  nce program and 
-0000bcc0: 7468 6520 6173 736f 6369 6174 6564 2070  the associated p
-0000bcd0: 6172 616d 6574 6572 7320 6361 6e20 6265  arameters can be
-0000bce0: 2073 6574 206a 7573 7420 6173 2061 626f   set just as abo
-0000bcf0: 7665 2062 7920 602d 7472 6565 6020 6f72  ve by `-tree` or
-0000bd00: 2060 2d2d 7472 6565 5f6d 6574 686f 6460   `--tree_method`
-0000bd10: 2061 6e64 2060 2d74 7360 206f 7220 602d   and `-ts` or `-
-0000bd20: 2d74 7265 6573 6574 602e 2050 6c65 6173  -treeset`. Pleas
-0000bd30: 6520 616c 736f 206d 616b 6520 7375 7265  e also make sure
-0000bd40: 2074 6865 2063 686f 7365 6e20 7472 6565   the chosen tree
-0000bd50: 2d69 6e66 6572 656e 6365 2070 726f 6772  -inference progr
-0000bd60: 616d 2069 7320 696e 7374 616c 6c65 6420  am is installed 
-0000bd70: 696e 2074 6865 2065 6e76 6972 6f6e 6d65  in the environme
-0000bd80: 6e74 2070 6174 682e 2054 6865 2070 726f  nt path. The pro
-0000bd90: 6772 616d 2060 6173 7472 616c 2d70 726f  gram `astral-pro
-0000bda0: 6020 6973 2072 6571 7569 7265 6420 746f  ` is required to
-0000bdb0: 2062 6520 696e 7374 616c 6c65 6420 696e   be installed in
-0000bdc0: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-0000bdd0: 2070 6174 6820 6966 2074 6865 2063 6f61   path if the coa
-0000bde0: 6c65 7363 656e 6365 2d62 6173 6564 206d  lescence-based m
-0000bdf0: 6574 686f 6420 6973 2063 686f 7365 6e2e  ethod is chosen.
-0000be00: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
-0000be10: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
-0000be20: 6f20 6163 6365 6c65 7261 7465 2074 6865  o accelerate the
-0000be30: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
-0000be40: 656e 7420 616e 6420 6765 6e65 2074 7265  ent and gene tre
-0000be50: 6520 696e 6665 7265 6e63 6520 666f 7220  e inference for 
-0000be60: 6561 6368 2067 656e 6520 6661 6d69 6c79  each gene family
-0000be70: 2074 6f6f 2061 6e64 2074 6875 7320 7375   too and thus su
-0000be80: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000be90: 7420 6173 206d 7563 6820 6173 2074 6865  t as much as the
-0000bea0: 206e 756d 6265 7220 6f66 2067 656e 6520   number of gene 
-0000beb0: 6661 6d69 6c69 6573 2e0a 0a2a 2a41 2073  families...**A s
-0000bec0: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
-0000bed0: 6720 7275 6e20 6f66 2074 6869 7320 616e  g run of this an
-0000bee0: 616c 7973 6973 2069 7320 7769 7468 2074  alysis is with t
-0000bef0: 6865 2073 696d 706c 6520 636f 6d6d 616e  he simple comman
-0000bf00: 6420 6265 6c6f 772e 2a2a 0a0a 6060 600a  d below.**..```.
-0000bf10: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000bf20: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000bf30: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000bf40: 202d 2d63 6f6e 6361 7465 6e61 7469 6f6e   --concatenation
-0000bf50: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
-0000bf60: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000bf70: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000bf80: 3320 2d2d 636f 616c 6573 6365 6e63 650a  3 --coalescence.
-0000bf90: 6060 600a 0a2a 2a54 6865 2066 756e 6374  ```..**The funct
-0000bfa0: 696f 6e61 6c20 616e 6e6f 7461 7469 6f6e  ional annotation
-0000bfb0: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
-0000bfc0: 732a 2a0a 6060 600a 7767 6420 666f 6375  s**.```.wgd focu
-0000bfd0: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000bfe0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000bff0: 6571 7565 6e63 6533 202d 2d61 6e6e 6f74  equence3 --annot
-0000c000: 6174 696f 6e20 6567 676e 6f67 202d 6564  ation eggnog -ed
-0000c010: 2065 6464 6174 6120 2d2d 646d 6e62 2064   eddata --dmnb d
-0000c020: 6264 6174 610a 6060 600a 0a54 6865 2070  bdata.```..The p
-0000c030: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
-0000c040: 7360 2061 6c73 6f20 6164 6465 6420 736f  s` also added so
-0000c050: 6d65 2077 7261 7070 696e 6720 6675 6e63  me wrapping func
-0000c060: 7469 6f6e 7320 666f 7220 6675 6e63 7469  tions for functi
-0000c070: 6f6e 616c 2061 6e6e 6f74 6174 696f 6e20  onal annotation 
-0000c080: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
-0000c090: 206f 6e20 7468 6520 686f 6f64 206f 6620   on the hood of 
-0000c0a0: 6461 7461 6261 7365 7320 616e 6420 736f  databases and so
-0000c0b0: 6674 7761 7265 7320 666f 7220 696e 7374  ftwares for inst
-0000c0c0: 616e 6365 2060 4567 674e 4f47 6020 616e  ance `EggNOG` an
-0000c0d0: 6420 6045 6767 4e4f 472d 6d61 7070 6572  d `EggNOG-mapper
-0000c0e0: 602e 2046 6f72 2074 6865 2061 6e6e 6f74  `. For the annot
-0000c0f0: 6174 696f 6e20 7573 696e 6720 6045 6767  ation using `Egg
-0000c100: 4e4f 472d 6d61 7070 6572 602c 2075 7365  NOG-mapper`, use
-0000c110: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
-0000c120: 6465 2074 6865 2070 6174 6820 746f 2074  de the path to t
-0000c130: 6865 2065 6767 4e4f 4720 616e 6e6f 7461  he eggNOG annota
-0000c140: 7469 6f6e 2064 6174 6162 6173 6520 7669  tion database vi
-0000c150: 6120 7468 6520 6f70 7469 6f6e 2060 2d65  a the option `-e
-0000c160: 6460 2c20 7468 6520 7061 7468 2074 6f20  d`, the path to 
-0000c170: 7468 6520 6469 616d 6f6e 642d 636f 6d70  the diamond-comp
-0000c180: 6174 6962 6c65 2064 6174 6162 6173 6520  atible database 
-0000c190: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-0000c1a0: 2d2d 646d 6e64 5f64 6260 2061 6e64 2074  --dmnd_db` and t
-0000c1b0: 6865 206f 7074 696f 6e20 602d 2d61 6e6e  he option `--ann
-0000c1c0: 6f74 6174 696f 6e60 2073 6574 2061 7320  otation` set as 
-0000c1d0: 2265 6767 6e6f 6722 2e20 5468 6520 6d61  "eggnog". The ma
-0000c1e0: 6e6e 6572 2068 6f77 2050 4641 4d20 616e  nner how PFAM an
-0000c1f0: 6e6f 7461 7469 6f6e 2077 696c 6c20 6265  notation will be
-0000c200: 2070 6572 666f 726d 6564 2063 616e 2062   performed can b
-0000c210: 6520 636f 6e74 726f 6c6c 6564 2076 6961  e controlled via
-0000c220: 2074 6865 206f 7074 696f 6e20 6070 6661   the option `pfa
-0000c230: 6d60 2c20 6569 7468 6572 2022 6e6f 6e65  m`, either "none
-0000c240: 222c 2022 7265 616c 6967 6e22 206f 7220  ", "realign" or 
-0000c250: 2264 656e 6f76 6f22 2c20 7468 6520 6465  "denovo", the de
-0000c260: 7461 696c 6564 2065 7870 6c61 6e61 7469  tailed explanati
-0000c270: 6f6e 2063 616e 2062 6520 666f 756e 6420  on can be found 
-0000c280: 6174 2074 6865 2077 696b 6920 6f66 205b  at the wiki of [
-0000c290: 4567 674e 4f47 2d6d 6170 7065 725d 2868  EggNOG-mapper](h
-0000c2a0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000c2b0: 6d2f 6567 676e 6f67 6462 2f65 6767 6e6f  m/eggnogdb/eggno
-0000c2c0: 672d 6d61 7070 6572 2f77 696b 6929 2e20  g-mapper/wiki). 
-0000c2d0: 506c 6561 7365 2070 7265 2d69 6e73 7461  Please pre-insta
-0000c2e0: 6c6c 2074 6865 2060 4567 674e 4f47 2d6d  ll the `EggNOG-m
-0000c2f0: 6170 7065 7260 2070 7974 686f 6e20 7061  apper` python pa
-0000c300: 636b 6167 6520 6966 2075 7369 6e67 2074  ckage if using t
-0000c310: 6869 7320 6675 6e63 7469 6f6e 2e20 466f  his function. Fo
-0000c320: 7220 7468 6520 616e 6e6f 7461 7469 6f6e  r the annotation
-0000c330: 2075 7369 6e67 2060 686d 6d73 6361 6e60   using `hmmscan`
-0000c340: 2c20 7768 6174 2069 7320 696d 706c 656d  , what is implem
-0000c350: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
-0000c360: 6020 6973 2061 2073 696d 706c 6520 6275  ` is a simple bu
-0000c370: 6e64 6c65 2066 756e 6374 696f 6e20 746f  ndle function to
-0000c380: 2070 6572 666f 726d 2060 686d 6d73 6361   perform `hmmsca
-0000c390: 6e60 2061 6e61 6c79 7369 7320 666f 7220  n` analysis for 
-0000c3a0: 6120 6769 7665 6e20 686d 6d20 7072 6f66  a given hmm prof
-0000c3b0: 696c 6520 616e 6420 7365 7420 6f66 2067  ile and set of g
-0000c3c0: 656e 6520 6661 6d69 6c69 6573 2073 7563  ene families suc
-0000c3d0: 6820 7468 6174 2075 7365 7273 206e 6565  h that users nee
-0000c3e0: 6420 746f 2073 6574 2074 6865 206f 7074  d to set the opt
-0000c3f0: 696f 6e20 602d 2d61 6e6e 6f74 6174 696f  ion `--annotatio
-0000c400: 6e60 2061 7320 2268 6d6d 7066 616d 2220  n` as "hmmpfam" 
-0000c410: 616e 6420 7072 6f76 6964 6520 7468 6520  and provide the 
-0000c420: 686d 6d70 726f 6669 6c65 2076 6961 2074  hmmprofile via t
-0000c430: 6865 206f 7074 696f 6e20 602d 2d68 6d6d  he option `--hmm
-0000c440: 602e 2046 6f72 2074 6865 2061 6e6e 6f74  `. For the annot
-0000c450: 6174 696f 6e20 7573 696e 6720 6069 6e74  ation using `int
-0000c460: 6572 7072 6f73 6361 6e60 2c20 7573 6572  erproscan`, user
-0000c470: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-0000c480: 6520 7468 6520 7061 7468 2074 6f20 7468  e the path to th
-0000c490: 6520 696e 7465 7270 726f 7363 616e 2069  e interproscan i
-0000c4a0: 6e73 7461 6c6c 6174 696f 6e20 666f 6c64  nstallation fold
-0000c4b0: 6572 2077 6865 7265 2074 6865 7265 2069  er where there i
-0000c4c0: 7320 6120 6069 6e74 6572 7072 6f73 6361  s a `interprosca
-0000c4d0: 6e2e 7368 6020 6669 6c65 2076 6961 2074  n.sh` file via t
-0000c4e0: 6865 206f 7074 696f 6e20 602d 2d65 7865  he option `--exe
-0000c4f0: 7061 7468 6020 616e 6420 7365 7420 7468  path` and set th
-0000c500: 6520 7468 6520 6f70 7469 6f6e 2060 2d2d  e the option `--
-0000c510: 616e 6e6f 7461 7469 6f6e 6020 6173 2022  annotation` as "
-0000c520: 696e 7465 7270 726f 7363 616e 222e 2054  interproscan". T
-0000c530: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000c540: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000c550: 7061 7261 6c6c 656c 697a 6520 7468 6520  parallelize the 
-0000c560: 616e 616c 7973 6973 2066 6f72 2065 6163  analysis for eac
-0000c570: 6820 6765 6e65 2066 616d 696c 7920 616e  h gene family an
-0000c580: 6420 7468 7573 2073 7567 6765 7374 6564  d thus suggested
-0000c590: 2074 6f20 6265 2073 6574 2061 7320 6d75   to be set as mu
-0000c5a0: 6368 2061 7320 7468 6520 6e75 6d62 6572  ch as the number
-0000c5b0: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
-0000c5c0: 732e 0a0a 2a2a 4120 7375 6767 6573 7465  s...**A suggeste
-0000c5d0: 6420 7374 6172 7469 6e67 2072 756e 206f  d starting run o
-0000c5e0: 6620 7468 6973 2061 6e61 6c79 7369 7320  f this analysis 
-0000c5f0: 6361 6e20 6265 2077 6974 6820 7468 6520  can be with the 
-0000c600: 636f 6d6d 616e 6420 6265 6c6f 772e 2a2a  command below.**
-0000c610: 0a0a 6060 600a 7767 6420 666f 6375 7320  ..```.wgd focus 
-0000c620: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-0000c630: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000c640: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
-0000c650: 696f 6e20 6567 676e 6f67 202d 6564 2065  ion eggnog -ed e
-0000c660: 6464 6174 6120 2d2d 646d 6e62 2064 6264  ddata --dmnb dbd
-0000c670: 6174 610a 7767 6420 666f 6375 7320 6661  ata.wgd focus fa
-0000c680: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
-0000c690: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000c6a0: 6e63 6533 202d 2d61 6e6e 6f74 6174 696f  nce3 --annotatio
-0000c6b0: 6e20 686d 6d70 6661 6d20 2d2d 686d 6d20  n hmmpfam --hmm 
-0000c6c0: 686d 6d64 6174 610a 7767 6420 666f 6375  hmmdata.wgd focu
-0000c6d0: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000c6e0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000c6f0: 6571 7565 6e63 6533 202d 2d61 6e6e 6f74  equence3 --annot
-0000c700: 6174 696f 6e20 696e 7465 7270 726f 7363  ation interprosc
-0000c710: 616e 202d 2d65 7865 7061 7468 2024 5041  an --exepath $PA
-0000c720: 5448 0a60 6060 0a0a 2a2a 5468 6520 7068  TH.```..**The ph
-0000c730: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-0000c740: 6720 6f66 2057 4744 732a 2a0a 6060 600a  g of WGDs**.```.
-0000c750: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000c760: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000c770: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000c780: 202d 6420 6d63 6d63 7472 6565 202d 7370   -d mcmctree -sp
-0000c790: 2073 7064 6174 6120 282d 2d70 726f 7463   spdata (--protc
-0000c7a0: 6f63 6461 7469 6e67 2920 282d 2d70 6172  ocdating) (--par
-0000c7b0: 7469 7469 6f6e 2920 282d 2d61 616d 6f64  tition) (--aamod
-0000c7c0: 656c 206c 6729 2028 2d64 7320 2762 7572  el lg) (-ds 'bur
-0000c7d0: 6e69 6e20 3d20 3230 3030 2729 2028 2d64  nin = 2000') (-d
-0000c7e0: 7320 2773 616d 7066 7265 7120 3d20 3130  s 'sampfreq = 10
-0000c7f0: 3030 2729 2028 2d64 7320 276e 7361 6d70  00') (-ds 'nsamp
-0000c800: 6c65 203d 2032 3030 3030 2729 2020 282d  le = 20000')  (-
-0000c810: 6e20 3429 2028 2d2d 746f 5f73 746f 7029  n 4) (--to_stop)
-0000c820: 2028 2d2d 6364 7329 2028 2d6f 2077 6764   (--cds) (-o wgd
-0000c830: 5f66 6f63 7573 2920 282d 7420 776f 726b  _focus) (-t work
-0000c840: 696e 675f 746d 7029 0a60 6060 0a0a 5468  ing_tmp).```..Th
-0000c850: 6520 6162 736f 6c75 7465 2064 6174 696e  e absolute datin
-0000c860: 6720 6f66 2057 4744 7320 6973 2061 2073  g of WGDs is a s
-0000c870: 7065 6369 6669 6320 7069 7065 6c69 6e65  pecific pipeline
-0000c880: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
-0000c890: 6077 6764 2076 3260 2075 7369 6e67 2074  `wgd v2` using t
-0000c8a0: 6865 206d 6574 686f 6420 6f66 2070 6879  he method of phy
-0000c8b0: 6c6f 6765 6e65 7469 6320 6461 7469 6e67  logenetic dating
-0000c8c0: 2e20 5468 6520 6661 6d69 6c69 6573 2075  . The families u
-0000c8d0: 7365 6420 696e 2074 6869 7320 7374 6570  sed in this step
-0000c8e0: 2063 616e 2062 6520 7072 6f64 7563 6564   can be produced
-0000c8f0: 2066 726f 6d20 6077 6764 2064 6d64 6020   from `wgd dmd` 
-0000c900: 616e 6420 6077 6764 2070 6561 6b60 2e20  and `wgd peak`. 
-0000c910: 4e6f 7465 2074 6861 7420 6865 7265 2077  Note that here w
-0000c920: 6520 6f6e 6c79 2064 6973 6375 7373 2068  e only discuss h
-0000c930: 6f77 2074 6f20 6461 7465 2057 4744 7320  ow to date WGDs 
-0000c940: 7769 7468 2067 656e 6f6d 6520 6173 7365  with genome asse
-0000c950: 6d62 6c79 2c20 696e 7374 6561 6420 6f66  mbly, instead of
-0000c960: 2074 7261 6e73 6372 6970 746f 6d65 2061   transcriptome a
-0000c970: 7373 656d 626c 7920 2877 6869 6368 2077  ssembly (which w
-0000c980: 696c 6c20 6265 2064 6973 6375 7373 6564  ill be discussed
-0000c990: 2069 6e20 6120 7365 7061 7261 7465 2073   in a separate s
-0000c9a0: 6563 7469 6f6e 2068 6572 6575 6e64 6572  ection hereunder
-0000c9b0: 292e 2054 6865 2061 7373 756d 7074 696f  ). The assumptio
-0000c9c0: 6e20 7765 206d 6164 6520 6865 7265 2069  n we made here i
-0000c9d0: 7320 7468 6174 206e 6f74 2061 6c6c 2061  s that not all a
-0000c9e0: 6e63 686f 7220 7061 6972 7320 2863 6f6c  nchor pairs (col
-0000c9f0: 6c69 6e65 6172 2064 7570 6c69 6361 7465  linear duplicate
-0000ca00: 7329 2061 7265 2073 7569 7461 626c 6520  s) are suitable 
-0000ca10: 666f 7220 7068 796c 6f67 656e 6574 6963  for phylogenetic
-0000ca20: 2064 6174 696e 672c 2066 6f72 2069 6e73   dating, for ins
-0000ca30: 7461 6e63 6520 7468 6f73 6520 6661 7374  tance those fast
-0000ca40: 6c79 206f 7220 736c 6f77 6c79 2065 766f  ly or slowly evo
-0000ca50: 6c76 696e 6720 6765 6e65 2064 7570 6c69  lving gene dupli
-0000ca60: 6361 7465 732c 2062 6563 6175 7365 2074  cates, because t
-0000ca70: 6865 7927 7265 2070 726f 6e65 2074 6f20  hey're prone to 
-0000ca80: 6769 7665 2062 6961 7365 6420 6461 7469  give biased dati
-0000ca90: 6e67 2065 7374 696d 6174 696f 6e2e 2053  ng estimation. S
-0000caa0: 6f20 6173 2074 6f20 7265 7472 6965 7665  o as to retrieve
-0000cab0: 2074 6865 2022 7265 6c69 6162 6c65 2220   the "reliable" 
-0000cac0: 616e 6368 6f72 2070 6169 7273 2c20 7765  anchor pairs, we
-0000cad0: 2069 6d70 6c65 6d65 6e74 6564 2073 6f6d   implemented som
-0000cae0: 6520 6d65 7468 6f64 7320 6f66 2069 6465  e methods of ide
-0000caf0: 6e74 6966 7969 6e67 2063 7265 6469 6162  ntifying crediab
-0000cb00: 6c65 2061 6e63 686f 7220 7061 6972 7320  le anchor pairs 
-0000cb10: 6261 7365 6420 6f6e 2074 6865 6972 204b  based on their K
-0000cb20: 7320 7661 6c75 6573 2061 6e64 2f6f 7220  s values and/or 
-0000cb30: 7265 7369 6469 6e67 2063 6f6c 6c69 6e65  residing colline
-0000cb40: 6172 2073 6567 6d65 6e74 732e 2046 6f72  ar segments. For
-0000cb50: 2061 2067 656e 6f6d 6520 7769 7468 2063   a genome with c
-0000cb60: 6c65 6172 2073 6967 6e61 6c73 206f 6620  lear signals of 
-0000cb70: 7075 7461 7469 7665 2057 4744 732c 2073  putative WGDs, s
-0000cb80: 7563 6820 6173 2074 6865 202a 4171 7569  uch as the *Aqui
-0000cb90: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-0000cba0: 696e 2074 6865 2065 7861 6d70 6c65 2c20  in the example, 
-0000cbb0: 6120 6865 7572 6973 7469 6320 6d65 7468  a heuristic meth
-0000cbc0: 6f64 2074 6861 7420 6170 706c 6965 7320  od that applies 
-0000cbd0: 7468 6520 7072 696e 6369 706c 6520 6f66  the principle of
-0000cbe0: 2068 6f77 205b 6b73 7261 7465 735d 2868   how [ksrates](h
-0000cbf0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000cc00: 6d2f 5649 422d 5053 422f 6b73 7261 7465  m/VIB-PSB/ksrate
-0000cc10: 7329 2066 696e 6420 7468 6520 696e 6974  s) find the init
-0000cc20: 6961 6c20 7065 616b 7320 616e 6420 7468  ial peaks and th
-0000cc30: 6569 7220 7061 7261 6d65 7465 7273 2077  eir parameters w
-0000cc40: 6173 2069 6d70 6c65 6d65 6e74 6564 2074  as implemented t
-0000cc50: 6f20 6669 6e64 2074 6865 2039 3525 2063  o find the 95% c
-0000cc60: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
-0000cc70: 6f66 2074 6865 2061 7373 756d 6564 206c  of the assumed l
-0000cc80: 6f67 6e6f 726d 616c 2064 6973 7472 6962  ognormal distrib
-0000cc90: 7574 696f 6e20 6f66 2074 6865 2061 6e63  ution of the anc
-0000cca0: 686f 7220 7061 6972 202a 4b2a 3c73 7562  hor pair *K*<sub
-0000ccb0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000ccc0: 7472 6962 7574 696f 6e20 746f 2066 696c  tribution to fil
-0000ccd0: 7465 7220 616e 6368 6f72 2070 6169 7273  ter anchor pairs
-0000cce0: 2077 6974 6820 746f 6f20 6869 6768 206f   with too high o
-0000ccf0: 7220 6c6f 7720 2a4b 2a3c 7375 623e 533c  r low *K*<sub>S<
-0000cd00: 2f73 7562 3e20 6167 6573 2c20 7468 6520  /sub> ages, the 
-0000cd10: 6578 616d 706c 6172 2063 6f6d 6d61 6e64  examplar command
-0000cd20: 206f 6620 7768 6963 6820 6973 2073 686f   of which is sho
-0000cd30: 7765 6420 696e 2074 6865 2060 7767 6420  wed in the `wgd 
-0000cd40: 7065 616b 6020 7365 6374 696f 6e20 6f66  peak` section of
-0000cd50: 205b 496c 6c75 7374 7261 7469 6f6e 5d28   [Illustration](
-0000cd60: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000cd70: 6f6d 2f68 6563 6865 2d70 7362 2f77 6764  om/heche-psb/wgd
-0000cd80: 3f74 6162 3d72 6561 646d 652d 6f76 2d66  ?tab=readme-ov-f
-0000cd90: 696c 6523 696c 6c75 7374 7261 7469 6f6e  ile#illustration
-0000cda0: 292e 2049 6620 7468 6973 2068 6575 7269  ). If this heuri
-0000cdb0: 7374 6963 206d 6574 686f 6420 6661 696c  stic method fail
-0000cdc0: 6564 2074 6f20 6769 7665 2072 6561 736f  ed to give reaso
-0000cdd0: 6e61 626c 6520 7265 7375 6c74 732c 2077  nable results, w
-0000cde0: 6869 6368 2069 7320 7573 7561 6c6c 7920  hich is usually 
-0000cdf0: 6475 6520 746f 2074 6865 2065 6666 6563  due to the effec
-0000ce00: 7420 6f66 206d 756c 7469 706c 6520 6164  t of multiple ad
-0000ce10: 6a61 6365 6e74 2057 4744 7320 7468 6174  jacent WGDs that
-0000ce20: 2062 6c75 7273 2074 6865 2070 6561 6b2d   blurs the peak-
-0000ce30: 6669 6e64 696e 6720 7072 6f63 6573 732c  finding process,
-0000ce40: 2075 7365 7273 2063 616e 2074 7572 6e20   users can turn 
-0000ce50: 746f 2074 6865 2063 6f6c 6c69 6e65 6172  to the collinear
-0000ce60: 2073 6567 6d65 6e74 732d 6775 6964 6564   segments-guided
-0000ce70: 2061 6e63 686f 7220 7061 6972 2063 6c75   anchor pair clu
-0000ce80: 7374 6572 696e 6720 696d 706c 656d 656e  stering implemen
-0000ce90: 7465 6420 696e 2060 7767 6420 7632 602c  ted in `wgd v2`,
-0000cea0: 2069 6e20 7768 6963 6820 6120 636f 6c6c   in which a coll
-0000ceb0: 696e 6561 7220 7365 676d 656e 742d 7769  inear segment-wi
-0000cec0: 7365 2047 4d4d 2063 6c75 7374 6572 696e  se GMM clusterin
-0000ced0: 6720 7769 6c6c 2062 6520 6669 7273 7420  g will be first 
-0000cee0: 636f 6e64 7563 7465 6420 6261 7365 6420  conducted based 
-0000cef0: 6f6e 2074 6865 2022 736f 2d63 616c 6c65  on the "so-calle
-0000cf00: 6422 2073 6567 6d65 6e74 202a 4b2a 3c73  d" segment *K*<s
-0000cf10: 7562 3e53 3c2f 7375 623e 2061 6765 2072  ub>S</sub> age r
-0000cf20: 6570 7265 7365 6e74 6564 2062 7920 7468  epresented by th
-0000cf30: 6520 6d65 6469 616e 202a 4b2a 3c73 7562  e median *K*<sub
-0000cf40: 3e53 3c2f 7375 623e 2061 6765 206f 6620  >S</sub> age of 
-0000cf50: 616c 6c20 7468 6520 7265 7369 6469 6e67  all the residing
-0000cf60: 2073 796e 7465 6c6f 6773 2028 6e6f 7465   syntelogs (note
-0000cf70: 2074 6861 7420 7468 6520 6765 6e65 2064   that the gene d
-0000cf80: 7570 6c69 6361 7465 2070 6169 7273 2061  uplicate pairs a
-0000cf90: 646f 7074 6564 2069 6e20 7468 6973 2073  dopted in this s
-0000cfa0: 7465 7020 6973 2066 726f 6d20 7468 6520  tep is from the 
-0000cfb0: 6669 6c65 2060 6d75 6c74 6970 6c69 636f  file `multiplico
-0000cfc0: 6e5f 7061 6972 732e 7478 7460 2077 6869  n_pairs.txt` whi
-0000cfd0: 6368 2063 6f6e 7461 696e 7320 7468 6520  ch contains the 
-0000cfe0: 6675 6c6c 2073 6574 206f 6620 7379 6e74  full set of synt
-0000cff0: 656c 6f67 7329 2c20 696e 7374 6561 6420  elogs), instead 
-0000d000: 6f66 2074 6865 2073 6d61 6c6c 6572 2067  of the smaller g
-0000d010: 656e 6520 7365 7420 6f66 2061 6e63 686f  ene set of ancho
-0000d020: 7220 7061 6972 732e 2054 6865 2064 6973  r pairs. The dis
-0000d030: 7469 6e63 7469 6f6e 2062 6574 7765 656e  tinction between
-0000d040: 2061 6e63 686f 7220 7061 6972 7320 616e   anchor pairs an
-0000d050: 6420 7379 6e74 656c 6f67 7320 6973 2074  d syntelogs is t
-0000d060: 6861 7420 7468 6520 6c61 7474 6572 2072  hat the latter r
-0000d070: 6566 6572 7320 746f 206d 756c 7469 706c  efers to multipl
-0000d080: 6520 7365 7473 206f 6620 6765 6e65 7320  e sets of genes 
-0000d090: 6465 7269 7665 6420 6672 6f6d 2074 6865  derived from the
-0000d0a0: 2073 616d 6520 616e 6365 7374 7261 6c20   same ancestral 
-0000d0b0: 6765 6e6f 6d69 6320 7265 6769 6f6e 2077  genomic region w
-0000d0c0: 6869 6c65 2074 6865 2066 6f72 6d65 7220  hile the former 
-0000d0d0: 696d 706c 6965 7320 7468 6520 6c61 7474  implies the latt
-0000d0e0: 6572 2062 7574 2072 6571 7569 7265 7320  er but requires 
-0000d0f0: 6164 6469 7469 6f6e 616c 6c79 2074 6865  additionally the
-0000d100: 2063 6f6e 7365 7276 6564 2067 656e 6520   conserved gene 
-0000d110: 6f72 6465 722c 2062 6f74 6820 6f66 2077  order, both of w
-0000d120: 6869 6368 2061 7265 2c20 7769 7468 696e  hich are, within
-0000d130: 2061 2067 656e 6f6d 652c 2061 7373 756d   a genome, assum
-0000d140: 6564 2074 6f20 6265 206f 7269 6769 6e61  ed to be origina
-0000d150: 7465 6420 6672 6f6d 2074 6865 2064 7570  ted from the dup
-0000d160: 6c69 6361 7469 6f6e 206f 6620 6120 636f  lication of a co
-0000d170: 6d6d 6f6e 2061 6e63 6573 7472 616c 2067  mmon ancestral g
-0000d180: 656e 6f6d 6963 2072 6567 696f 6e20 616e  enomic region an
-0000d190: 6420 6173 2073 7563 6820 6465 656d 6564  d as such deemed
-0000d1a0: 2065 7669 6465 6e63 6520 666f 7220 5747   evidence for WG
-0000d1b0: 442e 2054 6865 6e20 7468 6520 7379 6e74  D. Then the synt
-0000d1c0: 656c 6f67 7320 7769 6c6c 2062 6520 6d61  elogs will be ma
-0000d1d0: 7070 6564 2062 6163 6b20 6163 636f 7264  pped back accord
-0000d1e0: 696e 6720 746f 2074 6865 2063 6c75 7374  ing to the clust
-0000d1f0: 6572 696e 6720 7265 7375 6c74 7320 6f66  ering results of
-0000d200: 2074 6865 6972 2061 6666 696c 6961 7465   their affiliate
-0000d210: 6420 7365 676d 656e 7473 2e20 5369 6e63  d segments. Sinc
-0000d220: 6520 7468 6520 4761 7573 7369 616e 2073  e the Gaussian s
-0000d230: 6861 7065 206f 6620 7468 6520 7365 676d  hape of the segm
-0000d240: 656e 7420 636c 7573 7465 7220 646f 6573  ent cluster does
-0000d250: 6e27 7420 6e65 6365 7373 6172 696c 7920  n't necessarily 
-0000d260: 696d 6974 6174 6520 7468 6520 7368 6170  imitate the shap
-0000d270: 6520 6f66 2074 6865 2072 6573 6964 696e  e of the residin
-0000d280: 6720 7379 6e74 656c 6f67 732c 2073 6f20  g syntelogs, so 
-0000d290: 6173 2074 6f20 7265 7472 6965 7665 2074  as to retrieve t
-0000d2a0: 6865 2022 7265 6c69 6162 6c65 2220 6765  he "reliable" ge
-0000d2b0: 6e65 2073 6574 7320 666f 7220 7068 796c  ne sets for phyl
-0000d2c0: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
-0000d2d0: 2077 6520 6164 706f 7465 6420 7468 6520   we adpoted the 
-0000d2e0: 2839 3525 2920 6869 6768 6573 7420 6465  (95%) highest de
-0000d2f0: 6e73 6974 7920 7265 6769 6f6e 2028 4844  nsity region (HD
-0000d300: 5229 206f 6620 7468 6520 7379 6e74 656c  R) of the syntel
-0000d310: 6f67 202a 4b2a 3c73 7562 3e53 3c2f 7375  og *K*<sub>S</su
-0000d320: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000d330: 696f 6e20 666f 7220 7468 6520 7068 796c  ion for the phyl
-0000d340: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
-0000d350: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
-0000d360: 206f 6620 7468 6520 2839 3525 2920 4844   of the (95%) HD
-0000d370: 5220 696e 2074 6865 2066 756e 6374 696f  R in the functio
-0000d380: 6e20 5b63 616c 6375 6c61 7465 4850 445d  n [calculateHPD]
-0000d390: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-0000d3a0: 636f 6d2f 6865 6368 652d 7073 622f 7767  com/heche-psb/wg
-0000d3b0: 642f 626c 6f62 2f70 6879 6c6f 6461 7469  d/blob/phylodati
-0000d3c0: 6e67 2f77 6764 2f70 6561 6b2e 7079 234c  ng/wgd/peak.py#L
-0000d3d0: 3137 3537 2920 7365 656b 7320 7468 6520  1757) seeks the 
-0000d3e0: 7368 6f72 7465 7374 202a 4b2a 3c73 7562  shortest *K*<sub
-0000d3f0: 3e53 3c2f 7375 623e 2072 616e 6765 2028  >S</sub> range (
-0000d400: 612c 6229 2077 6869 6368 2073 6174 6973  a,b) which satis
-0000d410: 6669 6573 2074 6865 2072 6571 7569 7265  fies the require
-0000d420: 6d65 6e74 206f 6620 7370 616e 6e69 6e67  ment of spanning
-0000d430: 206d 6f72 6520 7468 616e 2028 3935 2529   more than (95%)
-0000d440: 206f 6620 616c 6c20 7468 6520 2a4b 2a3c   of all the *K*<
-0000d450: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-0000d460: 6573 2e20 4f6e 2074 6865 2070 7265 6d69  es. On the premi
-0000d470: 7365 206f 6620 6964 656e 7469 6669 6564  se of identified
-0000d480: 2061 6e63 686f 7220 7061 6972 7320 286e   anchor pairs (n
-0000d490: 6f74 6520 7468 6174 2074 6865 2073 796e  ote that the syn
-0000d4a0: 7465 6c6f 6773 2061 7265 2061 6c73 6f20  telogs are also 
-0000d4b0: 7265 6665 7272 6564 2061 7320 616e 6368  referred as anch
-0000d4c0: 6f72 2070 6169 7273 2068 6572 6561 6674  or pairs hereaft
-0000d4d0: 6572 292c 2077 6520 696d 706c 656d 656e  er), we implemen
-0000d4e0: 7465 6420 696e 2074 6865 2070 726f 6772  ted in the progr
-0000d4f0: 616d 2060 7767 6420 646d 6460 2074 6865  am `wgd dmd` the
-0000d500: 2073 6f2d 6361 6c6c 6564 2061 6e63 686f   so-called ancho
-0000d510: 722d 6177 6172 6520 6c6f 6361 6c20 4d52  r-aware local MR
-0000d520: 4248 7320 6f72 206f 7274 686f 6772 6f75  BHs or orthogrou
-0000d530: 7073 2c20 696e 2077 6869 6368 2074 6865  ps, in which the
-0000d540: 206f 7269 6769 6e61 6c20 6c6f 6361 6c20   original local 
-0000d550: 4d52 4248 7320 6172 6520 6675 7274 6865  MRBHs are furthe
-0000d560: 7220 6d65 7267 6564 2077 6974 6820 616e  r merged with an
-0000d570: 6368 6f72 2070 6169 7273 2073 7563 6820  chor pairs such 
-0000d580: 7468 6174 2065 6163 6820 6f72 7468 6f67  that each orthog
-0000d590: 726f 7570 2063 6f6e 7461 696e 7320 7468  roup contains th
-0000d5a0: 6520 616e 6368 6f72 2070 6169 7220 616e  e anchor pair an
-0000d5b0: 6420 7468 6520 6f72 7468 6f6c 6f67 7565  d the orthologue
-0000d5c0: 732e 2057 6974 6820 7468 6973 206f 7274  s. With this ort
-0000d5d0: 686f 6772 6f75 702c 2075 7365 7273 2074  hogroup, users t
-0000d5e0: 6865 6e20 6e65 6564 206f 6e65 2073 7461  hen need one sta
-0000d5f0: 7274 696e 6720 7472 6565 2066 696c 6520  rting tree file 
-0000d600: 2861 7320 7368 6f77 6e20 696e 2074 6865  (as shown in the
-0000d610: 205b 496c 6c75 7374 7261 7469 6f6e 5d28   [Illustration](
-0000d620: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000d630: 6f6d 2f68 6563 6865 2d70 7362 2f77 6764  om/heche-psb/wgd
-0000d640: 2f74 7265 652f 7068 796c 6f64 6174 696e  /tree/phylodatin
-0000d650: 673f 7461 623d 7265 6164 6d65 2d6f 762d  g?tab=readme-ov-
-0000d660: 6669 6c65 2369 6c6c 7573 7472 6174 696f  file#illustratio
-0000d670: 6e29 2070 6172 7429 2069 6e64 6963 6174  n) part) indicat
-0000d680: 696e 6720 7468 6520 7472 6565 2074 6f70  ing the tree top
-0000d690: 6f6c 6f67 7920 616e 6420 666f 7373 696c  ology and fossil
-0000d6a0: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
-0000d6b0: 6f72 6d61 7469 6f6e 2066 6f72 2074 6865  ormation for the
-0000d6c0: 2066 696e 616c 2070 6879 6c6f 6765 6e65   final phylogene
-0000d6d0: 7469 6320 6461 7469 6e67 2075 7369 6e67  tic dating using
-0000d6e0: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
-0000d6f0: 6420 666f 6375 7360 2e20 5573 6572 7320  d focus`. Users 
-0000d700: 6361 6e20 7365 7420 7468 6520 666c 6167  can set the flag
-0000d710: 206f 7074 696f 6e20 602d 2d70 726f 7463   option `--protc
-0000d720: 6f63 6461 7469 6e67 6020 746f 206f 6e6c  ocdating` to onl
-0000d730: 7920 636f 6e64 7563 7420 7468 6520 6461  y conduct the da
-0000d740: 7469 6e67 206f 6620 7468 6520 636f 6e63  ting of the conc
-0000d750: 6174 656e 6174 696f 6e20 7072 6f74 6569  atenation protei
-0000d760: 6e20 4d53 4120 6f72 2074 6865 2066 6c61  n MSA or the fla
-0000d770: 6720 6f70 7469 6f6e 2060 2d2d 7072 6f74  g option `--prot
-0000d780: 6461 7469 6e67 6020 746f 206f 6e6c 7920  dating` to only 
-0000d790: 636f 6e64 7563 7420 7468 6520 6461 7469  conduct the dati
-0000d7a0: 6e67 206f 6620 7468 6520 7072 6f74 6569  ng of the protei
-0000d7b0: 6e20 4d53 4173 2c20 6e6f 7465 6420 7468  n MSAs, noted th
-0000d7c0: 6174 2074 6865 7365 2074 776f 206f 7074  at these two opt
-0000d7d0: 696f 6e73 206f 6e6c 7920 776f 726b 2066  ions only work f
-0000d7e0: 6f72 2074 6865 2060 6d63 6d63 7472 6565  or the `mcmctree
-0000d7f0: 6020 6f70 7469 6f6e 2073 6f20 6661 722e  ` option so far.
-0000d800: 2054 6865 2066 6c61 6720 6f70 7469 6f6e   The flag option
-0000d810: 2060 2d2d 7061 7274 6974 696f 6e60 2063   `--partition` c
-0000d820: 616e 2062 6520 7365 7420 746f 2070 6572  an be set to per
-0000d830: 666f 726d 2060 6d63 6d63 7472 6565 6020  form `mcmctree` 
-0000d840: 616e 616c 7973 6973 2075 7369 6e67 2074  analysis using t
-0000d850: 6865 2070 6172 7469 7469 6f6e 6564 2064  he partitioned d
-0000d860: 6174 6120 2869 2e65 2e2c 2031 7374 2c20  ata (i.e., 1st, 
-0000d870: 326e 6420 616e 6420 3372 6420 706f 7369  2nd and 3rd posi
-0000d880: 7469 6f6e 206f 6620 636f 646f 6e29 2069  tion of codon) i
-0000d890: 6e73 7465 6164 206f 6620 7573 696e 6720  nstead of using 
-0000d8a0: 7468 6520 636f 646f 6e20 6173 2061 2077  the codon as a w
-0000d8b0: 686f 6c65 2e20 5468 6520 6f70 7469 6f6e  hole. The option
-0000d8c0: 2060 2d2d 6161 6d6f 6465 6c60 2063 616e   `--aamodel` can
-0000d8d0: 2062 6520 7365 7420 746f 2064 6574 6572   be set to deter
-0000d8e0: 6d69 6e65 2074 6865 2061 6d69 6e6f 2061  mine the amino a
-0000d8f0: 6369 6420 6d6f 6465 6c20 6170 706c 6965  cid model applie
-0000d900: 6420 696e 2060 6d63 6d63 7472 6565 6020  d in `mcmctree` 
-0000d910: 616e 616c 7973 6973 2e20 5468 6520 6f70  analysis. The op
-0000d920: 7469 6f6e 2060 2d64 7360 2063 616e 2062  tion `-ds` can b
-0000d930: 6520 7573 6564 2074 6f20 7365 7420 7061  e used to set pa
-0000d940: 7261 6d65 7465 7273 2066 6f72 2074 6865  rameters for the
-0000d950: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
-0000d960: 6720 7072 6f67 7261 6d2e 2054 6865 2070  g program. The p
-0000d970: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000d980: 6865 7265 2061 7265 2074 6f20 7061 7261  here are to para
-0000d990: 6c6c 656c 697a 6520 7468 6520 616e 616c  llelize the anal
-0000d9a0: 7973 6973 2066 6f72 2065 6163 6820 6765  ysis for each ge
-0000d9b0: 6e65 2066 616d 696c 7920 616e 6420 7468  ne family and th
-0000d9c0: 7573 2073 7567 6765 7374 6564 2074 6f20  us suggested to 
-0000d9d0: 6265 2073 6574 2061 7320 6d75 6368 2061  be set as much a
-0000d9e0: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000d9f0: 6765 6e65 2066 616d 696c 6965 732e 0a0a  gene families...
-0000da00: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-0000da10: 6172 7469 6e67 2072 756e 2063 616e 2062  arting run can b
-0000da20: 6520 7769 7468 2074 6865 2063 6f6d 6d61  e with the comma
-0000da30: 6e64 2062 656c 6f77 2e2a 2a0a 0a60 6060  nd below.**..```
-0000da40: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
-0000da50: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000da60: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000da70: 3320 2d64 206d 636d 6374 7265 6520 2d73  3 -d mcmctree -s
-0000da80: 7020 7370 6461 7461 0a77 6764 2066 6f63  p spdata.wgd foc
-0000da90: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000daa0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000dab0: 7365 7175 656e 6365 3320 2d64 2072 3873  sequence3 -d r8s
-0000dac0: 202d 7370 2073 7064 6174 6120 2d2d 6e73   -sp spdata --ns
-0000dad0: 6974 6573 206e 7369 7465 696e 666f 0a77  ites nsiteinfo.w
-0000dae0: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
-0000daf0: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000db00: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000db10: 2d64 2062 6561 7374 202d 2d66 6f73 7369  -d beast --fossi
-0000db20: 6c20 666f 7373 696c 696e 666f 202d 2d72  l fossilinfo --r
-0000db30: 6f6f 7468 6569 6768 7420 726f 6f74 6865  ootheight roothe
-0000db40: 6967 6874 696e 666f 202d 2d63 6861 696e  ightinfo --chain
-0000db50: 7365 7420 6368 6169 6e73 6574 696e 666f  set chainsetinfo
-0000db60: 202d 2d62 6561 7374 6c67 6a61 7220 2450   --beastlgjar $P
-0000db70: 4154 480a 6060 600a 0a23 2323 2077 6764  ATH.```..### wgd
-0000db80: 206b 7364 0a0a 2a2a 5468 6520 636f 6e73   ksd..**The cons
-0000db90: 7472 7563 7469 6f6e 206f 6620 7768 6f6c  truction of whol
-0000dba0: 6520 7061 7261 6e6f 6d65 202a 4b2a 3c73  e paranome *K*<s
-0000dbb0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000dbc0: 6973 7472 6962 7574 696f 6e2a 2a0a 6060  istribution**.``
-0000dbd0: 600a 7767 6420 6b73 6420 6661 6d69 6c69  `.wgd ksd famili
-0000dbe0: 6573 2073 6571 7565 6e63 6520 282d 6f20  es sequence (-o 
-0000dbf0: 7767 645f 6b73 6420 2d74 2077 6764 5f6b  wgd_ksd -t wgd_k
-0000dc00: 7364 5f74 6d70 202d 2d6e 7468 7265 6164  sd_tmp --nthread
-0000dc10: 7320 3420 2d2d 746f 5f73 746f 7020 2d2d  s 4 --to_stop --
-0000dc20: 6364 7320 2d2d 7061 6972 7769 7365 202d  cds --pairwise -
-0000dc30: 2d73 7472 6970 5f67 6170 7320 2d2d 616c  -strip_gaps --al
-0000dc40: 6967 6e65 7220 6d61 6666 7420 2d2d 7472  igner mafft --tr
-0000dc50: 6565 5f6d 6574 686f 6420 6661 7374 7472  ee_method fasttr
-0000dc60: 6565 202d 2d6e 6f64 655f 6176 6572 6167  ee --node_averag
-0000dc70: 6529 0a60 6060 0a0a 5468 6520 7072 6f67  e).```..The prog
-0000dc80: 7261 6d20 6077 6764 206b 7364 602c 2061  ram `wgd ksd`, a
-0000dc90: 7320 696d 7069 6564 2062 7920 6974 7320  s impied by its 
-0000dca0: 6e61 6d65 2c20 6973 2066 6f72 2074 6865  name, is for the
-0000dcb0: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-0000dcc0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000dcd0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000dce0: 6e2e 2045 7863 6570 7420 666f 7220 7468  n. Except for th
-0000dcf0: 6520 6166 6f72 656d 656e 7469 6f6e 6564  e aforementioned
-0000dd00: 2070 6172 616d 6574 6572 7320 7375 6368   parameters such
-0000dd10: 2061 7320 602d 2d74 6f5f 7374 6f70 6020   as `--to_stop` 
-0000dd20: 616e 6420 602d 2d63 6473 602c 2074 6865  and `--cds`, the
-0000dd30: 7265 2061 7265 2073 6f6d 6520 696d 706f  re are some impo
-0000dd40: 7274 616e 7420 7061 7261 6d65 7465 7273  rtant parameters
-0000dd50: 2074 6861 7420 6861 7665 2063 7275 6369   that have cruci
-0000dd60: 616c 2069 6d70 6163 7420 6f6e 2074 6865  al impact on the
-0000dd70: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000dd80: 2065 7374 696d 6174 696f 6e2e 2054 6865   estimation. The
-0000dd90: 206f 7074 696f 6e20 602d 2d70 6169 7277   option `--pairw
-0000dda0: 6973 6560 2069 7320 6120 7665 7279 2069  ise` is a very i
-0000ddb0: 6d70 6f72 7461 6e74 2070 6172 616d 6574  mportant paramet
-0000ddc0: 6572 2066 6f72 2074 6865 202a 4b2a 3c73  er for the *K*<s
-0000ddd0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
-0000dde0: 6174 696f 6e2c 2077 6974 6820 7768 6963  ation, with whic
-0000ddf0: 6820 7468 6520 6043 4f44 454d 4c60 2077  h the `CODEML` w
-0000de00: 696c 6c20 6361 6c63 756c 6174 6520 7468  ill calculate th
-0000de10: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000de20: 3e20 666f 7220 6561 6368 2067 656e 6520  > for each gene 
-0000de30: 7061 6972 2073 6570 6172 6174 656c 7920  pair separately 
-0000de40: 6261 7365 6420 6f6e 2074 6865 2061 6c69  based on the ali
-0000de50: 676e 6d65 6e74 206f 6620 6f6e 6c79 2074  gnment of only t
-0000de60: 6865 7365 2074 776f 2067 656e 6573 2069  hese two genes i
-0000de70: 6e73 7465 6164 206f 6620 7468 6520 7768  nstead of the wh
-0000de80: 6f6c 6520 616c 6967 6e6d 656e 7420 6f66  ole alignment of
-0000de90: 2074 6865 2066 616d 696c 792c 2073 7563   the family, suc
-0000dea0: 6820 7468 6174 206c 6573 7320 6761 7073  h that less gaps
-0000deb0: 2061 7265 2065 7870 6563 7465 6420 616e   are expected an
-0000dec0: 6420 7468 7573 2074 6865 2061 6c69 676e  d thus the align
-0000ded0: 6d65 6e74 2069 6e20 7468 6520 636f 6e73  ment in the cons
-0000dee0: 6964 6572 6174 696f 6e20 6f66 2060 434f  ideration of `CO
-0000def0: 4445 4d4c 6020 7769 6c6c 2062 6520 6c6f  DEML` will be lo
-0000df00: 6e67 6572 2028 6265 6361 7573 6520 6043  nger (because `C
-0000df10: 4f44 454d 4c60 2077 696c 6c20 6175 746f  ODEML` will auto
-0000df20: 6d61 7465 6c79 2073 6b69 7020 6576 6572  mately skip ever
-0000df30: 7920 636f 6c75 6d6e 2077 6974 6820 6761  y column with ga
-0000df40: 702c 2072 6567 6172 646c 6573 7320 6f66  p, regardless of
-0000df50: 2077 6865 7468 6572 2069 7427 7320 616e   whether it's an
-0000df60: 206f 7665 7261 6c6c 206f 7220 7061 7274   overall or part
-0000df70: 6961 6c20 6761 7029 2c20 7769 7468 6f75  ial gap), withou
-0000df80: 7420 7768 6963 6820 7468 6520 6043 4f44  t which the `COD
-0000df90: 454d 4c60 2077 696c 6c20 6361 6c63 756c  EML` will calcul
-0000dfa0: 6174 6520 7468 6520 2a4b 2a3c 7375 623e  ate the *K*<sub>
-0000dfb0: 533c 2f73 7562 3e20 6261 7365 6420 6f6e  S</sub> based on
-0000dfc0: 2074 6865 2077 686f 6c65 2061 6c69 676e   the whole align
-0000dfd0: 6d65 6e74 206f 6620 7468 6520 6661 6d69  ment of the fami
-0000dfe0: 6c79 2c20 7768 6963 6820 6d69 6768 7420  ly, which might 
-0000dff0: 6861 7665 206e 6f20 2a4b 2a3c 7375 623e  have no *K*<sub>
-0000e000: 533c 2f73 7562 3e20 7265 7375 6c74 2061  S</sub> result a
-0000e010: 7420 616c 6c20 6966 2074 6865 2073 7472  t all if the str
-0000e020: 6970 7065 6420 616c 6967 6e6d 656e 7420  ipped alignment 
-0000e030: 6c65 6e67 7468 2028 7265 6d6f 7669 6e67  length (removing
-0000e040: 2061 6c6c 2067 6170 2d63 6f6e 7461 696e   all gap-contain
-0000e050: 696e 6720 636f 6c75 6d6e 7329 2077 6173  ing columns) was
-0000e060: 207a 6572 6f2c 2061 2063 6175 7365 206f   zero, a cause o
-0000e070: 6620 6469 6666 6572 656e 7420 6e75 6d62  f different numb
-0000e080: 6572 206f 6620 2a4b 2a3c 7375 623e 533c  er of *K*<sub>S<
-0000e090: 2f73 7562 3e20 6573 7469 6d61 7465 7320  /sub> estimates 
-0000e0a0: 6265 7477 6565 6e20 2270 6169 7277 6973  between "pairwis
-0000e0b0: 6520 6d6f 6465 2220 616e 6420 226e 6f6e  e mode" and "non
-0000e0c0: 2d70 6169 7277 6973 6520 6d6f 6465 222e  -pairwise mode".
-0000e0d0: 2049 7427 7320 6469 6666 6963 756c 7420   It's difficult 
-0000e0e0: 746f 2073 6179 2077 6869 6368 206d 6f64  to say which mod
-0000e0f0: 6520 6973 206d 6f72 6520 6964 6561 6c2c  e is more ideal,
-0000e100: 2061 6c74 686f 7567 6820 7468 6520 226e   although the "n
-0000e110: 6f6e 2d70 6169 7277 6973 6520 6d6f 6465  on-pairwise mode
-0000e120: 2220 2864 6566 6175 6c74 2073 6574 7469  " (default setti
-0000e130: 6e67 2920 7768 6963 6820 7275 6e73 206f  ng) which runs o
-0000e140: 6e20 7468 6520 7768 6f6c 6520 616c 6967  n the whole alig
-0000e150: 6e6d 656e 7420 696e 7374 6561 6420 6f66  nment instead of
-0000e160: 2061 206c 6f63 616c 2061 6c69 676e 6d65   a local alignme
-0000e170: 6e74 2c20 6d69 6768 7420 6265 206d 6f72  nt, might be mor
-0000e180: 6520 6269 6f6c 6f67 6963 616c 6c79 2063  e biologically c
-0000e190: 6f6e 7365 7276 6564 2069 6e20 7468 6174  onserved in that
-0000e1a0: 2069 7420 6173 7375 7265 7320 7468 6520   it assures the 
-0000e1b0: 6576 6f6c 7574 696f 6e20 6f66 2065 6163  evolution of eac
-0000e1c0: 6820 636f 6c75 6d6e 2074 6f20 6265 2073  h column to be s
-0000e1d0: 7461 7274 6564 2066 726f 6d20 7468 6520  tarted from the 
-0000e1e0: 726f 6f74 206f 6620 7468 6520 6661 6d69  root of the fami
-0000e1f0: 6c79 2061 6e64 2061 6c6c 2074 6865 2067  ly and all the g
-0000e200: 656e 6520 6475 706c 6963 6174 6573 2061  ene duplicates a
-0000e210: 7265 2074 616b 656e 2069 6e74 6f20 6163  re taken into ac
-0000e220: 636f 756e 7420 696e 2074 6865 202a 4b2a  count in the *K*
-0000e230: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
-0000e240: 696d 6174 696f 6e20 7072 6f63 6573 732e  imation process.
-0000e250: 2054 6865 206f 7074 696f 6e20 602d 2d73   The option `--s
-0000e260: 7472 6970 5f67 6170 7360 2063 616e 2072  trip_gaps` can r
-0000e270: 656d 6f76 6520 616c 6c20 7468 6520 6761  emove all the ga
-0000e280: 702d 636f 6e74 6169 6e69 6e67 2063 6f6c  p-containing col
-0000e290: 756d 6e73 2c20 7769 7468 206f 7220 7769  umns, with or wi
-0000e2a0: 7468 6f75 7420 7768 6963 6820 7468 6520  thout which the 
-0000e2b0: 7265 7375 6c74 206f 6620 226e 6f6e 2d70  result of "non-p
-0000e2c0: 6169 7277 6973 6520 6d6f 6465 2220 776f  airwise mode" wo
-0000e2d0: 6e27 7420 6265 2061 6666 6563 7465 642c  n't be affected,
-0000e2e0: 2077 6869 6c65 2077 6974 6820 7768 6963   while with whic
-0000e2f0: 6820 7468 6520 7265 7375 6c74 206f 6620  h the result of 
-0000e300: 2270 6169 7277 6973 6520 6d6f 6465 2220  "pairwise mode" 
-0000e310: 7769 6c6c 2062 6520 616c 7465 7265 642e  will be altered.
-0000e320: 2054 6865 206f 7074 696f 6e20 602d 2d61   The option `--a
-0000e330: 6c69 676e 6572 6020 616e 6420 602d 2d61  ligner` and `--a
-0000e340: 6c6e 5f6f 7074 696f 6e73 6020 7768 6963  ln_options` whic
-0000e350: 6820 6465 6369 6465 2077 6869 6368 2061  h decide which a
-0000e360: 6c69 676e 6d65 6e74 2070 726f 6772 616d  lignment program
-0000e370: 2074 6f20 6265 2075 7365 6420 616e 6420   to be used and 
-0000e380: 7768 6963 6820 7061 7261 6d65 7465 7220  which parameter 
-0000e390: 746f 2062 6520 7365 742c 2077 696c 6c20  to be set, will 
-0000e3a0: 6861 7665 2069 6d70 6163 7420 6f6e 2074  have impact on t
-0000e3b0: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-0000e3c0: 623e 2072 6573 756c 7473 2c20 6e6f 7465  b> results, note
-0000e3d0: 6420 7468 6174 2074 6865 2064 6566 6175  d that the defau
-0000e3e0: 6c74 2070 726f 6772 616d 2069 7320 606d  lt program is `m
-0000e3f0: 6166 6674 6020 616e 6420 7468 6520 7061  afft` and the pa
-0000e400: 7261 6d65 7465 7220 6973 2060 2d2d 6175  rameter is `--au
-0000e410: 746f 602e 2054 6865 206f 7074 696f 6e20  to`. The option 
-0000e420: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
-0000e430: 616e 6420 602d 2d74 7265 655f 6f70 7469  and `--tree_opti
-0000e440: 6f6e 7360 2064 6563 6964 6520 7768 6963  ons` decide whic
-0000e450: 6820 6765 6e65 2074 7265 6520 696e 6665  h gene tree infe
-0000e460: 7265 6e63 6520 7072 6f67 7261 6d20 746f  rence program to
-0000e470: 2062 6520 7573 6564 2061 6e64 2077 6869   be used and whi
-0000e480: 6368 2070 6172 616d 6574 6572 2074 6f20  ch parameter to 
-0000e490: 6265 2073 6574 2c20 776f 6e27 7420 6166  be set, won't af
-0000e4a0: 6665 6374 2074 6865 202a 4b2a 3c73 7562  fect the *K*<sub
-0000e4b0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-0000e4c0: 696f 6e20 6974 7365 6c66 2062 7574 2074  ion itself but t
-0000e4d0: 6865 2072 6573 756c 7420 6f66 2064 652d  he result of de-
-0000e4e0: 7265 6475 6e64 616e 6379 2c20 6e6f 7465  redundancy, note
-0000e4f0: 6420 7468 6174 2077 6520 696d 706c 656d  d that we implem
-0000e500: 656e 7465 6420 6120 6275 696c 742d 696e  ented a built-in
-0000e510: 2067 656e 6520 7472 6565 2069 6e66 6572   gene tree infer
-0000e520: 656e 6365 206d 6574 686f 6420 6261 7365  ence method base
-0000e530: 6420 6f6e 2074 6865 2041 7665 7261 6765  d on the Average
-0000e540: 204c 696e 6b61 6765 2043 6c75 7374 6572   Linkage Cluster
-0000e550: 696e 6720 2841 4c43 2920 2874 6875 7320  ing (ALC) (thus 
-0000e560: 6120 6469 7374 616e 6365 2d62 6173 6564  a distance-based
-0000e570: 2074 7265 6529 2077 6974 6820 7468 6520   tree) with the 
-0000e580: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
-0000e590: 7365 7420 6173 2022 636c 7573 7465 7222  set as "cluster"
-0000e5a0: 2e20 5477 6f20 6d65 7468 6f64 7320 6f66  . Two methods of
-0000e5b0: 2064 652d 7265 6475 6e64 616e 6379 2061   de-redundancy a
-0000e5c0: 7265 2069 6d70 6c65 6d65 6e74 6564 2069  re implemented i
-0000e5d0: 6e20 6077 6764 2076 3260 2c20 6e61 6d65  n `wgd v2`, name
-0000e5e0: 6c79 206e 6f64 652d 7765 6967 6874 6564  ly node-weighted
-0000e5f0: 2061 6e64 206e 6f64 652d 6176 6572 6167   and node-averag
-0000e600: 6564 206d 6574 686f 6473 2e20 5468 6520  ed methods. The 
-0000e610: 6e6f 6465 2d77 6569 6768 7465 6420 6d65  node-weighted me
-0000e620: 7468 6f64 2061 6368 6965 7665 7320 7468  thod achieves th
-0000e630: 6520 6465 2d72 6564 756e 6461 6e63 7920  e de-redundancy 
-0000e640: 7669 6120 7765 6967 6869 6e67 2074 6865  via weighing the
-0000e650: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000e660: 2076 616c 7565 2061 7373 6f63 6961 7465   value associate
-0000e670: 6420 7769 7468 2065 6163 6820 6765 6e65  d with each gene
-0000e680: 2070 6169 7220 7375 6368 2074 6861 7420   pair such that 
-0000e690: 7468 6520 7765 6967 6874 206f 6620 6120  the weight of a 
-0000e6a0: 7369 6e67 6c65 2064 7570 6c69 6361 7469  single duplicati
-0000e6b0: 6f6e 2065 7665 6e74 2073 756d 7320 7570  on event sums up
-0000e6c0: 2074 6f20 3120 286e 6f74 6564 2074 6861   to 1 (noted tha
-0000e6d0: 7420 7468 6520 6e75 6d62 6572 206f 6620  t the number of 
-0000e6e0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e6f0: 6573 7469 6d61 7469 6f6e 2072 656d 6169  estimation remai
-0000e700: 6e73 2074 6865 2073 616d 6529 2077 6869  ns the same) whi
-0000e710: 6c65 2074 6865 206e 6f64 652d 6176 6572  le the node-aver
-0000e720: 6167 6564 206d 6574 686f 6420 7265 616c  aged method real
-0000e730: 697a 6573 2074 6865 2064 652d 7265 6475  izes the de-redu
-0000e740: 6e64 616e 6379 2076 6961 2063 616c 6375  ndancy via calcu
-0000e750: 6c61 7469 6e67 2070 6572 2067 656e 6520  lating per gene 
-0000e760: 7472 6565 206e 6f64 6520 6f6e 6520 6176  tree node one av
-0000e770: 6572 6167 6564 202a 4b2a 3c73 7562 3e53  eraged *K*<sub>S
-0000e780: 3c2f 7375 623e 2076 616c 7565 2074 6f20  </sub> value to 
-0000e790: 7265 7072 6573 656e 7420 7468 6520 6167  represent the ag
-0000e7a0: 6520 6f66 2065 6163 6820 6475 706c 6963  e of each duplic
-0000e7b0: 6174 696f 6e20 6576 656e 742e 2054 6865  ation event. The
-0000e7c0: 206f 7074 696f 6e20 602d 2d6e 6f64 655f   option `--node_
-0000e7d0: 6176 6572 6167 6560 2063 616e 2062 6520  average` can be 
-0000e7e0: 7365 7420 746f 2063 686f 6f73 6520 7468  set to choose th
-0000e7f0: 6520 6e6f 6465 2d61 7665 7261 6765 6420  e node-averaged 
-0000e800: 7761 7920 6f66 2064 652d 7265 6475 6e64  way of de-redund
-0000e810: 616e 6379 2e20 4469 6666 6572 656e 7420  ancy. Different 
-0000e820: 6d65 7468 6f64 7320 6f66 2064 652d 7265  methods of de-re
-0000e830: 6475 6e64 616e 6379 2077 696c 6c20 6861  dundancy will ha
-0000e840: 7665 2069 6d70 6163 7420 6f6e 2074 6865  ve impact on the
-0000e850: 2064 6574 6563 7469 6f6e 206f 6620 5747   detection of WG
-0000e860: 4420 7369 676e 616c 732c 2077 6869 6368  D signals, which
-0000e870: 2068 6173 2062 6565 6e20 696e 7665 7374   has been invest
-0000e880: 6967 6174 6564 2069 6e20 7468 6973 205b  igated in this [
-0000e890: 6c69 7465 7261 7475 7265 5d28 6874 7470  literature](http
-0000e8a0: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
-0000e8b0: 3039 332f 6762 652f 6576 7932 3030 292e  093/gbe/evy200).
-0000e8c0: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
-0000e8d0: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
-0000e8e0: 6f20 7061 7261 6c6c 656c 697a 6520 7468  o parallelize th
-0000e8f0: 6520 616e 616c 7973 6973 2066 6f72 2065  e analysis for e
-0000e900: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
-0000e910: 616e 6420 7468 7573 2073 7567 6765 7374  and thus suggest
-0000e920: 6564 2074 6f20 6265 2073 6574 2061 7320  ed to be set as 
-0000e930: 6d75 6368 2061 7320 7468 6520 6e75 6d62  much as the numb
-0000e940: 6572 206f 6620 6765 6e65 2066 616d 696c  er of gene famil
-0000e950: 6965 732e 0a0a 2a2a 4120 7375 6767 6573  ies...**A sugges
-0000e960: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
-0000e970: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
-0000e980: 2061 7320 6265 6c6f 772a 2a0a 0a60 6060   as below**..```
-0000e990: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
-0000e9a0: 7320 7365 7175 656e 6365 0a60 6060 0a0a  s sequence.```..
-0000e9b0: 2a2a 5468 6520 636f 6e73 7472 7563 7469  **The constructi
-0000e9c0: 6f6e 206f 6620 6f72 7468 6f6c 6f67 6f75  on of orthologou
-0000e9d0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
-0000e9e0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-0000e9f0: 6f6e 2a2a 0a60 6060 0a77 6764 206b 7364  on**.```.wgd ksd
-0000ea00: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000ea10: 6365 3120 7365 7175 656e 6365 3220 282d  ce1 sequence2 (-
-0000ea20: 2d72 6577 6569 6768 7429 0a60 6060 0a0a  -reweight).```..
-0000ea30: 4672 6f6d 2070 6172 616c 6f67 6f75 7320  From paralogous 
-0000ea40: 746f 206f 7274 686f 6c6f 676f 7573 202a  to orthologous *
-0000ea50: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000ea60: 6765 2064 6973 7472 6962 7574 696f 6e2c  ge distribution,
-0000ea70: 2075 7365 7273 206f 6e6c 7920 6e65 6564   users only need
-0000ea80: 2074 6f20 7072 6f76 6964 6520 6d6f 7265   to provide more
-0000ea90: 2063 6473 2066 696c 6573 2e20 4e6f 7465   cds files. Note
-0000eaa0: 2074 6861 7420 7769 7468 206f 7274 686f   that with ortho
-0000eab0: 6c6f 676f 7573 2067 656e 6520 6661 6d69  logous gene fami
-0000eac0: 6c69 6573 2074 6865 2077 6569 6768 7469  lies the weighti
-0000ead0: 6e67 206d 6574 686f 6420 6361 6e20 6265  ng method can be
-0000eae0: 2073 6574 2074 6f20 6265 2063 616c 6375   set to be calcu
-0000eaf0: 6c61 7465 6420 7065 7220 7370 6563 6965  lated per specie
-0000eb00: 7320 7061 6972 2069 6e73 7465 6164 206f  s pair instead o
-0000eb10: 6620 636f 6e73 6964 6572 696e 6720 7468  f considering th
-0000eb20: 6520 7768 6f6c 6520 6661 6d69 6c79 2062  e whole family b
-0000eb30: 6563 6175 7365 2077 6865 6e20 706c 6f74  ecause when plot
-0000eb40: 7469 6e67 206f 7274 686f 6c6f 676f 7573  ting orthologous
-0000eb50: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000eb60: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000eb70: 6e20 6265 7477 6565 6e20 7477 6f20 7370  n between two sp
-0000eb80: 6563 6965 7320 7468 6520 7765 6967 6874  ecies the weight
-0000eb90: 2063 616c 6375 6c61 7465 6420 6672 6f6d   calculated from
-0000eba0: 2074 6869 7320 7370 6563 6966 6963 2073   this specific s
-0000ebb0: 7065 6369 6573 2070 6169 7220 7368 6f75  pecies pair shou
-0000ebc0: 6c64 2062 6520 636f 6e73 6572 7665 6420  ld be conserved 
-0000ebd0: 7768 696c 6520 7468 6520 6f6e 6520 6361  while the one ca
-0000ebe0: 6c63 756c 6174 6564 2066 726f 6d20 7468  lculated from th
-0000ebf0: 6520 7768 6f6c 6520 6661 6d69 6c79 2077  e whole family w
-0000ec00: 696c 6c20 7661 7279 2077 6974 6820 7468  ill vary with th
-0000ec10: 6520 6e75 6d62 6572 206f 6620 7370 6563  e number of spec
-0000ec20: 6965 732e 2054 6f20 696e 6974 6961 7465  ies. To initiate
-0000ec30: 2074 6865 2077 6569 6768 7469 6e67 2070   the weighting p
-0000ec40: 6572 2073 7065 6369 6573 2070 6169 722c  er species pair,
-0000ec50: 2074 6865 206f 7074 696f 6e20 602d 2d72   the option `--r
-0000ec60: 6577 6569 6768 7460 2063 616e 2062 6520  eweight` can be 
-0000ec70: 7365 742e 0a0a 2a2a 4120 7375 6767 6573  set...**A sugges
-0000ec80: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
-0000ec90: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
-0000eca0: 2061 7320 6265 6c6f 772a 2a0a 0a60 6060   as below**..```
-0000ecb0: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
-0000ecc0: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000ecd0: 656e 6365 320a 6060 600a 0a2a 2a54 6865  ence2.```..**The
-0000ece0: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-0000ecf0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000ed00: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000ed10: 6e20 7769 7468 2072 6174 6520 636f 7272  n with rate corr
-0000ed20: 6563 7469 6f6e 2a2a 0a60 6060 0a77 6764  ection**.```.wgd
-0000ed30: 206b 7364 2066 616d 696c 6965 7320 7365   ksd families se
-0000ed40: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000ed50: 3220 7365 7175 656e 6365 3320 2d2d 666f  2 sequence3 --fo
-0000ed60: 6375 7332 616c 6c20 7365 7175 656e 6365  cus2all sequence
-0000ed70: 3120 2d73 7020 7370 6461 7461 0a60 6060  1 -sp spdata.```
-0000ed80: 0a0a 496e 7370 6972 6564 2062 7920 7468  ..Inspired by th
-0000ed90: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
-0000eda0: 6e20 616c 676f 7269 7468 6d20 696e 205b  n algorithm in [
-0000edb0: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
-0000edc0: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
-0000edd0: 5053 422f 6b73 7261 7465 7329 2c20 7765  PSB/ksrates), we
-0000ede0: 2069 6d70 6c65 6d65 6e74 6564 2074 6865   implemented the
-0000edf0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-0000ee00: 2061 6e61 6c79 7369 7320 616c 736f 2069   analysis also i
-0000ee10: 6e20 6077 6764 2076 3260 2c20 7768 6963  n `wgd v2`, whic
-0000ee20: 6820 6973 206d 6f73 746c 7920 7468 6520  h is mostly the 
-0000ee30: 7361 6d65 2061 7320 5b6b 7372 6174 6573  same as [ksrates
-0000ee40: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-0000ee50: 2e63 6f6d 2f56 4942 2d50 5342 2f6b 7372  .com/VIB-PSB/ksr
-0000ee60: 6174 6573 2920 6275 7420 6469 6666 6572  ates) but differ
-0000ee70: 7320 696e 2074 6865 2063 616c 6375 6c61  s in the calcula
-0000ee80: 7469 6f6e 206f 6620 7468 6520 7374 616e  tion of the stan
-0000ee90: 6461 7264 2064 6576 6961 7469 6f6e 206f  dard deviation o
-0000eea0: 6620 7265 7363 616c 6564 202a 4b2a 3c73  f rescaled *K*<s
-0000eeb0: 7562 3e53 3c2f 7375 623e 2061 6765 732e  ub>S</sub> ages.
-0000eec0: 2054 6f20 7065 7266 6f72 6d20 7468 6520   To perform the 
-0000eed0: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-0000eee0: 616e 616c 7973 6973 2c20 7573 6572 7320  analysis, users 
-0000eef0: 6361 6e20 7573 6520 626f 7468 2074 6865  can use both the
-0000ef00: 2060 7767 6420 6b73 6460 2061 6e64 2060   `wgd ksd` and `
-0000ef10: 7767 6420 7669 7a60 2070 726f 6772 616d  wgd viz` program
-0000ef20: 2e20 466f 7220 7468 6520 6077 6764 206b  . For the `wgd k
-0000ef30: 7364 6020 7072 6f67 7261 6d2c 2075 7365  sd` program, use
-0000ef40: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
-0000ef50: 6465 2061 2073 7065 6369 6573 2074 7265  de a species tre
-0000ef60: 6520 7669 6120 7468 6520 6f70 7469 6f6e  e via the option
-0000ef70: 2060 2d2d 7370 6563 6965 7374 7265 6560   `--speciestree`
-0000ef80: 206f 6e20 7768 6963 6820 7261 7465 2063   on which rate c
-0000ef90: 6f72 7265 6374 696f 6e20 6361 6e20 6265  orrection can be
-0000efa0: 2063 6f6e 6475 6374 6564 2e20 4e6f 7465   conducted. Note
-0000efb0: 2074 6861 7420 756e 6e65 6365 7373 6172   that unnecessar
-0000efc0: 7920 6272 6163 6b65 7473 206d 6967 6874  y brackets might
-0000efd0: 206c 6561 6420 746f 2075 6e65 7870 6563   lead to unexpec
-0000efe0: 7465 6420 6572 726f 7273 2c20 666f 7220  ted errors, for 
-0000eff0: 696e 7374 616e 6365 2061 2074 7265 6520  instance a tree 
-0000f000: 6028 412c 2842 2c43 2929 3b60 2073 686f  `(A,(B,C));` sho
-0000f010: 756c 6420 6e6f 7420 6265 2072 6570 7265  uld not be repre
-0000f020: 7365 6e74 6564 2061 7320 6028 412c 2828  sented as `(A,((
-0000f030: 422c 4329 2929 3b60 2e20 5468 6520 7365  B,C)));`. The se
-0000f040: 7420 6f66 2073 7065 6369 6573 2070 6169  t of species pai
-0000f050: 7273 2074 6f20 6265 2073 686f 776e 2069  rs to be shown i
-0000f060: 7320 666c 6578 6962 6c65 2074 6861 7420  s flexible that 
-0000f070: 7468 6520 6d6f 7374 2063 6f6e 7665 6e69  the most conveni
-0000f080: 656e 7420 6f70 7469 6f6e 2069 7320 602d  ent option is `-
-0000f090: 2d66 6f63 7573 3261 6c6c 6020 7768 6963  -focus2all` whic
-0000f0a0: 6820 7369 6d70 6c79 2073 686f 7773 2061  h simply shows a
-0000f0b0: 6c6c 2074 6865 2070 6f73 7369 626c 6520  ll the possible 
-0000f0c0: 666f 6361 6c2d 7369 7374 6572 2073 7065  focal-sister spe
-0000f0d0: 6369 6573 2070 6169 7273 2c20 6f72 2075  cies pairs, or u
-0000f0e0: 7365 7273 2063 616e 206d 616e 7561 6c6c  sers can manuall
-0000f0f0: 7920 7365 7420 7468 6520 7370 6563 6965  y set the specie
-0000f100: 7320 7061 6972 7320 7669 6120 7468 6520  s pairs via the 
-0000f110: 6f70 7469 6f6e 2060 2d2d 7370 6169 7260  option `--spair`
-0000f120: 2e20 4e6f 7465 2074 6861 7420 6966 2074  . Note that if t
-0000f130: 6865 2073 7065 6369 6573 2070 6169 7273  he species pairs
-0000f140: 2077 6572 6520 6d61 6e75 616c 6c79 2073   were manually s
-0000f150: 6574 2c20 6974 2077 6f75 6c64 2062 6520  et, it would be 
-0000f160: 6e65 6564 6564 2074 6f20 7365 7420 7468  needed to set th
-0000f170: 6520 6f70 7469 6f6e 2060 2d2d 636c 6173  e option `--clas
-0000f180: 7369 6360 2e20 5468 6520 6f70 7469 6f6e  sic`. The option
-0000f190: 2060 6f6e 6c79 726f 6f74 6f75 7460 2063   `onlyrootout` c
-0000f1a0: 616e 2062 6520 7365 7420 746f 206f 6e6c  an be set to onl
-0000f1b0: 7920 636f 6e73 6964 6572 206f 7574 6772  y consider outgr
-0000f1c0: 6f75 7020 6174 2074 6865 2072 6f6f 742c  oup at the root,
-0000f1d0: 2069 6e73 7465 6164 206f 6620 616c 6c20   instead of all 
-0000f1e0: 7468 6520 706f 7373 6962 6c65 206f 7574  the possible out
-0000f1f0: 6772 6f75 7073 2070 6572 2066 6f63 616c  groups per focal
-0000f200: 2d73 6973 7465 7220 7370 6563 6965 7320  -sister species 
-0000f210: 7061 6972 2c20 7768 6963 6820 6861 7320  pair, which has 
-0000f220: 696d 7061 6374 206f 6e20 7468 6520 6669  impact on the fi
-0000f230: 6e61 6c20 636f 7272 6563 7465 6420 2a4b  nal corrected *K
-0000f240: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000f250: 6573 2e20 536f 6d65 206f 7468 6572 206f  es. Some other o
-0000f260: 7074 696f 6e73 2077 6869 6368 2068 6176  ptions which hav
-0000f270: 6520 6e6f 2069 6d70 6163 7420 6f6e 2074  e no impact on t
-0000f280: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
-0000f290: 6f6e 2062 7574 2061 6464 206d 6f72 6520  on but add more 
-0000f2a0: 6c61 7965 7273 206f 7220 6368 616e 6765  layers or change
-0000f2b0: 2074 6865 2061 7070 6561 7261 6e63 6520   the appearance 
-0000f2c0: 6f6e 2074 6865 206d 6978 6564 202a 4b2a  on the mixed *K*
-0000f2d0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000f2e0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
-0000f2f0: 6e63 6c75 6465 2060 706c 6f74 6170 676d  nclude `plotapgm
-0000f300: 6d60 2061 6e64 2060 706c 6f74 656c 6d6d  m` and `plotelmm
-0000f310: 6020 6574 632e 2054 6865 206f 7074 696f  ` etc. The optio
-0000f320: 6e20 6070 6c6f 7461 7067 6d6d 6020 6361  n `plotapgmm` ca
-0000f330: 6e20 6265 2073 6574 2074 6f20 6361 6c6c  n be set to call
-0000f340: 2074 6865 2047 4d4d 2061 6e61 6c79 7369   the GMM analysi
-0000f350: 7320 7570 6f6e 2074 6865 2061 6e63 686f  s upon the ancho
-0000f360: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-0000f370: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000f380: 6962 7574 696f 6e20 616e 6420 706c 6f74  ibution and plot
-0000f390: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
-0000f3a0: 7265 7375 6c74 2075 706f 6e20 7468 6520  result upon the 
-0000f3b0: 6d69 7865 6420 2a4b 2a3c 7375 623e 533c  mixed *K*<sub>S<
-0000f3c0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-0000f3d0: 6275 7469 6f6e 2c20 7768 6963 6820 6861  bution, which ha
-0000f3e0: 7320 746f 2062 6520 636f 2d73 6574 2077  s to be co-set w
-0000f3f0: 6974 6820 7468 6520 6f70 7469 6f6e 2060  ith the option `
-0000f400: 2d2d 616e 6368 6f72 706f 696e 7473 6020  --anchorpoints` 
-0000f410: 7072 6f76 6964 696e 6720 7468 6520 616e  providing the an
-0000f420: 6368 6f72 2070 6169 7273 2069 6e66 6f72  chor pairs infor
-0000f430: 6d61 7469 6f6e 2e20 5468 6520 6f70 7469  mation. The opti
-0000f440: 6f6e 2060 2d2d 706c 6f74 656c 6d6d 6020  on `--plotelmm` 
-0000f450: 6361 6e20 6265 2073 6574 2074 6f20 6361  can be set to ca
-0000f460: 6c6c 2074 6865 2045 4c4d 4d20 616e 616c  ll the ELMM anal
-0000f470: 7973 6973 2075 706f 6e20 7468 6520 7061  ysis upon the pa
-0000f480: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-0000f490: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000f4a0: 7472 6962 7574 696f 6e2e 0a0a 5468 6572  tribution...Ther
-0000f4b0: 6520 6172 6520 3231 2063 6f6c 756d 6e73  e are 21 columns
-0000f4c0: 2069 6e20 7468 6520 7265 7375 6c74 2060   in the result `
-0000f4d0: 2a6b 732e 7473 7660 2066 696c 6520 6265  *ks.tsv` file be
-0000f4e0: 7369 6465 7320 7468 6520 696e 6465 7820  sides the index 
-0000f4f0: 636f 6c75 6d6e 7320 6070 6169 7260 2061  columns `pair` a
-0000f500: 7320 7468 6520 756e 6971 7565 2069 6465  s the unique ide
-0000f510: 6e74 6966 6965 7220 666f 7220 6561 6368  ntifier for each
-0000f520: 2067 656e 6520 7061 6972 2e20 5468 6520   gene pair. The 
-0000f530: 604e 602c 2060 5360 2c20 6064 4e60 2c20  `N`, `S`, `dN`, 
-0000f540: 6064 4e2f 6453 602c 2060 6453 602c 2060  `dN/dS`, `dS`, `
-0000f550: 6c60 2061 6e64 2060 7460 2061 7265 2066  l` and `t` are f
-0000f560: 726f 6d20 7468 6520 636f 6465 6d6c 2072  rom the codeml r
-0000f570: 6573 756c 7473 2c20 7265 7072 6573 656e  esults, represen
-0000f580: 7469 6e67 2074 6865 204e 2065 7374 696d  ting the N estim
-0000f590: 6174 652c 2074 6865 2053 2065 7374 696d  ate, the S estim
-0000f5a0: 6174 652c 2074 6865 2064 4e20 6573 7469  ate, the dN esti
-0000f5b0: 6d61 7465 2c20 7468 6520 644e 2f64 5320  mate, the dN/dS 
-0000f5c0: 286f 6d65 6761 2920 6573 7469 6d61 7465  (omega) estimate
-0000f5d0: 2c20 7468 6520 6453 2065 7374 696d 6174  , the dS estimat
-0000f5e0: 652c 2074 6865 206c 6f67 2d6c 696b 656c  e, the log-likel
-0000f5f0: 6968 6f6f 6420 616e 6420 7468 6520 7420  ihood and the t 
-0000f600: 6573 7469 6d61 7465 2c20 7265 7370 6563  estimate, respec
-0000f610: 7469 7665 6c79 2e20 5468 6520 6061 6c69  tively. The `ali
-0000f620: 676e 6d65 6e74 636f 7665 7261 6765 602c  gnmentcoverage`,
-0000f630: 2060 616c 6967 6e6d 656e 7469 6465 6e74   `alignmentident
-0000f640: 6974 7960 2061 6e64 2060 616c 6967 6e6d  ity` and `alignm
-0000f650: 656e 746c 656e 6774 6860 2061 7265 2074  entlength` are t
-0000f660: 6865 2069 6e66 6f72 6d61 7469 6f6e 2070  he information p
-0000f670: 6572 7461 696e 696e 6720 746f 2074 6865  ertaining to the
-0000f680: 2061 6c69 676e 6d65 6e74 2066 6f72 2065   alignment for e
-0000f690: 6163 6820 6661 6d69 6c79 2c20 7265 7072  ach family, repr
-0000f6a0: 6573 656e 7469 6e67 2074 6865 2072 6174  esenting the rat
-0000f6b0: 696f 206f 6620 7468 6520 7374 7269 7070  io of the stripp
-0000f6c0: 6564 2061 6c69 676e 6d65 6e74 206c 656e  ed alignment len
-0000f6d0: 6774 6820 636f 6d70 6172 6564 2074 6f20  gth compared to 
-0000f6e0: 7468 6520 6675 6c6c 2061 6c69 676e 6d65  the full alignme
-0000f6f0: 6e74 206c 656e 6774 682c 2074 6865 2072  nt length, the r
-0000f700: 6174 696f 206f 6620 636f 6c75 6d6e 7320  atio of columns 
-0000f710: 7769 7468 2069 6465 6e74 6963 616c 206e  with identical n
-0000f720: 7563 6c65 6f74 6964 6573 2063 6f6d 7061  ucleotides compa
-0000f730: 7265 6420 746f 2074 6865 206f 7665 7261  red to the overa
-0000f740: 6c6c 2063 6f6c 756d 6e73 206f 6620 7468  ll columns of th
-0000f750: 6520 7374 7269 7070 6564 2061 6c69 676e  e stripped align
-0000f760: 6d65 6e74 2c20 616e 6420 7468 6520 6c65  ment, and the le
-0000f770: 6e67 7468 206f 6620 7468 6520 6675 6c6c  ngth of the full
-0000f780: 2061 6c69 676e 6d65 6e74 2c20 7265 7370   alignment, resp
-0000f790: 6563 7469 7665 6c79 2e0a 2323 2320 7767  ectively..### wg
-0000f7a0: 6420 6d69 780a 0a2a 2a54 6865 206d 6978  d mix..**The mix
-0000f7b0: 7475 7265 206d 6f64 656c 2063 6c75 7374  ture model clust
-0000f7c0: 6572 696e 6720 616e 616c 7973 6973 206f  ering analysis o
-0000f7d0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
-0000f7e0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-0000f7f0: 6f6e 2a2a 0a60 6060 0a77 6764 206d 6978  on**.```.wgd mix
-0000f800: 206b 7364 6174 6120 282d 2d6e 5f69 6e69   ksdata (--n_ini
-0000f810: 7420 3230 3020 2d2d 6d61 785f 6974 6572  t 200 --max_iter
-0000f820: 2032 3030 202d 2d6b 735f 7261 6e67 6520   200 --ks_range 
-0000f830: 3020 3520 2d2d 6669 6c74 6572 7320 3330  0 5 --filters 30
-0000f840: 3020 2d2d 6269 6e73 2035 3020 2d2d 636f  0 --bins 50 --co
-0000f850: 6d70 6f6e 656e 7473 2031 2034 202d 2d67  mponents 1 4 --g
-0000f860: 616d 6d61 2030 2e30 3031 290a 6060 600a  amma 0.001).```.
-0000f870: 0a54 6869 7320 7061 7274 206f 6620 4761  .This part of Ga
-0000f880: 7573 7369 616e 206d 6978 7475 7265 206d  ussian mixture m
-0000f890: 6f64 656c 696e 6720 2847 4d4d 2920 616e  odeling (GMM) an
-0000f8a0: 616c 7973 6973 2069 7320 696e 6865 7269  alysis is inheri
-0000f8b0: 7465 6420 6672 6f6d 2074 6865 206f 7269  ted from the ori
-0000f8c0: 6769 6e61 6c20 6077 6764 6020 7072 6f67  ginal `wgd` prog
-0000f8d0: 7261 6d2c 2062 7574 2077 7269 7465 7320  ram, but writes 
-0000f8e0: 6164 6469 7469 6f6e 616c 6c79 2074 6865  additionally the
-0000f8f0: 2070 726f 6261 6269 6c69 7479 206f 6620   probability of 
-0000f900: 6561 6368 202a 4b2a 3c73 7562 3e53 3c2f  each *K*<sub>S</
-0000f910: 7375 623e 2076 616c 7565 2069 6e74 6f20  sub> value into 
-0000f920: 7468 6520 6669 6e61 6c20 6461 7461 6672  the final datafr
-0000f930: 616d 652e 2042 6173 6963 616c 6c79 2c20  ame. Basically, 
-0000f940: 7573 6572 7320 6e65 6564 2074 6f20 7072  users need to pr
-0000f950: 6f76 6964 6520 7769 7468 2061 2028 6e6f  ovide with a (no
-0000f960: 726d 616c 6c79 2066 726f 6d20 7768 6f6c  rmally from whol
-0000f970: 652d 7061 7261 6e6f 6d65 206f 7220 616e  e-paranome or an
-0000f980: 6368 6f72 2d70 6169 7273 2920 2a4b 2a3c  chor-pairs) *K*<
-0000f990: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-0000f9a0: 6669 6c65 2061 6e64 2074 6865 2047 4d4d  file and the GMM
-0000f9b0: 2061 6e61 6c79 7369 7320 7769 6c6c 2062   analysis will b
-0000f9c0: 6520 636f 6e64 7563 7465 6420 7570 6f6e  e conducted upon
-0000f9d0: 2074 6865 2064 6174 6166 696c 652e 2053   the datafile. S
-0000f9e0: 6f6d 6520 7061 7261 6d65 7465 7273 2063  ome parameters c
-0000f9f0: 616e 2061 6666 6563 7420 7468 6520 7265  an affect the re
-0000fa00: 7375 6c74 732c 2069 6e63 6c75 6469 6e67  sults, including
-0000fa10: 2060 2d2d 6e5f 696e 6974 602c 2077 6869   `--n_init`, whi
-0000fa20: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
-0000fa30: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
-0000fa40: 6974 6961 6c69 7a61 7469 6f6e 7320 2864  itializations (d
-0000fa50: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
-0000fa60: 6d61 785f 6974 6572 602c 2077 6869 6368  max_iter`, which
-0000fa70: 2073 6574 7320 7468 6520 6d61 7869 6d75   sets the maximu
-0000fa80: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
-0000fa90: 6174 696f 6e73 2028 6465 6661 756c 7420  ations (default 
-0000faa0: 3230 3029 2c20 602d 2d6d 6574 686f 6460  200), `--method`
-0000fab0: 2c20 7768 6963 6820 6465 7465 726d 696e  , which determin
-0000fac0: 6573 2077 6869 6368 2063 6c75 7374 6572  es which cluster
-0000fad0: 696e 6720 6d65 7468 6f64 2074 6f20 7573  ing method to us
-0000fae0: 6520 2864 6566 6175 6c74 2067 6d6d 292c  e (default gmm),
-0000faf0: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
-0000fb00: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
-0000fb10: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-0000fb20: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
-0000fb30: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
-0000fb40: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
-0000fb50: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
-0000fb60: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-0000fb70: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-0000fb80: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
-0000fb90: 3120 3429 2c20 616e 6420 7468 6520 6461  1 4), and the da
-0000fba0: 7461 2066 696c 7465 7269 6e67 2070 6172  ta filtering par
-0000fbb0: 616d 6574 6572 7320 602d 2d66 696c 7465  ameters `--filte
-0000fbc0: 7273 6020 7768 6963 6820 6669 6c74 6572  rs` which filter
-0000fbd0: 7320 6461 7461 2062 6173 6564 206f 6e20  s data based on 
-0000fbe0: 616c 6967 6e6d 656e 7420 6c65 6e67 7468  alignment length
-0000fbf0: 2c20 602d 2d6b 735f 7261 6e67 6560 2077  , `--ks_range` w
-0000fc00: 6869 6368 2066 696c 7465 7273 2064 6174  hich filters dat
-0000fc10: 6120 6261 7365 6420 6f6e 202a 4b2a 3c73  a based on *K*<s
-0000fc20: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-0000fc30: 7320 616e 6420 7468 6520 7061 7261 6d65  s and the parame
-0000fc40: 7465 7220 602d 2d62 696e 7360 2077 6869  ter `--bins` whi
-0000fc50: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
-0000fc60: 6572 206f 6620 6269 6e73 2069 6e20 2a4b  er of bins in *K
-0000fc70: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-0000fc80: 7374 7269 6275 7469 6f6e 2028 6465 6661  stribution (defa
-0000fc90: 756c 7420 3530 292e 0a0a 2a2a 4120 7375  ult 50)...**A su
-0000fca0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-0000fcb0: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-0000fcc0: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-0000fcd0: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-0000fce0: 6d69 7820 6b73 6461 7461 0a60 6060 0a0a  mix ksdata.```..
-0000fcf0: 2323 2320 7767 6420 7065 616b 0a0a 2a2a  ### wgd peak..**
-0000fd00: 5468 6520 7365 6172 6368 206f 6620 6372  The search of cr
-0000fd10: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
-0000fd20: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
-0000fd30: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
-0000fd40: 2a2a 0a60 6060 0a77 6764 2070 6561 6b20  **.```.wgd peak 
-0000fd50: 6b73 6461 7461 202d 6170 2061 7064 6174  ksdata -ap apdat
-0000fd60: 6120 2d73 6d20 736d 6461 7461 202d 6c65  a -sm smdata -le
-0000fd70: 206c 6564 6174 6120 2d6d 7020 6d70 6461   ledata -mp mpda
-0000fd80: 7461 202d 2d68 6575 7269 7374 6963 2028  ta --heuristic (
-0000fd90: 2d2d 616c 6967 6e66 696c 7465 7220 302e  --alignfilter 0.
-0000fda0: 3020 3020 302e 3020 2d2d 6b73 7261 6e67  0 0 0.0 --ksrang
-0000fdb0: 6520 3020 3520 2d2d 6269 6e5f 7769 6474  e 0 5 --bin_widt
-0000fdc0: 6820 302e 3120 2d2d 6775 6964 6520 7365  h 0.1 --guide se
-0000fdd0: 676d 656e 7420 2d2d 7072 6f6d 696e 656e  gment --prominen
-0000fde0: 6365 5f63 7574 6f66 6620 302e 3120 2d2d  ce_cutoff 0.1 --
-0000fdf0: 7265 6c5f 6865 6967 6874 2030 2e34 202d  rel_height 0.4 -
-0000fe00: 2d63 6920 3935 202d 2d68 6472 2039 3520  -ci 95 --hdr 95 
-0000fe10: 2d2d 6b73 6375 746f 6666 2035 290a 6060  --kscutoff 5).``
-0000fe20: 600a 0a41 7320 6d65 6e74 696f 6e65 6420  `..As mentioned 
-0000fe30: 7072 6576 696f 7573 6c79 2c20 6120 6865  previously, a he
-0000fe40: 7572 6973 7469 6320 6d65 7468 6f64 2061  uristic method a
-0000fe50: 6e64 2061 2063 6f6c 6c69 6e65 6172 2073  nd a collinear s
-0000fe60: 6567 6d65 6e74 732d 6775 6964 6564 2061  egments-guided a
-0000fe70: 6e63 686f 7220 7061 6972 2063 6c75 7374  nchor pair clust
-0000fe80: 6572 696e 6720 666f 7220 7468 6520 7365  ering for the se
-0000fe90: 6172 6368 206f 6620 6372 6564 6961 626c  arch of crediabl
-0000fea0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000feb0: 3e20 7261 6e67 6520 7573 6564 2069 6e20  > range used in 
-0000fec0: 5747 4420 6461 7469 6e67 2061 7265 2069  WGD dating are i
-0000fed0: 6d70 6c65 6d65 6e74 6564 2069 6e20 6077  mplemented in `w
-0000fee0: 6764 2076 3260 2e20 5573 6572 7320 6e65  gd v2`. Users ne
-0000fef0: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
-0000ff00: 6520 616e 6368 6f72 2070 6f69 6e74 732c  e anchor points,
-0000ff10: 2073 6567 6d65 6e74 732c 206c 6973 7473   segments, lists
-0000ff20: 6567 6d65 6e74 732c 206d 756c 7469 706c  egments, multipl
-0000ff30: 6963 6f6e 7061 6972 7320 6461 7461 6669  iconpairs datafi
-0000ff40: 6c65 2066 726f 6d20 6069 2d61 6468 6f72  le from `i-adhor
-0000ff50: 6560 2074 6f20 6163 6869 6576 6520 7468  e` to achieve th
-0000ff60: 6520 636c 7573 7465 7269 6e67 2066 756e  e clustering fun
-0000ff70: 6374 696f 6e2e 2053 6f6d 6520 7061 7261  ction. Some para
-0000ff80: 6d65 7465 7273 2074 6861 7420 6361 6e20  meters that can 
-0000ff90: 696d 7061 6374 2074 6865 2072 6573 756c  impact the resul
-0000ffa0: 7473 2069 6e63 6c75 6465 2060 2d2d 616c  ts include `--al
-0000ffb0: 6967 6e66 696c 7465 7260 2c20 7768 6963  ignfilter`, whic
-0000ffc0: 6820 6669 6c74 6572 7320 7468 6520 6461  h filters the da
-0000ffd0: 7461 2062 6173 6564 206f 6e20 616c 6967  ta based on alig
-0000ffe0: 6e6d 656e 7420 6964 656e 7469 7479 2c20  nment identity, 
-0000fff0: 6c65 6e67 7468 2061 6e64 2063 6f76 6572  length and cover
-00010000: 6167 652c 2060 2d2d 6b73 7261 6e67 6560  age, `--ksrange`
-00010010: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
-00010020: 2072 616e 6765 206f 6620 4b73 2074 6f20   range of Ks to 
-00010030: 6265 2061 6e61 6c79 7a65 642c 2060 2d2d  be analyzed, `--
-00010040: 6269 6e5f 7769 6474 6860 2c20 7768 6963  bin_width`, whic
-00010050: 6820 7365 7473 2074 6865 2062 616e 6477  h sets the bandw
-00010060: 6964 7468 206f 6620 2a4b 2a3c 7375 623e  idth of *K*<sub>
-00010070: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
-00010080: 7469 6f6e 2c20 602d 2d77 6569 6768 7473  tion, `--weights
-00010090: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
-000100a0: 6564 6020 7768 6963 6820 6465 7465 726d  ed` which determ
-000100b0: 696e 6573 2077 6865 7468 6572 2074 6f20  ines whether to 
-000100c0: 696e 636c 7564 6520 2a4b 2a3c 7375 623e  include *K*<sub>
-000100d0: 533c 2f73 7562 3e20 6f75 746c 6965 7273  S</sub> outliers
-000100e0: 2028 7768 6f73 6520 7661 6c75 6520 6973   (whose value is
-000100f0: 206f 7665 7220 3529 2069 6e20 616e 616c   over 5) in anal
-00010100: 7973 6973 2c20 602d 2d6d 6574 686f 6460  ysis, `--method`
-00010110: 2077 6869 6368 2064 6574 6572 6d69 6e65   which determine
-00010120: 7320 7768 6963 6820 636c 7573 7465 7269  s which clusteri
-00010130: 6e67 206d 6574 686f 6420 746f 2075 7365  ng method to use
-00010140: 2028 6465 6661 756c 7420 676d 6d29 2c20   (default gmm), 
-00010150: 602d 2d73 6565 6460 2077 6869 6368 2073  `--seed` which s
-00010160: 6574 7320 7468 6520 7261 6e64 6f6d 2073  ets the random s
-00010170: 6565 6420 6769 7665 6e20 746f 2069 6e69  eed given to ini
-00010180: 7469 616c 697a 6174 696f 6e20 2864 6566  tialization (def
-00010190: 6175 6c74 2032 3335 3238 3930 292c 2060  ault 2352890), `
-000101a0: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
-000101b0: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-000101c0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-000101d0: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
-000101e0: 6175 6c74 2032 3030 292c 2060 2d2d 656d  ault 200), `--em
-000101f0: 5f69 7465 7260 2c20 7768 6963 6820 7365  _iter`, which se
-00010200: 7473 2074 6865 206d 6178 696d 756d 206e  ts the maximum n
-00010210: 756d 6265 7220 6f66 2069 7465 7261 7469  umber of iterati
-00010220: 6f6e 7320 2864 6566 6175 6c74 2032 3030  ons (default 200
-00010230: 292c 2060 2d2d 6761 6d6d 6160 2c20 7768  ), `--gamma`, wh
-00010240: 6963 6820 7365 7473 2074 6865 2067 616d  ich sets the gam
-00010250: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
-00010260: 2074 6865 2062 676d 6d20 6d6f 6465 6c20   the bgmm model 
-00010270: 2864 6566 6175 6c74 2030 2e30 3031 292c  (default 0.001),
-00010280: 2060 2d2d 636f 6d70 6f6e 656e 7473 602c   `--components`,
-00010290: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
-000102a0: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
-000102b0: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
-000102c0: 7320 746f 2066 6974 2028 6465 6661 756c  s to fit (defaul
-000102d0: 7420 3120 3429 2c20 602d 2d77 6569 6768  t 1 4), `--weigh
-000102e0: 7465 6460 2077 6869 6368 2064 6574 6572  ted` which deter
-000102f0: 6d69 6e65 7320 7768 6574 6865 7220 746f  mines whether to
-00010300: 2075 7365 206e 6f64 652d 7765 6967 6874   use node-weight
-00010310: 6564 206d 6574 686f 6420 666f 7220 6465  ed method for de
-00010320: 2d72 6564 756e 6461 6e63 792c 2060 2d2d  -redundancy, `--
-00010330: 6775 6964 6560 2077 6869 6368 2064 6574  guide` which det
-00010340: 6572 6d69 6e65 7320 7768 6963 6820 7265  ermines which re
-00010350: 6769 6d65 2072 6573 6964 696e 6720 616e  gime residing an
-00010360: 6368 6f72 7320 746f 2062 6520 7573 6564  chors to be used
-00010370: 2028 6465 6661 756c 7420 7365 676d 656e   (default segmen
-00010380: 7429 2c20 602d 2d70 726f 6d69 6e65 6e63  t), `--prominenc
-00010390: 655f 6375 746f 6666 6020 7768 6963 6820  e_cutoff` which 
-000103a0: 7365 7473 2074 6865 2070 726f 6d69 6e65  sets the promine
-000103b0: 6e63 6520 6375 746f 6666 206f 6620 6163  nce cutoff of ac
-000103c0: 6365 7074 6162 6c65 2070 6561 6b73 2069  ceptable peaks i
-000103d0: 6e20 7065 616b 2066 696e 6469 6e67 2070  n peak finding p
-000103e0: 726f 6365 7373 2c20 602d 2d72 656c 5f68  rocess, `--rel_h
-000103f0: 6569 6768 7460 2077 6869 6368 2073 6574  eight` which set
-00010400: 7320 7468 6520 7265 6c61 7469 7665 2068  s the relative h
-00010410: 6569 6768 7420 6174 2077 6869 6368 2074  eight at which t
-00010420: 6865 2070 6561 6b20 7769 6474 6820 6973  he peak width is
-00010430: 206d 6561 7375 7265 642c 2060 2d2d 6b73   measured, `--ks
-00010440: 746f 6461 7465 6020 7768 6963 6820 6d61  todate` which ma
-00010450: 6e75 616c 6c79 2073 6574 7320 7468 6520  nually sets the 
-00010460: 7261 6e67 6520 6f66 202a 4b2a 3c73 7562  range of *K*<sub
-00010470: 3e53 3c2f 7375 623e 2074 6f20 6265 2064  >S</sub> to be d
-00010480: 6174 6564 2069 6e20 6865 7572 6973 7469  ated in heuristi
-00010490: 6320 7365 6172 6368 2061 6e64 206e 6565  c search and nee
-000104a0: 6473 2074 6f20 6265 2063 6f2d 7365 7420  ds to be co-set 
-000104b0: 7769 7468 206f 7074 696f 6e20 602d 2d6d  with option `--m
-000104c0: 616e 7561 6c73 6574 602c 2060 2d2d 786c  anualset`, `--xl
-000104d0: 696d 6020 616e 6420 602d 2d79 6c69 6d60  im` and `--ylim`
-000104e0: 2064 6574 6572 6d69 6e69 6e67 2074 6865   determining the
-000104f0: 2078 2061 6e64 2079 2061 7869 7320 6c69   x and y axis li
-00010500: 6d69 7420 6f66 2047 4d4d 202a 4b2a 3c73  mit of GMM *K*<s
-00010510: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-00010520: 6962 7574 696f 6e2c 2060 2d2d 6369 6020  ibution, `--ci` 
-00010530: 7365 7474 696e 6720 7468 6520 636f 6e66  setting the conf
-00010540: 6964 656e 6365 206c 6576 656c 206f 6620  idence level of 
-00010550: 6c6f 672d 6e6f 726d 616c 2064 6973 7472  log-normal distr
-00010560: 6962 7574 696f 6e20 746f 2064 6174 6520  ibution to date 
-00010570: 2864 6566 6175 6c74 2039 3529 2c20 602d  (default 95), `-
-00010580: 2d68 6472 6020 7365 7474 696e 6720 7468  -hdr` setting th
-00010590: 6520 6869 6768 6573 7420 6465 6e73 6974  e highest densit
-000105a0: 7920 7265 6769 6f6e 2028 4844 5229 2061  y region (HDR) a
-000105b0: 7070 6c69 6564 2069 6e20 7468 6520 7365  pplied in the se
-000105c0: 676d 656e 742d 6775 6964 6564 2061 6e63  gment-guided anc
-000105d0: 686f 7220 7061 6972 202a 4b2a 3c73 7562  hor pair *K*<sub
-000105e0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-000105f0: 7574 696f 6e2c 2060 2d2d 6865 7572 6973  ution, `--heuris
-00010600: 7469 6360 2064 6574 6572 6d69 6e69 6e67  tic` determining
-00010610: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00010620: 6961 7465 2068 6575 7269 7374 6963 206d  iate heuristic m
-00010630: 6574 686f 6420 6f66 2064 6566 696e 696e  ethod of definin
-00010640: 6720 4349 2066 6f72 2064 6174 696e 672c  g CI for dating,
-00010650: 2060 2d2d 6b73 6375 746f 6666 6020 7365   `--kscutoff` se
-00010660: 7474 696e 6720 7468 6520 2a4b 2a3c 7375  tting the *K*<su
-00010670: 623e 533c 2f73 7562 3e20 7361 7475 7261  b>S</sub> satura
-00010680: 7469 6f6e 2063 7574 6f66 6620 696e 2064  tion cutoff in d
-00010690: 6174 696e 6720 2864 6566 6175 6c74 2035  ating (default 5
-000106a0: 2920 616e 6420 602d 2d73 686f 7763 6960  ) and `--showci`
-000106b0: 2064 6574 6572 6d69 6e69 6e67 2077 6865   determining whe
-000106c0: 7468 6572 2074 6f20 7368 6f77 2043 4920  ther to show CI 
-000106d0: 666f 7220 7468 6520 6f72 6967 696e 616c  for the original
-000106e0: 2061 6e63 686f 7220 4b73 2047 4d4d 2061   anchor Ks GMM a
-000106f0: 6e61 6c79 7369 732e 0a0a 2a2a 4120 7375  nalysis...**A su
-00010700: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-00010710: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-00010720: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-00010730: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-00010740: 7065 616b 206b 7364 6174 6120 2d61 7020  peak ksdata -ap 
-00010750: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
-00010760: 6120 2d6c 6520 6c65 6461 7461 202d 6d70  a -le ledata -mp
-00010770: 206d 7064 6174 6120 2d2d 6865 7572 6973   mpdata --heuris
-00010780: 7469 630a 6060 600a 0a23 2323 2077 6764  tic.```..### wgd
-00010790: 2073 796e 0a0a 2a2a 5468 6520 696e 7472   syn..**The intr
-000107a0: 612d 7370 6563 6966 6963 2073 796e 7465  a-specific synte
-000107b0: 6e79 2069 6e66 6572 656e 6365 2a2a 0a60  ny inference**.`
-000107c0: 6060 0a77 6764 2073 796e 2066 616d 696c  ``.wgd syn famil
-000107d0: 6965 7320 6766 6620 282d 2d6b 735f 6469  ies gff (--ks_di
-000107e0: 7374 7269 6275 7469 6f6e 206b 7364 6174  stribution ksdat
-000107f0: 6120 2d66 2067 656e 6520 2d61 2049 4420  a -f gene -a ID 
-00010800: 2d2d 6d69 6e6c 656e 202d 3120 2d2d 6d69  --minlen -1 --mi
-00010810: 6e73 6567 6c65 6e20 3130 3030 3020 2d2d  nseglen 10000 --
-00010820: 6d69 6e67 656e 656e 756d 2033 3029 0a60  mingenenum 30).`
-00010830: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00010840: 6077 6764 2073 796e 6020 6973 206d 6169  `wgd syn` is mai
-00010850: 6e6c 7920 6465 616c 696e 6720 7769 7468  nly dealing with
-00010860: 2063 6f6c 6c69 6e65 6172 6974 7920 6f72   collinearity or
-00010870: 2073 796e 7465 6e79 2028 626f 7468 2072   synteny (both r
-00010880: 6566 6572 7265 6420 746f 2061 7320 7379  eferred to as sy
-00010890: 6e74 656e 7920 6865 7265 6166 7465 7229  nteny hereafter)
-000108a0: 2061 6e61 6c79 7369 732e 2054 776f 2069   analysis. Two i
-000108b0: 6e70 7574 2066 696c 6573 2061 7265 2065  nput files are e
-000108c0: 7373 656e 7469 616c 2c20 7468 6520 6765  ssential, the ge
-000108d0: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
-000108e0: 6e64 2074 6865 2067 6666 3320 6669 6c65  nd the gff3 file
-000108f0: 2e20 5468 6520 6765 6e65 2066 616d 696c  . The gene famil
-00010900: 7920 6669 6c65 2069 7320 696e 2074 6865  y file is in the
-00010910: 2066 6f72 6d61 7420 6173 2060 4f72 7468   format as `Orth
-00010920: 6f46 696e 6465 7260 2e20 5468 6520 736f  oFinder`. The so
-00010930: 6674 7761 7265 2060 692d 6164 686f 7265  ftware `i-adhore
-00010940: 6020 6973 2061 2070 7265 7265 7175 6973  ` is a prerequis
-00010950: 6974 652e 2057 6974 6820 6465 6661 756c  ite. With defaul
-00010960: 7420 7061 7261 6d65 7465 7273 2c20 7468  t parameters, th
-00010970: 6520 7072 6f67 7261 6d20 6261 7369 6361  e program basica
-00010980: 6c6c 7920 636f 6e64 7563 7473 2031 2920  lly conducts 1) 
-00010990: 6669 6c74 6572 696e 6720 6765 6e65 2066  filtering gene f
-000109a0: 616d 696c 6965 7320 6261 7365 6420 6f6e  amilies based on
-000109b0: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-000109c0: 7369 7a65 2032 2920 7265 7472 6965 7669  size 2) retrievi
-000109d0: 6e67 2067 656e 6520 706f 7369 7469 6f6e  ng gene position
-000109e0: 2061 6e64 2073 6361 6666 6f6c 6420 696e   and scaffold in
-000109f0: 666f 726d 6174 696f 6e20 6672 6f6d 2067  formation from g
-00010a00: 6666 3320 6669 6c65 2033 2920 7072 6f64  ff3 file 3) prod
-00010a10: 7563 696e 6720 7468 6520 636f 6e66 6967  ucing the config
-00010a20: 7572 6174 696f 6e20 6669 6c65 2061 6e64  uration file and
-00010a30: 2061 7373 6f63 6961 7465 6420 6461 7461   associated data
-00010a40: 6669 6c65 7320 666f 7220 6069 2d61 6468  files for `i-adh
-00010a50: 6f72 6560 2034 2920 6361 6c6c 696e 6720  ore` 4) calling 
-00010a60: 6069 2d61 6468 6f72 6560 2067 6976 656e  `i-adhore` given
-00010a70: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
-00010a80: 7365 7420 746f 2069 6e66 6572 2073 796e  set to infer syn
-00010a90: 7465 6e79 2035 2920 7669 7375 616c 697a  teny 5) visualiz
-00010aa0: 696e 6720 7468 6520 7379 6e74 656e 7920  ing the synteny 
-00010ab0: 696e 2022 646f 7470 6c6f 7422 2069 6e20  in "dotplot" in 
-00010ac0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
-00010ad0: 7320 616e 6420 6261 7365 732c 2069 6e20  s and bases, in 
-00010ae0: 2253 796e 6465 7074 6822 2070 6c6f 7420  "Syndepth" plot 
-00010af0: 7368 6f77 696e 6720 7468 6520 6469 7374  showing the dist
-00010b00: 7269 6275 7469 6f6e 206f 6620 6469 6666  ribution of diff
-00010b10: 6572 656e 7420 6361 7465 676f 7269 6573  erent categories
-00010b20: 206f 6620 636f 6c6c 696e 6561 7269 7479   of collinearity
-00010b30: 2072 6174 696f 7320 7769 7468 696e 2061   ratios within a
-00010b40: 6e64 2062 6574 7765 656e 2073 7065 6369  nd between speci
-00010b50: 6573 2c20 696e 2022 6475 7053 7461 636b  es, in "dupStack
-00010b60: 2220 706c 6f74 2073 686f 7769 6e67 206d  " plot showing m
-00010b70: 756c 7469 706c 6963 6f6e 7320 7769 7468  ultiplicons with
-00010b80: 2064 6966 6665 7265 6e74 206d 756c 7469   different multi
-00010b90: 706c 6963 6174 696f 6e20 6c65 7665 6c73  plication levels
-00010ba0: 2e20 3629 2069 6620 7769 7468 202a 4b2a  . 6) if with *K*
-00010bb0: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-00010bc0: 612c 2061 2022 2a4b 2a3c 7375 623e 533c  a, a "*K*<sub>S<
-00010bd0: 2f73 7562 3e20 646f 7470 6c6f 7422 2077  /sub> dotplot" w
-00010be0: 6974 6820 646f 7473 2061 6e6e 6f74 6174  ith dots annotat
-00010bf0: 6564 2069 6e20 2a4b 2a3c 7375 623e 533c  ed in *K*<sub>S<
-00010c00: 2f73 7562 3e20 7661 6c75 6573 2061 6e64  /sub> values and
-00010c10: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
-00010c20: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-00010c30: 7769 7468 2061 6e63 686f 7220 7061 6972  with anchor pair
-00010c40: 7320 6465 6e6f 7465 6420 7769 6c6c 2062  s denoted will b
-00010c50: 6520 7072 6f64 7563 6564 2e20 5468 6520  e produced. The 
-00010c60: 6765 6e65 2069 6e66 6f72 6d61 7469 6f6e  gene information
-00010c70: 2069 6e20 7468 6520 6765 6e65 2066 616d   in the gene fam
-00010c80: 696c 7920 6669 6c65 2061 6e64 2067 6666  ily file and gff
-00010c90: 3320 6669 6c65 2073 686f 756c 6420 6265  3 file should be
-00010ca0: 206d 6174 6368 6564 2077 6869 6368 2072   matched which r
-00010cb0: 6571 7569 7265 7320 7573 6572 7320 746f  equires users to
-00010cc0: 2073 6574 2070 726f 7065 7220 602d 2d66   set proper `--f
-00010cd0: 6561 7475 7265 6020 616e 6420 602d 2d61  eature` and `--a
-00010ce0: 7474 7269 6275 7465 602e 2054 6865 206d  ttribute`. The m
-00010cf0: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-00010d00: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
-00010d10: 6420 6361 6e20 6265 2073 6574 2076 6961  d can be set via
-00010d20: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
-00010d30: 6178 7369 7a65 602c 206e 6f74 6564 2074  axsize`, noted t
-00010d40: 6861 7420 7468 6973 2066 696c 7465 7269  hat this filteri
-00010d50: 6e67 2069 7320 6d61 696e 6c79 2074 6f20  ng is mainly to 
-00010d60: 6472 6f70 2074 686f 7365 2068 7567 6520  drop those huge 
-00010d70: 7461 6e64 656d 2064 7570 6c69 6361 7465  tandem duplicate
-00010d80: 7320 6661 6d69 6c79 2061 6e64 2074 7261  s family and tra
-00010d90: 6e73 706f 7361 626c 6520 656c 656d 656e  nsposable elemen
-00010da0: 7473 2028 5445 7329 2066 616d 696c 792c  ts (TEs) family,
-00010db0: 2061 6e64 206e 6f74 206d 616e 6461 746f   and not mandato
-00010dc0: 7279 2e20 5573 6572 7320 6361 6e20 6669  ry. Users can fi
-00010dd0: 6c74 6572 2074 686f 7365 2066 7261 676d  lter those fragm
-00010de0: 656e 7461 7279 2073 6361 6666 6f6c 6473  entary scaffolds
-00010df0: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
-00010e00: 602d 2d6d 696e 6c65 6e60 2e20 5468 6520  `--minlen`. The 
-00010e10: 6d69 6e69 6d75 6d20 6c65 6e67 7468 2061  minimum length a
-00010e20: 6e64 206e 756d 6265 7220 6f66 2067 656e  nd number of gen
-00010e30: 6573 2066 6f72 2061 2073 6567 6d65 6e74  es for a segment
-00010e40: 2074 6f20 6265 2063 6f6e 7369 6465 7265   to be considere
-00010e50: 6420 6361 6e20 6265 2073 6574 2076 6961  d can be set via
-00010e60: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
-00010e70: 696e 7365 676c 656e 6020 616e 6420 602d  inseglen` and `-
-00010e80: 2d6d 696e 6765 6e65 6e75 6d60 2e20 5265  -mingenenum`. Re
-00010e90: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
-00010ea0: 636f 6e73 2063 616e 2062 6520 6b65 7074  cons can be kept
-00010eb0: 2062 7920 7365 7420 7468 6520 666c 6167   by set the flag
-00010ec0: 206f 7074 696f 6e20 602d 2d6b 6565 7072   option `--keepr
-00010ed0: 6564 756e 602e 0a0a 2a2a 4120 7375 6767  edun`...**A sugg
-00010ee0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-00010ef0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-00010f00: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00010f10: 6f77 2a2a 0a0a 6060 600a 7767 6420 7379  ow**..```.wgd sy
-00010f20: 6e20 6661 6d69 6c69 6573 2067 6666 0a60  n families gff.`
-00010f30: 6060 0a0a 2a2a 5468 6520 696e 7465 722d  ``..**The inter-
-00010f40: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-00010f50: 2069 6e66 6572 656e 6365 2a2a 0a60 6060   inference**.```
-00010f60: 0a77 6764 2073 796e 2066 616d 696c 6965  .wgd syn familie
-00010f70: 7320 6766 6631 2067 6666 3220 282d 2d61  s gff1 gff2 (--a
-00010f80: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-00010f90: 2022 6d52 4e41 3b4e 616d 6522 202d 2d61   "mRNA;Name" --a
-00010fa0: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-00010fb0: 2022 6765 6e65 3b49 4422 290a 6060 600a   "gene;ID").```.
-00010fc0: 0a46 6f72 206d 756c 7469 2d73 7065 6369  .For multi-speci
-00010fd0: 6573 2073 796e 7465 6e79 2069 6e66 6572  es synteny infer
-00010fe0: 656e 6365 2c20 6966 2075 7365 7273 2068  ence, if users h
-00010ff0: 6176 6520 6766 6633 2066 696c 6573 2077  ave gff3 files w
-00011000: 6869 6368 2068 6176 6520 6469 6666 6572  hich have differ
-00011010: 656e 7420 6665 6174 7572 6573 206f 7220  ent features or 
-00011020: 6174 7472 6962 7574 6573 2066 6f72 2067  attributes for g
-00011030: 656e 6520 706f 7369 7469 6f6e 2069 6e66  ene position inf
-00011040: 6f72 6d61 7469 6f6e 2072 6574 7269 6576  ormation retriev
-00011050: 616c 2c20 7468 6520 6f70 7469 6f6e 2060  al, the option `
-00011060: 2d2d 6164 6469 7469 6f6e 616c 6766 6669  --additionalgffi
-00011070: 6e66 6f60 2063 616e 2062 6520 7365 7420  nfo` can be set 
-00011080: 746f 2070 726f 7669 6465 2074 6865 2061  to provide the a
-00011090: 6464 6974 696f 6e61 6c20 696e 666f 726d  dditional inform
-000110a0: 6174 696f 6e2e 2054 6865 2072 656d 6169  ation. The remai
-000110b0: 6e69 6e67 2070 6172 616d 6574 6572 2073  ning parameter s
-000110c0: 6574 7469 6e67 2069 7320 7468 6520 7361  etting is the sa
-000110d0: 6d65 2061 7320 7468 6520 696e 7472 612d  me as the intra-
-000110e0: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-000110f0: 2069 6e66 6572 656e 6365 2e0a 0a2a 2a41   inference...**A
-00011100: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
-00011110: 696e 6720 7275 6e20 6361 6e20 7573 6520  ing run can use 
-00011120: 636f 6d6d 616e 6420 7369 6d70 6c79 2061  command simply a
-00011130: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-00011140: 6764 2073 796e 2066 616d 696c 6965 7320  gd syn families 
-00011150: 6766 6631 2067 6666 320a 6060 600a 0a23  gff1 gff2.```..#
-00011160: 2323 2077 6764 2076 697a 0a0a 2a2a 5468  ## wgd viz..**Th
-00011170: 6520 7669 7375 616c 697a 6174 696f 6e20  e visualization 
-00011180: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-00011190: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000111a0: 696f 6e20 616e 6420 454c 4d4d 2061 6e61  ion and ELMM ana
-000111b0: 6c79 7369 732a 2a0a 6060 600a 7767 6420  lysis**.```.wgd 
-000111c0: 7669 7a20 2d64 206b 7364 6174 610a 6060  viz -d ksdata.``
-000111d0: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
-000111e0: 7767 6420 7669 7a60 2069 7320 6d61 696e  wgd viz` is main
-000111f0: 6c79 2066 6f72 2074 6865 2070 7572 706f  ly for the purpo
-00011200: 7365 206f 6620 2a4b 2a3c 7375 623e 533c  se of *K*<sub>S<
-00011210: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00011220: 6f6e 2061 6e64 2073 796e 7465 6e79 2076  on and synteny v
-00011230: 6973 7561 6c69 7a61 7469 6f6e 2c20 7769  isualization, wi
-00011240: 7468 2073 6f6d 6520 6f70 7469 6f6e 616c  th some optional
-00011250: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00011260: 6720 616e 616c 7973 6973 2e20 5468 6520  g analysis. The 
-00011270: 6261 7369 6320 6675 6e63 7469 6f6e 2069  basic function i
-00011280: 7320 6a75 7374 2074 6f20 706c 6f74 2074  s just to plot t
-00011290: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-000112a0: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-000112b0: 616e 6420 636f 6e64 7563 7420 616e 2045  and conduct an E
-000112c0: 4c4d 4d20 616e 616c 7973 6973 2069 6e20  LMM analysis in 
-000112d0: 7365 6172 6368 206f 6620 706f 7465 6e74  search of potent
-000112e0: 6961 6c20 5747 4420 636f 6d70 6f6e 656e  ial WGD componen
-000112f0: 7473 2e20 536f 6d65 206b 6579 2070 6172  ts. Some key par
-00011300: 616d 6574 6572 7320 6166 6665 6374 696e  ameters affectin
-00011310: 6720 7468 6520 454c 4d4d 2072 6573 756c  g the ELMM resul
-00011320: 7420 696e 636c 7564 6520 602d 2d70 726f  t include `--pro
-00011330: 6d69 6e65 6e63 655f 6375 746f 6666 602c  minence_cutoff`,
-00011340: 2060 2d2d 7265 6c5f 6865 6967 6874 602c   `--rel_height`,
-00011350: 2060 2d2d 656d 5f69 7465 7261 7469 6f6e   `--em_iteration
-00011360: 7360 2061 6e64 2060 2d2d 656d 5f69 6e69  s` and `--em_ini
-00011370: 7469 616c 697a 6174 696f 6e73 602c 2061  tializations`, a
-00011380: 6c6c 206f 6620 7768 6963 6820 6861 7665  ll of which have
-00011390: 2062 6565 6e20 6578 706c 6169 6e65 6420   been explained 
-000113a0: 6168 6561 642e 0a0a 2a2a 4120 7375 6767  ahead...**A sugg
-000113b0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-000113c0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-000113d0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-000113e0: 6f77 2a2a 0a0a 6060 600a 7767 6420 7669  ow**..```.wgd vi
-000113f0: 7a20 2d64 206b 7364 6174 610a 6060 600a  z -d ksdata.```.
-00011400: 0a2a 2a54 6865 2076 6973 7561 6c69 7a61  .**The visualiza
-00011410: 7469 6f6e 206f 6620 2a4b 2a3c 7375 623e  tion of *K*<sub>
-00011420: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00011430: 7269 6275 7469 6f6e 2077 6974 6820 7261  ribution with ra
-00011440: 7465 2063 6f72 7265 6374 696f 6e2a 2a0a  te correction**.
-00011450: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
-00011460: 7364 6174 6120 2d73 7020 7370 6461 7461  sdata -sp spdata
-00011470: 202d 2d66 6f63 7573 3261 6c6c 2066 6f63   --focus2all foc
-00011480: 616c 5f73 7065 6369 6573 202d 2d65 7874  al_species --ext
-00011490: 7261 7061 7261 6e6f 6d65 6b73 206b 7364  raparanomeks ksd
-000114a0: 6174 6120 282d 2d61 6e63 686f 7270 6f69  ata (--anchorpoi
-000114b0: 6e74 7320 6170 6461 7461 202d 2d70 6c6f  nts apdata --plo
-000114c0: 7461 7067 6d6d 202d 2d70 6c6f 7465 6c6d  tapgmm --plotelm
-000114d0: 6d29 0a60 6060 0a0a 4265 7369 6465 7320  m).```..Besides 
-000114e0: 7468 6520 6261 7369 6320 2a4b 2a3c 7375  the basic *K*<su
-000114f0: 623e 533c 2f73 7562 3e20 706c 6f74 2c20  b>S</sub> plot, 
-00011500: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00011510: 6520 636f 7272 6563 7469 6f6e 2063 616e  e correction can
-00011520: 2061 6c73 6f20 6265 2061 6368 6965 7665   also be achieve
-00011530: 6420 6769 7665 6e20 6174 206c 6561 7374  d given at least
-00011540: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
-00011550: 2876 6961 2074 6865 206f 7074 696f 6e20  (via the option 
-00011560: 602d 2d73 7065 6369 6573 7472 6565 6029  `--speciestree`)
-00011570: 2061 6e64 2061 2066 6f63 616c 2073 7065   and a focal spe
-00011580: 6369 6573 2028 6569 7468 6572 2076 6961  cies (either via
-00011590: 2074 6865 206f 7074 696f 6e20 602d 2d66   the option `--f
-000115a0: 6f63 7573 3261 6c6c 6020 6f72 2076 6961  ocus2all` or via
-000115b0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
-000115c0: 7061 6972 6020 696e 2074 6865 2066 6f72  pair` in the for
-000115d0: 6d20 6f66 2022 2466 6f63 616c 5f73 7065  m of "$focal_spe
-000115e0: 6369 6573 3b24 666f 6361 6c5f 7370 6563  cies;$focal_spec
-000115f0: 6965 7322 292e 2049 7427 7320 7375 6767  ies"). It's sugg
-00011600: 6573 7465 6420 7468 6174 2074 6865 206f  ested that the o
-00011610: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00011620: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
-00011630: 6973 2070 726f 7669 6465 6420 6279 2074  is provided by t
-00011640: 6865 2060 2d2d 6461 7461 6669 6c65 6020  he `--datafile` 
-00011650: 6f70 7469 6f6e 2061 6e64 2074 6865 2070  option and the p
-00011660: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
-00011670: 623e 533c 2f73 7562 3e20 6461 7461 2069  b>S</sub> data i
-00011680: 7320 7072 6f76 6964 6564 2062 7920 7468  s provided by th
-00011690: 6520 602d 2d65 7874 7261 7061 7261 6e6f  e `--extraparano
-000116a0: 6d65 6b73 6020 6f70 7469 6f6e 2c20 616c  meks` option, al
-000116b0: 7468 6f75 6768 2069 7427 7320 616c 6c6f  though it's allo
-000116c0: 7765 6420 746f 206f 6e6c 7920 7072 6f76  wed to only prov
-000116d0: 6964 6520 2a4b 2a3c 7375 623e 533c 2f73  ide *K*<sub>S</s
-000116e0: 7562 3e20 6461 7461 2076 6961 2074 6865  ub> data via the
-000116f0: 2060 2d2d 6461 7461 6669 6c65 6020 6f70   `--datafile` op
-00011700: 7469 6f6e 2061 6e64 2064 6570 6f73 6974  tion and deposit
-00011710: 2062 6f74 6820 6f72 7468 6f6c 6f67 6f75   both orthologou
-00011720: 7326 7061 7261 6c6f 676f 7573 202a 4b2a  s&paralogous *K*
-00011730: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-00011740: 6120 7468 6572 656f 6e2e 2054 6865 7265  a thereon. There
-00011750: 2061 7265 2074 776f 2074 7970 6573 206f   are two types o
-00011760: 6620 6d69 7865 6420 706c 6f74 7320 2874  f mixed plots (t
-00011770: 6865 2022 6d69 7865 6422 2068 6572 6520  he "mixed" here 
-00011780: 7265 6665 7273 2074 6f20 6d69 7865 6420  refers to mixed 
-00011790: 6f72 7468 6f6c 6f67 6f75 7320 616e 6420  orthologous and 
-000117a0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-000117b0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-000117c0: 6962 7574 696f 6e73 292c 206f 6e65 206f  ibutions), one o
-000117d0: 6620 7768 6963 6820 6973 2073 696d 696c  f which is simil
-000117e0: 6172 2074 6f20 7768 6174 2060 6b73 7261  ar to what `ksra
-000117f0: 7465 7360 2070 6c6f 7473 2c20 7768 696c  tes` plots, whil
-00011800: 6520 7468 6520 6f74 6865 7220 6f66 2077  e the other of w
-00011810: 6869 6368 2069 7320 6c69 6b65 2074 6865  hich is like the
-00011820: 2063 6f6e 7665 6e74 696f 6e61 6c20 2a4b   conventional *K
-00011830: 2a3c 7375 623e 533c 2f73 7562 3e20 706c  *<sub>S</sub> pl
-00011840: 6f74 2074 6861 7420 626f 7468 2074 6865  ot that both the
-00011850: 206f 7269 6769 6e61 6c20 6f72 7468 6f6c   original orthol
-00011860: 6f67 6f75 7320 616e 6420 7061 7261 6c6f  ogous and paralo
-00011870: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00011880: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
-00011890: 6e73 2061 7265 2074 7275 7468 6675 6c6c  ns are truthfull
-000118a0: 7920 706c 6f74 7465 6420 696e 7374 6561  y plotted instea
-000118b0: 6420 6f66 206a 7573 7420 6265 696e 6720  d of just being 
-000118c0: 7265 7072 6573 656e 7465 6420 6279 2073  represented by s
-000118d0: 6f6d 6520 7665 7274 6963 616c 206c 696e  ome vertical lin
-000118e0: 6573 2e20 5765 2073 7567 6765 7374 2075  es. We suggest u
-000118f0: 7365 7273 2074 6f20 6164 6f70 7420 7468  sers to adopt th
-00011900: 6520 606b 7372 6174 6573 602d 6c69 6b65  e `ksrates`-like
-00011910: 2070 6c6f 7473 2c20 7768 6963 6820 6973   plots, which is
-00011920: 2074 6865 2064 6566 6175 6c74 2e20 4f74   the default. Ot
-00011930: 6865 7277 6973 6520 7468 6520 6f70 7469  herwise the opti
-00011940: 6f6e 2060 2d2d 636c 6173 7369 6360 2077  on `--classic` w
-00011950: 696c 6c20 6265 206e 6565 6465 6420 746f  ill be needed to
-00011960: 2073 6574 2e20 4578 7472 6120 6d69 7874   set. Extra mixt
-00011970: 7572 6520 6d6f 6465 6c69 6e67 2061 6e61  ure modeling ana
-00011980: 6c79 7369 7320 6361 6e20 6265 2069 6e69  lysis can be ini
-00011990: 7469 6174 6564 2076 6961 2074 6865 206f  tiated via the o
-000119a0: 7074 696f 6e20 602d 2d70 6c6f 7465 6c6d  ption `--plotelm
-000119b0: 6d60 2061 6e64 2060 2d2d 706c 6f74 6170  m` and `--plotap
-000119c0: 676d 6d60 2077 6974 6820 7468 6520 616e  gmm` with the an
-000119d0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-000119e0: 6669 6c65 2070 726f 7669 6465 6420 6279  file provided by
-000119f0: 2074 6865 2060 2d2d 616e 6368 6f72 706f   the `--anchorpo
-00011a00: 696e 7473 6020 6f70 7469 6f6e 2e0a 0a2a  ints` option...*
-00011a10: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-00011a20: 7274 696e 6720 7275 6e20 6361 6e20 7573  rting run can us
-00011a30: 6520 636f 6d6d 616e 6420 7369 6d70 6c79  e command simply
-00011a40: 2061 7320 6265 6c6f 772a 2a0a 0a60 6060   as below**..```
-00011a50: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
-00011a60: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
-00011a70: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
-00011a80: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
-00011a90: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
-00011aa0: 0a60 6060 0a0a 2a2a 5468 6520 7669 7375  .```..**The visu
-00011ab0: 616c 697a 6174 696f 6e20 6f66 2073 796e  alization of syn
-00011ac0: 7465 6e79 2a2a 0a60 6060 0a77 6764 2076  teny**.```.wgd v
-00011ad0: 697a 202d 6170 2061 7064 6174 6120 2d73  iz -ap apdata -s
-00011ae0: 6d20 736d 6461 7461 202d 6d74 206d 7464  m smdata -mt mtd
-00011af0: 6174 6120 2d67 7420 6774 6461 7461 202d  ata -gt gtdata -
-00011b00: 2d70 6c6f 7473 796e 2028 2d2d 6d69 6e6c  -plotsyn (--minl
-00011b10: 656e 202d 3120 2d2d 6d69 6e73 6567 6c65  en -1 --minsegle
-00011b20: 6e20 3130 3030 3020 2d2d 6d69 6e67 656e  n 10000 --mingen
-00011b30: 656e 756d 2033 3029 0a60 6060 0a0a 436f  enum 30).```..Co
-00011b40: 6d70 6172 6564 2074 6f20 7468 6520 6f72  mpared to the or
-00011b50: 6967 696e 616c 2060 7767 6460 2c20 7468  iginal `wgd`, th
-00011b60: 6520 6077 6764 2076 697a 6020 7072 6f67  e `wgd viz` prog
-00011b70: 7261 6d20 6164 6465 6420 7379 6e74 656e  ram added synten
-00011b80: 7920 7669 7375 616c 697a 6174 696f 6e20  y visualization 
-00011b90: 7069 7065 6c69 6e65 2e20 5573 6572 7320  pipeline. Users 
-00011ba0: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-00011bb0: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-00011bc0: 602d 2d70 6c6f 7473 796e 6020 746f 2069  `--plotsyn` to i
-00011bd0: 6e69 7469 6174 6520 7468 6973 2070 6172  nitiate this par
-00011be0: 7420 6f66 2070 6970 656c 696e 652e 2054  t of pipeline. T
-00011bf0: 6869 7320 7374 6570 2061 7373 756d 6573  his step assumes
-00011c00: 2074 6861 7420 7573 6572 7320 6861 7665   that users have
-00011c10: 206f 6274 6169 6e65 6420 616c 7265 6164   obtained alread
-00011c20: 7920 7468 6520 7379 6e74 656e 6963 2072  y the syntenic r
-00011c30: 6573 756c 7473 2066 726f 6d20 6069 2d61  esults from `i-a
-00011c40: 6468 6f72 6560 2061 6e64 2075 7365 7320  dhore` and uses 
-00011c50: 7468 6f73 6520 7265 7375 6c74 2066 696c  those result fil
-00011c60: 6573 2074 6f20 7265 616c 697a 6520 7468  es to realize th
-00011c70: 6520 7379 6e74 656e 7920 7669 7375 616c  e synteny visual
-00011c80: 697a 6174 696f 6e2e 2053 696d 6c69 6172  ization. Simliar
-00011c90: 2074 6f20 6077 6764 2073 796e 602c 2074   to `wgd syn`, t
-00011ca0: 6865 2065 7874 7261 202a 4b2a 3c73 7562  he extra *K*<sub
-00011cb0: 3e53 3c2f 7375 623e 2064 6174 6120 6669  >S</sub> data fi
-00011cc0: 6c65 2063 616e 2062 6520 7472 616e 736d  le can be transm
-00011cd0: 6974 7465 6420 7669 6120 7468 6520 6f70  itted via the op
-00011ce0: 7469 6f6e 2060 2d2d 6461 7461 6669 6c65  tion `--datafile
-00011cf0: 6020 2869 6e73 7465 6164 206f 6620 602d  ` (instead of `-
-00011d00: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00011d10: 6020 6f70 7469 6f6e 292e 2042 6173 6963  ` option). Basic
-00011d20: 616c 6c79 2c20 7468 6520 7379 6e74 656e  ally, the synten
-00011d30: 6963 2072 6573 756c 7420 6669 6c65 7320  ic result files 
-00011d40: 7265 7175 6972 6564 2061 7265 2074 6865  required are the
-00011d50: 2061 6e63 686f 7220 706f 696e 7473 2064   anchor points d
-00011d60: 6174 6166 696c 652c 206d 756c 7469 706c  atafile, multipl
-00011d70: 6963 6f6e 7320 6461 7461 6669 6c65 2c20  icons datafile, 
-00011d80: 6765 6e65 2074 6162 6c65 2064 6174 6166  gene table dataf
-00011d90: 696c 6520 2861 7574 6f6d 6174 656c 7920  ile (automately 
-00011da0: 7072 6f64 7563 6564 2062 7920 6077 6764  produced by `wgd
-00011db0: 2073 796e 6029 2c20 616e 6420 7365 676d   syn`), and segm
-00011dc0: 656e 7473 2064 6174 6166 696c 652e 0a0a  ents datafile...
-00011dd0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-00011de0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-00011df0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-00011e00: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-00011e10: 600a 7767 6420 7669 7a20 2d61 7020 6170  `.wgd viz -ap ap
-00011e20: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
-00011e30: 2d6d 7420 6d74 6461 7461 202d 6774 2067  -mt mtdata -gt g
-00011e40: 7464 6174 6120 2d2d 706c 6f74 7379 6e0a  tdata --plotsyn.
-00011e50: 6060 600a 0a23 2320 496c 6c75 7374 7261  ```..## Illustra
-00011e60: 7469 6f6e 0a0a 5765 2069 6c6c 7573 7472  tion..We illustr
-00011e70: 6174 6520 6f75 7220 7072 6f67 7261 6d20  ate our program 
-00011e80: 6f6e 2061 6e20 6578 656d 706c 6172 7920  on an exemplary 
-00011e90: 5747 4420 696e 6665 7265 6e63 6520 616e  WGD inference an
-00011ea0: 6420 6461 7469 6e67 2075 706f 6e20 7370  d dating upon sp
-00011eb0: 6563 6965 7320 2a41 7175 696c 6567 6961  ecies *Aquilegia
-00011ec0: 2063 6f65 7275 6c65 612a 2e0a 0a54 6865   coerulea*...The
-00011ed0: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-00011ee0: 756c 6561 2a20 7761 7320 7265 706f 7274  ulea* was report
-00011ef0: 6564 2074 6f20 6578 7065 7269 656e 6365  ed to experience
-00011f00: 2061 6e20 7061 6c65 6f2d 706f 6c79 706c   an paleo-polypl
-00011f10: 6f69 6469 7a61 7469 6f6e 2065 7665 6e74  oidization event
-00011f20: 2061 6674 6572 2074 6865 2064 6976 6572   after the diver
-00011f30: 6765 6e63 6520 6f66 2063 6f72 6520 6575  gence of core eu
-00011f40: 6469 636f 7473 2c20 7768 6963 6820 6973  dicots, which is
-00011f50: 206c 696b 656c 7920 7368 6172 6564 2062   likely shared b
-00011f60: 7920 616c 6c20 5261 6e75 6e63 756c 616c  y all Ranunculal
-00011f70: 6573 2e0a 0a46 6972 7374 2061 626f 7665  es...First above
-00011f80: 2061 6c6c 2c20 6c65 7427 7320 6465 6c69   all, let's deli
-00011f90: 6e65 6174 6520 7468 6520 7768 6f6c 6520  neate the whole 
-00011fa0: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-00011fb0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-00011fc0: 7472 6962 7574 696f 6e20 616e 6420 6861  tribution and ha
-00011fd0: 7665 2061 2062 6173 6963 206f 6273 6572  ve a basic obser
-00011fe0: 7661 7469 6f6e 2066 6f72 2070 6f74 656e  vation for poten
-00011ff0: 7469 616c 6c79 2063 6f6e 6365 6976 6162  tially conceivab
-00012000: 6c65 2057 4744 732c 2075 7369 6e67 2074  le WGDs, using t
-00012010: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
-00012020: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
-00012030: 646d 6420 4171 7569 6c65 6769 615f 636f  dmd Aquilegia_co
-00012040: 6572 756c 6561 0a77 6764 206b 7364 2077  erulea.wgd ksd w
-00012050: 6764 5f64 6d64 2f41 7175 696c 6567 6961  gd_dmd/Aquilegia
-00012060: 5f63 6f65 7275 6c65 612e 7473 7620 4171  _coerulea.tsv Aq
-00012070: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012080: 0a60 6060 0a0a 5468 6520 636f 6e73 7472  .```..The constr
-00012090: 7563 7465 6420 7768 6f6c 6520 7061 7261  ucted whole para
-000120a0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-000120b0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000120c0: 7574 696f 6e20 6f66 202a 4171 7569 6c65  ution of *Aquile
-000120d0: 6769 6120 636f 6572 756c 6561 2a20 6973  gia coerulea* is
-000120e0: 2061 7320 6265 6c6f 772c 2077 6520 6361   as below, we ca
-000120f0: 6e20 7365 6520 7468 6174 2074 6865 7265  n see that there
-00012100: 2073 6565 6d73 2074 6f20 6265 2061 2068   seems to be a h
-00012110: 756d 7020 6174 202a 4b2a 3c73 7562 3e53  ump at *K*<sub>S
-00012120: 3c2f 7375 623e 2031 2062 7574 206e 6f74  </sub> 1 but not
-00012130: 2063 6c65 6172 2e0a 0a21 5b5d 2864 6174   clear...![](dat
-00012140: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00012150: 756c 6561 2e74 7376 2e6b 7364 5f77 702e  ulea.tsv.ksd_wp.
-00012160: 7376 6729 0a0a 5765 2074 6865 6e20 636f  svg)..We then co
-00012170: 6e73 7472 7563 7420 7468 6520 616e 6368  nstruct the anch
-00012180: 6f72 202a 4b2a 3c73 7562 3e53 3c2f 7375  or *K*<sub>S</su
-00012190: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000121a0: 696f 6e20 7573 696e 6720 7468 6520 636f  ion using the co
-000121b0: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
-000121c0: 2e0a 0a60 6060 0a77 6764 2073 796e 202d  ...```.wgd syn -
-000121d0: 6620 6d52 4e41 202d 6120 4e61 6d65 2077  f mRNA -a Name w
-000121e0: 6764 5f64 6d64 2f41 7175 696c 6567 6961  gd_dmd/Aquilegia
-000121f0: 5f63 6f65 7275 6c65 612e 7473 7620 4171  _coerulea.tsv Aq
-00012200: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012210: 2e67 6666 3320 2d6b 7320 7767 645f 6b73  .gff3 -ks wgd_ks
-00012220: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00012230: 756c 6561 2e74 7376 2e6b 732e 7473 760a  ulea.tsv.ks.tsv.
-00012240: 6060 600a 0a41 7320 7368 6f77 6e20 6265  ```..As shown be
-00012250: 6c6f 772c 2074 6865 7265 2061 7265 2073  low, there are s
-00012260: 6f6d 6520 7265 7461 696e 6564 2061 6e63  ome retained anc
-00012270: 686f 7220 7061 6972 7320 7769 7468 202a  hor pairs with *
-00012280: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2062  K*<sub>S</sub> b
-00012290: 6574 7765 656e 2031 2061 6e64 2032 2c20  etween 1 and 2, 
-000122a0: 7768 6963 6820 7365 656d 7320 746f 2073  which seems to s
-000122b0: 7567 6765 7374 2061 2057 4744 2065 7665  uggest a WGD eve
-000122c0: 6e74 2e0a 0a21 5b5d 2864 6174 612f 4171  nt...![](data/Aq
-000122d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000122e0: 2e74 7376 2e6b 7364 5f77 705f 6170 2e73  .tsv.ksd_wp_ap.s
-000122f0: 7667 290a 0a54 6865 2061 7373 6f63 6961  vg)..The associa
-00012300: 7465 6420 6064 7570 5374 6163 6b60 2070  ted `dupStack` p
-00012310: 6c6f 7420 7368 6f77 7320 7468 6174 2074  lot shows that t
-00012320: 6865 7265 2061 7265 206e 756d 6572 6f75  here are numerou
-00012330: 7320 6475 706c 6963 6174 6564 2073 6567  s duplicated seg
-00012340: 6d65 6e74 7320 6163 726f 7373 206d 6f73  ments across mos
-00012350: 7420 6f66 2074 6865 2063 6872 6f6d 6f73  t of the chromos
-00012360: 6f6d 6573 2e0a 0a21 5b5d 2864 6174 612f  omes...![](data/
-00012370: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012380: 6561 5f41 7175 696c 6567 6961 5f63 6f65  ea_Aquilegia_coe
-00012390: 7275 6c65 615f 6d75 6c74 6970 6c69 636f  rulea_multiplico
-000123a0: 6e73 5f6c 6576 656c 2e73 7667 290a 0a57  ns_level.svg)..W
-000123b0: 6520 696d 706c 656d 656e 7465 6420 7477  e implemented tw
-000123c0: 6f20 7479 7065 7320 6f66 2064 6f74 2070  o types of dot p
-000123d0: 6c6f 7473 2069 6e20 6f78 666f 7264 2067  lots in oxford g
-000123e0: 7269 643a 206f 6e65 2069 6e20 7468 6520  rid: one in the 
-000123f0: 756e 6974 206f 6620 6261 7365 7320 616e  unit of bases an
-00012400: 6420 7468 6520 6f74 6865 7220 696e 2074  d the other in t
-00012410: 6865 2075 6e69 7420 6f66 2067 656e 6573  he unit of genes
-00012420: 2c20 7768 6963 6820 6361 6e20 6265 2063  , which can be c
-00012430: 6f6c 6f72 6564 2062 7920 2a4b 2a3c 7375  olored by *K*<su
-00012440: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-00012450: 2067 6976 656e 202a 4b2a 3c73 7562 3e53   given *K*<sub>S
-00012460: 3c2f 7375 623e 2064 6174 612e 0a0a 215b  </sub> data...![
-00012470: 5d28 6461 7461 2f41 7175 696c 6567 6961  ](data/Aquilegia
-00012480: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
-00012490: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-000124a0: 4b73 2e64 6f74 5f75 6e69 745f 6765 6e65  Ks.dot_unit_gene
-000124b0: 2e70 6e67 290a 0a41 7320 7368 6f77 6e20  .png)..As shown 
-000124c0: 6162 6f76 652c 2074 6865 2064 6f74 2070  above, the dot p
-000124d0: 6c6f 7420 696e 2074 6865 2075 6e69 7420  lot in the unit 
-000124e0: 6f66 2067 656e 6573 2070 7265 7365 6e74  of genes present
-000124f0: 7320 6e75 6d65 726f 7573 2064 656e 7365  s numerous dense
-00012500: 6c79 2061 6767 7265 6761 7465 6420 286c  ly aggregated (l
-00012510: 696e 652d 6c69 6b65 2920 616e 6368 6f72  ine-like) anchor
-00012520: 2070 6f69 6e74 7320 6174 206d 6f73 7420   points at most 
-00012530: 6f66 2074 6865 2063 6872 6f6d 6f73 6f6d  of the chromosom
-00012540: 6573 2077 6974 6820 636f 6e73 6973 7465  es with consiste
-00012550: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
-00012560: 623e 2061 6765 2062 6574 7765 656e 2031  b> age between 1
-00012570: 2061 6e64 2032 2e20 5468 6520 646f 7420   and 2. The dot 
-00012580: 706c 6f74 2069 6e20 7468 6520 756e 6974  plot in the unit
-00012590: 206f 6620 6261 7365 7320 7368 6f77 7320   of bases shows 
-000125a0: 7468 6520 7361 6d65 2070 6174 7465 726e  the same pattern
-000125b0: 2c20 6173 206d 616e 6966 6573 7465 6420  , as manifested 
-000125c0: 6265 6c6f 772e 0a0a 215b 5d28 6461 7461  below...![](data
-000125d0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-000125e0: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
-000125f0: 5f63 6f65 7275 6c65 615f 4b73 2e64 6f74  _coerulea_Ks.dot
-00012600: 2e70 6e67 290a 0a54 6865 2064 6f74 2070  .png)..The dot p
-00012610: 6c6f 7473 2077 6974 686f 7574 202a 4b2a  lots without *K*
-00012620: 3c73 7562 3e53 3c2f 7375 623e 2061 6e6e  <sub>S</sub> ann
-00012630: 6f74 6174 696f 6e20 7769 6c6c 2061 6c73  otation will als
-00012640: 6f20 6265 2061 7574 6f6d 6174 656c 7920  o be automately 
-00012650: 7072 6f64 7563 6564 2c20 6173 2073 686f  produced, as sho
-00012660: 776e 2062 656c 6f77 2e0a 0a21 5b5d 2864  wn below...![](d
-00012670: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00012680: 6572 756c 6561 2d76 732d 4171 7569 6c65  erulea-vs-Aquile
-00012690: 6769 615f 636f 6572 756c 6561 2e64 6f74  gia_coerulea.dot
-000126a0: 5f75 6e69 745f 6765 6e65 2e70 6e67 290a  _unit_gene.png).
-000126b0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-000126c0: 6769 615f 636f 6572 756c 6561 2d76 732d  gia_coerulea-vs-
-000126d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000126e0: 6561 2e64 6f74 2e70 6e67 290a 0a4e 6f74  ea.dot.png)..Not
-000126f0: 6520 7468 6174 2074 6865 206f 7061 6369  e that the opaci
-00012700: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
-00012710: 7320 616e 6420 616c 6c20 686f 6d6f 6c6f  s and all homolo
-00012720: 6720 646f 7473 2063 616e 2062 6520 7365  g dots can be se
-00012730: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
-00012740: 6061 7061 6c70 6861 6020 616e 6420 6068  `apalpha` and `h
-00012750: 6f61 6c70 6861 6020 7365 7061 7261 7465  oalpha` separate
-00012760: 6c79 2e20 4966 206f 6e65 206a 7573 7420  ly. If one just 
-00012770: 7761 6e74 7320 746f 2073 6565 2074 6865  wants to see the
-00012780: 2061 6e63 686f 7220 646f 7473 2c20 7365   anchor dots, se
-00012790: 7474 696e 6720 7468 6520 6068 6f61 6c70  tting the `hoalp
-000127a0: 6861 6020 6173 2030 2028 6f72 206f 7468  ha` as 0 (or oth
-000127b0: 6572 206d 696e 7573 6375 6c65 2076 616c  er minuscule val
-000127c0: 7565 7329 2077 696c 6c20 646f 2e20 4966  ues) will do. If
-000127d0: 206f 6e65 2077 616e 7473 2074 6f20 7365   one wants to se
-000127e0: 6520 7468 6520 6469 7374 7269 6275 7469  e the distributi
-000127f0: 6f6e 206f 6620 7768 6f6c 6520 646f 7473  on of whole dots
-00012800: 2062 6574 7465 722c 2073 6574 7469 6e67   better, setting
-00012810: 2074 6865 2060 686f 616c 7068 6160 2068   the `hoalpha` h
-00012820: 6967 6865 7220 2861 6e64 2060 6170 616c  igher (and `apal
-00012830: 7068 6160 206c 6f77 6572 2920 7769 6c6c  pha` lower) will
-00012840: 2064 6f2e 2054 6865 2060 646f 7473 697a   do. The `dotsiz
-00012850: 6560 206f 7074 696f 6e20 6361 6e20 6265  e` option can be
-00012860: 2063 616c 6c65 6420 746f 2061 646a 7573   called to adjus
-00012870: 7420 7468 6520 7369 7a65 206f 6620 646f  t the size of do
-00012880: 7473 2e0a 0a41 2066 7572 7468 6572 2061  ts...A further a
-00012890: 7373 6f63 6961 7465 6420 5379 6e64 6570  ssociated Syndep
-000128a0: 7468 2070 6c6f 7420 7368 6f77 7320 7468  th plot shows th
-000128b0: 6174 2074 6865 7265 2061 7265 206d 6f72  at there are mor
-000128c0: 6520 7468 616e 2038 3020 6475 706c 6963  e than 80 duplic
-000128d0: 6174 6564 2073 6567 6d65 6e74 732c 2077  ated segments, w
-000128e0: 6869 6368 2064 6f6d 696e 6174 6573 2074  hich dominates t
-000128f0: 6865 2077 686f 6c65 2063 6f6c 6c69 6e65  he whole colline
-00012900: 6172 2072 6174 696f 2063 6174 6567 6f72  ar ratio categor
-00012910: 792e 0a0a 215b 5d28 6461 7461 2f53 796e  y...![](data/Syn
-00012920: 6465 7074 682e 7376 6729 0a0a 5765 2063  depth.svg)..We c
-00012930: 616e 2066 6974 2061 6e20 454c 4d4d 206d  an fit an ELMM m
-00012940: 6978 7475 7265 206d 6f64 656c 2075 706f  ixture model upo
-00012950: 6e20 7468 6520 7768 6f6c 6520 7061 7261  n the whole para
-00012960: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-00012970: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-00012980: 7574 696f 6e20 746f 2073 6565 206d 6f72  ution to see mor
-00012990: 6520 6163 6375 7261 7465 6c79 2074 6865  e accurately the
-000129a0: 2073 6967 6e69 6669 6361 6e63 6520 616e   significance an
-000129b0: 6420 6c6f 6361 7469 6f6e 206f 6620 706f  d location of po
-000129c0: 7465 6e74 6961 6c20 5747 4473 2c20 7573  tential WGDs, us
-000129d0: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
-000129e0: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
-000129f0: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-00012a00: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
-00012a10: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00012a20: 760a 6060 600a 0a54 6865 2072 6573 756c  v.```..The resul
-00012a30: 7420 6f66 2045 4c4d 4d20 6d69 7874 7572  t of ELMM mixtur
-00012a40: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
-00012a50: 6e67 2073 686f 7773 2074 6861 7420 7468  ng shows that th
-00012a60: 6572 6520 6973 2061 206c 696b 656c 7920  ere is a likely 
-00012a70: 5747 4420 636f 6d70 6f6e 656e 7420 6174  WGD component at
-00012a80: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012a90: 2031 2e31 392e 0a0a 215b 5d28 6461 7461   1.19...![](data
-00012aa0: 2f65 6c6d 6d5f 4171 7569 6c65 6769 615f  /elmm_Aquilegia_
-00012ab0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
-00012ac0: 7473 765f 6265 7374 5f6d 6f64 656c 735f  tsv_best_models_
-00012ad0: 7765 6967 6874 6564 2e73 7667 290a 0a4c  weighted.svg)..L
-00012ae0: 6574 2773 2064 6f20 6120 6d69 7874 7572  et's do a mixtur
-00012af0: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
-00012b00: 6e67 2066 6f72 2061 6e63 686f 7220 2a4b  ng for anchor *K
-00012b10: 2a3c 7375 623e 533c 2f73 7562 3e20 746f  *<sub>S</sub> to
-00012b20: 6f2c 2075 7369 6e67 2074 6865 2063 6f6d  o, using the com
-00012b30: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
-00012b40: 0a0a 6060 600a 7767 6420 7065 616b 2077  ..```.wgd peak w
-00012b50: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00012b60: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00012b70: 2e74 7376 202d 2d61 6e63 686f 7270 6f69  .tsv --anchorpoi
-00012b80: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-00012b90: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
-00012ba0: 696e 7473 2e74 7874 202d 2d73 6567 6d65  ints.txt --segme
-00012bb0: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-00012bc0: 6f72 652d 6f75 742f 7365 676d 656e 7473  ore-out/segments
-00012bd0: 2e74 7874 202d 2d6c 6973 7465 6c65 6d65  .txt --listeleme
-00012be0: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-00012bf0: 6f72 652d 6f75 742f 6c69 7374 5f65 6c65  ore-out/list_ele
-00012c00: 6d65 6e74 732e 7478 7420 2d2d 6d75 6c74  ments.txt --mult
-00012c10: 6970 6c69 636f 6e70 6169 7273 2077 6764  ipliconpairs wgd
-00012c20: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00012c30: 2f6d 756c 7469 706c 6963 6f6e 5f70 6169  /multiplicon_pai
-00012c40: 7273 2e74 7874 202d 2d77 6569 6768 7465  rs.txt --weighte
-00012c50: 640a 6060 600a 0a54 6865 2061 6e63 686f  d.```..The ancho
-00012c60: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00012c70: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-00012c80: 6f6e 2061 6c73 6f20 6861 7320 6120 6c69  on also has a li
-00012c90: 6b65 6c79 2057 4744 2063 6f6d 706f 6e65  kely WGD compone
-00012ca0: 6e74 2077 6974 6820 6d6f 6465 2031 2e32  nt with mode 1.2
-00012cb0: 382e 0a0a 215b 5d28 6461 7461 2f4f 7269  8...![](data/Ori
-00012cc0: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
-00012cd0: 4d4d 5f43 6f6d 706f 6e65 6e74 335f 6e6f  MM_Component3_no
-00012ce0: 6465 5f77 6569 6768 7465 645f 4c6f 676e  de_weighted_Logn
-00012cf0: 6f72 6d61 6c2e 7376 6729 0a0a 4e6f 7720  ormal.svg)..Now 
-00012d00: 7468 6174 2077 6520 6861 7665 2073 6565  that we have see
-00012d10: 6e20 7468 6520 6576 6964 656e 6365 206f  n the evidence o
-00012d20: 6620 6e75 6d65 726f 7573 2064 7570 6c69  f numerous dupli
-00012d30: 6361 7465 6420 7365 676d 656e 7473 2061  cated segments a
-00012d40: 6e64 2074 6865 2061 6767 7265 6761 7469  nd the aggregati
-00012d50: 6f6e 206f 6620 6475 706c 6963 6174 6573  on of duplicates
-00012d60: 2061 6765 2069 6e20 2a4b 2a3c 7375 623e   age in *K*<sub>
-00012d70: 533c 2f73 7562 3e20 6172 6f75 6e64 2031  S</sub> around 1
-00012d80: 2e32 2066 6f72 2061 6e63 686f 7220 7061  .2 for anchor pa
-00012d90: 6972 7320 616e 6420 6e6f 6e2d 616e 6368  irs and non-anch
-00012da0: 6f72 2070 6169 7273 2074 6872 6f75 6768  or pairs through
-00012db0: 6f75 7420 7468 6520 7768 6f6c 6520 6765  out the whole ge
-00012dc0: 6e6f 6d65 2e20 5765 2063 616e 2063 6c61  nome. We can cla
-00012dd0: 696d 2077 6974 6820 736f 6d65 2063 6f6e  im with some con
-00012de0: 6669 6465 6e63 6520 7468 6174 202a 4171  fidence that *Aq
-00012df0: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-00012e00: 2a20 6d69 6768 7420 6861 7665 2065 7870  * might have exp
-00012e10: 6572 6965 6e63 6564 2061 2070 616c 656f  erienced a paleo
-00012e20: 2d70 6f6c 7970 6c6f 6964 697a 6174 696f  -polyploidizatio
-00012e30: 6e20 6576 656e 742e 204e 6578 742c 204c  n event. Next, L
-00012e40: 6574 2773 2068 6176 6520 6120 6675 7274  et's have a furt
-00012e50: 6865 7220 6c6f 6f6b 2061 626f 7574 2069  her look about i
-00012e60: 7473 2070 6879 6c6f 6765 6e65 7469 6320  ts phylogenetic 
-00012e70: 6c6f 6361 7469 6f6e 2e20 5765 206b 6e6f  location. We kno
-00012e80: 7720 7468 6174 2074 6865 7265 2061 7265  w that there are
-00012e90: 2075 6e63 6572 7461 696e 7469 6573 2061   uncertainties a
-00012ea0: 626f 7574 2077 6865 7468 6572 2074 6869  bout whether thi
-00012eb0: 7320 7075 7461 7469 7665 2070 616c 656f  s putative paleo
-00012ec0: 2d70 6f6c 7970 6c6f 6964 697a 6174 696f  -polyploidizatio
-00012ed0: 6e20 6576 656e 7420 6973 2073 6861 7265  n event is share
-00012ee0: 6420 7769 7468 2061 6c6c 2065 7564 6963  d with all eudic
-00012ef0: 6f74 7320 6f72 206e 6f74 2e20 5765 2063  ots or not. We c
-00012f00: 616e 2063 686f 6f73 6520 736f 6d65 206f  an choose some o
-00012f10: 7468 6572 2065 7564 6963 6f74 2067 656e  ther eudicot gen
-00012f20: 6f6d 6573 2074 6f20 7365 6520 7468 6520  omes to see the 
-00012f30: 6f72 6465 7269 6e67 206f 6620 7370 6563  ordering of spec
-00012f40: 6961 7469 6f6e 2061 6e64 2070 6f6c 7970  iation and polyp
-00012f50: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-00012f60: 7473 2e20 4865 7265 2077 6520 6368 6f6f  ts. Here we choo
-00012f70: 7365 202a 5669 7469 7320 7669 6e69 6665  se *Vitis vinife
-00012f80: 7261 2a2c 202a 5072 6f74 6561 2063 796e  ra*, *Protea cyn
-00012f90: 6172 6f69 6465 732a 2061 6e64 202a 4163  aroides* and *Ac
-00012fa0: 6f72 7573 2061 6d65 7269 6361 6e75 732a  orus americanus*
-00012fb0: 2069 6e20 7468 6520 666f 6c6c 6f77 696e   in the followin
-00012fc0: 6720 2a4b 2a3c 7375 623e 533c 2f73 7562  g *K*<sub>S</sub
-00012fd0: 3e20 616e 616c 7973 6973 2e20 4669 7273  > analysis. Firs
-00012fe0: 742c 2077 6520 6275 696c 7420 6120 676c  t, we built a gl
-00012ff0: 6f62 616c 204d 5242 4820 6661 6d69 6c79  obal MRBH family
-00013000: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013010: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-00013020: 6764 2064 6d64 202d 2d67 6c6f 6261 6c6d  gd dmd --globalm
-00013030: 7262 6820 4171 7569 6c65 6769 615f 636f  rbh Aquilegia_co
-00013040: 6572 756c 6561 2050 726f 7465 615f 6379  erulea Protea_cy
-00013050: 6e61 726f 6964 6573 2041 636f 7275 735f  naroides Acorus_
-00013060: 616d 6572 6963 616e 7573 2056 6974 6973  americanus Vitis
-00013070: 5f76 696e 6966 6572 6120 2d6f 2077 6764  _vinifera -o wgd
-00013080: 5f67 6c6f 6261 6c6d 7262 680a 6060 600a  _globalmrbh.```.
-00013090: 0a49 6e20 7468 6520 676c 6f62 616c 204d  .In the global M
-000130a0: 5242 4820 6661 6d69 6c79 2c20 6576 6572  RBH family, ever
-000130b0: 7920 7061 6972 206f 6620 6f72 7468 6f6c  y pair of orthol
-000130c0: 6f67 6f75 7320 6765 6e65 7320 6973 2074  ogous genes is t
-000130d0: 6865 2072 6563 6970 726f 6361 6c20 6265  he reciprocal be
-000130e0: 7374 2068 6974 2c20 7375 6767 6573 7469  st hit, suggesti
-000130f0: 6e67 2074 7275 6520 6f72 7468 6f6c 6f67  ng true ortholog
-00013100: 6f75 7320 7265 6c61 7469 6f6e 7368 6970  ous relationship
-00013110: 732e 2057 6520 776f 756c 6420 7573 6520  s. We would use 
-00013120: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-00013130: 7562 3e20 7661 6c75 6573 2061 7373 6f63  ub> values assoc
-00013140: 6961 7465 6420 7769 7468 2074 6865 7365  iated with these
-00013150: 206f 7274 686f 6c6f 676f 7573 2070 6169   orthologous pai
-00013160: 7273 2074 6f20 6465 6c69 6d69 7420 7468  rs to delimit th
-00013170: 6520 6469 7665 7267 656e 6365 202a 4b2a  e divergence *K*
-00013180: 3c73 7562 3e53 3c2f 7375 623e 2070 6561  <sub>S</sub> pea
-00013190: 6b2e 2054 6f67 6574 6865 7220 7769 7468  k. Together with
-000131a0: 2074 6865 2077 686f 6c65 2070 6172 616e   the whole paran
-000131b0: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
-000131c0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
-000131d0: 2c20 7765 2063 6f6e 6475 6374 2074 6865  , we conduct the
-000131e0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-000131f0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013200: 6e64 2062 656c 6f77 2e0a 0a21 2153 696e  nd below...!!Sin
-00013210: 6365 2060 7767 6460 2076 6572 7369 6f6e  ce `wgd` version
-00013220: 2032 2e30 2e32 342c 2077 6520 7265 7772   2.0.24, we rewr
-00013230: 6f74 6520 6120 636c 6561 6e65 7220 616e  ote a cleaner an
-00013240: 6420 7175 6963 6b65 7220 7761 7920 6f66  d quicker way of
-00013250: 2064 6f69 6e67 2073 7562 7374 6974 7574   doing substitut
-00013260: 696f 6e20 7261 7465 2063 6f72 7265 6374  ion rate correct
-00013270: 696f 6e2e 2049 7427 7320 6e6f 7420 7265  ion. It's not re
-00013280: 7175 6972 6564 2074 6f20 7479 7065 2069  quired to type i
-00013290: 6e20 616e 7920 7370 6569 6365 7320 7061  n any speices pa
-000132a0: 6972 2061 6e64 2061 2073 6572 6965 7320  ir and a series 
-000132b0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-000132c0: 623e 2070 6c6f 7473 2077 696c 6c20 6265  b> plots will be
-000132d0: 2070 726f 6475 6365 642e 2054 6865 2072   produced. The r
-000132e0: 6571 7569 7265 6420 6669 6c65 7320 6172  equired files ar
-000132f0: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00013300: 2a3c 7375 623e 533c 2f73 7562 3e20 7461  *<sub>S</sub> ta
-00013310: 626c 652c 2070 6172 616c 6f67 6f75 7320  ble, paralogous 
-00013320: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013330: 7461 626c 652c 2061 2073 7065 6369 6573  table, a species
-00013340: 2074 7265 6520 616e 6420 6120 666f 6375   tree and a focu
-00013350: 7365 6420 7370 6563 6965 7320 2874 6865  sed species (the
-00013360: 206f 6e65 2069 6e70 7574 7465 6420 7769   one inputted wi
-00013370: 7468 2070 6172 616c 6f67 6f75 7320 2a4b  th paralogous *K
-00013380: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
-00013390: 7461 292e 2055 7365 7273 2063 616e 2063  ta). Users can c
-000133a0: 686f 6f73 6520 746f 2061 6464 206f 6e65  hoose to add one
-000133b0: 206d 6f72 6520 6c61 7965 7220 6f66 2065   more layer of e
-000133c0: 6c6d 6d20 6d6f 6465 6c69 6e67 206f 6e20  lmm modeling on 
-000133d0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-000133e0: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-000133f0: 7320 616e 642f 6f72 2067 6d6d 206d 6f64  s and/or gmm mod
-00013400: 656c 696e 6720 6f6e 2061 6e63 686f 7220  eling on anchor 
-00013410: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013420: 7661 6c75 6573 2e20 5468 6520 6f72 7468  values. The orth
-00013430: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00013440: 533c 2f73 7562 3e20 7661 6c75 6573 2063  S</sub> values c
-00013450: 616e 2062 6520 6361 6c63 756c 6174 6564  an be calculated
-00013460: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013470: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-00013480: 6764 206b 7364 2077 6764 5f67 6c6f 6261  gd ksd wgd_globa
-00013490: 6c6d 7262 682f 676c 6f62 616c 5f4d 5242  lmrbh/global_MRB
-000134a0: 482e 7473 7620 7365 7173 2a20 2d6f 2077  H.tsv seqs* -o w
-000134b0: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-000134c0: 0a60 6060 0a0a 5769 7468 2074 6865 2063  .```..With the c
-000134d0: 616c 6375 6c61 7465 6420 6f72 7468 6f6c  alculated orthol
-000134e0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-000134f0: 2f73 7562 3e20 7461 626c 652c 2077 6520  /sub> table, we 
-00013500: 6361 6e20 7573 6520 7468 6520 636f 6d6d  can use the comm
-00013510: 616e 6420 6265 6c6f 7720 746f 2063 6f6e  and below to con
-00013520: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
-00013530: 7272 6563 7469 6f6e 2061 6e64 2f6f 7220  rrection and/or 
-00013540: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00013550: 2061 6e61 6c79 7369 732e 0a0a 6060 600a   analysis...```.
-00013560: 7767 6420 7669 7a20 2d64 2077 6764 5f67  wgd viz -d wgd_g
-00013570: 6c6f 6261 6c6d 7262 685f 6b73 2f67 6c6f  lobalmrbh_ks/glo
-00013580: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
-00013590: 7473 7620 2d66 6120 4171 7569 6c65 6769  tsv -fa Aquilegi
-000135a0: 615f 636f 6572 756c 6561 202d 6570 6b20  a_coerulea -epk 
-000135b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000135c0: 6561 2e6b 732e 7473 7620 2d61 7020 616e  ea.ks.tsv -ap an
-000135d0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
-000135e0: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
-000135f0: 7720 2d6f 2077 6764 5f76 697a 5f6d 6978  w -o wgd_viz_mix
-00013600: 6564 5f4b 7320 2d2d 706c 6f74 656c 6d6d  ed_Ks --plotelmm
-00013610: 202d 2d70 6c6f 7461 7067 6d6d 0a60 6060   --plotapgmm.```
-00013620: 0a0a 6060 600a 7767 6420 6b73 6420 7767  ..```.wgd ksd wg
-00013630: 645f 676c 6f62 616c 6d72 6268 2f67 6c6f  d_globalmrbh/glo
-00013640: 6261 6c5f 4d52 4248 2e74 7376 2073 6571  bal_MRBH.tsv seq
-00013650: 732a 202d 2d65 7874 7261 7061 7261 6e6f  s* --extraparano
-00013660: 6d65 6b73 2077 6764 5f6b 7364 2f41 7175  meks wgd_ksd/Aqu
-00013670: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00013680: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00013690: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-000136a0: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
-000136b0: 676c 6f62 616c 6d72 6268 5f6b 7320 2d2d  globalmrbh_ks --
-000136c0: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-000136d0: 5f63 6f65 7275 6c65 613b 5072 6f74 6561  _coerulea;Protea
-000136e0: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-000136f0: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00013700: 636f 6572 756c 6561 3b56 6974 6973 5f76  coerulea;Vitis_v
-00013710: 696e 6966 6572 6122 202d 2d73 7061 6972  inifera" --spair
-00013720: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00013730: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
-00013740: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
-00013750: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00013760: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-00013770: 6572 756c 6561 2220 2d2d 706c 6f74 6b64  erulea" --plotkd
-00013780: 6520 282d 6170 2077 6764 5f73 796e 2f69  e (-ap wgd_syn/i
-00013790: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-000137a0: 7270 6f69 6e74 732e 7478 7429 0a60 6060  rpoints.txt).```
-000137b0: 0a0a 5468 6520 6669 6c65 2060 7370 6563  ..The file `spec
-000137c0: 6965 7374 7265 652e 6e77 6020 6973 2074  iestree.nw` is t
-000137d0: 6865 2074 6578 7420 6669 6c65 206f 6620  he text file of 
-000137e0: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
-000137f0: 6e65 7769 636b 2074 6861 7420 7261 7465  newick that rate
-00013800: 2063 6f72 7265 6374 696f 6e20 776f 756c   correction woul
-00013810: 6420 6265 2063 6f6e 6475 6374 6564 206f  d be conducted o
-00013820: 6e2e 2049 7473 2063 6f6e 7465 6e74 2069  n. Its content i
-00013830: 7320 6173 2062 656c 6f77 2e20 5573 6572  s as below. User
-00013840: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-00013850: 6520 7468 6520 7370 6563 6965 7320 7061  e the species pa
-00013860: 6972 7320 746f 2062 6520 706c 6f74 7465  irs to be plotte
-00013870: 642e 2057 6520 7375 6767 6573 7420 6164  d. We suggest ad
-00013880: 6469 6e67 2074 6865 206f 7074 696f 6e20  ding the option 
-00013890: 602d 2d72 6577 6569 6768 7460 2074 6f20  `--reweight` to 
-000138a0: 7265 6361 6c63 756c 6174 6520 7468 6520  recalculate the 
-000138b0: 7765 6967 6874 2070 6572 2073 7065 6369  weight per speci
-000138c0: 6573 2070 6169 7220 7375 6368 2074 6861  es pair such tha
-000138d0: 7420 7468 6520 7765 6967 6874 206f 6620  t the weight of 
-000138e0: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
-000138f0: 2070 6169 7273 2077 696c 6c20 6265 636f   pairs will beco
-00013900: 6d65 2031 2061 7320 7468 6520 7061 7261  me 1 as the para
-00013910: 6c6f 676f 7573 2067 656e 6520 7061 6972  logous gene pair
-00013920: 732e 2054 6865 2066 6c61 6720 602d 2d70  s. The flag `--p
-00013930: 6c6f 746b 6465 6020 6361 6e20 6265 2061  lotkde` can be a
-00013940: 6464 6564 2077 6865 6e20 7468 6520 6b64  dded when the kd
-00013950: 6520 6375 7276 6520 6f66 206f 7274 686f  e curve of ortho
-00013960: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00013970: 3c2f 7375 623e 2069 7320 6465 7369 7265  </sub> is desire
-00013980: 642e 2045 7874 7261 2063 6f6c 6c69 6e65  d. Extra colline
-00013990: 6172 2064 6174 6120 6361 6e20 6265 2061  ar data can be a
-000139a0: 6464 6564 2062 7920 7468 6520 6f70 7469  dded by the opti
-000139b0: 6f6e 2060 2d61 7060 2e0a 0a60 6060 0a28  on `-ap`...```.(
-000139c0: 2828 5669 7469 735f 7669 6e69 6665 7261  ((Vitis_vinifera
-000139d0: 2c50 726f 7465 615f 6379 6e61 726f 6964  ,Protea_cynaroid
-000139e0: 6573 292c 4171 7569 6c65 6769 615f 636f  es),Aquilegia_co
-000139f0: 6572 756c 6561 292c 4163 6f72 7573 5f61  erulea),Acorus_a
-00013a00: 6d65 7269 6361 6e75 7329 3b0a 6060 600a  mericanus);.```.
-00013a10: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00013a20: 6769 615f 636f 6572 756c 6561 5f47 6c6f  gia_coerulea_Glo
-00013a30: 6261 6c6d 7262 684b 735f 436f 7272 6563  balmrbhKs_Correc
-00013a40: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
-00013a50: 2073 686f 776e 2061 626f 7665 2c20 6265   shown above, be
-00013a60: 6361 7573 6520 6f66 2074 6865 2068 6967  cause of the hig
-00013a70: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
-00013a80: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
-00013a90: 6769 6120 636f 6572 756c 6561 2a2c 2074  gia coerulea*, t
-00013aa0: 6865 206f 7269 6769 6e61 6c20 6f72 7468  he original orth
-00013ab0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00013ac0: 533c 2f73 7562 3e20 7661 6c75 6573 2077  S</sub> values w
-00013ad0: 6572 6520 6163 7475 616c 6c79 2075 6e64  ere actually und
-00013ae0: 6572 6573 7469 6d61 7465 6420 696e 2074  erestimated in t
-00013af0: 6865 2074 696d 652d 6672 616d 6520 6f66  he time-frame of
-00013b00: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-00013b10: 756c 6561 2a2e 2057 6865 6e20 7765 2072  ulea*. When we r
-00013b20: 6563 6f76 6572 6564 2074 6865 2064 6976  ecovered the div
-00013b30: 6572 6765 6e63 6520 7375 6273 7469 7475  ergence substitu
-00013b40: 7469 6f6e 2064 6973 7461 6e63 6520 696e  tion distance in
-00013b50: 2074 6572 6d73 206f 6620 7477 6f20 7469   terms of two ti
-00013b60: 6d65 7320 6f66 2074 6865 2062 7261 6e63  mes of the branc
-00013b70: 682d 7370 6563 6966 6963 2063 6f6e 7472  h-specific contr
-00013b80: 6962 7574 696f 6e20 6f66 202a 4171 7569  ibution of *Aqui
-00013b90: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00013ba0: 7369 6e63 6520 6974 7320 6469 7665 7267  since its diverg
-00013bb0: 656e 6365 2077 6974 6820 7468 6520 7369  ence with the si
-00013bc0: 7374 6572 2073 7065 6369 6573 2070 6c75  ster species plu
-00013bd0: 7320 7468 6520 7368 6172 6564 2073 7562  s the shared sub
-00013be0: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
-00013bf0: 6365 2062 6566 6f72 6520 6469 7665 7267  ce before diverg
-00013c00: 656e 6365 2028 696e 2072 656c 6174 6976  ence (in relativ
-00013c10: 6520 746f 2074 6865 206f 7574 6772 6f75  e to the outgrou
-00013c20: 7029 2c20 7468 6520 636f 7272 6563 7465  p), the correcte
-00013c30: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
-00013c40: 3e20 6d6f 6465 2062 6563 616d 6520 6c61  > mode became la
-00013c50: 7267 6572 2e0a 0a49 6620 6f6e 6520 6861  rger...If one ha
-00013c60: 6420 7468 6520 6f72 7468 6f6c 6f67 6f75  d the orthologou
-00013c70: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
-00013c80: 3e20 6461 7461 2061 6c72 6561 6479 2c20  > data already, 
-00013c90: 6f6e 6520 636f 756c 6420 616c 736f 2061  one could also a
-00013ca0: 7070 6c79 2074 6865 2070 726f 6772 616d  pply the program
-00013cb0: 2060 7767 6420 7669 7a60 2074 6f20 636f   `wgd viz` to co
-00013cc0: 6e64 7563 7420 7468 6520 7261 7465 2063  nduct the rate c
-00013cd0: 6f72 7265 6374 696f 6e20 616e 616c 7973  orrection analys
-00013ce0: 6973 2075 7369 6e67 2074 6865 2063 6f6d  is using the com
-00013cf0: 6d61 6e64 2062 656c 6f77 2e20 4e6f 7465  mand below. Note
-00013d00: 2074 6861 7420 7468 6520 6f72 6465 7220   that the order 
-00013d10: 6f66 2067 6976 656e 2060 7370 6169 7260  of given `spair`
-00013d20: 206f 7074 696f 6e73 2064 6563 6964 6573   options decides
-00013d30: 2074 6865 2063 6f6c 6f72 206f 6620 7468   the color of th
-00013d40: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00013d50: 3e20 6469 7374 7269 6275 7469 6f6e 206f  > distribution o
-00013d60: 6620 6561 6368 2073 7065 6369 6573 2070  f each species p
-00013d70: 6169 722e 2054 6865 2061 6464 6974 696f  air. The additio
-00013d80: 6e61 6c20 6f70 7469 6f6e 2060 666f 6375  nal option `focu
-00013d90: 7332 616c 6c60 2069 6e20 6077 6764 206b  s2all` in `wgd k
-00013da0: 7364 6020 616e 6420 6077 6764 2076 697a  sd` and `wgd viz
-00013db0: 6020 6361 6e20 6265 2075 7365 6420 746f  ` can be used to
-00013dc0: 2074 656c 6c20 7468 6520 7072 6f67 7261   tell the progra
-00013dd0: 6d20 7468 6174 2074 6865 2073 7065 6369  m that the speci
-00013de0: 6573 2070 6169 7273 2061 7265 2062 6574  es pairs are bet
-00013df0: 7765 656e 2074 6865 2066 6f63 7573 2073  ween the focus s
-00013e00: 7065 6369 6573 2061 6e64 2061 6c6c 2074  pecies and all t
-00013e10: 6865 206f 7468 6572 2073 7065 6369 6573  he other species
-00013e20: 2073 7563 6820 7468 6174 2075 7365 7273   such that users
-00013e30: 2064 6f6e 2774 206e 6565 6420 746f 2074   don't need to t
-00013e40: 7970 6520 696e 2065 6163 6820 7061 6972  ype in each pair
-00013e50: 2069 6e64 6976 6964 7561 6c6c 792e 0a0a   individually...
-00013e60: 6060 600a 7767 6420 7669 7a20 2d64 2077  ```.wgd viz -d w
-00013e70: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-00013e80: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
-00013e90: 2e6b 732e 7473 7620 2d2d 6578 7472 6170  .ks.tsv --extrap
-00013ea0: 6172 616e 6f6d 656b 7320 7767 645f 6b73  aranomeks wgd_ks
-00013eb0: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00013ec0: 756c 6561 2e74 7376 2e6b 732e 7473 7620  ulea.tsv.ks.tsv 
-00013ed0: 2d73 7020 7370 6563 6965 7374 7265 652e  -sp speciestree.
-00013ee0: 6e77 202d 2d72 6577 6569 6768 7420 2d61  nw --reweight -a
-00013ef0: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-00013f00: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00013f10: 7473 2e74 7874 202d 6f20 7767 645f 7669  ts.txt -o wgd_vi
-00013f20: 7a5f 6d69 7865 645f 4b73 202d 2d73 7061  z_mixed_Ks --spa
-00013f30: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00013f40: 6572 756c 6561 3b50 726f 7465 615f 6379  erulea;Protea_cy
-00013f50: 6e61 726f 6964 6573 2220 2d2d 7370 6169  naroides" --spai
-00013f60: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-00013f70: 7275 6c65 613b 5669 7469 735f 7669 6e69  rulea;Vitis_vini
-00013f80: 6665 7261 2220 2d2d 7370 6169 7220 2241  fera" --spair "A
-00013f90: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00013fa0: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-00013fb0: 6e75 7322 202d 2d73 7061 6972 2022 4171  nus" --spair "Aq
-00013fc0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00013fd0: 3b41 7175 696c 6567 6961 5f63 6f65 7275  ;Aquilegia_coeru
-00013fe0: 6c65 6122 202d 2d70 6c6f 746b 6465 2028  lea" --plotkde (
-00013ff0: 2d2d 6773 6d61 7020 6765 6e65 5f73 7065  --gsmap gene_spe
-00014000: 6369 6573 2e6d 6170 290a 6060 600a 0a57  cies.map).```..W
-00014010: 6974 6820 7468 6520 6f70 7469 6f6e 2060  ith the option `
-00014020: 666f 6375 7332 616c 6c60 2c20 7468 6520  focus2all`, the 
-00014030: 636f 6d6d 616e 6420 6361 6e20 6265 2061  command can be a
-00014040: 6c73 6f20 6c69 6b65 2074 6869 732c 2073  lso like this, s
-00014050: 616d 6520 7769 7468 2060 7767 6420 6b73  ame with `wgd ks
-00014060: 6460 2e0a 0a60 6060 0a77 6764 2076 697a  d`...```.wgd viz
-00014070: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
-00014080: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
-00014090: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
-000140a0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-000140b0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-000140c0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-000140d0: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-000140e0: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-000140f0: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
-00014100: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00014110: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
-00014120: 6764 5f76 697a 5f6d 6978 6564 5f4b 7320  gd_viz_mixed_Ks 
-00014130: 2d2d 666f 6375 7332 616c 6c20 4171 7569  --focus2all Aqui
-00014140: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
-00014150: 2d70 6c6f 746b 6465 0a60 6060 0a0a 4e6f  -plotkde.```..No
-00014160: 7465 2074 6861 7420 7765 2063 616e 2065  te that we can e
-00014170: 6173 696c 7920 7368 6f77 2074 6861 7420  asily show that 
-00014180: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00014190: 6c65 612a 2068 6173 2068 6967 6865 7220  lea* has higher 
-000141a0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-000141b0: 6520 7468 616e 202a 5072 6f74 6561 2063  e than *Protea c
-000141c0: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
-000141d0: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
-000141e0: 6279 2063 6f6d 7061 7269 6e67 2074 6865  by comparing the
-000141f0: 6972 2073 7562 7374 6974 7574 696f 6e20  ir substitution 
-00014200: 6469 7374 616e 6365 2069 6e20 7265 6761  distance in rega
-00014210: 7264 2074 6f20 7468 6520 7361 6d65 2064  rd to the same d
-00014220: 6976 6572 6765 6e63 6520 6576 656e 7420  ivergence event 
-00014230: 7769 7468 206f 7574 6772 6f75 7020 7370  with outgroup sp
-00014240: 6563 6965 7320 2a41 636f 7275 735f 616d  ecies *Acorus_am
-00014250: 6572 6963 616e 7573 2a2c 2075 7369 6e67  ericanus*, using
-00014260: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
-00014270: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
-00014280: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
-00014290: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
-000142a0: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-000142b0: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-000142c0: 7765 6967 6874 202d 6f20 7767 645f 7669  weight -o wgd_vi
-000142d0: 7a5f 436f 6d70 6172 655f 7261 7465 202d  z_Compare_rate -
-000142e0: 2d73 7061 6972 2022 4163 6f72 7573 5f61  -spair "Acorus_a
-000142f0: 6d65 7269 6361 6e75 733b 5072 6f74 6561  mericanus;Protea
-00014300: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00014310: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00014320: 636f 6572 756c 6561 3b41 636f 7275 735f  coerulea;Acorus_
-00014330: 616d 6572 6963 616e 7573 2220 2d2d 7370  americanus" --sp
-00014340: 6169 7220 2256 6974 6973 5f76 696e 6966  air "Vitis_vinif
-00014350: 6572 613b 4163 6f72 7573 5f61 6d65 7269  era;Acorus_ameri
-00014360: 6361 6e75 7322 202d 2d67 736d 6170 2067  canus" --gsmap g
-00014370: 656e 655f 7370 6563 6965 732e 6d61 7020  ene_species.map 
-00014380: 2d2d 706c 6f74 6b64 650a 6060 600a 0a21  --plotkde.```..!
-00014390: 5b5d 2864 6174 612f 5261 775f 4f72 7468  [](data/Raw_Orth
-000143a0: 6f6c 6f67 7565 735f 436f 6d70 6172 655f  ologues_Compare_
-000143b0: 7261 7465 2e6b 7364 2e73 7667 290a 0a41  rate.ksd.svg)..A
-000143c0: 7320 6469 7370 6c61 7965 6420 6162 6f76  s displayed abov
-000143d0: 652c 2074 6865 206f 7274 686f 6c6f 676f  e, the orthologo
-000143e0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-000143f0: 623e 2076 616c 7565 7320 6265 7774 6565  b> values bewtee
-00014400: 6e20 2a41 7175 696c 6567 6961 2063 6f65  n *Aquilegia coe
-00014410: 7275 6c65 612a 2061 6e64 202a 4163 6f72  rulea* and *Acor
-00014420: 7573 2061 6d65 7269 6361 6e75 732a 2068  us americanus* h
-00014430: 6173 2074 6865 2068 6967 6865 7374 206d  as the highest m
-00014440: 6f64 652c 2069 6e64 6963 6174 696e 6774  ode, indicatingt
-00014450: 6865 2066 6173 7465 7220 7375 6273 7469  he faster substi
-00014460: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
-00014470: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00014480: 6561 2a20 636f 6d70 6172 6564 2074 6f20  ea* compared to 
-00014490: 2a50 726f 7465 6120 6379 6e61 726f 6964  *Protea cynaroid
-000144a0: 6573 2a20 616e 6420 2a56 6974 6973 2076  es* and *Vitis v
-000144b0: 696e 6966 6572 612a 2e0a 0a42 6566 6f72  inifera*...Befor
-000144c0: 6520 7632 2e30 2e32 312c 2074 6865 2067  e v2.0.21, the g
-000144d0: 656e 652d 7370 6563 6965 7320 6d61 7020  ene-species map 
-000144e0: 6669 6c65 2069 7320 6e65 6363 6573 7361  file is neccessa
-000144f0: 7269 6c79 206e 6565 6465 6420 666f 7220  rily needed for 
-00014500: 6974 7320 696d 706c 656d 656e 7461 7469  its implementati
-00014510: 6f6e 2069 6e20 6077 6764 2076 697a 602c  on in `wgd viz`,
-00014520: 2077 6869 6368 2073 686f 756c 6420 6265   which should be
-00014530: 2061 7574 6f6d 6174 656c 7920 7072 6f64   automately prod
-00014540: 7563 6564 2062 7920 7468 6520 6c61 7374  uced by the last
-00014550: 2060 7767 6420 6b73 6460 2073 7465 7020   `wgd ksd` step 
-00014560: 6769 7665 6e20 7468 6520 6073 7061 6972  given the `spair
-00014570: 6020 616e 6420 6073 7065 6369 6573 7472  ` and `speciestr
-00014580: 6565 6020 7061 7261 6d65 7465 7273 2e20  ee` parameters. 
-00014590: 5468 6520 6067 656e 655f 7370 6563 6965  The `gene_specie
-000145a0: 732e 6d61 7060 2068 6173 2063 6f6e 7465  s.map` has conte
-000145b0: 6e74 7320 6173 2062 656c 6f77 2069 6e20  nts as below in 
-000145c0: 7768 6963 6820 6561 6368 206c 696e 6520  which each line 
-000145d0: 6973 2074 6865 206a 6f69 6e65 6420 7374  is the joined st
-000145e0: 7269 6e67 206f 6620 6765 6e65 206e 616d  ring of gene nam
-000145f0: 6520 616e 6420 7370 6563 6965 7320 6e61  e and species na
-00014600: 6d65 2062 7920 7370 6163 652e 2041 6674  me by space. Aft
-00014610: 6572 2076 322e 302e 3231 2028 696e 636c  er v2.0.21 (incl
-00014620: 7564 6564 292c 2074 6865 2067 656e 652d  uded), the gene-
-00014630: 7370 6563 6965 7320 6d61 7020 6669 6c65  species map file
-00014640: 2069 7320 6e6f 7420 6e65 6363 6573 7361   is not neccessa
-00014650: 7269 6c79 206e 6565 6465 6420 616e 796d  rily needed anym
-00014660: 6f72 652e 0a0a 6060 600a 4171 636f 6536  ore...```.Aqcoe6
-00014670: 4730 3537 3830 302e 3120 4171 7569 6c65  G057800.1 Aquile
-00014680: 6769 615f 636f 6572 756c 6561 0a56 7669  gia_coerulea.Vvi
-00014690: 5f56 4954 5f32 3031 7330 3031 3167 3031  _VIT_201s0011g01
-000146a0: 3533 302e 3120 5669 7469 735f 7669 6e69  530.1 Vitis_vini
-000146b0: 6665 7261 0a50 6379 5f50 726f 6379 3031  fera.Pcy_Procy01
-000146c0: 6730 3835 3130 2050 726f 7465 615f 6379  g08510 Protea_cy
-000146d0: 6e61 726f 6964 6573 0a41 616d 5f41 636f  naroides.Aam_Aco
-000146e0: 7261 2e30 3447 3134 3239 3030 2e31 2041  ra.04G142900.1 A
-000146f0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00014700: 0a60 6060 0a0a 4120 6d6f 7265 2063 6f6d  .```..A more com
-00014710: 706c 6578 2070 6c6f 7420 6361 6e20 6265  plex plot can be
-00014720: 206d 6164 6520 6279 2061 6464 2074 6865   made by add the
-00014730: 2066 6c61 6720 602d 2d70 6c6f 7465 6c6d   flag `--plotelm
-00014740: 6d60 2073 7563 6820 7468 6174 2074 6865  m` such that the
-00014750: 2045 4c4d 4d20 6d69 7874 7572 6520 6d6f   ELMM mixture mo
-00014760: 6465 6c69 6e67 206f 6620 7072 6f76 6964  deling of provid
-00014770: 6564 2070 6172 616e 6f6d 6520 2a4b 2a3c  ed paranome *K*<
-00014780: 7375 623e 533c 2f73 7562 3e20 6361 6e20  sub>S</sub> can 
-00014790: 6265 2073 7570 6572 696d 706f 7365 642c  be superimposed,
-000147a0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-000147b0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-000147c0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-000147d0: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-000147e0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-000147f0: 7376 202d 2d65 7874 7261 7061 7261 6e6f  sv --extraparano
-00014800: 6d65 6b73 2077 6764 5f6b 7364 2f41 7175  meks wgd_ksd/Aqu
-00014810: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00014820: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00014830: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00014840: 7265 7765 6967 6874 202d 6170 2077 6764  reweight -ap wgd
-00014850: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00014860: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
-00014870: 7420 2d6f 2077 6764 5f76 697a 5f6d 6978  t -o wgd_viz_mix
-00014880: 6564 5f4b 735f 656c 6d6d 202d 2d73 7061  ed_Ks_elmm --spa
-00014890: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-000148a0: 6572 756c 6561 3b50 726f 7465 615f 6379  erulea;Protea_cy
-000148b0: 6e61 726f 6964 6573 2220 2d2d 7370 6169  naroides" --spai
-000148c0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-000148d0: 7275 6c65 613b 5669 7469 735f 7669 6e69  rulea;Vitis_vini
-000148e0: 6665 7261 2220 2d2d 7370 6169 7220 2241  fera" --spair "A
-000148f0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014900: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-00014910: 6e75 7322 202d 2d73 7061 6972 2022 4171  nus" --spair "Aq
-00014920: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014930: 3b41 7175 696c 6567 6961 5f63 6f65 7275  ;Aquilegia_coeru
-00014940: 6c65 6122 202d 2d67 736d 6170 2067 656e  lea" --gsmap gen
-00014950: 655f 7370 6563 6965 732e 6d61 7020 2d2d  e_species.map --
-00014960: 706c 6f74 6b64 6520 2d2d 706c 6f74 656c  plotkde --plotel
-00014970: 6d6d 0a60 6060 0a0a 215b 5d28 6461 7461  mm.```..![](data
-00014980: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00014990: 6c65 615f 476c 6f62 616c 6d72 6268 4b73  lea_GlobalmrbhKs
-000149a0: 5f45 6c6d 6d5f 436f 7272 6563 7465 642e  _Elmm_Corrected.
-000149b0: 6b73 642e 7376 6729 0a0a 4672 6f6d 2074  ksd.svg)..From t
-000149c0: 6865 206d 6978 6564 202a 4b2a 3c73 7562  he mixed *K*<sub
-000149d0: 3e53 3c2f 7375 623e 2070 6c6f 7420 6162  >S</sub> plot ab
-000149e0: 6f76 652c 2077 6520 6361 6e20 7365 6520  ove, we can see 
-000149f0: 7468 6174 2074 6865 206f 7074 696d 697a  that the optimiz
-00014a00: 6564 206c 6f67 6e6f 726d 616c 2063 6f6d  ed lognormal com
-00014a10: 706f 6e65 6e74 2062 2077 6974 6820 6d6f  ponent b with mo
-00014a20: 6465 2031 2e32 2069 7320 796f 756e 6765  de 1.2 is younge
-00014a30: 7220 7468 616e 2074 6865 2063 6f72 7265  r than the corre
-00014a40: 6374 6564 206f 7274 686f 6c6f 676f 7573  cted orthologous
-00014a50: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00014a60: 206d 6f64 6520 7769 7468 202a 5072 6f74   mode with *Prot
-00014a70: 6561 2063 796e 6172 6f69 6465 732a 2061  ea cynaroides* a
-00014a80: 6e64 202a 5669 7469 7320 7669 6e69 6665  nd *Vitis vinife
-00014a90: 7261 2a20 2831 2e33 3920 616e 6420 312e  ra* (1.39 and 1.
-00014aa0: 3437 2c20 7265 7370 6563 7469 7665 6c79  47, respectively
-00014ab0: 292e 0a0a 4265 7369 6465 732c 2077 6520  )...Besides, we 
-00014ac0: 6361 6e20 616c 736f 2061 6464 2074 6865  can also add the
-00014ad0: 2047 4d4d 206d 6978 7475 7265 206d 6f64   GMM mixture mod
-00014ae0: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
-00014af0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00014b00: 7661 6c75 6573 2077 6974 6820 7468 6520  values with the 
-00014b10: 666c 6167 2060 2d2d 706c 6f74 6170 676d  flag `--plotapgm
-00014b20: 6d60 2c20 7573 696e 6720 7468 6520 636f  m`, using the co
-00014b30: 6d6d 616e 6420 6265 6c6f 772e 0a0a 6060  mmand below...``
-00014b40: 600a 7767 6420 7669 7a20 2d64 2077 6764  `.wgd viz -d wgd
-00014b50: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
-00014b60: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
-00014b70: 732e 7473 7620 2d2d 6578 7472 6170 6172  s.tsv --extrapar
-00014b80: 616e 6f6d 656b 7320 7767 645f 6b73 642f  anomeks wgd_ksd/
-00014b90: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00014ba0: 6561 2e74 7376 2e6b 732e 7473 7620 2d73  ea.tsv.ks.tsv -s
-00014bb0: 7020 7370 6563 6965 7374 7265 652e 6e77  p speciestree.nw
-00014bc0: 202d 2d72 6577 6569 6768 7420 2d61 7020   --reweight -ap 
-00014bd0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00014be0: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-00014bf0: 2e74 7874 202d 6f20 7767 645f 7669 7a5f  .txt -o wgd_viz_
-00014c00: 6d69 7865 645f 4b73 5f65 6c6d 6d20 2d2d  mixed_Ks_elmm --
-00014c10: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00014c20: 5f63 6f65 7275 6c65 613b 5072 6f74 6561  _coerulea;Protea
-00014c30: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00014c40: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00014c50: 636f 6572 756c 6561 3b56 6974 6973 5f76  coerulea;Vitis_v
-00014c60: 696e 6966 6572 6122 202d 2d73 7061 6972  inifera" --spair
-00014c70: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014c80: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
-00014c90: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
-00014ca0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00014cb0: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-00014cc0: 6572 756c 6561 2220 2d2d 6773 6d61 7020  erulea" --gsmap 
-00014cd0: 6765 6e65 5f73 7065 6369 6573 2e6d 6170  gene_species.map
-00014ce0: 202d 2d70 6c6f 746b 6465 202d 2d70 6c6f   --plotkde --plo
-00014cf0: 7461 7067 6d6d 6d0a 6060 600a 0a21 5b5d  tapgmmm.```..![]
-00014d00: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
-00014d10: 636f 6572 756c 6561 5f47 6c6f 6261 6c6d  coerulea_Globalm
-00014d20: 7262 684b 735f 4170 676d 6d5f 436f 7272  rbhKs_Apgmm_Corr
-00014d30: 6563 7465 642e 6b73 642e 7376 6729 0a0a  ected.ksd.svg)..
-00014d40: 4173 206d 616e 6966 6573 7465 6420 6162  As manifested ab
-00014d50: 6f76 652c 2074 6865 2061 6e63 686f 7220  ove, the anchor 
-00014d60: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00014d70: 636f 6d70 6f6e 656e 7420 3220 7769 7468  component 2 with
-00014d80: 206d 6f64 6520 312e 3238 2069 7320 616c   mode 1.28 is al
-00014d90: 736f 2079 6f75 6e67 6572 2074 6861 6e20  so younger than 
-00014da0: 7468 6520 636f 7272 6563 7465 6420 6f72  the corrected or
-00014db0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-00014dc0: 623e 533c 2f73 7562 3e20 6d6f 6465 2077  b>S</sub> mode w
-00014dd0: 6974 6820 2a50 726f 7465 6120 6379 6e61  ith *Protea cyna
-00014de0: 726f 6964 6573 2a20 616e 6420 2a56 6974  roides* and *Vit
-00014df0: 6973 2076 696e 6966 6572 612a 2e20 4275  is vinifera*. Bu
-00014e00: 7420 7765 206e 6565 6420 746f 2062 6520  t we need to be 
-00014e10: 6f66 2063 6f75 7273 6520 6361 7574 696f  of course cautio
-00014e20: 7573 2074 6861 7420 7375 6368 2064 6973  us that such dis
-00014e30: 7469 6e63 7469 6f6e 2063 6f6d 6573 2077  tinction comes w
-00014e40: 6974 6820 7468 6520 756e 6365 7274 6169  ith the uncertai
-00014e50: 6e74 6965 7320 696e 7472 6f64 7563 6564  nties introduced
-00014e60: 2066 726f 6d20 7468 6520 6170 706c 6965   from the applie
-00014e70: 6420 6d69 7874 7572 6520 6d6f 6465 6c69  d mixture modeli
-00014e80: 6e67 206d 6574 686f 646f 6c6f 6779 2069  ng methodology i
-00014e90: 6e20 7465 726d 7320 6f66 2066 6f72 2069  n terms of for i
-00014ea0: 6e73 7461 6e63 6520 6469 6666 6572 656e  nstance differen
-00014eb0: 7420 696e 6974 6961 6c69 7a61 7469 6f6e  t initialization
-00014ec0: 2070 6f69 6e74 7320 616e 6420 7468 6520   points and the 
-00014ed0: 6973 7375 6520 6f66 206f 7665 7266 6974  issue of overfit
-00014ee0: 7469 6e67 2061 6e64 2074 6865 2073 6973  ting and the sis
-00014ef0: 7465 7220 7370 6563 6965 7373 2061 646f  ter speciess ado
-00014f00: 7074 6564 2069 6e20 7468 6174 2074 6865  pted in that the
-00014f10: 7265 206d 6967 6874 2062 6520 7370 6563  re might be spec
-00014f20: 6965 7320 7769 7468 206d 6f72 6520 6469  ies with more di
-00014f30: 7370 6172 6174 6520 7375 6273 7469 7475  sparate substitu
-00014f40: 7469 6f6e 2072 6174 6573 2074 6861 6e20  tion rates than 
-00014f50: 7468 6520 6f6e 6520 7765 2063 686f 7365  the one we chose
-00014f60: 2e0a 0a41 6464 696e 6720 626f 7468 2045  ...Adding both E
-00014f70: 4c4d 4d20 7265 7375 6c74 2066 6f72 2070  LMM result for p
-00014f80: 6172 616e 6f6d 6520 616e 6420 474d 4d20  aranome and GMM 
-00014f90: 7265 7375 6c74 2066 6f72 2061 6e63 686f  result for ancho
-00014fa0: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00014fb0: 3e20 6361 6e20 6265 2061 6368 6965 7665  > can be achieve
-00014fc0: 6420 6a75 7374 2062 7920 6164 6420 7468  d just by add th
-00014fd0: 6520 7477 6f20 666c 6167 7320 6d65 6e74  e two flags ment
-00014fe0: 696f 6e65 6420 6162 6f76 652c 2075 7369  ioned above, usi
-00014ff0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015000: 656c 6f77 2e0a 0a60 6060 0a77 6764 2076  elow...```.wgd v
-00015010: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
-00015020: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
-00015030: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
-00015040: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00015050: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-00015060: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00015070: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-00015080: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015090: 6967 6874 202d 6170 2077 6764 5f73 796e  ight -ap wgd_syn
-000150a0: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
-000150b0: 686f 7270 6f69 6e74 732e 7478 7420 2d6f  horpoints.txt -o
-000150c0: 2077 6764 5f76 697a 5f6d 6978 6564 5f4b   wgd_viz_mixed_K
-000150d0: 735f 656c 6d6d 202d 2d73 7061 6972 2022  s_elmm --spair "
-000150e0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000150f0: 6561 3b50 726f 7465 615f 6379 6e61 726f  ea;Protea_cynaro
-00015100: 6964 6573 2220 2d2d 7370 6169 7220 2241  ides" --spair "A
-00015110: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015120: 613b 5669 7469 735f 7669 6e69 6665 7261  a;Vitis_vinifera
-00015130: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
-00015140: 6567 6961 5f63 6f65 7275 6c65 613b 4163  egia_coerulea;Ac
-00015150: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
-00015160: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00015170: 6769 615f 636f 6572 756c 6561 3b41 7175  gia_coerulea;Aqu
-00015180: 696c 6567 6961 5f63 6f65 7275 6c65 6122  ilegia_coerulea"
-00015190: 202d 2d67 736d 6170 2067 656e 655f 7370   --gsmap gene_sp
-000151a0: 6563 6965 732e 6d61 7020 2d2d 706c 6f74  ecies.map --plot
-000151b0: 6b64 6520 2d2d 706c 6f74 656c 6d6d 202d  kde --plotelmm -
-000151c0: 2d70 6c6f 7461 7067 6d6d 6d0a 6060 600a  -plotapgmmm.```.
-000151d0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-000151e0: 6769 615f 636f 6572 756c 6561 5f47 6c6f  gia_coerulea_Glo
-000151f0: 6261 6c6d 7262 684b 735f 456c 6d6d 5f41  balmrbhKs_Elmm_A
-00015200: 7067 6d6d 5f43 6f72 7265 6374 6564 2e6b  pgmm_Corrected.k
-00015210: 7364 2e73 7667 290a 0a41 6e20 616c 7465  sd.svg)..An alte
-00015220: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
-00015230: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
-00015240: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00015250: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
-00015260: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
-00015270: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
-00015280: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
-00015290: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
-000152a0: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
-000152b0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
-000152c0: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
-000152d0: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
-000152e0: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
-000152f0: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
-00015300: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
-00015310: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
-00015320: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
-00015330: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
-00015340: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
-00015350: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015360: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00015370: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00015380: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
-00015390: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
-000153a0: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
-000153b0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
-000153c0: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
-000153d0: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
-000153e0: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
-000153f0: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
-00015400: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
-00015410: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
-00015420: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
-00015430: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
-00015440: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
-00015450: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
-00015460: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
-00015470: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
-00015480: 6c20 4d52 4248 2066 616d 696c 7920 6578  l MRBH family ex
-00015490: 6365 7074 2074 6861 7420 7765 2064 6f6e  cept that we don
-000154a0: 2774 2075 7365 2074 6865 2065 7874 7261  't use the extra
-000154b0: 2070 7265 2d69 6e66 6572 7265 6420 7061   pre-inferred pa
-000154c0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-000154d0: 3c2f 7375 623e 2061 6e79 6d6f 7265 2e20  </sub> anymore. 
-000154e0: 5765 2069 6e66 6572 2074 6865 202a 4b2a  We infer the *K*
-000154f0: 3c73 7562 3e53 3c2f 7375 623e 2075 7369  <sub>S</sub> usi
-00015500: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015510: 656c 6f77 2e20 4e6f 7465 2074 6861 7420  elow. Note that 
-00015520: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
-00015530: 2076 697a 6020 6361 6e20 706c 6f74 2074   viz` can plot t
-00015540: 6865 2073 616d 6520 6a75 7374 2061 7320  he same just as 
-00015550: 7368 6f77 6e20 6162 6f76 652e 0a0a 6060  shown above...``
-00015560: 600a 7767 6420 6b73 6420 7767 645f 6f72  `.wgd ksd wgd_or
-00015570: 7468 6f2f 4f72 7468 6f67 726f 7570 732e  tho/Orthogroups.
-00015580: 7370 2e74 7376 202d 7370 2073 7065 6369  sp.tsv -sp speci
-00015590: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-000155a0: 6967 6874 202d 6f20 7767 645f 6f72 7468  ight -o wgd_orth
-000155b0: 6f5f 6b73 202d 2d73 7061 6972 2022 4171  o_ks --spair "Aq
-000155c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000155d0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
-000155e0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
-000155f0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015600: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
-00015610: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00015620: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
-00015630: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015640: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015650: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
-00015660: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
-00015670: 2d70 6c6f 746b 6465 2028 2d61 7020 7767  -plotkde (-ap wg
-00015680: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00015690: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
-000156a0: 7874 290a 6060 600a 0a21 5b5d 2864 6174  xt).```..![](dat
-000156b0: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-000156c0: 756c 6561 5f4f 7274 686f 4b73 5f43 6f72  ulea_OrthoKs_Cor
-000156d0: 7265 6374 6564 2e6b 7364 2e73 7667 290a  rected.ksd.svg).
-000156e0: 0a41 7320 7368 6f77 6e20 6162 6f76 652c  .As shown above,
-000156f0: 2074 6865 206e 756d 6265 7220 6f66 2062   the number of b
-00015700: 6f74 6820 7061 7261 6c6f 676f 7573 2067  oth paralogous g
-00015710: 656e 6520 7061 6972 7320 616e 6420 6f72  ene pairs and or
-00015720: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-00015730: 6169 7273 2069 7320 6469 6666 6572 656e  airs is differen
-00015740: 7420 7468 616e 2074 6865 206f 6e65 2066  t than the one f
-00015750: 726f 6d20 676c 6f62 616c 204d 5242 4820  rom global MRBH 
-00015760: 6661 6d69 6c79 2069 6e20 7468 6174 2068  family in that h
-00015770: 6572 6520 7765 2070 6c6f 7474 6564 2061  ere we plotted a
-00015780: 6c6c 206f 7274 686f 6c6f 676f 7573 2067  ll orthologous g
-00015790: 656e 6520 7061 6972 7320 696e 7374 6561  ene pairs instea
-000157a0: 6420 6f66 206f 6e6c 7920 676c 6f62 616c  d of only global
-000157b0: 204d 5242 4820 616e 6420 706f 7465 6e74   MRBH and potent
-000157c0: 6961 6c6c 7920 6e65 7720 7061 7261 6c6f  ially new paralo
-000157d0: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
-000157e0: 6d69 6768 7420 6265 2070 726f 6475 6365  might be produce
-000157f0: 6420 696e 2074 6865 206f 7274 686f 6772  d in the orthogr
-00015800: 6f75 7020 696e 6665 7265 6e63 6520 7374  oup inference st
-00015810: 6570 2c20 746f 6765 7468 6572 2077 6974  ep, together wit
-00015820: 6820 6469 6666 6572 656e 7420 7265 6361  h different reca
-00015830: 6c63 756c 6174 6564 2077 6569 6768 7473  lculated weights
-00015840: 2e0a 0a49 6620 6f6e 6520 6f6e 6c79 2077  ...If one only w
-00015850: 616e 7465 6420 746f 2070 6c6f 7420 7468  anted to plot th
-00015860: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00015870: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00015880: 7374 7269 6275 7469 6f6e 2c20 6173 2077  stribution, as w
-00015890: 6520 6861 7665 2061 6c72 6561 6479 2073  e have already s
-000158a0: 686f 776e 2061 626f 7665 2c20 6974 2063  hown above, it c
-000158b0: 616e 2062 6520 6561 7369 6c79 2061 6368  an be easily ach
-000158c0: 6965 7665 6420 6279 2072 656d 6f76 696e  ieved by removin
-000158d0: 6720 7468 6520 7061 7261 6c6f 676f 7573  g the paralogous
-000158e0: 2073 7065 6369 6573 2070 6169 7220 6041   species pair `A
-000158f0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015900: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00015910: 756c 6561 602c 2075 7369 6e67 2074 6865  ulea`, using the
-00015920: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
-00015930: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
-00015940: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
-00015950: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
-00015960: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00015970: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00015980: 7765 6967 6874 202d 6f20 7767 645f 676c  weight -o wgd_gl
-00015990: 6f62 616c 6d72 6268 5f6f 6e6c 796f 7274  obalmrbh_onlyort
-000159a0: 686f 5f6b 7320 2d2d 7370 6169 7220 2241  ho_ks --spair "A
-000159b0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000159c0: 613b 5072 6f74 6561 5f63 796e 6172 6f69  a;Protea_cynaroi
-000159d0: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
-000159e0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000159f0: 3b56 6974 6973 5f76 696e 6966 6572 6122  ;Vitis_vinifera"
-00015a00: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00015a10: 6769 615f 636f 6572 756c 6561 3b41 636f  gia_coerulea;Aco
-00015a20: 7275 735f 616d 6572 6963 616e 7573 2220  rus_americanus" 
-00015a30: 2d2d 6773 6d61 7020 6765 6e65 5f73 7065  --gsmap gene_spe
-00015a40: 6369 6573 2e6d 6170 0a60 6060 0a0a 5765  cies.map.```..We
-00015a50: 2063 616e 2063 6c65 6172 6c79 2073 6565   can clearly see
-00015a60: 2074 6861 7420 2a56 6974 6973 2076 696e   that *Vitis vin
-00015a70: 6966 6572 612a 2068 6173 2068 6967 6865  ifera* has highe
-00015a80: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
-00015a90: 6174 6520 7468 616e 202a 5072 6f74 6561  ate than *Protea
-00015aa0: 2063 796e 6172 6f69 6465 732a 2069 6e20   cynaroides* in 
-00015ab0: 7468 6174 2074 6865 6972 206f 7274 686f  that their ortho
-00015ac0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00015ad0: 3c2f 7375 623e 2070 6561 6b73 2077 6974  </sub> peaks wit
-00015ae0: 6820 2a41 7175 696c 6567 6961 2063 6f65  h *Aquilegia coe
-00015af0: 7275 6c65 612a 2c20 616c 7468 6f75 6768  rulea*, although
-00015b00: 2072 6570 7265 7365 6e74 696e 6720 7468   representing th
-00015b10: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
-00015b20: 6520 6576 656e 742c 2064 6966 6665 7265  e event, differe
-00015b30: 6420 696e 2073 7562 7374 6974 7574 696f  d in substitutio
-00015b40: 6e20 6469 7374 616e 6365 2e0a 215b 5d28  n distance..![](
-00015b50: 6461 7461 2f52 6177 5f4f 7274 686f 6c6f  data/Raw_Ortholo
-00015b60: 6775 6573 2e6b 7364 2e73 7667 290a 0a41  gues.ksd.svg)..A
-00015b70: 6674 6572 2074 6865 2070 6879 6c6f 6765  fter the phyloge
-00015b80: 6e65 7469 6320 7469 6d69 6e67 206f 6620  netic timing of 
-00015b90: 7468 6520 5261 6e75 6e63 756c 616c 6573  the Ranunculales
-00015ba0: 2057 4744 2c20 7765 2063 616e 2066 7572   WGD, we can fur
-00015bb0: 7468 6572 2069 6e66 6572 2069 7473 2061  ther infer its a
-00015bc0: 6273 6f6c 7574 6520 6167 652e 2046 6972  bsolute age. Fir
-00015bd0: 7374 2077 6520 696e 6665 7220 7468 6520  st we infer the 
-00015be0: 6372 6564 6962 6c65 2072 616e 6765 206f  credible range o
-00015bf0: 6620 616e 6368 6f72 2070 6169 7273 2062  f anchor pairs b
-00015c00: 7920 2a4b 2a3c 7375 623e 533c 2f73 7562  y *K*<sub>S</sub
-00015c10: 3e20 6865 7572 6973 7469 6361 6c6c 7920  > heuristically 
-00015c20: 7573 696e 6720 7468 6520 7072 6f67 7261  using the progra
-00015c30: 6d20 6077 6764 2070 6561 6b60 2e0a 0a60  m `wgd peak`...`
-00015c40: 6060 0a77 6764 2070 6561 6b20 2d2d 6865  ``.wgd peak --he
-00015c50: 7572 6973 7469 6320 7767 645f 6b73 642f  uristic wgd_ksd/
-00015c60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015c70: 6561 2e74 7376 2e6b 732e 7473 7620 2d61  ea.tsv.ks.tsv -a
-00015c80: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-00015c90: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00015ca0: 7473 2e74 7874 202d 736d 2077 6764 5f73  ts.txt -sm wgd_s
-00015cb0: 796e 2f69 6164 686f 7265 2d6f 7574 2f73  yn/iadhore-out/s
-00015cc0: 6567 6d65 6e74 732e 7478 7420 2d6c 6520  egments.txt -le 
-00015cd0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00015ce0: 6f75 742f 6c69 7374 5f65 6c65 6d65 6e74  out/list_element
-00015cf0: 732e 7478 7420 2d6d 7020 7767 645f 7379  s.txt -mp wgd_sy
-00015d00: 6e2f 6961 6468 6f72 652d 6f75 742f 6d75  n/iadhore-out/mu
-00015d10: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
-00015d20: 7478 7420 2d6f 2077 6764 5f70 6561 6b0a  txt -o wgd_peak.
-00015d30: 6060 600a 0a21 5b5d 2864 6174 612f 416e  ```..![](data/An
-00015d40: 6368 6f72 4b73 5f50 6561 6b43 495f 4171  chorKs_PeakCI_Aq
-00015d50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015d60: 2e74 7376 2e6b 732e 7473 765f 6e6f 6465  .tsv.ks.tsv_node
-00015d70: 5f77 6569 6768 7465 642e 7376 6729 0a0a  _weighted.svg)..
-00015d80: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
-00015d90: 7765 2061 7373 756d 6564 2061 206c 6f67  we assumed a log
-00015da0: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-00015db0: 696f 6e20 6174 2074 6865 2070 6561 6b20  ion at the peak 
-00015dc0: 6c6f 6361 7469 6f6e 2064 6574 6563 7465  location detecte
-00015dd0: 6420 6279 2074 6865 2060 7369 676e 616c  d by the `signal
-00015de0: 6020 6d6f 6475 6c65 206f 6620 6073 6369  ` module of `sci
-00015df0: 7079 6020 6c69 6272 6172 792e 2054 6865  py` library. The
-00015e00: 2039 3525 2063 6f6e 6669 6465 6e63 6520   95% confidence 
-00015e10: 6c65 7665 6c20 6f66 2074 6865 206c 6f67  level of the log
-00015e20: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-00015e30: 696f 6e20 7761 7320 6170 706c 6965 642c  ion was applied,
-00015e40: 2069 2e65 2e2c 2030 2e36 382d 322e 3734   i.e., 0.68-2.74
-00015e50: 2c20 696e 2066 7572 7468 6572 206d 6f6c  , in further mol
-00015e60: 6563 756c 6172 2064 6174 696e 672e 2054  ecular dating. T
-00015e70: 6865 2066 696c 6520 6041 7175 696c 6567  he file `Aquileg
-00015e80: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00015e90: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
-00015ea0: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
-00015eb0: 7465 645f 666f 726d 6174 2e74 7376 6020  ted_format.tsv` 
-00015ec0: 6973 2077 6861 7420 7765 206e 6565 6420  is what we need 
-00015ed0: 666f 7220 6e65 7874 2073 7465 702e 2054  for next step. T
-00015ee0: 6f20 6275 696c 6420 7468 6520 6f72 7468  o build the orth
-00015ef0: 6f67 726f 7570 7320 7573 6564 2069 6e20  ogroups used in 
-00015f00: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
-00015f10: 696e 672c 2077 6520 6e65 6564 2074 6f20  ing, we need to 
-00015f20: 7365 6c65 6374 2073 6f6d 6520 7370 6563  select some spec
-00015f30: 6965 7320 616e 6420 666f 726d 2061 2073  ies and form a s
-00015f40: 7461 7274 696e 6720 7472 6565 2077 6974  tarting tree wit
-00015f50: 6820 7072 6f70 6572 2066 6f73 7369 6c20  h proper fossil 
-00015f60: 6361 6c69 6272 6174 696f 6e73 2e20 5765  calibrations. We
-00015f70: 2070 726f 7669 6465 206f 6e65 2069 6e20   provide one in 
-00015f80: 6d63 6d63 7472 6565 2066 6f72 6d61 7420  mcmctree format 
-00015f90: 6173 2062 656c 6f77 2e0a 0a60 6060 0a31  as below...```.1
-00015fa0: 3720 310a 2828 2828 506f 7461 6d6f 6765  7 1.((((Potamoge
-00015fb0: 746f 6e5f 6163 7574 6966 6f6c 6975 732c  ton_acutifolius,
-00015fc0: 2853 7069 726f 6465 6c61 5f69 6e74 6572  (Spirodela_inter
-00015fd0: 6d65 6469 612c 416d 6f72 7068 6f70 6861  media,Amorphopha
-00015fe0: 6c6c 7573 5f6b 6f6e 6a61 6329 292c 2841  llus_konjac)),(A
-00015ff0: 6361 6e74 686f 6368 6c61 6d79 735f 6272  canthochlamys_br
-00016000: 6163 7465 6174 612c 2844 696f 7363 6f72  acteata,(Dioscor
-00016010: 6561 5f61 6c61 7461 2c44 696f 7363 6f72  ea_alata,Dioscor
-00016020: 6561 5f72 6f74 756e 6461 7461 2929 273e  ea_rotundata))'>
-00016030: 302e 3536 3030 3c31 2e32 3836 3327 2927  0.5600<1.2863')'
-00016040: 3e30 2e38 3336 303c 312e 3238 3633 272c  >0.8360<1.2863',
-00016050: 2841 636f 7275 735f 616d 6572 6963 616e  (Acorus_american
-00016060: 7573 2c41 636f 7275 735f 7461 7461 7269  us,Acorus_tatari
-00016070: 6e6f 7769 6929 2927 3e30 2e38 3336 303c  nowii))'>0.8360<
-00016080: 312e 3238 3633 272c 2828 2828 5465 7472  1.2863',((((Tetr
-00016090: 6163 656e 7472 6f6e 5f73 696e 656e 7365  acentron_sinense
-000160a0: 2c54 726f 6368 6f64 656e 6472 6f6e 5f61  ,Trochodendron_a
-000160b0: 7261 6c69 6f69 6465 7329 2c28 4275 7875  ralioides),(Buxu
-000160c0: 735f 6175 7374 726f 7975 6e6e 616e 656e  s_austroyunnanen
-000160d0: 7369 732c 4275 7875 735f 7369 6e69 6361  sis,Buxus_sinica
-000160e0: 2929 273e 312e 3130 3830 3c31 2e32 3836  ))'>1.1080<1.286
-000160f0: 3327 2c28 4e65 6c75 6d62 6f5f 6e75 6369  3',(Nelumbo_nuci
-00016100: 6665 7261 2c28 5465 6c6f 7065 615f 7370  fera,(Telopea_sp
-00016110: 6563 696f 7369 7373 696d 612c 5072 6f74  eciosissima,Prot
-00016120: 6561 5f63 796e 6172 6f69 6465 7329 2929  ea_cynaroides)))
-00016130: 273e 312e 3130 3830 3c31 2e32 3836 3327  '>1.1080<1.2863'
-00016140: 2c28 4171 7569 6c65 6769 615f 636f 6572  ,(Aquilegia_coer
-00016150: 756c 6561 5f61 7031 2c41 7175 696c 6567  ulea_ap1,Aquileg
-00016160: 6961 5f63 6f65 7275 6c65 615f 6170 3229  ia_coerulea_ap2)
-00016170: 2927 3e31 2e31 3038 303c 312e 3238 3633  )'>1.1080<1.2863
-00016180: 2729 273e 312e 3237 3230 3c32 2e34 3732  ')'>1.2720<2.472
-00016190: 3027 3b0a 6060 600a 0a41 7320 7072 6573  0';.```..As pres
-000161a0: 656e 7465 6420 6162 6f76 652c 2074 6865  ented above, the
-000161b0: 2066 6f63 7573 2073 7065 6369 6573 2074   focus species t
-000161c0: 6861 7420 6973 2061 626f 7574 2074 6f20  hat is about to 
-000161d0: 6265 2064 6174 6564 206e 6565 6473 2074  be dated needs t
-000161e0: 6f20 6265 2072 6570 6c61 6365 6420 7769  o be replaced wi
-000161f0: 7468 2060 2841 7175 696c 6567 6961 5f63  th `(Aquilegia_c
-00016200: 6f65 7275 6c65 615f 6170 312c 4171 7569  oerulea_ap1,Aqui
-00016210: 6c65 6769 615f 636f 6572 756c 6561 5f61  legia_coerulea_a
-00016220: 7032 2960 2e20 5769 7468 2074 6869 7320  p2)`. With this 
-00016230: 7374 6172 7469 6e67 2074 7265 6520 616e  starting tree an
-00016240: 6420 7072 6564 6f77 6e6c 6f61 6465 6420  d predownloaded 
-00016250: 6364 7320 6669 6c65 7320 6f66 2061 6c6c  cds files of all
-00016260: 2074 6865 2073 7065 6369 6573 2c20 7765   the species, we
-00016270: 2063 616e 2062 7569 6c64 2074 6865 206f   can build the o
-00016280: 7274 686f 6772 6f75 7020 7573 6564 2069  rthogroup used i
-00016290: 6e20 7468 6520 6669 6e61 6c20 6d6f 6c65  n the final mole
-000162a0: 6375 6c61 7220 6461 7469 6e67 2075 7369  cular dating usi
-000162b0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2061  ng the command a
-000162c0: 7320 6265 6c6f 772e 0a0a 6060 600a 7767  s below...```.wg
-000162d0: 6420 646d 6420 2d66 2041 7175 696c 6567  d dmd -f Aquileg
-000162e0: 6961 5f63 6f65 7275 6c65 6120 2d61 7020  ia_coerulea -ap 
-000162f0: 7767 645f 7065 616b 2f41 7175 696c 6567  wgd_peak/Aquileg
-00016300: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00016310: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
-00016320: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
-00016330: 7465 645f 666f 726d 6174 2e74 7376 202d  ted_format.tsv -
-00016340: 6f20 7767 645f 646d 645f 6f72 7468 6f20  o wgd_dmd_ortho 
-00016350: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
-00016360: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
-00016370: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
-00016380: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
-00016390: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
-000163a0: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
-000163b0: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
-000163c0: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
-000163d0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-000163e0: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
-000163f0: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
-00016400: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
-00016410: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
-00016420: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
-00016430: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
-00016440: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
-00016450: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
-00016460: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
-00016470: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-00016480: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00016490: 6c65 610a 6060 600a 0a54 6865 2072 6573  lea.```..The res
-000164a0: 756c 7420 6669 6c65 2060 6d65 7267 655f  ult file `merge_
-000164b0: 666f 6375 735f 6170 2e74 7376 6020 6973  focus_ap.tsv` is
-000164c0: 2077 6861 7420 7765 206e 6565 6420 666f   what we need fo
-000164d0: 7220 7468 6520 6669 6e61 6c20 7374 6570  r the final step
-000164e0: 206f 6620 6d6f 6c65 6375 6c61 7220 6461   of molecular da
-000164f0: 7469 6e67 2069 6e20 7072 6f67 7261 6d20  ting in program 
-00016500: 6077 6764 2066 6f63 7573 602e 0a0a 6060  `wgd focus`...``
-00016510: 600a 7767 6420 666f 6375 7320 2d2d 7072  `.wgd focus --pr
-00016520: 6f74 6461 7469 6e67 202d 2d61 616d 6f64  otdating --aamod
-00016530: 656c 206c 6720 7767 645f 646d 645f 6f72  el lg wgd_dmd_or
-00016540: 7468 6f2f 6d65 7267 655f 666f 6375 735f  tho/merge_focus_
-00016550: 6170 2e74 7376 202d 7370 2064 6174 696e  ap.tsv -sp datin
-00016560: 675f 7472 6565 2e6e 7720 2d6f 2077 6764  g_tree.nw -o wgd
-00016570: 5f64 6174 696e 6720 2d64 206d 636d 6374  _dating -d mcmct
-00016580: 7265 6520 2d64 7320 2762 7572 6e69 6e20  ree -ds 'burnin 
-00016590: 3d20 3230 3030 2720 2d64 7320 2773 616d  = 2000' -ds 'sam
-000165a0: 7066 7265 7120 3d20 3130 3030 2720 2d64  pfreq = 1000' -d
-000165b0: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
-000165c0: 3030 2720 506f 7461 6d6f 6765 746f 6e5f  00' Potamogeton_
-000165d0: 6163 7574 6966 6f6c 6975 7320 5370 6972  acutifolius Spir
-000165e0: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
-000165f0: 2041 6d6f 7270 686f 7068 616c 6c75 735f   Amorphophallus_
-00016600: 6b6f 6e6a 6163 2041 6361 6e74 686f 6368  konjac Acanthoch
-00016610: 6c61 6d79 735f 6272 6163 7465 6174 6120  lamys_bracteata 
-00016620: 4469 6f73 636f 7265 615f 616c 6174 6120  Dioscorea_alata 
-00016630: 4469 6f73 636f 7265 615f 726f 7475 6e64  Dioscorea_rotund
-00016640: 6174 6120 4163 6f72 7573 5f61 6d65 7269  ata Acorus_ameri
-00016650: 6361 6e75 7320 4163 6f72 7573 5f74 6174  canus Acorus_tat
-00016660: 6172 696e 6f77 6969 2054 6574 7261 6365  arinowii Tetrace
-00016670: 6e74 726f 6e5f 7369 6e65 6e73 6520 5472  ntron_sinense Tr
-00016680: 6f63 686f 6465 6e64 726f 6e5f 6172 616c  ochodendron_aral
-00016690: 696f 6964 6573 2042 7578 7573 5f61 7573  ioides Buxus_aus
-000166a0: 7472 6f79 756e 6e61 6e65 6e73 6973 2042  troyunnanensis B
-000166b0: 7578 7573 5f73 696e 6963 6120 4e65 6c75  uxus_sinica Nelu
-000166c0: 6d62 6f5f 6e75 6369 6665 7261 2054 656c  mbo_nucifera Tel
-000166d0: 6f70 6561 5f73 7065 6369 6f73 6973 7369  opea_speciosissi
-000166e0: 6d61 2050 726f 7465 615f 6379 6e61 726f  ma Protea_cynaro
-000166f0: 6964 6573 2041 7175 696c 6567 6961 5f63  ides Aquilegia_c
-00016700: 6f65 7275 6c65 610a 6060 600a 0a48 6572  oerulea.```..Her
-00016710: 6520 7765 206f 6e6c 7920 696d 706c 656d  e we only implem
-00016720: 656e 7465 6420 7468 6520 636f 6e63 6174  ented the concat
-00016730: 656e 6174 696f 6e20 616e 616c 7973 6973  enation analysis
-00016740: 2075 7369 6e67 2070 726f 7465 696e 2073   using protein s
-00016750: 6571 7565 6e63 6520 6279 2061 6464 696e  equence by addin
-00016760: 6720 7468 6520 666c 6167 2060 2d2d 7072  g the flag `--pr
-00016770: 6f74 6461 7469 6e67 6020 616e 6420 7765  otdating` and we
-00016780: 2073 6574 2074 6865 2070 6172 616d 6574   set the paramet
-00016790: 6572 2066 6f72 2060 6d63 6d63 7472 6565  er for `mcmctree
-000167a0: 6020 7669 6120 7468 6520 6f70 7469 6f6e  ` via the option
-000167b0: 2060 2d64 7360 2e20 4e6f 7465 2074 6861   `-ds`. Note tha
-000167c0: 7420 6f74 6865 7220 6461 7469 6e67 2070  t other dating p
-000167d0: 726f 6772 616d 2073 7563 6820 6173 2060  rogram such as `
-000167e0: 7238 7360 2061 6e64 2060 6265 6173 7460  r8s` and `beast`
-000167f0: 2061 7265 2061 6c73 6f20 6176 6169 6c61   are also availa
-00016800: 626c 6520 6769 7665 6e20 736f 6d65 206d  ble given some m
-00016810: 616e 6461 746f 7279 2070 6172 616d 6574  andatory paramet
-00016820: 6572 732e 2054 6865 2066 696e 616c 206c  ers. The final l
-00016830: 6f67 206f 6620 7468 6520 7375 6363 6573  og of the succes
-00016840: 7366 756c 2072 756e 2069 7320 6173 2062  sful run is as b
-00016850: 656c 6f77 2e0a 0a60 6060 0a31 363a 3034  elow...```.16:04
-00016860: 3a32 3520 494e 464f 2020 2020 2052 756e  :25 INFO     Run
-00016870: 6e69 6e67 206d 636d 6374 7265 6520 7573  ning mcmctree us
-00016880: 696e 6720 4865 7373 6961 6e20 6d61 7472  ing Hessian matr
-00016890: 6978 206f 6620 4c47 2b47 616d 6d61 2020  ix of LG+Gamma  
-000168a0: 636f 7265 2e70 793a 3936 370a 2020 2020  core.py:967.    
-000168b0: 2020 2020 2020 2020 2020 2020 2020 666f                fo
-000168c0: 7220 7072 6f74 6569 6e20 6d6f 6465 6c0a  r protein model.
-000168d0: 3233 3a34 393a 3337 2049 4e46 4f20 2020  23:49:37 INFO   
-000168e0: 2020 506f 7374 6572 696f 7220 6d65 616e    Posterior mean
-000168f0: 2066 6f72 2074 6865 2061 6765 7320 6f66   for the ages of
-00016900: 2077 6764 2069 7320 3131 322e 3839 3435   wgd is 112.8945
-00016910: 206d 636d 6374 7265 652e 7079 3a32 3936   mcmctree.py:296
-00016920: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00016930: 2020 206d 696c 6c69 6f6e 2079 6561 7273     million years
-00016940: 2066 726f 6d20 436f 6e63 6174 656e 6174   from Concatenat
-00016950: 6564 2070 6570 7469 6465 0a20 2020 2020  ed peptide.     
-00016960: 2020 2020 2020 2020 2020 2020 2061 6c69               ali
-00016970: 676e 6d65 6e74 2061 6e64 2039 3525 2063  gnment and 95% c
-00016980: 7265 6469 6269 6c69 7479 2069 6e74 6572  redibility inter
-00016990: 7661 6c73 2028 4349 290a 2020 2020 2020  vals (CI).      
-000169a0: 2020 2020 2020 2020 2020 2020 6973 2031              is 1
-000169b0: 3031 2e32 3234 2d31 3233 2e31 3231 206d  01.224-123.121 m
-000169c0: 696c 6c69 6f6e 2079 6561 7273 0a20 2020  illion years.   
-000169d0: 2020 2020 2020 494e 464f 2020 2020 2054        INFO     T
-000169e0: 6f74 616c 2072 756e 2074 696d 653a 2032  otal run time: 2
-000169f0: 3931 3735 7320 2020 2020 2020 2020 2020  9175s           
-00016a00: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016a10: 2020 2063 6c69 2e70 793a 3234 310a 2020     cli.py:241.  
-00016a20: 2020 2020 2020 2049 4e46 4f20 2020 2020         INFO     
-00016a30: 446f 6e65 2020 2020 2020 2020 2020 2020  Done            
-00016a40: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016a50: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016a60: 2020 2020 636c 692e 7079 3a32 3432 0a60      cli.py:242.`
-00016a70: 6060 0a0a 546f 2076 6973 7561 6c69 7a65  ``..To visualize
-00016a80: 2074 6865 2064 6174 652c 2077 6520 616c   the date, we al
-00016a90: 736f 2070 726f 7669 6465 6420 6120 7079  so provided a py
-00016aa0: 7468 6f6e 2073 6372 6970 7420 746f 2070  thon script to p
-00016ab0: 6c6f 7420 7468 6520 5747 4420 6461 7465  lot the WGD date
-00016ac0: 7320 696e 2074 6865 2060 7767 6460 2066  s in the `wgd` f
-00016ad0: 6f6c 6465 722e 2055 7365 7273 206e 6565  older. Users nee
-00016ae0: 6420 746f 2065 7874 7261 6374 2074 6865  d to extract the
-00016af0: 2072 6177 2064 6174 6573 2066 726f 6d20   raw dates from 
-00016b00: 7468 6520 606d 636d 632e 7478 7460 2066  the `mcmc.txt` f
-00016b10: 6f72 2074 6865 2057 4744 206e 6f64 6520  or the WGD node 
-00016b20: 6669 7273 7420 616e 6420 7361 7665 2069  first and save i
-00016b30: 7420 6173 2066 696c 6520 6064 6174 6573  t as file `dates
-00016b40: 2e74 7874 6020 286f 7220 7768 6174 6576  .txt` (or whatev
-00016b50: 6572 2070 7265 6665 7272 6564 206e 616d  er preferred nam
-00016b60: 6529 2e20 416e 2065 7861 6d70 6c65 2063  e). An example c
-00016b70: 6f6d 6d61 6e64 2069 7320 6173 2062 656c  ommand is as bel
-00016b80: 6f77 2e0a 0a60 6060 0a70 7974 686f 6e20  ow...```.python 
-00016b90: 2450 4154 482f 706f 7374 706c 6f74 2e70  $PATH/postplot.p
-00016ba0: 7920 706f 7374 6469 7320 6461 7465 732e  y postdis dates.
-00016bb0: 7478 7420 2d2d 7065 7263 656e 7469 6c65  txt --percentile
-00016bc0: 2039 3020 2d2d 7469 746c 6520 2257 4744   90 --title "WGD
-00016bd0: 2064 6174 6522 202d 2d68 7064 202d 6f20   date" --hpd -o 
-00016be0: 2252 616e 756e 6375 6c61 6c65 735f 5747  "Ranunculales_WG
-00016bf0: 445f 6461 7465 2e73 7667 220a 6060 600a  D_date.svg".```.
-00016c00: 0a21 5b5d 2864 6174 612f 5261 6e75 6e63  .![](data/Ranunc
-00016c10: 756c 616c 6573 5f57 4744 5f64 6174 652e  ulales_WGD_date.
-00016c20: 7376 6729 0a0a 5468 6520 706f 7374 6572  svg)..The poster
-00016c30: 696f 7220 6d65 616e 2c20 6d65 6469 616e  ior mean, median
-00016c40: 2061 6e64 206d 6f64 6520 6f66 2074 6865   and mode of the
-00016c50: 2052 616e 756e 6375 6c61 6c65 7320 5747   Ranunculales WG
-00016c60: 4420 6167 6520 6973 2031 3132 2e39 322c  D age is 112.92,
-00016c70: 2031 3133 2e34 3420 616e 6420 3131 322e   113.44 and 112.
-00016c80: 3534 206d 7961 2c20 7769 7468 2039 3025  54 mya, with 90%
-00016c90: 2048 5044 2031 3035 2e30 3720 2d20 3132   HPD 105.07 - 12
-00016ca0: 322e 3332 206d 7961 2061 7320 6d61 6e69  2.32 mya as mani
-00016cb0: 6665 7374 6564 2061 626f 7665 2e0a 0a23  fested above...#
-00016cc0: 2323 204b 7374 7265 650a 0a49 6e20 6164  ## Kstree..In ad
-00016cd0: 6469 7469 6f6e 2074 6f20 7061 6972 7769  dition to pairwi
-00016ce0: 7365 202a 4b2a 3c73 7562 3e53 3c2f 7375  se *K*<sub>S</su
-00016cf0: 623e 2065 7374 696d 6174 696f 6e2c 2061  b> estimation, a
-00016d00: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00016d10: 2074 7265 6520 7769 7468 2062 7261 6e63   tree with branc
-00016d20: 6820 6c65 6e67 7468 2069 6e20 2a4b 2a3c  h length in *K*<
-00016d30: 7375 623e 533c 2f73 7562 3e20 756e 6974  sub>S</sub> unit
-00016d40: 2063 616e 2061 6c73 6f20 6265 2064 6572   can also be der
-00016d50: 6976 6564 2066 726f 6d20 7468 6520 7072  ived from the pr
-00016d60: 6f67 7261 6d20 6077 6764 206b 7364 6020  ogram `wgd ksd` 
-00016d70: 6769 7665 6e20 7468 6520 6f70 7469 6f6e  given the option
-00016d80: 2060 2d2d 6b73 7472 6565 6020 616e 6420   `--kstree` and 
-00016d90: 602d 2d73 7065 6369 6573 7472 6565 602e  `--speciestree`.
-00016da0: 204e 6f74 6520 7468 6174 2074 6865 2061   Note that the a
-00016db0: 6464 6974 696f 6e61 6c20 6f70 7469 6f6e  dditional option
-00016dc0: 2060 2d2d 6f6e 6c79 636f 6e63 6174 6b73   `--onlyconcatks
-00016dd0: 7472 6565 6020 7769 6c6c 206f 6e6c 7920  tree` will only 
-00016de0: 6361 6c6c 2074 6865 202a 4b2a 3c73 7562  call the *K*<sub
-00016df0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-00016e00: 696f 6e20 666f 7220 7468 6520 636f 6e63  ion for the conc
-00016e10: 6174 656e 6174 6564 2061 6c69 676e 6d65  atenated alignme
-00016e20: 6e74 2072 6174 6865 7220 7468 616e 2061  nt rather than a
-00016e30: 6c6c 2074 6865 2061 6c69 676e 6d65 6e74  ll the alignment
-00016e40: 732e 2055 7365 7273 206e 6565 6420 746f  s. Users need to
-00016e50: 2070 726f 7669 6465 2061 2070 7265 7365   provide a prese
-00016e60: 7420 7370 6563 6965 7320 7472 6565 2066  t species tree f
-00016e70: 6f72 2074 6865 202a 4b2a 3c73 7562 3e53  or the *K*<sub>S
-00016e80: 3c2f 7375 623e 2074 7265 6520 696e 6665  </sub> tree infe
-00016e90: 7265 6e63 6520 6f66 2074 6865 2063 6f6e  rence of the con
-00016ea0: 6361 7465 6e61 7465 6420 616c 6967 6e6d  catenated alignm
-00016eb0: 656e 7420 7768 696c 6520 7468 6520 7265  ent while the re
-00016ec0: 6d61 696e 696e 6720 616c 6967 6e6d 656e  maining alignmen
-00016ed0: 7473 2077 696c 6c20 6265 2061 6761 696e  ts will be again
-00016ee0: 7374 2061 6e20 6175 746f 6d61 7465 6c79  st an automately
-00016ef0: 2069 6e66 6572 7265 6420 7472 6565 2066   inferred tree f
-00016f00: 726f 6d20 6066 6173 7474 7265 6560 206f  rom `fasttree` o
-00016f10: 7220 6069 7174 7265 6560 2e20 496e 2074  r `iqtree`. In t
-00016f20: 6865 2065 6e64 2c20 7573 6572 7320 7769  he end, users wi
-00016f30: 6c6c 2067 6574 2061 202a 4b2a 3c73 7562  ll get a *K*<sub
-00016f40: 3e53 3c2f 7375 623e 2074 7265 652c 2061  >S</sub> tree, a
-00016f50: 202a 4b2a 3c73 7562 3e41 3c2f 7375 623e   *K*<sub>A</sub>
-00016f60: 2074 7265 6520 616e 6420 6120 cf89 2074   tree and a .. t
-00016f70: 7265 6520 7065 7220 6661 6d20 616e 6420  ree per fam and 
-00016f80: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
-00016f90: 6174 6564 2061 6c69 676e 6d65 6e74 2e0a  ated alignment..
-00016fa0: 0a60 6060 0a77 6764 206b 7364 2064 6174  .```.wgd ksd dat
-00016fb0: 612f 6b73 7472 6565 5f64 6174 612f 6661  a/kstree_data/fa
-00016fc0: 6d2e 7473 7620 6461 7461 2f6b 7374 7265  m.tsv data/kstre
-00016fd0: 655f 6461 7461 2f41 636f 7275 735f 7461  e_data/Acorus_ta
-00016fe0: 7461 7269 6e6f 7769 6920 6461 7461 2f6b  tarinowii data/k
-00016ff0: 7374 7265 655f 6461 7461 2f41 6d62 6f72  stree_data/Ambor
-00017000: 656c 6c61 5f74 7269 6368 6f70 6f64 6120  ella_trichopoda 
-00017010: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
-00017020: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00017030: 6c65 6120 6461 7461 2f6b 7374 7265 655f  lea data/kstree_
-00017040: 6461 7461 2f41 7269 7374 6f6c 6f63 6869  data/Aristolochi
-00017050: 615f 6669 6d62 7269 6174 6120 6461 7461  a_fimbriata data
-00017060: 2f6b 7374 7265 655f 6461 7461 2f43 7963  /kstree_data/Cyc
-00017070: 6173 5f70 616e 7a68 6968 7561 656e 7369  as_panzhihuaensi
-00017080: 202d 2d6b 7374 7265 6520 2d2d 7370 6563   --kstree --spec
-00017090: 6965 7374 7265 6520 6461 7461 2f6b 7374  iestree data/kst
-000170a0: 7265 655f 6461 7461 2f73 7065 6369 6573  ree_data/species
-000170b0: 5f74 7265 6531 2e6e 7720 2d2d 6f6e 6c79  _tree1.nw --only
-000170c0: 636f 6e63 6174 6b73 7472 6565 202d 6f20  concatkstree -o 
-000170d0: 7767 645f 6b73 7472 6565 5f74 6f70 6f6c  wgd_kstree_topol
-000170e0: 6f67 7931 0a60 6060 0a0a 215b 5d28 6461  ogy1.```..![](da
-000170f0: 7461 2f6b 7374 7265 655f 7265 7375 6c74  ta/kstree_result
-00017100: 732f 6b73 7472 6565 2e73 7667 290a 0a41  s/kstree.svg)..A
-00017110: 626f 7665 2077 6520 7573 6564 2074 6872  bove we used thr
-00017120: 6565 2061 6c74 6572 6e61 7469 7665 2074  ee alternative t
-00017130: 6f70 6f6c 6f67 6965 7320 746f 2069 6e66  opologies to inf
-00017140: 6572 2074 6865 202a 4b2a 3c73 7562 3e53  er the *K*<sub>S
-00017150: 3c2f 7375 623e 2074 7265 6520 7768 6963  </sub> tree whic
-00017160: 6820 6c65 6420 746f 2064 6966 6665 7265  h led to differe
-00017170: 6e74 2062 7261 6e63 6820 6c65 6e67 7468  nt branch length
-00017180: 2065 7374 696d 6174 696f 6e2e 204e 6f74   estimation. Not
-00017190: 6520 7468 6174 2074 6865 2066 616d 696c  e that the famil
-000171a0: 6965 7320 7765 2075 7365 6420 7765 7265  ies we used were
-000171b0: 206f 6e6c 7920 7477 6f20 676c 6f62 616c   only two global
-000171c0: 204d 5242 4820 6661 6d69 6c69 6573 2066   MRBH families f
-000171d0: 6f72 2074 6865 2070 7572 706f 7365 206f  or the purpose o
-000171e0: 6620 696c 6c75 7374 7261 7469 6f6e 2e20  f illustration. 
-000171f0: 546f 2061 6371 7569 7265 2061 6e20 6163  To acquire an ac
-00017200: 6375 7261 7465 2070 726f 6669 6c65 206f  curate profile o
-00017210: 6620 7468 6520 7375 6273 7469 7475 7469  f the substituti
-00017220: 6f6e 2072 6174 6520 7661 7269 6174 696f  on rate variatio
-00017230: 6e2c 206f 7274 686f 6c6f 6775 6573 2061  n, orthologues a
-00017240: 7420 7468 6520 7768 6f6c 6520 6765 6e6f  t the whole geno
-00017250: 6d65 2073 6361 6c65 2073 686f 756c 6420  me scale should 
-00017260: 6265 2075 7365 642e 0a0a 2323 2043 6974  be used...## Cit
-00017270: 6174 696f 6e0a 200a 506c 6561 7365 2063  ation. .Please c
-00017280: 6974 6520 7573 2061 7420 6874 7470 733a  ite us at https:
-00017290: 2f2f 646f 692e 6f72 672f 3130 2e31 3030  //doi.org/10.100
-000172a0: 372f 3937 382d 312d 3037 3136 2d32 3536  7/978-1-0716-256
-000172b0: 312d 335f 3120 616e 6420 6874 7470 733a  1-3_1 and https:
-000172c0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-000172d0: 332f 6269 6f69 6e66 6f72 6d61 7469 6373  3/bioinformatics
-000172e0: 2f62 7461 6532 3732 2e0a 0a60 6060 0a48  /btae272...```.H
-000172f0: 656e 6763 6869 2043 6865 6e2c 2041 7274  engchi Chen, Art
-00017300: 6875 7220 5a77 6165 6e65 706f 656c 2028  hur Zwaenepoel (
-00017310: 3230 3233 292e 2049 6e66 6572 656e 6365  2023). Inference
-00017320: 206f 6620 416e 6369 656e 7420 506f 6c79   of Ancient Poly
-00017330: 706c 6f69 6479 2066 726f 6d20 4765 6e6f  ploidy from Geno
-00017340: 6d69 6320 4461 7461 2e20 496e 3a20 5661  mic Data. In: Va
-00017350: 6e20 6465 2050 6565 722c 2059 2e20 2865  n de Peer, Y. (e
-00017360: 6473 2920 506f 6c79 706c 6f69 6479 2e20  ds) Polyploidy. 
-00017370: 4d65 7468 6f64 7320 696e 204d 6f6c 6563  Methods in Molec
-00017380: 756c 6172 2042 696f 6c6f 6779 2c20 766f  ular Biology, vo
-00017390: 6c20 3235 3435 2e20 4875 6d61 6e61 2c20  l 2545. Humana, 
-000173a0: 4e65 7720 596f 726b 2c20 4e59 2e20 6874  New York, NY. ht
-000173b0: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
-000173c0: 2e31 3030 372f 3937 382d 312d 3037 3136  .1007/978-1-0716
-000173d0: 2d32 3536 312d 335f 310a 4865 6e67 6368  -2561-3_1.Hengch
-000173e0: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
-000173f0: 7761 656e 6570 6f65 6c2c 2059 7665 7320  waenepoel, Yves 
-00017400: 5661 6e20 6465 2050 6565 7220 2832 3032  Van de Peer (202
-00017410: 3429 2e20 7767 6420 7632 3a20 6120 7375  4). wgd v2: a su
-00017420: 6974 6520 6f66 2074 6f6f 6c73 2074 6f20  ite of tools to 
-00017430: 756e 636f 7665 7220 616e 6420 6461 7465  uncover and date
-00017440: 2061 6e63 6965 6e74 2070 6f6c 7970 6c6f   ancient polyplo
-00017450: 6964 7920 616e 6420 7768 6f6c 652d 6765  idy and whole-ge
-00017460: 6e6f 6d65 2064 7570 6c69 6361 7469 6f6e  nome duplication
-00017470: 2e20 4269 6f69 6e66 6f72 6d61 7469 6373  . Bioinformatics
-00017480: 2c20 6874 7470 733a 2f2f 646f 692e 6f72  , https://doi.or
-00017490: 672f 3130 2e31 3039 332f 6269 6f69 6e66  g/10.1093/bioinf
-000174a0: 6f72 6d61 7469 6373 2f62 7461 6532 3732  ormatics/btae272
-000174b0: 0a60 6060 0a0a 466f 7220 6369 7461 7469  .```..For citati
-000174c0: 6f6e 206f 6620 7468 6520 746f 6f6c 7320  on of the tools 
-000174d0: 7573 6564 2069 6e20 7767 642c 2070 6c65  used in wgd, ple
-000174e0: 6173 6520 636f 6e73 756c 7420 7468 6520  ase consult the 
-000174f0: 646f 6375 6d65 6e74 6174 696f 6e20 6174  documentation at
-00017500: 0a68 7474 7073 3a2f 2f77 6764 7632 2e72  .https://wgdv2.r
-00017510: 6561 6474 6865 646f 6373 2e69 6f2f 656e  eadthedocs.io/en
-00017520: 2f6c 6174 6573 742f 6369 7461 7469 6f6e  /latest/citation
-00017530: 2e68 746d 6c2e 0a0a                      .html...
+00007420: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+00007430: 642c 202d 2d64 6174 6166 696c 652c 2074  d, --datafile, t
+00007440: 6865 204b 7320 6461 7461 6669 6c65 2c20  he Ks datafile, 
+00007450: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
+00007460: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+00007470: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00007480: 2064 6566 6175 6c74 2077 6764 5f76 697a   default wgd_viz
+00007490: 0a2d 7372 2c20 2d2d 7370 6169 722c 2074  .-sr, --spair, t
+000074a0: 6865 2073 7065 6369 6573 2070 6169 7220  he species pair 
+000074b0: 746f 2062 6520 706c 6f74 7465 642c 2064  to be plotted, d
+000074c0: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
+000074d0: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
+000074e0: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
+000074f0: 6520 7469 6d65 730a 2d66 612c 202d 2d66  e times.-fa, --f
+00007500: 6f63 7573 3261 6c6c 2c20 7365 7420 666f  ocus2all, set fo
+00007510: 6361 6c20 7370 6563 6965 7320 616e 6420  cal species and 
+00007520: 6c65 7420 7370 6563 6965 7320 7061 6972  let species pair
+00007530: 2074 6f20 6265 2062 6574 7765 656e 2066   to be between f
+00007540: 6f63 616c 2061 6e64 2061 6c6c 2074 6865  ocal and all the
+00007550: 2072 656d 6169 6e69 6e67 2073 7065 6369   remaining speci
+00007560: 6573 2c20 6465 6661 756c 7420 4e6f 6e65  es, default None
+00007570: 0a2d 6773 2c20 2d2d 6773 6d61 702c 2074  .-gs, --gsmap, t
+00007580: 6865 2067 656e 6520 6e61 6d65 2d73 7065  he gene name-spe
+00007590: 6369 6573 206e 616d 6520 6d61 702c 2064  cies name map, d
+000075a0: 6566 6175 6c74 204e 6f6e 650a 2d73 702c  efault None.-sp,
+000075b0: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
+000075c0: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+000075d0: 2074 6f20 7065 7266 6f72 6d20 7261 7465   to perform rate
+000075e0: 2063 6f72 7265 6374 696f 6e2c 2064 6566   correction, def
+000075f0: 6175 6c74 204e 6f6e 652c 2069 6620 4e6f  ault None, if No
+00007600: 6e65 2077 6173 2067 6976 656e 2c20 7468  ne was given, th
+00007610: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+00007620: 6e20 616e 616c 7973 6973 2077 6f75 6c64  n analysis would
+00007630: 2062 6520 6361 6c6c 6564 206f 6666 0a2d   be called off.-
+00007640: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
+00007650: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00007660: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
+00007670: 6375 7276 6520 7570 6f6e 2068 6973 746f  curve upon histo
+00007680: 6772 616d 2c20 6966 2074 6865 2066 6c61  gram, if the fla
+00007690: 6720 7761 7320 7365 742c 206b 6465 2063  g was set, kde c
+000076a0: 7572 7665 2077 6f75 6c64 2062 6520 6164  urve would be ad
+000076b0: 6465 640a 2d72 772c 202d 2d72 6577 6569  ded.-rw, --rewei
+000076c0: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
+000076d0: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
+000076e0: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
+000076f0: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
+00007700: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
+00007710: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
+00007720: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
+00007730: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
+00007740: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
+00007750: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007760: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
+00007770: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
+00007780: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
+00007790: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
+000077a0: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
+000077b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000077c0: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
+000077d0: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
+000077e0: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
+000077f0: 6772 6f75 700a 2d69 7465 722c 202d 2d65  group.-iter, --e
+00007800: 6d5f 6974 6572 6174 696f 6e73 2c20 7468  m_iterations, th
+00007810: 6520 6d61 7869 6d75 6d20 454d 2069 7465  e maximum EM ite
+00007820: 7261 7469 6f6e 732c 2064 6566 6175 6c74  rations, default
+00007830: 2032 3030 0a2d 696e 6974 2c20 2d2d 656d   200.-init, --em
+00007840: 5f69 6e69 7469 616c 697a 6174 696f 6e73  _initializations
+00007850: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
+00007860: 2069 6e69 7469 616c 697a 6174 696f 6e73   initializations
+00007870: 2c20 6465 6661 756c 7420 3230 300a 2d70  , default 200.-p
+00007880: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+00007890: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+000078a0: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+000078b0: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+000078c0: 616b 732c 2064 6566 6175 6c74 2030 2e31  aks, default 0.1
+000078d0: 0a2d 7268 2c20 2d2d 7265 6c5f 6865 6967  .-rh, --rel_heig
+000078e0: 6874 2c20 7468 6520 7265 6c61 7469 7665  ht, the relative
+000078f0: 2068 6569 6768 7420 6174 2077 6869 6368   height at which
+00007900: 2074 6865 2070 6561 6b20 7769 6474 6820   the peak width 
+00007910: 6973 206d 6561 7375 7265 642c 2064 6566  is measured, def
+00007920: 6175 6c74 2030 2e34 0a2d 736d 2c20 2d2d  ault 0.4.-sm, --
+00007930: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
+00007940: 676d 656e 7473 2064 6174 6166 696c 652c  gments datafile,
+00007950: 2064 6566 6175 6c74 204e 6f6e 650a 2d6d   default None.-m
+00007960: 6c2c 202d 2d6d 696e 6c65 6e2c 2074 6865  l, --minlen, the
+00007970: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+00007980: 6f66 2061 2073 6361 6666 6f6c 6420 746f  of a scaffold to
+00007990: 2062 6520 696e 636c 7564 6564 2069 6e20   be included in 
+000079a0: 646f 7470 6c6f 742c 2064 6566 6175 6c74  dotplot, default
+000079b0: 202d 312c 2069 6620 2d31 2077 6173 2073   -1, if -1 was s
+000079c0: 6574 2c20 7468 6520 3130 2520 6f66 2074  et, the 10% of t
+000079d0: 6865 206c 6f6e 6765 7374 2073 6361 6666  he longest scaff
+000079e0: 6f6c 6473 2077 696c 6c20 6265 2073 6574  olds will be set
+000079f0: 0a2d 6d73 2c20 2d2d 6d61 7873 697a 652c  .-ms, --maxsize,
+00007a00: 2074 6865 206d 6178 696d 756d 2066 616d   the maximum fam
+00007a10: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
+00007a20: 6e63 6c75 6465 642c 2064 6566 6175 6c74  ncluded, default
+00007a30: 2032 3030 0a2d 6170 2c20 2d2d 616e 6368   200.-ap, --anch
+00007a40: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+00007a50: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+00007a60: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00007a70: 6e65 0a2d 6d74 2c20 2d2d 6d75 6c74 6970  ne.-mt, --multip
+00007a80: 6c69 636f 6e2c 2074 6865 206d 756c 7469  licon, the multi
+00007a90: 706c 6963 6f6e 7320 6461 7461 6669 6c65  plicons datafile
+00007aa0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00007ab0: 6774 2c20 2d2d 6765 6e65 7461 626c 652c  gt, --genetable,
+00007ac0: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
+00007ad0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00007ae0: 7420 4e6f 6e65 0a2d 6d67 2c20 2d2d 6d69  t None.-mg, --mi
+00007af0: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
+00007b00: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
+00007b10: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
+00007b20: 6465 2c20 696e 2072 6174 696f 2069 6620  de, in ratio if 
+00007b30: 3c3d 2031 2c20 6465 6661 756c 7420 3130  <= 1, default 10
+00007b40: 3030 300a 2d6d 676e 2c20 2d2d 6d69 6e67  000.-mgn, --ming
+00007b50: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+00007b60: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+00007b70: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+00007b80: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+00007b90: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+00007ba0: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+00007bb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00007bc0: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+00007bd0: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+00007be0: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+00007bf0: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+00007c00: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+00007c10: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+00007c20: 6570 740a 2d65 706b 2c20 2d2d 6578 7472  ept.-epk, --extr
+00007c30: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
+00007c40: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
+00007c50: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
+00007c60: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+00007c70: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
+00007c80: 7420 4e6f 6e65 0a2d 7061 672c 202d 2d70  t None.-pag, --p
+00007c90: 6c6f 7461 7067 6d6d 2c20 666c 6167 206f  lotapgmm, flag o
+00007ca0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00007cb0: 6f20 636f 6e64 7563 7420 616e 6420 706c  o conduct and pl
+00007cc0: 6f74 206d 6978 7475 7265 206d 6f64 656c  ot mixture model
+00007cd0: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+00007ce0: 2069 6e20 7468 6520 6d69 7865 6420 4b73   in the mixed Ks
+00007cf0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
+00007d00: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00007d10: 6574 2c20 7468 6520 6d69 7874 7572 6520  et, the mixture 
+00007d20: 6d6f 6465 6c69 6e67 206f 6620 616e 6368  modeling of anch
+00007d30: 6f72 204b 7320 776f 756c 6420 6265 2063  or Ks would be c
+00007d40: 6f6e 6475 6374 6564 2061 6e64 2070 6c6f  onducted and plo
+00007d50: 7474 6564 0a2d 7065 6d2c 202d 2d70 6c6f  tted.-pem, --plo
+00007d60: 7465 6c6d 6d2c 2066 6c61 6720 6f70 7469  telmm, flag opti
+00007d70: 6f6e 2c20 7768 6574 6865 7220 746f 2063  on, whether to c
+00007d80: 6f6e 6475 6374 2061 6e64 2070 6c6f 7420  onduct and plot 
+00007d90: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
+00007da0: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
+00007db0: 6520 4b73 2069 6e20 7468 6520 6d69 7865  e Ks in the mixe
+00007dc0: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
+00007dd0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
+00007de0: 6173 2073 6574 2c20 7468 6520 656c 6d6d  as set, the elmm
+00007df0: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+00007e00: 6720 6f66 2070 6172 616e 6f6d 6520 4b73  g of paranome Ks
+00007e10: 2077 6f75 6c64 2062 6520 636f 6e64 7563   would be conduc
+00007e20: 7465 6420 616e 6420 706c 6f74 7465 640a  ted and plotted.
+00007e30: 2d63 2c20 2d2d 636f 6d70 6f6e 656e 7473  -c, --components
+00007e40: 2c20 7468 6520 7261 6e67 6520 6f66 2074  , the range of t
+00007e50: 6865 206e 756d 6265 7220 6f66 2063 6f6d  he number of com
+00007e60: 706f 6e65 6e74 7320 746f 2066 6974 2069  ponents to fit i
+00007e70: 6e20 616e 6368 6f72 204b 7320 6d69 7874  n anchor Ks mixt
+00007e80: 7572 6520 6d6f 6465 6c69 6e67 2c20 6465  ure modeling, de
+00007e90: 6661 756c 7420 2831 2c34 290a 2d70 7379  fault (1,4).-psy
+00007ea0: 2c20 2d2d 706c 6f74 7379 6e2c 2066 6c61  , --plotsyn, fla
+00007eb0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007ec0: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00007ed0: 6520 7379 6e74 656e 7920 706c 6f74 2c20  e synteny plot, 
+00007ee0: 6f6e 6c79 2077 6865 6e20 7468 6520 666c  only when the fl
+00007ef0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00007f00: 7379 6e74 656e 7920 706c 6f74 2077 6f75  synteny plot wou
+00007f10: 6c64 2062 6520 7072 6f64 7563 6564 0a2d  ld be produced.-
+00007f20: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+00007f30: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+00007f40: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+00007f50: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+00007f60: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+00007f70: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+00007f80: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+00007f90: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+00007fa0: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+00007fb0: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+00007fc0: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+00007fd0: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+00007fe0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00007ff0: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+00008000: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+00008010: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+00008020: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+00008030: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+00008040: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+00008050: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+00008060: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+00008070: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+00008080: 350a 2d78 6c2c 202d 2d78 6c69 6d2c 2074  5.-xl, --xlim, t
+00008090: 6865 2078 2061 7869 7320 6c69 6d69 7420  he x axis limit 
+000080a0: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+000080b0: 6f6e 0a2d 796c 2c20 2d2d 796c 696d 2c20  on.-yl, --ylim, 
+000080c0: 7468 6520 7920 6178 6973 206c 696d 6974  the y axis limit
+000080d0: 206f 6620 4b73 2064 6973 7472 6962 7574   of Ks distribut
+000080e0: 696f 6e0a 2d61 646f 2c20 2d2d 6164 6a75  ion.-ado, --adju
+000080f0: 7374 6f72 7468 6f2c 2066 6c61 6720 6f70  stortho, flag op
+00008100: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00008110: 2061 646a 7573 7420 7468 6520 6869 7374   adjust the hist
+00008120: 6f67 7261 6d20 6865 6967 6874 206f 6620  ogram height of 
+00008130: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2061  orthologous Ks a
+00008140: 7320 746f 206d 6174 6368 2074 6865 2068  s to match the h
+00008150: 6569 6768 7420 6f66 2070 6172 616c 6f67  eight of paralog
+00008160: 6f75 7320 4b73 2c20 6966 2074 6865 2066  ous Ks, if the f
+00008170: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00008180: 2061 646a 7573 746d 656e 7420 776f 756c   adjustment woul
+00008190: 6420 6265 2063 6f6e 6475 6374 6564 0a2d  d be conducted.-
+000081a0: 6164 662c 202d 2d61 646a 7573 7466 6163  adf, --adjustfac
+000081b0: 746f 722c 2074 6865 2061 646a 7573 746d  tor, the adjustm
+000081c0: 656e 7420 6661 6374 6f72 206f 6620 6f72  ent factor of or
+000081d0: 7468 6f6c 6f67 6f75 7320 4b73 2c20 6465  thologous Ks, de
+000081e0: 6661 756c 7420 302e 350a 2d6f 612c 202d  fault 0.5.-oa, -
+000081f0: 2d6f 6b61 6c70 6861 2c20 7468 6520 6f70  -okalpha, the op
+00008200: 6163 6974 7920 6f66 206f 7274 686f 6c6f  acity of ortholo
+00008210: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+00008220: 7469 6f6e 2069 6e20 6d69 7865 6420 706c  tion in mixed pl
+00008230: 6f74 2c20 6465 6661 756c 7420 302e 350a  ot, default 0.5.
+00008240: 2d63 732c 202d 2d63 6c61 7373 6963 2c20  -cs, --classic, 
+00008250: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00008260: 7468 6572 2074 6f20 6472 6177 206d 6978  ther to draw mix
+00008270: 6564 204b 7320 706c 6f74 2069 6e20 6120  ed Ks plot in a 
+00008280: 636c 6173 7369 6320 6d61 6e6e 6572 2077  classic manner w
+00008290: 6865 7265 2074 6865 2066 756c 6c20 6f72  here the full or
+000082a0: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
+000082b0: 7472 6962 7574 696f 6e20 6973 2064 7261  tribution is dra
+000082c0: 7765 642c 2069 6620 7468 6520 666c 6167  wed, if the flag
+000082d0: 2077 6173 2073 6574 2c20 7468 6520 636c   was set, the cl
+000082e0: 6173 7369 6320 6d69 7865 6420 4b73 2070  assic mixed Ks p
+000082f0: 6c6f 7420 776f 756c 6420 6265 2064 7261  lot would be dra
+00008300: 776e 0a2d 7461 2c20 2d2d 746f 7061 7272  wn.-ta, --toparr
+00008310: 6f77 2c20 666c 6167 206f 7074 696f 6e2c  ow, flag option,
+00008320: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
+00008330: 7374 2074 6865 2061 7272 6f77 2074 6f20  st the arrow to 
+00008340: 6265 2061 7420 7468 6520 746f 7020 6f66  be at the top of
+00008350: 2074 6865 2070 6c6f 742c 2069 6e73 7465   the plot, inste
+00008360: 6164 206f 6620 6265 696e 6720 636f 6f72  ad of being coor
+00008370: 6469 6e61 7465 6420 6173 2074 6865 204b  dinated as the K
+00008380: 4445 206f 6620 7468 6520 6f72 7468 6f6c  DE of the orthol
+00008390: 6f67 6f75 7320 4b73 2064 6973 7472 6962  ogous Ks distrib
+000083a0: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
+000083b0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+000083c0: 6172 726f 7720 776f 756c 6420 6265 2073  arrow would be s
+000083d0: 6574 2061 7420 7468 6520 746f 700a 2d6e  et at the top.-n
+000083e0: 612c 202d 2d6e 6f64 6561 7665 7261 6765  a, --nodeaverage
+000083f0: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+00008400: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+00008410: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
+00008420: 686f 6420 666f 7220 6465 2d72 6564 756e  hod for de-redun
+00008430: 6461 6e63 792c 2069 6620 7468 6520 666c  dancy, if the fl
+00008440: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00008450: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+00008460: 7468 6f64 2077 6f75 6c64 2062 6520 696e  thod would be in
+00008470: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
+00008480: 6f6f 7473 7472 6170 2c20 7468 6520 6e75  ootstrap, the nu
+00008490: 6d62 6572 206f 6620 626f 6f74 7374 7261  mber of bootstra
+000084a0: 7020 7265 706c 6963 6174 6573 206f 6620  p replicates of 
+000084b0: 6f72 7468 6f6c 6f67 204b 7320 6469 7374  ortholog Ks dist
+000084c0: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
+000084d0: 6420 706c 6f74 0a2d 6772 2c20 2d2d 6769  d plot.-gr, --gi
+000084e0: 7374 7262 2c20 666c 6167 206f 7074 696f  strb, flag optio
+000084f0: 6e2c 2077 6865 7468 6572 2074 6f20 7573  n, whether to us
+00008500: 6520 6769 7374 5f72 6169 6e62 6f77 2061  e gist_rainbow a
+00008510: 7320 636f 6c6f 7220 6d61 7020 6f66 2064  s color map of d
+00008520: 6f74 706c 6f74 0a2d 6e2c 202d 2d6e 7468  otplot.-n, --nth
+00008530: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+00008540: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+00008550: 7573 6520 696e 2062 6f6f 7473 7472 6170  use in bootstrap
+00008560: 2073 616d 706c 696e 672c 2064 6566 6175   sampling, defau
+00008570: 6c74 2031 0a60 6060 0a0a 2323 2055 7361  lt 1.```..## Usa
+00008580: 6765 0a0a 4865 7265 2077 6520 7072 6f76  ge..Here we prov
+00008590: 6964 6564 2074 6865 2062 6173 6963 2075  ided the basic u
+000085a0: 7361 6765 2066 6f72 2065 6163 6820 7072  sage for each pr
+000085b0: 6f67 7261 6d20 616e 6420 7468 6520 7265  ogram and the re
+000085c0: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
+000085d0: 7320 616e 6420 7375 6767 6573 7469 6f6e  s and suggestion
+000085e0: 7320 6f6e 2070 6172 616d 6574 6572 697a  s on parameteriz
+000085f0: 6174 696f 6e2e 2041 2072 656d 696e 6465  ation. A reminde
+00008600: 7220 7468 6174 2074 6865 2067 6976 656e  r that the given
+00008610: 2063 6f72 6573 2061 6e64 2074 6872 6561   cores and threa
+00008620: 6473 2063 616e 2073 6967 6e69 6669 6361  ds can significa
+00008630: 6e74 6c79 2069 6d70 6163 7420 7468 6520  ntly impact the 
+00008640: 7275 6e20 7469 6d65 2061 6e64 2074 6875  run time and thu
+00008650: 7320 7765 2061 6464 6564 2073 6f6d 6520  s we added some 
+00008660: 7265 706f 7274 2069 6e66 6f72 6d61 7469  report informati
+00008670: 6f6e 2070 6572 7461 696e 696e 6720 746f  on pertaining to
+00008680: 2074 6865 2073 7973 7465 6d20 6f66 2075   the system of u
+00008690: 7365 7273 2074 6f20 6661 6369 6c69 7461  sers to facilita
+000086a0: 7465 2074 6865 2065 6666 6963 6965 6e74  te the efficient
+000086b0: 2073 6574 7469 6e67 206f 6620 7468 7265   setting of thre
+000086c0: 6164 7320 616e 6420 6d65 6d6f 7279 2e20  ads and memory. 
+000086d0: 5468 6520 6c6f 6769 6361 6c20 4350 5573  The logical CPUs
+000086e0: 2072 6570 6f72 7465 6420 7265 7072 6573   reported repres
+000086f0: 656e 7473 2074 6865 206e 756d 6265 7220  ents the number 
+00008700: 6f66 2070 6879 7369 6361 6c20 636f 7265  of physical core
+00008710: 7320 6d75 6c74 6970 6c69 6564 2062 7920  s multiplied by 
+00008720: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
+00008730: 7265 6164 7320 7468 6174 2063 616e 2072  reads that can r
+00008740: 756e 206f 6e20 6561 6368 2063 6f72 652c  un on each core,
+00008750: 2061 6c73 6f20 6b6e 6f77 6e20 6173 2048   also known as H
+00008760: 7970 6572 2054 6872 6561 6469 6e67 2e20  yper Threading. 
+00008770: 5468 6520 6e75 6d62 6572 206f 6620 6c6f  The number of lo
+00008780: 6769 6361 6c20 4350 5573 206d 6179 206e  gical CPUs may n
+00008790: 6f74 206e 6563 6573 7361 7269 6c79 2062  ot necessarily b
+000087a0: 6520 6571 7569 7661 6c65 6e74 2074 6f20  e equivalent to 
+000087b0: 7468 6520 6163 7475 616c 206e 756d 6265  the actual numbe
+000087c0: 7220 6f66 2043 5055 7320 7468 6520 6375  r of CPUs the cu
+000087d0: 7272 656e 7420 7072 6f63 6573 7320 6361  rrent process ca
+000087e0: 6e20 7573 652e 2054 6865 2061 7661 696c  n use. The avail
+000087f0: 6162 6c65 206d 656d 6f72 7920 7265 6665  able memory refe
+00008800: 7273 2074 6f20 7468 6520 6d65 6d6f 7279  rs to the memory
+00008810: 2074 6861 7420 6361 6e20 6265 2067 6976   that can be giv
+00008820: 656e 2069 6e73 7461 6e74 6c79 2074 6f20  en instantly to 
+00008830: 7072 6f63 6573 7365 7320 7769 7468 6f75  processes withou
+00008840: 7420 7468 6520 7379 7374 656d 2067 6f69  t the system goi
+00008850: 6e67 2069 6e74 6f20 7377 6170 2061 6e64  ng into swap and
+00008860: 2072 6566 6c65 6374 7320 7468 6520 6163   reflects the ac
+00008870: 7475 616c 206d 656d 6f72 7920 6176 6169  tual memory avai
+00008880: 6c61 626c 652e 2054 6865 2066 7265 6520  lable. The free 
+00008890: 6d65 6d6f 7279 2072 6566 6572 7320 746f  memory refers to
+000088a0: 2074 6865 206d 656d 6f72 7920 6e6f 7420   the memory not 
+000088b0: 6265 696e 6720 7573 6564 2061 7420 616c  being used at al
+000088c0: 6c20 287a 6572 6f65 6429 2074 6861 7420  l (zeroed) that 
+000088d0: 6973 2072 6561 6469 6c79 2061 7661 696c  is readily avail
+000088e0: 6162 6c65 2e20 5468 6520 6465 7363 7269  able. The descri
+000088f0: 7074 696f 6e20 6162 6f76 6520 7265 6665  ption above refe
+00008900: 7273 2074 6f20 7468 6520 646f 6375 6d65  rs to the docume
+00008910: 6e74 6174 696f 6e20 6f66 2060 7073 7574  ntation of `psut
+00008920: 696c 602e 0a0a 2323 2320 7767 6420 646d  il`...### wgd dm
+00008930: 640a 0a2a 2a54 6865 2064 656c 696e 6561  d..**The delinea
+00008940: 7469 6f6e 206f 6620 7768 6f6c 6520 7061  tion of whole pa
+00008950: 7261 6e6f 6d65 2a2a 0a60 6060 0a77 6764  ranome**.```.wgd
+00008960: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00008970: 6f65 7275 6c65 6120 2d49 2032 202d 6520  oerulea -I 2 -e 
+00008980: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+00008990: 7020 3520 282d 6e6e 2920 282d 2d74 6f5f  p 5 (-nn) (--to_
+000089a0: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
+000089b0: 6e20 3429 2028 2d6f 2077 6764 5f64 6d64  n 4) (-o wgd_dmd
+000089c0: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
+000089d0: 7029 0a60 6060 200a 0a4e 6f74 6520 7468  p).``` ..Note th
+000089e0: 6174 2077 6520 646f 6e27 7420 7072 6f76  at we don't prov
+000089f0: 6964 6520 7468 6520 6461 7461 206f 6620  ide the data of 
+00008a00: 7468 6973 2063 6f64 696e 6720 7365 7175  this coding sequ
+00008a10: 656e 6365 2028 6364 7329 2066 696c 6520  ence (cds) file 
+00008a20: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+00008a30: 6c65 6160 2062 7574 2069 7420 6361 6e20  lea` but it can 
+00008a40: 6265 2065 6173 696c 7920 646f 776e 6c6f  be easily downlo
+00008a50: 6164 6564 2061 7420 5b50 6879 746f 7a6f  aded at [Phytozo
+00008a60: 6d65 5d28 6874 7470 733a 2f2f 7068 7974  me](https://phyt
+00008a70: 6f7a 6f6d 652d 6e65 7874 2e6a 6769 2e64  ozome-next.jgi.d
+00008a80: 6f65 2e67 6f76 2f69 6e66 6f2f 4163 6f65  oe.gov/info/Acoe
+00008a90: 7275 6c65 615f 7633 5f31 2920 2873 616d  rulea_v3_1) (sam
+00008aa0: 6520 666f 7220 6f74 6865 7220 5573 6167  e for other Usag
+00008ab0: 6520 646f 6329 2e20 4e6f 7465 2074 6861  e doc). Note tha
+00008ac0: 7420 696e 2074 6865 2069 7373 7565 7320  t in the issues 
+00008ad0: 736f 6d65 2075 7365 7273 2064 6964 6e27  some users didn'
+00008ae0: 7420 646f 776e 6c6f 6164 2074 6865 2060  t download the `
+00008af0: 4163 6f65 7275 6c65 615f 3332 325f 7633  Acoerulea_322_v3
+00008b00: 2e31 2e63 6473 5f70 7269 6d61 7279 5472  .1.cds_primaryTr
+00008b10: 616e 7363 7269 7074 4f6e 6c79 2e66 612e  anscriptOnly.fa.
+00008b20: 677a 6020 6275 7420 696e 7374 6561 6420  gz` but instead 
+00008b30: 7468 6520 6041 636f 6572 756c 6561 5f33  the `Acoerulea_3
+00008b40: 3232 5f76 332e 312e 6364 732e 6661 2e67  22_v3.1.cds.fa.g
+00008b50: 7a60 2066 696c 652e 2046 6f72 2074 6865  z` file. For the
+00008b60: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00008b70: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
+00008b80: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00008b90: 6469 7374 7269 6275 7469 6f6e 2c20 2a2a  distribution, **
+00008ba0: 6f6e 6c79 206f 6e65 2074 7261 6e73 6372  only one transcr
+00008bb0: 6970 7420 2874 6865 2070 7269 6d61 7279  ipt (the primary
+00008bc0: 206f 6e65 2920 7065 7220 6765 6e65 2073   one) per gene s
+00008bd0: 686f 756c 6420 6265 2069 6e63 6c75 6465  hould be include
+00008be0: 642a 2a20 7375 6368 2074 6861 7420 7468  d** such that th
+00008bf0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00008c00: 3e20 6973 2072 6561 6c6c 7920 696e 6469  > is really indi
+00008c10: 6361 7469 6e67 2074 6865 2061 6765 206f  cating the age o
+00008c20: 6620 6765 6e65 2064 7570 6c69 6361 7469  f gene duplicati
+00008c30: 6f6e 2065 7665 6e74 2c20 696e 7374 6561  on event, instea
+00008c40: 6420 6f66 2061 6c74 6572 6e61 7469 7665  d of alternative
+00008c50: 2073 706c 6963 696e 672e 2054 7261 6e73   splicing. Trans
+00008c60: 6372 6970 746f 6d65 2064 6174 6120 7368  criptome data sh
+00008c70: 6f75 6c64 2062 6520 6361 7265 6675 6c6c  ould be carefull
+00008c80: 7920 7472 6561 7465 6420 7769 7468 2064  y treated with d
+00008c90: 652d 7265 6475 6e64 616e 6379 2073 6f20  e-redundancy so 
+00008ca0: 6173 2074 6f20 7265 6475 6365 2074 6865  as to reduce the
+00008cb0: 2066 616c 7365 2070 6f73 6974 6976 6520   false positive 
+00008cc0: 6475 706c 6963 6174 696f 6e20 6275 6d70  duplication bump
+00008cd0: 2063 6175 7365 6420 6279 2070 6572 7661   caused by perva
+00008ce0: 7369 7665 2061 6c74 6572 6e61 7469 7665  sive alternative
+00008cf0: 2073 706c 6963 696e 672e 2048 6572 6520   splicing. Here 
+00008d00: 7468 6520 696e 666c 6174 696f 6e20 6661  the inflation fa
+00008d10: 6374 6f72 2070 6172 616d 6574 6572 2c20  ctor parameter, 
+00008d20: 6769 7665 6e20 6279 2060 2d49 6020 6f72  given by `-I` or
+00008d30: 2060 2d2d 696e 666c 6174 696f 6e60 2c20   `--inflation`, 
+00008d40: 6166 6665 6374 7320 7468 6520 6772 616e  affects the gran
+00008d50: 756c 6172 6974 7920 6f72 2072 6573 6f6c  ularity or resol
+00008d60: 7574 696f 6e20 6f66 2074 6865 2063 6c75  ution of the clu
+00008d70: 7374 6572 696e 6720 6f75 7463 6f6d 6520  stering outcome 
+00008d80: 616e 6420 696d 706c 6963 6974 6c79 2063  and implicitly c
+00008d90: 6f6e 7472 6f6c 6c73 2074 6865 206e 756d  ontrolls the num
+00008da0: 6265 7220 6f66 2063 6c75 7374 6572 732c  ber of clusters,
+00008db0: 2077 6974 6820 6c6f 7720 7661 6c75 6573   with low values
+00008dc0: 2073 7563 6820 6173 2031 2e33 206f 7220   such as 1.3 or 
+00008dd0: 312e 3420 6c65 6164 696e 6720 746f 2066  1.4 leading to f
+00008de0: 6577 6572 2062 7574 206c 6172 6765 7220  ewer but larger 
+00008df0: 636c 7573 7465 7273 2061 6e64 2068 6967  clusters and hig
+00008e00: 6820 7661 6c75 6573 2073 7563 6820 6173  h values such as
+00008e10: 2035 206f 7220 3620 6c65 6164 696e 6720   5 or 6 leading 
+00008e20: 746f 206d 6f72 6520 6275 7420 736d 616c  to more but smal
+00008e30: 6c65 7220 636c 7573 7465 7273 2e20 5765  ler clusters. We
+00008e40: 2073 6574 2074 6865 2064 6566 6175 6c74   set the default
+00008e50: 2076 616c 7565 2061 7320 3220 6173 2073   value as 2 as s
+00008e60: 7567 6765 7374 6564 2062 7920 5b4d 434c  uggested by [MCL
+00008e70: 5d28 6874 7470 733a 2f2f 6d69 6361 6e73  ](https://micans
+00008e80: 2e6f 7267 2f6d 636c 2f29 2e20 5468 6520  .org/mcl/). The 
+00008e90: 652d 7661 6c75 6520 6375 742d 6f66 6620  e-value cut-off 
+00008ea0: 666f 7220 7365 7175 656e 6365 2073 696d  for sequence sim
+00008eb0: 696c 6172 6974 792c 2067 6976 656e 2062  ilarity, given b
+00008ec0: 7920 602d 6560 206f 7220 602d 2d65 7661  y `-e` or `--eva
+00008ed0: 6c60 2c20 7768 6963 6820 6465 6e6f 7465  l`, which denote
+00008ee0: 7320 7468 6520 6578 7065 6374 6564 2076  s the expected v
+00008ef0: 616c 7565 206f 6620 7468 6520 6869 7420  alue of the hit 
+00008f00: 7175 616e 7469 6669 6573 2074 6865 206e  quantifies the n
+00008f10: 756d 6265 7220 6f66 2061 6c69 676e 6d65  umber of alignme
+00008f20: 6e74 7320 6f66 2073 696d 696c 6172 206f  nts of similar o
+00008f30: 7220 6265 7474 6572 2071 7561 6c69 7479  r better quality
+00008f40: 2074 6861 7420 796f 7520 6578 7065 6374   that you expect
+00008f50: 2074 6f20 6669 6e64 2073 6561 7263 6869   to find searchi
+00008f60: 6e67 2074 6869 7320 7175 6572 7920 6167  ng this query ag
+00008f70: 6169 6e73 7420 6120 6461 7461 6261 7365  ainst a database
+00008f80: 206f 6620 7261 6e64 6f6d 2073 6571 7565   of random seque
+00008f90: 6e63 6573 2074 6865 2073 616d 6520 7369  nces the same si
+00008fa0: 7a65 2061 7320 7468 6520 6163 7475 616c  ze as the actual
+00008fb0: 2074 6172 6765 7420 6461 7461 6261 7365   target database
+00008fc0: 2c20 6973 2074 6865 206b 6579 2070 6172  , is the key par
+00008fd0: 616d 6574 6572 206d 6561 7375 7269 6e67  ameter measuring
+00008fe0: 2074 6865 2073 6967 6e69 6669 6361 6e63   the significanc
+00008ff0: 6520 6f66 2061 2068 6974 2c20 7768 6963  e of a hit, whic
+00009000: 6820 6973 2073 6574 2068 6572 6520 6173  h is set here as
+00009010: 2064 6566 6175 6c74 2031 652d 3130 2e20   default 1e-10. 
+00009020: 4e6f 7465 2074 6861 7420 5b44 4941 4d4f  Note that [DIAMO
+00009030: 4e44 5d28 6874 7470 733a 2f2f 6769 7468  ND](https://gith
+00009040: 7562 2e63 6f6d 2f62 6275 6368 6669 6e6b  ub.com/bbuchfink
+00009050: 2f64 6961 6d6f 6e64 2f77 696b 692f 312e  /diamond/wiki/1.
+00009060: 2d54 7574 6f72 6961 6c29 2069 7473 656c  -Tutorial) itsel
+00009070: 6620 6279 2064 6566 6175 6c74 206f 6e6c  f by default onl
+00009080: 7920 7265 706f 7274 7320 616c 6c20 616c  y reports all al
+00009090: 6967 6e6d 656e 7473 2077 6974 6820 652d  ignments with e-
+000090a0: 7661 6c75 6520 3c20 302e 3030 312e 2054  value < 0.001. T
+000090b0: 6865 2070 6572 6365 6e74 6167 6520 6f66  he percentage of
+000090c0: 2075 7070 6572 2068 6974 7320 7573 6564   upper hits used
+000090d0: 2066 6f72 2067 656e 6520 6c65 6e67 7468   for gene length
+000090e0: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
+000090f0: 6769 7665 6e20 6279 2060 2d6e 7060 206f  given by `-np` o
+00009100: 7220 602d 2d6e 6f72 6d61 6c69 7a65 6470  r `--normalizedp
+00009110: 6572 6365 6e74 602c 2077 6869 6368 2064  ercent`, which d
+00009120: 6574 6572 6d69 6e65 7320 7468 6520 7570  etermines the up
+00009130: 7065 7220 7065 7263 656e 7469 6c65 206f  per percentile o
+00009140: 6620 6869 7473 2070 6572 2062 696e 2028  f hits per bin (
+00009150: 6361 7465 676f 7269 7a65 6420 6279 2067  categorized by g
+00009160: 656e 6520 6c65 6e67 7468 2920 7573 6564  ene length) used
+00009170: 2069 6e20 7468 6520 6669 7420 6f66 206c   in the fit of l
+00009180: 696e 6561 7220 7265 6772 6573 7369 6f6e  inear regression
+00009190: 2c20 636f 6e73 6964 6572 696e 6720 7468  , considering th
+000091a0: 6174 206e 6f74 2061 6c6c 2068 6974 7320  at not all hits 
+000091b0: 7065 7220 6269 6e20 7368 6f77 2061 7070  per bin show app
+000091c0: 6172 656e 7420 6c69 6e65 6172 2072 656c  arent linear rel
+000091d0: 6174 696f 6e73 6869 702c 2069 7320 7365  ationship, is se
+000091e0: 7420 6173 2064 6566 6175 6c74 2035 2c20  t as default 5, 
+000091f0: 696e 6469 6361 7469 6e67 2074 6865 2075  indicating the u
+00009200: 7361 6765 206f 6620 746f 7020 3525 2068  sage of top 5% h
+00009210: 6974 7320 7065 7220 6269 6e2e 2054 6865  its per bin. The
+00009220: 206e 756d 6265 7220 6f66 2062 696e 7320   number of bins 
+00009230: 6469 7669 6465 6420 696e 2067 656e 6520  divided in gene 
+00009240: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
+00009250: 7469 6f6e 2c20 6769 7665 6e20 6279 2060  tion, given by `
+00009260: 2d62 7360 206f 7220 602d 2d62 696e 7360  -bs` or `--bins`
+00009270: 2c20 6465 7465 726d 696e 6573 2074 6865  , determines the
+00009280: 206e 756d 6265 7220 6f66 2062 696e 7320   number of bins 
+00009290: 746f 2063 6174 6567 6f72 697a 6520 7468  to categorize th
+000092a0: 6520 6765 6e65 206c 656e 6774 682c 2069  e gene length, i
+000092b0: 7320 7365 7420 6173 2064 6566 6175 6c74  s set as default
+000092c0: 2031 3030 2e20 5468 6520 7061 7261 6d65   100. The parame
+000092d0: 7465 7220 602d 6e6e 6020 6f72 2060 2d2d  ter `-nn` or `--
+000092e0: 6e6f 6e6f 726d 616c 697a 6174 696f 6e60  nonormalization`
+000092f0: 2063 616e 2062 6520 7365 7420 746f 2063   can be set to c
+00009300: 616c 6c20 6f66 6620 7468 6520 6e6f 726d  all off the norm
+00009310: 616c 697a 6174 696f 6e20 7072 6f63 6573  alization proces
+00009320: 732c 2061 6c74 686f 7567 6820 6974 2773  s, although it's
+00009330: 2073 7567 6765 7374 6564 2074 6f20 636f   suggested to co
+00009340: 6e64 7563 7420 7468 6520 6e6f 726d 616c  nduct the normal
+00009350: 697a 6174 696f 6e20 746f 2061 6371 7569  ization to acqui
+00009360: 7265 206d 6f72 6520 6163 6375 7261 7465  re more accurate
+00009370: 2067 656e 6520 6661 6d69 6c79 2063 6c75   gene family clu
+00009380: 7374 6572 696e 6720 7265 7375 6c74 2e20  stering result. 
+00009390: 5468 6520 7061 7261 6d65 7465 7273 2060  The parameters `
+000093a0: 2d2d 746f 5f73 746f 7060 2061 6e64 2060  --to_stop` and `
+000093b0: 2d2d 6364 7360 2063 6f6e 7472 6f6c 2074  --cds` control t
+000093c0: 6865 2062 6568 6176 696f 7572 206f 6620  he behaviour of 
+000093d0: 7472 616e 736c 6174 696e 6720 636f 6469  translating codi
+000093e0: 6e67 2073 6571 7565 6e63 6520 696e 746f  ng sequence into
+000093f0: 2061 6d69 6e6f 2061 6369 6420 7365 7175   amino acid sequ
+00009400: 656e 6365 2e20 4966 2074 6865 2060 2d2d  ence. If the `--
+00009410: 746f 5f73 746f 7060 2077 6173 2073 6574  to_stop` was set
+00009420: 2c20 7468 6520 7472 616e 736c 6174 696f  , the translatio
+00009430: 6e20 776f 756c 6420 6265 2074 6572 6d69  n would be termi
+00009440: 6e61 7465 6420 6174 2074 6865 2066 6972  nated at the fir
+00009450: 7374 2069 6e2d 6672 616d 6520 7374 6f70  st in-frame stop
+00009460: 2063 6f64 6f6e 2c20 6f74 6865 7277 6973   codon, otherwis
+00009470: 6520 7468 6520 7472 616e 736c 6174 696f  e the translatio
+00009480: 6e20 776f 756c 6420 7369 6d70 6c79 2073  n would simply s
+00009490: 6b69 7020 616e 7920 7374 6f70 2063 6f64  kip any stop cod
+000094a0: 6f6e 732e 2049 6620 7468 6520 602d 2d63  ons. If the `--c
+000094b0: 6473 6020 7761 7320 7365 742c 2073 6571  ds` was set, seq
+000094c0: 7565 6e63 6573 2074 6861 7420 646f 6573  uences that does
+000094d0: 6e27 7420 7374 6172 7420 7769 7468 2061  n't start with a
+000094e0: 2076 616c 6964 2073 7461 7274 2063 6f64   valid start cod
+000094f0: 6f6e 2c20 6f72 2063 6f6e 7461 696e 7320  on, or contains 
+00009500: 6d6f 7265 2074 6861 6e20 6f6e 6520 696e  more than one in
+00009510: 2d66 7261 6d65 2073 746f 7020 636f 646f  -frame stop codo
+00009520: 6e2c 206f 7220 6973 206e 6f74 2064 6976  n, or is not div
+00009530: 6964 6162 6c65 2062 7920 332c 2077 6f75  idable by 3, wou
+00009540: 6c64 2062 6520 7369 6d70 6c79 2064 726f  ld be simply dro
+00009550: 7070 6564 2c20 7375 6368 2074 6861 7420  pped, such that 
+00009560: 6f6e 6c79 2073 7472 6963 7420 636f 6d70  only strict comp
+00009570: 6c65 7465 2063 6f64 696e 6720 7365 7175  lete coding sequ
+00009580: 656e 6365 7320 776f 756c 6420 6265 2069  ences would be i
+00009590: 6e63 6c75 6465 6420 696e 2074 6865 2073  ncluded in the s
+000095a0: 7562 7365 7175 656e 7420 616e 616c 7973  ubsequent analys
+000095b0: 6973 2e20 5468 6520 6e75 6d62 6572 206f  is. The number o
+000095c0: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
+000095d0: 6473 2062 7920 7468 6520 6f70 7469 6f6e  ds by the option
+000095e0: 2060 2d6e 6020 6f72 2060 2d2d 6e74 6872   `-n` or `--nthr
+000095f0: 6561 6473 6020 6361 6e20 6265 2073 6574  eads` can be set
+00009600: 2074 6f20 6163 6365 6c65 7261 7465 2074   to accelerate t
+00009610: 6865 2063 616c 6375 6c61 7469 6f6e 2077  he calculation w
+00009620: 6974 6869 6e20 6064 6961 6d6f 6e64 602e  ithin `diamond`.
+00009630: 2054 6865 2064 6972 6563 746f 7279 206f   The directory o
+00009640: 6620 6f75 7470 7574 206f 7220 696e 7465  f output or inte
+00009650: 726d 6564 6961 7465 2066 696c 6573 2069  rmediate files i
+00009660: 7320 6465 7465 726d 696e 6564 2062 7920  s determined by 
+00009670: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+00009680: 6f60 206f 7220 602d 2d6f 7574 6469 7260  o` or `--outdir`
+00009690: 2c20 616e 6420 602d 7460 206f 7220 602d  , and `-t` or `-
+000096a0: 2d74 6d70 6469 7260 2c20 7768 6963 6820  -tmpdir`, which 
+000096b0: 7769 6c6c 2062 6520 6372 6561 7465 6420  will be created 
+000096c0: 6279 2074 6865 2070 726f 6772 616d 2069  by the program i
+000096d0: 7473 656c 6620 616e 6420 6265 206f 7665  tself and be ove
+000096e0: 7277 7269 7474 656e 2069 6620 7468 6520  rwritten if the 
+000096f0: 666f 6c64 6572 2068 6173 2061 6c72 6561  folder has alrea
+00009700: 6479 2062 6565 6e20 6372 6561 7465 642e  dy been created.
+00009710: 204e 6f74 6520 7468 6174 2074 6865 2073   Note that the s
+00009720: 6f66 7477 6172 6520 6064 6961 6d6f 6e64  oftware `diamond
+00009730: 6020 7368 6f75 6c64 2062 6520 7072 652d  ` should be pre-
+00009740: 696e 7374 616c 6c65 6420 616e 6420 7365  installed and se
+00009750: 7420 696e 2074 6865 2065 6e76 6972 6f6e  t in the environ
+00009760: 6d65 6e74 2070 6174 6820 696e 2061 6c6c  ment path in all
+00009770: 2074 6865 2061 6e61 6c79 7369 7320 7065   the analysis pe
+00009780: 7266 6f72 6d65 6420 6279 2060 7767 6420  rformed by `wgd 
+00009790: 646d 6460 2065 7863 6570 7420 666f 7220  dmd` except for 
+000097a0: 7468 6520 636f 6c6c 696e 6561 7220 636f  the collinear co
+000097b0: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+000097c0: 6973 2e0a 0a2a 2a57 6520 7375 6767 6573  is...**We sugges
+000097d0: 7420 7468 6174 2074 6865 2064 6566 6175  t that the defau
+000097e0: 6c74 2073 6574 7469 6e67 2069 6e20 7768  lt setting in wh
+000097f0: 6963 6820 7468 6520 696e 666c 6174 696f  ich the inflatio
+00009800: 6e20 6661 6374 6f72 2069 7320 7365 7420  n factor is set 
+00009810: 6173 2032 2c20 652d 7661 6c75 6520 6375  as 2, e-value cu
+00009820: 742d 6f66 6620 6173 2031 652d 3130 2061  t-off as 1e-10 a
+00009830: 6e64 206f 7468 6572 2070 6172 616d 6574  nd other paramet
+00009840: 6572 7320 696e 2064 6566 6175 6c74 2069  ers in default i
+00009850: 7320 6120 676f 6f64 2073 7461 7274 696e  s a good startin
+00009860: 6720 706f 696e 742c 2075 6e6c 6573 7320  g point, unless 
+00009870: 796f 7520 7370 6563 6966 6963 616c 6c79  you specifically
+00009880: 2077 616e 7420 746f 2065 7870 6c6f 7265   want to explore
+00009890: 2074 6865 2065 6666 6563 7473 206f 6620   the effects of 
+000098a0: 6469 6666 6572 656e 7420 7061 7261 6d65  different parame
+000098b0: 7465 7273 2e2a 2a20 5375 6368 2074 6861  ters.** Such tha
+000098c0: 7420 7468 6520 636f 6d6d 616e 6420 666f  t the command fo
+000098d0: 7220 7468 6520 6465 6c69 6e65 6174 696f  r the delineatio
+000098e0: 6e20 6f66 2077 686f 6c65 2070 6172 616e  n of whole paran
+000098f0: 6f6d 6520 6973 2073 696d 706c 7920 6173  ome is simply as
+00009900: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+00009910: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00009920: 6f65 7275 6c65 610a 6060 600a 0a2a 2a54  oerulea.```..**T
+00009930: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
+00009940: 6620 5242 4873 2a2a 0a60 6060 0a77 6764  f RBHs**.```.wgd
+00009950: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+00009960: 6571 7565 6e63 6532 202d 6520 3165 2d31  equence2 -e 1e-1
+00009970: 3020 2d62 7320 3130 3020 2d6e 7020 3520  0 -bs 100 -np 5 
+00009980: 282d 6e6e 2920 282d 6e20 3429 2028 2d63  (-nn) (-n 4) (-c
+00009990: 2030 2e39 2920 282d 2d6f 6766 6f72 6d61   0.9) (--ogforma
+000099a0: 7429 2028 2d2d 746f 5f73 746f 7029 2028  t) (--to_stop) (
+000099b0: 2d2d 6364 7329 2028 2d6f 2077 6764 5f64  --cds) (-o wgd_d
+000099c0: 6d64 2920 282d 7420 776f 726b 696e 675f  md) (-t working_
+000099d0: 746d 7029 0a60 6060 0a0a 546f 2064 656c  tmp).```..To del
+000099e0: 696e 6561 7465 2052 4248 7320 6265 7477  ineate RBHs betw
+000099f0: 6565 6e20 7477 6f20 6364 7320 7365 7175  een two cds sequ
+00009a00: 656e 6365 2066 696c 6573 2c20 7468 6520  ence files, the 
+00009a10: 7265 6c65 7661 6e74 2070 6172 616d 6574  relevant paramet
+00009a20: 6572 2069 7320 6d6f 7374 6c79 2074 6865  er is mostly the
+00009a30: 2073 616d 6520 6173 2077 686f 6c65 2070   same as whole p
+00009a40: 6172 616e 6f6d 6520 696e 6665 7265 6e63  aranome inferenc
+00009a50: 652c 2065 7863 6570 7420 666f 7220 7468  e, except for th
+00009a60: 6520 7061 7261 6d65 7465 7220 602d 6360  e parameter `-c`
+00009a70: 206f 7220 602d 2d63 7363 6f72 6560 2c20   or `--cscore`, 
+00009a80: 7768 6963 6820 7261 6e67 6573 2062 6574  which ranges bet
+00009a90: 7765 656e 2030 2061 6e64 2031 2061 6e64  ween 0 and 1 and
+00009aa0: 2069 7320 7573 6564 2074 6f20 7265 6c61   is used to rela
+00009ab0: 7820 7468 6520 7369 6d69 6c61 7269 7479  x the similarity
+00009ac0: 2063 7574 6f66 6620 6672 6f6d 2074 6865   cutoff from the
+00009ad0: 2065 7863 6c75 7369 7665 2072 6563 6970   exclusive recip
+00009ae0: 726f 6361 6c20 6265 7374 2068 6974 7320  rocal best hits 
+00009af0: 746f 2061 2063 6572 7461 696e 2072 6174  to a certain rat
+00009b00: 696f 2061 7320 746f 2074 6865 2062 6573  io as to the bes
+00009b10: 7420 6869 7473 2e20 466f 7220 696e 7374  t hits. For inst
+00009b20: 616e 6365 2c20 6966 2074 6865 2067 656e  ance, if the gen
+00009b30: 6520 6231 2066 726f 6d20 6765 6e6f 6d65  e b1 from genome
+00009b40: 2042 2068 6173 2074 6865 2062 6573 7420   B has the best 
+00009b50: 6869 7420 6765 6e65 2061 3120 6672 6f6d  hit gene a1 from
+00009b60: 2067 656e 6f6d 6520 4120 7769 7468 2074   genome A with t
+00009b70: 6865 2062 6974 2073 636f 7265 2061 7320  he bit score as 
+00009b80: 3130 302c 2077 6869 6368 2069 7320 6120  100, which is a 
+00009b90: 7363 6f72 696e 6720 6d61 7472 6978 2069  scoring matrix i
+00009ba0: 6e64 6570 656e 6465 6e74 206d 6561 7375  ndependent measu
+00009bb0: 7265 206f 6620 7468 6520 286c 6f63 616c  re of the (local
+00009bc0: 2920 7369 6d69 6c61 7269 7479 206f 6620  ) similarity of 
+00009bd0: 7468 6520 7477 6f20 616c 6967 6e65 6420  the two aligned 
+00009be0: 7365 7175 656e 6365 732c 2077 6974 6820  sequences, with 
+00009bf0: 6c61 7267 6572 2076 616c 7565 7320 696e  larger values in
+00009c00: 6469 6361 7469 6e67 2068 6967 6865 7220  dicating higher 
+00009c10: 7369 6d69 6c61 7269 7469 6573 2c20 6769  similarities, gi
+00009c20: 7665 6e20 7468 6520 602d 6320 302e 3960  ven the `-c 0.9`
+00009c30: 2c20 6765 6e65 7320 6672 6f6d 2067 656e  , genes from gen
+00009c40: 6f6d 6520 4120 7768 6963 6820 6861 7320  ome A which has 
+00009c50: 7468 6520 6269 7420 7363 6f72 6520 7769  the bit score wi
+00009c60: 7468 2067 656e 6520 6231 2068 6967 6865  th gene b1 highe
+00009c70: 7220 7468 616e 2030 2e39 7831 3030 2077  r than 0.9x100 w
+00009c80: 696c 6c20 616c 736f 2062 6520 7772 6974  ill also be writ
+00009c90: 7465 6e20 696e 2074 6865 2072 6573 756c  ten in the resul
+00009ca0: 7420 6669 6c65 2c20 7768 6963 6820 696e  t file, which in
+00009cb0: 2061 2073 656e 7365 2061 7265 206e 6f74   a sense are not
+00009cc0: 2052 4248 7320 616e 796d 6f72 6520 6f66   RBHs anymore of
+00009cd0: 2063 6f75 7273 652c 2062 7574 2074 6865   course, but the
+00009ce0: 2068 6967 686c 7920 7369 6d69 6c61 7220   highly similar 
+00009cf0: 686f 6d6f 6c6f 6775 6520 7061 6972 732e  homologue pairs.
+00009d00: 2049 6620 6d6f 7265 2074 6861 6e20 3220   If more than 2 
+00009d10: 7365 7175 656e 6365 2066 696c 6573 2077  sequence files w
+00009d20: 6572 6520 7072 6f76 6964 6564 2c20 6576  ere provided, ev
+00009d30: 6572 7920 7061 6972 2d77 6973 6520 5242  ery pair-wise RB
+00009d40: 4873 2077 6f75 6c64 2062 6520 6361 6c63  Hs would be calc
+00009d50: 756c 6174 6564 2065 7863 6570 7420 666f  ulated except fo
+00009d60: 7220 7175 6572 7969 6e67 2074 6865 2073  r querying the s
+00009d70: 616d 6520 7365 7175 656e 6365 2069 7473  ame sequence its
+00009d80: 656c 662e 2054 6865 206e 756d 6265 7220  elf. The number 
+00009d90: 6f66 2070 6172 616c 6c65 6c20 7468 7265  of parallel thre
+00009da0: 6164 7320 746f 2062 6f6f 7374 6572 2074  ads to booster t
+00009db0: 6865 2072 756e 6e69 6e67 2073 7065 6564  he running speed
+00009dc0: 2063 616e 2062 6520 7365 7420 6279 2074   can be set by t
+00009dd0: 6865 206f 7074 696f 6e20 602d 6e60 206f  he option `-n` o
+00009de0: 7220 602d 2d6e 7468 7265 6164 7360 2077  r `--nthreads` w
+00009df0: 6869 6368 2069 7320 7375 6767 6573 7465  hich is suggeste
+00009e00: 6420 746f 2062 6520 7365 7420 6173 2028  d to be set as (
+00009e10: 4e2d 3129 4e2f 3220 7768 6572 6520 4e20  N-1)N/2 where N 
+00009e20: 6973 2074 6865 206e 756d 6265 7220 6f66  is the number of
+00009e30: 2063 6473 2066 696c 6573 2074 6f20 6163   cds files to ac
+00009e40: 6869 6576 6520 7468 6520 6869 6768 6573  hieve the highes
+00009e50: 7420 6566 6669 6369 656e 6379 2e20 5468  t efficiency. Th
+00009e60: 6520 6f70 7469 6f6e 2060 2d2d 6f67 666f  e option `--ogfo
+00009e70: 726d 6174 6020 6361 6e20 6265 2073 6574  rmat` can be set
+00009e80: 2074 6f20 6164 6420 696e 6465 7820 2866   to add index (f
+00009e90: 6f72 2069 6e73 7461 6e63 6520 4746 3030  or instance GF00
+00009ea0: 3030 3030 3031 2920 746f 2074 6865 206f  000001) to the o
+00009eb0: 7574 7075 7420 5242 4820 6765 6e65 2066  utput RBH gene f
+00009ec0: 616d 696c 6965 7320 7768 6963 6820 6361  amilies which ca
+00009ed0: 6e20 6265 2066 7572 7468 6572 2075 7365  n be further use
+00009ee0: 6420 696e 2074 6865 202a 4b2a 3c73 7562  d in the *K*<sub
+00009ef0: 3e53 3c2f 7375 623e 2063 616c 6375 6c61  >S</sub> calcula
+00009f00: 7469 6f6e 2062 7920 6077 6764 206b 7364  tion by `wgd ksd
+00009f10: 602e 0a0a 2a2a 5468 6520 7375 6767 6573  `...**The sugges
+00009f20: 7465 6420 636f 6d6d 616e 6420 746f 2073  ted command to s
+00009f30: 7461 7274 2077 6974 6820 6973 2061 6c73  tart with is als
+00009f40: 6f20 756e 6465 7220 7468 6520 6465 6661  o under the defa
+00009f50: 756c 7420 7365 7474 696e 6720 7769 7468  ult setting with
+00009f60: 2074 6865 2063 6f6d 6d61 6e64 2073 686f   the command sho
+00009f70: 776e 2062 656c 6f77 2a2a 0a0a 6060 600a  wn below**..```.
+00009f80: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+00009f90: 3120 7365 7175 656e 6365 320a 6060 600a  1 sequence2.```.
+00009fa0: 0a2a 2a54 6865 2064 656c 696e 6561 7469  .**The delineati
+00009fb0: 6f6e 206f 6620 6c6f 6361 6c20 4d52 4248  on of local MRBH
+00009fc0: 732a 2a0a 6060 600a 7767 6420 646d 6420  s**.```.wgd dmd 
+00009fd0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+00009fe0: 6365 3220 7365 7175 656e 6365 3320 2d66  ce2 sequence3 -f
+00009ff0: 2073 6571 7565 6e63 6531 202d 6520 3165   sequence1 -e 1e
+0000a000: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
+0000a010: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
+0000a020: 2d6b 6629 2028 2d6b 6429 2028 2d63 2030  -kf) (-kd) (-c 0
+0000a030: 2e39 2920 282d 2d74 6f5f 7374 6f70 2920  .9) (--to_stop) 
+0000a040: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
+0000a050: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
+0000a060: 5f74 6d70 290a 6060 600a 0a2a 2a54 6865  _tmp).```..**The
+0000a070: 2064 6973 7469 6e63 7469 6f6e 2062 6574   distinction bet
+0000a080: 7765 656e 206c 6f63 616c 2061 6e64 2067  ween local and g
+0000a090: 6c6f 6261 6c20 2868 6572 6575 6e64 6572  lobal (hereunder
+0000a0a0: 2920 4d52 4248 7320 6973 2074 6861 7420  ) MRBHs is that 
+0000a0b0: 6c6f 6361 6c20 4d52 4248 7320 6172 6520  local MRBHs are 
+0000a0c0: 7468 6520 7265 7375 6c74 7320 6f66 206d  the results of m
+0000a0d0: 6572 6765 6420 5242 4873 206f 6e20 6120  erged RBHs on a 
+0000a0e0: 6a6f 696e 7420 666f 6361 6c20 7370 6563  joint focal spec
+0000a0f0: 6965 732c 2066 6f72 2069 6e73 7461 6e63  ies, for instanc
+0000a100: 6520 696e 2061 2074 6872 6565 2073 7065  e in a three spe
+0000a110: 6369 6573 2073 7973 7465 6d20 2841 2c28  cies system (A,(
+0000a120: 422c 4329 292c 2074 6865 206c 6f63 616c  B,C)), the local
+0000a130: 204d 5242 4873 206f 6620 4120 6f6e 6c79   MRBHs of A only
+0000a140: 2072 6571 7569 7265 2074 6865 2063 616c   require the cal
+0000a150: 6375 6c61 7469 6f6e 206f 6620 5242 4873  culation of RBHs
+0000a160: 2062 6574 7765 656e 2041 2061 6e64 2043   between A and C
+0000a170: 2028 6465 6e6f 7465 6420 6173 2041 4329   (denoted as AC)
+0000a180: 2061 6e64 2041 4220 616e 6420 7468 656e   and AB and then
+0000a190: 2074 6865 206d 6572 6769 6e67 206f 6620   the merging of 
+0000a1a0: 4142 2061 6e64 2041 4320 6174 2074 6865  AB and AC at the
+0000a1b0: 2061 7869 7320 6f66 2041 2c20 7768 696c   axis of A, whil
+0000a1c0: 6520 7468 6520 676c 6f61 626c 204d 5242  e the gloabl MRB
+0000a1d0: 4873 2061 7265 2069 6e64 6570 656e 6465  Hs are independe
+0000a1e0: 6e74 206f 6620 666f 6361 6c20 7370 6563  nt of focal spec
+0000a1f0: 6965 7320 696e 2074 6861 7420 6974 206a  ies in that it j
+0000a200: 7573 7420 6361 6c6c 7320 7468 6520 6361  ust calls the ca
+0000a210: 6c63 756c 6174 696f 6e73 206f 6620 616c  lculations of al
+0000a220: 6c20 706f 7373 6962 6c65 2073 7065 6369  l possible speci
+0000a230: 6573 2070 6169 7220 286e 6f74 2073 656c  es pair (not sel
+0000a240: 6620 746f 2073 656c 6620 7370 6563 6965  f to self specie
+0000a250: 7320 7061 6972 292c 2073 7563 6820 7468  s pair), such th
+0000a260: 6174 2041 422c 2041 432c 2061 6e64 2042  at AB, AC, and B
+0000a270: 4320 6172 6520 616c 6c20 746f 2062 6520  C are all to be 
+0000a280: 6361 6c63 756c 6174 6564 2061 6e64 206d  calculated and m
+0000a290: 6572 6765 642e 2a2a 0a0a 5477 6f20 7479  erged.**..Two ty
+0000a2a0: 7065 7320 6f66 204d 5242 4873 2061 7320  pes of MRBHs as 
+0000a2b0: 696e 7465 7072 6574 6174 6564 2061 626f  intepretated abo
+0000a2c0: 7665 2063 616e 2062 6520 6465 6c69 6e65  ve can be deline
+0000a2d0: 6174 6564 2062 7920 6077 6764 2064 6d64  ated by `wgd dmd
+0000a2e0: 602c 2074 6865 206c 6f63 616c 204d 5242  `, the local MRB
+0000a2f0: 4873 2061 6e64 2074 6865 2067 6c6f 6261  Hs and the globa
+0000a300: 6c20 4d52 4248 732e 2054 6865 206c 6f63  l MRBHs. The loc
+0000a310: 616c 204d 5242 4873 2061 7265 2063 6f6e  al MRBHs are con
+0000a320: 7374 7275 6374 6564 2062 7920 6d65 7267  structed by merg
+0000a330: 696e 6720 616c 6c20 7468 6520 7265 6c65  ing all the rele
+0000a340: 7661 6e74 2052 4248 7320 6f6e 6c79 2077  vant RBHs only w
+0000a350: 6974 6820 7468 6520 666f 6361 6c20 7370  ith the focal sp
+0000a360: 6563 6965 732c 2077 6869 6368 2069 7320  ecies, which is 
+0000a370: 7365 7420 6279 2060 2d66 6020 6f72 2060  set by `-f` or `
+0000a380: 2d2d 666f 6375 7360 2e20 5468 6520 7061  --focus`. The pa
+0000a390: 7261 6d65 7465 7220 602d 6b66 6020 6f72  rameter `-kf` or
+0000a3a0: 2060 2d2d 6b65 6570 6661 7374 6160 2063   `--keepfasta` c
+0000a3b0: 616e 2062 6520 7365 7420 746f 2072 6574  an be set to ret
+0000a3c0: 6169 6e20 7468 6520 7365 7175 656e 6365  ain the sequence
+0000a3d0: 2069 6e66 6f72 6d61 7469 6f6e 206f 6620   information of 
+0000a3e0: 6561 6368 204d 5242 482e 2054 6865 2070  each MRBH. The p
+0000a3f0: 6172 616d 6574 6572 2060 2d6b 6460 206f  arameter `-kd` o
+0000a400: 7220 602d 2d6b 6565 7064 7570 6c69 6361  r `--keepduplica
+0000a410: 7465 7360 2064 6574 6572 6d69 6e65 7320  tes` determines 
+0000a420: 7768 6574 6865 7220 7468 6520 7361 6d65  whether the same
+0000a430: 2067 656e 6573 2063 616e 2061 7070 6561   genes can appea
+0000a440: 7220 696e 2064 6966 6665 7265 6e74 206c  r in different l
+0000a450: 6f63 616c 204d 5242 4873 2e20 4e6f 726d  ocal MRBHs. Norm
+0000a460: 616c 6c79 2074 6865 7265 2077 696c 6c20  ally there will 
+0000a470: 6265 206e 6f20 6475 706c 6963 6174 6573  be no duplicates
+0000a480: 2069 6e20 7468 6520 6c6f 6361 6c20 4d52   in the local MR
+0000a490: 4248 7320 6275 7420 6966 2075 7365 7273  BHs but if users
+0000a4a0: 2073 6574 2074 6865 2060 2d63 6020 6173   set the `-c` as
+0000a4b0: 2030 2e39 2028 6f72 2061 6e79 2076 616c   0.9 (or any val
+0000a4c0: 7565 2073 6d61 6c6c 6572 2074 6861 6e20  ue smaller than 
+0000a4d0: 3129 2c20 6974 2773 206c 696b 656c 7920  1), it's likely 
+0000a4e0: 7468 6174 2074 6865 2073 616d 6520 6765  that the same ge
+0000a4f0: 6e65 2077 696c 6c20 6861 7665 2063 6861  ne will have cha
+0000a500: 6e63 6520 746f 2061 7070 6561 7220 6d75  nce to appear mu
+0000a510: 6c74 6970 6c65 2074 696d 6573 2069 6e20  ltiple times in 
+0000a520: 6469 6666 6572 656e 7420 6c6f 6361 6c20  different local 
+0000a530: 4d52 4248 732e 2054 6861 7420 6973 2074  MRBHs. That is t
+0000a540: 6f20 7361 792c 2074 6865 2070 6172 616d  o say, the param
+0000a550: 6574 6572 2060 2d6b 6460 2069 7320 6d65  eter `-kd` is me
+0000a560: 616e 696e 6766 756c 206f 6e6c 7920 7768  aningful only wh
+0000a570: 656e 2069 7427 7320 7365 7420 746f 6765  en it's set toge
+0000a580: 7468 6572 2077 6974 6820 7468 6520 7061  ther with the pa
+0000a590: 7261 6d65 7465 7220 602d 6360 2e20 5468  rameter `-c`. Th
+0000a5a0: 6520 6e75 6d62 6572 206f 6620 7061 7261  e number of para
+0000a5b0: 6c6c 656c 2074 6872 6561 6473 2069 7320  llel threads is 
+0000a5c0: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+0000a5d0: 7365 7420 6173 2074 6865 206e 756d 6265  set as the numbe
+0000a5e0: 7220 6f66 2063 6473 2066 696c 6573 206d  r of cds files m
+0000a5f0: 696e 7573 2031 2e0a 0a2a 2a41 2073 7567  inus 1...**A sug
+0000a600: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+0000a610: 7275 6e20 6973 2075 6e64 6572 2074 6865  run is under the
+0000a620: 2064 6566 6175 6c74 2070 6172 616d 6574   default paramet
+0000a630: 6572 2077 6974 6820 7468 6520 636f 6d6d  er with the comm
+0000a640: 616e 6420 7368 6f77 6e20 6173 2062 656c  and shown as bel
+0000a650: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2064  ow.**..```.wgd d
+0000a660: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000a670: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000a680: 202d 6620 7365 7175 656e 6365 310a 6060   -f sequence1.``
+0000a690: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
+0000a6a0: 7469 6f6e 206f 6620 676c 6f62 616c 204d  tion of global M
+0000a6b0: 5242 4873 2a2a 0a60 6060 0a77 6764 2064  RBHs**.```.wgd d
+0000a6c0: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000a6d0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000a6e0: 202d 676d 202d 6520 3165 2d31 3020 2d62   -gm -e 1e-10 -b
+0000a6f0: 7320 3130 3020 2d6e 7020 3520 282d 6e6e  s 100 -np 5 (-nn
+0000a700: 2920 282d 6e20 3429 2028 2d6b 6629 2028  ) (-n 4) (-kf) (
+0000a710: 2d6b 6429 2028 2d63 2030 2e39 2920 282d  -kd) (-c 0.9) (-
+0000a720: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
+0000a730: 2920 282d 6f20 7767 645f 646d 6429 2028  ) (-o wgd_dmd) (
+0000a740: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
+0000a750: 6060 600a 0a54 6865 2067 6c6f 6261 6c20  ```..The global 
+0000a760: 4d52 4248 7320 6973 2063 6f6e 7374 7275  MRBHs is constru
+0000a770: 6374 6564 2062 7920 6578 6861 7573 7469  cted by exhausti
+0000a780: 7665 6c79 206d 6572 6769 6e67 2061 6c6c  vely merging all
+0000a790: 2074 6865 2070 6f73 7369 626c 6520 7061   the possible pa
+0000a7a0: 6972 2d77 6973 6520 5242 4873 2065 7863  ir-wise RBHs exc
+0000a7b0: 6570 7420 666f 7220 7175 6572 7969 6e67  ept for querying
+0000a7c0: 2074 6865 2073 6571 7565 6e63 6520 6974   the sequence it
+0000a7d0: 7365 6c66 2c20 7768 6963 6820 6361 6e20  self, which can 
+0000a7e0: 6265 2069 6e69 7469 6174 6564 2062 7920  be initiated by 
+0000a7f0: 6164 6420 7468 6520 666c 6167 2060 2d67  add the flag `-g
+0000a800: 6d60 206f 7220 602d 2d67 6c6f 6261 6c6d  m` or `--globalm
+0000a810: 7262 6860 2e20 5468 6520 7265 7374 206f  rbh`. The rest o
+0000a820: 6620 7265 6c65 7661 6e74 2070 6172 616d  f relevant param
+0000a830: 6574 6572 7320 7374 6179 7320 7468 6520  eters stays the 
+0000a840: 7361 6d65 2061 7320 7468 6520 6c6f 6361  same as the loca
+0000a850: 6c20 4d52 4248 732e 2054 6865 206e 756d  l MRBHs. The num
+0000a860: 6265 7220 6f66 2070 6172 616c 6c65 6c20  ber of parallel 
+0000a870: 7468 7265 6164 7320 6973 2073 7567 6765  threads is sugge
+0000a880: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
+0000a890: 7320 284e 2d31 294e 2f32 2074 6f6f 2077  s (N-1)N/2 too w
+0000a8a0: 6865 7265 204e 2069 7320 7468 6520 6e75  here N is the nu
+0000a8b0: 6d62 6572 206f 6620 6364 7320 6669 6c65  mber of cds file
+0000a8c0: 7320 746f 2061 6368 6965 7665 2074 6865  s to achieve the
+0000a8d0: 2068 6967 6865 7374 2065 6666 6963 6965   highest efficie
+0000a8e0: 6e63 792e 0a0a 2a2a 4120 7375 6767 6573  ncy...**A sugges
+0000a8f0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
+0000a900: 2069 7320 756e 6465 7220 7468 6520 6465   is under the de
+0000a910: 6661 756c 7420 7061 7261 6d65 7465 7220  fault parameter 
+0000a920: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
+0000a930: 2073 686f 776e 2061 7320 6265 6c6f 772e   shown as below.
+0000a940: 2a2a 0a0a 6060 600a 7767 6420 646d 6420  **..```.wgd dmd 
+0000a950: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000a960: 6365 3220 7365 7175 656e 6365 3320 2d67  ce2 sequence3 -g
+0000a970: 6d0a 6060 600a 0a2a 2a54 6865 2064 656c  m.```..**The del
+0000a980: 696e 6561 7469 6f6e 206f 6620 6f72 7468  ineation of orth
+0000a990: 6f67 726f 7570 732a 2a0a 6060 600a 7767  ogroups**.```.wg
+0000a9a0: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
+0000a9b0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000a9c0: 6365 3320 2d6f 6920 2d6f 6f20 2d65 2031  ce3 -oi -oo -e 1
+0000a9d0: 652d 3130 202d 6273 2031 3030 202d 6e70  e-10 -bs 100 -np
+0000a9e0: 2035 2028 2d6e 6e29 2028 2d63 6329 2028   5 (-nn) (-cc) (
+0000a9f0: 2d74 6529 2028 2d6d 6320 302e 3829 2028  -te) (-mc 0.8) (
+0000aa00: 2d67 6e29 2028 2d74 7265 6520 2766 6173  -gn) (-tree 'fas
+0000aa10: 7474 7265 6527 2920 282d 7473 2027 2d66  ttree') (-ts '-f
+0000aa20: 6173 7465 7374 2729 2028 2d6e 2034 2920  astest') (-n 4) 
+0000aa30: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+0000aa40: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
+0000aa50: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+0000aa60: 290a 6060 600a 0a49 6e20 6077 6764 2076  ).```..In `wgd v
+0000aa70: 3260 2c20 7765 2061 6c73 6f20 696d 706c  2`, we also impl
+0000aa80: 656d 656e 7465 6420 616e 2061 6c67 6f72  emented an algor
+0000aa90: 6974 686d 206f 6620 6465 6c69 6e65 6174  ithm of delineat
+0000aaa0: 696e 6720 6f72 7468 6f67 726f 7570 732c  ing orthogroups,
+0000aab0: 2077 6869 6368 2063 616e 2062 6520 696e   which can be in
+0000aac0: 6974 6961 7465 6420 7769 7468 2074 6865  itiated with the
+0000aad0: 2070 6172 616d 6574 6572 2060 2d6f 6960   parameter `-oi`
+0000aae0: 206f 7220 602d 2d6f 7274 686f 696e 6665   or `--orthoinfe
+0000aaf0: 7260 2e20 5477 6f20 7761 7973 206f 6620  r`. Two ways of 
+0000ab00: 6465 6c69 6e65 6174 696f 6e20 6361 6e20  delineation can 
+0000ab10: 6265 2063 686f 7365 6e2c 2074 6865 2063  be chosen, the c
+0000ab20: 6f6e 6361 7465 6e61 7469 6f6e 2077 6179  oncatenation way
+0000ab30: 2028 7365 7420 6279 2074 6865 2070 6172   (set by the par
+0000ab40: 616d 6574 6572 2060 2d63 6360 206f 7220  ameter `-cc` or 
+0000ab50: 602d 2d63 6f6e 6361 7460 2920 6f72 2074  `--concat`) or t
+0000ab60: 6865 206e 6f6e 2d63 6f6e 6361 7465 6e61  he non-concatena
+0000ab70: 7469 6f6e 2028 6465 6661 756c 7429 2077  tion (default) w
+0000ab80: 6179 2e20 496e 2062 7269 6566 2c20 7468  ay. In brief, th
+0000ab90: 6520 636f 6e63 6174 656e 6174 696f 6e20  e concatenation 
+0000aba0: 7761 7920 6f66 2064 656c 696e 6561 7469  way of delineati
+0000abb0: 6e67 206f 7274 686f 6772 6f75 7073 2073  ng orthogroups s
+0000abc0: 7461 7274 7320 7769 7468 2063 6f6e 6361  tarts with conca
+0000abd0: 7465 6e61 7469 6e67 2061 6c6c 2074 6865  tenating all the
+0000abe0: 2073 6571 7565 6e63 6573 2069 6e74 6f20   sequences into 
+0000abf0: 6120 7369 6e67 6c65 2073 6571 7565 6e63  a single sequenc
+0000ac00: 6520 6669 6c65 2061 6e64 2074 6865 6e20  e file and then 
+0000ac10: 696e 6665 7272 696e 6720 7468 6520 7768  inferring the wh
+0000ac20: 6f6c 6520 7061 7261 6e6f 6d65 206f 6620  ole paranome of 
+0000ac30: 7468 6973 2073 696e 676c 6520 7365 7175  this single sequ
+0000ac40: 656e 6365 2066 696c 6520 7769 7468 2074  ence file with t
+0000ac50: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
+0000ac60: 7375 6c74 7320 6d61 7070 6564 2062 6163  sults mapped bac
+0000ac70: 6b20 746f 2074 6865 2062 656c 6f6e 6769  k to the belongi
+0000ac80: 6e67 2073 7065 6369 6573 2e20 5768 696c  ng species. Whil
+0000ac90: 6520 7468 6520 6e6f 6e2d 636f 6e63 6174  e the non-concat
+0000aca0: 656e 6174 696f 6e20 7761 7920 7374 6172  enation way star
+0000acb0: 7473 2077 6974 6820 7265 7370 6563 7469  ts with respecti
+0000acc0: 7665 2070 6169 722d 7769 7365 2064 6961  ve pair-wise dia
+0000acd0: 6d6f 6e64 2073 6561 7263 6820 2869 6e63  mond search (inc
+0000ace0: 6c75 6469 6e67 2071 7565 7279 696e 6720  luding querying 
+0000acf0: 7468 6520 7361 6d65 2073 6571 7565 6e63  the same sequenc
+0000ad00: 6520 6974 7365 6c66 2920 616e 6420 7468  e itself) and th
+0000ad10: 656e 2061 6c6c 2074 6865 2073 6571 7565  en all the seque
+0000ad20: 6e63 6520 7369 6d69 6c61 7269 7479 2074  nce similarity t
+0000ad30: 6162 6c65 7320 7769 6c6c 2062 6520 636f  ables will be co
+0000ad40: 6e63 6174 656e 6174 6564 2061 6e64 2063  ncatenated and c
+0000ad50: 6c75 7374 6572 6564 2069 6e74 6f20 6f72  lustered into or
+0000ad60: 7468 6f67 726f 7570 732e 2053 6f6d 6520  thogroups. Some 
+0000ad70: 6f74 6865 7220 706f 7373 6962 6c79 2075  other possibly u
+0000ad80: 7365 6675 6c20 706f 7374 2d63 6c75 7374  seful post-clust
+0000ad90: 6572 696e 6720 6675 6e63 7469 6f6e 7320  ering functions 
+0000ada0: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
+0000adb0: 2c20 696e 636c 7564 696e 6720 7468 6520  , including the 
+0000adc0: 7061 7261 6d65 7465 7220 602d 7465 6020  parameter `-te` 
+0000add0: 6f72 2060 2d2d 7465 7374 736f 6760 2c20  or `--testsog`, 
+0000ade0: 7768 6963 6820 6361 6e20 6265 2073 6574  which can be set
+0000adf0: 2074 6f20 7374 6172 7420 7468 6520 756e   to start the un
+0000ae00: 6269 6173 6564 2074 6573 7420 6f66 2073  biased test of s
+0000ae10: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
+0000ae20: 6661 6d69 6c69 6573 2028 6e6f 7465 2074  families (note t
+0000ae30: 6861 7420 7468 6973 2066 756e 6374 696f  hat this functio
+0000ae40: 6e20 6e65 6564 7320 6068 6d6d 6572 6020  n needs `hmmer` 
+0000ae50: 2876 332e 3162 3229 2074 6f20 6265 2069  (v3.1b2) to be i
+0000ae60: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+0000ae70: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+0000ae80: 292c 2074 6865 2070 6172 616d 6574 6572  ), the parameter
+0000ae90: 2060 2d6d 6360 206f 7220 602d 2d6d 736f   `-mc` or `--mso
+0000aea0: 6763 7574 602c 2072 616e 6769 6e67 2062  gcut`, ranging b
+0000aeb0: 6574 7765 656e 2030 2074 6f20 312c 2077  etween 0 to 1, w
+0000aec0: 6869 6368 2063 616e 2062 6520 7365 7420  hich can be set 
+0000aed0: 746f 2073 6561 7263 6820 7468 6520 736f  to search the so
+0000aee0: 2d63 616c 6c65 6420 6d6f 7374 6c79 2073  -called mostly s
+0000aef0: 696e 676c 652d 636f 7079 2066 616d 696c  ingle-copy famil
+0000af00: 7920 7768 6963 6820 6861 7320 6869 6768  y which has high
+0000af10: 6572 2074 6861 6e20 6365 7274 6169 6e20  er than certain 
+0000af20: 6375 742d 6f66 6620 7065 7263 656e 7461  cut-off percenta
+0000af30: 6765 206f 6620 7370 6563 6965 7320 636f  ge of species co
+0000af40: 7665 7261 6765 2c20 7468 6520 7061 7261  verage, the para
+0000af50: 6d65 7465 7220 602d 676e 6020 6f72 2060  meter `-gn` or `
+0000af60: 2d2d 6765 746e 736f 6760 2c20 7768 6963  --getnsog`, whic
+0000af70: 6820 6361 6e20 6265 2073 6574 2074 6f20  h can be set to 
+0000af80: 7365 6172 6368 2066 6f72 206e 6573 7465  search for neste
+0000af90: 6420 7369 6e67 6c65 2d63 6f70 7920 6765  d single-copy ge
+0000afa0: 6e65 2066 616d 696c 6965 7320 284e 534f  ne families (NSO
+0000afb0: 4773 2920 7768 6963 6820 6973 206f 7269  Gs) which is ori
+0000afc0: 6769 6e61 6c6c 7920 6d75 6c74 6979 2d63  ginally multiy-c
+0000afd0: 6f70 7920 6275 7420 6861 7320 6120 286d  opy but has a (m
+0000afe0: 6f73 746c 7929 2073 696e 676c 652d 636f  ostly) single-co
+0000aff0: 7079 2062 7261 6e63 6820 2877 6869 6368  py branch (which
+0000b000: 2072 6571 7569 7265 7320 7468 6520 6368   requires the ch
+0000b010: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+0000b020: 6e63 6520 7072 6f67 7261 6d20 7365 7420  nce program set 
+0000b030: 6279 2060 2d74 7265 6560 206f 7220 602d  by `-tree` or `-
+0000b040: 2d74 7265 655f 6d65 7468 6f64 6020 746f  -tree_method` to
+0000b050: 2062 6520 7072 652d 696e 7374 616c 6c65   be pre-installe
+0000b060: 6420 696e 2074 6865 2065 6e76 6972 6f6e  d in the environ
+0000b070: 6d65 6e74 2070 6174 6820 7769 7468 2074  ment path with t
+0000b080: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
+0000b090: 7474 696e 6720 666f 7220 6765 6e65 2074  tting for gene t
+0000b0a0: 7265 6520 696e 6665 7265 6e63 6520 636f  ree inference co
+0000b0b0: 6e74 726f 6c6c 6564 2062 7920 602d 7473  ntrolled by `-ts
+0000b0c0: 6020 6f72 2060 2d2d 7472 6565 7365 7460  ` or `--treeset`
+0000b0d0: 292e 2054 6865 2070 726f 6772 616d 2060  ). The program `
+0000b0e0: 7767 6420 646d 6460 2077 6f75 6c64 2073  wgd dmd` would s
+0000b0f0: 7469 6c6c 2063 6f6e 6475 6374 2074 6865  till conduct the
+0000b100: 2052 4248 7320 6361 6c63 756c 6174 696f   RBHs calculatio
+0000b110: 6e20 756e 6c65 7373 2074 6865 2070 6172  n unless the par
+0000b120: 616d 6574 6572 2060 2d6f 6f60 206f 7220  ameter `-oo` or 
+0000b130: 602d 2d6f 6e6c 796f 7274 686f 6020 7761  `--onlyortho` wa
+0000b140: 7320 7365 742e 2049 6620 6f6e 6520 6f6e  s set. If one on
+0000b150: 6c79 2077 616e 7473 2074 6f20 696e 6665  ly wants to infe
+0000b160: 7220 7468 6520 6f72 7468 6f67 726f 7570  r the orthogroup
+0000b170: 732c 2069 7427 7320 7375 6767 6573 7465  s, it's suggeste
+0000b180: 6420 746f 2061 6464 2074 6865 2066 6c61  d to add the fla
+0000b190: 6720 602d 6f6f 6020 746f 206a 7573 7420  g `-oo` to just 
+0000b1a0: 696d 706c 656d 656e 7420 7468 6520 6f72  implement the or
+0000b1b0: 7468 6f67 726f 7570 7320 6465 6c69 6e65  thogroups deline
+0000b1c0: 6174 696f 6e20 616e 616c 7973 6973 2e20  ation analysis. 
+0000b1d0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
+0000b1e0: 7261 6c6c 656c 2074 6872 6561 6473 2069  rallel threads i
+0000b1f0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000b200: 6520 7365 7420 6173 2028 4e2b 3129 4e2f  e set as (N+1)N/
+0000b210: 3220 7768 6572 6520 4e20 6973 2074 6865  2 where N is the
+0000b220: 206e 756d 6265 7220 6f66 2063 6473 2066   number of cds f
+0000b230: 696c 6573 2074 6f20 6163 6869 6576 6520  iles to achieve 
+0000b240: 7468 6520 6869 6768 6573 7420 6566 6669  the highest effi
+0000b250: 6369 656e 6379 2073 696e 6365 2074 6865  ciency since the
+0000b260: 2073 656c 662d 636f 6d70 6172 6973 6f6e   self-comparison
+0000b270: 2069 7320 616c 736f 2069 6e63 6c75 6465   is also include
+0000b280: 642e 0a0a 2a2a 5468 6520 6465 6661 756c  d...**The defaul
+0000b290: 7420 7365 7474 696e 6720 6f66 2070 6172  t setting of par
+0000b2a0: 616d 6574 6572 7320 6973 2061 2072 6561  ameters is a rea
+0000b2b0: 736f 6e61 626c 6520 7374 6172 7469 6e67  sonable starting
+0000b2c0: 2070 6f69 6e74 2077 6974 6820 7468 6520   point with the 
+0000b2d0: 636f 6d6d 616e 6420 6173 2062 656c 6f77  command as below
+0000b2e0: 2e2a 2a0a 0a60 6060 0a77 6764 2064 6d64  .**..```.wgd dmd
+0000b2f0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
+0000b300: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
+0000b310: 6f69 202d 6f6f 0a60 6060 0a0a 2a2a 5468  oi -oo.```..**Th
+0000b320: 6520 636f 6c6c 696e 6561 7220 636f 616c  e collinear coal
+0000b330: 6573 6365 6e63 6520 696e 6665 7265 6e63  escence inferenc
+0000b340: 6520 6f66 2070 6879 6c6f 6765 6e79 2a2a  e of phylogeny**
+0000b350: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
+0000b360: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000b370: 2073 6571 7565 6e63 6533 202d 6170 2061   sequence3 -ap a
+0000b380: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
+0000b390: 202d 6c65 206c 6564 6174 6120 2d67 7420   -le ledata -gt 
+0000b3a0: 6774 6461 7461 202d 636f 6320 282d 7472  gtdata -coc (-tr
+0000b3b0: 6565 2027 6661 7374 7472 6565 2729 2028  ee 'fasttree') (
+0000b3c0: 2d74 7320 272d 6661 7374 6573 7427 2920  -ts '-fastest') 
+0000b3d0: 282d 6e20 3429 2028 2d2d 746f 5f73 746f  (-n 4) (--to_sto
+0000b3e0: 7029 2028 2d2d 6364 7329 2028 2d6f 2077  p) (--cds) (-o w
+0000b3f0: 6764 5f64 6d64 2920 282d 7420 776f 726b  gd_dmd) (-t work
+0000b400: 696e 675f 746d 7029 0a60 6060 0a0a 4120  ing_tmp).```..A 
+0000b410: 6e6f 7665 6c20 7068 796c 6f67 656e 6574  novel phylogenet
+0000b420: 6963 2069 6e66 6572 656e 6365 206d 6574  ic inference met
+0000b430: 686f 6420 6e61 6d65 6420 2263 6f6c 6c69  hod named "colli
+0000b440: 6e65 6172 2063 6f61 6c65 7363 656e 6365  near coalescence
+0000b450: 2069 6e66 6572 656e 6365 2220 6973 2061   inference" is a
+0000b460: 6c73 6f20 696d 706c 656d 656e 7465 6420  lso implemented 
+0000b470: 696e 2060 7767 6420 7632 602e 2046 6f72  in `wgd v2`. For
+0000b480: 2074 6869 7320 616e 616c 7973 6973 2c20   this analysis, 
+0000b490: 7573 6572 7320 6e65 6564 2074 6f20 7072  users need to pr
+0000b4a0: 6f76 6964 6520 7468 6520 616e 6368 6f72  ovide the anchor
+0000b4b0: 2070 6f69 6e74 7320 6669 6c65 2062 7920   points file by 
+0000b4c0: 602d 6170 6020 6f72 2060 2d2d 616e 6368  `-ap` or `--anch
+0000b4d0: 6f72 706f 696e 7473 602c 2074 6865 2063  orpoints`, the c
+0000b4e0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
+0000b4f0: 7320 6669 6c65 2062 7920 602d 736d 6020  s file by `-sm` 
+0000b500: 6f72 2060 2d2d 7365 676d 656e 7473 602c  or `--segments`,
+0000b510: 2074 6865 206c 6973 7473 6567 6d65 6e74   the listsegment
+0000b520: 7320 6669 6c65 2062 7920 602d 6c65 6020  s file by `-le` 
+0000b530: 6f72 2060 2d2d 6c69 7374 656c 656d 656e  or `--listelemen
+0000b540: 7473 602c 2061 6e64 2074 6865 2067 656e  ts`, and the gen
+0000b550: 6520 7461 626c 6520 6669 6c65 2062 7920  e table file by 
+0000b560: 602d 6774 6020 6f72 2060 2d2d 6765 6e65  `-gt` or `--gene
+0000b570: 7461 626c 6560 2c20 616c 6c20 6f66 2077  table`, all of w
+0000b580: 6869 6368 2063 616e 2062 6520 7072 6f64  hich can be prod
+0000b590: 7563 6564 2069 6e20 7468 6520 7072 6f67  uced in the prog
+0000b5a0: 7261 6d20 6077 6764 2073 796e 602e 2054  ram `wgd syn`. T
+0000b5b0: 6865 2070 6172 616d 6574 6572 2060 2d63  he parameter `-c
+0000b5c0: 6f63 6020 6f72 2060 2d2d 636f 6c6c 696e  oc` or `--collin
+0000b5d0: 6561 7263 6f61 6c65 7363 656e 6365 6020  earcoalescence` 
+0000b5e0: 6e65 6564 7320 746f 2062 6520 7365 7420  needs to be set 
+0000b5f0: 746f 2073 7461 7274 2074 6869 7320 616e  to start this an
+0000b600: 616c 7973 6973 2e20 5468 6520 7472 6565  alysis. The tree
+0000b610: 2d69 6e66 6572 656e 6365 2070 726f 6772  -inference progr
+0000b620: 616d 2061 6e64 2074 6865 2061 7373 6f63  am and the assoc
+0000b630: 6961 7465 6420 7061 7261 6d65 7465 7273  iated parameters
+0000b640: 2063 616e 2062 6520 7365 7420 6a75 7374   can be set just
+0000b650: 2061 7320 6162 6f76 6520 6279 2060 2d74   as above by `-t
+0000b660: 7265 6560 206f 7220 602d 2d74 7265 655f  ree` or `--tree_
+0000b670: 6d65 7468 6f64 6020 616e 6420 602d 7473  method` and `-ts
+0000b680: 6020 6f72 2060 2d2d 7472 6565 7365 7460  ` or `--treeset`
+0000b690: 2e20 506c 6561 7365 2061 6c73 6f20 6d61  . Please also ma
+0000b6a0: 6b65 2073 7572 6520 7468 6520 6368 6f73  ke sure the chos
+0000b6b0: 656e 2074 7265 652d 696e 6665 7265 6e63  en tree-inferenc
+0000b6c0: 6520 7072 6f67 7261 6d20 6973 2069 6e73  e program is ins
+0000b6d0: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+0000b6e0: 7669 726f 6e6d 656e 7420 7061 7468 2e20  vironment path. 
+0000b6f0: 5468 6520 7072 6f67 7261 6d20 6061 7374  The program `ast
+0000b700: 7261 6c2d 7072 6f60 2069 7320 7265 7175  ral-pro` is requ
+0000b710: 6972 6564 2074 6f20 6265 2069 6e73 7461  ired to be insta
+0000b720: 6c6c 6564 2069 6e20 7468 6520 656e 7669  lled in the envi
+0000b730: 726f 6e6d 656e 7420 7061 7468 2074 6f6f  ronment path too
+0000b740: 2e20 4e6f 7465 2074 6861 7420 7468 6572  . Note that ther
+0000b750: 6520 7368 6f75 6c64 2062 6520 6e6f 2064  e should be no d
+0000b760: 7570 6c69 6361 7465 6420 6765 6e65 2049  uplicated gene I
+0000b770: 4473 2069 6e20 7468 6520 7365 7175 656e  Ds in the sequen
+0000b780: 6365 2066 696c 652e 2054 6865 2070 6172  ce file. The par
+0000b790: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
+0000b7a0: 7265 2061 7265 2074 6f20 6163 6365 6c65  re are to accele
+0000b7b0: 7261 7465 2074 6865 2073 6571 7565 6e63  rate the sequenc
+0000b7c0: 6520 616c 6967 6e6d 656e 7420 616e 6420  e alignment and 
+0000b7d0: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
+0000b7e0: 6e63 6520 666f 7220 6561 6368 2067 656e  nce for each gen
+0000b7f0: 6520 6661 6d69 6c79 2061 6e64 2074 6875  e family and thu
+0000b800: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000b810: 6520 7365 7420 6173 206d 7563 6820 6173  e set as much as
+0000b820: 2074 6865 206e 756d 6265 7220 6f66 2067   the number of g
+0000b830: 656e 6520 6661 6d69 6c69 6573 2e0a 0a2a  ene families...*
+0000b840: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+0000b850: 7274 696e 6720 7275 6e20 6f66 2074 6869  rting run of thi
+0000b860: 7320 616e 616c 7973 6973 2069 7320 7769  s analysis is wi
+0000b870: 7468 2074 6865 2073 696d 706c 6520 636f  th the simple co
+0000b880: 6d6d 616e 6420 6265 6c6f 772e 2a2a 0a0a  mmand below.**..
+0000b890: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
+0000b8a0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000b8b0: 7365 7175 656e 6365 3320 2d61 7020 6170  sequence3 -ap ap
+0000b8c0: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
+0000b8d0: 2d6c 6520 6c65 6461 7461 202d 6774 2067  -le ledata -gt g
+0000b8e0: 7464 6174 6120 2d63 6f63 0a60 6060 0a0a  tdata -coc.```..
+0000b8f0: 2323 2320 7767 6420 666f 6375 730a 0a2a  ### wgd focus..*
+0000b900: 2a54 6865 2063 6f6e 6361 7465 6e61 7469  *The concatenati
+0000b910: 6f6e 2d62 6173 6564 2f63 6f61 6c65 7363  on-based/coalesc
+0000b920: 656e 6365 2d62 6173 6564 2070 6879 6c6f  ence-based phylo
+0000b930: 6765 6e65 7469 6320 696e 6665 7265 6e63  genetic inferenc
+0000b940: 652a 2a0a 6060 600a 7767 6420 666f 6375  e**.```.wgd focu
+0000b950: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000b960: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000b970: 6571 7565 6e63 6533 2028 2d2d 636f 6e63  equence3 (--conc
+0000b980: 6174 656e 6174 696f 6e29 2028 2d2d 636f  atenation) (--co
+0000b990: 616c 6573 6365 6e63 6529 2028 2d74 7265  alescence) (-tre
+0000b9a0: 6520 2766 6173 7474 7265 6527 2920 282d  e 'fasttree') (-
+0000b9b0: 7473 2027 2d66 6173 7465 7374 2729 2028  ts '-fastest') (
+0000b9c0: 2d6e 2034 2920 282d 2d74 6f5f 7374 6f70  -n 4) (--to_stop
+0000b9d0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+0000b9e0: 645f 666f 6375 7329 2028 2d74 2077 6f72  d_focus) (-t wor
+0000b9f0: 6b69 6e67 5f74 6d70 290a 6060 600a 0a54  king_tmp).```..T
+0000ba00: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+0000ba10: 666f 6375 7360 2069 6d70 6c65 6d65 6e74  focus` implement
+0000ba20: 6564 2074 776f 2062 6173 6963 2070 6879  ed two basic phy
+0000ba30: 6c6f 6765 6e65 7469 6320 696e 6665 7265  logenetic infere
+0000ba40: 6e63 6520 6d65 7468 6f64 732c 2069 2e65  nce methods, i.e
+0000ba50: 2e2c 2063 6f6e 6361 7465 6e61 7469 6f6e  ., concatenation
+0000ba60: 2d62 6173 6564 2061 6e64 2063 6f61 6c65  -based and coale
+0000ba70: 7363 656e 6365 2d62 6173 6564 206d 6574  scence-based met
+0000ba80: 686f 6473 2e20 546f 2069 6e69 7469 6174  hods. To initiat
+0000ba90: 6520 7468 6573 6520 616e 616c 7973 6973  e these analysis
+0000baa0: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000bab0: 7365 7420 7468 6520 666c 6167 206f 7074  set the flag opt
+0000bac0: 696f 6e20 602d 2d63 6f6e 6361 7465 6e61  ion `--concatena
+0000bad0: 7469 6f6e 6020 6f72 2060 2d2d 636f 616c  tion` or `--coal
+0000bae0: 6573 6365 6e63 6560 2e20 5468 6520 636f  escence`. The co
+0000baf0: 6e63 6174 656e 6174 696f 6e2d 6261 7365  ncatenation-base
+0000bb00: 6420 6d65 7468 6f64 2069 6e63 6c75 6465  d method include
+0000bb10: 7320 6120 6665 7720 6d61 6a6f 7220 7374  s a few major st
+0000bb20: 6570 732c 2069 2e65 2e2c 2074 6865 206d  eps, i.e., the m
+0000bb30: 756c 7469 706c 6520 7365 7175 656e 6365  ultiple sequence
+0000bb40: 2061 6c69 676e 6d65 6e74 2028 4d53 4129   alignment (MSA)
+0000bb50: 206f 6620 6561 6368 2067 656e 6520 6661   of each gene fa
+0000bb60: 6d69 6c79 2c20 7468 6520 636f 6e63 6174  mily, the concat
+0000bb70: 656e 6174 696f 6e20 6f66 2061 6c6c 2067  enation of all g
+0000bb80: 656e 6520 6661 6d69 6c69 6573 2061 6e64  ene families and
+0000bb90: 2074 6865 6e20 7468 6520 6765 6e65 2074   then the gene t
+0000bba0: 7265 6520 2861 6c73 6f20 7370 6563 6965  ree (also specie
+0000bbb0: 7320 7472 6565 2069 6e20 7468 6973 2063  s tree in this c
+0000bbc0: 6173 6529 2069 6e66 6572 656e 6365 2e20  ase) inference. 
+0000bbd0: 5468 6520 636f 616c 6573 6365 6e63 652d  The coalescence-
+0000bbe0: 6261 7365 6420 6d65 7468 6f64 2077 696c  based method wil
+0000bbf0: 6c20 696e 7374 6561 6420 7065 7266 6f72  l instead perfor
+0000bc00: 6d20 7468 6520 4d53 4120 6f66 2065 6163  m the MSA of eac
+0000bc10: 6820 6765 6e65 2066 616d 696c 7920 616e  h gene family an
+0000bc20: 6420 7468 6520 6765 6e65 2074 7265 6520  d the gene tree 
+0000bc30: 696e 6665 7265 6e63 6520 6261 7365 6420  inference based 
+0000bc40: 6f6e 2065 6163 6820 4d53 412c 2061 6e64  on each MSA, and
+0000bc50: 2074 6865 6e20 696e 6665 7220 7468 6520   then infer the 
+0000bc60: 7370 6563 6965 7320 7472 6565 2062 6173  species tree bas
+0000bc70: 6564 206f 6e20 7468 6573 6520 696e 6469  ed on these indi
+0000bc80: 7669 6475 616c 2067 656e 6520 7472 6565  vidual gene tree
+0000bc90: 732e 2054 6865 2074 7265 652d 696e 6665  s. The tree-infe
+0000bca0: 7265 6e63 6520 7072 6f67 7261 6d20 616e  rence program an
+0000bcb0: 6420 7468 6520 6173 736f 6369 6174 6564  d the associated
+0000bcc0: 2070 6172 616d 6574 6572 7320 6361 6e20   parameters can 
+0000bcd0: 6265 2073 6574 206a 7573 7420 6173 2061  be set just as a
+0000bce0: 626f 7665 2062 7920 602d 7472 6565 6020  bove by `-tree` 
+0000bcf0: 6f72 2060 2d2d 7472 6565 5f6d 6574 686f  or `--tree_metho
+0000bd00: 6460 2061 6e64 2060 2d74 7360 206f 7220  d` and `-ts` or 
+0000bd10: 602d 2d74 7265 6573 6574 602e 2050 6c65  `--treeset`. Ple
+0000bd20: 6173 6520 616c 736f 206d 616b 6520 7375  ase also make su
+0000bd30: 7265 2074 6865 2063 686f 7365 6e20 7472  re the chosen tr
+0000bd40: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+0000bd50: 6772 616d 2069 7320 696e 7374 616c 6c65  gram is installe
+0000bd60: 6420 696e 2074 6865 2065 6e76 6972 6f6e  d in the environ
+0000bd70: 6d65 6e74 2070 6174 682e 2054 6865 2070  ment path. The p
+0000bd80: 726f 6772 616d 2060 6173 7472 616c 2d70  rogram `astral-p
+0000bd90: 726f 6020 6973 2072 6571 7569 7265 6420  ro` is required 
+0000bda0: 746f 2062 6520 696e 7374 616c 6c65 6420  to be installed 
+0000bdb0: 696e 2074 6865 2065 6e76 6972 6f6e 6d65  in the environme
+0000bdc0: 6e74 2070 6174 6820 6966 2074 6865 2063  nt path if the c
+0000bdd0: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
+0000bde0: 206d 6574 686f 6420 6973 2063 686f 7365   method is chose
+0000bdf0: 6e2e 2054 6865 2070 6172 616c 6c65 6c20  n. The parallel 
+0000be00: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
+0000be10: 2074 6f20 6163 6365 6c65 7261 7465 2074   to accelerate t
+0000be20: 6865 2073 6571 7565 6e63 6520 616c 6967  he sequence alig
+0000be30: 6e6d 656e 7420 616e 6420 6765 6e65 2074  nment and gene t
+0000be40: 7265 6520 696e 6665 7265 6e63 6520 666f  ree inference fo
+0000be50: 7220 6561 6368 2067 656e 6520 6661 6d69  r each gene fami
+0000be60: 6c79 2074 6f6f 2061 6e64 2074 6875 7320  ly too and thus 
+0000be70: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+0000be80: 7365 7420 6173 206d 7563 6820 6173 2074  set as much as t
+0000be90: 6865 206e 756d 6265 7220 6f66 2067 656e  he number of gen
+0000bea0: 6520 6661 6d69 6c69 6573 2e0a 0a2a 2a41  e families...**A
+0000beb0: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
+0000bec0: 696e 6720 7275 6e20 6f66 2074 6869 7320  ing run of this 
+0000bed0: 616e 616c 7973 6973 2069 7320 7769 7468  analysis is with
+0000bee0: 2074 6865 2073 696d 706c 6520 636f 6d6d   the simple comm
+0000bef0: 616e 6420 6265 6c6f 772e 2a2a 0a0a 6060  and below.**..``
+0000bf00: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
+0000bf10: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+0000bf20: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000bf30: 6533 202d 2d63 6f6e 6361 7465 6e61 7469  e3 --concatenati
+0000bf40: 6f6e 0a77 6764 2066 6f63 7573 2066 616d  on.wgd focus fam
+0000bf50: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000bf60: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000bf70: 6365 3320 2d2d 636f 616c 6573 6365 6e63  ce3 --coalescenc
+0000bf80: 650a 6060 600a 0a2a 2a54 6865 2066 756e  e.```..**The fun
+0000bf90: 6374 696f 6e61 6c20 616e 6e6f 7461 7469  ctional annotati
+0000bfa0: 6f6e 206f 6620 6765 6e65 2066 616d 696c  on of gene famil
+0000bfb0: 6965 732a 2a0a 6060 600a 7767 6420 666f  ies**.```.wgd fo
+0000bfc0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000bfd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000bfe0: 2073 6571 7565 6e63 6533 202d 2d61 6e6e   sequence3 --ann
+0000bff0: 6f74 6174 696f 6e20 6567 676e 6f67 202d  otation eggnog -
+0000c000: 6564 2065 6464 6174 6120 2d2d 646d 6e62  ed eddata --dmnb
+0000c010: 2064 6264 6174 610a 6060 600a 0a54 6865   dbdata.```..The
+0000c020: 2070 726f 6772 616d 2060 7767 6420 666f   program `wgd fo
+0000c030: 6375 7360 2061 6c73 6f20 6164 6465 6420  cus` also added 
+0000c040: 736f 6d65 2077 7261 7070 696e 6720 6675  some wrapping fu
+0000c050: 6e63 7469 6f6e 7320 666f 7220 6675 6e63  nctions for func
+0000c060: 7469 6f6e 616c 2061 6e6e 6f74 6174 696f  tional annotatio
+0000c070: 6e20 6f66 2067 656e 6520 6661 6d69 6c69  n of gene famili
+0000c080: 6573 206f 6e20 7468 6520 686f 6f64 206f  es on the hood o
+0000c090: 6620 6461 7461 6261 7365 7320 616e 6420  f databases and 
+0000c0a0: 736f 6674 7761 7265 7320 666f 7220 696e  softwares for in
+0000c0b0: 7374 616e 6365 2060 4567 674e 4f47 6020  stance `EggNOG` 
+0000c0c0: 616e 6420 6045 6767 4e4f 472d 6d61 7070  and `EggNOG-mapp
+0000c0d0: 6572 602e 2046 6f72 2074 6865 2061 6e6e  er`. For the ann
+0000c0e0: 6f74 6174 696f 6e20 7573 696e 6720 6045  otation using `E
+0000c0f0: 6767 4e4f 472d 6d61 7070 6572 602c 2075  ggNOG-mapper`, u
+0000c100: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+0000c110: 7669 6465 2074 6865 2070 6174 6820 746f  vide the path to
+0000c120: 2074 6865 2065 6767 4e4f 4720 616e 6e6f   the eggNOG anno
+0000c130: 7461 7469 6f6e 2064 6174 6162 6173 6520  tation database 
+0000c140: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+0000c150: 2d65 6460 2c20 7468 6520 7061 7468 2074  -ed`, the path t
+0000c160: 6f20 7468 6520 6469 616d 6f6e 642d 636f  o the diamond-co
+0000c170: 6d70 6174 6962 6c65 2064 6174 6162 6173  mpatible databas
+0000c180: 6520 7669 6120 7468 6520 6f70 7469 6f6e  e via the option
+0000c190: 2060 2d2d 646d 6e64 5f64 6260 2061 6e64   `--dmnd_db` and
+0000c1a0: 2074 6865 206f 7074 696f 6e20 602d 2d61   the option `--a
+0000c1b0: 6e6e 6f74 6174 696f 6e60 2073 6574 2061  nnotation` set a
+0000c1c0: 7320 2265 6767 6e6f 6722 2e20 5468 6520  s "eggnog". The 
+0000c1d0: 6d61 6e6e 6572 2068 6f77 2050 4641 4d20  manner how PFAM 
+0000c1e0: 616e 6e6f 7461 7469 6f6e 2077 696c 6c20  annotation will 
+0000c1f0: 6265 2070 6572 666f 726d 6564 2063 616e  be performed can
+0000c200: 2062 6520 636f 6e74 726f 6c6c 6564 2076   be controlled v
+0000c210: 6961 2074 6865 206f 7074 696f 6e20 6070  ia the option `p
+0000c220: 6661 6d60 2c20 6569 7468 6572 2022 6e6f  fam`, either "no
+0000c230: 6e65 222c 2022 7265 616c 6967 6e22 206f  ne", "realign" o
+0000c240: 7220 2264 656e 6f76 6f22 2c20 7468 6520  r "denovo", the 
+0000c250: 6465 7461 696c 6564 2065 7870 6c61 6e61  detailed explana
+0000c260: 7469 6f6e 2063 616e 2062 6520 666f 756e  tion can be foun
+0000c270: 6420 6174 2074 6865 2077 696b 6920 6f66  d at the wiki of
+0000c280: 205b 4567 674e 4f47 2d6d 6170 7065 725d   [EggNOG-mapper]
+0000c290: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+0000c2a0: 636f 6d2f 6567 676e 6f67 6462 2f65 6767  com/eggnogdb/egg
+0000c2b0: 6e6f 672d 6d61 7070 6572 2f77 696b 6929  nog-mapper/wiki)
+0000c2c0: 2e20 506c 6561 7365 2070 7265 2d69 6e73  . Please pre-ins
+0000c2d0: 7461 6c6c 2074 6865 2060 4567 674e 4f47  tall the `EggNOG
+0000c2e0: 2d6d 6170 7065 7260 2070 7974 686f 6e20  -mapper` python 
+0000c2f0: 7061 636b 6167 6520 6966 2075 7369 6e67  package if using
+0000c300: 2074 6869 7320 6675 6e63 7469 6f6e 2e20   this function. 
+0000c310: 466f 7220 7468 6520 616e 6e6f 7461 7469  For the annotati
+0000c320: 6f6e 2075 7369 6e67 2060 686d 6d73 6361  on using `hmmsca
+0000c330: 6e60 2c20 7768 6174 2069 7320 696d 706c  n`, what is impl
+0000c340: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+0000c350: 7632 6020 6973 2061 2073 696d 706c 6520  v2` is a simple 
+0000c360: 6275 6e64 6c65 2066 756e 6374 696f 6e20  bundle function 
+0000c370: 746f 2070 6572 666f 726d 2060 686d 6d73  to perform `hmms
+0000c380: 6361 6e60 2061 6e61 6c79 7369 7320 666f  can` analysis fo
+0000c390: 7220 6120 6769 7665 6e20 686d 6d20 7072  r a given hmm pr
+0000c3a0: 6f66 696c 6520 616e 6420 7365 7420 6f66  ofile and set of
+0000c3b0: 2067 656e 6520 6661 6d69 6c69 6573 2073   gene families s
+0000c3c0: 7563 6820 7468 6174 2075 7365 7273 206e  uch that users n
+0000c3d0: 6565 6420 746f 2073 6574 2074 6865 206f  eed to set the o
+0000c3e0: 7074 696f 6e20 602d 2d61 6e6e 6f74 6174  ption `--annotat
+0000c3f0: 696f 6e60 2061 7320 2268 6d6d 7066 616d  ion` as "hmmpfam
+0000c400: 2220 616e 6420 7072 6f76 6964 6520 7468  " and provide th
+0000c410: 6520 686d 6d70 726f 6669 6c65 2076 6961  e hmmprofile via
+0000c420: 2074 6865 206f 7074 696f 6e20 602d 2d68   the option `--h
+0000c430: 6d6d 602e 2046 6f72 2074 6865 2061 6e6e  mm`. For the ann
+0000c440: 6f74 6174 696f 6e20 7573 696e 6720 6069  otation using `i
+0000c450: 6e74 6572 7072 6f73 6361 6e60 2c20 7573  nterproscan`, us
+0000c460: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
+0000c470: 6964 6520 7468 6520 7061 7468 2074 6f20  ide the path to 
+0000c480: 7468 6520 696e 7465 7270 726f 7363 616e  the interproscan
+0000c490: 2069 6e73 7461 6c6c 6174 696f 6e20 666f   installation fo
+0000c4a0: 6c64 6572 2077 6865 7265 2074 6865 7265  lder where there
+0000c4b0: 2069 7320 6120 6069 6e74 6572 7072 6f73   is a `interpros
+0000c4c0: 6361 6e2e 7368 6020 6669 6c65 2076 6961  can.sh` file via
+0000c4d0: 2074 6865 206f 7074 696f 6e20 602d 2d65   the option `--e
+0000c4e0: 7865 7061 7468 6020 616e 6420 7365 7420  xepath` and set 
+0000c4f0: 7468 6520 7468 6520 6f70 7469 6f6e 2060  the the option `
+0000c500: 2d2d 616e 6e6f 7461 7469 6f6e 6020 6173  --annotation` as
+0000c510: 2022 696e 7465 7270 726f 7363 616e 222e   "interproscan".
+0000c520: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
+0000c530: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
+0000c540: 6f20 7061 7261 6c6c 656c 697a 6520 7468  o parallelize th
+0000c550: 6520 616e 616c 7973 6973 2066 6f72 2065  e analysis for e
+0000c560: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
+0000c570: 616e 6420 7468 7573 2073 7567 6765 7374  and thus suggest
+0000c580: 6564 2074 6f20 6265 2073 6574 2061 7320  ed to be set as 
+0000c590: 6d75 6368 2061 7320 7468 6520 6e75 6d62  much as the numb
+0000c5a0: 6572 206f 6620 6765 6e65 2066 616d 696c  er of gene famil
+0000c5b0: 6965 732e 0a0a 2a2a 4120 7375 6767 6573  ies...**A sugges
+0000c5c0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
+0000c5d0: 206f 6620 7468 6973 2061 6e61 6c79 7369   of this analysi
+0000c5e0: 7320 6361 6e20 6265 2077 6974 6820 7468  s can be with th
+0000c5f0: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
+0000c600: 2a2a 0a0a 6060 600a 7767 6420 666f 6375  **..```.wgd focu
+0000c610: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000c620: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000c630: 6571 7565 6e63 6533 202d 2d61 6e6e 6f74  equence3 --annot
+0000c640: 6174 696f 6e20 6567 676e 6f67 202d 6564  ation eggnog -ed
+0000c650: 2065 6464 6174 6120 2d2d 646d 6e62 2064   eddata --dmnb d
+0000c660: 6264 6174 610a 7767 6420 666f 6375 7320  bdata.wgd focus 
+0000c670: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
+0000c680: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
+0000c690: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
+0000c6a0: 696f 6e20 686d 6d70 6661 6d20 2d2d 686d  ion hmmpfam --hm
+0000c6b0: 6d20 686d 6d64 6174 610a 7767 6420 666f  m hmmdata.wgd fo
+0000c6c0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000c6d0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000c6e0: 2073 6571 7565 6e63 6533 202d 2d61 6e6e   sequence3 --ann
+0000c6f0: 6f74 6174 696f 6e20 696e 7465 7270 726f  otation interpro
+0000c700: 7363 616e 202d 2d65 7865 7061 7468 2024  scan --exepath $
+0000c710: 5041 5448 0a60 6060 0a0a 2a2a 5468 6520  PATH.```..**The 
+0000c720: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+0000c730: 696e 6720 6f66 2057 4744 732a 2a0a 6060  ing of WGDs**.``
+0000c740: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
+0000c750: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+0000c760: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000c770: 6533 202d 6420 6d63 6d63 7472 6565 202d  e3 -d mcmctree -
+0000c780: 7370 2073 7064 6174 6120 282d 2d70 726f  sp spdata (--pro
+0000c790: 7463 6f63 6461 7469 6e67 2920 282d 2d70  tcocdating) (--p
+0000c7a0: 6172 7469 7469 6f6e 2920 282d 2d61 616d  artition) (--aam
+0000c7b0: 6f64 656c 206c 6729 2028 2d64 7320 2762  odel lg) (-ds 'b
+0000c7c0: 7572 6e69 6e20 3d20 3230 3030 2729 2028  urnin = 2000') (
+0000c7d0: 2d64 7320 2773 616d 7066 7265 7120 3d20  -ds 'sampfreq = 
+0000c7e0: 3130 3030 2729 2028 2d64 7320 276e 7361  1000') (-ds 'nsa
+0000c7f0: 6d70 6c65 203d 2032 3030 3030 2729 2020  mple = 20000')  
+0000c800: 282d 6e20 3429 2028 2d2d 746f 5f73 746f  (-n 4) (--to_sto
+0000c810: 7029 2028 2d2d 6364 7329 2028 2d6f 2077  p) (--cds) (-o w
+0000c820: 6764 5f66 6f63 7573 2920 282d 7420 776f  gd_focus) (-t wo
+0000c830: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
+0000c840: 5468 6520 6162 736f 6c75 7465 2064 6174  The absolute dat
+0000c850: 696e 6720 6f66 2057 4744 7320 6973 2061  ing of WGDs is a
+0000c860: 2073 7065 6369 6669 6320 7069 7065 6c69   specific pipeli
+0000c870: 6e65 2069 6d70 6c65 6d65 6e74 6564 2069  ne implemented i
+0000c880: 6e20 6077 6764 2076 3260 2075 7369 6e67  n `wgd v2` using
+0000c890: 2074 6865 206d 6574 686f 6420 6f66 2070   the method of p
+0000c8a0: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
+0000c8b0: 6e67 2e20 5468 6520 6661 6d69 6c69 6573  ng. The families
+0000c8c0: 2075 7365 6420 696e 2074 6869 7320 7374   used in this st
+0000c8d0: 6570 2063 616e 2062 6520 7072 6f64 7563  ep can be produc
+0000c8e0: 6564 2066 726f 6d20 6077 6764 2064 6d64  ed from `wgd dmd
+0000c8f0: 6020 616e 6420 6077 6764 2070 6561 6b60  ` and `wgd peak`
+0000c900: 2e20 4e6f 7465 2074 6861 7420 6865 7265  . Note that here
+0000c910: 2077 6520 6f6e 6c79 2064 6973 6375 7373   we only discuss
+0000c920: 2068 6f77 2074 6f20 6461 7465 2057 4744   how to date WGD
+0000c930: 7320 7769 7468 2067 656e 6f6d 6520 6173  s with genome as
+0000c940: 7365 6d62 6c79 2c20 696e 7374 6561 6420  sembly, instead 
+0000c950: 6f66 2074 7261 6e73 6372 6970 746f 6d65  of transcriptome
+0000c960: 2061 7373 656d 626c 7920 2877 6869 6368   assembly (which
+0000c970: 2077 696c 6c20 6265 2064 6973 6375 7373   will be discuss
+0000c980: 6564 2069 6e20 6120 7365 7061 7261 7465  ed in a separate
+0000c990: 2073 6563 7469 6f6e 2068 6572 6575 6e64   section hereund
+0000c9a0: 6572 292e 2054 6865 2061 7373 756d 7074  er). The assumpt
+0000c9b0: 696f 6e20 7765 206d 6164 6520 6865 7265  ion we made here
+0000c9c0: 2069 7320 7468 6174 206e 6f74 2061 6c6c   is that not all
+0000c9d0: 2061 6e63 686f 7220 7061 6972 7320 2863   anchor pairs (c
+0000c9e0: 6f6c 6c69 6e65 6172 2064 7570 6c69 6361  ollinear duplica
+0000c9f0: 7465 7329 2061 7265 2073 7569 7461 626c  tes) are suitabl
+0000ca00: 6520 666f 7220 7068 796c 6f67 656e 6574  e for phylogenet
+0000ca10: 6963 2064 6174 696e 672c 2066 6f72 2069  ic dating, for i
+0000ca20: 6e73 7461 6e63 6520 7468 6f73 6520 6661  nstance those fa
+0000ca30: 7374 6c79 206f 7220 736c 6f77 6c79 2065  stly or slowly e
+0000ca40: 766f 6c76 696e 6720 6765 6e65 2064 7570  volving gene dup
+0000ca50: 6c69 6361 7465 732c 2062 6563 6175 7365  licates, because
+0000ca60: 2074 6865 7927 7265 2070 726f 6e65 2074   they're prone t
+0000ca70: 6f20 6769 7665 2062 6961 7365 6420 6461  o give biased da
+0000ca80: 7469 6e67 2065 7374 696d 6174 696f 6e2e  ting estimation.
+0000ca90: 2053 6f20 6173 2074 6f20 7265 7472 6965   So as to retrie
+0000caa0: 7665 2074 6865 2022 7265 6c69 6162 6c65  ve the "reliable
+0000cab0: 2220 616e 6368 6f72 2070 6169 7273 2c20  " anchor pairs, 
+0000cac0: 7765 2069 6d70 6c65 6d65 6e74 6564 2073  we implemented s
+0000cad0: 6f6d 6520 6d65 7468 6f64 7320 6f66 2069  ome methods of i
+0000cae0: 6465 6e74 6966 7969 6e67 2063 7265 6469  dentifying credi
+0000caf0: 6162 6c65 2061 6e63 686f 7220 7061 6972  able anchor pair
+0000cb00: 7320 6261 7365 6420 6f6e 2074 6865 6972  s based on their
+0000cb10: 204b 7320 7661 6c75 6573 2061 6e64 2f6f   Ks values and/o
+0000cb20: 7220 7265 7369 6469 6e67 2063 6f6c 6c69  r residing colli
+0000cb30: 6e65 6172 2073 6567 6d65 6e74 732e 2046  near segments. F
+0000cb40: 6f72 2061 2067 656e 6f6d 6520 7769 7468  or a genome with
+0000cb50: 2063 6c65 6172 2073 6967 6e61 6c73 206f   clear signals o
+0000cb60: 6620 7075 7461 7469 7665 2057 4744 732c  f putative WGDs,
+0000cb70: 2073 7563 6820 6173 2074 6865 202a 4171   such as the *Aq
+0000cb80: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
+0000cb90: 2a20 696e 2074 6865 2065 7861 6d70 6c65  * in the example
+0000cba0: 2c20 6120 6865 7572 6973 7469 6320 6d65  , a heuristic me
+0000cbb0: 7468 6f64 2074 6861 7420 6170 706c 6965  thod that applie
+0000cbc0: 7320 7468 6520 7072 696e 6369 706c 6520  s the principle 
+0000cbd0: 6f66 2068 6f77 205b 6b73 7261 7465 735d  of how [ksrates]
+0000cbe0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+0000cbf0: 636f 6d2f 5649 422d 5053 422f 6b73 7261  com/VIB-PSB/ksra
+0000cc00: 7465 7329 2066 696e 6420 7468 6520 696e  tes) find the in
+0000cc10: 6974 6961 6c20 7065 616b 7320 616e 6420  itial peaks and 
+0000cc20: 7468 6569 7220 7061 7261 6d65 7465 7273  their parameters
+0000cc30: 2077 6173 2069 6d70 6c65 6d65 6e74 6564   was implemented
+0000cc40: 2074 6f20 6669 6e64 2074 6865 2039 3525   to find the 95%
+0000cc50: 2063 6f6e 6669 6465 6e63 6520 6c65 7665   confidence leve
+0000cc60: 6c20 6f66 2074 6865 2061 7373 756d 6564  l of the assumed
+0000cc70: 206c 6f67 6e6f 726d 616c 2064 6973 7472   lognormal distr
+0000cc80: 6962 7574 696f 6e20 6f66 2074 6865 2061  ibution of the a
+0000cc90: 6e63 686f 7220 7061 6972 202a 4b2a 3c73  nchor pair *K*<s
+0000cca0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000ccb0: 6973 7472 6962 7574 696f 6e20 746f 2066  istribution to f
+0000ccc0: 696c 7465 7220 616e 6368 6f72 2070 6169  ilter anchor pai
+0000ccd0: 7273 2077 6974 6820 746f 6f20 6869 6768  rs with too high
+0000cce0: 206f 7220 6c6f 7720 2a4b 2a3c 7375 623e   or low *K*<sub>
+0000ccf0: 533c 2f73 7562 3e20 6167 6573 2c20 7468  S</sub> ages, th
+0000cd00: 6520 6578 616d 706c 6172 2063 6f6d 6d61  e examplar comma
+0000cd10: 6e64 206f 6620 7768 6963 6820 6973 2073  nd of which is s
+0000cd20: 686f 7765 6420 696e 2074 6865 2060 7767  howed in the `wg
+0000cd30: 6420 7065 616b 6020 7365 6374 696f 6e20  d peak` section 
+0000cd40: 6f66 205b 496c 6c75 7374 7261 7469 6f6e  of [Illustration
+0000cd50: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+0000cd60: 2e63 6f6d 2f68 6563 6865 2d70 7362 2f77  .com/heche-psb/w
+0000cd70: 6764 3f74 6162 3d72 6561 646d 652d 6f76  gd?tab=readme-ov
+0000cd80: 2d66 696c 6523 696c 6c75 7374 7261 7469  -file#illustrati
+0000cd90: 6f6e 292e 2049 6620 7468 6973 2068 6575  on). If this heu
+0000cda0: 7269 7374 6963 206d 6574 686f 6420 6661  ristic method fa
+0000cdb0: 696c 6564 2074 6f20 6769 7665 2072 6561  iled to give rea
+0000cdc0: 736f 6e61 626c 6520 7265 7375 6c74 732c  sonable results,
+0000cdd0: 2077 6869 6368 2069 7320 7573 7561 6c6c   which is usuall
+0000cde0: 7920 6475 6520 746f 2074 6865 2065 6666  y due to the eff
+0000cdf0: 6563 7420 6f66 206d 756c 7469 706c 6520  ect of multiple 
+0000ce00: 6164 6a61 6365 6e74 2057 4744 7320 7468  adjacent WGDs th
+0000ce10: 6174 2062 6c75 7273 2074 6865 2070 6561  at blurs the pea
+0000ce20: 6b2d 6669 6e64 696e 6720 7072 6f63 6573  k-finding proces
+0000ce30: 732c 2075 7365 7273 2063 616e 2074 7572  s, users can tur
+0000ce40: 6e20 746f 2074 6865 2063 6f6c 6c69 6e65  n to the colline
+0000ce50: 6172 2073 6567 6d65 6e74 732d 6775 6964  ar segments-guid
+0000ce60: 6564 2061 6e63 686f 7220 7061 6972 2063  ed anchor pair c
+0000ce70: 6c75 7374 6572 696e 6720 696d 706c 656d  lustering implem
+0000ce80: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
+0000ce90: 602c 2069 6e20 7768 6963 6820 6120 636f  `, in which a co
+0000cea0: 6c6c 696e 6561 7220 7365 676d 656e 742d  llinear segment-
+0000ceb0: 7769 7365 2047 4d4d 2063 6c75 7374 6572  wise GMM cluster
+0000cec0: 696e 6720 7769 6c6c 2062 6520 6669 7273  ing will be firs
+0000ced0: 7420 636f 6e64 7563 7465 6420 6261 7365  t conducted base
+0000cee0: 6420 6f6e 2074 6865 2022 736f 2d63 616c  d on the "so-cal
+0000cef0: 6c65 6422 2073 6567 6d65 6e74 202a 4b2a  led" segment *K*
+0000cf00: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000cf10: 2072 6570 7265 7365 6e74 6564 2062 7920   represented by 
+0000cf20: 7468 6520 6d65 6469 616e 202a 4b2a 3c73  the median *K*<s
+0000cf30: 7562 3e53 3c2f 7375 623e 2061 6765 206f  ub>S</sub> age o
+0000cf40: 6620 616c 6c20 7468 6520 7265 7369 6469  f all the residi
+0000cf50: 6e67 2073 796e 7465 6c6f 6773 2028 6e6f  ng syntelogs (no
+0000cf60: 7465 2074 6861 7420 7468 6520 6765 6e65  te that the gene
+0000cf70: 2064 7570 6c69 6361 7465 2070 6169 7273   duplicate pairs
+0000cf80: 2061 646f 7074 6564 2069 6e20 7468 6973   adopted in this
+0000cf90: 2073 7465 7020 6973 2066 726f 6d20 7468   step is from th
+0000cfa0: 6520 6669 6c65 2060 6d75 6c74 6970 6c69  e file `multipli
+0000cfb0: 636f 6e5f 7061 6972 732e 7478 7460 2077  con_pairs.txt` w
+0000cfc0: 6869 6368 2063 6f6e 7461 696e 7320 7468  hich contains th
+0000cfd0: 6520 6675 6c6c 2073 6574 206f 6620 7379  e full set of sy
+0000cfe0: 6e74 656c 6f67 7329 2c20 696e 7374 6561  ntelogs), instea
+0000cff0: 6420 6f66 2074 6865 2073 6d61 6c6c 6572  d of the smaller
+0000d000: 2067 656e 6520 7365 7420 6f66 2061 6e63   gene set of anc
+0000d010: 686f 7220 7061 6972 732e 2054 6865 2064  hor pairs. The d
+0000d020: 6973 7469 6e63 7469 6f6e 2062 6574 7765  istinction betwe
+0000d030: 656e 2061 6e63 686f 7220 7061 6972 7320  en anchor pairs 
+0000d040: 616e 6420 7379 6e74 656c 6f67 7320 6973  and syntelogs is
+0000d050: 2074 6861 7420 7468 6520 6c61 7474 6572   that the latter
+0000d060: 2072 6566 6572 7320 746f 206d 756c 7469   refers to multi
+0000d070: 706c 6520 7365 7473 206f 6620 6765 6e65  ple sets of gene
+0000d080: 7320 6465 7269 7665 6420 6672 6f6d 2074  s derived from t
+0000d090: 6865 2073 616d 6520 616e 6365 7374 7261  he same ancestra
+0000d0a0: 6c20 6765 6e6f 6d69 6320 7265 6769 6f6e  l genomic region
+0000d0b0: 2077 6869 6c65 2074 6865 2066 6f72 6d65   while the forme
+0000d0c0: 7220 696d 706c 6965 7320 7468 6520 6c61  r implies the la
+0000d0d0: 7474 6572 2062 7574 2072 6571 7569 7265  tter but require
+0000d0e0: 7320 6164 6469 7469 6f6e 616c 6c79 2074  s additionally t
+0000d0f0: 6865 2063 6f6e 7365 7276 6564 2067 656e  he conserved gen
+0000d100: 6520 6f72 6465 722c 2062 6f74 6820 6f66  e order, both of
+0000d110: 2077 6869 6368 2061 7265 2c20 7769 7468   which are, with
+0000d120: 696e 2061 2067 656e 6f6d 652c 2061 7373  in a genome, ass
+0000d130: 756d 6564 2074 6f20 6265 206f 7269 6769  umed to be origi
+0000d140: 6e61 7465 6420 6672 6f6d 2074 6865 2064  nated from the d
+0000d150: 7570 6c69 6361 7469 6f6e 206f 6620 6120  uplication of a 
+0000d160: 636f 6d6d 6f6e 2061 6e63 6573 7472 616c  common ancestral
+0000d170: 2067 656e 6f6d 6963 2072 6567 696f 6e20   genomic region 
+0000d180: 616e 6420 6173 2073 7563 6820 6465 656d  and as such deem
+0000d190: 6564 2065 7669 6465 6e63 6520 666f 7220  ed evidence for 
+0000d1a0: 5747 442e 2054 6865 6e20 7468 6520 7379  WGD. Then the sy
+0000d1b0: 6e74 656c 6f67 7320 7769 6c6c 2062 6520  ntelogs will be 
+0000d1c0: 6d61 7070 6564 2062 6163 6b20 6163 636f  mapped back acco
+0000d1d0: 7264 696e 6720 746f 2074 6865 2063 6c75  rding to the clu
+0000d1e0: 7374 6572 696e 6720 7265 7375 6c74 7320  stering results 
+0000d1f0: 6f66 2074 6865 6972 2061 6666 696c 6961  of their affilia
+0000d200: 7465 6420 7365 676d 656e 7473 2e20 5369  ted segments. Si
+0000d210: 6e63 6520 7468 6520 4761 7573 7369 616e  nce the Gaussian
+0000d220: 2073 6861 7065 206f 6620 7468 6520 7365   shape of the se
+0000d230: 676d 656e 7420 636c 7573 7465 7220 646f  gment cluster do
+0000d240: 6573 6e27 7420 6e65 6365 7373 6172 696c  esn't necessaril
+0000d250: 7920 696d 6974 6174 6520 7468 6520 7368  y imitate the sh
+0000d260: 6170 6520 6f66 2074 6865 2072 6573 6964  ape of the resid
+0000d270: 696e 6720 7379 6e74 656c 6f67 732c 2073  ing syntelogs, s
+0000d280: 6f20 6173 2074 6f20 7265 7472 6965 7665  o as to retrieve
+0000d290: 2074 6865 2022 7265 6c69 6162 6c65 2220   the "reliable" 
+0000d2a0: 6765 6e65 2073 6574 7320 666f 7220 7068  gene sets for ph
+0000d2b0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+0000d2c0: 672c 2077 6520 6164 706f 7465 6420 7468  g, we adpoted th
+0000d2d0: 6520 2839 3525 2920 6869 6768 6573 7420  e (95%) highest 
+0000d2e0: 6465 6e73 6974 7920 7265 6769 6f6e 2028  density region (
+0000d2f0: 4844 5229 206f 6620 7468 6520 7379 6e74  HDR) of the synt
+0000d300: 656c 6f67 202a 4b2a 3c73 7562 3e53 3c2f  elog *K*<sub>S</
+0000d310: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000d320: 7574 696f 6e20 666f 7220 7468 6520 7068  ution for the ph
+0000d330: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+0000d340: 672c 2074 6865 2063 616c 6375 6c61 7469  g, the calculati
+0000d350: 6f6e 206f 6620 7468 6520 2839 3525 2920  on of the (95%) 
+0000d360: 4844 5220 696e 2074 6865 2066 756e 6374  HDR in the funct
+0000d370: 696f 6e20 5b63 616c 6375 6c61 7465 4850  ion [calculateHP
+0000d380: 445d 2868 7474 7073 3a2f 2f67 6974 6875  D](https://githu
+0000d390: 622e 636f 6d2f 6865 6368 652d 7073 622f  b.com/heche-psb/
+0000d3a0: 7767 642f 626c 6f62 2f70 6879 6c6f 6461  wgd/blob/phyloda
+0000d3b0: 7469 6e67 2f77 6764 2f70 6561 6b2e 7079  ting/wgd/peak.py
+0000d3c0: 234c 3137 3537 2920 7365 656b 7320 7468  #L1757) seeks th
+0000d3d0: 6520 7368 6f72 7465 7374 202a 4b2a 3c73  e shortest *K*<s
+0000d3e0: 7562 3e53 3c2f 7375 623e 2072 616e 6765  ub>S</sub> range
+0000d3f0: 2028 612c 6229 2077 6869 6368 2073 6174   (a,b) which sat
+0000d400: 6973 6669 6573 2074 6865 2072 6571 7569  isfies the requi
+0000d410: 7265 6d65 6e74 206f 6620 7370 616e 6e69  rement of spanni
+0000d420: 6e67 206d 6f72 6520 7468 616e 2028 3935  ng more than (95
+0000d430: 2529 206f 6620 616c 6c20 7468 6520 2a4b  %) of all the *K
+0000d440: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+0000d450: 6c75 6573 2e20 4f6e 2074 6865 2070 7265  lues. On the pre
+0000d460: 6d69 7365 206f 6620 6964 656e 7469 6669  mise of identifi
+0000d470: 6564 2061 6e63 686f 7220 7061 6972 7320  ed anchor pairs 
+0000d480: 286e 6f74 6520 7468 6174 2074 6865 2073  (note that the s
+0000d490: 796e 7465 6c6f 6773 2061 7265 2061 6c73  yntelogs are als
+0000d4a0: 6f20 7265 6665 7272 6564 2061 7320 616e  o referred as an
+0000d4b0: 6368 6f72 2070 6169 7273 2068 6572 6561  chor pairs herea
+0000d4c0: 6674 6572 292c 2077 6520 696d 706c 656d  fter), we implem
+0000d4d0: 656e 7465 6420 696e 2074 6865 2070 726f  ented in the pro
+0000d4e0: 6772 616d 2060 7767 6420 646d 6460 2074  gram `wgd dmd` t
+0000d4f0: 6865 2073 6f2d 6361 6c6c 6564 2061 6e63  he so-called anc
+0000d500: 686f 722d 6177 6172 6520 6c6f 6361 6c20  hor-aware local 
+0000d510: 4d52 4248 7320 6f72 206f 7274 686f 6772  MRBHs or orthogr
+0000d520: 6f75 7073 2c20 696e 2077 6869 6368 2074  oups, in which t
+0000d530: 6865 206f 7269 6769 6e61 6c20 6c6f 6361  he original loca
+0000d540: 6c20 4d52 4248 7320 6172 6520 6675 7274  l MRBHs are furt
+0000d550: 6865 7220 6d65 7267 6564 2077 6974 6820  her merged with 
+0000d560: 616e 6368 6f72 2070 6169 7273 2073 7563  anchor pairs suc
+0000d570: 6820 7468 6174 2065 6163 6820 6f72 7468  h that each orth
+0000d580: 6f67 726f 7570 2063 6f6e 7461 696e 7320  ogroup contains 
+0000d590: 7468 6520 616e 6368 6f72 2070 6169 7220  the anchor pair 
+0000d5a0: 616e 6420 7468 6520 6f72 7468 6f6c 6f67  and the ortholog
+0000d5b0: 7565 732e 2057 6974 6820 7468 6973 206f  ues. With this o
+0000d5c0: 7274 686f 6772 6f75 702c 2075 7365 7273  rthogroup, users
+0000d5d0: 2074 6865 6e20 6e65 6564 206f 6e65 2073   then need one s
+0000d5e0: 7461 7274 696e 6720 7472 6565 2066 696c  tarting tree fil
+0000d5f0: 6520 2861 7320 7368 6f77 6e20 696e 2074  e (as shown in t
+0000d600: 6865 205b 496c 6c75 7374 7261 7469 6f6e  he [Illustration
+0000d610: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+0000d620: 2e63 6f6d 2f68 6563 6865 2d70 7362 2f77  .com/heche-psb/w
+0000d630: 6764 2f74 7265 652f 7068 796c 6f64 6174  gd/tree/phylodat
+0000d640: 696e 673f 7461 623d 7265 6164 6d65 2d6f  ing?tab=readme-o
+0000d650: 762d 6669 6c65 2369 6c6c 7573 7472 6174  v-file#illustrat
+0000d660: 696f 6e29 2070 6172 7429 2069 6e64 6963  ion) part) indic
+0000d670: 6174 696e 6720 7468 6520 7472 6565 2074  ating the tree t
+0000d680: 6f70 6f6c 6f67 7920 616e 6420 666f 7373  opology and foss
+0000d690: 696c 2063 616c 6962 7261 7469 6f6e 2069  il calibration i
+0000d6a0: 6e66 6f72 6d61 7469 6f6e 2066 6f72 2074  nformation for t
+0000d6b0: 6865 2066 696e 616c 2070 6879 6c6f 6765  he final phyloge
+0000d6c0: 6e65 7469 6320 6461 7469 6e67 2075 7369  netic dating usi
+0000d6d0: 6e67 2074 6865 2070 726f 6772 616d 2060  ng the program `
+0000d6e0: 7767 6420 666f 6375 7360 2e20 5573 6572  wgd focus`. User
+0000d6f0: 7320 6361 6e20 7365 7420 7468 6520 666c  s can set the fl
+0000d700: 6167 206f 7074 696f 6e20 602d 2d70 726f  ag option `--pro
+0000d710: 7463 6f63 6461 7469 6e67 6020 746f 206f  tcocdating` to o
+0000d720: 6e6c 7920 636f 6e64 7563 7420 7468 6520  nly conduct the 
+0000d730: 6461 7469 6e67 206f 6620 7468 6520 636f  dating of the co
+0000d740: 6e63 6174 656e 6174 696f 6e20 7072 6f74  ncatenation prot
+0000d750: 6569 6e20 4d53 4120 6f72 2074 6865 2066  ein MSA or the f
+0000d760: 6c61 6720 6f70 7469 6f6e 2060 2d2d 7072  lag option `--pr
+0000d770: 6f74 6461 7469 6e67 6020 746f 206f 6e6c  otdating` to onl
+0000d780: 7920 636f 6e64 7563 7420 7468 6520 6461  y conduct the da
+0000d790: 7469 6e67 206f 6620 7468 6520 7072 6f74  ting of the prot
+0000d7a0: 6569 6e20 4d53 4173 2c20 6e6f 7465 6420  ein MSAs, noted 
+0000d7b0: 7468 6174 2074 6865 7365 2074 776f 206f  that these two o
+0000d7c0: 7074 696f 6e73 206f 6e6c 7920 776f 726b  ptions only work
+0000d7d0: 2066 6f72 2074 6865 2060 6d63 6d63 7472   for the `mcmctr
+0000d7e0: 6565 6020 6f70 7469 6f6e 2073 6f20 6661  ee` option so fa
+0000d7f0: 722e 2054 6865 2066 6c61 6720 6f70 7469  r. The flag opti
+0000d800: 6f6e 2060 2d2d 7061 7274 6974 696f 6e60  on `--partition`
+0000d810: 2063 616e 2062 6520 7365 7420 746f 2070   can be set to p
+0000d820: 6572 666f 726d 2060 6d63 6d63 7472 6565  erform `mcmctree
+0000d830: 6020 616e 616c 7973 6973 2075 7369 6e67  ` analysis using
+0000d840: 2074 6865 2070 6172 7469 7469 6f6e 6564   the partitioned
+0000d850: 2064 6174 6120 2869 2e65 2e2c 2031 7374   data (i.e., 1st
+0000d860: 2c20 326e 6420 616e 6420 3372 6420 706f  , 2nd and 3rd po
+0000d870: 7369 7469 6f6e 206f 6620 636f 646f 6e29  sition of codon)
+0000d880: 2069 6e73 7465 6164 206f 6620 7573 696e   instead of usin
+0000d890: 6720 7468 6520 636f 646f 6e20 6173 2061  g the codon as a
+0000d8a0: 2077 686f 6c65 2e20 5468 6520 6f70 7469   whole. The opti
+0000d8b0: 6f6e 2060 2d2d 6161 6d6f 6465 6c60 2063  on `--aamodel` c
+0000d8c0: 616e 2062 6520 7365 7420 746f 2064 6574  an be set to det
+0000d8d0: 6572 6d69 6e65 2074 6865 2061 6d69 6e6f  ermine the amino
+0000d8e0: 2061 6369 6420 6d6f 6465 6c20 6170 706c   acid model appl
+0000d8f0: 6965 6420 696e 2060 6d63 6d63 7472 6565  ied in `mcmctree
+0000d900: 6020 616e 616c 7973 6973 2e20 5468 6520  ` analysis. The 
+0000d910: 6f70 7469 6f6e 2060 2d64 7360 2063 616e  option `-ds` can
+0000d920: 2062 6520 7573 6564 2074 6f20 7365 7420   be used to set 
+0000d930: 7061 7261 6d65 7465 7273 2066 6f72 2074  parameters for t
+0000d940: 6865 206d 6f6c 6563 756c 6172 2064 6174  he molecular dat
+0000d950: 696e 6720 7072 6f67 7261 6d2e 2054 6865  ing program. The
+0000d960: 2070 6172 616c 6c65 6c20 7468 7265 6164   parallel thread
+0000d970: 7320 6865 7265 2061 7265 2074 6f20 7061  s here are to pa
+0000d980: 7261 6c6c 656c 697a 6520 7468 6520 616e  rallelize the an
+0000d990: 616c 7973 6973 2066 6f72 2065 6163 6820  alysis for each 
+0000d9a0: 6765 6e65 2066 616d 696c 7920 616e 6420  gene family and 
+0000d9b0: 7468 7573 2073 7567 6765 7374 6564 2074  thus suggested t
+0000d9c0: 6f20 6265 2073 6574 2061 7320 6d75 6368  o be set as much
+0000d9d0: 2061 7320 7468 6520 6e75 6d62 6572 206f   as the number o
+0000d9e0: 6620 6765 6e65 2066 616d 696c 6965 732e  f gene families.
+0000d9f0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+0000da00: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+0000da10: 2062 6520 7769 7468 2074 6865 2063 6f6d   be with the com
+0000da20: 6d61 6e64 2062 656c 6f77 2e2a 2a0a 0a60  mand below.**..`
+0000da30: 6060 0a77 6764 2066 6f63 7573 2066 616d  ``.wgd focus fam
+0000da40: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000da50: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000da60: 6365 3320 2d64 206d 636d 6374 7265 6520  ce3 -d mcmctree 
+0000da70: 2d73 7020 7370 6461 7461 0a77 6764 2066  -sp spdata.wgd f
+0000da80: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
+0000da90: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000daa0: 3220 7365 7175 656e 6365 3320 2d64 2072  2 sequence3 -d r
+0000dab0: 3873 202d 7370 2073 7064 6174 6120 2d2d  8s -sp spdata --
+0000dac0: 6e73 6974 6573 206e 7369 7465 696e 666f  nsites nsiteinfo
+0000dad0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+0000dae0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000daf0: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000db00: 3320 2d64 2062 6561 7374 202d 2d66 6f73  3 -d beast --fos
+0000db10: 7369 6c20 666f 7373 696c 696e 666f 202d  sil fossilinfo -
+0000db20: 2d72 6f6f 7468 6569 6768 7420 726f 6f74  -rootheight root
+0000db30: 6865 6967 6874 696e 666f 202d 2d63 6861  heightinfo --cha
+0000db40: 696e 7365 7420 6368 6169 6e73 6574 696e  inset chainsetin
+0000db50: 666f 202d 2d62 6561 7374 6c67 6a61 7220  fo --beastlgjar 
+0000db60: 2450 4154 480a 6060 600a 0a23 2323 2077  $PATH.```..### w
+0000db70: 6764 206b 7364 0a0a 2a2a 5468 6520 636f  gd ksd..**The co
+0000db80: 6e73 7472 7563 7469 6f6e 206f 6620 7768  nstruction of wh
+0000db90: 6f6c 6520 7061 7261 6e6f 6d65 202a 4b2a  ole paranome *K*
+0000dba0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000dbb0: 2064 6973 7472 6962 7574 696f 6e2a 2a0a   distribution**.
+0000dbc0: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
+0000dbd0: 6c69 6573 2073 6571 7565 6e63 6520 282d  lies sequence (-
+0000dbe0: 6f20 7767 645f 6b73 6420 2d74 2077 6764  o wgd_ksd -t wgd
+0000dbf0: 5f6b 7364 5f74 6d70 202d 2d6e 7468 7265  _ksd_tmp --nthre
+0000dc00: 6164 7320 3420 2d2d 746f 5f73 746f 7020  ads 4 --to_stop 
+0000dc10: 2d2d 6364 7320 2d2d 7061 6972 7769 7365  --cds --pairwise
+0000dc20: 202d 2d73 7472 6970 5f67 6170 7320 2d2d   --strip_gaps --
+0000dc30: 616c 6967 6e65 7220 6d61 6666 7420 2d2d  aligner mafft --
+0000dc40: 7472 6565 5f6d 6574 686f 6420 6661 7374  tree_method fast
+0000dc50: 7472 6565 202d 2d6e 6f64 655f 6176 6572  tree --node_aver
+0000dc60: 6167 6529 0a60 6060 0a0a 5468 6520 7072  age).```..The pr
+0000dc70: 6f67 7261 6d20 6077 6764 206b 7364 602c  ogram `wgd ksd`,
+0000dc80: 2061 7320 696d 7069 6564 2062 7920 6974   as impied by it
+0000dc90: 7320 6e61 6d65 2c20 6973 2066 6f72 2074  s name, is for t
+0000dca0: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+0000dcb0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
+0000dcc0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000dcd0: 696f 6e2e 2045 7863 6570 7420 666f 7220  ion. Except for 
+0000dce0: 7468 6520 6166 6f72 656d 656e 7469 6f6e  the aforemention
+0000dcf0: 6564 2070 6172 616d 6574 6572 7320 7375  ed parameters su
+0000dd00: 6368 2061 7320 602d 2d74 6f5f 7374 6f70  ch as `--to_stop
+0000dd10: 6020 616e 6420 602d 2d63 6473 602c 2074  ` and `--cds`, t
+0000dd20: 6865 7265 2061 7265 2073 6f6d 6520 696d  here are some im
+0000dd30: 706f 7274 616e 7420 7061 7261 6d65 7465  portant paramete
+0000dd40: 7273 2074 6861 7420 6861 7665 2063 7275  rs that have cru
+0000dd50: 6369 616c 2069 6d70 6163 7420 6f6e 2074  cial impact on t
+0000dd60: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000dd70: 623e 2065 7374 696d 6174 696f 6e2e 2054  b> estimation. T
+0000dd80: 6865 206f 7074 696f 6e20 602d 2d70 6169  he option `--pai
+0000dd90: 7277 6973 6560 2069 7320 6120 7665 7279  rwise` is a very
+0000dda0: 2069 6d70 6f72 7461 6e74 2070 6172 616d   important param
+0000ddb0: 6574 6572 2066 6f72 2074 6865 202a 4b2a  eter for the *K*
+0000ddc0: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+0000ddd0: 696d 6174 696f 6e2c 2077 6974 6820 7768  imation, with wh
+0000dde0: 6963 6820 7468 6520 6043 4f44 454d 4c60  ich the `CODEML`
+0000ddf0: 2077 696c 6c20 6361 6c63 756c 6174 6520   will calculate 
+0000de00: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+0000de10: 7562 3e20 666f 7220 6561 6368 2067 656e  ub> for each gen
+0000de20: 6520 7061 6972 2073 6570 6172 6174 656c  e pair separatel
+0000de30: 7920 6261 7365 6420 6f6e 2074 6865 2061  y based on the a
+0000de40: 6c69 676e 6d65 6e74 206f 6620 6f6e 6c79  lignment of only
+0000de50: 2074 6865 7365 2074 776f 2067 656e 6573   these two genes
+0000de60: 2069 6e73 7465 6164 206f 6620 7468 6520   instead of the 
+0000de70: 7768 6f6c 6520 616c 6967 6e6d 656e 7420  whole alignment 
+0000de80: 6f66 2074 6865 2066 616d 696c 792c 2073  of the family, s
+0000de90: 7563 6820 7468 6174 206c 6573 7320 6761  uch that less ga
+0000dea0: 7073 2061 7265 2065 7870 6563 7465 6420  ps are expected 
+0000deb0: 616e 6420 7468 7573 2074 6865 2061 6c69  and thus the ali
+0000dec0: 676e 6d65 6e74 2069 6e20 7468 6520 636f  gnment in the co
+0000ded0: 6e73 6964 6572 6174 696f 6e20 6f66 2060  nsideration of `
+0000dee0: 434f 4445 4d4c 6020 7769 6c6c 2062 6520  CODEML` will be 
+0000def0: 6c6f 6e67 6572 2028 6265 6361 7573 6520  longer (because 
+0000df00: 6043 4f44 454d 4c60 2077 696c 6c20 6175  `CODEML` will au
+0000df10: 746f 6d61 7465 6c79 2073 6b69 7020 6576  tomately skip ev
+0000df20: 6572 7920 636f 6c75 6d6e 2077 6974 6820  ery column with 
+0000df30: 6761 702c 2072 6567 6172 646c 6573 7320  gap, regardless 
+0000df40: 6f66 2077 6865 7468 6572 2069 7427 7320  of whether it's 
+0000df50: 616e 206f 7665 7261 6c6c 206f 7220 7061  an overall or pa
+0000df60: 7274 6961 6c20 6761 7029 2c20 7769 7468  rtial gap), with
+0000df70: 6f75 7420 7768 6963 6820 7468 6520 6043  out which the `C
+0000df80: 4f44 454d 4c60 2077 696c 6c20 6361 6c63  ODEML` will calc
+0000df90: 756c 6174 6520 7468 6520 2a4b 2a3c 7375  ulate the *K*<su
+0000dfa0: 623e 533c 2f73 7562 3e20 6261 7365 6420  b>S</sub> based 
+0000dfb0: 6f6e 2074 6865 2077 686f 6c65 2061 6c69  on the whole ali
+0000dfc0: 676e 6d65 6e74 206f 6620 7468 6520 6661  gnment of the fa
+0000dfd0: 6d69 6c79 2c20 7768 6963 6820 6d69 6768  mily, which migh
+0000dfe0: 7420 6861 7665 206e 6f20 2a4b 2a3c 7375  t have no *K*<su
+0000dff0: 623e 533c 2f73 7562 3e20 7265 7375 6c74  b>S</sub> result
+0000e000: 2061 7420 616c 6c20 6966 2074 6865 2073   at all if the s
+0000e010: 7472 6970 7065 6420 616c 6967 6e6d 656e  tripped alignmen
+0000e020: 7420 6c65 6e67 7468 2028 7265 6d6f 7669  t length (removi
+0000e030: 6e67 2061 6c6c 2067 6170 2d63 6f6e 7461  ng all gap-conta
+0000e040: 696e 696e 6720 636f 6c75 6d6e 7329 2077  ining columns) w
+0000e050: 6173 207a 6572 6f2c 2061 2063 6175 7365  as zero, a cause
+0000e060: 206f 6620 6469 6666 6572 656e 7420 6e75   of different nu
+0000e070: 6d62 6572 206f 6620 2a4b 2a3c 7375 623e  mber of *K*<sub>
+0000e080: 533c 2f73 7562 3e20 6573 7469 6d61 7465  S</sub> estimate
+0000e090: 7320 6265 7477 6565 6e20 2270 6169 7277  s between "pairw
+0000e0a0: 6973 6520 6d6f 6465 2220 616e 6420 226e  ise mode" and "n
+0000e0b0: 6f6e 2d70 6169 7277 6973 6520 6d6f 6465  on-pairwise mode
+0000e0c0: 222e 2049 7427 7320 6469 6666 6963 756c  ". It's difficul
+0000e0d0: 7420 746f 2073 6179 2077 6869 6368 206d  t to say which m
+0000e0e0: 6f64 6520 6973 206d 6f72 6520 6964 6561  ode is more idea
+0000e0f0: 6c2c 2061 6c74 686f 7567 6820 7468 6520  l, although the 
+0000e100: 226e 6f6e 2d70 6169 7277 6973 6520 6d6f  "non-pairwise mo
+0000e110: 6465 2220 2864 6566 6175 6c74 2073 6574  de" (default set
+0000e120: 7469 6e67 2920 7768 6963 6820 7275 6e73  ting) which runs
+0000e130: 206f 6e20 7468 6520 7768 6f6c 6520 616c   on the whole al
+0000e140: 6967 6e6d 656e 7420 696e 7374 6561 6420  ignment instead 
+0000e150: 6f66 2061 206c 6f63 616c 2061 6c69 676e  of a local align
+0000e160: 6d65 6e74 2c20 6d69 6768 7420 6265 206d  ment, might be m
+0000e170: 6f72 6520 6269 6f6c 6f67 6963 616c 6c79  ore biologically
+0000e180: 2063 6f6e 7365 7276 6564 2069 6e20 7468   conserved in th
+0000e190: 6174 2069 7420 6173 7375 7265 7320 7468  at it assures th
+0000e1a0: 6520 6576 6f6c 7574 696f 6e20 6f66 2065  e evolution of e
+0000e1b0: 6163 6820 636f 6c75 6d6e 2074 6f20 6265  ach column to be
+0000e1c0: 2073 7461 7274 6564 2066 726f 6d20 7468   started from th
+0000e1d0: 6520 726f 6f74 206f 6620 7468 6520 6661  e root of the fa
+0000e1e0: 6d69 6c79 2061 6e64 2061 6c6c 2074 6865  mily and all the
+0000e1f0: 2067 656e 6520 6475 706c 6963 6174 6573   gene duplicates
+0000e200: 2061 7265 2074 616b 656e 2069 6e74 6f20   are taken into 
+0000e210: 6163 636f 756e 7420 696e 2074 6865 202a  account in the *
+0000e220: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+0000e230: 7374 696d 6174 696f 6e20 7072 6f63 6573  stimation proces
+0000e240: 732e 2054 6865 206f 7074 696f 6e20 602d  s. The option `-
+0000e250: 2d73 7472 6970 5f67 6170 7360 2063 616e  -strip_gaps` can
+0000e260: 2072 656d 6f76 6520 616c 6c20 7468 6520   remove all the 
+0000e270: 6761 702d 636f 6e74 6169 6e69 6e67 2063  gap-containing c
+0000e280: 6f6c 756d 6e73 2c20 7769 7468 206f 7220  olumns, with or 
+0000e290: 7769 7468 6f75 7420 7768 6963 6820 7468  without which th
+0000e2a0: 6520 7265 7375 6c74 206f 6620 226e 6f6e  e result of "non
+0000e2b0: 2d70 6169 7277 6973 6520 6d6f 6465 2220  -pairwise mode" 
+0000e2c0: 776f 6e27 7420 6265 2061 6666 6563 7465  won't be affecte
+0000e2d0: 642c 2077 6869 6c65 2077 6974 6820 7768  d, while with wh
+0000e2e0: 6963 6820 7468 6520 7265 7375 6c74 206f  ich the result o
+0000e2f0: 6620 2270 6169 7277 6973 6520 6d6f 6465  f "pairwise mode
+0000e300: 2220 7769 6c6c 2062 6520 616c 7465 7265  " will be altere
+0000e310: 642e 2054 6865 206f 7074 696f 6e20 602d  d. The option `-
+0000e320: 2d61 6c69 676e 6572 6020 616e 6420 602d  -aligner` and `-
+0000e330: 2d61 6c6e 5f6f 7074 696f 6e73 6020 7768  -aln_options` wh
+0000e340: 6963 6820 6465 6369 6465 2077 6869 6368  ich decide which
+0000e350: 2061 6c69 676e 6d65 6e74 2070 726f 6772   alignment progr
+0000e360: 616d 2074 6f20 6265 2075 7365 6420 616e  am to be used an
+0000e370: 6420 7768 6963 6820 7061 7261 6d65 7465  d which paramete
+0000e380: 7220 746f 2062 6520 7365 742c 2077 696c  r to be set, wil
+0000e390: 6c20 6861 7665 2069 6d70 6163 7420 6f6e  l have impact on
+0000e3a0: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0000e3b0: 7375 623e 2072 6573 756c 7473 2c20 6e6f  sub> results, no
+0000e3c0: 7465 6420 7468 6174 2074 6865 2064 6566  ted that the def
+0000e3d0: 6175 6c74 2070 726f 6772 616d 2069 7320  ault program is 
+0000e3e0: 606d 6166 6674 6020 616e 6420 7468 6520  `mafft` and the 
+0000e3f0: 7061 7261 6d65 7465 7220 6973 2060 2d2d  parameter is `--
+0000e400: 6175 746f 602e 2054 6865 206f 7074 696f  auto`. The optio
+0000e410: 6e20 602d 2d74 7265 655f 6d65 7468 6f64  n `--tree_method
+0000e420: 6020 616e 6420 602d 2d74 7265 655f 6f70  ` and `--tree_op
+0000e430: 7469 6f6e 7360 2064 6563 6964 6520 7768  tions` decide wh
+0000e440: 6963 6820 6765 6e65 2074 7265 6520 696e  ich gene tree in
+0000e450: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+0000e460: 746f 2062 6520 7573 6564 2061 6e64 2077  to be used and w
+0000e470: 6869 6368 2070 6172 616d 6574 6572 2074  hich parameter t
+0000e480: 6f20 6265 2073 6574 2c20 776f 6e27 7420  o be set, won't 
+0000e490: 6166 6665 6374 2074 6865 202a 4b2a 3c73  affect the *K*<s
+0000e4a0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
+0000e4b0: 6174 696f 6e20 6974 7365 6c66 2062 7574  ation itself but
+0000e4c0: 2074 6865 2072 6573 756c 7420 6f66 2064   the result of d
+0000e4d0: 652d 7265 6475 6e64 616e 6379 2c20 6e6f  e-redundancy, no
+0000e4e0: 7465 6420 7468 6174 2077 6520 696d 706c  ted that we impl
+0000e4f0: 656d 656e 7465 6420 6120 6275 696c 742d  emented a built-
+0000e500: 696e 2067 656e 6520 7472 6565 2069 6e66  in gene tree inf
+0000e510: 6572 656e 6365 206d 6574 686f 6420 6261  erence method ba
+0000e520: 7365 6420 6f6e 2074 6865 2041 7665 7261  sed on the Avera
+0000e530: 6765 204c 696e 6b61 6765 2043 6c75 7374  ge Linkage Clust
+0000e540: 6572 696e 6720 2841 4c43 2920 2874 6875  ering (ALC) (thu
+0000e550: 7320 6120 6469 7374 616e 6365 2d62 6173  s a distance-bas
+0000e560: 6564 2074 7265 6529 2077 6974 6820 7468  ed tree) with th
+0000e570: 6520 602d 2d74 7265 655f 6d65 7468 6f64  e `--tree_method
+0000e580: 6020 7365 7420 6173 2022 636c 7573 7465  ` set as "cluste
+0000e590: 7222 2e20 5477 6f20 6d65 7468 6f64 7320  r". Two methods 
+0000e5a0: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
+0000e5b0: 2061 7265 2069 6d70 6c65 6d65 6e74 6564   are implemented
+0000e5c0: 2069 6e20 6077 6764 2076 3260 2c20 6e61   in `wgd v2`, na
+0000e5d0: 6d65 6c79 206e 6f64 652d 7765 6967 6874  mely node-weight
+0000e5e0: 6564 2061 6e64 206e 6f64 652d 6176 6572  ed and node-aver
+0000e5f0: 6167 6564 206d 6574 686f 6473 2e20 5468  aged methods. Th
+0000e600: 6520 6e6f 6465 2d77 6569 6768 7465 6420  e node-weighted 
+0000e610: 6d65 7468 6f64 2061 6368 6965 7665 7320  method achieves 
+0000e620: 7468 6520 6465 2d72 6564 756e 6461 6e63  the de-redundanc
+0000e630: 7920 7669 6120 7765 6967 6869 6e67 2074  y via weighing t
+0000e640: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000e650: 623e 2076 616c 7565 2061 7373 6f63 6961  b> value associa
+0000e660: 7465 6420 7769 7468 2065 6163 6820 6765  ted with each ge
+0000e670: 6e65 2070 6169 7220 7375 6368 2074 6861  ne pair such tha
+0000e680: 7420 7468 6520 7765 6967 6874 206f 6620  t the weight of 
+0000e690: 6120 7369 6e67 6c65 2064 7570 6c69 6361  a single duplica
+0000e6a0: 7469 6f6e 2065 7665 6e74 2073 756d 7320  tion event sums 
+0000e6b0: 7570 2074 6f20 3120 286e 6f74 6564 2074  up to 1 (noted t
+0000e6c0: 6861 7420 7468 6520 6e75 6d62 6572 206f  hat the number o
+0000e6d0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+0000e6e0: 3e20 6573 7469 6d61 7469 6f6e 2072 656d  > estimation rem
+0000e6f0: 6169 6e73 2074 6865 2073 616d 6529 2077  ains the same) w
+0000e700: 6869 6c65 2074 6865 206e 6f64 652d 6176  hile the node-av
+0000e710: 6572 6167 6564 206d 6574 686f 6420 7265  eraged method re
+0000e720: 616c 697a 6573 2074 6865 2064 652d 7265  alizes the de-re
+0000e730: 6475 6e64 616e 6379 2076 6961 2063 616c  dundancy via cal
+0000e740: 6375 6c61 7469 6e67 2070 6572 2067 656e  culating per gen
+0000e750: 6520 7472 6565 206e 6f64 6520 6f6e 6520  e tree node one 
+0000e760: 6176 6572 6167 6564 202a 4b2a 3c73 7562  averaged *K*<sub
+0000e770: 3e53 3c2f 7375 623e 2076 616c 7565 2074  >S</sub> value t
+0000e780: 6f20 7265 7072 6573 656e 7420 7468 6520  o represent the 
+0000e790: 6167 6520 6f66 2065 6163 6820 6475 706c  age of each dupl
+0000e7a0: 6963 6174 696f 6e20 6576 656e 742e 2054  ication event. T
+0000e7b0: 6865 206f 7074 696f 6e20 602d 2d6e 6f64  he option `--nod
+0000e7c0: 655f 6176 6572 6167 6560 2063 616e 2062  e_average` can b
+0000e7d0: 6520 7365 7420 746f 2063 686f 6f73 6520  e set to choose 
+0000e7e0: 7468 6520 6e6f 6465 2d61 7665 7261 6765  the node-average
+0000e7f0: 6420 7761 7920 6f66 2064 652d 7265 6475  d way of de-redu
+0000e800: 6e64 616e 6379 2e20 4469 6666 6572 656e  ndancy. Differen
+0000e810: 7420 6d65 7468 6f64 7320 6f66 2064 652d  t methods of de-
+0000e820: 7265 6475 6e64 616e 6379 2077 696c 6c20  redundancy will 
+0000e830: 6861 7665 2069 6d70 6163 7420 6f6e 2074  have impact on t
+0000e840: 6865 2064 6574 6563 7469 6f6e 206f 6620  he detection of 
+0000e850: 5747 4420 7369 676e 616c 732c 2077 6869  WGD signals, whi
+0000e860: 6368 2068 6173 2062 6565 6e20 696e 7665  ch has been inve
+0000e870: 7374 6967 6174 6564 2069 6e20 7468 6973  stigated in this
+0000e880: 205b 6c69 7465 7261 7475 7265 5d28 6874   [literature](ht
+0000e890: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
+0000e8a0: 2e31 3039 332f 6762 652f 6576 7932 3030  .1093/gbe/evy200
+0000e8b0: 292e 2054 6865 2070 6172 616c 6c65 6c20  ). The parallel 
+0000e8c0: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
+0000e8d0: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
+0000e8e0: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
+0000e8f0: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
+0000e900: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
+0000e910: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
+0000e920: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
+0000e930: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
+0000e940: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
+0000e950: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000e960: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+0000e970: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
+0000e980: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000e990: 6965 7320 7365 7175 656e 6365 0a60 6060  ies sequence.```
+0000e9a0: 0a0a 2a2a 5468 6520 636f 6e73 7472 7563  ..**The construc
+0000e9b0: 7469 6f6e 206f 6620 6f72 7468 6f6c 6f67  tion of ortholog
+0000e9c0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+0000e9d0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000e9e0: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
+0000e9f0: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
+0000ea00: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000ea10: 282d 2d72 6577 6569 6768 7429 0a60 6060  (--reweight).```
+0000ea20: 0a0a 4672 6f6d 2070 6172 616c 6f67 6f75  ..From paralogou
+0000ea30: 7320 746f 206f 7274 686f 6c6f 676f 7573  s to orthologous
+0000ea40: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000ea50: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+0000ea60: 6e2c 2075 7365 7273 206f 6e6c 7920 6e65  n, users only ne
+0000ea70: 6564 2074 6f20 7072 6f76 6964 6520 6d6f  ed to provide mo
+0000ea80: 7265 2063 6473 2066 696c 6573 2e20 4e6f  re cds files. No
+0000ea90: 7465 2074 6861 7420 7769 7468 206f 7274  te that with ort
+0000eaa0: 686f 6c6f 676f 7573 2067 656e 6520 6661  hologous gene fa
+0000eab0: 6d69 6c69 6573 2074 6865 2077 6569 6768  milies the weigh
+0000eac0: 7469 6e67 206d 6574 686f 6420 6361 6e20  ting method can 
+0000ead0: 6265 2073 6574 2074 6f20 6265 2063 616c  be set to be cal
+0000eae0: 6375 6c61 7465 6420 7065 7220 7370 6563  culated per spec
+0000eaf0: 6965 7320 7061 6972 2069 6e73 7465 6164  ies pair instead
+0000eb00: 206f 6620 636f 6e73 6964 6572 696e 6720   of considering 
+0000eb10: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000eb20: 2062 6563 6175 7365 2077 6865 6e20 706c   because when pl
+0000eb30: 6f74 7469 6e67 206f 7274 686f 6c6f 676f  otting orthologo
+0000eb40: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+0000eb50: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000eb60: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
+0000eb70: 7370 6563 6965 7320 7468 6520 7765 6967  species the weig
+0000eb80: 6874 2063 616c 6375 6c61 7465 6420 6672  ht calculated fr
+0000eb90: 6f6d 2074 6869 7320 7370 6563 6966 6963  om this specific
+0000eba0: 2073 7065 6369 6573 2070 6169 7220 7368   species pair sh
+0000ebb0: 6f75 6c64 2062 6520 636f 6e73 6572 7665  ould be conserve
+0000ebc0: 6420 7768 696c 6520 7468 6520 6f6e 6520  d while the one 
+0000ebd0: 6361 6c63 756c 6174 6564 2066 726f 6d20  calculated from 
+0000ebe0: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000ebf0: 2077 696c 6c20 7661 7279 2077 6974 6820   will vary with 
+0000ec00: 7468 6520 6e75 6d62 6572 206f 6620 7370  the number of sp
+0000ec10: 6563 6965 732e 2054 6f20 696e 6974 6961  ecies. To initia
+0000ec20: 7465 2074 6865 2077 6569 6768 7469 6e67  te the weighting
+0000ec30: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
+0000ec40: 722c 2074 6865 206f 7074 696f 6e20 602d  r, the option `-
+0000ec50: 2d72 6577 6569 6768 7460 2063 616e 2062  -reweight` can b
+0000ec60: 6520 7365 742e 0a0a 2a2a 4120 7375 6767  e set...**A sugg
+0000ec70: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000ec80: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+0000ec90: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
+0000eca0: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000ecb0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000ecc0: 7175 656e 6365 320a 6060 600a 0a2a 2a54  quence2.```..**T
+0000ecd0: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+0000ece0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
+0000ecf0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000ed00: 696f 6e20 7769 7468 2072 6174 6520 636f  ion with rate co
+0000ed10: 7272 6563 7469 6f6e 2a2a 0a60 6060 0a77  rrection**.```.w
+0000ed20: 6764 206b 7364 2066 616d 696c 6965 7320  gd ksd families 
+0000ed30: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000ed40: 6365 3220 7365 7175 656e 6365 3320 2d2d  ce2 sequence3 --
+0000ed50: 666f 6375 7332 616c 6c20 7365 7175 656e  focus2all sequen
+0000ed60: 6365 3120 2d73 7020 7370 6461 7461 0a60  ce1 -sp spdata.`
+0000ed70: 6060 0a0a 496e 7370 6972 6564 2062 7920  ``..Inspired by 
+0000ed80: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+0000ed90: 696f 6e20 616c 676f 7269 7468 6d20 696e  ion algorithm in
+0000eda0: 205b 6b73 7261 7465 735d 2868 7474 7073   [ksrates](https
+0000edb0: 3a2f 2f67 6974 6875 622e 636f 6d2f 5649  ://github.com/VI
+0000edc0: 422d 5053 422f 6b73 7261 7465 7329 2c20  B-PSB/ksrates), 
+0000edd0: 7765 2069 6d70 6c65 6d65 6e74 6564 2074  we implemented t
+0000ede0: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+0000edf0: 6f6e 2061 6e61 6c79 7369 7320 616c 736f  on analysis also
+0000ee00: 2069 6e20 6077 6764 2076 3260 2c20 7768   in `wgd v2`, wh
+0000ee10: 6963 6820 6973 206d 6f73 746c 7920 7468  ich is mostly th
+0000ee20: 6520 7361 6d65 2061 7320 5b6b 7372 6174  e same as [ksrat
+0000ee30: 6573 5d28 6874 7470 733a 2f2f 6769 7468  es](https://gith
+0000ee40: 7562 2e63 6f6d 2f56 4942 2d50 5342 2f6b  ub.com/VIB-PSB/k
+0000ee50: 7372 6174 6573 2920 6275 7420 6469 6666  srates) but diff
+0000ee60: 6572 7320 696e 2074 6865 2063 616c 6375  ers in the calcu
+0000ee70: 6c61 7469 6f6e 206f 6620 7468 6520 7374  lation of the st
+0000ee80: 616e 6461 7264 2064 6576 6961 7469 6f6e  andard deviation
+0000ee90: 206f 6620 7265 7363 616c 6564 202a 4b2a   of rescaled *K*
+0000eea0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000eeb0: 732e 2054 6f20 7065 7266 6f72 6d20 7468  s. To perform th
+0000eec0: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+0000eed0: 6e20 616e 616c 7973 6973 2c20 7573 6572  n analysis, user
+0000eee0: 7320 6361 6e20 7573 6520 626f 7468 2074  s can use both t
+0000eef0: 6865 2060 7767 6420 6b73 6460 2061 6e64  he `wgd ksd` and
+0000ef00: 2060 7767 6420 7669 7a60 2070 726f 6772   `wgd viz` progr
+0000ef10: 616d 2e20 466f 7220 7468 6520 6077 6764  am. For the `wgd
+0000ef20: 206b 7364 6020 7072 6f67 7261 6d2c 2075   ksd` program, u
+0000ef30: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+0000ef40: 7669 6465 2061 2073 7065 6369 6573 2074  vide a species t
+0000ef50: 7265 6520 7669 6120 7468 6520 6f70 7469  ree via the opti
+0000ef60: 6f6e 2060 2d2d 7370 6563 6965 7374 7265  on `--speciestre
+0000ef70: 6560 206f 6e20 7768 6963 6820 7261 7465  e` on which rate
+0000ef80: 2063 6f72 7265 6374 696f 6e20 6361 6e20   correction can 
+0000ef90: 6265 2063 6f6e 6475 6374 6564 2e20 4e6f  be conducted. No
+0000efa0: 7465 2074 6861 7420 756e 6e65 6365 7373  te that unnecess
+0000efb0: 6172 7920 6272 6163 6b65 7473 206d 6967  ary brackets mig
+0000efc0: 6874 206c 6561 6420 746f 2075 6e65 7870  ht lead to unexp
+0000efd0: 6563 7465 6420 6572 726f 7273 2c20 666f  ected errors, fo
+0000efe0: 7220 696e 7374 616e 6365 2061 2074 7265  r instance a tre
+0000eff0: 6520 6028 412c 2842 2c43 2929 3b60 2073  e `(A,(B,C));` s
+0000f000: 686f 756c 6420 6e6f 7420 6265 2072 6570  hould not be rep
+0000f010: 7265 7365 6e74 6564 2061 7320 6028 412c  resented as `(A,
+0000f020: 2828 422c 4329 2929 3b60 2e20 5468 6520  ((B,C)));`. The 
+0000f030: 7365 7420 6f66 2073 7065 6369 6573 2070  set of species p
+0000f040: 6169 7273 2074 6f20 6265 2073 686f 776e  airs to be shown
+0000f050: 2069 7320 666c 6578 6962 6c65 2074 6861   is flexible tha
+0000f060: 7420 7468 6520 6d6f 7374 2063 6f6e 7665  t the most conve
+0000f070: 6e69 656e 7420 6f70 7469 6f6e 2069 7320  nient option is 
+0000f080: 602d 2d66 6f63 7573 3261 6c6c 6020 7768  `--focus2all` wh
+0000f090: 6963 6820 7369 6d70 6c79 2073 686f 7773  ich simply shows
+0000f0a0: 2061 6c6c 2074 6865 2070 6f73 7369 626c   all the possibl
+0000f0b0: 6520 666f 6361 6c2d 7369 7374 6572 2073  e focal-sister s
+0000f0c0: 7065 6369 6573 2070 6169 7273 2c20 6f72  pecies pairs, or
+0000f0d0: 2075 7365 7273 2063 616e 206d 616e 7561   users can manua
+0000f0e0: 6c6c 7920 7365 7420 7468 6520 7370 6563  lly set the spec
+0000f0f0: 6965 7320 7061 6972 7320 7669 6120 7468  ies pairs via th
+0000f100: 6520 6f70 7469 6f6e 2060 2d2d 7370 6169  e option `--spai
+0000f110: 7260 2e20 4e6f 7465 2074 6861 7420 6966  r`. Note that if
+0000f120: 2074 6865 2073 7065 6369 6573 2070 6169   the species pai
+0000f130: 7273 2077 6572 6520 6d61 6e75 616c 6c79  rs were manually
+0000f140: 2073 6574 2c20 6974 2077 6f75 6c64 2062   set, it would b
+0000f150: 6520 6e65 6564 6564 2074 6f20 7365 7420  e needed to set 
+0000f160: 7468 6520 6f70 7469 6f6e 2060 2d2d 636c  the option `--cl
+0000f170: 6173 7369 6360 2e20 5468 6520 6f70 7469  assic`. The opti
+0000f180: 6f6e 2060 6f6e 6c79 726f 6f74 6f75 7460  on `onlyrootout`
+0000f190: 2063 616e 2062 6520 7365 7420 746f 206f   can be set to o
+0000f1a0: 6e6c 7920 636f 6e73 6964 6572 206f 7574  nly consider out
+0000f1b0: 6772 6f75 7020 6174 2074 6865 2072 6f6f  group at the roo
+0000f1c0: 742c 2069 6e73 7465 6164 206f 6620 616c  t, instead of al
+0000f1d0: 6c20 7468 6520 706f 7373 6962 6c65 206f  l the possible o
+0000f1e0: 7574 6772 6f75 7073 2070 6572 2066 6f63  utgroups per foc
+0000f1f0: 616c 2d73 6973 7465 7220 7370 6563 6965  al-sister specie
+0000f200: 7320 7061 6972 2c20 7768 6963 6820 6861  s pair, which ha
+0000f210: 7320 696d 7061 6374 206f 6e20 7468 6520  s impact on the 
+0000f220: 6669 6e61 6c20 636f 7272 6563 7465 6420  final corrected 
+0000f230: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000f240: 6167 6573 2e20 536f 6d65 206f 7468 6572  ages. Some other
+0000f250: 206f 7074 696f 6e73 2077 6869 6368 2068   options which h
+0000f260: 6176 6520 6e6f 2069 6d70 6163 7420 6f6e  ave no impact on
+0000f270: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+0000f280: 7469 6f6e 2062 7574 2061 6464 206d 6f72  tion but add mor
+0000f290: 6520 6c61 7965 7273 206f 7220 6368 616e  e layers or chan
+0000f2a0: 6765 2074 6865 2061 7070 6561 7261 6e63  ge the appearanc
+0000f2b0: 6520 6f6e 2074 6865 206d 6978 6564 202a  e on the mixed *
+0000f2c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+0000f2d0: 6765 2064 6973 7472 6962 7574 696f 6e2c  ge distribution,
+0000f2e0: 2069 6e63 6c75 6465 2060 706c 6f74 6170   include `plotap
+0000f2f0: 676d 6d60 2061 6e64 2060 706c 6f74 656c  gmm` and `plotel
+0000f300: 6d6d 6020 6574 632e 2054 6865 206f 7074  mm` etc. The opt
+0000f310: 696f 6e20 6070 6c6f 7461 7067 6d6d 6020  ion `plotapgmm` 
+0000f320: 6361 6e20 6265 2073 6574 2074 6f20 6361  can be set to ca
+0000f330: 6c6c 2074 6865 2047 4d4d 2061 6e61 6c79  ll the GMM analy
+0000f340: 7369 7320 7570 6f6e 2074 6865 2061 6e63  sis upon the anc
+0000f350: 686f 7220 7061 6972 202a 4b2a 3c73 7562  hor pair *K*<sub
+0000f360: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+0000f370: 7472 6962 7574 696f 6e20 616e 6420 706c  tribution and pl
+0000f380: 6f74 2074 6865 2063 6c75 7374 6572 696e  ot the clusterin
+0000f390: 6720 7265 7375 6c74 2075 706f 6e20 7468  g result upon th
+0000f3a0: 6520 6d69 7865 6420 2a4b 2a3c 7375 623e  e mixed *K*<sub>
+0000f3b0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+0000f3c0: 7269 6275 7469 6f6e 2c20 7768 6963 6820  ribution, which 
+0000f3d0: 6861 7320 746f 2062 6520 636f 2d73 6574  has to be co-set
+0000f3e0: 2077 6974 6820 7468 6520 6f70 7469 6f6e   with the option
+0000f3f0: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
+0000f400: 6020 7072 6f76 6964 696e 6720 7468 6520  ` providing the 
+0000f410: 616e 6368 6f72 2070 6169 7273 2069 6e66  anchor pairs inf
+0000f420: 6f72 6d61 7469 6f6e 2e20 5468 6520 6f70  ormation. The op
+0000f430: 7469 6f6e 2060 2d2d 706c 6f74 656c 6d6d  tion `--plotelmm
+0000f440: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
+0000f450: 6361 6c6c 2074 6865 2045 4c4d 4d20 616e  call the ELMM an
+0000f460: 616c 7973 6973 2075 706f 6e20 7468 6520  alysis upon the 
+0000f470: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
+0000f480: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000f490: 6973 7472 6962 7574 696f 6e2e 0a0a 5468  istribution...Th
+0000f4a0: 6572 6520 6172 6520 3231 2063 6f6c 756d  ere are 21 colum
+0000f4b0: 6e73 2069 6e20 7468 6520 7265 7375 6c74  ns in the result
+0000f4c0: 2060 2a6b 732e 7473 7660 2066 696c 6520   `*ks.tsv` file 
+0000f4d0: 6265 7369 6465 7320 7468 6520 696e 6465  besides the inde
+0000f4e0: 7820 636f 6c75 6d6e 7320 6070 6169 7260  x columns `pair`
+0000f4f0: 2061 7320 7468 6520 756e 6971 7565 2069   as the unique i
+0000f500: 6465 6e74 6966 6965 7220 666f 7220 6561  dentifier for ea
+0000f510: 6368 2067 656e 6520 7061 6972 2e20 5468  ch gene pair. Th
+0000f520: 6520 604e 602c 2060 5360 2c20 6064 4e60  e `N`, `S`, `dN`
+0000f530: 2c20 6064 4e2f 6453 602c 2060 6453 602c  , `dN/dS`, `dS`,
+0000f540: 2060 6c60 2061 6e64 2060 7460 2061 7265   `l` and `t` are
+0000f550: 2066 726f 6d20 7468 6520 636f 6465 6d6c   from the codeml
+0000f560: 2072 6573 756c 7473 2c20 7265 7072 6573   results, repres
+0000f570: 656e 7469 6e67 2074 6865 204e 2065 7374  enting the N est
+0000f580: 696d 6174 652c 2074 6865 2053 2065 7374  imate, the S est
+0000f590: 696d 6174 652c 2074 6865 2064 4e20 6573  imate, the dN es
+0000f5a0: 7469 6d61 7465 2c20 7468 6520 644e 2f64  timate, the dN/d
+0000f5b0: 5320 286f 6d65 6761 2920 6573 7469 6d61  S (omega) estima
+0000f5c0: 7465 2c20 7468 6520 6453 2065 7374 696d  te, the dS estim
+0000f5d0: 6174 652c 2074 6865 206c 6f67 2d6c 696b  ate, the log-lik
+0000f5e0: 656c 6968 6f6f 6420 616e 6420 7468 6520  elihood and the 
+0000f5f0: 7420 6573 7469 6d61 7465 2c20 7265 7370  t estimate, resp
+0000f600: 6563 7469 7665 6c79 2e20 5468 6520 6061  ectively. The `a
+0000f610: 6c69 676e 6d65 6e74 636f 7665 7261 6765  lignmentcoverage
+0000f620: 602c 2060 616c 6967 6e6d 656e 7469 6465  `, `alignmentide
+0000f630: 6e74 6974 7960 2061 6e64 2060 616c 6967  ntity` and `alig
+0000f640: 6e6d 656e 746c 656e 6774 6860 2061 7265  nmentlength` are
+0000f650: 2074 6865 2069 6e66 6f72 6d61 7469 6f6e   the information
+0000f660: 2070 6572 7461 696e 696e 6720 746f 2074   pertaining to t
+0000f670: 6865 2061 6c69 676e 6d65 6e74 2066 6f72  he alignment for
+0000f680: 2065 6163 6820 6661 6d69 6c79 2c20 7265   each family, re
+0000f690: 7072 6573 656e 7469 6e67 2074 6865 2072  presenting the r
+0000f6a0: 6174 696f 206f 6620 7468 6520 7374 7269  atio of the stri
+0000f6b0: 7070 6564 2061 6c69 676e 6d65 6e74 206c  pped alignment l
+0000f6c0: 656e 6774 6820 636f 6d70 6172 6564 2074  ength compared t
+0000f6d0: 6f20 7468 6520 6675 6c6c 2061 6c69 676e  o the full align
+0000f6e0: 6d65 6e74 206c 656e 6774 682c 2074 6865  ment length, the
+0000f6f0: 2072 6174 696f 206f 6620 636f 6c75 6d6e   ratio of column
+0000f700: 7320 7769 7468 2069 6465 6e74 6963 616c  s with identical
+0000f710: 206e 7563 6c65 6f74 6964 6573 2063 6f6d   nucleotides com
+0000f720: 7061 7265 6420 746f 2074 6865 206f 7665  pared to the ove
+0000f730: 7261 6c6c 2063 6f6c 756d 6e73 206f 6620  rall columns of 
+0000f740: 7468 6520 7374 7269 7070 6564 2061 6c69  the stripped ali
+0000f750: 676e 6d65 6e74 2c20 616e 6420 7468 6520  gnment, and the 
+0000f760: 6c65 6e67 7468 206f 6620 7468 6520 6675  length of the fu
+0000f770: 6c6c 2061 6c69 676e 6d65 6e74 2c20 7265  ll alignment, re
+0000f780: 7370 6563 7469 7665 6c79 2e0a 2323 2320  spectively..### 
+0000f790: 7767 6420 6d69 780a 0a2a 2a54 6865 206d  wgd mix..**The m
+0000f7a0: 6978 7475 7265 206d 6f64 656c 2063 6c75  ixture model clu
+0000f7b0: 7374 6572 696e 6720 616e 616c 7973 6973  stering analysis
+0000f7c0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
+0000f7d0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000f7e0: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206d  tion**.```.wgd m
+0000f7f0: 6978 206b 7364 6174 6120 282d 2d6e 5f69  ix ksdata (--n_i
+0000f800: 6e69 7420 3230 3020 2d2d 6d61 785f 6974  nit 200 --max_it
+0000f810: 6572 2032 3030 202d 2d6b 735f 7261 6e67  er 200 --ks_rang
+0000f820: 6520 3020 3520 2d2d 6669 6c74 6572 7320  e 0 5 --filters 
+0000f830: 3330 3020 2d2d 6269 6e73 2035 3020 2d2d  300 --bins 50 --
+0000f840: 636f 6d70 6f6e 656e 7473 2031 2034 202d  components 1 4 -
+0000f850: 2d67 616d 6d61 2030 2e30 3031 290a 6060  -gamma 0.001).``
+0000f860: 600a 0a54 6869 7320 7061 7274 206f 6620  `..This part of 
+0000f870: 4761 7573 7369 616e 206d 6978 7475 7265  Gaussian mixture
+0000f880: 206d 6f64 656c 696e 6720 2847 4d4d 2920   modeling (GMM) 
+0000f890: 616e 616c 7973 6973 2069 7320 696e 6865  analysis is inhe
+0000f8a0: 7269 7465 6420 6672 6f6d 2074 6865 206f  rited from the o
+0000f8b0: 7269 6769 6e61 6c20 6077 6764 6020 7072  riginal `wgd` pr
+0000f8c0: 6f67 7261 6d2c 2062 7574 2077 7269 7465  ogram, but write
+0000f8d0: 7320 6164 6469 7469 6f6e 616c 6c79 2074  s additionally t
+0000f8e0: 6865 2070 726f 6261 6269 6c69 7479 206f  he probability o
+0000f8f0: 6620 6561 6368 202a 4b2a 3c73 7562 3e53  f each *K*<sub>S
+0000f900: 3c2f 7375 623e 2076 616c 7565 2069 6e74  </sub> value int
+0000f910: 6f20 7468 6520 6669 6e61 6c20 6461 7461  o the final data
+0000f920: 6672 616d 652e 2042 6173 6963 616c 6c79  frame. Basically
+0000f930: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000f940: 7072 6f76 6964 6520 7769 7468 2061 2028  provide with a (
+0000f950: 6e6f 726d 616c 6c79 2066 726f 6d20 7768  normally from wh
+0000f960: 6f6c 652d 7061 7261 6e6f 6d65 206f 7220  ole-paranome or 
+0000f970: 616e 6368 6f72 2d70 6169 7273 2920 2a4b  anchor-pairs) *K
+0000f980: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+0000f990: 7461 6669 6c65 2061 6e64 2074 6865 2047  tafile and the G
+0000f9a0: 4d4d 2061 6e61 6c79 7369 7320 7769 6c6c  MM analysis will
+0000f9b0: 2062 6520 636f 6e64 7563 7465 6420 7570   be conducted up
+0000f9c0: 6f6e 2074 6865 2064 6174 6166 696c 652e  on the datafile.
+0000f9d0: 2053 6f6d 6520 7061 7261 6d65 7465 7273   Some parameters
+0000f9e0: 2063 616e 2061 6666 6563 7420 7468 6520   can affect the 
+0000f9f0: 7265 7375 6c74 732c 2069 6e63 6c75 6469  results, includi
+0000fa00: 6e67 2060 2d2d 6e5f 696e 6974 602c 2077  ng `--n_init`, w
+0000fa10: 6869 6368 2073 6574 7320 7468 6520 6e75  hich sets the nu
+0000fa20: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
+0000fa30: 696e 6974 6961 6c69 7a61 7469 6f6e 7320  initializations 
+0000fa40: 2864 6566 6175 6c74 2032 3030 292c 2060  (default 200), `
+0000fa50: 2d2d 6d61 785f 6974 6572 602c 2077 6869  --max_iter`, whi
+0000fa60: 6368 2073 6574 7320 7468 6520 6d61 7869  ch sets the maxi
+0000fa70: 6d75 6d20 6e75 6d62 6572 206f 6620 6974  mum number of it
+0000fa80: 6572 6174 696f 6e73 2028 6465 6661 756c  erations (defaul
+0000fa90: 7420 3230 3029 2c20 602d 2d6d 6574 686f  t 200), `--metho
+0000faa0: 6460 2c20 7768 6963 6820 6465 7465 726d  d`, which determ
+0000fab0: 696e 6573 2077 6869 6368 2063 6c75 7374  ines which clust
+0000fac0: 6572 696e 6720 6d65 7468 6f64 2074 6f20  ering method to 
+0000fad0: 7573 6520 2864 6566 6175 6c74 2067 6d6d  use (default gmm
+0000fae0: 292c 2060 2d2d 6761 6d6d 6160 2c20 7768  ), `--gamma`, wh
+0000faf0: 6963 6820 7365 7473 2074 6865 2067 616d  ich sets the gam
+0000fb00: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
+0000fb10: 2074 6865 2062 676d 6d20 6d6f 6465 6c20   the bgmm model 
+0000fb20: 2864 6566 6175 6c74 2030 2e30 3031 292c  (default 0.001),
+0000fb30: 2060 2d2d 636f 6d70 6f6e 656e 7473 602c   `--components`,
+0000fb40: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+0000fb50: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
+0000fb60: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
+0000fb70: 7320 746f 2066 6974 2028 6465 6661 756c  s to fit (defaul
+0000fb80: 7420 3120 3429 2c20 616e 6420 7468 6520  t 1 4), and the 
+0000fb90: 6461 7461 2066 696c 7465 7269 6e67 2070  data filtering p
+0000fba0: 6172 616d 6574 6572 7320 602d 2d66 696c  arameters `--fil
+0000fbb0: 7465 7273 6020 7768 6963 6820 6669 6c74  ters` which filt
+0000fbc0: 6572 7320 6461 7461 2062 6173 6564 206f  ers data based o
+0000fbd0: 6e20 616c 6967 6e6d 656e 7420 6c65 6e67  n alignment leng
+0000fbe0: 7468 2c20 602d 2d6b 735f 7261 6e67 6560  th, `--ks_range`
+0000fbf0: 2077 6869 6368 2066 696c 7465 7273 2064   which filters d
+0000fc00: 6174 6120 6261 7365 6420 6f6e 202a 4b2a  ata based on *K*
+0000fc10: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
+0000fc20: 7565 7320 616e 6420 7468 6520 7061 7261  ues and the para
+0000fc30: 6d65 7465 7220 602d 2d62 696e 7360 2077  meter `--bins` w
+0000fc40: 6869 6368 2073 6574 7320 7468 6520 6e75  hich sets the nu
+0000fc50: 6d62 6572 206f 6620 6269 6e73 2069 6e20  mber of bins in 
+0000fc60: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000fc70: 6469 7374 7269 6275 7469 6f6e 2028 6465  distribution (de
+0000fc80: 6661 756c 7420 3530 292e 0a0a 2a2a 4120  fault 50)...**A 
+0000fc90: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+0000fca0: 6e67 2072 756e 2063 616e 2075 7365 2063  ng run can use c
+0000fcb0: 6f6d 6d61 6e64 2073 696d 706c 7920 6173  ommand simply as
+0000fcc0: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
+0000fcd0: 6420 6d69 7820 6b73 6461 7461 0a60 6060  d mix ksdata.```
+0000fce0: 0a0a 2323 2320 7767 6420 7065 616b 0a0a  ..### wgd peak..
+0000fcf0: 2a2a 5468 6520 7365 6172 6368 206f 6620  **The search of 
+0000fd00: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
+0000fd10: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
+0000fd20: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
+0000fd30: 6e67 2a2a 0a60 6060 0a77 6764 2070 6561  ng**.```.wgd pea
+0000fd40: 6b20 6b73 6461 7461 202d 6170 2061 7064  k ksdata -ap apd
+0000fd50: 6174 6120 2d73 6d20 736d 6461 7461 202d  ata -sm smdata -
+0000fd60: 6c65 206c 6564 6174 6120 2d6d 7020 6d70  le ledata -mp mp
+0000fd70: 6461 7461 202d 2d68 6575 7269 7374 6963  data --heuristic
+0000fd80: 2028 2d2d 616c 6967 6e66 696c 7465 7220   (--alignfilter 
+0000fd90: 302e 3020 3020 302e 3020 2d2d 6b73 7261  0.0 0 0.0 --ksra
+0000fda0: 6e67 6520 3020 3520 2d2d 6269 6e5f 7769  nge 0 5 --bin_wi
+0000fdb0: 6474 6820 302e 3120 2d2d 6775 6964 6520  dth 0.1 --guide 
+0000fdc0: 7365 676d 656e 7420 2d2d 7072 6f6d 696e  segment --promin
+0000fdd0: 656e 6365 5f63 7574 6f66 6620 302e 3120  ence_cutoff 0.1 
+0000fde0: 2d2d 7265 6c5f 6865 6967 6874 2030 2e34  --rel_height 0.4
+0000fdf0: 202d 2d63 6920 3935 202d 2d68 6472 2039   --ci 95 --hdr 9
+0000fe00: 3520 2d2d 6b73 6375 746f 6666 2035 290a  5 --kscutoff 5).
+0000fe10: 6060 600a 0a41 7320 6d65 6e74 696f 6e65  ```..As mentione
+0000fe20: 6420 7072 6576 696f 7573 6c79 2c20 6120  d previously, a 
+0000fe30: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
+0000fe40: 2061 6e64 2061 2063 6f6c 6c69 6e65 6172   and a collinear
+0000fe50: 2073 6567 6d65 6e74 732d 6775 6964 6564   segments-guided
+0000fe60: 2061 6e63 686f 7220 7061 6972 2063 6c75   anchor pair clu
+0000fe70: 7374 6572 696e 6720 666f 7220 7468 6520  stering for the 
+0000fe80: 7365 6172 6368 206f 6620 6372 6564 6961  search of credia
+0000fe90: 626c 6520 2a4b 2a3c 7375 623e 533c 2f73  ble *K*<sub>S</s
+0000fea0: 7562 3e20 7261 6e67 6520 7573 6564 2069  ub> range used i
+0000feb0: 6e20 5747 4420 6461 7469 6e67 2061 7265  n WGD dating are
+0000fec0: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
+0000fed0: 6077 6764 2076 3260 2e20 5573 6572 7320  `wgd v2`. Users 
+0000fee0: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
+0000fef0: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
+0000ff00: 732c 2073 6567 6d65 6e74 732c 206c 6973  s, segments, lis
+0000ff10: 7473 6567 6d65 6e74 732c 206d 756c 7469  tsegments, multi
+0000ff20: 706c 6963 6f6e 7061 6972 7320 6461 7461  pliconpairs data
+0000ff30: 6669 6c65 2066 726f 6d20 6069 2d61 6468  file from `i-adh
+0000ff40: 6f72 6560 2074 6f20 6163 6869 6576 6520  ore` to achieve 
+0000ff50: 7468 6520 636c 7573 7465 7269 6e67 2066  the clustering f
+0000ff60: 756e 6374 696f 6e2e 2053 6f6d 6520 7061  unction. Some pa
+0000ff70: 7261 6d65 7465 7273 2074 6861 7420 6361  rameters that ca
+0000ff80: 6e20 696d 7061 6374 2074 6865 2072 6573  n impact the res
+0000ff90: 756c 7473 2069 6e63 6c75 6465 2060 2d2d  ults include `--
+0000ffa0: 616c 6967 6e66 696c 7465 7260 2c20 7768  alignfilter`, wh
+0000ffb0: 6963 6820 6669 6c74 6572 7320 7468 6520  ich filters the 
+0000ffc0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
+0000ffd0: 6967 6e6d 656e 7420 6964 656e 7469 7479  ignment identity
+0000ffe0: 2c20 6c65 6e67 7468 2061 6e64 2063 6f76  , length and cov
+0000fff0: 6572 6167 652c 2060 2d2d 6b73 7261 6e67  erage, `--ksrang
+00010000: 6560 2c20 7768 6963 6820 7365 7473 2074  e`, which sets t
+00010010: 6865 2072 616e 6765 206f 6620 4b73 2074  he range of Ks t
+00010020: 6f20 6265 2061 6e61 6c79 7a65 642c 2060  o be analyzed, `
+00010030: 2d2d 6269 6e5f 7769 6474 6860 2c20 7768  --bin_width`, wh
+00010040: 6963 6820 7365 7473 2074 6865 2062 616e  ich sets the ban
+00010050: 6477 6964 7468 206f 6620 2a4b 2a3c 7375  dwidth of *K*<su
+00010060: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00010070: 6275 7469 6f6e 2c20 602d 2d77 6569 6768  bution, `--weigh
+00010080: 7473 5f6f 7574 6c69 6572 735f 696e 636c  ts_outliers_incl
+00010090: 7564 6564 6020 7768 6963 6820 6465 7465  uded` which dete
+000100a0: 726d 696e 6573 2077 6865 7468 6572 2074  rmines whether t
+000100b0: 6f20 696e 636c 7564 6520 2a4b 2a3c 7375  o include *K*<su
+000100c0: 623e 533c 2f73 7562 3e20 6f75 746c 6965  b>S</sub> outlie
+000100d0: 7273 2028 7768 6f73 6520 7661 6c75 6520  rs (whose value 
+000100e0: 6973 206f 7665 7220 3529 2069 6e20 616e  is over 5) in an
+000100f0: 616c 7973 6973 2c20 602d 2d6d 6574 686f  alysis, `--metho
+00010100: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
+00010110: 6e65 7320 7768 6963 6820 636c 7573 7465  nes which cluste
+00010120: 7269 6e67 206d 6574 686f 6420 746f 2075  ring method to u
+00010130: 7365 2028 6465 6661 756c 7420 676d 6d29  se (default gmm)
+00010140: 2c20 602d 2d73 6565 6460 2077 6869 6368  , `--seed` which
+00010150: 2073 6574 7320 7468 6520 7261 6e64 6f6d   sets the random
+00010160: 2073 6565 6420 6769 7665 6e20 746f 2069   seed given to i
+00010170: 6e69 7469 616c 697a 6174 696f 6e20 2864  nitialization (d
+00010180: 6566 6175 6c74 2032 3335 3238 3930 292c  efault 2352890),
+00010190: 2060 2d2d 6e5f 696e 6974 602c 2077 6869   `--n_init`, whi
+000101a0: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
+000101b0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
+000101c0: 6974 6961 6c69 7a61 7469 6f6e 7320 2864  itializations (d
+000101d0: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
+000101e0: 656d 5f69 7465 7260 2c20 7768 6963 6820  em_iter`, which 
+000101f0: 7365 7473 2074 6865 206d 6178 696d 756d  sets the maximum
+00010200: 206e 756d 6265 7220 6f66 2069 7465 7261   number of itera
+00010210: 7469 6f6e 7320 2864 6566 6175 6c74 2032  tions (default 2
+00010220: 3030 292c 2060 2d2d 6761 6d6d 6160 2c20  00), `--gamma`, 
+00010230: 7768 6963 6820 7365 7473 2074 6865 2067  which sets the g
+00010240: 616d 6d61 2070 6172 616d 6574 6572 2066  amma parameter f
+00010250: 6f72 2074 6865 2062 676d 6d20 6d6f 6465  or the bgmm mode
+00010260: 6c20 2864 6566 6175 6c74 2030 2e30 3031  l (default 0.001
+00010270: 292c 2060 2d2d 636f 6d70 6f6e 656e 7473  ), `--components
+00010280: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+00010290: 6520 7261 6e67 6520 6f66 2074 6865 206e  e range of the n
+000102a0: 756d 6265 7220 6f66 2063 6f6d 706f 6e65  umber of compone
+000102b0: 6e74 7320 746f 2066 6974 2028 6465 6661  nts to fit (defa
+000102c0: 756c 7420 3120 3429 2c20 602d 2d77 6569  ult 1 4), `--wei
+000102d0: 6768 7465 6460 2077 6869 6368 2064 6574  ghted` which det
+000102e0: 6572 6d69 6e65 7320 7768 6574 6865 7220  ermines whether 
+000102f0: 746f 2075 7365 206e 6f64 652d 7765 6967  to use node-weig
+00010300: 6874 6564 206d 6574 686f 6420 666f 7220  hted method for 
+00010310: 6465 2d72 6564 756e 6461 6e63 792c 2060  de-redundancy, `
+00010320: 2d2d 6775 6964 6560 2077 6869 6368 2064  --guide` which d
+00010330: 6574 6572 6d69 6e65 7320 7768 6963 6820  etermines which 
+00010340: 7265 6769 6d65 2072 6573 6964 696e 6720  regime residing 
+00010350: 616e 6368 6f72 7320 746f 2062 6520 7573  anchors to be us
+00010360: 6564 2028 6465 6661 756c 7420 5365 676d  ed (default Segm
+00010370: 656e 7429 2c20 602d 2d70 726f 6d69 6e65  ent), `--promine
+00010380: 6e63 655f 6375 746f 6666 6020 7768 6963  nce_cutoff` whic
+00010390: 6820 7365 7473 2074 6865 2070 726f 6d69  h sets the promi
+000103a0: 6e65 6e63 6520 6375 746f 6666 206f 6620  nence cutoff of 
+000103b0: 6163 6365 7074 6162 6c65 2070 6561 6b73  acceptable peaks
+000103c0: 2069 6e20 7065 616b 2066 696e 6469 6e67   in peak finding
+000103d0: 2070 726f 6365 7373 2c20 602d 2d72 656c   process, `--rel
+000103e0: 5f68 6569 6768 7460 2077 6869 6368 2073  _height` which s
+000103f0: 6574 7320 7468 6520 7265 6c61 7469 7665  ets the relative
+00010400: 2068 6569 6768 7420 6174 2077 6869 6368   height at which
+00010410: 2074 6865 2070 6561 6b20 7769 6474 6820   the peak width 
+00010420: 6973 206d 6561 7375 7265 642c 2060 2d2d  is measured, `--
+00010430: 6b73 746f 6461 7465 6020 7768 6963 6820  kstodate` which 
+00010440: 6d61 6e75 616c 6c79 2073 6574 7320 7468  manually sets th
+00010450: 6520 7261 6e67 6520 6f66 202a 4b2a 3c73  e range of *K*<s
+00010460: 7562 3e53 3c2f 7375 623e 2074 6f20 6265  ub>S</sub> to be
+00010470: 2064 6174 6564 2069 6e20 6865 7572 6973   dated in heuris
+00010480: 7469 6320 7365 6172 6368 2061 6e64 206e  tic search and n
+00010490: 6565 6473 2074 6f20 6265 2063 6f2d 7365  eeds to be co-se
+000104a0: 7420 7769 7468 206f 7074 696f 6e20 602d  t with option `-
+000104b0: 2d6d 616e 7561 6c73 6574 602c 2060 2d2d  -manualset`, `--
+000104c0: 786c 696d 6020 616e 6420 602d 2d79 6c69  xlim` and `--yli
+000104d0: 6d60 2064 6574 6572 6d69 6e69 6e67 2074  m` determining t
+000104e0: 6865 2078 2061 6e64 2079 2061 7869 7320  he x and y axis 
+000104f0: 6c69 6d69 7420 6f66 2047 4d4d 202a 4b2a  limit of GMM *K*
+00010500: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+00010510: 7472 6962 7574 696f 6e2c 2060 2d2d 6369  tribution, `--ci
+00010520: 6020 7365 7474 696e 6720 7468 6520 636f  ` setting the co
+00010530: 6e66 6964 656e 6365 206c 6576 656c 206f  nfidence level o
+00010540: 6620 6c6f 672d 6e6f 726d 616c 2064 6973  f log-normal dis
+00010550: 7472 6962 7574 696f 6e20 746f 2064 6174  tribution to dat
+00010560: 6520 2864 6566 6175 6c74 2039 3529 2c20  e (default 95), 
+00010570: 602d 2d68 6472 6020 7365 7474 696e 6720  `--hdr` setting 
+00010580: 7468 6520 6869 6768 6573 7420 6465 6e73  the highest dens
+00010590: 6974 7920 7265 6769 6f6e 2028 4844 5229  ity region (HDR)
+000105a0: 2061 7070 6c69 6564 2069 6e20 7468 6520   applied in the 
+000105b0: 7365 676d 656e 742d 6775 6964 6564 2061  segment-guided a
+000105c0: 6e63 686f 7220 7061 6972 202a 4b2a 3c73  nchor pair *K*<s
+000105d0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
+000105e0: 6962 7574 696f 6e20 2864 6566 6175 6c74  ibution (default
+000105f0: 2039 3529 2c20 602d 2d68 6575 7269 7374   95), `--heurist
+00010600: 6963 6020 6465 7465 726d 696e 696e 6720  ic` determining 
+00010610: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00010620: 6174 6520 6865 7572 6973 7469 6320 6d65  ate heuristic me
+00010630: 7468 6f64 206f 6620 6465 6669 6e69 6e67  thod of defining
+00010640: 2043 4920 666f 7220 6461 7469 6e67 2c20   CI for dating, 
+00010650: 602d 2d6b 7363 7574 6f66 6660 2073 6574  `--kscutoff` set
+00010660: 7469 6e67 2074 6865 202a 4b2a 3c73 7562  ting the *K*<sub
+00010670: 3e53 3c2f 7375 623e 2073 6174 7572 6174  >S</sub> saturat
+00010680: 696f 6e20 6375 746f 6666 2069 6e20 6461  ion cutoff in da
+00010690: 7469 6e67 2028 6465 6661 756c 7420 3529  ting (default 5)
+000106a0: 2e0a 0a46 6f75 7220 7265 7375 6c74 2073  ...Four result s
+000106b0: 7562 666c 6f64 6572 7320 7769 6c6c 2062  ubfloders will b
+000106c0: 6520 7072 6f64 7563 6564 2c20 6e61 6d65  e produced, name
+000106d0: 6c79 2060 416e 6368 6f72 4b73 5f46 696e  ly `AnchorKs_Fin
+000106e0: 6450 6561 6b60 2c20 6041 6e63 686f 724b  dPeak`, `AnchorK
+000106f0: 735f 474d 4d60 2c20 6053 6567 6d65 6e74  s_GMM`, `Segment
+00010700: 4775 6964 654b 735f 474d 4d60 2061 6e64  GuideKs_GMM` and
+00010710: 2060 5365 676d 656e 744b 735f 4669 6e64   `SegmentKs_Find
+00010720: 5065 616b 602e 2054 6865 2060 416e 6368  Peak`. The `Anch
+00010730: 6f72 4b73 5f46 696e 6450 6561 6b60 2073  orKs_FindPeak` s
+00010740: 7562 666c 6f64 6572 2063 6f6e 7461 696e  ubfloder contain
+00010750: 7320 7265 7375 6c74 7320 6f66 2074 6865  s results of the
+00010760: 2064 6574 6563 7465 6420 7065 616b 7320   detected peaks 
+00010770: 6279 2074 6865 2060 7369 676e 616c 6020  by the `signal` 
+00010780: 6d6f 6475 6c65 206f 6620 6053 6369 5079  module of `SciPy
+00010790: 6020 6c69 6272 6172 7920 616e 6420 7468  ` library and th
+000107a0: 6520 6173 7375 6d65 6420 6869 6768 6573  e assumed highes
+000107b0: 7420 6d61 7373 2070 6172 7420 2872 6566  t mass part (ref
+000107c0: 6572 7265 6420 746f 2061 7320 4869 6768  erred to as High
+000107d0: 4d61 7373 2068 6572 6561 6674 6572 2920  Mass hereafter) 
+000107e0: 6f66 2065 6163 6820 7065 616b 2c20 7768  of each peak, wh
+000107f0: 6963 6820 6361 6e20 6265 2075 7365 6420  ich can be used 
+00010800: 666f 7220 6675 7274 6865 7220 5747 4420  for further WGD 
+00010810: 6461 7469 6e67 2e20 5468 6520 6041 6e63  dating. The `Anc
+00010820: 686f 724b 735f 474d 4d60 2073 686f 7773  horKs_GMM` shows
+00010830: 2074 6865 2047 4d4d 2072 6573 756c 7473   the GMM results
+00010840: 2075 706f 6e20 7468 6520 6f72 6967 696e   upon the origin
+00010850: 616c 2061 6e63 686f 7220 2a4b 2a3c 7375  al anchor *K*<su
+00010860: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00010870: 6275 7469 6f6e 2062 7920 7468 6520 606d  bution by the `m
+00010880: 6978 7475 7265 6020 6d6f 6475 6c65 206f  ixture` module o
+00010890: 6620 6073 6369 6b69 742d 6c65 6172 6e60  f `scikit-learn`
+000108a0: 206c 6962 7261 7279 2061 6e64 2074 776f   library and two
+000108b0: 2073 7562 666c 6f64 6572 732c 2060 4c6f   subfloders, `Lo
+000108c0: 6747 4d4d 5f43 4960 2063 6f6e 7461 696e  gGMM_CI` contain
+000108d0: 696e 6720 7468 6520 7265 7375 6c74 7320  ing the results 
+000108e0: 6f66 2039 3525 2043 4920 6f66 2065 6163  of 95% CI of eac
+000108f0: 6820 636f 6d70 6f6e 656e 742c 2060 4869  h component, `Hi
+00010900: 6768 4d61 7373 5f43 4960 2063 6f6e 7461  ghMass_CI` conta
+00010910: 696e 696e 6720 7468 6520 4869 6768 4d61  ining the HighMa
+00010920: 7373 206f 6620 6561 6368 2063 6f6d 706f  ss of each compo
+00010930: 6e65 6e74 2c20 7768 6963 6820 6361 6e20  nent, which can 
+00010940: 6265 2075 7365 6420 666f 7220 6675 7274  be used for furt
+00010950: 6865 7220 5747 4420 6461 7469 6e67 2e20  her WGD dating. 
+00010960: 5468 6520 6053 6567 6d65 6e74 4775 6964  The `SegmentGuid
+00010970: 654b 735f 474d 4d60 2073 7562 666f 6c64  eKs_GMM` subfold
+00010980: 6572 2070 7265 7365 6e74 7320 7265 7375  er presents resu
+00010990: 6c74 7320 6f66 2073 6567 6d65 6e74 202a  lts of segment *
+000109a0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2047  K*<sub>S</sub> G
+000109b0: 4d4d 2077 6869 6368 2061 7265 206d 6170  MM which are map
+000109c0: 7065 6420 6261 636b 2074 6f20 7468 6520  ped back to the 
+000109d0: 7265 7369 6469 6e67 2061 6e63 686f 7220  residing anchor 
+000109e0: 7061 6972 7320 616e 6420 7468 6520 6173  pairs and the as
+000109f0: 736f 6369 6174 6564 2039 3525 2048 4452  sociated 95% HDR
+00010a00: 2061 6e64 2048 6967 684d 6173 7320 6f66   and HighMass of
+00010a10: 2065 6163 6820 7365 676d 656e 7420 636c   each segment cl
+00010a20: 7573 7465 7220 696e 2073 7562 666c 6f64  uster in subflod
+00010a30: 6572 7320 6f66 2060 4844 525f 4349 6020  ers of `HDR_CI` 
+00010a40: 616e 6420 6048 6967 684d 6173 735f 4349  and `HighMass_CI
+00010a50: 602e 2054 6865 2060 5365 676d 656e 744b  `. The `SegmentK
+00010a60: 735f 4669 6e64 5065 616b 6020 7375 6266  s_FindPeak` subf
+00010a70: 6f6c 6465 7220 6973 2073 696d 696c 6172  older is similar
+00010a80: 2074 6f20 6041 6e63 686f 724b 735f 4669   to `AnchorKs_Fi
+00010a90: 6e64 5065 616b 6020 6275 7420 7769 7468  ndPeak` but with
+00010aa0: 2073 6567 6d65 6e74 202a 4b2a 3c73 7562   segment *K*<sub
+00010ab0: 3e53 3c2f 7375 623e 2069 6e73 7465 6164  >S</sub> instead
+00010ac0: 2e20 5468 6520 2a4b 2a3c 7375 623e 533c  . The *K*<sub>S<
+00010ad0: 2f73 7562 3e20 696e 2060 4d75 6c74 6970  /sub> in `Multip
+00010ae0: 6c69 636f 6e60 2063 616e 2061 6c73 6f20  licon` can also 
+00010af0: 6265 2063 616c 6375 6c61 7465 6420 696e  be calculated in
+00010b00: 2070 6c61 6365 206f 6620 6053 6567 6d65   place of `Segme
+00010b10: 6e74 6020 7573 696e 6720 7468 6520 6f70  nt` using the op
+00010b20: 7469 6f6e 2060 2d2d 6775 6964 6560 2061  tion `--guide` a
+00010b30: 7320 7375 6368 2074 6865 2072 6573 756c  s such the resul
+00010b40: 7420 7469 746c 652c 206c 6162 656c 2c20  t title, label, 
+00010b50: 6669 6c65 2061 6e64 2066 6f6c 6465 7220  file and folder 
+00010b60: 6e61 6d65 7320 7769 6c6c 2062 6520 6368  names will be ch
+00010b70: 616e 6765 6420 6163 636f 7264 696e 676c  anged accordingl
+00010b80: 792e 0a0a 2a2a 4120 7375 6767 6573 7465  y...**A suggeste
+00010b90: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
+00010ba0: 616e 2075 7365 2063 6f6d 6d61 6e64 2073  an use command s
+00010bb0: 696d 706c 7920 6173 2062 656c 6f77 2a2a  imply as below**
+00010bc0: 0a0a 6060 600a 7767 6420 7065 616b 206b  ..```.wgd peak k
+00010bd0: 7364 6174 6120 2d61 7020 6170 6461 7461  sdata -ap apdata
+00010be0: 202d 736d 2073 6d64 6174 6120 2d6c 6520   -sm smdata -le 
+00010bf0: 6c65 6461 7461 202d 6d70 206d 7064 6174  ledata -mp mpdat
+00010c00: 6120 2d2d 6865 7572 6973 7469 630a 6060  a --heuristic.``
+00010c10: 600a 0a23 2323 2077 6764 2073 796e 0a0a  `..### wgd syn..
+00010c20: 2a2a 5468 6520 696e 7472 612d 7370 6563  **The intra-spec
+00010c30: 6966 6963 2073 796e 7465 6e79 2069 6e66  ific synteny inf
+00010c40: 6572 656e 6365 2a2a 0a60 6060 0a77 6764  erence**.```.wgd
+00010c50: 2073 796e 2066 616d 696c 6965 7320 6766   syn families gf
+00010c60: 6620 282d 2d6b 735f 6469 7374 7269 6275  f (--ks_distribu
+00010c70: 7469 6f6e 206b 7364 6174 6120 2d66 2067  tion ksdata -f g
+00010c80: 656e 6520 2d61 2049 4420 2d2d 6d69 6e6c  ene -a ID --minl
+00010c90: 656e 202d 3120 2d2d 6d69 6e73 6567 6c65  en -1 --minsegle
+00010ca0: 6e20 3130 3030 3020 2d2d 6d69 6e67 656e  n 10000 --mingen
+00010cb0: 656e 756d 2033 3029 0a60 6060 0a0a 5468  enum 30).```..Th
+00010cc0: 6520 7072 6f67 7261 6d20 6077 6764 2073  e program `wgd s
+00010cd0: 796e 6020 6973 206d 6169 6e6c 7920 6465  yn` is mainly de
+00010ce0: 616c 696e 6720 7769 7468 2063 6f6c 6c69  aling with colli
+00010cf0: 6e65 6172 6974 7920 6f72 2073 796e 7465  nearity or synte
+00010d00: 6e79 2028 626f 7468 2072 6566 6572 7265  ny (both referre
+00010d10: 6420 746f 2061 7320 7379 6e74 656e 7920  d to as synteny 
+00010d20: 6865 7265 6166 7465 7229 2061 6e61 6c79  hereafter) analy
+00010d30: 7369 732e 2054 776f 2069 6e70 7574 2066  sis. Two input f
+00010d40: 696c 6573 2061 7265 2065 7373 656e 7469  iles are essenti
+00010d50: 616c 2c20 7468 6520 6765 6e65 2066 616d  al, the gene fam
+00010d60: 696c 7920 6669 6c65 2061 6e64 2074 6865  ily file and the
+00010d70: 2067 6666 3320 6669 6c65 2e20 5468 6520   gff3 file. The 
+00010d80: 6765 6e65 2066 616d 696c 7920 6669 6c65  gene family file
+00010d90: 2069 7320 696e 2074 6865 2066 6f72 6d61   is in the forma
+00010da0: 7420 6173 2060 4f72 7468 6f46 696e 6465  t as `OrthoFinde
+00010db0: 7260 2e20 5468 6520 736f 6674 7761 7265  r`. The software
+00010dc0: 2060 692d 6164 686f 7265 6020 6973 2061   `i-adhore` is a
+00010dd0: 2070 7265 7265 7175 6973 6974 652e 2057   prerequisite. W
+00010de0: 6974 6820 6465 6661 756c 7420 7061 7261  ith default para
+00010df0: 6d65 7465 7273 2c20 7468 6520 7072 6f67  meters, the prog
+00010e00: 7261 6d20 6261 7369 6361 6c6c 7920 636f  ram basically co
+00010e10: 6e64 7563 7473 2031 2920 6669 6c74 6572  nducts 1) filter
+00010e20: 696e 6720 6765 6e65 2066 616d 696c 6965  ing gene familie
+00010e30: 7320 6261 7365 6420 6f6e 206d 6178 696d  s based on maxim
+00010e40: 756d 2066 616d 696c 7920 7369 7a65 2032  um family size 2
+00010e50: 2920 7265 7472 6965 7669 6e67 2067 656e  ) retrieving gen
+00010e60: 6520 706f 7369 7469 6f6e 2061 6e64 2073  e position and s
+00010e70: 6361 6666 6f6c 6420 696e 666f 726d 6174  caffold informat
+00010e80: 696f 6e20 6672 6f6d 2067 6666 3320 6669  ion from gff3 fi
+00010e90: 6c65 2033 2920 7072 6f64 7563 696e 6720  le 3) producing 
+00010ea0: 7468 6520 636f 6e66 6967 7572 6174 696f  the configuratio
+00010eb0: 6e20 6669 6c65 2061 6e64 2061 7373 6f63  n file and assoc
+00010ec0: 6961 7465 6420 6461 7461 6669 6c65 7320  iated datafiles 
+00010ed0: 666f 7220 6069 2d61 6468 6f72 6560 2034  for `i-adhore` 4
+00010ee0: 2920 6361 6c6c 696e 6720 6069 2d61 6468  ) calling `i-adh
+00010ef0: 6f72 6560 2067 6976 656e 2074 6865 2070  ore` given the p
+00010f00: 6172 616d 6574 6572 7320 7365 7420 746f  arameters set to
+00010f10: 2069 6e66 6572 2073 796e 7465 6e79 2035   infer synteny 5
+00010f20: 2920 7669 7375 616c 697a 696e 6720 7468  ) visualizing th
+00010f30: 6520 7379 6e74 656e 7920 696e 2022 646f  e synteny in "do
+00010f40: 7470 6c6f 7422 2069 6e20 7468 6520 756e  tplot" in the un
+00010f50: 6974 206f 6620 6765 6e65 7320 616e 6420  it of genes and 
+00010f60: 6261 7365 732c 2069 6e20 2253 796e 6465  bases, in "Synde
+00010f70: 7074 6822 2070 6c6f 7420 7368 6f77 696e  pth" plot showin
+00010f80: 6720 7468 6520 6469 7374 7269 6275 7469  g the distributi
+00010f90: 6f6e 206f 6620 6469 6666 6572 656e 7420  on of different 
+00010fa0: 6361 7465 676f 7269 6573 206f 6620 636f  categories of co
+00010fb0: 6c6c 696e 6561 7269 7479 2072 6174 696f  llinearity ratio
+00010fc0: 7320 7769 7468 696e 2061 6e64 2062 6574  s within and bet
+00010fd0: 7765 656e 2073 7065 6369 6573 2c20 696e  ween species, in
+00010fe0: 2022 6475 7053 7461 636b 2220 706c 6f74   "dupStack" plot
+00010ff0: 2073 686f 7769 6e67 206d 756c 7469 706c   showing multipl
+00011000: 6963 6f6e 7320 7769 7468 2064 6966 6665  icons with diffe
+00011010: 7265 6e74 206d 756c 7469 706c 6963 6174  rent multiplicat
+00011020: 696f 6e20 6c65 7665 6c73 2e20 3629 2069  ion levels. 6) i
+00011030: 6620 7769 7468 202a 4b2a 3c73 7562 3e53  f with *K*<sub>S
+00011040: 3c2f 7375 623e 2064 6174 612c 2061 2022  </sub> data, a "
+00011050: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00011060: 646f 7470 6c6f 7422 2077 6974 6820 646f  dotplot" with do
+00011070: 7473 2061 6e6e 6f74 6174 6564 2069 6e20  ts annotated in 
+00011080: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00011090: 7661 6c75 6573 2061 6e64 2061 202a 4b2a  values and a *K*
+000110a0: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+000110b0: 7472 6962 7574 696f 6e20 7769 7468 2061  tribution with a
+000110c0: 6e63 686f 7220 7061 6972 7320 6465 6e6f  nchor pairs deno
+000110d0: 7465 6420 7769 6c6c 2062 6520 7072 6f64  ted will be prod
+000110e0: 7563 6564 2e20 5468 6520 6765 6e65 2069  uced. The gene i
+000110f0: 6e66 6f72 6d61 7469 6f6e 2069 6e20 7468  nformation in th
+00011100: 6520 6765 6e65 2066 616d 696c 7920 6669  e gene family fi
+00011110: 6c65 2061 6e64 2067 6666 3320 6669 6c65  le and gff3 file
+00011120: 2073 686f 756c 6420 6265 206d 6174 6368   should be match
+00011130: 6564 2077 6869 6368 2072 6571 7569 7265  ed which require
+00011140: 7320 7573 6572 7320 746f 2073 6574 2070  s users to set p
+00011150: 726f 7065 7220 602d 2d66 6561 7475 7265  roper `--feature
+00011160: 6020 616e 6420 602d 2d61 7474 7269 6275  ` and `--attribu
+00011170: 7465 602e 2054 6865 206d 6178 696d 756d  te`. The maximum
+00011180: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
+00011190: 6265 2069 6e63 6c75 6465 6420 6361 6e20  be included can 
+000111a0: 6265 2073 6574 2076 6961 2074 6865 206f  be set via the o
+000111b0: 7074 696f 6e20 602d 2d6d 6178 7369 7a65  ption `--maxsize
+000111c0: 602c 206e 6f74 6564 2074 6861 7420 7468  `, noted that th
+000111d0: 6973 2066 696c 7465 7269 6e67 2069 7320  is filtering is 
+000111e0: 6d61 696e 6c79 2074 6f20 6472 6f70 2074  mainly to drop t
+000111f0: 686f 7365 2068 7567 6520 7461 6e64 656d  hose huge tandem
+00011200: 2064 7570 6c69 6361 7465 7320 6661 6d69   duplicates fami
+00011210: 6c79 2061 6e64 2074 7261 6e73 706f 7361  ly and transposa
+00011220: 626c 6520 656c 656d 656e 7473 2028 5445  ble elements (TE
+00011230: 7329 2066 616d 696c 792c 2061 6e64 206e  s) family, and n
+00011240: 6f74 206d 616e 6461 746f 7279 2e20 5573  ot mandatory. Us
+00011250: 6572 7320 6361 6e20 6669 6c74 6572 2074  ers can filter t
+00011260: 686f 7365 2066 7261 676d 656e 7461 7279  hose fragmentary
+00011270: 2073 6361 6666 6f6c 6473 2076 6961 2074   scaffolds via t
+00011280: 6865 206f 7074 696f 6e20 602d 2d6d 696e  he option `--min
+00011290: 6c65 6e60 2e20 5468 6520 6d69 6e69 6d75  len`. The minimu
+000112a0: 6d20 6c65 6e67 7468 2061 6e64 206e 756d  m length and num
+000112b0: 6265 7220 6f66 2067 656e 6573 2066 6f72  ber of genes for
+000112c0: 2061 2073 6567 6d65 6e74 2074 6f20 6265   a segment to be
+000112d0: 2063 6f6e 7369 6465 7265 6420 6361 6e20   considered can 
+000112e0: 6265 2073 6574 2076 6961 2074 6865 206f  be set via the o
+000112f0: 7074 696f 6e20 602d 2d6d 696e 7365 676c  ption `--minsegl
+00011300: 656e 6020 616e 6420 602d 2d6d 696e 6765  en` and `--minge
+00011310: 6e65 6e75 6d60 2e20 5265 6475 6e64 616e  nenum`. Redundan
+00011320: 7420 6d75 6c74 6970 6c69 636f 6e73 2063  t multiplicons c
+00011330: 616e 2062 6520 6b65 7074 2062 7920 7365  an be kept by se
+00011340: 7420 7468 6520 666c 6167 206f 7074 696f  t the flag optio
+00011350: 6e20 602d 2d6b 6565 7072 6564 756e 602e  n `--keepredun`.
+00011360: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+00011370: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+00011380: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+00011390: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+000113a0: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+000113b0: 6c69 6573 2067 6666 0a60 6060 0a0a 2a2a  lies gff.```..**
+000113c0: 5468 6520 696e 7465 722d 7370 6563 6966  The inter-specif
+000113d0: 6963 2073 796e 7465 6e79 2069 6e66 6572  ic synteny infer
+000113e0: 656e 6365 2a2a 0a60 6060 0a77 6764 2073  ence**.```.wgd s
+000113f0: 796e 2066 616d 696c 6965 7320 6766 6631  yn families gff1
+00011400: 2067 6666 3220 282d 2d61 6464 6974 696f   gff2 (--additio
+00011410: 6e61 6c67 6666 696e 666f 2022 6d52 4e41  nalgffinfo "mRNA
+00011420: 3b4e 616d 6522 202d 2d61 6464 6974 696f  ;Name" --additio
+00011430: 6e61 6c67 6666 696e 666f 2022 6765 6e65  nalgffinfo "gene
+00011440: 3b49 4422 290a 6060 600a 0a46 6f72 206d  ;ID").```..For m
+00011450: 756c 7469 2d73 7065 6369 6573 2073 796e  ulti-species syn
+00011460: 7465 6e79 2069 6e66 6572 656e 6365 2c20  teny inference, 
+00011470: 6966 2075 7365 7273 2068 6176 6520 6766  if users have gf
+00011480: 6633 2066 696c 6573 2077 6869 6368 2068  f3 files which h
+00011490: 6176 6520 6469 6666 6572 656e 7420 6665  ave different fe
+000114a0: 6174 7572 6573 206f 7220 6174 7472 6962  atures or attrib
+000114b0: 7574 6573 2066 6f72 2067 656e 6520 706f  utes for gene po
+000114c0: 7369 7469 6f6e 2069 6e66 6f72 6d61 7469  sition informati
+000114d0: 6f6e 2072 6574 7269 6576 616c 2c20 7468  on retrieval, th
+000114e0: 6520 6f70 7469 6f6e 2060 2d2d 6164 6469  e option `--addi
+000114f0: 7469 6f6e 616c 6766 6669 6e66 6f60 2063  tionalgffinfo` c
+00011500: 616e 2062 6520 7365 7420 746f 2070 726f  an be set to pro
+00011510: 7669 6465 2074 6865 2061 6464 6974 696f  vide the additio
+00011520: 6e61 6c20 696e 666f 726d 6174 696f 6e2e  nal information.
+00011530: 2054 6865 2072 656d 6169 6e69 6e67 2070   The remaining p
+00011540: 6172 616d 6574 6572 2073 6574 7469 6e67  arameter setting
+00011550: 2069 7320 7468 6520 7361 6d65 2061 7320   is the same as 
+00011560: 7468 6520 696e 7472 612d 7370 6563 6966  the intra-specif
+00011570: 6963 2073 796e 7465 6e79 2069 6e66 6572  ic synteny infer
+00011580: 656e 6365 2e0a 0a2a 2a41 2073 7567 6765  ence...**A sugge
+00011590: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
+000115a0: 6e20 6361 6e20 7573 6520 636f 6d6d 616e  n can use comman
+000115b0: 6420 7369 6d70 6c79 2061 7320 6265 6c6f  d simply as belo
+000115c0: 772a 2a0a 0a60 6060 0a77 6764 2073 796e  w**..```.wgd syn
+000115d0: 2066 616d 696c 6965 7320 6766 6631 2067   families gff1 g
+000115e0: 6666 320a 6060 600a 0a23 2323 2077 6764  ff2.```..### wgd
+000115f0: 2076 697a 0a0a 2a2a 5468 6520 7669 7375   viz..**The visu
+00011600: 616c 697a 6174 696f 6e20 6f66 202a 4b2a  alization of *K*
+00011610: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00011620: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+00011630: 6420 454c 4d4d 2061 6e61 6c79 7369 732a  d ELMM analysis*
+00011640: 2a0a 6060 600a 7767 6420 7669 7a20 2d64  *.```.wgd viz -d
+00011650: 206b 7364 6174 610a 6060 600a 0a54 6865   ksdata.```..The
+00011660: 2070 726f 6772 616d 2060 7767 6420 7669   program `wgd vi
+00011670: 7a60 2069 7320 6d61 696e 6c79 2066 6f72  z` is mainly for
+00011680: 2074 6865 2070 7572 706f 7365 206f 6620   the purpose of 
+00011690: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000116a0: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+000116b0: 2073 796e 7465 6e79 2076 6973 7561 6c69   synteny visuali
+000116c0: 7a61 7469 6f6e 2c20 7769 7468 2073 6f6d  zation, with som
+000116d0: 6520 6f70 7469 6f6e 616c 206d 6978 7475  e optional mixtu
+000116e0: 7265 206d 6f64 656c 696e 6720 616e 616c  re modeling anal
+000116f0: 7973 6973 2e20 5468 6520 6261 7369 6320  ysis. The basic 
+00011700: 6675 6e63 7469 6f6e 2069 7320 6a75 7374  function is just
+00011710: 2074 6f20 706c 6f74 2074 6865 202a 4b2a   to plot the *K*
+00011720: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+00011730: 7472 6962 7574 696f 6e20 616e 6420 636f  tribution and co
+00011740: 6e64 7563 7420 616e 2045 4c4d 4d20 616e  nduct an ELMM an
+00011750: 616c 7973 6973 2069 6e20 7365 6172 6368  alysis in search
+00011760: 206f 6620 706f 7465 6e74 6961 6c20 5747   of potential WG
+00011770: 4420 636f 6d70 6f6e 656e 7473 2e20 536f  D components. So
+00011780: 6d65 206b 6579 2070 6172 616d 6574 6572  me key parameter
+00011790: 7320 6166 6665 6374 696e 6720 7468 6520  s affecting the 
+000117a0: 454c 4d4d 2072 6573 756c 7420 696e 636c  ELMM result incl
+000117b0: 7564 6520 602d 2d70 726f 6d69 6e65 6e63  ude `--prominenc
+000117c0: 655f 6375 746f 6666 602c 2060 2d2d 7265  e_cutoff`, `--re
+000117d0: 6c5f 6865 6967 6874 602c 2060 2d2d 656d  l_height`, `--em
+000117e0: 5f69 7465 7261 7469 6f6e 7360 2061 6e64  _iterations` and
+000117f0: 2060 2d2d 656d 5f69 6e69 7469 616c 697a   `--em_initializ
+00011800: 6174 696f 6e73 602c 2061 6c6c 206f 6620  ations`, all of 
+00011810: 7768 6963 6820 6861 7665 2062 6565 6e20  which have been 
+00011820: 6578 706c 6169 6e65 6420 6168 6561 642e  explained ahead.
+00011830: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+00011840: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+00011850: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+00011860: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+00011870: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
+00011880: 7364 6174 610a 6060 600a 0a2a 2a54 6865  sdata.```..**The
+00011890: 2076 6973 7561 6c69 7a61 7469 6f6e 206f   visualization o
+000118a0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+000118b0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+000118c0: 6f6e 2077 6974 6820 7261 7465 2063 6f72  on with rate cor
+000118d0: 7265 6374 696f 6e2a 2a0a 6060 600a 7767  rection**.```.wg
+000118e0: 6420 7669 7a20 2d64 206b 7364 6174 6120  d viz -d ksdata 
+000118f0: 2d73 7020 7370 6461 7461 202d 2d66 6f63  -sp spdata --foc
+00011900: 7573 3261 6c6c 2066 6f63 616c 5f73 7065  us2all focal_spe
+00011910: 6369 6573 202d 2d65 7874 7261 7061 7261  cies --extrapara
+00011920: 6e6f 6d65 6b73 206b 7364 6174 6120 282d  nomeks ksdata (-
+00011930: 2d61 6e63 686f 7270 6f69 6e74 7320 6170  -anchorpoints ap
+00011940: 6461 7461 202d 2d70 6c6f 7461 7067 6d6d  data --plotapgmm
+00011950: 202d 2d70 6c6f 7465 6c6d 6d29 0a60 6060   --plotelmm).```
+00011960: 0a0a 4265 7369 6465 7320 7468 6520 6261  ..Besides the ba
+00011970: 7369 6320 2a4b 2a3c 7375 623e 533c 2f73  sic *K*<sub>S</s
+00011980: 7562 3e20 706c 6f74 2c20 7375 6273 7469  ub> plot, substi
+00011990: 7475 7469 6f6e 2072 6174 6520 636f 7272  tution rate corr
+000119a0: 6563 7469 6f6e 2063 616e 2061 6c73 6f20  ection can also 
+000119b0: 6265 2061 6368 6965 7665 6420 6769 7665  be achieved give
+000119c0: 6e20 6174 206c 6561 7374 2061 2073 7065  n at least a spe
+000119d0: 6369 6573 2074 7265 6520 2876 6961 2074  cies tree (via t
+000119e0: 6865 206f 7074 696f 6e20 602d 2d73 7065  he option `--spe
+000119f0: 6369 6573 7472 6565 6029 2061 6e64 2061  ciestree`) and a
+00011a00: 2066 6f63 616c 2073 7065 6369 6573 2028   focal species (
+00011a10: 6569 7468 6572 2076 6961 2074 6865 206f  either via the o
+00011a20: 7074 696f 6e20 602d 2d66 6f63 7573 3261  ption `--focus2a
+00011a30: 6c6c 6020 6f72 2076 6961 2074 6865 206f  ll` or via the o
+00011a40: 7074 696f 6e20 602d 2d73 7061 6972 6020  ption `--spair` 
+00011a50: 696e 2074 6865 2066 6f72 6d20 6f66 2022  in the form of "
+00011a60: 2466 6f63 616c 5f73 7065 6369 6573 3b24  $focal_species;$
+00011a70: 666f 6361 6c5f 7370 6563 6965 7322 292e  focal_species").
+00011a80: 2049 7427 7320 7375 6767 6573 7465 6420   It's suggested 
+00011a90: 7468 6174 2074 6865 206f 7274 686f 6c6f  that the ortholo
+00011aa0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+00011ab0: 7375 623e 2064 6174 6120 6973 2070 726f  sub> data is pro
+00011ac0: 7669 6465 6420 6279 2074 6865 2060 2d2d  vided by the `--
+00011ad0: 6461 7461 6669 6c65 6020 6f70 7469 6f6e  datafile` option
+00011ae0: 2061 6e64 2074 6865 2070 6172 616c 6f67   and the paralog
+00011af0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00011b00: 7562 3e20 6461 7461 2069 7320 7072 6f76  ub> data is prov
+00011b10: 6964 6564 2062 7920 7468 6520 602d 2d65  ided by the `--e
+00011b20: 7874 7261 7061 7261 6e6f 6d65 6b73 6020  xtraparanomeks` 
+00011b30: 6f70 7469 6f6e 2c20 616c 7468 6f75 6768  option, although
+00011b40: 2069 7427 7320 616c 6c6f 7765 6420 746f   it's allowed to
+00011b50: 206f 6e6c 7920 7072 6f76 6964 6520 2a4b   only provide *K
+00011b60: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+00011b70: 7461 2076 6961 2074 6865 2060 2d2d 6461  ta via the `--da
+00011b80: 7461 6669 6c65 6020 6f70 7469 6f6e 2061  tafile` option a
+00011b90: 6e64 2064 6570 6f73 6974 2062 6f74 6820  nd deposit both 
+00011ba0: 6f72 7468 6f6c 6f67 6f75 7326 7061 7261  orthologous&para
+00011bb0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00011bc0: 3c2f 7375 623e 2064 6174 6120 7468 6572  </sub> data ther
+00011bd0: 656f 6e2e 2054 6865 7265 2061 7265 2074  eon. There are t
+00011be0: 776f 2074 7970 6573 206f 6620 6d69 7865  wo types of mixe
+00011bf0: 6420 706c 6f74 7320 2874 6865 2022 6d69  d plots (the "mi
+00011c00: 7865 6422 2068 6572 6520 7265 6665 7273  xed" here refers
+00011c10: 2074 6f20 6d69 7865 6420 6f72 7468 6f6c   to mixed orthol
+00011c20: 6f67 6f75 7320 616e 6420 7061 7261 6c6f  ogous and paralo
+00011c30: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+00011c40: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
+00011c50: 6e73 292c 206f 6e65 206f 6620 7768 6963  ns), one of whic
+00011c60: 6820 6973 2073 696d 696c 6172 2074 6f20  h is similar to 
+00011c70: 7768 6174 2060 6b73 7261 7465 7360 2070  what `ksrates` p
+00011c80: 6c6f 7473 2c20 7768 696c 6520 7468 6520  lots, while the 
+00011c90: 6f74 6865 7220 6f66 2077 6869 6368 2069  other of which i
+00011ca0: 7320 6c69 6b65 2074 6865 2063 6f6e 7665  s like the conve
+00011cb0: 6e74 696f 6e61 6c20 2a4b 2a3c 7375 623e  ntional *K*<sub>
+00011cc0: 533c 2f73 7562 3e20 706c 6f74 2074 6861  S</sub> plot tha
+00011cd0: 7420 626f 7468 2074 6865 206f 7269 6769  t both the origi
+00011ce0: 6e61 6c20 6f72 7468 6f6c 6f67 6f75 7320  nal orthologous 
+00011cf0: 616e 6420 7061 7261 6c6f 676f 7573 202a  and paralogous *
+00011d00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00011d10: 6973 7472 6962 7574 696f 6e73 2061 7265  istributions are
+00011d20: 2074 7275 7468 6675 6c6c 7920 706c 6f74   truthfully plot
+00011d30: 7465 6420 696e 7374 6561 6420 6f66 206a  ted instead of j
+00011d40: 7573 7420 6265 696e 6720 7265 7072 6573  ust being repres
+00011d50: 656e 7465 6420 6279 2073 6f6d 6520 7665  ented by some ve
+00011d60: 7274 6963 616c 206c 696e 6573 2e20 5765  rtical lines. We
+00011d70: 2073 7567 6765 7374 2075 7365 7273 2074   suggest users t
+00011d80: 6f20 6164 6f70 7420 7468 6520 606b 7372  o adopt the `ksr
+00011d90: 6174 6573 602d 6c69 6b65 2070 6c6f 7473  ates`-like plots
+00011da0: 2c20 7768 6963 6820 6973 2074 6865 2064  , which is the d
+00011db0: 6566 6175 6c74 2e20 4f74 6865 7277 6973  efault. Otherwis
+00011dc0: 6520 7468 6520 6f70 7469 6f6e 2060 2d2d  e the option `--
+00011dd0: 636c 6173 7369 6360 2077 696c 6c20 6265  classic` will be
+00011de0: 206e 6565 6465 6420 746f 2073 6574 2e20   needed to set. 
+00011df0: 4578 7472 6120 6d69 7874 7572 6520 6d6f  Extra mixture mo
+00011e00: 6465 6c69 6e67 2061 6e61 6c79 7369 7320  deling analysis 
+00011e10: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
+00011e20: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
+00011e30: 602d 2d70 6c6f 7465 6c6d 6d60 2061 6e64  `--plotelmm` and
+00011e40: 2060 2d2d 706c 6f74 6170 676d 6d60 2077   `--plotapgmm` w
+00011e50: 6974 6820 7468 6520 616e 6368 6f72 2070  ith the anchor p
+00011e60: 6f69 6e74 7320 6461 7461 6669 6c65 2070  oints datafile p
+00011e70: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
+00011e80: 2d2d 616e 6368 6f72 706f 696e 7473 6020  --anchorpoints` 
+00011e90: 6f70 7469 6f6e 2e0a 0a2a 2a41 2073 7567  option...**A sug
+00011ea0: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00011eb0: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+00011ec0: 616e 6420 7369 6d70 6c79 2061 7320 6265  and simply as be
+00011ed0: 6c6f 772a 2a0a 0a60 6060 0a77 6764 2076  low**..```.wgd v
+00011ee0: 697a 202d 6420 6b73 6461 7461 202d 7370  iz -d ksdata -sp
+00011ef0: 2073 7064 6174 6120 2d2d 666f 6375 7332   spdata --focus2
+00011f00: 616c 6c20 666f 6361 6c5f 7370 6563 6965  all focal_specie
+00011f10: 7320 2d2d 6578 7472 6170 6172 616e 6f6d  s --extraparanom
+00011f20: 656b 7320 6b73 6461 7461 0a60 6060 0a0a  eks ksdata.```..
+00011f30: 2a2a 5468 6520 7669 7375 616c 697a 6174  **The visualizat
+00011f40: 696f 6e20 6f66 2073 796e 7465 6e79 2a2a  ion of synteny**
+00011f50: 0a60 6060 0a77 6764 2076 697a 202d 6170  .```.wgd viz -ap
+00011f60: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
+00011f70: 7461 202d 6d74 206d 7464 6174 6120 2d67  ta -mt mtdata -g
+00011f80: 7420 6774 6461 7461 202d 2d70 6c6f 7473  t gtdata --plots
+00011f90: 796e 2028 2d2d 6d69 6e6c 656e 202d 3120  yn (--minlen -1 
+00011fa0: 2d2d 6d69 6e73 6567 6c65 6e20 3130 3030  --minseglen 1000
+00011fb0: 3020 2d2d 6d69 6e67 656e 656e 756d 2033  0 --mingenenum 3
+00011fc0: 3029 0a60 6060 0a0a 436f 6d70 6172 6564  0).```..Compared
+00011fd0: 2074 6f20 7468 6520 6f72 6967 696e 616c   to the original
+00011fe0: 2060 7767 6460 2c20 7468 6520 6077 6764   `wgd`, the `wgd
+00011ff0: 2076 697a 6020 7072 6f67 7261 6d20 6164   viz` program ad
+00012000: 6465 6420 7379 6e74 656e 7920 7669 7375  ded synteny visu
+00012010: 616c 697a 6174 696f 6e20 7069 7065 6c69  alization pipeli
+00012020: 6e65 2e20 5573 6572 7320 6e65 6564 2074  ne. Users need t
+00012030: 6f20 7072 6f76 6964 6520 7468 6520 666c  o provide the fl
+00012040: 6167 206f 7074 696f 6e20 602d 2d70 6c6f  ag option `--plo
+00012050: 7473 796e 6020 746f 2069 6e69 7469 6174  tsyn` to initiat
+00012060: 6520 7468 6973 2070 6172 7420 6f66 2070  e this part of p
+00012070: 6970 656c 696e 652e 2054 6869 7320 7374  ipeline. This st
+00012080: 6570 2061 7373 756d 6573 2074 6861 7420  ep assumes that 
+00012090: 7573 6572 7320 6861 7665 206f 6274 6169  users have obtai
+000120a0: 6e65 6420 616c 7265 6164 7920 7468 6520  ned already the 
+000120b0: 7379 6e74 656e 6963 2072 6573 756c 7473  syntenic results
+000120c0: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
+000120d0: 2061 6e64 2075 7365 7320 7468 6f73 6520   and uses those 
+000120e0: 7265 7375 6c74 2066 696c 6573 2074 6f20  result files to 
+000120f0: 7265 616c 697a 6520 7468 6520 7379 6e74  realize the synt
+00012100: 656e 7920 7669 7375 616c 697a 6174 696f  eny visualizatio
+00012110: 6e2e 2053 696d 6c69 6172 2074 6f20 6077  n. Simliar to `w
+00012120: 6764 2073 796e 602c 2074 6865 2065 7874  gd syn`, the ext
+00012130: 7261 202a 4b2a 3c73 7562 3e53 3c2f 7375  ra *K*<sub>S</su
+00012140: 623e 2064 6174 6120 6669 6c65 2063 616e  b> data file can
+00012150: 2062 6520 7472 616e 736d 6974 7465 6420   be transmitted 
+00012160: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00012170: 2d2d 6461 7461 6669 6c65 6020 2869 6e73  --datafile` (ins
+00012180: 7465 6164 206f 6620 602d 2d65 7874 7261  tead of `--extra
+00012190: 7061 7261 6e6f 6d65 6b73 6020 6f70 7469  paranomeks` opti
+000121a0: 6f6e 292e 2042 6173 6963 616c 6c79 2c20  on). Basically, 
+000121b0: 7468 6520 7379 6e74 656e 6963 2072 6573  the syntenic res
+000121c0: 756c 7420 6669 6c65 7320 7265 7175 6972  ult files requir
+000121d0: 6564 2061 7265 2074 6865 2061 6e63 686f  ed are the ancho
+000121e0: 7220 706f 696e 7473 2064 6174 6166 696c  r points datafil
+000121f0: 652c 206d 756c 7469 706c 6963 6f6e 7320  e, multiplicons 
+00012200: 6461 7461 6669 6c65 2c20 6765 6e65 2074  datafile, gene t
+00012210: 6162 6c65 2064 6174 6166 696c 6520 2861  able datafile (a
+00012220: 7574 6f6d 6174 656c 7920 7072 6f64 7563  utomately produc
+00012230: 6564 2062 7920 6077 6764 2073 796e 6029  ed by `wgd syn`)
+00012240: 2c20 616e 6420 7365 676d 656e 7473 2064  , and segments d
+00012250: 6174 6166 696c 652e 0a0a 2a2a 4120 7375  atafile...**A su
+00012260: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+00012270: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+00012280: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
+00012290: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
+000122a0: 7669 7a20 2d61 7020 6170 6461 7461 202d  viz -ap apdata -
+000122b0: 736d 2073 6d64 6174 6120 2d6d 7420 6d74  sm smdata -mt mt
+000122c0: 6461 7461 202d 6774 2067 7464 6174 6120  data -gt gtdata 
+000122d0: 2d2d 706c 6f74 7379 6e0a 6060 600a 0a23  --plotsyn.```..#
+000122e0: 2320 496c 6c75 7374 7261 7469 6f6e 0a0a  # Illustration..
+000122f0: 5765 2069 6c6c 7573 7472 6174 6520 6f75  We illustrate ou
+00012300: 7220 7072 6f67 7261 6d20 6f6e 2061 6e20  r program on an 
+00012310: 6578 656d 706c 6172 7920 5747 4420 696e  exemplary WGD in
+00012320: 6665 7265 6e63 6520 616e 6420 6461 7469  ference and dati
+00012330: 6e67 2075 706f 6e20 7370 6563 6965 7320  ng upon species 
+00012340: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
+00012350: 6c65 612a 2e0a 0a54 6865 202a 4171 7569  lea*...The *Aqui
+00012360: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
+00012370: 7761 7320 7265 706f 7274 6564 2074 6f20  was reported to 
+00012380: 6578 7065 7269 656e 6365 2061 6e20 7061  experience an pa
+00012390: 6c65 6f2d 706f 6c79 706c 6f69 6469 7a61  leo-polyploidiza
+000123a0: 7469 6f6e 2065 7665 6e74 2061 6674 6572  tion event after
+000123b0: 2074 6865 2064 6976 6572 6765 6e63 6520   the divergence 
+000123c0: 6f66 2063 6f72 6520 6575 6469 636f 7473  of core eudicots
+000123d0: 2c20 7768 6963 6820 6973 206c 696b 656c  , which is likel
+000123e0: 7920 7368 6172 6564 2062 7920 616c 6c20  y shared by all 
+000123f0: 5261 6e75 6e63 756c 616c 6573 2e0a 0a46  Ranunculales...F
+00012400: 6972 7374 2061 626f 7665 2061 6c6c 2c20  irst above all, 
+00012410: 6c65 7427 7320 6465 6c69 6e65 6174 6520  let's delineate 
+00012420: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
+00012430: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+00012440: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+00012450: 696f 6e20 616e 6420 6861 7665 2061 2062  ion and have a b
+00012460: 6173 6963 206f 6273 6572 7661 7469 6f6e  asic observation
+00012470: 2066 6f72 2070 6f74 656e 7469 616c 6c79   for potentially
+00012480: 2063 6f6e 6365 6976 6162 6c65 2057 4744   conceivable WGD
+00012490: 732c 2075 7369 6e67 2074 6865 2063 6f6d  s, using the com
+000124a0: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
+000124b0: 0a0a 6060 600a 7767 6420 646d 6420 4171  ..```.wgd dmd Aq
+000124c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000124d0: 0a77 6764 206b 7364 2077 6764 5f64 6d64  .wgd ksd wgd_dmd
+000124e0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000124f0: 6c65 612e 7473 7620 4171 7569 6c65 6769  lea.tsv Aquilegi
+00012500: 615f 636f 6572 756c 6561 0a60 6060 0a0a  a_coerulea.```..
+00012510: 5468 6520 636f 6e73 7472 7563 7465 6420  The constructed 
+00012520: 7768 6f6c 6520 7061 7261 6e6f 6d65 202a  whole paranome *
+00012530: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00012540: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00012550: 6f66 202a 4171 7569 6c65 6769 6120 636f  of *Aquilegia co
+00012560: 6572 756c 6561 2a20 6973 2061 7320 6265  erulea* is as be
+00012570: 6c6f 772c 2077 6520 6361 6e20 7365 6520  low, we can see 
+00012580: 7468 6174 2074 6865 7265 2073 6565 6d73  that there seems
+00012590: 2074 6f20 6265 2061 2068 756d 7020 6174   to be a hump at
+000125a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000125b0: 2031 2062 7574 206e 6f74 2063 6c65 6172   1 but not clear
+000125c0: 2e0a 0a21 5b5d 2864 6174 612f 4171 7569  ...![](data/Aqui
+000125d0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000125e0: 7376 2e6b 7364 5f77 702e 7376 6729 0a0a  sv.ksd_wp.svg)..
+000125f0: 5765 2074 6865 6e20 636f 6e73 7472 7563  We then construc
+00012600: 7420 7468 6520 616e 6368 6f72 202a 4b2a  t the anchor *K*
+00012610: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00012620: 2064 6973 7472 6962 7574 696f 6e20 7573   distribution us
+00012630: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00012640: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
+00012650: 0a77 6764 2073 796e 202d 6620 6d52 4e41  .wgd syn -f mRNA
+00012660: 202d 6120 4e61 6d65 2077 6764 5f64 6d64   -a Name wgd_dmd
+00012670: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012680: 6c65 612e 7473 7620 4171 7569 6c65 6769  lea.tsv Aquilegi
+00012690: 615f 636f 6572 756c 6561 2e67 6666 3320  a_coerulea.gff3 
+000126a0: 2d6b 7320 7767 645f 6b73 642f 4171 7569  -ks wgd_ksd/Aqui
+000126b0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000126c0: 7376 2e6b 732e 7473 760a 6060 600a 0a41  sv.ks.tsv.```..A
+000126d0: 7320 7368 6f77 6e20 6265 6c6f 772c 2074  s shown below, t
+000126e0: 6865 7265 2061 7265 2073 6f6d 6520 7265  here are some re
+000126f0: 7461 696e 6564 2061 6e63 686f 7220 7061  tained anchor pa
+00012700: 6972 7320 7769 7468 202a 4b2a 3c73 7562  irs with *K*<sub
+00012710: 3e53 3c2f 7375 623e 2062 6574 7765 656e  >S</sub> between
+00012720: 2031 2061 6e64 2032 2c20 7768 6963 6820   1 and 2, which 
+00012730: 7365 656d 7320 746f 2073 7567 6765 7374  seems to suggest
+00012740: 2061 2057 4744 2065 7665 6e74 2e0a 0a21   a WGD event...!
+00012750: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
+00012760: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00012770: 7364 5f77 705f 6170 2e73 7667 290a 0a54  sd_wp_ap.svg)..T
+00012780: 6865 2061 7373 6f63 6961 7465 6420 6064  he associated `d
+00012790: 7570 5374 6163 6b60 2070 6c6f 7420 7368  upStack` plot sh
+000127a0: 6f77 7320 7468 6174 2074 6865 7265 2061  ows that there a
+000127b0: 7265 206e 756d 6572 6f75 7320 6475 706c  re numerous dupl
+000127c0: 6963 6174 6564 2073 6567 6d65 6e74 7320  icated segments 
+000127d0: 6163 726f 7373 206d 6f73 7420 6f66 2074  across most of t
+000127e0: 6865 2063 6872 6f6d 6f73 6f6d 6573 2e0a  he chromosomes..
+000127f0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
+00012800: 6769 615f 636f 6572 756c 6561 5f41 7175  gia_coerulea_Aqu
+00012810: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00012820: 6d75 6c74 6970 6c69 636f 6e73 5f6c 6576  multiplicons_lev
+00012830: 656c 2e73 7667 290a 0a57 6520 696d 706c  el.svg)..We impl
+00012840: 656d 656e 7465 6420 7477 6f20 7479 7065  emented two type
+00012850: 7320 6f66 2064 6f74 2070 6c6f 7473 2069  s of dot plots i
+00012860: 6e20 6f78 666f 7264 2067 7269 643a 206f  n oxford grid: o
+00012870: 6e65 2069 6e20 7468 6520 756e 6974 206f  ne in the unit o
+00012880: 6620 6261 7365 7320 616e 6420 7468 6520  f bases and the 
+00012890: 6f74 6865 7220 696e 2074 6865 2075 6e69  other in the uni
+000128a0: 7420 6f66 2067 656e 6573 2c20 7768 6963  t of genes, whic
+000128b0: 6820 6361 6e20 6265 2063 6f6c 6f72 6564  h can be colored
+000128c0: 2062 7920 2a4b 2a3c 7375 623e 533c 2f73   by *K*<sub>S</s
+000128d0: 7562 3e20 7661 6c75 6573 2067 6976 656e  ub> values given
+000128e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000128f0: 2064 6174 612e 0a0a 215b 5d28 6461 7461   data...![](data
+00012900: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012910: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+00012920: 5f63 6f65 7275 6c65 615f 4b73 2e64 6f74  _coerulea_Ks.dot
+00012930: 5f75 6e69 745f 6765 6e65 2e70 6e67 290a  _unit_gene.png).
+00012940: 0a41 7320 7368 6f77 6e20 6162 6f76 652c  .As shown above,
+00012950: 2074 6865 2064 6f74 2070 6c6f 7420 696e   the dot plot in
+00012960: 2074 6865 2075 6e69 7420 6f66 2067 656e   the unit of gen
+00012970: 6573 2070 7265 7365 6e74 7320 6e75 6d65  es presents nume
+00012980: 726f 7573 2064 656e 7365 6c79 2061 6767  rous densely agg
+00012990: 7265 6761 7465 6420 286c 696e 652d 6c69  regated (line-li
+000129a0: 6b65 2920 616e 6368 6f72 2070 6f69 6e74  ke) anchor point
+000129b0: 7320 6174 206d 6f73 7420 6f66 2074 6865  s at most of the
+000129c0: 2063 6872 6f6d 6f73 6f6d 6573 2077 6974   chromosomes wit
+000129d0: 6820 636f 6e73 6973 7465 6e74 202a 4b2a  h consistent *K*
+000129e0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+000129f0: 2062 6574 7765 656e 2031 2061 6e64 2032   between 1 and 2
+00012a00: 2e20 5468 6520 646f 7420 706c 6f74 2069  . The dot plot i
+00012a10: 6e20 7468 6520 756e 6974 206f 6620 6261  n the unit of ba
+00012a20: 7365 7320 7368 6f77 7320 7468 6520 7361  ses shows the sa
+00012a30: 6d65 2070 6174 7465 726e 2c20 6173 206d  me pattern, as m
+00012a40: 616e 6966 6573 7465 6420 6265 6c6f 772e  anifested below.
+00012a50: 0a0a 215b 5d28 6461 7461 2f41 7175 696c  ..![](data/Aquil
+00012a60: 6567 6961 5f63 6f65 7275 6c65 612d 7673  egia_coerulea-vs
+00012a70: 2d41 7175 696c 6567 6961 5f63 6f65 7275  -Aquilegia_coeru
+00012a80: 6c65 615f 4b73 2e64 6f74 2e70 6e67 290a  lea_Ks.dot.png).
+00012a90: 0a54 6865 2064 6f74 2070 6c6f 7473 2077  .The dot plots w
+00012aa0: 6974 686f 7574 202a 4b2a 3c73 7562 3e53  ithout *K*<sub>S
+00012ab0: 3c2f 7375 623e 2061 6e6e 6f74 6174 696f  </sub> annotatio
+00012ac0: 6e20 7769 6c6c 2061 6c73 6f20 6265 2061  n will also be a
+00012ad0: 7574 6f6d 6174 656c 7920 7072 6f64 7563  utomately produc
+00012ae0: 6564 2c20 6173 2073 686f 776e 2062 656c  ed, as shown bel
+00012af0: 6f77 2e0a 0a21 5b5d 2864 6174 612f 4171  ow...![](data/Aq
+00012b00: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00012b10: 2d76 732d 4171 7569 6c65 6769 615f 636f  -vs-Aquilegia_co
+00012b20: 6572 756c 6561 2e64 6f74 5f75 6e69 745f  erulea.dot_unit_
+00012b30: 6765 6e65 2e70 6e67 290a 0a21 5b5d 2864  gene.png)..![](d
+00012b40: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
+00012b50: 6572 756c 6561 2d76 732d 4171 7569 6c65  erulea-vs-Aquile
+00012b60: 6769 615f 636f 6572 756c 6561 2e64 6f74  gia_coerulea.dot
+00012b70: 2e70 6e67 290a 0a4e 6f74 6520 7468 6174  .png)..Note that
+00012b80: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
+00012b90: 616e 6368 6f72 2064 6f74 7320 616e 6420  anchor dots and 
+00012ba0: 616c 6c20 686f 6d6f 6c6f 6720 646f 7473  all homolog dots
+00012bb0: 2063 616e 2062 6520 7365 7420 6279 2074   can be set by t
+00012bc0: 6865 206f 7074 696f 6e20 6061 7061 6c70  he option `apalp
+00012bd0: 6861 6020 616e 6420 6068 6f61 6c70 6861  ha` and `hoalpha
+00012be0: 6020 7365 7061 7261 7465 6c79 2e20 4966  ` separately. If
+00012bf0: 206f 6e65 206a 7573 7420 7761 6e74 7320   one just wants 
+00012c00: 746f 2073 6565 2074 6865 2061 6e63 686f  to see the ancho
+00012c10: 7220 646f 7473 2c20 7365 7474 696e 6720  r dots, setting 
+00012c20: 7468 6520 6068 6f61 6c70 6861 6020 6173  the `hoalpha` as
+00012c30: 2030 2028 6f72 206f 7468 6572 206d 696e   0 (or other min
+00012c40: 7573 6375 6c65 2076 616c 7565 7329 2077  uscule values) w
+00012c50: 696c 6c20 646f 2e20 4966 206f 6e65 2077  ill do. If one w
+00012c60: 616e 7473 2074 6f20 7365 6520 7468 6520  ants to see the 
+00012c70: 6469 7374 7269 6275 7469 6f6e 206f 6620  distribution of 
+00012c80: 7768 6f6c 6520 646f 7473 2062 6574 7465  whole dots bette
+00012c90: 722c 2073 6574 7469 6e67 2074 6865 2060  r, setting the `
+00012ca0: 686f 616c 7068 6160 2068 6967 6865 7220  hoalpha` higher 
+00012cb0: 2861 6e64 2060 6170 616c 7068 6160 206c  (and `apalpha` l
+00012cc0: 6f77 6572 2920 7769 6c6c 2064 6f2e 2054  ower) will do. T
+00012cd0: 6865 2060 646f 7473 697a 6560 206f 7074  he `dotsize` opt
+00012ce0: 696f 6e20 6361 6e20 6265 2063 616c 6c65  ion can be calle
+00012cf0: 6420 746f 2061 646a 7573 7420 7468 6520  d to adjust the 
+00012d00: 7369 7a65 206f 6620 646f 7473 2e0a 0a41  size of dots...A
+00012d10: 2066 7572 7468 6572 2061 7373 6f63 6961   further associa
+00012d20: 7465 6420 5379 6e64 6570 7468 2070 6c6f  ted Syndepth plo
+00012d30: 7420 7368 6f77 7320 7468 6174 2074 6865  t shows that the
+00012d40: 7265 2061 7265 206d 6f72 6520 7468 616e  re are more than
+00012d50: 2038 3020 6475 706c 6963 6174 6564 2073   80 duplicated s
+00012d60: 6567 6d65 6e74 732c 2077 6869 6368 2064  egments, which d
+00012d70: 6f6d 696e 6174 6573 2074 6865 2077 686f  ominates the who
+00012d80: 6c65 2063 6f6c 6c69 6e65 6172 2072 6174  le collinear rat
+00012d90: 696f 2063 6174 6567 6f72 792e 0a0a 215b  io category...![
+00012da0: 5d28 6461 7461 2f53 796e 6465 7074 682e  ](data/Syndepth.
+00012db0: 7376 6729 0a0a 5765 2063 616e 2066 6974  svg)..We can fit
+00012dc0: 2061 6e20 454c 4d4d 206d 6978 7475 7265   an ELMM mixture
+00012dd0: 206d 6f64 656c 2075 706f 6e20 7468 6520   model upon the 
+00012de0: 7768 6f6c 6520 7061 7261 6e6f 6d65 202a  whole paranome *
+00012df0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00012e00: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00012e10: 746f 2073 6565 206d 6f72 6520 6163 6375  to see more accu
+00012e20: 7261 7465 6c79 2074 6865 2073 6967 6e69  rately the signi
+00012e30: 6669 6361 6e63 6520 616e 6420 6c6f 6361  ficance and loca
+00012e40: 7469 6f6e 206f 6620 706f 7465 6e74 6961  tion of potentia
+00012e50: 6c20 5747 4473 2c20 7573 696e 6720 7468  l WGDs, using th
+00012e60: 6520 636f 6d6d 616e 6420 6c69 6e65 2062  e command line b
+00012e70: 656c 6f77 2e0a 0a60 6060 0a77 6764 2076  elow...```.wgd v
+00012e80: 697a 202d 6420 7767 645f 6b73 642f 4171  iz -d wgd_ksd/Aq
+00012e90: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00012ea0: 2e74 7376 2e6b 732e 7473 760a 6060 600a  .tsv.ks.tsv.```.
+00012eb0: 0a54 6865 2072 6573 756c 7420 6f66 2045  .The result of E
+00012ec0: 4c4d 4d20 6d69 7874 7572 6520 6d6f 6465  LMM mixture mode
+00012ed0: 6c20 636c 7573 7465 7269 6e67 2073 686f  l clustering sho
+00012ee0: 7773 2074 6861 7420 7468 6572 6520 6973  ws that there is
+00012ef0: 2061 206c 696b 656c 7920 5747 4420 636f   a likely WGD co
+00012f00: 6d70 6f6e 656e 7420 6174 202a 4b2a 3c73  mponent at *K*<s
+00012f10: 7562 3e53 3c2f 7375 623e 2031 2e31 392e  ub>S</sub> 1.19.
+00012f20: 0a0a 215b 5d28 6461 7461 2f65 6c6d 6d5f  ..![](data/elmm_
+00012f30: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00012f40: 6561 2e74 7376 2e6b 732e 7473 765f 6265  ea.tsv.ks.tsv_be
+00012f50: 7374 5f6d 6f64 656c 735f 7765 6967 6874  st_models_weight
+00012f60: 6564 2e73 7667 290a 0a4c 6574 2773 2064  ed.svg)..Let's d
+00012f70: 6f20 6120 6d69 7874 7572 6520 6d6f 6465  o a mixture mode
+00012f80: 6c20 636c 7573 7465 7269 6e67 2066 6f72  l clustering for
+00012f90: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
+00012fa0: 533c 2f73 7562 3e20 746f 6f2c 2075 7369  S</sub> too, usi
+00012fb0: 6e67 2074 6865 2063 6f6d 6d61 6e64 206c  ng the command l
+00012fc0: 696e 6520 6265 6c6f 772e 0a0a 6060 600a  ine below...```.
+00012fd0: 7767 6420 7065 616b 2077 6764 5f6b 7364  wgd peak wgd_ksd
+00012fe0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012ff0: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00013000: 2d61 6e63 686f 7270 6f69 6e74 7320 7767  -anchorpoints wg
+00013010: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00013020: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
+00013030: 7874 202d 2d73 6567 6d65 6e74 7320 7767  xt --segments wg
+00013040: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00013050: 742f 7365 676d 656e 7473 2e74 7874 202d  t/segments.txt -
+00013060: 2d6c 6973 7465 6c65 6d65 6e74 7320 7767  -listelements wg
+00013070: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00013080: 742f 6c69 7374 5f65 6c65 6d65 6e74 732e  t/list_elements.
+00013090: 7478 7420 2d2d 6d75 6c74 6970 6c69 636f  txt --multiplico
+000130a0: 6e70 6169 7273 2077 6764 5f73 796e 2f69  npairs wgd_syn/i
+000130b0: 6164 686f 7265 2d6f 7574 2f6d 756c 7469  adhore-out/multi
+000130c0: 706c 6963 6f6e 5f70 6169 7273 2e74 7874  plicon_pairs.txt
+000130d0: 202d 2d77 6569 6768 7465 640a 6060 600a   --weighted.```.
+000130e0: 0a54 6865 2061 6e63 686f 7220 2a4b 2a3c  .The anchor *K*<
+000130f0: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00013100: 6469 7374 7269 6275 7469 6f6e 2061 6c73  distribution als
+00013110: 6f20 6861 7320 6120 6c69 6b65 6c79 2057  o has a likely W
+00013120: 4744 2063 6f6d 706f 6e65 6e74 2077 6974  GD component wit
+00013130: 6820 6d6f 6465 2031 2e32 382e 0a0a 215b  h mode 1.28...![
+00013140: 5d28 6461 7461 2f4f 7269 6769 6e61 6c5f  ](data/Original_
+00013150: 416e 6368 6f72 4b73 5f47 4d4d 5f43 6f6d  AnchorKs_GMM_Com
+00013160: 706f 6e65 6e74 335f 6e6f 6465 5f77 6569  ponent3_node_wei
+00013170: 6768 7465 645f 4c6f 676e 6f72 6d61 6c2e  ghted_Lognormal.
+00013180: 7376 6729 0a0a 4e6f 7720 7468 6174 2077  svg)..Now that w
+00013190: 6520 6861 7665 2073 6565 6e20 7468 6520  e have seen the 
+000131a0: 6576 6964 656e 6365 206f 6620 6e75 6d65  evidence of nume
+000131b0: 726f 7573 2064 7570 6c69 6361 7465 6420  rous duplicated 
+000131c0: 7365 676d 656e 7473 2061 6e64 2074 6865  segments and the
+000131d0: 2061 6767 7265 6761 7469 6f6e 206f 6620   aggregation of 
+000131e0: 6475 706c 6963 6174 6573 2061 6765 2069  duplicates age i
+000131f0: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+00013200: 3e20 6172 6f75 6e64 2031 2e32 2066 6f72  > around 1.2 for
+00013210: 2061 6e63 686f 7220 7061 6972 7320 616e   anchor pairs an
+00013220: 6420 6e6f 6e2d 616e 6368 6f72 2070 6169  d non-anchor pai
+00013230: 7273 2074 6872 6f75 6768 6f75 7420 7468  rs throughout th
+00013240: 6520 7768 6f6c 6520 6765 6e6f 6d65 2e20  e whole genome. 
+00013250: 5765 2063 616e 2063 6c61 696d 2077 6974  We can claim wit
+00013260: 6820 736f 6d65 2063 6f6e 6669 6465 6e63  h some confidenc
+00013270: 6520 7468 6174 202a 4171 7569 6c65 6769  e that *Aquilegi
+00013280: 6120 636f 6572 756c 6561 2a20 6d69 6768  a coerulea* migh
+00013290: 7420 6861 7665 2065 7870 6572 6965 6e63  t have experienc
+000132a0: 6564 2061 2070 616c 656f 2d70 6f6c 7970  ed a paleo-polyp
+000132b0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+000132c0: 742e 204e 6578 742c 204c 6574 2773 2068  t. Next, Let's h
+000132d0: 6176 6520 6120 6675 7274 6865 7220 6c6f  ave a further lo
+000132e0: 6f6b 2061 626f 7574 2069 7473 2070 6879  ok about its phy
+000132f0: 6c6f 6765 6e65 7469 6320 6c6f 6361 7469  logenetic locati
+00013300: 6f6e 2e20 5765 206b 6e6f 7720 7468 6174  on. We know that
+00013310: 2074 6865 7265 2061 7265 2075 6e63 6572   there are uncer
+00013320: 7461 696e 7469 6573 2061 626f 7574 2077  tainties about w
+00013330: 6865 7468 6572 2074 6869 7320 7075 7461  hether this puta
+00013340: 7469 7665 2070 616c 656f 2d70 6f6c 7970  tive paleo-polyp
+00013350: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+00013360: 7420 6973 2073 6861 7265 6420 7769 7468  t is shared with
+00013370: 2061 6c6c 2065 7564 6963 6f74 7320 6f72   all eudicots or
+00013380: 206e 6f74 2e20 5765 2063 616e 2063 686f   not. We can cho
+00013390: 6f73 6520 736f 6d65 206f 7468 6572 2065  ose some other e
+000133a0: 7564 6963 6f74 2067 656e 6f6d 6573 2074  udicot genomes t
+000133b0: 6f20 7365 6520 7468 6520 6f72 6465 7269  o see the orderi
+000133c0: 6e67 206f 6620 7370 6563 6961 7469 6f6e  ng of speciation
+000133d0: 2061 6e64 2070 6f6c 7970 6c6f 6964 697a   and polyploidiz
+000133e0: 6174 696f 6e20 6576 656e 7473 2e20 4865  ation events. He
+000133f0: 7265 2077 6520 6368 6f6f 7365 202a 5669  re we choose *Vi
+00013400: 7469 7320 7669 6e69 6665 7261 2a2c 202a  tis vinifera*, *
+00013410: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
+00013420: 732a 2061 6e64 202a 4163 6f72 7573 2061  s* and *Acorus a
+00013430: 6d65 7269 6361 6e75 732a 2069 6e20 7468  mericanus* in th
+00013440: 6520 666f 6c6c 6f77 696e 6720 2a4b 2a3c  e following *K*<
+00013450: 7375 623e 533c 2f73 7562 3e20 616e 616c  sub>S</sub> anal
+00013460: 7973 6973 2e20 4669 7273 742c 2077 6520  ysis. First, we 
+00013470: 6275 696c 7420 6120 676c 6f62 616c 204d  built a global M
+00013480: 5242 4820 6661 6d69 6c79 2075 7369 6e67  RBH family using
+00013490: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+000134a0: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+000134b0: 202d 2d67 6c6f 6261 6c6d 7262 6820 4171   --globalmrbh Aq
+000134c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000134d0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+000134e0: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
+000134f0: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
+00013500: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
+00013510: 6c6d 7262 680a 6060 600a 0a49 6e20 7468  lmrbh.```..In th
+00013520: 6520 676c 6f62 616c 204d 5242 4820 6661  e global MRBH fa
+00013530: 6d69 6c79 2c20 6576 6572 7920 7061 6972  mily, every pair
+00013540: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+00013550: 6765 6e65 7320 6973 2074 6865 2072 6563  genes is the rec
+00013560: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
+00013570: 2c20 7375 6767 6573 7469 6e67 2074 7275  , suggesting tru
+00013580: 6520 6f72 7468 6f6c 6f67 6f75 7320 7265  e orthologous re
+00013590: 6c61 7469 6f6e 7368 6970 732e 2057 6520  lationships. We 
+000135a0: 776f 756c 6420 7573 6520 7468 6520 2a4b  would use the *K
+000135b0: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+000135c0: 6c75 6573 2061 7373 6f63 6961 7465 6420  lues associated 
+000135d0: 7769 7468 2074 6865 7365 206f 7274 686f  with these ortho
+000135e0: 6c6f 676f 7573 2070 6169 7273 2074 6f20  logous pairs to 
+000135f0: 6465 6c69 6d69 7420 7468 6520 6469 7665  delimit the dive
+00013600: 7267 656e 6365 202a 4b2a 3c73 7562 3e53  rgence *K*<sub>S
+00013610: 3c2f 7375 623e 2070 6561 6b2e 2054 6f67  </sub> peak. Tog
+00013620: 6574 6865 7220 7769 7468 2074 6865 2077  ether with the w
+00013630: 686f 6c65 2070 6172 616e 6f6d 6520 2a4b  hole paranome *K
+00013640: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00013650: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
+00013660: 6f6e 6475 6374 2074 6865 2072 6174 6520  onduct the rate 
+00013670: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
+00013680: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00013690: 6f77 2e0a 0a21 2153 696e 6365 2060 7767  ow...!!Since `wg
+000136a0: 6460 2076 6572 7369 6f6e 2032 2e30 2e32  d` version 2.0.2
+000136b0: 342c 2077 6520 7265 7772 6f74 6520 6120  4, we rewrote a 
+000136c0: 636c 6561 6e65 7220 616e 6420 7175 6963  cleaner and quic
+000136d0: 6b65 7220 7761 7920 6f66 2064 6f69 6e67  ker way of doing
+000136e0: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
+000136f0: 7465 2063 6f72 7265 6374 696f 6e2e 2049  te correction. I
+00013700: 7427 7320 6e6f 7420 7265 7175 6972 6564  t's not required
+00013710: 2074 6f20 7479 7065 2069 6e20 616e 7920   to type in any 
+00013720: 7370 6569 6365 7320 7061 6972 2061 6e64  speices pair and
+00013730: 2061 2073 6572 6965 7320 6f66 202a 4b2a   a series of *K*
+00013740: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
+00013750: 7473 2077 696c 6c20 6265 2070 726f 6475  ts will be produ
+00013760: 6365 642e 2054 6865 2072 6571 7569 7265  ced. The require
+00013770: 6420 6669 6c65 7320 6172 6520 6f72 7468  d files are orth
+00013780: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00013790: 533c 2f73 7562 3e20 7461 626c 652c 2070  S</sub> table, p
+000137a0: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
+000137b0: 623e 533c 2f73 7562 3e20 7461 626c 652c  b>S</sub> table,
+000137c0: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
+000137d0: 616e 6420 6120 666f 6375 7365 6420 7370  and a focused sp
+000137e0: 6563 6965 7320 2874 6865 206f 6e65 2069  ecies (the one i
+000137f0: 6e70 7574 7465 6420 7769 7468 2070 6172  nputted with par
+00013800: 616c 6f67 6f75 7320 2a4b 2a3c 7375 623e  alogous *K*<sub>
+00013810: 533c 2f73 7562 3e20 6461 7461 292e 2055  S</sub> data). U
+00013820: 7365 7273 2063 616e 2063 686f 6f73 6520  sers can choose 
+00013830: 746f 2061 6464 206f 6e65 206d 6f72 6520  to add one more 
+00013840: 6c61 7965 7220 6f66 2065 6c6d 6d20 6d6f  layer of elmm mo
+00013850: 6465 6c69 6e67 206f 6e20 7061 7261 6c6f  deling on paralo
+00013860: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+00013870: 7375 623e 2076 616c 7565 7320 616e 642f  sub> values and/
+00013880: 6f72 2067 6d6d 206d 6f64 656c 696e 6720  or gmm modeling 
+00013890: 6f6e 2061 6e63 686f 7220 2a4b 2a3c 7375  on anchor *K*<su
+000138a0: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
+000138b0: 2e20 5468 6520 6f72 7468 6f6c 6f67 6f75  . The orthologou
+000138c0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000138d0: 3e20 7661 6c75 6573 2063 616e 2062 6520  > values can be 
+000138e0: 6361 6c63 756c 6174 6564 2075 7369 6e67  calculated using
+000138f0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00013900: 6f77 2e0a 0a60 6060 0a77 6764 206b 7364  ow...```.wgd ksd
+00013910: 2077 6764 5f67 6c6f 6261 6c6d 7262 682f   wgd_globalmrbh/
+00013920: 676c 6f62 616c 5f4d 5242 482e 7473 7620  global_MRBH.tsv 
+00013930: 7365 7173 2a20 2d6f 2077 6764 5f67 6c6f  seqs* -o wgd_glo
+00013940: 6261 6c6d 7262 685f 6b73 0a60 6060 0a0a  balmrbh_ks.```..
+00013950: 5769 7468 2074 6865 2063 616c 6375 6c61  With the calcula
+00013960: 7465 6420 6f72 7468 6f6c 6f67 6f75 7320  ted orthologous 
+00013970: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013980: 7461 626c 652c 2077 6520 6361 6e20 7573  table, we can us
+00013990: 6520 7468 6520 636f 6d6d 616e 6420 6265  e the command be
+000139a0: 6c6f 7720 746f 2063 6f6e 6475 6374 2074  low to conduct t
+000139b0: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+000139c0: 6f6e 2061 6e64 2f6f 7220 6d69 7874 7572  on and/or mixtur
+000139d0: 6520 6d6f 6465 6c69 6e67 2061 6e61 6c79  e modeling analy
+000139e0: 7369 732e 0a0a 6060 600a 7767 6420 7669  sis...```.wgd vi
+000139f0: 7a20 2d64 2077 6764 5f67 6c6f 6261 6c6d  z -d wgd_globalm
+00013a00: 7262 685f 6b73 2f67 6c6f 6261 6c5f 4d52  rbh_ks/global_MR
+00013a10: 4248 2e74 7376 2e6b 732e 7473 7620 2d66  BH.tsv.ks.tsv -f
+00013a20: 6120 4171 7569 6c65 6769 615f 636f 6572  a Aquilegia_coer
+00013a30: 756c 6561 202d 6570 6b20 4171 7569 6c65  ulea -epk Aquile
+00013a40: 6769 615f 636f 6572 756c 6561 2e6b 732e  gia_coerulea.ks.
+00013a50: 7473 7620 2d61 7020 616e 6368 6f72 706f  tsv -ap anchorpo
+00013a60: 696e 7473 2e74 7874 202d 7370 2073 7065  ints.txt -sp spe
+00013a70: 6369 6573 7472 6565 2e6e 7720 2d6f 2077  ciestree.nw -o w
+00013a80: 6764 5f76 697a 5f6d 6978 6564 5f4b 7320  gd_viz_mixed_Ks 
+00013a90: 2d2d 706c 6f74 656c 6d6d 202d 2d70 6c6f  --plotelmm --plo
+00013aa0: 7461 7067 6d6d 0a60 6060 0a0a 6060 600a  tapgmm.```..```.
+00013ab0: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
+00013ac0: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
+00013ad0: 4248 2e74 7376 2073 6571 732a 202d 2d65  BH.tsv seqs* --e
+00013ae0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
+00013af0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00013b00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00013b10: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00013b20: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00013b30: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
+00013b40: 6d72 6268 5f6b 7320 2d2d 7370 6169 7220  mrbh_ks --spair 
+00013b50: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
+00013b60: 6c65 613b 5072 6f74 6561 5f63 796e 6172  lea;Protea_cynar
+00013b70: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+00013b80: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013b90: 6561 3b56 6974 6973 5f76 696e 6966 6572  ea;Vitis_vinifer
+00013ba0: 6122 202d 2d73 7061 6972 2022 4171 7569  a" --spair "Aqui
+00013bb0: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
+00013bc0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00013bd0: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+00013be0: 6567 6961 5f63 6f65 7275 6c65 613b 4171  egia_coerulea;Aq
+00013bf0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00013c00: 2220 2d2d 706c 6f74 6b64 6520 282d 6170  " --plotkde (-ap
+00013c10: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00013c20: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+00013c30: 732e 7478 7429 0a60 6060 0a0a 5468 6520  s.txt).```..The 
+00013c40: 6669 6c65 2060 7370 6563 6965 7374 7265  file `speciestre
+00013c50: 652e 6e77 6020 6973 2074 6865 2074 6578  e.nw` is the tex
+00013c60: 7420 6669 6c65 206f 6620 7370 6563 6965  t file of specie
+00013c70: 7320 7472 6565 2069 6e20 6e65 7769 636b  s tree in newick
+00013c80: 2074 6861 7420 7261 7465 2063 6f72 7265   that rate corre
+00013c90: 6374 696f 6e20 776f 756c 6420 6265 2063  ction would be c
+00013ca0: 6f6e 6475 6374 6564 206f 6e2e 2049 7473  onducted on. Its
+00013cb0: 2063 6f6e 7465 6e74 2069 7320 6173 2062   content is as b
+00013cc0: 656c 6f77 2e20 5573 6572 7320 6e65 6564  elow. Users need
+00013cd0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00013ce0: 7370 6563 6965 7320 7061 6972 7320 746f  species pairs to
+00013cf0: 2062 6520 706c 6f74 7465 642e 2057 6520   be plotted. We 
+00013d00: 7375 6767 6573 7420 6164 6469 6e67 2074  suggest adding t
+00013d10: 6865 206f 7074 696f 6e20 602d 2d72 6577  he option `--rew
+00013d20: 6569 6768 7460 2074 6f20 7265 6361 6c63  eight` to recalc
+00013d30: 756c 6174 6520 7468 6520 7765 6967 6874  ulate the weight
+00013d40: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
+00013d50: 7220 7375 6368 2074 6861 7420 7468 6520  r such that the 
+00013d60: 7765 6967 6874 206f 6620 6f72 7468 6f6c  weight of orthol
+00013d70: 6f67 6f75 7320 6765 6e65 2070 6169 7273  ogous gene pairs
+00013d80: 2077 696c 6c20 6265 636f 6d65 2031 2061   will become 1 a
+00013d90: 7320 7468 6520 7061 7261 6c6f 676f 7573  s the paralogous
+00013da0: 2067 656e 6520 7061 6972 732e 2054 6865   gene pairs. The
+00013db0: 2066 6c61 6720 602d 2d70 6c6f 746b 6465   flag `--plotkde
+00013dc0: 6020 6361 6e20 6265 2061 6464 6564 2077  ` can be added w
+00013dd0: 6865 6e20 7468 6520 6b64 6520 6375 7276  hen the kde curv
+00013de0: 6520 6f66 206f 7274 686f 6c6f 676f 7573  e of orthologous
+00013df0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00013e00: 2069 7320 6465 7369 7265 642e 2045 7874   is desired. Ext
+00013e10: 7261 2063 6f6c 6c69 6e65 6172 2064 6174  ra collinear dat
+00013e20: 6120 6361 6e20 6265 2061 6464 6564 2062  a can be added b
+00013e30: 7920 7468 6520 6f70 7469 6f6e 2060 2d61  y the option `-a
+00013e40: 7060 2e0a 0a60 6060 0a28 2828 5669 7469  p`...```.(((Viti
+00013e50: 735f 7669 6e69 6665 7261 2c50 726f 7465  s_vinifera,Prote
+00013e60: 615f 6379 6e61 726f 6964 6573 292c 4171  a_cynaroides),Aq
+00013e70: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00013e80: 292c 4163 6f72 7573 5f61 6d65 7269 6361  ),Acorus_america
+00013e90: 6e75 7329 3b0a 6060 600a 0a21 5b5d 2864  nus);.```..![](d
+00013ea0: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
+00013eb0: 6572 756c 6561 5f47 6c6f 6261 6c6d 7262  erulea_Globalmrb
+00013ec0: 684b 735f 436f 7272 6563 7465 642e 6b73  hKs_Corrected.ks
+00013ed0: 642e 7376 6729 0a0a 4173 2073 686f 776e  d.svg)..As shown
+00013ee0: 2061 626f 7665 2c20 6265 6361 7573 6520   above, because 
+00013ef0: 6f66 2074 6865 2068 6967 6865 7220 7375  of the higher su
+00013f00: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
+00013f10: 6f66 202a 4171 7569 6c65 6769 6120 636f  of *Aquilegia co
+00013f20: 6572 756c 6561 2a2c 2074 6865 206f 7269  erulea*, the ori
+00013f30: 6769 6e61 6c20 6f72 7468 6f6c 6f67 6f75  ginal orthologou
+00013f40: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+00013f50: 3e20 7661 6c75 6573 2077 6572 6520 6163  > values were ac
+00013f60: 7475 616c 6c79 2075 6e64 6572 6573 7469  tually underesti
+00013f70: 6d61 7465 6420 696e 2074 6865 2074 696d  mated in the tim
+00013f80: 652d 6672 616d 6520 6f66 202a 4171 7569  e-frame of *Aqui
+00013f90: 6c65 6769 6120 636f 6572 756c 6561 2a2e  legia coerulea*.
+00013fa0: 2057 6865 6e20 7765 2072 6563 6f76 6572   When we recover
+00013fb0: 6564 2074 6865 2064 6976 6572 6765 6e63  ed the divergenc
+00013fc0: 6520 7375 6273 7469 7475 7469 6f6e 2064  e substitution d
+00013fd0: 6973 7461 6e63 6520 696e 2074 6572 6d73  istance in terms
+00013fe0: 206f 6620 7477 6f20 7469 6d65 7320 6f66   of two times of
+00013ff0: 2074 6865 2062 7261 6e63 682d 7370 6563   the branch-spec
+00014000: 6966 6963 2063 6f6e 7472 6962 7574 696f  ific contributio
+00014010: 6e20 6f66 202a 4171 7569 6c65 6769 6120  n of *Aquilegia 
+00014020: 636f 6572 756c 6561 2a20 7369 6e63 6520  coerulea* since 
+00014030: 6974 7320 6469 7665 7267 656e 6365 2077  its divergence w
+00014040: 6974 6820 7468 6520 7369 7374 6572 2073  ith the sister s
+00014050: 7065 6369 6573 2070 6c75 7320 7468 6520  pecies plus the 
+00014060: 7368 6172 6564 2073 7562 7374 6974 7574  shared substitut
+00014070: 696f 6e20 6469 7374 616e 6365 2062 6566  ion distance bef
+00014080: 6f72 6520 6469 7665 7267 656e 6365 2028  ore divergence (
+00014090: 696e 2072 656c 6174 6976 6520 746f 2074  in relative to t
+000140a0: 6865 206f 7574 6772 6f75 7029 2c20 7468  he outgroup), th
+000140b0: 6520 636f 7272 6563 7465 6420 2a4b 2a3c  e corrected *K*<
+000140c0: 7375 623e 533c 2f73 7562 3e20 6d6f 6465  sub>S</sub> mode
+000140d0: 2062 6563 616d 6520 6c61 7267 6572 2e0a   became larger..
+000140e0: 0a49 6620 6f6e 6520 6861 6420 7468 6520  .If one had the 
+000140f0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
+00014100: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+00014110: 2061 6c72 6561 6479 2c20 6f6e 6520 636f   already, one co
+00014120: 756c 6420 616c 736f 2061 7070 6c79 2074  uld also apply t
+00014130: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+00014140: 7669 7a60 2074 6f20 636f 6e64 7563 7420  viz` to conduct 
+00014150: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+00014160: 696f 6e20 616e 616c 7973 6973 2075 7369  ion analysis usi
+00014170: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00014180: 656c 6f77 2e20 4e6f 7465 2074 6861 7420  elow. Note that 
+00014190: 7468 6520 6f72 6465 7220 6f66 2067 6976  the order of giv
+000141a0: 656e 2060 7370 6169 7260 206f 7074 696f  en `spair` optio
+000141b0: 6e73 2064 6563 6964 6573 2074 6865 2063  ns decides the c
+000141c0: 6f6c 6f72 206f 6620 7468 6520 2a4b 2a3c  olor of the *K*<
+000141d0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+000141e0: 7269 6275 7469 6f6e 206f 6620 6561 6368  ribution of each
+000141f0: 2073 7065 6369 6573 2070 6169 722e 2054   species pair. T
+00014200: 6865 2061 6464 6974 696f 6e61 6c20 6f70  he additional op
+00014210: 7469 6f6e 2060 666f 6375 7332 616c 6c60  tion `focus2all`
+00014220: 2069 6e20 6077 6764 206b 7364 6020 616e   in `wgd ksd` an
+00014230: 6420 6077 6764 2076 697a 6020 6361 6e20  d `wgd viz` can 
+00014240: 6265 2075 7365 6420 746f 2074 656c 6c20  be used to tell 
+00014250: 7468 6520 7072 6f67 7261 6d20 7468 6174  the program that
+00014260: 2074 6865 2073 7065 6369 6573 2070 6169   the species pai
+00014270: 7273 2061 7265 2062 6574 7765 656e 2074  rs are between t
+00014280: 6865 2066 6f63 7573 2073 7065 6369 6573  he focus species
+00014290: 2061 6e64 2061 6c6c 2074 6865 206f 7468   and all the oth
+000142a0: 6572 2073 7065 6369 6573 2073 7563 6820  er species such 
+000142b0: 7468 6174 2075 7365 7273 2064 6f6e 2774  that users don't
+000142c0: 206e 6565 6420 746f 2074 7970 6520 696e   need to type in
+000142d0: 2065 6163 6820 7061 6972 2069 6e64 6976   each pair indiv
+000142e0: 6964 7561 6c6c 792e 0a0a 6060 600a 7767  idually...```.wg
+000142f0: 6420 7669 7a20 2d64 2077 6764 5f67 6c6f  d viz -d wgd_glo
+00014300: 6261 6c6d 7262 685f 6b73 2f67 6c6f 6261  balmrbh_ks/globa
+00014310: 6c5f 4d52 4248 2e74 7376 2e6b 732e 7473  l_MRBH.tsv.ks.ts
+00014320: 7620 2d2d 6578 7472 6170 6172 616e 6f6d  v --extraparanom
+00014330: 656b 7320 7767 645f 6b73 642f 4171 7569  eks wgd_ksd/Aqui
+00014340: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00014350: 7376 2e6b 732e 7473 7620 2d73 7020 7370  sv.ks.tsv -sp sp
+00014360: 6563 6965 7374 7265 652e 6e77 202d 2d72  eciestree.nw --r
+00014370: 6577 6569 6768 7420 2d61 7020 7767 645f  eweight -ap wgd_
+00014380: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00014390: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
+000143a0: 202d 6f20 7767 645f 7669 7a5f 6d69 7865   -o wgd_viz_mixe
+000143b0: 645f 4b73 202d 2d73 7061 6972 2022 4171  d_Ks --spair "Aq
+000143c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000143d0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
+000143e0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
+000143f0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
+00014400: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
+00014410: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00014420: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
+00014430: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
+00014440: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00014450: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
+00014460: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
+00014470: 2d70 6c6f 746b 6465 2028 2d2d 6773 6d61  -plotkde (--gsma
+00014480: 7020 6765 6e65 5f73 7065 6369 6573 2e6d  p gene_species.m
+00014490: 6170 290a 6060 600a 0a57 6974 6820 7468  ap).```..With th
+000144a0: 6520 6f70 7469 6f6e 2060 666f 6375 7332  e option `focus2
+000144b0: 616c 6c60 2c20 7468 6520 636f 6d6d 616e  all`, the comman
+000144c0: 6420 6361 6e20 6265 2061 6c73 6f20 6c69  d can be also li
+000144d0: 6b65 2074 6869 732c 2073 616d 6520 7769  ke this, same wi
+000144e0: 7468 2060 7767 6420 6b73 6460 2e0a 0a60  th `wgd ksd`...`
+000144f0: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
+00014500: 645f 676c 6f62 616c 6d72 6268 5f6b 732f  d_globalmrbh_ks/
+00014510: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
+00014520: 6b73 2e74 7376 202d 2d65 7874 7261 7061  ks.tsv --extrapa
+00014530: 7261 6e6f 6d65 6b73 2077 6764 5f6b 7364  ranomeks wgd_ksd
+00014540: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00014550: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00014560: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
+00014570: 7720 2d2d 7265 7765 6967 6874 202d 6170  w --reweight -ap
+00014580: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00014590: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+000145a0: 732e 7478 7420 2d6f 2077 6764 5f76 697a  s.txt -o wgd_viz
+000145b0: 5f6d 6978 6564 5f4b 7320 2d2d 666f 6375  _mixed_Ks --focu
+000145c0: 7332 616c 6c20 4171 7569 6c65 6769 615f  s2all Aquilegia_
+000145d0: 636f 6572 756c 6561 202d 2d70 6c6f 746b  coerulea --plotk
+000145e0: 6465 0a60 6060 0a0a 4e6f 7465 2074 6861  de.```..Note tha
+000145f0: 7420 7765 2063 616e 2065 6173 696c 7920  t we can easily 
+00014600: 7368 6f77 2074 6861 7420 2a41 7175 696c  show that *Aquil
+00014610: 6567 6961 2063 6f65 7275 6c65 612a 2068  egia coerulea* h
+00014620: 6173 2068 6967 6865 7220 7375 6273 7469  as higher substi
+00014630: 7475 7469 6f6e 2072 6174 6520 7468 616e  tution rate than
+00014640: 202a 5072 6f74 6561 2063 796e 6172 6f69   *Protea cynaroi
+00014650: 6465 732a 2061 6e64 202a 5669 7469 7320  des* and *Vitis 
+00014660: 7669 6e69 6665 7261 2a20 6279 2063 6f6d  vinifera* by com
+00014670: 7061 7269 6e67 2074 6865 6972 2073 7562  paring their sub
+00014680: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
+00014690: 6365 2069 6e20 7265 6761 7264 2074 6f20  ce in regard to 
+000146a0: 7468 6520 7361 6d65 2064 6976 6572 6765  the same diverge
+000146b0: 6e63 6520 6576 656e 7420 7769 7468 206f  nce event with o
+000146c0: 7574 6772 6f75 7020 7370 6563 6965 7320  utgroup species 
+000146d0: 2a41 636f 7275 735f 616d 6572 6963 616e  *Acorus_american
+000146e0: 7573 2a2c 2075 7369 6e67 2063 6f6d 6d61  us*, using comma
+000146f0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
+00014700: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00014710: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00014720: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00014730: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00014740: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00014750: 202d 6f20 7767 645f 7669 7a5f 436f 6d70   -o wgd_viz_Comp
+00014760: 6172 655f 7261 7465 202d 2d73 7061 6972  are_rate --spair
+00014770: 2022 4163 6f72 7573 5f61 6d65 7269 6361   "Acorus_america
+00014780: 6e75 733b 5072 6f74 6561 5f63 796e 6172  nus;Protea_cynar
+00014790: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+000147a0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000147b0: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
+000147c0: 616e 7573 2220 2d2d 7370 6169 7220 2256  anus" --spair "V
+000147d0: 6974 6973 5f76 696e 6966 6572 613b 4163  itis_vinifera;Ac
+000147e0: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
+000147f0: 202d 2d67 736d 6170 2067 656e 655f 7370   --gsmap gene_sp
+00014800: 6563 6965 732e 6d61 7020 2d2d 706c 6f74  ecies.map --plot
+00014810: 6b64 650a 6060 600a 0a21 5b5d 2864 6174  kde.```..![](dat
+00014820: 612f 5261 775f 4f72 7468 6f6c 6f67 7565  a/Raw_Orthologue
+00014830: 735f 436f 6d70 6172 655f 7261 7465 2e6b  s_Compare_rate.k
+00014840: 7364 2e73 7667 290a 0a41 7320 6469 7370  sd.svg)..As disp
+00014850: 6c61 7965 6420 6162 6f76 652c 2074 6865  layed above, the
+00014860: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
+00014870: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
+00014880: 7565 7320 6265 7774 6565 6e20 2a41 7175  ues bewteen *Aqu
+00014890: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+000148a0: 2061 6e64 202a 4163 6f72 7573 2061 6d65   and *Acorus ame
+000148b0: 7269 6361 6e75 732a 2068 6173 2074 6865  ricanus* has the
+000148c0: 2068 6967 6865 7374 206d 6f64 652c 2069   highest mode, i
+000148d0: 6e64 6963 6174 696e 6774 6865 2066 6173  ndicatingthe fas
+000148e0: 7465 7220 7375 6273 7469 7475 7469 6f6e  ter substitution
+000148f0: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
+00014900: 6769 6120 636f 6572 756c 6561 2a20 636f  gia coerulea* co
+00014910: 6d70 6172 6564 2074 6f20 2a50 726f 7465  mpared to *Prote
+00014920: 6120 6379 6e61 726f 6964 6573 2a20 616e  a cynaroides* an
+00014930: 6420 2a56 6974 6973 2076 696e 6966 6572  d *Vitis vinifer
+00014940: 612a 2e0a 0a42 6566 6f72 6520 7632 2e30  a*...Before v2.0
+00014950: 2e32 312c 2074 6865 2067 656e 652d 7370  .21, the gene-sp
+00014960: 6563 6965 7320 6d61 7020 6669 6c65 2069  ecies map file i
+00014970: 7320 6e65 6363 6573 7361 7269 6c79 206e  s neccessarily n
+00014980: 6565 6465 6420 666f 7220 6974 7320 696d  eeded for its im
+00014990: 706c 656d 656e 7461 7469 6f6e 2069 6e20  plementation in 
+000149a0: 6077 6764 2076 697a 602c 2077 6869 6368  `wgd viz`, which
+000149b0: 2073 686f 756c 6420 6265 2061 7574 6f6d   should be autom
+000149c0: 6174 656c 7920 7072 6f64 7563 6564 2062  ately produced b
+000149d0: 7920 7468 6520 6c61 7374 2060 7767 6420  y the last `wgd 
+000149e0: 6b73 6460 2073 7465 7020 6769 7665 6e20  ksd` step given 
+000149f0: 7468 6520 6073 7061 6972 6020 616e 6420  the `spair` and 
+00014a00: 6073 7065 6369 6573 7472 6565 6020 7061  `speciestree` pa
+00014a10: 7261 6d65 7465 7273 2e20 5468 6520 6067  rameters. The `g
+00014a20: 656e 655f 7370 6563 6965 732e 6d61 7060  ene_species.map`
+00014a30: 2068 6173 2063 6f6e 7465 6e74 7320 6173   has contents as
+00014a40: 2062 656c 6f77 2069 6e20 7768 6963 6820   below in which 
+00014a50: 6561 6368 206c 696e 6520 6973 2074 6865  each line is the
+00014a60: 206a 6f69 6e65 6420 7374 7269 6e67 206f   joined string o
+00014a70: 6620 6765 6e65 206e 616d 6520 616e 6420  f gene name and 
+00014a80: 7370 6563 6965 7320 6e61 6d65 2062 7920  species name by 
+00014a90: 7370 6163 652e 2041 6674 6572 2076 322e  space. After v2.
+00014aa0: 302e 3231 2028 696e 636c 7564 6564 292c  0.21 (included),
+00014ab0: 2074 6865 2067 656e 652d 7370 6563 6965   the gene-specie
+00014ac0: 7320 6d61 7020 6669 6c65 2069 7320 6e6f  s map file is no
+00014ad0: 7420 6e65 6363 6573 7361 7269 6c79 206e  t neccessarily n
+00014ae0: 6565 6465 6420 616e 796d 6f72 652e 0a0a  eeded anymore...
+00014af0: 6060 600a 4171 636f 6536 4730 3537 3830  ```.Aqcoe6G05780
+00014b00: 302e 3120 4171 7569 6c65 6769 615f 636f  0.1 Aquilegia_co
+00014b10: 6572 756c 6561 0a56 7669 5f56 4954 5f32  erulea.Vvi_VIT_2
+00014b20: 3031 7330 3031 3167 3031 3533 302e 3120  01s0011g01530.1 
+00014b30: 5669 7469 735f 7669 6e69 6665 7261 0a50  Vitis_vinifera.P
+00014b40: 6379 5f50 726f 6379 3031 6730 3835 3130  cy_Procy01g08510
+00014b50: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+00014b60: 6573 0a41 616d 5f41 636f 7261 2e30 3447  es.Aam_Acora.04G
+00014b70: 3134 3239 3030 2e31 2041 636f 7275 735f  142900.1 Acorus_
+00014b80: 616d 6572 6963 616e 7573 0a60 6060 0a0a  americanus.```..
+00014b90: 4120 6d6f 7265 2063 6f6d 706c 6578 2070  A more complex p
+00014ba0: 6c6f 7420 6361 6e20 6265 206d 6164 6520  lot can be made 
+00014bb0: 6279 2061 6464 2074 6865 2066 6c61 6720  by add the flag 
+00014bc0: 602d 2d70 6c6f 7465 6c6d 6d60 2073 7563  `--plotelmm` suc
+00014bd0: 6820 7468 6174 2074 6865 2045 4c4d 4d20  h that the ELMM 
+00014be0: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+00014bf0: 206f 6620 7072 6f76 6964 6564 2070 6172   of provided par
+00014c00: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00014c10: 2f73 7562 3e20 6361 6e20 6265 2073 7570  /sub> can be sup
+00014c20: 6572 696d 706f 7365 642c 2075 7369 6e67  erimposed, using
+00014c30: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00014c40: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
+00014c50: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00014c60: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+00014c70: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
+00014c80: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
+00014c90: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00014ca0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00014cb0: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00014cc0: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00014cd0: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
+00014ce0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00014cf0: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
+00014d00: 6764 5f76 697a 5f6d 6978 6564 5f4b 735f  gd_viz_mixed_Ks_
+00014d10: 656c 6d6d 202d 2d73 7061 6972 2022 4171  elmm --spair "Aq
+00014d20: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014d30: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
+00014d40: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
+00014d50: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
+00014d60: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
+00014d70: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00014d80: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
+00014d90: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
+00014da0: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00014db0: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
+00014dc0: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
+00014dd0: 2d67 736d 6170 2067 656e 655f 7370 6563  -gsmap gene_spec
+00014de0: 6965 732e 6d61 7020 2d2d 706c 6f74 6b64  ies.map --plotkd
+00014df0: 6520 2d2d 706c 6f74 656c 6d6d 0a60 6060  e --plotelmm.```
+00014e00: 0a0a 215b 5d28 6461 7461 2f41 7175 696c  ..![](data/Aquil
+00014e10: 6567 6961 5f63 6f65 7275 6c65 615f 476c  egia_coerulea_Gl
+00014e20: 6f62 616c 6d72 6268 4b73 5f45 6c6d 6d5f  obalmrbhKs_Elmm_
+00014e30: 436f 7272 6563 7465 642e 6b73 642e 7376  Corrected.ksd.sv
+00014e40: 6729 0a0a 4672 6f6d 2074 6865 206d 6978  g)..From the mix
+00014e50: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
+00014e60: 623e 2070 6c6f 7420 6162 6f76 652c 2077  b> plot above, w
+00014e70: 6520 6361 6e20 7365 6520 7468 6174 2074  e can see that t
+00014e80: 6865 206f 7074 696d 697a 6564 206c 6f67  he optimized log
+00014e90: 6e6f 726d 616c 2063 6f6d 706f 6e65 6e74  normal component
+00014ea0: 2062 2077 6974 6820 6d6f 6465 2031 2e32   b with mode 1.2
+00014eb0: 2069 7320 796f 756e 6765 7220 7468 616e   is younger than
+00014ec0: 2074 6865 2063 6f72 7265 6374 6564 206f   the corrected o
+00014ed0: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00014ee0: 7562 3e53 3c2f 7375 623e 206d 6f64 6520  ub>S</sub> mode 
+00014ef0: 7769 7468 202a 5072 6f74 6561 2063 796e  with *Protea cyn
+00014f00: 6172 6f69 6465 732a 2061 6e64 202a 5669  aroides* and *Vi
+00014f10: 7469 7320 7669 6e69 6665 7261 2a20 2831  tis vinifera* (1
+00014f20: 2e33 3920 616e 6420 312e 3437 2c20 7265  .39 and 1.47, re
+00014f30: 7370 6563 7469 7665 6c79 292e 0a0a 4265  spectively)...Be
+00014f40: 7369 6465 732c 2077 6520 6361 6e20 616c  sides, we can al
+00014f50: 736f 2061 6464 2074 6865 2047 4d4d 206d  so add the GMM m
+00014f60: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
+00014f70: 6f66 2061 6e63 686f 7220 2a4b 2a3c 7375  of anchor *K*<su
+00014f80: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
+00014f90: 2077 6974 6820 7468 6520 666c 6167 2060   with the flag `
+00014fa0: 2d2d 706c 6f74 6170 676d 6d60 2c20 7573  --plotapgmm`, us
+00014fb0: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00014fc0: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
+00014fd0: 7669 7a20 2d64 2077 6764 5f67 6c6f 6261  viz -d wgd_globa
+00014fe0: 6c6d 7262 685f 6b73 2f67 6c6f 6261 6c5f  lmrbh_ks/global_
+00014ff0: 4d52 4248 2e74 7376 2e6b 732e 7473 7620  MRBH.tsv.ks.tsv 
+00015000: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+00015010: 7320 7767 645f 6b73 642f 4171 7569 6c65  s wgd_ksd/Aquile
+00015020: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00015030: 2e6b 732e 7473 7620 2d73 7020 7370 6563  .ks.tsv -sp spec
+00015040: 6965 7374 7265 652e 6e77 202d 2d72 6577  iestree.nw --rew
+00015050: 6569 6768 7420 2d61 7020 7767 645f 7379  eight -ap wgd_sy
+00015060: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+00015070: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
+00015080: 6f20 7767 645f 7669 7a5f 6d69 7865 645f  o wgd_viz_mixed_
+00015090: 4b73 5f65 6c6d 6d20 2d2d 7370 6169 7220  Ks_elmm --spair 
+000150a0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
+000150b0: 6c65 613b 5072 6f74 6561 5f63 796e 6172  lea;Protea_cynar
+000150c0: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+000150d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000150e0: 6561 3b56 6974 6973 5f76 696e 6966 6572  ea;Vitis_vinifer
+000150f0: 6122 202d 2d73 7061 6972 2022 4171 7569  a" --spair "Aqui
+00015100: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
+00015110: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00015120: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+00015130: 6567 6961 5f63 6f65 7275 6c65 613b 4171  egia_coerulea;Aq
+00015140: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015150: 2220 2d2d 6773 6d61 7020 6765 6e65 5f73  " --gsmap gene_s
+00015160: 7065 6369 6573 2e6d 6170 202d 2d70 6c6f  pecies.map --plo
+00015170: 746b 6465 202d 2d70 6c6f 7461 7067 6d6d  tkde --plotapgmm
+00015180: 6d0a 6060 600a 0a21 5b5d 2864 6174 612f  m.```..![](data/
+00015190: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000151a0: 6561 5f47 6c6f 6261 6c6d 7262 684b 735f  ea_GlobalmrbhKs_
+000151b0: 4170 676d 6d5f 436f 7272 6563 7465 642e  Apgmm_Corrected.
+000151c0: 6b73 642e 7376 6729 0a0a 4173 206d 616e  ksd.svg)..As man
+000151d0: 6966 6573 7465 6420 6162 6f76 652c 2074  ifested above, t
+000151e0: 6865 2061 6e63 686f 7220 2a4b 2a3c 7375  he anchor *K*<su
+000151f0: 623e 533c 2f73 7562 3e20 636f 6d70 6f6e  b>S</sub> compon
+00015200: 656e 7420 3220 7769 7468 206d 6f64 6520  ent 2 with mode 
+00015210: 312e 3238 2069 7320 616c 736f 2079 6f75  1.28 is also you
+00015220: 6e67 6572 2074 6861 6e20 7468 6520 636f  nger than the co
+00015230: 7272 6563 7465 6420 6f72 7468 6f6c 6f67  rrected ortholog
+00015240: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00015250: 7562 3e20 6d6f 6465 2077 6974 6820 2a50  ub> mode with *P
+00015260: 726f 7465 6120 6379 6e61 726f 6964 6573  rotea cynaroides
+00015270: 2a20 616e 6420 2a56 6974 6973 2076 696e  * and *Vitis vin
+00015280: 6966 6572 612a 2e20 4275 7420 7765 206e  ifera*. But we n
+00015290: 6565 6420 746f 2062 6520 6f66 2063 6f75  eed to be of cou
+000152a0: 7273 6520 6361 7574 696f 7573 2074 6861  rse cautious tha
+000152b0: 7420 7375 6368 2064 6973 7469 6e63 7469  t such distincti
+000152c0: 6f6e 2063 6f6d 6573 2077 6974 6820 7468  on comes with th
+000152d0: 6520 756e 6365 7274 6169 6e74 6965 7320  e uncertainties 
+000152e0: 696e 7472 6f64 7563 6564 2066 726f 6d20  introduced from 
+000152f0: 7468 6520 6170 706c 6965 6420 6d69 7874  the applied mixt
+00015300: 7572 6520 6d6f 6465 6c69 6e67 206d 6574  ure modeling met
+00015310: 686f 646f 6c6f 6779 2069 6e20 7465 726d  hodology in term
+00015320: 7320 6f66 2066 6f72 2069 6e73 7461 6e63  s of for instanc
+00015330: 6520 6469 6666 6572 656e 7420 696e 6974  e different init
+00015340: 6961 6c69 7a61 7469 6f6e 2070 6f69 6e74  ialization point
+00015350: 7320 616e 6420 7468 6520 6973 7375 6520  s and the issue 
+00015360: 6f66 206f 7665 7266 6974 7469 6e67 2061  of overfitting a
+00015370: 6e64 2074 6865 2073 6973 7465 7220 7370  nd the sister sp
+00015380: 6563 6965 7373 2061 646f 7074 6564 2069  eciess adopted i
+00015390: 6e20 7468 6174 2074 6865 7265 206d 6967  n that there mig
+000153a0: 6874 2062 6520 7370 6563 6965 7320 7769  ht be species wi
+000153b0: 7468 206d 6f72 6520 6469 7370 6172 6174  th more disparat
+000153c0: 6520 7375 6273 7469 7475 7469 6f6e 2072  e substitution r
+000153d0: 6174 6573 2074 6861 6e20 7468 6520 6f6e  ates than the on
+000153e0: 6520 7765 2063 686f 7365 2e0a 0a41 6464  e we chose...Add
+000153f0: 696e 6720 626f 7468 2045 4c4d 4d20 7265  ing both ELMM re
+00015400: 7375 6c74 2066 6f72 2070 6172 616e 6f6d  sult for paranom
+00015410: 6520 616e 6420 474d 4d20 7265 7375 6c74  e and GMM result
+00015420: 2066 6f72 2061 6e63 686f 7220 2a4b 2a3c   for anchor *K*<
+00015430: 7375 623e 533c 2f73 7562 3e20 6361 6e20  sub>S</sub> can 
+00015440: 6265 2061 6368 6965 7665 6420 6a75 7374  be achieved just
+00015450: 2062 7920 6164 6420 7468 6520 7477 6f20   by add the two 
+00015460: 666c 6167 7320 6d65 6e74 696f 6e65 6420  flags mentioned 
+00015470: 6162 6f76 652c 2075 7369 6e67 2074 6865  above, using the
+00015480: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
+00015490: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
+000154a0: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+000154b0: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
+000154c0: 762e 6b73 2e74 7376 202d 2d65 7874 7261  v.ks.tsv --extra
+000154d0: 7061 7261 6e6f 6d65 6b73 2077 6764 5f6b  paranomeks wgd_k
+000154e0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+000154f0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00015500: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00015510: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00015520: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+00015530: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+00015540: 6e74 732e 7478 7420 2d6f 2077 6764 5f76  nts.txt -o wgd_v
+00015550: 697a 5f6d 6978 6564 5f4b 735f 656c 6d6d  iz_mixed_Ks_elmm
+00015560: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00015570: 6769 615f 636f 6572 756c 6561 3b50 726f  gia_coerulea;Pro
+00015580: 7465 615f 6379 6e61 726f 6964 6573 2220  tea_cynaroides" 
+00015590: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+000155a0: 6961 5f63 6f65 7275 6c65 613b 5669 7469  ia_coerulea;Viti
+000155b0: 735f 7669 6e69 6665 7261 2220 2d2d 7370  s_vinifera" --sp
+000155c0: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
+000155d0: 6f65 7275 6c65 613b 4163 6f72 7573 5f61  oerulea;Acorus_a
+000155e0: 6d65 7269 6361 6e75 7322 202d 2d73 7061  mericanus" --spa
+000155f0: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
+00015600: 6572 756c 6561 3b41 7175 696c 6567 6961  erulea;Aquilegia
+00015610: 5f63 6f65 7275 6c65 6122 202d 2d67 736d  _coerulea" --gsm
+00015620: 6170 2067 656e 655f 7370 6563 6965 732e  ap gene_species.
+00015630: 6d61 7020 2d2d 706c 6f74 6b64 6520 2d2d  map --plotkde --
+00015640: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
+00015650: 7067 6d6d 6d0a 6060 600a 0a21 5b5d 2864  pgmmm.```..![](d
+00015660: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
+00015670: 6572 756c 6561 5f47 6c6f 6261 6c6d 7262  erulea_Globalmrb
+00015680: 684b 735f 456c 6d6d 5f41 7067 6d6d 5f43  hKs_Elmm_Apgmm_C
+00015690: 6f72 7265 6374 6564 2e6b 7364 2e73 7667  orrected.ksd.svg
+000156a0: 290a 0a41 6e20 616c 7465 726e 6174 6976  )..An alternativ
+000156b0: 6520 7761 7920 746f 2063 616c 6375 6c61  e way to calcula
+000156c0: 7465 2074 6865 206f 7274 686f 6c6f 676f  te the orthologo
+000156d0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+000156e0: 623e 2069 7320 746f 2064 6972 6563 746c  b> is to directl
+000156f0: 7920 7573 6520 7468 6520 6f72 7468 6f67  y use the orthog
+00015700: 726f 7570 7320 696e 7374 6561 6420 6f66  roups instead of
+00015710: 2067 6c6f 6261 6c20 4d52 4248 2066 616d   global MRBH fam
+00015720: 696c 792e 2054 6861 7420 7761 7920 7765  ily. That way we
+00015730: 2064 6f6e 2774 2075 7365 2074 6865 2070   don't use the p
+00015740: 7265 2d69 6e66 6572 7265 6420 7061 7261  re-inferred para
+00015750: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00015760: 7375 623e 2062 7574 2074 6865 2070 6172  sub> but the par
+00015770: 616c 6f67 6f75 7320 6765 6e65 2070 6169  alogous gene pai
+00015780: 7273 2069 6e73 6964 6520 6561 6368 206f  rs inside each o
+00015790: 7274 686f 6772 6f75 7020 696e 7374 6561  rthogroup instea
+000157a0: 642e 2054 6f20 6163 6869 6576 6520 7468  d. To achieve th
+000157b0: 6174 2c20 7765 2066 6972 7374 206e 6565  at, we first nee
+000157c0: 6420 746f 2069 6e66 6572 206f 7274 686f  d to infer ortho
+000157d0: 6772 6f75 7073 2075 7369 6e67 2074 6865  groups using the
+000157e0: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
+000157f0: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00015800: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00015810: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+00015820: 7320 4163 6f72 7573 5f61 6d65 7269 6361  s Acorus_america
+00015830: 6e75 7320 5669 7469 735f 7669 6e69 6665  nus Vitis_vinife
+00015840: 7261 202d 2d6f 7274 686f 696e 6665 7220  ra --orthoinfer 
+00015850: 2d6f 2077 6764 5f6f 7274 686f 2028 2d2d  -o wgd_ortho (--
+00015860: 6f6e 6c79 6f72 7468 6f29 200a 6060 600a  onlyortho) .```.
+00015870: 0a55 7365 7273 2063 616e 2064 6563 6964  .Users can decid
+00015880: 6520 746f 206f 6e6c 7920 636f 6e64 7563  e to only conduc
+00015890: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
+000158a0: 2061 6e61 6c79 7369 7320 7768 696c 6520   analysis while 
+000158b0: 736b 6970 7069 6e67 206f 7468 6572 2061  skipping other a
+000158c0: 6e61 6c79 7369 7320 6279 2061 6464 696e  nalysis by addin
+000158d0: 6720 7468 6520 666c 6167 2060 2d2d 6f6e  g the flag `--on
+000158e0: 6c79 6f72 7468 6f60 2e20 4e65 7874 2073  lyortho`. Next s
+000158f0: 7465 7020 6973 2074 6865 2073 616d 6520  tep is the same 
+00015900: 7769 7468 2067 6c6f 6261 6c20 4d52 4248  with global MRBH
+00015910: 2066 616d 696c 7920 6578 6365 7074 2074   family except t
+00015920: 6861 7420 7765 2064 6f6e 2774 2075 7365  hat we don't use
+00015930: 2074 6865 2065 7874 7261 2070 7265 2d69   the extra pre-i
+00015940: 6e66 6572 7265 6420 7061 7261 6e6f 6d65  nferred paranome
+00015950: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00015960: 2061 6e79 6d6f 7265 2e20 5765 2069 6e66   anymore. We inf
+00015970: 6572 2074 6865 202a 4b2a 3c73 7562 3e53  er the *K*<sub>S
+00015980: 3c2f 7375 623e 2075 7369 6e67 2074 6865  </sub> using the
+00015990: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e20   command below. 
+000159a0: 4e6f 7465 2074 6861 7420 7468 6520 7072  Note that the pr
+000159b0: 6f67 7261 6d20 6077 6764 2076 697a 6020  ogram `wgd viz` 
+000159c0: 6361 6e20 706c 6f74 2074 6865 2073 616d  can plot the sam
+000159d0: 6520 6a75 7374 2061 7320 7368 6f77 6e20  e just as shown 
+000159e0: 6162 6f76 652e 0a0a 6060 600a 7767 6420  above...```.wgd 
+000159f0: 6b73 6420 7767 645f 6f72 7468 6f2f 4f72  ksd wgd_ortho/Or
+00015a00: 7468 6f67 726f 7570 732e 7370 2e74 7376  thogroups.sp.tsv
+00015a10: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00015a20: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00015a30: 6f20 7767 645f 6f72 7468 6f5f 6b73 202d  o wgd_ortho_ks -
+00015a40: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00015a50: 615f 636f 6572 756c 6561 3b50 726f 7465  a_coerulea;Prote
+00015a60: 615f 6379 6e61 726f 6964 6573 2220 2d2d  a_cynaroides" --
+00015a70: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
+00015a80: 5f63 6f65 7275 6c65 613b 5669 7469 735f  _coerulea;Vitis_
+00015a90: 7669 6e69 6665 7261 2220 2d2d 7370 6169  vinifera" --spai
+00015aa0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
+00015ab0: 7275 6c65 613b 4163 6f72 7573 5f61 6d65  rulea;Acorus_ame
+00015ac0: 7269 6361 6e75 7322 202d 2d73 7061 6972  ricanus" --spair
+00015ad0: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
+00015ae0: 756c 6561 3b41 7175 696c 6567 6961 5f63  ulea;Aquilegia_c
+00015af0: 6f65 7275 6c65 6122 202d 2d70 6c6f 746b  oerulea" --plotk
+00015b00: 6465 2028 2d61 7020 7767 645f 7379 6e2f  de (-ap wgd_syn/
+00015b10: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
+00015b20: 6f72 706f 696e 7473 2e74 7874 290a 6060  orpoints.txt).``
+00015b30: 600a 0a21 5b5d 2864 6174 612f 4171 7569  `..![](data/Aqui
+00015b40: 6c65 6769 615f 636f 6572 756c 6561 5f4f  legia_coerulea_O
+00015b50: 7274 686f 4b73 5f43 6f72 7265 6374 6564  rthoKs_Corrected
+00015b60: 2e6b 7364 2e73 7667 290a 0a41 7320 7368  .ksd.svg)..As sh
+00015b70: 6f77 6e20 6162 6f76 652c 2074 6865 206e  own above, the n
+00015b80: 756d 6265 7220 6f66 2062 6f74 6820 7061  umber of both pa
+00015b90: 7261 6c6f 676f 7573 2067 656e 6520 7061  ralogous gene pa
+00015ba0: 6972 7320 616e 6420 6f72 7468 6f6c 6f67  irs and ortholog
+00015bb0: 6f75 7320 6765 6e65 2070 6169 7273 2069  ous gene pairs i
+00015bc0: 7320 6469 6666 6572 656e 7420 7468 616e  s different than
+00015bd0: 2074 6865 206f 6e65 2066 726f 6d20 676c   the one from gl
+00015be0: 6f62 616c 204d 5242 4820 6661 6d69 6c79  obal MRBH family
+00015bf0: 2069 6e20 7468 6174 2068 6572 6520 7765   in that here we
+00015c00: 2070 6c6f 7474 6564 2061 6c6c 206f 7274   plotted all ort
+00015c10: 686f 6c6f 676f 7573 2067 656e 6520 7061  hologous gene pa
+00015c20: 6972 7320 696e 7374 6561 6420 6f66 206f  irs instead of o
+00015c30: 6e6c 7920 676c 6f62 616c 204d 5242 4820  nly global MRBH 
+00015c40: 616e 6420 706f 7465 6e74 6961 6c6c 7920  and potentially 
+00015c50: 6e65 7720 7061 7261 6c6f 676f 7573 2067  new paralogous g
+00015c60: 656e 6520 7061 6972 7320 6d69 6768 7420  ene pairs might 
+00015c70: 6265 2070 726f 6475 6365 6420 696e 2074  be produced in t
+00015c80: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00015c90: 6665 7265 6e63 6520 7374 6570 2c20 746f  ference step, to
+00015ca0: 6765 7468 6572 2077 6974 6820 6469 6666  gether with diff
+00015cb0: 6572 656e 7420 7265 6361 6c63 756c 6174  erent recalculat
+00015cc0: 6564 2077 6569 6768 7473 2e0a 0a49 6620  ed weights...If 
+00015cd0: 6f6e 6520 6f6e 6c79 2077 616e 7465 6420  one only wanted 
+00015ce0: 746f 2070 6c6f 7420 7468 6520 6f72 7468  to plot the orth
+00015cf0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00015d00: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00015d10: 7469 6f6e 2c20 6173 2077 6520 6861 7665  tion, as we have
+00015d20: 2061 6c72 6561 6479 2073 686f 776e 2061   already shown a
+00015d30: 626f 7665 2c20 6974 2063 616e 2062 6520  bove, it can be 
+00015d40: 6561 7369 6c79 2061 6368 6965 7665 6420  easily achieved 
+00015d50: 6279 2072 656d 6f76 696e 6720 7468 6520  by removing the 
+00015d60: 7061 7261 6c6f 676f 7573 2073 7065 6369  paralogous speci
+00015d70: 6573 2070 6169 7220 6041 7175 696c 6567  es pair `Aquileg
+00015d80: 6961 5f63 6f65 7275 6c65 613b 4171 7569  ia_coerulea;Aqui
+00015d90: 6c65 6769 615f 636f 6572 756c 6561 602c  legia_coerulea`,
+00015da0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
+00015db0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
+00015dc0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00015dd0: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00015de0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00015df0: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00015e00: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00015e10: 202d 6f20 7767 645f 676c 6f62 616c 6d72   -o wgd_globalmr
+00015e20: 6268 5f6f 6e6c 796f 7274 686f 5f6b 7320  bh_onlyortho_ks 
+00015e30: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00015e40: 6961 5f63 6f65 7275 6c65 613b 5072 6f74  ia_coerulea;Prot
+00015e50: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
+00015e60: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00015e70: 615f 636f 6572 756c 6561 3b56 6974 6973  a_coerulea;Vitis
+00015e80: 5f76 696e 6966 6572 6122 202d 2d73 7061  _vinifera" --spa
+00015e90: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
+00015ea0: 6572 756c 6561 3b41 636f 7275 735f 616d  erulea;Acorus_am
+00015eb0: 6572 6963 616e 7573 2220 2d2d 6773 6d61  ericanus" --gsma
+00015ec0: 7020 6765 6e65 5f73 7065 6369 6573 2e6d  p gene_species.m
+00015ed0: 6170 0a60 6060 0a0a 5765 2063 616e 2063  ap.```..We can c
+00015ee0: 6c65 6172 6c79 2073 6565 2074 6861 7420  learly see that 
+00015ef0: 2a56 6974 6973 2076 696e 6966 6572 612a  *Vitis vinifera*
+00015f00: 2068 6173 2068 6967 6865 7220 7375 6273   has higher subs
+00015f10: 7469 7475 7469 6f6e 2072 6174 6520 7468  titution rate th
+00015f20: 616e 202a 5072 6f74 6561 2063 796e 6172  an *Protea cynar
+00015f30: 6f69 6465 732a 2069 6e20 7468 6174 2074  oides* in that t
+00015f40: 6865 6972 206f 7274 686f 6c6f 676f 7573  heir orthologous
+00015f50: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00015f60: 2070 6561 6b73 2077 6974 6820 2a41 7175   peaks with *Aqu
+00015f70: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+00015f80: 2c20 616c 7468 6f75 6768 2072 6570 7265  , although repre
+00015f90: 7365 6e74 696e 6720 7468 6520 7361 6d65  senting the same
+00015fa0: 2064 6976 6572 6765 6e63 6520 6576 656e   divergence even
+00015fb0: 742c 2064 6966 6665 7265 6420 696e 2073  t, differed in s
+00015fc0: 7562 7374 6974 7574 696f 6e20 6469 7374  ubstitution dist
+00015fd0: 616e 6365 2e0a 215b 5d28 6461 7461 2f52  ance..![](data/R
+00015fe0: 6177 5f4f 7274 686f 6c6f 6775 6573 2e6b  aw_Orthologues.k
+00015ff0: 7364 2e73 7667 290a 0a41 6674 6572 2074  sd.svg)..After t
+00016000: 6865 2070 6879 6c6f 6765 6e65 7469 6320  he phylogenetic 
+00016010: 7469 6d69 6e67 206f 6620 7468 6520 5261  timing of the Ra
+00016020: 6e75 6e63 756c 616c 6573 2057 4744 2c20  nunculales WGD, 
+00016030: 7765 2063 616e 2066 7572 7468 6572 2069  we can further i
+00016040: 6e66 6572 2069 7473 2061 6273 6f6c 7574  nfer its absolut
+00016050: 6520 6167 652e 2046 6972 7374 2077 6520  e age. First we 
+00016060: 696e 6665 7220 7468 6520 6372 6564 6962  infer the credib
+00016070: 6c65 2072 616e 6765 206f 6620 616e 6368  le range of anch
+00016080: 6f72 2070 6169 7273 2062 7920 2a4b 2a3c  or pairs by *K*<
+00016090: 7375 623e 533c 2f73 7562 3e20 6865 7572  sub>S</sub> heur
+000160a0: 6973 7469 6361 6c6c 7920 7573 696e 6720  istically using 
+000160b0: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
+000160c0: 2070 6561 6b60 2e0a 0a60 6060 0a77 6764   peak`...```.wgd
+000160d0: 2070 6561 6b20 2d2d 6865 7572 6973 7469   peak --heuristi
+000160e0: 6320 7767 645f 6b73 642f 4171 7569 6c65  c wgd_ksd/Aquile
+000160f0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00016100: 2e6b 732e 7473 7620 2d61 7020 7767 645f  .ks.tsv -ap wgd_
+00016110: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00016120: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
+00016130: 202d 736d 2077 6764 5f73 796e 2f69 6164   -sm wgd_syn/iad
+00016140: 686f 7265 2d6f 7574 2f73 6567 6d65 6e74  hore-out/segment
+00016150: 732e 7478 7420 2d6c 6520 7767 645f 7379  s.txt -le wgd_sy
+00016160: 6e2f 6961 6468 6f72 652d 6f75 742f 6c69  n/iadhore-out/li
+00016170: 7374 5f65 6c65 6d65 6e74 732e 7478 7420  st_elements.txt 
+00016180: 2d6d 7020 7767 645f 7379 6e2f 6961 6468  -mp wgd_syn/iadh
+00016190: 6f72 652d 6f75 742f 6d75 6c74 6970 6c69  ore-out/multipli
+000161a0: 636f 6e5f 7061 6972 732e 7478 7420 2d6f  con_pairs.txt -o
+000161b0: 2077 6764 5f70 6561 6b0a 6060 600a 0a21   wgd_peak.```..!
+000161c0: 5b5d 2864 6174 612f 416e 6368 6f72 4b73  [](data/AnchorKs
+000161d0: 5f50 6561 6b43 495f 4171 7569 6c65 6769  _PeakCI_Aquilegi
+000161e0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+000161f0: 732e 7473 765f 6e6f 6465 5f77 6569 6768  s.tsv_node_weigh
+00016200: 7465 642e 7376 6729 0a0a 4173 2073 686f  ted.svg)..As sho
+00016210: 776e 2061 626f 7665 2c20 7765 2061 7373  wn above, we ass
+00016220: 756d 6564 2061 206c 6f67 6e6f 726d 616c  umed a lognormal
+00016230: 2064 6973 7472 6962 7574 696f 6e20 6174   distribution at
+00016240: 2074 6865 2070 6561 6b20 6c6f 6361 7469   the peak locati
+00016250: 6f6e 2064 6574 6563 7465 6420 6279 2074  on detected by t
+00016260: 6865 2060 7369 676e 616c 6020 6d6f 6475  he `signal` modu
+00016270: 6c65 206f 6620 6073 6369 7079 6020 6c69  le of `scipy` li
+00016280: 6272 6172 792e 2054 6865 2039 3525 2063  brary. The 95% c
+00016290: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
+000162a0: 6f66 2074 6865 206c 6f67 6e6f 726d 616c  of the lognormal
+000162b0: 2064 6973 7472 6962 7574 696f 6e20 7761   distribution wa
+000162c0: 7320 6170 706c 6965 642c 2069 2e65 2e2c  s applied, i.e.,
+000162d0: 2030 2e36 382d 322e 3734 2c20 696e 2066   0.68-2.74, in f
+000162e0: 7572 7468 6572 206d 6f6c 6563 756c 6172  urther molecular
+000162f0: 2064 6174 696e 672e 2054 6865 2066 696c   dating. The fil
+00016300: 6520 6041 7175 696c 6567 6961 5f63 6f65  e `Aquilegia_coe
+00016310: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00016320: 5f39 3525 4349 5f41 505f 666f 725f 6461  _95%CI_AP_for_da
+00016330: 7469 6e67 5f77 6569 6768 7465 645f 666f  ting_weighted_fo
+00016340: 726d 6174 2e74 7376 6020 6973 2077 6861  rmat.tsv` is wha
+00016350: 7420 7765 206e 6565 6420 666f 7220 6e65  t we need for ne
+00016360: 7874 2073 7465 702e 2054 6f20 6275 696c  xt step. To buil
+00016370: 6420 7468 6520 6f72 7468 6f67 726f 7570  d the orthogroup
+00016380: 7320 7573 6564 2069 6e20 7068 796c 6f67  s used in phylog
+00016390: 656e 6574 6963 2064 6174 696e 672c 2077  enetic dating, w
+000163a0: 6520 6e65 6564 2074 6f20 7365 6c65 6374  e need to select
+000163b0: 2073 6f6d 6520 7370 6563 6965 7320 616e   some species an
+000163c0: 6420 666f 726d 2061 2073 7461 7274 696e  d form a startin
+000163d0: 6720 7472 6565 2077 6974 6820 7072 6f70  g tree with prop
+000163e0: 6572 2066 6f73 7369 6c20 6361 6c69 6272  er fossil calibr
+000163f0: 6174 696f 6e73 2e20 5765 2070 726f 7669  ations. We provi
+00016400: 6465 206f 6e65 2069 6e20 6d63 6d63 7472  de one in mcmctr
+00016410: 6565 2066 6f72 6d61 7420 6173 2062 656c  ee format as bel
+00016420: 6f77 2e0a 0a60 6060 0a31 3720 310a 2828  ow...```.17 1.((
+00016430: 2828 506f 7461 6d6f 6765 746f 6e5f 6163  ((Potamogeton_ac
+00016440: 7574 6966 6f6c 6975 732c 2853 7069 726f  utifolius,(Spiro
+00016450: 6465 6c61 5f69 6e74 6572 6d65 6469 612c  dela_intermedia,
+00016460: 416d 6f72 7068 6f70 6861 6c6c 7573 5f6b  Amorphophallus_k
+00016470: 6f6e 6a61 6329 292c 2841 6361 6e74 686f  onjac)),(Acantho
+00016480: 6368 6c61 6d79 735f 6272 6163 7465 6174  chlamys_bracteat
+00016490: 612c 2844 696f 7363 6f72 6561 5f61 6c61  a,(Dioscorea_ala
+000164a0: 7461 2c44 696f 7363 6f72 6561 5f72 6f74  ta,Dioscorea_rot
+000164b0: 756e 6461 7461 2929 273e 302e 3536 3030  undata))'>0.5600
+000164c0: 3c31 2e32 3836 3327 2927 3e30 2e38 3336  <1.2863')'>0.836
+000164d0: 303c 312e 3238 3633 272c 2841 636f 7275  0<1.2863',(Acoru
+000164e0: 735f 616d 6572 6963 616e 7573 2c41 636f  s_americanus,Aco
+000164f0: 7275 735f 7461 7461 7269 6e6f 7769 6929  rus_tatarinowii)
+00016500: 2927 3e30 2e38 3336 303c 312e 3238 3633  )'>0.8360<1.2863
+00016510: 272c 2828 2828 5465 7472 6163 656e 7472  ',((((Tetracentr
+00016520: 6f6e 5f73 696e 656e 7365 2c54 726f 6368  on_sinense,Troch
+00016530: 6f64 656e 6472 6f6e 5f61 7261 6c69 6f69  odendron_aralioi
+00016540: 6465 7329 2c28 4275 7875 735f 6175 7374  des),(Buxus_aust
+00016550: 726f 7975 6e6e 616e 656e 7369 732c 4275  royunnanensis,Bu
+00016560: 7875 735f 7369 6e69 6361 2929 273e 312e  xus_sinica))'>1.
+00016570: 3130 3830 3c31 2e32 3836 3327 2c28 4e65  1080<1.2863',(Ne
+00016580: 6c75 6d62 6f5f 6e75 6369 6665 7261 2c28  lumbo_nucifera,(
+00016590: 5465 6c6f 7065 615f 7370 6563 696f 7369  Telopea_speciosi
+000165a0: 7373 696d 612c 5072 6f74 6561 5f63 796e  ssima,Protea_cyn
+000165b0: 6172 6f69 6465 7329 2929 273e 312e 3130  aroides)))'>1.10
+000165c0: 3830 3c31 2e32 3836 3327 2c28 4171 7569  80<1.2863',(Aqui
+000165d0: 6c65 6769 615f 636f 6572 756c 6561 5f61  legia_coerulea_a
+000165e0: 7031 2c41 7175 696c 6567 6961 5f63 6f65  p1,Aquilegia_coe
+000165f0: 7275 6c65 615f 6170 3229 2927 3e31 2e31  rulea_ap2))'>1.1
+00016600: 3038 303c 312e 3238 3633 2729 273e 312e  080<1.2863')'>1.
+00016610: 3237 3230 3c32 2e34 3732 3027 3b0a 6060  2720<2.4720';.``
+00016620: 600a 0a41 7320 7072 6573 656e 7465 6420  `..As presented 
+00016630: 6162 6f76 652c 2074 6865 2066 6f63 7573  above, the focus
+00016640: 2073 7065 6369 6573 2074 6861 7420 6973   species that is
+00016650: 2061 626f 7574 2074 6f20 6265 2064 6174   about to be dat
+00016660: 6564 206e 6565 6473 2074 6f20 6265 2072  ed needs to be r
+00016670: 6570 6c61 6365 6420 7769 7468 2060 2841  eplaced with `(A
+00016680: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00016690: 615f 6170 312c 4171 7569 6c65 6769 615f  a_ap1,Aquilegia_
+000166a0: 636f 6572 756c 6561 5f61 7032 2960 2e20  coerulea_ap2)`. 
+000166b0: 5769 7468 2074 6869 7320 7374 6172 7469  With this starti
+000166c0: 6e67 2074 7265 6520 616e 6420 7072 6564  ng tree and pred
+000166d0: 6f77 6e6c 6f61 6465 6420 6364 7320 6669  ownloaded cds fi
+000166e0: 6c65 7320 6f66 2061 6c6c 2074 6865 2073  les of all the s
+000166f0: 7065 6369 6573 2c20 7765 2063 616e 2062  pecies, we can b
+00016700: 7569 6c64 2074 6865 206f 7274 686f 6772  uild the orthogr
+00016710: 6f75 7020 7573 6564 2069 6e20 7468 6520  oup used in the 
+00016720: 6669 6e61 6c20 6d6f 6c65 6375 6c61 7220  final molecular 
+00016730: 6461 7469 6e67 2075 7369 6e67 2074 6865  dating using the
+00016740: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
+00016750: 772e 0a0a 6060 600a 7767 6420 646d 6420  w...```.wgd dmd 
+00016760: 2d66 2041 7175 696c 6567 6961 5f63 6f65  -f Aquilegia_coe
+00016770: 7275 6c65 6120 2d61 7020 7767 645f 7065  rulea -ap wgd_pe
+00016780: 616b 2f41 7175 696c 6567 6961 5f63 6f65  ak/Aquilegia_coe
+00016790: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000167a0: 5f39 3525 4349 5f41 505f 666f 725f 6461  _95%CI_AP_for_da
+000167b0: 7469 6e67 5f77 6569 6768 7465 645f 666f  ting_weighted_fo
+000167c0: 726d 6174 2e74 7376 202d 6f20 7767 645f  rmat.tsv -o wgd_
+000167d0: 646d 645f 6f72 7468 6f20 506f 7461 6d6f  dmd_ortho Potamo
+000167e0: 6765 746f 6e5f 6163 7574 6966 6f6c 6975  geton_acutifoliu
+000167f0: 7320 5370 6972 6f64 656c 615f 696e 7465  s Spirodela_inte
+00016800: 726d 6564 6961 2041 6d6f 7270 686f 7068  rmedia Amorphoph
+00016810: 616c 6c75 735f 6b6f 6e6a 6163 2041 6361  allus_konjac Aca
+00016820: 6e74 686f 6368 6c61 6d79 735f 6272 6163  nthochlamys_brac
+00016830: 7465 6174 6120 4469 6f73 636f 7265 615f  teata Dioscorea_
+00016840: 616c 6174 6120 4469 6f73 636f 7265 615f  alata Dioscorea_
+00016850: 726f 7475 6e64 6174 6120 4163 6f72 7573  rotundata Acorus
+00016860: 5f61 6d65 7269 6361 6e75 7320 4163 6f72  _americanus Acor
+00016870: 7573 5f74 6174 6172 696e 6f77 6969 2054  us_tatarinowii T
+00016880: 6574 7261 6365 6e74 726f 6e5f 7369 6e65  etracentron_sine
+00016890: 6e73 6520 5472 6f63 686f 6465 6e64 726f  nse Trochodendro
+000168a0: 6e5f 6172 616c 696f 6964 6573 2042 7578  n_aralioides Bux
+000168b0: 7573 5f61 7573 7472 6f79 756e 6e61 6e65  us_austroyunnane
+000168c0: 6e73 6973 2042 7578 7573 5f73 696e 6963  nsis Buxus_sinic
+000168d0: 6120 4e65 6c75 6d62 6f5f 6e75 6369 6665  a Nelumbo_nucife
+000168e0: 7261 2054 656c 6f70 6561 5f73 7065 6369  ra Telopea_speci
+000168f0: 6f73 6973 7369 6d61 2050 726f 7465 615f  osissima Protea_
+00016900: 6379 6e61 726f 6964 6573 2041 7175 696c  cynaroides Aquil
+00016910: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
+00016920: 600a 0a54 6865 2072 6573 756c 7420 6669  `..The result fi
+00016930: 6c65 2060 6d65 7267 655f 666f 6375 735f  le `merge_focus_
+00016940: 6170 2e74 7376 6020 6973 2077 6861 7420  ap.tsv` is what 
+00016950: 7765 206e 6565 6420 666f 7220 7468 6520  we need for the 
+00016960: 6669 6e61 6c20 7374 6570 206f 6620 6d6f  final step of mo
+00016970: 6c65 6375 6c61 7220 6461 7469 6e67 2069  lecular dating i
+00016980: 6e20 7072 6f67 7261 6d20 6077 6764 2066  n program `wgd f
+00016990: 6f63 7573 602e 0a0a 6060 600a 7767 6420  ocus`...```.wgd 
+000169a0: 666f 6375 7320 2d2d 7072 6f74 6461 7469  focus --protdati
+000169b0: 6e67 202d 2d61 616d 6f64 656c 206c 6720  ng --aamodel lg 
+000169c0: 7767 645f 646d 645f 6f72 7468 6f2f 6d65  wgd_dmd_ortho/me
+000169d0: 7267 655f 666f 6375 735f 6170 2e74 7376  rge_focus_ap.tsv
+000169e0: 202d 7370 2064 6174 696e 675f 7472 6565   -sp dating_tree
+000169f0: 2e6e 7720 2d6f 2077 6764 5f64 6174 696e  .nw -o wgd_datin
+00016a00: 6720 2d64 206d 636d 6374 7265 6520 2d64  g -d mcmctree -d
+00016a10: 7320 2762 7572 6e69 6e20 3d20 3230 3030  s 'burnin = 2000
+00016a20: 2720 2d64 7320 2773 616d 7066 7265 7120  ' -ds 'sampfreq 
+00016a30: 3d20 3130 3030 2720 2d64 7320 276e 7361  = 1000' -ds 'nsa
+00016a40: 6d70 6c65 203d 2032 3030 3030 2720 506f  mple = 20000' Po
+00016a50: 7461 6d6f 6765 746f 6e5f 6163 7574 6966  tamogeton_acutif
+00016a60: 6f6c 6975 7320 5370 6972 6f64 656c 615f  olius Spirodela_
+00016a70: 696e 7465 726d 6564 6961 2041 6d6f 7270  intermedia Amorp
+00016a80: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
+00016a90: 2041 6361 6e74 686f 6368 6c61 6d79 735f   Acanthochlamys_
+00016aa0: 6272 6163 7465 6174 6120 4469 6f73 636f  bracteata Diosco
+00016ab0: 7265 615f 616c 6174 6120 4469 6f73 636f  rea_alata Diosco
+00016ac0: 7265 615f 726f 7475 6e64 6174 6120 4163  rea_rotundata Ac
+00016ad0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+00016ae0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
+00016af0: 6969 2054 6574 7261 6365 6e74 726f 6e5f  ii Tetracentron_
+00016b00: 7369 6e65 6e73 6520 5472 6f63 686f 6465  sinense Trochode
+00016b10: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
+00016b20: 2042 7578 7573 5f61 7573 7472 6f79 756e   Buxus_austroyun
+00016b30: 6e61 6e65 6e73 6973 2042 7578 7573 5f73  nanensis Buxus_s
+00016b40: 696e 6963 6120 4e65 6c75 6d62 6f5f 6e75  inica Nelumbo_nu
+00016b50: 6369 6665 7261 2054 656c 6f70 6561 5f73  cifera Telopea_s
+00016b60: 7065 6369 6f73 6973 7369 6d61 2050 726f  peciosissima Pro
+00016b70: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
+00016b80: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00016b90: 610a 6060 600a 0a48 6572 6520 7765 206f  a.```..Here we o
+00016ba0: 6e6c 7920 696d 706c 656d 656e 7465 6420  nly implemented 
+00016bb0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+00016bc0: 6e20 616e 616c 7973 6973 2075 7369 6e67  n analysis using
+00016bd0: 2070 726f 7465 696e 2073 6571 7565 6e63   protein sequenc
+00016be0: 6520 6279 2061 6464 696e 6720 7468 6520  e by adding the 
+00016bf0: 666c 6167 2060 2d2d 7072 6f74 6461 7469  flag `--protdati
+00016c00: 6e67 6020 616e 6420 7765 2073 6574 2074  ng` and we set t
+00016c10: 6865 2070 6172 616d 6574 6572 2066 6f72  he parameter for
+00016c20: 2060 6d63 6d63 7472 6565 6020 7669 6120   `mcmctree` via 
+00016c30: 7468 6520 6f70 7469 6f6e 2060 2d64 7360  the option `-ds`
+00016c40: 2e20 4e6f 7465 2074 6861 7420 6f74 6865  . Note that othe
+00016c50: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
+00016c60: 2073 7563 6820 6173 2060 7238 7360 2061   such as `r8s` a
+00016c70: 6e64 2060 6265 6173 7460 2061 7265 2061  nd `beast` are a
+00016c80: 6c73 6f20 6176 6169 6c61 626c 6520 6769  lso available gi
+00016c90: 7665 6e20 736f 6d65 206d 616e 6461 746f  ven some mandato
+00016ca0: 7279 2070 6172 616d 6574 6572 732e 2054  ry parameters. T
+00016cb0: 6865 2066 696e 616c 206c 6f67 206f 6620  he final log of 
+00016cc0: 7468 6520 7375 6363 6573 7366 756c 2072  the successful r
+00016cd0: 756e 2069 7320 6173 2062 656c 6f77 2e0a  un is as below..
+00016ce0: 0a60 6060 0a31 363a 3034 3a32 3520 494e  .```.16:04:25 IN
+00016cf0: 464f 2020 2020 2052 756e 6e69 6e67 206d  FO     Running m
+00016d00: 636d 6374 7265 6520 7573 696e 6720 4865  cmctree using He
+00016d10: 7373 6961 6e20 6d61 7472 6978 206f 6620  ssian matrix of 
+00016d20: 4c47 2b47 616d 6d61 2020 636f 7265 2e70  LG+Gamma  core.p
+00016d30: 793a 3936 370a 2020 2020 2020 2020 2020  y:967.          
+00016d40: 2020 2020 2020 2020 666f 7220 7072 6f74          for prot
+00016d50: 6569 6e20 6d6f 6465 6c0a 3233 3a34 393a  ein model.23:49:
+00016d60: 3337 2049 4e46 4f20 2020 2020 506f 7374  37 INFO     Post
+00016d70: 6572 696f 7220 6d65 616e 2066 6f72 2074  erior mean for t
+00016d80: 6865 2061 6765 7320 6f66 2077 6764 2069  he ages of wgd i
+00016d90: 7320 3131 322e 3839 3435 206d 636d 6374  s 112.8945 mcmct
+00016da0: 7265 652e 7079 3a32 3936 0a20 2020 2020  ree.py:296.     
+00016db0: 2020 2020 2020 2020 2020 2020 206d 696c               mil
+00016dc0: 6c69 6f6e 2079 6561 7273 2066 726f 6d20  lion years from 
+00016dd0: 436f 6e63 6174 656e 6174 6564 2070 6570  Concatenated pep
+00016de0: 7469 6465 0a20 2020 2020 2020 2020 2020  tide.           
+00016df0: 2020 2020 2020 2061 6c69 676e 6d65 6e74         alignment
+00016e00: 2061 6e64 2039 3525 2063 7265 6469 6269   and 95% credibi
+00016e10: 6c69 7479 2069 6e74 6572 7661 6c73 2028  lity intervals (
+00016e20: 4349 290a 2020 2020 2020 2020 2020 2020  CI).            
+00016e30: 2020 2020 2020 6973 2031 3031 2e32 3234        is 101.224
+00016e40: 2d31 3233 2e31 3231 206d 696c 6c69 6f6e  -123.121 million
+00016e50: 2079 6561 7273 0a20 2020 2020 2020 2020   years.         
+00016e60: 494e 464f 2020 2020 2054 6f74 616c 2072  INFO     Total r
+00016e70: 756e 2074 696d 653a 2032 3931 3735 7320  un time: 29175s 
+00016e80: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016e90: 2020 2020 2020 2020 2020 2020 2063 6c69               cli
+00016ea0: 2e70 793a 3234 310a 2020 2020 2020 2020  .py:241.        
+00016eb0: 2049 4e46 4f20 2020 2020 446f 6e65 2020   INFO     Done  
+00016ec0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016ed0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016ee0: 2020 2020 2020 2020 2020 2020 2020 636c                cl
+00016ef0: 692e 7079 3a32 3432 0a60 6060 0a0a 546f  i.py:242.```..To
+00016f00: 2076 6973 7561 6c69 7a65 2074 6865 2064   visualize the d
+00016f10: 6174 652c 2077 6520 616c 736f 2070 726f  ate, we also pro
+00016f20: 7669 6465 6420 6120 7079 7468 6f6e 2073  vided a python s
+00016f30: 6372 6970 7420 746f 2070 6c6f 7420 7468  cript to plot th
+00016f40: 6520 5747 4420 6461 7465 7320 696e 2074  e WGD dates in t
+00016f50: 6865 2060 7767 6460 2066 6f6c 6465 722e  he `wgd` folder.
+00016f60: 2055 7365 7273 206e 6565 6420 746f 2065   Users need to e
+00016f70: 7874 7261 6374 2074 6865 2072 6177 2064  xtract the raw d
+00016f80: 6174 6573 2066 726f 6d20 7468 6520 606d  ates from the `m
+00016f90: 636d 632e 7478 7460 2066 6f72 2074 6865  cmc.txt` for the
+00016fa0: 2057 4744 206e 6f64 6520 6669 7273 7420   WGD node first 
+00016fb0: 616e 6420 7361 7665 2069 7420 6173 2066  and save it as f
+00016fc0: 696c 6520 6064 6174 6573 2e74 7874 6020  ile `dates.txt` 
+00016fd0: 286f 7220 7768 6174 6576 6572 2070 7265  (or whatever pre
+00016fe0: 6665 7272 6564 206e 616d 6529 2e20 416e  ferred name). An
+00016ff0: 2065 7861 6d70 6c65 2063 6f6d 6d61 6e64   example command
+00017000: 2069 7320 6173 2062 656c 6f77 2e0a 0a60   is as below...`
+00017010: 6060 0a70 7974 686f 6e20 2450 4154 482f  ``.python $PATH/
+00017020: 706f 7374 706c 6f74 2e70 7920 706f 7374  postplot.py post
+00017030: 6469 7320 6461 7465 732e 7478 7420 2d2d  dis dates.txt --
+00017040: 7065 7263 656e 7469 6c65 2039 3020 2d2d  percentile 90 --
+00017050: 7469 746c 6520 2257 4744 2064 6174 6522  title "WGD date"
+00017060: 202d 2d68 7064 202d 6f20 2252 616e 756e   --hpd -o "Ranun
+00017070: 6375 6c61 6c65 735f 5747 445f 6461 7465  culales_WGD_date
+00017080: 2e73 7667 220a 6060 600a 0a21 5b5d 2864  .svg".```..![](d
+00017090: 6174 612f 5261 6e75 6e63 756c 616c 6573  ata/Ranunculales
+000170a0: 5f57 4744 5f64 6174 652e 7376 6729 0a0a  _WGD_date.svg)..
+000170b0: 5468 6520 706f 7374 6572 696f 7220 6d65  The posterior me
+000170c0: 616e 2c20 6d65 6469 616e 2061 6e64 206d  an, median and m
+000170d0: 6f64 6520 6f66 2074 6865 2052 616e 756e  ode of the Ranun
+000170e0: 6375 6c61 6c65 7320 5747 4420 6167 6520  culales WGD age 
+000170f0: 6973 2031 3132 2e39 322c 2031 3133 2e34  is 112.92, 113.4
+00017100: 3420 616e 6420 3131 322e 3534 206d 7961  4 and 112.54 mya
+00017110: 2c20 7769 7468 2039 3025 2048 5044 2031  , with 90% HPD 1
+00017120: 3035 2e30 3720 2d20 3132 322e 3332 206d  05.07 - 122.32 m
+00017130: 7961 2061 7320 6d61 6e69 6665 7374 6564  ya as manifested
+00017140: 2061 626f 7665 2e0a 0a23 2323 204b 7374   above...### Kst
+00017150: 7265 650a 0a49 6e20 6164 6469 7469 6f6e  ree..In addition
+00017160: 2074 6f20 7061 6972 7769 7365 202a 4b2a   to pairwise *K*
+00017170: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+00017180: 696d 6174 696f 6e2c 2061 202a 4b2a 3c73  imation, a *K*<s
+00017190: 7562 3e53 3c2f 7375 623e 2074 7265 6520  ub>S</sub> tree 
+000171a0: 7769 7468 2062 7261 6e63 6820 6c65 6e67  with branch leng
+000171b0: 7468 2069 6e20 2a4b 2a3c 7375 623e 533c  th in *K*<sub>S<
+000171c0: 2f73 7562 3e20 756e 6974 2063 616e 2061  /sub> unit can a
+000171d0: 6c73 6f20 6265 2064 6572 6976 6564 2066  lso be derived f
+000171e0: 726f 6d20 7468 6520 7072 6f67 7261 6d20  rom the program 
+000171f0: 6077 6764 206b 7364 6020 6769 7665 6e20  `wgd ksd` given 
+00017200: 7468 6520 6f70 7469 6f6e 2060 2d2d 6b73  the option `--ks
+00017210: 7472 6565 6020 616e 6420 602d 2d73 7065  tree` and `--spe
+00017220: 6369 6573 7472 6565 602e 204e 6f74 6520  ciestree`. Note 
+00017230: 7468 6174 2074 6865 2061 6464 6974 696f  that the additio
+00017240: 6e61 6c20 6f70 7469 6f6e 2060 2d2d 6f6e  nal option `--on
+00017250: 6c79 636f 6e63 6174 6b73 7472 6565 6020  lyconcatkstree` 
+00017260: 7769 6c6c 206f 6e6c 7920 6361 6c6c 2074  will only call t
+00017270: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+00017280: 623e 2065 7374 696d 6174 696f 6e20 666f  b> estimation fo
+00017290: 7220 7468 6520 636f 6e63 6174 656e 6174  r the concatenat
+000172a0: 6564 2061 6c69 676e 6d65 6e74 2072 6174  ed alignment rat
+000172b0: 6865 7220 7468 616e 2061 6c6c 2074 6865  her than all the
+000172c0: 2061 6c69 676e 6d65 6e74 732e 2055 7365   alignments. Use
+000172d0: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
+000172e0: 6465 2061 2070 7265 7365 7420 7370 6563  de a preset spec
+000172f0: 6965 7320 7472 6565 2066 6f72 2074 6865  ies tree for the
+00017300: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00017310: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+00017320: 6f66 2074 6865 2063 6f6e 6361 7465 6e61  of the concatena
+00017330: 7465 6420 616c 6967 6e6d 656e 7420 7768  ted alignment wh
+00017340: 696c 6520 7468 6520 7265 6d61 696e 696e  ile the remainin
+00017350: 6720 616c 6967 6e6d 656e 7473 2077 696c  g alignments wil
+00017360: 6c20 6265 2061 6761 696e 7374 2061 6e20  l be against an 
+00017370: 6175 746f 6d61 7465 6c79 2069 6e66 6572  automately infer
+00017380: 7265 6420 7472 6565 2066 726f 6d20 6066  red tree from `f
+00017390: 6173 7474 7265 6560 206f 7220 6069 7174  asttree` or `iqt
+000173a0: 7265 6560 2e20 496e 2074 6865 2065 6e64  ree`. In the end
+000173b0: 2c20 7573 6572 7320 7769 6c6c 2067 6574  , users will get
+000173c0: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
+000173d0: 623e 2074 7265 652c 2061 202a 4b2a 3c73  b> tree, a *K*<s
+000173e0: 7562 3e41 3c2f 7375 623e 2074 7265 6520  ub>A</sub> tree 
+000173f0: 616e 6420 6120 cf89 2074 7265 6520 7065  and a .. tree pe
+00017400: 7220 6661 6d20 616e 6420 666f 7220 7468  r fam and for th
+00017410: 6520 636f 6e63 6174 656e 6174 6564 2061  e concatenated a
+00017420: 6c69 676e 6d65 6e74 2e0a 0a60 6060 0a77  lignment...```.w
+00017430: 6764 206b 7364 2064 6174 612f 6b73 7472  gd ksd data/kstr
+00017440: 6565 5f64 6174 612f 6661 6d2e 7473 7620  ee_data/fam.tsv 
+00017450: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
+00017460: 2f41 636f 7275 735f 7461 7461 7269 6e6f  /Acorus_tatarino
+00017470: 7769 6920 6461 7461 2f6b 7374 7265 655f  wii data/kstree_
+00017480: 6461 7461 2f41 6d62 6f72 656c 6c61 5f74  data/Amborella_t
+00017490: 7269 6368 6f70 6f64 6120 6461 7461 2f6b  richopoda data/k
+000174a0: 7374 7265 655f 6461 7461 2f41 7175 696c  stree_data/Aquil
+000174b0: 6567 6961 5f63 6f65 7275 6c65 6120 6461  egia_coerulea da
+000174c0: 7461 2f6b 7374 7265 655f 6461 7461 2f41  ta/kstree_data/A
+000174d0: 7269 7374 6f6c 6f63 6869 615f 6669 6d62  ristolochia_fimb
+000174e0: 7269 6174 6120 6461 7461 2f6b 7374 7265  riata data/kstre
+000174f0: 655f 6461 7461 2f43 7963 6173 5f70 616e  e_data/Cycas_pan
+00017500: 7a68 6968 7561 656e 7369 202d 2d6b 7374  zhihuaensi --kst
+00017510: 7265 6520 2d2d 7370 6563 6965 7374 7265  ree --speciestre
+00017520: 6520 6461 7461 2f6b 7374 7265 655f 6461  e data/kstree_da
+00017530: 7461 2f73 7065 6369 6573 5f74 7265 6531  ta/species_tree1
+00017540: 2e6e 7720 2d2d 6f6e 6c79 636f 6e63 6174  .nw --onlyconcat
+00017550: 6b73 7472 6565 202d 6f20 7767 645f 6b73  kstree -o wgd_ks
+00017560: 7472 6565 5f74 6f70 6f6c 6f67 7931 0a60  tree_topology1.`
+00017570: 6060 0a0a 215b 5d28 6461 7461 2f6b 7374  ``..![](data/kst
+00017580: 7265 655f 7265 7375 6c74 732f 6b73 7472  ree_results/kstr
+00017590: 6565 2e73 7667 290a 0a41 626f 7665 2077  ee.svg)..Above w
+000175a0: 6520 7573 6564 2074 6872 6565 2061 6c74  e used three alt
+000175b0: 6572 6e61 7469 7665 2074 6f70 6f6c 6f67  ernative topolog
+000175c0: 6965 7320 746f 2069 6e66 6572 2074 6865  ies to infer the
+000175d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000175e0: 2074 7265 6520 7768 6963 6820 6c65 6420   tree which led 
+000175f0: 746f 2064 6966 6665 7265 6e74 2062 7261  to different bra
+00017600: 6e63 6820 6c65 6e67 7468 2065 7374 696d  nch length estim
+00017610: 6174 696f 6e2e 204e 6f74 6520 7468 6174  ation. Note that
+00017620: 2074 6865 2066 616d 696c 6965 7320 7765   the families we
+00017630: 2075 7365 6420 7765 7265 206f 6e6c 7920   used were only 
+00017640: 7477 6f20 676c 6f62 616c 204d 5242 4820  two global MRBH 
+00017650: 6661 6d69 6c69 6573 2066 6f72 2074 6865  families for the
+00017660: 2070 7572 706f 7365 206f 6620 696c 6c75   purpose of illu
+00017670: 7374 7261 7469 6f6e 2e20 546f 2061 6371  stration. To acq
+00017680: 7569 7265 2061 6e20 6163 6375 7261 7465  uire an accurate
+00017690: 2070 726f 6669 6c65 206f 6620 7468 6520   profile of the 
+000176a0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+000176b0: 6520 7661 7269 6174 696f 6e2c 206f 7274  e variation, ort
+000176c0: 686f 6c6f 6775 6573 2061 7420 7468 6520  hologues at the 
+000176d0: 7768 6f6c 6520 6765 6e6f 6d65 2073 6361  whole genome sca
+000176e0: 6c65 2073 686f 756c 6420 6265 2075 7365  le should be use
+000176f0: 642e 0a0a 2323 2043 6974 6174 696f 6e0a  d...## Citation.
+00017700: 200a 506c 6561 7365 2063 6974 6520 7573   .Please cite us
+00017710: 2061 7420 6874 7470 733a 2f2f 646f 692e   at https://doi.
+00017720: 6f72 672f 3130 2e31 3030 372f 3937 382d  org/10.1007/978-
+00017730: 312d 3037 3136 2d32 3536 312d 335f 3120  1-0716-2561-3_1 
+00017740: 616e 6420 6874 7470 733a 2f2f 646f 692e  and https://doi.
+00017750: 6f72 672f 3130 2e31 3039 332f 6269 6f69  org/10.1093/bioi
+00017760: 6e66 6f72 6d61 7469 6373 2f62 7461 6532  nformatics/btae2
+00017770: 3732 2e0a 0a60 6060 0a48 656e 6763 6869  72...```.Hengchi
+00017780: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
+00017790: 6165 6e65 706f 656c 2028 3230 3233 292e  aenepoel (2023).
+000177a0: 2049 6e66 6572 656e 6365 206f 6620 416e   Inference of An
+000177b0: 6369 656e 7420 506f 6c79 706c 6f69 6479  cient Polyploidy
+000177c0: 2066 726f 6d20 4765 6e6f 6d69 6320 4461   from Genomic Da
+000177d0: 7461 2e20 496e 3a20 5661 6e20 6465 2050  ta. In: Van de P
+000177e0: 6565 722c 2059 2e20 2865 6473 2920 506f  eer, Y. (eds) Po
+000177f0: 6c79 706c 6f69 6479 2e20 4d65 7468 6f64  lyploidy. Method
+00017800: 7320 696e 204d 6f6c 6563 756c 6172 2042  s in Molecular B
+00017810: 696f 6c6f 6779 2c20 766f 6c20 3235 3435  iology, vol 2545
+00017820: 2e20 4875 6d61 6e61 2c20 4e65 7720 596f  . Humana, New Yo
+00017830: 726b 2c20 4e59 2e20 6874 7470 733a 2f2f  rk, NY. https://
+00017840: 646f 692e 6f72 672f 3130 2e31 3030 372f  doi.org/10.1007/
+00017850: 3937 382d 312d 3037 3136 2d32 3536 312d  978-1-0716-2561-
+00017860: 335f 310a 4865 6e67 6368 6920 4368 656e  3_1.Hengchi Chen
+00017870: 2c20 4172 7468 7572 205a 7761 656e 6570  , Arthur Zwaenep
+00017880: 6f65 6c2c 2059 7665 7320 5661 6e20 6465  oel, Yves Van de
+00017890: 2050 6565 7220 2832 3032 3429 2e20 7767   Peer (2024). wg
+000178a0: 6420 7632 3a20 6120 7375 6974 6520 6f66  d v2: a suite of
+000178b0: 2074 6f6f 6c73 2074 6f20 756e 636f 7665   tools to uncove
+000178c0: 7220 616e 6420 6461 7465 2061 6e63 6965  r and date ancie
+000178d0: 6e74 2070 6f6c 7970 6c6f 6964 7920 616e  nt polyploidy an
+000178e0: 6420 7768 6f6c 652d 6765 6e6f 6d65 2064  d whole-genome d
+000178f0: 7570 6c69 6361 7469 6f6e 2e20 4269 6f69  uplication. Bioi
+00017900: 6e66 6f72 6d61 7469 6373 2c20 6874 7470  nformatics, http
+00017910: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
+00017920: 3039 332f 6269 6f69 6e66 6f72 6d61 7469  093/bioinformati
+00017930: 6373 2f62 7461 6532 3732 0a60 6060 0a0a  cs/btae272.```..
+00017940: 466f 7220 6369 7461 7469 6f6e 206f 6620  For citation of 
+00017950: 7468 6520 746f 6f6c 7320 7573 6564 2069  the tools used i
+00017960: 6e20 7767 642c 2070 6c65 6173 6520 636f  n wgd, please co
+00017970: 6e73 756c 7420 7468 6520 646f 6375 6d65  nsult the docume
+00017980: 6e74 6174 696f 6e20 6174 0a68 7474 7073  ntation at.https
+00017990: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
+000179a0: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
+000179b0: 742f 6369 7461 7469 6f6e 2e68 746d 6c2e  t/citation.html.
+000179c0: 0a0a                                     ..
```

### Comparing `wgd-2.0.32/cli.py` & `wgd-2.0.33/cli.py`

 * *Files 1% similar despite different names*

```diff
@@ -328,36 +328,34 @@
 @click.option('--boots', type=int, default=200, show_default=True, help="number of bootstrap replicates of kde")
 @click.option('--weighted', is_flag=True,help="node-weighted instead of node-averaged method")
 @click.option('--plot', '-p', type=click.Choice(['stacked', 'identical']), default='identical', show_default=True, help="plotting method")
 @click.option('--bw_method', '-bm', type=click.Choice(['silverman', 'ISJ']), default='silverman', show_default=True, help="bandwidth method")
 @click.option('--n_medoids', type=int, default=2, show_default=True, help="number of medoids to generate")
 @click.option('--kdemethod', '-km', type=click.Choice(['scipy', 'naivekde', 'treekde', 'fftkde']), default='scipy', show_default=True, help="kde method")
 @click.option('--n_clusters',type=int, default=5, show_default=True, help="number of clusters to plot Elbow loss function")
-@click.option('--kmedoids', is_flag=True,help="K-Medoids clustering method")
-@click.option('--guide', '-gd', type=click.Choice(['multiplicon', 'basecluster', 'segment']), default='segment', show_default=True, help="regime residing anchors")
+@click.option('--guide', '-gd', type=click.Choice(['Multiplicon','Segment']), default='Segment', show_default=True, help="regime residing anchors")
 @click.option('--prominence_cutoff', '-prct', type=float, default=0.1, show_default=True, help='prominence cutoff of acceptable peaks')
 @click.option('--rel_height', '-rh', type=float, default=0.4, show_default=True, help='relative height at which the peak width is measured')
 @click.option('--kstodate', '-kd', nargs=2, type=float, default=(0.5, 1.5), show_default=True, help='range of Ks to be dated')
 @click.option('--xlim', '-xl', nargs=2, type=float, default=(None, None), show_default=True, help='xlim of GMM Ks distribution')
 @click.option('--ylim', '-yl', nargs=2, type=float, default=(None, None), show_default=True, help='ylim of GMM Ks distribution')
 @click.option('--manualset', is_flag=True,help="Manually set Ks range of anchor pairs or multiplicons as CI")
 @click.option('--ci', default=95, show_default=True,type=int, help='confidence level of log-normal distribution to date')
-@click.option('--hdr', default=95, show_default=True,type=int, help='highest density region (HDR) in a given distribution to date')
+@click.option('--hdr', default=95, show_default=True,type=int, help='highest density region (HDR)')
 @click.option('--heuristic', is_flag=True,help="heuristic CI for dating")
 @click.option('--kscutoff', '-kc', default=5, show_default=True, type=float, help='Ks Saturation cutoff for genes in Dating')
-@click.option('--showci', is_flag=True,help="show CI for original anchor Ks gmm analysis")
 @click.option('--keeptmpfig', is_flag=True,help="keep temporary figures in peak finding process")
 def peak(**kwargs):
     """
     Infer peak and CI of Ks distribution.
     """
     _peak(**kwargs)
 
-def _peak(ks_distribution, anchorpoints, outdir, alignfilter, ksrange, bin_width, weights_outliers_included, method, seed, em_iter, n_init, components, boots, weighted, plot, bw_method, n_medoids, kdemethod, n_clusters, kmedoids, guide, prominence_cutoff, kstodate, rel_height, ci,manualset,segments,hdr,heuristic,listelements,multipliconpairs,kscutoff,gamma,showci,xlim,ylim,keeptmpfig,family = None):
-    from wgd.peak import alnfilter, group_dS, log_trans, fit_gmm, fit_bgmm, add_prediction, bootstrap_kde, default_plot, get_kde, draw_kde_CI, draw_components_kde_bootstrap, fit_kmedoids, default_plot_kde, fit_apgmm_guide, fit_apgmm_ap, find_apeak, find_mpeak, retreive95CI
+def _peak(ks_distribution, anchorpoints, outdir, alignfilter, ksrange, bin_width, weights_outliers_included, method, seed, em_iter, n_init, components, boots, weighted, plot, bw_method, n_medoids, kdemethod, n_clusters, guide, prominence_cutoff, kstodate, rel_height, ci,manualset,segments,hdr,heuristic,listelements,multipliconpairs,kscutoff,gamma,xlim,ylim,keeptmpfig,family = None):
+    from wgd.peak import alnfilter, group_dS, log_trans, fit_gmm, fit_bgmm, add_prediction, bootstrap_kde, default_plot, get_kde, draw_kde_CI, draw_components_kde_bootstrap, default_plot_kde, fit_apgmm_guide, fit_apgmm_ap, find_apeak, find_mpeak, retreive95CI
     from wgd.core import _mkdir,endtime
     from wgd.utils import formatv2
     start = timer()
     outpath = _mkdir(outdir)
     ksdf = pd.read_csv(ks_distribution,header=0,index_col=0,sep='\t')
     ksdf = formatv2(ksdf)
     if len(ksdf.columns) <4:
@@ -367,19 +365,16 @@
     ksdf_filtered = alnfilter(ksdf,weights_outliers_included,alignfilter[0],alignfilter[1],alignfilter[2],ksrange[0],ksrange[1])
     if family != None:
         retreive95CI(family,ksdf_filtered,outdir,kstodate[0],kstodate[1])
         endtime(start)
     fn_ksdf, weight_col = group_dS(ksdf_filtered)
     train_in = log_trans(fn_ksdf)
     if anchorpoints!= None:
-        if kmedoids:
-            df_ap = fit_kmedoids(guide, anchorpoints, boots, kdemethod, bin_width, weighted, ksdf, ksdf_filtered, outdir, seed, n_medoids, em_iter=em_iter, plot=plot, n_kmedoids = n_clusters, segment = segments, multipliconpairs=multipliconpairs,listelement=listelements,user_xlim=xlim,user_ylim=ylim)
-        else:
-            df_ap_mp = fit_apgmm_guide(hdr,guide,anchorpoints,ksdf,ksdf_filtered,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,segment=segments,multipliconpairs=multipliconpairs,listelement=listelements,cutoff = kscutoff,user_xlim=xlim,user_ylim=ylim)
-            df_ap = fit_apgmm_ap(hdr,anchorpoints,ksdf_filtered,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,heuristic,showCI=showci,cutoff = kscutoff,peak_threshold=prominence_cutoff,rel_height=rel_height,user_xlim=xlim,user_ylim=ylim)
+        df_ap_mp = fit_apgmm_guide(hdr,guide,anchorpoints,ksdf,ksdf_filtered,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,segment=segments,multipliconpairs=multipliconpairs,listelement=listelements,cutoff = kscutoff,user_xlim=xlim,user_ylim=ylim,peak_threshold=prominence_cutoff,rel_height=rel_height,keeptmp=keeptmpfig)
+        df_ap = fit_apgmm_ap(hdr,anchorpoints,ksdf_filtered,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,cutoff = kscutoff,peak_threshold=prominence_cutoff,rel_height=rel_height,user_xlim=xlim,user_ylim=ylim)
         if heuristic:
             find_apeak(df_ap,anchorpoints,os.path.basename(ks_distribution),outdir,peak_threshold=prominence_cutoff,na=False,rel_height=rel_height,ci=ci,user_low=kstodate[0],user_upp=kstodate[1],user=manualset, kscutoff=kscutoff,keeptmp=keeptmpfig)
             find_apeak(df_ap,anchorpoints,os.path.basename(ks_distribution),outdir,peak_threshold=prominence_cutoff,na=True,rel_height=rel_height,ci=ci,user_low=kstodate[0],user_upp=kstodate[1],user=manualset, kscutoff=kscutoff,keeptmp=keeptmpfig)
             find_mpeak(df_ap_mp,anchorpoints,os.path.basename(ks_distribution),outdir,guide,peak_threshold=prominence_cutoff,rel_height=rel_height,ci=ci,user_low=kstodate[0],user_upp=kstodate[1],user=manualset,kscutoff=kscutoff,keeptmp=keeptmpfig)
         endtime(start)
     get_kde(kdemethod,outdir,fn_ksdf,ksdf_filtered,weighted,ksrange[0],ksrange[1])
     if method == 'gmm':
@@ -392,19 +387,19 @@
         ksdf_predict = add_prediction(ksdf,fn_ksdf,train_in,m)
         ksdf_predict.to_csv(fname,header=True,index=True,sep='\t')
         logging.info("Plotting components-annotated Ks distribution for {} components model".format(n))
         #fig = default_plot(ksdf_predict, title=os.path.basename(fname), bins=50, ylabel="Duplication events", nums = int(n),plot = plot)
         #fig.savefig(fname + "_Ks.svg")
         #fig.savefig(fname + "_Ks.pdf")
         #plt.close()
-        fig,ylim,yticks = default_plot_kde(ksdf_predict, title=os.path.basename(fname), bins=50, ylabel="Duplication events", nums = int(n),plot = 'identical',user_xlim=xlim,user_ylim=ylim)
+        fig,safe_ylim,yticks = default_plot_kde(ksdf_predict, title=os.path.basename(fname), bins=50, ylabel="Duplication events", nums = int(n),plot = 'identical',user_xlim=xlim,user_ylim=ylim)
         fig.savefig(fname + "_Ks_kde.svg")
         fig.savefig(fname + "_Ks_kde.pdf")
         plt.close()
-        fig = default_plot(ksdf_predict, title=os.path.basename(fname), bins=50, ylabel="Duplication events", nums = int(n),plot = plot, ylim=ylim,yticks=yticks,user_xlim=xlim,user_ylim=ylim)
+        fig = default_plot(ksdf_predict, title=os.path.basename(fname), bins=50, ylabel="Duplication events", nums = int(n),plot = plot, ylim=safe_ylim,yticks=yticks,user_xlim=xlim,user_ylim=ylim)
         fig.savefig(fname + "_Ks.svg")
         fig.savefig(fname + "_Ks.pdf")
         plt.close()
         #ksdf_predict_filter = alnfilter(ksdf_predict,weights_outliers_included,alignfilter[0],alignfilter[1],alignfilter[2],ksrange[0],ksrange[1])
         #draw_components_kde_bootstrap(kdemethod,outdir,int(n),ksdf_predict_filter,weighted,boots,bin_width)
     #mean_modes, std_modes, mean_medians, std_medians = bootstrap_kde(kdemethod,outdir, train_in, ksrange[0], ksrange[1], boots, bin_width, ksdf_filtered, weight_col, weighted = weighted)
     endtime(start)
```

### Comparing `wgd-2.0.32/setup.py` & `wgd-2.0.33/setup.py`

 * *Files 6% similar despite different names*

```diff
@@ -5,15 +5,15 @@
 from setuptools import setup
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
 setup(
     name='wgd',
-    version='2.0.32',
+    version='2.0.33',
     packages=['wgd'],
     url='http://github.com/heche-psb/wgd',
     license='GPL',
     author='Hengchi Chen',
     author_email='heche@psb.vib-ugent.be',
     description='wgd',
     long_description=long_description,
```

### Comparing `wgd-2.0.32/test/test_core.py` & `wgd-2.0.33/test/test_core.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/__init__.py` & `wgd-2.0.33/wgd/__init__.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/beast.py` & `wgd-2.0.33/wgd/beast.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/cluster.py` & `wgd-2.0.33/wgd/cluster.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/codeml.py` & `wgd-2.0.33/wgd/codeml.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/core.py` & `wgd-2.0.33/wgd/core.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/mcmctree.py` & `wgd-2.0.33/wgd/mcmctree.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/mix.py` & `wgd-2.0.33/wgd/mix.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/peak.py` & `wgd-2.0.33/wgd/peak.py`

 * *Files 3% similar despite different names*

```diff
@@ -176,31 +176,27 @@
     Hs_max = max(Hs)
     y_lim_beforekde = ax.get_ylim()[1]
     CHF = get_totalH(Hs)
     scaling = CHF*0.1
     for kde_x,kde_y,mode,color,i in zip(kdexs,kdeys,modes,colors,range(len(modes))):
         ax.plot(kde_x,scaling*kde_y, alpha=0)
         y_lim_afterkde = ax.get_ylim()[1]
-        #print(y_lim_afterkde)
         scalingtmp=y_lim_beforekde/y_lim_afterkde*scaling
-        #print((scalingtmp,mode))
         ax.plot(kde_x,scalingtmp*kde_y, c=color, ls='-', lw=1.5, alpha=0.8, label='{} component {} mode {:.2f}'.format(guide,i,mode))
     ax.set_ylim(0, max([Hs_max,y_lim_beforekde]))
     ax.legend(loc='upper right', frameon=False)
     ax.set_xlabel("$K_\mathrm{S}$")
     ax.set_ylabel("Duplication events")
     ax.set_xticks([0,1,2,3,4,5])
     sns.despine(offset=1)
     plt.title('Anchor '+'$K_\mathrm{S}$'+'components')
     fig.tight_layout()
     return fig
 
 def plot_mp_component_lognormal(X,hdr,means,stds,weights,labels,n,bins=50,ylabel="Counts",regime='multiplicon',user_xlim=None,user_ylim=None):
-    #labels = labels[(X<5) & (X>0)]
-    #X = X[(X<5) & (X>0)]
     colors = cm.viridis(np.linspace(0, 1, n))
     kdesity = 100
     kde_x = np.linspace(0,5,num=bins*kdesity)
     fig, ax = plt.subplots()
     df = pd.DataFrame.from_dict({'label':labels,'dS':X})
     Hs_maxs,y_lim_beforekde_s = [],[]
     kdexs,kdeys,modes=[],[],[]
@@ -213,19 +209,15 @@
         Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, alpha=0.5, rwidth=0.8, label = "component {}".format(i))
         Hs_max = max(Hs)
         Hs_maxs.append(Hs_max)
         y_lim_beforekde = ax.get_ylim()[1]
         y_lim_beforekde_s.append(y_lim_beforekde)
         CHF = get_totalH(Hs)
         scaling = CHF*0.1
-        #upper_HPD,lower_HPD = calculateHPD(y,hdr)
-        #plt.axvline(x = upper_HPD, color = color, alpha = 0.8, ls = '-.', lw = 1,label="{}% HDR CI Upper {:.2f}".format(hdr,upper_HPD))
-        #plt.axvline(x = lower_HPD, color = color, alpha = 0.8, ls = '-.', lw = 1,label="{}% HDR CI Lower {:.2f}".format(hdr,lower_HPD))
         kde_y = np.array([stats.lognorm.pdf(j, scale=np.exp(mean),s=std) for j in kde_x])
-        #ax.plot(kde_x,scaling*stats.lognorm.pdf(kde_x, scale=np.exp(mean),s=std), c=color, ls='-', lw=1.5, alpha=0.8, label='component {} mode {:.2f}'.format(i,np.exp(mean - std**2)))
         ax.plot(kde_x,scaling*kde_y, c=color, ls='-', lw=1.5, alpha=0.8, label='component {} mode {:.2f}'.format(i,np.exp(mean - std**2)))
         kdexs.append(kde_x)
         kdeys.append(kde_y)
         modes.append(np.exp(mean - std**2))
         y_lim_afterkde = ax.get_ylim()[1]
         if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
     safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)]) * 1.1
@@ -233,17 +225,17 @@
     ax.legend(loc='upper right', fontsize='small',frameon=False)
     ax.set_xlabel("$K_\mathrm{S}$")
     ax.set_ylabel(ylabel)
     ax.set_xticks([0,1,2,3,4,5])
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime== 'multiplicon': plt.title('Multiplicon $K_\mathrm{S}$ GMM modeling')
-    elif regime== 'segment': plt.title('Segment $K_\mathrm{S}$ GMM modeling')
-    else: plt.title('Basecluster $K_\mathrm{S}$ GMM modeling')
+    if regime== 'Multiplicon': plt.title('Multiplicon $K_\mathrm{S}$ GMM modeling')
+    elif regime== 'Segment': plt.title('Segment $K_\mathrm{S}$ GMM modeling')
+    #else: plt.title('Basecluster $K_\mathrm{S}$ GMM modeling')
     fig.tight_layout()
     return fig, kdexs, kdeys, modes
 
 def plot_mp_component(X,labels,n,bins=50,plot = 'identical',ylabel="Counts",regime='multiplicon',user_xlim=None,user_ylim=None):
     labels = labels[(X<5) & (X>0)]
     X = X[(X<5) & (X>0)]
     colors = cm.viridis(np.linspace(0, 1, n))
@@ -265,28 +257,27 @@
     ax.legend(loc='upper right', fontsize='small',frameon=False)
     ax.set_xlabel("$K_\mathrm{S}$")
     ax.set_ylabel(ylabel)
     ax.set_xticks([0,1,2,3,4,5])
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime== 'multiplicon': plt.title('Multiplicon $K_\mathrm{S}$ GMM modeling')
-    elif regime== 'segment': plt.title('Segment $K_\mathrm{S}$ GMM modeling')
-    else: plt.title('Basecluster $K_\mathrm{S}$ GMM modeling')
+    if regime== 'Multiplicon': plt.title('Multiplicon $K_\mathrm{S}$ GMM modeling')
+    elif regime== 'Segment': plt.title('Segment $K_\mathrm{S}$ GMM modeling')
+    #else: plt.title('Basecluster $K_\mathrm{S}$ GMM modeling')
     fig.tight_layout()
     return fig
 
 def plot_ak_component(df,nums,bins=50,plot = 'identical',ylabel="Duplication events",weighted=True,regime='multiplicon',user_xlim=None,user_ylim=None):
     colors = cm.viridis(np.linspace(0, 1, nums))
     fig, ax = plt.subplots()
     if plot == 'identical':
         if weighted:
             for num,color in zip(range(nums),colors):
                 if nums == 1: df_comp = df
-                #if nums == 1: df_comp = df.drop_duplicates(subset=['family','node'])
                 else: df_comp = df[df['AnchorKs_GMM_Component']==num]
                 w = df_comp['weightoutlierexcluded']
                 x = np.array(list(df_comp['dS']))
                 y = x[np.isfinite(x)]
                 w = w[np.isfinite(x)]
                 ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, weights=w, alpha=0.5, rwidth=0.8, label = "component {}".format(num))
         else:
@@ -314,18 +305,18 @@
     ax.legend(loc='upper right', fontsize='small',frameon=False)
     ax.set_xlabel("$K_\mathrm{S}$")
     ax.set_ylabel(ylabel)
     ax.set_xticks([0,1,2,3,4,5])
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime== 'multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ GMM modeling')
-    elif regime== 'segment': plt.title('Segment-guided Anchor Pairs $K_\mathrm{S}$ GMM modeling')
+    if regime== 'Multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ GMM modeling')
+    elif regime== 'Segment': plt.title('Segment-guided Anchor Pairs $K_\mathrm{S}$ GMM modeling')
     elif regime== 'original': plt.title('Original Anchor $K_\mathrm{S}$ GMM modeling')
-    else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ GMM modeling')
+    #else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ GMM modeling')
     fig.tight_layout()
     return fig
 
 def getcompheuri(df,peak_threshold,rel_height,na=False,ci=95):
     gs_ks = df.loc[:,['gene1','gene2','dS']]
     if na:
         df = df.drop_duplicates(subset=['family','node'])
@@ -341,16 +332,14 @@
     kde_x = np.linspace(min_ks-cutoff, max_ks+cutoff,num=500)
     kde = stats.gaussian_kde(ks_refed,weights=w_refed,bw_method="scott")
     kde_y = kde(kde_x)
     spl = interpolate.UnivariateSpline(kde_x, kde_y)
     spl.set_smoothing_factor(0.01)
     spl_x = np.linspace(min_ks, max_ks+0.1, num=int(round((abs(min_ks) + (max_ks+0.1)) *100)))
     spl_y = spl(spl_x)
-    #if na: logging.info('Detecting likely peaks from node-averaged data')
-    #else: logging.info('Detecting likely peaks from node-weighted data')
     init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,peak_threshold=peak_threshold,rel_height=rel_height)
     x_points_strictly_positive = np.linspace(0, 5, int(5 * 100))
     if len(init_means) == 0:
         return None
     else:
         mean,std = init_means[0],init_stdevs[0]
         lowc, upperc = (1-ci/100)/2, 1-(1-ci/100)/2
@@ -390,21 +379,22 @@
             init_means.append(spl_x[peaks[i]])
             best = np.argmax((prominences_refed_R1[i], prominences_refed_L1[i]))
             good_prominences.append(max([prominences_refed_R1[i], prominences_refed_L1[i]]))
             width = [width_refed_R1[i], width_refed_L1[i]][best]
             init_stdevs.append(width/2/100)
     return init_means, init_stdevs,good_prominences
 
-def plot_ak_component_lognormal(df,means,stds,weights,nums,bins=50,ylabel="Duplication events",weighted=True,regime='multiplicon',showCI=False,heuristic=False, peak_threshold=0.1, rel_height=0.4, user_xlim=None,user_ylim=None):
+def plot_ak_component_lognormal(df,means,stds,weights,nums,bins=50,ylabel="Duplication events",weighted=True,regime='multiplicon',showCI=False,heuristic=False, peak_threshold=0.1, rel_height=0.4, user_xlim=None,user_ylim=None,hideci=False,safeylim=False):
     colors = cm.viridis(np.linspace(0, 1, nums))
     kdesity = 100
     kde_x = np.linspace(0,5,num=bins*kdesity)
     fig, ax = plt.subplots()
     CI_dict, CI_dict_heri = {},{}
     Hs_maxs,y_lim_beforekde_s=[],[]
+    scalings = []
     if weighted:
         for num,color in zip(range(nums),colors):
             # here I recover the std by sqrt
             mean,std,weight = means[num][0],np.sqrt(stds[num][0][0]),weights[num]
             if nums == 1: df_comp = df.copy()
             else: df_comp = df[df['AnchorKs_GMM_Component']==num]
             w = df_comp['weightoutlierexcluded']
@@ -413,34 +403,28 @@
             w = w[np.isfinite(x)]
             if len(y) < 2:
                 logging.info("Detected one component with less than 2 elements, will skip it")
                 continue
             Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, weights=w, alpha=0.5, rwidth=0.8, label = "component {}".format(num))
             CHF = get_totalH(Hs)
             scaling = CHF*0.1
-            Hs_max = max(Hs)
-            Hs_maxs.append(Hs_max)
-            y_lim_beforekde = ax.get_ylim()[1]
-            y_lim_beforekde_s.append(y_lim_beforekde)
+            scalings.append(scaling)
+            if safeylim:
+                Hs_max = max(Hs)
+                Hs_maxs.append(Hs_max)
+                y_lim_beforekde = ax.get_ylim()[1]
+                y_lim_beforekde_s.append(y_lim_beforekde)
             ax.plot(kde_x,scaling*stats.lognorm.pdf(kde_x, scale=np.exp(mean),s=std), c=color, ls='-', lw=1.5, alpha=0.8, label='component {} mode {:.2f}'.format(num,np.exp(mean - std**2)))
-            y_lim_afterkde = ax.get_ylim()[1]
-            if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
-            safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
-            ax.set_ylim(0, safe_max)
-            #if showCI and heuristic:
-            #    CI_95 = getcompheuri(df_comp,peak_threshold,rel_height,na=False,ci=95)
-            #    CI_dict_heri[num]=CI_95
-            #    if not (CI_95 is None):
-            #        CI_95_y0 = scaling*stats.lognorm.pdf(CI_95[0], scale=np.exp(mean),s=std)
-            #        CI_95_y1 = scaling*stats.lognorm.pdf(CI_95[1], scale=np.exp(mean),s=std)
-            #        plt.plot([CI_95[0],CI_95[0]],[0,CI_95_y0], color = color, alpha = 0.8, ls = ':', lw = 1,label='Peak {} lower {}%CI {:.2f}'.format(num,95,CI_95[0]))
-            #        plt.plot([CI_95[1],CI_95[1]],[0,CI_95_y1],color = color, alpha = 0.8, ls = ':', lw = 1,label='Peak {} upper {}%CI {:.2f}'.format(num,95,CI_95[1]))
-            if showCI:
+            if safeylim:
+                y_lim_afterkde = ax.get_ylim()[1]
+                if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
+                safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
+                ax.set_ylim(0, safe_max)
+            if showCI and not hideci:
                 CI_95 = stats.lognorm.ppf([0.025, 0.975], scale=np.exp(mean), s=std)
-                #CI_95 = stats.lognorm(std,loc=0,scale=np.exp(mean)).interval(0.95)
                 CI_dict[num]=CI_95
                 CI_95_y0 = scaling*stats.lognorm.pdf(CI_95[0], scale=np.exp(mean),s=std)
                 CI_95_y1 = scaling*stats.lognorm.pdf(CI_95[1], scale=np.exp(mean),s=std)
                 plt.plot([CI_95[0],CI_95[0]],[0,CI_95_y0], color = color, alpha = 0.8, ls = ':', lw = 1,label='component {} lower {}%CI {:.2f}'.format(num,95,CI_95[0]))
                 plt.plot([CI_95[1],CI_95[1]],[0,CI_95_y1],color = color, alpha = 0.8, ls = ':', lw = 1,label='component {} upper {}%CI {:.2f}'.format(num,95,CI_95[1]))
     else:
         for num,color in zip(range(nums),colors):
@@ -451,269 +435,235 @@
             y = x[np.isfinite(x)]
             if len(y) < 2:
                 logging.info("Detected one component with less than 2 elements, will skip it")
                 continue
             Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, alpha=0.5, rwidth=0.8, label = "component {}".format(num))
             CHF = get_totalH(Hs)
             scaling = CHF*0.1
-            Hs_max = max(Hs)
-            Hs_maxs.append(Hs_max)
-            y_lim_beforekde = ax.get_ylim()[1]
-            y_lim_beforekde_s.append(y_lim_beforekde)
+            scalings.append(scaling)
+            if safeylim:
+                Hs_max = max(Hs)
+                Hs_maxs.append(Hs_max)
+                y_lim_beforekde = ax.get_ylim()[1]
+                y_lim_beforekde_s.append(y_lim_beforekde)
             ax.plot(kde_x,scaling*stats.lognorm.pdf(kde_x, scale=np.exp(mean),s=std), c=color, ls='-', lw=1.5, alpha=0.8, label='component {} mode {:.2f}'.format(num,np.exp(mean - std**2)))
-            y_lim_afterkde = ax.get_ylim()[1]
-            if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
-            safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
-            ax.set_ylim(0, safe_max)
+            if safeylim:
+                y_lim_afterkde = ax.get_ylim()[1]
+                if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
+                safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
+                ax.set_ylim(0, safe_max)
             # Here I cancel the heuristic peak searching for each component
             #if showCI and heuristic:
             #    CI_95 = getcompheuri(df_comp,peak_threshold,rel_height,na=True,ci=95)
             #    CI_dict_heri[num]=CI_95
             #    if not (CI_95 is None):
             #        CI_95_y0 = scaling*stats.lognorm.pdf(CI_95[0], scale=np.exp(mean),s=std)
             #        CI_95_y1 = scaling*stats.lognorm.pdf(CI_95[1], scale=np.exp(mean),s=std)
             #        plt.plot([CI_95[0],CI_95[0]],[0,CI_95_y0], color = color, alpha = 0.8, ls = ':', lw = 1,label='Peak {} lower {}%CI {:.2f}'.format(num,95,CI_95[0]))
             #        plt.plot([CI_95[1],CI_95[1]],[0,CI_95_y1],color = color, alpha = 0.8, ls = ':', lw = 1,label='Peak {} upper {}%CI {:.2f}'.format(num,95,CI_95[1]))
-            if showCI:
+            if showCI and not hideci:
                 CI_95 = stats.lognorm.ppf([0.025, 0.975], scale=np.exp(mean), s=std)
                 CI_dict[num]=CI_95
                 CI_95_y0 = scaling*stats.lognorm.pdf(CI_95[0], scale=np.exp(mean),s=std)
                 CI_95_y1 = scaling*stats.lognorm.pdf(CI_95[1], scale=np.exp(mean),s=std)
                 plt.plot([CI_95[0],CI_95[0]],[0,CI_95_y0], color = color, alpha = 0.8, ls = ':', lw = 1,label='component {} lower {}%CI {:.2f}'.format(num,95,CI_95[0]))
                 plt.plot([CI_95[1],CI_95[1]],[0,CI_95_y1],color = color, alpha = 0.8, ls = ':', lw = 1,label='component {} upper {}%CI {:.2f}'.format(num,95,CI_95[1]))
-    #ax.legend(loc='upper right', fontsize='small',frameon=False)
     ax.set_xlim(0, 5)
-    if nums<2: ax.legend(loc='upper right',frameon=False)
-    else: ax.legend(loc='upper right',frameon=False,fontsize='x-small')
+    ax.legend(loc='upper right',frameon=False)
+    #if nums<2: ax.legend(loc='upper right',frameon=False)
+    #else: ax.legend(loc='upper right',frameon=False,fontsize='x-small')
     #ax.legend(loc='center left',bbox_to_anchor=(1.0, 0.5),frameon=False)
     ax.set_xlabel("$K_\mathrm{S}$")
     ax.set_ylabel(ylabel)
     ax.set_xticks([0,1,2,3,4,5])
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime=='multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ GMM modeling')
-    elif regime=='segment': plt.title('Segment-guided Anchor Pairs $K_\mathrm{S}$ GMM modeling')
+    if regime=='Multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ GMM modeling')
+    elif regime=='Segment': plt.title('Segment-guided Anchor Pairs $K_\mathrm{S}$ GMM modeling')
     elif regime== 'original': plt.title('Original Anchor $K_\mathrm{S}$ GMM modeling')
-    else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ GMM modeling')
+    #else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ GMM modeling')
     fig.tight_layout()
-    return fig, {'showci':CI_dict,'heuristic':CI_dict_heri}
+    return fig, {'showci':CI_dict,'heuristic':CI_dict_heri},ax,scalings
 
 def getmediankdexy(kdex,kdey):
     weights = [kdex[i]*kdey[i] for i in range(len(kdex))]
     half_weight = sum(weights)/2
     cum_weight = 0
     medianx = 0
     for i,weight in enumerate(weights):
         cum_weight = cum_weight + weight
         if cum_weight >= half_weight:
             medianx = kdex[i]
             break
     return medianx
 
-def plot_ak_component_kde(df,nums,hdr,bins=50,ylabel="Duplication events",weighted=True,regime='multiplicon',user_xlim=None,user_ylim=None):
+def plot_ak_component_kde(df,nums,hdr,bins=50,ylabel="Duplication events",weighted=True,regime='multiplicon',user_xlim=None,user_ylim=None,hidehdr=False,notscale=False,safeylim=False):
     colors = cm.viridis(np.linspace(0, 1, nums))
     kdesity = 100
     kde_x = np.linspace(0,5,num=bins*kdesity)
     fig, ax = plt.subplots()
     Hs_maxs,y_lim_beforekde_s,HDRs = [],[],{}
+    scalings,kdes = [],[]
     if weighted:
         for num,color in zip(range(nums),colors):
             #if nums == 1: df_comp = df.drop_duplicates(subset=['family','node'])
             if nums == 1: df_comp = df.copy()
             else: df_comp = df[df['AnchorKs_GMM_Component']==num]
             w = df_comp['weightoutlierexcluded']
             x = np.array(list(df_comp['dS']))
             y = x[np.isfinite(x)]
             w = w[np.isfinite(x)]
             if len(y) < 2:
                 logging.info("Detected one component with less than 2 elements, will skip it")
                 continue
             kde = stats.gaussian_kde(y,weights=w,bw_method='scott')
+            kdes.append(kde)
             kde_y = kde(kde_x)
             mode, maxim = kde_mode(kde_x, kde_y)
             #median = getmediankdexy(kde_x,kde_y)
             Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, weights=w, alpha=0.5, rwidth=0.8, label = "component {} (mode {:.2f})".format(num,mode))
-            #Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, weights=w, alpha=0.5, rwidth=0.8, label = "component {} (mode {:.2f} median {:.2f})".format(num,mode,median))
-            Hs_max = max(Hs)
-            Hs_maxs.append(Hs_max)
-            y_lim_beforekde = ax.get_ylim()[1]
-            y_lim_beforekde_s.append(y_lim_beforekde)
+            if safeylim:
+                Hs_max = max(Hs)
+                Hs_maxs.append(Hs_max)
+                y_lim_beforekde = ax.get_ylim()[1]
+                y_lim_beforekde_s.append(y_lim_beforekde)
             CHF = get_totalH(Hs)
             scaling = CHF*0.1
+            scalings.append(scaling)
             ax.plot(kde_x, kde_y*scaling, color=color,alpha=0.4, ls = '--',lw = 1)
-            y_lim_afterkde = ax.get_ylim()[1]
-            if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
-            safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
-            ax.set_ylim(0, safe_max)
+            if safeylim:
+                y_lim_afterkde = ax.get_ylim()[1]
+                if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
+                safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
+                ax.set_ylim(0, safe_max)
             #ax.axvline(x = mode, ymin=0,ymax=scaling*maxim/Hs_max, color = color, alpha = 0.8, ls = ':', lw = 1)
             plt.plot([mode,mode], [0,scaling*maxim], color=color,alpha = 0.8,linestyle='-.')
             #plt.plot([median,median], [0,scaling*kde(median)], color=color,alpha = 0.8,linestyle='-.')
-            upper_HPD,lower_HPD = calculateHPD(y,hdr)
-            plt.plot([upper_HPD,upper_HPD], [0,scaling*kde(upper_HPD)], color=color,alpha = 0.8,linestyle=':')
-            plt.plot([lower_HPD,lower_HPD], [0,scaling*kde(lower_HPD)], color=color,alpha = 0.8,linestyle=':')
+            if not hidehdr:
+                upper_HPD,lower_HPD = calculateHPD(y,hdr)
+                plt.plot([upper_HPD,upper_HPD], [0,scaling*kde(upper_HPD)], color=color,alpha = 0.8,linestyle=':')
+                plt.plot([lower_HPD,lower_HPD], [0,scaling*kde(lower_HPD)], color=color,alpha = 0.8,linestyle=':')
             #ax.axvline(x = upper_HPD, ymin=0, ymax=upper_HPD_y, color = color, alpha = 0.8, ls = '-.', lw = 1,label='HDR {:.2f}-{:.2f}'.format(lower_HPD,upper_HPD))
             #ax.axvline(x = lower_HPD, ymin=0, ymax=lower_HPD_y, color = color, alpha = 0.8, ls = '-.', lw = 1)
-            HDRs[num] = (lower_HPD,upper_HPD)
+                HDRs[num] = (lower_HPD,upper_HPD)
     else:
         for num,color in zip(range(nums),colors):
             if nums == 1: df_comp = df.drop_duplicates(subset=['family','node'])
             else: df_comp = df[df['AnchorKs_GMM_Component']==num].drop_duplicates(subset=['family','node'])
             x = np.array(list(df_comp['node_averaged_dS_outlierexcluded']))
             y = x[np.isfinite(x)]
             if len(y) < 2:
                 logging.info("Detected one component with less than 2 elements, will skip it")
                 continue
             kde = stats.gaussian_kde(y,bw_method='scott')
+            kdes.append(kde)
             kde_y = kde(kde_x)
             mode, maxim = kde_mode(kde_x, kde_y)
             #median = getmediankdexy(kde_x,kde_y)
             Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, alpha=0.5, rwidth=0.8, label = "component {} (mode {:.2f})".format(num,mode))
             #Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, alpha=0.5, rwidth=0.8, label = "component {} (mode {:.2f} median {:.2f})".format(num,mode,median))
-            Hs_max = max(Hs)
-            Hs_maxs.append(Hs_max)
-            y_lim_beforekde = ax.get_ylim()[1]
-            y_lim_beforekde_s.append(y_lim_beforekde)
+            if safeylim:
+                Hs_max = max(Hs)
+                Hs_maxs.append(Hs_max)
+                y_lim_beforekde = ax.get_ylim()[1]
+                y_lim_beforekde_s.append(y_lim_beforekde)
             CHF = get_totalH(Hs)
             scaling = CHF*0.1
+            scalings.append(scaling)
             ax.plot(kde_x, kde_y*scaling, color=color,alpha=0.4, ls = '--',lw = 1)
-            y_lim_afterkde = ax.get_ylim()[1]
-            if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
-            safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
-            ax.set_ylim(0, safe_max)
+            if safeylim:
+                y_lim_afterkde = ax.get_ylim()[1]
+                if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
+                safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
+                ax.set_ylim(0, safe_max)
             plt.plot([mode,mode], [0,scaling*maxim], color=color,alpha = 0.8,linestyle='-.')
             #plt.plot([median,median], [0,scaling*kde(median)], color=color,alpha = 0.8,linestyle='-.')
             #ax.axvline(x = mode, ymin=0, ymax=scaling*maxim/Hs_max, color = color, alpha = 0.8, ls = ':', lw = 1)
-            upper_HPD,lower_HPD = calculateHPD(y,hdr)
-            #upper_HPD_y = scaling*kde(upper_HPD)/Hs_max
-            #lower_HPD_y = scaling*kde(lower_HPD)/Hs_max
-            plt.plot([upper_HPD,upper_HPD], [0,scaling*kde(upper_HPD)], color=color,alpha = 0.8,linestyle=':')
-            plt.plot([lower_HPD,lower_HPD], [0,scaling*kde(lower_HPD)], color=color,alpha = 0.8,linestyle=':')
+            if not hidehdr:
+                upper_HPD,lower_HPD = calculateHPD(y,hdr)
+                #upper_HPD_y = scaling*kde(upper_HPD)/Hs_max
+                #lower_HPD_y = scaling*kde(lower_HPD)/Hs_max
+                plt.plot([upper_HPD,upper_HPD], [0,scaling*kde(upper_HPD)], color=color,alpha = 0.8,linestyle=':')
+                plt.plot([lower_HPD,lower_HPD], [0,scaling*kde(lower_HPD)], color=color,alpha = 0.8,linestyle=':')
             #ax.axvline(x = upper_HPD, ymin=0, ymax=upper_HPD_y, color = color, alpha = 0.8, ls = '-.', lw = 1,label='HDR {:.2f}-{:.2f}'.format(lower_HPD,upper_HPD))
             #ax.axvline(x = lower_HPD, ymin=0, ymax=lower_HPD_y, color = color, alpha = 0.8, ls = '-.', lw = 1)
-            HDRs[num] = (lower_HPD,upper_HPD)
+                HDRs[num] = (lower_HPD,upper_HPD)
     #ax.legend(loc='upper right', fontsize='small',frameon=False)
     ax.legend(loc='upper right', frameon=False)
     #ax.legend(loc='center left',bbox_to_anchor=(1.0, 0.5),frameon=False)
     ax.set_xlabel("$K_\mathrm{S}$")
     ax.set_ylabel(ylabel)
-    ax.set_xticks([0,1,2,3,4,5])
-    if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
-    if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
+    if not notscale:
+        ax.set_xticks([0,1,2,3,4,5])
+        if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
+        if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime=='multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ GMM modeling')
-    elif regime=='segment': plt.title('Segment-guided Anchor Pairs $K_\mathrm{S}$ GMM modeling')
+    if regime=='Multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ GMM modeling')
+    elif regime=='Segment': plt.title('Segment-guided Anchor Pairs $K_\mathrm{S}$ GMM modeling')
     elif regime== 'original': plt.title('Original Anchor $K_\mathrm{S}$ GMM modeling')
-    else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ GMM modeling')
-    fig.tight_layout()
-    return fig,HDRs
+    #else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ GMM modeling')
+    if not notscale: fig.tight_layout()
+    if not hidehdr: return fig,HDRs
+    else: return fig,ax,kdes,scalings
 
-def default_plot_kde(*args,bins=50,alphas=None,colors=None,weighted=True,title="",ylabel="Duplication events",nums = "", plot = 'identical',user_xlim=None,user_ylim=None,**kwargs):
+def default_plot_kde(*args,bins=50,alphas=None,colors=None,weighted=True,title="",ylabel="Duplication events",nums = "", plot = 'identical',user_xlim=None,user_ylim=None, safeylim=False,**kwargs):
     ndists = len(args)
     alphas = alphas or list(np.linspace(0.2, 1, ndists))
     colors = colors or ['black'] * ndists
     # assemble panels
     keys = ["dS", "dS", "dN", "dN/dS"]
     np.seterr(divide='ignore')
     funs = [lambda x: x, np.log10, np.log10, np.log10]
     fig, axs = plt.subplots(2, 2)
     _labels = {"dS" : "$K_\mathrm{S}$","dN" : "$K_\mathrm{A}$","dN/dS": "$\omega$"}
     bins = 50
     kdesity = 100
     kde_x = np.linspace(0,5,num=bins*kdesity)
-    #color_table = ['blue','orange','green','red','purple','brown','pink','gray','olive','cyan']
     for (c, a, dist) in zip(colors, alphas, args):
         dis = dist.dropna(subset=['weightoutlierexcluded'])
-        #y_lim = []
-        #time = 0
         y_lim_beforekde_s,Hs_maxs = [],[]
         for ax, k, f in zip(axs.flatten(), keys, funs):
-            #color_random = cm.rainbow(np.linspace(0, 1, n))
-            #time = time + 1
-            #comp_time = 0
             for num, color in zip(range(nums),cm.viridis(np.linspace(0, 1, nums))):
                 dist_comp = dis[dis['component']==num]
                 w = dist_comp['weightoutlierexcluded']
                 x = f(dist_comp[k])
                 y = x[np.isfinite(x)]
                 w = w[np.isfinite(x)]
                 if funs[0] == f:
-                    #print(y.shape)
                     Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, weights=w, alpha=a, rwidth=0.8, label = "component {}".format(num),**kwargs)
-                    #print(Hs[0])
                     Hs_max = max(Hs)
                     Hs_maxs.append(Hs_max)
-                    #y_lim.append(ax.get_ylim()[1])
-                    #ax.set_ylim(0, max(y_lim) * 1.2)
                     y_lim_beforekde = ax.get_ylim()[1]
-                    #print(y_lim_beforekde)
                     y_lim_beforekde_s.append(y_lim_beforekde)
                     kde = stats.gaussian_kde(y,weights=w,bw_method=0.1)
                     kde_y = kde(kde_x)
                     mode, maxim = kde_mode(kde_x, kde_y)
                     CHF = get_totalH(Hs)
                     scale = CHF*0.1
                     ax.plot(kde_x, kde_y*scale, color=color,alpha=0.4, ls = '--',lw = 1)
                     y_lim_afterkde = ax.get_ylim()[1]
-                    #print(y_lim_afterkde)
-                    if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
-                    #y_lim_done = ax.get_ylim()[1]
-                    #y_lim_dones.append(y_lim_done)
-                    #safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs),max(y_lim_dones)])
+                    if safeylim:
+                        if y_lim_afterkde > y_lim_beforekde: ax.set_ylim(0, y_lim_beforekde)
                     safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
-                    ax.set_ylim(0, safe_max)
-                    #yticks = ax.get_yticks()
-                    #if y_lim_beforekde < max(y_lim_beforekde_s):
-                    #    if max(y_lim_beforekde_s) > max(y_lim_dones): ax.set_ylim(0, max(y_lim_beforekde_s))
-                    #    else: ax.set_ylim(0, max(y_lim_dones))
+                    if safeylim: ax.set_ylim(0, safe_max)
                     ax.axvline(x = mode, color = color, alpha = 0.8, ls = ':', lw = 1)
                     ax.legend(loc='upper right', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1,frameon=False)
                 else:
-                    #comp_time = comp_time + 1
-                    #if comp_time < 2: Hs, Bins, patches = ax.hist(y, bins = 50, color = color, weights=w, alpha=a, rwidth=0.8, label = "component{}".format(num),**kwargs)
                     ax.hist(y, bins = np.linspace(-50, 20, num=70+1,dtype=int)/10, color = color, weights=w, alpha=a, rwidth=0.8, label = "component {}".format(num),**kwargs)
                     ax.legend(loc='upper left', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1,frameon=False)
-                    #Hss.append(Hs)
-                    #Binss.append(Bins)
-                    #patchess.append(patches)
-                #kde_x, kde_y = get_kde(train_in,ax)
-                #ax.plot(kde_x,kde_y)
-            #else:
-            #    cs = [color for color in cm.rainbow(np.linspace(0, 1, nums))]
-            #    dist_comps = [dis[dis['component']==num] for num in range(nums)]
-            #    ws = [i['weightoutlierexcluded'] for i in dist_comps]
-            #    xs = [f(i[k]) for i in dist_comps]
-            #    ys = [x[np.isfinite(x)] for x in xs]
-            #    ws = [w[np.isfinite(x)] for w,x in zip(ws,xs)]
-            #    if funs[0] == f:
-            #        ax.hist(ys, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color=[cs[i] for i in range(nums)], stacked=True, weights=ws, alpha=a, rwidth=0.8, label = ["component{}".format(int(i)) for i in range(nums)],**kwargs)
-            #        ax.legend(loc='upper right', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1)
-            #    else:
-                    #maxi = int(max(list(itertools.chain.from_iterable(xs)))*100)
-                    #mini = int(min(list(itertools.chain.from_iterable(xs)))*100)
-            #        ax.hist(ys, bins = bins, color=[cs[i] for i in range(nums)], weights=ws, alpha=a, rwidth=0.8, label = ["component{}".format(int(i)) for i in range(nums)],**kwargs)
-            #        ax.legend(loc='upper left', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1)
-                    #ax.set_xticks(np.round_(np.linspace(mini, maxi, num=5,dtype=int)/100,decimals = 1))
-                #kde_x, kde_y = get_kde(train_in,ax)
-            #leg = ax.legend(loc='upper right', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1)
-            #for lh in leg.legendHandles: lh.set_alpha(0.1)
         xlabel = _labels[k]
         if f == np.log10:
             xlabel = "$\log_{10}" + xlabel[1:-1] + "$"
         ax.set_xlabel(xlabel)
-        #original_y_lim = ax.get_ylim()[1]
-        #ax.set_ylim(0, original_y_lim * 1.2)
     yticks = axs[0,0].get_yticks()
     yticklabels = axs[0,0].get_yticklabels()
-    #print(yticks)
-    #print(safe_max)
-    #print(yticklabels)
     axs[0,0].set_ylabel(ylabel)
     axs[0,0].set_yticks(axs[0,0].get_yticks())
-    axs[0,0].set_ylim(0, safe_max)
+    if safeylim: axs[0,0].set_ylim(0, safe_max)
     axs[1,0].set_ylabel(ylabel)
     axs[0,0].set_xticks([0,1,2,3,4,5])
     if not (user_ylim[0]) is None: axs[0,0].set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: axs[0,0].set_xlim(user_xlim[0],user_xlim[1])
     # finalize plot
     sns.despine(offset=1)
     fig.suptitle(title, x=0.125, y=0.9, ha="left", va="top")
@@ -727,70 +677,45 @@
     colors = colors or ['black'] * ndists
     # assemble panels
     keys = ["dS", "dS", "dN", "dN/dS"]
     np.seterr(divide='ignore')
     funs = [lambda x: x, np.log10, np.log10, np.log10]
     fig, axs = plt.subplots(2, 2)
     _labels = {"dS" : "$K_\mathrm{S}$","dN" : "$K_\mathrm{A}$","dN/dS": "$\omega$"}
-    #color_table = ['blue','orange','green','red','purple','brown','pink','gray','olive','cyan']
     for (c, a, dist) in zip(colors, alphas, args):
         dis = dist.dropna(subset=['weightoutlierexcluded'])
-        #time = 0
         for ax, k, f in zip(axs.flatten(), keys, funs):
-            #color_random = cm.rainbow(np.linspace(0, 1, n))
-            #time = time + 1
-            #comp_time = 0
             if plot == 'identical':
                 for num, color in zip(range(nums),cm.viridis(np.linspace(0, 1, nums))):
                     dist_comp = dis[dis['component']==num]
                     w = dist_comp['weightoutlierexcluded']
                     x = f(dist_comp[k])
                     y = x[np.isfinite(x)]
                     w = w[np.isfinite(x)]
                     if funs[0] == f:
                         ax.hist(y, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color = color, weights=w, alpha=a, rwidth=0.8, label = "component {}".format(num),**kwargs)
-                        #Hs_max = max(Hs)
-                        #Hs_maxs.append(Hs_max)
-                        #y_lim_beforekde = ax.get_ylim()[1]
-                        #y_lim_beforekde_s.append(y_lim_beforekde)
-                        #y_lim.append(ax.get_ylim()[1])
-                        #ax.set_ylim(0, max(y_lim) * 1.2)
-                        #safe_max = max([max(y_lim_beforekde_s),max(Hs_maxs)])
                         ax.set_yticks(yticks)
                         ax.set_ylim(0, ylim)
                         ax.legend(loc='upper right', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1)
                     else:
-                        #comp_time = comp_time + 1
-                        #if comp_time < 2: Hs, Bins, patches = ax.hist(y, bins = 50, color = color, weights=w, alpha=a, rwidth=0.8, label = "component{}".format(num),**kwargs)
                         ax.hist(y, bins = np.linspace(-50, 20, num=70+1,dtype=int)/10, color = color, weights=w, alpha=a, rwidth=0.8, label = "component {}".format(num),**kwargs)
                         ax.legend(loc='upper left', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1)
-                        #Hss.append(Hs)
-                        #Binss.append(Bins)
-                        #patchess.append(patches)
-                    #kde_x, kde_y = get_kde(train_in,ax)
-                    #ax.plot(kde_x,kde_y)
             else:
                 cs = [color for color in cm.viridis(np.linspace(0, 1, nums))]
                 dist_comps = [dis[dis['component']==num] for num in range(nums)]
                 ws = [i['weightoutlierexcluded'] for i in dist_comps]
                 xs = [f(i[k]) for i in dist_comps]
                 ys = [x[np.isfinite(x)] for x in xs]
                 ws = [w[np.isfinite(x)] for w,x in zip(ws,xs)]
                 if funs[0] == f:
                     ax.hist(ys, bins = np.linspace(0, 50, num=bins+1,dtype=int)/10, color=[cs[i] for i in range(nums)], stacked=True, weights=ws, alpha=a, rwidth=0.8, label = ["component {}".format(int(i)) for i in range(nums)],**kwargs)
                     ax.legend(loc='upper right', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1,frameon=False)
                 else:
-                    #maxi = int(max(list(itertools.chain.from_iterable(xs)))*100)
-                    #mini = int(min(list(itertools.chain.from_iterable(xs)))*100)
                     ax.hist(ys, bins = bins, color=[cs[i] for i in range(nums)], weights=ws, alpha=a, rwidth=0.8, label = ["component {}".format(int(i)) for i in range(nums)],**kwargs)
                     ax.legend(loc='upper left', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1,frameon=False)
-                    #ax.set_xticks(np.round_(np.linspace(mini, maxi, num=5,dtype=int)/100,decimals = 1))
-                #kde_x, kde_y = get_kde(train_in,ax)
-            #leg = ax.legend(loc='upper right', fontsize=5,fancybox=True, framealpha=0.1,labelspacing=0.1,handlelength=2,handletextpad=0.1)
-            #for lh in leg.legendHandles: lh.set_alpha(0.1)
             xlabel = _labels[k]
             if f == np.log10:
                 xlabel = "$\log_{10}" + xlabel[1:-1] + "$"
             ax.set_xlabel(xlabel)
     axs[0,0].set_ylabel(ylabel)
     axs[1,0].set_ylabel(ylabel)
     axs[0,0].set_xticks([0,1,2,3,4,5])
@@ -871,36 +796,32 @@
     return kde_x, kde_y
 
 def kde_mode(kde_x, kde_y):
     maxy_iloc = np.argmax(kde_y)
     mode = kde_x[maxy_iloc]
     return mode, max(kde_y)
 
-#def bootstrap_dates(nd,)
 def bootstrap_kde(kdemethod,outdir,train_in, ks_lower, ks_upper, boots, bin_width, ksdf_filtered, weight_col, weighted = False):
     train_nonan = train_in[~np.isnan(train_in)]
     modes = []
     medians = []
     kde_x = np.linspace(ks_lower, ks_upper, num=500)
     for r in range(boots):
         if weighted:
-            #bootstrap_sample = random.choices(ksdf_filtered['dS'], weights = weight_col, k=len(weight_col))
             bootstrap_sample = random.choices(ksdf_filtered['dS'], k=len(weight_col))
-            #kde_x, kde_y = get_gaussian_kde(bootstrap_sample, ks_lower, ks_upper, bin_width, ksdf_filtered, weight_col, weighted)
             if kdemethod == 'scipy':kde_y=stats.gaussian_kde(bootstrap_sample,bw_method=bin_width,weights=weight_col.tolist()).pdf(kde_x)
             if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bin_width).fit(bootstrap_sample,weights=weight_col.tolist()).evaluate(kde_x)
             if kdemethod == 'treekde': kde_y = TreeKDE(bw=bin_width).fit(bootstrap_sample,weights=weight_col.tolist()).evaluate(kde_x)
             if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bin_width).fit(bootstrap_sample,weights=weight_col.tolist()).evaluate(kde_x)
         else:
             bootstrap_sample = random.choices(train_nonan, k=len(train_nonan))
             if kdemethod == 'scipy': kde_y=stats.gaussian_kde(bootstrap_sample,bw_method=bin_width).pdf(kde_x)
             if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bin_width).fit(bootstrap_sample,weights=None).evaluate(kde_x)
             if kdemethod == 'treekde': kde_y = TreeKDE(bw=bin_width).fit(bootstrap_sample,weights=None).evaluate(kde_x)
             if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bin_width).fit(bootstrap_sample,weights=None).evaluate(kde_x)
-            #kde_x, kde_y = get_gaussian_kde(bootstrap_sample, ks_lower, ks_upper, bin_width, ksdf_filtered, weight_col, weighted)
         plt.plot(kde_x, kde_y, color = 'black',alpha=0.1)
         mode, maxim = kde_mode(kde_x, kde_y)
         modes.append(mode)
         medians.append(np.median(bootstrap_sample))
     if weighted:
         plt.hist(ksdf_filtered['dS'],bins = np.linspace(0, 50, num=51,dtype=int)/10,density=True,color = 'black', alpha=0.1, rwidth=0.8)
     else:
@@ -964,23 +885,20 @@
     kde_x = np.linspace(minm,maxm,num=500)
     modes = []
     f, ax = plt.subplots()
     if bw_method == 'silverman': logging.info("Assmuing the dates is unimodal and close to normal, applying silverman’s rule of thumb")
     else: logging.info("Assmuing the dates is far from normal or multimodal, applying the Improved Sheather Jones (ISJ) algorithm")
     for i,color in zip(range(boots),cm.viridis(np.linspace(0, 1, boots))):
         sample = random.choices(train_in, k=len(train_in))
-        #kde = stats.gaussian_kde(sample)
         if kdemethod == 'scipy': kde_y=stats.gaussian_kde(sample,bw_method=bw_method).pdf(kde_x)
         if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bw_method).fit(sample,weights=None).evaluate(kde_x)
         if kdemethod == 'treekde': kde_y = TreeKDE(bw=bw_method).fit(sample,weights=None).evaluate(kde_x)
         if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bw_method).fit(sample,weights=None).evaluate(kde_x)
-        #kde_y = NaiveKDE(bw=bw_method).fit(sample,weights=None).evaluate(kde_x)
         mode, maxim = kde_mode(kde_x, kde_y)
         modes.append(mode)
-        #kde_y = kde(kde_x)
         plt.plot(kde_x, kde_y, color = color,alpha=0.7)
     lower, upper = get_empirical_CI(0.95,modes)
     plt.axvline(x = lower, color = 'green', alpha = 0.8, ls = '--', lw = 1)
     plt.axvline(x = upper, color = 'green', alpha = 0.8, ls = '--', lw = 1)
     #https://machinelearningmastery.com/calculate-bootstrap-confidence-intervals-machine-learning-results-python/
     logging.info("The 95% empirical confidence interval (CI) of WGD dates is {0} - {1} billion years".format(lower,upper))
     if kdemethod == 'scipy': kde_y=stats.gaussian_kde(train_in.tolist(),bw_method=bw_method).pdf(kde_x)
@@ -994,15 +912,14 @@
     if kdemethod == 'treekde': kde_y = TreeKDE(bw=bw_method).fit(modes).evaluate(kde_x)
     if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bw_method).fit(modes).evaluate(kde_x)
     mode_of_modes, maxim_of_modes = kde_mode(kde_x, kde_y)
     plt.axvline(x = mode_of_modes, color = 'red', alpha = 0.8, ls = '-.', lw = 1)
     logging.info("The kde-mode of original WGD dates is {} billion years".format(mode_orig))
     plt.xlabel("Billion years ago", fontsize = 10)
     plt.ylabel("Frequency", fontsize = 10)
-    #print(Ten_multi(int(maxim_orig)))
     plt.yticks(np.linspace(0,Ten_multi(int(maxim_orig))+10,num=10,dtype=int))
     plt.hist(train_in,bins = np.linspace(minm, maxm, num=50),density=True,color = 'black', alpha=0.15, rwidth=0.8)
     props = dict(boxstyle='round', facecolor='grey', alpha=0.1)
     text = "\n".join(["Raw mode: {:4.4f}".format(mode_of_modes),"Peak: {:4.4f}".format(mode_orig),"CI: {:4.4f}-{:4.4f}".format(lower, upper),"OGs: {}".format(len(train_in))])
     plt.text(0.75,0.95,text,transform=ax.transAxes,fontsize=8,verticalalignment='top',bbox=props)
     fname = os.path.join(outdir, "WGD_peak.pdf")
     plt.tight_layout()
@@ -1021,15 +938,14 @@
         minm = float(pretrain_in['dS'].min())
         kde_x = np.linspace(0,5,num=500)
         modes = []
         f, ax = plt.subplots()
         if weighted:
             train_in = pretrain_in.dropna(subset=['weightoutlierexcluded'])
             for i,color in zip(range(boots),cm.viridis(np.linspace(0, 1, boots))):
-                #sample = random.choices(train_in['dS'],weights = train_in['weightoutlierexcluded'], k=len(train_in))
                 sample = random.choices(train_in['dS'], k=len(train_in))
                 if kdemethod == 'scipy': kde_y=stats.gaussian_kde(sample,weights=train_in['weightoutlierexcluded'].tolist(),bw_method=bin_width).pdf(kde_x)
                 if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bin_width).fit(sample,weights=train_in['weightoutlierexcluded'].tolist()).evaluate(kde_x)
                 if kdemethod == 'treekde': kde_y = TreeKDE(bw=bin_width).fit(sample,weights=train_in['weightoutlierexcluded'].tolist()).evaluate(kde_x)
                 if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bin_width).fit(sample,weights=train_in['weightoutlierexcluded'].tolist()).evaluate(kde_x)
                 mode, maxim = kde_mode(kde_x, kde_y)
                 modes.append(mode)
@@ -1043,16 +959,14 @@
                 if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bin_width).fit(sample).evaluate(kde_x)
                 if kdemethod == 'treekde': kde_y = TreeKDE(bw=bin_width).fit(sample).evaluate(kde_x)
                 if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bin_width).fit(sample).evaluate(kde_x)
                 mode, maxim = kde_mode(kde_x, kde_y)
                 modes.append(mode)
                 plt.plot(kde_x, kde_y, color = color,alpha=0.1)
             plt.hist(train_in['node_averaged_dS_outlierexcluded'], bins = np.linspace(0, 50, num=51,dtype=int)/10,density=True, color = 'black', alpha=0.15, rwidth=0.8)
-        #ax.set_xticks([0,1,2,3,4,5])
-        #plt.xlim(0, 5)
         plt.xlabel("$K_\mathrm{S}$", fontsize = 10)
         plt.ylabel("Frequency", fontsize = 10)
         lower, upper = get_empirical_CI(0.95,modes)
         plt.axvline(x = lower, color = 'green', alpha = 0.8, ls = '--', lw = 1)
         plt.axvline(x = upper, color = 'green', alpha = 0.8, ls = '--', lw = 1)
         if weighted:
             train_in = pretrain_in.dropna(subset=['weightoutlierexcluded'])
@@ -1063,17 +977,15 @@
         else:
             train_in = pretrain_in.dropna(subset=['node_averaged_dS_outlierexcluded'])
             if kdemethod == 'scipy': kde_y=stats.gaussian_kde(train_in['node_averaged_dS_outlierexcluded'].tolist(),bw_method=bin_width).pdf(kde_x)
             if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bin_width).fit(train_in['node_averaged_dS_outlierexcluded'].tolist()).evaluate(kde_x)
             if kdemethod == 'treekde': kde_y = TreeKDE(bw=bin_width).fit(train_in['node_averaged_dS_outlierexcluded'].tolist()).evaluate(kde_x)
             if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bin_width).fit(train_in['node_averaged_dS_outlierexcluded'].tolist()).evaluate(kde_x)
         mode_orig, maxim_orig = kde_mode(kde_x, kde_y)
-        #print(modes)
         if all([i==0 for i in modes]):
-            #modes = np.array(modes)+1e-10
             modes = [i+np.random.choice(np.arange(1e-10, 1e-8, 1e-10)) for i in modes]
         if kdemethod == 'scipy': kde_y=stats.gaussian_kde(np.array(modes),bw_method=bin_width).pdf(kde_x)
         if kdemethod == 'naivekde': kde_y = NaiveKDE(bw=bin_width).fit(modes).evaluate(kde_x)
         if kdemethod == 'treekde': kde_y = TreeKDE(bw=bin_width).fit(modes).evaluate(kde_x)
         if kdemethod == 'fftkde': kde_y = FFTKDE(bw=bin_width).fit(modes).evaluate(kde_x)
         mode_of_modes, maxim_of_modes = kde_mode(kde_x, kde_y)
         plt.axvline(x = mode_orig, color = 'black', alpha = 0.8, ls = ':', lw = 1)
@@ -1206,42 +1118,43 @@
         plt.hist(Xs,bins = np.linspace(0, 5, num=int(5/bin_width)+1),color=cs,alpha=0.5,rwidth=0.8,stacked=True,label=labels)
     plt.xlabel("$K_\mathrm{S}$", fontsize = 10)
     plt.ylabel("Frequency", fontsize = 10)
     ax.legend(loc=1,fontsize='large',frameon=False)
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime=='multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
-    elif regime=='segment': plt.title('Segment-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
-    else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
+    if regime=='Multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
+    elif regime=='Segment': plt.title('Segment-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
+    #else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
     plt.tight_layout()
     plt.savefig(fname,format ='pdf', bbox_inches='tight')
     plt.close()
 
-def plot_segment_kmedoids(labels,X,outdir,bin_width,n,regime='segment',user_xlim=None,user_ylim=None):
+def plot_segment_kmedoids(labels,X,outdir,bin_width,n,regime='Segment',user_xlim=None,user_ylim=None):
     fname = os.path.join(outdir,"{}_Ks_KMedoids_Clustering_{}components.pdf".format(regime,n))
     f, ax = plt.subplots()
     df = pd.DataFrame(labels,columns=['KMedoids_Cluster'])
     df.index.name = 'label'
-    df.loc[:,['Segment_Ks']] = X
+    df.loc[:,['{}_Ks'.format(regime)]] = X
     df = df.reset_index()
     for i,c in zip(range(n),cm.viridis(np.linspace(0, 1, n))):
         if n == 1: dfo = df.copy()
         else: dfo = df[df['KMedoids_Cluster']==i]
-        data = getX(dfo,'Segment_Ks')
+        data = getX(dfo,'{}_Ks'.format(regime))
         ax.hist(data,bins = np.linspace(0, 5, num=int(5/bin_width)+1),color=c,alpha=0.5,rwidth=0.8,label='component {}'.format(i))
     plt.xlabel("$K_\mathrm{S}$", fontsize = 10)
-    plt.ylabel("Number of segment pair", fontsize = 10)
+    y = lambda x:x[0].lower()+x[1:]
+    plt.ylabel("Number of {} pair".format(y(regime)), fontsize = 10)
     ax.legend(loc=1,fontsize='small',frameon=False)
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime=='multiplicon': plt.title('Multilplicon $K_\mathrm{S}$ KMedoid modeling')
-    elif regime=='segment': plt.title('Segment  $K_\mathrm{S}$ KMedoid modeling')
-    else: plt.title('Basecluster $K_\mathrm{S}$ KMedoid modeling')
+    if regime=='Multiplicon': plt.title('Multilplicon $K_\mathrm{S}$ KMedoid modeling')
+    elif regime=='Segment': plt.title('Segment $K_\mathrm{S}$ KMedoid modeling')
+    #else: plt.title('Basecluster $K_\mathrm{S}$ KMedoid modeling')
     plt.tight_layout()
     plt.savefig(fname,format ='pdf', bbox_inches='tight')
     plt.close()
 
 def plot_kmedoids(boots,kdemethod,dfo,outdir,n,bin_width,bins=50,weighted=True,title="",plot='identical',alpha=0.50,regime='multiplicon',user_xlim=None,user_ylim=None):
     fname = os.path.join(outdir,"{}-guided_AnchorKs_KMedoids_Clustering_{}components.pdf".format(regime,n))
     f, ax = plt.subplots()
@@ -1282,17 +1195,17 @@
         plt.hist(Xs,bins = np.linspace(0, 5, num=int(5/bin_width)+1),color=cs,alpha=0.5,rwidth=0.8,stacked=True,label='component {}'.format(i))
     plt.xlabel("$K_\mathrm{S}$", fontsize = 10)
     plt.ylabel("Frequency", fontsize = 10)
     ax.legend(loc=1,fontsize='large',frameon=False)
     if not (user_ylim[0]) is None: ax.set_ylim(user_ylim[0],user_ylim[1])
     if not (user_xlim[0]) is None: ax.set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
-    if regime=='multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
-    elif regime=='segment': plt.title('Segment-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
-    else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
+    if regime=='Multiplicon': plt.title('Multilplicon-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
+    elif regime=='Segment': plt.title('Segment-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
+    #else: plt.title('Basecluster-guided Anchor $K_\mathrm{S}$ KMedoid modeling')
     plt.tight_layout()
     plt.savefig(fname,format ='pdf', bbox_inches='tight')
     plt.close()
 
 def getX(df,column,cutoff=5.0):
     X = np.array(df.loc[:,column].dropna())
     X = X[np.isfinite(X)]
@@ -1307,16 +1220,17 @@
             if l == i:
                 sum_d = sum_d + (x - c)**2
         D.append(sum_d)
     Loss = sum(D)
     return Loss
 
 def find_mpeak(df,anchor,sp,outdir,guide,peak_threshold=0.1,rel_height=0.4,ci=95,user_low=0,user_upp=1,user=False,kscutoff=5, keeptmp=False):
-    outdir = _mkdir(os.path.join(outdir,"SegmentKs_FindPeak"))
-    gs_ks = df.loc[:,['gene1','gene2',guide,'dS']]
+    outdir = _mkdir(os.path.join(outdir,"{}Ks_FindPeak".format(guide)))
+    y = lambda x:x[0].lower()+x[1:]
+    gs_ks = df.loc[:,['gene1','gene2',y(guide),'dS']]
     df_withindex,ks_or = bc_group_anchor(df,regime=guide)
     mpKs = pd.DataFrame.from_dict({guide:df_withindex.index,'Median_Ks':ks_or}).set_index(guide)
     df_m = df_withindex.copy()
     df_m['weightoutlierexcluded'] = 1
     w = np.array(df_m['weightoutlierexcluded'])
     ks = np.log(ks_or)
     fig, ax = plt.subplots(figsize=(10, 10))
@@ -1342,69 +1256,77 @@
     ax.set_ylim(0, ax.get_ylim()[1] * 1.1)
     fig.tight_layout()
     if keeptmp: fig.savefig(os.path.join(outdir, "{}_guided_{}_Ks_spline.pdf".format(guide,sp)))
     plt.close(fig)
     logging.info('Detecting likely peaks from {}-guided Ks data '.format(guide))
     if keeptmp: init_means, init_stdevs, good_prominences = find_peak_init_parameters(spl_x,spl_y,sp,outdir,peak_threshold=peak_threshold,guide=guide,rel_height=rel_height)
     else: init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,peak_threshold=peak_threshold,rel_height=rel_height)
-    #init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,sp,outdir,peak_threshold=peak_threshold,guide=guide,rel_height=rel_height)
     lower95CI,upper95CI = plot_95CI_lognorm_hist(init_means, init_stdevs, ks_or, w, outdir, False, sp, ci=ci,guide=guide,keeptmp=True)
     if user: get95CIap_MP(user_low,user_upp,anchor,gs_ks,outdir,sp,ci,guide,mpKs,user=user,kscutoff=kscutoff)
     else: get95CIap_MP(lower95CI,upper95CI,anchor,gs_ks,outdir,sp,ci,guide,mpKs,user=user,kscutoff=kscutoff)
 
-def find_apeak(df,anchor,sp,outdir,peak_threshold=0.1,na=False,rel_height=0.4,ci=95,user_low=0,user_upp=1,user=False,kscutoff=5,keeptmp=False):
-    outdir = _mkdir(os.path.join(outdir,"AnchorKs_FindPeak"))
+def find_apeak(df,anchor,sp,outdir,peak_threshold=0.1,na=False,rel_height=0.4,ci=95,user_low=0,user_upp=1,user=False,kscutoff=5,keeptmp=False,withfig=True,loglabel=None):
+    if withfig: outdir = _mkdir(os.path.join(outdir,"AnchorKs_FindPeak"))
     gs_ks = df.loc[:,['gene1','gene2','dS']]
     if na:
         df = df.drop_duplicates(subset=['family','node'])
         df = df.loc[:,['node_averaged_dS_outlierexcluded']].copy().rename(columns={'node_averaged_dS_outlierexcluded':'dS'})
         df['weightoutlierexcluded'] = 1
     df = df.dropna(subset=['dS','weightoutlierexcluded'])
     df = df.loc[(df['dS']>0) & (df['dS']<5),:]
     ks_or = np.array(df['dS'])
     w = np.array(df['weightoutlierexcluded'])
     ks = np.log(ks_or)
-    fig, ax = plt.subplots(figsize=(10, 10))
-    ax.set_xlim(-5,2)
-    ax.set_title('KDE and spline of log-transformed anchor $K_\mathrm{S}$ of species '+ '{}'.format(sp))
-    ax.set_xlabel("ln $K_\mathrm{S}$")
-    ax.set_ylabel("Density of retained duplicates")
+    if withfig:
+        fig, ax = plt.subplots(figsize=(10, 10))
+        ax.set_xlim(-5,2)
+        ax.set_title('KDE and spline of log-transformed anchor $K_\mathrm{S}$ of species '+ '{}'.format(sp))
+        ax.set_xlabel("ln $K_\mathrm{S}$")
+        ax.set_ylabel("Density of retained duplicates")
     max_ks,min_ks = ks.max(),ks.min()
     ks_refed,cutoff,w_refed = reflect_logks(ks,w)
     kde = stats.gaussian_kde(ks_refed, bw_method="scott", weights=w_refed)
     bw_modifier = 0.4
     kde.set_bandwidth(kde.factor * bw_modifier)
     kde_x = np.linspace(min_ks-cutoff, max_ks+cutoff,num=500)
     kde_y = kde(kde_x)
-    ax.plot(kde_x, kde_y, color="k", lw=1, label="KDE")
+    if withfig: ax.plot(kde_x, kde_y, color="k", lw=1, label="KDE")
     spl = interpolate.UnivariateSpline(kde_x, kde_y)
     spl.set_smoothing_factor(0.01)
     spl_x = np.linspace(min_ks, max_ks+0.1, num=int(round((abs(min_ks) + (max_ks+0.1)) *100)))
     spl_y = spl(spl_x)
-    ax.plot(kde_x, spl(kde_x), 'b', lw=1, label="Spline on KDE")
-    ax.hist(ks_refed, weights=w_refed, bins=np.arange(-5, 2.1, 0.1), color="gray", alpha=0.8, density=True,rwidth=0.8)
-    ax.legend(loc=2,fontsize='large',frameon=False)
-    ax.set_ylim(0, ax.get_ylim()[1] * 1.1)
-    fig.tight_layout()
+    if withfig:
+        ax.plot(kde_x, spl(kde_x), 'b', lw=1, label="Spline on KDE")
+        ax.hist(ks_refed, weights=w_refed, bins=np.arange(-5, 2.1, 0.1), color="gray", alpha=0.8, density=True,rwidth=0.8)
+        ax.legend(loc=2,fontsize='large',frameon=False)
+        ax.set_ylim(0, ax.get_ylim()[1] * 1.1)
+        fig.tight_layout()
+    if withfig:
+        if na:
+            if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_node_averaged.svg".format(sp)))
+            if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_node_averaged.pdf".format(sp)))
+        else:
+            if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_weighted.svg".format(sp)))
+            if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_weighted.pdf".format(sp)))
+        plt.close(fig)
     if na:
-        if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_node_averaged.svg".format(sp)))
-        if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_node_averaged.pdf".format(sp)))
+        if loglabel is None: logging.info('Detecting likely peaks from node-averaged data of the whole anchor Ks distribution')
+        else: logging.info('Detecting likely peaks from node-averaged data of {}'.format(loglabel))
     else:
-        if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_weighted.svg".format(sp)))
-        if keeptmp: fig.savefig(os.path.join(outdir, "{}.spline_weighted.pdf".format(sp)))
-    plt.close(fig)
-    if na: logging.info('Detecting likely peaks from node-averaged data')
-    else: logging.info('Detecting likely peaks from node-weighted data')
+        if loglabel is None: logging.info('Detecting likely peaks from node-weighted data of the whole anchor Ks distribution')
+        else: logging.info('Detecting likely peaks from node-weighted data of {}'.format(loglabel))
     if keeptmp: init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,sp,outdir,peak_threshold=peak_threshold,na=na,rel_height=rel_height)
     else: init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,peak_threshold=peak_threshold,rel_height=rel_height)
-    #init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,sp,outdir,peak_threshold=peak_threshold,na=na,rel_height=rel_height)
-    #lower95CI,upper95CI = plot_95CI_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp)
-    lower95CI,upper95CI = plot_95CI_lognorm_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp, ci=ci, keeptmp=True)
-    if user: get95CIap(user_low,user_upp,anchor,gs_ks,outdir,na,sp,ci,user=user,kscutoff=kscutoff)
-    else: get95CIap(lower95CI,upper95CI,anchor,gs_ks,outdir,na,sp,ci,user=user,kscutoff=kscutoff)
+    if withfig:
+        lower95CI,upper95CI = plot_95CI_lognorm_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp, ci=ci, keeptmp=True)
+        if user: get95CIap(user_low,user_upp,anchor,gs_ks,outdir,na,sp,ci,user=user,kscutoff=kscutoff)
+        else: get95CIap(lower95CI,upper95CI,anchor,gs_ks,outdir,na,sp,ci,user=user,kscutoff=kscutoff)
+    else:
+        lower95CI,upper95CI = noplot_95CI_lognorm_hist(init_means, init_stdevs, ci=ci)
+        return lower95CI,upper95CI
 
 def get95CIap(lower,upper,anchor,gs_ks,outdir,na,sp,ci,user=False,kscutoff=5):
     if type(lower) == float:
         upper = min([upper,kscutoff])
         ap_95CI = gs_ks.loc[(gs_ks['dS']<=upper) & (gs_ks['dS']>=lower),:]
         sp_m = '{}'.format(sp)
         if na: fname = os.path.join(outdir, "{}_Manual_CI_AP_for_dating_node_averaged.tsv".format(sp_m))
@@ -1470,15 +1392,16 @@
     df = df.reset_index().merge(predict_column, on = [regime])
     df = df.set_index('pair')
     fname = os.path.join(outdir,'AnchorKs_GMM_{}components_prediction.tsv'.format(n))
     df.to_csv(fname,header=True,index=True,sep='\t')
     return df
 
 def get95CIap_MP(lower,upper,anchor,gs_ks,outdir,sp,ci,guide,mpKs,user=False,kscutoff=5):
-    gs_ks = gs_ks.reset_index().set_index(guide).join(mpKs)
+    y = lambda x:x[0].lower()+x[1:]
+    gs_ks = gs_ks.reset_index().set_index(y(guide)).join(mpKs)
     if type(lower) == float:
         upper = min([kscutoff,upper])
         ap_95CI = gs_ks.loc[(gs_ks['Median_Ks']<=upper) & (gs_ks['Median_Ks']>=lower),:]
         sp_m = '{}_guided_{}'.format(guide,sp)
         fname = os.path.join(outdir, "{}_Manual_CI_MP_for_dating.tsv".format(sp_m))
         ap_95CI.to_csv(fname,header=True,index=True,sep='\t')
         anchors = pd.read_csv(anchor, sep="\t", index_col=0)
@@ -1517,15 +1440,14 @@
             if user: fname = os.path.join(outdir, "{}_Manual_CI_MP_for_dating_format.tsv".format(sp_m))
             else: fname = os.path.join(outdir, "{}_{}%CI_MP_for_dating_format.tsv".format(sp_m,ci))
             ap_format.to_csv(fname,header=True,index=True,sep='\t')
 
 
 def plot_95CI_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp, guide = None):
     text = "AnchorKs_PeakCI_"
-    #if guide != None: text = "AnchorKs_PeakCI_{}_guided_".format(guide)
     if guide != None: fname = os.path.join(outdir, "{}{}_guided_{}.pdf".format(text,guide,sp))
     elif na: fname = os.path.join(outdir, "{}{}_node_averaged.pdf".format(text,sp))
     else: fname = os.path.join(outdir, "{}{}_node_weighted.pdf".format(text,sp))
     f, ax = plt.subplots()
     kdesity = 100
     bins = 50
     bin_width = 0.1
@@ -1545,14 +1467,23 @@
     sns.despine(offset=1)
     plt.title('Anchor $K_\mathrm{S}$'+' distribution of {}'.format(sp))
     plt.tight_layout()
     plt.savefig(fname,format ='pdf', bbox_inches='tight')
     plt.close()
     return np.exp(mean)-std*2,np.exp(mean)+std*2
 
+def noplot_95CI_lognorm_hist(init_means, init_stdevs, ci=95):
+    ci_l = (1-ci/100)/2
+    ci_u = 1-(1-ci/100)/2
+    CI_95s = []
+    for mean,std,i in zip(init_means, init_stdevs,range(len(init_means))):
+        CI_95 = stats.lognorm.ppf([ci_l, ci_u], scale=np.exp(mean), s=std)
+        CI_95s.append(CI_95) # There is supposed to be only one peak detected provided the GMM component
+    return CI_95s[0][0],CI_95s[0][1]
+
 def plot_95CI_lognorm_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp, guide = None, ci=95, keeptmp=False):
     if guide != None: fname = os.path.join(outdir, "{}Ks_PeakCI_{}.pdf".format(guide,sp))
     elif na: fname = os.path.join(outdir, "AnchorKs_PeakCI_{}_node_averaged.pdf".format(sp))
     else: fname = os.path.join(outdir, "AnchorKs_PeakCI_{}_node_weighted.pdf".format(sp))
     f, ax = plt.subplots()
     x_points_strictly_positive = np.linspace(0, 5, int(5 * 100))
     bin_width = 0.1
@@ -1569,16 +1500,17 @@
         CI_95 = stats.lognorm.ppf([ci_l, ci_u], scale=np.exp(mean), s=std)
         CI_95s.append(CI_95)
         CI_95_y0 = scaling*stats.lognorm.pdf(CI_95[0], scale=np.exp(mean),s=std)/ax.get_ylim()[1]
         CI_95_y1 = scaling*stats.lognorm.pdf(CI_95[1], scale=np.exp(mean),s=std)/ax.get_ylim()[1]
         plt.axvline(x = CI_95[0], ymin=0, ymax=CI_95_y0, color = cs[i], alpha = 0.8, ls = ':', lw = 1,label='Peak {} lower {}%CI {:.2f}'.format(i+1,ci,CI_95[0]))
         plt.axvline(x = CI_95[1], ymin=0, ymax=CI_95_y1, color = cs[i], alpha = 0.8, ls = ':', lw = 1,label='Peak {} upper {}%CI {:.2f}'.format(i+1,ci,CI_95[1]))
     plt.xlabel("$K_\mathrm{S}$", fontsize = 10)
-    if guide == 'segment': plt.ylabel("Number of segment pair", fontsize = 10)
-    elif guide != None: plt.ylabel("Number of {}".format(guide), fontsize = 10)
+    y = lambda x:x[0].lower()+x[1:]
+    if guide == 'Segment': plt.ylabel("Number of segment pair", fontsize = 10)
+    elif guide != None: plt.ylabel("Number of {}".format(y(guide)), fontsize = 10)
     elif na: plt.ylabel("Number of retained duplicates (node averaged)", fontsize = 10)
     else: plt.ylabel("Number of retained duplicates (weighted)", fontsize = 10)
     ax.legend(loc=1,fontsize='large',frameon=False)
     ax.set_xlim(0, 5)
     sns.despine(offset=1)
     plt.title('Anchor $K_\mathrm{S}$'+' distribution of {}'.format(sp))
     plt.tight_layout()
@@ -1614,36 +1546,40 @@
     df = anchors[["pair", "basecluster",'multiplicon']].drop_duplicates("pair").set_index("pair")
     return df
 
 def get_anchor_ksd(ksdf, apdf):
     return ksdf.join(apdf).dropna()
     # here any NA occurred per row is dropped, normally a row has all NaN or only weightexclu and nodeKsexcl NaN
 
-def bc_group_anchor(df,regime='multiplicon'):
+def bc_group_anchor(df,regime='Multiplicon'):
     #median_df = df.groupby(["basecluster"]).median()
+    y = lambda x: x[0].lower()+x[1:]
+    df = df.rename(columns={y(regime):regime})
     median_df = df.groupby([regime]).median()
     X = getX(median_df,'dS')
     return median_df,X
 
-def add_apgmmlabels(df,df_index,labels,outdir,n,regime='multiplicon'):
+def add_apgmmlabels(df,df_index,labels,outdir,n,regime='Multiplicon',guided=False):
     predict_column = pd.DataFrame(labels,index=df_index.index,columns=['AnchorKs_GMM_Component']).reset_index()
+    y = lambda x: x[0].lower()+x[1:]
+    df = df.rename(columns={y(regime):regime})
     df = df.reset_index().merge(predict_column, on = [regime])
     df = df.set_index('pair')
     fname = os.path.join(outdir,'{}-guided_AnchorKs_GMM_{}components_prediction.tsv'.format(regime,n))
-    df.to_csv(fname,header=True,index=True,sep='\t')
+    if guided: df.copy().rename(columns={"AnchorKs_GMM_Component":"{}_GMM_Component".format(regime)}).to_csv(fname,header=True,index=True,sep='\t')
     return df
 
 def add_apgmmlabels_pairs(df,df_index,labels,outdir,n):
     predict_column = pd.DataFrame(labels,index=df_index,columns=['AnchorKs_GMM_Component'])
     df = df.join(predict_column)
     fname = os.path.join(outdir,'Original_AnchorKs_GMM_{}components_prediction.tsv'.format(n))
     df.to_csv(fname,header=True,index=True,sep='\t')
     return df
 
-def add_seg(df,listelement,multipliconpairs,segment):
+def add_seg(df,listelement,multipliconpairs,segment,mul=False):
     #Here the df should be Ks 0-5 filter
     #It hasn't to be ap in listelement note that a same pair of syntelogs can be in different multiplicon
     mp = pd.read_csv(multipliconpairs, sep="\t", index_col=0)
     if len(mp.columns) == 5: mp = mp.drop(columns=['gene_y']).rename(columns = {'gene_x':'gene_y'}).rename(columns = {'Unnamed: 2':'gene_x'})
     mp = mp.loc[:,['multiplicon','gene_x','gene_y']]
     mp.loc[:,['pair']] = ["__".join(sorted([x,y])) for x,y in zip(mp['gene_x'],mp['gene_y'])]
     #mp = mp.drop_duplicates(subset=['pair']) # at this step half of the data will be filtered out
@@ -1663,20 +1599,28 @@
     df_mp_le_r = df_mp_le_r.rename(columns = {'segment':'segment_y'})
     #df_mp_le_r = df_mp_le_r.drop_duplicates(subset=['id'],keep=False)
     df_mp_le_r = df_mp_le_r.drop_duplicates() # only drop those fully duplicated
     df_mp_le_r = df_mp_le_r.loc[:,['multiplicon','gene_x','gene_y','segment_x','segment_y','pair']].set_index('pair')
     df_mp_le_r.loc[:,['segment_pair']] = ["__".join(sorted([x,y])) for x,y in zip(list(df_mp_le_r['segment_x'].astype(str)),list(df_mp_le_r['segment_y'].astype(str)))]
     df_cal_medKs = df_mp_le_r.loc[:,['segment_pair','multiplicon']].join(df).rename(columns = {'segment_pair':'segment'}).dropna(subset=['dS'])
     # here I filter  Ks >= 5
-    dS_median = df_cal_medKs.groupby(['segment'])['dS'].median()
-    seg_weight = {seg:len(df_tmp) for seg, df_tmp in list(df_cal_medKs.groupby(['segment']))}
-    dS_median = dS_median[dS_median<5]
-    dS_median.name = 'Segment_dS'
-    df_cal_medKs = df_cal_medKs.reset_index().merge(dS_median.reset_index(),on='segment').dropna(subset=['Segment_dS']).set_index('pair')
-    segment_weight = list(df_cal_medKs['segment'].apply(lambda x:seg_weight[x]))
+    if mul:
+        dS_median = df_cal_medKs.groupby(['multiplicon'])['dS'].median()
+        mul_weight = {ml:len(df_tmp) for ml, df_tmp in list(df_cal_medKs.groupby(['multiplicon']))}
+        dS_median = dS_median[dS_median<5]
+        dS_median.name = 'Multiplicon_dS'
+        df_cal_medKs = df_cal_medKs.reset_index().merge(dS_median.reset_index(),on='multiplicon').dropna(subset=['Multiplicon_dS']).set_index('pair')
+        multiplicon_weight = list(df_cal_medKs['multiplicon'].apply(lambda x:mul_weight[x]))
+    else:
+        dS_median = df_cal_medKs.groupby(['segment'])['dS'].median()
+        seg_weight = {seg:len(df_tmp) for seg, df_tmp in list(df_cal_medKs.groupby(['segment']))}
+        dS_median = dS_median[dS_median<5]
+        dS_median.name = 'Segment_dS'
+        df_cal_medKs = df_cal_medKs.reset_index().merge(dS_median.reset_index(),on='segment').dropna(subset=['Segment_dS']).set_index('pair')
+        segment_weight = list(df_cal_medKs['segment'].apply(lambda x:seg_weight[x]))
     #df_cal_medKs = df.join(df_mp_le_r.loc[:,['segment_pair','multiplicon']]).rename(columns = {'segment_pair':'segment'}) # here I did keep the NaN value, which impacted and changed the median Ks values afterwards
     #df_mp_le_r = mp.merge(df_le,left_on='gene_y',right_on='gene')
     #df_mp_le_r = df_mp_le_r[df_mp_le_r['multiplicon_x'] == df_mp_le_r['multiplicon']]
     #df_mp_le_r = df_mp_le_r.rename(columns = {'segment':'segment_y'})
     #df_mp_le_lr = df_mp_le.loc[:,['multiplicon','gene_x','gene_y','segment_x','pair']].merge(df_mp_le_r.loc[:,['pair','segment_y']],on='pair').set_index('pair')
     #df_mp_le_lr.loc[:,['segment_pair']] = ["__".join(sorted([x,y])) for x,y in zip(list(df_mp_le_lr['segment_x'].astype(str)),list(df_mp_le_lr['segment_y'].astype(str)))]
     #df_cal_medKs = df.join(df_mp_le_lr.loc[:,['segment_pair','multiplicon']]).dropna().rename(columns = {'segment_pair':'segment'})
@@ -1688,15 +1632,18 @@
     #df_cal_medKs = df_withKs.drop_duplicates(subset=['segment','gene_y']).drop_duplicates(subset=['segment','gene_x'])
     #df_cal_medKs = df_cal_medKs.drop(columns=['gene_y','multiplicon_x','gene_x','gene'])
     #Here we retrive the duplicates information for segment age clustering
     #dup_le = df_withKs[df_withKs.duplicated(subset=['segment','gene_y'], keep=False)]
     #dup_ri = df_withKs[df_withKs.duplicated(subset=['segment','gene_x'], keep=False)]
     #dup_combined = pd.concat([dup_le,dup_ri], ignore_index=True).drop_duplicates()
     #dup_combined_work = dup_combined.loc[:,['segment','gene_x','gene_y','dS']]
-    return df_cal_medKs, segment_weight
+    if mul:
+        return df_cal_medKs, multiplicon_weight
+    else:
+        return df_cal_medKs, segment_weight
 
 def getDM(X):
     dm = np.zeros((len(X), len(X)))
     for i in range(len(X)):
         for j in range(i+1, len(X)):
             dm[i,j] = dm[j,i] = (abs(X[i][0] - X[j][0]))**2
     return dm
@@ -1786,92 +1733,96 @@
 
 def weighttransform(X_log,seg_weight):
     new_X_log = np.array([])
     for ks,times in zip(X_log,seg_weight):
         new_X_log = np.append(new_X_log,np.repeat(ks, times))
     return new_X_log.reshape(-1, 1)
 
-def fit_apgmm_guide(hdr,guide,anchor,df_nofilter,dfor,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,segment=None,multipliconpairs=None,listelement=None,cutoff=None,user_xlim=None,user_ylim=None):
-    outdir = _mkdir(os.path.join(outdir,"SegmentGuideKs_GMM"))
-    logging.info("GMM modeling on Log-scale segment Ks data")
+def fit_apgmm_guide(hdr,guide,anchor,df_nofilter,dfor,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,segment=None,multipliconpairs=None,listelement=None,cutoff=None,user_xlim=None,user_ylim=None,peak_threshold=0.1,rel_height=0.4,keeptmp=False):
+    outdir = _mkdir(os.path.join(outdir,"{}GuideKs_GMM".format(guide)))
+    logging.info("Gaussian Mixture Modeling (GMM) on Log-scale {} Ks data".format(guide))
     if anchor == None:
         logging.error('Please provide anchorpoints.txt file for Anchor Ks GMM Clustering')
         exit(0)
     df_ap = get_anchors(anchor)
     df = get_anchor_ksd(dfor, df_ap)
     df_mp = get_mp_ksd(multipliconpairs,df_nofilter)
     df_nofilter = df_nofilter[df_nofilter['dS']>0]
     if segment!= None:
-        df, seg_weight = add_seg(df_nofilter,listelement,multipliconpairs,segment)
-        df.to_csv(os.path.join(outdir, "Segment_Ks.tsv"),header=True,index=True,sep='\t')
+        if guide == 'Multiplicon':
+            df, mul_weight = add_seg(df_nofilter,listelement,multipliconpairs,segment,mul=True)
+            df.to_csv(os.path.join(outdir, "Multiplicon_Ks.tsv"),header=True,index=True,sep='\t')
+        else:
+            df, seg_weight = add_seg(df_nofilter,listelement,multipliconpairs,segment)
+            df.to_csv(os.path.join(outdir, "Segment_Ks.tsv"),header=True,index=True,sep='\t')
     df_withindex,X = bc_group_anchor(df,regime=guide)
     X_log = np.log(X).reshape(-1, 1)
     out_file = os.path.join(outdir, "{}_Ks_GMM_AIC_BIC.pdf".format(guide))
     #X_log = weighttransform(X_log,seg_weight)
     if method == 'gmm': models, aic, bic, besta, bestb, N = fit_gmm(out_file, X_log, seed, components[0], components[1], em_iter=em_iter, n_init=n_init)
     if method == 'bgmm': models, N = fit_bgmm(X_log, seed, gamma, components[0], components[1], em_iter=em_iter, n_init=n_init)
     if components[0] == 1 and components[1] > 1:
         plot_silhouette_score(X_log,components[0]+1,components[1],[m.predict(X_log) for m in models][1:],outdir,guide+'_Ks','GMM')
         #significance_test_cluster(X_log,components[0]+1,components[1],[m.predict(X_log) for m in models][1:])
     else:
         plot_silhouette_score(X_log,components[0],components[1],[m.predict(X_log) for m in models],outdir,guide+'_Ks','GMM')
         #significance_test_cluster(X_log,components[0],components[1],[m.predict(X_log) for m in models])
     Losses = []
+    y = lambda x:x[0].lower()+x[1:]
     for n, m in zip(N,models):
         labels = m.predict(X_log)
-        df_c = add_apgmmlabels(df,df_withindex,labels,outdir,n,regime=guide)
+        df_c = add_apgmmlabels(df,df_withindex,labels,outdir,n,regime=guide,guided=True)
         means,stds,weights = m.means_,m.covariances_,m.weights_
         Losses.append(Elbow_lossf(X_log,[i[0] for i in means],labels))
-        fig = plot_mp_component(X,labels,n,bins=50,plot = plot,ylabel="Number of segment pair",regime=guide,user_xlim=user_xlim,user_ylim=user_ylim)
+        fig = plot_mp_component(X,labels,n,bins=50,plot = plot,ylabel="Number of {} pair".format(y(guide)),regime=guide,user_xlim=user_xlim,user_ylim=user_ylim)
         fname = os.path.join(outdir, "{}_Ks_Clusters_GMM_Component{}.pdf".format(guide,n))
         fig.savefig(fname)
         plt.close()
-        fig, kdexs, kdeys, modes = plot_mp_component_lognormal(X,hdr,means,stds,weights,labels,n,bins=50,ylabel="Number of segment pair",regime=guide,user_xlim=user_xlim,user_ylim=user_ylim)
+        fig, kdexs, kdeys, modes = plot_mp_component_lognormal(X,hdr,means,stds,weights,labels,n,bins=50,ylabel="Number of {} pair".format(y(guide)),regime=guide,user_xlim=user_xlim,user_ylim=user_ylim)
         #hdr_ap(hdr,df,df_withindex)
         fname = os.path.join(outdir, "{}_Ks_Clusters_Lognormal_GMM_Component{}.pdf".format(guide,n))
         fig.savefig(fname)
         plt.close()
-        #fig = plot_mpbackap_lognormal(df_mp,kdexs,kdeys,modes,guide)
-        #fname = os.path.join(outdir, "Syntelogs_Ks_Clusters_From_{}_Component{}.pdf".format(guide,n))
-        #fig.savefig(fname)
-        plt.close()
-    #for n, m in zip(N,models):
         fig = plot_ak_component(df_c.dropna(),n,bins=50,plot = plot,ylabel="Duplication events",weighted=weighted,regime=guide,user_xlim=user_xlim,user_ylim=user_ylim)
-        if weighted: fname = os.path.join(outdir, "{}-guided_AnchorKs_GMM_Component{}_node_weighted.pdf".format(guide,n))
-        else: fname = os.path.join(outdir, "{}-guided_AnchorKs_GMM_Component{}_node_averaged.pdf".format(guide,n))
+        outdir2 = _mkdir(os.path.join(outdir,"HDR_CI"))
+        if weighted: fname = os.path.join(outdir2, "{}_guided_AnchorKs_GMM_Component{}_node_weighted.pdf".format(guide,n))
+        else: fname = os.path.join(outdir2, "{}_guided_AnchorKs_GMM_Component{}_node_averaged.pdf".format(guide,n))
         fig.savefig(fname)
         plt.close()
         fig,HDRs = plot_ak_component_kde(df_c.dropna(),n,hdr,bins=50,ylabel="Duplication events",weighted=weighted,regime=guide,user_xlim=user_xlim,user_ylim=user_ylim)
-        getGuided_AP_HDR(HDRs,hdr,n,df_c,outdir,guide,cutoff)
-        if weighted: fname = os.path.join(outdir, "{}-guided_AnchorKs_GMM_Component{}_node_weighted_kde.pdf".format(guide,n))
-        else: fname = os.path.join(outdir, "{}-guided_AnchorKs_GMM_Component{}_node_averaged_kde.pdf".format(guide,n))
+        getGuided_AP_HDR(HDRs,hdr,n,df_c,outdir2,guide,cutoff)
+        if weighted: fname = os.path.join(outdir2, "{}_guided_AnchorKs_GMM_Component{}_node_weighted_kde.pdf".format(guide,n))
+        else: fname = os.path.join(outdir2, "{}_guided_AnchorKs_GMM_Component{}_node_averaged_kde.pdf".format(guide,n))
         fig.savefig(fname)
         plt.close()
+        fig2,ax2,kdes,scalings = plot_ak_component_kde(df_c.dropna(),n,hdr,bins=50,ylabel="Duplication events",weighted=weighted,regime=guide,user_xlim=user_xlim,user_ylim=user_ylim,hidehdr=True,notscale=True)
+        get_peak_SegGuidedGMM(guide,kdes,scalings,fig2,ax2,df_c,n,outdir,peak_threshold,weighted,rel_height,keeptmp=keeptmp)
     plot_Elbow_loss(Losses,outdir,n1=components[0],n2=components[1],method='GMM',regime=guide)
     return df
 
 def getGuided_AP_HDR(HDRs,hdr,n,df_c,outdir,regime,cutoff):
     for num in HDRs.keys():
         df_tmp = df_c[df_c['AnchorKs_GMM_Component']==num]
         df_tmp = df_tmp.loc[(df_tmp['dS']<=min([cutoff,HDRs[num][1]])) & (df_tmp['dS']>=HDRs[num][0]),:]
-        df_tmp = df_tmp.loc[:,['gene1','gene2','dS','Segment_dS','AnchorKs_GMM_Component']].rename(columns={"gene1":"gene_x","gene2":"gene_y"})
-        fname = os.path.join(outdir,"{}_guided_{}%HDR_Anchor_Pairs_Component{}_Model{}_WGDating.tsv".format(regime,hdr,num,n))
+        df_tmp = df_tmp.loc[:,['gene1','gene2','dS','{}_dS'.format(regime),'AnchorKs_GMM_Component']].rename(columns={"gene1":"gene_x","gene2":"gene_y"})
+        fname = os.path.join(outdir,"{}_guided_{}%HDR_AP_{}components_C{}.tsv".format(regime,hdr,n,num))
+        df_tmp = df_tmp.rename(columns={'AnchorKs_GMM_Component':'{}Ks_GMM_Component'.format(regime)})
         df_tmp.to_csv(fname,sep='\t',header=True,index=True)
 
-def fit_apgmm_ap(hdr,anchor,df,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,heuristic,showCI=False,cutoff = 5,peak_threshold=0.1,rel_height=0.4,user_xlim=None,user_ylim=None):
+def fit_apgmm_ap(hdr,anchor,df,seed,components,em_iter,n_init,outdir,method,gamma,weighted,plot,showCI=True,cutoff = 5,peak_threshold=0.1,rel_height=0.4,user_xlim=None,user_ylim=None,keeptmp=False):
     outdir = _mkdir(os.path.join(outdir,"AnchorKs_GMM"))
     if anchor == None:
         logging.error('Please provide anchorpoints.txt file for Anchor Ks GMM Clustering')
         exit(0)
     df_ap = get_anchors(anchor)
     df = get_anchor_ksd(df, df_ap)
     df_withindex,X = df.index,getX(df,'dS')
     X_log = np.log(X).reshape(-1, 1)
     out_file = os.path.join(outdir, "Original_AnchorKs_GMM_AIC_BIC.pdf")
-    logging.info("GMM modeling on Log-scale original anchor Ks data")
+    logging.info("Gaussian Mixture Modeling (GMM) on Log-scale original anchor Ks data")
     if method == 'gmm': models, aic, bic, besta, bestb, N = fit_gmm(out_file, X_log, seed, components[0], components[1], em_iter=em_iter, n_init=n_init)
     if method == 'bgmm': models, N = fit_bgmm(X_log, seed, gamma, components[0], components[1], em_iter=em_iter, n_init=n_init)
     if components[0] == 1 and components[1] > 1:
         plot_silhouette_score(X_log,components[0]+1,components[1],[m.predict(X_log) for m in models][1:],outdir,'Original_AnchorKs','GMM')
         #significance_test_cluster(X_log,components[0]+1,components[1],[m.predict(X_log) for m in models][1:])
     else:
         plot_silhouette_score(X_log,components[0],components[1],[m.predict(X_log) for m in models],outdir,'Original_AnchorKs','GMM')
@@ -1883,44 +1834,103 @@
         Losses.append(Elbow_lossf(X_log,[i[0] for i in means],labels))
         df_c = add_apgmmlabels_pairs(df,df_withindex,labels,outdir,n)
         fig = plot_ak_component(df_c,n,bins=50,plot = plot,ylabel="Duplication events",weighted=weighted,regime='original',user_xlim=user_xlim,user_ylim=user_ylim)
         if weighted: fname = os.path.join(outdir, "Original_AnchorKs_GMM_Component{}_node_weighted.pdf".format(n))
         else: fname = os.path.join(outdir, "Original_AnchorKs_GMM_Component{}_node_averaged.pdf".format(n))
         fig.savefig(fname)
         plt.close()
-        #fig = plot_ak_component_kde(df_c,n,bins=50,ylabel="Duplication events",weighted=weighted,regime='original')
-        #if weighted: fname = os.path.join(outdir, "Original_AnchorKs_GMM_Component{}_node_weighted_kde.pdf".format(n))
-        #else: fname = os.path.join(outdir, "Original_AnchorKs_GMM_Component{}_node_averaged_kde.pdf".format(n))
-        #fig.savefig(fname)
-        #plt.close()
-        fig, CI = plot_ak_component_lognormal(df_c.dropna(),means,stds,weights,n,bins=50,ylabel="Duplication events",weighted=weighted,regime='original',showCI=showCI,heuristic=heuristic,peak_threshold=peak_threshold,rel_height=rel_height,user_xlim=user_xlim,user_ylim=user_ylim)
-        if showCI or heuristic: df_c_CI = add_apCI(df_c,outdir,CI,n,cutoff)
+        fig, CI, ax, _ = plot_ak_component_lognormal(df_c.dropna(),means,stds,weights,n,bins=50,ylabel="Duplication events",weighted=weighted,regime='original',showCI=showCI,peak_threshold=peak_threshold,rel_height=rel_height,user_xlim=user_xlim,user_ylim=user_ylim,hideci=True)
         if weighted: fname = os.path.join(outdir, "Original_AnchorKs_GMM_Component{}_node_weighted_Lognormal.pdf".format(n))
         else: fname = os.path.join(outdir, "Original_AnchorKs_GMM_Component{}_node_averaged_Lognormal.pdf".format(n))
         fig.savefig(fname)
         plt.close()
+        fig2, CI, ax2, _ = plot_ak_component_lognormal(df_c.dropna(),means,stds,weights,n,bins=50,ylabel="Duplication events",weighted=weighted,regime='original',showCI=showCI,peak_threshold=peak_threshold,rel_height=rel_height,user_xlim=user_xlim,user_ylim=user_ylim)
+        if weighted: fname = os.path.join(_mkdir(os.path.join(outdir,"LogGMM_CI")), "GMM_Component{}_node_weighted_Lognormal.pdf".format(n))
+        else: fname = os.path.join(outdir,"LogGMM_CI", "GMM_Component{}_node_averaged_Lognormal.pdf".format(n))
+        if showCI: df_c_CI = add_apCI(df_c,outdir,CI,n,cutoff,weighted)
+        fig2.savefig(fname)
+        plt.close()
+        fig3, CI, ax3, scalings = plot_ak_component_lognormal(df_c.dropna(),means,stds,weights,n,bins=50,ylabel="Duplication events",weighted=weighted,regime='original',showCI=showCI,peak_threshold=peak_threshold,rel_height=rel_height,user_xlim=user_xlim,user_ylim=user_ylim,hideci=True)
+        get_peak_AnchGMM(means,stds,scalings,fig3,ax3,df_c,n,outdir,peak_threshold,weighted,rel_height,keeptmp=keeptmp)
+        plt.close()
     plot_Elbow_loss(Losses,outdir,n1=components[0],n2=components[1],method='GMM',regime='original')
     return df
 
-def add_apCI(df,outdir,CI,n,cutoff):
+def get_peak_SegGuidedGMM(regime,kdes,scalings,fig,ax,df_c,n,outdir,peak_threshold,weighted,rel_height,keeptmp=False):
+    outdir = _mkdir(os.path.join(outdir,"HighMass_CI"))
+    na=False if weighted else True
+    #Components = list(set(df_c['AnchorKs_GMM_Component']))
+    colors = cm.viridis(np.linspace(0, 1, n))
+    for cp,color,scaling,kde in zip(range(n),colors,scalings,kdes):
+        df = df_c[df_c['AnchorKs_GMM_Component']==cp]
+        lower_CI, upper_CI = find_apeak(df,None,None,outdir,peak_threshold=peak_threshold,na=na,rel_height=rel_height,withfig=False,loglabel="{}-Components Model Peak {}".format(n,cp),keeptmp=keeptmp)
+        if upper_CI> 5: upper_CI=5
+        #ax.axvline(lower_CI, color = color, alpha = 0.8, ls = '--', lw = 0.8,label='Peak of c{} {}%CI {:.2f}-{:.2f}'.format(cp,95,lower_CI,upper_CI))
+        #ax.axvline(upper_CI, color = color, alpha = 0.8, ls = '--', lw = 0.8)
+        lower_CI_y0 = scaling*kde(lower_CI)
+        upper_CI_y1 = scaling*kde(upper_CI)
+        ax.plot([lower_CI,lower_CI],[0,lower_CI_y0], color = color, alpha = 0.8, ls = '--', lw = 1,label='HighMass of c{} {}%CI {:.2f}-{:.2f}'.format(cp,95,lower_CI,upper_CI))
+        ax.plot([upper_CI,upper_CI],[0,upper_CI_y1], color = color, alpha = 0.8, ls = '--', lw = 1)
+        if weighted: df = df[(df['dS']>=lower_CI)&(df['dS']<=upper_CI)]
+        else: df = df[(df['node_averaged_dS_outlierexcluded']>=lower_CI)&(df['node_averaged_dS_outlierexcluded']<=upper_CI)]
+        fname = os.path.join(outdir,'{}_guided_{}components_C{}_HighMass_95%CI.tsv'.format(regime,n,cp))
+        df = df.rename(columns={'gene1':'gene_x','gene2':'gene_y','AnchorKs_GMM_Component':'SegmentKs_GMM_Component'})
+        df.to_csv(fname,header=True,index=True,sep='\t')
+    fname = os.path.join(outdir,'{}_guided_{}components_HighMass_95%CI.pdf'.format(regime,n))
+    ax.legend(loc='upper right',frameon=False)
+    fig.tight_layout()
+    fig.savefig(fname)
+
+def get_peak_AnchGMM(means,stds,scalings,fig,ax,df_c,n,outdir,peak_threshold,weighted,rel_height,keeptmp=False):
+    outdir = _mkdir(os.path.join(outdir,"HighMass_CI"))
+    na=False if weighted else True
+    #Components = sorted(list(set(df_c['AnchorKs_GMM_Component'])))
+    colors = cm.viridis(np.linspace(0, 1, n))
+    for cp,color,scaling in zip(range(n),colors,scalings):
+        mean,std = means[cp][0],np.sqrt(stds[cp][0][0])
+        df = df_c[df_c['AnchorKs_GMM_Component']==cp]
+        lower_CI, upper_CI = find_apeak(df,None,None,outdir,peak_threshold=peak_threshold,na=na,rel_height=rel_height,withfig=False,loglabel="{}-Components Model Peak {}".format(n,cp),keeptmp=keeptmp)
+        if upper_CI > 5: upper_CI=5
+        #ax.axvline(lower_CI, color = color, alpha = 0.8, ls = '--', lw = 0.8,label='Peak of c{} {}%CI {:.2f}-{:.2f}'.format(cp,95,lower_CI,upper_CI))
+        #ax.axvline(upper_CI, color = color, alpha = 0.8, ls = '--', lw = 0.8)
+        lower_CI_y0 = scaling*stats.lognorm.pdf(lower_CI, scale=np.exp(mean),s=std)
+        upper_CI_y1 = scaling*stats.lognorm.pdf(upper_CI, scale=np.exp(mean),s=std)
+        ax.plot([lower_CI,lower_CI],[0,lower_CI_y0], color = color, alpha = 0.8, ls = '--', lw = 1,label='HighMass of c{} {}%CI {:.2f}-{:.2f}'.format(cp,95,lower_CI,upper_CI))
+        ax.plot([upper_CI,upper_CI],[0,upper_CI_y1], color = color, alpha = 0.8, ls = '--', lw = 1)
+        if weighted: df = df[(df['dS']>=lower_CI)&(df['dS']<=upper_CI)]
+        else: df = df[(df['node_averaged_dS_outlierexcluded']>=lower_CI)&(df['node_averaged_dS_outlierexcluded']<=upper_CI)]
+        fname = os.path.join(outdir,'GMM_{}components_C{}_HighMass_95%CI.tsv'.format(n,cp))
+        df = df.rename(columns={'gene1':'gene_x','gene2':'gene_y'})
+        df.to_csv(fname,header=True,index=True,sep='\t')
+    fname = os.path.join(outdir,'GMM_{}components_HighMass_95%CI.pdf'.format(n))
+    ax.legend(loc='upper right',frameon=False)
+    fig.tight_layout()
+    fig.savefig(fname)
+
+def add_apCI(df,outdir,CI,n,cutoff,weighted):
+    outdir = _mkdir(os.path.join(outdir,"LogGMM_CI"))
     for categ, content in CI.items():
         if len(content) == 0:
             continue
         for comp, ci in content.items():
             if ci is None:
                 continue
             df_tmp = df[df['AnchorKs_GMM_Component']==comp]
             maxi = min([ci[1],cutoff])
-            df_tmp = df_tmp[ci[0]<=df_tmp['dS']]
-            df_tmp = df_tmp[df_tmp['dS']<=maxi].loc[:,['dS']].copy()
+            if weighted:
+                df_tmp = df_tmp[ci[0]<=df_tmp['dS']]
+                df_tmp = df_tmp[df_tmp['dS']<=maxi].loc[:,['dS']].copy()
+            else:
+                df_tmp = df_tmp[ci[0]<=df_tmp['node_averaged_dS_outlierexcluded']]
+                df_tmp = df_tmp[df_tmp['node_averaged_dS_outlierexcluded']<=maxi].loc[:,['dS']].copy()
             df_tmp['gene_x'], df_tmp['gene_y'] = [pair.split("__")[0] for pair in df_tmp.index], [pair.split("__")[1] for pair in df_tmp.index]
             if categ == 'showci':
-                fname = os.path.join(outdir,'Original_AnchorKs_GMM_{0}components_C{1}_95%CI.tsv'.format(n,comp))
+                fname = os.path.join(outdir,'GMM_{0}components_C{1}_95%CI.tsv'.format(n,comp))
             elif categ == 'heuristic':
-                fname = os.path.join(outdir,'Original_AnchorKs_GMM_{0}components_P{1}_95%CI.tsv'.format(n,comp))
+                fname = os.path.join(outdir,'GMM_{0}components_P{1}_95%CI.tsv'.format(n,comp))
             df_tmp.to_csv(fname,header=True,index=True,sep='\t')
 
 
 def fit_kmedoids(guide,anchor, boots, kdemethod, bin_width, weighted, df_nofilter, df, outdir, seed, n, em_iter=100, metric='euclidean', method='pam', init ='k-medoids++', plot = 'identical', n_kmedoids = 5, segment= None, multipliconpairs=None,listelement=None,user_xlim=None,user_ylim=None):
     """
     Clustering with KMedoids to delineate different anchor groups from anchor Ks distribution
     """
@@ -1928,16 +1938,20 @@
     if anchor == None:
         logging.error('Please provide anchorpoints.txt file for Anchor Ks KMedoids Clustering')
         exit(0)
     df_ap = get_anchors(anchor)
     df = get_anchor_ksd(df, df_ap)
     df_nofilter = df_nofilter[df_nofilter['dS']>0]
     if segment!= None:
-        df = add_seg(df_nofilter,listelement,multipliconpairs,segment)
-        df.to_csv(os.path.join(outdir, "Segment_Ks.tsv"),header=True,index=True,sep='\t')
+        if guide == 'Multiplicon':
+            df, mul_weight = add_seg(df_nofilter,listelement,multipliconpairs,segment,mul=True)
+            df.to_csv(os.path.join(outdir, "Multiplicon_Ks.tsv"),header=True,index=True,sep='\t')
+        else:
+            df, seg_weight = add_seg(df_nofilter,listelement,multipliconpairs,segment)
+            df.to_csv(os.path.join(outdir, "Segment_Ks.tsv"),header=True,index=True,sep='\t')
     df_withindex,X = bc_group_anchor(df,regime=guide)
     #df = df.dropna(subset=['node_averaged_dS_outlierexcluded'])
     #df_rmdup = df.drop_duplicates(subset=['family', 'node'])
     #X = getX(df_rmdup,'node_averaged_dS_outlierexcluded')
     X_log = np.log(X).reshape(-1, 1)
     logging.info("KMedoids clustering with {} component".format(n))
     if n > 1: kmedoids = KMedoids(n_clusters=n,metric=metric,method=method,init=init,max_iter=em_iter,random_state=seed).fit(X_log)
```

### Comparing `wgd-2.0.32/wgd/postplot.py` & `wgd-2.0.33/wgd/postplot.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/ratecorrect.py` & `wgd-2.0.33/wgd/ratecorrect.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/syn.py` & `wgd-2.0.33/wgd/syn.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/tree.py` & `wgd-2.0.33/wgd/tree.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/utils.py` & `wgd-2.0.33/wgd/utils.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd/viz.py` & `wgd-2.0.33/wgd/viz.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.32/wgd.egg-info/PKG-INFO` & `wgd-2.0.33/wgd.egg-info/PKG-INFO`

 * *Files 1% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
 00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
-00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 320a  Version: 2.0.32.
+00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 330a  Version: 2.0.33.
 00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
 00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
 00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
 00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
 00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
 00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
 00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
@@ -1655,4333 +1655,4391 @@
 00006760: 7220 7468 6520 6261 7369 6320 4b73 2070  r the basic Ks p
 00006770: 6c6f 7474 696e 672c 2064 6566 6175 6c74  lotting, default
 00006780: 2073 6369 7079 0a2d 2d6e 5f63 6c75 7374   scipy.--n_clust
 00006790: 6572 732c 2074 6865 206e 756d 6265 7220  ers, the number 
 000067a0: 6f66 2063 6c75 7374 6572 7320 746f 2070  of clusters to p
 000067b0: 6c6f 7420 456c 626f 7720 6c6f 7373 2066  lot Elbow loss f
 000067c0: 756e 6374 696f 6e2c 2064 6566 6175 6c74  unction, default
-000067d0: 2035 0a2d 2d6b 6d65 646f 6964 732c 2066   5.--kmedoids, f
-000067e0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000067f0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00006800: 4b2d 4d65 646f 6964 7320 636c 7573 7465  K-Medoids cluste
-00006810: 7269 6e67 2061 6e61 6c79 7369 732c 2069  ring analysis, i
-00006820: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00006830: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00006840: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00006850: 7465 640a 2d67 642c 202d 2d67 7569 6465  ted.-gd, --guide
-00006860: 2c20 7468 6520 7265 6769 6d65 2072 6573  , the regime res
-00006870: 6964 696e 6720 616e 6368 6f72 732c 2064  iding anchors, d
-00006880: 6566 6175 6c74 2073 6567 6d65 6e74 0a2d  efault segment.-
-00006890: 7072 6374 2c20 2d2d 7072 6f6d 696e 656e  prct, --prominen
-000068a0: 6365 5f63 7574 6f66 662c 2074 6865 2070  ce_cutoff, the p
-000068b0: 726f 6d69 6e65 6e63 6520 6375 746f 6666  rominence cutoff
-000068c0: 206f 6620 6163 6365 7074 6162 6c65 2070   of acceptable p
-000068d0: 6561 6b73 2069 6e20 7065 616b 2066 696e  eaks in peak fin
-000068e0: 6469 6e67 2073 7465 7073 2c20 6465 6661  ding steps, defa
-000068f0: 756c 7420 302e 310a 2d72 682c 202d 2d72  ult 0.1.-rh, --r
-00006900: 656c 5f68 6569 6768 742c 2074 6865 2072  el_height, the r
-00006910: 656c 6174 6976 6520 6865 6967 6874 2061  elative height a
-00006920: 7420 7768 6963 6820 7468 6520 7065 616b  t which the peak
-00006930: 2077 6964 7468 2069 7320 6d65 6173 7572   width is measur
-00006940: 6564 2c20 6465 6661 756c 7420 302e 340a  ed, default 0.4.
-00006950: 2d6b 642c 202d 2d6b 7374 6f64 6174 652c  -kd, --kstodate,
-00006960: 2074 6865 2072 616e 6765 206f 6620 4b73   the range of Ks
-00006970: 2074 6f20 6265 2064 6174 6564 2069 6e20   to be dated in 
-00006980: 6865 7572 6973 7469 6320 7365 6172 6368  heuristic search
-00006990: 2c20 6465 6661 756c 7420 2830 2e35 2c20  , default (0.5, 
-000069a0: 312e 3529 0a2d 786c 2c20 2d2d 786c 696d  1.5).-xl, --xlim
-000069b0: 2c20 7468 6520 7820 6178 6973 206c 696d  , the x axis lim
-000069c0: 6974 206f 6620 474d 4d20 4b73 2064 6973  it of GMM Ks dis
-000069d0: 7472 6962 7574 696f 6e0a 2d79 6c2c 202d  tribution.-yl, -
-000069e0: 2d79 6c69 6d2c 2074 6865 2079 2061 7869  -ylim, the y axi
-000069f0: 7320 6c69 6d69 7420 6f66 2047 4d4d 204b  s limit of GMM K
-00006a00: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-00006a10: 2d6d 616e 7561 6c73 6574 2c20 666c 6167  -manualset, flag
-00006a20: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00006a30: 2074 6f20 6f75 7470 7574 2061 6e63 686f   to output ancho
-00006a40: 7220 7061 6972 7320 7769 7468 206d 616e  r pairs with man
-00006a50: 7561 6c6c 7920 7365 7420 4b73 2072 616e  ually set Ks ran
-00006a60: 6765 2c20 6966 2074 6865 2066 6c61 6720  ge, if the flag 
-00006a70: 7761 7320 7365 742c 206d 616e 7561 6c6c  was set, manuall
-00006a80: 7920 7365 7420 4b73 2072 616e 6765 2077  y set Ks range w
-00006a90: 6f75 6c64 2062 6520 6f75 7470 7574 7465  ould be outputte
-00006aa0: 640a 2d2d 6369 2c20 7468 6520 636f 6e66  d.--ci, the conf
-00006ab0: 6964 656e 6365 206c 6576 656c 206f 6620  idence level of 
-00006ac0: 6c6f 672d 6e6f 726d 616c 2064 6973 7472  log-normal distr
-00006ad0: 6962 7574 696f 6e20 746f 2064 6174 652c  ibution to date,
-00006ae0: 2064 6566 6175 6c74 2039 350a 2d2d 6864   default 95.--hd
-00006af0: 722c 2074 6865 2068 6967 6865 7374 2064  r, the highest d
-00006b00: 656e 7369 7479 2072 6567 696f 6e20 2848  ensity region (H
-00006b10: 4452 2920 6170 706c 6965 6420 696e 2074  DR) applied in t
-00006b20: 6865 2073 6567 6d65 6e74 2d67 7569 6465  he segment-guide
-00006b30: 6420 616e 6368 6f72 2070 6169 7220 4b73  d anchor pair Ks
-00006b40: 2064 6973 7472 6962 7574 696f 6e2c 2064   distribution, d
-00006b50: 6566 6175 6c74 2039 350a 2d2d 6865 7572  efault 95.--heur
-00006b60: 6973 7469 632c 2066 6c61 6720 6f70 7469  istic, flag opti
-00006b70: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00006b80: 6e69 7469 6174 6520 6865 7572 6973 7469  nitiate heuristi
-00006b90: 6320 6d65 7468 6f64 206f 6620 6465 6669  c method of defi
-00006ba0: 6e69 6e67 2043 4920 666f 7220 6461 7469  ning CI for dati
-00006bb0: 6e67 2c20 6966 2074 6865 2066 6c61 6720  ng, if the flag 
-00006bc0: 7761 7320 7365 742c 2074 6865 2068 6575  was set, the heu
-00006bd0: 7269 7374 6963 206d 6574 686f 6420 776f  ristic method wo
-00006be0: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
-00006bf0: 0a2d 6b63 2c20 2d2d 6b73 6375 746f 6666  .-kc, --kscutoff
-00006c00: 2c20 7468 6520 4b73 2073 6174 7572 6174  , the Ks saturat
-00006c10: 696f 6e20 6375 746f 6666 2069 6e20 6461  ion cutoff in da
-00006c20: 7469 6e67 2c20 6465 6661 756c 7420 350a  ting, default 5.
-00006c30: 2d2d 7368 6f77 6369 2c20 666c 6167 206f  --showci, flag o
-00006c40: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-00006c50: 6f20 7368 6f77 2043 4920 666f 7220 7468  o show CI for th
-00006c60: 6520 6f72 6967 696e 616c 2061 6e63 686f  e original ancho
-00006c70: 7220 4b73 2047 4d4d 2061 6e61 6c79 7369  r Ks GMM analysi
-00006c80: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00006c90: 6173 2073 6574 2c20 6974 2077 6f75 6c64  as set, it would
-00006ca0: 2062 6520 7368 6f77 6564 0a2d 2d6b 6565   be showed.--kee
-00006cb0: 7074 6d70 6669 672c 2066 6c61 6720 6f70  ptmpfig, flag op
-00006cc0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00006cd0: 206b 6565 7020 7465 6d70 6f72 6172 7920   keep temporary 
-00006ce0: 6669 6775 7265 7320 696e 2070 6561 6b20  figures in peak 
-00006cf0: 6669 6e64 696e 6720 7072 6f63 6573 732c  finding process,
-00006d00: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00006d10: 2073 6574 2c20 7468 6f73 6520 6669 6775   set, those figu
-00006d20: 7265 7320 776f 756c 6420 6265 206b 6570  res would be kep
-00006d30: 740a 6060 600a 0a54 6865 2070 726f 6772  t.```..The progr
-00006d40: 616d 2060 7767 6420 7379 6e60 2063 616e  am `wgd syn` can
-00006d50: 2072 6561 6c69 7a65 2074 6865 2069 6e74   realize the int
-00006d60: 7261 2d20 616e 6420 696e 7465 722d 7370  ra- and inter-sp
-00006d70: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
-00006d80: 6e66 6572 656e 6365 2e0a 6060 600a 7767  nference..```.wg
-00006d90: 6420 7379 6e20 6661 6d69 6c69 6573 2067  d syn families g
-00006da0: 6666 7320 286f 7074 696f 6e29 0a2d 2d2d  ffs (option).---
-00006db0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006dc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006dd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006de0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006df0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d6b  -------------.-k
-00006e00: 732c 202d 2d6b 735f 6469 7374 7269 6275  s, --ks_distribu
-00006e10: 7469 6f6e 2c20 6b73 2064 6973 7472 6962  tion, ks distrib
-00006e20: 7574 696f 6e20 6461 7461 6669 6c65 2c20  ution datafile, 
-00006e30: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
-00006e40: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
-00006e50: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
-00006e60: 2064 6566 6175 6c74 2077 6764 5f73 796e   default wgd_syn
-00006e70: 0a2d 662c 202d 2d66 6561 7475 7265 2c20  .-f, --feature, 
-00006e80: 7468 6520 6665 6174 7572 6520 666f 7220  the feature for 
-00006e90: 7061 7273 696e 6720 6765 6e65 2049 4473  parsing gene IDs
-00006ea0: 2066 726f 6d20 4746 4620 6669 6c65 732c   from GFF files,
-00006eb0: 2064 6566 6175 6c74 2067 656e 650a 2d61   default gene.-a
-00006ec0: 2c20 2d2d 6174 7472 6962 7574 652c 2074  , --attribute, t
-00006ed0: 6865 2061 7474 7269 6275 7465 2066 6f72  he attribute for
-00006ee0: 2070 6172 7369 6e67 2074 6865 2067 656e   parsing the gen
-00006ef0: 6520 4944 7320 6672 6f6d 2074 6865 2047  e IDs from the G
-00006f00: 4646 2066 696c 6573 2c20 6465 6661 756c  FF files, defaul
-00006f10: 7420 4944 0a2d 6174 672c 202d 2d61 6464  t ID.-atg, --add
-00006f20: 6974 696f 6e61 6c67 6666 696e 666f 2c20  itionalgffinfo, 
-00006f30: 7468 6520 6665 6174 7572 6520 616e 6420  the feature and 
-00006f40: 6174 7472 6962 7574 6520 696e 666f 726d  attribute inform
-00006f50: 6174 696f 6e20 6f66 2061 6464 6974 696f  ation of additio
-00006f60: 6e61 6c20 6766 6633 2066 696c 6573 2069  nal gff3 files i
-00006f70: 6620 6469 6666 6572 656e 7420 696e 2074  f different in t
-00006f80: 6865 2066 6f72 6d61 7420 6f66 2028 6665  he format of (fe
-00006f90: 6174 7572 653b 6174 7472 6962 7574 6529  ature;attribute)
-00006fa0: 272c 2064 6566 6175 6c74 204e 6f6e 650a  ', default None.
-00006fb0: 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c 2074  -ml, --minlen, t
-00006fc0: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
-00006fd0: 6820 6f66 2061 2073 6361 6666 6f6c 6420  h of a scaffold 
-00006fe0: 746f 2062 6520 696e 636c 7564 6564 2069  to be included i
-00006ff0: 6e20 646f 7470 6c6f 742c 2064 6566 6175  n dotplot, defau
-00007000: 6c74 202d 312c 2069 6620 2d31 2077 6173  lt -1, if -1 was
-00007010: 2073 6574 2c20 7468 6520 3130 2520 6f66   set, the 10% of
-00007020: 2074 6865 206c 6f6e 6765 7374 2073 6361   the longest sca
-00007030: 6666 6f6c 6420 776f 756c 6420 6265 2073  ffold would be s
-00007040: 6574 0a2d 6d73 2c20 2d2d 6d61 7873 697a  et.-ms, --maxsiz
-00007050: 652c 2074 6865 206d 6178 696d 756d 2066  e, the maximum f
-00007060: 616d 696c 7920 7369 7a65 2074 6f20 6265  amily size to be
-00007070: 2069 6e63 6c75 6465 642c 2064 6566 6175   included, defau
-00007080: 6c74 2032 3030 0a2d 722c 202d 2d6b 735f  lt 200.-r, --ks_
-00007090: 7261 6e67 652c 2074 6865 204b 7320 7261  range, the Ks ra
-000070a0: 6e67 6520 696e 2063 6f6c 6f72 6564 2064  nge in colored d
-000070b0: 6f74 706c 6f74 2c20 6465 6661 756c 7420  otplot, default 
-000070c0: 2830 2c20 3529 0a2d 2d69 6164 686f 7265  (0, 5).--iadhore
-000070d0: 5f6f 7074 696f 6e73 2c20 7468 6520 7061  _options, the pa
-000070e0: 7261 6d65 7465 7220 7365 7474 696e 6720  rameter setting 
-000070f0: 696e 2069 6164 686f 7265 2c20 6465 6661  in iadhore, defa
-00007100: 756c 7420 6173 2061 2073 7472 696e 6720  ult as a string 
-00007110: 6f66 206c 656e 6774 6820 7a65 726f 0a2d  of length zero.-
-00007120: 6d67 2c20 2d2d 6d69 6e73 6567 6c65 6e2c  mg, --minseglen,
-00007130: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
-00007140: 6774 6820 6f66 2073 6567 6d65 6e74 7320  gth of segments 
-00007150: 746f 2069 6e63 6c75 6465 2069 6e20 7261  to include in ra
-00007160: 7469 6f20 6966 203c 3d20 312c 2064 6566  tio if <= 1, def
-00007170: 6175 6c74 2031 3030 3030 0a2d 6b72 2c20  ault 10000.-kr, 
-00007180: 2d2d 6b65 6570 7265 6475 6e2c 2066 6c61  --keepredun, fla
-00007190: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000071a0: 7220 746f 206b 6565 7020 7265 6475 6e64  r to keep redund
-000071b0: 616e 7420 6d75 6c74 6970 6c69 636f 6e73  ant multiplicons
-000071c0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-000071d0: 7320 7365 742c 2074 6865 2072 6564 756e  s set, the redun
-000071e0: 6461 6e74 206d 756c 7469 706c 6963 6f6e  dant multiplicon
-000071f0: 7320 776f 756c 6420 6265 206b 6570 740a  s would be kept.
-00007200: 2d6d 676e 2c20 2d2d 6d69 6e67 656e 656e  -mgn, --mingenen
-00007210: 756d 2c20 7468 6520 6d69 6e69 6d75 6d20  um, the minimum 
-00007220: 6e75 6d62 6572 206f 6620 6765 6e65 7320  number of genes 
-00007230: 666f 7220 6120 7365 676d 656e 7420 746f  for a segment to
-00007240: 2062 6520 636f 6e73 6964 6572 6564 2c20   be considered, 
-00007250: 6465 6661 756c 7420 3330 0a2d 6473 2c20  default 30.-ds, 
-00007260: 2d2d 646f 7473 697a 652c 2074 6865 2064  --dotsize, the d
-00007270: 6f74 2073 697a 6520 696e 2064 6f74 2070  ot size in dot p
-00007280: 6c6f 742c 2064 6566 6175 6c74 2030 2e33  lot, default 0.3
-00007290: 0a2d 6161 2c20 2d2d 6170 616c 7068 612c  .-aa, --apalpha,
-000072a0: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-000072b0: 616e 6368 6f72 2064 6f74 732c 2064 6566  anchor dots, def
-000072c0: 6175 6c74 2031 0a2d 6861 2c20 2d2d 686f  ault 1.-ha, --ho
-000072d0: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-000072e0: 7479 206f 6620 686f 6d6f 6c6f 6720 646f  ty of homolog do
-000072f0: 7473 2c20 6465 6661 756c 7420 300a 2d73  ts, default 0.-s
-00007300: 7274 2c20 2d2d 7368 6f77 7265 616c 7469  rt, --showrealti
-00007310: 636b 2c20 666c 6167 206f 7074 696f 6e2c  ck, flag option,
-00007320: 2077 6865 7468 6572 2074 6f20 7368 6f77   whether to show
-00007330: 2074 6865 2072 6561 6c20 7469 636b 2069   the real tick i
-00007340: 6e20 6765 6e65 7320 6f72 2062 6173 6573  n genes or bases
-00007350: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007360: 7320 7365 742c 2074 6865 2072 6561 6c20  s set, the real 
-00007370: 7469 636b 2077 6f75 6c64 2062 6520 7368  tick would be sh
-00007380: 6f77 6564 0a2d 746c 732c 202d 2d74 6963  owed.-tls, --tic
-00007390: 6b6c 6162 656c 7369 7a65 2c20 7468 6520  klabelsize, the 
-000073a0: 6c61 6265 6c20 7369 7a65 206f 6620 7469  label size of ti
-000073b0: 636b 2c20 6465 6661 756c 7420 350a 2d67  ck, default 5.-g
-000073c0: 722c 202d 2d67 6973 7472 622c 2066 6c61  r, --gistrb, fla
-000073d0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000073e0: 7220 746f 2075 7365 2067 6973 745f 7261  r to use gist_ra
-000073f0: 696e 626f 7720 6173 2063 6f6c 6f72 206d  inbow as color m
-00007400: 6170 206f 6620 646f 7470 6c6f 740a 2d6e  ap of dotplot.-n
-00007410: 2c20 2d2d 6e74 6872 6561 6473 2c20 7468  , --nthreads, th
-00007420: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
-00007430: 6164 7320 746f 2075 7365 2069 6e20 7379  ads to use in sy
-00007440: 6e74 656e 7920 696e 6665 7265 6e63 652c  nteny inference,
-00007450: 2064 6566 6175 6c74 2034 0a60 6060 0a0a   default 4.```..
-00007460: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
-00007470: 2076 697a 6020 6361 6e20 7265 616c 697a   viz` can realiz
-00007480: 6520 7468 6520 7669 7375 616c 697a 6174  e the visualizat
-00007490: 696f 6e20 6f66 202a 4b2a 3c73 7562 3e53  ion of *K*<sub>S
-000074a0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-000074b0: 6962 7574 696f 6e20 616e 6420 7379 6e74  ibution and synt
-000074c0: 656e 792e 0a60 6060 0a77 6764 2076 697a  eny..```.wgd viz
-000074d0: 2028 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d   (option).------
-000074e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000074f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007500: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007510: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007520: 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 642c 202d  ----------.-d, -
-00007530: 2d64 6174 6166 696c 652c 2074 6865 204b  -datafile, the K
-00007540: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
-00007550: 756c 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f  ult None.-o, --o
-00007560: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
-00007570: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
-00007580: 6175 6c74 2077 6764 5f76 697a 0a2d 7372  ault wgd_viz.-sr
-00007590: 2c20 2d2d 7370 6169 722c 2074 6865 2073  , --spair, the s
-000075a0: 7065 6369 6573 2070 6169 7220 746f 2062  pecies pair to b
-000075b0: 6520 706c 6f74 7465 642c 2064 6566 6175  e plotted, defau
-000075c0: 6c74 204e 6f6e 652c 2074 6869 7320 6f70  lt None, this op
-000075d0: 7469 6f6e 2063 616e 2062 6520 7072 6f76  tion can be prov
-000075e0: 6964 6564 206d 756c 7469 706c 6520 7469  ided multiple ti
-000075f0: 6d65 730a 2d66 612c 202d 2d66 6f63 7573  mes.-fa, --focus
-00007600: 3261 6c6c 2c20 7365 7420 666f 6361 6c20  2all, set focal 
-00007610: 7370 6563 6965 7320 616e 6420 6c65 7420  species and let 
-00007620: 7370 6563 6965 7320 7061 6972 2074 6f20  species pair to 
-00007630: 6265 2062 6574 7765 656e 2066 6f63 616c  be between focal
-00007640: 2061 6e64 2061 6c6c 2074 6865 2072 656d   and all the rem
-00007650: 6169 6e69 6e67 2073 7065 6369 6573 2c20  aining species, 
-00007660: 6465 6661 756c 7420 4e6f 6e65 0a2d 6773  default None.-gs
-00007670: 2c20 2d2d 6773 6d61 702c 2074 6865 2067  , --gsmap, the g
-00007680: 656e 6520 6e61 6d65 2d73 7065 6369 6573  ene name-species
-00007690: 206e 616d 6520 6d61 702c 2064 6566 6175   name map, defau
-000076a0: 6c74 204e 6f6e 650a 2d73 702c 202d 2d73  lt None.-sp, --s
-000076b0: 7065 6369 6573 7472 6565 2c20 7468 6520  peciestree, the 
-000076c0: 7370 6563 6965 7320 7472 6565 2074 6f20  species tree to 
-000076d0: 7065 7266 6f72 6d20 7261 7465 2063 6f72  perform rate cor
-000076e0: 7265 6374 696f 6e2c 2064 6566 6175 6c74  rection, default
-000076f0: 204e 6f6e 652c 2069 6620 4e6f 6e65 2077   None, if None w
-00007700: 6173 2067 6976 656e 2c20 7468 6520 7261  as given, the ra
-00007710: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
-00007720: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-00007730: 6361 6c6c 6564 206f 6666 0a2d 706b 2c20  called off.-pk, 
-00007740: 2d2d 706c 6f74 6b64 652c 2066 6c61 6720  --plotkde, flag 
-00007750: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00007760: 746f 2070 6c6f 7420 6b64 6520 6375 7276  to plot kde curv
-00007770: 6520 7570 6f6e 2068 6973 746f 6772 616d  e upon histogram
-00007780: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007790: 7320 7365 742c 206b 6465 2063 7572 7665  s set, kde curve
-000077a0: 2077 6f75 6c64 2062 6520 6164 6465 640a   would be added.
-000077b0: 2d72 772c 202d 2d72 6577 6569 6768 742c  -rw, --reweight,
-000077c0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-000077d0: 6574 6865 7220 746f 2072 6563 616c 6375  ether to recalcu
-000077e0: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
-000077f0: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
-00007800: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007810: 7320 7365 742c 2074 6865 2077 6569 6768  s set, the weigh
-00007820: 7420 776f 756c 6420 6265 2072 6563 616c  t would be recal
-00007830: 6375 6c61 7465 640a 2d6f 722c 202d 2d6f  culated.-or, --o
-00007840: 6e6c 7972 6f6f 746f 7574 2c20 666c 6167  nlyrootout, flag
-00007850: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00007860: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
-00007870: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00007880: 2075 7369 6e67 2074 6865 206f 7574 6772   using the outgr
-00007890: 6f75 7020 6174 2072 6f6f 7420 6173 206f  oup at root as o
-000078a0: 7574 6772 6f75 702c 2069 6620 7468 6520  utgroup, if the 
-000078b0: 666c 6167 2077 6173 2073 6574 2c20 6f6e  flag was set, on
-000078c0: 6c79 2074 6865 206f 7574 6772 6f75 7020  ly the outgroup 
-000078d0: 6174 2072 6f6f 7420 776f 756c 6420 6265  at root would be
-000078e0: 2075 7365 6420 6173 206f 7574 6772 6f75   used as outgrou
-000078f0: 700a 2d69 7465 722c 202d 2d65 6d5f 6974  p.-iter, --em_it
-00007900: 6572 6174 696f 6e73 2c20 7468 6520 6d61  erations, the ma
-00007910: 7869 6d75 6d20 454d 2069 7465 7261 7469  ximum EM iterati
-00007920: 6f6e 732c 2064 6566 6175 6c74 2032 3030  ons, default 200
-00007930: 0a2d 696e 6974 2c20 2d2d 656d 5f69 6e69  .-init, --em_ini
-00007940: 7469 616c 697a 6174 696f 6e73 2c20 7468  tializations, th
-00007950: 6520 6d61 7869 6d75 6d20 454d 2069 6e69  e maximum EM ini
-00007960: 7469 616c 697a 6174 696f 6e73 2c20 6465  tializations, de
-00007970: 6661 756c 7420 3230 300a 2d70 7263 742c  fault 200.-prct,
-00007980: 202d 2d70 726f 6d69 6e65 6e63 655f 6375   --prominence_cu
-00007990: 746f 6666 2c20 7468 6520 7072 6f6d 696e  toff, the promin
-000079a0: 656e 6365 2063 7574 6f66 6620 6f66 2061  ence cutoff of a
-000079b0: 6363 6570 7461 626c 6520 7065 616b 732c  cceptable peaks,
-000079c0: 2064 6566 6175 6c74 2030 2e31 0a2d 7268   default 0.1.-rh
-000079d0: 2c20 2d2d 7265 6c5f 6865 6967 6874 2c20  , --rel_height, 
-000079e0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-000079f0: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00007a00: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00007a10: 6561 7375 7265 642c 2064 6566 6175 6c74  easured, default
-00007a20: 2030 2e34 0a2d 736d 2c20 2d2d 7365 676d   0.4.-sm, --segm
-00007a30: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
-00007a40: 7473 2064 6174 6166 696c 652c 2064 6566  ts datafile, def
-00007a50: 6175 6c74 204e 6f6e 650a 2d6d 6c2c 202d  ault None.-ml, -
-00007a60: 2d6d 696e 6c65 6e2c 2074 6865 206d 696e  -minlen, the min
-00007a70: 696d 756d 206c 656e 6774 6820 6f66 2061  imum length of a
-00007a80: 2073 6361 6666 6f6c 6420 746f 2062 6520   scaffold to be 
-00007a90: 696e 636c 7564 6564 2069 6e20 646f 7470  included in dotp
-00007aa0: 6c6f 742c 2064 6566 6175 6c74 202d 312c  lot, default -1,
-00007ab0: 2069 6620 2d31 2077 6173 2073 6574 2c20   if -1 was set, 
-00007ac0: 7468 6520 3130 2520 6f66 2074 6865 206c  the 10% of the l
-00007ad0: 6f6e 6765 7374 2073 6361 6666 6f6c 6473  ongest scaffolds
-00007ae0: 2077 696c 6c20 6265 2073 6574 0a2d 6d73   will be set.-ms
-00007af0: 2c20 2d2d 6d61 7873 697a 652c 2074 6865  , --maxsize, the
-00007b00: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-00007b10: 7369 7a65 2074 6f20 6265 2069 6e63 6c75  size to be inclu
-00007b20: 6465 642c 2064 6566 6175 6c74 2032 3030  ded, default 200
-00007b30: 0a2d 6170 2c20 2d2d 616e 6368 6f72 706f  .-ap, --anchorpo
-00007b40: 696e 7473 2c20 7468 6520 616e 6368 6f72  ints, the anchor
-00007b50: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
-00007b60: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00007b70: 6d74 2c20 2d2d 6d75 6c74 6970 6c69 636f  mt, --multiplico
-00007b80: 6e2c 2074 6865 206d 756c 7469 706c 6963  n, the multiplic
-00007b90: 6f6e 7320 6461 7461 6669 6c65 2c20 6465  ons datafile, de
-00007ba0: 6661 756c 7420 4e6f 6e65 0a2d 6774 2c20  fault None.-gt, 
-00007bb0: 2d2d 6765 6e65 7461 626c 652c 2074 6865  --genetable, the
-00007bc0: 2067 656e 6520 7461 626c 6520 6461 7461   gene table data
-00007bd0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007be0: 6e65 0a2d 6d67 2c20 2d2d 6d69 6e73 6567  ne.-mg, --minseg
-00007bf0: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
-00007c00: 206c 656e 6774 6820 6f66 2073 6567 6d65   length of segme
-00007c10: 6e74 7320 746f 2069 6e63 6c75 6465 2c20  nts to include, 
-00007c20: 696e 2072 6174 696f 2069 6620 3c3d 2031  in ratio if <= 1
-00007c30: 2c20 6465 6661 756c 7420 3130 3030 300a  , default 10000.
-00007c40: 2d6d 676e 2c20 2d2d 6d69 6e67 656e 656e  -mgn, --mingenen
-00007c50: 756d 2c20 7468 6520 6d69 6e69 6d75 6d20  um, the minimum 
-00007c60: 6e75 6d62 6572 206f 6620 6765 6e65 7320  number of genes 
-00007c70: 666f 7220 6120 7365 676d 656e 7420 746f  for a segment to
-00007c80: 2062 6520 636f 6e73 6964 6572 6564 2c20   be considered, 
-00007c90: 6465 6661 756c 7420 3330 0a2d 6b72 2c20  default 30.-kr, 
-00007ca0: 2d2d 6b65 6570 7265 6475 6e2c 2066 6c61  --keepredun, fla
-00007cb0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00007cc0: 7220 746f 206b 6565 7020 7265 6475 6e64  r to keep redund
-00007cd0: 616e 7420 6d75 6c74 6970 6c69 636f 6e73  ant multiplicons
-00007ce0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00007cf0: 7320 7365 742c 2074 6865 2072 6564 756e  s set, the redun
-00007d00: 6461 6e74 206d 756c 7469 706c 6963 6f6e  dant multiplicon
-00007d10: 7320 776f 756c 6420 6265 206b 6570 740a  s would be kept.
-00007d20: 2d65 706b 2c20 2d2d 6578 7472 6170 6172  -epk, --extrapar
-00007d30: 616e 6f6d 656b 732c 2065 7874 7261 2070  anomeks, extra p
-00007d40: 6172 616e 6f6d 6520 4b73 2064 6174 6120  aranome Ks data 
-00007d50: 746f 2070 6c6f 7420 696e 2074 6865 206d  to plot in the m
-00007d60: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
-00007d70: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
-00007d80: 6e65 0a2d 7061 672c 202d 2d70 6c6f 7461  ne.-pag, --plota
-00007d90: 7067 6d6d 2c20 666c 6167 206f 7074 696f  pgmm, flag optio
-00007da0: 6e2c 2077 6865 7468 6572 2074 6f20 636f  n, whether to co
-00007db0: 6e64 7563 7420 616e 6420 706c 6f74 206d  nduct and plot m
-00007dc0: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00007dd0: 6f66 2061 6e63 686f 7220 4b73 2069 6e20  of anchor Ks in 
-00007de0: 7468 6520 6d69 7865 6420 4b73 2064 6973  the mixed Ks dis
-00007df0: 7472 6962 7574 696f 6e2c 2069 6620 7468  tribution, if th
-00007e00: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00007e10: 7468 6520 6d69 7874 7572 6520 6d6f 6465  the mixture mode
-00007e20: 6c69 6e67 206f 6620 616e 6368 6f72 204b  ling of anchor K
-00007e30: 7320 776f 756c 6420 6265 2063 6f6e 6475  s would be condu
-00007e40: 6374 6564 2061 6e64 2070 6c6f 7474 6564  cted and plotted
-00007e50: 0a2d 7065 6d2c 202d 2d70 6c6f 7465 6c6d  .-pem, --plotelm
-00007e60: 6d2c 2066 6c61 6720 6f70 7469 6f6e 2c20  m, flag option, 
-00007e70: 7768 6574 6865 7220 746f 2063 6f6e 6475  whether to condu
-00007e80: 6374 2061 6e64 2070 6c6f 7420 656c 6d6d  ct and plot elmm
-00007e90: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00007ea0: 6720 6f66 2070 6172 616e 6f6d 6520 4b73  g of paranome Ks
-00007eb0: 2069 6e20 7468 6520 6d69 7865 6420 4b73   in the mixed Ks
-00007ec0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
-00007ed0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00007ee0: 6574 2c20 7468 6520 656c 6d6d 206d 6978  et, the elmm mix
-00007ef0: 7475 7265 206d 6f64 656c 696e 6720 6f66  ture modeling of
-00007f00: 2070 6172 616e 6f6d 6520 4b73 2077 6f75   paranome Ks wou
-00007f10: 6c64 2062 6520 636f 6e64 7563 7465 6420  ld be conducted 
-00007f20: 616e 6420 706c 6f74 7465 640a 2d63 2c20  and plotted.-c, 
-00007f30: 2d2d 636f 6d70 6f6e 656e 7473 2c20 7468  --components, th
-00007f40: 6520 7261 6e67 6520 6f66 2074 6865 206e  e range of the n
-00007f50: 756d 6265 7220 6f66 2063 6f6d 706f 6e65  umber of compone
-00007f60: 6e74 7320 746f 2066 6974 2069 6e20 616e  nts to fit in an
-00007f70: 6368 6f72 204b 7320 6d69 7874 7572 6520  chor Ks mixture 
-00007f80: 6d6f 6465 6c69 6e67 2c20 6465 6661 756c  modeling, defaul
-00007f90: 7420 2831 2c34 290a 2d70 7379 2c20 2d2d  t (1,4).-psy, --
-00007fa0: 706c 6f74 7379 6e2c 2066 6c61 6720 6f70  plotsyn, flag op
-00007fb0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00007fc0: 2069 6e69 7469 6174 6520 7468 6520 7379   initiate the sy
-00007fd0: 6e74 656e 7920 706c 6f74 2c20 6f6e 6c79  nteny plot, only
-00007fe0: 2077 6865 6e20 7468 6520 666c 6167 2077   when the flag w
-00007ff0: 6173 2073 6574 2c20 7468 6520 7379 6e74  as set, the synt
-00008000: 656e 7920 706c 6f74 2077 6f75 6c64 2062  eny plot would b
-00008010: 6520 7072 6f64 7563 6564 0a2d 6473 2c20  e produced.-ds, 
-00008020: 2d2d 646f 7473 697a 652c 2074 6865 2064  --dotsize, the d
-00008030: 6f74 2073 697a 6520 696e 2064 6f74 2070  ot size in dot p
-00008040: 6c6f 742c 2064 6566 6175 6c74 2030 2e33  lot, default 0.3
-00008050: 0a2d 6161 2c20 2d2d 6170 616c 7068 612c  .-aa, --apalpha,
-00008060: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-00008070: 616e 6368 6f72 2064 6f74 732c 2064 6566  anchor dots, def
-00008080: 6175 6c74 2031 0a2d 6861 2c20 2d2d 686f  ault 1.-ha, --ho
-00008090: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-000080a0: 7479 206f 6620 686f 6d6f 6c6f 6720 646f  ty of homolog do
-000080b0: 7473 2c20 6465 6661 756c 7420 300a 2d73  ts, default 0.-s
-000080c0: 7274 2c20 2d2d 7368 6f77 7265 616c 7469  rt, --showrealti
-000080d0: 636b 2c20 666c 6167 206f 7074 696f 6e2c  ck, flag option,
-000080e0: 2077 6865 7468 6572 2074 6f20 7368 6f77   whether to show
-000080f0: 2074 6865 2072 6561 6c20 7469 636b 2069   the real tick i
-00008100: 6e20 6765 6e65 7320 6f72 2062 6173 6573  n genes or bases
-00008110: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00008120: 7320 7365 742c 2074 6865 2072 6561 6c20  s set, the real 
-00008130: 7469 636b 2077 6f75 6c64 2062 6520 7368  tick would be sh
-00008140: 6f77 6564 0a2d 746c 732c 202d 2d74 6963  owed.-tls, --tic
-00008150: 6b6c 6162 656c 7369 7a65 2c20 7468 6520  klabelsize, the 
-00008160: 6c61 6265 6c20 7369 7a65 206f 6620 7469  label size of ti
-00008170: 636b 2c20 6465 6661 756c 7420 350a 2d78  ck, default 5.-x
-00008180: 6c2c 202d 2d78 6c69 6d2c 2074 6865 2078  l, --xlim, the x
-00008190: 2061 7869 7320 6c69 6d69 7420 6f66 204b   axis limit of K
-000081a0: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-000081b0: 796c 2c20 2d2d 796c 696d 2c20 7468 6520  yl, --ylim, the 
-000081c0: 7920 6178 6973 206c 696d 6974 206f 6620  y axis limit of 
-000081d0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
-000081e0: 2d61 646f 2c20 2d2d 6164 6a75 7374 6f72  -ado, --adjustor
-000081f0: 7468 6f2c 2066 6c61 6720 6f70 7469 6f6e  tho, flag option
-00008200: 2c20 7768 6574 6865 7220 746f 2061 646a  , whether to adj
-00008210: 7573 7420 7468 6520 6869 7374 6f67 7261  ust the histogra
-00008220: 6d20 6865 6967 6874 206f 6620 6f72 7468  m height of orth
-00008230: 6f6c 6f67 6f75 7320 4b73 2061 7320 746f  ologous Ks as to
-00008240: 206d 6174 6368 2074 6865 2068 6569 6768   match the heigh
-00008250: 7420 6f66 2070 6172 616c 6f67 6f75 7320  t of paralogous 
-00008260: 4b73 2c20 6966 2074 6865 2066 6c61 6720  Ks, if the flag 
-00008270: 7761 7320 7365 742c 2074 6865 2061 646a  was set, the adj
-00008280: 7573 746d 656e 7420 776f 756c 6420 6265  ustment would be
-00008290: 2063 6f6e 6475 6374 6564 0a2d 6164 662c   conducted.-adf,
-000082a0: 202d 2d61 646a 7573 7466 6163 746f 722c   --adjustfactor,
-000082b0: 2074 6865 2061 646a 7573 746d 656e 7420   the adjustment 
-000082c0: 6661 6374 6f72 206f 6620 6f72 7468 6f6c  factor of orthol
-000082d0: 6f67 6f75 7320 4b73 2c20 6465 6661 756c  ogous Ks, defaul
-000082e0: 7420 302e 350a 2d6f 612c 202d 2d6f 6b61  t 0.5.-oa, --oka
-000082f0: 6c70 6861 2c20 7468 6520 6f70 6163 6974  lpha, the opacit
-00008300: 7920 6f66 206f 7274 686f 6c6f 676f 7573  y of orthologous
-00008310: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00008320: 2069 6e20 6d69 7865 6420 706c 6f74 2c20   in mixed plot, 
-00008330: 6465 6661 756c 7420 302e 350a 2d63 732c  default 0.5.-cs,
-00008340: 202d 2d63 6c61 7373 6963 2c20 666c 6167   --classic, flag
-00008350: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00008360: 2074 6f20 6472 6177 206d 6978 6564 204b   to draw mixed K
-00008370: 7320 706c 6f74 2069 6e20 6120 636c 6173  s plot in a clas
-00008380: 7369 6320 6d61 6e6e 6572 2077 6865 7265  sic manner where
-00008390: 2074 6865 2066 756c 6c20 6f72 7468 6f6c   the full orthol
-000083a0: 6f67 6f75 7320 4b73 2064 6973 7472 6962  ogous Ks distrib
-000083b0: 7574 696f 6e20 6973 2064 7261 7765 642c  ution is drawed,
-000083c0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-000083d0: 2073 6574 2c20 7468 6520 636c 6173 7369   set, the classi
-000083e0: 6320 6d69 7865 6420 4b73 2070 6c6f 7420  c mixed Ks plot 
-000083f0: 776f 756c 6420 6265 2064 7261 776e 0a2d  would be drawn.-
-00008400: 7461 2c20 2d2d 746f 7061 7272 6f77 2c20  ta, --toparrow, 
-00008410: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00008420: 7468 6572 2074 6f20 6164 6a75 7374 2074  ther to adjust t
-00008430: 6865 2061 7272 6f77 2074 6f20 6265 2061  he arrow to be a
-00008440: 7420 7468 6520 746f 7020 6f66 2074 6865  t the top of the
-00008450: 2070 6c6f 742c 2069 6e73 7465 6164 206f   plot, instead o
-00008460: 6620 6265 696e 6720 636f 6f72 6469 6e61  f being coordina
-00008470: 7465 6420 6173 2074 6865 204b 4445 206f  ted as the KDE o
-00008480: 6620 7468 6520 6f72 7468 6f6c 6f67 6f75  f the orthologou
-00008490: 7320 4b73 2064 6973 7472 6962 7574 696f  s Ks distributio
-000084a0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-000084b0: 6173 2073 6574 2c20 7468 6520 6172 726f  as set, the arro
-000084c0: 7720 776f 756c 6420 6265 2073 6574 2061  w would be set a
-000084d0: 7420 7468 6520 746f 700a 2d6e 612c 202d  t the top.-na, -
-000084e0: 2d6e 6f64 6561 7665 7261 6765 642c 2066  -nodeaveraged, f
-000084f0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00008500: 6865 7220 746f 2075 7365 206e 6f64 652d  her to use node-
-00008510: 6176 6572 6167 6564 206d 6574 686f 6420  averaged method 
-00008520: 666f 7220 6465 2d72 6564 756e 6461 6e63  for de-redundanc
-00008530: 792c 2069 6620 7468 6520 666c 6167 2077  y, if the flag w
-00008540: 6173 2073 6574 2c20 7468 6520 6e6f 6465  as set, the node
-00008550: 2d61 7665 7261 6765 6420 6d65 7468 6f64  -averaged method
-00008560: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00008570: 7465 640a 2d62 732c 202d 2d62 6f6f 7473  ted.-bs, --boots
-00008580: 7472 6170 2c20 7468 6520 6e75 6d62 6572  trap, the number
-00008590: 206f 6620 626f 6f74 7374 7261 7020 7265   of bootstrap re
-000085a0: 706c 6963 6174 6573 206f 6620 6f72 7468  plicates of orth
-000085b0: 6f6c 6f67 204b 7320 6469 7374 7269 6275  olog Ks distribu
-000085c0: 7469 6f6e 2069 6e20 6d69 7865 6420 706c  tion in mixed pl
-000085d0: 6f74 0a2d 6772 2c20 2d2d 6769 7374 7262  ot.-gr, --gistrb
-000085e0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-000085f0: 6865 7468 6572 2074 6f20 7573 6520 6769  hether to use gi
-00008600: 7374 5f72 6169 6e62 6f77 2061 7320 636f  st_rainbow as co
-00008610: 6c6f 7220 6d61 7020 6f66 2064 6f74 706c  lor map of dotpl
-00008620: 6f74 0a2d 6e2c 202d 2d6e 7468 7265 6164  ot.-n, --nthread
-00008630: 732c 2074 6865 206e 756d 6265 7220 6f66  s, the number of
-00008640: 2074 6872 6561 6473 2074 6f20 7573 6520   threads to use 
-00008650: 696e 2062 6f6f 7473 7472 6170 2073 616d  in bootstrap sam
-00008660: 706c 696e 672c 2064 6566 6175 6c74 2031  pling, default 1
-00008670: 0a60 6060 0a0a 2323 2055 7361 6765 0a0a  .```..## Usage..
-00008680: 4865 7265 2077 6520 7072 6f76 6964 6564  Here we provided
-00008690: 2074 6865 2062 6173 6963 2075 7361 6765   the basic usage
-000086a0: 2066 6f72 2065 6163 6820 7072 6f67 7261   for each progra
-000086b0: 6d20 616e 6420 7468 6520 7265 6c65 7661  m and the releva
-000086c0: 6e74 2070 6172 616d 6574 6572 7320 616e  nt parameters an
-000086d0: 6420 7375 6767 6573 7469 6f6e 7320 6f6e  d suggestions on
-000086e0: 2070 6172 616d 6574 6572 697a 6174 696f   parameterizatio
-000086f0: 6e2e 2041 2072 656d 696e 6465 7220 7468  n. A reminder th
-00008700: 6174 2074 6865 2067 6976 656e 2063 6f72  at the given cor
-00008710: 6573 2061 6e64 2074 6872 6561 6473 2063  es and threads c
-00008720: 616e 2073 6967 6e69 6669 6361 6e74 6c79  an significantly
-00008730: 2069 6d70 6163 7420 7468 6520 7275 6e20   impact the run 
-00008740: 7469 6d65 2061 6e64 2074 6875 7320 7765  time and thus we
-00008750: 2061 6464 6564 2073 6f6d 6520 7265 706f   added some repo
-00008760: 7274 2069 6e66 6f72 6d61 7469 6f6e 2070  rt information p
-00008770: 6572 7461 696e 696e 6720 746f 2074 6865  ertaining to the
-00008780: 2073 7973 7465 6d20 6f66 2075 7365 7273   system of users
-00008790: 2074 6f20 6661 6369 6c69 7461 7465 2074   to facilitate t
-000087a0: 6865 2065 6666 6963 6965 6e74 2073 6574  he efficient set
-000087b0: 7469 6e67 206f 6620 7468 7265 6164 7320  ting of threads 
-000087c0: 616e 6420 6d65 6d6f 7279 2e20 5468 6520  and memory. The 
-000087d0: 6c6f 6769 6361 6c20 4350 5573 2072 6570  logical CPUs rep
-000087e0: 6f72 7465 6420 7265 7072 6573 656e 7473  orted represents
-000087f0: 2074 6865 206e 756d 6265 7220 6f66 2070   the number of p
-00008800: 6879 7369 6361 6c20 636f 7265 7320 6d75  hysical cores mu
-00008810: 6c74 6970 6c69 6564 2062 7920 7468 6520  ltiplied by the 
-00008820: 6e75 6d62 6572 206f 6620 7468 7265 6164  number of thread
-00008830: 7320 7468 6174 2063 616e 2072 756e 206f  s that can run o
-00008840: 6e20 6561 6368 2063 6f72 652c 2061 6c73  n each core, als
-00008850: 6f20 6b6e 6f77 6e20 6173 2048 7970 6572  o known as Hyper
-00008860: 2054 6872 6561 6469 6e67 2e20 5468 6520   Threading. The 
-00008870: 6e75 6d62 6572 206f 6620 6c6f 6769 6361  number of logica
-00008880: 6c20 4350 5573 206d 6179 206e 6f74 206e  l CPUs may not n
-00008890: 6563 6573 7361 7269 6c79 2062 6520 6571  ecessarily be eq
-000088a0: 7569 7661 6c65 6e74 2074 6f20 7468 6520  uivalent to the 
-000088b0: 6163 7475 616c 206e 756d 6265 7220 6f66  actual number of
-000088c0: 2043 5055 7320 7468 6520 6375 7272 656e   CPUs the curren
-000088d0: 7420 7072 6f63 6573 7320 6361 6e20 7573  t process can us
-000088e0: 652e 2054 6865 2061 7661 696c 6162 6c65  e. The available
-000088f0: 206d 656d 6f72 7920 7265 6665 7273 2074   memory refers t
-00008900: 6f20 7468 6520 6d65 6d6f 7279 2074 6861  o the memory tha
-00008910: 7420 6361 6e20 6265 2067 6976 656e 2069  t can be given i
-00008920: 6e73 7461 6e74 6c79 2074 6f20 7072 6f63  nstantly to proc
-00008930: 6573 7365 7320 7769 7468 6f75 7420 7468  esses without th
-00008940: 6520 7379 7374 656d 2067 6f69 6e67 2069  e system going i
-00008950: 6e74 6f20 7377 6170 2061 6e64 2072 6566  nto swap and ref
-00008960: 6c65 6374 7320 7468 6520 6163 7475 616c  lects the actual
-00008970: 206d 656d 6f72 7920 6176 6169 6c61 626c   memory availabl
-00008980: 652e 2054 6865 2066 7265 6520 6d65 6d6f  e. The free memo
-00008990: 7279 2072 6566 6572 7320 746f 2074 6865  ry refers to the
-000089a0: 206d 656d 6f72 7920 6e6f 7420 6265 696e   memory not bein
-000089b0: 6720 7573 6564 2061 7420 616c 6c20 287a  g used at all (z
-000089c0: 6572 6f65 6429 2074 6861 7420 6973 2072  eroed) that is r
-000089d0: 6561 6469 6c79 2061 7661 696c 6162 6c65  eadily available
-000089e0: 2e20 5468 6520 6465 7363 7269 7074 696f  . The descriptio
-000089f0: 6e20 6162 6f76 6520 7265 6665 7273 2074  n above refers t
-00008a00: 6f20 7468 6520 646f 6375 6d65 6e74 6174  o the documentat
-00008a10: 696f 6e20 6f66 2060 7073 7574 696c 602e  ion of `psutil`.
-00008a20: 0a0a 2323 2320 7767 6420 646d 640a 0a2a  ..### wgd dmd..*
-00008a30: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-00008a40: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-00008a50: 6d65 2a2a 0a60 6060 0a77 6764 2064 6d64  me**.```.wgd dmd
-00008a60: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00008a70: 6c65 6120 2d49 2032 202d 6520 3165 2d31  lea -I 2 -e 1e-1
-00008a80: 3020 2d62 7320 3130 3020 2d6e 7020 3520  0 -bs 100 -np 5 
-00008a90: 282d 6e6e 2920 282d 2d74 6f5f 7374 6f70  (-nn) (--to_stop
-00008aa0: 2920 282d 2d63 6473 2920 282d 6e20 3429  ) (--cds) (-n 4)
-00008ab0: 2028 2d6f 2077 6764 5f64 6d64 2920 282d   (-o wgd_dmd) (-
-00008ac0: 7420 776f 726b 696e 675f 746d 7029 0a60  t working_tmp).`
-00008ad0: 6060 200a 0a4e 6f74 6520 7468 6174 2077  `` ..Note that w
-00008ae0: 6520 646f 6e27 7420 7072 6f76 6964 6520  e don't provide 
-00008af0: 7468 6520 6461 7461 206f 6620 7468 6973  the data of this
-00008b00: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
-00008b10: 2028 6364 7329 2066 696c 6520 6041 7175   (cds) file `Aqu
-00008b20: 696c 6567 6961 5f63 6f65 7275 6c65 6160  ilegia_coerulea`
-00008b30: 2062 7574 2069 7420 6361 6e20 6265 2065   but it can be e
-00008b40: 6173 696c 7920 646f 776e 6c6f 6164 6564  asily downloaded
-00008b50: 2061 7420 5b50 6879 746f 7a6f 6d65 5d28   at [Phytozome](
-00008b60: 6874 7470 733a 2f2f 7068 7974 6f7a 6f6d  https://phytozom
-00008b70: 652d 6e65 7874 2e6a 6769 2e64 6f65 2e67  e-next.jgi.doe.g
-00008b80: 6f76 2f69 6e66 6f2f 4163 6f65 7275 6c65  ov/info/Acoerule
-00008b90: 615f 7633 5f31 2920 2873 616d 6520 666f  a_v3_1) (same fo
-00008ba0: 7220 6f74 6865 7220 5573 6167 6520 646f  r other Usage do
-00008bb0: 6329 2e20 4e6f 7465 2074 6861 7420 696e  c). Note that in
-00008bc0: 2074 6865 2069 7373 7565 7320 736f 6d65   the issues some
-00008bd0: 2075 7365 7273 2064 6964 6e27 7420 646f   users didn't do
-00008be0: 776e 6c6f 6164 2074 6865 2060 4163 6f65  wnload the `Acoe
-00008bf0: 7275 6c65 615f 3332 325f 7633 2e31 2e63  rulea_322_v3.1.c
-00008c00: 6473 5f70 7269 6d61 7279 5472 616e 7363  ds_primaryTransc
-00008c10: 7269 7074 4f6e 6c79 2e66 612e 677a 6020  riptOnly.fa.gz` 
-00008c20: 6275 7420 696e 7374 6561 6420 7468 6520  but instead the 
-00008c30: 6041 636f 6572 756c 6561 5f33 3232 5f76  `Acoerulea_322_v
-00008c40: 332e 312e 6364 732e 6661 2e67 7a60 2066  3.1.cds.fa.gz` f
-00008c50: 696c 652e 2046 6f72 2074 6865 2063 6f6e  ile. For the con
-00008c60: 7374 7275 6374 696f 6e20 6f66 2077 686f  struction of who
-00008c70: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00008c80: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00008c90: 7269 6275 7469 6f6e 2c20 2a2a 6f6e 6c79  ribution, **only
-00008ca0: 206f 6e65 2074 7261 6e73 6372 6970 7420   one transcript 
-00008cb0: 2874 6865 2070 7269 6d61 7279 206f 6e65  (the primary one
-00008cc0: 2920 7065 7220 6765 6e65 2073 686f 756c  ) per gene shoul
-00008cd0: 6420 6265 2069 6e63 6c75 6465 642a 2a20  d be included** 
-00008ce0: 7375 6368 2074 6861 7420 7468 6520 2a4b  such that the *K
-00008cf0: 2a3c 7375 623e 533c 2f73 7562 3e20 6973  *<sub>S</sub> is
-00008d00: 2072 6561 6c6c 7920 696e 6469 6361 7469   really indicati
-00008d10: 6e67 2074 6865 2061 6765 206f 6620 6765  ng the age of ge
-00008d20: 6e65 2064 7570 6c69 6361 7469 6f6e 2065  ne duplication e
-00008d30: 7665 6e74 2c20 696e 7374 6561 6420 6f66  vent, instead of
-00008d40: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008d50: 6963 696e 672e 2054 7261 6e73 6372 6970  icing. Transcrip
-00008d60: 746f 6d65 2064 6174 6120 7368 6f75 6c64  tome data should
-00008d70: 2062 6520 6361 7265 6675 6c6c 7920 7472   be carefully tr
-00008d80: 6561 7465 6420 7769 7468 2064 652d 7265  eated with de-re
-00008d90: 6475 6e64 616e 6379 2073 6f20 6173 2074  dundancy so as t
-00008da0: 6f20 7265 6475 6365 2074 6865 2066 616c  o reduce the fal
-00008db0: 7365 2070 6f73 6974 6976 6520 6475 706c  se positive dupl
-00008dc0: 6963 6174 696f 6e20 6275 6d70 2063 6175  ication bump cau
-00008dd0: 7365 6420 6279 2070 6572 7661 7369 7665  sed by pervasive
-00008de0: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008df0: 6963 696e 672e 2048 6572 6520 7468 6520  icing. Here the 
-00008e00: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
-00008e10: 2070 6172 616d 6574 6572 2c20 6769 7665   parameter, give
-00008e20: 6e20 6279 2060 2d49 6020 6f72 2060 2d2d  n by `-I` or `--
-00008e30: 696e 666c 6174 696f 6e60 2c20 6166 6665  inflation`, affe
-00008e40: 6374 7320 7468 6520 6772 616e 756c 6172  cts the granular
-00008e50: 6974 7920 6f72 2072 6573 6f6c 7574 696f  ity or resolutio
-00008e60: 6e20 6f66 2074 6865 2063 6c75 7374 6572  n of the cluster
-00008e70: 696e 6720 6f75 7463 6f6d 6520 616e 6420  ing outcome and 
-00008e80: 696d 706c 6963 6974 6c79 2063 6f6e 7472  implicitly contr
-00008e90: 6f6c 6c73 2074 6865 206e 756d 6265 7220  olls the number 
-00008ea0: 6f66 2063 6c75 7374 6572 732c 2077 6974  of clusters, wit
-00008eb0: 6820 6c6f 7720 7661 6c75 6573 2073 7563  h low values suc
-00008ec0: 6820 6173 2031 2e33 206f 7220 312e 3420  h as 1.3 or 1.4 
-00008ed0: 6c65 6164 696e 6720 746f 2066 6577 6572  leading to fewer
-00008ee0: 2062 7574 206c 6172 6765 7220 636c 7573   but larger clus
-00008ef0: 7465 7273 2061 6e64 2068 6967 6820 7661  ters and high va
-00008f00: 6c75 6573 2073 7563 6820 6173 2035 206f  lues such as 5 o
-00008f10: 7220 3620 6c65 6164 696e 6720 746f 206d  r 6 leading to m
-00008f20: 6f72 6520 6275 7420 736d 616c 6c65 7220  ore but smaller 
-00008f30: 636c 7573 7465 7273 2e20 5765 2073 6574  clusters. We set
-00008f40: 2074 6865 2064 6566 6175 6c74 2076 616c   the default val
-00008f50: 7565 2061 7320 3220 6173 2073 7567 6765  ue as 2 as sugge
-00008f60: 7374 6564 2062 7920 5b4d 434c 5d28 6874  sted by [MCL](ht
-00008f70: 7470 733a 2f2f 6d69 6361 6e73 2e6f 7267  tps://micans.org
-00008f80: 2f6d 636c 2f29 2e20 5468 6520 652d 7661  /mcl/). The e-va
-00008f90: 6c75 6520 6375 742d 6f66 6620 666f 7220  lue cut-off for 
-00008fa0: 7365 7175 656e 6365 2073 696d 696c 6172  sequence similar
-00008fb0: 6974 792c 2067 6976 656e 2062 7920 602d  ity, given by `-
-00008fc0: 6560 206f 7220 602d 2d65 7661 6c60 2c20  e` or `--eval`, 
-00008fd0: 7768 6963 6820 6465 6e6f 7465 7320 7468  which denotes th
-00008fe0: 6520 6578 7065 6374 6564 2076 616c 7565  e expected value
-00008ff0: 206f 6620 7468 6520 6869 7420 7175 616e   of the hit quan
-00009000: 7469 6669 6573 2074 6865 206e 756d 6265  tifies the numbe
-00009010: 7220 6f66 2061 6c69 676e 6d65 6e74 7320  r of alignments 
-00009020: 6f66 2073 696d 696c 6172 206f 7220 6265  of similar or be
-00009030: 7474 6572 2071 7561 6c69 7479 2074 6861  tter quality tha
-00009040: 7420 796f 7520 6578 7065 6374 2074 6f20  t you expect to 
-00009050: 6669 6e64 2073 6561 7263 6869 6e67 2074  find searching t
-00009060: 6869 7320 7175 6572 7920 6167 6169 6e73  his query agains
-00009070: 7420 6120 6461 7461 6261 7365 206f 6620  t a database of 
-00009080: 7261 6e64 6f6d 2073 6571 7565 6e63 6573  random sequences
-00009090: 2074 6865 2073 616d 6520 7369 7a65 2061   the same size a
-000090a0: 7320 7468 6520 6163 7475 616c 2074 6172  s the actual tar
-000090b0: 6765 7420 6461 7461 6261 7365 2c20 6973  get database, is
-000090c0: 2074 6865 206b 6579 2070 6172 616d 6574   the key paramet
-000090d0: 6572 206d 6561 7375 7269 6e67 2074 6865  er measuring the
-000090e0: 2073 6967 6e69 6669 6361 6e63 6520 6f66   significance of
-000090f0: 2061 2068 6974 2c20 7768 6963 6820 6973   a hit, which is
-00009100: 2073 6574 2068 6572 6520 6173 2064 6566   set here as def
-00009110: 6175 6c74 2031 652d 3130 2e20 4e6f 7465  ault 1e-10. Note
-00009120: 2074 6861 7420 5b44 4941 4d4f 4e44 5d28   that [DIAMOND](
-00009130: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00009140: 6f6d 2f62 6275 6368 6669 6e6b 2f64 6961  om/bbuchfink/dia
-00009150: 6d6f 6e64 2f77 696b 692f 312e 2d54 7574  mond/wiki/1.-Tut
-00009160: 6f72 6961 6c29 2069 7473 656c 6620 6279  orial) itself by
-00009170: 2064 6566 6175 6c74 206f 6e6c 7920 7265   default only re
-00009180: 706f 7274 7320 616c 6c20 616c 6967 6e6d  ports all alignm
-00009190: 656e 7473 2077 6974 6820 652d 7661 6c75  ents with e-valu
-000091a0: 6520 3c20 302e 3030 312e 2054 6865 2070  e < 0.001. The p
-000091b0: 6572 6365 6e74 6167 6520 6f66 2075 7070  ercentage of upp
-000091c0: 6572 2068 6974 7320 7573 6564 2066 6f72  er hits used for
-000091d0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
-000091e0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
-000091f0: 6e20 6279 2060 2d6e 7060 206f 7220 602d  n by `-np` or `-
-00009200: 2d6e 6f72 6d61 6c69 7a65 6470 6572 6365  -normalizedperce
-00009210: 6e74 602c 2077 6869 6368 2064 6574 6572  nt`, which deter
-00009220: 6d69 6e65 7320 7468 6520 7570 7065 7220  mines the upper 
-00009230: 7065 7263 656e 7469 6c65 206f 6620 6869  percentile of hi
-00009240: 7473 2070 6572 2062 696e 2028 6361 7465  ts per bin (cate
-00009250: 676f 7269 7a65 6420 6279 2067 656e 6520  gorized by gene 
-00009260: 6c65 6e67 7468 2920 7573 6564 2069 6e20  length) used in 
-00009270: 7468 6520 6669 7420 6f66 206c 696e 6561  the fit of linea
-00009280: 7220 7265 6772 6573 7369 6f6e 2c20 636f  r regression, co
-00009290: 6e73 6964 6572 696e 6720 7468 6174 206e  nsidering that n
-000092a0: 6f74 2061 6c6c 2068 6974 7320 7065 7220  ot all hits per 
-000092b0: 6269 6e20 7368 6f77 2061 7070 6172 656e  bin show apparen
-000092c0: 7420 6c69 6e65 6172 2072 656c 6174 696f  t linear relatio
-000092d0: 6e73 6869 702c 2069 7320 7365 7420 6173  nship, is set as
-000092e0: 2064 6566 6175 6c74 2035 2c20 696e 6469   default 5, indi
-000092f0: 6361 7469 6e67 2074 6865 2075 7361 6765  cating the usage
-00009300: 206f 6620 746f 7020 3525 2068 6974 7320   of top 5% hits 
-00009310: 7065 7220 6269 6e2e 2054 6865 206e 756d  per bin. The num
-00009320: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
-00009330: 6465 6420 696e 2067 656e 6520 6c65 6e67  ded in gene leng
-00009340: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
-00009350: 2c20 6769 7665 6e20 6279 2060 2d62 7360  , given by `-bs`
-00009360: 206f 7220 602d 2d62 696e 7360 2c20 6465   or `--bins`, de
-00009370: 7465 726d 696e 6573 2074 6865 206e 756d  termines the num
-00009380: 6265 7220 6f66 2062 696e 7320 746f 2063  ber of bins to c
-00009390: 6174 6567 6f72 697a 6520 7468 6520 6765  ategorize the ge
-000093a0: 6e65 206c 656e 6774 682c 2069 7320 7365  ne length, is se
-000093b0: 7420 6173 2064 6566 6175 6c74 2031 3030  t as default 100
-000093c0: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
-000093d0: 602d 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f  `-nn` or `--nono
-000093e0: 726d 616c 697a 6174 696f 6e60 2063 616e  rmalization` can
-000093f0: 2062 6520 7365 7420 746f 2063 616c 6c20   be set to call 
-00009400: 6f66 6620 7468 6520 6e6f 726d 616c 697a  off the normaliz
-00009410: 6174 696f 6e20 7072 6f63 6573 732c 2061  ation process, a
-00009420: 6c74 686f 7567 6820 6974 2773 2073 7567  lthough it's sug
-00009430: 6765 7374 6564 2074 6f20 636f 6e64 7563  gested to conduc
-00009440: 7420 7468 6520 6e6f 726d 616c 697a 6174  t the normalizat
-00009450: 696f 6e20 746f 2061 6371 7569 7265 206d  ion to acquire m
-00009460: 6f72 6520 6163 6375 7261 7465 2067 656e  ore accurate gen
-00009470: 6520 6661 6d69 6c79 2063 6c75 7374 6572  e family cluster
-00009480: 696e 6720 7265 7375 6c74 2e20 5468 6520  ing result. The 
-00009490: 7061 7261 6d65 7465 7273 2060 2d2d 746f  parameters `--to
-000094a0: 5f73 746f 7060 2061 6e64 2060 2d2d 6364  _stop` and `--cd
-000094b0: 7360 2063 6f6e 7472 6f6c 2074 6865 2062  s` control the b
-000094c0: 6568 6176 696f 7572 206f 6620 7472 616e  ehaviour of tran
-000094d0: 736c 6174 696e 6720 636f 6469 6e67 2073  slating coding s
-000094e0: 6571 7565 6e63 6520 696e 746f 2061 6d69  equence into ami
-000094f0: 6e6f 2061 6369 6420 7365 7175 656e 6365  no acid sequence
-00009500: 2e20 4966 2074 6865 2060 2d2d 746f 5f73  . If the `--to_s
-00009510: 746f 7060 2077 6173 2073 6574 2c20 7468  top` was set, th
-00009520: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009530: 756c 6420 6265 2074 6572 6d69 6e61 7465  uld be terminate
-00009540: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
-00009550: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00009560: 6f6e 2c20 6f74 6865 7277 6973 6520 7468  on, otherwise th
-00009570: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009580: 756c 6420 7369 6d70 6c79 2073 6b69 7020  uld simply skip 
-00009590: 616e 7920 7374 6f70 2063 6f64 6f6e 732e  any stop codons.
-000095a0: 2049 6620 7468 6520 602d 2d63 6473 6020   If the `--cds` 
-000095b0: 7761 7320 7365 742c 2073 6571 7565 6e63  was set, sequenc
-000095c0: 6573 2074 6861 7420 646f 6573 6e27 7420  es that doesn't 
-000095d0: 7374 6172 7420 7769 7468 2061 2076 616c  start with a val
-000095e0: 6964 2073 7461 7274 2063 6f64 6f6e 2c20  id start codon, 
-000095f0: 6f72 2063 6f6e 7461 696e 7320 6d6f 7265  or contains more
-00009600: 2074 6861 6e20 6f6e 6520 696e 2d66 7261   than one in-fra
-00009610: 6d65 2073 746f 7020 636f 646f 6e2c 206f  me stop codon, o
-00009620: 7220 6973 206e 6f74 2064 6976 6964 6162  r is not dividab
-00009630: 6c65 2062 7920 332c 2077 6f75 6c64 2062  le by 3, would b
-00009640: 6520 7369 6d70 6c79 2064 726f 7070 6564  e simply dropped
-00009650: 2c20 7375 6368 2074 6861 7420 6f6e 6c79  , such that only
-00009660: 2073 7472 6963 7420 636f 6d70 6c65 7465   strict complete
-00009670: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
-00009680: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00009690: 6465 6420 696e 2074 6865 2073 7562 7365  ded in the subse
-000096a0: 7175 656e 7420 616e 616c 7973 6973 2e20  quent analysis. 
-000096b0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
-000096c0: 7261 6c6c 656c 2074 6872 6561 6473 2062  rallel threads b
-000096d0: 7920 7468 6520 6f70 7469 6f6e 2060 2d6e  y the option `-n
-000096e0: 6020 6f72 2060 2d2d 6e74 6872 6561 6473  ` or `--nthreads
-000096f0: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
-00009700: 6163 6365 6c65 7261 7465 2074 6865 2063  accelerate the c
-00009710: 616c 6375 6c61 7469 6f6e 2077 6974 6869  alculation withi
-00009720: 6e20 6064 6961 6d6f 6e64 602e 2054 6865  n `diamond`. The
-00009730: 2064 6972 6563 746f 7279 206f 6620 6f75   directory of ou
-00009740: 7470 7574 206f 7220 696e 7465 726d 6564  tput or intermed
-00009750: 6961 7465 2066 696c 6573 2069 7320 6465  iate files is de
-00009760: 7465 726d 696e 6564 2062 7920 7468 6520  termined by the 
-00009770: 7061 7261 6d65 7465 7220 602d 6f60 206f  parameter `-o` o
-00009780: 7220 602d 2d6f 7574 6469 7260 2c20 616e  r `--outdir`, an
-00009790: 6420 602d 7460 206f 7220 602d 2d74 6d70  d `-t` or `--tmp
-000097a0: 6469 7260 2c20 7768 6963 6820 7769 6c6c  dir`, which will
-000097b0: 2062 6520 6372 6561 7465 6420 6279 2074   be created by t
-000097c0: 6865 2070 726f 6772 616d 2069 7473 656c  he program itsel
-000097d0: 6620 616e 6420 6265 206f 7665 7277 7269  f and be overwri
-000097e0: 7474 656e 2069 6620 7468 6520 666f 6c64  tten if the fold
-000097f0: 6572 2068 6173 2061 6c72 6561 6479 2062  er has already b
-00009800: 6565 6e20 6372 6561 7465 642e 204e 6f74  een created. Not
-00009810: 6520 7468 6174 2074 6865 2073 6f66 7477  e that the softw
-00009820: 6172 6520 6064 6961 6d6f 6e64 6020 7368  are `diamond` sh
-00009830: 6f75 6c64 2062 6520 7072 652d 696e 7374  ould be pre-inst
-00009840: 616c 6c65 6420 616e 6420 7365 7420 696e  alled and set in
-00009850: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-00009860: 2070 6174 6820 696e 2061 6c6c 2074 6865   path in all the
-00009870: 2061 6e61 6c79 7369 7320 7065 7266 6f72   analysis perfor
-00009880: 6d65 6420 6279 2060 7767 6420 646d 6460  med by `wgd dmd`
-00009890: 2065 7863 6570 7420 666f 7220 7468 6520   except for the 
-000098a0: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
-000098b0: 6365 6e63 6520 616e 616c 7973 6973 2e0a  cence analysis..
-000098c0: 0a2a 2a57 6520 7375 6767 6573 7420 7468  .**We suggest th
-000098d0: 6174 2074 6865 2064 6566 6175 6c74 2073  at the default s
-000098e0: 6574 7469 6e67 2069 6e20 7768 6963 6820  etting in which 
-000098f0: 7468 6520 696e 666c 6174 696f 6e20 6661  the inflation fa
-00009900: 6374 6f72 2069 7320 7365 7420 6173 2032  ctor is set as 2
-00009910: 2c20 652d 7661 6c75 6520 6375 742d 6f66  , e-value cut-of
-00009920: 6620 6173 2031 652d 3130 2061 6e64 206f  f as 1e-10 and o
-00009930: 7468 6572 2070 6172 616d 6574 6572 7320  ther parameters 
-00009940: 696e 2064 6566 6175 6c74 2069 7320 6120  in default is a 
-00009950: 676f 6f64 2073 7461 7274 696e 6720 706f  good starting po
-00009960: 696e 742c 2075 6e6c 6573 7320 796f 7520  int, unless you 
-00009970: 7370 6563 6966 6963 616c 6c79 2077 616e  specifically wan
-00009980: 7420 746f 2065 7870 6c6f 7265 2074 6865  t to explore the
-00009990: 2065 6666 6563 7473 206f 6620 6469 6666   effects of diff
-000099a0: 6572 656e 7420 7061 7261 6d65 7465 7273  erent parameters
-000099b0: 2e2a 2a20 5375 6368 2074 6861 7420 7468  .** Such that th
-000099c0: 6520 636f 6d6d 616e 6420 666f 7220 7468  e command for th
-000099d0: 6520 6465 6c69 6e65 6174 696f 6e20 6f66  e delineation of
-000099e0: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
-000099f0: 6973 2073 696d 706c 7920 6173 2062 656c  is simply as bel
-00009a00: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-00009a10: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00009a20: 6c65 610a 6060 600a 0a2a 2a54 6865 2064  lea.```..**The d
-00009a30: 656c 696e 6561 7469 6f6e 206f 6620 5242  elineation of RB
-00009a40: 4873 2a2a 0a60 6060 0a77 6764 2064 6d64  Hs**.```.wgd dmd
-00009a50: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-00009a60: 6e63 6532 202d 6520 3165 2d31 3020 2d62  nce2 -e 1e-10 -b
-00009a70: 7320 3130 3020 2d6e 7020 3520 282d 6e6e  s 100 -np 5 (-nn
-00009a80: 2920 282d 6e20 3429 2028 2d63 2030 2e39  ) (-n 4) (-c 0.9
-00009a90: 2920 282d 2d6f 6766 6f72 6d61 7429 2028  ) (--ogformat) (
-00009aa0: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
-00009ab0: 7329 2028 2d6f 2077 6764 5f64 6d64 2920  s) (-o wgd_dmd) 
-00009ac0: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-00009ad0: 0a60 6060 0a0a 546f 2064 656c 696e 6561  .```..To delinea
-00009ae0: 7465 2052 4248 7320 6265 7477 6565 6e20  te RBHs between 
-00009af0: 7477 6f20 6364 7320 7365 7175 656e 6365  two cds sequence
-00009b00: 2066 696c 6573 2c20 7468 6520 7265 6c65   files, the rele
-00009b10: 7661 6e74 2070 6172 616d 6574 6572 2069  vant parameter i
-00009b20: 7320 6d6f 7374 6c79 2074 6865 2073 616d  s mostly the sam
-00009b30: 6520 6173 2077 686f 6c65 2070 6172 616e  e as whole paran
-00009b40: 6f6d 6520 696e 6665 7265 6e63 652c 2065  ome inference, e
-00009b50: 7863 6570 7420 666f 7220 7468 6520 7061  xcept for the pa
-00009b60: 7261 6d65 7465 7220 602d 6360 206f 7220  rameter `-c` or 
-00009b70: 602d 2d63 7363 6f72 6560 2c20 7768 6963  `--cscore`, whic
-00009b80: 6820 7261 6e67 6573 2062 6574 7765 656e  h ranges between
-00009b90: 2030 2061 6e64 2031 2061 6e64 2069 7320   0 and 1 and is 
-00009ba0: 7573 6564 2074 6f20 7265 6c61 7820 7468  used to relax th
-00009bb0: 6520 7369 6d69 6c61 7269 7479 2063 7574  e similarity cut
-00009bc0: 6f66 6620 6672 6f6d 2074 6865 2065 7863  off from the exc
-00009bd0: 6c75 7369 7665 2072 6563 6970 726f 6361  lusive reciproca
-00009be0: 6c20 6265 7374 2068 6974 7320 746f 2061  l best hits to a
-00009bf0: 2063 6572 7461 696e 2072 6174 696f 2061   certain ratio a
-00009c00: 7320 746f 2074 6865 2062 6573 7420 6869  s to the best hi
-00009c10: 7473 2e20 466f 7220 696e 7374 616e 6365  ts. For instance
-00009c20: 2c20 6966 2074 6865 2067 656e 6520 6231  , if the gene b1
-00009c30: 2066 726f 6d20 6765 6e6f 6d65 2042 2068   from genome B h
-00009c40: 6173 2074 6865 2062 6573 7420 6869 7420  as the best hit 
-00009c50: 6765 6e65 2061 3120 6672 6f6d 2067 656e  gene a1 from gen
-00009c60: 6f6d 6520 4120 7769 7468 2074 6865 2062  ome A with the b
-00009c70: 6974 2073 636f 7265 2061 7320 3130 302c  it score as 100,
-00009c80: 2077 6869 6368 2069 7320 6120 7363 6f72   which is a scor
-00009c90: 696e 6720 6d61 7472 6978 2069 6e64 6570  ing matrix indep
-00009ca0: 656e 6465 6e74 206d 6561 7375 7265 206f  endent measure o
-00009cb0: 6620 7468 6520 286c 6f63 616c 2920 7369  f the (local) si
-00009cc0: 6d69 6c61 7269 7479 206f 6620 7468 6520  milarity of the 
-00009cd0: 7477 6f20 616c 6967 6e65 6420 7365 7175  two aligned sequ
-00009ce0: 656e 6365 732c 2077 6974 6820 6c61 7267  ences, with larg
-00009cf0: 6572 2076 616c 7565 7320 696e 6469 6361  er values indica
-00009d00: 7469 6e67 2068 6967 6865 7220 7369 6d69  ting higher simi
-00009d10: 6c61 7269 7469 6573 2c20 6769 7665 6e20  larities, given 
-00009d20: 7468 6520 602d 6320 302e 3960 2c20 6765  the `-c 0.9`, ge
-00009d30: 6e65 7320 6672 6f6d 2067 656e 6f6d 6520  nes from genome 
-00009d40: 4120 7768 6963 6820 6861 7320 7468 6520  A which has the 
-00009d50: 6269 7420 7363 6f72 6520 7769 7468 2067  bit score with g
-00009d60: 656e 6520 6231 2068 6967 6865 7220 7468  ene b1 higher th
-00009d70: 616e 2030 2e39 7831 3030 2077 696c 6c20  an 0.9x100 will 
-00009d80: 616c 736f 2062 6520 7772 6974 7465 6e20  also be written 
-00009d90: 696e 2074 6865 2072 6573 756c 7420 6669  in the result fi
-00009da0: 6c65 2c20 7768 6963 6820 696e 2061 2073  le, which in a s
-00009db0: 656e 7365 2061 7265 206e 6f74 2052 4248  ense are not RBH
-00009dc0: 7320 616e 796d 6f72 6520 6f66 2063 6f75  s anymore of cou
-00009dd0: 7273 652c 2062 7574 2074 6865 2068 6967  rse, but the hig
-00009de0: 686c 7920 7369 6d69 6c61 7220 686f 6d6f  hly similar homo
-00009df0: 6c6f 6775 6520 7061 6972 732e 2049 6620  logue pairs. If 
-00009e00: 6d6f 7265 2074 6861 6e20 3220 7365 7175  more than 2 sequ
-00009e10: 656e 6365 2066 696c 6573 2077 6572 6520  ence files were 
-00009e20: 7072 6f76 6964 6564 2c20 6576 6572 7920  provided, every 
-00009e30: 7061 6972 2d77 6973 6520 5242 4873 2077  pair-wise RBHs w
-00009e40: 6f75 6c64 2062 6520 6361 6c63 756c 6174  ould be calculat
-00009e50: 6564 2065 7863 6570 7420 666f 7220 7175  ed except for qu
-00009e60: 6572 7969 6e67 2074 6865 2073 616d 6520  erying the same 
-00009e70: 7365 7175 656e 6365 2069 7473 656c 662e  sequence itself.
-00009e80: 2054 6865 206e 756d 6265 7220 6f66 2070   The number of p
-00009e90: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-00009ea0: 746f 2062 6f6f 7374 6572 2074 6865 2072  to booster the r
-00009eb0: 756e 6e69 6e67 2073 7065 6564 2063 616e  unning speed can
-00009ec0: 2062 6520 7365 7420 6279 2074 6865 206f   be set by the o
-00009ed0: 7074 696f 6e20 602d 6e60 206f 7220 602d  ption `-n` or `-
-00009ee0: 2d6e 7468 7265 6164 7360 2077 6869 6368  -nthreads` which
-00009ef0: 2069 7320 7375 6767 6573 7465 6420 746f   is suggested to
-00009f00: 2062 6520 7365 7420 6173 2028 4e2d 3129   be set as (N-1)
-00009f10: 4e2f 3220 7768 6572 6520 4e20 6973 2074  N/2 where N is t
-00009f20: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
-00009f30: 2066 696c 6573 2074 6f20 6163 6869 6576   files to achiev
-00009f40: 6520 7468 6520 6869 6768 6573 7420 6566  e the highest ef
-00009f50: 6669 6369 656e 6379 2e20 5468 6520 6f70  ficiency. The op
-00009f60: 7469 6f6e 2060 2d2d 6f67 666f 726d 6174  tion `--ogformat
-00009f70: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
-00009f80: 6164 6420 696e 6465 7820 2866 6f72 2069  add index (for i
-00009f90: 6e73 7461 6e63 6520 4746 3030 3030 3030  nstance GF000000
-00009fa0: 3031 2920 746f 2074 6865 206f 7574 7075  01) to the outpu
-00009fb0: 7420 5242 4820 6765 6e65 2066 616d 696c  t RBH gene famil
-00009fc0: 6965 7320 7768 6963 6820 6361 6e20 6265  ies which can be
-00009fd0: 2066 7572 7468 6572 2075 7365 6420 696e   further used in
-00009fe0: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00009ff0: 7375 623e 2063 616c 6375 6c61 7469 6f6e  sub> calculation
-0000a000: 2062 7920 6077 6764 206b 7364 602e 0a0a   by `wgd ksd`...
-0000a010: 2a2a 5468 6520 7375 6767 6573 7465 6420  **The suggested 
-0000a020: 636f 6d6d 616e 6420 746f 2073 7461 7274  command to start
-0000a030: 2077 6974 6820 6973 2061 6c73 6f20 756e   with is also un
-0000a040: 6465 7220 7468 6520 6465 6661 756c 7420  der the default 
-0000a050: 7365 7474 696e 6720 7769 7468 2074 6865  setting with the
-0000a060: 2063 6f6d 6d61 6e64 2073 686f 776e 2062   command shown b
-0000a070: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-0000a080: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
-0000a090: 7175 656e 6365 320a 6060 600a 0a2a 2a54  quence2.```..**T
-0000a0a0: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
-0000a0b0: 6620 6c6f 6361 6c20 4d52 4248 732a 2a0a  f local MRBHs**.
-0000a0c0: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000a0d0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000a0e0: 7365 7175 656e 6365 3320 2d66 2073 6571  sequence3 -f seq
-0000a0f0: 7565 6e63 6531 202d 6520 3165 2d31 3020  uence1 -e 1e-10 
-0000a100: 2d62 7320 3130 3020 2d6e 7020 3520 282d  -bs 100 -np 5 (-
-0000a110: 6e6e 2920 282d 6e20 3429 2028 2d6b 6629  nn) (-n 4) (-kf)
-0000a120: 2028 2d6b 6429 2028 2d63 2030 2e39 2920   (-kd) (-c 0.9) 
-0000a130: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
-0000a140: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
-0000a150: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
-0000a160: 290a 6060 600a 0a2a 2a54 6865 2064 6973  ).```..**The dis
-0000a170: 7469 6e63 7469 6f6e 2062 6574 7765 656e  tinction between
-0000a180: 206c 6f63 616c 2061 6e64 2067 6c6f 6261   local and globa
-0000a190: 6c20 2868 6572 6575 6e64 6572 2920 4d52  l (hereunder) MR
-0000a1a0: 4248 7320 6973 2074 6861 7420 6c6f 6361  BHs is that loca
-0000a1b0: 6c20 4d52 4248 7320 6172 6520 7468 6520  l MRBHs are the 
-0000a1c0: 7265 7375 6c74 7320 6f66 206d 6572 6765  results of merge
-0000a1d0: 6420 5242 4873 206f 6e20 6120 6a6f 696e  d RBHs on a join
-0000a1e0: 7420 666f 6361 6c20 7370 6563 6965 732c  t focal species,
-0000a1f0: 2066 6f72 2069 6e73 7461 6e63 6520 696e   for instance in
-0000a200: 2061 2074 6872 6565 2073 7065 6369 6573   a three species
-0000a210: 2073 7973 7465 6d20 2841 2c28 422c 4329   system (A,(B,C)
-0000a220: 292c 2074 6865 206c 6f63 616c 204d 5242  ), the local MRB
-0000a230: 4873 206f 6620 4120 6f6e 6c79 2072 6571  Hs of A only req
-0000a240: 7569 7265 2074 6865 2063 616c 6375 6c61  uire the calcula
-0000a250: 7469 6f6e 206f 6620 5242 4873 2062 6574  tion of RBHs bet
-0000a260: 7765 656e 2041 2061 6e64 2043 2028 6465  ween A and C (de
-0000a270: 6e6f 7465 6420 6173 2041 4329 2061 6e64  noted as AC) and
-0000a280: 2041 4220 616e 6420 7468 656e 2074 6865   AB and then the
-0000a290: 206d 6572 6769 6e67 206f 6620 4142 2061   merging of AB a
-0000a2a0: 6e64 2041 4320 6174 2074 6865 2061 7869  nd AC at the axi
-0000a2b0: 7320 6f66 2041 2c20 7768 696c 6520 7468  s of A, while th
-0000a2c0: 6520 676c 6f61 626c 204d 5242 4873 2061  e gloabl MRBHs a
-0000a2d0: 7265 2069 6e64 6570 656e 6465 6e74 206f  re independent o
-0000a2e0: 6620 666f 6361 6c20 7370 6563 6965 7320  f focal species 
-0000a2f0: 696e 2074 6861 7420 6974 206a 7573 7420  in that it just 
-0000a300: 6361 6c6c 7320 7468 6520 6361 6c63 756c  calls the calcul
-0000a310: 6174 696f 6e73 206f 6620 616c 6c20 706f  ations of all po
-0000a320: 7373 6962 6c65 2073 7065 6369 6573 2070  ssible species p
-0000a330: 6169 7220 286e 6f74 2073 656c 6620 746f  air (not self to
-0000a340: 2073 656c 6620 7370 6563 6965 7320 7061   self species pa
-0000a350: 6972 292c 2073 7563 6820 7468 6174 2041  ir), such that A
-0000a360: 422c 2041 432c 2061 6e64 2042 4320 6172  B, AC, and BC ar
-0000a370: 6520 616c 6c20 746f 2062 6520 6361 6c63  e all to be calc
-0000a380: 756c 6174 6564 2061 6e64 206d 6572 6765  ulated and merge
-0000a390: 642e 2a2a 0a0a 5477 6f20 7479 7065 7320  d.**..Two types 
-0000a3a0: 6f66 204d 5242 4873 2061 7320 696e 7465  of MRBHs as inte
-0000a3b0: 7072 6574 6174 6564 2061 626f 7665 2063  pretated above c
-0000a3c0: 616e 2062 6520 6465 6c69 6e65 6174 6564  an be delineated
-0000a3d0: 2062 7920 6077 6764 2064 6d64 602c 2074   by `wgd dmd`, t
-0000a3e0: 6865 206c 6f63 616c 204d 5242 4873 2061  he local MRBHs a
-0000a3f0: 6e64 2074 6865 2067 6c6f 6261 6c20 4d52  nd the global MR
-0000a400: 4248 732e 2054 6865 206c 6f63 616c 204d  BHs. The local M
-0000a410: 5242 4873 2061 7265 2063 6f6e 7374 7275  RBHs are constru
-0000a420: 6374 6564 2062 7920 6d65 7267 696e 6720  cted by merging 
-0000a430: 616c 6c20 7468 6520 7265 6c65 7661 6e74  all the relevant
-0000a440: 2052 4248 7320 6f6e 6c79 2077 6974 6820   RBHs only with 
-0000a450: 7468 6520 666f 6361 6c20 7370 6563 6965  the focal specie
-0000a460: 732c 2077 6869 6368 2069 7320 7365 7420  s, which is set 
-0000a470: 6279 2060 2d66 6020 6f72 2060 2d2d 666f  by `-f` or `--fo
-0000a480: 6375 7360 2e20 5468 6520 7061 7261 6d65  cus`. The parame
-0000a490: 7465 7220 602d 6b66 6020 6f72 2060 2d2d  ter `-kf` or `--
-0000a4a0: 6b65 6570 6661 7374 6160 2063 616e 2062  keepfasta` can b
-0000a4b0: 6520 7365 7420 746f 2072 6574 6169 6e20  e set to retain 
-0000a4c0: 7468 6520 7365 7175 656e 6365 2069 6e66  the sequence inf
-0000a4d0: 6f72 6d61 7469 6f6e 206f 6620 6561 6368  ormation of each
-0000a4e0: 204d 5242 482e 2054 6865 2070 6172 616d   MRBH. The param
-0000a4f0: 6574 6572 2060 2d6b 6460 206f 7220 602d  eter `-kd` or `-
-0000a500: 2d6b 6565 7064 7570 6c69 6361 7465 7360  -keepduplicates`
-0000a510: 2064 6574 6572 6d69 6e65 7320 7768 6574   determines whet
-0000a520: 6865 7220 7468 6520 7361 6d65 2067 656e  her the same gen
-0000a530: 6573 2063 616e 2061 7070 6561 7220 696e  es can appear in
-0000a540: 2064 6966 6665 7265 6e74 206c 6f63 616c   different local
-0000a550: 204d 5242 4873 2e20 4e6f 726d 616c 6c79   MRBHs. Normally
-0000a560: 2074 6865 7265 2077 696c 6c20 6265 206e   there will be n
-0000a570: 6f20 6475 706c 6963 6174 6573 2069 6e20  o duplicates in 
-0000a580: 7468 6520 6c6f 6361 6c20 4d52 4248 7320  the local MRBHs 
-0000a590: 6275 7420 6966 2075 7365 7273 2073 6574  but if users set
-0000a5a0: 2074 6865 2060 2d63 6020 6173 2030 2e39   the `-c` as 0.9
-0000a5b0: 2028 6f72 2061 6e79 2076 616c 7565 2073   (or any value s
-0000a5c0: 6d61 6c6c 6572 2074 6861 6e20 3129 2c20  maller than 1), 
-0000a5d0: 6974 2773 206c 696b 656c 7920 7468 6174  it's likely that
-0000a5e0: 2074 6865 2073 616d 6520 6765 6e65 2077   the same gene w
-0000a5f0: 696c 6c20 6861 7665 2063 6861 6e63 6520  ill have chance 
-0000a600: 746f 2061 7070 6561 7220 6d75 6c74 6970  to appear multip
-0000a610: 6c65 2074 696d 6573 2069 6e20 6469 6666  le times in diff
-0000a620: 6572 656e 7420 6c6f 6361 6c20 4d52 4248  erent local MRBH
-0000a630: 732e 2054 6861 7420 6973 2074 6f20 7361  s. That is to sa
-0000a640: 792c 2074 6865 2070 6172 616d 6574 6572  y, the parameter
-0000a650: 2060 2d6b 6460 2069 7320 6d65 616e 696e   `-kd` is meanin
-0000a660: 6766 756c 206f 6e6c 7920 7768 656e 2069  gful only when i
-0000a670: 7427 7320 7365 7420 746f 6765 7468 6572  t's set together
-0000a680: 2077 6974 6820 7468 6520 7061 7261 6d65   with the parame
-0000a690: 7465 7220 602d 6360 2e20 5468 6520 6e75  ter `-c`. The nu
-0000a6a0: 6d62 6572 206f 6620 7061 7261 6c6c 656c  mber of parallel
-0000a6b0: 2074 6872 6561 6473 2069 7320 7375 6767   threads is sugg
-0000a6c0: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
-0000a6d0: 6173 2074 6865 206e 756d 6265 7220 6f66  as the number of
-0000a6e0: 2063 6473 2066 696c 6573 206d 696e 7573   cds files minus
-0000a6f0: 2031 2e0a 0a2a 2a41 2073 7567 6765 7374   1...**A suggest
-0000a700: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000a710: 6973 2075 6e64 6572 2074 6865 2064 6566  is under the def
-0000a720: 6175 6c74 2070 6172 616d 6574 6572 2077  ault parameter w
-0000a730: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000a740: 7368 6f77 6e20 6173 2062 656c 6f77 2e2a  shown as below.*
-0000a750: 2a0a 0a60 6060 0a77 6764 2064 6d64 2073  *..```.wgd dmd s
-0000a760: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000a770: 6532 2073 6571 7565 6e63 6533 202d 6620  e2 sequence3 -f 
-0000a780: 7365 7175 656e 6365 310a 6060 600a 0a2a  sequence1.```..*
-0000a790: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-0000a7a0: 206f 6620 676c 6f62 616c 204d 5242 4873   of global MRBHs
-0000a7b0: 2a2a 0a60 6060 0a77 6764 2064 6d64 2073  **.```.wgd dmd s
-0000a7c0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000a7d0: 6532 2073 6571 7565 6e63 6533 202d 676d  e2 sequence3 -gm
-0000a7e0: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-0000a7f0: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-0000a800: 6e20 3429 2028 2d6b 6629 2028 2d6b 6429  n 4) (-kf) (-kd)
-0000a810: 2028 2d63 2030 2e39 2920 282d 2d74 6f5f   (-c 0.9) (--to_
-0000a820: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
-0000a830: 6f20 7767 645f 646d 6429 2028 2d74 2077  o wgd_dmd) (-t w
-0000a840: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
-0000a850: 0a54 6865 2067 6c6f 6261 6c20 4d52 4248  .The global MRBH
-0000a860: 7320 6973 2063 6f6e 7374 7275 6374 6564  s is constructed
-0000a870: 2062 7920 6578 6861 7573 7469 7665 6c79   by exhaustively
-0000a880: 206d 6572 6769 6e67 2061 6c6c 2074 6865   merging all the
-0000a890: 2070 6f73 7369 626c 6520 7061 6972 2d77   possible pair-w
-0000a8a0: 6973 6520 5242 4873 2065 7863 6570 7420  ise RBHs except 
-0000a8b0: 666f 7220 7175 6572 7969 6e67 2074 6865  for querying the
-0000a8c0: 2073 6571 7565 6e63 6520 6974 7365 6c66   sequence itself
-0000a8d0: 2c20 7768 6963 6820 6361 6e20 6265 2069  , which can be i
-0000a8e0: 6e69 7469 6174 6564 2062 7920 6164 6420  nitiated by add 
-0000a8f0: 7468 6520 666c 6167 2060 2d67 6d60 206f  the flag `-gm` o
-0000a900: 7220 602d 2d67 6c6f 6261 6c6d 7262 6860  r `--globalmrbh`
-0000a910: 2e20 5468 6520 7265 7374 206f 6620 7265  . The rest of re
-0000a920: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
-0000a930: 7320 7374 6179 7320 7468 6520 7361 6d65  s stays the same
-0000a940: 2061 7320 7468 6520 6c6f 6361 6c20 4d52   as the local MR
-0000a950: 4248 732e 2054 6865 206e 756d 6265 7220  BHs. The number 
-0000a960: 6f66 2070 6172 616c 6c65 6c20 7468 7265  of parallel thre
-0000a970: 6164 7320 6973 2073 7567 6765 7374 6564  ads is suggested
-0000a980: 2074 6f20 6265 2073 6574 2061 7320 284e   to be set as (N
-0000a990: 2d31 294e 2f32 2074 6f6f 2077 6865 7265  -1)N/2 too where
-0000a9a0: 204e 2069 7320 7468 6520 6e75 6d62 6572   N is the number
-0000a9b0: 206f 6620 6364 7320 6669 6c65 7320 746f   of cds files to
-0000a9c0: 2061 6368 6965 7665 2074 6865 2068 6967   achieve the hig
-0000a9d0: 6865 7374 2065 6666 6963 6965 6e63 792e  hest efficiency.
-0000a9e0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-0000a9f0: 7374 6172 7469 6e67 2072 756e 2069 7320  starting run is 
-0000aa00: 756e 6465 7220 7468 6520 6465 6661 756c  under the defaul
-0000aa10: 7420 7061 7261 6d65 7465 7220 7769 7468  t parameter with
-0000aa20: 2074 6865 2063 6f6d 6d61 6e64 2073 686f   the command sho
-0000aa30: 776e 2061 7320 6265 6c6f 772e 2a2a 0a0a  wn as below.**..
-0000aa40: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000aa50: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000aa60: 7365 7175 656e 6365 3320 2d67 6d0a 6060  sequence3 -gm.``
-0000aa70: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
-0000aa80: 7469 6f6e 206f 6620 6f72 7468 6f67 726f  tion of orthogro
-0000aa90: 7570 732a 2a0a 6060 600a 7767 6420 646d  ups**.```.wgd dm
-0000aaa0: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
-0000aab0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000aac0: 2d6f 6920 2d6f 6f20 2d65 2031 652d 3130  -oi -oo -e 1e-10
-0000aad0: 202d 6273 2031 3030 202d 6e70 2035 2028   -bs 100 -np 5 (
-0000aae0: 2d6e 6e29 2028 2d63 6329 2028 2d74 6529  -nn) (-cc) (-te)
-0000aaf0: 2028 2d6d 6320 302e 3829 2028 2d67 6e29   (-mc 0.8) (-gn)
-0000ab00: 2028 2d74 7265 6520 2766 6173 7474 7265   (-tree 'fasttre
-0000ab10: 6527 2920 282d 7473 2027 2d66 6173 7465  e') (-ts '-faste
-0000ab20: 7374 2729 2028 2d6e 2034 2920 282d 2d74  st') (-n 4) (--t
-0000ab30: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000ab40: 282d 6f20 7767 645f 646d 6429 2028 2d74  (-o wgd_dmd) (-t
-0000ab50: 2077 6f72 6b69 6e67 5f74 6d70 290a 6060   working_tmp).``
-0000ab60: 600a 0a49 6e20 6077 6764 2076 3260 2c20  `..In `wgd v2`, 
-0000ab70: 7765 2061 6c73 6f20 696d 706c 656d 656e  we also implemen
-0000ab80: 7465 6420 616e 2061 6c67 6f72 6974 686d  ted an algorithm
-0000ab90: 206f 6620 6465 6c69 6e65 6174 696e 6720   of delineating 
-0000aba0: 6f72 7468 6f67 726f 7570 732c 2077 6869  orthogroups, whi
-0000abb0: 6368 2063 616e 2062 6520 696e 6974 6961  ch can be initia
-0000abc0: 7465 6420 7769 7468 2074 6865 2070 6172  ted with the par
-0000abd0: 616d 6574 6572 2060 2d6f 6960 206f 7220  ameter `-oi` or 
-0000abe0: 602d 2d6f 7274 686f 696e 6665 7260 2e20  `--orthoinfer`. 
-0000abf0: 5477 6f20 7761 7973 206f 6620 6465 6c69  Two ways of deli
-0000ac00: 6e65 6174 696f 6e20 6361 6e20 6265 2063  neation can be c
-0000ac10: 686f 7365 6e2c 2074 6865 2063 6f6e 6361  hosen, the conca
-0000ac20: 7465 6e61 7469 6f6e 2077 6179 2028 7365  tenation way (se
-0000ac30: 7420 6279 2074 6865 2070 6172 616d 6574  t by the paramet
-0000ac40: 6572 2060 2d63 6360 206f 7220 602d 2d63  er `-cc` or `--c
-0000ac50: 6f6e 6361 7460 2920 6f72 2074 6865 206e  oncat`) or the n
-0000ac60: 6f6e 2d63 6f6e 6361 7465 6e61 7469 6f6e  on-concatenation
-0000ac70: 2028 6465 6661 756c 7429 2077 6179 2e20   (default) way. 
-0000ac80: 496e 2062 7269 6566 2c20 7468 6520 636f  In brief, the co
-0000ac90: 6e63 6174 656e 6174 696f 6e20 7761 7920  ncatenation way 
-0000aca0: 6f66 2064 656c 696e 6561 7469 6e67 206f  of delineating o
-0000acb0: 7274 686f 6772 6f75 7073 2073 7461 7274  rthogroups start
-0000acc0: 7320 7769 7468 2063 6f6e 6361 7465 6e61  s with concatena
-0000acd0: 7469 6e67 2061 6c6c 2074 6865 2073 6571  ting all the seq
-0000ace0: 7565 6e63 6573 2069 6e74 6f20 6120 7369  uences into a si
-0000acf0: 6e67 6c65 2073 6571 7565 6e63 6520 6669  ngle sequence fi
-0000ad00: 6c65 2061 6e64 2074 6865 6e20 696e 6665  le and then infe
-0000ad10: 7272 696e 6720 7468 6520 7768 6f6c 6520  rring the whole 
-0000ad20: 7061 7261 6e6f 6d65 206f 6620 7468 6973  paranome of this
-0000ad30: 2073 696e 676c 6520 7365 7175 656e 6365   single sequence
-0000ad40: 2066 696c 6520 7769 7468 2074 6865 2063   file with the c
-0000ad50: 6c75 7374 6572 696e 6720 7265 7375 6c74  lustering result
-0000ad60: 7320 6d61 7070 6564 2062 6163 6b20 746f  s mapped back to
-0000ad70: 2074 6865 2062 656c 6f6e 6769 6e67 2073   the belonging s
-0000ad80: 7065 6369 6573 2e20 5768 696c 6520 7468  pecies. While th
-0000ad90: 6520 6e6f 6e2d 636f 6e63 6174 656e 6174  e non-concatenat
-0000ada0: 696f 6e20 7761 7920 7374 6172 7473 2077  ion way starts w
-0000adb0: 6974 6820 7265 7370 6563 7469 7665 2070  ith respective p
-0000adc0: 6169 722d 7769 7365 2064 6961 6d6f 6e64  air-wise diamond
-0000add0: 2073 6561 7263 6820 2869 6e63 6c75 6469   search (includi
-0000ade0: 6e67 2071 7565 7279 696e 6720 7468 6520  ng querying the 
-0000adf0: 7361 6d65 2073 6571 7565 6e63 6520 6974  same sequence it
-0000ae00: 7365 6c66 2920 616e 6420 7468 656e 2061  self) and then a
-0000ae10: 6c6c 2074 6865 2073 6571 7565 6e63 6520  ll the sequence 
-0000ae20: 7369 6d69 6c61 7269 7479 2074 6162 6c65  similarity table
-0000ae30: 7320 7769 6c6c 2062 6520 636f 6e63 6174  s will be concat
-0000ae40: 656e 6174 6564 2061 6e64 2063 6c75 7374  enated and clust
-0000ae50: 6572 6564 2069 6e74 6f20 6f72 7468 6f67  ered into orthog
-0000ae60: 726f 7570 732e 2053 6f6d 6520 6f74 6865  roups. Some othe
-0000ae70: 7220 706f 7373 6962 6c79 2075 7365 6675  r possibly usefu
-0000ae80: 6c20 706f 7374 2d63 6c75 7374 6572 696e  l post-clusterin
-0000ae90: 6720 6675 6e63 7469 6f6e 7320 6361 6e20  g functions can 
-0000aea0: 6265 2069 6e69 7469 6174 6564 2c20 696e  be initiated, in
-0000aeb0: 636c 7564 696e 6720 7468 6520 7061 7261  cluding the para
-0000aec0: 6d65 7465 7220 602d 7465 6020 6f72 2060  meter `-te` or `
-0000aed0: 2d2d 7465 7374 736f 6760 2c20 7768 6963  --testsog`, whic
-0000aee0: 6820 6361 6e20 6265 2073 6574 2074 6f20  h can be set to 
-0000aef0: 7374 6172 7420 7468 6520 756e 6269 6173  start the unbias
-0000af00: 6564 2074 6573 7420 6f66 2073 696e 676c  ed test of singl
-0000af10: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-0000af20: 6c69 6573 2028 6e6f 7465 2074 6861 7420  lies (note that 
-0000af30: 7468 6973 2066 756e 6374 696f 6e20 6e65  this function ne
-0000af40: 6564 7320 6068 6d6d 6572 6020 2876 332e  eds `hmmer` (v3.
-0000af50: 3162 3229 2074 6f20 6265 2069 6e73 7461  1b2) to be insta
-0000af60: 6c6c 6564 2069 6e20 7468 6520 656e 7669  lled in the envi
-0000af70: 726f 6e6d 656e 7420 7061 7468 292c 2074  ronment path), t
-0000af80: 6865 2070 6172 616d 6574 6572 2060 2d6d  he parameter `-m
-0000af90: 6360 206f 7220 602d 2d6d 736f 6763 7574  c` or `--msogcut
-0000afa0: 602c 2072 616e 6769 6e67 2062 6574 7765  `, ranging betwe
-0000afb0: 656e 2030 2074 6f20 312c 2077 6869 6368  en 0 to 1, which
-0000afc0: 2063 616e 2062 6520 7365 7420 746f 2073   can be set to s
-0000afd0: 6561 7263 6820 7468 6520 736f 2d63 616c  earch the so-cal
-0000afe0: 6c65 6420 6d6f 7374 6c79 2073 696e 676c  led mostly singl
-0000aff0: 652d 636f 7079 2066 616d 696c 7920 7768  e-copy family wh
-0000b000: 6963 6820 6861 7320 6869 6768 6572 2074  ich has higher t
-0000b010: 6861 6e20 6365 7274 6169 6e20 6375 742d  han certain cut-
-0000b020: 6f66 6620 7065 7263 656e 7461 6765 206f  off percentage o
-0000b030: 6620 7370 6563 6965 7320 636f 7665 7261  f species covera
-0000b040: 6765 2c20 7468 6520 7061 7261 6d65 7465  ge, the paramete
-0000b050: 7220 602d 676e 6020 6f72 2060 2d2d 6765  r `-gn` or `--ge
-0000b060: 746e 736f 6760 2c20 7768 6963 6820 6361  tnsog`, which ca
-0000b070: 6e20 6265 2073 6574 2074 6f20 7365 6172  n be set to sear
-0000b080: 6368 2066 6f72 206e 6573 7465 6420 7369  ch for nested si
-0000b090: 6e67 6c65 2d63 6f70 7920 6765 6e65 2066  ngle-copy gene f
-0000b0a0: 616d 696c 6965 7320 284e 534f 4773 2920  amilies (NSOGs) 
-0000b0b0: 7768 6963 6820 6973 206f 7269 6769 6e61  which is origina
-0000b0c0: 6c6c 7920 6d75 6c74 6979 2d63 6f70 7920  lly multiy-copy 
-0000b0d0: 6275 7420 6861 7320 6120 286d 6f73 746c  but has a (mostl
-0000b0e0: 7929 2073 696e 676c 652d 636f 7079 2062  y) single-copy b
-0000b0f0: 7261 6e63 6820 2877 6869 6368 2072 6571  ranch (which req
-0000b100: 7569 7265 7320 7468 6520 6368 6f73 656e  uires the chosen
-0000b110: 2074 7265 652d 696e 6665 7265 6e63 6520   tree-inference 
-0000b120: 7072 6f67 7261 6d20 7365 7420 6279 2060  program set by `
-0000b130: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
-0000b140: 655f 6d65 7468 6f64 6020 746f 2062 6520  e_method` to be 
-0000b150: 7072 652d 696e 7374 616c 6c65 6420 696e  pre-installed in
-0000b160: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-0000b170: 2070 6174 6820 7769 7468 2074 6865 2070   path with the p
-0000b180: 6172 616d 6574 6572 7320 7365 7474 696e  arameters settin
-0000b190: 6720 666f 7220 6765 6e65 2074 7265 6520  g for gene tree 
-0000b1a0: 696e 6665 7265 6e63 6520 636f 6e74 726f  inference contro
-0000b1b0: 6c6c 6564 2062 7920 602d 7473 6020 6f72  lled by `-ts` or
-0000b1c0: 2060 2d2d 7472 6565 7365 7460 292e 2054   `--treeset`). T
-0000b1d0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000b1e0: 646d 6460 2077 6f75 6c64 2073 7469 6c6c  dmd` would still
-0000b1f0: 2063 6f6e 6475 6374 2074 6865 2052 4248   conduct the RBH
-0000b200: 7320 6361 6c63 756c 6174 696f 6e20 756e  s calculation un
-0000b210: 6c65 7373 2074 6865 2070 6172 616d 6574  less the paramet
-0000b220: 6572 2060 2d6f 6f60 206f 7220 602d 2d6f  er `-oo` or `--o
-0000b230: 6e6c 796f 7274 686f 6020 7761 7320 7365  nlyortho` was se
-0000b240: 742e 2049 6620 6f6e 6520 6f6e 6c79 2077  t. If one only w
-0000b250: 616e 7473 2074 6f20 696e 6665 7220 7468  ants to infer th
-0000b260: 6520 6f72 7468 6f67 726f 7570 732c 2069  e orthogroups, i
-0000b270: 7427 7320 7375 6767 6573 7465 6420 746f  t's suggested to
-0000b280: 2061 6464 2074 6865 2066 6c61 6720 602d   add the flag `-
-0000b290: 6f6f 6020 746f 206a 7573 7420 696d 706c  oo` to just impl
-0000b2a0: 656d 656e 7420 7468 6520 6f72 7468 6f67  ement the orthog
-0000b2b0: 726f 7570 7320 6465 6c69 6e65 6174 696f  roups delineatio
-0000b2c0: 6e20 616e 616c 7973 6973 2e20 5468 6520  n analysis. The 
-0000b2d0: 6e75 6d62 6572 206f 6620 7061 7261 6c6c  number of parall
-0000b2e0: 656c 2074 6872 6561 6473 2069 7320 7375  el threads is su
-0000b2f0: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000b300: 7420 6173 2028 4e2b 3129 4e2f 3220 7768  t as (N+1)N/2 wh
-0000b310: 6572 6520 4e20 6973 2074 6865 206e 756d  ere N is the num
-0000b320: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
-0000b330: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-0000b340: 6869 6768 6573 7420 6566 6669 6369 656e  highest efficien
-0000b350: 6379 2073 696e 6365 2074 6865 2073 656c  cy since the sel
-0000b360: 662d 636f 6d70 6172 6973 6f6e 2069 7320  f-comparison is 
-0000b370: 616c 736f 2069 6e63 6c75 6465 642e 0a0a  also included...
-0000b380: 2a2a 5468 6520 6465 6661 756c 7420 7365  **The default se
-0000b390: 7474 696e 6720 6f66 2070 6172 616d 6574  tting of paramet
-0000b3a0: 6572 7320 6973 2061 2072 6561 736f 6e61  ers is a reasona
-0000b3b0: 626c 6520 7374 6172 7469 6e67 2070 6f69  ble starting poi
-0000b3c0: 6e74 2077 6974 6820 7468 6520 636f 6d6d  nt with the comm
-0000b3d0: 616e 6420 6173 2062 656c 6f77 2e2a 2a0a  and as below.**.
-0000b3e0: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-0000b3f0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-0000b400: 2073 6571 7565 6e63 6533 202d 6f69 202d   sequence3 -oi -
-0000b410: 6f6f 0a60 6060 0a0a 2a2a 5468 6520 636f  oo.```..**The co
-0000b420: 6c6c 696e 6561 7220 636f 616c 6573 6365  llinear coalesce
-0000b430: 6e63 6520 696e 6665 7265 6e63 6520 6f66  nce inference of
-0000b440: 2070 6879 6c6f 6765 6e79 2a2a 0a60 6060   phylogeny**.```
-0000b450: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000b460: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000b470: 7565 6e63 6533 202d 6170 2061 7064 6174  uence3 -ap apdat
-0000b480: 6120 2d73 6d20 736d 6461 7461 202d 6c65  a -sm smdata -le
-0000b490: 206c 6564 6174 6120 2d67 7420 6774 6461   ledata -gt gtda
-0000b4a0: 7461 202d 636f 6320 282d 7472 6565 2027  ta -coc (-tree '
-0000b4b0: 6661 7374 7472 6565 2729 2028 2d74 7320  fasttree') (-ts 
-0000b4c0: 272d 6661 7374 6573 7427 2920 282d 6e20  '-fastest') (-n 
-0000b4d0: 3429 2028 2d2d 746f 5f73 746f 7029 2028  4) (--to_stop) (
-0000b4e0: 2d2d 6364 7329 2028 2d6f 2077 6764 5f64  --cds) (-o wgd_d
-0000b4f0: 6d64 2920 282d 7420 776f 726b 696e 675f  md) (-t working_
-0000b500: 746d 7029 0a60 6060 0a0a 4120 6e6f 7665  tmp).```..A nove
-0000b510: 6c20 7068 796c 6f67 656e 6574 6963 2069  l phylogenetic i
-0000b520: 6e66 6572 656e 6365 206d 6574 686f 6420  nference method 
-0000b530: 6e61 6d65 6420 2263 6f6c 6c69 6e65 6172  named "collinear
-0000b540: 2063 6f61 6c65 7363 656e 6365 2069 6e66   coalescence inf
-0000b550: 6572 656e 6365 2220 6973 2061 6c73 6f20  erence" is also 
-0000b560: 696d 706c 656d 656e 7465 6420 696e 2060  implemented in `
-0000b570: 7767 6420 7632 602e 2046 6f72 2074 6869  wgd v2`. For thi
-0000b580: 7320 616e 616c 7973 6973 2c20 7573 6572  s analysis, user
-0000b590: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-0000b5a0: 6520 7468 6520 616e 6368 6f72 2070 6f69  e the anchor poi
-0000b5b0: 6e74 7320 6669 6c65 2062 7920 602d 6170  nts file by `-ap
-0000b5c0: 6020 6f72 2060 2d2d 616e 6368 6f72 706f  ` or `--anchorpo
-0000b5d0: 696e 7473 602c 2074 6865 2063 6f6c 6c69  ints`, the colli
-0000b5e0: 6e65 6172 2073 6567 6d65 6e74 7320 6669  near segments fi
-0000b5f0: 6c65 2062 7920 602d 736d 6020 6f72 2060  le by `-sm` or `
-0000b600: 2d2d 7365 676d 656e 7473 602c 2074 6865  --segments`, the
-0000b610: 206c 6973 7473 6567 6d65 6e74 7320 6669   listsegments fi
-0000b620: 6c65 2062 7920 602d 6c65 6020 6f72 2060  le by `-le` or `
-0000b630: 2d2d 6c69 7374 656c 656d 656e 7473 602c  --listelements`,
-0000b640: 2061 6e64 2074 6865 2067 656e 6520 7461   and the gene ta
-0000b650: 626c 6520 6669 6c65 2062 7920 602d 6774  ble file by `-gt
-0000b660: 6020 6f72 2060 2d2d 6765 6e65 7461 626c  ` or `--genetabl
-0000b670: 6560 2c20 616c 6c20 6f66 2077 6869 6368  e`, all of which
-0000b680: 2063 616e 2062 6520 7072 6f64 7563 6564   can be produced
-0000b690: 2069 6e20 7468 6520 7072 6f67 7261 6d20   in the program 
-0000b6a0: 6077 6764 2073 796e 602e 2054 6865 2070  `wgd syn`. The p
-0000b6b0: 6172 616d 6574 6572 2060 2d63 6f63 6020  arameter `-coc` 
-0000b6c0: 6f72 2060 2d2d 636f 6c6c 696e 6561 7263  or `--collinearc
-0000b6d0: 6f61 6c65 7363 656e 6365 6020 6e65 6564  oalescence` need
-0000b6e0: 7320 746f 2062 6520 7365 7420 746f 2073  s to be set to s
-0000b6f0: 7461 7274 2074 6869 7320 616e 616c 7973  tart this analys
-0000b700: 6973 2e20 5468 6520 7472 6565 2d69 6e66  is. The tree-inf
-0000b710: 6572 656e 6365 2070 726f 6772 616d 2061  erence program a
-0000b720: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
-0000b730: 6420 7061 7261 6d65 7465 7273 2063 616e  d parameters can
-0000b740: 2062 6520 7365 7420 6a75 7374 2061 7320   be set just as 
-0000b750: 6162 6f76 6520 6279 2060 2d74 7265 6560  above by `-tree`
-0000b760: 206f 7220 602d 2d74 7265 655f 6d65 7468   or `--tree_meth
-0000b770: 6f64 6020 616e 6420 602d 7473 6020 6f72  od` and `-ts` or
-0000b780: 2060 2d2d 7472 6565 7365 7460 2e20 506c   `--treeset`. Pl
-0000b790: 6561 7365 2061 6c73 6f20 6d61 6b65 2073  ease also make s
-0000b7a0: 7572 6520 7468 6520 6368 6f73 656e 2074  ure the chosen t
-0000b7b0: 7265 652d 696e 6665 7265 6e63 6520 7072  ree-inference pr
-0000b7c0: 6f67 7261 6d20 6973 2069 6e73 7461 6c6c  ogram is install
-0000b7d0: 6564 2069 6e20 7468 6520 656e 7669 726f  ed in the enviro
-0000b7e0: 6e6d 656e 7420 7061 7468 2e20 5468 6520  nment path. The 
-0000b7f0: 7072 6f67 7261 6d20 6061 7374 7261 6c2d  program `astral-
-0000b800: 7072 6f60 2069 7320 7265 7175 6972 6564  pro` is required
-0000b810: 2074 6f20 6265 2069 6e73 7461 6c6c 6564   to be installed
-0000b820: 2069 6e20 7468 6520 656e 7669 726f 6e6d   in the environm
-0000b830: 656e 7420 7061 7468 2074 6f6f 2e20 4e6f  ent path too. No
-0000b840: 7465 2074 6861 7420 7468 6572 6520 7368  te that there sh
-0000b850: 6f75 6c64 2062 6520 6e6f 2064 7570 6c69  ould be no dupli
-0000b860: 6361 7465 6420 6765 6e65 2049 4473 2069  cated gene IDs i
-0000b870: 6e20 7468 6520 7365 7175 656e 6365 2066  n the sequence f
-0000b880: 696c 652e 2054 6865 2070 6172 616c 6c65  ile. The paralle
-0000b890: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
-0000b8a0: 7265 2074 6f20 6163 6365 6c65 7261 7465  re to accelerate
-0000b8b0: 2074 6865 2073 6571 7565 6e63 6520 616c   the sequence al
-0000b8c0: 6967 6e6d 656e 7420 616e 6420 6765 6e65  ignment and gene
-0000b8d0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000b8e0: 666f 7220 6561 6368 2067 656e 6520 6661  for each gene fa
-0000b8f0: 6d69 6c79 2061 6e64 2074 6875 7320 7375  mily and thus su
-0000b900: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000b910: 7420 6173 206d 7563 6820 6173 2074 6865  t as much as the
-0000b920: 206e 756d 6265 7220 6f66 2067 656e 6520   number of gene 
-0000b930: 6661 6d69 6c69 6573 2e0a 0a2a 2a41 2073  families...**A s
-0000b940: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
-0000b950: 6720 7275 6e20 6f66 2074 6869 7320 616e  g run of this an
-0000b960: 616c 7973 6973 2069 7320 7769 7468 2074  alysis is with t
-0000b970: 6865 2073 696d 706c 6520 636f 6d6d 616e  he simple comman
-0000b980: 6420 6265 6c6f 772e 2a2a 0a0a 6060 600a  d below.**..```.
-0000b990: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
-0000b9a0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000b9b0: 656e 6365 3320 2d61 7020 6170 6461 7461  ence3 -ap apdata
-0000b9c0: 202d 736d 2073 6d64 6174 6120 2d6c 6520   -sm smdata -le 
-0000b9d0: 6c65 6461 7461 202d 6774 2067 7464 6174  ledata -gt gtdat
-0000b9e0: 6120 2d63 6f63 0a60 6060 0a0a 2323 2320  a -coc.```..### 
-0000b9f0: 7767 6420 666f 6375 730a 0a2a 2a54 6865  wgd focus..**The
-0000ba00: 2063 6f6e 6361 7465 6e61 7469 6f6e 2d62   concatenation-b
-0000ba10: 6173 6564 2f63 6f61 6c65 7363 656e 6365  ased/coalescence
-0000ba20: 2d62 6173 6564 2070 6879 6c6f 6765 6e65  -based phylogene
-0000ba30: 7469 6320 696e 6665 7265 6e63 652a 2a0a  tic inference**.
-0000ba40: 6060 600a 7767 6420 666f 6375 7320 6661  ```.wgd focus fa
-0000ba50: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
-0000ba60: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000ba70: 6e63 6533 2028 2d2d 636f 6e63 6174 656e  nce3 (--concaten
-0000ba80: 6174 696f 6e29 2028 2d2d 636f 616c 6573  ation) (--coales
-0000ba90: 6365 6e63 6529 2028 2d74 7265 6520 2766  cence) (-tree 'f
-0000baa0: 6173 7474 7265 6527 2920 282d 7473 2027  asttree') (-ts '
-0000bab0: 2d66 6173 7465 7374 2729 2028 2d6e 2034  -fastest') (-n 4
-0000bac0: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
-0000bad0: 2d63 6473 2920 282d 6f20 7767 645f 666f  -cds) (-o wgd_fo
-0000bae0: 6375 7329 2028 2d74 2077 6f72 6b69 6e67  cus) (-t working
-0000baf0: 5f74 6d70 290a 6060 600a 0a54 6865 2070  _tmp).```..The p
-0000bb00: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
-0000bb10: 7360 2069 6d70 6c65 6d65 6e74 6564 2074  s` implemented t
-0000bb20: 776f 2062 6173 6963 2070 6879 6c6f 6765  wo basic phyloge
-0000bb30: 6e65 7469 6320 696e 6665 7265 6e63 6520  netic inference 
-0000bb40: 6d65 7468 6f64 732c 2069 2e65 2e2c 2063  methods, i.e., c
-0000bb50: 6f6e 6361 7465 6e61 7469 6f6e 2d62 6173  oncatenation-bas
-0000bb60: 6564 2061 6e64 2063 6f61 6c65 7363 656e  ed and coalescen
-0000bb70: 6365 2d62 6173 6564 206d 6574 686f 6473  ce-based methods
-0000bb80: 2e20 546f 2069 6e69 7469 6174 6520 7468  . To initiate th
-0000bb90: 6573 6520 616e 616c 7973 6973 2c20 7573  ese analysis, us
-0000bba0: 6572 7320 6e65 6564 2074 6f20 7365 7420  ers need to set 
-0000bbb0: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-0000bbc0: 602d 2d63 6f6e 6361 7465 6e61 7469 6f6e  `--concatenation
-0000bbd0: 6020 6f72 2060 2d2d 636f 616c 6573 6365  ` or `--coalesce
-0000bbe0: 6e63 6560 2e20 5468 6520 636f 6e63 6174  nce`. The concat
-0000bbf0: 656e 6174 696f 6e2d 6261 7365 6420 6d65  enation-based me
-0000bc00: 7468 6f64 2069 6e63 6c75 6465 7320 6120  thod includes a 
-0000bc10: 6665 7720 6d61 6a6f 7220 7374 6570 732c  few major steps,
-0000bc20: 2069 2e65 2e2c 2074 6865 206d 756c 7469   i.e., the multi
-0000bc30: 706c 6520 7365 7175 656e 6365 2061 6c69  ple sequence ali
-0000bc40: 676e 6d65 6e74 2028 4d53 4129 206f 6620  gnment (MSA) of 
-0000bc50: 6561 6368 2067 656e 6520 6661 6d69 6c79  each gene family
-0000bc60: 2c20 7468 6520 636f 6e63 6174 656e 6174  , the concatenat
-0000bc70: 696f 6e20 6f66 2061 6c6c 2067 656e 6520  ion of all gene 
-0000bc80: 6661 6d69 6c69 6573 2061 6e64 2074 6865  families and the
-0000bc90: 6e20 7468 6520 6765 6e65 2074 7265 6520  n the gene tree 
-0000bca0: 2861 6c73 6f20 7370 6563 6965 7320 7472  (also species tr
-0000bcb0: 6565 2069 6e20 7468 6973 2063 6173 6529  ee in this case)
-0000bcc0: 2069 6e66 6572 656e 6365 2e20 5468 6520   inference. The 
-0000bcd0: 636f 616c 6573 6365 6e63 652d 6261 7365  coalescence-base
-0000bce0: 6420 6d65 7468 6f64 2077 696c 6c20 696e  d method will in
-0000bcf0: 7374 6561 6420 7065 7266 6f72 6d20 7468  stead perform th
-0000bd00: 6520 4d53 4120 6f66 2065 6163 6820 6765  e MSA of each ge
-0000bd10: 6e65 2066 616d 696c 7920 616e 6420 7468  ne family and th
-0000bd20: 6520 6765 6e65 2074 7265 6520 696e 6665  e gene tree infe
-0000bd30: 7265 6e63 6520 6261 7365 6420 6f6e 2065  rence based on e
-0000bd40: 6163 6820 4d53 412c 2061 6e64 2074 6865  ach MSA, and the
-0000bd50: 6e20 696e 6665 7220 7468 6520 7370 6563  n infer the spec
-0000bd60: 6965 7320 7472 6565 2062 6173 6564 206f  ies tree based o
-0000bd70: 6e20 7468 6573 6520 696e 6469 7669 6475  n these individu
-0000bd80: 616c 2067 656e 6520 7472 6565 732e 2054  al gene trees. T
-0000bd90: 6865 2074 7265 652d 696e 6665 7265 6e63  he tree-inferenc
-0000bda0: 6520 7072 6f67 7261 6d20 616e 6420 7468  e program and th
-0000bdb0: 6520 6173 736f 6369 6174 6564 2070 6172  e associated par
-0000bdc0: 616d 6574 6572 7320 6361 6e20 6265 2073  ameters can be s
-0000bdd0: 6574 206a 7573 7420 6173 2061 626f 7665  et just as above
-0000bde0: 2062 7920 602d 7472 6565 6020 6f72 2060   by `-tree` or `
-0000bdf0: 2d2d 7472 6565 5f6d 6574 686f 6460 2061  --tree_method` a
-0000be00: 6e64 2060 2d74 7360 206f 7220 602d 2d74  nd `-ts` or `--t
-0000be10: 7265 6573 6574 602e 2050 6c65 6173 6520  reeset`. Please 
-0000be20: 616c 736f 206d 616b 6520 7375 7265 2074  also make sure t
-0000be30: 6865 2063 686f 7365 6e20 7472 6565 2d69  he chosen tree-i
-0000be40: 6e66 6572 656e 6365 2070 726f 6772 616d  nference program
-0000be50: 2069 7320 696e 7374 616c 6c65 6420 696e   is installed in
-0000be60: 2074 6865 2065 6e76 6972 6f6e 6d65 6e74   the environment
-0000be70: 2070 6174 682e 2054 6865 2070 726f 6772   path. The progr
-0000be80: 616d 2060 6173 7472 616c 2d70 726f 6020  am `astral-pro` 
-0000be90: 6973 2072 6571 7569 7265 6420 746f 2062  is required to b
-0000bea0: 6520 696e 7374 616c 6c65 6420 696e 2074  e installed in t
-0000beb0: 6865 2065 6e76 6972 6f6e 6d65 6e74 2070  he environment p
-0000bec0: 6174 6820 6966 2074 6865 2063 6f61 6c65  ath if the coale
-0000bed0: 7363 656e 6365 2d62 6173 6564 206d 6574  scence-based met
-0000bee0: 686f 6420 6973 2063 686f 7365 6e2e 2054  hod is chosen. T
-0000bef0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000bf00: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000bf10: 6163 6365 6c65 7261 7465 2074 6865 2073  accelerate the s
-0000bf20: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
-0000bf30: 7420 616e 6420 6765 6e65 2074 7265 6520  t and gene tree 
-0000bf40: 696e 6665 7265 6e63 6520 666f 7220 6561  inference for ea
-0000bf50: 6368 2067 656e 6520 6661 6d69 6c79 2074  ch gene family t
-0000bf60: 6f6f 2061 6e64 2074 6875 7320 7375 6767  oo and thus sugg
-0000bf70: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
-0000bf80: 6173 206d 7563 6820 6173 2074 6865 206e  as much as the n
-0000bf90: 756d 6265 7220 6f66 2067 656e 6520 6661  umber of gene fa
-0000bfa0: 6d69 6c69 6573 2e0a 0a2a 2a41 2073 7567  milies...**A sug
-0000bfb0: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
-0000bfc0: 7275 6e20 6f66 2074 6869 7320 616e 616c  run of this anal
-0000bfd0: 7973 6973 2069 7320 7769 7468 2074 6865  ysis is with the
-0000bfe0: 2073 696d 706c 6520 636f 6d6d 616e 6420   simple command 
-0000bff0: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000c000: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000c010: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000c020: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000c030: 2d63 6f6e 6361 7465 6e61 7469 6f6e 0a77  -concatenation.w
-0000c040: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
-0000c050: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000c060: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000c070: 2d2d 636f 616c 6573 6365 6e63 650a 6060  --coalescence.``
-0000c080: 600a 0a2a 2a54 6865 2066 756e 6374 696f  `..**The functio
-0000c090: 6e61 6c20 616e 6e6f 7461 7469 6f6e 206f  nal annotation o
-0000c0a0: 6620 6765 6e65 2066 616d 696c 6965 732a  f gene families*
-0000c0b0: 2a0a 6060 600a 7767 6420 666f 6375 7320  *.```.wgd focus 
-0000c0c0: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-0000c0d0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000c0e0: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
-0000c0f0: 696f 6e20 6567 676e 6f67 202d 6564 2065  ion eggnog -ed e
-0000c100: 6464 6174 6120 2d2d 646d 6e62 2064 6264  ddata --dmnb dbd
-0000c110: 6174 610a 6060 600a 0a54 6865 2070 726f  ata.```..The pro
-0000c120: 6772 616d 2060 7767 6420 666f 6375 7360  gram `wgd focus`
-0000c130: 2061 6c73 6f20 6164 6465 6420 736f 6d65   also added some
-0000c140: 2077 7261 7070 696e 6720 6675 6e63 7469   wrapping functi
-0000c150: 6f6e 7320 666f 7220 6675 6e63 7469 6f6e  ons for function
-0000c160: 616c 2061 6e6e 6f74 6174 696f 6e20 6f66  al annotation of
-0000c170: 2067 656e 6520 6661 6d69 6c69 6573 206f   gene families o
-0000c180: 6e20 7468 6520 686f 6f64 206f 6620 6461  n the hood of da
-0000c190: 7461 6261 7365 7320 616e 6420 736f 6674  tabases and soft
-0000c1a0: 7761 7265 7320 666f 7220 696e 7374 616e  wares for instan
-0000c1b0: 6365 2060 4567 674e 4f47 6020 616e 6420  ce `EggNOG` and 
-0000c1c0: 6045 6767 4e4f 472d 6d61 7070 6572 602e  `EggNOG-mapper`.
-0000c1d0: 2046 6f72 2074 6865 2061 6e6e 6f74 6174   For the annotat
-0000c1e0: 696f 6e20 7573 696e 6720 6045 6767 4e4f  ion using `EggNO
-0000c1f0: 472d 6d61 7070 6572 602c 2075 7365 7273  G-mapper`, users
-0000c200: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
-0000c210: 2074 6865 2070 6174 6820 746f 2074 6865   the path to the
-0000c220: 2065 6767 4e4f 4720 616e 6e6f 7461 7469   eggNOG annotati
-0000c230: 6f6e 2064 6174 6162 6173 6520 7669 6120  on database via 
-0000c240: 7468 6520 6f70 7469 6f6e 2060 2d65 6460  the option `-ed`
-0000c250: 2c20 7468 6520 7061 7468 2074 6f20 7468  , the path to th
-0000c260: 6520 6469 616d 6f6e 642d 636f 6d70 6174  e diamond-compat
-0000c270: 6962 6c65 2064 6174 6162 6173 6520 7669  ible database vi
-0000c280: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
-0000c290: 646d 6e64 5f64 6260 2061 6e64 2074 6865  dmnd_db` and the
-0000c2a0: 206f 7074 696f 6e20 602d 2d61 6e6e 6f74   option `--annot
-0000c2b0: 6174 696f 6e60 2073 6574 2061 7320 2265  ation` set as "e
-0000c2c0: 6767 6e6f 6722 2e20 5468 6520 6d61 6e6e  ggnog". The mann
-0000c2d0: 6572 2068 6f77 2050 4641 4d20 616e 6e6f  er how PFAM anno
-0000c2e0: 7461 7469 6f6e 2077 696c 6c20 6265 2070  tation will be p
-0000c2f0: 6572 666f 726d 6564 2063 616e 2062 6520  erformed can be 
-0000c300: 636f 6e74 726f 6c6c 6564 2076 6961 2074  controlled via t
-0000c310: 6865 206f 7074 696f 6e20 6070 6661 6d60  he option `pfam`
-0000c320: 2c20 6569 7468 6572 2022 6e6f 6e65 222c  , either "none",
-0000c330: 2022 7265 616c 6967 6e22 206f 7220 2264   "realign" or "d
-0000c340: 656e 6f76 6f22 2c20 7468 6520 6465 7461  enovo", the deta
-0000c350: 696c 6564 2065 7870 6c61 6e61 7469 6f6e  iled explanation
-0000c360: 2063 616e 2062 6520 666f 756e 6420 6174   can be found at
-0000c370: 2074 6865 2077 696b 6920 6f66 205b 4567   the wiki of [Eg
-0000c380: 674e 4f47 2d6d 6170 7065 725d 2868 7474  gNOG-mapper](htt
-0000c390: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-0000c3a0: 6567 676e 6f67 6462 2f65 6767 6e6f 672d  eggnogdb/eggnog-
-0000c3b0: 6d61 7070 6572 2f77 696b 6929 2e20 506c  mapper/wiki). Pl
-0000c3c0: 6561 7365 2070 7265 2d69 6e73 7461 6c6c  ease pre-install
-0000c3d0: 2074 6865 2060 4567 674e 4f47 2d6d 6170   the `EggNOG-map
-0000c3e0: 7065 7260 2070 7974 686f 6e20 7061 636b  per` python pack
-0000c3f0: 6167 6520 6966 2075 7369 6e67 2074 6869  age if using thi
-0000c400: 7320 6675 6e63 7469 6f6e 2e20 466f 7220  s function. For 
-0000c410: 7468 6520 616e 6e6f 7461 7469 6f6e 2075  the annotation u
-0000c420: 7369 6e67 2060 686d 6d73 6361 6e60 2c20  sing `hmmscan`, 
-0000c430: 7768 6174 2069 7320 696d 706c 656d 656e  what is implemen
-0000c440: 7465 6420 696e 2060 7767 6420 7632 6020  ted in `wgd v2` 
-0000c450: 6973 2061 2073 696d 706c 6520 6275 6e64  is a simple bund
-0000c460: 6c65 2066 756e 6374 696f 6e20 746f 2070  le function to p
-0000c470: 6572 666f 726d 2060 686d 6d73 6361 6e60  erform `hmmscan`
-0000c480: 2061 6e61 6c79 7369 7320 666f 7220 6120   analysis for a 
-0000c490: 6769 7665 6e20 686d 6d20 7072 6f66 696c  given hmm profil
-0000c4a0: 6520 616e 6420 7365 7420 6f66 2067 656e  e and set of gen
-0000c4b0: 6520 6661 6d69 6c69 6573 2073 7563 6820  e families such 
-0000c4c0: 7468 6174 2075 7365 7273 206e 6565 6420  that users need 
-0000c4d0: 746f 2073 6574 2074 6865 206f 7074 696f  to set the optio
-0000c4e0: 6e20 602d 2d61 6e6e 6f74 6174 696f 6e60  n `--annotation`
-0000c4f0: 2061 7320 2268 6d6d 7066 616d 2220 616e   as "hmmpfam" an
-0000c500: 6420 7072 6f76 6964 6520 7468 6520 686d  d provide the hm
-0000c510: 6d70 726f 6669 6c65 2076 6961 2074 6865  mprofile via the
-0000c520: 206f 7074 696f 6e20 602d 2d68 6d6d 602e   option `--hmm`.
-0000c530: 2046 6f72 2074 6865 2061 6e6e 6f74 6174   For the annotat
-0000c540: 696f 6e20 7573 696e 6720 6069 6e74 6572  ion using `inter
-0000c550: 7072 6f73 6361 6e60 2c20 7573 6572 7320  proscan`, users 
-0000c560: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-0000c570: 7468 6520 7061 7468 2074 6f20 7468 6520  the path to the 
-0000c580: 696e 7465 7270 726f 7363 616e 2069 6e73  interproscan ins
-0000c590: 7461 6c6c 6174 696f 6e20 666f 6c64 6572  tallation folder
-0000c5a0: 2077 6865 7265 2074 6865 7265 2069 7320   where there is 
-0000c5b0: 6120 6069 6e74 6572 7072 6f73 6361 6e2e  a `interproscan.
-0000c5c0: 7368 6020 6669 6c65 2076 6961 2074 6865  sh` file via the
-0000c5d0: 206f 7074 696f 6e20 602d 2d65 7865 7061   option `--exepa
-0000c5e0: 7468 6020 616e 6420 7365 7420 7468 6520  th` and set the 
-0000c5f0: 7468 6520 6f70 7469 6f6e 2060 2d2d 616e  the option `--an
-0000c600: 6e6f 7461 7469 6f6e 6020 6173 2022 696e  notation` as "in
-0000c610: 7465 7270 726f 7363 616e 222e 2054 6865  terproscan". The
-0000c620: 2070 6172 616c 6c65 6c20 7468 7265 6164   parallel thread
-0000c630: 7320 6865 7265 2061 7265 2074 6f20 7061  s here are to pa
-0000c640: 7261 6c6c 656c 697a 6520 7468 6520 616e  rallelize the an
-0000c650: 616c 7973 6973 2066 6f72 2065 6163 6820  alysis for each 
-0000c660: 6765 6e65 2066 616d 696c 7920 616e 6420  gene family and 
-0000c670: 7468 7573 2073 7567 6765 7374 6564 2074  thus suggested t
-0000c680: 6f20 6265 2073 6574 2061 7320 6d75 6368  o be set as much
-0000c690: 2061 7320 7468 6520 6e75 6d62 6572 206f   as the number o
-0000c6a0: 6620 6765 6e65 2066 616d 696c 6965 732e  f gene families.
-0000c6b0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-0000c6c0: 7374 6172 7469 6e67 2072 756e 206f 6620  starting run of 
-0000c6d0: 7468 6973 2061 6e61 6c79 7369 7320 6361  this analysis ca
-0000c6e0: 6e20 6265 2077 6974 6820 7468 6520 636f  n be with the co
-0000c6f0: 6d6d 616e 6420 6265 6c6f 772e 2a2a 0a0a  mmand below.**..
-0000c700: 6060 600a 7767 6420 666f 6375 7320 6661  ```.wgd focus fa
-0000c710: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
-0000c720: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000c730: 6e63 6533 202d 2d61 6e6e 6f74 6174 696f  nce3 --annotatio
-0000c740: 6e20 6567 676e 6f67 202d 6564 2065 6464  n eggnog -ed edd
-0000c750: 6174 6120 2d2d 646d 6e62 2064 6264 6174  ata --dmnb dbdat
-0000c760: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
-0000c770: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
-0000c780: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000c790: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
-0000c7a0: 686d 6d70 6661 6d20 2d2d 686d 6d20 686d  hmmpfam --hmm hm
-0000c7b0: 6d64 6174 610a 7767 6420 666f 6375 7320  mdata.wgd focus 
-0000c7c0: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
-0000c7d0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000c7e0: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
-0000c7f0: 696f 6e20 696e 7465 7270 726f 7363 616e  ion interproscan
-0000c800: 202d 2d65 7865 7061 7468 2024 5041 5448   --exepath $PATH
-0000c810: 0a60 6060 0a0a 2a2a 5468 6520 7068 796c  .```..**The phyl
-0000c820: 6f67 656e 6574 6963 2064 6174 696e 6720  ogenetic dating 
-0000c830: 6f66 2057 4744 732a 2a0a 6060 600a 7767  of WGDs**.```.wg
-0000c840: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000c850: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000c860: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000c870: 6420 6d63 6d63 7472 6565 202d 7370 2073  d mcmctree -sp s
-0000c880: 7064 6174 6120 282d 2d70 726f 7463 6f63  pdata (--protcoc
-0000c890: 6461 7469 6e67 2920 282d 2d70 6172 7469  dating) (--parti
-0000c8a0: 7469 6f6e 2920 282d 2d61 616d 6f64 656c  tion) (--aamodel
-0000c8b0: 206c 6729 2028 2d64 7320 2762 7572 6e69   lg) (-ds 'burni
-0000c8c0: 6e20 3d20 3230 3030 2729 2028 2d64 7320  n = 2000') (-ds 
-0000c8d0: 2773 616d 7066 7265 7120 3d20 3130 3030  'sampfreq = 1000
-0000c8e0: 2729 2028 2d64 7320 276e 7361 6d70 6c65  ') (-ds 'nsample
-0000c8f0: 203d 2032 3030 3030 2729 2020 282d 6e20   = 20000')  (-n 
-0000c900: 3429 2028 2d2d 746f 5f73 746f 7029 2028  4) (--to_stop) (
-0000c910: 2d2d 6364 7329 2028 2d6f 2077 6764 5f66  --cds) (-o wgd_f
-0000c920: 6f63 7573 2920 282d 7420 776f 726b 696e  ocus) (-t workin
-0000c930: 675f 746d 7029 0a60 6060 0a0a 5468 6520  g_tmp).```..The 
-0000c940: 6162 736f 6c75 7465 2064 6174 696e 6720  absolute dating 
-0000c950: 6f66 2057 4744 7320 6973 2061 2073 7065  of WGDs is a spe
-0000c960: 6369 6669 6320 7069 7065 6c69 6e65 2069  cific pipeline i
-0000c970: 6d70 6c65 6d65 6e74 6564 2069 6e20 6077  mplemented in `w
-0000c980: 6764 2076 3260 2075 7369 6e67 2074 6865  gd v2` using the
-0000c990: 206d 6574 686f 6420 6f66 2070 6879 6c6f   method of phylo
-0000c9a0: 6765 6e65 7469 6320 6461 7469 6e67 2e20  genetic dating. 
-0000c9b0: 5468 6520 6661 6d69 6c69 6573 2075 7365  The families use
-0000c9c0: 6420 696e 2074 6869 7320 7374 6570 2063  d in this step c
-0000c9d0: 616e 2062 6520 7072 6f64 7563 6564 2066  an be produced f
-0000c9e0: 726f 6d20 6077 6764 2064 6d64 6020 616e  rom `wgd dmd` an
-0000c9f0: 6420 6077 6764 2070 6561 6b60 2e20 4e6f  d `wgd peak`. No
-0000ca00: 7465 2074 6861 7420 6865 7265 2077 6520  te that here we 
-0000ca10: 6f6e 6c79 2064 6973 6375 7373 2068 6f77  only discuss how
-0000ca20: 2074 6f20 6461 7465 2057 4744 7320 7769   to date WGDs wi
-0000ca30: 7468 2067 656e 6f6d 6520 6173 7365 6d62  th genome assemb
-0000ca40: 6c79 2c20 696e 7374 6561 6420 6f66 2074  ly, instead of t
-0000ca50: 7261 6e73 6372 6970 746f 6d65 2061 7373  ranscriptome ass
-0000ca60: 656d 626c 7920 2877 6869 6368 2077 696c  embly (which wil
-0000ca70: 6c20 6265 2064 6973 6375 7373 6564 2069  l be discussed i
-0000ca80: 6e20 6120 7365 7061 7261 7465 2073 6563  n a separate sec
-0000ca90: 7469 6f6e 2068 6572 6575 6e64 6572 292e  tion hereunder).
-0000caa0: 2054 6865 2061 7373 756d 7074 696f 6e20   The assumption 
-0000cab0: 7765 206d 6164 6520 6865 7265 2069 7320  we made here is 
-0000cac0: 7468 6174 206e 6f74 2061 6c6c 2061 6e63  that not all anc
-0000cad0: 686f 7220 7061 6972 7320 2863 6f6c 6c69  hor pairs (colli
-0000cae0: 6e65 6172 2064 7570 6c69 6361 7465 7329  near duplicates)
-0000caf0: 2061 7265 2073 7569 7461 626c 6520 666f   are suitable fo
-0000cb00: 7220 7068 796c 6f67 656e 6574 6963 2064  r phylogenetic d
-0000cb10: 6174 696e 672c 2066 6f72 2069 6e73 7461  ating, for insta
-0000cb20: 6e63 6520 7468 6f73 6520 6661 7374 6c79  nce those fastly
-0000cb30: 206f 7220 736c 6f77 6c79 2065 766f 6c76   or slowly evolv
-0000cb40: 696e 6720 6765 6e65 2064 7570 6c69 6361  ing gene duplica
-0000cb50: 7465 732c 2062 6563 6175 7365 2074 6865  tes, because the
-0000cb60: 7927 7265 2070 726f 6e65 2074 6f20 6769  y're prone to gi
-0000cb70: 7665 2062 6961 7365 6420 6461 7469 6e67  ve biased dating
-0000cb80: 2065 7374 696d 6174 696f 6e2e 2053 6f20   estimation. So 
-0000cb90: 6173 2074 6f20 7265 7472 6965 7665 2074  as to retrieve t
-0000cba0: 6865 2022 7265 6c69 6162 6c65 2220 616e  he "reliable" an
-0000cbb0: 6368 6f72 2070 6169 7273 2c20 7765 2069  chor pairs, we i
-0000cbc0: 6d70 6c65 6d65 6e74 6564 2073 6f6d 6520  mplemented some 
-0000cbd0: 6d65 7468 6f64 7320 6f66 2069 6465 6e74  methods of ident
-0000cbe0: 6966 7969 6e67 2063 7265 6469 6162 6c65  ifying crediable
-0000cbf0: 2061 6e63 686f 7220 7061 6972 7320 6261   anchor pairs ba
-0000cc00: 7365 6420 6f6e 2074 6865 6972 204b 7320  sed on their Ks 
-0000cc10: 7661 6c75 6573 2061 6e64 2f6f 7220 7265  values and/or re
-0000cc20: 7369 6469 6e67 2063 6f6c 6c69 6e65 6172  siding collinear
-0000cc30: 2073 6567 6d65 6e74 732e 2046 6f72 2061   segments. For a
-0000cc40: 2067 656e 6f6d 6520 7769 7468 2063 6c65   genome with cle
-0000cc50: 6172 2073 6967 6e61 6c73 206f 6620 7075  ar signals of pu
-0000cc60: 7461 7469 7665 2057 4744 732c 2073 7563  tative WGDs, suc
-0000cc70: 6820 6173 2074 6865 202a 4171 7569 6c65  h as the *Aquile
-0000cc80: 6769 6120 636f 6572 756c 6561 2a20 696e  gia coerulea* in
-0000cc90: 2074 6865 2065 7861 6d70 6c65 2c20 6120   the example, a 
-0000cca0: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-0000ccb0: 2074 6861 7420 6170 706c 6965 7320 7468   that applies th
-0000ccc0: 6520 7072 696e 6369 706c 6520 6f66 2068  e principle of h
-0000ccd0: 6f77 205b 6b73 7261 7465 735d 2868 7474  ow [ksrates](htt
-0000cce0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-0000ccf0: 5649 422d 5053 422f 6b73 7261 7465 7329  VIB-PSB/ksrates)
-0000cd00: 2066 696e 6420 7468 6520 696e 6974 6961   find the initia
-0000cd10: 6c20 7065 616b 7320 616e 6420 7468 6569  l peaks and thei
-0000cd20: 7220 7061 7261 6d65 7465 7273 2077 6173  r parameters was
-0000cd30: 2069 6d70 6c65 6d65 6e74 6564 2074 6f20   implemented to 
-0000cd40: 6669 6e64 2074 6865 2039 3525 2063 6f6e  find the 95% con
-0000cd50: 6669 6465 6e63 6520 6c65 7665 6c20 6f66  fidence level of
-0000cd60: 2074 6865 2061 7373 756d 6564 206c 6f67   the assumed log
-0000cd70: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-0000cd80: 696f 6e20 6f66 2074 6865 2061 6e63 686f  ion of the ancho
-0000cd90: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-0000cda0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000cdb0: 6962 7574 696f 6e20 746f 2066 696c 7465  ibution to filte
-0000cdc0: 7220 616e 6368 6f72 2070 6169 7273 2077  r anchor pairs w
-0000cdd0: 6974 6820 746f 6f20 6869 6768 206f 7220  ith too high or 
-0000cde0: 6c6f 7720 2a4b 2a3c 7375 623e 533c 2f73  low *K*<sub>S</s
-0000cdf0: 7562 3e20 6167 6573 2c20 7468 6520 6578  ub> ages, the ex
-0000ce00: 616d 706c 6172 2063 6f6d 6d61 6e64 206f  amplar command o
-0000ce10: 6620 7768 6963 6820 6973 2073 686f 7765  f which is showe
-0000ce20: 6420 696e 2074 6865 2060 7767 6420 7065  d in the `wgd pe
-0000ce30: 616b 6020 7365 6374 696f 6e20 6f66 205b  ak` section of [
-0000ce40: 496c 6c75 7374 7261 7469 6f6e 5d28 6874  Illustration](ht
-0000ce50: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000ce60: 2f68 6563 6865 2d70 7362 2f77 6764 3f74  /heche-psb/wgd?t
-0000ce70: 6162 3d72 6561 646d 652d 6f76 2d66 696c  ab=readme-ov-fil
-0000ce80: 6523 696c 6c75 7374 7261 7469 6f6e 292e  e#illustration).
-0000ce90: 2049 6620 7468 6973 2068 6575 7269 7374   If this heurist
-0000cea0: 6963 206d 6574 686f 6420 6661 696c 6564  ic method failed
-0000ceb0: 2074 6f20 6769 7665 2072 6561 736f 6e61   to give reasona
-0000cec0: 626c 6520 7265 7375 6c74 732c 2077 6869  ble results, whi
-0000ced0: 6368 2069 7320 7573 7561 6c6c 7920 6475  ch is usually du
-0000cee0: 6520 746f 2074 6865 2065 6666 6563 7420  e to the effect 
-0000cef0: 6f66 206d 756c 7469 706c 6520 6164 6a61  of multiple adja
-0000cf00: 6365 6e74 2057 4744 7320 7468 6174 2062  cent WGDs that b
-0000cf10: 6c75 7273 2074 6865 2070 6561 6b2d 6669  lurs the peak-fi
-0000cf20: 6e64 696e 6720 7072 6f63 6573 732c 2075  nding process, u
-0000cf30: 7365 7273 2063 616e 2074 7572 6e20 746f  sers can turn to
-0000cf40: 2074 6865 2063 6f6c 6c69 6e65 6172 2073   the collinear s
-0000cf50: 6567 6d65 6e74 732d 6775 6964 6564 2061  egments-guided a
-0000cf60: 6e63 686f 7220 7061 6972 2063 6c75 7374  nchor pair clust
-0000cf70: 6572 696e 6720 696d 706c 656d 656e 7465  ering implemente
-0000cf80: 6420 696e 2060 7767 6420 7632 602c 2069  d in `wgd v2`, i
-0000cf90: 6e20 7768 6963 6820 6120 636f 6c6c 696e  n which a collin
-0000cfa0: 6561 7220 7365 676d 656e 742d 7769 7365  ear segment-wise
-0000cfb0: 2047 4d4d 2063 6c75 7374 6572 696e 6720   GMM clustering 
-0000cfc0: 7769 6c6c 2062 6520 6669 7273 7420 636f  will be first co
-0000cfd0: 6e64 7563 7465 6420 6261 7365 6420 6f6e  nducted based on
-0000cfe0: 2074 6865 2022 736f 2d63 616c 6c65 6422   the "so-called"
-0000cff0: 2073 6567 6d65 6e74 202a 4b2a 3c73 7562   segment *K*<sub
-0000d000: 3e53 3c2f 7375 623e 2061 6765 2072 6570  >S</sub> age rep
-0000d010: 7265 7365 6e74 6564 2062 7920 7468 6520  resented by the 
-0000d020: 6d65 6469 616e 202a 4b2a 3c73 7562 3e53  median *K*<sub>S
-0000d030: 3c2f 7375 623e 2061 6765 206f 6620 616c  </sub> age of al
-0000d040: 6c20 7468 6520 7265 7369 6469 6e67 2073  l the residing s
-0000d050: 796e 7465 6c6f 6773 2028 6e6f 7465 2074  yntelogs (note t
-0000d060: 6861 7420 7468 6520 6765 6e65 2064 7570  hat the gene dup
-0000d070: 6c69 6361 7465 2070 6169 7273 2061 646f  licate pairs ado
-0000d080: 7074 6564 2069 6e20 7468 6973 2073 7465  pted in this ste
-0000d090: 7020 6973 2066 726f 6d20 7468 6520 6669  p is from the fi
-0000d0a0: 6c65 2060 6d75 6c74 6970 6c69 636f 6e5f  le `multiplicon_
-0000d0b0: 7061 6972 732e 7478 7460 2077 6869 6368  pairs.txt` which
-0000d0c0: 2063 6f6e 7461 696e 7320 7468 6520 6675   contains the fu
-0000d0d0: 6c6c 2073 6574 206f 6620 7379 6e74 656c  ll set of syntel
-0000d0e0: 6f67 7329 2c20 696e 7374 6561 6420 6f66  ogs), instead of
-0000d0f0: 2074 6865 2073 6d61 6c6c 6572 2067 656e   the smaller gen
-0000d100: 6520 7365 7420 6f66 2061 6e63 686f 7220  e set of anchor 
-0000d110: 7061 6972 732e 2054 6865 2064 6973 7469  pairs. The disti
-0000d120: 6e63 7469 6f6e 2062 6574 7765 656e 2061  nction between a
-0000d130: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
-0000d140: 7379 6e74 656c 6f67 7320 6973 2074 6861  syntelogs is tha
-0000d150: 7420 7468 6520 6c61 7474 6572 2072 6566  t the latter ref
-0000d160: 6572 7320 746f 206d 756c 7469 706c 6520  ers to multiple 
-0000d170: 7365 7473 206f 6620 6765 6e65 7320 6465  sets of genes de
-0000d180: 7269 7665 6420 6672 6f6d 2074 6865 2073  rived from the s
-0000d190: 616d 6520 616e 6365 7374 7261 6c20 6765  ame ancestral ge
-0000d1a0: 6e6f 6d69 6320 7265 6769 6f6e 2077 6869  nomic region whi
-0000d1b0: 6c65 2074 6865 2066 6f72 6d65 7220 696d  le the former im
-0000d1c0: 706c 6965 7320 7468 6520 6c61 7474 6572  plies the latter
-0000d1d0: 2062 7574 2072 6571 7569 7265 7320 6164   but requires ad
-0000d1e0: 6469 7469 6f6e 616c 6c79 2074 6865 2063  ditionally the c
-0000d1f0: 6f6e 7365 7276 6564 2067 656e 6520 6f72  onserved gene or
-0000d200: 6465 722c 2062 6f74 6820 6f66 2077 6869  der, both of whi
-0000d210: 6368 2061 7265 2c20 7769 7468 696e 2061  ch are, within a
-0000d220: 2067 656e 6f6d 652c 2061 7373 756d 6564   genome, assumed
-0000d230: 2074 6f20 6265 206f 7269 6769 6e61 7465   to be originate
-0000d240: 6420 6672 6f6d 2074 6865 2064 7570 6c69  d from the dupli
-0000d250: 6361 7469 6f6e 206f 6620 6120 636f 6d6d  cation of a comm
-0000d260: 6f6e 2061 6e63 6573 7472 616c 2067 656e  on ancestral gen
-0000d270: 6f6d 6963 2072 6567 696f 6e20 616e 6420  omic region and 
-0000d280: 6173 2073 7563 6820 6465 656d 6564 2065  as such deemed e
-0000d290: 7669 6465 6e63 6520 666f 7220 5747 442e  vidence for WGD.
-0000d2a0: 2054 6865 6e20 7468 6520 7379 6e74 656c   Then the syntel
-0000d2b0: 6f67 7320 7769 6c6c 2062 6520 6d61 7070  ogs will be mapp
-0000d2c0: 6564 2062 6163 6b20 6163 636f 7264 696e  ed back accordin
-0000d2d0: 6720 746f 2074 6865 2063 6c75 7374 6572  g to the cluster
-0000d2e0: 696e 6720 7265 7375 6c74 7320 6f66 2074  ing results of t
-0000d2f0: 6865 6972 2061 6666 696c 6961 7465 6420  heir affiliated 
-0000d300: 7365 676d 656e 7473 2e20 5369 6e63 6520  segments. Since 
-0000d310: 7468 6520 4761 7573 7369 616e 2073 6861  the Gaussian sha
-0000d320: 7065 206f 6620 7468 6520 7365 676d 656e  pe of the segmen
-0000d330: 7420 636c 7573 7465 7220 646f 6573 6e27  t cluster doesn'
-0000d340: 7420 6e65 6365 7373 6172 696c 7920 696d  t necessarily im
-0000d350: 6974 6174 6520 7468 6520 7368 6170 6520  itate the shape 
-0000d360: 6f66 2074 6865 2072 6573 6964 696e 6720  of the residing 
-0000d370: 7379 6e74 656c 6f67 732c 2073 6f20 6173  syntelogs, so as
-0000d380: 2074 6f20 7265 7472 6965 7665 2074 6865   to retrieve the
-0000d390: 2022 7265 6c69 6162 6c65 2220 6765 6e65   "reliable" gene
-0000d3a0: 2073 6574 7320 666f 7220 7068 796c 6f67   sets for phylog
-0000d3b0: 656e 6574 6963 2064 6174 696e 672c 2077  enetic dating, w
-0000d3c0: 6520 6164 706f 7465 6420 7468 6520 2839  e adpoted the (9
-0000d3d0: 3525 2920 6869 6768 6573 7420 6465 6e73  5%) highest dens
-0000d3e0: 6974 7920 7265 6769 6f6e 2028 4844 5229  ity region (HDR)
-0000d3f0: 206f 6620 7468 6520 7379 6e74 656c 6f67   of the syntelog
-0000d400: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d410: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000d420: 6e20 666f 7220 7468 6520 7068 796c 6f67  n for the phylog
-0000d430: 656e 6574 6963 2064 6174 696e 672c 2074  enetic dating, t
-0000d440: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-0000d450: 6620 7468 6520 2839 3525 2920 4844 5220  f the (95%) HDR 
-0000d460: 696e 2074 6865 2066 756e 6374 696f 6e20  in the function 
-0000d470: 5b63 616c 6375 6c61 7465 4850 445d 2868  [calculateHPD](h
-0000d480: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000d490: 6d2f 6865 6368 652d 7073 622f 7767 642f  m/heche-psb/wgd/
-0000d4a0: 626c 6f62 2f70 6879 6c6f 6461 7469 6e67  blob/phylodating
-0000d4b0: 2f77 6764 2f70 6561 6b2e 7079 234c 3137  /wgd/peak.py#L17
-0000d4c0: 3537 2920 7365 656b 7320 7468 6520 7368  57) seeks the sh
-0000d4d0: 6f72 7465 7374 202a 4b2a 3c73 7562 3e53  ortest *K*<sub>S
-0000d4e0: 3c2f 7375 623e 2072 616e 6765 2028 612c  </sub> range (a,
-0000d4f0: 6229 2077 6869 6368 2073 6174 6973 6669  b) which satisfi
-0000d500: 6573 2074 6865 2072 6571 7569 7265 6d65  es the requireme
-0000d510: 6e74 206f 6620 7370 616e 6e69 6e67 206d  nt of spanning m
-0000d520: 6f72 6520 7468 616e 2028 3935 2529 206f  ore than (95%) o
-0000d530: 6620 616c 6c20 7468 6520 2a4b 2a3c 7375  f all the *K*<su
-0000d540: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-0000d550: 2e20 4f6e 2074 6865 2070 7265 6d69 7365  . On the premise
-0000d560: 206f 6620 6964 656e 7469 6669 6564 2061   of identified a
-0000d570: 6e63 686f 7220 7061 6972 7320 286e 6f74  nchor pairs (not
-0000d580: 6520 7468 6174 2074 6865 2073 796e 7465  e that the synte
-0000d590: 6c6f 6773 2061 7265 2061 6c73 6f20 7265  logs are also re
-0000d5a0: 6665 7272 6564 2061 7320 616e 6368 6f72  ferred as anchor
-0000d5b0: 2070 6169 7273 2068 6572 6561 6674 6572   pairs hereafter
-0000d5c0: 292c 2077 6520 696d 706c 656d 656e 7465  ), we implemente
-0000d5d0: 6420 696e 2074 6865 2070 726f 6772 616d  d in the program
-0000d5e0: 2060 7767 6420 646d 6460 2074 6865 2073   `wgd dmd` the s
-0000d5f0: 6f2d 6361 6c6c 6564 2061 6e63 686f 722d  o-called anchor-
-0000d600: 6177 6172 6520 6c6f 6361 6c20 4d52 4248  aware local MRBH
-0000d610: 7320 6f72 206f 7274 686f 6772 6f75 7073  s or orthogroups
-0000d620: 2c20 696e 2077 6869 6368 2074 6865 206f  , in which the o
-0000d630: 7269 6769 6e61 6c20 6c6f 6361 6c20 4d52  riginal local MR
-0000d640: 4248 7320 6172 6520 6675 7274 6865 7220  BHs are further 
-0000d650: 6d65 7267 6564 2077 6974 6820 616e 6368  merged with anch
-0000d660: 6f72 2070 6169 7273 2073 7563 6820 7468  or pairs such th
-0000d670: 6174 2065 6163 6820 6f72 7468 6f67 726f  at each orthogro
-0000d680: 7570 2063 6f6e 7461 696e 7320 7468 6520  up contains the 
-0000d690: 616e 6368 6f72 2070 6169 7220 616e 6420  anchor pair and 
-0000d6a0: 7468 6520 6f72 7468 6f6c 6f67 7565 732e  the orthologues.
-0000d6b0: 2057 6974 6820 7468 6973 206f 7274 686f   With this ortho
-0000d6c0: 6772 6f75 702c 2075 7365 7273 2074 6865  group, users the
-0000d6d0: 6e20 6e65 6564 206f 6e65 2073 7461 7274  n need one start
-0000d6e0: 696e 6720 7472 6565 2066 696c 6520 2861  ing tree file (a
-0000d6f0: 7320 7368 6f77 6e20 696e 2074 6865 205b  s shown in the [
-0000d700: 496c 6c75 7374 7261 7469 6f6e 5d28 6874  Illustration](ht
-0000d710: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000d720: 2f68 6563 6865 2d70 7362 2f77 6764 2f74  /heche-psb/wgd/t
-0000d730: 7265 652f 7068 796c 6f64 6174 696e 673f  ree/phylodating?
-0000d740: 7461 623d 7265 6164 6d65 2d6f 762d 6669  tab=readme-ov-fi
-0000d750: 6c65 2369 6c6c 7573 7472 6174 696f 6e29  le#illustration)
-0000d760: 2070 6172 7429 2069 6e64 6963 6174 696e   part) indicatin
-0000d770: 6720 7468 6520 7472 6565 2074 6f70 6f6c  g the tree topol
-0000d780: 6f67 7920 616e 6420 666f 7373 696c 2063  ogy and fossil c
-0000d790: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-0000d7a0: 6d61 7469 6f6e 2066 6f72 2074 6865 2066  mation for the f
-0000d7b0: 696e 616c 2070 6879 6c6f 6765 6e65 7469  inal phylogeneti
-0000d7c0: 6320 6461 7469 6e67 2075 7369 6e67 2074  c dating using t
-0000d7d0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000d7e0: 666f 6375 7360 2e20 5573 6572 7320 6361  focus`. Users ca
-0000d7f0: 6e20 7365 7420 7468 6520 666c 6167 206f  n set the flag o
-0000d800: 7074 696f 6e20 602d 2d70 726f 7463 6f63  ption `--protcoc
-0000d810: 6461 7469 6e67 6020 746f 206f 6e6c 7920  dating` to only 
-0000d820: 636f 6e64 7563 7420 7468 6520 6461 7469  conduct the dati
-0000d830: 6e67 206f 6620 7468 6520 636f 6e63 6174  ng of the concat
-0000d840: 656e 6174 696f 6e20 7072 6f74 6569 6e20  enation protein 
-0000d850: 4d53 4120 6f72 2074 6865 2066 6c61 6720  MSA or the flag 
-0000d860: 6f70 7469 6f6e 2060 2d2d 7072 6f74 6461  option `--protda
-0000d870: 7469 6e67 6020 746f 206f 6e6c 7920 636f  ting` to only co
-0000d880: 6e64 7563 7420 7468 6520 6461 7469 6e67  nduct the dating
-0000d890: 206f 6620 7468 6520 7072 6f74 6569 6e20   of the protein 
-0000d8a0: 4d53 4173 2c20 6e6f 7465 6420 7468 6174  MSAs, noted that
-0000d8b0: 2074 6865 7365 2074 776f 206f 7074 696f   these two optio
-0000d8c0: 6e73 206f 6e6c 7920 776f 726b 2066 6f72  ns only work for
-0000d8d0: 2074 6865 2060 6d63 6d63 7472 6565 6020   the `mcmctree` 
-0000d8e0: 6f70 7469 6f6e 2073 6f20 6661 722e 2054  option so far. T
-0000d8f0: 6865 2066 6c61 6720 6f70 7469 6f6e 2060  he flag option `
-0000d900: 2d2d 7061 7274 6974 696f 6e60 2063 616e  --partition` can
-0000d910: 2062 6520 7365 7420 746f 2070 6572 666f   be set to perfo
-0000d920: 726d 2060 6d63 6d63 7472 6565 6020 616e  rm `mcmctree` an
-0000d930: 616c 7973 6973 2075 7369 6e67 2074 6865  alysis using the
-0000d940: 2070 6172 7469 7469 6f6e 6564 2064 6174   partitioned dat
-0000d950: 6120 2869 2e65 2e2c 2031 7374 2c20 326e  a (i.e., 1st, 2n
-0000d960: 6420 616e 6420 3372 6420 706f 7369 7469  d and 3rd positi
-0000d970: 6f6e 206f 6620 636f 646f 6e29 2069 6e73  on of codon) ins
-0000d980: 7465 6164 206f 6620 7573 696e 6720 7468  tead of using th
-0000d990: 6520 636f 646f 6e20 6173 2061 2077 686f  e codon as a who
-0000d9a0: 6c65 2e20 5468 6520 6f70 7469 6f6e 2060  le. The option `
-0000d9b0: 2d2d 6161 6d6f 6465 6c60 2063 616e 2062  --aamodel` can b
-0000d9c0: 6520 7365 7420 746f 2064 6574 6572 6d69  e set to determi
-0000d9d0: 6e65 2074 6865 2061 6d69 6e6f 2061 6369  ne the amino aci
-0000d9e0: 6420 6d6f 6465 6c20 6170 706c 6965 6420  d model applied 
-0000d9f0: 696e 2060 6d63 6d63 7472 6565 6020 616e  in `mcmctree` an
-0000da00: 616c 7973 6973 2e20 5468 6520 6f70 7469  alysis. The opti
-0000da10: 6f6e 2060 2d64 7360 2063 616e 2062 6520  on `-ds` can be 
-0000da20: 7573 6564 2074 6f20 7365 7420 7061 7261  used to set para
-0000da30: 6d65 7465 7273 2066 6f72 2074 6865 206d  meters for the m
-0000da40: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
-0000da50: 7072 6f67 7261 6d2e 2054 6865 2070 6172  program. The par
-0000da60: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000da70: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000da80: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000da90: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000daa0: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000dab0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000dac0: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000dad0: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000dae0: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000daf0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000db00: 7469 6e67 2072 756e 2063 616e 2062 6520  ting run can be 
-0000db10: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
-0000db20: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
-0000db30: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
-0000db40: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000db50: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000db60: 2d64 206d 636d 6374 7265 6520 2d73 7020  -d mcmctree -sp 
-0000db70: 7370 6461 7461 0a77 6764 2066 6f63 7573  spdata.wgd focus
-0000db80: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000db90: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000dba0: 7175 656e 6365 3320 2d64 2072 3873 202d  quence3 -d r8s -
-0000dbb0: 7370 2073 7064 6174 6120 2d2d 6e73 6974  sp spdata --nsit
-0000dbc0: 6573 206e 7369 7465 696e 666f 0a77 6764  es nsiteinfo.wgd
-0000dbd0: 2066 6f63 7573 2066 616d 696c 6965 7320   focus families 
-0000dbe0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
-0000dbf0: 6365 3220 7365 7175 656e 6365 3320 2d64  ce2 sequence3 -d
-0000dc00: 2062 6561 7374 202d 2d66 6f73 7369 6c20   beast --fossil 
-0000dc10: 666f 7373 696c 696e 666f 202d 2d72 6f6f  fossilinfo --roo
-0000dc20: 7468 6569 6768 7420 726f 6f74 6865 6967  theight rootheig
-0000dc30: 6874 696e 666f 202d 2d63 6861 696e 7365  htinfo --chainse
-0000dc40: 7420 6368 6169 6e73 6574 696e 666f 202d  t chainsetinfo -
-0000dc50: 2d62 6561 7374 6c67 6a61 7220 2450 4154  -beastlgjar $PAT
-0000dc60: 480a 6060 600a 0a23 2323 2077 6764 206b  H.```..### wgd k
-0000dc70: 7364 0a0a 2a2a 5468 6520 636f 6e73 7472  sd..**The constr
-0000dc80: 7563 7469 6f6e 206f 6620 7768 6f6c 6520  uction of whole 
-0000dc90: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-0000dca0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000dcb0: 7472 6962 7574 696f 6e2a 2a0a 6060 600a  tribution**.```.
-0000dcc0: 7767 6420 6b73 6420 6661 6d69 6c69 6573  wgd ksd families
-0000dcd0: 2073 6571 7565 6e63 6520 282d 6f20 7767   sequence (-o wg
-0000dce0: 645f 6b73 6420 2d74 2077 6764 5f6b 7364  d_ksd -t wgd_ksd
-0000dcf0: 5f74 6d70 202d 2d6e 7468 7265 6164 7320  _tmp --nthreads 
-0000dd00: 3420 2d2d 746f 5f73 746f 7020 2d2d 6364  4 --to_stop --cd
-0000dd10: 7320 2d2d 7061 6972 7769 7365 202d 2d73  s --pairwise --s
-0000dd20: 7472 6970 5f67 6170 7320 2d2d 616c 6967  trip_gaps --alig
-0000dd30: 6e65 7220 6d61 6666 7420 2d2d 7472 6565  ner mafft --tree
-0000dd40: 5f6d 6574 686f 6420 6661 7374 7472 6565  _method fasttree
-0000dd50: 202d 2d6e 6f64 655f 6176 6572 6167 6529   --node_average)
-0000dd60: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
-0000dd70: 6d20 6077 6764 206b 7364 602c 2061 7320  m `wgd ksd`, as 
-0000dd80: 696d 7069 6564 2062 7920 6974 7320 6e61  impied by its na
-0000dd90: 6d65 2c20 6973 2066 6f72 2074 6865 2063  me, is for the c
-0000dda0: 6f6e 7374 7275 6374 696f 6e20 6f66 202a  onstruction of *
-0000ddb0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000ddc0: 6765 2064 6973 7472 6962 7574 696f 6e2e  ge distribution.
-0000ddd0: 2045 7863 6570 7420 666f 7220 7468 6520   Except for the 
-0000dde0: 6166 6f72 656d 656e 7469 6f6e 6564 2070  aforementioned p
-0000ddf0: 6172 616d 6574 6572 7320 7375 6368 2061  arameters such a
-0000de00: 7320 602d 2d74 6f5f 7374 6f70 6020 616e  s `--to_stop` an
-0000de10: 6420 602d 2d63 6473 602c 2074 6865 7265  d `--cds`, there
-0000de20: 2061 7265 2073 6f6d 6520 696d 706f 7274   are some import
-0000de30: 616e 7420 7061 7261 6d65 7465 7273 2074  ant parameters t
-0000de40: 6861 7420 6861 7665 2063 7275 6369 616c  hat have crucial
-0000de50: 2069 6d70 6163 7420 6f6e 2074 6865 202a   impact on the *
-0000de60: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
-0000de70: 7374 696d 6174 696f 6e2e 2054 6865 206f  stimation. The o
-0000de80: 7074 696f 6e20 602d 2d70 6169 7277 6973  ption `--pairwis
-0000de90: 6560 2069 7320 6120 7665 7279 2069 6d70  e` is a very imp
-0000dea0: 6f72 7461 6e74 2070 6172 616d 6574 6572  ortant parameter
-0000deb0: 2066 6f72 2074 6865 202a 4b2a 3c73 7562   for the *K*<sub
-0000dec0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-0000ded0: 696f 6e2c 2077 6974 6820 7768 6963 6820  ion, with which 
-0000dee0: 7468 6520 6043 4f44 454d 4c60 2077 696c  the `CODEML` wil
-0000def0: 6c20 6361 6c63 756c 6174 6520 7468 6520  l calculate the 
-0000df00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000df10: 666f 7220 6561 6368 2067 656e 6520 7061  for each gene pa
-0000df20: 6972 2073 6570 6172 6174 656c 7920 6261  ir separately ba
-0000df30: 7365 6420 6f6e 2074 6865 2061 6c69 676e  sed on the align
-0000df40: 6d65 6e74 206f 6620 6f6e 6c79 2074 6865  ment of only the
-0000df50: 7365 2074 776f 2067 656e 6573 2069 6e73  se two genes ins
-0000df60: 7465 6164 206f 6620 7468 6520 7768 6f6c  tead of the whol
-0000df70: 6520 616c 6967 6e6d 656e 7420 6f66 2074  e alignment of t
-0000df80: 6865 2066 616d 696c 792c 2073 7563 6820  he family, such 
-0000df90: 7468 6174 206c 6573 7320 6761 7073 2061  that less gaps a
-0000dfa0: 7265 2065 7870 6563 7465 6420 616e 6420  re expected and 
-0000dfb0: 7468 7573 2074 6865 2061 6c69 676e 6d65  thus the alignme
-0000dfc0: 6e74 2069 6e20 7468 6520 636f 6e73 6964  nt in the consid
-0000dfd0: 6572 6174 696f 6e20 6f66 2060 434f 4445  eration of `CODE
-0000dfe0: 4d4c 6020 7769 6c6c 2062 6520 6c6f 6e67  ML` will be long
-0000dff0: 6572 2028 6265 6361 7573 6520 6043 4f44  er (because `COD
-0000e000: 454d 4c60 2077 696c 6c20 6175 746f 6d61  EML` will automa
-0000e010: 7465 6c79 2073 6b69 7020 6576 6572 7920  tely skip every 
-0000e020: 636f 6c75 6d6e 2077 6974 6820 6761 702c  column with gap,
-0000e030: 2072 6567 6172 646c 6573 7320 6f66 2077   regardless of w
-0000e040: 6865 7468 6572 2069 7427 7320 616e 206f  hether it's an o
-0000e050: 7665 7261 6c6c 206f 7220 7061 7274 6961  verall or partia
-0000e060: 6c20 6761 7029 2c20 7769 7468 6f75 7420  l gap), without 
-0000e070: 7768 6963 6820 7468 6520 6043 4f44 454d  which the `CODEM
-0000e080: 4c60 2077 696c 6c20 6361 6c63 756c 6174  L` will calculat
-0000e090: 6520 7468 6520 2a4b 2a3c 7375 623e 533c  e the *K*<sub>S<
-0000e0a0: 2f73 7562 3e20 6261 7365 6420 6f6e 2074  /sub> based on t
-0000e0b0: 6865 2077 686f 6c65 2061 6c69 676e 6d65  he whole alignme
-0000e0c0: 6e74 206f 6620 7468 6520 6661 6d69 6c79  nt of the family
-0000e0d0: 2c20 7768 6963 6820 6d69 6768 7420 6861  , which might ha
-0000e0e0: 7665 206e 6f20 2a4b 2a3c 7375 623e 533c  ve no *K*<sub>S<
-0000e0f0: 2f73 7562 3e20 7265 7375 6c74 2061 7420  /sub> result at 
-0000e100: 616c 6c20 6966 2074 6865 2073 7472 6970  all if the strip
-0000e110: 7065 6420 616c 6967 6e6d 656e 7420 6c65  ped alignment le
-0000e120: 6e67 7468 2028 7265 6d6f 7669 6e67 2061  ngth (removing a
-0000e130: 6c6c 2067 6170 2d63 6f6e 7461 696e 696e  ll gap-containin
-0000e140: 6720 636f 6c75 6d6e 7329 2077 6173 207a  g columns) was z
-0000e150: 6572 6f2c 2061 2063 6175 7365 206f 6620  ero, a cause of 
-0000e160: 6469 6666 6572 656e 7420 6e75 6d62 6572  different number
-0000e170: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
-0000e180: 7562 3e20 6573 7469 6d61 7465 7320 6265  ub> estimates be
-0000e190: 7477 6565 6e20 2270 6169 7277 6973 6520  tween "pairwise 
-0000e1a0: 6d6f 6465 2220 616e 6420 226e 6f6e 2d70  mode" and "non-p
-0000e1b0: 6169 7277 6973 6520 6d6f 6465 222e 2049  airwise mode". I
-0000e1c0: 7427 7320 6469 6666 6963 756c 7420 746f  t's difficult to
-0000e1d0: 2073 6179 2077 6869 6368 206d 6f64 6520   say which mode 
-0000e1e0: 6973 206d 6f72 6520 6964 6561 6c2c 2061  is more ideal, a
-0000e1f0: 6c74 686f 7567 6820 7468 6520 226e 6f6e  lthough the "non
-0000e200: 2d70 6169 7277 6973 6520 6d6f 6465 2220  -pairwise mode" 
-0000e210: 2864 6566 6175 6c74 2073 6574 7469 6e67  (default setting
-0000e220: 2920 7768 6963 6820 7275 6e73 206f 6e20  ) which runs on 
-0000e230: 7468 6520 7768 6f6c 6520 616c 6967 6e6d  the whole alignm
-0000e240: 656e 7420 696e 7374 6561 6420 6f66 2061  ent instead of a
-0000e250: 206c 6f63 616c 2061 6c69 676e 6d65 6e74   local alignment
-0000e260: 2c20 6d69 6768 7420 6265 206d 6f72 6520  , might be more 
-0000e270: 6269 6f6c 6f67 6963 616c 6c79 2063 6f6e  biologically con
-0000e280: 7365 7276 6564 2069 6e20 7468 6174 2069  served in that i
-0000e290: 7420 6173 7375 7265 7320 7468 6520 6576  t assures the ev
-0000e2a0: 6f6c 7574 696f 6e20 6f66 2065 6163 6820  olution of each 
-0000e2b0: 636f 6c75 6d6e 2074 6f20 6265 2073 7461  column to be sta
-0000e2c0: 7274 6564 2066 726f 6d20 7468 6520 726f  rted from the ro
-0000e2d0: 6f74 206f 6620 7468 6520 6661 6d69 6c79  ot of the family
-0000e2e0: 2061 6e64 2061 6c6c 2074 6865 2067 656e   and all the gen
-0000e2f0: 6520 6475 706c 6963 6174 6573 2061 7265  e duplicates are
-0000e300: 2074 616b 656e 2069 6e74 6f20 6163 636f   taken into acco
-0000e310: 756e 7420 696e 2074 6865 202a 4b2a 3c73  unt in the *K*<s
-0000e320: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
-0000e330: 6174 696f 6e20 7072 6f63 6573 732e 2054  ation process. T
-0000e340: 6865 206f 7074 696f 6e20 602d 2d73 7472  he option `--str
-0000e350: 6970 5f67 6170 7360 2063 616e 2072 656d  ip_gaps` can rem
-0000e360: 6f76 6520 616c 6c20 7468 6520 6761 702d  ove all the gap-
-0000e370: 636f 6e74 6169 6e69 6e67 2063 6f6c 756d  containing colum
-0000e380: 6e73 2c20 7769 7468 206f 7220 7769 7468  ns, with or with
-0000e390: 6f75 7420 7768 6963 6820 7468 6520 7265  out which the re
-0000e3a0: 7375 6c74 206f 6620 226e 6f6e 2d70 6169  sult of "non-pai
-0000e3b0: 7277 6973 6520 6d6f 6465 2220 776f 6e27  rwise mode" won'
-0000e3c0: 7420 6265 2061 6666 6563 7465 642c 2077  t be affected, w
-0000e3d0: 6869 6c65 2077 6974 6820 7768 6963 6820  hile with which 
-0000e3e0: 7468 6520 7265 7375 6c74 206f 6620 2270  the result of "p
-0000e3f0: 6169 7277 6973 6520 6d6f 6465 2220 7769  airwise mode" wi
-0000e400: 6c6c 2062 6520 616c 7465 7265 642e 2054  ll be altered. T
-0000e410: 6865 206f 7074 696f 6e20 602d 2d61 6c69  he option `--ali
-0000e420: 676e 6572 6020 616e 6420 602d 2d61 6c6e  gner` and `--aln
-0000e430: 5f6f 7074 696f 6e73 6020 7768 6963 6820  _options` which 
-0000e440: 6465 6369 6465 2077 6869 6368 2061 6c69  decide which ali
-0000e450: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
-0000e460: 6f20 6265 2075 7365 6420 616e 6420 7768  o be used and wh
-0000e470: 6963 6820 7061 7261 6d65 7465 7220 746f  ich parameter to
-0000e480: 2062 6520 7365 742c 2077 696c 6c20 6861   be set, will ha
-0000e490: 7665 2069 6d70 6163 7420 6f6e 2074 6865  ve impact on the
-0000e4a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000e4b0: 2072 6573 756c 7473 2c20 6e6f 7465 6420   results, noted 
-0000e4c0: 7468 6174 2074 6865 2064 6566 6175 6c74  that the default
-0000e4d0: 2070 726f 6772 616d 2069 7320 606d 6166   program is `maf
-0000e4e0: 6674 6020 616e 6420 7468 6520 7061 7261  ft` and the para
-0000e4f0: 6d65 7465 7220 6973 2060 2d2d 6175 746f  meter is `--auto
-0000e500: 602e 2054 6865 206f 7074 696f 6e20 602d  `. The option `-
-0000e510: 2d74 7265 655f 6d65 7468 6f64 6020 616e  -tree_method` an
-0000e520: 6420 602d 2d74 7265 655f 6f70 7469 6f6e  d `--tree_option
-0000e530: 7360 2064 6563 6964 6520 7768 6963 6820  s` decide which 
-0000e540: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000e550: 6e63 6520 7072 6f67 7261 6d20 746f 2062  nce program to b
-0000e560: 6520 7573 6564 2061 6e64 2077 6869 6368  e used and which
-0000e570: 2070 6172 616d 6574 6572 2074 6f20 6265   parameter to be
-0000e580: 2073 6574 2c20 776f 6e27 7420 6166 6665   set, won't affe
-0000e590: 6374 2074 6865 202a 4b2a 3c73 7562 3e53  ct the *K*<sub>S
-0000e5a0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
-0000e5b0: 6e20 6974 7365 6c66 2062 7574 2074 6865  n itself but the
-0000e5c0: 2072 6573 756c 7420 6f66 2064 652d 7265   result of de-re
-0000e5d0: 6475 6e64 616e 6379 2c20 6e6f 7465 6420  dundancy, noted 
-0000e5e0: 7468 6174 2077 6520 696d 706c 656d 656e  that we implemen
-0000e5f0: 7465 6420 6120 6275 696c 742d 696e 2067  ted a built-in g
-0000e600: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
-0000e610: 6365 206d 6574 686f 6420 6261 7365 6420  ce method based 
-0000e620: 6f6e 2074 6865 2041 7665 7261 6765 204c  on the Average L
-0000e630: 696e 6b61 6765 2043 6c75 7374 6572 696e  inkage Clusterin
-0000e640: 6720 2841 4c43 2920 2874 6875 7320 6120  g (ALC) (thus a 
-0000e650: 6469 7374 616e 6365 2d62 6173 6564 2074  distance-based t
-0000e660: 7265 6529 2077 6974 6820 7468 6520 602d  ree) with the `-
-0000e670: 2d74 7265 655f 6d65 7468 6f64 6020 7365  -tree_method` se
-0000e680: 7420 6173 2022 636c 7573 7465 7222 2e20  t as "cluster". 
-0000e690: 5477 6f20 6d65 7468 6f64 7320 6f66 2064  Two methods of d
-0000e6a0: 652d 7265 6475 6e64 616e 6379 2061 7265  e-redundancy are
-0000e6b0: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
-0000e6c0: 6077 6764 2076 3260 2c20 6e61 6d65 6c79  `wgd v2`, namely
-0000e6d0: 206e 6f64 652d 7765 6967 6874 6564 2061   node-weighted a
-0000e6e0: 6e64 206e 6f64 652d 6176 6572 6167 6564  nd node-averaged
-0000e6f0: 206d 6574 686f 6473 2e20 5468 6520 6e6f   methods. The no
-0000e700: 6465 2d77 6569 6768 7465 6420 6d65 7468  de-weighted meth
-0000e710: 6f64 2061 6368 6965 7665 7320 7468 6520  od achieves the 
-0000e720: 6465 2d72 6564 756e 6461 6e63 7920 7669  de-redundancy vi
-0000e730: 6120 7765 6967 6869 6e67 2074 6865 202a  a weighing the *
-0000e740: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
-0000e750: 616c 7565 2061 7373 6f63 6961 7465 6420  alue associated 
-0000e760: 7769 7468 2065 6163 6820 6765 6e65 2070  with each gene p
-0000e770: 6169 7220 7375 6368 2074 6861 7420 7468  air such that th
-0000e780: 6520 7765 6967 6874 206f 6620 6120 7369  e weight of a si
-0000e790: 6e67 6c65 2064 7570 6c69 6361 7469 6f6e  ngle duplication
-0000e7a0: 2065 7665 6e74 2073 756d 7320 7570 2074   event sums up t
-0000e7b0: 6f20 3120 286e 6f74 6564 2074 6861 7420  o 1 (noted that 
-0000e7c0: 7468 6520 6e75 6d62 6572 206f 6620 2a4b  the number of *K
-0000e7d0: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-0000e7e0: 7469 6d61 7469 6f6e 2072 656d 6169 6e73  timation remains
-0000e7f0: 2074 6865 2073 616d 6529 2077 6869 6c65   the same) while
-0000e800: 2074 6865 206e 6f64 652d 6176 6572 6167   the node-averag
-0000e810: 6564 206d 6574 686f 6420 7265 616c 697a  ed method realiz
-0000e820: 6573 2074 6865 2064 652d 7265 6475 6e64  es the de-redund
-0000e830: 616e 6379 2076 6961 2063 616c 6375 6c61  ancy via calcula
-0000e840: 7469 6e67 2070 6572 2067 656e 6520 7472  ting per gene tr
-0000e850: 6565 206e 6f64 6520 6f6e 6520 6176 6572  ee node one aver
-0000e860: 6167 6564 202a 4b2a 3c73 7562 3e53 3c2f  aged *K*<sub>S</
-0000e870: 7375 623e 2076 616c 7565 2074 6f20 7265  sub> value to re
-0000e880: 7072 6573 656e 7420 7468 6520 6167 6520  present the age 
-0000e890: 6f66 2065 6163 6820 6475 706c 6963 6174  of each duplicat
-0000e8a0: 696f 6e20 6576 656e 742e 2054 6865 206f  ion event. The o
-0000e8b0: 7074 696f 6e20 602d 2d6e 6f64 655f 6176  ption `--node_av
-0000e8c0: 6572 6167 6560 2063 616e 2062 6520 7365  erage` can be se
-0000e8d0: 7420 746f 2063 686f 6f73 6520 7468 6520  t to choose the 
-0000e8e0: 6e6f 6465 2d61 7665 7261 6765 6420 7761  node-averaged wa
-0000e8f0: 7920 6f66 2064 652d 7265 6475 6e64 616e  y of de-redundan
-0000e900: 6379 2e20 4469 6666 6572 656e 7420 6d65  cy. Different me
-0000e910: 7468 6f64 7320 6f66 2064 652d 7265 6475  thods of de-redu
-0000e920: 6e64 616e 6379 2077 696c 6c20 6861 7665  ndancy will have
-0000e930: 2069 6d70 6163 7420 6f6e 2074 6865 2064   impact on the d
-0000e940: 6574 6563 7469 6f6e 206f 6620 5747 4420  etection of WGD 
-0000e950: 7369 676e 616c 732c 2077 6869 6368 2068  signals, which h
-0000e960: 6173 2062 6565 6e20 696e 7665 7374 6967  as been investig
-0000e970: 6174 6564 2069 6e20 7468 6973 205b 6c69  ated in this [li
-0000e980: 7465 7261 7475 7265 5d28 6874 7470 733a  terature](https:
-0000e990: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-0000e9a0: 332f 6762 652f 6576 7932 3030 292e 2054  3/gbe/evy200). T
-0000e9b0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000e9c0: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000e9d0: 7061 7261 6c6c 656c 697a 6520 7468 6520  parallelize the 
-0000e9e0: 616e 616c 7973 6973 2066 6f72 2065 6163  analysis for eac
-0000e9f0: 6820 6765 6e65 2066 616d 696c 7920 616e  h gene family an
-0000ea00: 6420 7468 7573 2073 7567 6765 7374 6564  d thus suggested
-0000ea10: 2074 6f20 6265 2073 6574 2061 7320 6d75   to be set as mu
-0000ea20: 6368 2061 7320 7468 6520 6e75 6d62 6572  ch as the number
-0000ea30: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
-0000ea40: 732e 0a0a 2a2a 4120 7375 6767 6573 7465  s...**A suggeste
-0000ea50: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
-0000ea60: 616e 2075 7365 2063 6f6d 6d61 6e64 2061  an use command a
-0000ea70: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-0000ea80: 6764 206b 7364 2066 616d 696c 6965 7320  gd ksd families 
-0000ea90: 7365 7175 656e 6365 0a60 6060 0a0a 2a2a  sequence.```..**
-0000eaa0: 5468 6520 636f 6e73 7472 7563 7469 6f6e  The construction
-0000eab0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-0000eac0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000ead0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000eae0: 2a2a 0a60 6060 0a77 6764 206b 7364 2066  **.```.wgd ksd f
-0000eaf0: 616d 696c 6965 7320 7365 7175 656e 6365  amilies sequence
-0000eb00: 3120 7365 7175 656e 6365 3220 282d 2d72  1 sequence2 (--r
-0000eb10: 6577 6569 6768 7429 0a60 6060 0a0a 4672  eweight).```..Fr
-0000eb20: 6f6d 2070 6172 616c 6f67 6f75 7320 746f  om paralogous to
-0000eb30: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
-0000eb40: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000eb50: 2064 6973 7472 6962 7574 696f 6e2c 2075   distribution, u
-0000eb60: 7365 7273 206f 6e6c 7920 6e65 6564 2074  sers only need t
-0000eb70: 6f20 7072 6f76 6964 6520 6d6f 7265 2063  o provide more c
-0000eb80: 6473 2066 696c 6573 2e20 4e6f 7465 2074  ds files. Note t
-0000eb90: 6861 7420 7769 7468 206f 7274 686f 6c6f  hat with ortholo
-0000eba0: 676f 7573 2067 656e 6520 6661 6d69 6c69  gous gene famili
-0000ebb0: 6573 2074 6865 2077 6569 6768 7469 6e67  es the weighting
-0000ebc0: 206d 6574 686f 6420 6361 6e20 6265 2073   method can be s
-0000ebd0: 6574 2074 6f20 6265 2063 616c 6375 6c61  et to be calcula
-0000ebe0: 7465 6420 7065 7220 7370 6563 6965 7320  ted per species 
-0000ebf0: 7061 6972 2069 6e73 7465 6164 206f 6620  pair instead of 
-0000ec00: 636f 6e73 6964 6572 696e 6720 7468 6520  considering the 
-0000ec10: 7768 6f6c 6520 6661 6d69 6c79 2062 6563  whole family bec
-0000ec20: 6175 7365 2077 6865 6e20 706c 6f74 7469  ause when plotti
-0000ec30: 6e67 206f 7274 686f 6c6f 676f 7573 202a  ng orthologous *
-0000ec40: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000ec50: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000ec60: 6265 7477 6565 6e20 7477 6f20 7370 6563  between two spec
-0000ec70: 6965 7320 7468 6520 7765 6967 6874 2063  ies the weight c
-0000ec80: 616c 6375 6c61 7465 6420 6672 6f6d 2074  alculated from t
-0000ec90: 6869 7320 7370 6563 6966 6963 2073 7065  his specific spe
-0000eca0: 6369 6573 2070 6169 7220 7368 6f75 6c64  cies pair should
-0000ecb0: 2062 6520 636f 6e73 6572 7665 6420 7768   be conserved wh
-0000ecc0: 696c 6520 7468 6520 6f6e 6520 6361 6c63  ile the one calc
-0000ecd0: 756c 6174 6564 2066 726f 6d20 7468 6520  ulated from the 
-0000ece0: 7768 6f6c 6520 6661 6d69 6c79 2077 696c  whole family wil
-0000ecf0: 6c20 7661 7279 2077 6974 6820 7468 6520  l vary with the 
-0000ed00: 6e75 6d62 6572 206f 6620 7370 6563 6965  number of specie
-0000ed10: 732e 2054 6f20 696e 6974 6961 7465 2074  s. To initiate t
-0000ed20: 6865 2077 6569 6768 7469 6e67 2070 6572  he weighting per
-0000ed30: 2073 7065 6369 6573 2070 6169 722c 2074   species pair, t
-0000ed40: 6865 206f 7074 696f 6e20 602d 2d72 6577  he option `--rew
-0000ed50: 6569 6768 7460 2063 616e 2062 6520 7365  eight` can be se
-0000ed60: 742e 0a0a 2a2a 4120 7375 6767 6573 7465  t...**A suggeste
-0000ed70: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
-0000ed80: 616e 2075 7365 2063 6f6d 6d61 6e64 2061  an use command a
-0000ed90: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-0000eda0: 6764 206b 7364 2066 616d 696c 6965 7320  gd ksd families 
-0000edb0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
-0000edc0: 6365 320a 6060 600a 0a2a 2a54 6865 2063  ce2.```..**The c
-0000edd0: 6f6e 7374 7275 6374 696f 6e20 6f66 202a  onstruction of *
-0000ede0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000edf0: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000ee00: 7769 7468 2072 6174 6520 636f 7272 6563  with rate correc
-0000ee10: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
-0000ee20: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
-0000ee30: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000ee40: 7365 7175 656e 6365 3320 2d2d 666f 6375  sequence3 --focu
-0000ee50: 7332 616c 6c20 7365 7175 656e 6365 3120  s2all sequence1 
-0000ee60: 2d73 7020 7370 6461 7461 0a60 6060 0a0a  -sp spdata.```..
-0000ee70: 496e 7370 6972 6564 2062 7920 7468 6520  Inspired by the 
-0000ee80: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-0000ee90: 616c 676f 7269 7468 6d20 696e 205b 6b73  algorithm in [ks
-0000eea0: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000eeb0: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000eec0: 422f 6b73 7261 7465 7329 2c20 7765 2069  B/ksrates), we i
-0000eed0: 6d70 6c65 6d65 6e74 6564 2074 6865 2072  mplemented the r
-0000eee0: 6174 6520 636f 7272 6563 7469 6f6e 2061  ate correction a
-0000eef0: 6e61 6c79 7369 7320 616c 736f 2069 6e20  nalysis also in 
-0000ef00: 6077 6764 2076 3260 2c20 7768 6963 6820  `wgd v2`, which 
-0000ef10: 6973 206d 6f73 746c 7920 7468 6520 7361  is mostly the sa
-0000ef20: 6d65 2061 7320 5b6b 7372 6174 6573 5d28  me as [ksrates](
-0000ef30: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000ef40: 6f6d 2f56 4942 2d50 5342 2f6b 7372 6174  om/VIB-PSB/ksrat
-0000ef50: 6573 2920 6275 7420 6469 6666 6572 7320  es) but differs 
-0000ef60: 696e 2074 6865 2063 616c 6375 6c61 7469  in the calculati
-0000ef70: 6f6e 206f 6620 7468 6520 7374 616e 6461  on of the standa
-0000ef80: 7264 2064 6576 6961 7469 6f6e 206f 6620  rd deviation of 
-0000ef90: 7265 7363 616c 6564 202a 4b2a 3c73 7562  rescaled *K*<sub
-0000efa0: 3e53 3c2f 7375 623e 2061 6765 732e 2054  >S</sub> ages. T
-0000efb0: 6f20 7065 7266 6f72 6d20 7468 6520 7261  o perform the ra
-0000efc0: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
-0000efd0: 616c 7973 6973 2c20 7573 6572 7320 6361  alysis, users ca
-0000efe0: 6e20 7573 6520 626f 7468 2074 6865 2060  n use both the `
-0000eff0: 7767 6420 6b73 6460 2061 6e64 2060 7767  wgd ksd` and `wg
-0000f000: 6420 7669 7a60 2070 726f 6772 616d 2e20  d viz` program. 
-0000f010: 466f 7220 7468 6520 6077 6764 206b 7364  For the `wgd ksd
-0000f020: 6020 7072 6f67 7261 6d2c 2075 7365 7273  ` program, users
-0000f030: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
-0000f040: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
-0000f050: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-0000f060: 2d2d 7370 6563 6965 7374 7265 6560 206f  --speciestree` o
-0000f070: 6e20 7768 6963 6820 7261 7465 2063 6f72  n which rate cor
-0000f080: 7265 6374 696f 6e20 6361 6e20 6265 2063  rection can be c
-0000f090: 6f6e 6475 6374 6564 2e20 4e6f 7465 2074  onducted. Note t
-0000f0a0: 6861 7420 756e 6e65 6365 7373 6172 7920  hat unnecessary 
-0000f0b0: 6272 6163 6b65 7473 206d 6967 6874 206c  brackets might l
-0000f0c0: 6561 6420 746f 2075 6e65 7870 6563 7465  ead to unexpecte
-0000f0d0: 6420 6572 726f 7273 2c20 666f 7220 696e  d errors, for in
-0000f0e0: 7374 616e 6365 2061 2074 7265 6520 6028  stance a tree `(
-0000f0f0: 412c 2842 2c43 2929 3b60 2073 686f 756c  A,(B,C));` shoul
-0000f100: 6420 6e6f 7420 6265 2072 6570 7265 7365  d not be represe
-0000f110: 6e74 6564 2061 7320 6028 412c 2828 422c  nted as `(A,((B,
-0000f120: 4329 2929 3b60 2e20 5468 6520 7365 7420  C)));`. The set 
-0000f130: 6f66 2073 7065 6369 6573 2070 6169 7273  of species pairs
-0000f140: 2074 6f20 6265 2073 686f 776e 2069 7320   to be shown is 
-0000f150: 666c 6578 6962 6c65 2074 6861 7420 7468  flexible that th
-0000f160: 6520 6d6f 7374 2063 6f6e 7665 6e69 656e  e most convenien
-0000f170: 7420 6f70 7469 6f6e 2069 7320 602d 2d66  t option is `--f
-0000f180: 6f63 7573 3261 6c6c 6020 7768 6963 6820  ocus2all` which 
-0000f190: 7369 6d70 6c79 2073 686f 7773 2061 6c6c  simply shows all
-0000f1a0: 2074 6865 2070 6f73 7369 626c 6520 666f   the possible fo
-0000f1b0: 6361 6c2d 7369 7374 6572 2073 7065 6369  cal-sister speci
-0000f1c0: 6573 2070 6169 7273 2c20 6f72 2075 7365  es pairs, or use
-0000f1d0: 7273 2063 616e 206d 616e 7561 6c6c 7920  rs can manually 
-0000f1e0: 7365 7420 7468 6520 7370 6563 6965 7320  set the species 
-0000f1f0: 7061 6972 7320 7669 6120 7468 6520 6f70  pairs via the op
-0000f200: 7469 6f6e 2060 2d2d 7370 6169 7260 2e20  tion `--spair`. 
-0000f210: 4e6f 7465 2074 6861 7420 6966 2074 6865  Note that if the
-0000f220: 2073 7065 6369 6573 2070 6169 7273 2077   species pairs w
-0000f230: 6572 6520 6d61 6e75 616c 6c79 2073 6574  ere manually set
-0000f240: 2c20 6974 2077 6f75 6c64 2062 6520 6e65  , it would be ne
-0000f250: 6564 6564 2074 6f20 7365 7420 7468 6520  eded to set the 
-0000f260: 6f70 7469 6f6e 2060 2d2d 636c 6173 7369  option `--classi
-0000f270: 6360 2e20 5468 6520 6f70 7469 6f6e 2060  c`. The option `
-0000f280: 6f6e 6c79 726f 6f74 6f75 7460 2063 616e  onlyrootout` can
-0000f290: 2062 6520 7365 7420 746f 206f 6e6c 7920   be set to only 
-0000f2a0: 636f 6e73 6964 6572 206f 7574 6772 6f75  consider outgrou
-0000f2b0: 7020 6174 2074 6865 2072 6f6f 742c 2069  p at the root, i
-0000f2c0: 6e73 7465 6164 206f 6620 616c 6c20 7468  nstead of all th
-0000f2d0: 6520 706f 7373 6962 6c65 206f 7574 6772  e possible outgr
-0000f2e0: 6f75 7073 2070 6572 2066 6f63 616c 2d73  oups per focal-s
-0000f2f0: 6973 7465 7220 7370 6563 6965 7320 7061  ister species pa
-0000f300: 6972 2c20 7768 6963 6820 6861 7320 696d  ir, which has im
-0000f310: 7061 6374 206f 6e20 7468 6520 6669 6e61  pact on the fina
-0000f320: 6c20 636f 7272 6563 7465 6420 2a4b 2a3c  l corrected *K*<
-0000f330: 7375 623e 533c 2f73 7562 3e20 6167 6573  sub>S</sub> ages
-0000f340: 2e20 536f 6d65 206f 7468 6572 206f 7074  . Some other opt
-0000f350: 696f 6e73 2077 6869 6368 2068 6176 6520  ions which have 
-0000f360: 6e6f 2069 6d70 6163 7420 6f6e 2074 6865  no impact on the
-0000f370: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-0000f380: 2062 7574 2061 6464 206d 6f72 6520 6c61   but add more la
-0000f390: 7965 7273 206f 7220 6368 616e 6765 2074  yers or change t
-0000f3a0: 6865 2061 7070 6561 7261 6e63 6520 6f6e  he appearance on
-0000f3b0: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
-0000f3c0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000f3d0: 6973 7472 6962 7574 696f 6e2c 2069 6e63  istribution, inc
-0000f3e0: 6c75 6465 2060 706c 6f74 6170 676d 6d60  lude `plotapgmm`
-0000f3f0: 2061 6e64 2060 706c 6f74 656c 6d6d 6020   and `plotelmm` 
-0000f400: 6574 632e 2054 6865 206f 7074 696f 6e20  etc. The option 
-0000f410: 6070 6c6f 7461 7067 6d6d 6020 6361 6e20  `plotapgmm` can 
-0000f420: 6265 2073 6574 2074 6f20 6361 6c6c 2074  be set to call t
-0000f430: 6865 2047 4d4d 2061 6e61 6c79 7369 7320  he GMM analysis 
-0000f440: 7570 6f6e 2074 6865 2061 6e63 686f 7220  upon the anchor 
-0000f450: 7061 6972 202a 4b2a 3c73 7562 3e53 3c2f  pair *K*<sub>S</
-0000f460: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000f470: 7574 696f 6e20 616e 6420 706c 6f74 2074  ution and plot t
-0000f480: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
-0000f490: 7375 6c74 2075 706f 6e20 7468 6520 6d69  sult upon the mi
-0000f4a0: 7865 6420 2a4b 2a3c 7375 623e 533c 2f73  xed *K*<sub>S</s
-0000f4b0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
-0000f4c0: 7469 6f6e 2c20 7768 6963 6820 6861 7320  tion, which has 
-0000f4d0: 746f 2062 6520 636f 2d73 6574 2077 6974  to be co-set wit
-0000f4e0: 6820 7468 6520 6f70 7469 6f6e 2060 2d2d  h the option `--
-0000f4f0: 616e 6368 6f72 706f 696e 7473 6020 7072  anchorpoints` pr
-0000f500: 6f76 6964 696e 6720 7468 6520 616e 6368  oviding the anch
-0000f510: 6f72 2070 6169 7273 2069 6e66 6f72 6d61  or pairs informa
-0000f520: 7469 6f6e 2e20 5468 6520 6f70 7469 6f6e  tion. The option
-0000f530: 2060 2d2d 706c 6f74 656c 6d6d 6020 6361   `--plotelmm` ca
-0000f540: 6e20 6265 2073 6574 2074 6f20 6361 6c6c  n be set to call
-0000f550: 2074 6865 2045 4c4d 4d20 616e 616c 7973   the ELMM analys
-0000f560: 6973 2075 706f 6e20 7468 6520 7061 7261  is upon the para
-0000f570: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-0000f580: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000f590: 6962 7574 696f 6e2e 0a0a 5468 6572 6520  ibution...There 
-0000f5a0: 6172 6520 3231 2063 6f6c 756d 6e73 2069  are 21 columns i
-0000f5b0: 6e20 7468 6520 7265 7375 6c74 2060 2a6b  n the result `*k
-0000f5c0: 732e 7473 7660 2066 696c 6520 6265 7369  s.tsv` file besi
-0000f5d0: 6465 7320 7468 6520 696e 6465 7820 636f  des the index co
-0000f5e0: 6c75 6d6e 7320 6070 6169 7260 2061 7320  lumns `pair` as 
-0000f5f0: 7468 6520 756e 6971 7565 2069 6465 6e74  the unique ident
-0000f600: 6966 6965 7220 666f 7220 6561 6368 2067  ifier for each g
-0000f610: 656e 6520 7061 6972 2e20 5468 6520 604e  ene pair. The `N
-0000f620: 602c 2060 5360 2c20 6064 4e60 2c20 6064  `, `S`, `dN`, `d
-0000f630: 4e2f 6453 602c 2060 6453 602c 2060 6c60  N/dS`, `dS`, `l`
-0000f640: 2061 6e64 2060 7460 2061 7265 2066 726f   and `t` are fro
-0000f650: 6d20 7468 6520 636f 6465 6d6c 2072 6573  m the codeml res
-0000f660: 756c 7473 2c20 7265 7072 6573 656e 7469  ults, representi
-0000f670: 6e67 2074 6865 204e 2065 7374 696d 6174  ng the N estimat
-0000f680: 652c 2074 6865 2053 2065 7374 696d 6174  e, the S estimat
-0000f690: 652c 2074 6865 2064 4e20 6573 7469 6d61  e, the dN estima
-0000f6a0: 7465 2c20 7468 6520 644e 2f64 5320 286f  te, the dN/dS (o
-0000f6b0: 6d65 6761 2920 6573 7469 6d61 7465 2c20  mega) estimate, 
-0000f6c0: 7468 6520 6453 2065 7374 696d 6174 652c  the dS estimate,
-0000f6d0: 2074 6865 206c 6f67 2d6c 696b 656c 6968   the log-likelih
-0000f6e0: 6f6f 6420 616e 6420 7468 6520 7420 6573  ood and the t es
-0000f6f0: 7469 6d61 7465 2c20 7265 7370 6563 7469  timate, respecti
-0000f700: 7665 6c79 2e20 5468 6520 6061 6c69 676e  vely. The `align
-0000f710: 6d65 6e74 636f 7665 7261 6765 602c 2060  mentcoverage`, `
-0000f720: 616c 6967 6e6d 656e 7469 6465 6e74 6974  alignmentidentit
-0000f730: 7960 2061 6e64 2060 616c 6967 6e6d 656e  y` and `alignmen
-0000f740: 746c 656e 6774 6860 2061 7265 2074 6865  tlength` are the
-0000f750: 2069 6e66 6f72 6d61 7469 6f6e 2070 6572   information per
-0000f760: 7461 696e 696e 6720 746f 2074 6865 2061  taining to the a
-0000f770: 6c69 676e 6d65 6e74 2066 6f72 2065 6163  lignment for eac
-0000f780: 6820 6661 6d69 6c79 2c20 7265 7072 6573  h family, repres
-0000f790: 656e 7469 6e67 2074 6865 2072 6174 696f  enting the ratio
-0000f7a0: 206f 6620 7468 6520 7374 7269 7070 6564   of the stripped
-0000f7b0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
-0000f7c0: 6820 636f 6d70 6172 6564 2074 6f20 7468  h compared to th
-0000f7d0: 6520 6675 6c6c 2061 6c69 676e 6d65 6e74  e full alignment
-0000f7e0: 206c 656e 6774 682c 2074 6865 2072 6174   length, the rat
-0000f7f0: 696f 206f 6620 636f 6c75 6d6e 7320 7769  io of columns wi
-0000f800: 7468 2069 6465 6e74 6963 616c 206e 7563  th identical nuc
-0000f810: 6c65 6f74 6964 6573 2063 6f6d 7061 7265  leotides compare
-0000f820: 6420 746f 2074 6865 206f 7665 7261 6c6c  d to the overall
-0000f830: 2063 6f6c 756d 6e73 206f 6620 7468 6520   columns of the 
-0000f840: 7374 7269 7070 6564 2061 6c69 676e 6d65  stripped alignme
-0000f850: 6e74 2c20 616e 6420 7468 6520 6c65 6e67  nt, and the leng
-0000f860: 7468 206f 6620 7468 6520 6675 6c6c 2061  th of the full a
-0000f870: 6c69 676e 6d65 6e74 2c20 7265 7370 6563  lignment, respec
-0000f880: 7469 7665 6c79 2e0a 2323 2320 7767 6420  tively..### wgd 
-0000f890: 6d69 780a 0a2a 2a54 6865 206d 6978 7475  mix..**The mixtu
-0000f8a0: 7265 206d 6f64 656c 2063 6c75 7374 6572  re model cluster
-0000f8b0: 696e 6720 616e 616c 7973 6973 206f 6620  ing analysis of 
-0000f8c0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000f8d0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000f8e0: 2a2a 0a60 6060 0a77 6764 206d 6978 206b  **.```.wgd mix k
-0000f8f0: 7364 6174 6120 282d 2d6e 5f69 6e69 7420  sdata (--n_init 
-0000f900: 3230 3020 2d2d 6d61 785f 6974 6572 2032  200 --max_iter 2
-0000f910: 3030 202d 2d6b 735f 7261 6e67 6520 3020  00 --ks_range 0 
-0000f920: 3520 2d2d 6669 6c74 6572 7320 3330 3020  5 --filters 300 
-0000f930: 2d2d 6269 6e73 2035 3020 2d2d 636f 6d70  --bins 50 --comp
-0000f940: 6f6e 656e 7473 2031 2034 202d 2d67 616d  onents 1 4 --gam
-0000f950: 6d61 2030 2e30 3031 290a 6060 600a 0a54  ma 0.001).```..T
-0000f960: 6869 7320 7061 7274 206f 6620 4761 7573  his part of Gaus
-0000f970: 7369 616e 206d 6978 7475 7265 206d 6f64  sian mixture mod
-0000f980: 656c 696e 6720 2847 4d4d 2920 616e 616c  eling (GMM) anal
-0000f990: 7973 6973 2069 7320 696e 6865 7269 7465  ysis is inherite
-0000f9a0: 6420 6672 6f6d 2074 6865 206f 7269 6769  d from the origi
-0000f9b0: 6e61 6c20 6077 6764 6020 7072 6f67 7261  nal `wgd` progra
-0000f9c0: 6d2c 2062 7574 2077 7269 7465 7320 6164  m, but writes ad
-0000f9d0: 6469 7469 6f6e 616c 6c79 2074 6865 2070  ditionally the p
-0000f9e0: 726f 6261 6269 6c69 7479 206f 6620 6561  robability of ea
-0000f9f0: 6368 202a 4b2a 3c73 7562 3e53 3c2f 7375  ch *K*<sub>S</su
-0000fa00: 623e 2076 616c 7565 2069 6e74 6f20 7468  b> value into th
-0000fa10: 6520 6669 6e61 6c20 6461 7461 6672 616d  e final datafram
-0000fa20: 652e 2042 6173 6963 616c 6c79 2c20 7573  e. Basically, us
-0000fa30: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-0000fa40: 6964 6520 7769 7468 2061 2028 6e6f 726d  ide with a (norm
-0000fa50: 616c 6c79 2066 726f 6d20 7768 6f6c 652d  ally from whole-
-0000fa60: 7061 7261 6e6f 6d65 206f 7220 616e 6368  paranome or anch
-0000fa70: 6f72 2d70 6169 7273 2920 2a4b 2a3c 7375  or-pairs) *K*<su
-0000fa80: 623e 533c 2f73 7562 3e20 6461 7461 6669  b>S</sub> datafi
-0000fa90: 6c65 2061 6e64 2074 6865 2047 4d4d 2061  le and the GMM a
-0000faa0: 6e61 6c79 7369 7320 7769 6c6c 2062 6520  nalysis will be 
-0000fab0: 636f 6e64 7563 7465 6420 7570 6f6e 2074  conducted upon t
-0000fac0: 6865 2064 6174 6166 696c 652e 2053 6f6d  he datafile. Som
-0000fad0: 6520 7061 7261 6d65 7465 7273 2063 616e  e parameters can
-0000fae0: 2061 6666 6563 7420 7468 6520 7265 7375   affect the resu
-0000faf0: 6c74 732c 2069 6e63 6c75 6469 6e67 2060  lts, including `
-0000fb00: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
-0000fb10: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-0000fb20: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-0000fb30: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
-0000fb40: 6175 6c74 2032 3030 292c 2060 2d2d 6d61  ault 200), `--ma
-0000fb50: 785f 6974 6572 602c 2077 6869 6368 2073  x_iter`, which s
-0000fb60: 6574 7320 7468 6520 6d61 7869 6d75 6d20  ets the maximum 
-0000fb70: 6e75 6d62 6572 206f 6620 6974 6572 6174  number of iterat
-0000fb80: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
-0000fb90: 3029 2c20 602d 2d6d 6574 686f 6460 2c20  0), `--method`, 
-0000fba0: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
-0000fbb0: 2077 6869 6368 2063 6c75 7374 6572 696e   which clusterin
-0000fbc0: 6720 6d65 7468 6f64 2074 6f20 7573 6520  g method to use 
-0000fbd0: 2864 6566 6175 6c74 2067 6d6d 292c 2060  (default gmm), `
-0000fbe0: 2d2d 6761 6d6d 6160 2c20 7768 6963 6820  --gamma`, which 
-0000fbf0: 7365 7473 2074 6865 2067 616d 6d61 2070  sets the gamma p
-0000fc00: 6172 616d 6574 6572 2066 6f72 2074 6865  arameter for the
-0000fc10: 2062 676d 6d20 6d6f 6465 6c20 2864 6566   bgmm model (def
-0000fc20: 6175 6c74 2030 2e30 3031 292c 2060 2d2d  ault 0.001), `--
-0000fc30: 636f 6d70 6f6e 656e 7473 602c 2077 6869  components`, whi
-0000fc40: 6368 2073 6574 7320 7468 6520 7261 6e67  ch sets the rang
-0000fc50: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-0000fc60: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-0000fc70: 2066 6974 2028 6465 6661 756c 7420 3120   fit (default 1 
-0000fc80: 3429 2c20 616e 6420 7468 6520 6461 7461  4), and the data
-0000fc90: 2066 696c 7465 7269 6e67 2070 6172 616d   filtering param
-0000fca0: 6574 6572 7320 602d 2d66 696c 7465 7273  eters `--filters
-0000fcb0: 6020 7768 6963 6820 6669 6c74 6572 7320  ` which filters 
-0000fcc0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
-0000fcd0: 6967 6e6d 656e 7420 6c65 6e67 7468 2c20  ignment length, 
-0000fce0: 602d 2d6b 735f 7261 6e67 6560 2077 6869  `--ks_range` whi
-0000fcf0: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
-0000fd00: 6261 7365 6420 6f6e 202a 4b2a 3c73 7562  based on *K*<sub
-0000fd10: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-0000fd20: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
-0000fd30: 7220 602d 2d62 696e 7360 2077 6869 6368  r `--bins` which
-0000fd40: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-0000fd50: 206f 6620 6269 6e73 2069 6e20 2a4b 2a3c   of bins in *K*<
-0000fd60: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-0000fd70: 7269 6275 7469 6f6e 2028 6465 6661 756c  ribution (defaul
-0000fd80: 7420 3530 292e 0a0a 2a2a 4120 7375 6767  t 50)...**A sugg
-0000fd90: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000fda0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-0000fdb0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-0000fdc0: 6f77 2a2a 0a0a 6060 600a 7767 6420 6d69  ow**..```.wgd mi
-0000fdd0: 7820 6b73 6461 7461 0a60 6060 0a0a 2323  x ksdata.```..##
-0000fde0: 2320 7767 6420 7065 616b 0a0a 2a2a 5468  # wgd peak..**Th
-0000fdf0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
-0000fe00: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
-0000fe10: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
-0000fe20: 2069 6e20 5747 4420 6461 7469 6e67 2a2a   in WGD dating**
-0000fe30: 0a60 6060 0a77 6764 2070 6561 6b20 6b73  .```.wgd peak ks
-0000fe40: 6461 7461 202d 6170 2061 7064 6174 6120  data -ap apdata 
-0000fe50: 2d73 6d20 736d 6461 7461 202d 6c65 206c  -sm smdata -le l
-0000fe60: 6564 6174 6120 2d6d 7020 6d70 6461 7461  edata -mp mpdata
-0000fe70: 202d 2d68 6575 7269 7374 6963 2028 2d2d   --heuristic (--
-0000fe80: 616c 6967 6e66 696c 7465 7220 302e 3020  alignfilter 0.0 
-0000fe90: 3020 302e 3020 2d2d 6b73 7261 6e67 6520  0 0.0 --ksrange 
-0000fea0: 3020 3520 2d2d 6269 6e5f 7769 6474 6820  0 5 --bin_width 
-0000feb0: 302e 3120 2d2d 6775 6964 6520 7365 676d  0.1 --guide segm
-0000fec0: 656e 7420 2d2d 7072 6f6d 696e 656e 6365  ent --prominence
-0000fed0: 5f63 7574 6f66 6620 302e 3120 2d2d 7265  _cutoff 0.1 --re
-0000fee0: 6c5f 6865 6967 6874 2030 2e34 202d 2d63  l_height 0.4 --c
-0000fef0: 6920 3935 202d 2d68 6472 2039 3520 2d2d  i 95 --hdr 95 --
-0000ff00: 6b73 6375 746f 6666 2035 290a 6060 600a  kscutoff 5).```.
-0000ff10: 0a41 7320 6d65 6e74 696f 6e65 6420 7072  .As mentioned pr
-0000ff20: 6576 696f 7573 6c79 2c20 6120 6865 7572  eviously, a heur
-0000ff30: 6973 7469 6320 6d65 7468 6f64 2061 6e64  istic method and
-0000ff40: 2061 2063 6f6c 6c69 6e65 6172 2073 6567   a collinear seg
-0000ff50: 6d65 6e74 732d 6775 6964 6564 2061 6e63  ments-guided anc
-0000ff60: 686f 7220 7061 6972 2063 6c75 7374 6572  hor pair cluster
-0000ff70: 696e 6720 666f 7220 7468 6520 7365 6172  ing for the sear
-0000ff80: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
-0000ff90: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000ffa0: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
-0000ffb0: 4420 6461 7469 6e67 2061 7265 2069 6d70  D dating are imp
-0000ffc0: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
-0000ffd0: 2076 3260 2e20 5573 6572 7320 6e65 6564   v2`. Users need
-0000ffe0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000fff0: 616e 6368 6f72 2070 6f69 6e74 732c 2073  anchor points, s
-00010000: 6567 6d65 6e74 732c 206c 6973 7473 6567  egments, listseg
-00010010: 6d65 6e74 732c 206d 756c 7469 706c 6963  ments, multiplic
-00010020: 6f6e 7061 6972 7320 6461 7461 6669 6c65  onpairs datafile
-00010030: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
-00010040: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-00010050: 636c 7573 7465 7269 6e67 2066 756e 6374  clustering funct
-00010060: 696f 6e2e 2053 6f6d 6520 7061 7261 6d65  ion. Some parame
-00010070: 7465 7273 2074 6861 7420 6361 6e20 696d  ters that can im
-00010080: 7061 6374 2074 6865 2072 6573 756c 7473  pact the results
-00010090: 2069 6e63 6c75 6465 2060 2d2d 616c 6967   include `--alig
-000100a0: 6e66 696c 7465 7260 2c20 7768 6963 6820  nfilter`, which 
-000100b0: 6669 6c74 6572 7320 7468 6520 6461 7461  filters the data
-000100c0: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
-000100d0: 656e 7420 6964 656e 7469 7479 2c20 6c65  ent identity, le
-000100e0: 6e67 7468 2061 6e64 2063 6f76 6572 6167  ngth and coverag
-000100f0: 652c 2060 2d2d 6b73 7261 6e67 6560 2c20  e, `--ksrange`, 
-00010100: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
-00010110: 616e 6765 206f 6620 4b73 2074 6f20 6265  ange of Ks to be
-00010120: 2061 6e61 6c79 7a65 642c 2060 2d2d 6269   analyzed, `--bi
-00010130: 6e5f 7769 6474 6860 2c20 7768 6963 6820  n_width`, which 
-00010140: 7365 7473 2074 6865 2062 616e 6477 6964  sets the bandwid
-00010150: 7468 206f 6620 2a4b 2a3c 7375 623e 533c  th of *K*<sub>S<
-00010160: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00010170: 6f6e 2c20 602d 2d77 6569 6768 7473 5f6f  on, `--weights_o
-00010180: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
-00010190: 6020 7768 6963 6820 6465 7465 726d 696e  ` which determin
-000101a0: 6573 2077 6865 7468 6572 2074 6f20 696e  es whether to in
-000101b0: 636c 7564 6520 2a4b 2a3c 7375 623e 533c  clude *K*<sub>S<
-000101c0: 2f73 7562 3e20 6f75 746c 6965 7273 2028  /sub> outliers (
-000101d0: 7768 6f73 6520 7661 6c75 6520 6973 206f  whose value is o
-000101e0: 7665 7220 3529 2069 6e20 616e 616c 7973  ver 5) in analys
-000101f0: 6973 2c20 602d 2d6d 6574 686f 6460 2077  is, `--method` w
-00010200: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-00010210: 7768 6963 6820 636c 7573 7465 7269 6e67  which clustering
-00010220: 206d 6574 686f 6420 746f 2075 7365 2028   method to use (
-00010230: 6465 6661 756c 7420 676d 6d29 2c20 602d  default gmm), `-
-00010240: 2d73 6565 6460 2077 6869 6368 2073 6574  -seed` which set
-00010250: 7320 7468 6520 7261 6e64 6f6d 2073 6565  s the random see
-00010260: 6420 6769 7665 6e20 746f 2069 6e69 7469  d given to initi
-00010270: 616c 697a 6174 696f 6e20 2864 6566 6175  alization (defau
-00010280: 6c74 2032 3335 3238 3930 292c 2060 2d2d  lt 2352890), `--
-00010290: 6e5f 696e 6974 602c 2077 6869 6368 2073  n_init`, which s
-000102a0: 6574 7320 7468 6520 6e75 6d62 6572 206f  ets the number o
-000102b0: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
-000102c0: 6c69 7a61 7469 6f6e 7320 2864 6566 6175  lizations (defau
-000102d0: 6c74 2032 3030 292c 2060 2d2d 656d 5f69  lt 200), `--em_i
-000102e0: 7465 7260 2c20 7768 6963 6820 7365 7473  ter`, which sets
-000102f0: 2074 6865 206d 6178 696d 756d 206e 756d   the maximum num
-00010300: 6265 7220 6f66 2069 7465 7261 7469 6f6e  ber of iteration
-00010310: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
-00010320: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
-00010330: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
-00010340: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-00010350: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
-00010360: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
-00010370: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
-00010380: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
-00010390: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-000103a0: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-000103b0: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
-000103c0: 3120 3429 2c20 602d 2d77 6569 6768 7465  1 4), `--weighte
-000103d0: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
-000103e0: 6e65 7320 7768 6574 6865 7220 746f 2075  nes whether to u
-000103f0: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00010400: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-00010410: 6564 756e 6461 6e63 792c 2060 2d2d 6775  edundancy, `--gu
-00010420: 6964 6560 2077 6869 6368 2064 6574 6572  ide` which deter
-00010430: 6d69 6e65 7320 7768 6963 6820 7265 6769  mines which regi
-00010440: 6d65 2072 6573 6964 696e 6720 616e 6368  me residing anch
-00010450: 6f72 7320 746f 2062 6520 7573 6564 2028  ors to be used (
-00010460: 6465 6661 756c 7420 7365 676d 656e 7429  default segment)
-00010470: 2c20 602d 2d70 726f 6d69 6e65 6e63 655f  , `--prominence_
-00010480: 6375 746f 6666 6020 7768 6963 6820 7365  cutoff` which se
-00010490: 7473 2074 6865 2070 726f 6d69 6e65 6e63  ts the prominenc
-000104a0: 6520 6375 746f 6666 206f 6620 6163 6365  e cutoff of acce
-000104b0: 7074 6162 6c65 2070 6561 6b73 2069 6e20  ptable peaks in 
-000104c0: 7065 616b 2066 696e 6469 6e67 2070 726f  peak finding pro
-000104d0: 6365 7373 2c20 602d 2d72 656c 5f68 6569  cess, `--rel_hei
-000104e0: 6768 7460 2077 6869 6368 2073 6574 7320  ght` which sets 
-000104f0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-00010500: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00010510: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00010520: 6561 7375 7265 642c 2060 2d2d 6b73 746f  easured, `--ksto
-00010530: 6461 7465 6020 7768 6963 6820 6d61 6e75  date` which manu
-00010540: 616c 6c79 2073 6574 7320 7468 6520 7261  ally sets the ra
-00010550: 6e67 6520 6f66 202a 4b2a 3c73 7562 3e53  nge of *K*<sub>S
-00010560: 3c2f 7375 623e 2074 6f20 6265 2064 6174  </sub> to be dat
-00010570: 6564 2069 6e20 6865 7572 6973 7469 6320  ed in heuristic 
-00010580: 7365 6172 6368 2061 6e64 206e 6565 6473  search and needs
-00010590: 2074 6f20 6265 2063 6f2d 7365 7420 7769   to be co-set wi
-000105a0: 7468 206f 7074 696f 6e20 602d 2d6d 616e  th option `--man
-000105b0: 7561 6c73 6574 602c 2060 2d2d 786c 696d  ualset`, `--xlim
-000105c0: 6020 616e 6420 602d 2d79 6c69 6d60 2064  ` and `--ylim` d
-000105d0: 6574 6572 6d69 6e69 6e67 2074 6865 2078  etermining the x
-000105e0: 2061 6e64 2079 2061 7869 7320 6c69 6d69   and y axis limi
-000105f0: 7420 6f66 2047 4d4d 202a 4b2a 3c73 7562  t of GMM *K*<sub
-00010600: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00010610: 7574 696f 6e2c 2060 2d2d 6369 6020 7365  ution, `--ci` se
-00010620: 7474 696e 6720 7468 6520 636f 6e66 6964  tting the confid
-00010630: 656e 6365 206c 6576 656c 206f 6620 6c6f  ence level of lo
-00010640: 672d 6e6f 726d 616c 2064 6973 7472 6962  g-normal distrib
-00010650: 7574 696f 6e20 746f 2064 6174 6520 2864  ution to date (d
-00010660: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
-00010670: 6472 6020 7365 7474 696e 6720 7468 6520  dr` setting the 
-00010680: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
-00010690: 7265 6769 6f6e 2028 4844 5229 2061 7070  region (HDR) app
-000106a0: 6c69 6564 2069 6e20 7468 6520 7365 676d  lied in the segm
-000106b0: 656e 742d 6775 6964 6564 2061 6e63 686f  ent-guided ancho
-000106c0: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-000106d0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
-000106e0: 696f 6e2c 2060 2d2d 6865 7572 6973 7469  ion, `--heuristi
-000106f0: 6360 2064 6574 6572 6d69 6e69 6e67 2077  c` determining w
-00010700: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00010710: 7465 2068 6575 7269 7374 6963 206d 6574  te heuristic met
-00010720: 686f 6420 6f66 2064 6566 696e 696e 6720  hod of defining 
-00010730: 4349 2066 6f72 2064 6174 696e 672c 2060  CI for dating, `
-00010740: 2d2d 6b73 6375 746f 6666 6020 7365 7474  --kscutoff` sett
-00010750: 696e 6720 7468 6520 2a4b 2a3c 7375 623e  ing the *K*<sub>
-00010760: 533c 2f73 7562 3e20 7361 7475 7261 7469  S</sub> saturati
-00010770: 6f6e 2063 7574 6f66 6620 696e 2064 6174  on cutoff in dat
-00010780: 696e 6720 2864 6566 6175 6c74 2035 2920  ing (default 5) 
-00010790: 616e 6420 602d 2d73 686f 7763 6960 2064  and `--showci` d
-000107a0: 6574 6572 6d69 6e69 6e67 2077 6865 7468  etermining wheth
-000107b0: 6572 2074 6f20 7368 6f77 2043 4920 666f  er to show CI fo
-000107c0: 7220 7468 6520 6f72 6967 696e 616c 2061  r the original a
-000107d0: 6e63 686f 7220 4b73 2047 4d4d 2061 6e61  nchor Ks GMM ana
-000107e0: 6c79 7369 732e 0a0a 2a2a 4120 7375 6767  lysis...**A sugg
-000107f0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-00010800: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-00010810: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00010820: 6f77 2a2a 0a0a 6060 600a 7767 6420 7065  ow**..```.wgd pe
-00010830: 616b 206b 7364 6174 6120 2d61 7020 6170  ak ksdata -ap ap
-00010840: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
-00010850: 2d6c 6520 6c65 6461 7461 202d 6d70 206d  -le ledata -mp m
-00010860: 7064 6174 6120 2d2d 6865 7572 6973 7469  pdata --heuristi
-00010870: 630a 6060 600a 0a23 2323 2077 6764 2073  c.```..### wgd s
-00010880: 796e 0a0a 2a2a 5468 6520 696e 7472 612d  yn..**The intra-
-00010890: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-000108a0: 2069 6e66 6572 656e 6365 2a2a 0a60 6060   inference**.```
-000108b0: 0a77 6764 2073 796e 2066 616d 696c 6965  .wgd syn familie
-000108c0: 7320 6766 6620 282d 2d6b 735f 6469 7374  s gff (--ks_dist
-000108d0: 7269 6275 7469 6f6e 206b 7364 6174 6120  ribution ksdata 
-000108e0: 2d66 2067 656e 6520 2d61 2049 4420 2d2d  -f gene -a ID --
-000108f0: 6d69 6e6c 656e 202d 3120 2d2d 6d69 6e73  minlen -1 --mins
-00010900: 6567 6c65 6e20 3130 3030 3020 2d2d 6d69  eglen 10000 --mi
-00010910: 6e67 656e 656e 756d 2033 3029 0a60 6060  ngenenum 30).```
-00010920: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
-00010930: 6764 2073 796e 6020 6973 206d 6169 6e6c  gd syn` is mainl
-00010940: 7920 6465 616c 696e 6720 7769 7468 2063  y dealing with c
-00010950: 6f6c 6c69 6e65 6172 6974 7920 6f72 2073  ollinearity or s
-00010960: 796e 7465 6e79 2028 626f 7468 2072 6566  ynteny (both ref
-00010970: 6572 7265 6420 746f 2061 7320 7379 6e74  erred to as synt
-00010980: 656e 7920 6865 7265 6166 7465 7229 2061  eny hereafter) a
-00010990: 6e61 6c79 7369 732e 2054 776f 2069 6e70  nalysis. Two inp
-000109a0: 7574 2066 696c 6573 2061 7265 2065 7373  ut files are ess
-000109b0: 656e 7469 616c 2c20 7468 6520 6765 6e65  ential, the gene
-000109c0: 2066 616d 696c 7920 6669 6c65 2061 6e64   family file and
-000109d0: 2074 6865 2067 6666 3320 6669 6c65 2e20   the gff3 file. 
-000109e0: 5468 6520 6765 6e65 2066 616d 696c 7920  The gene family 
-000109f0: 6669 6c65 2069 7320 696e 2074 6865 2066  file is in the f
-00010a00: 6f72 6d61 7420 6173 2060 4f72 7468 6f46  ormat as `OrthoF
-00010a10: 696e 6465 7260 2e20 5468 6520 736f 6674  inder`. The soft
-00010a20: 7761 7265 2060 692d 6164 686f 7265 6020  ware `i-adhore` 
-00010a30: 6973 2061 2070 7265 7265 7175 6973 6974  is a prerequisit
-00010a40: 652e 2057 6974 6820 6465 6661 756c 7420  e. With default 
-00010a50: 7061 7261 6d65 7465 7273 2c20 7468 6520  parameters, the 
-00010a60: 7072 6f67 7261 6d20 6261 7369 6361 6c6c  program basicall
-00010a70: 7920 636f 6e64 7563 7473 2031 2920 6669  y conducts 1) fi
-00010a80: 6c74 6572 696e 6720 6765 6e65 2066 616d  ltering gene fam
-00010a90: 696c 6965 7320 6261 7365 6420 6f6e 206d  ilies based on m
-00010aa0: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-00010ab0: 7a65 2032 2920 7265 7472 6965 7669 6e67  ze 2) retrieving
-00010ac0: 2067 656e 6520 706f 7369 7469 6f6e 2061   gene position a
-00010ad0: 6e64 2073 6361 6666 6f6c 6420 696e 666f  nd scaffold info
-00010ae0: 726d 6174 696f 6e20 6672 6f6d 2067 6666  rmation from gff
-00010af0: 3320 6669 6c65 2033 2920 7072 6f64 7563  3 file 3) produc
-00010b00: 696e 6720 7468 6520 636f 6e66 6967 7572  ing the configur
-00010b10: 6174 696f 6e20 6669 6c65 2061 6e64 2061  ation file and a
-00010b20: 7373 6f63 6961 7465 6420 6461 7461 6669  ssociated datafi
-00010b30: 6c65 7320 666f 7220 6069 2d61 6468 6f72  les for `i-adhor
-00010b40: 6560 2034 2920 6361 6c6c 696e 6720 6069  e` 4) calling `i
-00010b50: 2d61 6468 6f72 6560 2067 6976 656e 2074  -adhore` given t
-00010b60: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
-00010b70: 7420 746f 2069 6e66 6572 2073 796e 7465  t to infer synte
-00010b80: 6e79 2035 2920 7669 7375 616c 697a 696e  ny 5) visualizin
-00010b90: 6720 7468 6520 7379 6e74 656e 7920 696e  g the synteny in
-00010ba0: 2022 646f 7470 6c6f 7422 2069 6e20 7468   "dotplot" in th
-00010bb0: 6520 756e 6974 206f 6620 6765 6e65 7320  e unit of genes 
-00010bc0: 616e 6420 6261 7365 732c 2069 6e20 2253  and bases, in "S
-00010bd0: 796e 6465 7074 6822 2070 6c6f 7420 7368  yndepth" plot sh
-00010be0: 6f77 696e 6720 7468 6520 6469 7374 7269  owing the distri
-00010bf0: 6275 7469 6f6e 206f 6620 6469 6666 6572  bution of differ
-00010c00: 656e 7420 6361 7465 676f 7269 6573 206f  ent categories o
-00010c10: 6620 636f 6c6c 696e 6561 7269 7479 2072  f collinearity r
-00010c20: 6174 696f 7320 7769 7468 696e 2061 6e64  atios within and
-00010c30: 2062 6574 7765 656e 2073 7065 6369 6573   between species
-00010c40: 2c20 696e 2022 6475 7053 7461 636b 2220  , in "dupStack" 
-00010c50: 706c 6f74 2073 686f 7769 6e67 206d 756c  plot showing mul
-00010c60: 7469 706c 6963 6f6e 7320 7769 7468 2064  tiplicons with d
-00010c70: 6966 6665 7265 6e74 206d 756c 7469 706c  ifferent multipl
-00010c80: 6963 6174 696f 6e20 6c65 7665 6c73 2e20  ication levels. 
-00010c90: 3629 2069 6620 7769 7468 202a 4b2a 3c73  6) if with *K*<s
-00010ca0: 7562 3e53 3c2f 7375 623e 2064 6174 612c  ub>S</sub> data,
-00010cb0: 2061 2022 2a4b 2a3c 7375 623e 533c 2f73   a "*K*<sub>S</s
-00010cc0: 7562 3e20 646f 7470 6c6f 7422 2077 6974  ub> dotplot" wit
-00010cd0: 6820 646f 7473 2061 6e6e 6f74 6174 6564  h dots annotated
-00010ce0: 2069 6e20 2a4b 2a3c 7375 623e 533c 2f73   in *K*<sub>S</s
-00010cf0: 7562 3e20 7661 6c75 6573 2061 6e64 2061  ub> values and a
-00010d00: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00010d10: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
-00010d20: 7468 2061 6e63 686f 7220 7061 6972 7320  th anchor pairs 
-00010d30: 6465 6e6f 7465 6420 7769 6c6c 2062 6520  denoted will be 
-00010d40: 7072 6f64 7563 6564 2e20 5468 6520 6765  produced. The ge
-00010d50: 6e65 2069 6e66 6f72 6d61 7469 6f6e 2069  ne information i
-00010d60: 6e20 7468 6520 6765 6e65 2066 616d 696c  n the gene famil
-00010d70: 7920 6669 6c65 2061 6e64 2067 6666 3320  y file and gff3 
-00010d80: 6669 6c65 2073 686f 756c 6420 6265 206d  file should be m
-00010d90: 6174 6368 6564 2077 6869 6368 2072 6571  atched which req
-00010da0: 7569 7265 7320 7573 6572 7320 746f 2073  uires users to s
-00010db0: 6574 2070 726f 7065 7220 602d 2d66 6561  et proper `--fea
-00010dc0: 7475 7265 6020 616e 6420 602d 2d61 7474  ture` and `--att
-00010dd0: 7269 6275 7465 602e 2054 6865 206d 6178  ribute`. The max
-00010de0: 696d 756d 2066 616d 696c 7920 7369 7a65  imum family size
-00010df0: 2074 6f20 6265 2069 6e63 6c75 6465 6420   to be included 
-00010e00: 6361 6e20 6265 2073 6574 2076 6961 2074  can be set via t
-00010e10: 6865 206f 7074 696f 6e20 602d 2d6d 6178  he option `--max
-00010e20: 7369 7a65 602c 206e 6f74 6564 2074 6861  size`, noted tha
-00010e30: 7420 7468 6973 2066 696c 7465 7269 6e67  t this filtering
-00010e40: 2069 7320 6d61 696e 6c79 2074 6f20 6472   is mainly to dr
-00010e50: 6f70 2074 686f 7365 2068 7567 6520 7461  op those huge ta
-00010e60: 6e64 656d 2064 7570 6c69 6361 7465 7320  ndem duplicates 
-00010e70: 6661 6d69 6c79 2061 6e64 2074 7261 6e73  family and trans
-00010e80: 706f 7361 626c 6520 656c 656d 656e 7473  posable elements
-00010e90: 2028 5445 7329 2066 616d 696c 792c 2061   (TEs) family, a
-00010ea0: 6e64 206e 6f74 206d 616e 6461 746f 7279  nd not mandatory
-00010eb0: 2e20 5573 6572 7320 6361 6e20 6669 6c74  . Users can filt
-00010ec0: 6572 2074 686f 7365 2066 7261 676d 656e  er those fragmen
-00010ed0: 7461 7279 2073 6361 6666 6f6c 6473 2076  tary scaffolds v
-00010ee0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00010ef0: 2d6d 696e 6c65 6e60 2e20 5468 6520 6d69  -minlen`. The mi
-00010f00: 6e69 6d75 6d20 6c65 6e67 7468 2061 6e64  nimum length and
-00010f10: 206e 756d 6265 7220 6f66 2067 656e 6573   number of genes
-00010f20: 2066 6f72 2061 2073 6567 6d65 6e74 2074   for a segment t
-00010f30: 6f20 6265 2063 6f6e 7369 6465 7265 6420  o be considered 
-00010f40: 6361 6e20 6265 2073 6574 2076 6961 2074  can be set via t
-00010f50: 6865 206f 7074 696f 6e20 602d 2d6d 696e  he option `--min
-00010f60: 7365 676c 656e 6020 616e 6420 602d 2d6d  seglen` and `--m
-00010f70: 696e 6765 6e65 6e75 6d60 2e20 5265 6475  ingenenum`. Redu
-00010f80: 6e64 616e 7420 6d75 6c74 6970 6c69 636f  ndant multiplico
-00010f90: 6e73 2063 616e 2062 6520 6b65 7074 2062  ns can be kept b
-00010fa0: 7920 7365 7420 7468 6520 666c 6167 206f  y set the flag o
-00010fb0: 7074 696f 6e20 602d 2d6b 6565 7072 6564  ption `--keepred
-00010fc0: 756e 602e 0a0a 2a2a 4120 7375 6767 6573  un`...**A sugges
-00010fd0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
-00010fe0: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
-00010ff0: 2073 696d 706c 7920 6173 2062 656c 6f77   simply as below
-00011000: 2a2a 0a0a 6060 600a 7767 6420 7379 6e20  **..```.wgd syn 
-00011010: 6661 6d69 6c69 6573 2067 6666 0a60 6060  families gff.```
-00011020: 0a0a 2a2a 5468 6520 696e 7465 722d 7370  ..**The inter-sp
-00011030: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
-00011040: 6e66 6572 656e 6365 2a2a 0a60 6060 0a77  nference**.```.w
-00011050: 6764 2073 796e 2066 616d 696c 6965 7320  gd syn families 
-00011060: 6766 6631 2067 6666 3220 282d 2d61 6464  gff1 gff2 (--add
-00011070: 6974 696f 6e61 6c67 6666 696e 666f 2022  itionalgffinfo "
-00011080: 6d52 4e41 3b4e 616d 6522 202d 2d61 6464  mRNA;Name" --add
-00011090: 6974 696f 6e61 6c67 6666 696e 666f 2022  itionalgffinfo "
-000110a0: 6765 6e65 3b49 4422 290a 6060 600a 0a46  gene;ID").```..F
-000110b0: 6f72 206d 756c 7469 2d73 7065 6369 6573  or multi-species
-000110c0: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-000110d0: 6365 2c20 6966 2075 7365 7273 2068 6176  ce, if users hav
-000110e0: 6520 6766 6633 2066 696c 6573 2077 6869  e gff3 files whi
-000110f0: 6368 2068 6176 6520 6469 6666 6572 656e  ch have differen
-00011100: 7420 6665 6174 7572 6573 206f 7220 6174  t features or at
-00011110: 7472 6962 7574 6573 2066 6f72 2067 656e  tributes for gen
-00011120: 6520 706f 7369 7469 6f6e 2069 6e66 6f72  e position infor
-00011130: 6d61 7469 6f6e 2072 6574 7269 6576 616c  mation retrieval
-00011140: 2c20 7468 6520 6f70 7469 6f6e 2060 2d2d  , the option `--
-00011150: 6164 6469 7469 6f6e 616c 6766 6669 6e66  additionalgffinf
-00011160: 6f60 2063 616e 2062 6520 7365 7420 746f  o` can be set to
-00011170: 2070 726f 7669 6465 2074 6865 2061 6464   provide the add
-00011180: 6974 696f 6e61 6c20 696e 666f 726d 6174  itional informat
-00011190: 696f 6e2e 2054 6865 2072 656d 6169 6e69  ion. The remaini
-000111a0: 6e67 2070 6172 616d 6574 6572 2073 6574  ng parameter set
-000111b0: 7469 6e67 2069 7320 7468 6520 7361 6d65  ting is the same
-000111c0: 2061 7320 7468 6520 696e 7472 612d 7370   as the intra-sp
-000111d0: 6563 6966 6963 2073 796e 7465 6e79 2069  ecific synteny i
-000111e0: 6e66 6572 656e 6365 2e0a 0a2a 2a41 2073  nference...**A s
-000111f0: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
-00011200: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
-00011210: 6d6d 616e 6420 7369 6d70 6c79 2061 7320  mmand simply as 
-00011220: 6265 6c6f 772a 2a0a 0a60 6060 0a77 6764  below**..```.wgd
-00011230: 2073 796e 2066 616d 696c 6965 7320 6766   syn families gf
-00011240: 6631 2067 6666 320a 6060 600a 0a23 2323  f1 gff2.```..###
-00011250: 2077 6764 2076 697a 0a0a 2a2a 5468 6520   wgd viz..**The 
-00011260: 7669 7375 616c 697a 6174 696f 6e20 6f66  visualization of
-00011270: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00011280: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-00011290: 6e20 616e 6420 454c 4d4d 2061 6e61 6c79  n and ELMM analy
-000112a0: 7369 732a 2a0a 6060 600a 7767 6420 7669  sis**.```.wgd vi
-000112b0: 7a20 2d64 206b 7364 6174 610a 6060 600a  z -d ksdata.```.
-000112c0: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-000112d0: 6420 7669 7a60 2069 7320 6d61 696e 6c79  d viz` is mainly
-000112e0: 2066 6f72 2074 6865 2070 7572 706f 7365   for the purpose
-000112f0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
-00011300: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
-00011310: 2061 6e64 2073 796e 7465 6e79 2076 6973   and synteny vis
-00011320: 7561 6c69 7a61 7469 6f6e 2c20 7769 7468  ualization, with
-00011330: 2073 6f6d 6520 6f70 7469 6f6e 616c 206d   some optional m
-00011340: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00011350: 616e 616c 7973 6973 2e20 5468 6520 6261  analysis. The ba
-00011360: 7369 6320 6675 6e63 7469 6f6e 2069 7320  sic function is 
-00011370: 6a75 7374 2074 6f20 706c 6f74 2074 6865  just to plot the
-00011380: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00011390: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
-000113a0: 6420 636f 6e64 7563 7420 616e 2045 4c4d  d conduct an ELM
-000113b0: 4d20 616e 616c 7973 6973 2069 6e20 7365  M analysis in se
-000113c0: 6172 6368 206f 6620 706f 7465 6e74 6961  arch of potentia
-000113d0: 6c20 5747 4420 636f 6d70 6f6e 656e 7473  l WGD components
-000113e0: 2e20 536f 6d65 206b 6579 2070 6172 616d  . Some key param
-000113f0: 6574 6572 7320 6166 6665 6374 696e 6720  eters affecting 
-00011400: 7468 6520 454c 4d4d 2072 6573 756c 7420  the ELMM result 
-00011410: 696e 636c 7564 6520 602d 2d70 726f 6d69  include `--promi
-00011420: 6e65 6e63 655f 6375 746f 6666 602c 2060  nence_cutoff`, `
-00011430: 2d2d 7265 6c5f 6865 6967 6874 602c 2060  --rel_height`, `
-00011440: 2d2d 656d 5f69 7465 7261 7469 6f6e 7360  --em_iterations`
-00011450: 2061 6e64 2060 2d2d 656d 5f69 6e69 7469   and `--em_initi
-00011460: 616c 697a 6174 696f 6e73 602c 2061 6c6c  alizations`, all
-00011470: 206f 6620 7768 6963 6820 6861 7665 2062   of which have b
-00011480: 6565 6e20 6578 706c 6169 6e65 6420 6168  een explained ah
-00011490: 6561 642e 0a0a 2a2a 4120 7375 6767 6573  ead...**A sugges
-000114a0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
-000114b0: 2063 616e 2075 7365 2063 6f6d 6d61 6e64   can use command
-000114c0: 2073 696d 706c 7920 6173 2062 656c 6f77   simply as below
-000114d0: 2a2a 0a0a 6060 600a 7767 6420 7669 7a20  **..```.wgd viz 
-000114e0: 2d64 206b 7364 6174 610a 6060 600a 0a2a  -d ksdata.```..*
-000114f0: 2a54 6865 2076 6973 7561 6c69 7a61 7469  *The visualizati
-00011500: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
-00011510: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00011520: 6275 7469 6f6e 2077 6974 6820 7261 7465  bution with rate
-00011530: 2063 6f72 7265 6374 696f 6e2a 2a0a 6060   correction**.``
-00011540: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
-00011550: 6174 6120 2d73 7020 7370 6461 7461 202d  ata -sp spdata -
-00011560: 2d66 6f63 7573 3261 6c6c 2066 6f63 616c  -focus2all focal
-00011570: 5f73 7065 6369 6573 202d 2d65 7874 7261  _species --extra
-00011580: 7061 7261 6e6f 6d65 6b73 206b 7364 6174  paranomeks ksdat
-00011590: 6120 282d 2d61 6e63 686f 7270 6f69 6e74  a (--anchorpoint
-000115a0: 7320 6170 6461 7461 202d 2d70 6c6f 7461  s apdata --plota
-000115b0: 7067 6d6d 202d 2d70 6c6f 7465 6c6d 6d29  pgmm --plotelmm)
-000115c0: 0a60 6060 0a0a 4265 7369 6465 7320 7468  .```..Besides th
-000115d0: 6520 6261 7369 6320 2a4b 2a3c 7375 623e  e basic *K*<sub>
-000115e0: 533c 2f73 7562 3e20 706c 6f74 2c20 7375  S</sub> plot, su
-000115f0: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
-00011600: 636f 7272 6563 7469 6f6e 2063 616e 2061  correction can a
-00011610: 6c73 6f20 6265 2061 6368 6965 7665 6420  lso be achieved 
-00011620: 6769 7665 6e20 6174 206c 6561 7374 2061  given at least a
-00011630: 2073 7065 6369 6573 2074 7265 6520 2876   species tree (v
-00011640: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00011650: 2d73 7065 6369 6573 7472 6565 6029 2061  -speciestree`) a
-00011660: 6e64 2061 2066 6f63 616c 2073 7065 6369  nd a focal speci
-00011670: 6573 2028 6569 7468 6572 2076 6961 2074  es (either via t
-00011680: 6865 206f 7074 696f 6e20 602d 2d66 6f63  he option `--foc
-00011690: 7573 3261 6c6c 6020 6f72 2076 6961 2074  us2all` or via t
-000116a0: 6865 206f 7074 696f 6e20 602d 2d73 7061  he option `--spa
-000116b0: 6972 6020 696e 2074 6865 2066 6f72 6d20  ir` in the form 
-000116c0: 6f66 2022 2466 6f63 616c 5f73 7065 6369  of "$focal_speci
-000116d0: 6573 3b24 666f 6361 6c5f 7370 6563 6965  es;$focal_specie
-000116e0: 7322 292e 2049 7427 7320 7375 6767 6573  s"). It's sugges
-000116f0: 7465 6420 7468 6174 2074 6865 206f 7274  ted that the ort
-00011700: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00011710: 3e53 3c2f 7375 623e 2064 6174 6120 6973  >S</sub> data is
-00011720: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
-00011730: 2060 2d2d 6461 7461 6669 6c65 6020 6f70   `--datafile` op
-00011740: 7469 6f6e 2061 6e64 2074 6865 2070 6172  tion and the par
-00011750: 616c 6f67 6f75 7320 2a4b 2a3c 7375 623e  alogous *K*<sub>
-00011760: 533c 2f73 7562 3e20 6461 7461 2069 7320  S</sub> data is 
-00011770: 7072 6f76 6964 6564 2062 7920 7468 6520  provided by the 
-00011780: 602d 2d65 7874 7261 7061 7261 6e6f 6d65  `--extraparanome
-00011790: 6b73 6020 6f70 7469 6f6e 2c20 616c 7468  ks` option, alth
-000117a0: 6f75 6768 2069 7427 7320 616c 6c6f 7765  ough it's allowe
-000117b0: 6420 746f 206f 6e6c 7920 7072 6f76 6964  d to only provid
-000117c0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000117d0: 3e20 6461 7461 2076 6961 2074 6865 2060  > data via the `
-000117e0: 2d2d 6461 7461 6669 6c65 6020 6f70 7469  --datafile` opti
-000117f0: 6f6e 2061 6e64 2064 6570 6f73 6974 2062  on and deposit b
-00011800: 6f74 6820 6f72 7468 6f6c 6f67 6f75 7326  oth orthologous&
-00011810: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-00011820: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
-00011830: 7468 6572 656f 6e2e 2054 6865 7265 2061  thereon. There a
-00011840: 7265 2074 776f 2074 7970 6573 206f 6620  re two types of 
-00011850: 6d69 7865 6420 706c 6f74 7320 2874 6865  mixed plots (the
-00011860: 2022 6d69 7865 6422 2068 6572 6520 7265   "mixed" here re
-00011870: 6665 7273 2074 6f20 6d69 7865 6420 6f72  fers to mixed or
-00011880: 7468 6f6c 6f67 6f75 7320 616e 6420 7061  thologous and pa
-00011890: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-000118a0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-000118b0: 7574 696f 6e73 292c 206f 6e65 206f 6620  utions), one of 
-000118c0: 7768 6963 6820 6973 2073 696d 696c 6172  which is similar
-000118d0: 2074 6f20 7768 6174 2060 6b73 7261 7465   to what `ksrate
-000118e0: 7360 2070 6c6f 7473 2c20 7768 696c 6520  s` plots, while 
-000118f0: 7468 6520 6f74 6865 7220 6f66 2077 6869  the other of whi
-00011900: 6368 2069 7320 6c69 6b65 2074 6865 2063  ch is like the c
-00011910: 6f6e 7665 6e74 696f 6e61 6c20 2a4b 2a3c  onventional *K*<
-00011920: 7375 623e 533c 2f73 7562 3e20 706c 6f74  sub>S</sub> plot
-00011930: 2074 6861 7420 626f 7468 2074 6865 206f   that both the o
-00011940: 7269 6769 6e61 6c20 6f72 7468 6f6c 6f67  riginal ortholog
-00011950: 6f75 7320 616e 6420 7061 7261 6c6f 676f  ous and paralogo
-00011960: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-00011970: 623e 2064 6973 7472 6962 7574 696f 6e73  b> distributions
-00011980: 2061 7265 2074 7275 7468 6675 6c6c 7920   are truthfully 
-00011990: 706c 6f74 7465 6420 696e 7374 6561 6420  plotted instead 
-000119a0: 6f66 206a 7573 7420 6265 696e 6720 7265  of just being re
-000119b0: 7072 6573 656e 7465 6420 6279 2073 6f6d  presented by som
-000119c0: 6520 7665 7274 6963 616c 206c 696e 6573  e vertical lines
-000119d0: 2e20 5765 2073 7567 6765 7374 2075 7365  . We suggest use
-000119e0: 7273 2074 6f20 6164 6f70 7420 7468 6520  rs to adopt the 
-000119f0: 606b 7372 6174 6573 602d 6c69 6b65 2070  `ksrates`-like p
-00011a00: 6c6f 7473 2c20 7768 6963 6820 6973 2074  lots, which is t
-00011a10: 6865 2064 6566 6175 6c74 2e20 4f74 6865  he default. Othe
-00011a20: 7277 6973 6520 7468 6520 6f70 7469 6f6e  rwise the option
-00011a30: 2060 2d2d 636c 6173 7369 6360 2077 696c   `--classic` wil
-00011a40: 6c20 6265 206e 6565 6465 6420 746f 2073  l be needed to s
-00011a50: 6574 2e20 4578 7472 6120 6d69 7874 7572  et. Extra mixtur
-00011a60: 6520 6d6f 6465 6c69 6e67 2061 6e61 6c79  e modeling analy
-00011a70: 7369 7320 6361 6e20 6265 2069 6e69 7469  sis can be initi
-00011a80: 6174 6564 2076 6961 2074 6865 206f 7074  ated via the opt
-00011a90: 696f 6e20 602d 2d70 6c6f 7465 6c6d 6d60  ion `--plotelmm`
-00011aa0: 2061 6e64 2060 2d2d 706c 6f74 6170 676d   and `--plotapgm
-00011ab0: 6d60 2077 6974 6820 7468 6520 616e 6368  m` with the anch
-00011ac0: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
-00011ad0: 6c65 2070 726f 7669 6465 6420 6279 2074  le provided by t
-00011ae0: 6865 2060 2d2d 616e 6368 6f72 706f 696e  he `--anchorpoin
-00011af0: 7473 6020 6f70 7469 6f6e 2e0a 0a2a 2a41  ts` option...**A
-00011b00: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
-00011b10: 696e 6720 7275 6e20 6361 6e20 7573 6520  ing run can use 
-00011b20: 636f 6d6d 616e 6420 7369 6d70 6c79 2061  command simply a
-00011b30: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-00011b40: 6764 2076 697a 202d 6420 6b73 6461 7461  gd viz -d ksdata
-00011b50: 202d 7370 2073 7064 6174 6120 2d2d 666f   -sp spdata --fo
-00011b60: 6375 7332 616c 6c20 666f 6361 6c5f 7370  cus2all focal_sp
-00011b70: 6563 6965 7320 2d2d 6578 7472 6170 6172  ecies --extrapar
-00011b80: 616e 6f6d 656b 7320 6b73 6461 7461 0a60  anomeks ksdata.`
-00011b90: 6060 0a0a 2a2a 5468 6520 7669 7375 616c  ``..**The visual
-00011ba0: 697a 6174 696f 6e20 6f66 2073 796e 7465  ization of synte
-00011bb0: 6e79 2a2a 0a60 6060 0a77 6764 2076 697a  ny**.```.wgd viz
-00011bc0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
-00011bd0: 736d 6461 7461 202d 6d74 206d 7464 6174  smdata -mt mtdat
-00011be0: 6120 2d67 7420 6774 6461 7461 202d 2d70  a -gt gtdata --p
-00011bf0: 6c6f 7473 796e 2028 2d2d 6d69 6e6c 656e  lotsyn (--minlen
-00011c00: 202d 3120 2d2d 6d69 6e73 6567 6c65 6e20   -1 --minseglen 
-00011c10: 3130 3030 3020 2d2d 6d69 6e67 656e 656e  10000 --mingenen
-00011c20: 756d 2033 3029 0a60 6060 0a0a 436f 6d70  um 30).```..Comp
-00011c30: 6172 6564 2074 6f20 7468 6520 6f72 6967  ared to the orig
-00011c40: 696e 616c 2060 7767 6460 2c20 7468 6520  inal `wgd`, the 
-00011c50: 6077 6764 2076 697a 6020 7072 6f67 7261  `wgd viz` progra
-00011c60: 6d20 6164 6465 6420 7379 6e74 656e 7920  m added synteny 
-00011c70: 7669 7375 616c 697a 6174 696f 6e20 7069  visualization pi
-00011c80: 7065 6c69 6e65 2e20 5573 6572 7320 6e65  peline. Users ne
-00011c90: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
-00011ca0: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
-00011cb0: 2d70 6c6f 7473 796e 6020 746f 2069 6e69  -plotsyn` to ini
-00011cc0: 7469 6174 6520 7468 6973 2070 6172 7420  tiate this part 
-00011cd0: 6f66 2070 6970 656c 696e 652e 2054 6869  of pipeline. Thi
-00011ce0: 7320 7374 6570 2061 7373 756d 6573 2074  s step assumes t
-00011cf0: 6861 7420 7573 6572 7320 6861 7665 206f  hat users have o
-00011d00: 6274 6169 6e65 6420 616c 7265 6164 7920  btained already 
-00011d10: 7468 6520 7379 6e74 656e 6963 2072 6573  the syntenic res
-00011d20: 756c 7473 2066 726f 6d20 6069 2d61 6468  ults from `i-adh
-00011d30: 6f72 6560 2061 6e64 2075 7365 7320 7468  ore` and uses th
-00011d40: 6f73 6520 7265 7375 6c74 2066 696c 6573  ose result files
-00011d50: 2074 6f20 7265 616c 697a 6520 7468 6520   to realize the 
-00011d60: 7379 6e74 656e 7920 7669 7375 616c 697a  synteny visualiz
-00011d70: 6174 696f 6e2e 2053 696d 6c69 6172 2074  ation. Simliar t
-00011d80: 6f20 6077 6764 2073 796e 602c 2074 6865  o `wgd syn`, the
-00011d90: 2065 7874 7261 202a 4b2a 3c73 7562 3e53   extra *K*<sub>S
-00011da0: 3c2f 7375 623e 2064 6174 6120 6669 6c65  </sub> data file
-00011db0: 2063 616e 2062 6520 7472 616e 736d 6974   can be transmit
-00011dc0: 7465 6420 7669 6120 7468 6520 6f70 7469  ted via the opti
-00011dd0: 6f6e 2060 2d2d 6461 7461 6669 6c65 6020  on `--datafile` 
-00011de0: 2869 6e73 7465 6164 206f 6620 602d 2d65  (instead of `--e
-00011df0: 7874 7261 7061 7261 6e6f 6d65 6b73 6020  xtraparanomeks` 
-00011e00: 6f70 7469 6f6e 292e 2042 6173 6963 616c  option). Basical
-00011e10: 6c79 2c20 7468 6520 7379 6e74 656e 6963  ly, the syntenic
-00011e20: 2072 6573 756c 7420 6669 6c65 7320 7265   result files re
-00011e30: 7175 6972 6564 2061 7265 2074 6865 2061  quired are the a
-00011e40: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
-00011e50: 6166 696c 652c 206d 756c 7469 706c 6963  afile, multiplic
-00011e60: 6f6e 7320 6461 7461 6669 6c65 2c20 6765  ons datafile, ge
-00011e70: 6e65 2074 6162 6c65 2064 6174 6166 696c  ne table datafil
-00011e80: 6520 2861 7574 6f6d 6174 656c 7920 7072  e (automately pr
-00011e90: 6f64 7563 6564 2062 7920 6077 6764 2073  oduced by `wgd s
-00011ea0: 796e 6029 2c20 616e 6420 7365 676d 656e  yn`), and segmen
-00011eb0: 7473 2064 6174 6166 696c 652e 0a0a 2a2a  ts datafile...**
-00011ec0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-00011ed0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-00011ee0: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
-00011ef0: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
-00011f00: 7767 6420 7669 7a20 2d61 7020 6170 6461  wgd viz -ap apda
-00011f10: 7461 202d 736d 2073 6d64 6174 6120 2d6d  ta -sm smdata -m
-00011f20: 7420 6d74 6461 7461 202d 6774 2067 7464  t mtdata -gt gtd
-00011f30: 6174 6120 2d2d 706c 6f74 7379 6e0a 6060  ata --plotsyn.``
-00011f40: 600a 0a23 2320 496c 6c75 7374 7261 7469  `..## Illustrati
-00011f50: 6f6e 0a0a 5765 2069 6c6c 7573 7472 6174  on..We illustrat
-00011f60: 6520 6f75 7220 7072 6f67 7261 6d20 6f6e  e our program on
-00011f70: 2061 6e20 6578 656d 706c 6172 7920 5747   an exemplary WG
-00011f80: 4420 696e 6665 7265 6e63 6520 616e 6420  D inference and 
-00011f90: 6461 7469 6e67 2075 706f 6e20 7370 6563  dating upon spec
-00011fa0: 6965 7320 2a41 7175 696c 6567 6961 2063  ies *Aquilegia c
-00011fb0: 6f65 7275 6c65 612a 2e0a 0a54 6865 202a  oerulea*...The *
-00011fc0: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00011fd0: 6561 2a20 7761 7320 7265 706f 7274 6564  ea* was reported
-00011fe0: 2074 6f20 6578 7065 7269 656e 6365 2061   to experience a
-00011ff0: 6e20 7061 6c65 6f2d 706f 6c79 706c 6f69  n paleo-polyploi
-00012000: 6469 7a61 7469 6f6e 2065 7665 6e74 2061  dization event a
-00012010: 6674 6572 2074 6865 2064 6976 6572 6765  fter the diverge
-00012020: 6e63 6520 6f66 2063 6f72 6520 6575 6469  nce of core eudi
-00012030: 636f 7473 2c20 7768 6963 6820 6973 206c  cots, which is l
-00012040: 696b 656c 7920 7368 6172 6564 2062 7920  ikely shared by 
-00012050: 616c 6c20 5261 6e75 6e63 756c 616c 6573  all Ranunculales
-00012060: 2e0a 0a46 6972 7374 2061 626f 7665 2061  ...First above a
-00012070: 6c6c 2c20 6c65 7427 7320 6465 6c69 6e65  ll, let's deline
-00012080: 6174 6520 7468 6520 7768 6f6c 6520 7061  ate the whole pa
-00012090: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-000120a0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-000120b0: 6962 7574 696f 6e20 616e 6420 6861 7665  ibution and have
-000120c0: 2061 2062 6173 6963 206f 6273 6572 7661   a basic observa
-000120d0: 7469 6f6e 2066 6f72 2070 6f74 656e 7469  tion for potenti
-000120e0: 616c 6c79 2063 6f6e 6365 6976 6162 6c65  ally conceivable
-000120f0: 2057 4744 732c 2075 7369 6e67 2074 6865   WGDs, using the
-00012100: 2063 6f6d 6d61 6e64 206c 696e 6520 6265   command line be
-00012110: 6c6f 772e 0a0a 6060 600a 7767 6420 646d  low...```.wgd dm
-00012120: 6420 4171 7569 6c65 6769 615f 636f 6572  d Aquilegia_coer
-00012130: 756c 6561 0a77 6764 206b 7364 2077 6764  ulea.wgd ksd wgd
-00012140: 5f64 6d64 2f41 7175 696c 6567 6961 5f63  _dmd/Aquilegia_c
-00012150: 6f65 7275 6c65 612e 7473 7620 4171 7569  oerulea.tsv Aqui
-00012160: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
-00012170: 6060 0a0a 5468 6520 636f 6e73 7472 7563  ``..The construc
-00012180: 7465 6420 7768 6f6c 6520 7061 7261 6e6f  ted whole parano
-00012190: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-000121a0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000121b0: 696f 6e20 6f66 202a 4171 7569 6c65 6769  ion of *Aquilegi
-000121c0: 6120 636f 6572 756c 6561 2a20 6973 2061  a coerulea* is a
-000121d0: 7320 6265 6c6f 772c 2077 6520 6361 6e20  s below, we can 
-000121e0: 7365 6520 7468 6174 2074 6865 7265 2073  see that there s
-000121f0: 6565 6d73 2074 6f20 6265 2061 2068 756d  eems to be a hum
-00012200: 7020 6174 202a 4b2a 3c73 7562 3e53 3c2f  p at *K*<sub>S</
-00012210: 7375 623e 2031 2062 7574 206e 6f74 2063  sub> 1 but not c
-00012220: 6c65 6172 2e0a 0a21 5b5d 2864 6174 612f  lear...![](data/
-00012230: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012240: 6561 2e74 7376 2e6b 7364 5f77 702e 7376  ea.tsv.ksd_wp.sv
-00012250: 6729 0a0a 5765 2074 6865 6e20 636f 6e73  g)..We then cons
-00012260: 7472 7563 7420 7468 6520 616e 6368 6f72  truct the anchor
-00012270: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012280: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-00012290: 6e20 7573 696e 6720 7468 6520 636f 6d6d  n using the comm
-000122a0: 616e 6420 6c69 6e65 2062 656c 6f77 2e0a  and line below..
-000122b0: 0a60 6060 0a77 6764 2073 796e 202d 6620  .```.wgd syn -f 
-000122c0: 6d52 4e41 202d 6120 4e61 6d65 2077 6764  mRNA -a Name wgd
-000122d0: 5f64 6d64 2f41 7175 696c 6567 6961 5f63  _dmd/Aquilegia_c
-000122e0: 6f65 7275 6c65 612e 7473 7620 4171 7569  oerulea.tsv Aqui
-000122f0: 6c65 6769 615f 636f 6572 756c 6561 2e67  legia_coerulea.g
-00012300: 6666 3320 2d6b 7320 7767 645f 6b73 642f  ff3 -ks wgd_ksd/
-00012310: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012320: 6561 2e74 7376 2e6b 732e 7473 760a 6060  ea.tsv.ks.tsv.``
-00012330: 600a 0a41 7320 7368 6f77 6e20 6265 6c6f  `..As shown belo
-00012340: 772c 2074 6865 7265 2061 7265 2073 6f6d  w, there are som
-00012350: 6520 7265 7461 696e 6564 2061 6e63 686f  e retained ancho
-00012360: 7220 7061 6972 7320 7769 7468 202a 4b2a  r pairs with *K*
-00012370: 3c73 7562 3e53 3c2f 7375 623e 2062 6574  <sub>S</sub> bet
-00012380: 7765 656e 2031 2061 6e64 2032 2c20 7768  ween 1 and 2, wh
-00012390: 6963 6820 7365 656d 7320 746f 2073 7567  ich seems to sug
-000123a0: 6765 7374 2061 2057 4744 2065 7665 6e74  gest a WGD event
-000123b0: 2e0a 0a21 5b5d 2864 6174 612f 4171 7569  ...![](data/Aqui
-000123c0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-000123d0: 7376 2e6b 7364 5f77 705f 6170 2e73 7667  sv.ksd_wp_ap.svg
-000123e0: 290a 0a54 6865 2061 7373 6f63 6961 7465  )..The associate
-000123f0: 6420 6064 7570 5374 6163 6b60 2070 6c6f  d `dupStack` plo
-00012400: 7420 7368 6f77 7320 7468 6174 2074 6865  t shows that the
-00012410: 7265 2061 7265 206e 756d 6572 6f75 7320  re are numerous 
-00012420: 6475 706c 6963 6174 6564 2073 6567 6d65  duplicated segme
-00012430: 6e74 7320 6163 726f 7373 206d 6f73 7420  nts across most 
-00012440: 6f66 2074 6865 2063 6872 6f6d 6f73 6f6d  of the chromosom
-00012450: 6573 2e0a 0a21 5b5d 2864 6174 612f 4171  es...![](data/Aq
-00012460: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012470: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
-00012480: 6c65 615f 6d75 6c74 6970 6c69 636f 6e73  lea_multiplicons
-00012490: 5f6c 6576 656c 2e73 7667 290a 0a57 6520  _level.svg)..We 
-000124a0: 696d 706c 656d 656e 7465 6420 7477 6f20  implemented two 
-000124b0: 7479 7065 7320 6f66 2064 6f74 2070 6c6f  types of dot plo
-000124c0: 7473 2069 6e20 6f78 666f 7264 2067 7269  ts in oxford gri
-000124d0: 643a 206f 6e65 2069 6e20 7468 6520 756e  d: one in the un
-000124e0: 6974 206f 6620 6261 7365 7320 616e 6420  it of bases and 
-000124f0: 7468 6520 6f74 6865 7220 696e 2074 6865  the other in the
-00012500: 2075 6e69 7420 6f66 2067 656e 6573 2c20   unit of genes, 
-00012510: 7768 6963 6820 6361 6e20 6265 2063 6f6c  which can be col
-00012520: 6f72 6564 2062 7920 2a4b 2a3c 7375 623e  ored by *K*<sub>
-00012530: 533c 2f73 7562 3e20 7661 6c75 6573 2067  S</sub> values g
-00012540: 6976 656e 202a 4b2a 3c73 7562 3e53 3c2f  iven *K*<sub>S</
-00012550: 7375 623e 2064 6174 612e 0a0a 215b 5d28  sub> data...![](
-00012560: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
-00012570: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
-00012580: 6567 6961 5f63 6f65 7275 6c65 615f 4b73  egia_coerulea_Ks
-00012590: 2e64 6f74 5f75 6e69 745f 6765 6e65 2e70  .dot_unit_gene.p
-000125a0: 6e67 290a 0a41 7320 7368 6f77 6e20 6162  ng)..As shown ab
-000125b0: 6f76 652c 2074 6865 2064 6f74 2070 6c6f  ove, the dot plo
-000125c0: 7420 696e 2074 6865 2075 6e69 7420 6f66  t in the unit of
-000125d0: 2067 656e 6573 2070 7265 7365 6e74 7320   genes presents 
-000125e0: 6e75 6d65 726f 7573 2064 656e 7365 6c79  numerous densely
-000125f0: 2061 6767 7265 6761 7465 6420 286c 696e   aggregated (lin
-00012600: 652d 6c69 6b65 2920 616e 6368 6f72 2070  e-like) anchor p
-00012610: 6f69 6e74 7320 6174 206d 6f73 7420 6f66  oints at most of
-00012620: 2074 6865 2063 6872 6f6d 6f73 6f6d 6573   the chromosomes
-00012630: 2077 6974 6820 636f 6e73 6973 7465 6e74   with consistent
-00012640: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012650: 2061 6765 2062 6574 7765 656e 2031 2061   age between 1 a
-00012660: 6e64 2032 2e20 5468 6520 646f 7420 706c  nd 2. The dot pl
-00012670: 6f74 2069 6e20 7468 6520 756e 6974 206f  ot in the unit o
-00012680: 6620 6261 7365 7320 7368 6f77 7320 7468  f bases shows th
-00012690: 6520 7361 6d65 2070 6174 7465 726e 2c20  e same pattern, 
-000126a0: 6173 206d 616e 6966 6573 7465 6420 6265  as manifested be
-000126b0: 6c6f 772e 0a0a 215b 5d28 6461 7461 2f41  low...![](data/A
-000126c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000126d0: 612d 7673 2d41 7175 696c 6567 6961 5f63  a-vs-Aquilegia_c
-000126e0: 6f65 7275 6c65 615f 4b73 2e64 6f74 2e70  oerulea_Ks.dot.p
-000126f0: 6e67 290a 0a54 6865 2064 6f74 2070 6c6f  ng)..The dot plo
-00012700: 7473 2077 6974 686f 7574 202a 4b2a 3c73  ts without *K*<s
-00012710: 7562 3e53 3c2f 7375 623e 2061 6e6e 6f74  ub>S</sub> annot
-00012720: 6174 696f 6e20 7769 6c6c 2061 6c73 6f20  ation will also 
-00012730: 6265 2061 7574 6f6d 6174 656c 7920 7072  be automately pr
-00012740: 6f64 7563 6564 2c20 6173 2073 686f 776e  oduced, as shown
-00012750: 2062 656c 6f77 2e0a 0a21 5b5d 2864 6174   below...![](dat
-00012760: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00012770: 756c 6561 2d76 732d 4171 7569 6c65 6769  ulea-vs-Aquilegi
-00012780: 615f 636f 6572 756c 6561 2e64 6f74 5f75  a_coerulea.dot_u
-00012790: 6e69 745f 6765 6e65 2e70 6e67 290a 0a21  nit_gene.png)..!
-000127a0: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-000127b0: 615f 636f 6572 756c 6561 2d76 732d 4171  a_coerulea-vs-Aq
-000127c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000127d0: 2e64 6f74 2e70 6e67 290a 0a4e 6f74 6520  .dot.png)..Note 
-000127e0: 7468 6174 2074 6865 206f 7061 6369 7479  that the opacity
-000127f0: 206f 6620 616e 6368 6f72 2064 6f74 7320   of anchor dots 
-00012800: 616e 6420 616c 6c20 686f 6d6f 6c6f 6720  and all homolog 
-00012810: 646f 7473 2063 616e 2062 6520 7365 7420  dots can be set 
-00012820: 6279 2074 6865 206f 7074 696f 6e20 6061  by the option `a
-00012830: 7061 6c70 6861 6020 616e 6420 6068 6f61  palpha` and `hoa
-00012840: 6c70 6861 6020 7365 7061 7261 7465 6c79  lpha` separately
-00012850: 2e20 4966 206f 6e65 206a 7573 7420 7761  . If one just wa
-00012860: 6e74 7320 746f 2073 6565 2074 6865 2061  nts to see the a
-00012870: 6e63 686f 7220 646f 7473 2c20 7365 7474  nchor dots, sett
-00012880: 696e 6720 7468 6520 6068 6f61 6c70 6861  ing the `hoalpha
-00012890: 6020 6173 2030 2028 6f72 206f 7468 6572  ` as 0 (or other
-000128a0: 206d 696e 7573 6375 6c65 2076 616c 7565   minuscule value
-000128b0: 7329 2077 696c 6c20 646f 2e20 4966 206f  s) will do. If o
-000128c0: 6e65 2077 616e 7473 2074 6f20 7365 6520  ne wants to see 
-000128d0: 7468 6520 6469 7374 7269 6275 7469 6f6e  the distribution
-000128e0: 206f 6620 7768 6f6c 6520 646f 7473 2062   of whole dots b
-000128f0: 6574 7465 722c 2073 6574 7469 6e67 2074  etter, setting t
-00012900: 6865 2060 686f 616c 7068 6160 2068 6967  he `hoalpha` hig
-00012910: 6865 7220 2861 6e64 2060 6170 616c 7068  her (and `apalph
-00012920: 6160 206c 6f77 6572 2920 7769 6c6c 2064  a` lower) will d
-00012930: 6f2e 2054 6865 2060 646f 7473 697a 6560  o. The `dotsize`
-00012940: 206f 7074 696f 6e20 6361 6e20 6265 2063   option can be c
-00012950: 616c 6c65 6420 746f 2061 646a 7573 7420  alled to adjust 
-00012960: 7468 6520 7369 7a65 206f 6620 646f 7473  the size of dots
-00012970: 2e0a 0a41 2066 7572 7468 6572 2061 7373  ...A further ass
-00012980: 6f63 6961 7465 6420 5379 6e64 6570 7468  ociated Syndepth
-00012990: 2070 6c6f 7420 7368 6f77 7320 7468 6174   plot shows that
-000129a0: 2074 6865 7265 2061 7265 206d 6f72 6520   there are more 
-000129b0: 7468 616e 2038 3020 6475 706c 6963 6174  than 80 duplicat
-000129c0: 6564 2073 6567 6d65 6e74 732c 2077 6869  ed segments, whi
-000129d0: 6368 2064 6f6d 696e 6174 6573 2074 6865  ch dominates the
-000129e0: 2077 686f 6c65 2063 6f6c 6c69 6e65 6172   whole collinear
-000129f0: 2072 6174 696f 2063 6174 6567 6f72 792e   ratio category.
-00012a00: 0a0a 215b 5d28 6461 7461 2f53 796e 6465  ..![](data/Synde
-00012a10: 7074 682e 7376 6729 0a0a 5765 2063 616e  pth.svg)..We can
-00012a20: 2066 6974 2061 6e20 454c 4d4d 206d 6978   fit an ELMM mix
-00012a30: 7475 7265 206d 6f64 656c 2075 706f 6e20  ture model upon 
-00012a40: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
-00012a50: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-00012a60: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00012a70: 696f 6e20 746f 2073 6565 206d 6f72 6520  ion to see more 
-00012a80: 6163 6375 7261 7465 6c79 2074 6865 2073  accurately the s
-00012a90: 6967 6e69 6669 6361 6e63 6520 616e 6420  ignificance and 
-00012aa0: 6c6f 6361 7469 6f6e 206f 6620 706f 7465  location of pote
-00012ab0: 6e74 6961 6c20 5747 4473 2c20 7573 696e  ntial WGDs, usin
-00012ac0: 6720 7468 6520 636f 6d6d 616e 6420 6c69  g the command li
-00012ad0: 6e65 2062 656c 6f77 2e0a 0a60 6060 0a77  ne below...```.w
-00012ae0: 6764 2076 697a 202d 6420 7767 645f 6b73  gd viz -d wgd_ks
-00012af0: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00012b00: 756c 6561 2e74 7376 2e6b 732e 7473 760a  ulea.tsv.ks.tsv.
-00012b10: 6060 600a 0a54 6865 2072 6573 756c 7420  ```..The result 
-00012b20: 6f66 2045 4c4d 4d20 6d69 7874 7572 6520  of ELMM mixture 
-00012b30: 6d6f 6465 6c20 636c 7573 7465 7269 6e67  model clustering
-00012b40: 2073 686f 7773 2074 6861 7420 7468 6572   shows that ther
-00012b50: 6520 6973 2061 206c 696b 656c 7920 5747  e is a likely WG
-00012b60: 4420 636f 6d70 6f6e 656e 7420 6174 202a  D component at *
-00012b70: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2031  K*<sub>S</sub> 1
-00012b80: 2e31 392e 0a0a 215b 5d28 6461 7461 2f65  .19...![](data/e
-00012b90: 6c6d 6d5f 4171 7569 6c65 6769 615f 636f  lmm_Aquilegia_co
-00012ba0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00012bb0: 765f 6265 7374 5f6d 6f64 656c 735f 7765  v_best_models_we
-00012bc0: 6967 6874 6564 2e73 7667 290a 0a4c 6574  ighted.svg)..Let
-00012bd0: 2773 2064 6f20 6120 6d69 7874 7572 6520  's do a mixture 
-00012be0: 6d6f 6465 6c20 636c 7573 7465 7269 6e67  model clustering
-00012bf0: 2066 6f72 2061 6e63 686f 7220 2a4b 2a3c   for anchor *K*<
-00012c00: 7375 623e 533c 2f73 7562 3e20 746f 6f2c  sub>S</sub> too,
-00012c10: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00012c20: 6e64 206c 696e 6520 6265 6c6f 772e 0a0a  nd line below...
-00012c30: 6060 600a 7767 6420 7065 616b 2077 6764  ```.wgd peak wgd
-00012c40: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
-00012c50: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
-00012c60: 7376 202d 2d61 6e63 686f 7270 6f69 6e74  sv --anchorpoint
-00012c70: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
-00012c80: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00012c90: 7473 2e74 7874 202d 2d73 6567 6d65 6e74  ts.txt --segment
-00012ca0: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
-00012cb0: 652d 6f75 742f 7365 676d 656e 7473 2e74  e-out/segments.t
-00012cc0: 7874 202d 2d6c 6973 7465 6c65 6d65 6e74  xt --listelement
-00012cd0: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
-00012ce0: 652d 6f75 742f 6c69 7374 5f65 6c65 6d65  e-out/list_eleme
-00012cf0: 6e74 732e 7478 7420 2d2d 6d75 6c74 6970  nts.txt --multip
-00012d00: 6c69 636f 6e70 6169 7273 2077 6764 5f73  liconpairs wgd_s
-00012d10: 796e 2f69 6164 686f 7265 2d6f 7574 2f6d  yn/iadhore-out/m
-00012d20: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
-00012d30: 2e74 7874 202d 2d77 6569 6768 7465 640a  .txt --weighted.
-00012d40: 6060 600a 0a54 6865 2061 6e63 686f 7220  ```..The anchor 
-00012d50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00012d60: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-00012d70: 2061 6c73 6f20 6861 7320 6120 6c69 6b65   also has a like
-00012d80: 6c79 2057 4744 2063 6f6d 706f 6e65 6e74  ly WGD component
-00012d90: 2077 6974 6820 6d6f 6465 2031 2e32 382e   with mode 1.28.
-00012da0: 0a0a 215b 5d28 6461 7461 2f4f 7269 6769  ..![](data/Origi
-00012db0: 6e61 6c5f 416e 6368 6f72 4b73 5f47 4d4d  nal_AnchorKs_GMM
-00012dc0: 5f43 6f6d 706f 6e65 6e74 335f 6e6f 6465  _Component3_node
-00012dd0: 5f77 6569 6768 7465 645f 4c6f 676e 6f72  _weighted_Lognor
-00012de0: 6d61 6c2e 7376 6729 0a0a 4e6f 7720 7468  mal.svg)..Now th
-00012df0: 6174 2077 6520 6861 7665 2073 6565 6e20  at we have seen 
-00012e00: 7468 6520 6576 6964 656e 6365 206f 6620  the evidence of 
-00012e10: 6e75 6d65 726f 7573 2064 7570 6c69 6361  numerous duplica
-00012e20: 7465 6420 7365 676d 656e 7473 2061 6e64  ted segments and
-00012e30: 2074 6865 2061 6767 7265 6761 7469 6f6e   the aggregation
-00012e40: 206f 6620 6475 706c 6963 6174 6573 2061   of duplicates a
-00012e50: 6765 2069 6e20 2a4b 2a3c 7375 623e 533c  ge in *K*<sub>S<
-00012e60: 2f73 7562 3e20 6172 6f75 6e64 2031 2e32  /sub> around 1.2
-00012e70: 2066 6f72 2061 6e63 686f 7220 7061 6972   for anchor pair
-00012e80: 7320 616e 6420 6e6f 6e2d 616e 6368 6f72  s and non-anchor
-00012e90: 2070 6169 7273 2074 6872 6f75 6768 6f75   pairs throughou
-00012ea0: 7420 7468 6520 7768 6f6c 6520 6765 6e6f  t the whole geno
-00012eb0: 6d65 2e20 5765 2063 616e 2063 6c61 696d  me. We can claim
-00012ec0: 2077 6974 6820 736f 6d65 2063 6f6e 6669   with some confi
-00012ed0: 6465 6e63 6520 7468 6174 202a 4171 7569  dence that *Aqui
-00012ee0: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00012ef0: 6d69 6768 7420 6861 7665 2065 7870 6572  might have exper
-00012f00: 6965 6e63 6564 2061 2070 616c 656f 2d70  ienced a paleo-p
-00012f10: 6f6c 7970 6c6f 6964 697a 6174 696f 6e20  olyploidization 
-00012f20: 6576 656e 742e 204e 6578 742c 204c 6574  event. Next, Let
-00012f30: 2773 2068 6176 6520 6120 6675 7274 6865  's have a furthe
-00012f40: 7220 6c6f 6f6b 2061 626f 7574 2069 7473  r look about its
-00012f50: 2070 6879 6c6f 6765 6e65 7469 6320 6c6f   phylogenetic lo
-00012f60: 6361 7469 6f6e 2e20 5765 206b 6e6f 7720  cation. We know 
-00012f70: 7468 6174 2074 6865 7265 2061 7265 2075  that there are u
-00012f80: 6e63 6572 7461 696e 7469 6573 2061 626f  ncertainties abo
-00012f90: 7574 2077 6865 7468 6572 2074 6869 7320  ut whether this 
-00012fa0: 7075 7461 7469 7665 2070 616c 656f 2d70  putative paleo-p
-00012fb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e20  olyploidization 
-00012fc0: 6576 656e 7420 6973 2073 6861 7265 6420  event is shared 
-00012fd0: 7769 7468 2061 6c6c 2065 7564 6963 6f74  with all eudicot
-00012fe0: 7320 6f72 206e 6f74 2e20 5765 2063 616e  s or not. We can
-00012ff0: 2063 686f 6f73 6520 736f 6d65 206f 7468   choose some oth
-00013000: 6572 2065 7564 6963 6f74 2067 656e 6f6d  er eudicot genom
-00013010: 6573 2074 6f20 7365 6520 7468 6520 6f72  es to see the or
-00013020: 6465 7269 6e67 206f 6620 7370 6563 6961  dering of specia
-00013030: 7469 6f6e 2061 6e64 2070 6f6c 7970 6c6f  tion and polyplo
-00013040: 6964 697a 6174 696f 6e20 6576 656e 7473  idization events
-00013050: 2e20 4865 7265 2077 6520 6368 6f6f 7365  . Here we choose
-00013060: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
-00013070: 2a2c 202a 5072 6f74 6561 2063 796e 6172  *, *Protea cynar
-00013080: 6f69 6465 732a 2061 6e64 202a 4163 6f72  oides* and *Acor
-00013090: 7573 2061 6d65 7269 6361 6e75 732a 2069  us americanus* i
-000130a0: 6e20 7468 6520 666f 6c6c 6f77 696e 6720  n the following 
-000130b0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000130c0: 616e 616c 7973 6973 2e20 4669 7273 742c  analysis. First,
-000130d0: 2077 6520 6275 696c 7420 6120 676c 6f62   we built a glob
-000130e0: 616c 204d 5242 4820 6661 6d69 6c79 2075  al MRBH family u
-000130f0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00013100: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00013110: 2064 6d64 202d 2d67 6c6f 6261 6c6d 7262   dmd --globalmrb
-00013120: 6820 4171 7569 6c65 6769 615f 636f 6572  h Aquilegia_coer
-00013130: 756c 6561 2050 726f 7465 615f 6379 6e61  ulea Protea_cyna
-00013140: 726f 6964 6573 2041 636f 7275 735f 616d  roides Acorus_am
-00013150: 6572 6963 616e 7573 2056 6974 6973 5f76  ericanus Vitis_v
-00013160: 696e 6966 6572 6120 2d6f 2077 6764 5f67  inifera -o wgd_g
-00013170: 6c6f 6261 6c6d 7262 680a 6060 600a 0a49  lobalmrbh.```..I
-00013180: 6e20 7468 6520 676c 6f62 616c 204d 5242  n the global MRB
-00013190: 4820 6661 6d69 6c79 2c20 6576 6572 7920  H family, every 
-000131a0: 7061 6972 206f 6620 6f72 7468 6f6c 6f67  pair of ortholog
-000131b0: 6f75 7320 6765 6e65 7320 6973 2074 6865  ous genes is the
-000131c0: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-000131d0: 2068 6974 2c20 7375 6767 6573 7469 6e67   hit, suggesting
-000131e0: 2074 7275 6520 6f72 7468 6f6c 6f67 6f75   true orthologou
-000131f0: 7320 7265 6c61 7469 6f6e 7368 6970 732e  s relationships.
-00013200: 2057 6520 776f 756c 6420 7573 6520 7468   We would use th
-00013210: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00013220: 3e20 7661 6c75 6573 2061 7373 6f63 6961  > values associa
-00013230: 7465 6420 7769 7468 2074 6865 7365 206f  ted with these o
-00013240: 7274 686f 6c6f 676f 7573 2070 6169 7273  rthologous pairs
-00013250: 2074 6f20 6465 6c69 6d69 7420 7468 6520   to delimit the 
-00013260: 6469 7665 7267 656e 6365 202a 4b2a 3c73  divergence *K*<s
-00013270: 7562 3e53 3c2f 7375 623e 2070 6561 6b2e  ub>S</sub> peak.
-00013280: 2054 6f67 6574 6865 7220 7769 7468 2074   Together with t
-00013290: 6865 2077 686f 6c65 2070 6172 616e 6f6d  he whole paranom
-000132a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000132b0: 3e20 6469 7374 7269 6275 7469 6f6e 2c20  > distribution, 
-000132c0: 7765 2063 6f6e 6475 6374 2074 6865 2072  we conduct the r
-000132d0: 6174 6520 636f 7272 6563 7469 6f6e 2075  ate correction u
-000132e0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-000132f0: 2062 656c 6f77 2e0a 0a21 2153 696e 6365   below...!!Since
-00013300: 2060 7767 6460 2076 6572 7369 6f6e 2032   `wgd` version 2
-00013310: 2e30 2e32 342c 2077 6520 7265 7772 6f74  .0.24, we rewrot
-00013320: 6520 6120 636c 6561 6e65 7220 616e 6420  e a cleaner and 
-00013330: 7175 6963 6b65 7220 7761 7920 6f66 2064  quicker way of d
-00013340: 6f69 6e67 2073 7562 7374 6974 7574 696f  oing substitutio
-00013350: 6e20 7261 7465 2063 6f72 7265 6374 696f  n rate correctio
-00013360: 6e2e 2049 7427 7320 6e6f 7420 7265 7175  n. It's not requ
-00013370: 6972 6564 2074 6f20 7479 7065 2069 6e20  ired to type in 
-00013380: 616e 7920 7370 6569 6365 7320 7061 6972  any speices pair
-00013390: 2061 6e64 2061 2073 6572 6965 7320 6f66   and a series of
-000133a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000133b0: 2070 6c6f 7473 2077 696c 6c20 6265 2070   plots will be p
-000133c0: 726f 6475 6365 642e 2054 6865 2072 6571  roduced. The req
-000133d0: 7569 7265 6420 6669 6c65 7320 6172 6520  uired files are 
-000133e0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-000133f0: 7375 623e 533c 2f73 7562 3e20 7461 626c  sub>S</sub> tabl
-00013400: 652c 2070 6172 616c 6f67 6f75 7320 2a4b  e, paralogous *K
-00013410: 2a3c 7375 623e 533c 2f73 7562 3e20 7461  *<sub>S</sub> ta
-00013420: 626c 652c 2061 2073 7065 6369 6573 2074  ble, a species t
-00013430: 7265 6520 616e 6420 6120 666f 6375 7365  ree and a focuse
-00013440: 6420 7370 6563 6965 7320 2874 6865 206f  d species (the o
-00013450: 6e65 2069 6e70 7574 7465 6420 7769 7468  ne inputted with
-00013460: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
-00013470: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-00013480: 292e 2055 7365 7273 2063 616e 2063 686f  ). Users can cho
-00013490: 6f73 6520 746f 2061 6464 206f 6e65 206d  ose to add one m
-000134a0: 6f72 6520 6c61 7965 7220 6f66 2065 6c6d  ore layer of elm
-000134b0: 6d20 6d6f 6465 6c69 6e67 206f 6e20 7061  m modeling on pa
-000134c0: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-000134d0: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-000134e0: 616e 642f 6f72 2067 6d6d 206d 6f64 656c  and/or gmm model
-000134f0: 696e 6720 6f6e 2061 6e63 686f 7220 2a4b  ing on anchor *K
-00013500: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00013510: 6c75 6573 2e20 5468 6520 6f72 7468 6f6c  lues. The orthol
-00013520: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00013530: 2f73 7562 3e20 7661 6c75 6573 2063 616e  /sub> values can
-00013540: 2062 6520 6361 6c63 756c 6174 6564 2075   be calculated u
-00013550: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00013560: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00013570: 206b 7364 2077 6764 5f67 6c6f 6261 6c6d   ksd wgd_globalm
-00013580: 7262 682f 676c 6f62 616c 5f4d 5242 482e  rbh/global_MRBH.
-00013590: 7473 7620 7365 7173 2a20 2d6f 2077 6764  tsv seqs* -o wgd
-000135a0: 5f67 6c6f 6261 6c6d 7262 685f 6b73 0a60  _globalmrbh_ks.`
-000135b0: 6060 0a0a 5769 7468 2074 6865 2063 616c  ``..With the cal
-000135c0: 6375 6c61 7465 6420 6f72 7468 6f6c 6f67  culated ortholog
-000135d0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
-000135e0: 7562 3e20 7461 626c 652c 2077 6520 6361  ub> table, we ca
-000135f0: 6e20 7573 6520 7468 6520 636f 6d6d 616e  n use the comman
-00013600: 6420 6265 6c6f 7720 746f 2063 6f6e 6475  d below to condu
-00013610: 6374 2074 6865 2072 6174 6520 636f 7272  ct the rate corr
-00013620: 6563 7469 6f6e 2061 6e64 2f6f 7220 6d69  ection and/or mi
-00013630: 7874 7572 6520 6d6f 6465 6c69 6e67 2061  xture modeling a
-00013640: 6e61 6c79 7369 732e 0a0a 6060 600a 7767  nalysis...```.wg
-00013650: 6420 7669 7a20 2d64 2077 6764 5f67 6c6f  d viz -d wgd_glo
-00013660: 6261 6c6d 7262 685f 6b73 2f67 6c6f 6261  balmrbh_ks/globa
-00013670: 6c5f 4d52 4248 2e74 7376 2e6b 732e 7473  l_MRBH.tsv.ks.ts
-00013680: 7620 2d66 6120 4171 7569 6c65 6769 615f  v -fa Aquilegia_
-00013690: 636f 6572 756c 6561 202d 6570 6b20 4171  coerulea -epk Aq
-000136a0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000136b0: 2e6b 732e 7473 7620 2d61 7020 616e 6368  .ks.tsv -ap anch
-000136c0: 6f72 706f 696e 7473 2e74 7874 202d 7370  orpoints.txt -sp
-000136d0: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
-000136e0: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
-000136f0: 5f4b 7320 2d2d 706c 6f74 656c 6d6d 202d  _Ks --plotelmm -
-00013700: 2d70 6c6f 7461 7067 6d6d 0a60 6060 0a0a  -plotapgmm.```..
-00013710: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
-00013720: 676c 6f62 616c 6d72 6268 2f67 6c6f 6261  globalmrbh/globa
-00013730: 6c5f 4d52 4248 2e74 7376 2073 6571 732a  l_MRBH.tsv seqs*
-00013740: 202d 2d65 7874 7261 7061 7261 6e6f 6d65   --extraparanome
-00013750: 6b73 2077 6764 5f6b 7364 2f41 7175 696c  ks wgd_ksd/Aquil
-00013760: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00013770: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00013780: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00013790: 7765 6967 6874 202d 6f20 7767 645f 676c  weight -o wgd_gl
-000137a0: 6f62 616c 6d72 6268 5f6b 7320 2d2d 7370  obalmrbh_ks --sp
-000137b0: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
-000137c0: 6f65 7275 6c65 613b 5072 6f74 6561 5f63  oerulea;Protea_c
-000137d0: 796e 6172 6f69 6465 7322 202d 2d73 7061  ynaroides" --spa
-000137e0: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-000137f0: 6572 756c 6561 3b56 6974 6973 5f76 696e  erulea;Vitis_vin
-00013800: 6966 6572 6122 202d 2d73 7061 6972 2022  ifera" --spair "
-00013810: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013820: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
-00013830: 616e 7573 2220 2d2d 7370 6169 7220 2241  anus" --spair "A
-00013840: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00013850: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00013860: 756c 6561 2220 2d2d 706c 6f74 6b64 6520  ulea" --plotkde 
-00013870: 282d 6170 2077 6764 5f73 796e 2f69 6164  (-ap wgd_syn/iad
-00013880: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00013890: 6f69 6e74 732e 7478 7429 0a60 6060 0a0a  oints.txt).```..
-000138a0: 5468 6520 6669 6c65 2060 7370 6563 6965  The file `specie
-000138b0: 7374 7265 652e 6e77 6020 6973 2074 6865  stree.nw` is the
-000138c0: 2074 6578 7420 6669 6c65 206f 6620 7370   text file of sp
-000138d0: 6563 6965 7320 7472 6565 2069 6e20 6e65  ecies tree in ne
-000138e0: 7769 636b 2074 6861 7420 7261 7465 2063  wick that rate c
-000138f0: 6f72 7265 6374 696f 6e20 776f 756c 6420  orrection would 
-00013900: 6265 2063 6f6e 6475 6374 6564 206f 6e2e  be conducted on.
-00013910: 2049 7473 2063 6f6e 7465 6e74 2069 7320   Its content is 
-00013920: 6173 2062 656c 6f77 2e20 5573 6572 7320  as below. Users 
-00013930: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-00013940: 7468 6520 7370 6563 6965 7320 7061 6972  the species pair
-00013950: 7320 746f 2062 6520 706c 6f74 7465 642e  s to be plotted.
-00013960: 2057 6520 7375 6767 6573 7420 6164 6469   We suggest addi
-00013970: 6e67 2074 6865 206f 7074 696f 6e20 602d  ng the option `-
-00013980: 2d72 6577 6569 6768 7460 2074 6f20 7265  -reweight` to re
-00013990: 6361 6c63 756c 6174 6520 7468 6520 7765  calculate the we
-000139a0: 6967 6874 2070 6572 2073 7065 6369 6573  ight per species
-000139b0: 2070 6169 7220 7375 6368 2074 6861 7420   pair such that 
-000139c0: 7468 6520 7765 6967 6874 206f 6620 6f72  the weight of or
-000139d0: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-000139e0: 6169 7273 2077 696c 6c20 6265 636f 6d65  airs will become
-000139f0: 2031 2061 7320 7468 6520 7061 7261 6c6f   1 as the paralo
-00013a00: 676f 7573 2067 656e 6520 7061 6972 732e  gous gene pairs.
-00013a10: 2054 6865 2066 6c61 6720 602d 2d70 6c6f   The flag `--plo
-00013a20: 746b 6465 6020 6361 6e20 6265 2061 6464  tkde` can be add
-00013a30: 6564 2077 6865 6e20 7468 6520 6b64 6520  ed when the kde 
-00013a40: 6375 7276 6520 6f66 206f 7274 686f 6c6f  curve of ortholo
-00013a50: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00013a60: 7375 623e 2069 7320 6465 7369 7265 642e  sub> is desired.
-00013a70: 2045 7874 7261 2063 6f6c 6c69 6e65 6172   Extra collinear
-00013a80: 2064 6174 6120 6361 6e20 6265 2061 6464   data can be add
-00013a90: 6564 2062 7920 7468 6520 6f70 7469 6f6e  ed by the option
-00013aa0: 2060 2d61 7060 2e0a 0a60 6060 0a28 2828   `-ap`...```.(((
-00013ab0: 5669 7469 735f 7669 6e69 6665 7261 2c50  Vitis_vinifera,P
-00013ac0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-00013ad0: 292c 4171 7569 6c65 6769 615f 636f 6572  ),Aquilegia_coer
-00013ae0: 756c 6561 292c 4163 6f72 7573 5f61 6d65  ulea),Acorus_ame
-00013af0: 7269 6361 6e75 7329 3b0a 6060 600a 0a21  ricanus);.```..!
-00013b00: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-00013b10: 615f 636f 6572 756c 6561 5f47 6c6f 6261  a_coerulea_Globa
-00013b20: 6c6d 7262 684b 735f 436f 7272 6563 7465  lmrbhKs_Correcte
-00013b30: 642e 6b73 642e 7376 6729 0a0a 4173 2073  d.ksd.svg)..As s
-00013b40: 686f 776e 2061 626f 7665 2c20 6265 6361  hown above, beca
-00013b50: 7573 6520 6f66 2074 6865 2068 6967 6865  use of the highe
-00013b60: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
-00013b70: 6174 6520 6f66 202a 4171 7569 6c65 6769  ate of *Aquilegi
-00013b80: 6120 636f 6572 756c 6561 2a2c 2074 6865  a coerulea*, the
-00013b90: 206f 7269 6769 6e61 6c20 6f72 7468 6f6c   original orthol
-00013ba0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00013bb0: 2f73 7562 3e20 7661 6c75 6573 2077 6572  /sub> values wer
-00013bc0: 6520 6163 7475 616c 6c79 2075 6e64 6572  e actually under
-00013bd0: 6573 7469 6d61 7465 6420 696e 2074 6865  estimated in the
-00013be0: 2074 696d 652d 6672 616d 6520 6f66 202a   time-frame of *
-00013bf0: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00013c00: 6561 2a2e 2057 6865 6e20 7765 2072 6563  ea*. When we rec
-00013c10: 6f76 6572 6564 2074 6865 2064 6976 6572  overed the diver
-00013c20: 6765 6e63 6520 7375 6273 7469 7475 7469  gence substituti
-00013c30: 6f6e 2064 6973 7461 6e63 6520 696e 2074  on distance in t
-00013c40: 6572 6d73 206f 6620 7477 6f20 7469 6d65  erms of two time
-00013c50: 7320 6f66 2074 6865 2062 7261 6e63 682d  s of the branch-
-00013c60: 7370 6563 6966 6963 2063 6f6e 7472 6962  specific contrib
-00013c70: 7574 696f 6e20 6f66 202a 4171 7569 6c65  ution of *Aquile
-00013c80: 6769 6120 636f 6572 756c 6561 2a20 7369  gia coerulea* si
-00013c90: 6e63 6520 6974 7320 6469 7665 7267 656e  nce its divergen
-00013ca0: 6365 2077 6974 6820 7468 6520 7369 7374  ce with the sist
-00013cb0: 6572 2073 7065 6369 6573 2070 6c75 7320  er species plus 
-00013cc0: 7468 6520 7368 6172 6564 2073 7562 7374  the shared subst
-00013cd0: 6974 7574 696f 6e20 6469 7374 616e 6365  itution distance
-00013ce0: 2062 6566 6f72 6520 6469 7665 7267 656e   before divergen
-00013cf0: 6365 2028 696e 2072 656c 6174 6976 6520  ce (in relative 
-00013d00: 746f 2074 6865 206f 7574 6772 6f75 7029  to the outgroup)
-00013d10: 2c20 7468 6520 636f 7272 6563 7465 6420  , the corrected 
-00013d20: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013d30: 6d6f 6465 2062 6563 616d 6520 6c61 7267  mode became larg
-00013d40: 6572 2e0a 0a49 6620 6f6e 6520 6861 6420  er...If one had 
-00013d50: 7468 6520 6f72 7468 6f6c 6f67 6f75 7320  the orthologous 
-00013d60: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013d70: 6461 7461 2061 6c72 6561 6479 2c20 6f6e  data already, on
-00013d80: 6520 636f 756c 6420 616c 736f 2061 7070  e could also app
-00013d90: 6c79 2074 6865 2070 726f 6772 616d 2060  ly the program `
-00013da0: 7767 6420 7669 7a60 2074 6f20 636f 6e64  wgd viz` to cond
-00013db0: 7563 7420 7468 6520 7261 7465 2063 6f72  uct the rate cor
-00013dc0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-00013dd0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013de0: 6e64 2062 656c 6f77 2e20 4e6f 7465 2074  nd below. Note t
-00013df0: 6861 7420 7468 6520 6f72 6465 7220 6f66  hat the order of
-00013e00: 2067 6976 656e 2060 7370 6169 7260 206f   given `spair` o
-00013e10: 7074 696f 6e73 2064 6563 6964 6573 2074  ptions decides t
-00013e20: 6865 2063 6f6c 6f72 206f 6620 7468 6520  he color of the 
-00013e30: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013e40: 6469 7374 7269 6275 7469 6f6e 206f 6620  distribution of 
-00013e50: 6561 6368 2073 7065 6369 6573 2070 6169  each species pai
-00013e60: 722e 2054 6865 2061 6464 6974 696f 6e61  r. The additiona
-00013e70: 6c20 6f70 7469 6f6e 2060 666f 6375 7332  l option `focus2
-00013e80: 616c 6c60 2069 6e20 6077 6764 206b 7364  all` in `wgd ksd
-00013e90: 6020 616e 6420 6077 6764 2076 697a 6020  ` and `wgd viz` 
-00013ea0: 6361 6e20 6265 2075 7365 6420 746f 2074  can be used to t
-00013eb0: 656c 6c20 7468 6520 7072 6f67 7261 6d20  ell the program 
-00013ec0: 7468 6174 2074 6865 2073 7065 6369 6573  that the species
-00013ed0: 2070 6169 7273 2061 7265 2062 6574 7765   pairs are betwe
-00013ee0: 656e 2074 6865 2066 6f63 7573 2073 7065  en the focus spe
-00013ef0: 6369 6573 2061 6e64 2061 6c6c 2074 6865  cies and all the
-00013f00: 206f 7468 6572 2073 7065 6369 6573 2073   other species s
-00013f10: 7563 6820 7468 6174 2075 7365 7273 2064  uch that users d
-00013f20: 6f6e 2774 206e 6565 6420 746f 2074 7970  on't need to typ
-00013f30: 6520 696e 2065 6163 6820 7061 6972 2069  e in each pair i
-00013f40: 6e64 6976 6964 7561 6c6c 792e 0a0a 6060  ndividually...``
-00013f50: 600a 7767 6420 7669 7a20 2d64 2077 6764  `.wgd viz -d wgd
-00013f60: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
-00013f70: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
-00013f80: 732e 7473 7620 2d2d 6578 7472 6170 6172  s.tsv --extrapar
-00013f90: 616e 6f6d 656b 7320 7767 645f 6b73 642f  anomeks wgd_ksd/
-00013fa0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013fb0: 6561 2e74 7376 2e6b 732e 7473 7620 2d73  ea.tsv.ks.tsv -s
-00013fc0: 7020 7370 6563 6965 7374 7265 652e 6e77  p speciestree.nw
-00013fd0: 202d 2d72 6577 6569 6768 7420 2d61 7020   --reweight -ap 
-00013fe0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00013ff0: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-00014000: 2e74 7874 202d 6f20 7767 645f 7669 7a5f  .txt -o wgd_viz_
-00014010: 6d69 7865 645f 4b73 202d 2d73 7061 6972  mixed_Ks --spair
-00014020: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014030: 756c 6561 3b50 726f 7465 615f 6379 6e61  ulea;Protea_cyna
-00014040: 726f 6964 6573 2220 2d2d 7370 6169 7220  roides" --spair 
-00014050: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00014060: 6c65 613b 5669 7469 735f 7669 6e69 6665  lea;Vitis_vinife
-00014070: 7261 2220 2d2d 7370 6169 7220 2241 7175  ra" --spair "Aqu
-00014080: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00014090: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-000140a0: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-000140b0: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
-000140c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000140d0: 6122 202d 2d70 6c6f 746b 6465 2028 2d2d  a" --plotkde (--
-000140e0: 6773 6d61 7020 6765 6e65 5f73 7065 6369  gsmap gene_speci
-000140f0: 6573 2e6d 6170 290a 6060 600a 0a57 6974  es.map).```..Wit
-00014100: 6820 7468 6520 6f70 7469 6f6e 2060 666f  h the option `fo
-00014110: 6375 7332 616c 6c60 2c20 7468 6520 636f  cus2all`, the co
-00014120: 6d6d 616e 6420 6361 6e20 6265 2061 6c73  mmand can be als
-00014130: 6f20 6c69 6b65 2074 6869 732c 2073 616d  o like this, sam
-00014140: 6520 7769 7468 2060 7767 6420 6b73 6460  e with `wgd ksd`
-00014150: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
-00014160: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
-00014170: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
-00014180: 7473 762e 6b73 2e74 7376 202d 2d65 7874  tsv.ks.tsv --ext
-00014190: 7261 7061 7261 6e6f 6d65 6b73 2077 6764  raparanomeks wgd
-000141a0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
-000141b0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
-000141c0: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
-000141d0: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
-000141e0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-000141f0: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00014200: 6f69 6e74 732e 7478 7420 2d6f 2077 6764  oints.txt -o wgd
-00014210: 5f76 697a 5f6d 6978 6564 5f4b 7320 2d2d  _viz_mixed_Ks --
-00014220: 666f 6375 7332 616c 6c20 4171 7569 6c65  focus2all Aquile
-00014230: 6769 615f 636f 6572 756c 6561 202d 2d70  gia_coerulea --p
-00014240: 6c6f 746b 6465 0a60 6060 0a0a 4e6f 7465  lotkde.```..Note
-00014250: 2074 6861 7420 7765 2063 616e 2065 6173   that we can eas
-00014260: 696c 7920 7368 6f77 2074 6861 7420 2a41  ily show that *A
-00014270: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
-00014280: 612a 2068 6173 2068 6967 6865 7220 7375  a* has higher su
-00014290: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
-000142a0: 7468 616e 202a 5072 6f74 6561 2063 796e  than *Protea cyn
-000142b0: 6172 6f69 6465 732a 2061 6e64 202a 5669  aroides* and *Vi
-000142c0: 7469 7320 7669 6e69 6665 7261 2a20 6279  tis vinifera* by
-000142d0: 2063 6f6d 7061 7269 6e67 2074 6865 6972   comparing their
-000142e0: 2073 7562 7374 6974 7574 696f 6e20 6469   substitution di
-000142f0: 7374 616e 6365 2069 6e20 7265 6761 7264  stance in regard
-00014300: 2074 6f20 7468 6520 7361 6d65 2064 6976   to the same div
-00014310: 6572 6765 6e63 6520 6576 656e 7420 7769  ergence event wi
-00014320: 7468 206f 7574 6772 6f75 7020 7370 6563  th outgroup spec
-00014330: 6965 7320 2a41 636f 7275 735f 616d 6572  ies *Acorus_amer
-00014340: 6963 616e 7573 2a2c 2075 7369 6e67 2063  icanus*, using c
-00014350: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a60  ommand below...`
-00014360: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
-00014370: 645f 676c 6f62 616c 6d72 6268 5f6b 732f  d_globalmrbh_ks/
-00014380: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
-00014390: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-000143a0: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-000143b0: 6967 6874 202d 6f20 7767 645f 7669 7a5f  ight -o wgd_viz_
-000143c0: 436f 6d70 6172 655f 7261 7465 202d 2d73  Compare_rate --s
-000143d0: 7061 6972 2022 4163 6f72 7573 5f61 6d65  pair "Acorus_ame
-000143e0: 7269 6361 6e75 733b 5072 6f74 6561 5f63  ricanus;Protea_c
-000143f0: 796e 6172 6f69 6465 7322 202d 2d73 7061  ynaroides" --spa
-00014400: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00014410: 6572 756c 6561 3b41 636f 7275 735f 616d  erulea;Acorus_am
-00014420: 6572 6963 616e 7573 2220 2d2d 7370 6169  ericanus" --spai
-00014430: 7220 2256 6974 6973 5f76 696e 6966 6572  r "Vitis_vinifer
-00014440: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-00014450: 6e75 7322 202d 2d67 736d 6170 2067 656e  nus" --gsmap gen
-00014460: 655f 7370 6563 6965 732e 6d61 7020 2d2d  e_species.map --
-00014470: 706c 6f74 6b64 650a 6060 600a 0a21 5b5d  plotkde.```..![]
-00014480: 2864 6174 612f 5261 775f 4f72 7468 6f6c  (data/Raw_Orthol
-00014490: 6f67 7565 735f 436f 6d70 6172 655f 7261  ogues_Compare_ra
-000144a0: 7465 2e6b 7364 2e73 7667 290a 0a41 7320  te.ksd.svg)..As 
-000144b0: 6469 7370 6c61 7965 6420 6162 6f76 652c  displayed above,
-000144c0: 2074 6865 206f 7274 686f 6c6f 676f 7573   the orthologous
-000144d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000144e0: 2076 616c 7565 7320 6265 7774 6565 6e20   values bewteen 
-000144f0: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00014500: 6c65 612a 2061 6e64 202a 4163 6f72 7573  lea* and *Acorus
-00014510: 2061 6d65 7269 6361 6e75 732a 2068 6173   americanus* has
-00014520: 2074 6865 2068 6967 6865 7374 206d 6f64   the highest mod
-00014530: 652c 2069 6e64 6963 6174 696e 6774 6865  e, indicatingthe
-00014540: 2066 6173 7465 7220 7375 6273 7469 7475   faster substitu
-00014550: 7469 6f6e 2072 6174 6520 6f66 202a 4171  tion rate of *Aq
-00014560: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-00014570: 2a20 636f 6d70 6172 6564 2074 6f20 2a50  * compared to *P
-00014580: 726f 7465 6120 6379 6e61 726f 6964 6573  rotea cynaroides
-00014590: 2a20 616e 6420 2a56 6974 6973 2076 696e  * and *Vitis vin
-000145a0: 6966 6572 612a 2e0a 0a42 6566 6f72 6520  ifera*...Before 
-000145b0: 7632 2e30 2e32 312c 2074 6865 2067 656e  v2.0.21, the gen
-000145c0: 652d 7370 6563 6965 7320 6d61 7020 6669  e-species map fi
-000145d0: 6c65 2069 7320 6e65 6363 6573 7361 7269  le is neccessari
-000145e0: 6c79 206e 6565 6465 6420 666f 7220 6974  ly needed for it
-000145f0: 7320 696d 706c 656d 656e 7461 7469 6f6e  s implementation
-00014600: 2069 6e20 6077 6764 2076 697a 602c 2077   in `wgd viz`, w
-00014610: 6869 6368 2073 686f 756c 6420 6265 2061  hich should be a
-00014620: 7574 6f6d 6174 656c 7920 7072 6f64 7563  utomately produc
-00014630: 6564 2062 7920 7468 6520 6c61 7374 2060  ed by the last `
-00014640: 7767 6420 6b73 6460 2073 7465 7020 6769  wgd ksd` step gi
-00014650: 7665 6e20 7468 6520 6073 7061 6972 6020  ven the `spair` 
-00014660: 616e 6420 6073 7065 6369 6573 7472 6565  and `speciestree
-00014670: 6020 7061 7261 6d65 7465 7273 2e20 5468  ` parameters. Th
-00014680: 6520 6067 656e 655f 7370 6563 6965 732e  e `gene_species.
-00014690: 6d61 7060 2068 6173 2063 6f6e 7465 6e74  map` has content
-000146a0: 7320 6173 2062 656c 6f77 2069 6e20 7768  s as below in wh
-000146b0: 6963 6820 6561 6368 206c 696e 6520 6973  ich each line is
-000146c0: 2074 6865 206a 6f69 6e65 6420 7374 7269   the joined stri
-000146d0: 6e67 206f 6620 6765 6e65 206e 616d 6520  ng of gene name 
-000146e0: 616e 6420 7370 6563 6965 7320 6e61 6d65  and species name
-000146f0: 2062 7920 7370 6163 652e 2041 6674 6572   by space. After
-00014700: 2076 322e 302e 3231 2028 696e 636c 7564   v2.0.21 (includ
-00014710: 6564 292c 2074 6865 2067 656e 652d 7370  ed), the gene-sp
-00014720: 6563 6965 7320 6d61 7020 6669 6c65 2069  ecies map file i
-00014730: 7320 6e6f 7420 6e65 6363 6573 7361 7269  s not neccessari
-00014740: 6c79 206e 6565 6465 6420 616e 796d 6f72  ly needed anymor
-00014750: 652e 0a0a 6060 600a 4171 636f 6536 4730  e...```.Aqcoe6G0
-00014760: 3537 3830 302e 3120 4171 7569 6c65 6769  57800.1 Aquilegi
-00014770: 615f 636f 6572 756c 6561 0a56 7669 5f56  a_coerulea.Vvi_V
-00014780: 4954 5f32 3031 7330 3031 3167 3031 3533  IT_201s0011g0153
-00014790: 302e 3120 5669 7469 735f 7669 6e69 6665  0.1 Vitis_vinife
-000147a0: 7261 0a50 6379 5f50 726f 6379 3031 6730  ra.Pcy_Procy01g0
-000147b0: 3835 3130 2050 726f 7465 615f 6379 6e61  8510 Protea_cyna
-000147c0: 726f 6964 6573 0a41 616d 5f41 636f 7261  roides.Aam_Acora
-000147d0: 2e30 3447 3134 3239 3030 2e31 2041 636f  .04G142900.1 Aco
-000147e0: 7275 735f 616d 6572 6963 616e 7573 0a60  rus_americanus.`
-000147f0: 6060 0a0a 4120 6d6f 7265 2063 6f6d 706c  ``..A more compl
-00014800: 6578 2070 6c6f 7420 6361 6e20 6265 206d  ex plot can be m
-00014810: 6164 6520 6279 2061 6464 2074 6865 2066  ade by add the f
-00014820: 6c61 6720 602d 2d70 6c6f 7465 6c6d 6d60  lag `--plotelmm`
-00014830: 2073 7563 6820 7468 6174 2074 6865 2045   such that the E
-00014840: 4c4d 4d20 6d69 7874 7572 6520 6d6f 6465  LMM mixture mode
-00014850: 6c69 6e67 206f 6620 7072 6f76 6964 6564  ling of provided
-00014860: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00014870: 623e 533c 2f73 7562 3e20 6361 6e20 6265  b>S</sub> can be
-00014880: 2073 7570 6572 696d 706f 7365 642c 2075   superimposed, u
-00014890: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-000148a0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-000148b0: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
-000148c0: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
-000148d0: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
-000148e0: 202d 2d65 7874 7261 7061 7261 6e6f 6d65   --extraparanome
-000148f0: 6b73 2077 6764 5f6b 7364 2f41 7175 696c  ks wgd_ksd/Aquil
-00014900: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00014910: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00014920: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00014930: 7765 6967 6874 202d 6170 2077 6764 5f73  weight -ap wgd_s
-00014940: 796e 2f69 6164 686f 7265 2d6f 7574 2f61  yn/iadhore-out/a
-00014950: 6e63 686f 7270 6f69 6e74 732e 7478 7420  nchorpoints.txt 
-00014960: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
-00014970: 5f4b 735f 656c 6d6d 202d 2d73 7061 6972  _Ks_elmm --spair
-00014980: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014990: 756c 6561 3b50 726f 7465 615f 6379 6e61  ulea;Protea_cyna
-000149a0: 726f 6964 6573 2220 2d2d 7370 6169 7220  roides" --spair 
-000149b0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-000149c0: 6c65 613b 5669 7469 735f 7669 6e69 6665  lea;Vitis_vinife
-000149d0: 7261 2220 2d2d 7370 6169 7220 2241 7175  ra" --spair "Aqu
-000149e0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-000149f0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00014a00: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-00014a10: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
-00014a20: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014a30: 6122 202d 2d67 736d 6170 2067 656e 655f  a" --gsmap gene_
-00014a40: 7370 6563 6965 732e 6d61 7020 2d2d 706c  species.map --pl
-00014a50: 6f74 6b64 6520 2d2d 706c 6f74 656c 6d6d  otkde --plotelmm
-00014a60: 0a60 6060 0a0a 215b 5d28 6461 7461 2f41  .```..![](data/A
-00014a70: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014a80: 615f 476c 6f62 616c 6d72 6268 4b73 5f45  a_GlobalmrbhKs_E
-00014a90: 6c6d 6d5f 436f 7272 6563 7465 642e 6b73  lmm_Corrected.ks
-00014aa0: 642e 7376 6729 0a0a 4672 6f6d 2074 6865  d.svg)..From the
-00014ab0: 206d 6978 6564 202a 4b2a 3c73 7562 3e53   mixed *K*<sub>S
-00014ac0: 3c2f 7375 623e 2070 6c6f 7420 6162 6f76  </sub> plot abov
-00014ad0: 652c 2077 6520 6361 6e20 7365 6520 7468  e, we can see th
-00014ae0: 6174 2074 6865 206f 7074 696d 697a 6564  at the optimized
-00014af0: 206c 6f67 6e6f 726d 616c 2063 6f6d 706f   lognormal compo
-00014b00: 6e65 6e74 2062 2077 6974 6820 6d6f 6465  nent b with mode
-00014b10: 2031 2e32 2069 7320 796f 756e 6765 7220   1.2 is younger 
-00014b20: 7468 616e 2074 6865 2063 6f72 7265 6374  than the correct
-00014b30: 6564 206f 7274 686f 6c6f 676f 7573 202a  ed orthologous *
-00014b40: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
-00014b50: 6f64 6520 7769 7468 202a 5072 6f74 6561  ode with *Protea
-00014b60: 2063 796e 6172 6f69 6465 732a 2061 6e64   cynaroides* and
-00014b70: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
-00014b80: 2a20 2831 2e33 3920 616e 6420 312e 3437  * (1.39 and 1.47
-00014b90: 2c20 7265 7370 6563 7469 7665 6c79 292e  , respectively).
-00014ba0: 0a0a 4265 7369 6465 732c 2077 6520 6361  ..Besides, we ca
-00014bb0: 6e20 616c 736f 2061 6464 2074 6865 2047  n also add the G
-00014bc0: 4d4d 206d 6978 7475 7265 206d 6f64 656c  MM mixture model
-00014bd0: 696e 6720 6f66 2061 6e63 686f 7220 2a4b  ing of anchor *K
-00014be0: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00014bf0: 6c75 6573 2077 6974 6820 7468 6520 666c  lues with the fl
-00014c00: 6167 2060 2d2d 706c 6f74 6170 676d 6d60  ag `--plotapgmm`
-00014c10: 2c20 7573 696e 6720 7468 6520 636f 6d6d  , using the comm
-00014c20: 616e 6420 6265 6c6f 772e 0a0a 6060 600a  and below...```.
-00014c30: 7767 6420 7669 7a20 2d64 2077 6764 5f67  wgd viz -d wgd_g
-00014c40: 6c6f 6261 6c6d 7262 685f 6b73 2f67 6c6f  lobalmrbh_ks/glo
-00014c50: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
-00014c60: 7473 7620 2d2d 6578 7472 6170 6172 616e  tsv --extraparan
-00014c70: 6f6d 656b 7320 7767 645f 6b73 642f 4171  omeks wgd_ksd/Aq
-00014c80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014c90: 2e74 7376 2e6b 732e 7473 7620 2d73 7020  .tsv.ks.tsv -sp 
-00014ca0: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
-00014cb0: 2d72 6577 6569 6768 7420 2d61 7020 7767  -reweight -ap wg
-00014cc0: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00014cd0: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
-00014ce0: 7874 202d 6f20 7767 645f 7669 7a5f 6d69  xt -o wgd_viz_mi
-00014cf0: 7865 645f 4b73 5f65 6c6d 6d20 2d2d 7370  xed_Ks_elmm --sp
-00014d00: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
-00014d10: 6f65 7275 6c65 613b 5072 6f74 6561 5f63  oerulea;Protea_c
-00014d20: 796e 6172 6f69 6465 7322 202d 2d73 7061  ynaroides" --spa
-00014d30: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00014d40: 6572 756c 6561 3b56 6974 6973 5f76 696e  erulea;Vitis_vin
-00014d50: 6966 6572 6122 202d 2d73 7061 6972 2022  ifera" --spair "
-00014d60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00014d70: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
-00014d80: 616e 7573 2220 2d2d 7370 6169 7220 2241  anus" --spair "A
-00014d90: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014da0: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00014db0: 756c 6561 2220 2d2d 6773 6d61 7020 6765  ulea" --gsmap ge
-00014dc0: 6e65 5f73 7065 6369 6573 2e6d 6170 202d  ne_species.map -
-00014dd0: 2d70 6c6f 746b 6465 202d 2d70 6c6f 7461  -plotkde --plota
-00014de0: 7067 6d6d 6d0a 6060 600a 0a21 5b5d 2864  pgmmm.```..![](d
-00014df0: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00014e00: 6572 756c 6561 5f47 6c6f 6261 6c6d 7262  erulea_Globalmrb
-00014e10: 684b 735f 4170 676d 6d5f 436f 7272 6563  hKs_Apgmm_Correc
-00014e20: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
-00014e30: 206d 616e 6966 6573 7465 6420 6162 6f76   manifested abov
-00014e40: 652c 2074 6865 2061 6e63 686f 7220 2a4b  e, the anchor *K
-00014e50: 2a3c 7375 623e 533c 2f73 7562 3e20 636f  *<sub>S</sub> co
-00014e60: 6d70 6f6e 656e 7420 3220 7769 7468 206d  mponent 2 with m
-00014e70: 6f64 6520 312e 3238 2069 7320 616c 736f  ode 1.28 is also
-00014e80: 2079 6f75 6e67 6572 2074 6861 6e20 7468   younger than th
-00014e90: 6520 636f 7272 6563 7465 6420 6f72 7468  e corrected orth
-00014ea0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00014eb0: 533c 2f73 7562 3e20 6d6f 6465 2077 6974  S</sub> mode wit
-00014ec0: 6820 2a50 726f 7465 6120 6379 6e61 726f  h *Protea cynaro
-00014ed0: 6964 6573 2a20 616e 6420 2a56 6974 6973  ides* and *Vitis
-00014ee0: 2076 696e 6966 6572 612a 2e20 4275 7420   vinifera*. But 
-00014ef0: 7765 206e 6565 6420 746f 2062 6520 6f66  we need to be of
-00014f00: 2063 6f75 7273 6520 6361 7574 696f 7573   course cautious
-00014f10: 2074 6861 7420 7375 6368 2064 6973 7469   that such disti
-00014f20: 6e63 7469 6f6e 2063 6f6d 6573 2077 6974  nction comes wit
-00014f30: 6820 7468 6520 756e 6365 7274 6169 6e74  h the uncertaint
-00014f40: 6965 7320 696e 7472 6f64 7563 6564 2066  ies introduced f
-00014f50: 726f 6d20 7468 6520 6170 706c 6965 6420  rom the applied 
-00014f60: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00014f70: 206d 6574 686f 646f 6c6f 6779 2069 6e20   methodology in 
-00014f80: 7465 726d 7320 6f66 2066 6f72 2069 6e73  terms of for ins
-00014f90: 7461 6e63 6520 6469 6666 6572 656e 7420  tance different 
-00014fa0: 696e 6974 6961 6c69 7a61 7469 6f6e 2070  initialization p
-00014fb0: 6f69 6e74 7320 616e 6420 7468 6520 6973  oints and the is
-00014fc0: 7375 6520 6f66 206f 7665 7266 6974 7469  sue of overfitti
-00014fd0: 6e67 2061 6e64 2074 6865 2073 6973 7465  ng and the siste
-00014fe0: 7220 7370 6563 6965 7373 2061 646f 7074  r speciess adopt
-00014ff0: 6564 2069 6e20 7468 6174 2074 6865 7265  ed in that there
-00015000: 206d 6967 6874 2062 6520 7370 6563 6965   might be specie
-00015010: 7320 7769 7468 206d 6f72 6520 6469 7370  s with more disp
-00015020: 6172 6174 6520 7375 6273 7469 7475 7469  arate substituti
-00015030: 6f6e 2072 6174 6573 2074 6861 6e20 7468  on rates than th
-00015040: 6520 6f6e 6520 7765 2063 686f 7365 2e0a  e one we chose..
-00015050: 0a41 6464 696e 6720 626f 7468 2045 4c4d  .Adding both ELM
-00015060: 4d20 7265 7375 6c74 2066 6f72 2070 6172  M result for par
-00015070: 616e 6f6d 6520 616e 6420 474d 4d20 7265  anome and GMM re
-00015080: 7375 6c74 2066 6f72 2061 6e63 686f 7220  sult for anchor 
-00015090: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000150a0: 6361 6e20 6265 2061 6368 6965 7665 6420  can be achieved 
-000150b0: 6a75 7374 2062 7920 6164 6420 7468 6520  just by add the 
-000150c0: 7477 6f20 666c 6167 7320 6d65 6e74 696f  two flags mentio
-000150d0: 6e65 6420 6162 6f76 652c 2075 7369 6e67  ned above, using
-000150e0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-000150f0: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
-00015100: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
-00015110: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
-00015120: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
-00015130: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-00015140: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00015150: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00015160: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-00015170: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-00015180: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
-00015190: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-000151a0: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
-000151b0: 6764 5f76 697a 5f6d 6978 6564 5f4b 735f  gd_viz_mixed_Ks_
-000151c0: 656c 6d6d 202d 2d73 7061 6972 2022 4171  elmm --spair "Aq
-000151d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000151e0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
-000151f0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
-00015200: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015210: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
-00015220: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00015230: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
-00015240: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015250: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015260: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
-00015270: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
-00015280: 2d67 736d 6170 2067 656e 655f 7370 6563  -gsmap gene_spec
-00015290: 6965 732e 6d61 7020 2d2d 706c 6f74 6b64  ies.map --plotkd
-000152a0: 6520 2d2d 706c 6f74 656c 6d6d 202d 2d70  e --plotelmm --p
-000152b0: 6c6f 7461 7067 6d6d 6d0a 6060 600a 0a21  lotapgmmm.```..!
-000152c0: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-000152d0: 615f 636f 6572 756c 6561 5f47 6c6f 6261  a_coerulea_Globa
-000152e0: 6c6d 7262 684b 735f 456c 6d6d 5f41 7067  lmrbhKs_Elmm_Apg
-000152f0: 6d6d 5f43 6f72 7265 6374 6564 2e6b 7364  mm_Corrected.ksd
-00015300: 2e73 7667 290a 0a41 6e20 616c 7465 726e  .svg)..An altern
-00015310: 6174 6976 6520 7761 7920 746f 2063 616c  ative way to cal
-00015320: 6375 6c61 7465 2074 6865 206f 7274 686f  culate the ortho
-00015330: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00015340: 3c2f 7375 623e 2069 7320 746f 2064 6972  </sub> is to dir
-00015350: 6563 746c 7920 7573 6520 7468 6520 6f72  ectly use the or
-00015360: 7468 6f67 726f 7570 7320 696e 7374 6561  thogroups instea
-00015370: 6420 6f66 2067 6c6f 6261 6c20 4d52 4248  d of global MRBH
-00015380: 2066 616d 696c 792e 2054 6861 7420 7761   family. That wa
-00015390: 7920 7765 2064 6f6e 2774 2075 7365 2074  y we don't use t
-000153a0: 6865 2070 7265 2d69 6e66 6572 7265 6420  he pre-inferred 
-000153b0: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-000153c0: 3e53 3c2f 7375 623e 2062 7574 2074 6865  >S</sub> but the
-000153d0: 2070 6172 616c 6f67 6f75 7320 6765 6e65   paralogous gene
-000153e0: 2070 6169 7273 2069 6e73 6964 6520 6561   pairs inside ea
-000153f0: 6368 206f 7274 686f 6772 6f75 7020 696e  ch orthogroup in
-00015400: 7374 6561 642e 2054 6f20 6163 6869 6576  stead. To achiev
-00015410: 6520 7468 6174 2c20 7765 2066 6972 7374  e that, we first
-00015420: 206e 6565 6420 746f 2069 6e66 6572 206f   need to infer o
-00015430: 7274 686f 6772 6f75 7073 2075 7369 6e67  rthogroups using
-00015440: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00015450: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-00015460: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00015470: 6c65 6120 5072 6f74 6561 5f63 796e 6172  lea Protea_cynar
-00015480: 6f69 6465 7320 4163 6f72 7573 5f61 6d65  oides Acorus_ame
-00015490: 7269 6361 6e75 7320 5669 7469 735f 7669  ricanus Vitis_vi
-000154a0: 6e69 6665 7261 202d 2d6f 7274 686f 696e  nifera --orthoin
-000154b0: 6665 7220 2d6f 2077 6764 5f6f 7274 686f  fer -o wgd_ortho
-000154c0: 2028 2d2d 6f6e 6c79 6f72 7468 6f29 200a   (--onlyortho) .
-000154d0: 6060 600a 0a55 7365 7273 2063 616e 2064  ```..Users can d
-000154e0: 6563 6964 6520 746f 206f 6e6c 7920 636f  ecide to only co
-000154f0: 6e64 7563 7420 7468 6520 6f72 7468 6f67  nduct the orthog
-00015500: 726f 7570 2061 6e61 6c79 7369 7320 7768  roup analysis wh
-00015510: 696c 6520 736b 6970 7069 6e67 206f 7468  ile skipping oth
-00015520: 6572 2061 6e61 6c79 7369 7320 6279 2061  er analysis by a
-00015530: 6464 696e 6720 7468 6520 666c 6167 2060  dding the flag `
-00015540: 2d2d 6f6e 6c79 6f72 7468 6f60 2e20 4e65  --onlyortho`. Ne
-00015550: 7874 2073 7465 7020 6973 2074 6865 2073  xt step is the s
-00015560: 616d 6520 7769 7468 2067 6c6f 6261 6c20  ame with global 
-00015570: 4d52 4248 2066 616d 696c 7920 6578 6365  MRBH family exce
-00015580: 7074 2074 6861 7420 7765 2064 6f6e 2774  pt that we don't
-00015590: 2075 7365 2074 6865 2065 7874 7261 2070   use the extra p
-000155a0: 7265 2d69 6e66 6572 7265 6420 7061 7261  re-inferred para
-000155b0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-000155c0: 7375 623e 2061 6e79 6d6f 7265 2e20 5765  sub> anymore. We
-000155d0: 2069 6e66 6572 2074 6865 202a 4b2a 3c73   infer the *K*<s
-000155e0: 7562 3e53 3c2f 7375 623e 2075 7369 6e67  ub>S</sub> using
-000155f0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00015600: 6f77 2e20 4e6f 7465 2074 6861 7420 7468  ow. Note that th
-00015610: 6520 7072 6f67 7261 6d20 6077 6764 2076  e program `wgd v
-00015620: 697a 6020 6361 6e20 706c 6f74 2074 6865  iz` can plot the
-00015630: 2073 616d 6520 6a75 7374 2061 7320 7368   same just as sh
-00015640: 6f77 6e20 6162 6f76 652e 0a0a 6060 600a  own above...```.
-00015650: 7767 6420 6b73 6420 7767 645f 6f72 7468  wgd ksd wgd_orth
-00015660: 6f2f 4f72 7468 6f67 726f 7570 732e 7370  o/Orthogroups.sp
-00015670: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-00015680: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-00015690: 6874 202d 6f20 7767 645f 6f72 7468 6f5f  ht -o wgd_ortho_
-000156a0: 6b73 202d 2d73 7061 6972 2022 4171 7569  ks --spair "Aqui
-000156b0: 6c65 6769 615f 636f 6572 756c 6561 3b50  legia_coerulea;P
-000156c0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-000156d0: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
-000156e0: 6567 6961 5f63 6f65 7275 6c65 613b 5669  egia_coerulea;Vi
-000156f0: 7469 735f 7669 6e69 6665 7261 2220 2d2d  tis_vinifera" --
-00015700: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00015710: 5f63 6f65 7275 6c65 613b 4163 6f72 7573  _coerulea;Acorus
-00015720: 5f61 6d65 7269 6361 6e75 7322 202d 2d73  _americanus" --s
-00015730: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00015740: 636f 6572 756c 6561 3b41 7175 696c 6567  coerulea;Aquileg
-00015750: 6961 5f63 6f65 7275 6c65 6122 202d 2d70  ia_coerulea" --p
-00015760: 6c6f 746b 6465 2028 2d61 7020 7767 645f  lotkde (-ap wgd_
-00015770: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
-00015780: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
-00015790: 290a 6060 600a 0a21 5b5d 2864 6174 612f  ).```..![](data/
-000157a0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000157b0: 6561 5f4f 7274 686f 4b73 5f43 6f72 7265  ea_OrthoKs_Corre
-000157c0: 6374 6564 2e6b 7364 2e73 7667 290a 0a41  cted.ksd.svg)..A
-000157d0: 7320 7368 6f77 6e20 6162 6f76 652c 2074  s shown above, t
-000157e0: 6865 206e 756d 6265 7220 6f66 2062 6f74  he number of bot
-000157f0: 6820 7061 7261 6c6f 676f 7573 2067 656e  h paralogous gen
-00015800: 6520 7061 6972 7320 616e 6420 6f72 7468  e pairs and orth
-00015810: 6f6c 6f67 6f75 7320 6765 6e65 2070 6169  ologous gene pai
-00015820: 7273 2069 7320 6469 6666 6572 656e 7420  rs is different 
-00015830: 7468 616e 2074 6865 206f 6e65 2066 726f  than the one fro
-00015840: 6d20 676c 6f62 616c 204d 5242 4820 6661  m global MRBH fa
-00015850: 6d69 6c79 2069 6e20 7468 6174 2068 6572  mily in that her
-00015860: 6520 7765 2070 6c6f 7474 6564 2061 6c6c  e we plotted all
-00015870: 206f 7274 686f 6c6f 676f 7573 2067 656e   orthologous gen
-00015880: 6520 7061 6972 7320 696e 7374 6561 6420  e pairs instead 
-00015890: 6f66 206f 6e6c 7920 676c 6f62 616c 204d  of only global M
-000158a0: 5242 4820 616e 6420 706f 7465 6e74 6961  RBH and potentia
-000158b0: 6c6c 7920 6e65 7720 7061 7261 6c6f 676f  lly new paralogo
-000158c0: 7573 2067 656e 6520 7061 6972 7320 6d69  us gene pairs mi
-000158d0: 6768 7420 6265 2070 726f 6475 6365 6420  ght be produced 
-000158e0: 696e 2074 6865 206f 7274 686f 6772 6f75  in the orthogrou
-000158f0: 7020 696e 6665 7265 6e63 6520 7374 6570  p inference step
-00015900: 2c20 746f 6765 7468 6572 2077 6974 6820  , together with 
-00015910: 6469 6666 6572 656e 7420 7265 6361 6c63  different recalc
-00015920: 756c 6174 6564 2077 6569 6768 7473 2e0a  ulated weights..
-00015930: 0a49 6620 6f6e 6520 6f6e 6c79 2077 616e  .If one only wan
-00015940: 7465 6420 746f 2070 6c6f 7420 7468 6520  ted to plot the 
-00015950: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-00015960: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00015970: 7269 6275 7469 6f6e 2c20 6173 2077 6520  ribution, as we 
-00015980: 6861 7665 2061 6c72 6561 6479 2073 686f  have already sho
-00015990: 776e 2061 626f 7665 2c20 6974 2063 616e  wn above, it can
-000159a0: 2062 6520 6561 7369 6c79 2061 6368 6965   be easily achie
-000159b0: 7665 6420 6279 2072 656d 6f76 696e 6720  ved by removing 
-000159c0: 7468 6520 7061 7261 6c6f 676f 7573 2073  the paralogous s
-000159d0: 7065 6369 6573 2070 6169 7220 6041 7175  pecies pair `Aqu
-000159e0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-000159f0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015a00: 6561 602c 2075 7369 6e67 2074 6865 2063  ea`, using the c
-00015a10: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a60  ommand below...`
-00015a20: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
-00015a30: 645f 676c 6f62 616c 6d72 6268 5f6b 732f  d_globalmrbh_ks/
-00015a40: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
-00015a50: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-00015a60: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015a70: 6967 6874 202d 6f20 7767 645f 676c 6f62  ight -o wgd_glob
-00015a80: 616c 6d72 6268 5f6f 6e6c 796f 7274 686f  almrbh_onlyortho
-00015a90: 5f6b 7320 2d2d 7370 6169 7220 2241 7175  _ks --spair "Aqu
-00015aa0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015ab0: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
-00015ac0: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-00015ad0: 6c65 6769 615f 636f 6572 756c 6561 3b56  legia_coerulea;V
-00015ae0: 6974 6973 5f76 696e 6966 6572 6122 202d  itis_vinifera" -
-00015af0: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015b00: 615f 636f 6572 756c 6561 3b41 636f 7275  a_coerulea;Acoru
-00015b10: 735f 616d 6572 6963 616e 7573 2220 2d2d  s_americanus" --
-00015b20: 6773 6d61 7020 6765 6e65 5f73 7065 6369  gsmap gene_speci
-00015b30: 6573 2e6d 6170 0a60 6060 0a0a 5765 2063  es.map.```..We c
-00015b40: 616e 2063 6c65 6172 6c79 2073 6565 2074  an clearly see t
-00015b50: 6861 7420 2a56 6974 6973 2076 696e 6966  hat *Vitis vinif
-00015b60: 6572 612a 2068 6173 2068 6967 6865 7220  era* has higher 
-00015b70: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00015b80: 6520 7468 616e 202a 5072 6f74 6561 2063  e than *Protea c
-00015b90: 796e 6172 6f69 6465 732a 2069 6e20 7468  ynaroides* in th
-00015ba0: 6174 2074 6865 6972 206f 7274 686f 6c6f  at their ortholo
-00015bb0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00015bc0: 7375 623e 2070 6561 6b73 2077 6974 6820  sub> peaks with 
-00015bd0: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00015be0: 6c65 612a 2c20 616c 7468 6f75 6768 2072  lea*, although r
-00015bf0: 6570 7265 7365 6e74 696e 6720 7468 6520  epresenting the 
-00015c00: 7361 6d65 2064 6976 6572 6765 6e63 6520  same divergence 
-00015c10: 6576 656e 742c 2064 6966 6665 7265 6420  event, differed 
-00015c20: 696e 2073 7562 7374 6974 7574 696f 6e20  in substitution 
-00015c30: 6469 7374 616e 6365 2e0a 215b 5d28 6461  distance..![](da
-00015c40: 7461 2f52 6177 5f4f 7274 686f 6c6f 6775  ta/Raw_Orthologu
-00015c50: 6573 2e6b 7364 2e73 7667 290a 0a41 6674  es.ksd.svg)..Aft
-00015c60: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
-00015c70: 7469 6320 7469 6d69 6e67 206f 6620 7468  tic timing of th
-00015c80: 6520 5261 6e75 6e63 756c 616c 6573 2057  e Ranunculales W
-00015c90: 4744 2c20 7765 2063 616e 2066 7572 7468  GD, we can furth
-00015ca0: 6572 2069 6e66 6572 2069 7473 2061 6273  er infer its abs
-00015cb0: 6f6c 7574 6520 6167 652e 2046 6972 7374  olute age. First
-00015cc0: 2077 6520 696e 6665 7220 7468 6520 6372   we infer the cr
-00015cd0: 6564 6962 6c65 2072 616e 6765 206f 6620  edible range of 
-00015ce0: 616e 6368 6f72 2070 6169 7273 2062 7920  anchor pairs by 
-00015cf0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00015d00: 6865 7572 6973 7469 6361 6c6c 7920 7573  heuristically us
-00015d10: 696e 6720 7468 6520 7072 6f67 7261 6d20  ing the program 
-00015d20: 6077 6764 2070 6561 6b60 2e0a 0a60 6060  `wgd peak`...```
-00015d30: 0a77 6764 2070 6561 6b20 2d2d 6865 7572  .wgd peak --heur
-00015d40: 6973 7469 6320 7767 645f 6b73 642f 4171  istic wgd_ksd/Aq
-00015d50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015d60: 2e74 7376 2e6b 732e 7473 7620 2d61 7020  .tsv.ks.tsv -ap 
-00015d70: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00015d80: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-00015d90: 2e74 7874 202d 736d 2077 6764 5f73 796e  .txt -sm wgd_syn
-00015da0: 2f69 6164 686f 7265 2d6f 7574 2f73 6567  /iadhore-out/seg
-00015db0: 6d65 6e74 732e 7478 7420 2d6c 6520 7767  ments.txt -le wg
-00015dc0: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00015dd0: 742f 6c69 7374 5f65 6c65 6d65 6e74 732e  t/list_elements.
-00015de0: 7478 7420 2d6d 7020 7767 645f 7379 6e2f  txt -mp wgd_syn/
-00015df0: 6961 6468 6f72 652d 6f75 742f 6d75 6c74  iadhore-out/mult
-00015e00: 6970 6c69 636f 6e5f 7061 6972 732e 7478  iplicon_pairs.tx
-00015e10: 7420 2d6f 2077 6764 5f70 6561 6b0a 6060  t -o wgd_peak.``
-00015e20: 600a 0a21 5b5d 2864 6174 612f 416e 6368  `..![](data/Anch
-00015e30: 6f72 4b73 5f50 6561 6b43 495f 4171 7569  orKs_PeakCI_Aqui
-00015e40: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-00015e50: 7376 2e6b 732e 7473 765f 6e6f 6465 5f77  sv.ks.tsv_node_w
-00015e60: 6569 6768 7465 642e 7376 6729 0a0a 4173  eighted.svg)..As
-00015e70: 2073 686f 776e 2061 626f 7665 2c20 7765   shown above, we
-00015e80: 2061 7373 756d 6564 2061 206c 6f67 6e6f   assumed a logno
-00015e90: 726d 616c 2064 6973 7472 6962 7574 696f  rmal distributio
-00015ea0: 6e20 6174 2074 6865 2070 6561 6b20 6c6f  n at the peak lo
-00015eb0: 6361 7469 6f6e 2064 6574 6563 7465 6420  cation detected 
-00015ec0: 6279 2074 6865 2060 7369 676e 616c 6020  by the `signal` 
-00015ed0: 6d6f 6475 6c65 206f 6620 6073 6369 7079  module of `scipy
-00015ee0: 6020 6c69 6272 6172 792e 2054 6865 2039  ` library. The 9
-00015ef0: 3525 2063 6f6e 6669 6465 6e63 6520 6c65  5% confidence le
-00015f00: 7665 6c20 6f66 2074 6865 206c 6f67 6e6f  vel of the logno
-00015f10: 726d 616c 2064 6973 7472 6962 7574 696f  rmal distributio
-00015f20: 6e20 7761 7320 6170 706c 6965 642c 2069  n was applied, i
-00015f30: 2e65 2e2c 2030 2e36 382d 322e 3734 2c20  .e., 0.68-2.74, 
-00015f40: 696e 2066 7572 7468 6572 206d 6f6c 6563  in further molec
-00015f50: 756c 6172 2064 6174 696e 672e 2054 6865  ular dating. The
-00015f60: 2066 696c 6520 6041 7175 696c 6567 6961   file `Aquilegia
-00015f70: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00015f80: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
-00015f90: 725f 6461 7469 6e67 5f77 6569 6768 7465  r_dating_weighte
-00015fa0: 645f 666f 726d 6174 2e74 7376 6020 6973  d_format.tsv` is
-00015fb0: 2077 6861 7420 7765 206e 6565 6420 666f   what we need fo
-00015fc0: 7220 6e65 7874 2073 7465 702e 2054 6f20  r next step. To 
-00015fd0: 6275 696c 6420 7468 6520 6f72 7468 6f67  build the orthog
-00015fe0: 726f 7570 7320 7573 6564 2069 6e20 7068  roups used in ph
-00015ff0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-00016000: 672c 2077 6520 6e65 6564 2074 6f20 7365  g, we need to se
-00016010: 6c65 6374 2073 6f6d 6520 7370 6563 6965  lect some specie
-00016020: 7320 616e 6420 666f 726d 2061 2073 7461  s and form a sta
-00016030: 7274 696e 6720 7472 6565 2077 6974 6820  rting tree with 
-00016040: 7072 6f70 6572 2066 6f73 7369 6c20 6361  proper fossil ca
-00016050: 6c69 6272 6174 696f 6e73 2e20 5765 2070  librations. We p
-00016060: 726f 7669 6465 206f 6e65 2069 6e20 6d63  rovide one in mc
-00016070: 6d63 7472 6565 2066 6f72 6d61 7420 6173  mctree format as
-00016080: 2062 656c 6f77 2e0a 0a60 6060 0a31 3720   below...```.17 
-00016090: 310a 2828 2828 506f 7461 6d6f 6765 746f  1.((((Potamogeto
-000160a0: 6e5f 6163 7574 6966 6f6c 6975 732c 2853  n_acutifolius,(S
-000160b0: 7069 726f 6465 6c61 5f69 6e74 6572 6d65  pirodela_interme
-000160c0: 6469 612c 416d 6f72 7068 6f70 6861 6c6c  dia,Amorphophall
-000160d0: 7573 5f6b 6f6e 6a61 6329 292c 2841 6361  us_konjac)),(Aca
-000160e0: 6e74 686f 6368 6c61 6d79 735f 6272 6163  nthochlamys_brac
-000160f0: 7465 6174 612c 2844 696f 7363 6f72 6561  teata,(Dioscorea
-00016100: 5f61 6c61 7461 2c44 696f 7363 6f72 6561  _alata,Dioscorea
-00016110: 5f72 6f74 756e 6461 7461 2929 273e 302e  _rotundata))'>0.
-00016120: 3536 3030 3c31 2e32 3836 3327 2927 3e30  5600<1.2863')'>0
-00016130: 2e38 3336 303c 312e 3238 3633 272c 2841  .8360<1.2863',(A
-00016140: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00016150: 2c41 636f 7275 735f 7461 7461 7269 6e6f  ,Acorus_tatarino
-00016160: 7769 6929 2927 3e30 2e38 3336 303c 312e  wii))'>0.8360<1.
-00016170: 3238 3633 272c 2828 2828 5465 7472 6163  2863',((((Tetrac
-00016180: 656e 7472 6f6e 5f73 696e 656e 7365 2c54  entron_sinense,T
-00016190: 726f 6368 6f64 656e 6472 6f6e 5f61 7261  rochodendron_ara
-000161a0: 6c69 6f69 6465 7329 2c28 4275 7875 735f  lioides),(Buxus_
-000161b0: 6175 7374 726f 7975 6e6e 616e 656e 7369  austroyunnanensi
-000161c0: 732c 4275 7875 735f 7369 6e69 6361 2929  s,Buxus_sinica))
-000161d0: 273e 312e 3130 3830 3c31 2e32 3836 3327  '>1.1080<1.2863'
-000161e0: 2c28 4e65 6c75 6d62 6f5f 6e75 6369 6665  ,(Nelumbo_nucife
-000161f0: 7261 2c28 5465 6c6f 7065 615f 7370 6563  ra,(Telopea_spec
-00016200: 696f 7369 7373 696d 612c 5072 6f74 6561  iosissima,Protea
-00016210: 5f63 796e 6172 6f69 6465 7329 2929 273e  _cynaroides)))'>
-00016220: 312e 3130 3830 3c31 2e32 3836 3327 2c28  1.1080<1.2863',(
-00016230: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016240: 6561 5f61 7031 2c41 7175 696c 6567 6961  ea_ap1,Aquilegia
-00016250: 5f63 6f65 7275 6c65 615f 6170 3229 2927  _coerulea_ap2))'
-00016260: 3e31 2e31 3038 303c 312e 3238 3633 2729  >1.1080<1.2863')
-00016270: 273e 312e 3237 3230 3c32 2e34 3732 3027  '>1.2720<2.4720'
-00016280: 3b0a 6060 600a 0a41 7320 7072 6573 656e  ;.```..As presen
-00016290: 7465 6420 6162 6f76 652c 2074 6865 2066  ted above, the f
-000162a0: 6f63 7573 2073 7065 6369 6573 2074 6861  ocus species tha
-000162b0: 7420 6973 2061 626f 7574 2074 6f20 6265  t is about to be
-000162c0: 2064 6174 6564 206e 6565 6473 2074 6f20   dated needs to 
-000162d0: 6265 2072 6570 6c61 6365 6420 7769 7468  be replaced with
-000162e0: 2060 2841 7175 696c 6567 6961 5f63 6f65   `(Aquilegia_coe
-000162f0: 7275 6c65 615f 6170 312c 4171 7569 6c65  rulea_ap1,Aquile
-00016300: 6769 615f 636f 6572 756c 6561 5f61 7032  gia_coerulea_ap2
-00016310: 2960 2e20 5769 7468 2074 6869 7320 7374  )`. With this st
-00016320: 6172 7469 6e67 2074 7265 6520 616e 6420  arting tree and 
-00016330: 7072 6564 6f77 6e6c 6f61 6465 6420 6364  predownloaded cd
-00016340: 7320 6669 6c65 7320 6f66 2061 6c6c 2074  s files of all t
-00016350: 6865 2073 7065 6369 6573 2c20 7765 2063  he species, we c
-00016360: 616e 2062 7569 6c64 2074 6865 206f 7274  an build the ort
-00016370: 686f 6772 6f75 7020 7573 6564 2069 6e20  hogroup used in 
-00016380: 7468 6520 6669 6e61 6c20 6d6f 6c65 6375  the final molecu
-00016390: 6c61 7220 6461 7469 6e67 2075 7369 6e67  lar dating using
-000163a0: 2074 6865 2063 6f6d 6d61 6e64 2061 7320   the command as 
-000163b0: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
-000163c0: 646d 6420 2d66 2041 7175 696c 6567 6961  dmd -f Aquilegia
-000163d0: 5f63 6f65 7275 6c65 6120 2d61 7020 7767  _coerulea -ap wg
-000163e0: 645f 7065 616b 2f41 7175 696c 6567 6961  d_peak/Aquilegia
-000163f0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00016400: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
-00016410: 725f 6461 7469 6e67 5f77 6569 6768 7465  r_dating_weighte
-00016420: 645f 666f 726d 6174 2e74 7376 202d 6f20  d_format.tsv -o 
-00016430: 7767 645f 646d 645f 6f72 7468 6f20 506f  wgd_dmd_ortho Po
-00016440: 7461 6d6f 6765 746f 6e5f 6163 7574 6966  tamogeton_acutif
-00016450: 6f6c 6975 7320 5370 6972 6f64 656c 615f  olius Spirodela_
-00016460: 696e 7465 726d 6564 6961 2041 6d6f 7270  intermedia Amorp
-00016470: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
-00016480: 2041 6361 6e74 686f 6368 6c61 6d79 735f   Acanthochlamys_
-00016490: 6272 6163 7465 6174 6120 4469 6f73 636f  bracteata Diosco
-000164a0: 7265 615f 616c 6174 6120 4469 6f73 636f  rea_alata Diosco
-000164b0: 7265 615f 726f 7475 6e64 6174 6120 4163  rea_rotundata Ac
-000164c0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
-000164d0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
-000164e0: 6969 2054 6574 7261 6365 6e74 726f 6e5f  ii Tetracentron_
-000164f0: 7369 6e65 6e73 6520 5472 6f63 686f 6465  sinense Trochode
-00016500: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
-00016510: 2042 7578 7573 5f61 7573 7472 6f79 756e   Buxus_austroyun
-00016520: 6e61 6e65 6e73 6973 2042 7578 7573 5f73  nanensis Buxus_s
-00016530: 696e 6963 6120 4e65 6c75 6d62 6f5f 6e75  inica Nelumbo_nu
-00016540: 6369 6665 7261 2054 656c 6f70 6561 5f73  cifera Telopea_s
-00016550: 7065 6369 6f73 6973 7369 6d61 2050 726f  peciosissima Pro
-00016560: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
-00016570: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00016580: 610a 6060 600a 0a54 6865 2072 6573 756c  a.```..The resul
-00016590: 7420 6669 6c65 2060 6d65 7267 655f 666f  t file `merge_fo
-000165a0: 6375 735f 6170 2e74 7376 6020 6973 2077  cus_ap.tsv` is w
-000165b0: 6861 7420 7765 206e 6565 6420 666f 7220  hat we need for 
-000165c0: 7468 6520 6669 6e61 6c20 7374 6570 206f  the final step o
-000165d0: 6620 6d6f 6c65 6375 6c61 7220 6461 7469  f molecular dati
-000165e0: 6e67 2069 6e20 7072 6f67 7261 6d20 6077  ng in program `w
-000165f0: 6764 2066 6f63 7573 602e 0a0a 6060 600a  gd focus`...```.
-00016600: 7767 6420 666f 6375 7320 2d2d 7072 6f74  wgd focus --prot
-00016610: 6461 7469 6e67 202d 2d61 616d 6f64 656c  dating --aamodel
-00016620: 206c 6720 7767 645f 646d 645f 6f72 7468   lg wgd_dmd_orth
-00016630: 6f2f 6d65 7267 655f 666f 6375 735f 6170  o/merge_focus_ap
-00016640: 2e74 7376 202d 7370 2064 6174 696e 675f  .tsv -sp dating_
-00016650: 7472 6565 2e6e 7720 2d6f 2077 6764 5f64  tree.nw -o wgd_d
-00016660: 6174 696e 6720 2d64 206d 636d 6374 7265  ating -d mcmctre
-00016670: 6520 2d64 7320 2762 7572 6e69 6e20 3d20  e -ds 'burnin = 
-00016680: 3230 3030 2720 2d64 7320 2773 616d 7066  2000' -ds 'sampf
-00016690: 7265 7120 3d20 3130 3030 2720 2d64 7320  req = 1000' -ds 
-000166a0: 276e 7361 6d70 6c65 203d 2032 3030 3030  'nsample = 20000
-000166b0: 2720 506f 7461 6d6f 6765 746f 6e5f 6163  ' Potamogeton_ac
-000166c0: 7574 6966 6f6c 6975 7320 5370 6972 6f64  utifolius Spirod
-000166d0: 656c 615f 696e 7465 726d 6564 6961 2041  ela_intermedia A
-000166e0: 6d6f 7270 686f 7068 616c 6c75 735f 6b6f  morphophallus_ko
-000166f0: 6e6a 6163 2041 6361 6e74 686f 6368 6c61  njac Acanthochla
-00016700: 6d79 735f 6272 6163 7465 6174 6120 4469  mys_bracteata Di
-00016710: 6f73 636f 7265 615f 616c 6174 6120 4469  oscorea_alata Di
-00016720: 6f73 636f 7265 615f 726f 7475 6e64 6174  oscorea_rotundat
-00016730: 6120 4163 6f72 7573 5f61 6d65 7269 6361  a Acorus_america
-00016740: 6e75 7320 4163 6f72 7573 5f74 6174 6172  nus Acorus_tatar
-00016750: 696e 6f77 6969 2054 6574 7261 6365 6e74  inowii Tetracent
-00016760: 726f 6e5f 7369 6e65 6e73 6520 5472 6f63  ron_sinense Troc
-00016770: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
-00016780: 6964 6573 2042 7578 7573 5f61 7573 7472  ides Buxus_austr
-00016790: 6f79 756e 6e61 6e65 6e73 6973 2042 7578  oyunnanensis Bux
-000167a0: 7573 5f73 696e 6963 6120 4e65 6c75 6d62  us_sinica Nelumb
-000167b0: 6f5f 6e75 6369 6665 7261 2054 656c 6f70  o_nucifera Telop
-000167c0: 6561 5f73 7065 6369 6f73 6973 7369 6d61  ea_speciosissima
-000167d0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
-000167e0: 6573 2041 7175 696c 6567 6961 5f63 6f65  es Aquilegia_coe
-000167f0: 7275 6c65 610a 6060 600a 0a48 6572 6520  rulea.```..Here 
-00016800: 7765 206f 6e6c 7920 696d 706c 656d 656e  we only implemen
-00016810: 7465 6420 7468 6520 636f 6e63 6174 656e  ted the concaten
-00016820: 6174 696f 6e20 616e 616c 7973 6973 2075  ation analysis u
-00016830: 7369 6e67 2070 726f 7465 696e 2073 6571  sing protein seq
-00016840: 7565 6e63 6520 6279 2061 6464 696e 6720  uence by adding 
-00016850: 7468 6520 666c 6167 2060 2d2d 7072 6f74  the flag `--prot
-00016860: 6461 7469 6e67 6020 616e 6420 7765 2073  dating` and we s
-00016870: 6574 2074 6865 2070 6172 616d 6574 6572  et the parameter
-00016880: 2066 6f72 2060 6d63 6d63 7472 6565 6020   for `mcmctree` 
-00016890: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-000168a0: 2d64 7360 2e20 4e6f 7465 2074 6861 7420  -ds`. Note that 
-000168b0: 6f74 6865 7220 6461 7469 6e67 2070 726f  other dating pro
-000168c0: 6772 616d 2073 7563 6820 6173 2060 7238  gram such as `r8
-000168d0: 7360 2061 6e64 2060 6265 6173 7460 2061  s` and `beast` a
-000168e0: 7265 2061 6c73 6f20 6176 6169 6c61 626c  re also availabl
-000168f0: 6520 6769 7665 6e20 736f 6d65 206d 616e  e given some man
-00016900: 6461 746f 7279 2070 6172 616d 6574 6572  datory parameter
-00016910: 732e 2054 6865 2066 696e 616c 206c 6f67  s. The final log
-00016920: 206f 6620 7468 6520 7375 6363 6573 7366   of the successf
-00016930: 756c 2072 756e 2069 7320 6173 2062 656c  ul run is as bel
-00016940: 6f77 2e0a 0a60 6060 0a31 363a 3034 3a32  ow...```.16:04:2
-00016950: 3520 494e 464f 2020 2020 2052 756e 6e69  5 INFO     Runni
-00016960: 6e67 206d 636d 6374 7265 6520 7573 696e  ng mcmctree usin
-00016970: 6720 4865 7373 6961 6e20 6d61 7472 6978  g Hessian matrix
-00016980: 206f 6620 4c47 2b47 616d 6d61 2020 636f   of LG+Gamma  co
-00016990: 7265 2e70 793a 3936 370a 2020 2020 2020  re.py:967.      
-000169a0: 2020 2020 2020 2020 2020 2020 666f 7220              for 
-000169b0: 7072 6f74 6569 6e20 6d6f 6465 6c0a 3233  protein model.23
-000169c0: 3a34 393a 3337 2049 4e46 4f20 2020 2020  :49:37 INFO     
-000169d0: 506f 7374 6572 696f 7220 6d65 616e 2066  Posterior mean f
-000169e0: 6f72 2074 6865 2061 6765 7320 6f66 2077  or the ages of w
-000169f0: 6764 2069 7320 3131 322e 3839 3435 206d  gd is 112.8945 m
-00016a00: 636d 6374 7265 652e 7079 3a32 3936 0a20  cmctree.py:296. 
-00016a10: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016a20: 206d 696c 6c69 6f6e 2079 6561 7273 2066   million years f
-00016a30: 726f 6d20 436f 6e63 6174 656e 6174 6564  rom Concatenated
-00016a40: 2070 6570 7469 6465 0a20 2020 2020 2020   peptide.       
-00016a50: 2020 2020 2020 2020 2020 2061 6c69 676e             align
-00016a60: 6d65 6e74 2061 6e64 2039 3525 2063 7265  ment and 95% cre
-00016a70: 6469 6269 6c69 7479 2069 6e74 6572 7661  dibility interva
-00016a80: 6c73 2028 4349 290a 2020 2020 2020 2020  ls (CI).        
-00016a90: 2020 2020 2020 2020 2020 6973 2031 3031            is 101
-00016aa0: 2e32 3234 2d31 3233 2e31 3231 206d 696c  .224-123.121 mil
-00016ab0: 6c69 6f6e 2079 6561 7273 0a20 2020 2020  lion years.     
-00016ac0: 2020 2020 494e 464f 2020 2020 2054 6f74      INFO     Tot
-00016ad0: 616c 2072 756e 2074 696d 653a 2032 3931  al run time: 291
-00016ae0: 3735 7320 2020 2020 2020 2020 2020 2020  75s             
-00016af0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016b00: 2063 6c69 2e70 793a 3234 310a 2020 2020   cli.py:241.    
-00016b10: 2020 2020 2049 4e46 4f20 2020 2020 446f       INFO     Do
-00016b20: 6e65 2020 2020 2020 2020 2020 2020 2020  ne              
-00016b30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016b40: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00016b50: 2020 636c 692e 7079 3a32 3432 0a60 6060    cli.py:242.```
-00016b60: 0a0a 546f 2076 6973 7561 6c69 7a65 2074  ..To visualize t
-00016b70: 6865 2064 6174 652c 2077 6520 616c 736f  he date, we also
-00016b80: 2070 726f 7669 6465 6420 6120 7079 7468   provided a pyth
-00016b90: 6f6e 2073 6372 6970 7420 746f 2070 6c6f  on script to plo
-00016ba0: 7420 7468 6520 5747 4420 6461 7465 7320  t the WGD dates 
-00016bb0: 696e 2074 6865 2060 7767 6460 2066 6f6c  in the `wgd` fol
-00016bc0: 6465 722e 2055 7365 7273 206e 6565 6420  der. Users need 
-00016bd0: 746f 2065 7874 7261 6374 2074 6865 2072  to extract the r
-00016be0: 6177 2064 6174 6573 2066 726f 6d20 7468  aw dates from th
-00016bf0: 6520 606d 636d 632e 7478 7460 2066 6f72  e `mcmc.txt` for
-00016c00: 2074 6865 2057 4744 206e 6f64 6520 6669   the WGD node fi
-00016c10: 7273 7420 616e 6420 7361 7665 2069 7420  rst and save it 
-00016c20: 6173 2066 696c 6520 6064 6174 6573 2e74  as file `dates.t
-00016c30: 7874 6020 286f 7220 7768 6174 6576 6572  xt` (or whatever
-00016c40: 2070 7265 6665 7272 6564 206e 616d 6529   preferred name)
-00016c50: 2e20 416e 2065 7861 6d70 6c65 2063 6f6d  . An example com
-00016c60: 6d61 6e64 2069 7320 6173 2062 656c 6f77  mand is as below
-00016c70: 2e0a 0a60 6060 0a70 7974 686f 6e20 2450  ...```.python $P
-00016c80: 4154 482f 706f 7374 706c 6f74 2e70 7920  ATH/postplot.py 
-00016c90: 706f 7374 6469 7320 6461 7465 732e 7478  postdis dates.tx
-00016ca0: 7420 2d2d 7065 7263 656e 7469 6c65 2039  t --percentile 9
-00016cb0: 3020 2d2d 7469 746c 6520 2257 4744 2064  0 --title "WGD d
-00016cc0: 6174 6522 202d 2d68 7064 202d 6f20 2252  ate" --hpd -o "R
-00016cd0: 616e 756e 6375 6c61 6c65 735f 5747 445f  anunculales_WGD_
-00016ce0: 6461 7465 2e73 7667 220a 6060 600a 0a21  date.svg".```..!
-00016cf0: 5b5d 2864 6174 612f 5261 6e75 6e63 756c  [](data/Ranuncul
-00016d00: 616c 6573 5f57 4744 5f64 6174 652e 7376  ales_WGD_date.sv
-00016d10: 6729 0a0a 5468 6520 706f 7374 6572 696f  g)..The posterio
-00016d20: 7220 6d65 616e 2c20 6d65 6469 616e 2061  r mean, median a
-00016d30: 6e64 206d 6f64 6520 6f66 2074 6865 2052  nd mode of the R
-00016d40: 616e 756e 6375 6c61 6c65 7320 5747 4420  anunculales WGD 
-00016d50: 6167 6520 6973 2031 3132 2e39 322c 2031  age is 112.92, 1
-00016d60: 3133 2e34 3420 616e 6420 3131 322e 3534  13.44 and 112.54
-00016d70: 206d 7961 2c20 7769 7468 2039 3025 2048   mya, with 90% H
-00016d80: 5044 2031 3035 2e30 3720 2d20 3132 322e  PD 105.07 - 122.
-00016d90: 3332 206d 7961 2061 7320 6d61 6e69 6665  32 mya as manife
-00016da0: 7374 6564 2061 626f 7665 2e0a 0a23 2323  sted above...###
-00016db0: 204b 7374 7265 650a 0a49 6e20 6164 6469   Kstree..In addi
-00016dc0: 7469 6f6e 2074 6f20 7061 6972 7769 7365  tion to pairwise
-00016dd0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00016de0: 2065 7374 696d 6174 696f 6e2c 2061 202a   estimation, a *
-00016df0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2074  K*<sub>S</sub> t
-00016e00: 7265 6520 7769 7468 2062 7261 6e63 6820  ree with branch 
-00016e10: 6c65 6e67 7468 2069 6e20 2a4b 2a3c 7375  length in *K*<su
-00016e20: 623e 533c 2f73 7562 3e20 756e 6974 2063  b>S</sub> unit c
-00016e30: 616e 2061 6c73 6f20 6265 2064 6572 6976  an also be deriv
-00016e40: 6564 2066 726f 6d20 7468 6520 7072 6f67  ed from the prog
-00016e50: 7261 6d20 6077 6764 206b 7364 6020 6769  ram `wgd ksd` gi
-00016e60: 7665 6e20 7468 6520 6f70 7469 6f6e 2060  ven the option `
-00016e70: 2d2d 6b73 7472 6565 6020 616e 6420 602d  --kstree` and `-
-00016e80: 2d73 7065 6369 6573 7472 6565 602e 204e  -speciestree`. N
-00016e90: 6f74 6520 7468 6174 2074 6865 2061 6464  ote that the add
-00016ea0: 6974 696f 6e61 6c20 6f70 7469 6f6e 2060  itional option `
-00016eb0: 2d2d 6f6e 6c79 636f 6e63 6174 6b73 7472  --onlyconcatkstr
-00016ec0: 6565 6020 7769 6c6c 206f 6e6c 7920 6361  ee` will only ca
-00016ed0: 6c6c 2074 6865 202a 4b2a 3c73 7562 3e53  ll the *K*<sub>S
-00016ee0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
-00016ef0: 6e20 666f 7220 7468 6520 636f 6e63 6174  n for the concat
-00016f00: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
-00016f10: 2072 6174 6865 7220 7468 616e 2061 6c6c   rather than all
-00016f20: 2074 6865 2061 6c69 676e 6d65 6e74 732e   the alignments.
-00016f30: 2055 7365 7273 206e 6565 6420 746f 2070   Users need to p
-00016f40: 726f 7669 6465 2061 2070 7265 7365 7420  rovide a preset 
-00016f50: 7370 6563 6965 7320 7472 6565 2066 6f72  species tree for
-00016f60: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00016f70: 7375 623e 2074 7265 6520 696e 6665 7265  sub> tree infere
-00016f80: 6e63 6520 6f66 2074 6865 2063 6f6e 6361  nce of the conca
-00016f90: 7465 6e61 7465 6420 616c 6967 6e6d 656e  tenated alignmen
-00016fa0: 7420 7768 696c 6520 7468 6520 7265 6d61  t while the rema
-00016fb0: 696e 696e 6720 616c 6967 6e6d 656e 7473  ining alignments
-00016fc0: 2077 696c 6c20 6265 2061 6761 696e 7374   will be against
-00016fd0: 2061 6e20 6175 746f 6d61 7465 6c79 2069   an automately i
-00016fe0: 6e66 6572 7265 6420 7472 6565 2066 726f  nferred tree fro
-00016ff0: 6d20 6066 6173 7474 7265 6560 206f 7220  m `fasttree` or 
-00017000: 6069 7174 7265 6560 2e20 496e 2074 6865  `iqtree`. In the
-00017010: 2065 6e64 2c20 7573 6572 7320 7769 6c6c   end, users will
-00017020: 2067 6574 2061 202a 4b2a 3c73 7562 3e53   get a *K*<sub>S
-00017030: 3c2f 7375 623e 2074 7265 652c 2061 202a  </sub> tree, a *
-00017040: 4b2a 3c73 7562 3e41 3c2f 7375 623e 2074  K*<sub>A</sub> t
-00017050: 7265 6520 616e 6420 6120 cf89 2074 7265  ree and a .. tre
-00017060: 6520 7065 7220 6661 6d20 616e 6420 666f  e per fam and fo
-00017070: 7220 7468 6520 636f 6e63 6174 656e 6174  r the concatenat
-00017080: 6564 2061 6c69 676e 6d65 6e74 2e0a 0a60  ed alignment...`
-00017090: 6060 0a77 6764 206b 7364 2064 6174 612f  ``.wgd ksd data/
-000170a0: 6b73 7472 6565 5f64 6174 612f 6661 6d2e  kstree_data/fam.
-000170b0: 7473 7620 6461 7461 2f6b 7374 7265 655f  tsv data/kstree_
-000170c0: 6461 7461 2f41 636f 7275 735f 7461 7461  data/Acorus_tata
-000170d0: 7269 6e6f 7769 6920 6461 7461 2f6b 7374  rinowii data/kst
-000170e0: 7265 655f 6461 7461 2f41 6d62 6f72 656c  ree_data/Amborel
-000170f0: 6c61 5f74 7269 6368 6f70 6f64 6120 6461  la_trichopoda da
-00017100: 7461 2f6b 7374 7265 655f 6461 7461 2f41  ta/kstree_data/A
-00017110: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00017120: 6120 6461 7461 2f6b 7374 7265 655f 6461  a data/kstree_da
-00017130: 7461 2f41 7269 7374 6f6c 6f63 6869 615f  ta/Aristolochia_
-00017140: 6669 6d62 7269 6174 6120 6461 7461 2f6b  fimbriata data/k
-00017150: 7374 7265 655f 6461 7461 2f43 7963 6173  stree_data/Cycas
-00017160: 5f70 616e 7a68 6968 7561 656e 7369 202d  _panzhihuaensi -
-00017170: 2d6b 7374 7265 6520 2d2d 7370 6563 6965  -kstree --specie
-00017180: 7374 7265 6520 6461 7461 2f6b 7374 7265  stree data/kstre
-00017190: 655f 6461 7461 2f73 7065 6369 6573 5f74  e_data/species_t
-000171a0: 7265 6531 2e6e 7720 2d2d 6f6e 6c79 636f  ree1.nw --onlyco
-000171b0: 6e63 6174 6b73 7472 6565 202d 6f20 7767  ncatkstree -o wg
-000171c0: 645f 6b73 7472 6565 5f74 6f70 6f6c 6f67  d_kstree_topolog
-000171d0: 7931 0a60 6060 0a0a 215b 5d28 6461 7461  y1.```..![](data
-000171e0: 2f6b 7374 7265 655f 7265 7375 6c74 732f  /kstree_results/
-000171f0: 6b73 7472 6565 2e73 7667 290a 0a41 626f  kstree.svg)..Abo
-00017200: 7665 2077 6520 7573 6564 2074 6872 6565  ve we used three
-00017210: 2061 6c74 6572 6e61 7469 7665 2074 6f70   alternative top
-00017220: 6f6c 6f67 6965 7320 746f 2069 6e66 6572  ologies to infer
-00017230: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00017240: 7375 623e 2074 7265 6520 7768 6963 6820  sub> tree which 
-00017250: 6c65 6420 746f 2064 6966 6665 7265 6e74  led to different
-00017260: 2062 7261 6e63 6820 6c65 6e67 7468 2065   branch length e
-00017270: 7374 696d 6174 696f 6e2e 204e 6f74 6520  stimation. Note 
-00017280: 7468 6174 2074 6865 2066 616d 696c 6965  that the familie
-00017290: 7320 7765 2075 7365 6420 7765 7265 206f  s we used were o
-000172a0: 6e6c 7920 7477 6f20 676c 6f62 616c 204d  nly two global M
-000172b0: 5242 4820 6661 6d69 6c69 6573 2066 6f72  RBH families for
-000172c0: 2074 6865 2070 7572 706f 7365 206f 6620   the purpose of 
-000172d0: 696c 6c75 7374 7261 7469 6f6e 2e20 546f  illustration. To
-000172e0: 2061 6371 7569 7265 2061 6e20 6163 6375   acquire an accu
-000172f0: 7261 7465 2070 726f 6669 6c65 206f 6620  rate profile of 
-00017300: 7468 6520 7375 6273 7469 7475 7469 6f6e  the substitution
-00017310: 2072 6174 6520 7661 7269 6174 696f 6e2c   rate variation,
-00017320: 206f 7274 686f 6c6f 6775 6573 2061 7420   orthologues at 
-00017330: 7468 6520 7768 6f6c 6520 6765 6e6f 6d65  the whole genome
-00017340: 2073 6361 6c65 2073 686f 756c 6420 6265   scale should be
-00017350: 2075 7365 642e 0a0a 2323 2043 6974 6174   used...## Citat
-00017360: 696f 6e0a 200a 506c 6561 7365 2063 6974  ion. .Please cit
-00017370: 6520 7573 2061 7420 6874 7470 733a 2f2f  e us at https://
-00017380: 646f 692e 6f72 672f 3130 2e31 3030 372f  doi.org/10.1007/
-00017390: 3937 382d 312d 3037 3136 2d32 3536 312d  978-1-0716-2561-
-000173a0: 335f 3120 616e 6420 6874 7470 733a 2f2f  3_1 and https://
-000173b0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
-000173c0: 6269 6f69 6e66 6f72 6d61 7469 6373 2f62  bioinformatics/b
-000173d0: 7461 6532 3732 2e0a 0a60 6060 0a48 656e  tae272...```.Hen
-000173e0: 6763 6869 2043 6865 6e2c 2041 7274 6875  gchi Chen, Arthu
-000173f0: 7220 5a77 6165 6e65 706f 656c 2028 3230  r Zwaenepoel (20
-00017400: 3233 292e 2049 6e66 6572 656e 6365 206f  23). Inference o
-00017410: 6620 416e 6369 656e 7420 506f 6c79 706c  f Ancient Polypl
-00017420: 6f69 6479 2066 726f 6d20 4765 6e6f 6d69  oidy from Genomi
-00017430: 6320 4461 7461 2e20 496e 3a20 5661 6e20  c Data. In: Van 
-00017440: 6465 2050 6565 722c 2059 2e20 2865 6473  de Peer, Y. (eds
-00017450: 2920 506f 6c79 706c 6f69 6479 2e20 4d65  ) Polyploidy. Me
-00017460: 7468 6f64 7320 696e 204d 6f6c 6563 756c  thods in Molecul
-00017470: 6172 2042 696f 6c6f 6779 2c20 766f 6c20  ar Biology, vol 
-00017480: 3235 3435 2e20 4875 6d61 6e61 2c20 4e65  2545. Humana, Ne
-00017490: 7720 596f 726b 2c20 4e59 2e20 6874 7470  w York, NY. http
-000174a0: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
-000174b0: 3030 372f 3937 382d 312d 3037 3136 2d32  007/978-1-0716-2
-000174c0: 3536 312d 335f 310a 4865 6e67 6368 6920  561-3_1.Hengchi 
-000174d0: 4368 656e 2c20 4172 7468 7572 205a 7761  Chen, Arthur Zwa
-000174e0: 656e 6570 6f65 6c2c 2059 7665 7320 5661  enepoel, Yves Va
-000174f0: 6e20 6465 2050 6565 7220 2832 3032 3429  n de Peer (2024)
-00017500: 2e20 7767 6420 7632 3a20 6120 7375 6974  . wgd v2: a suit
-00017510: 6520 6f66 2074 6f6f 6c73 2074 6f20 756e  e of tools to un
-00017520: 636f 7665 7220 616e 6420 6461 7465 2061  cover and date a
-00017530: 6e63 6965 6e74 2070 6f6c 7970 6c6f 6964  ncient polyploid
-00017540: 7920 616e 6420 7768 6f6c 652d 6765 6e6f  y and whole-geno
-00017550: 6d65 2064 7570 6c69 6361 7469 6f6e 2e20  me duplication. 
-00017560: 4269 6f69 6e66 6f72 6d61 7469 6373 2c20  Bioinformatics, 
-00017570: 6874 7470 733a 2f2f 646f 692e 6f72 672f  https://doi.org/
-00017580: 3130 2e31 3039 332f 6269 6f69 6e66 6f72  10.1093/bioinfor
-00017590: 6d61 7469 6373 2f62 7461 6532 3732 0a60  matics/btae272.`
-000175a0: 6060 0a0a 466f 7220 6369 7461 7469 6f6e  ``..For citation
-000175b0: 206f 6620 7468 6520 746f 6f6c 7320 7573   of the tools us
-000175c0: 6564 2069 6e20 7767 642c 2070 6c65 6173  ed in wgd, pleas
-000175d0: 6520 636f 6e73 756c 7420 7468 6520 646f  e consult the do
-000175e0: 6375 6d65 6e74 6174 696f 6e20 6174 0a68  cumentation at.h
-000175f0: 7474 7073 3a2f 2f77 6764 7632 2e72 6561  ttps://wgdv2.rea
-00017600: 6474 6865 646f 6373 2e69 6f2f 656e 2f6c  dthedocs.io/en/l
-00017610: 6174 6573 742f 6369 7461 7469 6f6e 2e68  atest/citation.h
-00017620: 746d 6c2e 0a0a                           tml...
+000067d0: 2035 0a2d 6764 2c20 2d2d 6775 6964 652c   5.-gd, --guide,
+000067e0: 2074 6865 2072 6567 696d 6520 7265 7369   the regime resi
+000067f0: 6469 6e67 2061 6e63 686f 7273 2c20 6465  ding anchors, de
+00006800: 6661 756c 7420 5365 676d 656e 740a 2d70  fault Segment.-p
+00006810: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+00006820: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+00006830: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+00006840: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+00006850: 616b 7320 696e 2070 6561 6b20 6669 6e64  aks in peak find
+00006860: 696e 6720 7374 6570 732c 2064 6566 6175  ing steps, defau
+00006870: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
+00006880: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
+00006890: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
+000068a0: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
+000068b0: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
+000068c0: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
+000068d0: 6b64 2c20 2d2d 6b73 746f 6461 7465 2c20  kd, --kstodate, 
+000068e0: 7468 6520 7261 6e67 6520 6f66 204b 7320  the range of Ks 
+000068f0: 746f 2062 6520 6461 7465 6420 696e 2068  to be dated in h
+00006900: 6575 7269 7374 6963 2073 6561 7263 682c  euristic search,
+00006910: 2064 6566 6175 6c74 2028 302e 352c 2031   default (0.5, 1
+00006920: 2e35 290a 2d78 6c2c 202d 2d78 6c69 6d2c  .5).-xl, --xlim,
+00006930: 2074 6865 2078 2061 7869 7320 6c69 6d69   the x axis limi
+00006940: 7420 6f66 2047 4d4d 204b 7320 6469 7374  t of GMM Ks dist
+00006950: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
+00006960: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
+00006970: 206c 696d 6974 206f 6620 474d 4d20 4b73   limit of GMM Ks
+00006980: 2064 6973 7472 6962 7574 696f 6e0a 2d2d   distribution.--
+00006990: 6d61 6e75 616c 7365 742c 2066 6c61 6720  manualset, flag 
+000069a0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000069b0: 746f 206f 7574 7075 7420 616e 6368 6f72  to output anchor
+000069c0: 2070 6169 7273 2077 6974 6820 6d61 6e75   pairs with manu
+000069d0: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
+000069e0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+000069f0: 6173 2073 6574 2c20 6d61 6e75 616c 6c79  as set, manually
+00006a00: 2073 6574 204b 7320 7261 6e67 6520 776f   set Ks range wo
+00006a10: 756c 6420 6265 206f 7574 7075 7474 6564  uld be outputted
+00006a20: 0a2d 2d63 692c 2074 6865 2063 6f6e 6669  .--ci, the confi
+00006a30: 6465 6e63 6520 6c65 7665 6c20 6f66 206c  dence level of l
+00006a40: 6f67 2d6e 6f72 6d61 6c20 6469 7374 7269  og-normal distri
+00006a50: 6275 7469 6f6e 2074 6f20 6461 7465 2c20  bution to date, 
+00006a60: 6465 6661 756c 7420 3935 0a2d 2d68 6472  default 95.--hdr
+00006a70: 2c20 7468 6520 6869 6768 6573 7420 6465  , the highest de
+00006a80: 6e73 6974 7920 7265 6769 6f6e 2028 4844  nsity region (HD
+00006a90: 5229 2061 7070 6c69 6564 2069 6e20 7468  R) applied in th
+00006aa0: 6520 7365 676d 656e 742d 6775 6964 6564  e segment-guided
+00006ab0: 2061 6e63 686f 7220 7061 6972 204b 7320   anchor pair Ks 
+00006ac0: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+00006ad0: 6661 756c 7420 3935 0a2d 2d68 6575 7269  fault 95.--heuri
+00006ae0: 7374 6963 2c20 666c 6167 206f 7074 696f  stic, flag optio
+00006af0: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00006b00: 6974 6961 7465 2068 6575 7269 7374 6963  itiate heuristic
+00006b10: 206d 6574 686f 6420 6f66 2064 6566 696e   method of defin
+00006b20: 696e 6720 4349 2066 6f72 2064 6174 696e  ing CI for datin
+00006b30: 672c 2069 6620 7468 6520 666c 6167 2077  g, if the flag w
+00006b40: 6173 2073 6574 2c20 7468 6520 6865 7572  as set, the heur
+00006b50: 6973 7469 6320 6d65 7468 6f64 2077 6f75  istic method wou
+00006b60: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+00006b70: 2d6b 632c 202d 2d6b 7363 7574 6f66 662c  -kc, --kscutoff,
+00006b80: 2074 6865 204b 7320 7361 7475 7261 7469   the Ks saturati
+00006b90: 6f6e 2063 7574 6f66 6620 696e 2064 6174  on cutoff in dat
+00006ba0: 696e 672c 2064 6566 6175 6c74 2035 0a2d  ing, default 5.-
+00006bb0: 2d6b 6565 7074 6d70 6669 672c 2066 6c61  -keeptmpfig, fla
+00006bc0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00006bd0: 7220 746f 206b 6565 7020 7465 6d70 6f72  r to keep tempor
+00006be0: 6172 7920 6669 6775 7265 7320 696e 2070  ary figures in p
+00006bf0: 6561 6b20 6669 6e64 696e 6720 7072 6f63  eak finding proc
+00006c00: 6573 732c 2069 6620 7468 6520 666c 6167  ess, if the flag
+00006c10: 2077 6173 2073 6574 2c20 7468 6f73 6520   was set, those 
+00006c20: 6669 6775 7265 7320 776f 756c 6420 6265  figures would be
+00006c30: 206b 6570 740a 6060 600a 0a54 6865 2070   kept.```..The p
+00006c40: 726f 6772 616d 2060 7767 6420 7379 6e60  rogram `wgd syn`
+00006c50: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+00006c60: 2069 6e74 7261 2d20 616e 6420 696e 7465   intra- and inte
+00006c70: 722d 7370 6563 6966 6963 2073 796e 7465  r-specific synte
+00006c80: 6e79 2069 6e66 6572 656e 6365 2e0a 6060  ny inference..``
+00006c90: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+00006ca0: 6573 2067 6666 7320 286f 7074 696f 6e29  es gffs (option)
+00006cb0: 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  .---------------
+00006cc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006cd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006ce0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006cf0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006d00: 2d0a 2d6b 732c 202d 2d6b 735f 6469 7374  -.-ks, --ks_dist
+00006d10: 7269 6275 7469 6f6e 2c20 6b73 2064 6973  ribution, ks dis
+00006d20: 7472 6962 7574 696f 6e20 6461 7461 6669  tribution datafi
+00006d30: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00006d40: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+00006d50: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+00006d60: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+00006d70: 5f73 796e 0a2d 662c 202d 2d66 6561 7475  _syn.-f, --featu
+00006d80: 7265 2c20 7468 6520 6665 6174 7572 6520  re, the feature 
+00006d90: 666f 7220 7061 7273 696e 6720 6765 6e65  for parsing gene
+00006da0: 2049 4473 2066 726f 6d20 4746 4620 6669   IDs from GFF fi
+00006db0: 6c65 732c 2064 6566 6175 6c74 2067 656e  les, default gen
+00006dc0: 650a 2d61 2c20 2d2d 6174 7472 6962 7574  e.-a, --attribut
+00006dd0: 652c 2074 6865 2061 7474 7269 6275 7465  e, the attribute
+00006de0: 2066 6f72 2070 6172 7369 6e67 2074 6865   for parsing the
+00006df0: 2067 656e 6520 4944 7320 6672 6f6d 2074   gene IDs from t
+00006e00: 6865 2047 4646 2066 696c 6573 2c20 6465  he GFF files, de
+00006e10: 6661 756c 7420 4944 0a2d 6174 672c 202d  fault ID.-atg, -
+00006e20: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+00006e30: 666f 2c20 7468 6520 6665 6174 7572 6520  fo, the feature 
+00006e40: 616e 6420 6174 7472 6962 7574 6520 696e  and attribute in
+00006e50: 666f 726d 6174 696f 6e20 6f66 2061 6464  formation of add
+00006e60: 6974 696f 6e61 6c20 6766 6633 2066 696c  itional gff3 fil
+00006e70: 6573 2069 6620 6469 6666 6572 656e 7420  es if different 
+00006e80: 696e 2074 6865 2066 6f72 6d61 7420 6f66  in the format of
+00006e90: 2028 6665 6174 7572 653b 6174 7472 6962   (feature;attrib
+00006ea0: 7574 6529 272c 2064 6566 6175 6c74 204e  ute)', default N
+00006eb0: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
+00006ec0: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
+00006ed0: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
+00006ee0: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
+00006ef0: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
+00006f00: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
+00006f10: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
+00006f20: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
+00006f30: 2073 6361 6666 6f6c 6420 776f 756c 6420   scaffold would 
+00006f40: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
+00006f50: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
+00006f60: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
+00006f70: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
+00006f80: 6566 6175 6c74 2032 3030 0a2d 722c 202d  efault 200.-r, -
+00006f90: 2d6b 735f 7261 6e67 652c 2074 6865 204b  -ks_range, the K
+00006fa0: 7320 7261 6e67 6520 696e 2063 6f6c 6f72  s range in color
+00006fb0: 6564 2064 6f74 706c 6f74 2c20 6465 6661  ed dotplot, defa
+00006fc0: 756c 7420 2830 2c20 3529 0a2d 2d69 6164  ult (0, 5).--iad
+00006fd0: 686f 7265 5f6f 7074 696f 6e73 2c20 7468  hore_options, th
+00006fe0: 6520 7061 7261 6d65 7465 7220 7365 7474  e parameter sett
+00006ff0: 696e 6720 696e 2069 6164 686f 7265 2c20  ing in iadhore, 
+00007000: 6465 6661 756c 7420 6173 2061 2073 7472  default as a str
+00007010: 696e 6720 6f66 206c 656e 6774 6820 7a65  ing of length ze
+00007020: 726f 0a2d 6d67 2c20 2d2d 6d69 6e73 6567  ro.-mg, --minseg
+00007030: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
+00007040: 206c 656e 6774 6820 6f66 2073 6567 6d65   length of segme
+00007050: 6e74 7320 746f 2069 6e63 6c75 6465 2069  nts to include i
+00007060: 6e20 7261 7469 6f20 6966 203c 3d20 312c  n ratio if <= 1,
+00007070: 2064 6566 6175 6c74 2031 3030 3030 0a2d   default 10000.-
+00007080: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+00007090: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000070a0: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+000070b0: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+000070c0: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+000070d0: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+000070e0: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+000070f0: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+00007100: 6570 740a 2d6d 676e 2c20 2d2d 6d69 6e67  ept.-mgn, --ming
+00007110: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+00007120: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+00007130: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+00007140: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+00007150: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+00007160: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+00007170: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+00007180: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+00007190: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+000071a0: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+000071b0: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+000071c0: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+000071d0: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+000071e0: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+000071f0: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+00007200: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+00007210: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+00007220: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00007230: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+00007240: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+00007250: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+00007260: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+00007270: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+00007280: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+00007290: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+000072a0: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+000072b0: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+000072c0: 350a 2d67 722c 202d 2d67 6973 7472 622c  5.-gr, --gistrb,
+000072d0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000072e0: 6574 6865 7220 746f 2075 7365 2067 6973  ether to use gis
+000072f0: 745f 7261 696e 626f 7720 6173 2063 6f6c  t_rainbow as col
+00007300: 6f72 206d 6170 206f 6620 646f 7470 6c6f  or map of dotplo
+00007310: 740a 2d6e 2c20 2d2d 6e74 6872 6561 6473  t.-n, --nthreads
+00007320: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00007330: 7468 7265 6164 7320 746f 2075 7365 2069  threads to use i
+00007340: 6e20 7379 6e74 656e 7920 696e 6665 7265  n synteny infere
+00007350: 6e63 652c 2064 6566 6175 6c74 2034 0a60  nce, default 4.`
+00007360: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+00007370: 6077 6764 2076 697a 6020 6361 6e20 7265  `wgd viz` can re
+00007380: 616c 697a 6520 7468 6520 7669 7375 616c  alize the visual
+00007390: 697a 6174 696f 6e20 6f66 202a 4b2a 3c73  ization of *K*<s
+000073a0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+000073b0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+000073c0: 7379 6e74 656e 792e 0a60 6060 0a77 6764  synteny..```.wgd
+000073d0: 2076 697a 2028 6f70 7469 6f6e 290a 2d2d   viz (option).--
+000073e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000073f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007400: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007410: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007420: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+00007430: 642c 202d 2d64 6174 6166 696c 652c 2074  d, --datafile, t
+00007440: 6865 204b 7320 6461 7461 6669 6c65 2c20  he Ks datafile, 
+00007450: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
+00007460: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+00007470: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00007480: 2064 6566 6175 6c74 2077 6764 5f76 697a   default wgd_viz
+00007490: 0a2d 7372 2c20 2d2d 7370 6169 722c 2074  .-sr, --spair, t
+000074a0: 6865 2073 7065 6369 6573 2070 6169 7220  he species pair 
+000074b0: 746f 2062 6520 706c 6f74 7465 642c 2064  to be plotted, d
+000074c0: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
+000074d0: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
+000074e0: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
+000074f0: 6520 7469 6d65 730a 2d66 612c 202d 2d66  e times.-fa, --f
+00007500: 6f63 7573 3261 6c6c 2c20 7365 7420 666f  ocus2all, set fo
+00007510: 6361 6c20 7370 6563 6965 7320 616e 6420  cal species and 
+00007520: 6c65 7420 7370 6563 6965 7320 7061 6972  let species pair
+00007530: 2074 6f20 6265 2062 6574 7765 656e 2066   to be between f
+00007540: 6f63 616c 2061 6e64 2061 6c6c 2074 6865  ocal and all the
+00007550: 2072 656d 6169 6e69 6e67 2073 7065 6369   remaining speci
+00007560: 6573 2c20 6465 6661 756c 7420 4e6f 6e65  es, default None
+00007570: 0a2d 6773 2c20 2d2d 6773 6d61 702c 2074  .-gs, --gsmap, t
+00007580: 6865 2067 656e 6520 6e61 6d65 2d73 7065  he gene name-spe
+00007590: 6369 6573 206e 616d 6520 6d61 702c 2064  cies name map, d
+000075a0: 6566 6175 6c74 204e 6f6e 650a 2d73 702c  efault None.-sp,
+000075b0: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
+000075c0: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+000075d0: 2074 6f20 7065 7266 6f72 6d20 7261 7465   to perform rate
+000075e0: 2063 6f72 7265 6374 696f 6e2c 2064 6566   correction, def
+000075f0: 6175 6c74 204e 6f6e 652c 2069 6620 4e6f  ault None, if No
+00007600: 6e65 2077 6173 2067 6976 656e 2c20 7468  ne was given, th
+00007610: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+00007620: 6e20 616e 616c 7973 6973 2077 6f75 6c64  n analysis would
+00007630: 2062 6520 6361 6c6c 6564 206f 6666 0a2d   be called off.-
+00007640: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
+00007650: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00007660: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
+00007670: 6375 7276 6520 7570 6f6e 2068 6973 746f  curve upon histo
+00007680: 6772 616d 2c20 6966 2074 6865 2066 6c61  gram, if the fla
+00007690: 6720 7761 7320 7365 742c 206b 6465 2063  g was set, kde c
+000076a0: 7572 7665 2077 6f75 6c64 2062 6520 6164  urve would be ad
+000076b0: 6465 640a 2d72 772c 202d 2d72 6577 6569  ded.-rw, --rewei
+000076c0: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
+000076d0: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
+000076e0: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
+000076f0: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
+00007700: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
+00007710: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
+00007720: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
+00007730: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
+00007740: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
+00007750: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007760: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
+00007770: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
+00007780: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
+00007790: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
+000077a0: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
+000077b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000077c0: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
+000077d0: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
+000077e0: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
+000077f0: 6772 6f75 700a 2d69 7465 722c 202d 2d65  group.-iter, --e
+00007800: 6d5f 6974 6572 6174 696f 6e73 2c20 7468  m_iterations, th
+00007810: 6520 6d61 7869 6d75 6d20 454d 2069 7465  e maximum EM ite
+00007820: 7261 7469 6f6e 732c 2064 6566 6175 6c74  rations, default
+00007830: 2032 3030 0a2d 696e 6974 2c20 2d2d 656d   200.-init, --em
+00007840: 5f69 6e69 7469 616c 697a 6174 696f 6e73  _initializations
+00007850: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
+00007860: 2069 6e69 7469 616c 697a 6174 696f 6e73   initializations
+00007870: 2c20 6465 6661 756c 7420 3230 300a 2d70  , default 200.-p
+00007880: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+00007890: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+000078a0: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+000078b0: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+000078c0: 616b 732c 2064 6566 6175 6c74 2030 2e31  aks, default 0.1
+000078d0: 0a2d 7268 2c20 2d2d 7265 6c5f 6865 6967  .-rh, --rel_heig
+000078e0: 6874 2c20 7468 6520 7265 6c61 7469 7665  ht, the relative
+000078f0: 2068 6569 6768 7420 6174 2077 6869 6368   height at which
+00007900: 2074 6865 2070 6561 6b20 7769 6474 6820   the peak width 
+00007910: 6973 206d 6561 7375 7265 642c 2064 6566  is measured, def
+00007920: 6175 6c74 2030 2e34 0a2d 736d 2c20 2d2d  ault 0.4.-sm, --
+00007930: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
+00007940: 676d 656e 7473 2064 6174 6166 696c 652c  gments datafile,
+00007950: 2064 6566 6175 6c74 204e 6f6e 650a 2d6d   default None.-m
+00007960: 6c2c 202d 2d6d 696e 6c65 6e2c 2074 6865  l, --minlen, the
+00007970: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+00007980: 6f66 2061 2073 6361 6666 6f6c 6420 746f  of a scaffold to
+00007990: 2062 6520 696e 636c 7564 6564 2069 6e20   be included in 
+000079a0: 646f 7470 6c6f 742c 2064 6566 6175 6c74  dotplot, default
+000079b0: 202d 312c 2069 6620 2d31 2077 6173 2073   -1, if -1 was s
+000079c0: 6574 2c20 7468 6520 3130 2520 6f66 2074  et, the 10% of t
+000079d0: 6865 206c 6f6e 6765 7374 2073 6361 6666  he longest scaff
+000079e0: 6f6c 6473 2077 696c 6c20 6265 2073 6574  olds will be set
+000079f0: 0a2d 6d73 2c20 2d2d 6d61 7873 697a 652c  .-ms, --maxsize,
+00007a00: 2074 6865 206d 6178 696d 756d 2066 616d   the maximum fam
+00007a10: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
+00007a20: 6e63 6c75 6465 642c 2064 6566 6175 6c74  ncluded, default
+00007a30: 2032 3030 0a2d 6170 2c20 2d2d 616e 6368   200.-ap, --anch
+00007a40: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+00007a50: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+00007a60: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00007a70: 6e65 0a2d 6d74 2c20 2d2d 6d75 6c74 6970  ne.-mt, --multip
+00007a80: 6c69 636f 6e2c 2074 6865 206d 756c 7469  licon, the multi
+00007a90: 706c 6963 6f6e 7320 6461 7461 6669 6c65  plicons datafile
+00007aa0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00007ab0: 6774 2c20 2d2d 6765 6e65 7461 626c 652c  gt, --genetable,
+00007ac0: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
+00007ad0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00007ae0: 7420 4e6f 6e65 0a2d 6d67 2c20 2d2d 6d69  t None.-mg, --mi
+00007af0: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
+00007b00: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
+00007b10: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
+00007b20: 6465 2c20 696e 2072 6174 696f 2069 6620  de, in ratio if 
+00007b30: 3c3d 2031 2c20 6465 6661 756c 7420 3130  <= 1, default 10
+00007b40: 3030 300a 2d6d 676e 2c20 2d2d 6d69 6e67  000.-mgn, --ming
+00007b50: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+00007b60: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+00007b70: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+00007b80: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+00007b90: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+00007ba0: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+00007bb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00007bc0: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+00007bd0: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+00007be0: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+00007bf0: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+00007c00: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+00007c10: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+00007c20: 6570 740a 2d65 706b 2c20 2d2d 6578 7472  ept.-epk, --extr
+00007c30: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
+00007c40: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
+00007c50: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
+00007c60: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+00007c70: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
+00007c80: 7420 4e6f 6e65 0a2d 7061 672c 202d 2d70  t None.-pag, --p
+00007c90: 6c6f 7461 7067 6d6d 2c20 666c 6167 206f  lotapgmm, flag o
+00007ca0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00007cb0: 6f20 636f 6e64 7563 7420 616e 6420 706c  o conduct and pl
+00007cc0: 6f74 206d 6978 7475 7265 206d 6f64 656c  ot mixture model
+00007cd0: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+00007ce0: 2069 6e20 7468 6520 6d69 7865 6420 4b73   in the mixed Ks
+00007cf0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
+00007d00: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00007d10: 6574 2c20 7468 6520 6d69 7874 7572 6520  et, the mixture 
+00007d20: 6d6f 6465 6c69 6e67 206f 6620 616e 6368  modeling of anch
+00007d30: 6f72 204b 7320 776f 756c 6420 6265 2063  or Ks would be c
+00007d40: 6f6e 6475 6374 6564 2061 6e64 2070 6c6f  onducted and plo
+00007d50: 7474 6564 0a2d 7065 6d2c 202d 2d70 6c6f  tted.-pem, --plo
+00007d60: 7465 6c6d 6d2c 2066 6c61 6720 6f70 7469  telmm, flag opti
+00007d70: 6f6e 2c20 7768 6574 6865 7220 746f 2063  on, whether to c
+00007d80: 6f6e 6475 6374 2061 6e64 2070 6c6f 7420  onduct and plot 
+00007d90: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
+00007da0: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
+00007db0: 6520 4b73 2069 6e20 7468 6520 6d69 7865  e Ks in the mixe
+00007dc0: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
+00007dd0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
+00007de0: 6173 2073 6574 2c20 7468 6520 656c 6d6d  as set, the elmm
+00007df0: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+00007e00: 6720 6f66 2070 6172 616e 6f6d 6520 4b73  g of paranome Ks
+00007e10: 2077 6f75 6c64 2062 6520 636f 6e64 7563   would be conduc
+00007e20: 7465 6420 616e 6420 706c 6f74 7465 640a  ted and plotted.
+00007e30: 2d63 2c20 2d2d 636f 6d70 6f6e 656e 7473  -c, --components
+00007e40: 2c20 7468 6520 7261 6e67 6520 6f66 2074  , the range of t
+00007e50: 6865 206e 756d 6265 7220 6f66 2063 6f6d  he number of com
+00007e60: 706f 6e65 6e74 7320 746f 2066 6974 2069  ponents to fit i
+00007e70: 6e20 616e 6368 6f72 204b 7320 6d69 7874  n anchor Ks mixt
+00007e80: 7572 6520 6d6f 6465 6c69 6e67 2c20 6465  ure modeling, de
+00007e90: 6661 756c 7420 2831 2c34 290a 2d70 7379  fault (1,4).-psy
+00007ea0: 2c20 2d2d 706c 6f74 7379 6e2c 2066 6c61  , --plotsyn, fla
+00007eb0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007ec0: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00007ed0: 6520 7379 6e74 656e 7920 706c 6f74 2c20  e synteny plot, 
+00007ee0: 6f6e 6c79 2077 6865 6e20 7468 6520 666c  only when the fl
+00007ef0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00007f00: 7379 6e74 656e 7920 706c 6f74 2077 6f75  synteny plot wou
+00007f10: 6c64 2062 6520 7072 6f64 7563 6564 0a2d  ld be produced.-
+00007f20: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+00007f30: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+00007f40: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+00007f50: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+00007f60: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+00007f70: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+00007f80: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+00007f90: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+00007fa0: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+00007fb0: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+00007fc0: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+00007fd0: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+00007fe0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00007ff0: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+00008000: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+00008010: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+00008020: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+00008030: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+00008040: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+00008050: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+00008060: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+00008070: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+00008080: 350a 2d78 6c2c 202d 2d78 6c69 6d2c 2074  5.-xl, --xlim, t
+00008090: 6865 2078 2061 7869 7320 6c69 6d69 7420  he x axis limit 
+000080a0: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+000080b0: 6f6e 0a2d 796c 2c20 2d2d 796c 696d 2c20  on.-yl, --ylim, 
+000080c0: 7468 6520 7920 6178 6973 206c 696d 6974  the y axis limit
+000080d0: 206f 6620 4b73 2064 6973 7472 6962 7574   of Ks distribut
+000080e0: 696f 6e0a 2d61 646f 2c20 2d2d 6164 6a75  ion.-ado, --adju
+000080f0: 7374 6f72 7468 6f2c 2066 6c61 6720 6f70  stortho, flag op
+00008100: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00008110: 2061 646a 7573 7420 7468 6520 6869 7374   adjust the hist
+00008120: 6f67 7261 6d20 6865 6967 6874 206f 6620  ogram height of 
+00008130: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2061  orthologous Ks a
+00008140: 7320 746f 206d 6174 6368 2074 6865 2068  s to match the h
+00008150: 6569 6768 7420 6f66 2070 6172 616c 6f67  eight of paralog
+00008160: 6f75 7320 4b73 2c20 6966 2074 6865 2066  ous Ks, if the f
+00008170: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00008180: 2061 646a 7573 746d 656e 7420 776f 756c   adjustment woul
+00008190: 6420 6265 2063 6f6e 6475 6374 6564 0a2d  d be conducted.-
+000081a0: 6164 662c 202d 2d61 646a 7573 7466 6163  adf, --adjustfac
+000081b0: 746f 722c 2074 6865 2061 646a 7573 746d  tor, the adjustm
+000081c0: 656e 7420 6661 6374 6f72 206f 6620 6f72  ent factor of or
+000081d0: 7468 6f6c 6f67 6f75 7320 4b73 2c20 6465  thologous Ks, de
+000081e0: 6661 756c 7420 302e 350a 2d6f 612c 202d  fault 0.5.-oa, -
+000081f0: 2d6f 6b61 6c70 6861 2c20 7468 6520 6f70  -okalpha, the op
+00008200: 6163 6974 7920 6f66 206f 7274 686f 6c6f  acity of ortholo
+00008210: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+00008220: 7469 6f6e 2069 6e20 6d69 7865 6420 706c  tion in mixed pl
+00008230: 6f74 2c20 6465 6661 756c 7420 302e 350a  ot, default 0.5.
+00008240: 2d63 732c 202d 2d63 6c61 7373 6963 2c20  -cs, --classic, 
+00008250: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00008260: 7468 6572 2074 6f20 6472 6177 206d 6978  ther to draw mix
+00008270: 6564 204b 7320 706c 6f74 2069 6e20 6120  ed Ks plot in a 
+00008280: 636c 6173 7369 6320 6d61 6e6e 6572 2077  classic manner w
+00008290: 6865 7265 2074 6865 2066 756c 6c20 6f72  here the full or
+000082a0: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
+000082b0: 7472 6962 7574 696f 6e20 6973 2064 7261  tribution is dra
+000082c0: 7765 642c 2069 6620 7468 6520 666c 6167  wed, if the flag
+000082d0: 2077 6173 2073 6574 2c20 7468 6520 636c   was set, the cl
+000082e0: 6173 7369 6320 6d69 7865 6420 4b73 2070  assic mixed Ks p
+000082f0: 6c6f 7420 776f 756c 6420 6265 2064 7261  lot would be dra
+00008300: 776e 0a2d 7461 2c20 2d2d 746f 7061 7272  wn.-ta, --toparr
+00008310: 6f77 2c20 666c 6167 206f 7074 696f 6e2c  ow, flag option,
+00008320: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
+00008330: 7374 2074 6865 2061 7272 6f77 2074 6f20  st the arrow to 
+00008340: 6265 2061 7420 7468 6520 746f 7020 6f66  be at the top of
+00008350: 2074 6865 2070 6c6f 742c 2069 6e73 7465   the plot, inste
+00008360: 6164 206f 6620 6265 696e 6720 636f 6f72  ad of being coor
+00008370: 6469 6e61 7465 6420 6173 2074 6865 204b  dinated as the K
+00008380: 4445 206f 6620 7468 6520 6f72 7468 6f6c  DE of the orthol
+00008390: 6f67 6f75 7320 4b73 2064 6973 7472 6962  ogous Ks distrib
+000083a0: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
+000083b0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+000083c0: 6172 726f 7720 776f 756c 6420 6265 2073  arrow would be s
+000083d0: 6574 2061 7420 7468 6520 746f 700a 2d6e  et at the top.-n
+000083e0: 612c 202d 2d6e 6f64 6561 7665 7261 6765  a, --nodeaverage
+000083f0: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+00008400: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+00008410: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
+00008420: 686f 6420 666f 7220 6465 2d72 6564 756e  hod for de-redun
+00008430: 6461 6e63 792c 2069 6620 7468 6520 666c  dancy, if the fl
+00008440: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00008450: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+00008460: 7468 6f64 2077 6f75 6c64 2062 6520 696e  thod would be in
+00008470: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
+00008480: 6f6f 7473 7472 6170 2c20 7468 6520 6e75  ootstrap, the nu
+00008490: 6d62 6572 206f 6620 626f 6f74 7374 7261  mber of bootstra
+000084a0: 7020 7265 706c 6963 6174 6573 206f 6620  p replicates of 
+000084b0: 6f72 7468 6f6c 6f67 204b 7320 6469 7374  ortholog Ks dist
+000084c0: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
+000084d0: 6420 706c 6f74 0a2d 6772 2c20 2d2d 6769  d plot.-gr, --gi
+000084e0: 7374 7262 2c20 666c 6167 206f 7074 696f  strb, flag optio
+000084f0: 6e2c 2077 6865 7468 6572 2074 6f20 7573  n, whether to us
+00008500: 6520 6769 7374 5f72 6169 6e62 6f77 2061  e gist_rainbow a
+00008510: 7320 636f 6c6f 7220 6d61 7020 6f66 2064  s color map of d
+00008520: 6f74 706c 6f74 0a2d 6e2c 202d 2d6e 7468  otplot.-n, --nth
+00008530: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+00008540: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+00008550: 7573 6520 696e 2062 6f6f 7473 7472 6170  use in bootstrap
+00008560: 2073 616d 706c 696e 672c 2064 6566 6175   sampling, defau
+00008570: 6c74 2031 0a60 6060 0a0a 2323 2055 7361  lt 1.```..## Usa
+00008580: 6765 0a0a 4865 7265 2077 6520 7072 6f76  ge..Here we prov
+00008590: 6964 6564 2074 6865 2062 6173 6963 2075  ided the basic u
+000085a0: 7361 6765 2066 6f72 2065 6163 6820 7072  sage for each pr
+000085b0: 6f67 7261 6d20 616e 6420 7468 6520 7265  ogram and the re
+000085c0: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
+000085d0: 7320 616e 6420 7375 6767 6573 7469 6f6e  s and suggestion
+000085e0: 7320 6f6e 2070 6172 616d 6574 6572 697a  s on parameteriz
+000085f0: 6174 696f 6e2e 2041 2072 656d 696e 6465  ation. A reminde
+00008600: 7220 7468 6174 2074 6865 2067 6976 656e  r that the given
+00008610: 2063 6f72 6573 2061 6e64 2074 6872 6561   cores and threa
+00008620: 6473 2063 616e 2073 6967 6e69 6669 6361  ds can significa
+00008630: 6e74 6c79 2069 6d70 6163 7420 7468 6520  ntly impact the 
+00008640: 7275 6e20 7469 6d65 2061 6e64 2074 6875  run time and thu
+00008650: 7320 7765 2061 6464 6564 2073 6f6d 6520  s we added some 
+00008660: 7265 706f 7274 2069 6e66 6f72 6d61 7469  report informati
+00008670: 6f6e 2070 6572 7461 696e 696e 6720 746f  on pertaining to
+00008680: 2074 6865 2073 7973 7465 6d20 6f66 2075   the system of u
+00008690: 7365 7273 2074 6f20 6661 6369 6c69 7461  sers to facilita
+000086a0: 7465 2074 6865 2065 6666 6963 6965 6e74  te the efficient
+000086b0: 2073 6574 7469 6e67 206f 6620 7468 7265   setting of thre
+000086c0: 6164 7320 616e 6420 6d65 6d6f 7279 2e20  ads and memory. 
+000086d0: 5468 6520 6c6f 6769 6361 6c20 4350 5573  The logical CPUs
+000086e0: 2072 6570 6f72 7465 6420 7265 7072 6573   reported repres
+000086f0: 656e 7473 2074 6865 206e 756d 6265 7220  ents the number 
+00008700: 6f66 2070 6879 7369 6361 6c20 636f 7265  of physical core
+00008710: 7320 6d75 6c74 6970 6c69 6564 2062 7920  s multiplied by 
+00008720: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
+00008730: 7265 6164 7320 7468 6174 2063 616e 2072  reads that can r
+00008740: 756e 206f 6e20 6561 6368 2063 6f72 652c  un on each core,
+00008750: 2061 6c73 6f20 6b6e 6f77 6e20 6173 2048   also known as H
+00008760: 7970 6572 2054 6872 6561 6469 6e67 2e20  yper Threading. 
+00008770: 5468 6520 6e75 6d62 6572 206f 6620 6c6f  The number of lo
+00008780: 6769 6361 6c20 4350 5573 206d 6179 206e  gical CPUs may n
+00008790: 6f74 206e 6563 6573 7361 7269 6c79 2062  ot necessarily b
+000087a0: 6520 6571 7569 7661 6c65 6e74 2074 6f20  e equivalent to 
+000087b0: 7468 6520 6163 7475 616c 206e 756d 6265  the actual numbe
+000087c0: 7220 6f66 2043 5055 7320 7468 6520 6375  r of CPUs the cu
+000087d0: 7272 656e 7420 7072 6f63 6573 7320 6361  rrent process ca
+000087e0: 6e20 7573 652e 2054 6865 2061 7661 696c  n use. The avail
+000087f0: 6162 6c65 206d 656d 6f72 7920 7265 6665  able memory refe
+00008800: 7273 2074 6f20 7468 6520 6d65 6d6f 7279  rs to the memory
+00008810: 2074 6861 7420 6361 6e20 6265 2067 6976   that can be giv
+00008820: 656e 2069 6e73 7461 6e74 6c79 2074 6f20  en instantly to 
+00008830: 7072 6f63 6573 7365 7320 7769 7468 6f75  processes withou
+00008840: 7420 7468 6520 7379 7374 656d 2067 6f69  t the system goi
+00008850: 6e67 2069 6e74 6f20 7377 6170 2061 6e64  ng into swap and
+00008860: 2072 6566 6c65 6374 7320 7468 6520 6163   reflects the ac
+00008870: 7475 616c 206d 656d 6f72 7920 6176 6169  tual memory avai
+00008880: 6c61 626c 652e 2054 6865 2066 7265 6520  lable. The free 
+00008890: 6d65 6d6f 7279 2072 6566 6572 7320 746f  memory refers to
+000088a0: 2074 6865 206d 656d 6f72 7920 6e6f 7420   the memory not 
+000088b0: 6265 696e 6720 7573 6564 2061 7420 616c  being used at al
+000088c0: 6c20 287a 6572 6f65 6429 2074 6861 7420  l (zeroed) that 
+000088d0: 6973 2072 6561 6469 6c79 2061 7661 696c  is readily avail
+000088e0: 6162 6c65 2e20 5468 6520 6465 7363 7269  able. The descri
+000088f0: 7074 696f 6e20 6162 6f76 6520 7265 6665  ption above refe
+00008900: 7273 2074 6f20 7468 6520 646f 6375 6d65  rs to the docume
+00008910: 6e74 6174 696f 6e20 6f66 2060 7073 7574  ntation of `psut
+00008920: 696c 602e 0a0a 2323 2320 7767 6420 646d  il`...### wgd dm
+00008930: 640a 0a2a 2a54 6865 2064 656c 696e 6561  d..**The delinea
+00008940: 7469 6f6e 206f 6620 7768 6f6c 6520 7061  tion of whole pa
+00008950: 7261 6e6f 6d65 2a2a 0a60 6060 0a77 6764  ranome**.```.wgd
+00008960: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00008970: 6f65 7275 6c65 6120 2d49 2032 202d 6520  oerulea -I 2 -e 
+00008980: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+00008990: 7020 3520 282d 6e6e 2920 282d 2d74 6f5f  p 5 (-nn) (--to_
+000089a0: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
+000089b0: 6e20 3429 2028 2d6f 2077 6764 5f64 6d64  n 4) (-o wgd_dmd
+000089c0: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
+000089d0: 7029 0a60 6060 200a 0a4e 6f74 6520 7468  p).``` ..Note th
+000089e0: 6174 2077 6520 646f 6e27 7420 7072 6f76  at we don't prov
+000089f0: 6964 6520 7468 6520 6461 7461 206f 6620  ide the data of 
+00008a00: 7468 6973 2063 6f64 696e 6720 7365 7175  this coding sequ
+00008a10: 656e 6365 2028 6364 7329 2066 696c 6520  ence (cds) file 
+00008a20: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+00008a30: 6c65 6160 2062 7574 2069 7420 6361 6e20  lea` but it can 
+00008a40: 6265 2065 6173 696c 7920 646f 776e 6c6f  be easily downlo
+00008a50: 6164 6564 2061 7420 5b50 6879 746f 7a6f  aded at [Phytozo
+00008a60: 6d65 5d28 6874 7470 733a 2f2f 7068 7974  me](https://phyt
+00008a70: 6f7a 6f6d 652d 6e65 7874 2e6a 6769 2e64  ozome-next.jgi.d
+00008a80: 6f65 2e67 6f76 2f69 6e66 6f2f 4163 6f65  oe.gov/info/Acoe
+00008a90: 7275 6c65 615f 7633 5f31 2920 2873 616d  rulea_v3_1) (sam
+00008aa0: 6520 666f 7220 6f74 6865 7220 5573 6167  e for other Usag
+00008ab0: 6520 646f 6329 2e20 4e6f 7465 2074 6861  e doc). Note tha
+00008ac0: 7420 696e 2074 6865 2069 7373 7565 7320  t in the issues 
+00008ad0: 736f 6d65 2075 7365 7273 2064 6964 6e27  some users didn'
+00008ae0: 7420 646f 776e 6c6f 6164 2074 6865 2060  t download the `
+00008af0: 4163 6f65 7275 6c65 615f 3332 325f 7633  Acoerulea_322_v3
+00008b00: 2e31 2e63 6473 5f70 7269 6d61 7279 5472  .1.cds_primaryTr
+00008b10: 616e 7363 7269 7074 4f6e 6c79 2e66 612e  anscriptOnly.fa.
+00008b20: 677a 6020 6275 7420 696e 7374 6561 6420  gz` but instead 
+00008b30: 7468 6520 6041 636f 6572 756c 6561 5f33  the `Acoerulea_3
+00008b40: 3232 5f76 332e 312e 6364 732e 6661 2e67  22_v3.1.cds.fa.g
+00008b50: 7a60 2066 696c 652e 2046 6f72 2074 6865  z` file. For the
+00008b60: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00008b70: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
+00008b80: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00008b90: 6469 7374 7269 6275 7469 6f6e 2c20 2a2a  distribution, **
+00008ba0: 6f6e 6c79 206f 6e65 2074 7261 6e73 6372  only one transcr
+00008bb0: 6970 7420 2874 6865 2070 7269 6d61 7279  ipt (the primary
+00008bc0: 206f 6e65 2920 7065 7220 6765 6e65 2073   one) per gene s
+00008bd0: 686f 756c 6420 6265 2069 6e63 6c75 6465  hould be include
+00008be0: 642a 2a20 7375 6368 2074 6861 7420 7468  d** such that th
+00008bf0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00008c00: 3e20 6973 2072 6561 6c6c 7920 696e 6469  > is really indi
+00008c10: 6361 7469 6e67 2074 6865 2061 6765 206f  cating the age o
+00008c20: 6620 6765 6e65 2064 7570 6c69 6361 7469  f gene duplicati
+00008c30: 6f6e 2065 7665 6e74 2c20 696e 7374 6561  on event, instea
+00008c40: 6420 6f66 2061 6c74 6572 6e61 7469 7665  d of alternative
+00008c50: 2073 706c 6963 696e 672e 2054 7261 6e73   splicing. Trans
+00008c60: 6372 6970 746f 6d65 2064 6174 6120 7368  criptome data sh
+00008c70: 6f75 6c64 2062 6520 6361 7265 6675 6c6c  ould be carefull
+00008c80: 7920 7472 6561 7465 6420 7769 7468 2064  y treated with d
+00008c90: 652d 7265 6475 6e64 616e 6379 2073 6f20  e-redundancy so 
+00008ca0: 6173 2074 6f20 7265 6475 6365 2074 6865  as to reduce the
+00008cb0: 2066 616c 7365 2070 6f73 6974 6976 6520   false positive 
+00008cc0: 6475 706c 6963 6174 696f 6e20 6275 6d70  duplication bump
+00008cd0: 2063 6175 7365 6420 6279 2070 6572 7661   caused by perva
+00008ce0: 7369 7665 2061 6c74 6572 6e61 7469 7665  sive alternative
+00008cf0: 2073 706c 6963 696e 672e 2048 6572 6520   splicing. Here 
+00008d00: 7468 6520 696e 666c 6174 696f 6e20 6661  the inflation fa
+00008d10: 6374 6f72 2070 6172 616d 6574 6572 2c20  ctor parameter, 
+00008d20: 6769 7665 6e20 6279 2060 2d49 6020 6f72  given by `-I` or
+00008d30: 2060 2d2d 696e 666c 6174 696f 6e60 2c20   `--inflation`, 
+00008d40: 6166 6665 6374 7320 7468 6520 6772 616e  affects the gran
+00008d50: 756c 6172 6974 7920 6f72 2072 6573 6f6c  ularity or resol
+00008d60: 7574 696f 6e20 6f66 2074 6865 2063 6c75  ution of the clu
+00008d70: 7374 6572 696e 6720 6f75 7463 6f6d 6520  stering outcome 
+00008d80: 616e 6420 696d 706c 6963 6974 6c79 2063  and implicitly c
+00008d90: 6f6e 7472 6f6c 6c73 2074 6865 206e 756d  ontrolls the num
+00008da0: 6265 7220 6f66 2063 6c75 7374 6572 732c  ber of clusters,
+00008db0: 2077 6974 6820 6c6f 7720 7661 6c75 6573   with low values
+00008dc0: 2073 7563 6820 6173 2031 2e33 206f 7220   such as 1.3 or 
+00008dd0: 312e 3420 6c65 6164 696e 6720 746f 2066  1.4 leading to f
+00008de0: 6577 6572 2062 7574 206c 6172 6765 7220  ewer but larger 
+00008df0: 636c 7573 7465 7273 2061 6e64 2068 6967  clusters and hig
+00008e00: 6820 7661 6c75 6573 2073 7563 6820 6173  h values such as
+00008e10: 2035 206f 7220 3620 6c65 6164 696e 6720   5 or 6 leading 
+00008e20: 746f 206d 6f72 6520 6275 7420 736d 616c  to more but smal
+00008e30: 6c65 7220 636c 7573 7465 7273 2e20 5765  ler clusters. We
+00008e40: 2073 6574 2074 6865 2064 6566 6175 6c74   set the default
+00008e50: 2076 616c 7565 2061 7320 3220 6173 2073   value as 2 as s
+00008e60: 7567 6765 7374 6564 2062 7920 5b4d 434c  uggested by [MCL
+00008e70: 5d28 6874 7470 733a 2f2f 6d69 6361 6e73  ](https://micans
+00008e80: 2e6f 7267 2f6d 636c 2f29 2e20 5468 6520  .org/mcl/). The 
+00008e90: 652d 7661 6c75 6520 6375 742d 6f66 6620  e-value cut-off 
+00008ea0: 666f 7220 7365 7175 656e 6365 2073 696d  for sequence sim
+00008eb0: 696c 6172 6974 792c 2067 6976 656e 2062  ilarity, given b
+00008ec0: 7920 602d 6560 206f 7220 602d 2d65 7661  y `-e` or `--eva
+00008ed0: 6c60 2c20 7768 6963 6820 6465 6e6f 7465  l`, which denote
+00008ee0: 7320 7468 6520 6578 7065 6374 6564 2076  s the expected v
+00008ef0: 616c 7565 206f 6620 7468 6520 6869 7420  alue of the hit 
+00008f00: 7175 616e 7469 6669 6573 2074 6865 206e  quantifies the n
+00008f10: 756d 6265 7220 6f66 2061 6c69 676e 6d65  umber of alignme
+00008f20: 6e74 7320 6f66 2073 696d 696c 6172 206f  nts of similar o
+00008f30: 7220 6265 7474 6572 2071 7561 6c69 7479  r better quality
+00008f40: 2074 6861 7420 796f 7520 6578 7065 6374   that you expect
+00008f50: 2074 6f20 6669 6e64 2073 6561 7263 6869   to find searchi
+00008f60: 6e67 2074 6869 7320 7175 6572 7920 6167  ng this query ag
+00008f70: 6169 6e73 7420 6120 6461 7461 6261 7365  ainst a database
+00008f80: 206f 6620 7261 6e64 6f6d 2073 6571 7565   of random seque
+00008f90: 6e63 6573 2074 6865 2073 616d 6520 7369  nces the same si
+00008fa0: 7a65 2061 7320 7468 6520 6163 7475 616c  ze as the actual
+00008fb0: 2074 6172 6765 7420 6461 7461 6261 7365   target database
+00008fc0: 2c20 6973 2074 6865 206b 6579 2070 6172  , is the key par
+00008fd0: 616d 6574 6572 206d 6561 7375 7269 6e67  ameter measuring
+00008fe0: 2074 6865 2073 6967 6e69 6669 6361 6e63   the significanc
+00008ff0: 6520 6f66 2061 2068 6974 2c20 7768 6963  e of a hit, whic
+00009000: 6820 6973 2073 6574 2068 6572 6520 6173  h is set here as
+00009010: 2064 6566 6175 6c74 2031 652d 3130 2e20   default 1e-10. 
+00009020: 4e6f 7465 2074 6861 7420 5b44 4941 4d4f  Note that [DIAMO
+00009030: 4e44 5d28 6874 7470 733a 2f2f 6769 7468  ND](https://gith
+00009040: 7562 2e63 6f6d 2f62 6275 6368 6669 6e6b  ub.com/bbuchfink
+00009050: 2f64 6961 6d6f 6e64 2f77 696b 692f 312e  /diamond/wiki/1.
+00009060: 2d54 7574 6f72 6961 6c29 2069 7473 656c  -Tutorial) itsel
+00009070: 6620 6279 2064 6566 6175 6c74 206f 6e6c  f by default onl
+00009080: 7920 7265 706f 7274 7320 616c 6c20 616c  y reports all al
+00009090: 6967 6e6d 656e 7473 2077 6974 6820 652d  ignments with e-
+000090a0: 7661 6c75 6520 3c20 302e 3030 312e 2054  value < 0.001. T
+000090b0: 6865 2070 6572 6365 6e74 6167 6520 6f66  he percentage of
+000090c0: 2075 7070 6572 2068 6974 7320 7573 6564   upper hits used
+000090d0: 2066 6f72 2067 656e 6520 6c65 6e67 7468   for gene length
+000090e0: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
+000090f0: 6769 7665 6e20 6279 2060 2d6e 7060 206f  given by `-np` o
+00009100: 7220 602d 2d6e 6f72 6d61 6c69 7a65 6470  r `--normalizedp
+00009110: 6572 6365 6e74 602c 2077 6869 6368 2064  ercent`, which d
+00009120: 6574 6572 6d69 6e65 7320 7468 6520 7570  etermines the up
+00009130: 7065 7220 7065 7263 656e 7469 6c65 206f  per percentile o
+00009140: 6620 6869 7473 2070 6572 2062 696e 2028  f hits per bin (
+00009150: 6361 7465 676f 7269 7a65 6420 6279 2067  categorized by g
+00009160: 656e 6520 6c65 6e67 7468 2920 7573 6564  ene length) used
+00009170: 2069 6e20 7468 6520 6669 7420 6f66 206c   in the fit of l
+00009180: 696e 6561 7220 7265 6772 6573 7369 6f6e  inear regression
+00009190: 2c20 636f 6e73 6964 6572 696e 6720 7468  , considering th
+000091a0: 6174 206e 6f74 2061 6c6c 2068 6974 7320  at not all hits 
+000091b0: 7065 7220 6269 6e20 7368 6f77 2061 7070  per bin show app
+000091c0: 6172 656e 7420 6c69 6e65 6172 2072 656c  arent linear rel
+000091d0: 6174 696f 6e73 6869 702c 2069 7320 7365  ationship, is se
+000091e0: 7420 6173 2064 6566 6175 6c74 2035 2c20  t as default 5, 
+000091f0: 696e 6469 6361 7469 6e67 2074 6865 2075  indicating the u
+00009200: 7361 6765 206f 6620 746f 7020 3525 2068  sage of top 5% h
+00009210: 6974 7320 7065 7220 6269 6e2e 2054 6865  its per bin. The
+00009220: 206e 756d 6265 7220 6f66 2062 696e 7320   number of bins 
+00009230: 6469 7669 6465 6420 696e 2067 656e 6520  divided in gene 
+00009240: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
+00009250: 7469 6f6e 2c20 6769 7665 6e20 6279 2060  tion, given by `
+00009260: 2d62 7360 206f 7220 602d 2d62 696e 7360  -bs` or `--bins`
+00009270: 2c20 6465 7465 726d 696e 6573 2074 6865  , determines the
+00009280: 206e 756d 6265 7220 6f66 2062 696e 7320   number of bins 
+00009290: 746f 2063 6174 6567 6f72 697a 6520 7468  to categorize th
+000092a0: 6520 6765 6e65 206c 656e 6774 682c 2069  e gene length, i
+000092b0: 7320 7365 7420 6173 2064 6566 6175 6c74  s set as default
+000092c0: 2031 3030 2e20 5468 6520 7061 7261 6d65   100. The parame
+000092d0: 7465 7220 602d 6e6e 6020 6f72 2060 2d2d  ter `-nn` or `--
+000092e0: 6e6f 6e6f 726d 616c 697a 6174 696f 6e60  nonormalization`
+000092f0: 2063 616e 2062 6520 7365 7420 746f 2063   can be set to c
+00009300: 616c 6c20 6f66 6620 7468 6520 6e6f 726d  all off the norm
+00009310: 616c 697a 6174 696f 6e20 7072 6f63 6573  alization proces
+00009320: 732c 2061 6c74 686f 7567 6820 6974 2773  s, although it's
+00009330: 2073 7567 6765 7374 6564 2074 6f20 636f   suggested to co
+00009340: 6e64 7563 7420 7468 6520 6e6f 726d 616c  nduct the normal
+00009350: 697a 6174 696f 6e20 746f 2061 6371 7569  ization to acqui
+00009360: 7265 206d 6f72 6520 6163 6375 7261 7465  re more accurate
+00009370: 2067 656e 6520 6661 6d69 6c79 2063 6c75   gene family clu
+00009380: 7374 6572 696e 6720 7265 7375 6c74 2e20  stering result. 
+00009390: 5468 6520 7061 7261 6d65 7465 7273 2060  The parameters `
+000093a0: 2d2d 746f 5f73 746f 7060 2061 6e64 2060  --to_stop` and `
+000093b0: 2d2d 6364 7360 2063 6f6e 7472 6f6c 2074  --cds` control t
+000093c0: 6865 2062 6568 6176 696f 7572 206f 6620  he behaviour of 
+000093d0: 7472 616e 736c 6174 696e 6720 636f 6469  translating codi
+000093e0: 6e67 2073 6571 7565 6e63 6520 696e 746f  ng sequence into
+000093f0: 2061 6d69 6e6f 2061 6369 6420 7365 7175   amino acid sequ
+00009400: 656e 6365 2e20 4966 2074 6865 2060 2d2d  ence. If the `--
+00009410: 746f 5f73 746f 7060 2077 6173 2073 6574  to_stop` was set
+00009420: 2c20 7468 6520 7472 616e 736c 6174 696f  , the translatio
+00009430: 6e20 776f 756c 6420 6265 2074 6572 6d69  n would be termi
+00009440: 6e61 7465 6420 6174 2074 6865 2066 6972  nated at the fir
+00009450: 7374 2069 6e2d 6672 616d 6520 7374 6f70  st in-frame stop
+00009460: 2063 6f64 6f6e 2c20 6f74 6865 7277 6973   codon, otherwis
+00009470: 6520 7468 6520 7472 616e 736c 6174 696f  e the translatio
+00009480: 6e20 776f 756c 6420 7369 6d70 6c79 2073  n would simply s
+00009490: 6b69 7020 616e 7920 7374 6f70 2063 6f64  kip any stop cod
+000094a0: 6f6e 732e 2049 6620 7468 6520 602d 2d63  ons. If the `--c
+000094b0: 6473 6020 7761 7320 7365 742c 2073 6571  ds` was set, seq
+000094c0: 7565 6e63 6573 2074 6861 7420 646f 6573  uences that does
+000094d0: 6e27 7420 7374 6172 7420 7769 7468 2061  n't start with a
+000094e0: 2076 616c 6964 2073 7461 7274 2063 6f64   valid start cod
+000094f0: 6f6e 2c20 6f72 2063 6f6e 7461 696e 7320  on, or contains 
+00009500: 6d6f 7265 2074 6861 6e20 6f6e 6520 696e  more than one in
+00009510: 2d66 7261 6d65 2073 746f 7020 636f 646f  -frame stop codo
+00009520: 6e2c 206f 7220 6973 206e 6f74 2064 6976  n, or is not div
+00009530: 6964 6162 6c65 2062 7920 332c 2077 6f75  idable by 3, wou
+00009540: 6c64 2062 6520 7369 6d70 6c79 2064 726f  ld be simply dro
+00009550: 7070 6564 2c20 7375 6368 2074 6861 7420  pped, such that 
+00009560: 6f6e 6c79 2073 7472 6963 7420 636f 6d70  only strict comp
+00009570: 6c65 7465 2063 6f64 696e 6720 7365 7175  lete coding sequ
+00009580: 656e 6365 7320 776f 756c 6420 6265 2069  ences would be i
+00009590: 6e63 6c75 6465 6420 696e 2074 6865 2073  ncluded in the s
+000095a0: 7562 7365 7175 656e 7420 616e 616c 7973  ubsequent analys
+000095b0: 6973 2e20 5468 6520 6e75 6d62 6572 206f  is. The number o
+000095c0: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
+000095d0: 6473 2062 7920 7468 6520 6f70 7469 6f6e  ds by the option
+000095e0: 2060 2d6e 6020 6f72 2060 2d2d 6e74 6872   `-n` or `--nthr
+000095f0: 6561 6473 6020 6361 6e20 6265 2073 6574  eads` can be set
+00009600: 2074 6f20 6163 6365 6c65 7261 7465 2074   to accelerate t
+00009610: 6865 2063 616c 6375 6c61 7469 6f6e 2077  he calculation w
+00009620: 6974 6869 6e20 6064 6961 6d6f 6e64 602e  ithin `diamond`.
+00009630: 2054 6865 2064 6972 6563 746f 7279 206f   The directory o
+00009640: 6620 6f75 7470 7574 206f 7220 696e 7465  f output or inte
+00009650: 726d 6564 6961 7465 2066 696c 6573 2069  rmediate files i
+00009660: 7320 6465 7465 726d 696e 6564 2062 7920  s determined by 
+00009670: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+00009680: 6f60 206f 7220 602d 2d6f 7574 6469 7260  o` or `--outdir`
+00009690: 2c20 616e 6420 602d 7460 206f 7220 602d  , and `-t` or `-
+000096a0: 2d74 6d70 6469 7260 2c20 7768 6963 6820  -tmpdir`, which 
+000096b0: 7769 6c6c 2062 6520 6372 6561 7465 6420  will be created 
+000096c0: 6279 2074 6865 2070 726f 6772 616d 2069  by the program i
+000096d0: 7473 656c 6620 616e 6420 6265 206f 7665  tself and be ove
+000096e0: 7277 7269 7474 656e 2069 6620 7468 6520  rwritten if the 
+000096f0: 666f 6c64 6572 2068 6173 2061 6c72 6561  folder has alrea
+00009700: 6479 2062 6565 6e20 6372 6561 7465 642e  dy been created.
+00009710: 204e 6f74 6520 7468 6174 2074 6865 2073   Note that the s
+00009720: 6f66 7477 6172 6520 6064 6961 6d6f 6e64  oftware `diamond
+00009730: 6020 7368 6f75 6c64 2062 6520 7072 652d  ` should be pre-
+00009740: 696e 7374 616c 6c65 6420 616e 6420 7365  installed and se
+00009750: 7420 696e 2074 6865 2065 6e76 6972 6f6e  t in the environ
+00009760: 6d65 6e74 2070 6174 6820 696e 2061 6c6c  ment path in all
+00009770: 2074 6865 2061 6e61 6c79 7369 7320 7065   the analysis pe
+00009780: 7266 6f72 6d65 6420 6279 2060 7767 6420  rformed by `wgd 
+00009790: 646d 6460 2065 7863 6570 7420 666f 7220  dmd` except for 
+000097a0: 7468 6520 636f 6c6c 696e 6561 7220 636f  the collinear co
+000097b0: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+000097c0: 6973 2e0a 0a2a 2a57 6520 7375 6767 6573  is...**We sugges
+000097d0: 7420 7468 6174 2074 6865 2064 6566 6175  t that the defau
+000097e0: 6c74 2073 6574 7469 6e67 2069 6e20 7768  lt setting in wh
+000097f0: 6963 6820 7468 6520 696e 666c 6174 696f  ich the inflatio
+00009800: 6e20 6661 6374 6f72 2069 7320 7365 7420  n factor is set 
+00009810: 6173 2032 2c20 652d 7661 6c75 6520 6375  as 2, e-value cu
+00009820: 742d 6f66 6620 6173 2031 652d 3130 2061  t-off as 1e-10 a
+00009830: 6e64 206f 7468 6572 2070 6172 616d 6574  nd other paramet
+00009840: 6572 7320 696e 2064 6566 6175 6c74 2069  ers in default i
+00009850: 7320 6120 676f 6f64 2073 7461 7274 696e  s a good startin
+00009860: 6720 706f 696e 742c 2075 6e6c 6573 7320  g point, unless 
+00009870: 796f 7520 7370 6563 6966 6963 616c 6c79  you specifically
+00009880: 2077 616e 7420 746f 2065 7870 6c6f 7265   want to explore
+00009890: 2074 6865 2065 6666 6563 7473 206f 6620   the effects of 
+000098a0: 6469 6666 6572 656e 7420 7061 7261 6d65  different parame
+000098b0: 7465 7273 2e2a 2a20 5375 6368 2074 6861  ters.** Such tha
+000098c0: 7420 7468 6520 636f 6d6d 616e 6420 666f  t the command fo
+000098d0: 7220 7468 6520 6465 6c69 6e65 6174 696f  r the delineatio
+000098e0: 6e20 6f66 2077 686f 6c65 2070 6172 616e  n of whole paran
+000098f0: 6f6d 6520 6973 2073 696d 706c 7920 6173  ome is simply as
+00009900: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+00009910: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00009920: 6f65 7275 6c65 610a 6060 600a 0a2a 2a54  oerulea.```..**T
+00009930: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
+00009940: 6620 5242 4873 2a2a 0a60 6060 0a77 6764  f RBHs**.```.wgd
+00009950: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+00009960: 6571 7565 6e63 6532 202d 6520 3165 2d31  equence2 -e 1e-1
+00009970: 3020 2d62 7320 3130 3020 2d6e 7020 3520  0 -bs 100 -np 5 
+00009980: 282d 6e6e 2920 282d 6e20 3429 2028 2d63  (-nn) (-n 4) (-c
+00009990: 2030 2e39 2920 282d 2d6f 6766 6f72 6d61   0.9) (--ogforma
+000099a0: 7429 2028 2d2d 746f 5f73 746f 7029 2028  t) (--to_stop) (
+000099b0: 2d2d 6364 7329 2028 2d6f 2077 6764 5f64  --cds) (-o wgd_d
+000099c0: 6d64 2920 282d 7420 776f 726b 696e 675f  md) (-t working_
+000099d0: 746d 7029 0a60 6060 0a0a 546f 2064 656c  tmp).```..To del
+000099e0: 696e 6561 7465 2052 4248 7320 6265 7477  ineate RBHs betw
+000099f0: 6565 6e20 7477 6f20 6364 7320 7365 7175  een two cds sequ
+00009a00: 656e 6365 2066 696c 6573 2c20 7468 6520  ence files, the 
+00009a10: 7265 6c65 7661 6e74 2070 6172 616d 6574  relevant paramet
+00009a20: 6572 2069 7320 6d6f 7374 6c79 2074 6865  er is mostly the
+00009a30: 2073 616d 6520 6173 2077 686f 6c65 2070   same as whole p
+00009a40: 6172 616e 6f6d 6520 696e 6665 7265 6e63  aranome inferenc
+00009a50: 652c 2065 7863 6570 7420 666f 7220 7468  e, except for th
+00009a60: 6520 7061 7261 6d65 7465 7220 602d 6360  e parameter `-c`
+00009a70: 206f 7220 602d 2d63 7363 6f72 6560 2c20   or `--cscore`, 
+00009a80: 7768 6963 6820 7261 6e67 6573 2062 6574  which ranges bet
+00009a90: 7765 656e 2030 2061 6e64 2031 2061 6e64  ween 0 and 1 and
+00009aa0: 2069 7320 7573 6564 2074 6f20 7265 6c61   is used to rela
+00009ab0: 7820 7468 6520 7369 6d69 6c61 7269 7479  x the similarity
+00009ac0: 2063 7574 6f66 6620 6672 6f6d 2074 6865   cutoff from the
+00009ad0: 2065 7863 6c75 7369 7665 2072 6563 6970   exclusive recip
+00009ae0: 726f 6361 6c20 6265 7374 2068 6974 7320  rocal best hits 
+00009af0: 746f 2061 2063 6572 7461 696e 2072 6174  to a certain rat
+00009b00: 696f 2061 7320 746f 2074 6865 2062 6573  io as to the bes
+00009b10: 7420 6869 7473 2e20 466f 7220 696e 7374  t hits. For inst
+00009b20: 616e 6365 2c20 6966 2074 6865 2067 656e  ance, if the gen
+00009b30: 6520 6231 2066 726f 6d20 6765 6e6f 6d65  e b1 from genome
+00009b40: 2042 2068 6173 2074 6865 2062 6573 7420   B has the best 
+00009b50: 6869 7420 6765 6e65 2061 3120 6672 6f6d  hit gene a1 from
+00009b60: 2067 656e 6f6d 6520 4120 7769 7468 2074   genome A with t
+00009b70: 6865 2062 6974 2073 636f 7265 2061 7320  he bit score as 
+00009b80: 3130 302c 2077 6869 6368 2069 7320 6120  100, which is a 
+00009b90: 7363 6f72 696e 6720 6d61 7472 6978 2069  scoring matrix i
+00009ba0: 6e64 6570 656e 6465 6e74 206d 6561 7375  ndependent measu
+00009bb0: 7265 206f 6620 7468 6520 286c 6f63 616c  re of the (local
+00009bc0: 2920 7369 6d69 6c61 7269 7479 206f 6620  ) similarity of 
+00009bd0: 7468 6520 7477 6f20 616c 6967 6e65 6420  the two aligned 
+00009be0: 7365 7175 656e 6365 732c 2077 6974 6820  sequences, with 
+00009bf0: 6c61 7267 6572 2076 616c 7565 7320 696e  larger values in
+00009c00: 6469 6361 7469 6e67 2068 6967 6865 7220  dicating higher 
+00009c10: 7369 6d69 6c61 7269 7469 6573 2c20 6769  similarities, gi
+00009c20: 7665 6e20 7468 6520 602d 6320 302e 3960  ven the `-c 0.9`
+00009c30: 2c20 6765 6e65 7320 6672 6f6d 2067 656e  , genes from gen
+00009c40: 6f6d 6520 4120 7768 6963 6820 6861 7320  ome A which has 
+00009c50: 7468 6520 6269 7420 7363 6f72 6520 7769  the bit score wi
+00009c60: 7468 2067 656e 6520 6231 2068 6967 6865  th gene b1 highe
+00009c70: 7220 7468 616e 2030 2e39 7831 3030 2077  r than 0.9x100 w
+00009c80: 696c 6c20 616c 736f 2062 6520 7772 6974  ill also be writ
+00009c90: 7465 6e20 696e 2074 6865 2072 6573 756c  ten in the resul
+00009ca0: 7420 6669 6c65 2c20 7768 6963 6820 696e  t file, which in
+00009cb0: 2061 2073 656e 7365 2061 7265 206e 6f74   a sense are not
+00009cc0: 2052 4248 7320 616e 796d 6f72 6520 6f66   RBHs anymore of
+00009cd0: 2063 6f75 7273 652c 2062 7574 2074 6865   course, but the
+00009ce0: 2068 6967 686c 7920 7369 6d69 6c61 7220   highly similar 
+00009cf0: 686f 6d6f 6c6f 6775 6520 7061 6972 732e  homologue pairs.
+00009d00: 2049 6620 6d6f 7265 2074 6861 6e20 3220   If more than 2 
+00009d10: 7365 7175 656e 6365 2066 696c 6573 2077  sequence files w
+00009d20: 6572 6520 7072 6f76 6964 6564 2c20 6576  ere provided, ev
+00009d30: 6572 7920 7061 6972 2d77 6973 6520 5242  ery pair-wise RB
+00009d40: 4873 2077 6f75 6c64 2062 6520 6361 6c63  Hs would be calc
+00009d50: 756c 6174 6564 2065 7863 6570 7420 666f  ulated except fo
+00009d60: 7220 7175 6572 7969 6e67 2074 6865 2073  r querying the s
+00009d70: 616d 6520 7365 7175 656e 6365 2069 7473  ame sequence its
+00009d80: 656c 662e 2054 6865 206e 756d 6265 7220  elf. The number 
+00009d90: 6f66 2070 6172 616c 6c65 6c20 7468 7265  of parallel thre
+00009da0: 6164 7320 746f 2062 6f6f 7374 6572 2074  ads to booster t
+00009db0: 6865 2072 756e 6e69 6e67 2073 7065 6564  he running speed
+00009dc0: 2063 616e 2062 6520 7365 7420 6279 2074   can be set by t
+00009dd0: 6865 206f 7074 696f 6e20 602d 6e60 206f  he option `-n` o
+00009de0: 7220 602d 2d6e 7468 7265 6164 7360 2077  r `--nthreads` w
+00009df0: 6869 6368 2069 7320 7375 6767 6573 7465  hich is suggeste
+00009e00: 6420 746f 2062 6520 7365 7420 6173 2028  d to be set as (
+00009e10: 4e2d 3129 4e2f 3220 7768 6572 6520 4e20  N-1)N/2 where N 
+00009e20: 6973 2074 6865 206e 756d 6265 7220 6f66  is the number of
+00009e30: 2063 6473 2066 696c 6573 2074 6f20 6163   cds files to ac
+00009e40: 6869 6576 6520 7468 6520 6869 6768 6573  hieve the highes
+00009e50: 7420 6566 6669 6369 656e 6379 2e20 5468  t efficiency. Th
+00009e60: 6520 6f70 7469 6f6e 2060 2d2d 6f67 666f  e option `--ogfo
+00009e70: 726d 6174 6020 6361 6e20 6265 2073 6574  rmat` can be set
+00009e80: 2074 6f20 6164 6420 696e 6465 7820 2866   to add index (f
+00009e90: 6f72 2069 6e73 7461 6e63 6520 4746 3030  or instance GF00
+00009ea0: 3030 3030 3031 2920 746f 2074 6865 206f  000001) to the o
+00009eb0: 7574 7075 7420 5242 4820 6765 6e65 2066  utput RBH gene f
+00009ec0: 616d 696c 6965 7320 7768 6963 6820 6361  amilies which ca
+00009ed0: 6e20 6265 2066 7572 7468 6572 2075 7365  n be further use
+00009ee0: 6420 696e 2074 6865 202a 4b2a 3c73 7562  d in the *K*<sub
+00009ef0: 3e53 3c2f 7375 623e 2063 616c 6375 6c61  >S</sub> calcula
+00009f00: 7469 6f6e 2062 7920 6077 6764 206b 7364  tion by `wgd ksd
+00009f10: 602e 0a0a 2a2a 5468 6520 7375 6767 6573  `...**The sugges
+00009f20: 7465 6420 636f 6d6d 616e 6420 746f 2073  ted command to s
+00009f30: 7461 7274 2077 6974 6820 6973 2061 6c73  tart with is als
+00009f40: 6f20 756e 6465 7220 7468 6520 6465 6661  o under the defa
+00009f50: 756c 7420 7365 7474 696e 6720 7769 7468  ult setting with
+00009f60: 2074 6865 2063 6f6d 6d61 6e64 2073 686f   the command sho
+00009f70: 776e 2062 656c 6f77 2a2a 0a0a 6060 600a  wn below**..```.
+00009f80: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+00009f90: 3120 7365 7175 656e 6365 320a 6060 600a  1 sequence2.```.
+00009fa0: 0a2a 2a54 6865 2064 656c 696e 6561 7469  .**The delineati
+00009fb0: 6f6e 206f 6620 6c6f 6361 6c20 4d52 4248  on of local MRBH
+00009fc0: 732a 2a0a 6060 600a 7767 6420 646d 6420  s**.```.wgd dmd 
+00009fd0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+00009fe0: 6365 3220 7365 7175 656e 6365 3320 2d66  ce2 sequence3 -f
+00009ff0: 2073 6571 7565 6e63 6531 202d 6520 3165   sequence1 -e 1e
+0000a000: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
+0000a010: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
+0000a020: 2d6b 6629 2028 2d6b 6429 2028 2d63 2030  -kf) (-kd) (-c 0
+0000a030: 2e39 2920 282d 2d74 6f5f 7374 6f70 2920  .9) (--to_stop) 
+0000a040: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
+0000a050: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
+0000a060: 5f74 6d70 290a 6060 600a 0a2a 2a54 6865  _tmp).```..**The
+0000a070: 2064 6973 7469 6e63 7469 6f6e 2062 6574   distinction bet
+0000a080: 7765 656e 206c 6f63 616c 2061 6e64 2067  ween local and g
+0000a090: 6c6f 6261 6c20 2868 6572 6575 6e64 6572  lobal (hereunder
+0000a0a0: 2920 4d52 4248 7320 6973 2074 6861 7420  ) MRBHs is that 
+0000a0b0: 6c6f 6361 6c20 4d52 4248 7320 6172 6520  local MRBHs are 
+0000a0c0: 7468 6520 7265 7375 6c74 7320 6f66 206d  the results of m
+0000a0d0: 6572 6765 6420 5242 4873 206f 6e20 6120  erged RBHs on a 
+0000a0e0: 6a6f 696e 7420 666f 6361 6c20 7370 6563  joint focal spec
+0000a0f0: 6965 732c 2066 6f72 2069 6e73 7461 6e63  ies, for instanc
+0000a100: 6520 696e 2061 2074 6872 6565 2073 7065  e in a three spe
+0000a110: 6369 6573 2073 7973 7465 6d20 2841 2c28  cies system (A,(
+0000a120: 422c 4329 292c 2074 6865 206c 6f63 616c  B,C)), the local
+0000a130: 204d 5242 4873 206f 6620 4120 6f6e 6c79   MRBHs of A only
+0000a140: 2072 6571 7569 7265 2074 6865 2063 616c   require the cal
+0000a150: 6375 6c61 7469 6f6e 206f 6620 5242 4873  culation of RBHs
+0000a160: 2062 6574 7765 656e 2041 2061 6e64 2043   between A and C
+0000a170: 2028 6465 6e6f 7465 6420 6173 2041 4329   (denoted as AC)
+0000a180: 2061 6e64 2041 4220 616e 6420 7468 656e   and AB and then
+0000a190: 2074 6865 206d 6572 6769 6e67 206f 6620   the merging of 
+0000a1a0: 4142 2061 6e64 2041 4320 6174 2074 6865  AB and AC at the
+0000a1b0: 2061 7869 7320 6f66 2041 2c20 7768 696c   axis of A, whil
+0000a1c0: 6520 7468 6520 676c 6f61 626c 204d 5242  e the gloabl MRB
+0000a1d0: 4873 2061 7265 2069 6e64 6570 656e 6465  Hs are independe
+0000a1e0: 6e74 206f 6620 666f 6361 6c20 7370 6563  nt of focal spec
+0000a1f0: 6965 7320 696e 2074 6861 7420 6974 206a  ies in that it j
+0000a200: 7573 7420 6361 6c6c 7320 7468 6520 6361  ust calls the ca
+0000a210: 6c63 756c 6174 696f 6e73 206f 6620 616c  lculations of al
+0000a220: 6c20 706f 7373 6962 6c65 2073 7065 6369  l possible speci
+0000a230: 6573 2070 6169 7220 286e 6f74 2073 656c  es pair (not sel
+0000a240: 6620 746f 2073 656c 6620 7370 6563 6965  f to self specie
+0000a250: 7320 7061 6972 292c 2073 7563 6820 7468  s pair), such th
+0000a260: 6174 2041 422c 2041 432c 2061 6e64 2042  at AB, AC, and B
+0000a270: 4320 6172 6520 616c 6c20 746f 2062 6520  C are all to be 
+0000a280: 6361 6c63 756c 6174 6564 2061 6e64 206d  calculated and m
+0000a290: 6572 6765 642e 2a2a 0a0a 5477 6f20 7479  erged.**..Two ty
+0000a2a0: 7065 7320 6f66 204d 5242 4873 2061 7320  pes of MRBHs as 
+0000a2b0: 696e 7465 7072 6574 6174 6564 2061 626f  intepretated abo
+0000a2c0: 7665 2063 616e 2062 6520 6465 6c69 6e65  ve can be deline
+0000a2d0: 6174 6564 2062 7920 6077 6764 2064 6d64  ated by `wgd dmd
+0000a2e0: 602c 2074 6865 206c 6f63 616c 204d 5242  `, the local MRB
+0000a2f0: 4873 2061 6e64 2074 6865 2067 6c6f 6261  Hs and the globa
+0000a300: 6c20 4d52 4248 732e 2054 6865 206c 6f63  l MRBHs. The loc
+0000a310: 616c 204d 5242 4873 2061 7265 2063 6f6e  al MRBHs are con
+0000a320: 7374 7275 6374 6564 2062 7920 6d65 7267  structed by merg
+0000a330: 696e 6720 616c 6c20 7468 6520 7265 6c65  ing all the rele
+0000a340: 7661 6e74 2052 4248 7320 6f6e 6c79 2077  vant RBHs only w
+0000a350: 6974 6820 7468 6520 666f 6361 6c20 7370  ith the focal sp
+0000a360: 6563 6965 732c 2077 6869 6368 2069 7320  ecies, which is 
+0000a370: 7365 7420 6279 2060 2d66 6020 6f72 2060  set by `-f` or `
+0000a380: 2d2d 666f 6375 7360 2e20 5468 6520 7061  --focus`. The pa
+0000a390: 7261 6d65 7465 7220 602d 6b66 6020 6f72  rameter `-kf` or
+0000a3a0: 2060 2d2d 6b65 6570 6661 7374 6160 2063   `--keepfasta` c
+0000a3b0: 616e 2062 6520 7365 7420 746f 2072 6574  an be set to ret
+0000a3c0: 6169 6e20 7468 6520 7365 7175 656e 6365  ain the sequence
+0000a3d0: 2069 6e66 6f72 6d61 7469 6f6e 206f 6620   information of 
+0000a3e0: 6561 6368 204d 5242 482e 2054 6865 2070  each MRBH. The p
+0000a3f0: 6172 616d 6574 6572 2060 2d6b 6460 206f  arameter `-kd` o
+0000a400: 7220 602d 2d6b 6565 7064 7570 6c69 6361  r `--keepduplica
+0000a410: 7465 7360 2064 6574 6572 6d69 6e65 7320  tes` determines 
+0000a420: 7768 6574 6865 7220 7468 6520 7361 6d65  whether the same
+0000a430: 2067 656e 6573 2063 616e 2061 7070 6561   genes can appea
+0000a440: 7220 696e 2064 6966 6665 7265 6e74 206c  r in different l
+0000a450: 6f63 616c 204d 5242 4873 2e20 4e6f 726d  ocal MRBHs. Norm
+0000a460: 616c 6c79 2074 6865 7265 2077 696c 6c20  ally there will 
+0000a470: 6265 206e 6f20 6475 706c 6963 6174 6573  be no duplicates
+0000a480: 2069 6e20 7468 6520 6c6f 6361 6c20 4d52   in the local MR
+0000a490: 4248 7320 6275 7420 6966 2075 7365 7273  BHs but if users
+0000a4a0: 2073 6574 2074 6865 2060 2d63 6020 6173   set the `-c` as
+0000a4b0: 2030 2e39 2028 6f72 2061 6e79 2076 616c   0.9 (or any val
+0000a4c0: 7565 2073 6d61 6c6c 6572 2074 6861 6e20  ue smaller than 
+0000a4d0: 3129 2c20 6974 2773 206c 696b 656c 7920  1), it's likely 
+0000a4e0: 7468 6174 2074 6865 2073 616d 6520 6765  that the same ge
+0000a4f0: 6e65 2077 696c 6c20 6861 7665 2063 6861  ne will have cha
+0000a500: 6e63 6520 746f 2061 7070 6561 7220 6d75  nce to appear mu
+0000a510: 6c74 6970 6c65 2074 696d 6573 2069 6e20  ltiple times in 
+0000a520: 6469 6666 6572 656e 7420 6c6f 6361 6c20  different local 
+0000a530: 4d52 4248 732e 2054 6861 7420 6973 2074  MRBHs. That is t
+0000a540: 6f20 7361 792c 2074 6865 2070 6172 616d  o say, the param
+0000a550: 6574 6572 2060 2d6b 6460 2069 7320 6d65  eter `-kd` is me
+0000a560: 616e 696e 6766 756c 206f 6e6c 7920 7768  aningful only wh
+0000a570: 656e 2069 7427 7320 7365 7420 746f 6765  en it's set toge
+0000a580: 7468 6572 2077 6974 6820 7468 6520 7061  ther with the pa
+0000a590: 7261 6d65 7465 7220 602d 6360 2e20 5468  rameter `-c`. Th
+0000a5a0: 6520 6e75 6d62 6572 206f 6620 7061 7261  e number of para
+0000a5b0: 6c6c 656c 2074 6872 6561 6473 2069 7320  llel threads is 
+0000a5c0: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+0000a5d0: 7365 7420 6173 2074 6865 206e 756d 6265  set as the numbe
+0000a5e0: 7220 6f66 2063 6473 2066 696c 6573 206d  r of cds files m
+0000a5f0: 696e 7573 2031 2e0a 0a2a 2a41 2073 7567  inus 1...**A sug
+0000a600: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+0000a610: 7275 6e20 6973 2075 6e64 6572 2074 6865  run is under the
+0000a620: 2064 6566 6175 6c74 2070 6172 616d 6574   default paramet
+0000a630: 6572 2077 6974 6820 7468 6520 636f 6d6d  er with the comm
+0000a640: 616e 6420 7368 6f77 6e20 6173 2062 656c  and shown as bel
+0000a650: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2064  ow.**..```.wgd d
+0000a660: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000a670: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000a680: 202d 6620 7365 7175 656e 6365 310a 6060   -f sequence1.``
+0000a690: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
+0000a6a0: 7469 6f6e 206f 6620 676c 6f62 616c 204d  tion of global M
+0000a6b0: 5242 4873 2a2a 0a60 6060 0a77 6764 2064  RBHs**.```.wgd d
+0000a6c0: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000a6d0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000a6e0: 202d 676d 202d 6520 3165 2d31 3020 2d62   -gm -e 1e-10 -b
+0000a6f0: 7320 3130 3020 2d6e 7020 3520 282d 6e6e  s 100 -np 5 (-nn
+0000a700: 2920 282d 6e20 3429 2028 2d6b 6629 2028  ) (-n 4) (-kf) (
+0000a710: 2d6b 6429 2028 2d63 2030 2e39 2920 282d  -kd) (-c 0.9) (-
+0000a720: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
+0000a730: 2920 282d 6f20 7767 645f 646d 6429 2028  ) (-o wgd_dmd) (
+0000a740: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
+0000a750: 6060 600a 0a54 6865 2067 6c6f 6261 6c20  ```..The global 
+0000a760: 4d52 4248 7320 6973 2063 6f6e 7374 7275  MRBHs is constru
+0000a770: 6374 6564 2062 7920 6578 6861 7573 7469  cted by exhausti
+0000a780: 7665 6c79 206d 6572 6769 6e67 2061 6c6c  vely merging all
+0000a790: 2074 6865 2070 6f73 7369 626c 6520 7061   the possible pa
+0000a7a0: 6972 2d77 6973 6520 5242 4873 2065 7863  ir-wise RBHs exc
+0000a7b0: 6570 7420 666f 7220 7175 6572 7969 6e67  ept for querying
+0000a7c0: 2074 6865 2073 6571 7565 6e63 6520 6974   the sequence it
+0000a7d0: 7365 6c66 2c20 7768 6963 6820 6361 6e20  self, which can 
+0000a7e0: 6265 2069 6e69 7469 6174 6564 2062 7920  be initiated by 
+0000a7f0: 6164 6420 7468 6520 666c 6167 2060 2d67  add the flag `-g
+0000a800: 6d60 206f 7220 602d 2d67 6c6f 6261 6c6d  m` or `--globalm
+0000a810: 7262 6860 2e20 5468 6520 7265 7374 206f  rbh`. The rest o
+0000a820: 6620 7265 6c65 7661 6e74 2070 6172 616d  f relevant param
+0000a830: 6574 6572 7320 7374 6179 7320 7468 6520  eters stays the 
+0000a840: 7361 6d65 2061 7320 7468 6520 6c6f 6361  same as the loca
+0000a850: 6c20 4d52 4248 732e 2054 6865 206e 756d  l MRBHs. The num
+0000a860: 6265 7220 6f66 2070 6172 616c 6c65 6c20  ber of parallel 
+0000a870: 7468 7265 6164 7320 6973 2073 7567 6765  threads is sugge
+0000a880: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
+0000a890: 7320 284e 2d31 294e 2f32 2074 6f6f 2077  s (N-1)N/2 too w
+0000a8a0: 6865 7265 204e 2069 7320 7468 6520 6e75  here N is the nu
+0000a8b0: 6d62 6572 206f 6620 6364 7320 6669 6c65  mber of cds file
+0000a8c0: 7320 746f 2061 6368 6965 7665 2074 6865  s to achieve the
+0000a8d0: 2068 6967 6865 7374 2065 6666 6963 6965   highest efficie
+0000a8e0: 6e63 792e 0a0a 2a2a 4120 7375 6767 6573  ncy...**A sugges
+0000a8f0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
+0000a900: 2069 7320 756e 6465 7220 7468 6520 6465   is under the de
+0000a910: 6661 756c 7420 7061 7261 6d65 7465 7220  fault parameter 
+0000a920: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
+0000a930: 2073 686f 776e 2061 7320 6265 6c6f 772e   shown as below.
+0000a940: 2a2a 0a0a 6060 600a 7767 6420 646d 6420  **..```.wgd dmd 
+0000a950: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000a960: 6365 3220 7365 7175 656e 6365 3320 2d67  ce2 sequence3 -g
+0000a970: 6d0a 6060 600a 0a2a 2a54 6865 2064 656c  m.```..**The del
+0000a980: 696e 6561 7469 6f6e 206f 6620 6f72 7468  ineation of orth
+0000a990: 6f67 726f 7570 732a 2a0a 6060 600a 7767  ogroups**.```.wg
+0000a9a0: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
+0000a9b0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000a9c0: 6365 3320 2d6f 6920 2d6f 6f20 2d65 2031  ce3 -oi -oo -e 1
+0000a9d0: 652d 3130 202d 6273 2031 3030 202d 6e70  e-10 -bs 100 -np
+0000a9e0: 2035 2028 2d6e 6e29 2028 2d63 6329 2028   5 (-nn) (-cc) (
+0000a9f0: 2d74 6529 2028 2d6d 6320 302e 3829 2028  -te) (-mc 0.8) (
+0000aa00: 2d67 6e29 2028 2d74 7265 6520 2766 6173  -gn) (-tree 'fas
+0000aa10: 7474 7265 6527 2920 282d 7473 2027 2d66  ttree') (-ts '-f
+0000aa20: 6173 7465 7374 2729 2028 2d6e 2034 2920  astest') (-n 4) 
+0000aa30: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+0000aa40: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
+0000aa50: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+0000aa60: 290a 6060 600a 0a49 6e20 6077 6764 2076  ).```..In `wgd v
+0000aa70: 3260 2c20 7765 2061 6c73 6f20 696d 706c  2`, we also impl
+0000aa80: 656d 656e 7465 6420 616e 2061 6c67 6f72  emented an algor
+0000aa90: 6974 686d 206f 6620 6465 6c69 6e65 6174  ithm of delineat
+0000aaa0: 696e 6720 6f72 7468 6f67 726f 7570 732c  ing orthogroups,
+0000aab0: 2077 6869 6368 2063 616e 2062 6520 696e   which can be in
+0000aac0: 6974 6961 7465 6420 7769 7468 2074 6865  itiated with the
+0000aad0: 2070 6172 616d 6574 6572 2060 2d6f 6960   parameter `-oi`
+0000aae0: 206f 7220 602d 2d6f 7274 686f 696e 6665   or `--orthoinfe
+0000aaf0: 7260 2e20 5477 6f20 7761 7973 206f 6620  r`. Two ways of 
+0000ab00: 6465 6c69 6e65 6174 696f 6e20 6361 6e20  delineation can 
+0000ab10: 6265 2063 686f 7365 6e2c 2074 6865 2063  be chosen, the c
+0000ab20: 6f6e 6361 7465 6e61 7469 6f6e 2077 6179  oncatenation way
+0000ab30: 2028 7365 7420 6279 2074 6865 2070 6172   (set by the par
+0000ab40: 616d 6574 6572 2060 2d63 6360 206f 7220  ameter `-cc` or 
+0000ab50: 602d 2d63 6f6e 6361 7460 2920 6f72 2074  `--concat`) or t
+0000ab60: 6865 206e 6f6e 2d63 6f6e 6361 7465 6e61  he non-concatena
+0000ab70: 7469 6f6e 2028 6465 6661 756c 7429 2077  tion (default) w
+0000ab80: 6179 2e20 496e 2062 7269 6566 2c20 7468  ay. In brief, th
+0000ab90: 6520 636f 6e63 6174 656e 6174 696f 6e20  e concatenation 
+0000aba0: 7761 7920 6f66 2064 656c 696e 6561 7469  way of delineati
+0000abb0: 6e67 206f 7274 686f 6772 6f75 7073 2073  ng orthogroups s
+0000abc0: 7461 7274 7320 7769 7468 2063 6f6e 6361  tarts with conca
+0000abd0: 7465 6e61 7469 6e67 2061 6c6c 2074 6865  tenating all the
+0000abe0: 2073 6571 7565 6e63 6573 2069 6e74 6f20   sequences into 
+0000abf0: 6120 7369 6e67 6c65 2073 6571 7565 6e63  a single sequenc
+0000ac00: 6520 6669 6c65 2061 6e64 2074 6865 6e20  e file and then 
+0000ac10: 696e 6665 7272 696e 6720 7468 6520 7768  inferring the wh
+0000ac20: 6f6c 6520 7061 7261 6e6f 6d65 206f 6620  ole paranome of 
+0000ac30: 7468 6973 2073 696e 676c 6520 7365 7175  this single sequ
+0000ac40: 656e 6365 2066 696c 6520 7769 7468 2074  ence file with t
+0000ac50: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
+0000ac60: 7375 6c74 7320 6d61 7070 6564 2062 6163  sults mapped bac
+0000ac70: 6b20 746f 2074 6865 2062 656c 6f6e 6769  k to the belongi
+0000ac80: 6e67 2073 7065 6369 6573 2e20 5768 696c  ng species. Whil
+0000ac90: 6520 7468 6520 6e6f 6e2d 636f 6e63 6174  e the non-concat
+0000aca0: 656e 6174 696f 6e20 7761 7920 7374 6172  enation way star
+0000acb0: 7473 2077 6974 6820 7265 7370 6563 7469  ts with respecti
+0000acc0: 7665 2070 6169 722d 7769 7365 2064 6961  ve pair-wise dia
+0000acd0: 6d6f 6e64 2073 6561 7263 6820 2869 6e63  mond search (inc
+0000ace0: 6c75 6469 6e67 2071 7565 7279 696e 6720  luding querying 
+0000acf0: 7468 6520 7361 6d65 2073 6571 7565 6e63  the same sequenc
+0000ad00: 6520 6974 7365 6c66 2920 616e 6420 7468  e itself) and th
+0000ad10: 656e 2061 6c6c 2074 6865 2073 6571 7565  en all the seque
+0000ad20: 6e63 6520 7369 6d69 6c61 7269 7479 2074  nce similarity t
+0000ad30: 6162 6c65 7320 7769 6c6c 2062 6520 636f  ables will be co
+0000ad40: 6e63 6174 656e 6174 6564 2061 6e64 2063  ncatenated and c
+0000ad50: 6c75 7374 6572 6564 2069 6e74 6f20 6f72  lustered into or
+0000ad60: 7468 6f67 726f 7570 732e 2053 6f6d 6520  thogroups. Some 
+0000ad70: 6f74 6865 7220 706f 7373 6962 6c79 2075  other possibly u
+0000ad80: 7365 6675 6c20 706f 7374 2d63 6c75 7374  seful post-clust
+0000ad90: 6572 696e 6720 6675 6e63 7469 6f6e 7320  ering functions 
+0000ada0: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
+0000adb0: 2c20 696e 636c 7564 696e 6720 7468 6520  , including the 
+0000adc0: 7061 7261 6d65 7465 7220 602d 7465 6020  parameter `-te` 
+0000add0: 6f72 2060 2d2d 7465 7374 736f 6760 2c20  or `--testsog`, 
+0000ade0: 7768 6963 6820 6361 6e20 6265 2073 6574  which can be set
+0000adf0: 2074 6f20 7374 6172 7420 7468 6520 756e   to start the un
+0000ae00: 6269 6173 6564 2074 6573 7420 6f66 2073  biased test of s
+0000ae10: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
+0000ae20: 6661 6d69 6c69 6573 2028 6e6f 7465 2074  families (note t
+0000ae30: 6861 7420 7468 6973 2066 756e 6374 696f  hat this functio
+0000ae40: 6e20 6e65 6564 7320 6068 6d6d 6572 6020  n needs `hmmer` 
+0000ae50: 2876 332e 3162 3229 2074 6f20 6265 2069  (v3.1b2) to be i
+0000ae60: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+0000ae70: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+0000ae80: 292c 2074 6865 2070 6172 616d 6574 6572  ), the parameter
+0000ae90: 2060 2d6d 6360 206f 7220 602d 2d6d 736f   `-mc` or `--mso
+0000aea0: 6763 7574 602c 2072 616e 6769 6e67 2062  gcut`, ranging b
+0000aeb0: 6574 7765 656e 2030 2074 6f20 312c 2077  etween 0 to 1, w
+0000aec0: 6869 6368 2063 616e 2062 6520 7365 7420  hich can be set 
+0000aed0: 746f 2073 6561 7263 6820 7468 6520 736f  to search the so
+0000aee0: 2d63 616c 6c65 6420 6d6f 7374 6c79 2073  -called mostly s
+0000aef0: 696e 676c 652d 636f 7079 2066 616d 696c  ingle-copy famil
+0000af00: 7920 7768 6963 6820 6861 7320 6869 6768  y which has high
+0000af10: 6572 2074 6861 6e20 6365 7274 6169 6e20  er than certain 
+0000af20: 6375 742d 6f66 6620 7065 7263 656e 7461  cut-off percenta
+0000af30: 6765 206f 6620 7370 6563 6965 7320 636f  ge of species co
+0000af40: 7665 7261 6765 2c20 7468 6520 7061 7261  verage, the para
+0000af50: 6d65 7465 7220 602d 676e 6020 6f72 2060  meter `-gn` or `
+0000af60: 2d2d 6765 746e 736f 6760 2c20 7768 6963  --getnsog`, whic
+0000af70: 6820 6361 6e20 6265 2073 6574 2074 6f20  h can be set to 
+0000af80: 7365 6172 6368 2066 6f72 206e 6573 7465  search for neste
+0000af90: 6420 7369 6e67 6c65 2d63 6f70 7920 6765  d single-copy ge
+0000afa0: 6e65 2066 616d 696c 6965 7320 284e 534f  ne families (NSO
+0000afb0: 4773 2920 7768 6963 6820 6973 206f 7269  Gs) which is ori
+0000afc0: 6769 6e61 6c6c 7920 6d75 6c74 6979 2d63  ginally multiy-c
+0000afd0: 6f70 7920 6275 7420 6861 7320 6120 286d  opy but has a (m
+0000afe0: 6f73 746c 7929 2073 696e 676c 652d 636f  ostly) single-co
+0000aff0: 7079 2062 7261 6e63 6820 2877 6869 6368  py branch (which
+0000b000: 2072 6571 7569 7265 7320 7468 6520 6368   requires the ch
+0000b010: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+0000b020: 6e63 6520 7072 6f67 7261 6d20 7365 7420  nce program set 
+0000b030: 6279 2060 2d74 7265 6560 206f 7220 602d  by `-tree` or `-
+0000b040: 2d74 7265 655f 6d65 7468 6f64 6020 746f  -tree_method` to
+0000b050: 2062 6520 7072 652d 696e 7374 616c 6c65   be pre-installe
+0000b060: 6420 696e 2074 6865 2065 6e76 6972 6f6e  d in the environ
+0000b070: 6d65 6e74 2070 6174 6820 7769 7468 2074  ment path with t
+0000b080: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
+0000b090: 7474 696e 6720 666f 7220 6765 6e65 2074  tting for gene t
+0000b0a0: 7265 6520 696e 6665 7265 6e63 6520 636f  ree inference co
+0000b0b0: 6e74 726f 6c6c 6564 2062 7920 602d 7473  ntrolled by `-ts
+0000b0c0: 6020 6f72 2060 2d2d 7472 6565 7365 7460  ` or `--treeset`
+0000b0d0: 292e 2054 6865 2070 726f 6772 616d 2060  ). The program `
+0000b0e0: 7767 6420 646d 6460 2077 6f75 6c64 2073  wgd dmd` would s
+0000b0f0: 7469 6c6c 2063 6f6e 6475 6374 2074 6865  till conduct the
+0000b100: 2052 4248 7320 6361 6c63 756c 6174 696f   RBHs calculatio
+0000b110: 6e20 756e 6c65 7373 2074 6865 2070 6172  n unless the par
+0000b120: 616d 6574 6572 2060 2d6f 6f60 206f 7220  ameter `-oo` or 
+0000b130: 602d 2d6f 6e6c 796f 7274 686f 6020 7761  `--onlyortho` wa
+0000b140: 7320 7365 742e 2049 6620 6f6e 6520 6f6e  s set. If one on
+0000b150: 6c79 2077 616e 7473 2074 6f20 696e 6665  ly wants to infe
+0000b160: 7220 7468 6520 6f72 7468 6f67 726f 7570  r the orthogroup
+0000b170: 732c 2069 7427 7320 7375 6767 6573 7465  s, it's suggeste
+0000b180: 6420 746f 2061 6464 2074 6865 2066 6c61  d to add the fla
+0000b190: 6720 602d 6f6f 6020 746f 206a 7573 7420  g `-oo` to just 
+0000b1a0: 696d 706c 656d 656e 7420 7468 6520 6f72  implement the or
+0000b1b0: 7468 6f67 726f 7570 7320 6465 6c69 6e65  thogroups deline
+0000b1c0: 6174 696f 6e20 616e 616c 7973 6973 2e20  ation analysis. 
+0000b1d0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
+0000b1e0: 7261 6c6c 656c 2074 6872 6561 6473 2069  rallel threads i
+0000b1f0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000b200: 6520 7365 7420 6173 2028 4e2b 3129 4e2f  e set as (N+1)N/
+0000b210: 3220 7768 6572 6520 4e20 6973 2074 6865  2 where N is the
+0000b220: 206e 756d 6265 7220 6f66 2063 6473 2066   number of cds f
+0000b230: 696c 6573 2074 6f20 6163 6869 6576 6520  iles to achieve 
+0000b240: 7468 6520 6869 6768 6573 7420 6566 6669  the highest effi
+0000b250: 6369 656e 6379 2073 696e 6365 2074 6865  ciency since the
+0000b260: 2073 656c 662d 636f 6d70 6172 6973 6f6e   self-comparison
+0000b270: 2069 7320 616c 736f 2069 6e63 6c75 6465   is also include
+0000b280: 642e 0a0a 2a2a 5468 6520 6465 6661 756c  d...**The defaul
+0000b290: 7420 7365 7474 696e 6720 6f66 2070 6172  t setting of par
+0000b2a0: 616d 6574 6572 7320 6973 2061 2072 6561  ameters is a rea
+0000b2b0: 736f 6e61 626c 6520 7374 6172 7469 6e67  sonable starting
+0000b2c0: 2070 6f69 6e74 2077 6974 6820 7468 6520   point with the 
+0000b2d0: 636f 6d6d 616e 6420 6173 2062 656c 6f77  command as below
+0000b2e0: 2e2a 2a0a 0a60 6060 0a77 6764 2064 6d64  .**..```.wgd dmd
+0000b2f0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
+0000b300: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
+0000b310: 6f69 202d 6f6f 0a60 6060 0a0a 2a2a 5468  oi -oo.```..**Th
+0000b320: 6520 636f 6c6c 696e 6561 7220 636f 616c  e collinear coal
+0000b330: 6573 6365 6e63 6520 696e 6665 7265 6e63  escence inferenc
+0000b340: 6520 6f66 2070 6879 6c6f 6765 6e79 2a2a  e of phylogeny**
+0000b350: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
+0000b360: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000b370: 2073 6571 7565 6e63 6533 202d 6170 2061   sequence3 -ap a
+0000b380: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
+0000b390: 202d 6c65 206c 6564 6174 6120 2d67 7420   -le ledata -gt 
+0000b3a0: 6774 6461 7461 202d 636f 6320 282d 7472  gtdata -coc (-tr
+0000b3b0: 6565 2027 6661 7374 7472 6565 2729 2028  ee 'fasttree') (
+0000b3c0: 2d74 7320 272d 6661 7374 6573 7427 2920  -ts '-fastest') 
+0000b3d0: 282d 6e20 3429 2028 2d2d 746f 5f73 746f  (-n 4) (--to_sto
+0000b3e0: 7029 2028 2d2d 6364 7329 2028 2d6f 2077  p) (--cds) (-o w
+0000b3f0: 6764 5f64 6d64 2920 282d 7420 776f 726b  gd_dmd) (-t work
+0000b400: 696e 675f 746d 7029 0a60 6060 0a0a 4120  ing_tmp).```..A 
+0000b410: 6e6f 7665 6c20 7068 796c 6f67 656e 6574  novel phylogenet
+0000b420: 6963 2069 6e66 6572 656e 6365 206d 6574  ic inference met
+0000b430: 686f 6420 6e61 6d65 6420 2263 6f6c 6c69  hod named "colli
+0000b440: 6e65 6172 2063 6f61 6c65 7363 656e 6365  near coalescence
+0000b450: 2069 6e66 6572 656e 6365 2220 6973 2061   inference" is a
+0000b460: 6c73 6f20 696d 706c 656d 656e 7465 6420  lso implemented 
+0000b470: 696e 2060 7767 6420 7632 602e 2046 6f72  in `wgd v2`. For
+0000b480: 2074 6869 7320 616e 616c 7973 6973 2c20   this analysis, 
+0000b490: 7573 6572 7320 6e65 6564 2074 6f20 7072  users need to pr
+0000b4a0: 6f76 6964 6520 7468 6520 616e 6368 6f72  ovide the anchor
+0000b4b0: 2070 6f69 6e74 7320 6669 6c65 2062 7920   points file by 
+0000b4c0: 602d 6170 6020 6f72 2060 2d2d 616e 6368  `-ap` or `--anch
+0000b4d0: 6f72 706f 696e 7473 602c 2074 6865 2063  orpoints`, the c
+0000b4e0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
+0000b4f0: 7320 6669 6c65 2062 7920 602d 736d 6020  s file by `-sm` 
+0000b500: 6f72 2060 2d2d 7365 676d 656e 7473 602c  or `--segments`,
+0000b510: 2074 6865 206c 6973 7473 6567 6d65 6e74   the listsegment
+0000b520: 7320 6669 6c65 2062 7920 602d 6c65 6020  s file by `-le` 
+0000b530: 6f72 2060 2d2d 6c69 7374 656c 656d 656e  or `--listelemen
+0000b540: 7473 602c 2061 6e64 2074 6865 2067 656e  ts`, and the gen
+0000b550: 6520 7461 626c 6520 6669 6c65 2062 7920  e table file by 
+0000b560: 602d 6774 6020 6f72 2060 2d2d 6765 6e65  `-gt` or `--gene
+0000b570: 7461 626c 6560 2c20 616c 6c20 6f66 2077  table`, all of w
+0000b580: 6869 6368 2063 616e 2062 6520 7072 6f64  hich can be prod
+0000b590: 7563 6564 2069 6e20 7468 6520 7072 6f67  uced in the prog
+0000b5a0: 7261 6d20 6077 6764 2073 796e 602e 2054  ram `wgd syn`. T
+0000b5b0: 6865 2070 6172 616d 6574 6572 2060 2d63  he parameter `-c
+0000b5c0: 6f63 6020 6f72 2060 2d2d 636f 6c6c 696e  oc` or `--collin
+0000b5d0: 6561 7263 6f61 6c65 7363 656e 6365 6020  earcoalescence` 
+0000b5e0: 6e65 6564 7320 746f 2062 6520 7365 7420  needs to be set 
+0000b5f0: 746f 2073 7461 7274 2074 6869 7320 616e  to start this an
+0000b600: 616c 7973 6973 2e20 5468 6520 7472 6565  alysis. The tree
+0000b610: 2d69 6e66 6572 656e 6365 2070 726f 6772  -inference progr
+0000b620: 616d 2061 6e64 2074 6865 2061 7373 6f63  am and the assoc
+0000b630: 6961 7465 6420 7061 7261 6d65 7465 7273  iated parameters
+0000b640: 2063 616e 2062 6520 7365 7420 6a75 7374   can be set just
+0000b650: 2061 7320 6162 6f76 6520 6279 2060 2d74   as above by `-t
+0000b660: 7265 6560 206f 7220 602d 2d74 7265 655f  ree` or `--tree_
+0000b670: 6d65 7468 6f64 6020 616e 6420 602d 7473  method` and `-ts
+0000b680: 6020 6f72 2060 2d2d 7472 6565 7365 7460  ` or `--treeset`
+0000b690: 2e20 506c 6561 7365 2061 6c73 6f20 6d61  . Please also ma
+0000b6a0: 6b65 2073 7572 6520 7468 6520 6368 6f73  ke sure the chos
+0000b6b0: 656e 2074 7265 652d 696e 6665 7265 6e63  en tree-inferenc
+0000b6c0: 6520 7072 6f67 7261 6d20 6973 2069 6e73  e program is ins
+0000b6d0: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+0000b6e0: 7669 726f 6e6d 656e 7420 7061 7468 2e20  vironment path. 
+0000b6f0: 5468 6520 7072 6f67 7261 6d20 6061 7374  The program `ast
+0000b700: 7261 6c2d 7072 6f60 2069 7320 7265 7175  ral-pro` is requ
+0000b710: 6972 6564 2074 6f20 6265 2069 6e73 7461  ired to be insta
+0000b720: 6c6c 6564 2069 6e20 7468 6520 656e 7669  lled in the envi
+0000b730: 726f 6e6d 656e 7420 7061 7468 2074 6f6f  ronment path too
+0000b740: 2e20 4e6f 7465 2074 6861 7420 7468 6572  . Note that ther
+0000b750: 6520 7368 6f75 6c64 2062 6520 6e6f 2064  e should be no d
+0000b760: 7570 6c69 6361 7465 6420 6765 6e65 2049  uplicated gene I
+0000b770: 4473 2069 6e20 7468 6520 7365 7175 656e  Ds in the sequen
+0000b780: 6365 2066 696c 652e 2054 6865 2070 6172  ce file. The par
+0000b790: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
+0000b7a0: 7265 2061 7265 2074 6f20 6163 6365 6c65  re are to accele
+0000b7b0: 7261 7465 2074 6865 2073 6571 7565 6e63  rate the sequenc
+0000b7c0: 6520 616c 6967 6e6d 656e 7420 616e 6420  e alignment and 
+0000b7d0: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
+0000b7e0: 6e63 6520 666f 7220 6561 6368 2067 656e  nce for each gen
+0000b7f0: 6520 6661 6d69 6c79 2061 6e64 2074 6875  e family and thu
+0000b800: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000b810: 6520 7365 7420 6173 206d 7563 6820 6173  e set as much as
+0000b820: 2074 6865 206e 756d 6265 7220 6f66 2067   the number of g
+0000b830: 656e 6520 6661 6d69 6c69 6573 2e0a 0a2a  ene families...*
+0000b840: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+0000b850: 7274 696e 6720 7275 6e20 6f66 2074 6869  rting run of thi
+0000b860: 7320 616e 616c 7973 6973 2069 7320 7769  s analysis is wi
+0000b870: 7468 2074 6865 2073 696d 706c 6520 636f  th the simple co
+0000b880: 6d6d 616e 6420 6265 6c6f 772e 2a2a 0a0a  mmand below.**..
+0000b890: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
+0000b8a0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000b8b0: 7365 7175 656e 6365 3320 2d61 7020 6170  sequence3 -ap ap
+0000b8c0: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
+0000b8d0: 2d6c 6520 6c65 6461 7461 202d 6774 2067  -le ledata -gt g
+0000b8e0: 7464 6174 6120 2d63 6f63 0a60 6060 0a0a  tdata -coc.```..
+0000b8f0: 2323 2320 7767 6420 666f 6375 730a 0a2a  ### wgd focus..*
+0000b900: 2a54 6865 2063 6f6e 6361 7465 6e61 7469  *The concatenati
+0000b910: 6f6e 2d62 6173 6564 2f63 6f61 6c65 7363  on-based/coalesc
+0000b920: 656e 6365 2d62 6173 6564 2070 6879 6c6f  ence-based phylo
+0000b930: 6765 6e65 7469 6320 696e 6665 7265 6e63  genetic inferenc
+0000b940: 652a 2a0a 6060 600a 7767 6420 666f 6375  e**.```.wgd focu
+0000b950: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000b960: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000b970: 6571 7565 6e63 6533 2028 2d2d 636f 6e63  equence3 (--conc
+0000b980: 6174 656e 6174 696f 6e29 2028 2d2d 636f  atenation) (--co
+0000b990: 616c 6573 6365 6e63 6529 2028 2d74 7265  alescence) (-tre
+0000b9a0: 6520 2766 6173 7474 7265 6527 2920 282d  e 'fasttree') (-
+0000b9b0: 7473 2027 2d66 6173 7465 7374 2729 2028  ts '-fastest') (
+0000b9c0: 2d6e 2034 2920 282d 2d74 6f5f 7374 6f70  -n 4) (--to_stop
+0000b9d0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+0000b9e0: 645f 666f 6375 7329 2028 2d74 2077 6f72  d_focus) (-t wor
+0000b9f0: 6b69 6e67 5f74 6d70 290a 6060 600a 0a54  king_tmp).```..T
+0000ba00: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+0000ba10: 666f 6375 7360 2069 6d70 6c65 6d65 6e74  focus` implement
+0000ba20: 6564 2074 776f 2062 6173 6963 2070 6879  ed two basic phy
+0000ba30: 6c6f 6765 6e65 7469 6320 696e 6665 7265  logenetic infere
+0000ba40: 6e63 6520 6d65 7468 6f64 732c 2069 2e65  nce methods, i.e
+0000ba50: 2e2c 2063 6f6e 6361 7465 6e61 7469 6f6e  ., concatenation
+0000ba60: 2d62 6173 6564 2061 6e64 2063 6f61 6c65  -based and coale
+0000ba70: 7363 656e 6365 2d62 6173 6564 206d 6574  scence-based met
+0000ba80: 686f 6473 2e20 546f 2069 6e69 7469 6174  hods. To initiat
+0000ba90: 6520 7468 6573 6520 616e 616c 7973 6973  e these analysis
+0000baa0: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000bab0: 7365 7420 7468 6520 666c 6167 206f 7074  set the flag opt
+0000bac0: 696f 6e20 602d 2d63 6f6e 6361 7465 6e61  ion `--concatena
+0000bad0: 7469 6f6e 6020 6f72 2060 2d2d 636f 616c  tion` or `--coal
+0000bae0: 6573 6365 6e63 6560 2e20 5468 6520 636f  escence`. The co
+0000baf0: 6e63 6174 656e 6174 696f 6e2d 6261 7365  ncatenation-base
+0000bb00: 6420 6d65 7468 6f64 2069 6e63 6c75 6465  d method include
+0000bb10: 7320 6120 6665 7720 6d61 6a6f 7220 7374  s a few major st
+0000bb20: 6570 732c 2069 2e65 2e2c 2074 6865 206d  eps, i.e., the m
+0000bb30: 756c 7469 706c 6520 7365 7175 656e 6365  ultiple sequence
+0000bb40: 2061 6c69 676e 6d65 6e74 2028 4d53 4129   alignment (MSA)
+0000bb50: 206f 6620 6561 6368 2067 656e 6520 6661   of each gene fa
+0000bb60: 6d69 6c79 2c20 7468 6520 636f 6e63 6174  mily, the concat
+0000bb70: 656e 6174 696f 6e20 6f66 2061 6c6c 2067  enation of all g
+0000bb80: 656e 6520 6661 6d69 6c69 6573 2061 6e64  ene families and
+0000bb90: 2074 6865 6e20 7468 6520 6765 6e65 2074   then the gene t
+0000bba0: 7265 6520 2861 6c73 6f20 7370 6563 6965  ree (also specie
+0000bbb0: 7320 7472 6565 2069 6e20 7468 6973 2063  s tree in this c
+0000bbc0: 6173 6529 2069 6e66 6572 656e 6365 2e20  ase) inference. 
+0000bbd0: 5468 6520 636f 616c 6573 6365 6e63 652d  The coalescence-
+0000bbe0: 6261 7365 6420 6d65 7468 6f64 2077 696c  based method wil
+0000bbf0: 6c20 696e 7374 6561 6420 7065 7266 6f72  l instead perfor
+0000bc00: 6d20 7468 6520 4d53 4120 6f66 2065 6163  m the MSA of eac
+0000bc10: 6820 6765 6e65 2066 616d 696c 7920 616e  h gene family an
+0000bc20: 6420 7468 6520 6765 6e65 2074 7265 6520  d the gene tree 
+0000bc30: 696e 6665 7265 6e63 6520 6261 7365 6420  inference based 
+0000bc40: 6f6e 2065 6163 6820 4d53 412c 2061 6e64  on each MSA, and
+0000bc50: 2074 6865 6e20 696e 6665 7220 7468 6520   then infer the 
+0000bc60: 7370 6563 6965 7320 7472 6565 2062 6173  species tree bas
+0000bc70: 6564 206f 6e20 7468 6573 6520 696e 6469  ed on these indi
+0000bc80: 7669 6475 616c 2067 656e 6520 7472 6565  vidual gene tree
+0000bc90: 732e 2054 6865 2074 7265 652d 696e 6665  s. The tree-infe
+0000bca0: 7265 6e63 6520 7072 6f67 7261 6d20 616e  rence program an
+0000bcb0: 6420 7468 6520 6173 736f 6369 6174 6564  d the associated
+0000bcc0: 2070 6172 616d 6574 6572 7320 6361 6e20   parameters can 
+0000bcd0: 6265 2073 6574 206a 7573 7420 6173 2061  be set just as a
+0000bce0: 626f 7665 2062 7920 602d 7472 6565 6020  bove by `-tree` 
+0000bcf0: 6f72 2060 2d2d 7472 6565 5f6d 6574 686f  or `--tree_metho
+0000bd00: 6460 2061 6e64 2060 2d74 7360 206f 7220  d` and `-ts` or 
+0000bd10: 602d 2d74 7265 6573 6574 602e 2050 6c65  `--treeset`. Ple
+0000bd20: 6173 6520 616c 736f 206d 616b 6520 7375  ase also make su
+0000bd30: 7265 2074 6865 2063 686f 7365 6e20 7472  re the chosen tr
+0000bd40: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+0000bd50: 6772 616d 2069 7320 696e 7374 616c 6c65  gram is installe
+0000bd60: 6420 696e 2074 6865 2065 6e76 6972 6f6e  d in the environ
+0000bd70: 6d65 6e74 2070 6174 682e 2054 6865 2070  ment path. The p
+0000bd80: 726f 6772 616d 2060 6173 7472 616c 2d70  rogram `astral-p
+0000bd90: 726f 6020 6973 2072 6571 7569 7265 6420  ro` is required 
+0000bda0: 746f 2062 6520 696e 7374 616c 6c65 6420  to be installed 
+0000bdb0: 696e 2074 6865 2065 6e76 6972 6f6e 6d65  in the environme
+0000bdc0: 6e74 2070 6174 6820 6966 2074 6865 2063  nt path if the c
+0000bdd0: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
+0000bde0: 206d 6574 686f 6420 6973 2063 686f 7365   method is chose
+0000bdf0: 6e2e 2054 6865 2070 6172 616c 6c65 6c20  n. The parallel 
+0000be00: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
+0000be10: 2074 6f20 6163 6365 6c65 7261 7465 2074   to accelerate t
+0000be20: 6865 2073 6571 7565 6e63 6520 616c 6967  he sequence alig
+0000be30: 6e6d 656e 7420 616e 6420 6765 6e65 2074  nment and gene t
+0000be40: 7265 6520 696e 6665 7265 6e63 6520 666f  ree inference fo
+0000be50: 7220 6561 6368 2067 656e 6520 6661 6d69  r each gene fami
+0000be60: 6c79 2074 6f6f 2061 6e64 2074 6875 7320  ly too and thus 
+0000be70: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+0000be80: 7365 7420 6173 206d 7563 6820 6173 2074  set as much as t
+0000be90: 6865 206e 756d 6265 7220 6f66 2067 656e  he number of gen
+0000bea0: 6520 6661 6d69 6c69 6573 2e0a 0a2a 2a41  e families...**A
+0000beb0: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
+0000bec0: 696e 6720 7275 6e20 6f66 2074 6869 7320  ing run of this 
+0000bed0: 616e 616c 7973 6973 2069 7320 7769 7468  analysis is with
+0000bee0: 2074 6865 2073 696d 706c 6520 636f 6d6d   the simple comm
+0000bef0: 616e 6420 6265 6c6f 772e 2a2a 0a0a 6060  and below.**..``
+0000bf00: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
+0000bf10: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+0000bf20: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000bf30: 6533 202d 2d63 6f6e 6361 7465 6e61 7469  e3 --concatenati
+0000bf40: 6f6e 0a77 6764 2066 6f63 7573 2066 616d  on.wgd focus fam
+0000bf50: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000bf60: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000bf70: 6365 3320 2d2d 636f 616c 6573 6365 6e63  ce3 --coalescenc
+0000bf80: 650a 6060 600a 0a2a 2a54 6865 2066 756e  e.```..**The fun
+0000bf90: 6374 696f 6e61 6c20 616e 6e6f 7461 7469  ctional annotati
+0000bfa0: 6f6e 206f 6620 6765 6e65 2066 616d 696c  on of gene famil
+0000bfb0: 6965 732a 2a0a 6060 600a 7767 6420 666f  ies**.```.wgd fo
+0000bfc0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000bfd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000bfe0: 2073 6571 7565 6e63 6533 202d 2d61 6e6e   sequence3 --ann
+0000bff0: 6f74 6174 696f 6e20 6567 676e 6f67 202d  otation eggnog -
+0000c000: 6564 2065 6464 6174 6120 2d2d 646d 6e62  ed eddata --dmnb
+0000c010: 2064 6264 6174 610a 6060 600a 0a54 6865   dbdata.```..The
+0000c020: 2070 726f 6772 616d 2060 7767 6420 666f   program `wgd fo
+0000c030: 6375 7360 2061 6c73 6f20 6164 6465 6420  cus` also added 
+0000c040: 736f 6d65 2077 7261 7070 696e 6720 6675  some wrapping fu
+0000c050: 6e63 7469 6f6e 7320 666f 7220 6675 6e63  nctions for func
+0000c060: 7469 6f6e 616c 2061 6e6e 6f74 6174 696f  tional annotatio
+0000c070: 6e20 6f66 2067 656e 6520 6661 6d69 6c69  n of gene famili
+0000c080: 6573 206f 6e20 7468 6520 686f 6f64 206f  es on the hood o
+0000c090: 6620 6461 7461 6261 7365 7320 616e 6420  f databases and 
+0000c0a0: 736f 6674 7761 7265 7320 666f 7220 696e  softwares for in
+0000c0b0: 7374 616e 6365 2060 4567 674e 4f47 6020  stance `EggNOG` 
+0000c0c0: 616e 6420 6045 6767 4e4f 472d 6d61 7070  and `EggNOG-mapp
+0000c0d0: 6572 602e 2046 6f72 2074 6865 2061 6e6e  er`. For the ann
+0000c0e0: 6f74 6174 696f 6e20 7573 696e 6720 6045  otation using `E
+0000c0f0: 6767 4e4f 472d 6d61 7070 6572 602c 2075  ggNOG-mapper`, u
+0000c100: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+0000c110: 7669 6465 2074 6865 2070 6174 6820 746f  vide the path to
+0000c120: 2074 6865 2065 6767 4e4f 4720 616e 6e6f   the eggNOG anno
+0000c130: 7461 7469 6f6e 2064 6174 6162 6173 6520  tation database 
+0000c140: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+0000c150: 2d65 6460 2c20 7468 6520 7061 7468 2074  -ed`, the path t
+0000c160: 6f20 7468 6520 6469 616d 6f6e 642d 636f  o the diamond-co
+0000c170: 6d70 6174 6962 6c65 2064 6174 6162 6173  mpatible databas
+0000c180: 6520 7669 6120 7468 6520 6f70 7469 6f6e  e via the option
+0000c190: 2060 2d2d 646d 6e64 5f64 6260 2061 6e64   `--dmnd_db` and
+0000c1a0: 2074 6865 206f 7074 696f 6e20 602d 2d61   the option `--a
+0000c1b0: 6e6e 6f74 6174 696f 6e60 2073 6574 2061  nnotation` set a
+0000c1c0: 7320 2265 6767 6e6f 6722 2e20 5468 6520  s "eggnog". The 
+0000c1d0: 6d61 6e6e 6572 2068 6f77 2050 4641 4d20  manner how PFAM 
+0000c1e0: 616e 6e6f 7461 7469 6f6e 2077 696c 6c20  annotation will 
+0000c1f0: 6265 2070 6572 666f 726d 6564 2063 616e  be performed can
+0000c200: 2062 6520 636f 6e74 726f 6c6c 6564 2076   be controlled v
+0000c210: 6961 2074 6865 206f 7074 696f 6e20 6070  ia the option `p
+0000c220: 6661 6d60 2c20 6569 7468 6572 2022 6e6f  fam`, either "no
+0000c230: 6e65 222c 2022 7265 616c 6967 6e22 206f  ne", "realign" o
+0000c240: 7220 2264 656e 6f76 6f22 2c20 7468 6520  r "denovo", the 
+0000c250: 6465 7461 696c 6564 2065 7870 6c61 6e61  detailed explana
+0000c260: 7469 6f6e 2063 616e 2062 6520 666f 756e  tion can be foun
+0000c270: 6420 6174 2074 6865 2077 696b 6920 6f66  d at the wiki of
+0000c280: 205b 4567 674e 4f47 2d6d 6170 7065 725d   [EggNOG-mapper]
+0000c290: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+0000c2a0: 636f 6d2f 6567 676e 6f67 6462 2f65 6767  com/eggnogdb/egg
+0000c2b0: 6e6f 672d 6d61 7070 6572 2f77 696b 6929  nog-mapper/wiki)
+0000c2c0: 2e20 506c 6561 7365 2070 7265 2d69 6e73  . Please pre-ins
+0000c2d0: 7461 6c6c 2074 6865 2060 4567 674e 4f47  tall the `EggNOG
+0000c2e0: 2d6d 6170 7065 7260 2070 7974 686f 6e20  -mapper` python 
+0000c2f0: 7061 636b 6167 6520 6966 2075 7369 6e67  package if using
+0000c300: 2074 6869 7320 6675 6e63 7469 6f6e 2e20   this function. 
+0000c310: 466f 7220 7468 6520 616e 6e6f 7461 7469  For the annotati
+0000c320: 6f6e 2075 7369 6e67 2060 686d 6d73 6361  on using `hmmsca
+0000c330: 6e60 2c20 7768 6174 2069 7320 696d 706c  n`, what is impl
+0000c340: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+0000c350: 7632 6020 6973 2061 2073 696d 706c 6520  v2` is a simple 
+0000c360: 6275 6e64 6c65 2066 756e 6374 696f 6e20  bundle function 
+0000c370: 746f 2070 6572 666f 726d 2060 686d 6d73  to perform `hmms
+0000c380: 6361 6e60 2061 6e61 6c79 7369 7320 666f  can` analysis fo
+0000c390: 7220 6120 6769 7665 6e20 686d 6d20 7072  r a given hmm pr
+0000c3a0: 6f66 696c 6520 616e 6420 7365 7420 6f66  ofile and set of
+0000c3b0: 2067 656e 6520 6661 6d69 6c69 6573 2073   gene families s
+0000c3c0: 7563 6820 7468 6174 2075 7365 7273 206e  uch that users n
+0000c3d0: 6565 6420 746f 2073 6574 2074 6865 206f  eed to set the o
+0000c3e0: 7074 696f 6e20 602d 2d61 6e6e 6f74 6174  ption `--annotat
+0000c3f0: 696f 6e60 2061 7320 2268 6d6d 7066 616d  ion` as "hmmpfam
+0000c400: 2220 616e 6420 7072 6f76 6964 6520 7468  " and provide th
+0000c410: 6520 686d 6d70 726f 6669 6c65 2076 6961  e hmmprofile via
+0000c420: 2074 6865 206f 7074 696f 6e20 602d 2d68   the option `--h
+0000c430: 6d6d 602e 2046 6f72 2074 6865 2061 6e6e  mm`. For the ann
+0000c440: 6f74 6174 696f 6e20 7573 696e 6720 6069  otation using `i
+0000c450: 6e74 6572 7072 6f73 6361 6e60 2c20 7573  nterproscan`, us
+0000c460: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
+0000c470: 6964 6520 7468 6520 7061 7468 2074 6f20  ide the path to 
+0000c480: 7468 6520 696e 7465 7270 726f 7363 616e  the interproscan
+0000c490: 2069 6e73 7461 6c6c 6174 696f 6e20 666f   installation fo
+0000c4a0: 6c64 6572 2077 6865 7265 2074 6865 7265  lder where there
+0000c4b0: 2069 7320 6120 6069 6e74 6572 7072 6f73   is a `interpros
+0000c4c0: 6361 6e2e 7368 6020 6669 6c65 2076 6961  can.sh` file via
+0000c4d0: 2074 6865 206f 7074 696f 6e20 602d 2d65   the option `--e
+0000c4e0: 7865 7061 7468 6020 616e 6420 7365 7420  xepath` and set 
+0000c4f0: 7468 6520 7468 6520 6f70 7469 6f6e 2060  the the option `
+0000c500: 2d2d 616e 6e6f 7461 7469 6f6e 6020 6173  --annotation` as
+0000c510: 2022 696e 7465 7270 726f 7363 616e 222e   "interproscan".
+0000c520: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
+0000c530: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
+0000c540: 6f20 7061 7261 6c6c 656c 697a 6520 7468  o parallelize th
+0000c550: 6520 616e 616c 7973 6973 2066 6f72 2065  e analysis for e
+0000c560: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
+0000c570: 616e 6420 7468 7573 2073 7567 6765 7374  and thus suggest
+0000c580: 6564 2074 6f20 6265 2073 6574 2061 7320  ed to be set as 
+0000c590: 6d75 6368 2061 7320 7468 6520 6e75 6d62  much as the numb
+0000c5a0: 6572 206f 6620 6765 6e65 2066 616d 696c  er of gene famil
+0000c5b0: 6965 732e 0a0a 2a2a 4120 7375 6767 6573  ies...**A sugges
+0000c5c0: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
+0000c5d0: 206f 6620 7468 6973 2061 6e61 6c79 7369   of this analysi
+0000c5e0: 7320 6361 6e20 6265 2077 6974 6820 7468  s can be with th
+0000c5f0: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
+0000c600: 2a2a 0a0a 6060 600a 7767 6420 666f 6375  **..```.wgd focu
+0000c610: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000c620: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000c630: 6571 7565 6e63 6533 202d 2d61 6e6e 6f74  equence3 --annot
+0000c640: 6174 696f 6e20 6567 676e 6f67 202d 6564  ation eggnog -ed
+0000c650: 2065 6464 6174 6120 2d2d 646d 6e62 2064   eddata --dmnb d
+0000c660: 6264 6174 610a 7767 6420 666f 6375 7320  bdata.wgd focus 
+0000c670: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
+0000c680: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
+0000c690: 7565 6e63 6533 202d 2d61 6e6e 6f74 6174  uence3 --annotat
+0000c6a0: 696f 6e20 686d 6d70 6661 6d20 2d2d 686d  ion hmmpfam --hm
+0000c6b0: 6d20 686d 6d64 6174 610a 7767 6420 666f  m hmmdata.wgd fo
+0000c6c0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000c6d0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000c6e0: 2073 6571 7565 6e63 6533 202d 2d61 6e6e   sequence3 --ann
+0000c6f0: 6f74 6174 696f 6e20 696e 7465 7270 726f  otation interpro
+0000c700: 7363 616e 202d 2d65 7865 7061 7468 2024  scan --exepath $
+0000c710: 5041 5448 0a60 6060 0a0a 2a2a 5468 6520  PATH.```..**The 
+0000c720: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+0000c730: 696e 6720 6f66 2057 4744 732a 2a0a 6060  ing of WGDs**.``
+0000c740: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
+0000c750: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+0000c760: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000c770: 6533 202d 6420 6d63 6d63 7472 6565 202d  e3 -d mcmctree -
+0000c780: 7370 2073 7064 6174 6120 282d 2d70 726f  sp spdata (--pro
+0000c790: 7463 6f63 6461 7469 6e67 2920 282d 2d70  tcocdating) (--p
+0000c7a0: 6172 7469 7469 6f6e 2920 282d 2d61 616d  artition) (--aam
+0000c7b0: 6f64 656c 206c 6729 2028 2d64 7320 2762  odel lg) (-ds 'b
+0000c7c0: 7572 6e69 6e20 3d20 3230 3030 2729 2028  urnin = 2000') (
+0000c7d0: 2d64 7320 2773 616d 7066 7265 7120 3d20  -ds 'sampfreq = 
+0000c7e0: 3130 3030 2729 2028 2d64 7320 276e 7361  1000') (-ds 'nsa
+0000c7f0: 6d70 6c65 203d 2032 3030 3030 2729 2020  mple = 20000')  
+0000c800: 282d 6e20 3429 2028 2d2d 746f 5f73 746f  (-n 4) (--to_sto
+0000c810: 7029 2028 2d2d 6364 7329 2028 2d6f 2077  p) (--cds) (-o w
+0000c820: 6764 5f66 6f63 7573 2920 282d 7420 776f  gd_focus) (-t wo
+0000c830: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
+0000c840: 5468 6520 6162 736f 6c75 7465 2064 6174  The absolute dat
+0000c850: 696e 6720 6f66 2057 4744 7320 6973 2061  ing of WGDs is a
+0000c860: 2073 7065 6369 6669 6320 7069 7065 6c69   specific pipeli
+0000c870: 6e65 2069 6d70 6c65 6d65 6e74 6564 2069  ne implemented i
+0000c880: 6e20 6077 6764 2076 3260 2075 7369 6e67  n `wgd v2` using
+0000c890: 2074 6865 206d 6574 686f 6420 6f66 2070   the method of p
+0000c8a0: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
+0000c8b0: 6e67 2e20 5468 6520 6661 6d69 6c69 6573  ng. The families
+0000c8c0: 2075 7365 6420 696e 2074 6869 7320 7374   used in this st
+0000c8d0: 6570 2063 616e 2062 6520 7072 6f64 7563  ep can be produc
+0000c8e0: 6564 2066 726f 6d20 6077 6764 2064 6d64  ed from `wgd dmd
+0000c8f0: 6020 616e 6420 6077 6764 2070 6561 6b60  ` and `wgd peak`
+0000c900: 2e20 4e6f 7465 2074 6861 7420 6865 7265  . Note that here
+0000c910: 2077 6520 6f6e 6c79 2064 6973 6375 7373   we only discuss
+0000c920: 2068 6f77 2074 6f20 6461 7465 2057 4744   how to date WGD
+0000c930: 7320 7769 7468 2067 656e 6f6d 6520 6173  s with genome as
+0000c940: 7365 6d62 6c79 2c20 696e 7374 6561 6420  sembly, instead 
+0000c950: 6f66 2074 7261 6e73 6372 6970 746f 6d65  of transcriptome
+0000c960: 2061 7373 656d 626c 7920 2877 6869 6368   assembly (which
+0000c970: 2077 696c 6c20 6265 2064 6973 6375 7373   will be discuss
+0000c980: 6564 2069 6e20 6120 7365 7061 7261 7465  ed in a separate
+0000c990: 2073 6563 7469 6f6e 2068 6572 6575 6e64   section hereund
+0000c9a0: 6572 292e 2054 6865 2061 7373 756d 7074  er). The assumpt
+0000c9b0: 696f 6e20 7765 206d 6164 6520 6865 7265  ion we made here
+0000c9c0: 2069 7320 7468 6174 206e 6f74 2061 6c6c   is that not all
+0000c9d0: 2061 6e63 686f 7220 7061 6972 7320 2863   anchor pairs (c
+0000c9e0: 6f6c 6c69 6e65 6172 2064 7570 6c69 6361  ollinear duplica
+0000c9f0: 7465 7329 2061 7265 2073 7569 7461 626c  tes) are suitabl
+0000ca00: 6520 666f 7220 7068 796c 6f67 656e 6574  e for phylogenet
+0000ca10: 6963 2064 6174 696e 672c 2066 6f72 2069  ic dating, for i
+0000ca20: 6e73 7461 6e63 6520 7468 6f73 6520 6661  nstance those fa
+0000ca30: 7374 6c79 206f 7220 736c 6f77 6c79 2065  stly or slowly e
+0000ca40: 766f 6c76 696e 6720 6765 6e65 2064 7570  volving gene dup
+0000ca50: 6c69 6361 7465 732c 2062 6563 6175 7365  licates, because
+0000ca60: 2074 6865 7927 7265 2070 726f 6e65 2074   they're prone t
+0000ca70: 6f20 6769 7665 2062 6961 7365 6420 6461  o give biased da
+0000ca80: 7469 6e67 2065 7374 696d 6174 696f 6e2e  ting estimation.
+0000ca90: 2053 6f20 6173 2074 6f20 7265 7472 6965   So as to retrie
+0000caa0: 7665 2074 6865 2022 7265 6c69 6162 6c65  ve the "reliable
+0000cab0: 2220 616e 6368 6f72 2070 6169 7273 2c20  " anchor pairs, 
+0000cac0: 7765 2069 6d70 6c65 6d65 6e74 6564 2073  we implemented s
+0000cad0: 6f6d 6520 6d65 7468 6f64 7320 6f66 2069  ome methods of i
+0000cae0: 6465 6e74 6966 7969 6e67 2063 7265 6469  dentifying credi
+0000caf0: 6162 6c65 2061 6e63 686f 7220 7061 6972  able anchor pair
+0000cb00: 7320 6261 7365 6420 6f6e 2074 6865 6972  s based on their
+0000cb10: 204b 7320 7661 6c75 6573 2061 6e64 2f6f   Ks values and/o
+0000cb20: 7220 7265 7369 6469 6e67 2063 6f6c 6c69  r residing colli
+0000cb30: 6e65 6172 2073 6567 6d65 6e74 732e 2046  near segments. F
+0000cb40: 6f72 2061 2067 656e 6f6d 6520 7769 7468  or a genome with
+0000cb50: 2063 6c65 6172 2073 6967 6e61 6c73 206f   clear signals o
+0000cb60: 6620 7075 7461 7469 7665 2057 4744 732c  f putative WGDs,
+0000cb70: 2073 7563 6820 6173 2074 6865 202a 4171   such as the *Aq
+0000cb80: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
+0000cb90: 2a20 696e 2074 6865 2065 7861 6d70 6c65  * in the example
+0000cba0: 2c20 6120 6865 7572 6973 7469 6320 6d65  , a heuristic me
+0000cbb0: 7468 6f64 2074 6861 7420 6170 706c 6965  thod that applie
+0000cbc0: 7320 7468 6520 7072 696e 6369 706c 6520  s the principle 
+0000cbd0: 6f66 2068 6f77 205b 6b73 7261 7465 735d  of how [ksrates]
+0000cbe0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+0000cbf0: 636f 6d2f 5649 422d 5053 422f 6b73 7261  com/VIB-PSB/ksra
+0000cc00: 7465 7329 2066 696e 6420 7468 6520 696e  tes) find the in
+0000cc10: 6974 6961 6c20 7065 616b 7320 616e 6420  itial peaks and 
+0000cc20: 7468 6569 7220 7061 7261 6d65 7465 7273  their parameters
+0000cc30: 2077 6173 2069 6d70 6c65 6d65 6e74 6564   was implemented
+0000cc40: 2074 6f20 6669 6e64 2074 6865 2039 3525   to find the 95%
+0000cc50: 2063 6f6e 6669 6465 6e63 6520 6c65 7665   confidence leve
+0000cc60: 6c20 6f66 2074 6865 2061 7373 756d 6564  l of the assumed
+0000cc70: 206c 6f67 6e6f 726d 616c 2064 6973 7472   lognormal distr
+0000cc80: 6962 7574 696f 6e20 6f66 2074 6865 2061  ibution of the a
+0000cc90: 6e63 686f 7220 7061 6972 202a 4b2a 3c73  nchor pair *K*<s
+0000cca0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000ccb0: 6973 7472 6962 7574 696f 6e20 746f 2066  istribution to f
+0000ccc0: 696c 7465 7220 616e 6368 6f72 2070 6169  ilter anchor pai
+0000ccd0: 7273 2077 6974 6820 746f 6f20 6869 6768  rs with too high
+0000cce0: 206f 7220 6c6f 7720 2a4b 2a3c 7375 623e   or low *K*<sub>
+0000ccf0: 533c 2f73 7562 3e20 6167 6573 2c20 7468  S</sub> ages, th
+0000cd00: 6520 6578 616d 706c 6172 2063 6f6d 6d61  e examplar comma
+0000cd10: 6e64 206f 6620 7768 6963 6820 6973 2073  nd of which is s
+0000cd20: 686f 7765 6420 696e 2074 6865 2060 7767  howed in the `wg
+0000cd30: 6420 7065 616b 6020 7365 6374 696f 6e20  d peak` section 
+0000cd40: 6f66 205b 496c 6c75 7374 7261 7469 6f6e  of [Illustration
+0000cd50: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+0000cd60: 2e63 6f6d 2f68 6563 6865 2d70 7362 2f77  .com/heche-psb/w
+0000cd70: 6764 3f74 6162 3d72 6561 646d 652d 6f76  gd?tab=readme-ov
+0000cd80: 2d66 696c 6523 696c 6c75 7374 7261 7469  -file#illustrati
+0000cd90: 6f6e 292e 2049 6620 7468 6973 2068 6575  on). If this heu
+0000cda0: 7269 7374 6963 206d 6574 686f 6420 6661  ristic method fa
+0000cdb0: 696c 6564 2074 6f20 6769 7665 2072 6561  iled to give rea
+0000cdc0: 736f 6e61 626c 6520 7265 7375 6c74 732c  sonable results,
+0000cdd0: 2077 6869 6368 2069 7320 7573 7561 6c6c   which is usuall
+0000cde0: 7920 6475 6520 746f 2074 6865 2065 6666  y due to the eff
+0000cdf0: 6563 7420 6f66 206d 756c 7469 706c 6520  ect of multiple 
+0000ce00: 6164 6a61 6365 6e74 2057 4744 7320 7468  adjacent WGDs th
+0000ce10: 6174 2062 6c75 7273 2074 6865 2070 6561  at blurs the pea
+0000ce20: 6b2d 6669 6e64 696e 6720 7072 6f63 6573  k-finding proces
+0000ce30: 732c 2075 7365 7273 2063 616e 2074 7572  s, users can tur
+0000ce40: 6e20 746f 2074 6865 2063 6f6c 6c69 6e65  n to the colline
+0000ce50: 6172 2073 6567 6d65 6e74 732d 6775 6964  ar segments-guid
+0000ce60: 6564 2061 6e63 686f 7220 7061 6972 2063  ed anchor pair c
+0000ce70: 6c75 7374 6572 696e 6720 696d 706c 656d  lustering implem
+0000ce80: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
+0000ce90: 602c 2069 6e20 7768 6963 6820 6120 636f  `, in which a co
+0000cea0: 6c6c 696e 6561 7220 7365 676d 656e 742d  llinear segment-
+0000ceb0: 7769 7365 2047 4d4d 2063 6c75 7374 6572  wise GMM cluster
+0000cec0: 696e 6720 7769 6c6c 2062 6520 6669 7273  ing will be firs
+0000ced0: 7420 636f 6e64 7563 7465 6420 6261 7365  t conducted base
+0000cee0: 6420 6f6e 2074 6865 2022 736f 2d63 616c  d on the "so-cal
+0000cef0: 6c65 6422 2073 6567 6d65 6e74 202a 4b2a  led" segment *K*
+0000cf00: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000cf10: 2072 6570 7265 7365 6e74 6564 2062 7920   represented by 
+0000cf20: 7468 6520 6d65 6469 616e 202a 4b2a 3c73  the median *K*<s
+0000cf30: 7562 3e53 3c2f 7375 623e 2061 6765 206f  ub>S</sub> age o
+0000cf40: 6620 616c 6c20 7468 6520 7265 7369 6469  f all the residi
+0000cf50: 6e67 2073 796e 7465 6c6f 6773 2028 6e6f  ng syntelogs (no
+0000cf60: 7465 2074 6861 7420 7468 6520 6765 6e65  te that the gene
+0000cf70: 2064 7570 6c69 6361 7465 2070 6169 7273   duplicate pairs
+0000cf80: 2061 646f 7074 6564 2069 6e20 7468 6973   adopted in this
+0000cf90: 2073 7465 7020 6973 2066 726f 6d20 7468   step is from th
+0000cfa0: 6520 6669 6c65 2060 6d75 6c74 6970 6c69  e file `multipli
+0000cfb0: 636f 6e5f 7061 6972 732e 7478 7460 2077  con_pairs.txt` w
+0000cfc0: 6869 6368 2063 6f6e 7461 696e 7320 7468  hich contains th
+0000cfd0: 6520 6675 6c6c 2073 6574 206f 6620 7379  e full set of sy
+0000cfe0: 6e74 656c 6f67 7329 2c20 696e 7374 6561  ntelogs), instea
+0000cff0: 6420 6f66 2074 6865 2073 6d61 6c6c 6572  d of the smaller
+0000d000: 2067 656e 6520 7365 7420 6f66 2061 6e63   gene set of anc
+0000d010: 686f 7220 7061 6972 732e 2054 6865 2064  hor pairs. The d
+0000d020: 6973 7469 6e63 7469 6f6e 2062 6574 7765  istinction betwe
+0000d030: 656e 2061 6e63 686f 7220 7061 6972 7320  en anchor pairs 
+0000d040: 616e 6420 7379 6e74 656c 6f67 7320 6973  and syntelogs is
+0000d050: 2074 6861 7420 7468 6520 6c61 7474 6572   that the latter
+0000d060: 2072 6566 6572 7320 746f 206d 756c 7469   refers to multi
+0000d070: 706c 6520 7365 7473 206f 6620 6765 6e65  ple sets of gene
+0000d080: 7320 6465 7269 7665 6420 6672 6f6d 2074  s derived from t
+0000d090: 6865 2073 616d 6520 616e 6365 7374 7261  he same ancestra
+0000d0a0: 6c20 6765 6e6f 6d69 6320 7265 6769 6f6e  l genomic region
+0000d0b0: 2077 6869 6c65 2074 6865 2066 6f72 6d65   while the forme
+0000d0c0: 7220 696d 706c 6965 7320 7468 6520 6c61  r implies the la
+0000d0d0: 7474 6572 2062 7574 2072 6571 7569 7265  tter but require
+0000d0e0: 7320 6164 6469 7469 6f6e 616c 6c79 2074  s additionally t
+0000d0f0: 6865 2063 6f6e 7365 7276 6564 2067 656e  he conserved gen
+0000d100: 6520 6f72 6465 722c 2062 6f74 6820 6f66  e order, both of
+0000d110: 2077 6869 6368 2061 7265 2c20 7769 7468   which are, with
+0000d120: 696e 2061 2067 656e 6f6d 652c 2061 7373  in a genome, ass
+0000d130: 756d 6564 2074 6f20 6265 206f 7269 6769  umed to be origi
+0000d140: 6e61 7465 6420 6672 6f6d 2074 6865 2064  nated from the d
+0000d150: 7570 6c69 6361 7469 6f6e 206f 6620 6120  uplication of a 
+0000d160: 636f 6d6d 6f6e 2061 6e63 6573 7472 616c  common ancestral
+0000d170: 2067 656e 6f6d 6963 2072 6567 696f 6e20   genomic region 
+0000d180: 616e 6420 6173 2073 7563 6820 6465 656d  and as such deem
+0000d190: 6564 2065 7669 6465 6e63 6520 666f 7220  ed evidence for 
+0000d1a0: 5747 442e 2054 6865 6e20 7468 6520 7379  WGD. Then the sy
+0000d1b0: 6e74 656c 6f67 7320 7769 6c6c 2062 6520  ntelogs will be 
+0000d1c0: 6d61 7070 6564 2062 6163 6b20 6163 636f  mapped back acco
+0000d1d0: 7264 696e 6720 746f 2074 6865 2063 6c75  rding to the clu
+0000d1e0: 7374 6572 696e 6720 7265 7375 6c74 7320  stering results 
+0000d1f0: 6f66 2074 6865 6972 2061 6666 696c 6961  of their affilia
+0000d200: 7465 6420 7365 676d 656e 7473 2e20 5369  ted segments. Si
+0000d210: 6e63 6520 7468 6520 4761 7573 7369 616e  nce the Gaussian
+0000d220: 2073 6861 7065 206f 6620 7468 6520 7365   shape of the se
+0000d230: 676d 656e 7420 636c 7573 7465 7220 646f  gment cluster do
+0000d240: 6573 6e27 7420 6e65 6365 7373 6172 696c  esn't necessaril
+0000d250: 7920 696d 6974 6174 6520 7468 6520 7368  y imitate the sh
+0000d260: 6170 6520 6f66 2074 6865 2072 6573 6964  ape of the resid
+0000d270: 696e 6720 7379 6e74 656c 6f67 732c 2073  ing syntelogs, s
+0000d280: 6f20 6173 2074 6f20 7265 7472 6965 7665  o as to retrieve
+0000d290: 2074 6865 2022 7265 6c69 6162 6c65 2220   the "reliable" 
+0000d2a0: 6765 6e65 2073 6574 7320 666f 7220 7068  gene sets for ph
+0000d2b0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+0000d2c0: 672c 2077 6520 6164 706f 7465 6420 7468  g, we adpoted th
+0000d2d0: 6520 2839 3525 2920 6869 6768 6573 7420  e (95%) highest 
+0000d2e0: 6465 6e73 6974 7920 7265 6769 6f6e 2028  density region (
+0000d2f0: 4844 5229 206f 6620 7468 6520 7379 6e74  HDR) of the synt
+0000d300: 656c 6f67 202a 4b2a 3c73 7562 3e53 3c2f  elog *K*<sub>S</
+0000d310: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000d320: 7574 696f 6e20 666f 7220 7468 6520 7068  ution for the ph
+0000d330: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+0000d340: 672c 2074 6865 2063 616c 6375 6c61 7469  g, the calculati
+0000d350: 6f6e 206f 6620 7468 6520 2839 3525 2920  on of the (95%) 
+0000d360: 4844 5220 696e 2074 6865 2066 756e 6374  HDR in the funct
+0000d370: 696f 6e20 5b63 616c 6375 6c61 7465 4850  ion [calculateHP
+0000d380: 445d 2868 7474 7073 3a2f 2f67 6974 6875  D](https://githu
+0000d390: 622e 636f 6d2f 6865 6368 652d 7073 622f  b.com/heche-psb/
+0000d3a0: 7767 642f 626c 6f62 2f70 6879 6c6f 6461  wgd/blob/phyloda
+0000d3b0: 7469 6e67 2f77 6764 2f70 6561 6b2e 7079  ting/wgd/peak.py
+0000d3c0: 234c 3137 3537 2920 7365 656b 7320 7468  #L1757) seeks th
+0000d3d0: 6520 7368 6f72 7465 7374 202a 4b2a 3c73  e shortest *K*<s
+0000d3e0: 7562 3e53 3c2f 7375 623e 2072 616e 6765  ub>S</sub> range
+0000d3f0: 2028 612c 6229 2077 6869 6368 2073 6174   (a,b) which sat
+0000d400: 6973 6669 6573 2074 6865 2072 6571 7569  isfies the requi
+0000d410: 7265 6d65 6e74 206f 6620 7370 616e 6e69  rement of spanni
+0000d420: 6e67 206d 6f72 6520 7468 616e 2028 3935  ng more than (95
+0000d430: 2529 206f 6620 616c 6c20 7468 6520 2a4b  %) of all the *K
+0000d440: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+0000d450: 6c75 6573 2e20 4f6e 2074 6865 2070 7265  lues. On the pre
+0000d460: 6d69 7365 206f 6620 6964 656e 7469 6669  mise of identifi
+0000d470: 6564 2061 6e63 686f 7220 7061 6972 7320  ed anchor pairs 
+0000d480: 286e 6f74 6520 7468 6174 2074 6865 2073  (note that the s
+0000d490: 796e 7465 6c6f 6773 2061 7265 2061 6c73  yntelogs are als
+0000d4a0: 6f20 7265 6665 7272 6564 2061 7320 616e  o referred as an
+0000d4b0: 6368 6f72 2070 6169 7273 2068 6572 6561  chor pairs herea
+0000d4c0: 6674 6572 292c 2077 6520 696d 706c 656d  fter), we implem
+0000d4d0: 656e 7465 6420 696e 2074 6865 2070 726f  ented in the pro
+0000d4e0: 6772 616d 2060 7767 6420 646d 6460 2074  gram `wgd dmd` t
+0000d4f0: 6865 2073 6f2d 6361 6c6c 6564 2061 6e63  he so-called anc
+0000d500: 686f 722d 6177 6172 6520 6c6f 6361 6c20  hor-aware local 
+0000d510: 4d52 4248 7320 6f72 206f 7274 686f 6772  MRBHs or orthogr
+0000d520: 6f75 7073 2c20 696e 2077 6869 6368 2074  oups, in which t
+0000d530: 6865 206f 7269 6769 6e61 6c20 6c6f 6361  he original loca
+0000d540: 6c20 4d52 4248 7320 6172 6520 6675 7274  l MRBHs are furt
+0000d550: 6865 7220 6d65 7267 6564 2077 6974 6820  her merged with 
+0000d560: 616e 6368 6f72 2070 6169 7273 2073 7563  anchor pairs suc
+0000d570: 6820 7468 6174 2065 6163 6820 6f72 7468  h that each orth
+0000d580: 6f67 726f 7570 2063 6f6e 7461 696e 7320  ogroup contains 
+0000d590: 7468 6520 616e 6368 6f72 2070 6169 7220  the anchor pair 
+0000d5a0: 616e 6420 7468 6520 6f72 7468 6f6c 6f67  and the ortholog
+0000d5b0: 7565 732e 2057 6974 6820 7468 6973 206f  ues. With this o
+0000d5c0: 7274 686f 6772 6f75 702c 2075 7365 7273  rthogroup, users
+0000d5d0: 2074 6865 6e20 6e65 6564 206f 6e65 2073   then need one s
+0000d5e0: 7461 7274 696e 6720 7472 6565 2066 696c  tarting tree fil
+0000d5f0: 6520 2861 7320 7368 6f77 6e20 696e 2074  e (as shown in t
+0000d600: 6865 205b 496c 6c75 7374 7261 7469 6f6e  he [Illustration
+0000d610: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+0000d620: 2e63 6f6d 2f68 6563 6865 2d70 7362 2f77  .com/heche-psb/w
+0000d630: 6764 2f74 7265 652f 7068 796c 6f64 6174  gd/tree/phylodat
+0000d640: 696e 673f 7461 623d 7265 6164 6d65 2d6f  ing?tab=readme-o
+0000d650: 762d 6669 6c65 2369 6c6c 7573 7472 6174  v-file#illustrat
+0000d660: 696f 6e29 2070 6172 7429 2069 6e64 6963  ion) part) indic
+0000d670: 6174 696e 6720 7468 6520 7472 6565 2074  ating the tree t
+0000d680: 6f70 6f6c 6f67 7920 616e 6420 666f 7373  opology and foss
+0000d690: 696c 2063 616c 6962 7261 7469 6f6e 2069  il calibration i
+0000d6a0: 6e66 6f72 6d61 7469 6f6e 2066 6f72 2074  nformation for t
+0000d6b0: 6865 2066 696e 616c 2070 6879 6c6f 6765  he final phyloge
+0000d6c0: 6e65 7469 6320 6461 7469 6e67 2075 7369  netic dating usi
+0000d6d0: 6e67 2074 6865 2070 726f 6772 616d 2060  ng the program `
+0000d6e0: 7767 6420 666f 6375 7360 2e20 5573 6572  wgd focus`. User
+0000d6f0: 7320 6361 6e20 7365 7420 7468 6520 666c  s can set the fl
+0000d700: 6167 206f 7074 696f 6e20 602d 2d70 726f  ag option `--pro
+0000d710: 7463 6f63 6461 7469 6e67 6020 746f 206f  tcocdating` to o
+0000d720: 6e6c 7920 636f 6e64 7563 7420 7468 6520  nly conduct the 
+0000d730: 6461 7469 6e67 206f 6620 7468 6520 636f  dating of the co
+0000d740: 6e63 6174 656e 6174 696f 6e20 7072 6f74  ncatenation prot
+0000d750: 6569 6e20 4d53 4120 6f72 2074 6865 2066  ein MSA or the f
+0000d760: 6c61 6720 6f70 7469 6f6e 2060 2d2d 7072  lag option `--pr
+0000d770: 6f74 6461 7469 6e67 6020 746f 206f 6e6c  otdating` to onl
+0000d780: 7920 636f 6e64 7563 7420 7468 6520 6461  y conduct the da
+0000d790: 7469 6e67 206f 6620 7468 6520 7072 6f74  ting of the prot
+0000d7a0: 6569 6e20 4d53 4173 2c20 6e6f 7465 6420  ein MSAs, noted 
+0000d7b0: 7468 6174 2074 6865 7365 2074 776f 206f  that these two o
+0000d7c0: 7074 696f 6e73 206f 6e6c 7920 776f 726b  ptions only work
+0000d7d0: 2066 6f72 2074 6865 2060 6d63 6d63 7472   for the `mcmctr
+0000d7e0: 6565 6020 6f70 7469 6f6e 2073 6f20 6661  ee` option so fa
+0000d7f0: 722e 2054 6865 2066 6c61 6720 6f70 7469  r. The flag opti
+0000d800: 6f6e 2060 2d2d 7061 7274 6974 696f 6e60  on `--partition`
+0000d810: 2063 616e 2062 6520 7365 7420 746f 2070   can be set to p
+0000d820: 6572 666f 726d 2060 6d63 6d63 7472 6565  erform `mcmctree
+0000d830: 6020 616e 616c 7973 6973 2075 7369 6e67  ` analysis using
+0000d840: 2074 6865 2070 6172 7469 7469 6f6e 6564   the partitioned
+0000d850: 2064 6174 6120 2869 2e65 2e2c 2031 7374   data (i.e., 1st
+0000d860: 2c20 326e 6420 616e 6420 3372 6420 706f  , 2nd and 3rd po
+0000d870: 7369 7469 6f6e 206f 6620 636f 646f 6e29  sition of codon)
+0000d880: 2069 6e73 7465 6164 206f 6620 7573 696e   instead of usin
+0000d890: 6720 7468 6520 636f 646f 6e20 6173 2061  g the codon as a
+0000d8a0: 2077 686f 6c65 2e20 5468 6520 6f70 7469   whole. The opti
+0000d8b0: 6f6e 2060 2d2d 6161 6d6f 6465 6c60 2063  on `--aamodel` c
+0000d8c0: 616e 2062 6520 7365 7420 746f 2064 6574  an be set to det
+0000d8d0: 6572 6d69 6e65 2074 6865 2061 6d69 6e6f  ermine the amino
+0000d8e0: 2061 6369 6420 6d6f 6465 6c20 6170 706c   acid model appl
+0000d8f0: 6965 6420 696e 2060 6d63 6d63 7472 6565  ied in `mcmctree
+0000d900: 6020 616e 616c 7973 6973 2e20 5468 6520  ` analysis. The 
+0000d910: 6f70 7469 6f6e 2060 2d64 7360 2063 616e  option `-ds` can
+0000d920: 2062 6520 7573 6564 2074 6f20 7365 7420   be used to set 
+0000d930: 7061 7261 6d65 7465 7273 2066 6f72 2074  parameters for t
+0000d940: 6865 206d 6f6c 6563 756c 6172 2064 6174  he molecular dat
+0000d950: 696e 6720 7072 6f67 7261 6d2e 2054 6865  ing program. The
+0000d960: 2070 6172 616c 6c65 6c20 7468 7265 6164   parallel thread
+0000d970: 7320 6865 7265 2061 7265 2074 6f20 7061  s here are to pa
+0000d980: 7261 6c6c 656c 697a 6520 7468 6520 616e  rallelize the an
+0000d990: 616c 7973 6973 2066 6f72 2065 6163 6820  alysis for each 
+0000d9a0: 6765 6e65 2066 616d 696c 7920 616e 6420  gene family and 
+0000d9b0: 7468 7573 2073 7567 6765 7374 6564 2074  thus suggested t
+0000d9c0: 6f20 6265 2073 6574 2061 7320 6d75 6368  o be set as much
+0000d9d0: 2061 7320 7468 6520 6e75 6d62 6572 206f   as the number o
+0000d9e0: 6620 6765 6e65 2066 616d 696c 6965 732e  f gene families.
+0000d9f0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+0000da00: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+0000da10: 2062 6520 7769 7468 2074 6865 2063 6f6d   be with the com
+0000da20: 6d61 6e64 2062 656c 6f77 2e2a 2a0a 0a60  mand below.**..`
+0000da30: 6060 0a77 6764 2066 6f63 7573 2066 616d  ``.wgd focus fam
+0000da40: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000da50: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000da60: 6365 3320 2d64 206d 636d 6374 7265 6520  ce3 -d mcmctree 
+0000da70: 2d73 7020 7370 6461 7461 0a77 6764 2066  -sp spdata.wgd f
+0000da80: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
+0000da90: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000daa0: 3220 7365 7175 656e 6365 3320 2d64 2072  2 sequence3 -d r
+0000dab0: 3873 202d 7370 2073 7064 6174 6120 2d2d  8s -sp spdata --
+0000dac0: 6e73 6974 6573 206e 7369 7465 696e 666f  nsites nsiteinfo
+0000dad0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+0000dae0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000daf0: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000db00: 3320 2d64 2062 6561 7374 202d 2d66 6f73  3 -d beast --fos
+0000db10: 7369 6c20 666f 7373 696c 696e 666f 202d  sil fossilinfo -
+0000db20: 2d72 6f6f 7468 6569 6768 7420 726f 6f74  -rootheight root
+0000db30: 6865 6967 6874 696e 666f 202d 2d63 6861  heightinfo --cha
+0000db40: 696e 7365 7420 6368 6169 6e73 6574 696e  inset chainsetin
+0000db50: 666f 202d 2d62 6561 7374 6c67 6a61 7220  fo --beastlgjar 
+0000db60: 2450 4154 480a 6060 600a 0a23 2323 2077  $PATH.```..### w
+0000db70: 6764 206b 7364 0a0a 2a2a 5468 6520 636f  gd ksd..**The co
+0000db80: 6e73 7472 7563 7469 6f6e 206f 6620 7768  nstruction of wh
+0000db90: 6f6c 6520 7061 7261 6e6f 6d65 202a 4b2a  ole paranome *K*
+0000dba0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000dbb0: 2064 6973 7472 6962 7574 696f 6e2a 2a0a   distribution**.
+0000dbc0: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
+0000dbd0: 6c69 6573 2073 6571 7565 6e63 6520 282d  lies sequence (-
+0000dbe0: 6f20 7767 645f 6b73 6420 2d74 2077 6764  o wgd_ksd -t wgd
+0000dbf0: 5f6b 7364 5f74 6d70 202d 2d6e 7468 7265  _ksd_tmp --nthre
+0000dc00: 6164 7320 3420 2d2d 746f 5f73 746f 7020  ads 4 --to_stop 
+0000dc10: 2d2d 6364 7320 2d2d 7061 6972 7769 7365  --cds --pairwise
+0000dc20: 202d 2d73 7472 6970 5f67 6170 7320 2d2d   --strip_gaps --
+0000dc30: 616c 6967 6e65 7220 6d61 6666 7420 2d2d  aligner mafft --
+0000dc40: 7472 6565 5f6d 6574 686f 6420 6661 7374  tree_method fast
+0000dc50: 7472 6565 202d 2d6e 6f64 655f 6176 6572  tree --node_aver
+0000dc60: 6167 6529 0a60 6060 0a0a 5468 6520 7072  age).```..The pr
+0000dc70: 6f67 7261 6d20 6077 6764 206b 7364 602c  ogram `wgd ksd`,
+0000dc80: 2061 7320 696d 7069 6564 2062 7920 6974   as impied by it
+0000dc90: 7320 6e61 6d65 2c20 6973 2066 6f72 2074  s name, is for t
+0000dca0: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+0000dcb0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
+0000dcc0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000dcd0: 696f 6e2e 2045 7863 6570 7420 666f 7220  ion. Except for 
+0000dce0: 7468 6520 6166 6f72 656d 656e 7469 6f6e  the aforemention
+0000dcf0: 6564 2070 6172 616d 6574 6572 7320 7375  ed parameters su
+0000dd00: 6368 2061 7320 602d 2d74 6f5f 7374 6f70  ch as `--to_stop
+0000dd10: 6020 616e 6420 602d 2d63 6473 602c 2074  ` and `--cds`, t
+0000dd20: 6865 7265 2061 7265 2073 6f6d 6520 696d  here are some im
+0000dd30: 706f 7274 616e 7420 7061 7261 6d65 7465  portant paramete
+0000dd40: 7273 2074 6861 7420 6861 7665 2063 7275  rs that have cru
+0000dd50: 6369 616c 2069 6d70 6163 7420 6f6e 2074  cial impact on t
+0000dd60: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000dd70: 623e 2065 7374 696d 6174 696f 6e2e 2054  b> estimation. T
+0000dd80: 6865 206f 7074 696f 6e20 602d 2d70 6169  he option `--pai
+0000dd90: 7277 6973 6560 2069 7320 6120 7665 7279  rwise` is a very
+0000dda0: 2069 6d70 6f72 7461 6e74 2070 6172 616d   important param
+0000ddb0: 6574 6572 2066 6f72 2074 6865 202a 4b2a  eter for the *K*
+0000ddc0: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+0000ddd0: 696d 6174 696f 6e2c 2077 6974 6820 7768  imation, with wh
+0000dde0: 6963 6820 7468 6520 6043 4f44 454d 4c60  ich the `CODEML`
+0000ddf0: 2077 696c 6c20 6361 6c63 756c 6174 6520   will calculate 
+0000de00: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+0000de10: 7562 3e20 666f 7220 6561 6368 2067 656e  ub> for each gen
+0000de20: 6520 7061 6972 2073 6570 6172 6174 656c  e pair separatel
+0000de30: 7920 6261 7365 6420 6f6e 2074 6865 2061  y based on the a
+0000de40: 6c69 676e 6d65 6e74 206f 6620 6f6e 6c79  lignment of only
+0000de50: 2074 6865 7365 2074 776f 2067 656e 6573   these two genes
+0000de60: 2069 6e73 7465 6164 206f 6620 7468 6520   instead of the 
+0000de70: 7768 6f6c 6520 616c 6967 6e6d 656e 7420  whole alignment 
+0000de80: 6f66 2074 6865 2066 616d 696c 792c 2073  of the family, s
+0000de90: 7563 6820 7468 6174 206c 6573 7320 6761  uch that less ga
+0000dea0: 7073 2061 7265 2065 7870 6563 7465 6420  ps are expected 
+0000deb0: 616e 6420 7468 7573 2074 6865 2061 6c69  and thus the ali
+0000dec0: 676e 6d65 6e74 2069 6e20 7468 6520 636f  gnment in the co
+0000ded0: 6e73 6964 6572 6174 696f 6e20 6f66 2060  nsideration of `
+0000dee0: 434f 4445 4d4c 6020 7769 6c6c 2062 6520  CODEML` will be 
+0000def0: 6c6f 6e67 6572 2028 6265 6361 7573 6520  longer (because 
+0000df00: 6043 4f44 454d 4c60 2077 696c 6c20 6175  `CODEML` will au
+0000df10: 746f 6d61 7465 6c79 2073 6b69 7020 6576  tomately skip ev
+0000df20: 6572 7920 636f 6c75 6d6e 2077 6974 6820  ery column with 
+0000df30: 6761 702c 2072 6567 6172 646c 6573 7320  gap, regardless 
+0000df40: 6f66 2077 6865 7468 6572 2069 7427 7320  of whether it's 
+0000df50: 616e 206f 7665 7261 6c6c 206f 7220 7061  an overall or pa
+0000df60: 7274 6961 6c20 6761 7029 2c20 7769 7468  rtial gap), with
+0000df70: 6f75 7420 7768 6963 6820 7468 6520 6043  out which the `C
+0000df80: 4f44 454d 4c60 2077 696c 6c20 6361 6c63  ODEML` will calc
+0000df90: 756c 6174 6520 7468 6520 2a4b 2a3c 7375  ulate the *K*<su
+0000dfa0: 623e 533c 2f73 7562 3e20 6261 7365 6420  b>S</sub> based 
+0000dfb0: 6f6e 2074 6865 2077 686f 6c65 2061 6c69  on the whole ali
+0000dfc0: 676e 6d65 6e74 206f 6620 7468 6520 6661  gnment of the fa
+0000dfd0: 6d69 6c79 2c20 7768 6963 6820 6d69 6768  mily, which migh
+0000dfe0: 7420 6861 7665 206e 6f20 2a4b 2a3c 7375  t have no *K*<su
+0000dff0: 623e 533c 2f73 7562 3e20 7265 7375 6c74  b>S</sub> result
+0000e000: 2061 7420 616c 6c20 6966 2074 6865 2073   at all if the s
+0000e010: 7472 6970 7065 6420 616c 6967 6e6d 656e  tripped alignmen
+0000e020: 7420 6c65 6e67 7468 2028 7265 6d6f 7669  t length (removi
+0000e030: 6e67 2061 6c6c 2067 6170 2d63 6f6e 7461  ng all gap-conta
+0000e040: 696e 696e 6720 636f 6c75 6d6e 7329 2077  ining columns) w
+0000e050: 6173 207a 6572 6f2c 2061 2063 6175 7365  as zero, a cause
+0000e060: 206f 6620 6469 6666 6572 656e 7420 6e75   of different nu
+0000e070: 6d62 6572 206f 6620 2a4b 2a3c 7375 623e  mber of *K*<sub>
+0000e080: 533c 2f73 7562 3e20 6573 7469 6d61 7465  S</sub> estimate
+0000e090: 7320 6265 7477 6565 6e20 2270 6169 7277  s between "pairw
+0000e0a0: 6973 6520 6d6f 6465 2220 616e 6420 226e  ise mode" and "n
+0000e0b0: 6f6e 2d70 6169 7277 6973 6520 6d6f 6465  on-pairwise mode
+0000e0c0: 222e 2049 7427 7320 6469 6666 6963 756c  ". It's difficul
+0000e0d0: 7420 746f 2073 6179 2077 6869 6368 206d  t to say which m
+0000e0e0: 6f64 6520 6973 206d 6f72 6520 6964 6561  ode is more idea
+0000e0f0: 6c2c 2061 6c74 686f 7567 6820 7468 6520  l, although the 
+0000e100: 226e 6f6e 2d70 6169 7277 6973 6520 6d6f  "non-pairwise mo
+0000e110: 6465 2220 2864 6566 6175 6c74 2073 6574  de" (default set
+0000e120: 7469 6e67 2920 7768 6963 6820 7275 6e73  ting) which runs
+0000e130: 206f 6e20 7468 6520 7768 6f6c 6520 616c   on the whole al
+0000e140: 6967 6e6d 656e 7420 696e 7374 6561 6420  ignment instead 
+0000e150: 6f66 2061 206c 6f63 616c 2061 6c69 676e  of a local align
+0000e160: 6d65 6e74 2c20 6d69 6768 7420 6265 206d  ment, might be m
+0000e170: 6f72 6520 6269 6f6c 6f67 6963 616c 6c79  ore biologically
+0000e180: 2063 6f6e 7365 7276 6564 2069 6e20 7468   conserved in th
+0000e190: 6174 2069 7420 6173 7375 7265 7320 7468  at it assures th
+0000e1a0: 6520 6576 6f6c 7574 696f 6e20 6f66 2065  e evolution of e
+0000e1b0: 6163 6820 636f 6c75 6d6e 2074 6f20 6265  ach column to be
+0000e1c0: 2073 7461 7274 6564 2066 726f 6d20 7468   started from th
+0000e1d0: 6520 726f 6f74 206f 6620 7468 6520 6661  e root of the fa
+0000e1e0: 6d69 6c79 2061 6e64 2061 6c6c 2074 6865  mily and all the
+0000e1f0: 2067 656e 6520 6475 706c 6963 6174 6573   gene duplicates
+0000e200: 2061 7265 2074 616b 656e 2069 6e74 6f20   are taken into 
+0000e210: 6163 636f 756e 7420 696e 2074 6865 202a  account in the *
+0000e220: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+0000e230: 7374 696d 6174 696f 6e20 7072 6f63 6573  stimation proces
+0000e240: 732e 2054 6865 206f 7074 696f 6e20 602d  s. The option `-
+0000e250: 2d73 7472 6970 5f67 6170 7360 2063 616e  -strip_gaps` can
+0000e260: 2072 656d 6f76 6520 616c 6c20 7468 6520   remove all the 
+0000e270: 6761 702d 636f 6e74 6169 6e69 6e67 2063  gap-containing c
+0000e280: 6f6c 756d 6e73 2c20 7769 7468 206f 7220  olumns, with or 
+0000e290: 7769 7468 6f75 7420 7768 6963 6820 7468  without which th
+0000e2a0: 6520 7265 7375 6c74 206f 6620 226e 6f6e  e result of "non
+0000e2b0: 2d70 6169 7277 6973 6520 6d6f 6465 2220  -pairwise mode" 
+0000e2c0: 776f 6e27 7420 6265 2061 6666 6563 7465  won't be affecte
+0000e2d0: 642c 2077 6869 6c65 2077 6974 6820 7768  d, while with wh
+0000e2e0: 6963 6820 7468 6520 7265 7375 6c74 206f  ich the result o
+0000e2f0: 6620 2270 6169 7277 6973 6520 6d6f 6465  f "pairwise mode
+0000e300: 2220 7769 6c6c 2062 6520 616c 7465 7265  " will be altere
+0000e310: 642e 2054 6865 206f 7074 696f 6e20 602d  d. The option `-
+0000e320: 2d61 6c69 676e 6572 6020 616e 6420 602d  -aligner` and `-
+0000e330: 2d61 6c6e 5f6f 7074 696f 6e73 6020 7768  -aln_options` wh
+0000e340: 6963 6820 6465 6369 6465 2077 6869 6368  ich decide which
+0000e350: 2061 6c69 676e 6d65 6e74 2070 726f 6772   alignment progr
+0000e360: 616d 2074 6f20 6265 2075 7365 6420 616e  am to be used an
+0000e370: 6420 7768 6963 6820 7061 7261 6d65 7465  d which paramete
+0000e380: 7220 746f 2062 6520 7365 742c 2077 696c  r to be set, wil
+0000e390: 6c20 6861 7665 2069 6d70 6163 7420 6f6e  l have impact on
+0000e3a0: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0000e3b0: 7375 623e 2072 6573 756c 7473 2c20 6e6f  sub> results, no
+0000e3c0: 7465 6420 7468 6174 2074 6865 2064 6566  ted that the def
+0000e3d0: 6175 6c74 2070 726f 6772 616d 2069 7320  ault program is 
+0000e3e0: 606d 6166 6674 6020 616e 6420 7468 6520  `mafft` and the 
+0000e3f0: 7061 7261 6d65 7465 7220 6973 2060 2d2d  parameter is `--
+0000e400: 6175 746f 602e 2054 6865 206f 7074 696f  auto`. The optio
+0000e410: 6e20 602d 2d74 7265 655f 6d65 7468 6f64  n `--tree_method
+0000e420: 6020 616e 6420 602d 2d74 7265 655f 6f70  ` and `--tree_op
+0000e430: 7469 6f6e 7360 2064 6563 6964 6520 7768  tions` decide wh
+0000e440: 6963 6820 6765 6e65 2074 7265 6520 696e  ich gene tree in
+0000e450: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+0000e460: 746f 2062 6520 7573 6564 2061 6e64 2077  to be used and w
+0000e470: 6869 6368 2070 6172 616d 6574 6572 2074  hich parameter t
+0000e480: 6f20 6265 2073 6574 2c20 776f 6e27 7420  o be set, won't 
+0000e490: 6166 6665 6374 2074 6865 202a 4b2a 3c73  affect the *K*<s
+0000e4a0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
+0000e4b0: 6174 696f 6e20 6974 7365 6c66 2062 7574  ation itself but
+0000e4c0: 2074 6865 2072 6573 756c 7420 6f66 2064   the result of d
+0000e4d0: 652d 7265 6475 6e64 616e 6379 2c20 6e6f  e-redundancy, no
+0000e4e0: 7465 6420 7468 6174 2077 6520 696d 706c  ted that we impl
+0000e4f0: 656d 656e 7465 6420 6120 6275 696c 742d  emented a built-
+0000e500: 696e 2067 656e 6520 7472 6565 2069 6e66  in gene tree inf
+0000e510: 6572 656e 6365 206d 6574 686f 6420 6261  erence method ba
+0000e520: 7365 6420 6f6e 2074 6865 2041 7665 7261  sed on the Avera
+0000e530: 6765 204c 696e 6b61 6765 2043 6c75 7374  ge Linkage Clust
+0000e540: 6572 696e 6720 2841 4c43 2920 2874 6875  ering (ALC) (thu
+0000e550: 7320 6120 6469 7374 616e 6365 2d62 6173  s a distance-bas
+0000e560: 6564 2074 7265 6529 2077 6974 6820 7468  ed tree) with th
+0000e570: 6520 602d 2d74 7265 655f 6d65 7468 6f64  e `--tree_method
+0000e580: 6020 7365 7420 6173 2022 636c 7573 7465  ` set as "cluste
+0000e590: 7222 2e20 5477 6f20 6d65 7468 6f64 7320  r". Two methods 
+0000e5a0: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
+0000e5b0: 2061 7265 2069 6d70 6c65 6d65 6e74 6564   are implemented
+0000e5c0: 2069 6e20 6077 6764 2076 3260 2c20 6e61   in `wgd v2`, na
+0000e5d0: 6d65 6c79 206e 6f64 652d 7765 6967 6874  mely node-weight
+0000e5e0: 6564 2061 6e64 206e 6f64 652d 6176 6572  ed and node-aver
+0000e5f0: 6167 6564 206d 6574 686f 6473 2e20 5468  aged methods. Th
+0000e600: 6520 6e6f 6465 2d77 6569 6768 7465 6420  e node-weighted 
+0000e610: 6d65 7468 6f64 2061 6368 6965 7665 7320  method achieves 
+0000e620: 7468 6520 6465 2d72 6564 756e 6461 6e63  the de-redundanc
+0000e630: 7920 7669 6120 7765 6967 6869 6e67 2074  y via weighing t
+0000e640: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000e650: 623e 2076 616c 7565 2061 7373 6f63 6961  b> value associa
+0000e660: 7465 6420 7769 7468 2065 6163 6820 6765  ted with each ge
+0000e670: 6e65 2070 6169 7220 7375 6368 2074 6861  ne pair such tha
+0000e680: 7420 7468 6520 7765 6967 6874 206f 6620  t the weight of 
+0000e690: 6120 7369 6e67 6c65 2064 7570 6c69 6361  a single duplica
+0000e6a0: 7469 6f6e 2065 7665 6e74 2073 756d 7320  tion event sums 
+0000e6b0: 7570 2074 6f20 3120 286e 6f74 6564 2074  up to 1 (noted t
+0000e6c0: 6861 7420 7468 6520 6e75 6d62 6572 206f  hat the number o
+0000e6d0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+0000e6e0: 3e20 6573 7469 6d61 7469 6f6e 2072 656d  > estimation rem
+0000e6f0: 6169 6e73 2074 6865 2073 616d 6529 2077  ains the same) w
+0000e700: 6869 6c65 2074 6865 206e 6f64 652d 6176  hile the node-av
+0000e710: 6572 6167 6564 206d 6574 686f 6420 7265  eraged method re
+0000e720: 616c 697a 6573 2074 6865 2064 652d 7265  alizes the de-re
+0000e730: 6475 6e64 616e 6379 2076 6961 2063 616c  dundancy via cal
+0000e740: 6375 6c61 7469 6e67 2070 6572 2067 656e  culating per gen
+0000e750: 6520 7472 6565 206e 6f64 6520 6f6e 6520  e tree node one 
+0000e760: 6176 6572 6167 6564 202a 4b2a 3c73 7562  averaged *K*<sub
+0000e770: 3e53 3c2f 7375 623e 2076 616c 7565 2074  >S</sub> value t
+0000e780: 6f20 7265 7072 6573 656e 7420 7468 6520  o represent the 
+0000e790: 6167 6520 6f66 2065 6163 6820 6475 706c  age of each dupl
+0000e7a0: 6963 6174 696f 6e20 6576 656e 742e 2054  ication event. T
+0000e7b0: 6865 206f 7074 696f 6e20 602d 2d6e 6f64  he option `--nod
+0000e7c0: 655f 6176 6572 6167 6560 2063 616e 2062  e_average` can b
+0000e7d0: 6520 7365 7420 746f 2063 686f 6f73 6520  e set to choose 
+0000e7e0: 7468 6520 6e6f 6465 2d61 7665 7261 6765  the node-average
+0000e7f0: 6420 7761 7920 6f66 2064 652d 7265 6475  d way of de-redu
+0000e800: 6e64 616e 6379 2e20 4469 6666 6572 656e  ndancy. Differen
+0000e810: 7420 6d65 7468 6f64 7320 6f66 2064 652d  t methods of de-
+0000e820: 7265 6475 6e64 616e 6379 2077 696c 6c20  redundancy will 
+0000e830: 6861 7665 2069 6d70 6163 7420 6f6e 2074  have impact on t
+0000e840: 6865 2064 6574 6563 7469 6f6e 206f 6620  he detection of 
+0000e850: 5747 4420 7369 676e 616c 732c 2077 6869  WGD signals, whi
+0000e860: 6368 2068 6173 2062 6565 6e20 696e 7665  ch has been inve
+0000e870: 7374 6967 6174 6564 2069 6e20 7468 6973  stigated in this
+0000e880: 205b 6c69 7465 7261 7475 7265 5d28 6874   [literature](ht
+0000e890: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
+0000e8a0: 2e31 3039 332f 6762 652f 6576 7932 3030  .1093/gbe/evy200
+0000e8b0: 292e 2054 6865 2070 6172 616c 6c65 6c20  ). The parallel 
+0000e8c0: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
+0000e8d0: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
+0000e8e0: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
+0000e8f0: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
+0000e900: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
+0000e910: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
+0000e920: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
+0000e930: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
+0000e940: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
+0000e950: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000e960: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+0000e970: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
+0000e980: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000e990: 6965 7320 7365 7175 656e 6365 0a60 6060  ies sequence.```
+0000e9a0: 0a0a 2a2a 5468 6520 636f 6e73 7472 7563  ..**The construc
+0000e9b0: 7469 6f6e 206f 6620 6f72 7468 6f6c 6f67  tion of ortholog
+0000e9c0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+0000e9d0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000e9e0: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
+0000e9f0: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
+0000ea00: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000ea10: 282d 2d72 6577 6569 6768 7429 0a60 6060  (--reweight).```
+0000ea20: 0a0a 4672 6f6d 2070 6172 616c 6f67 6f75  ..From paralogou
+0000ea30: 7320 746f 206f 7274 686f 6c6f 676f 7573  s to orthologous
+0000ea40: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000ea50: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+0000ea60: 6e2c 2075 7365 7273 206f 6e6c 7920 6e65  n, users only ne
+0000ea70: 6564 2074 6f20 7072 6f76 6964 6520 6d6f  ed to provide mo
+0000ea80: 7265 2063 6473 2066 696c 6573 2e20 4e6f  re cds files. No
+0000ea90: 7465 2074 6861 7420 7769 7468 206f 7274  te that with ort
+0000eaa0: 686f 6c6f 676f 7573 2067 656e 6520 6661  hologous gene fa
+0000eab0: 6d69 6c69 6573 2074 6865 2077 6569 6768  milies the weigh
+0000eac0: 7469 6e67 206d 6574 686f 6420 6361 6e20  ting method can 
+0000ead0: 6265 2073 6574 2074 6f20 6265 2063 616c  be set to be cal
+0000eae0: 6375 6c61 7465 6420 7065 7220 7370 6563  culated per spec
+0000eaf0: 6965 7320 7061 6972 2069 6e73 7465 6164  ies pair instead
+0000eb00: 206f 6620 636f 6e73 6964 6572 696e 6720   of considering 
+0000eb10: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000eb20: 2062 6563 6175 7365 2077 6865 6e20 706c   because when pl
+0000eb30: 6f74 7469 6e67 206f 7274 686f 6c6f 676f  otting orthologo
+0000eb40: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+0000eb50: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000eb60: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
+0000eb70: 7370 6563 6965 7320 7468 6520 7765 6967  species the weig
+0000eb80: 6874 2063 616c 6375 6c61 7465 6420 6672  ht calculated fr
+0000eb90: 6f6d 2074 6869 7320 7370 6563 6966 6963  om this specific
+0000eba0: 2073 7065 6369 6573 2070 6169 7220 7368   species pair sh
+0000ebb0: 6f75 6c64 2062 6520 636f 6e73 6572 7665  ould be conserve
+0000ebc0: 6420 7768 696c 6520 7468 6520 6f6e 6520  d while the one 
+0000ebd0: 6361 6c63 756c 6174 6564 2066 726f 6d20  calculated from 
+0000ebe0: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000ebf0: 2077 696c 6c20 7661 7279 2077 6974 6820   will vary with 
+0000ec00: 7468 6520 6e75 6d62 6572 206f 6620 7370  the number of sp
+0000ec10: 6563 6965 732e 2054 6f20 696e 6974 6961  ecies. To initia
+0000ec20: 7465 2074 6865 2077 6569 6768 7469 6e67  te the weighting
+0000ec30: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
+0000ec40: 722c 2074 6865 206f 7074 696f 6e20 602d  r, the option `-
+0000ec50: 2d72 6577 6569 6768 7460 2063 616e 2062  -reweight` can b
+0000ec60: 6520 7365 742e 0a0a 2a2a 4120 7375 6767  e set...**A sugg
+0000ec70: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000ec80: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+0000ec90: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
+0000eca0: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000ecb0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000ecc0: 7175 656e 6365 320a 6060 600a 0a2a 2a54  quence2.```..**T
+0000ecd0: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+0000ece0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
+0000ecf0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000ed00: 696f 6e20 7769 7468 2072 6174 6520 636f  ion with rate co
+0000ed10: 7272 6563 7469 6f6e 2a2a 0a60 6060 0a77  rrection**.```.w
+0000ed20: 6764 206b 7364 2066 616d 696c 6965 7320  gd ksd families 
+0000ed30: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000ed40: 6365 3220 7365 7175 656e 6365 3320 2d2d  ce2 sequence3 --
+0000ed50: 666f 6375 7332 616c 6c20 7365 7175 656e  focus2all sequen
+0000ed60: 6365 3120 2d73 7020 7370 6461 7461 0a60  ce1 -sp spdata.`
+0000ed70: 6060 0a0a 496e 7370 6972 6564 2062 7920  ``..Inspired by 
+0000ed80: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+0000ed90: 696f 6e20 616c 676f 7269 7468 6d20 696e  ion algorithm in
+0000eda0: 205b 6b73 7261 7465 735d 2868 7474 7073   [ksrates](https
+0000edb0: 3a2f 2f67 6974 6875 622e 636f 6d2f 5649  ://github.com/VI
+0000edc0: 422d 5053 422f 6b73 7261 7465 7329 2c20  B-PSB/ksrates), 
+0000edd0: 7765 2069 6d70 6c65 6d65 6e74 6564 2074  we implemented t
+0000ede0: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+0000edf0: 6f6e 2061 6e61 6c79 7369 7320 616c 736f  on analysis also
+0000ee00: 2069 6e20 6077 6764 2076 3260 2c20 7768   in `wgd v2`, wh
+0000ee10: 6963 6820 6973 206d 6f73 746c 7920 7468  ich is mostly th
+0000ee20: 6520 7361 6d65 2061 7320 5b6b 7372 6174  e same as [ksrat
+0000ee30: 6573 5d28 6874 7470 733a 2f2f 6769 7468  es](https://gith
+0000ee40: 7562 2e63 6f6d 2f56 4942 2d50 5342 2f6b  ub.com/VIB-PSB/k
+0000ee50: 7372 6174 6573 2920 6275 7420 6469 6666  srates) but diff
+0000ee60: 6572 7320 696e 2074 6865 2063 616c 6375  ers in the calcu
+0000ee70: 6c61 7469 6f6e 206f 6620 7468 6520 7374  lation of the st
+0000ee80: 616e 6461 7264 2064 6576 6961 7469 6f6e  andard deviation
+0000ee90: 206f 6620 7265 7363 616c 6564 202a 4b2a   of rescaled *K*
+0000eea0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000eeb0: 732e 2054 6f20 7065 7266 6f72 6d20 7468  s. To perform th
+0000eec0: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+0000eed0: 6e20 616e 616c 7973 6973 2c20 7573 6572  n analysis, user
+0000eee0: 7320 6361 6e20 7573 6520 626f 7468 2074  s can use both t
+0000eef0: 6865 2060 7767 6420 6b73 6460 2061 6e64  he `wgd ksd` and
+0000ef00: 2060 7767 6420 7669 7a60 2070 726f 6772   `wgd viz` progr
+0000ef10: 616d 2e20 466f 7220 7468 6520 6077 6764  am. For the `wgd
+0000ef20: 206b 7364 6020 7072 6f67 7261 6d2c 2075   ksd` program, u
+0000ef30: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+0000ef40: 7669 6465 2061 2073 7065 6369 6573 2074  vide a species t
+0000ef50: 7265 6520 7669 6120 7468 6520 6f70 7469  ree via the opti
+0000ef60: 6f6e 2060 2d2d 7370 6563 6965 7374 7265  on `--speciestre
+0000ef70: 6560 206f 6e20 7768 6963 6820 7261 7465  e` on which rate
+0000ef80: 2063 6f72 7265 6374 696f 6e20 6361 6e20   correction can 
+0000ef90: 6265 2063 6f6e 6475 6374 6564 2e20 4e6f  be conducted. No
+0000efa0: 7465 2074 6861 7420 756e 6e65 6365 7373  te that unnecess
+0000efb0: 6172 7920 6272 6163 6b65 7473 206d 6967  ary brackets mig
+0000efc0: 6874 206c 6561 6420 746f 2075 6e65 7870  ht lead to unexp
+0000efd0: 6563 7465 6420 6572 726f 7273 2c20 666f  ected errors, fo
+0000efe0: 7220 696e 7374 616e 6365 2061 2074 7265  r instance a tre
+0000eff0: 6520 6028 412c 2842 2c43 2929 3b60 2073  e `(A,(B,C));` s
+0000f000: 686f 756c 6420 6e6f 7420 6265 2072 6570  hould not be rep
+0000f010: 7265 7365 6e74 6564 2061 7320 6028 412c  resented as `(A,
+0000f020: 2828 422c 4329 2929 3b60 2e20 5468 6520  ((B,C)));`. The 
+0000f030: 7365 7420 6f66 2073 7065 6369 6573 2070  set of species p
+0000f040: 6169 7273 2074 6f20 6265 2073 686f 776e  airs to be shown
+0000f050: 2069 7320 666c 6578 6962 6c65 2074 6861   is flexible tha
+0000f060: 7420 7468 6520 6d6f 7374 2063 6f6e 7665  t the most conve
+0000f070: 6e69 656e 7420 6f70 7469 6f6e 2069 7320  nient option is 
+0000f080: 602d 2d66 6f63 7573 3261 6c6c 6020 7768  `--focus2all` wh
+0000f090: 6963 6820 7369 6d70 6c79 2073 686f 7773  ich simply shows
+0000f0a0: 2061 6c6c 2074 6865 2070 6f73 7369 626c   all the possibl
+0000f0b0: 6520 666f 6361 6c2d 7369 7374 6572 2073  e focal-sister s
+0000f0c0: 7065 6369 6573 2070 6169 7273 2c20 6f72  pecies pairs, or
+0000f0d0: 2075 7365 7273 2063 616e 206d 616e 7561   users can manua
+0000f0e0: 6c6c 7920 7365 7420 7468 6520 7370 6563  lly set the spec
+0000f0f0: 6965 7320 7061 6972 7320 7669 6120 7468  ies pairs via th
+0000f100: 6520 6f70 7469 6f6e 2060 2d2d 7370 6169  e option `--spai
+0000f110: 7260 2e20 4e6f 7465 2074 6861 7420 6966  r`. Note that if
+0000f120: 2074 6865 2073 7065 6369 6573 2070 6169   the species pai
+0000f130: 7273 2077 6572 6520 6d61 6e75 616c 6c79  rs were manually
+0000f140: 2073 6574 2c20 6974 2077 6f75 6c64 2062   set, it would b
+0000f150: 6520 6e65 6564 6564 2074 6f20 7365 7420  e needed to set 
+0000f160: 7468 6520 6f70 7469 6f6e 2060 2d2d 636c  the option `--cl
+0000f170: 6173 7369 6360 2e20 5468 6520 6f70 7469  assic`. The opti
+0000f180: 6f6e 2060 6f6e 6c79 726f 6f74 6f75 7460  on `onlyrootout`
+0000f190: 2063 616e 2062 6520 7365 7420 746f 206f   can be set to o
+0000f1a0: 6e6c 7920 636f 6e73 6964 6572 206f 7574  nly consider out
+0000f1b0: 6772 6f75 7020 6174 2074 6865 2072 6f6f  group at the roo
+0000f1c0: 742c 2069 6e73 7465 6164 206f 6620 616c  t, instead of al
+0000f1d0: 6c20 7468 6520 706f 7373 6962 6c65 206f  l the possible o
+0000f1e0: 7574 6772 6f75 7073 2070 6572 2066 6f63  utgroups per foc
+0000f1f0: 616c 2d73 6973 7465 7220 7370 6563 6965  al-sister specie
+0000f200: 7320 7061 6972 2c20 7768 6963 6820 6861  s pair, which ha
+0000f210: 7320 696d 7061 6374 206f 6e20 7468 6520  s impact on the 
+0000f220: 6669 6e61 6c20 636f 7272 6563 7465 6420  final corrected 
+0000f230: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000f240: 6167 6573 2e20 536f 6d65 206f 7468 6572  ages. Some other
+0000f250: 206f 7074 696f 6e73 2077 6869 6368 2068   options which h
+0000f260: 6176 6520 6e6f 2069 6d70 6163 7420 6f6e  ave no impact on
+0000f270: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+0000f280: 7469 6f6e 2062 7574 2061 6464 206d 6f72  tion but add mor
+0000f290: 6520 6c61 7965 7273 206f 7220 6368 616e  e layers or chan
+0000f2a0: 6765 2074 6865 2061 7070 6561 7261 6e63  ge the appearanc
+0000f2b0: 6520 6f6e 2074 6865 206d 6978 6564 202a  e on the mixed *
+0000f2c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+0000f2d0: 6765 2064 6973 7472 6962 7574 696f 6e2c  ge distribution,
+0000f2e0: 2069 6e63 6c75 6465 2060 706c 6f74 6170   include `plotap
+0000f2f0: 676d 6d60 2061 6e64 2060 706c 6f74 656c  gmm` and `plotel
+0000f300: 6d6d 6020 6574 632e 2054 6865 206f 7074  mm` etc. The opt
+0000f310: 696f 6e20 6070 6c6f 7461 7067 6d6d 6020  ion `plotapgmm` 
+0000f320: 6361 6e20 6265 2073 6574 2074 6f20 6361  can be set to ca
+0000f330: 6c6c 2074 6865 2047 4d4d 2061 6e61 6c79  ll the GMM analy
+0000f340: 7369 7320 7570 6f6e 2074 6865 2061 6e63  sis upon the anc
+0000f350: 686f 7220 7061 6972 202a 4b2a 3c73 7562  hor pair *K*<sub
+0000f360: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+0000f370: 7472 6962 7574 696f 6e20 616e 6420 706c  tribution and pl
+0000f380: 6f74 2074 6865 2063 6c75 7374 6572 696e  ot the clusterin
+0000f390: 6720 7265 7375 6c74 2075 706f 6e20 7468  g result upon th
+0000f3a0: 6520 6d69 7865 6420 2a4b 2a3c 7375 623e  e mixed *K*<sub>
+0000f3b0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+0000f3c0: 7269 6275 7469 6f6e 2c20 7768 6963 6820  ribution, which 
+0000f3d0: 6861 7320 746f 2062 6520 636f 2d73 6574  has to be co-set
+0000f3e0: 2077 6974 6820 7468 6520 6f70 7469 6f6e   with the option
+0000f3f0: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
+0000f400: 6020 7072 6f76 6964 696e 6720 7468 6520  ` providing the 
+0000f410: 616e 6368 6f72 2070 6169 7273 2069 6e66  anchor pairs inf
+0000f420: 6f72 6d61 7469 6f6e 2e20 5468 6520 6f70  ormation. The op
+0000f430: 7469 6f6e 2060 2d2d 706c 6f74 656c 6d6d  tion `--plotelmm
+0000f440: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
+0000f450: 6361 6c6c 2074 6865 2045 4c4d 4d20 616e  call the ELMM an
+0000f460: 616c 7973 6973 2075 706f 6e20 7468 6520  alysis upon the 
+0000f470: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
+0000f480: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000f490: 6973 7472 6962 7574 696f 6e2e 0a0a 5468  istribution...Th
+0000f4a0: 6572 6520 6172 6520 3231 2063 6f6c 756d  ere are 21 colum
+0000f4b0: 6e73 2069 6e20 7468 6520 7265 7375 6c74  ns in the result
+0000f4c0: 2060 2a6b 732e 7473 7660 2066 696c 6520   `*ks.tsv` file 
+0000f4d0: 6265 7369 6465 7320 7468 6520 696e 6465  besides the inde
+0000f4e0: 7820 636f 6c75 6d6e 7320 6070 6169 7260  x columns `pair`
+0000f4f0: 2061 7320 7468 6520 756e 6971 7565 2069   as the unique i
+0000f500: 6465 6e74 6966 6965 7220 666f 7220 6561  dentifier for ea
+0000f510: 6368 2067 656e 6520 7061 6972 2e20 5468  ch gene pair. Th
+0000f520: 6520 604e 602c 2060 5360 2c20 6064 4e60  e `N`, `S`, `dN`
+0000f530: 2c20 6064 4e2f 6453 602c 2060 6453 602c  , `dN/dS`, `dS`,
+0000f540: 2060 6c60 2061 6e64 2060 7460 2061 7265   `l` and `t` are
+0000f550: 2066 726f 6d20 7468 6520 636f 6465 6d6c   from the codeml
+0000f560: 2072 6573 756c 7473 2c20 7265 7072 6573   results, repres
+0000f570: 656e 7469 6e67 2074 6865 204e 2065 7374  enting the N est
+0000f580: 696d 6174 652c 2074 6865 2053 2065 7374  imate, the S est
+0000f590: 696d 6174 652c 2074 6865 2064 4e20 6573  imate, the dN es
+0000f5a0: 7469 6d61 7465 2c20 7468 6520 644e 2f64  timate, the dN/d
+0000f5b0: 5320 286f 6d65 6761 2920 6573 7469 6d61  S (omega) estima
+0000f5c0: 7465 2c20 7468 6520 6453 2065 7374 696d  te, the dS estim
+0000f5d0: 6174 652c 2074 6865 206c 6f67 2d6c 696b  ate, the log-lik
+0000f5e0: 656c 6968 6f6f 6420 616e 6420 7468 6520  elihood and the 
+0000f5f0: 7420 6573 7469 6d61 7465 2c20 7265 7370  t estimate, resp
+0000f600: 6563 7469 7665 6c79 2e20 5468 6520 6061  ectively. The `a
+0000f610: 6c69 676e 6d65 6e74 636f 7665 7261 6765  lignmentcoverage
+0000f620: 602c 2060 616c 6967 6e6d 656e 7469 6465  `, `alignmentide
+0000f630: 6e74 6974 7960 2061 6e64 2060 616c 6967  ntity` and `alig
+0000f640: 6e6d 656e 746c 656e 6774 6860 2061 7265  nmentlength` are
+0000f650: 2074 6865 2069 6e66 6f72 6d61 7469 6f6e   the information
+0000f660: 2070 6572 7461 696e 696e 6720 746f 2074   pertaining to t
+0000f670: 6865 2061 6c69 676e 6d65 6e74 2066 6f72  he alignment for
+0000f680: 2065 6163 6820 6661 6d69 6c79 2c20 7265   each family, re
+0000f690: 7072 6573 656e 7469 6e67 2074 6865 2072  presenting the r
+0000f6a0: 6174 696f 206f 6620 7468 6520 7374 7269  atio of the stri
+0000f6b0: 7070 6564 2061 6c69 676e 6d65 6e74 206c  pped alignment l
+0000f6c0: 656e 6774 6820 636f 6d70 6172 6564 2074  ength compared t
+0000f6d0: 6f20 7468 6520 6675 6c6c 2061 6c69 676e  o the full align
+0000f6e0: 6d65 6e74 206c 656e 6774 682c 2074 6865  ment length, the
+0000f6f0: 2072 6174 696f 206f 6620 636f 6c75 6d6e   ratio of column
+0000f700: 7320 7769 7468 2069 6465 6e74 6963 616c  s with identical
+0000f710: 206e 7563 6c65 6f74 6964 6573 2063 6f6d   nucleotides com
+0000f720: 7061 7265 6420 746f 2074 6865 206f 7665  pared to the ove
+0000f730: 7261 6c6c 2063 6f6c 756d 6e73 206f 6620  rall columns of 
+0000f740: 7468 6520 7374 7269 7070 6564 2061 6c69  the stripped ali
+0000f750: 676e 6d65 6e74 2c20 616e 6420 7468 6520  gnment, and the 
+0000f760: 6c65 6e67 7468 206f 6620 7468 6520 6675  length of the fu
+0000f770: 6c6c 2061 6c69 676e 6d65 6e74 2c20 7265  ll alignment, re
+0000f780: 7370 6563 7469 7665 6c79 2e0a 2323 2320  spectively..### 
+0000f790: 7767 6420 6d69 780a 0a2a 2a54 6865 206d  wgd mix..**The m
+0000f7a0: 6978 7475 7265 206d 6f64 656c 2063 6c75  ixture model clu
+0000f7b0: 7374 6572 696e 6720 616e 616c 7973 6973  stering analysis
+0000f7c0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
+0000f7d0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000f7e0: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206d  tion**.```.wgd m
+0000f7f0: 6978 206b 7364 6174 6120 282d 2d6e 5f69  ix ksdata (--n_i
+0000f800: 6e69 7420 3230 3020 2d2d 6d61 785f 6974  nit 200 --max_it
+0000f810: 6572 2032 3030 202d 2d6b 735f 7261 6e67  er 200 --ks_rang
+0000f820: 6520 3020 3520 2d2d 6669 6c74 6572 7320  e 0 5 --filters 
+0000f830: 3330 3020 2d2d 6269 6e73 2035 3020 2d2d  300 --bins 50 --
+0000f840: 636f 6d70 6f6e 656e 7473 2031 2034 202d  components 1 4 -
+0000f850: 2d67 616d 6d61 2030 2e30 3031 290a 6060  -gamma 0.001).``
+0000f860: 600a 0a54 6869 7320 7061 7274 206f 6620  `..This part of 
+0000f870: 4761 7573 7369 616e 206d 6978 7475 7265  Gaussian mixture
+0000f880: 206d 6f64 656c 696e 6720 2847 4d4d 2920   modeling (GMM) 
+0000f890: 616e 616c 7973 6973 2069 7320 696e 6865  analysis is inhe
+0000f8a0: 7269 7465 6420 6672 6f6d 2074 6865 206f  rited from the o
+0000f8b0: 7269 6769 6e61 6c20 6077 6764 6020 7072  riginal `wgd` pr
+0000f8c0: 6f67 7261 6d2c 2062 7574 2077 7269 7465  ogram, but write
+0000f8d0: 7320 6164 6469 7469 6f6e 616c 6c79 2074  s additionally t
+0000f8e0: 6865 2070 726f 6261 6269 6c69 7479 206f  he probability o
+0000f8f0: 6620 6561 6368 202a 4b2a 3c73 7562 3e53  f each *K*<sub>S
+0000f900: 3c2f 7375 623e 2076 616c 7565 2069 6e74  </sub> value int
+0000f910: 6f20 7468 6520 6669 6e61 6c20 6461 7461  o the final data
+0000f920: 6672 616d 652e 2042 6173 6963 616c 6c79  frame. Basically
+0000f930: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000f940: 7072 6f76 6964 6520 7769 7468 2061 2028  provide with a (
+0000f950: 6e6f 726d 616c 6c79 2066 726f 6d20 7768  normally from wh
+0000f960: 6f6c 652d 7061 7261 6e6f 6d65 206f 7220  ole-paranome or 
+0000f970: 616e 6368 6f72 2d70 6169 7273 2920 2a4b  anchor-pairs) *K
+0000f980: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+0000f990: 7461 6669 6c65 2061 6e64 2074 6865 2047  tafile and the G
+0000f9a0: 4d4d 2061 6e61 6c79 7369 7320 7769 6c6c  MM analysis will
+0000f9b0: 2062 6520 636f 6e64 7563 7465 6420 7570   be conducted up
+0000f9c0: 6f6e 2074 6865 2064 6174 6166 696c 652e  on the datafile.
+0000f9d0: 2053 6f6d 6520 7061 7261 6d65 7465 7273   Some parameters
+0000f9e0: 2063 616e 2061 6666 6563 7420 7468 6520   can affect the 
+0000f9f0: 7265 7375 6c74 732c 2069 6e63 6c75 6469  results, includi
+0000fa00: 6e67 2060 2d2d 6e5f 696e 6974 602c 2077  ng `--n_init`, w
+0000fa10: 6869 6368 2073 6574 7320 7468 6520 6e75  hich sets the nu
+0000fa20: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
+0000fa30: 696e 6974 6961 6c69 7a61 7469 6f6e 7320  initializations 
+0000fa40: 2864 6566 6175 6c74 2032 3030 292c 2060  (default 200), `
+0000fa50: 2d2d 6d61 785f 6974 6572 602c 2077 6869  --max_iter`, whi
+0000fa60: 6368 2073 6574 7320 7468 6520 6d61 7869  ch sets the maxi
+0000fa70: 6d75 6d20 6e75 6d62 6572 206f 6620 6974  mum number of it
+0000fa80: 6572 6174 696f 6e73 2028 6465 6661 756c  erations (defaul
+0000fa90: 7420 3230 3029 2c20 602d 2d6d 6574 686f  t 200), `--metho
+0000faa0: 6460 2c20 7768 6963 6820 6465 7465 726d  d`, which determ
+0000fab0: 696e 6573 2077 6869 6368 2063 6c75 7374  ines which clust
+0000fac0: 6572 696e 6720 6d65 7468 6f64 2074 6f20  ering method to 
+0000fad0: 7573 6520 2864 6566 6175 6c74 2067 6d6d  use (default gmm
+0000fae0: 292c 2060 2d2d 6761 6d6d 6160 2c20 7768  ), `--gamma`, wh
+0000faf0: 6963 6820 7365 7473 2074 6865 2067 616d  ich sets the gam
+0000fb00: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
+0000fb10: 2074 6865 2062 676d 6d20 6d6f 6465 6c20   the bgmm model 
+0000fb20: 2864 6566 6175 6c74 2030 2e30 3031 292c  (default 0.001),
+0000fb30: 2060 2d2d 636f 6d70 6f6e 656e 7473 602c   `--components`,
+0000fb40: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+0000fb50: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
+0000fb60: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
+0000fb70: 7320 746f 2066 6974 2028 6465 6661 756c  s to fit (defaul
+0000fb80: 7420 3120 3429 2c20 616e 6420 7468 6520  t 1 4), and the 
+0000fb90: 6461 7461 2066 696c 7465 7269 6e67 2070  data filtering p
+0000fba0: 6172 616d 6574 6572 7320 602d 2d66 696c  arameters `--fil
+0000fbb0: 7465 7273 6020 7768 6963 6820 6669 6c74  ters` which filt
+0000fbc0: 6572 7320 6461 7461 2062 6173 6564 206f  ers data based o
+0000fbd0: 6e20 616c 6967 6e6d 656e 7420 6c65 6e67  n alignment leng
+0000fbe0: 7468 2c20 602d 2d6b 735f 7261 6e67 6560  th, `--ks_range`
+0000fbf0: 2077 6869 6368 2066 696c 7465 7273 2064   which filters d
+0000fc00: 6174 6120 6261 7365 6420 6f6e 202a 4b2a  ata based on *K*
+0000fc10: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
+0000fc20: 7565 7320 616e 6420 7468 6520 7061 7261  ues and the para
+0000fc30: 6d65 7465 7220 602d 2d62 696e 7360 2077  meter `--bins` w
+0000fc40: 6869 6368 2073 6574 7320 7468 6520 6e75  hich sets the nu
+0000fc50: 6d62 6572 206f 6620 6269 6e73 2069 6e20  mber of bins in 
+0000fc60: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000fc70: 6469 7374 7269 6275 7469 6f6e 2028 6465  distribution (de
+0000fc80: 6661 756c 7420 3530 292e 0a0a 2a2a 4120  fault 50)...**A 
+0000fc90: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+0000fca0: 6e67 2072 756e 2063 616e 2075 7365 2063  ng run can use c
+0000fcb0: 6f6d 6d61 6e64 2073 696d 706c 7920 6173  ommand simply as
+0000fcc0: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
+0000fcd0: 6420 6d69 7820 6b73 6461 7461 0a60 6060  d mix ksdata.```
+0000fce0: 0a0a 2323 2320 7767 6420 7065 616b 0a0a  ..### wgd peak..
+0000fcf0: 2a2a 5468 6520 7365 6172 6368 206f 6620  **The search of 
+0000fd00: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
+0000fd10: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
+0000fd20: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
+0000fd30: 6e67 2a2a 0a60 6060 0a77 6764 2070 6561  ng**.```.wgd pea
+0000fd40: 6b20 6b73 6461 7461 202d 6170 2061 7064  k ksdata -ap apd
+0000fd50: 6174 6120 2d73 6d20 736d 6461 7461 202d  ata -sm smdata -
+0000fd60: 6c65 206c 6564 6174 6120 2d6d 7020 6d70  le ledata -mp mp
+0000fd70: 6461 7461 202d 2d68 6575 7269 7374 6963  data --heuristic
+0000fd80: 2028 2d2d 616c 6967 6e66 696c 7465 7220   (--alignfilter 
+0000fd90: 302e 3020 3020 302e 3020 2d2d 6b73 7261  0.0 0 0.0 --ksra
+0000fda0: 6e67 6520 3020 3520 2d2d 6269 6e5f 7769  nge 0 5 --bin_wi
+0000fdb0: 6474 6820 302e 3120 2d2d 6775 6964 6520  dth 0.1 --guide 
+0000fdc0: 7365 676d 656e 7420 2d2d 7072 6f6d 696e  segment --promin
+0000fdd0: 656e 6365 5f63 7574 6f66 6620 302e 3120  ence_cutoff 0.1 
+0000fde0: 2d2d 7265 6c5f 6865 6967 6874 2030 2e34  --rel_height 0.4
+0000fdf0: 202d 2d63 6920 3935 202d 2d68 6472 2039   --ci 95 --hdr 9
+0000fe00: 3520 2d2d 6b73 6375 746f 6666 2035 290a  5 --kscutoff 5).
+0000fe10: 6060 600a 0a41 7320 6d65 6e74 696f 6e65  ```..As mentione
+0000fe20: 6420 7072 6576 696f 7573 6c79 2c20 6120  d previously, a 
+0000fe30: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
+0000fe40: 2061 6e64 2061 2063 6f6c 6c69 6e65 6172   and a collinear
+0000fe50: 2073 6567 6d65 6e74 732d 6775 6964 6564   segments-guided
+0000fe60: 2061 6e63 686f 7220 7061 6972 2063 6c75   anchor pair clu
+0000fe70: 7374 6572 696e 6720 666f 7220 7468 6520  stering for the 
+0000fe80: 7365 6172 6368 206f 6620 6372 6564 6961  search of credia
+0000fe90: 626c 6520 2a4b 2a3c 7375 623e 533c 2f73  ble *K*<sub>S</s
+0000fea0: 7562 3e20 7261 6e67 6520 7573 6564 2069  ub> range used i
+0000feb0: 6e20 5747 4420 6461 7469 6e67 2061 7265  n WGD dating are
+0000fec0: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
+0000fed0: 6077 6764 2076 3260 2e20 5573 6572 7320  `wgd v2`. Users 
+0000fee0: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
+0000fef0: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
+0000ff00: 732c 2073 6567 6d65 6e74 732c 206c 6973  s, segments, lis
+0000ff10: 7473 6567 6d65 6e74 732c 206d 756c 7469  tsegments, multi
+0000ff20: 706c 6963 6f6e 7061 6972 7320 6461 7461  pliconpairs data
+0000ff30: 6669 6c65 2066 726f 6d20 6069 2d61 6468  file from `i-adh
+0000ff40: 6f72 6560 2074 6f20 6163 6869 6576 6520  ore` to achieve 
+0000ff50: 7468 6520 636c 7573 7465 7269 6e67 2066  the clustering f
+0000ff60: 756e 6374 696f 6e2e 2053 6f6d 6520 7061  unction. Some pa
+0000ff70: 7261 6d65 7465 7273 2074 6861 7420 6361  rameters that ca
+0000ff80: 6e20 696d 7061 6374 2074 6865 2072 6573  n impact the res
+0000ff90: 756c 7473 2069 6e63 6c75 6465 2060 2d2d  ults include `--
+0000ffa0: 616c 6967 6e66 696c 7465 7260 2c20 7768  alignfilter`, wh
+0000ffb0: 6963 6820 6669 6c74 6572 7320 7468 6520  ich filters the 
+0000ffc0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
+0000ffd0: 6967 6e6d 656e 7420 6964 656e 7469 7479  ignment identity
+0000ffe0: 2c20 6c65 6e67 7468 2061 6e64 2063 6f76  , length and cov
+0000fff0: 6572 6167 652c 2060 2d2d 6b73 7261 6e67  erage, `--ksrang
+00010000: 6560 2c20 7768 6963 6820 7365 7473 2074  e`, which sets t
+00010010: 6865 2072 616e 6765 206f 6620 4b73 2074  he range of Ks t
+00010020: 6f20 6265 2061 6e61 6c79 7a65 642c 2060  o be analyzed, `
+00010030: 2d2d 6269 6e5f 7769 6474 6860 2c20 7768  --bin_width`, wh
+00010040: 6963 6820 7365 7473 2074 6865 2062 616e  ich sets the ban
+00010050: 6477 6964 7468 206f 6620 2a4b 2a3c 7375  dwidth of *K*<su
+00010060: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00010070: 6275 7469 6f6e 2c20 602d 2d77 6569 6768  bution, `--weigh
+00010080: 7473 5f6f 7574 6c69 6572 735f 696e 636c  ts_outliers_incl
+00010090: 7564 6564 6020 7768 6963 6820 6465 7465  uded` which dete
+000100a0: 726d 696e 6573 2077 6865 7468 6572 2074  rmines whether t
+000100b0: 6f20 696e 636c 7564 6520 2a4b 2a3c 7375  o include *K*<su
+000100c0: 623e 533c 2f73 7562 3e20 6f75 746c 6965  b>S</sub> outlie
+000100d0: 7273 2028 7768 6f73 6520 7661 6c75 6520  rs (whose value 
+000100e0: 6973 206f 7665 7220 3529 2069 6e20 616e  is over 5) in an
+000100f0: 616c 7973 6973 2c20 602d 2d6d 6574 686f  alysis, `--metho
+00010100: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
+00010110: 6e65 7320 7768 6963 6820 636c 7573 7465  nes which cluste
+00010120: 7269 6e67 206d 6574 686f 6420 746f 2075  ring method to u
+00010130: 7365 2028 6465 6661 756c 7420 676d 6d29  se (default gmm)
+00010140: 2c20 602d 2d73 6565 6460 2077 6869 6368  , `--seed` which
+00010150: 2073 6574 7320 7468 6520 7261 6e64 6f6d   sets the random
+00010160: 2073 6565 6420 6769 7665 6e20 746f 2069   seed given to i
+00010170: 6e69 7469 616c 697a 6174 696f 6e20 2864  nitialization (d
+00010180: 6566 6175 6c74 2032 3335 3238 3930 292c  efault 2352890),
+00010190: 2060 2d2d 6e5f 696e 6974 602c 2077 6869   `--n_init`, whi
+000101a0: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
+000101b0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
+000101c0: 6974 6961 6c69 7a61 7469 6f6e 7320 2864  itializations (d
+000101d0: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
+000101e0: 656d 5f69 7465 7260 2c20 7768 6963 6820  em_iter`, which 
+000101f0: 7365 7473 2074 6865 206d 6178 696d 756d  sets the maximum
+00010200: 206e 756d 6265 7220 6f66 2069 7465 7261   number of itera
+00010210: 7469 6f6e 7320 2864 6566 6175 6c74 2032  tions (default 2
+00010220: 3030 292c 2060 2d2d 6761 6d6d 6160 2c20  00), `--gamma`, 
+00010230: 7768 6963 6820 7365 7473 2074 6865 2067  which sets the g
+00010240: 616d 6d61 2070 6172 616d 6574 6572 2066  amma parameter f
+00010250: 6f72 2074 6865 2062 676d 6d20 6d6f 6465  or the bgmm mode
+00010260: 6c20 2864 6566 6175 6c74 2030 2e30 3031  l (default 0.001
+00010270: 292c 2060 2d2d 636f 6d70 6f6e 656e 7473  ), `--components
+00010280: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+00010290: 6520 7261 6e67 6520 6f66 2074 6865 206e  e range of the n
+000102a0: 756d 6265 7220 6f66 2063 6f6d 706f 6e65  umber of compone
+000102b0: 6e74 7320 746f 2066 6974 2028 6465 6661  nts to fit (defa
+000102c0: 756c 7420 3120 3429 2c20 602d 2d77 6569  ult 1 4), `--wei
+000102d0: 6768 7465 6460 2077 6869 6368 2064 6574  ghted` which det
+000102e0: 6572 6d69 6e65 7320 7768 6574 6865 7220  ermines whether 
+000102f0: 746f 2075 7365 206e 6f64 652d 7765 6967  to use node-weig
+00010300: 6874 6564 206d 6574 686f 6420 666f 7220  hted method for 
+00010310: 6465 2d72 6564 756e 6461 6e63 792c 2060  de-redundancy, `
+00010320: 2d2d 6775 6964 6560 2077 6869 6368 2064  --guide` which d
+00010330: 6574 6572 6d69 6e65 7320 7768 6963 6820  etermines which 
+00010340: 7265 6769 6d65 2072 6573 6964 696e 6720  regime residing 
+00010350: 616e 6368 6f72 7320 746f 2062 6520 7573  anchors to be us
+00010360: 6564 2028 6465 6661 756c 7420 5365 676d  ed (default Segm
+00010370: 656e 7429 2c20 602d 2d70 726f 6d69 6e65  ent), `--promine
+00010380: 6e63 655f 6375 746f 6666 6020 7768 6963  nce_cutoff` whic
+00010390: 6820 7365 7473 2074 6865 2070 726f 6d69  h sets the promi
+000103a0: 6e65 6e63 6520 6375 746f 6666 206f 6620  nence cutoff of 
+000103b0: 6163 6365 7074 6162 6c65 2070 6561 6b73  acceptable peaks
+000103c0: 2069 6e20 7065 616b 2066 696e 6469 6e67   in peak finding
+000103d0: 2070 726f 6365 7373 2c20 602d 2d72 656c   process, `--rel
+000103e0: 5f68 6569 6768 7460 2077 6869 6368 2073  _height` which s
+000103f0: 6574 7320 7468 6520 7265 6c61 7469 7665  ets the relative
+00010400: 2068 6569 6768 7420 6174 2077 6869 6368   height at which
+00010410: 2074 6865 2070 6561 6b20 7769 6474 6820   the peak width 
+00010420: 6973 206d 6561 7375 7265 642c 2060 2d2d  is measured, `--
+00010430: 6b73 746f 6461 7465 6020 7768 6963 6820  kstodate` which 
+00010440: 6d61 6e75 616c 6c79 2073 6574 7320 7468  manually sets th
+00010450: 6520 7261 6e67 6520 6f66 202a 4b2a 3c73  e range of *K*<s
+00010460: 7562 3e53 3c2f 7375 623e 2074 6f20 6265  ub>S</sub> to be
+00010470: 2064 6174 6564 2069 6e20 6865 7572 6973   dated in heuris
+00010480: 7469 6320 7365 6172 6368 2061 6e64 206e  tic search and n
+00010490: 6565 6473 2074 6f20 6265 2063 6f2d 7365  eeds to be co-se
+000104a0: 7420 7769 7468 206f 7074 696f 6e20 602d  t with option `-
+000104b0: 2d6d 616e 7561 6c73 6574 602c 2060 2d2d  -manualset`, `--
+000104c0: 786c 696d 6020 616e 6420 602d 2d79 6c69  xlim` and `--yli
+000104d0: 6d60 2064 6574 6572 6d69 6e69 6e67 2074  m` determining t
+000104e0: 6865 2078 2061 6e64 2079 2061 7869 7320  he x and y axis 
+000104f0: 6c69 6d69 7420 6f66 2047 4d4d 202a 4b2a  limit of GMM *K*
+00010500: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+00010510: 7472 6962 7574 696f 6e2c 2060 2d2d 6369  tribution, `--ci
+00010520: 6020 7365 7474 696e 6720 7468 6520 636f  ` setting the co
+00010530: 6e66 6964 656e 6365 206c 6576 656c 206f  nfidence level o
+00010540: 6620 6c6f 672d 6e6f 726d 616c 2064 6973  f log-normal dis
+00010550: 7472 6962 7574 696f 6e20 746f 2064 6174  tribution to dat
+00010560: 6520 2864 6566 6175 6c74 2039 3529 2c20  e (default 95), 
+00010570: 602d 2d68 6472 6020 7365 7474 696e 6720  `--hdr` setting 
+00010580: 7468 6520 6869 6768 6573 7420 6465 6e73  the highest dens
+00010590: 6974 7920 7265 6769 6f6e 2028 4844 5229  ity region (HDR)
+000105a0: 2061 7070 6c69 6564 2069 6e20 7468 6520   applied in the 
+000105b0: 7365 676d 656e 742d 6775 6964 6564 2061  segment-guided a
+000105c0: 6e63 686f 7220 7061 6972 202a 4b2a 3c73  nchor pair *K*<s
+000105d0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
+000105e0: 6962 7574 696f 6e20 2864 6566 6175 6c74  ibution (default
+000105f0: 2039 3529 2c20 602d 2d68 6575 7269 7374   95), `--heurist
+00010600: 6963 6020 6465 7465 726d 696e 696e 6720  ic` determining 
+00010610: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00010620: 6174 6520 6865 7572 6973 7469 6320 6d65  ate heuristic me
+00010630: 7468 6f64 206f 6620 6465 6669 6e69 6e67  thod of defining
+00010640: 2043 4920 666f 7220 6461 7469 6e67 2c20   CI for dating, 
+00010650: 602d 2d6b 7363 7574 6f66 6660 2073 6574  `--kscutoff` set
+00010660: 7469 6e67 2074 6865 202a 4b2a 3c73 7562  ting the *K*<sub
+00010670: 3e53 3c2f 7375 623e 2073 6174 7572 6174  >S</sub> saturat
+00010680: 696f 6e20 6375 746f 6666 2069 6e20 6461  ion cutoff in da
+00010690: 7469 6e67 2028 6465 6661 756c 7420 3529  ting (default 5)
+000106a0: 2e0a 0a46 6f75 7220 7265 7375 6c74 2073  ...Four result s
+000106b0: 7562 666c 6f64 6572 7320 7769 6c6c 2062  ubfloders will b
+000106c0: 6520 7072 6f64 7563 6564 2c20 6e61 6d65  e produced, name
+000106d0: 6c79 2060 416e 6368 6f72 4b73 5f46 696e  ly `AnchorKs_Fin
+000106e0: 6450 6561 6b60 2c20 6041 6e63 686f 724b  dPeak`, `AnchorK
+000106f0: 735f 474d 4d60 2c20 6053 6567 6d65 6e74  s_GMM`, `Segment
+00010700: 4775 6964 654b 735f 474d 4d60 2061 6e64  GuideKs_GMM` and
+00010710: 2060 5365 676d 656e 744b 735f 4669 6e64   `SegmentKs_Find
+00010720: 5065 616b 602e 2054 6865 2060 416e 6368  Peak`. The `Anch
+00010730: 6f72 4b73 5f46 696e 6450 6561 6b60 2073  orKs_FindPeak` s
+00010740: 7562 666c 6f64 6572 2063 6f6e 7461 696e  ubfloder contain
+00010750: 7320 7265 7375 6c74 7320 6f66 2074 6865  s results of the
+00010760: 2064 6574 6563 7465 6420 7065 616b 7320   detected peaks 
+00010770: 6279 2074 6865 2060 7369 676e 616c 6020  by the `signal` 
+00010780: 6d6f 6475 6c65 206f 6620 6053 6369 5079  module of `SciPy
+00010790: 6020 6c69 6272 6172 7920 616e 6420 7468  ` library and th
+000107a0: 6520 6173 7375 6d65 6420 6869 6768 6573  e assumed highes
+000107b0: 7420 6d61 7373 2070 6172 7420 2872 6566  t mass part (ref
+000107c0: 6572 7265 6420 746f 2061 7320 4869 6768  erred to as High
+000107d0: 4d61 7373 2068 6572 6561 6674 6572 2920  Mass hereafter) 
+000107e0: 6f66 2065 6163 6820 7065 616b 2c20 7768  of each peak, wh
+000107f0: 6963 6820 6361 6e20 6265 2075 7365 6420  ich can be used 
+00010800: 666f 7220 6675 7274 6865 7220 5747 4420  for further WGD 
+00010810: 6461 7469 6e67 2e20 5468 6520 6041 6e63  dating. The `Anc
+00010820: 686f 724b 735f 474d 4d60 2073 686f 7773  horKs_GMM` shows
+00010830: 2074 6865 2047 4d4d 2072 6573 756c 7473   the GMM results
+00010840: 2075 706f 6e20 7468 6520 6f72 6967 696e   upon the origin
+00010850: 616c 2061 6e63 686f 7220 2a4b 2a3c 7375  al anchor *K*<su
+00010860: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00010870: 6275 7469 6f6e 2062 7920 7468 6520 606d  bution by the `m
+00010880: 6978 7475 7265 6020 6d6f 6475 6c65 206f  ixture` module o
+00010890: 6620 6073 6369 6b69 742d 6c65 6172 6e60  f `scikit-learn`
+000108a0: 206c 6962 7261 7279 2061 6e64 2074 776f   library and two
+000108b0: 2073 7562 666c 6f64 6572 732c 2060 4c6f   subfloders, `Lo
+000108c0: 6747 4d4d 5f43 4960 2063 6f6e 7461 696e  gGMM_CI` contain
+000108d0: 696e 6720 7468 6520 7265 7375 6c74 7320  ing the results 
+000108e0: 6f66 2039 3525 2043 4920 6f66 2065 6163  of 95% CI of eac
+000108f0: 6820 636f 6d70 6f6e 656e 742c 2060 4869  h component, `Hi
+00010900: 6768 4d61 7373 5f43 4960 2063 6f6e 7461  ghMass_CI` conta
+00010910: 696e 696e 6720 7468 6520 4869 6768 4d61  ining the HighMa
+00010920: 7373 206f 6620 6561 6368 2063 6f6d 706f  ss of each compo
+00010930: 6e65 6e74 2c20 7768 6963 6820 6361 6e20  nent, which can 
+00010940: 6265 2075 7365 6420 666f 7220 6675 7274  be used for furt
+00010950: 6865 7220 5747 4420 6461 7469 6e67 2e20  her WGD dating. 
+00010960: 5468 6520 6053 6567 6d65 6e74 4775 6964  The `SegmentGuid
+00010970: 654b 735f 474d 4d60 2073 7562 666f 6c64  eKs_GMM` subfold
+00010980: 6572 2070 7265 7365 6e74 7320 7265 7375  er presents resu
+00010990: 6c74 7320 6f66 2073 6567 6d65 6e74 202a  lts of segment *
+000109a0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2047  K*<sub>S</sub> G
+000109b0: 4d4d 2077 6869 6368 2061 7265 206d 6170  MM which are map
+000109c0: 7065 6420 6261 636b 2074 6f20 7468 6520  ped back to the 
+000109d0: 7265 7369 6469 6e67 2061 6e63 686f 7220  residing anchor 
+000109e0: 7061 6972 7320 616e 6420 7468 6520 6173  pairs and the as
+000109f0: 736f 6369 6174 6564 2039 3525 2048 4452  sociated 95% HDR
+00010a00: 2061 6e64 2048 6967 684d 6173 7320 6f66   and HighMass of
+00010a10: 2065 6163 6820 7365 676d 656e 7420 636c   each segment cl
+00010a20: 7573 7465 7220 696e 2073 7562 666c 6f64  uster in subflod
+00010a30: 6572 7320 6f66 2060 4844 525f 4349 6020  ers of `HDR_CI` 
+00010a40: 616e 6420 6048 6967 684d 6173 735f 4349  and `HighMass_CI
+00010a50: 602e 2054 6865 2060 5365 676d 656e 744b  `. The `SegmentK
+00010a60: 735f 4669 6e64 5065 616b 6020 7375 6266  s_FindPeak` subf
+00010a70: 6f6c 6465 7220 6973 2073 696d 696c 6172  older is similar
+00010a80: 2074 6f20 6041 6e63 686f 724b 735f 4669   to `AnchorKs_Fi
+00010a90: 6e64 5065 616b 6020 6275 7420 7769 7468  ndPeak` but with
+00010aa0: 2073 6567 6d65 6e74 202a 4b2a 3c73 7562   segment *K*<sub
+00010ab0: 3e53 3c2f 7375 623e 2069 6e73 7465 6164  >S</sub> instead
+00010ac0: 2e20 5468 6520 2a4b 2a3c 7375 623e 533c  . The *K*<sub>S<
+00010ad0: 2f73 7562 3e20 696e 2060 4d75 6c74 6970  /sub> in `Multip
+00010ae0: 6c69 636f 6e60 2063 616e 2061 6c73 6f20  licon` can also 
+00010af0: 6265 2063 616c 6375 6c61 7465 6420 696e  be calculated in
+00010b00: 2070 6c61 6365 206f 6620 6053 6567 6d65   place of `Segme
+00010b10: 6e74 6020 7573 696e 6720 7468 6520 6f70  nt` using the op
+00010b20: 7469 6f6e 2060 2d2d 6775 6964 6560 2061  tion `--guide` a
+00010b30: 7320 7375 6368 2074 6865 2072 6573 756c  s such the resul
+00010b40: 7420 7469 746c 652c 206c 6162 656c 2c20  t title, label, 
+00010b50: 6669 6c65 2061 6e64 2066 6f6c 6465 7220  file and folder 
+00010b60: 6e61 6d65 7320 7769 6c6c 2062 6520 6368  names will be ch
+00010b70: 616e 6765 6420 6163 636f 7264 696e 676c  anged accordingl
+00010b80: 792e 0a0a 2a2a 4120 7375 6767 6573 7465  y...**A suggeste
+00010b90: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
+00010ba0: 616e 2075 7365 2063 6f6d 6d61 6e64 2073  an use command s
+00010bb0: 696d 706c 7920 6173 2062 656c 6f77 2a2a  imply as below**
+00010bc0: 0a0a 6060 600a 7767 6420 7065 616b 206b  ..```.wgd peak k
+00010bd0: 7364 6174 6120 2d61 7020 6170 6461 7461  sdata -ap apdata
+00010be0: 202d 736d 2073 6d64 6174 6120 2d6c 6520   -sm smdata -le 
+00010bf0: 6c65 6461 7461 202d 6d70 206d 7064 6174  ledata -mp mpdat
+00010c00: 6120 2d2d 6865 7572 6973 7469 630a 6060  a --heuristic.``
+00010c10: 600a 0a23 2323 2077 6764 2073 796e 0a0a  `..### wgd syn..
+00010c20: 2a2a 5468 6520 696e 7472 612d 7370 6563  **The intra-spec
+00010c30: 6966 6963 2073 796e 7465 6e79 2069 6e66  ific synteny inf
+00010c40: 6572 656e 6365 2a2a 0a60 6060 0a77 6764  erence**.```.wgd
+00010c50: 2073 796e 2066 616d 696c 6965 7320 6766   syn families gf
+00010c60: 6620 282d 2d6b 735f 6469 7374 7269 6275  f (--ks_distribu
+00010c70: 7469 6f6e 206b 7364 6174 6120 2d66 2067  tion ksdata -f g
+00010c80: 656e 6520 2d61 2049 4420 2d2d 6d69 6e6c  ene -a ID --minl
+00010c90: 656e 202d 3120 2d2d 6d69 6e73 6567 6c65  en -1 --minsegle
+00010ca0: 6e20 3130 3030 3020 2d2d 6d69 6e67 656e  n 10000 --mingen
+00010cb0: 656e 756d 2033 3029 0a60 6060 0a0a 5468  enum 30).```..Th
+00010cc0: 6520 7072 6f67 7261 6d20 6077 6764 2073  e program `wgd s
+00010cd0: 796e 6020 6973 206d 6169 6e6c 7920 6465  yn` is mainly de
+00010ce0: 616c 696e 6720 7769 7468 2063 6f6c 6c69  aling with colli
+00010cf0: 6e65 6172 6974 7920 6f72 2073 796e 7465  nearity or synte
+00010d00: 6e79 2028 626f 7468 2072 6566 6572 7265  ny (both referre
+00010d10: 6420 746f 2061 7320 7379 6e74 656e 7920  d to as synteny 
+00010d20: 6865 7265 6166 7465 7229 2061 6e61 6c79  hereafter) analy
+00010d30: 7369 732e 2054 776f 2069 6e70 7574 2066  sis. Two input f
+00010d40: 696c 6573 2061 7265 2065 7373 656e 7469  iles are essenti
+00010d50: 616c 2c20 7468 6520 6765 6e65 2066 616d  al, the gene fam
+00010d60: 696c 7920 6669 6c65 2061 6e64 2074 6865  ily file and the
+00010d70: 2067 6666 3320 6669 6c65 2e20 5468 6520   gff3 file. The 
+00010d80: 6765 6e65 2066 616d 696c 7920 6669 6c65  gene family file
+00010d90: 2069 7320 696e 2074 6865 2066 6f72 6d61   is in the forma
+00010da0: 7420 6173 2060 4f72 7468 6f46 696e 6465  t as `OrthoFinde
+00010db0: 7260 2e20 5468 6520 736f 6674 7761 7265  r`. The software
+00010dc0: 2060 692d 6164 686f 7265 6020 6973 2061   `i-adhore` is a
+00010dd0: 2070 7265 7265 7175 6973 6974 652e 2057   prerequisite. W
+00010de0: 6974 6820 6465 6661 756c 7420 7061 7261  ith default para
+00010df0: 6d65 7465 7273 2c20 7468 6520 7072 6f67  meters, the prog
+00010e00: 7261 6d20 6261 7369 6361 6c6c 7920 636f  ram basically co
+00010e10: 6e64 7563 7473 2031 2920 6669 6c74 6572  nducts 1) filter
+00010e20: 696e 6720 6765 6e65 2066 616d 696c 6965  ing gene familie
+00010e30: 7320 6261 7365 6420 6f6e 206d 6178 696d  s based on maxim
+00010e40: 756d 2066 616d 696c 7920 7369 7a65 2032  um family size 2
+00010e50: 2920 7265 7472 6965 7669 6e67 2067 656e  ) retrieving gen
+00010e60: 6520 706f 7369 7469 6f6e 2061 6e64 2073  e position and s
+00010e70: 6361 6666 6f6c 6420 696e 666f 726d 6174  caffold informat
+00010e80: 696f 6e20 6672 6f6d 2067 6666 3320 6669  ion from gff3 fi
+00010e90: 6c65 2033 2920 7072 6f64 7563 696e 6720  le 3) producing 
+00010ea0: 7468 6520 636f 6e66 6967 7572 6174 696f  the configuratio
+00010eb0: 6e20 6669 6c65 2061 6e64 2061 7373 6f63  n file and assoc
+00010ec0: 6961 7465 6420 6461 7461 6669 6c65 7320  iated datafiles 
+00010ed0: 666f 7220 6069 2d61 6468 6f72 6560 2034  for `i-adhore` 4
+00010ee0: 2920 6361 6c6c 696e 6720 6069 2d61 6468  ) calling `i-adh
+00010ef0: 6f72 6560 2067 6976 656e 2074 6865 2070  ore` given the p
+00010f00: 6172 616d 6574 6572 7320 7365 7420 746f  arameters set to
+00010f10: 2069 6e66 6572 2073 796e 7465 6e79 2035   infer synteny 5
+00010f20: 2920 7669 7375 616c 697a 696e 6720 7468  ) visualizing th
+00010f30: 6520 7379 6e74 656e 7920 696e 2022 646f  e synteny in "do
+00010f40: 7470 6c6f 7422 2069 6e20 7468 6520 756e  tplot" in the un
+00010f50: 6974 206f 6620 6765 6e65 7320 616e 6420  it of genes and 
+00010f60: 6261 7365 732c 2069 6e20 2253 796e 6465  bases, in "Synde
+00010f70: 7074 6822 2070 6c6f 7420 7368 6f77 696e  pth" plot showin
+00010f80: 6720 7468 6520 6469 7374 7269 6275 7469  g the distributi
+00010f90: 6f6e 206f 6620 6469 6666 6572 656e 7420  on of different 
+00010fa0: 6361 7465 676f 7269 6573 206f 6620 636f  categories of co
+00010fb0: 6c6c 696e 6561 7269 7479 2072 6174 696f  llinearity ratio
+00010fc0: 7320 7769 7468 696e 2061 6e64 2062 6574  s within and bet
+00010fd0: 7765 656e 2073 7065 6369 6573 2c20 696e  ween species, in
+00010fe0: 2022 6475 7053 7461 636b 2220 706c 6f74   "dupStack" plot
+00010ff0: 2073 686f 7769 6e67 206d 756c 7469 706c   showing multipl
+00011000: 6963 6f6e 7320 7769 7468 2064 6966 6665  icons with diffe
+00011010: 7265 6e74 206d 756c 7469 706c 6963 6174  rent multiplicat
+00011020: 696f 6e20 6c65 7665 6c73 2e20 3629 2069  ion levels. 6) i
+00011030: 6620 7769 7468 202a 4b2a 3c73 7562 3e53  f with *K*<sub>S
+00011040: 3c2f 7375 623e 2064 6174 612c 2061 2022  </sub> data, a "
+00011050: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00011060: 646f 7470 6c6f 7422 2077 6974 6820 646f  dotplot" with do
+00011070: 7473 2061 6e6e 6f74 6174 6564 2069 6e20  ts annotated in 
+00011080: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00011090: 7661 6c75 6573 2061 6e64 2061 202a 4b2a  values and a *K*
+000110a0: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+000110b0: 7472 6962 7574 696f 6e20 7769 7468 2061  tribution with a
+000110c0: 6e63 686f 7220 7061 6972 7320 6465 6e6f  nchor pairs deno
+000110d0: 7465 6420 7769 6c6c 2062 6520 7072 6f64  ted will be prod
+000110e0: 7563 6564 2e20 5468 6520 6765 6e65 2069  uced. The gene i
+000110f0: 6e66 6f72 6d61 7469 6f6e 2069 6e20 7468  nformation in th
+00011100: 6520 6765 6e65 2066 616d 696c 7920 6669  e gene family fi
+00011110: 6c65 2061 6e64 2067 6666 3320 6669 6c65  le and gff3 file
+00011120: 2073 686f 756c 6420 6265 206d 6174 6368   should be match
+00011130: 6564 2077 6869 6368 2072 6571 7569 7265  ed which require
+00011140: 7320 7573 6572 7320 746f 2073 6574 2070  s users to set p
+00011150: 726f 7065 7220 602d 2d66 6561 7475 7265  roper `--feature
+00011160: 6020 616e 6420 602d 2d61 7474 7269 6275  ` and `--attribu
+00011170: 7465 602e 2054 6865 206d 6178 696d 756d  te`. The maximum
+00011180: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
+00011190: 6265 2069 6e63 6c75 6465 6420 6361 6e20  be included can 
+000111a0: 6265 2073 6574 2076 6961 2074 6865 206f  be set via the o
+000111b0: 7074 696f 6e20 602d 2d6d 6178 7369 7a65  ption `--maxsize
+000111c0: 602c 206e 6f74 6564 2074 6861 7420 7468  `, noted that th
+000111d0: 6973 2066 696c 7465 7269 6e67 2069 7320  is filtering is 
+000111e0: 6d61 696e 6c79 2074 6f20 6472 6f70 2074  mainly to drop t
+000111f0: 686f 7365 2068 7567 6520 7461 6e64 656d  hose huge tandem
+00011200: 2064 7570 6c69 6361 7465 7320 6661 6d69   duplicates fami
+00011210: 6c79 2061 6e64 2074 7261 6e73 706f 7361  ly and transposa
+00011220: 626c 6520 656c 656d 656e 7473 2028 5445  ble elements (TE
+00011230: 7329 2066 616d 696c 792c 2061 6e64 206e  s) family, and n
+00011240: 6f74 206d 616e 6461 746f 7279 2e20 5573  ot mandatory. Us
+00011250: 6572 7320 6361 6e20 6669 6c74 6572 2074  ers can filter t
+00011260: 686f 7365 2066 7261 676d 656e 7461 7279  hose fragmentary
+00011270: 2073 6361 6666 6f6c 6473 2076 6961 2074   scaffolds via t
+00011280: 6865 206f 7074 696f 6e20 602d 2d6d 696e  he option `--min
+00011290: 6c65 6e60 2e20 5468 6520 6d69 6e69 6d75  len`. The minimu
+000112a0: 6d20 6c65 6e67 7468 2061 6e64 206e 756d  m length and num
+000112b0: 6265 7220 6f66 2067 656e 6573 2066 6f72  ber of genes for
+000112c0: 2061 2073 6567 6d65 6e74 2074 6f20 6265   a segment to be
+000112d0: 2063 6f6e 7369 6465 7265 6420 6361 6e20   considered can 
+000112e0: 6265 2073 6574 2076 6961 2074 6865 206f  be set via the o
+000112f0: 7074 696f 6e20 602d 2d6d 696e 7365 676c  ption `--minsegl
+00011300: 656e 6020 616e 6420 602d 2d6d 696e 6765  en` and `--minge
+00011310: 6e65 6e75 6d60 2e20 5265 6475 6e64 616e  nenum`. Redundan
+00011320: 7420 6d75 6c74 6970 6c69 636f 6e73 2063  t multiplicons c
+00011330: 616e 2062 6520 6b65 7074 2062 7920 7365  an be kept by se
+00011340: 7420 7468 6520 666c 6167 206f 7074 696f  t the flag optio
+00011350: 6e20 602d 2d6b 6565 7072 6564 756e 602e  n `--keepredun`.
+00011360: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+00011370: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+00011380: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+00011390: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+000113a0: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+000113b0: 6c69 6573 2067 6666 0a60 6060 0a0a 2a2a  lies gff.```..**
+000113c0: 5468 6520 696e 7465 722d 7370 6563 6966  The inter-specif
+000113d0: 6963 2073 796e 7465 6e79 2069 6e66 6572  ic synteny infer
+000113e0: 656e 6365 2a2a 0a60 6060 0a77 6764 2073  ence**.```.wgd s
+000113f0: 796e 2066 616d 696c 6965 7320 6766 6631  yn families gff1
+00011400: 2067 6666 3220 282d 2d61 6464 6974 696f   gff2 (--additio
+00011410: 6e61 6c67 6666 696e 666f 2022 6d52 4e41  nalgffinfo "mRNA
+00011420: 3b4e 616d 6522 202d 2d61 6464 6974 696f  ;Name" --additio
+00011430: 6e61 6c67 6666 696e 666f 2022 6765 6e65  nalgffinfo "gene
+00011440: 3b49 4422 290a 6060 600a 0a46 6f72 206d  ;ID").```..For m
+00011450: 756c 7469 2d73 7065 6369 6573 2073 796e  ulti-species syn
+00011460: 7465 6e79 2069 6e66 6572 656e 6365 2c20  teny inference, 
+00011470: 6966 2075 7365 7273 2068 6176 6520 6766  if users have gf
+00011480: 6633 2066 696c 6573 2077 6869 6368 2068  f3 files which h
+00011490: 6176 6520 6469 6666 6572 656e 7420 6665  ave different fe
+000114a0: 6174 7572 6573 206f 7220 6174 7472 6962  atures or attrib
+000114b0: 7574 6573 2066 6f72 2067 656e 6520 706f  utes for gene po
+000114c0: 7369 7469 6f6e 2069 6e66 6f72 6d61 7469  sition informati
+000114d0: 6f6e 2072 6574 7269 6576 616c 2c20 7468  on retrieval, th
+000114e0: 6520 6f70 7469 6f6e 2060 2d2d 6164 6469  e option `--addi
+000114f0: 7469 6f6e 616c 6766 6669 6e66 6f60 2063  tionalgffinfo` c
+00011500: 616e 2062 6520 7365 7420 746f 2070 726f  an be set to pro
+00011510: 7669 6465 2074 6865 2061 6464 6974 696f  vide the additio
+00011520: 6e61 6c20 696e 666f 726d 6174 696f 6e2e  nal information.
+00011530: 2054 6865 2072 656d 6169 6e69 6e67 2070   The remaining p
+00011540: 6172 616d 6574 6572 2073 6574 7469 6e67  arameter setting
+00011550: 2069 7320 7468 6520 7361 6d65 2061 7320   is the same as 
+00011560: 7468 6520 696e 7472 612d 7370 6563 6966  the intra-specif
+00011570: 6963 2073 796e 7465 6e79 2069 6e66 6572  ic synteny infer
+00011580: 656e 6365 2e0a 0a2a 2a41 2073 7567 6765  ence...**A sugge
+00011590: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
+000115a0: 6e20 6361 6e20 7573 6520 636f 6d6d 616e  n can use comman
+000115b0: 6420 7369 6d70 6c79 2061 7320 6265 6c6f  d simply as belo
+000115c0: 772a 2a0a 0a60 6060 0a77 6764 2073 796e  w**..```.wgd syn
+000115d0: 2066 616d 696c 6965 7320 6766 6631 2067   families gff1 g
+000115e0: 6666 320a 6060 600a 0a23 2323 2077 6764  ff2.```..### wgd
+000115f0: 2076 697a 0a0a 2a2a 5468 6520 7669 7375   viz..**The visu
+00011600: 616c 697a 6174 696f 6e20 6f66 202a 4b2a  alization of *K*
+00011610: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00011620: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+00011630: 6420 454c 4d4d 2061 6e61 6c79 7369 732a  d ELMM analysis*
+00011640: 2a0a 6060 600a 7767 6420 7669 7a20 2d64  *.```.wgd viz -d
+00011650: 206b 7364 6174 610a 6060 600a 0a54 6865   ksdata.```..The
+00011660: 2070 726f 6772 616d 2060 7767 6420 7669   program `wgd vi
+00011670: 7a60 2069 7320 6d61 696e 6c79 2066 6f72  z` is mainly for
+00011680: 2074 6865 2070 7572 706f 7365 206f 6620   the purpose of 
+00011690: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000116a0: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+000116b0: 2073 796e 7465 6e79 2076 6973 7561 6c69   synteny visuali
+000116c0: 7a61 7469 6f6e 2c20 7769 7468 2073 6f6d  zation, with som
+000116d0: 6520 6f70 7469 6f6e 616c 206d 6978 7475  e optional mixtu
+000116e0: 7265 206d 6f64 656c 696e 6720 616e 616c  re modeling anal
+000116f0: 7973 6973 2e20 5468 6520 6261 7369 6320  ysis. The basic 
+00011700: 6675 6e63 7469 6f6e 2069 7320 6a75 7374  function is just
+00011710: 2074 6f20 706c 6f74 2074 6865 202a 4b2a   to plot the *K*
+00011720: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
+00011730: 7472 6962 7574 696f 6e20 616e 6420 636f  tribution and co
+00011740: 6e64 7563 7420 616e 2045 4c4d 4d20 616e  nduct an ELMM an
+00011750: 616c 7973 6973 2069 6e20 7365 6172 6368  alysis in search
+00011760: 206f 6620 706f 7465 6e74 6961 6c20 5747   of potential WG
+00011770: 4420 636f 6d70 6f6e 656e 7473 2e20 536f  D components. So
+00011780: 6d65 206b 6579 2070 6172 616d 6574 6572  me key parameter
+00011790: 7320 6166 6665 6374 696e 6720 7468 6520  s affecting the 
+000117a0: 454c 4d4d 2072 6573 756c 7420 696e 636c  ELMM result incl
+000117b0: 7564 6520 602d 2d70 726f 6d69 6e65 6e63  ude `--prominenc
+000117c0: 655f 6375 746f 6666 602c 2060 2d2d 7265  e_cutoff`, `--re
+000117d0: 6c5f 6865 6967 6874 602c 2060 2d2d 656d  l_height`, `--em
+000117e0: 5f69 7465 7261 7469 6f6e 7360 2061 6e64  _iterations` and
+000117f0: 2060 2d2d 656d 5f69 6e69 7469 616c 697a   `--em_initializ
+00011800: 6174 696f 6e73 602c 2061 6c6c 206f 6620  ations`, all of 
+00011810: 7768 6963 6820 6861 7665 2062 6565 6e20  which have been 
+00011820: 6578 706c 6169 6e65 6420 6168 6561 642e  explained ahead.
+00011830: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+00011840: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+00011850: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+00011860: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+00011870: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
+00011880: 7364 6174 610a 6060 600a 0a2a 2a54 6865  sdata.```..**The
+00011890: 2076 6973 7561 6c69 7a61 7469 6f6e 206f   visualization o
+000118a0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+000118b0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+000118c0: 6f6e 2077 6974 6820 7261 7465 2063 6f72  on with rate cor
+000118d0: 7265 6374 696f 6e2a 2a0a 6060 600a 7767  rection**.```.wg
+000118e0: 6420 7669 7a20 2d64 206b 7364 6174 6120  d viz -d ksdata 
+000118f0: 2d73 7020 7370 6461 7461 202d 2d66 6f63  -sp spdata --foc
+00011900: 7573 3261 6c6c 2066 6f63 616c 5f73 7065  us2all focal_spe
+00011910: 6369 6573 202d 2d65 7874 7261 7061 7261  cies --extrapara
+00011920: 6e6f 6d65 6b73 206b 7364 6174 6120 282d  nomeks ksdata (-
+00011930: 2d61 6e63 686f 7270 6f69 6e74 7320 6170  -anchorpoints ap
+00011940: 6461 7461 202d 2d70 6c6f 7461 7067 6d6d  data --plotapgmm
+00011950: 202d 2d70 6c6f 7465 6c6d 6d29 0a60 6060   --plotelmm).```
+00011960: 0a0a 4265 7369 6465 7320 7468 6520 6261  ..Besides the ba
+00011970: 7369 6320 2a4b 2a3c 7375 623e 533c 2f73  sic *K*<sub>S</s
+00011980: 7562 3e20 706c 6f74 2c20 7375 6273 7469  ub> plot, substi
+00011990: 7475 7469 6f6e 2072 6174 6520 636f 7272  tution rate corr
+000119a0: 6563 7469 6f6e 2063 616e 2061 6c73 6f20  ection can also 
+000119b0: 6265 2061 6368 6965 7665 6420 6769 7665  be achieved give
+000119c0: 6e20 6174 206c 6561 7374 2061 2073 7065  n at least a spe
+000119d0: 6369 6573 2074 7265 6520 2876 6961 2074  cies tree (via t
+000119e0: 6865 206f 7074 696f 6e20 602d 2d73 7065  he option `--spe
+000119f0: 6369 6573 7472 6565 6029 2061 6e64 2061  ciestree`) and a
+00011a00: 2066 6f63 616c 2073 7065 6369 6573 2028   focal species (
+00011a10: 6569 7468 6572 2076 6961 2074 6865 206f  either via the o
+00011a20: 7074 696f 6e20 602d 2d66 6f63 7573 3261  ption `--focus2a
+00011a30: 6c6c 6020 6f72 2076 6961 2074 6865 206f  ll` or via the o
+00011a40: 7074 696f 6e20 602d 2d73 7061 6972 6020  ption `--spair` 
+00011a50: 696e 2074 6865 2066 6f72 6d20 6f66 2022  in the form of "
+00011a60: 2466 6f63 616c 5f73 7065 6369 6573 3b24  $focal_species;$
+00011a70: 666f 6361 6c5f 7370 6563 6965 7322 292e  focal_species").
+00011a80: 2049 7427 7320 7375 6767 6573 7465 6420   It's suggested 
+00011a90: 7468 6174 2074 6865 206f 7274 686f 6c6f  that the ortholo
+00011aa0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+00011ab0: 7375 623e 2064 6174 6120 6973 2070 726f  sub> data is pro
+00011ac0: 7669 6465 6420 6279 2074 6865 2060 2d2d  vided by the `--
+00011ad0: 6461 7461 6669 6c65 6020 6f70 7469 6f6e  datafile` option
+00011ae0: 2061 6e64 2074 6865 2070 6172 616c 6f67   and the paralog
+00011af0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00011b00: 7562 3e20 6461 7461 2069 7320 7072 6f76  ub> data is prov
+00011b10: 6964 6564 2062 7920 7468 6520 602d 2d65  ided by the `--e
+00011b20: 7874 7261 7061 7261 6e6f 6d65 6b73 6020  xtraparanomeks` 
+00011b30: 6f70 7469 6f6e 2c20 616c 7468 6f75 6768  option, although
+00011b40: 2069 7427 7320 616c 6c6f 7765 6420 746f   it's allowed to
+00011b50: 206f 6e6c 7920 7072 6f76 6964 6520 2a4b   only provide *K
+00011b60: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+00011b70: 7461 2076 6961 2074 6865 2060 2d2d 6461  ta via the `--da
+00011b80: 7461 6669 6c65 6020 6f70 7469 6f6e 2061  tafile` option a
+00011b90: 6e64 2064 6570 6f73 6974 2062 6f74 6820  nd deposit both 
+00011ba0: 6f72 7468 6f6c 6f67 6f75 7326 7061 7261  orthologous&para
+00011bb0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00011bc0: 3c2f 7375 623e 2064 6174 6120 7468 6572  </sub> data ther
+00011bd0: 656f 6e2e 2054 6865 7265 2061 7265 2074  eon. There are t
+00011be0: 776f 2074 7970 6573 206f 6620 6d69 7865  wo types of mixe
+00011bf0: 6420 706c 6f74 7320 2874 6865 2022 6d69  d plots (the "mi
+00011c00: 7865 6422 2068 6572 6520 7265 6665 7273  xed" here refers
+00011c10: 2074 6f20 6d69 7865 6420 6f72 7468 6f6c   to mixed orthol
+00011c20: 6f67 6f75 7320 616e 6420 7061 7261 6c6f  ogous and paralo
+00011c30: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+00011c40: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
+00011c50: 6e73 292c 206f 6e65 206f 6620 7768 6963  ns), one of whic
+00011c60: 6820 6973 2073 696d 696c 6172 2074 6f20  h is similar to 
+00011c70: 7768 6174 2060 6b73 7261 7465 7360 2070  what `ksrates` p
+00011c80: 6c6f 7473 2c20 7768 696c 6520 7468 6520  lots, while the 
+00011c90: 6f74 6865 7220 6f66 2077 6869 6368 2069  other of which i
+00011ca0: 7320 6c69 6b65 2074 6865 2063 6f6e 7665  s like the conve
+00011cb0: 6e74 696f 6e61 6c20 2a4b 2a3c 7375 623e  ntional *K*<sub>
+00011cc0: 533c 2f73 7562 3e20 706c 6f74 2074 6861  S</sub> plot tha
+00011cd0: 7420 626f 7468 2074 6865 206f 7269 6769  t both the origi
+00011ce0: 6e61 6c20 6f72 7468 6f6c 6f67 6f75 7320  nal orthologous 
+00011cf0: 616e 6420 7061 7261 6c6f 676f 7573 202a  and paralogous *
+00011d00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00011d10: 6973 7472 6962 7574 696f 6e73 2061 7265  istributions are
+00011d20: 2074 7275 7468 6675 6c6c 7920 706c 6f74   truthfully plot
+00011d30: 7465 6420 696e 7374 6561 6420 6f66 206a  ted instead of j
+00011d40: 7573 7420 6265 696e 6720 7265 7072 6573  ust being repres
+00011d50: 656e 7465 6420 6279 2073 6f6d 6520 7665  ented by some ve
+00011d60: 7274 6963 616c 206c 696e 6573 2e20 5765  rtical lines. We
+00011d70: 2073 7567 6765 7374 2075 7365 7273 2074   suggest users t
+00011d80: 6f20 6164 6f70 7420 7468 6520 606b 7372  o adopt the `ksr
+00011d90: 6174 6573 602d 6c69 6b65 2070 6c6f 7473  ates`-like plots
+00011da0: 2c20 7768 6963 6820 6973 2074 6865 2064  , which is the d
+00011db0: 6566 6175 6c74 2e20 4f74 6865 7277 6973  efault. Otherwis
+00011dc0: 6520 7468 6520 6f70 7469 6f6e 2060 2d2d  e the option `--
+00011dd0: 636c 6173 7369 6360 2077 696c 6c20 6265  classic` will be
+00011de0: 206e 6565 6465 6420 746f 2073 6574 2e20   needed to set. 
+00011df0: 4578 7472 6120 6d69 7874 7572 6520 6d6f  Extra mixture mo
+00011e00: 6465 6c69 6e67 2061 6e61 6c79 7369 7320  deling analysis 
+00011e10: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
+00011e20: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
+00011e30: 602d 2d70 6c6f 7465 6c6d 6d60 2061 6e64  `--plotelmm` and
+00011e40: 2060 2d2d 706c 6f74 6170 676d 6d60 2077   `--plotapgmm` w
+00011e50: 6974 6820 7468 6520 616e 6368 6f72 2070  ith the anchor p
+00011e60: 6f69 6e74 7320 6461 7461 6669 6c65 2070  oints datafile p
+00011e70: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
+00011e80: 2d2d 616e 6368 6f72 706f 696e 7473 6020  --anchorpoints` 
+00011e90: 6f70 7469 6f6e 2e0a 0a2a 2a41 2073 7567  option...**A sug
+00011ea0: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00011eb0: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+00011ec0: 616e 6420 7369 6d70 6c79 2061 7320 6265  and simply as be
+00011ed0: 6c6f 772a 2a0a 0a60 6060 0a77 6764 2076  low**..```.wgd v
+00011ee0: 697a 202d 6420 6b73 6461 7461 202d 7370  iz -d ksdata -sp
+00011ef0: 2073 7064 6174 6120 2d2d 666f 6375 7332   spdata --focus2
+00011f00: 616c 6c20 666f 6361 6c5f 7370 6563 6965  all focal_specie
+00011f10: 7320 2d2d 6578 7472 6170 6172 616e 6f6d  s --extraparanom
+00011f20: 656b 7320 6b73 6461 7461 0a60 6060 0a0a  eks ksdata.```..
+00011f30: 2a2a 5468 6520 7669 7375 616c 697a 6174  **The visualizat
+00011f40: 696f 6e20 6f66 2073 796e 7465 6e79 2a2a  ion of synteny**
+00011f50: 0a60 6060 0a77 6764 2076 697a 202d 6170  .```.wgd viz -ap
+00011f60: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
+00011f70: 7461 202d 6d74 206d 7464 6174 6120 2d67  ta -mt mtdata -g
+00011f80: 7420 6774 6461 7461 202d 2d70 6c6f 7473  t gtdata --plots
+00011f90: 796e 2028 2d2d 6d69 6e6c 656e 202d 3120  yn (--minlen -1 
+00011fa0: 2d2d 6d69 6e73 6567 6c65 6e20 3130 3030  --minseglen 1000
+00011fb0: 3020 2d2d 6d69 6e67 656e 656e 756d 2033  0 --mingenenum 3
+00011fc0: 3029 0a60 6060 0a0a 436f 6d70 6172 6564  0).```..Compared
+00011fd0: 2074 6f20 7468 6520 6f72 6967 696e 616c   to the original
+00011fe0: 2060 7767 6460 2c20 7468 6520 6077 6764   `wgd`, the `wgd
+00011ff0: 2076 697a 6020 7072 6f67 7261 6d20 6164   viz` program ad
+00012000: 6465 6420 7379 6e74 656e 7920 7669 7375  ded synteny visu
+00012010: 616c 697a 6174 696f 6e20 7069 7065 6c69  alization pipeli
+00012020: 6e65 2e20 5573 6572 7320 6e65 6564 2074  ne. Users need t
+00012030: 6f20 7072 6f76 6964 6520 7468 6520 666c  o provide the fl
+00012040: 6167 206f 7074 696f 6e20 602d 2d70 6c6f  ag option `--plo
+00012050: 7473 796e 6020 746f 2069 6e69 7469 6174  tsyn` to initiat
+00012060: 6520 7468 6973 2070 6172 7420 6f66 2070  e this part of p
+00012070: 6970 656c 696e 652e 2054 6869 7320 7374  ipeline. This st
+00012080: 6570 2061 7373 756d 6573 2074 6861 7420  ep assumes that 
+00012090: 7573 6572 7320 6861 7665 206f 6274 6169  users have obtai
+000120a0: 6e65 6420 616c 7265 6164 7920 7468 6520  ned already the 
+000120b0: 7379 6e74 656e 6963 2072 6573 756c 7473  syntenic results
+000120c0: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
+000120d0: 2061 6e64 2075 7365 7320 7468 6f73 6520   and uses those 
+000120e0: 7265 7375 6c74 2066 696c 6573 2074 6f20  result files to 
+000120f0: 7265 616c 697a 6520 7468 6520 7379 6e74  realize the synt
+00012100: 656e 7920 7669 7375 616c 697a 6174 696f  eny visualizatio
+00012110: 6e2e 2053 696d 6c69 6172 2074 6f20 6077  n. Simliar to `w
+00012120: 6764 2073 796e 602c 2074 6865 2065 7874  gd syn`, the ext
+00012130: 7261 202a 4b2a 3c73 7562 3e53 3c2f 7375  ra *K*<sub>S</su
+00012140: 623e 2064 6174 6120 6669 6c65 2063 616e  b> data file can
+00012150: 2062 6520 7472 616e 736d 6974 7465 6420   be transmitted 
+00012160: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00012170: 2d2d 6461 7461 6669 6c65 6020 2869 6e73  --datafile` (ins
+00012180: 7465 6164 206f 6620 602d 2d65 7874 7261  tead of `--extra
+00012190: 7061 7261 6e6f 6d65 6b73 6020 6f70 7469  paranomeks` opti
+000121a0: 6f6e 292e 2042 6173 6963 616c 6c79 2c20  on). Basically, 
+000121b0: 7468 6520 7379 6e74 656e 6963 2072 6573  the syntenic res
+000121c0: 756c 7420 6669 6c65 7320 7265 7175 6972  ult files requir
+000121d0: 6564 2061 7265 2074 6865 2061 6e63 686f  ed are the ancho
+000121e0: 7220 706f 696e 7473 2064 6174 6166 696c  r points datafil
+000121f0: 652c 206d 756c 7469 706c 6963 6f6e 7320  e, multiplicons 
+00012200: 6461 7461 6669 6c65 2c20 6765 6e65 2074  datafile, gene t
+00012210: 6162 6c65 2064 6174 6166 696c 6520 2861  able datafile (a
+00012220: 7574 6f6d 6174 656c 7920 7072 6f64 7563  utomately produc
+00012230: 6564 2062 7920 6077 6764 2073 796e 6029  ed by `wgd syn`)
+00012240: 2c20 616e 6420 7365 676d 656e 7473 2064  , and segments d
+00012250: 6174 6166 696c 652e 0a0a 2a2a 4120 7375  atafile...**A su
+00012260: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+00012270: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+00012280: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
+00012290: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
+000122a0: 7669 7a20 2d61 7020 6170 6461 7461 202d  viz -ap apdata -
+000122b0: 736d 2073 6d64 6174 6120 2d6d 7420 6d74  sm smdata -mt mt
+000122c0: 6461 7461 202d 6774 2067 7464 6174 6120  data -gt gtdata 
+000122d0: 2d2d 706c 6f74 7379 6e0a 6060 600a 0a23  --plotsyn.```..#
+000122e0: 2320 496c 6c75 7374 7261 7469 6f6e 0a0a  # Illustration..
+000122f0: 5765 2069 6c6c 7573 7472 6174 6520 6f75  We illustrate ou
+00012300: 7220 7072 6f67 7261 6d20 6f6e 2061 6e20  r program on an 
+00012310: 6578 656d 706c 6172 7920 5747 4420 696e  exemplary WGD in
+00012320: 6665 7265 6e63 6520 616e 6420 6461 7469  ference and dati
+00012330: 6e67 2075 706f 6e20 7370 6563 6965 7320  ng upon species 
+00012340: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
+00012350: 6c65 612a 2e0a 0a54 6865 202a 4171 7569  lea*...The *Aqui
+00012360: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
+00012370: 7761 7320 7265 706f 7274 6564 2074 6f20  was reported to 
+00012380: 6578 7065 7269 656e 6365 2061 6e20 7061  experience an pa
+00012390: 6c65 6f2d 706f 6c79 706c 6f69 6469 7a61  leo-polyploidiza
+000123a0: 7469 6f6e 2065 7665 6e74 2061 6674 6572  tion event after
+000123b0: 2074 6865 2064 6976 6572 6765 6e63 6520   the divergence 
+000123c0: 6f66 2063 6f72 6520 6575 6469 636f 7473  of core eudicots
+000123d0: 2c20 7768 6963 6820 6973 206c 696b 656c  , which is likel
+000123e0: 7920 7368 6172 6564 2062 7920 616c 6c20  y shared by all 
+000123f0: 5261 6e75 6e63 756c 616c 6573 2e0a 0a46  Ranunculales...F
+00012400: 6972 7374 2061 626f 7665 2061 6c6c 2c20  irst above all, 
+00012410: 6c65 7427 7320 6465 6c69 6e65 6174 6520  let's delineate 
+00012420: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
+00012430: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+00012440: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+00012450: 696f 6e20 616e 6420 6861 7665 2061 2062  ion and have a b
+00012460: 6173 6963 206f 6273 6572 7661 7469 6f6e  asic observation
+00012470: 2066 6f72 2070 6f74 656e 7469 616c 6c79   for potentially
+00012480: 2063 6f6e 6365 6976 6162 6c65 2057 4744   conceivable WGD
+00012490: 732c 2075 7369 6e67 2074 6865 2063 6f6d  s, using the com
+000124a0: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
+000124b0: 0a0a 6060 600a 7767 6420 646d 6420 4171  ..```.wgd dmd Aq
+000124c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000124d0: 0a77 6764 206b 7364 2077 6764 5f64 6d64  .wgd ksd wgd_dmd
+000124e0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000124f0: 6c65 612e 7473 7620 4171 7569 6c65 6769  lea.tsv Aquilegi
+00012500: 615f 636f 6572 756c 6561 0a60 6060 0a0a  a_coerulea.```..
+00012510: 5468 6520 636f 6e73 7472 7563 7465 6420  The constructed 
+00012520: 7768 6f6c 6520 7061 7261 6e6f 6d65 202a  whole paranome *
+00012530: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00012540: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00012550: 6f66 202a 4171 7569 6c65 6769 6120 636f  of *Aquilegia co
+00012560: 6572 756c 6561 2a20 6973 2061 7320 6265  erulea* is as be
+00012570: 6c6f 772c 2077 6520 6361 6e20 7365 6520  low, we can see 
+00012580: 7468 6174 2074 6865 7265 2073 6565 6d73  that there seems
+00012590: 2074 6f20 6265 2061 2068 756d 7020 6174   to be a hump at
+000125a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000125b0: 2031 2062 7574 206e 6f74 2063 6c65 6172   1 but not clear
+000125c0: 2e0a 0a21 5b5d 2864 6174 612f 4171 7569  ...![](data/Aqui
+000125d0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000125e0: 7376 2e6b 7364 5f77 702e 7376 6729 0a0a  sv.ksd_wp.svg)..
+000125f0: 5765 2074 6865 6e20 636f 6e73 7472 7563  We then construc
+00012600: 7420 7468 6520 616e 6368 6f72 202a 4b2a  t the anchor *K*
+00012610: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00012620: 2064 6973 7472 6962 7574 696f 6e20 7573   distribution us
+00012630: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00012640: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
+00012650: 0a77 6764 2073 796e 202d 6620 6d52 4e41  .wgd syn -f mRNA
+00012660: 202d 6120 4e61 6d65 2077 6764 5f64 6d64   -a Name wgd_dmd
+00012670: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012680: 6c65 612e 7473 7620 4171 7569 6c65 6769  lea.tsv Aquilegi
+00012690: 615f 636f 6572 756c 6561 2e67 6666 3320  a_coerulea.gff3 
+000126a0: 2d6b 7320 7767 645f 6b73 642f 4171 7569  -ks wgd_ksd/Aqui
+000126b0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000126c0: 7376 2e6b 732e 7473 760a 6060 600a 0a41  sv.ks.tsv.```..A
+000126d0: 7320 7368 6f77 6e20 6265 6c6f 772c 2074  s shown below, t
+000126e0: 6865 7265 2061 7265 2073 6f6d 6520 7265  here are some re
+000126f0: 7461 696e 6564 2061 6e63 686f 7220 7061  tained anchor pa
+00012700: 6972 7320 7769 7468 202a 4b2a 3c73 7562  irs with *K*<sub
+00012710: 3e53 3c2f 7375 623e 2062 6574 7765 656e  >S</sub> between
+00012720: 2031 2061 6e64 2032 2c20 7768 6963 6820   1 and 2, which 
+00012730: 7365 656d 7320 746f 2073 7567 6765 7374  seems to suggest
+00012740: 2061 2057 4744 2065 7665 6e74 2e0a 0a21   a WGD event...!
+00012750: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
+00012760: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00012770: 7364 5f77 705f 6170 2e73 7667 290a 0a54  sd_wp_ap.svg)..T
+00012780: 6865 2061 7373 6f63 6961 7465 6420 6064  he associated `d
+00012790: 7570 5374 6163 6b60 2070 6c6f 7420 7368  upStack` plot sh
+000127a0: 6f77 7320 7468 6174 2074 6865 7265 2061  ows that there a
+000127b0: 7265 206e 756d 6572 6f75 7320 6475 706c  re numerous dupl
+000127c0: 6963 6174 6564 2073 6567 6d65 6e74 7320  icated segments 
+000127d0: 6163 726f 7373 206d 6f73 7420 6f66 2074  across most of t
+000127e0: 6865 2063 6872 6f6d 6f73 6f6d 6573 2e0a  he chromosomes..
+000127f0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
+00012800: 6769 615f 636f 6572 756c 6561 5f41 7175  gia_coerulea_Aqu
+00012810: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00012820: 6d75 6c74 6970 6c69 636f 6e73 5f6c 6576  multiplicons_lev
+00012830: 656c 2e73 7667 290a 0a57 6520 696d 706c  el.svg)..We impl
+00012840: 656d 656e 7465 6420 7477 6f20 7479 7065  emented two type
+00012850: 7320 6f66 2064 6f74 2070 6c6f 7473 2069  s of dot plots i
+00012860: 6e20 6f78 666f 7264 2067 7269 643a 206f  n oxford grid: o
+00012870: 6e65 2069 6e20 7468 6520 756e 6974 206f  ne in the unit o
+00012880: 6620 6261 7365 7320 616e 6420 7468 6520  f bases and the 
+00012890: 6f74 6865 7220 696e 2074 6865 2075 6e69  other in the uni
+000128a0: 7420 6f66 2067 656e 6573 2c20 7768 6963  t of genes, whic
+000128b0: 6820 6361 6e20 6265 2063 6f6c 6f72 6564  h can be colored
+000128c0: 2062 7920 2a4b 2a3c 7375 623e 533c 2f73   by *K*<sub>S</s
+000128d0: 7562 3e20 7661 6c75 6573 2067 6976 656e  ub> values given
+000128e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000128f0: 2064 6174 612e 0a0a 215b 5d28 6461 7461   data...![](data
+00012900: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012910: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+00012920: 5f63 6f65 7275 6c65 615f 4b73 2e64 6f74  _coerulea_Ks.dot
+00012930: 5f75 6e69 745f 6765 6e65 2e70 6e67 290a  _unit_gene.png).
+00012940: 0a41 7320 7368 6f77 6e20 6162 6f76 652c  .As shown above,
+00012950: 2074 6865 2064 6f74 2070 6c6f 7420 696e   the dot plot in
+00012960: 2074 6865 2075 6e69 7420 6f66 2067 656e   the unit of gen
+00012970: 6573 2070 7265 7365 6e74 7320 6e75 6d65  es presents nume
+00012980: 726f 7573 2064 656e 7365 6c79 2061 6767  rous densely agg
+00012990: 7265 6761 7465 6420 286c 696e 652d 6c69  regated (line-li
+000129a0: 6b65 2920 616e 6368 6f72 2070 6f69 6e74  ke) anchor point
+000129b0: 7320 6174 206d 6f73 7420 6f66 2074 6865  s at most of the
+000129c0: 2063 6872 6f6d 6f73 6f6d 6573 2077 6974   chromosomes wit
+000129d0: 6820 636f 6e73 6973 7465 6e74 202a 4b2a  h consistent *K*
+000129e0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+000129f0: 2062 6574 7765 656e 2031 2061 6e64 2032   between 1 and 2
+00012a00: 2e20 5468 6520 646f 7420 706c 6f74 2069  . The dot plot i
+00012a10: 6e20 7468 6520 756e 6974 206f 6620 6261  n the unit of ba
+00012a20: 7365 7320 7368 6f77 7320 7468 6520 7361  ses shows the sa
+00012a30: 6d65 2070 6174 7465 726e 2c20 6173 206d  me pattern, as m
+00012a40: 616e 6966 6573 7465 6420 6265 6c6f 772e  anifested below.
+00012a50: 0a0a 215b 5d28 6461 7461 2f41 7175 696c  ..![](data/Aquil
+00012a60: 6567 6961 5f63 6f65 7275 6c65 612d 7673  egia_coerulea-vs
+00012a70: 2d41 7175 696c 6567 6961 5f63 6f65 7275  -Aquilegia_coeru
+00012a80: 6c65 615f 4b73 2e64 6f74 2e70 6e67 290a  lea_Ks.dot.png).
+00012a90: 0a54 6865 2064 6f74 2070 6c6f 7473 2077  .The dot plots w
+00012aa0: 6974 686f 7574 202a 4b2a 3c73 7562 3e53  ithout *K*<sub>S
+00012ab0: 3c2f 7375 623e 2061 6e6e 6f74 6174 696f  </sub> annotatio
+00012ac0: 6e20 7769 6c6c 2061 6c73 6f20 6265 2061  n will also be a
+00012ad0: 7574 6f6d 6174 656c 7920 7072 6f64 7563  utomately produc
+00012ae0: 6564 2c20 6173 2073 686f 776e 2062 656c  ed, as shown bel
+00012af0: 6f77 2e0a 0a21 5b5d 2864 6174 612f 4171  ow...![](data/Aq
+00012b00: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00012b10: 2d76 732d 4171 7569 6c65 6769 615f 636f  -vs-Aquilegia_co
+00012b20: 6572 756c 6561 2e64 6f74 5f75 6e69 745f  erulea.dot_unit_
+00012b30: 6765 6e65 2e70 6e67 290a 0a21 5b5d 2864  gene.png)..![](d
+00012b40: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
+00012b50: 6572 756c 6561 2d76 732d 4171 7569 6c65  erulea-vs-Aquile
+00012b60: 6769 615f 636f 6572 756c 6561 2e64 6f74  gia_coerulea.dot
+00012b70: 2e70 6e67 290a 0a4e 6f74 6520 7468 6174  .png)..Note that
+00012b80: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
+00012b90: 616e 6368 6f72 2064 6f74 7320 616e 6420  anchor dots and 
+00012ba0: 616c 6c20 686f 6d6f 6c6f 6720 646f 7473  all homolog dots
+00012bb0: 2063 616e 2062 6520 7365 7420 6279 2074   can be set by t
+00012bc0: 6865 206f 7074 696f 6e20 6061 7061 6c70  he option `apalp
+00012bd0: 6861 6020 616e 6420 6068 6f61 6c70 6861  ha` and `hoalpha
+00012be0: 6020 7365 7061 7261 7465 6c79 2e20 4966  ` separately. If
+00012bf0: 206f 6e65 206a 7573 7420 7761 6e74 7320   one just wants 
+00012c00: 746f 2073 6565 2074 6865 2061 6e63 686f  to see the ancho
+00012c10: 7220 646f 7473 2c20 7365 7474 696e 6720  r dots, setting 
+00012c20: 7468 6520 6068 6f61 6c70 6861 6020 6173  the `hoalpha` as
+00012c30: 2030 2028 6f72 206f 7468 6572 206d 696e   0 (or other min
+00012c40: 7573 6375 6c65 2076 616c 7565 7329 2077  uscule values) w
+00012c50: 696c 6c20 646f 2e20 4966 206f 6e65 2077  ill do. If one w
+00012c60: 616e 7473 2074 6f20 7365 6520 7468 6520  ants to see the 
+00012c70: 6469 7374 7269 6275 7469 6f6e 206f 6620  distribution of 
+00012c80: 7768 6f6c 6520 646f 7473 2062 6574 7465  whole dots bette
+00012c90: 722c 2073 6574 7469 6e67 2074 6865 2060  r, setting the `
+00012ca0: 686f 616c 7068 6160 2068 6967 6865 7220  hoalpha` higher 
+00012cb0: 2861 6e64 2060 6170 616c 7068 6160 206c  (and `apalpha` l
+00012cc0: 6f77 6572 2920 7769 6c6c 2064 6f2e 2054  ower) will do. T
+00012cd0: 6865 2060 646f 7473 697a 6560 206f 7074  he `dotsize` opt
+00012ce0: 696f 6e20 6361 6e20 6265 2063 616c 6c65  ion can be calle
+00012cf0: 6420 746f 2061 646a 7573 7420 7468 6520  d to adjust the 
+00012d00: 7369 7a65 206f 6620 646f 7473 2e0a 0a41  size of dots...A
+00012d10: 2066 7572 7468 6572 2061 7373 6f63 6961   further associa
+00012d20: 7465 6420 5379 6e64 6570 7468 2070 6c6f  ted Syndepth plo
+00012d30: 7420 7368 6f77 7320 7468 6174 2074 6865  t shows that the
+00012d40: 7265 2061 7265 206d 6f72 6520 7468 616e  re are more than
+00012d50: 2038 3020 6475 706c 6963 6174 6564 2073   80 duplicated s
+00012d60: 6567 6d65 6e74 732c 2077 6869 6368 2064  egments, which d
+00012d70: 6f6d 696e 6174 6573 2074 6865 2077 686f  ominates the who
+00012d80: 6c65 2063 6f6c 6c69 6e65 6172 2072 6174  le collinear rat
+00012d90: 696f 2063 6174 6567 6f72 792e 0a0a 215b  io category...![
+00012da0: 5d28 6461 7461 2f53 796e 6465 7074 682e  ](data/Syndepth.
+00012db0: 7376 6729 0a0a 5765 2063 616e 2066 6974  svg)..We can fit
+00012dc0: 2061 6e20 454c 4d4d 206d 6978 7475 7265   an ELMM mixture
+00012dd0: 206d 6f64 656c 2075 706f 6e20 7468 6520   model upon the 
+00012de0: 7768 6f6c 6520 7061 7261 6e6f 6d65 202a  whole paranome *
+00012df0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00012e00: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00012e10: 746f 2073 6565 206d 6f72 6520 6163 6375  to see more accu
+00012e20: 7261 7465 6c79 2074 6865 2073 6967 6e69  rately the signi
+00012e30: 6669 6361 6e63 6520 616e 6420 6c6f 6361  ficance and loca
+00012e40: 7469 6f6e 206f 6620 706f 7465 6e74 6961  tion of potentia
+00012e50: 6c20 5747 4473 2c20 7573 696e 6720 7468  l WGDs, using th
+00012e60: 6520 636f 6d6d 616e 6420 6c69 6e65 2062  e command line b
+00012e70: 656c 6f77 2e0a 0a60 6060 0a77 6764 2076  elow...```.wgd v
+00012e80: 697a 202d 6420 7767 645f 6b73 642f 4171  iz -d wgd_ksd/Aq
+00012e90: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00012ea0: 2e74 7376 2e6b 732e 7473 760a 6060 600a  .tsv.ks.tsv.```.
+00012eb0: 0a54 6865 2072 6573 756c 7420 6f66 2045  .The result of E
+00012ec0: 4c4d 4d20 6d69 7874 7572 6520 6d6f 6465  LMM mixture mode
+00012ed0: 6c20 636c 7573 7465 7269 6e67 2073 686f  l clustering sho
+00012ee0: 7773 2074 6861 7420 7468 6572 6520 6973  ws that there is
+00012ef0: 2061 206c 696b 656c 7920 5747 4420 636f   a likely WGD co
+00012f00: 6d70 6f6e 656e 7420 6174 202a 4b2a 3c73  mponent at *K*<s
+00012f10: 7562 3e53 3c2f 7375 623e 2031 2e31 392e  ub>S</sub> 1.19.
+00012f20: 0a0a 215b 5d28 6461 7461 2f65 6c6d 6d5f  ..![](data/elmm_
+00012f30: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00012f40: 6561 2e74 7376 2e6b 732e 7473 765f 6265  ea.tsv.ks.tsv_be
+00012f50: 7374 5f6d 6f64 656c 735f 7765 6967 6874  st_models_weight
+00012f60: 6564 2e73 7667 290a 0a4c 6574 2773 2064  ed.svg)..Let's d
+00012f70: 6f20 6120 6d69 7874 7572 6520 6d6f 6465  o a mixture mode
+00012f80: 6c20 636c 7573 7465 7269 6e67 2066 6f72  l clustering for
+00012f90: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
+00012fa0: 533c 2f73 7562 3e20 746f 6f2c 2075 7369  S</sub> too, usi
+00012fb0: 6e67 2074 6865 2063 6f6d 6d61 6e64 206c  ng the command l
+00012fc0: 696e 6520 6265 6c6f 772e 0a0a 6060 600a  ine below...```.
+00012fd0: 7767 6420 7065 616b 2077 6764 5f6b 7364  wgd peak wgd_ksd
+00012fe0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012ff0: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00013000: 2d61 6e63 686f 7270 6f69 6e74 7320 7767  -anchorpoints wg
+00013010: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00013020: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
+00013030: 7874 202d 2d73 6567 6d65 6e74 7320 7767  xt --segments wg
+00013040: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00013050: 742f 7365 676d 656e 7473 2e74 7874 202d  t/segments.txt -
+00013060: 2d6c 6973 7465 6c65 6d65 6e74 7320 7767  -listelements wg
+00013070: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00013080: 742f 6c69 7374 5f65 6c65 6d65 6e74 732e  t/list_elements.
+00013090: 7478 7420 2d2d 6d75 6c74 6970 6c69 636f  txt --multiplico
+000130a0: 6e70 6169 7273 2077 6764 5f73 796e 2f69  npairs wgd_syn/i
+000130b0: 6164 686f 7265 2d6f 7574 2f6d 756c 7469  adhore-out/multi
+000130c0: 706c 6963 6f6e 5f70 6169 7273 2e74 7874  plicon_pairs.txt
+000130d0: 202d 2d77 6569 6768 7465 640a 6060 600a   --weighted.```.
+000130e0: 0a54 6865 2061 6e63 686f 7220 2a4b 2a3c  .The anchor *K*<
+000130f0: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00013100: 6469 7374 7269 6275 7469 6f6e 2061 6c73  distribution als
+00013110: 6f20 6861 7320 6120 6c69 6b65 6c79 2057  o has a likely W
+00013120: 4744 2063 6f6d 706f 6e65 6e74 2077 6974  GD component wit
+00013130: 6820 6d6f 6465 2031 2e32 382e 0a0a 215b  h mode 1.28...![
+00013140: 5d28 6461 7461 2f4f 7269 6769 6e61 6c5f  ](data/Original_
+00013150: 416e 6368 6f72 4b73 5f47 4d4d 5f43 6f6d  AnchorKs_GMM_Com
+00013160: 706f 6e65 6e74 335f 6e6f 6465 5f77 6569  ponent3_node_wei
+00013170: 6768 7465 645f 4c6f 676e 6f72 6d61 6c2e  ghted_Lognormal.
+00013180: 7376 6729 0a0a 4e6f 7720 7468 6174 2077  svg)..Now that w
+00013190: 6520 6861 7665 2073 6565 6e20 7468 6520  e have seen the 
+000131a0: 6576 6964 656e 6365 206f 6620 6e75 6d65  evidence of nume
+000131b0: 726f 7573 2064 7570 6c69 6361 7465 6420  rous duplicated 
+000131c0: 7365 676d 656e 7473 2061 6e64 2074 6865  segments and the
+000131d0: 2061 6767 7265 6761 7469 6f6e 206f 6620   aggregation of 
+000131e0: 6475 706c 6963 6174 6573 2061 6765 2069  duplicates age i
+000131f0: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+00013200: 3e20 6172 6f75 6e64 2031 2e32 2066 6f72  > around 1.2 for
+00013210: 2061 6e63 686f 7220 7061 6972 7320 616e   anchor pairs an
+00013220: 6420 6e6f 6e2d 616e 6368 6f72 2070 6169  d non-anchor pai
+00013230: 7273 2074 6872 6f75 6768 6f75 7420 7468  rs throughout th
+00013240: 6520 7768 6f6c 6520 6765 6e6f 6d65 2e20  e whole genome. 
+00013250: 5765 2063 616e 2063 6c61 696d 2077 6974  We can claim wit
+00013260: 6820 736f 6d65 2063 6f6e 6669 6465 6e63  h some confidenc
+00013270: 6520 7468 6174 202a 4171 7569 6c65 6769  e that *Aquilegi
+00013280: 6120 636f 6572 756c 6561 2a20 6d69 6768  a coerulea* migh
+00013290: 7420 6861 7665 2065 7870 6572 6965 6e63  t have experienc
+000132a0: 6564 2061 2070 616c 656f 2d70 6f6c 7970  ed a paleo-polyp
+000132b0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+000132c0: 742e 204e 6578 742c 204c 6574 2773 2068  t. Next, Let's h
+000132d0: 6176 6520 6120 6675 7274 6865 7220 6c6f  ave a further lo
+000132e0: 6f6b 2061 626f 7574 2069 7473 2070 6879  ok about its phy
+000132f0: 6c6f 6765 6e65 7469 6320 6c6f 6361 7469  logenetic locati
+00013300: 6f6e 2e20 5765 206b 6e6f 7720 7468 6174  on. We know that
+00013310: 2074 6865 7265 2061 7265 2075 6e63 6572   there are uncer
+00013320: 7461 696e 7469 6573 2061 626f 7574 2077  tainties about w
+00013330: 6865 7468 6572 2074 6869 7320 7075 7461  hether this puta
+00013340: 7469 7665 2070 616c 656f 2d70 6f6c 7970  tive paleo-polyp
+00013350: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+00013360: 7420 6973 2073 6861 7265 6420 7769 7468  t is shared with
+00013370: 2061 6c6c 2065 7564 6963 6f74 7320 6f72   all eudicots or
+00013380: 206e 6f74 2e20 5765 2063 616e 2063 686f   not. We can cho
+00013390: 6f73 6520 736f 6d65 206f 7468 6572 2065  ose some other e
+000133a0: 7564 6963 6f74 2067 656e 6f6d 6573 2074  udicot genomes t
+000133b0: 6f20 7365 6520 7468 6520 6f72 6465 7269  o see the orderi
+000133c0: 6e67 206f 6620 7370 6563 6961 7469 6f6e  ng of speciation
+000133d0: 2061 6e64 2070 6f6c 7970 6c6f 6964 697a   and polyploidiz
+000133e0: 6174 696f 6e20 6576 656e 7473 2e20 4865  ation events. He
+000133f0: 7265 2077 6520 6368 6f6f 7365 202a 5669  re we choose *Vi
+00013400: 7469 7320 7669 6e69 6665 7261 2a2c 202a  tis vinifera*, *
+00013410: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
+00013420: 732a 2061 6e64 202a 4163 6f72 7573 2061  s* and *Acorus a
+00013430: 6d65 7269 6361 6e75 732a 2069 6e20 7468  mericanus* in th
+00013440: 6520 666f 6c6c 6f77 696e 6720 2a4b 2a3c  e following *K*<
+00013450: 7375 623e 533c 2f73 7562 3e20 616e 616c  sub>S</sub> anal
+00013460: 7973 6973 2e20 4669 7273 742c 2077 6520  ysis. First, we 
+00013470: 6275 696c 7420 6120 676c 6f62 616c 204d  built a global M
+00013480: 5242 4820 6661 6d69 6c79 2075 7369 6e67  RBH family using
+00013490: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+000134a0: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+000134b0: 202d 2d67 6c6f 6261 6c6d 7262 6820 4171   --globalmrbh Aq
+000134c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000134d0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+000134e0: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
+000134f0: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
+00013500: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
+00013510: 6c6d 7262 680a 6060 600a 0a49 6e20 7468  lmrbh.```..In th
+00013520: 6520 676c 6f62 616c 204d 5242 4820 6661  e global MRBH fa
+00013530: 6d69 6c79 2c20 6576 6572 7920 7061 6972  mily, every pair
+00013540: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+00013550: 6765 6e65 7320 6973 2074 6865 2072 6563  genes is the rec
+00013560: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
+00013570: 2c20 7375 6767 6573 7469 6e67 2074 7275  , suggesting tru
+00013580: 6520 6f72 7468 6f6c 6f67 6f75 7320 7265  e orthologous re
+00013590: 6c61 7469 6f6e 7368 6970 732e 2057 6520  lationships. We 
+000135a0: 776f 756c 6420 7573 6520 7468 6520 2a4b  would use the *K
+000135b0: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+000135c0: 6c75 6573 2061 7373 6f63 6961 7465 6420  lues associated 
+000135d0: 7769 7468 2074 6865 7365 206f 7274 686f  with these ortho
+000135e0: 6c6f 676f 7573 2070 6169 7273 2074 6f20  logous pairs to 
+000135f0: 6465 6c69 6d69 7420 7468 6520 6469 7665  delimit the dive
+00013600: 7267 656e 6365 202a 4b2a 3c73 7562 3e53  rgence *K*<sub>S
+00013610: 3c2f 7375 623e 2070 6561 6b2e 2054 6f67  </sub> peak. Tog
+00013620: 6574 6865 7220 7769 7468 2074 6865 2077  ether with the w
+00013630: 686f 6c65 2070 6172 616e 6f6d 6520 2a4b  hole paranome *K
+00013640: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00013650: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
+00013660: 6f6e 6475 6374 2074 6865 2072 6174 6520  onduct the rate 
+00013670: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
+00013680: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00013690: 6f77 2e0a 0a21 2153 696e 6365 2060 7767  ow...!!Since `wg
+000136a0: 6460 2076 6572 7369 6f6e 2032 2e30 2e32  d` version 2.0.2
+000136b0: 342c 2077 6520 7265 7772 6f74 6520 6120  4, we rewrote a 
+000136c0: 636c 6561 6e65 7220 616e 6420 7175 6963  cleaner and quic
+000136d0: 6b65 7220 7761 7920 6f66 2064 6f69 6e67  ker way of doing
+000136e0: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
+000136f0: 7465 2063 6f72 7265 6374 696f 6e2e 2049  te correction. I
+00013700: 7427 7320 6e6f 7420 7265 7175 6972 6564  t's not required
+00013710: 2074 6f20 7479 7065 2069 6e20 616e 7920   to type in any 
+00013720: 7370 6569 6365 7320 7061 6972 2061 6e64  speices pair and
+00013730: 2061 2073 6572 6965 7320 6f66 202a 4b2a   a series of *K*
+00013740: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
+00013750: 7473 2077 696c 6c20 6265 2070 726f 6475  ts will be produ
+00013760: 6365 642e 2054 6865 2072 6571 7569 7265  ced. The require
+00013770: 6420 6669 6c65 7320 6172 6520 6f72 7468  d files are orth
+00013780: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00013790: 533c 2f73 7562 3e20 7461 626c 652c 2070  S</sub> table, p
+000137a0: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
+000137b0: 623e 533c 2f73 7562 3e20 7461 626c 652c  b>S</sub> table,
+000137c0: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
+000137d0: 616e 6420 6120 666f 6375 7365 6420 7370  and a focused sp
+000137e0: 6563 6965 7320 2874 6865 206f 6e65 2069  ecies (the one i
+000137f0: 6e70 7574 7465 6420 7769 7468 2070 6172  nputted with par
+00013800: 616c 6f67 6f75 7320 2a4b 2a3c 7375 623e  alogous *K*<sub>
+00013810: 533c 2f73 7562 3e20 6461 7461 292e 2055  S</sub> data). U
+00013820: 7365 7273 2063 616e 2063 686f 6f73 6520  sers can choose 
+00013830: 746f 2061 6464 206f 6e65 206d 6f72 6520  to add one more 
+00013840: 6c61 7965 7220 6f66 2065 6c6d 6d20 6d6f  layer of elmm mo
+00013850: 6465 6c69 6e67 206f 6e20 7061 7261 6c6f  deling on paralo
+00013860: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+00013870: 7375 623e 2076 616c 7565 7320 616e 642f  sub> values and/
+00013880: 6f72 2067 6d6d 206d 6f64 656c 696e 6720  or gmm modeling 
+00013890: 6f6e 2061 6e63 686f 7220 2a4b 2a3c 7375  on anchor *K*<su
+000138a0: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
+000138b0: 2e20 5468 6520 6f72 7468 6f6c 6f67 6f75  . The orthologou
+000138c0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000138d0: 3e20 7661 6c75 6573 2063 616e 2062 6520  > values can be 
+000138e0: 6361 6c63 756c 6174 6564 2075 7369 6e67  calculated using
+000138f0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00013900: 6f77 2e0a 0a60 6060 0a77 6764 206b 7364  ow...```.wgd ksd
+00013910: 2077 6764 5f67 6c6f 6261 6c6d 7262 682f   wgd_globalmrbh/
+00013920: 676c 6f62 616c 5f4d 5242 482e 7473 7620  global_MRBH.tsv 
+00013930: 7365 7173 2a20 2d6f 2077 6764 5f67 6c6f  seqs* -o wgd_glo
+00013940: 6261 6c6d 7262 685f 6b73 0a60 6060 0a0a  balmrbh_ks.```..
+00013950: 5769 7468 2074 6865 2063 616c 6375 6c61  With the calcula
+00013960: 7465 6420 6f72 7468 6f6c 6f67 6f75 7320  ted orthologous 
+00013970: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013980: 7461 626c 652c 2077 6520 6361 6e20 7573  table, we can us
+00013990: 6520 7468 6520 636f 6d6d 616e 6420 6265  e the command be
+000139a0: 6c6f 7720 746f 2063 6f6e 6475 6374 2074  low to conduct t
+000139b0: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+000139c0: 6f6e 2061 6e64 2f6f 7220 6d69 7874 7572  on and/or mixtur
+000139d0: 6520 6d6f 6465 6c69 6e67 2061 6e61 6c79  e modeling analy
+000139e0: 7369 732e 0a0a 6060 600a 7767 6420 7669  sis...```.wgd vi
+000139f0: 7a20 2d64 2077 6764 5f67 6c6f 6261 6c6d  z -d wgd_globalm
+00013a00: 7262 685f 6b73 2f67 6c6f 6261 6c5f 4d52  rbh_ks/global_MR
+00013a10: 4248 2e74 7376 2e6b 732e 7473 7620 2d66  BH.tsv.ks.tsv -f
+00013a20: 6120 4171 7569 6c65 6769 615f 636f 6572  a Aquilegia_coer
+00013a30: 756c 6561 202d 6570 6b20 4171 7569 6c65  ulea -epk Aquile
+00013a40: 6769 615f 636f 6572 756c 6561 2e6b 732e  gia_coerulea.ks.
+00013a50: 7473 7620 2d61 7020 616e 6368 6f72 706f  tsv -ap anchorpo
+00013a60: 696e 7473 2e74 7874 202d 7370 2073 7065  ints.txt -sp spe
+00013a70: 6369 6573 7472 6565 2e6e 7720 2d6f 2077  ciestree.nw -o w
+00013a80: 6764 5f76 697a 5f6d 6978 6564 5f4b 7320  gd_viz_mixed_Ks 
+00013a90: 2d2d 706c 6f74 656c 6d6d 202d 2d70 6c6f  --plotelmm --plo
+00013aa0: 7461 7067 6d6d 0a60 6060 0a0a 6060 600a  tapgmm.```..```.
+00013ab0: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
+00013ac0: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
+00013ad0: 4248 2e74 7376 2073 6571 732a 202d 2d65  BH.tsv seqs* --e
+00013ae0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
+00013af0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00013b00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00013b10: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00013b20: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00013b30: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
+00013b40: 6d72 6268 5f6b 7320 2d2d 7370 6169 7220  mrbh_ks --spair 
+00013b50: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
+00013b60: 6c65 613b 5072 6f74 6561 5f63 796e 6172  lea;Protea_cynar
+00013b70: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+00013b80: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013b90: 6561 3b56 6974 6973 5f76 696e 6966 6572  ea;Vitis_vinifer
+00013ba0: 6122 202d 2d73 7061 6972 2022 4171 7569  a" --spair "Aqui
+00013bb0: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
+00013bc0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00013bd0: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+00013be0: 6567 6961 5f63 6f65 7275 6c65 613b 4171  egia_coerulea;Aq
+00013bf0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00013c00: 2220 2d2d 706c 6f74 6b64 6520 282d 6170  " --plotkde (-ap
+00013c10: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00013c20: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+00013c30: 732e 7478 7429 0a60 6060 0a0a 5468 6520  s.txt).```..The 
+00013c40: 6669 6c65 2060 7370 6563 6965 7374 7265  file `speciestre
+00013c50: 652e 6e77 6020 6973 2074 6865 2074 6578  e.nw` is the tex
+00013c60: 7420 6669 6c65 206f 6620 7370 6563 6965  t file of specie
+00013c70: 7320 7472 6565 2069 6e20 6e65 7769 636b  s tree in newick
+00013c80: 2074 6861 7420 7261 7465 2063 6f72 7265   that rate corre
+00013c90: 6374 696f 6e20 776f 756c 6420 6265 2063  ction would be c
+00013ca0: 6f6e 6475 6374 6564 206f 6e2e 2049 7473  onducted on. Its
+00013cb0: 2063 6f6e 7465 6e74 2069 7320 6173 2062   content is as b
+00013cc0: 656c 6f77 2e20 5573 6572 7320 6e65 6564  elow. Users need
+00013cd0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00013ce0: 7370 6563 6965 7320 7061 6972 7320 746f  species pairs to
+00013cf0: 2062 6520 706c 6f74 7465 642e 2057 6520   be plotted. We 
+00013d00: 7375 6767 6573 7420 6164 6469 6e67 2074  suggest adding t
+00013d10: 6865 206f 7074 696f 6e20 602d 2d72 6577  he option `--rew
+00013d20: 6569 6768 7460 2074 6f20 7265 6361 6c63  eight` to recalc
+00013d30: 756c 6174 6520 7468 6520 7765 6967 6874  ulate the weight
+00013d40: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
+00013d50: 7220 7375 6368 2074 6861 7420 7468 6520  r such that the 
+00013d60: 7765 6967 6874 206f 6620 6f72 7468 6f6c  weight of orthol
+00013d70: 6f67 6f75 7320 6765 6e65 2070 6169 7273  ogous gene pairs
+00013d80: 2077 696c 6c20 6265 636f 6d65 2031 2061   will become 1 a
+00013d90: 7320 7468 6520 7061 7261 6c6f 676f 7573  s the paralogous
+00013da0: 2067 656e 6520 7061 6972 732e 2054 6865   gene pairs. The
+00013db0: 2066 6c61 6720 602d 2d70 6c6f 746b 6465   flag `--plotkde
+00013dc0: 6020 6361 6e20 6265 2061 6464 6564 2077  ` can be added w
+00013dd0: 6865 6e20 7468 6520 6b64 6520 6375 7276  hen the kde curv
+00013de0: 6520 6f66 206f 7274 686f 6c6f 676f 7573  e of orthologous
+00013df0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00013e00: 2069 7320 6465 7369 7265 642e 2045 7874   is desired. Ext
+00013e10: 7261 2063 6f6c 6c69 6e65 6172 2064 6174  ra collinear dat
+00013e20: 6120 6361 6e20 6265 2061 6464 6564 2062  a can be added b
+00013e30: 7920 7468 6520 6f70 7469 6f6e 2060 2d61  y the option `-a
+00013e40: 7060 2e0a 0a60 6060 0a28 2828 5669 7469  p`...```.(((Viti
+00013e50: 735f 7669 6e69 6665 7261 2c50 726f 7465  s_vinifera,Prote
+00013e60: 615f 6379 6e61 726f 6964 6573 292c 4171  a_cynaroides),Aq
+00013e70: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00013e80: 292c 4163 6f72 7573 5f61 6d65 7269 6361  ),Acorus_america
+00013e90: 6e75 7329 3b0a 6060 600a 0a21 5b5d 2864  nus);.```..![](d
+00013ea0: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
+00013eb0: 6572 756c 6561 5f47 6c6f 6261 6c6d 7262  erulea_Globalmrb
+00013ec0: 684b 735f 436f 7272 6563 7465 642e 6b73  hKs_Corrected.ks
+00013ed0: 642e 7376 6729 0a0a 4173 2073 686f 776e  d.svg)..As shown
+00013ee0: 2061 626f 7665 2c20 6265 6361 7573 6520   above, because 
+00013ef0: 6f66 2074 6865 2068 6967 6865 7220 7375  of the higher su
+00013f00: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
+00013f10: 6f66 202a 4171 7569 6c65 6769 6120 636f  of *Aquilegia co
+00013f20: 6572 756c 6561 2a2c 2074 6865 206f 7269  erulea*, the ori
+00013f30: 6769 6e61 6c20 6f72 7468 6f6c 6f67 6f75  ginal orthologou
+00013f40: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+00013f50: 3e20 7661 6c75 6573 2077 6572 6520 6163  > values were ac
+00013f60: 7475 616c 6c79 2075 6e64 6572 6573 7469  tually underesti
+00013f70: 6d61 7465 6420 696e 2074 6865 2074 696d  mated in the tim
+00013f80: 652d 6672 616d 6520 6f66 202a 4171 7569  e-frame of *Aqui
+00013f90: 6c65 6769 6120 636f 6572 756c 6561 2a2e  legia coerulea*.
+00013fa0: 2057 6865 6e20 7765 2072 6563 6f76 6572   When we recover
+00013fb0: 6564 2074 6865 2064 6976 6572 6765 6e63  ed the divergenc
+00013fc0: 6520 7375 6273 7469 7475 7469 6f6e 2064  e substitution d
+00013fd0: 6973 7461 6e63 6520 696e 2074 6572 6d73  istance in terms
+00013fe0: 206f 6620 7477 6f20 7469 6d65 7320 6f66   of two times of
+00013ff0: 2074 6865 2062 7261 6e63 682d 7370 6563   the branch-spec
+00014000: 6966 6963 2063 6f6e 7472 6962 7574 696f  ific contributio
+00014010: 6e20 6f66 202a 4171 7569 6c65 6769 6120  n of *Aquilegia 
+00014020: 636f 6572 756c 6561 2a20 7369 6e63 6520  coerulea* since 
+00014030: 6974 7320 6469 7665 7267 656e 6365 2077  its divergence w
+00014040: 6974 6820 7468 6520 7369 7374 6572 2073  ith the sister s
+00014050: 7065 6369 6573 2070 6c75 7320 7468 6520  pecies plus the 
+00014060: 7368 6172 6564 2073 7562 7374 6974 7574  shared substitut
+00014070: 696f 6e20 6469 7374 616e 6365 2062 6566  ion distance bef
+00014080: 6f72 6520 6469 7665 7267 656e 6365 2028  ore divergence (
+00014090: 696e 2072 656c 6174 6976 6520 746f 2074  in relative to t
+000140a0: 6865 206f 7574 6772 6f75 7029 2c20 7468  he outgroup), th
+000140b0: 6520 636f 7272 6563 7465 6420 2a4b 2a3c  e corrected *K*<
+000140c0: 7375 623e 533c 2f73 7562 3e20 6d6f 6465  sub>S</sub> mode
+000140d0: 2062 6563 616d 6520 6c61 7267 6572 2e0a   became larger..
+000140e0: 0a49 6620 6f6e 6520 6861 6420 7468 6520  .If one had the 
+000140f0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
+00014100: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+00014110: 2061 6c72 6561 6479 2c20 6f6e 6520 636f   already, one co
+00014120: 756c 6420 616c 736f 2061 7070 6c79 2074  uld also apply t
+00014130: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+00014140: 7669 7a60 2074 6f20 636f 6e64 7563 7420  viz` to conduct 
+00014150: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+00014160: 696f 6e20 616e 616c 7973 6973 2075 7369  ion analysis usi
+00014170: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00014180: 656c 6f77 2e20 4e6f 7465 2074 6861 7420  elow. Note that 
+00014190: 7468 6520 6f72 6465 7220 6f66 2067 6976  the order of giv
+000141a0: 656e 2060 7370 6169 7260 206f 7074 696f  en `spair` optio
+000141b0: 6e73 2064 6563 6964 6573 2074 6865 2063  ns decides the c
+000141c0: 6f6c 6f72 206f 6620 7468 6520 2a4b 2a3c  olor of the *K*<
+000141d0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+000141e0: 7269 6275 7469 6f6e 206f 6620 6561 6368  ribution of each
+000141f0: 2073 7065 6369 6573 2070 6169 722e 2054   species pair. T
+00014200: 6865 2061 6464 6974 696f 6e61 6c20 6f70  he additional op
+00014210: 7469 6f6e 2060 666f 6375 7332 616c 6c60  tion `focus2all`
+00014220: 2069 6e20 6077 6764 206b 7364 6020 616e   in `wgd ksd` an
+00014230: 6420 6077 6764 2076 697a 6020 6361 6e20  d `wgd viz` can 
+00014240: 6265 2075 7365 6420 746f 2074 656c 6c20  be used to tell 
+00014250: 7468 6520 7072 6f67 7261 6d20 7468 6174  the program that
+00014260: 2074 6865 2073 7065 6369 6573 2070 6169   the species pai
+00014270: 7273 2061 7265 2062 6574 7765 656e 2074  rs are between t
+00014280: 6865 2066 6f63 7573 2073 7065 6369 6573  he focus species
+00014290: 2061 6e64 2061 6c6c 2074 6865 206f 7468   and all the oth
+000142a0: 6572 2073 7065 6369 6573 2073 7563 6820  er species such 
+000142b0: 7468 6174 2075 7365 7273 2064 6f6e 2774  that users don't
+000142c0: 206e 6565 6420 746f 2074 7970 6520 696e   need to type in
+000142d0: 2065 6163 6820 7061 6972 2069 6e64 6976   each pair indiv
+000142e0: 6964 7561 6c6c 792e 0a0a 6060 600a 7767  idually...```.wg
+000142f0: 6420 7669 7a20 2d64 2077 6764 5f67 6c6f  d viz -d wgd_glo
+00014300: 6261 6c6d 7262 685f 6b73 2f67 6c6f 6261  balmrbh_ks/globa
+00014310: 6c5f 4d52 4248 2e74 7376 2e6b 732e 7473  l_MRBH.tsv.ks.ts
+00014320: 7620 2d2d 6578 7472 6170 6172 616e 6f6d  v --extraparanom
+00014330: 656b 7320 7767 645f 6b73 642f 4171 7569  eks wgd_ksd/Aqui
+00014340: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00014350: 7376 2e6b 732e 7473 7620 2d73 7020 7370  sv.ks.tsv -sp sp
+00014360: 6563 6965 7374 7265 652e 6e77 202d 2d72  eciestree.nw --r
+00014370: 6577 6569 6768 7420 2d61 7020 7767 645f  eweight -ap wgd_
+00014380: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00014390: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
+000143a0: 202d 6f20 7767 645f 7669 7a5f 6d69 7865   -o wgd_viz_mixe
+000143b0: 645f 4b73 202d 2d73 7061 6972 2022 4171  d_Ks --spair "Aq
+000143c0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000143d0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
+000143e0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
+000143f0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
+00014400: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
+00014410: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00014420: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
+00014430: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
+00014440: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00014450: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
+00014460: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
+00014470: 2d70 6c6f 746b 6465 2028 2d2d 6773 6d61  -plotkde (--gsma
+00014480: 7020 6765 6e65 5f73 7065 6369 6573 2e6d  p gene_species.m
+00014490: 6170 290a 6060 600a 0a57 6974 6820 7468  ap).```..With th
+000144a0: 6520 6f70 7469 6f6e 2060 666f 6375 7332  e option `focus2
+000144b0: 616c 6c60 2c20 7468 6520 636f 6d6d 616e  all`, the comman
+000144c0: 6420 6361 6e20 6265 2061 6c73 6f20 6c69  d can be also li
+000144d0: 6b65 2074 6869 732c 2073 616d 6520 7769  ke this, same wi
+000144e0: 7468 2060 7767 6420 6b73 6460 2e0a 0a60  th `wgd ksd`...`
+000144f0: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
+00014500: 645f 676c 6f62 616c 6d72 6268 5f6b 732f  d_globalmrbh_ks/
+00014510: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
+00014520: 6b73 2e74 7376 202d 2d65 7874 7261 7061  ks.tsv --extrapa
+00014530: 7261 6e6f 6d65 6b73 2077 6764 5f6b 7364  ranomeks wgd_ksd
+00014540: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00014550: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00014560: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
+00014570: 7720 2d2d 7265 7765 6967 6874 202d 6170  w --reweight -ap
+00014580: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00014590: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+000145a0: 732e 7478 7420 2d6f 2077 6764 5f76 697a  s.txt -o wgd_viz
+000145b0: 5f6d 6978 6564 5f4b 7320 2d2d 666f 6375  _mixed_Ks --focu
+000145c0: 7332 616c 6c20 4171 7569 6c65 6769 615f  s2all Aquilegia_
+000145d0: 636f 6572 756c 6561 202d 2d70 6c6f 746b  coerulea --plotk
+000145e0: 6465 0a60 6060 0a0a 4e6f 7465 2074 6861  de.```..Note tha
+000145f0: 7420 7765 2063 616e 2065 6173 696c 7920  t we can easily 
+00014600: 7368 6f77 2074 6861 7420 2a41 7175 696c  show that *Aquil
+00014610: 6567 6961 2063 6f65 7275 6c65 612a 2068  egia coerulea* h
+00014620: 6173 2068 6967 6865 7220 7375 6273 7469  as higher substi
+00014630: 7475 7469 6f6e 2072 6174 6520 7468 616e  tution rate than
+00014640: 202a 5072 6f74 6561 2063 796e 6172 6f69   *Protea cynaroi
+00014650: 6465 732a 2061 6e64 202a 5669 7469 7320  des* and *Vitis 
+00014660: 7669 6e69 6665 7261 2a20 6279 2063 6f6d  vinifera* by com
+00014670: 7061 7269 6e67 2074 6865 6972 2073 7562  paring their sub
+00014680: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
+00014690: 6365 2069 6e20 7265 6761 7264 2074 6f20  ce in regard to 
+000146a0: 7468 6520 7361 6d65 2064 6976 6572 6765  the same diverge
+000146b0: 6e63 6520 6576 656e 7420 7769 7468 206f  nce event with o
+000146c0: 7574 6772 6f75 7020 7370 6563 6965 7320  utgroup species 
+000146d0: 2a41 636f 7275 735f 616d 6572 6963 616e  *Acorus_american
+000146e0: 7573 2a2c 2075 7369 6e67 2063 6f6d 6d61  us*, using comma
+000146f0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
+00014700: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00014710: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00014720: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00014730: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00014740: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00014750: 202d 6f20 7767 645f 7669 7a5f 436f 6d70   -o wgd_viz_Comp
+00014760: 6172 655f 7261 7465 202d 2d73 7061 6972  are_rate --spair
+00014770: 2022 4163 6f72 7573 5f61 6d65 7269 6361   "Acorus_america
+00014780: 6e75 733b 5072 6f74 6561 5f63 796e 6172  nus;Protea_cynar
+00014790: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+000147a0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000147b0: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
+000147c0: 616e 7573 2220 2d2d 7370 6169 7220 2256  anus" --spair "V
+000147d0: 6974 6973 5f76 696e 6966 6572 613b 4163  itis_vinifera;Ac
+000147e0: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
+000147f0: 202d 2d67 736d 6170 2067 656e 655f 7370   --gsmap gene_sp
+00014800: 6563 6965 732e 6d61 7020 2d2d 706c 6f74  ecies.map --plot
+00014810: 6b64 650a 6060 600a 0a21 5b5d 2864 6174  kde.```..![](dat
+00014820: 612f 5261 775f 4f72 7468 6f6c 6f67 7565  a/Raw_Orthologue
+00014830: 735f 436f 6d70 6172 655f 7261 7465 2e6b  s_Compare_rate.k
+00014840: 7364 2e73 7667 290a 0a41 7320 6469 7370  sd.svg)..As disp
+00014850: 6c61 7965 6420 6162 6f76 652c 2074 6865  layed above, the
+00014860: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
+00014870: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
+00014880: 7565 7320 6265 7774 6565 6e20 2a41 7175  ues bewteen *Aqu
+00014890: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+000148a0: 2061 6e64 202a 4163 6f72 7573 2061 6d65   and *Acorus ame
+000148b0: 7269 6361 6e75 732a 2068 6173 2074 6865  ricanus* has the
+000148c0: 2068 6967 6865 7374 206d 6f64 652c 2069   highest mode, i
+000148d0: 6e64 6963 6174 696e 6774 6865 2066 6173  ndicatingthe fas
+000148e0: 7465 7220 7375 6273 7469 7475 7469 6f6e  ter substitution
+000148f0: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
+00014900: 6769 6120 636f 6572 756c 6561 2a20 636f  gia coerulea* co
+00014910: 6d70 6172 6564 2074 6f20 2a50 726f 7465  mpared to *Prote
+00014920: 6120 6379 6e61 726f 6964 6573 2a20 616e  a cynaroides* an
+00014930: 6420 2a56 6974 6973 2076 696e 6966 6572  d *Vitis vinifer
+00014940: 612a 2e0a 0a42 6566 6f72 6520 7632 2e30  a*...Before v2.0
+00014950: 2e32 312c 2074 6865 2067 656e 652d 7370  .21, the gene-sp
+00014960: 6563 6965 7320 6d61 7020 6669 6c65 2069  ecies map file i
+00014970: 7320 6e65 6363 6573 7361 7269 6c79 206e  s neccessarily n
+00014980: 6565 6465 6420 666f 7220 6974 7320 696d  eeded for its im
+00014990: 706c 656d 656e 7461 7469 6f6e 2069 6e20  plementation in 
+000149a0: 6077 6764 2076 697a 602c 2077 6869 6368  `wgd viz`, which
+000149b0: 2073 686f 756c 6420 6265 2061 7574 6f6d   should be autom
+000149c0: 6174 656c 7920 7072 6f64 7563 6564 2062  ately produced b
+000149d0: 7920 7468 6520 6c61 7374 2060 7767 6420  y the last `wgd 
+000149e0: 6b73 6460 2073 7465 7020 6769 7665 6e20  ksd` step given 
+000149f0: 7468 6520 6073 7061 6972 6020 616e 6420  the `spair` and 
+00014a00: 6073 7065 6369 6573 7472 6565 6020 7061  `speciestree` pa
+00014a10: 7261 6d65 7465 7273 2e20 5468 6520 6067  rameters. The `g
+00014a20: 656e 655f 7370 6563 6965 732e 6d61 7060  ene_species.map`
+00014a30: 2068 6173 2063 6f6e 7465 6e74 7320 6173   has contents as
+00014a40: 2062 656c 6f77 2069 6e20 7768 6963 6820   below in which 
+00014a50: 6561 6368 206c 696e 6520 6973 2074 6865  each line is the
+00014a60: 206a 6f69 6e65 6420 7374 7269 6e67 206f   joined string o
+00014a70: 6620 6765 6e65 206e 616d 6520 616e 6420  f gene name and 
+00014a80: 7370 6563 6965 7320 6e61 6d65 2062 7920  species name by 
+00014a90: 7370 6163 652e 2041 6674 6572 2076 322e  space. After v2.
+00014aa0: 302e 3231 2028 696e 636c 7564 6564 292c  0.21 (included),
+00014ab0: 2074 6865 2067 656e 652d 7370 6563 6965   the gene-specie
+00014ac0: 7320 6d61 7020 6669 6c65 2069 7320 6e6f  s map file is no
+00014ad0: 7420 6e65 6363 6573 7361 7269 6c79 206e  t neccessarily n
+00014ae0: 6565 6465 6420 616e 796d 6f72 652e 0a0a  eeded anymore...
+00014af0: 6060 600a 4171 636f 6536 4730 3537 3830  ```.Aqcoe6G05780
+00014b00: 302e 3120 4171 7569 6c65 6769 615f 636f  0.1 Aquilegia_co
+00014b10: 6572 756c 6561 0a56 7669 5f56 4954 5f32  erulea.Vvi_VIT_2
+00014b20: 3031 7330 3031 3167 3031 3533 302e 3120  01s0011g01530.1 
+00014b30: 5669 7469 735f 7669 6e69 6665 7261 0a50  Vitis_vinifera.P
+00014b40: 6379 5f50 726f 6379 3031 6730 3835 3130  cy_Procy01g08510
+00014b50: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+00014b60: 6573 0a41 616d 5f41 636f 7261 2e30 3447  es.Aam_Acora.04G
+00014b70: 3134 3239 3030 2e31 2041 636f 7275 735f  142900.1 Acorus_
+00014b80: 616d 6572 6963 616e 7573 0a60 6060 0a0a  americanus.```..
+00014b90: 4120 6d6f 7265 2063 6f6d 706c 6578 2070  A more complex p
+00014ba0: 6c6f 7420 6361 6e20 6265 206d 6164 6520  lot can be made 
+00014bb0: 6279 2061 6464 2074 6865 2066 6c61 6720  by add the flag 
+00014bc0: 602d 2d70 6c6f 7465 6c6d 6d60 2073 7563  `--plotelmm` suc
+00014bd0: 6820 7468 6174 2074 6865 2045 4c4d 4d20  h that the ELMM 
+00014be0: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+00014bf0: 206f 6620 7072 6f76 6964 6564 2070 6172   of provided par
+00014c00: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00014c10: 2f73 7562 3e20 6361 6e20 6265 2073 7570  /sub> can be sup
+00014c20: 6572 696d 706f 7365 642c 2075 7369 6e67  erimposed, using
+00014c30: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00014c40: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
+00014c50: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00014c60: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+00014c70: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
+00014c80: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
+00014c90: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00014ca0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00014cb0: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00014cc0: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00014cd0: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
+00014ce0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00014cf0: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
+00014d00: 6764 5f76 697a 5f6d 6978 6564 5f4b 735f  gd_viz_mixed_Ks_
+00014d10: 656c 6d6d 202d 2d73 7061 6972 2022 4171  elmm --spair "Aq
+00014d20: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014d30: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
+00014d40: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
+00014d50: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
+00014d60: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
+00014d70: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00014d80: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
+00014d90: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
+00014da0: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00014db0: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
+00014dc0: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
+00014dd0: 2d67 736d 6170 2067 656e 655f 7370 6563  -gsmap gene_spec
+00014de0: 6965 732e 6d61 7020 2d2d 706c 6f74 6b64  ies.map --plotkd
+00014df0: 6520 2d2d 706c 6f74 656c 6d6d 0a60 6060  e --plotelmm.```
+00014e00: 0a0a 215b 5d28 6461 7461 2f41 7175 696c  ..![](data/Aquil
+00014e10: 6567 6961 5f63 6f65 7275 6c65 615f 476c  egia_coerulea_Gl
+00014e20: 6f62 616c 6d72 6268 4b73 5f45 6c6d 6d5f  obalmrbhKs_Elmm_
+00014e30: 436f 7272 6563 7465 642e 6b73 642e 7376  Corrected.ksd.sv
+00014e40: 6729 0a0a 4672 6f6d 2074 6865 206d 6978  g)..From the mix
+00014e50: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
+00014e60: 623e 2070 6c6f 7420 6162 6f76 652c 2077  b> plot above, w
+00014e70: 6520 6361 6e20 7365 6520 7468 6174 2074  e can see that t
+00014e80: 6865 206f 7074 696d 697a 6564 206c 6f67  he optimized log
+00014e90: 6e6f 726d 616c 2063 6f6d 706f 6e65 6e74  normal component
+00014ea0: 2062 2077 6974 6820 6d6f 6465 2031 2e32   b with mode 1.2
+00014eb0: 2069 7320 796f 756e 6765 7220 7468 616e   is younger than
+00014ec0: 2074 6865 2063 6f72 7265 6374 6564 206f   the corrected o
+00014ed0: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00014ee0: 7562 3e53 3c2f 7375 623e 206d 6f64 6520  ub>S</sub> mode 
+00014ef0: 7769 7468 202a 5072 6f74 6561 2063 796e  with *Protea cyn
+00014f00: 6172 6f69 6465 732a 2061 6e64 202a 5669  aroides* and *Vi
+00014f10: 7469 7320 7669 6e69 6665 7261 2a20 2831  tis vinifera* (1
+00014f20: 2e33 3920 616e 6420 312e 3437 2c20 7265  .39 and 1.47, re
+00014f30: 7370 6563 7469 7665 6c79 292e 0a0a 4265  spectively)...Be
+00014f40: 7369 6465 732c 2077 6520 6361 6e20 616c  sides, we can al
+00014f50: 736f 2061 6464 2074 6865 2047 4d4d 206d  so add the GMM m
+00014f60: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
+00014f70: 6f66 2061 6e63 686f 7220 2a4b 2a3c 7375  of anchor *K*<su
+00014f80: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
+00014f90: 2077 6974 6820 7468 6520 666c 6167 2060   with the flag `
+00014fa0: 2d2d 706c 6f74 6170 676d 6d60 2c20 7573  --plotapgmm`, us
+00014fb0: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00014fc0: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
+00014fd0: 7669 7a20 2d64 2077 6764 5f67 6c6f 6261  viz -d wgd_globa
+00014fe0: 6c6d 7262 685f 6b73 2f67 6c6f 6261 6c5f  lmrbh_ks/global_
+00014ff0: 4d52 4248 2e74 7376 2e6b 732e 7473 7620  MRBH.tsv.ks.tsv 
+00015000: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+00015010: 7320 7767 645f 6b73 642f 4171 7569 6c65  s wgd_ksd/Aquile
+00015020: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00015030: 2e6b 732e 7473 7620 2d73 7020 7370 6563  .ks.tsv -sp spec
+00015040: 6965 7374 7265 652e 6e77 202d 2d72 6577  iestree.nw --rew
+00015050: 6569 6768 7420 2d61 7020 7767 645f 7379  eight -ap wgd_sy
+00015060: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+00015070: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
+00015080: 6f20 7767 645f 7669 7a5f 6d69 7865 645f  o wgd_viz_mixed_
+00015090: 4b73 5f65 6c6d 6d20 2d2d 7370 6169 7220  Ks_elmm --spair 
+000150a0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
+000150b0: 6c65 613b 5072 6f74 6561 5f63 796e 6172  lea;Protea_cynar
+000150c0: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+000150d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000150e0: 6561 3b56 6974 6973 5f76 696e 6966 6572  ea;Vitis_vinifer
+000150f0: 6122 202d 2d73 7061 6972 2022 4171 7569  a" --spair "Aqui
+00015100: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
+00015110: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00015120: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
+00015130: 6567 6961 5f63 6f65 7275 6c65 613b 4171  egia_coerulea;Aq
+00015140: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015150: 2220 2d2d 6773 6d61 7020 6765 6e65 5f73  " --gsmap gene_s
+00015160: 7065 6369 6573 2e6d 6170 202d 2d70 6c6f  pecies.map --plo
+00015170: 746b 6465 202d 2d70 6c6f 7461 7067 6d6d  tkde --plotapgmm
+00015180: 6d0a 6060 600a 0a21 5b5d 2864 6174 612f  m.```..![](data/
+00015190: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000151a0: 6561 5f47 6c6f 6261 6c6d 7262 684b 735f  ea_GlobalmrbhKs_
+000151b0: 4170 676d 6d5f 436f 7272 6563 7465 642e  Apgmm_Corrected.
+000151c0: 6b73 642e 7376 6729 0a0a 4173 206d 616e  ksd.svg)..As man
+000151d0: 6966 6573 7465 6420 6162 6f76 652c 2074  ifested above, t
+000151e0: 6865 2061 6e63 686f 7220 2a4b 2a3c 7375  he anchor *K*<su
+000151f0: 623e 533c 2f73 7562 3e20 636f 6d70 6f6e  b>S</sub> compon
+00015200: 656e 7420 3220 7769 7468 206d 6f64 6520  ent 2 with mode 
+00015210: 312e 3238 2069 7320 616c 736f 2079 6f75  1.28 is also you
+00015220: 6e67 6572 2074 6861 6e20 7468 6520 636f  nger than the co
+00015230: 7272 6563 7465 6420 6f72 7468 6f6c 6f67  rrected ortholog
+00015240: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00015250: 7562 3e20 6d6f 6465 2077 6974 6820 2a50  ub> mode with *P
+00015260: 726f 7465 6120 6379 6e61 726f 6964 6573  rotea cynaroides
+00015270: 2a20 616e 6420 2a56 6974 6973 2076 696e  * and *Vitis vin
+00015280: 6966 6572 612a 2e20 4275 7420 7765 206e  ifera*. But we n
+00015290: 6565 6420 746f 2062 6520 6f66 2063 6f75  eed to be of cou
+000152a0: 7273 6520 6361 7574 696f 7573 2074 6861  rse cautious tha
+000152b0: 7420 7375 6368 2064 6973 7469 6e63 7469  t such distincti
+000152c0: 6f6e 2063 6f6d 6573 2077 6974 6820 7468  on comes with th
+000152d0: 6520 756e 6365 7274 6169 6e74 6965 7320  e uncertainties 
+000152e0: 696e 7472 6f64 7563 6564 2066 726f 6d20  introduced from 
+000152f0: 7468 6520 6170 706c 6965 6420 6d69 7874  the applied mixt
+00015300: 7572 6520 6d6f 6465 6c69 6e67 206d 6574  ure modeling met
+00015310: 686f 646f 6c6f 6779 2069 6e20 7465 726d  hodology in term
+00015320: 7320 6f66 2066 6f72 2069 6e73 7461 6e63  s of for instanc
+00015330: 6520 6469 6666 6572 656e 7420 696e 6974  e different init
+00015340: 6961 6c69 7a61 7469 6f6e 2070 6f69 6e74  ialization point
+00015350: 7320 616e 6420 7468 6520 6973 7375 6520  s and the issue 
+00015360: 6f66 206f 7665 7266 6974 7469 6e67 2061  of overfitting a
+00015370: 6e64 2074 6865 2073 6973 7465 7220 7370  nd the sister sp
+00015380: 6563 6965 7373 2061 646f 7074 6564 2069  eciess adopted i
+00015390: 6e20 7468 6174 2074 6865 7265 206d 6967  n that there mig
+000153a0: 6874 2062 6520 7370 6563 6965 7320 7769  ht be species wi
+000153b0: 7468 206d 6f72 6520 6469 7370 6172 6174  th more disparat
+000153c0: 6520 7375 6273 7469 7475 7469 6f6e 2072  e substitution r
+000153d0: 6174 6573 2074 6861 6e20 7468 6520 6f6e  ates than the on
+000153e0: 6520 7765 2063 686f 7365 2e0a 0a41 6464  e we chose...Add
+000153f0: 696e 6720 626f 7468 2045 4c4d 4d20 7265  ing both ELMM re
+00015400: 7375 6c74 2066 6f72 2070 6172 616e 6f6d  sult for paranom
+00015410: 6520 616e 6420 474d 4d20 7265 7375 6c74  e and GMM result
+00015420: 2066 6f72 2061 6e63 686f 7220 2a4b 2a3c   for anchor *K*<
+00015430: 7375 623e 533c 2f73 7562 3e20 6361 6e20  sub>S</sub> can 
+00015440: 6265 2061 6368 6965 7665 6420 6a75 7374  be achieved just
+00015450: 2062 7920 6164 6420 7468 6520 7477 6f20   by add the two 
+00015460: 666c 6167 7320 6d65 6e74 696f 6e65 6420  flags mentioned 
+00015470: 6162 6f76 652c 2075 7369 6e67 2074 6865  above, using the
+00015480: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
+00015490: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
+000154a0: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+000154b0: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
+000154c0: 762e 6b73 2e74 7376 202d 2d65 7874 7261  v.ks.tsv --extra
+000154d0: 7061 7261 6e6f 6d65 6b73 2077 6764 5f6b  paranomeks wgd_k
+000154e0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+000154f0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00015500: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00015510: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00015520: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+00015530: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+00015540: 6e74 732e 7478 7420 2d6f 2077 6764 5f76  nts.txt -o wgd_v
+00015550: 697a 5f6d 6978 6564 5f4b 735f 656c 6d6d  iz_mixed_Ks_elmm
+00015560: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00015570: 6769 615f 636f 6572 756c 6561 3b50 726f  gia_coerulea;Pro
+00015580: 7465 615f 6379 6e61 726f 6964 6573 2220  tea_cynaroides" 
+00015590: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+000155a0: 6961 5f63 6f65 7275 6c65 613b 5669 7469  ia_coerulea;Viti
+000155b0: 735f 7669 6e69 6665 7261 2220 2d2d 7370  s_vinifera" --sp
+000155c0: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
+000155d0: 6f65 7275 6c65 613b 4163 6f72 7573 5f61  oerulea;Acorus_a
+000155e0: 6d65 7269 6361 6e75 7322 202d 2d73 7061  mericanus" --spa
+000155f0: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
+00015600: 6572 756c 6561 3b41 7175 696c 6567 6961  erulea;Aquilegia
+00015610: 5f63 6f65 7275 6c65 6122 202d 2d67 736d  _coerulea" --gsm
+00015620: 6170 2067 656e 655f 7370 6563 6965 732e  ap gene_species.
+00015630: 6d61 7020 2d2d 706c 6f74 6b64 6520 2d2d  map --plotkde --
+00015640: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
+00015650: 7067 6d6d 6d0a 6060 600a 0a21 5b5d 2864  pgmmm.```..![](d
+00015660: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
+00015670: 6572 756c 6561 5f47 6c6f 6261 6c6d 7262  erulea_Globalmrb
+00015680: 684b 735f 456c 6d6d 5f41 7067 6d6d 5f43  hKs_Elmm_Apgmm_C
+00015690: 6f72 7265 6374 6564 2e6b 7364 2e73 7667  orrected.ksd.svg
+000156a0: 290a 0a41 6e20 616c 7465 726e 6174 6976  )..An alternativ
+000156b0: 6520 7761 7920 746f 2063 616c 6375 6c61  e way to calcula
+000156c0: 7465 2074 6865 206f 7274 686f 6c6f 676f  te the orthologo
+000156d0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+000156e0: 623e 2069 7320 746f 2064 6972 6563 746c  b> is to directl
+000156f0: 7920 7573 6520 7468 6520 6f72 7468 6f67  y use the orthog
+00015700: 726f 7570 7320 696e 7374 6561 6420 6f66  roups instead of
+00015710: 2067 6c6f 6261 6c20 4d52 4248 2066 616d   global MRBH fam
+00015720: 696c 792e 2054 6861 7420 7761 7920 7765  ily. That way we
+00015730: 2064 6f6e 2774 2075 7365 2074 6865 2070   don't use the p
+00015740: 7265 2d69 6e66 6572 7265 6420 7061 7261  re-inferred para
+00015750: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00015760: 7375 623e 2062 7574 2074 6865 2070 6172  sub> but the par
+00015770: 616c 6f67 6f75 7320 6765 6e65 2070 6169  alogous gene pai
+00015780: 7273 2069 6e73 6964 6520 6561 6368 206f  rs inside each o
+00015790: 7274 686f 6772 6f75 7020 696e 7374 6561  rthogroup instea
+000157a0: 642e 2054 6f20 6163 6869 6576 6520 7468  d. To achieve th
+000157b0: 6174 2c20 7765 2066 6972 7374 206e 6565  at, we first nee
+000157c0: 6420 746f 2069 6e66 6572 206f 7274 686f  d to infer ortho
+000157d0: 6772 6f75 7073 2075 7369 6e67 2074 6865  groups using the
+000157e0: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
+000157f0: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00015800: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00015810: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+00015820: 7320 4163 6f72 7573 5f61 6d65 7269 6361  s Acorus_america
+00015830: 6e75 7320 5669 7469 735f 7669 6e69 6665  nus Vitis_vinife
+00015840: 7261 202d 2d6f 7274 686f 696e 6665 7220  ra --orthoinfer 
+00015850: 2d6f 2077 6764 5f6f 7274 686f 2028 2d2d  -o wgd_ortho (--
+00015860: 6f6e 6c79 6f72 7468 6f29 200a 6060 600a  onlyortho) .```.
+00015870: 0a55 7365 7273 2063 616e 2064 6563 6964  .Users can decid
+00015880: 6520 746f 206f 6e6c 7920 636f 6e64 7563  e to only conduc
+00015890: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
+000158a0: 2061 6e61 6c79 7369 7320 7768 696c 6520   analysis while 
+000158b0: 736b 6970 7069 6e67 206f 7468 6572 2061  skipping other a
+000158c0: 6e61 6c79 7369 7320 6279 2061 6464 696e  nalysis by addin
+000158d0: 6720 7468 6520 666c 6167 2060 2d2d 6f6e  g the flag `--on
+000158e0: 6c79 6f72 7468 6f60 2e20 4e65 7874 2073  lyortho`. Next s
+000158f0: 7465 7020 6973 2074 6865 2073 616d 6520  tep is the same 
+00015900: 7769 7468 2067 6c6f 6261 6c20 4d52 4248  with global MRBH
+00015910: 2066 616d 696c 7920 6578 6365 7074 2074   family except t
+00015920: 6861 7420 7765 2064 6f6e 2774 2075 7365  hat we don't use
+00015930: 2074 6865 2065 7874 7261 2070 7265 2d69   the extra pre-i
+00015940: 6e66 6572 7265 6420 7061 7261 6e6f 6d65  nferred paranome
+00015950: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00015960: 2061 6e79 6d6f 7265 2e20 5765 2069 6e66   anymore. We inf
+00015970: 6572 2074 6865 202a 4b2a 3c73 7562 3e53  er the *K*<sub>S
+00015980: 3c2f 7375 623e 2075 7369 6e67 2074 6865  </sub> using the
+00015990: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e20   command below. 
+000159a0: 4e6f 7465 2074 6861 7420 7468 6520 7072  Note that the pr
+000159b0: 6f67 7261 6d20 6077 6764 2076 697a 6020  ogram `wgd viz` 
+000159c0: 6361 6e20 706c 6f74 2074 6865 2073 616d  can plot the sam
+000159d0: 6520 6a75 7374 2061 7320 7368 6f77 6e20  e just as shown 
+000159e0: 6162 6f76 652e 0a0a 6060 600a 7767 6420  above...```.wgd 
+000159f0: 6b73 6420 7767 645f 6f72 7468 6f2f 4f72  ksd wgd_ortho/Or
+00015a00: 7468 6f67 726f 7570 732e 7370 2e74 7376  thogroups.sp.tsv
+00015a10: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00015a20: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00015a30: 6f20 7767 645f 6f72 7468 6f5f 6b73 202d  o wgd_ortho_ks -
+00015a40: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00015a50: 615f 636f 6572 756c 6561 3b50 726f 7465  a_coerulea;Prote
+00015a60: 615f 6379 6e61 726f 6964 6573 2220 2d2d  a_cynaroides" --
+00015a70: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
+00015a80: 5f63 6f65 7275 6c65 613b 5669 7469 735f  _coerulea;Vitis_
+00015a90: 7669 6e69 6665 7261 2220 2d2d 7370 6169  vinifera" --spai
+00015aa0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
+00015ab0: 7275 6c65 613b 4163 6f72 7573 5f61 6d65  rulea;Acorus_ame
+00015ac0: 7269 6361 6e75 7322 202d 2d73 7061 6972  ricanus" --spair
+00015ad0: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
+00015ae0: 756c 6561 3b41 7175 696c 6567 6961 5f63  ulea;Aquilegia_c
+00015af0: 6f65 7275 6c65 6122 202d 2d70 6c6f 746b  oerulea" --plotk
+00015b00: 6465 2028 2d61 7020 7767 645f 7379 6e2f  de (-ap wgd_syn/
+00015b10: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
+00015b20: 6f72 706f 696e 7473 2e74 7874 290a 6060  orpoints.txt).``
+00015b30: 600a 0a21 5b5d 2864 6174 612f 4171 7569  `..![](data/Aqui
+00015b40: 6c65 6769 615f 636f 6572 756c 6561 5f4f  legia_coerulea_O
+00015b50: 7274 686f 4b73 5f43 6f72 7265 6374 6564  rthoKs_Corrected
+00015b60: 2e6b 7364 2e73 7667 290a 0a41 7320 7368  .ksd.svg)..As sh
+00015b70: 6f77 6e20 6162 6f76 652c 2074 6865 206e  own above, the n
+00015b80: 756d 6265 7220 6f66 2062 6f74 6820 7061  umber of both pa
+00015b90: 7261 6c6f 676f 7573 2067 656e 6520 7061  ralogous gene pa
+00015ba0: 6972 7320 616e 6420 6f72 7468 6f6c 6f67  irs and ortholog
+00015bb0: 6f75 7320 6765 6e65 2070 6169 7273 2069  ous gene pairs i
+00015bc0: 7320 6469 6666 6572 656e 7420 7468 616e  s different than
+00015bd0: 2074 6865 206f 6e65 2066 726f 6d20 676c   the one from gl
+00015be0: 6f62 616c 204d 5242 4820 6661 6d69 6c79  obal MRBH family
+00015bf0: 2069 6e20 7468 6174 2068 6572 6520 7765   in that here we
+00015c00: 2070 6c6f 7474 6564 2061 6c6c 206f 7274   plotted all ort
+00015c10: 686f 6c6f 676f 7573 2067 656e 6520 7061  hologous gene pa
+00015c20: 6972 7320 696e 7374 6561 6420 6f66 206f  irs instead of o
+00015c30: 6e6c 7920 676c 6f62 616c 204d 5242 4820  nly global MRBH 
+00015c40: 616e 6420 706f 7465 6e74 6961 6c6c 7920  and potentially 
+00015c50: 6e65 7720 7061 7261 6c6f 676f 7573 2067  new paralogous g
+00015c60: 656e 6520 7061 6972 7320 6d69 6768 7420  ene pairs might 
+00015c70: 6265 2070 726f 6475 6365 6420 696e 2074  be produced in t
+00015c80: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00015c90: 6665 7265 6e63 6520 7374 6570 2c20 746f  ference step, to
+00015ca0: 6765 7468 6572 2077 6974 6820 6469 6666  gether with diff
+00015cb0: 6572 656e 7420 7265 6361 6c63 756c 6174  erent recalculat
+00015cc0: 6564 2077 6569 6768 7473 2e0a 0a49 6620  ed weights...If 
+00015cd0: 6f6e 6520 6f6e 6c79 2077 616e 7465 6420  one only wanted 
+00015ce0: 746f 2070 6c6f 7420 7468 6520 6f72 7468  to plot the orth
+00015cf0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00015d00: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00015d10: 7469 6f6e 2c20 6173 2077 6520 6861 7665  tion, as we have
+00015d20: 2061 6c72 6561 6479 2073 686f 776e 2061   already shown a
+00015d30: 626f 7665 2c20 6974 2063 616e 2062 6520  bove, it can be 
+00015d40: 6561 7369 6c79 2061 6368 6965 7665 6420  easily achieved 
+00015d50: 6279 2072 656d 6f76 696e 6720 7468 6520  by removing the 
+00015d60: 7061 7261 6c6f 676f 7573 2073 7065 6369  paralogous speci
+00015d70: 6573 2070 6169 7220 6041 7175 696c 6567  es pair `Aquileg
+00015d80: 6961 5f63 6f65 7275 6c65 613b 4171 7569  ia_coerulea;Aqui
+00015d90: 6c65 6769 615f 636f 6572 756c 6561 602c  legia_coerulea`,
+00015da0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
+00015db0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
+00015dc0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00015dd0: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00015de0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00015df0: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00015e00: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00015e10: 202d 6f20 7767 645f 676c 6f62 616c 6d72   -o wgd_globalmr
+00015e20: 6268 5f6f 6e6c 796f 7274 686f 5f6b 7320  bh_onlyortho_ks 
+00015e30: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
+00015e40: 6961 5f63 6f65 7275 6c65 613b 5072 6f74  ia_coerulea;Prot
+00015e50: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
+00015e60: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00015e70: 615f 636f 6572 756c 6561 3b56 6974 6973  a_coerulea;Vitis
+00015e80: 5f76 696e 6966 6572 6122 202d 2d73 7061  _vinifera" --spa
+00015e90: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
+00015ea0: 6572 756c 6561 3b41 636f 7275 735f 616d  erulea;Acorus_am
+00015eb0: 6572 6963 616e 7573 2220 2d2d 6773 6d61  ericanus" --gsma
+00015ec0: 7020 6765 6e65 5f73 7065 6369 6573 2e6d  p gene_species.m
+00015ed0: 6170 0a60 6060 0a0a 5765 2063 616e 2063  ap.```..We can c
+00015ee0: 6c65 6172 6c79 2073 6565 2074 6861 7420  learly see that 
+00015ef0: 2a56 6974 6973 2076 696e 6966 6572 612a  *Vitis vinifera*
+00015f00: 2068 6173 2068 6967 6865 7220 7375 6273   has higher subs
+00015f10: 7469 7475 7469 6f6e 2072 6174 6520 7468  titution rate th
+00015f20: 616e 202a 5072 6f74 6561 2063 796e 6172  an *Protea cynar
+00015f30: 6f69 6465 732a 2069 6e20 7468 6174 2074  oides* in that t
+00015f40: 6865 6972 206f 7274 686f 6c6f 676f 7573  heir orthologous
+00015f50: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00015f60: 2070 6561 6b73 2077 6974 6820 2a41 7175   peaks with *Aqu
+00015f70: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+00015f80: 2c20 616c 7468 6f75 6768 2072 6570 7265  , although repre
+00015f90: 7365 6e74 696e 6720 7468 6520 7361 6d65  senting the same
+00015fa0: 2064 6976 6572 6765 6e63 6520 6576 656e   divergence even
+00015fb0: 742c 2064 6966 6665 7265 6420 696e 2073  t, differed in s
+00015fc0: 7562 7374 6974 7574 696f 6e20 6469 7374  ubstitution dist
+00015fd0: 616e 6365 2e0a 215b 5d28 6461 7461 2f52  ance..![](data/R
+00015fe0: 6177 5f4f 7274 686f 6c6f 6775 6573 2e6b  aw_Orthologues.k
+00015ff0: 7364 2e73 7667 290a 0a41 6674 6572 2074  sd.svg)..After t
+00016000: 6865 2070 6879 6c6f 6765 6e65 7469 6320  he phylogenetic 
+00016010: 7469 6d69 6e67 206f 6620 7468 6520 5261  timing of the Ra
+00016020: 6e75 6e63 756c 616c 6573 2057 4744 2c20  nunculales WGD, 
+00016030: 7765 2063 616e 2066 7572 7468 6572 2069  we can further i
+00016040: 6e66 6572 2069 7473 2061 6273 6f6c 7574  nfer its absolut
+00016050: 6520 6167 652e 2046 6972 7374 2077 6520  e age. First we 
+00016060: 696e 6665 7220 7468 6520 6372 6564 6962  infer the credib
+00016070: 6c65 2072 616e 6765 206f 6620 616e 6368  le range of anch
+00016080: 6f72 2070 6169 7273 2062 7920 2a4b 2a3c  or pairs by *K*<
+00016090: 7375 623e 533c 2f73 7562 3e20 6865 7572  sub>S</sub> heur
+000160a0: 6973 7469 6361 6c6c 7920 7573 696e 6720  istically using 
+000160b0: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
+000160c0: 2070 6561 6b60 2e0a 0a60 6060 0a77 6764   peak`...```.wgd
+000160d0: 2070 6561 6b20 2d2d 6865 7572 6973 7469   peak --heuristi
+000160e0: 6320 7767 645f 6b73 642f 4171 7569 6c65  c wgd_ksd/Aquile
+000160f0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00016100: 2e6b 732e 7473 7620 2d61 7020 7767 645f  .ks.tsv -ap wgd_
+00016110: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00016120: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
+00016130: 202d 736d 2077 6764 5f73 796e 2f69 6164   -sm wgd_syn/iad
+00016140: 686f 7265 2d6f 7574 2f73 6567 6d65 6e74  hore-out/segment
+00016150: 732e 7478 7420 2d6c 6520 7767 645f 7379  s.txt -le wgd_sy
+00016160: 6e2f 6961 6468 6f72 652d 6f75 742f 6c69  n/iadhore-out/li
+00016170: 7374 5f65 6c65 6d65 6e74 732e 7478 7420  st_elements.txt 
+00016180: 2d6d 7020 7767 645f 7379 6e2f 6961 6468  -mp wgd_syn/iadh
+00016190: 6f72 652d 6f75 742f 6d75 6c74 6970 6c69  ore-out/multipli
+000161a0: 636f 6e5f 7061 6972 732e 7478 7420 2d6f  con_pairs.txt -o
+000161b0: 2077 6764 5f70 6561 6b0a 6060 600a 0a21   wgd_peak.```..!
+000161c0: 5b5d 2864 6174 612f 416e 6368 6f72 4b73  [](data/AnchorKs
+000161d0: 5f50 6561 6b43 495f 4171 7569 6c65 6769  _PeakCI_Aquilegi
+000161e0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+000161f0: 732e 7473 765f 6e6f 6465 5f77 6569 6768  s.tsv_node_weigh
+00016200: 7465 642e 7376 6729 0a0a 4173 2073 686f  ted.svg)..As sho
+00016210: 776e 2061 626f 7665 2c20 7765 2061 7373  wn above, we ass
+00016220: 756d 6564 2061 206c 6f67 6e6f 726d 616c  umed a lognormal
+00016230: 2064 6973 7472 6962 7574 696f 6e20 6174   distribution at
+00016240: 2074 6865 2070 6561 6b20 6c6f 6361 7469   the peak locati
+00016250: 6f6e 2064 6574 6563 7465 6420 6279 2074  on detected by t
+00016260: 6865 2060 7369 676e 616c 6020 6d6f 6475  he `signal` modu
+00016270: 6c65 206f 6620 6073 6369 7079 6020 6c69  le of `scipy` li
+00016280: 6272 6172 792e 2054 6865 2039 3525 2063  brary. The 95% c
+00016290: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
+000162a0: 6f66 2074 6865 206c 6f67 6e6f 726d 616c  of the lognormal
+000162b0: 2064 6973 7472 6962 7574 696f 6e20 7761   distribution wa
+000162c0: 7320 6170 706c 6965 642c 2069 2e65 2e2c  s applied, i.e.,
+000162d0: 2030 2e36 382d 322e 3734 2c20 696e 2066   0.68-2.74, in f
+000162e0: 7572 7468 6572 206d 6f6c 6563 756c 6172  urther molecular
+000162f0: 2064 6174 696e 672e 2054 6865 2066 696c   dating. The fil
+00016300: 6520 6041 7175 696c 6567 6961 5f63 6f65  e `Aquilegia_coe
+00016310: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00016320: 5f39 3525 4349 5f41 505f 666f 725f 6461  _95%CI_AP_for_da
+00016330: 7469 6e67 5f77 6569 6768 7465 645f 666f  ting_weighted_fo
+00016340: 726d 6174 2e74 7376 6020 6973 2077 6861  rmat.tsv` is wha
+00016350: 7420 7765 206e 6565 6420 666f 7220 6e65  t we need for ne
+00016360: 7874 2073 7465 702e 2054 6f20 6275 696c  xt step. To buil
+00016370: 6420 7468 6520 6f72 7468 6f67 726f 7570  d the orthogroup
+00016380: 7320 7573 6564 2069 6e20 7068 796c 6f67  s used in phylog
+00016390: 656e 6574 6963 2064 6174 696e 672c 2077  enetic dating, w
+000163a0: 6520 6e65 6564 2074 6f20 7365 6c65 6374  e need to select
+000163b0: 2073 6f6d 6520 7370 6563 6965 7320 616e   some species an
+000163c0: 6420 666f 726d 2061 2073 7461 7274 696e  d form a startin
+000163d0: 6720 7472 6565 2077 6974 6820 7072 6f70  g tree with prop
+000163e0: 6572 2066 6f73 7369 6c20 6361 6c69 6272  er fossil calibr
+000163f0: 6174 696f 6e73 2e20 5765 2070 726f 7669  ations. We provi
+00016400: 6465 206f 6e65 2069 6e20 6d63 6d63 7472  de one in mcmctr
+00016410: 6565 2066 6f72 6d61 7420 6173 2062 656c  ee format as bel
+00016420: 6f77 2e0a 0a60 6060 0a31 3720 310a 2828  ow...```.17 1.((
+00016430: 2828 506f 7461 6d6f 6765 746f 6e5f 6163  ((Potamogeton_ac
+00016440: 7574 6966 6f6c 6975 732c 2853 7069 726f  utifolius,(Spiro
+00016450: 6465 6c61 5f69 6e74 6572 6d65 6469 612c  dela_intermedia,
+00016460: 416d 6f72 7068 6f70 6861 6c6c 7573 5f6b  Amorphophallus_k
+00016470: 6f6e 6a61 6329 292c 2841 6361 6e74 686f  onjac)),(Acantho
+00016480: 6368 6c61 6d79 735f 6272 6163 7465 6174  chlamys_bracteat
+00016490: 612c 2844 696f 7363 6f72 6561 5f61 6c61  a,(Dioscorea_ala
+000164a0: 7461 2c44 696f 7363 6f72 6561 5f72 6f74  ta,Dioscorea_rot
+000164b0: 756e 6461 7461 2929 273e 302e 3536 3030  undata))'>0.5600
+000164c0: 3c31 2e32 3836 3327 2927 3e30 2e38 3336  <1.2863')'>0.836
+000164d0: 303c 312e 3238 3633 272c 2841 636f 7275  0<1.2863',(Acoru
+000164e0: 735f 616d 6572 6963 616e 7573 2c41 636f  s_americanus,Aco
+000164f0: 7275 735f 7461 7461 7269 6e6f 7769 6929  rus_tatarinowii)
+00016500: 2927 3e30 2e38 3336 303c 312e 3238 3633  )'>0.8360<1.2863
+00016510: 272c 2828 2828 5465 7472 6163 656e 7472  ',((((Tetracentr
+00016520: 6f6e 5f73 696e 656e 7365 2c54 726f 6368  on_sinense,Troch
+00016530: 6f64 656e 6472 6f6e 5f61 7261 6c69 6f69  odendron_aralioi
+00016540: 6465 7329 2c28 4275 7875 735f 6175 7374  des),(Buxus_aust
+00016550: 726f 7975 6e6e 616e 656e 7369 732c 4275  royunnanensis,Bu
+00016560: 7875 735f 7369 6e69 6361 2929 273e 312e  xus_sinica))'>1.
+00016570: 3130 3830 3c31 2e32 3836 3327 2c28 4e65  1080<1.2863',(Ne
+00016580: 6c75 6d62 6f5f 6e75 6369 6665 7261 2c28  lumbo_nucifera,(
+00016590: 5465 6c6f 7065 615f 7370 6563 696f 7369  Telopea_speciosi
+000165a0: 7373 696d 612c 5072 6f74 6561 5f63 796e  ssima,Protea_cyn
+000165b0: 6172 6f69 6465 7329 2929 273e 312e 3130  aroides)))'>1.10
+000165c0: 3830 3c31 2e32 3836 3327 2c28 4171 7569  80<1.2863',(Aqui
+000165d0: 6c65 6769 615f 636f 6572 756c 6561 5f61  legia_coerulea_a
+000165e0: 7031 2c41 7175 696c 6567 6961 5f63 6f65  p1,Aquilegia_coe
+000165f0: 7275 6c65 615f 6170 3229 2927 3e31 2e31  rulea_ap2))'>1.1
+00016600: 3038 303c 312e 3238 3633 2729 273e 312e  080<1.2863')'>1.
+00016610: 3237 3230 3c32 2e34 3732 3027 3b0a 6060  2720<2.4720';.``
+00016620: 600a 0a41 7320 7072 6573 656e 7465 6420  `..As presented 
+00016630: 6162 6f76 652c 2074 6865 2066 6f63 7573  above, the focus
+00016640: 2073 7065 6369 6573 2074 6861 7420 6973   species that is
+00016650: 2061 626f 7574 2074 6f20 6265 2064 6174   about to be dat
+00016660: 6564 206e 6565 6473 2074 6f20 6265 2072  ed needs to be r
+00016670: 6570 6c61 6365 6420 7769 7468 2060 2841  eplaced with `(A
+00016680: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00016690: 615f 6170 312c 4171 7569 6c65 6769 615f  a_ap1,Aquilegia_
+000166a0: 636f 6572 756c 6561 5f61 7032 2960 2e20  coerulea_ap2)`. 
+000166b0: 5769 7468 2074 6869 7320 7374 6172 7469  With this starti
+000166c0: 6e67 2074 7265 6520 616e 6420 7072 6564  ng tree and pred
+000166d0: 6f77 6e6c 6f61 6465 6420 6364 7320 6669  ownloaded cds fi
+000166e0: 6c65 7320 6f66 2061 6c6c 2074 6865 2073  les of all the s
+000166f0: 7065 6369 6573 2c20 7765 2063 616e 2062  pecies, we can b
+00016700: 7569 6c64 2074 6865 206f 7274 686f 6772  uild the orthogr
+00016710: 6f75 7020 7573 6564 2069 6e20 7468 6520  oup used in the 
+00016720: 6669 6e61 6c20 6d6f 6c65 6375 6c61 7220  final molecular 
+00016730: 6461 7469 6e67 2075 7369 6e67 2074 6865  dating using the
+00016740: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
+00016750: 772e 0a0a 6060 600a 7767 6420 646d 6420  w...```.wgd dmd 
+00016760: 2d66 2041 7175 696c 6567 6961 5f63 6f65  -f Aquilegia_coe
+00016770: 7275 6c65 6120 2d61 7020 7767 645f 7065  rulea -ap wgd_pe
+00016780: 616b 2f41 7175 696c 6567 6961 5f63 6f65  ak/Aquilegia_coe
+00016790: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000167a0: 5f39 3525 4349 5f41 505f 666f 725f 6461  _95%CI_AP_for_da
+000167b0: 7469 6e67 5f77 6569 6768 7465 645f 666f  ting_weighted_fo
+000167c0: 726d 6174 2e74 7376 202d 6f20 7767 645f  rmat.tsv -o wgd_
+000167d0: 646d 645f 6f72 7468 6f20 506f 7461 6d6f  dmd_ortho Potamo
+000167e0: 6765 746f 6e5f 6163 7574 6966 6f6c 6975  geton_acutifoliu
+000167f0: 7320 5370 6972 6f64 656c 615f 696e 7465  s Spirodela_inte
+00016800: 726d 6564 6961 2041 6d6f 7270 686f 7068  rmedia Amorphoph
+00016810: 616c 6c75 735f 6b6f 6e6a 6163 2041 6361  allus_konjac Aca
+00016820: 6e74 686f 6368 6c61 6d79 735f 6272 6163  nthochlamys_brac
+00016830: 7465 6174 6120 4469 6f73 636f 7265 615f  teata Dioscorea_
+00016840: 616c 6174 6120 4469 6f73 636f 7265 615f  alata Dioscorea_
+00016850: 726f 7475 6e64 6174 6120 4163 6f72 7573  rotundata Acorus
+00016860: 5f61 6d65 7269 6361 6e75 7320 4163 6f72  _americanus Acor
+00016870: 7573 5f74 6174 6172 696e 6f77 6969 2054  us_tatarinowii T
+00016880: 6574 7261 6365 6e74 726f 6e5f 7369 6e65  etracentron_sine
+00016890: 6e73 6520 5472 6f63 686f 6465 6e64 726f  nse Trochodendro
+000168a0: 6e5f 6172 616c 696f 6964 6573 2042 7578  n_aralioides Bux
+000168b0: 7573 5f61 7573 7472 6f79 756e 6e61 6e65  us_austroyunnane
+000168c0: 6e73 6973 2042 7578 7573 5f73 696e 6963  nsis Buxus_sinic
+000168d0: 6120 4e65 6c75 6d62 6f5f 6e75 6369 6665  a Nelumbo_nucife
+000168e0: 7261 2054 656c 6f70 6561 5f73 7065 6369  ra Telopea_speci
+000168f0: 6f73 6973 7369 6d61 2050 726f 7465 615f  osissima Protea_
+00016900: 6379 6e61 726f 6964 6573 2041 7175 696c  cynaroides Aquil
+00016910: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
+00016920: 600a 0a54 6865 2072 6573 756c 7420 6669  `..The result fi
+00016930: 6c65 2060 6d65 7267 655f 666f 6375 735f  le `merge_focus_
+00016940: 6170 2e74 7376 6020 6973 2077 6861 7420  ap.tsv` is what 
+00016950: 7765 206e 6565 6420 666f 7220 7468 6520  we need for the 
+00016960: 6669 6e61 6c20 7374 6570 206f 6620 6d6f  final step of mo
+00016970: 6c65 6375 6c61 7220 6461 7469 6e67 2069  lecular dating i
+00016980: 6e20 7072 6f67 7261 6d20 6077 6764 2066  n program `wgd f
+00016990: 6f63 7573 602e 0a0a 6060 600a 7767 6420  ocus`...```.wgd 
+000169a0: 666f 6375 7320 2d2d 7072 6f74 6461 7469  focus --protdati
+000169b0: 6e67 202d 2d61 616d 6f64 656c 206c 6720  ng --aamodel lg 
+000169c0: 7767 645f 646d 645f 6f72 7468 6f2f 6d65  wgd_dmd_ortho/me
+000169d0: 7267 655f 666f 6375 735f 6170 2e74 7376  rge_focus_ap.tsv
+000169e0: 202d 7370 2064 6174 696e 675f 7472 6565   -sp dating_tree
+000169f0: 2e6e 7720 2d6f 2077 6764 5f64 6174 696e  .nw -o wgd_datin
+00016a00: 6720 2d64 206d 636d 6374 7265 6520 2d64  g -d mcmctree -d
+00016a10: 7320 2762 7572 6e69 6e20 3d20 3230 3030  s 'burnin = 2000
+00016a20: 2720 2d64 7320 2773 616d 7066 7265 7120  ' -ds 'sampfreq 
+00016a30: 3d20 3130 3030 2720 2d64 7320 276e 7361  = 1000' -ds 'nsa
+00016a40: 6d70 6c65 203d 2032 3030 3030 2720 506f  mple = 20000' Po
+00016a50: 7461 6d6f 6765 746f 6e5f 6163 7574 6966  tamogeton_acutif
+00016a60: 6f6c 6975 7320 5370 6972 6f64 656c 615f  olius Spirodela_
+00016a70: 696e 7465 726d 6564 6961 2041 6d6f 7270  intermedia Amorp
+00016a80: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
+00016a90: 2041 6361 6e74 686f 6368 6c61 6d79 735f   Acanthochlamys_
+00016aa0: 6272 6163 7465 6174 6120 4469 6f73 636f  bracteata Diosco
+00016ab0: 7265 615f 616c 6174 6120 4469 6f73 636f  rea_alata Diosco
+00016ac0: 7265 615f 726f 7475 6e64 6174 6120 4163  rea_rotundata Ac
+00016ad0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+00016ae0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
+00016af0: 6969 2054 6574 7261 6365 6e74 726f 6e5f  ii Tetracentron_
+00016b00: 7369 6e65 6e73 6520 5472 6f63 686f 6465  sinense Trochode
+00016b10: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
+00016b20: 2042 7578 7573 5f61 7573 7472 6f79 756e   Buxus_austroyun
+00016b30: 6e61 6e65 6e73 6973 2042 7578 7573 5f73  nanensis Buxus_s
+00016b40: 696e 6963 6120 4e65 6c75 6d62 6f5f 6e75  inica Nelumbo_nu
+00016b50: 6369 6665 7261 2054 656c 6f70 6561 5f73  cifera Telopea_s
+00016b60: 7065 6369 6f73 6973 7369 6d61 2050 726f  peciosissima Pro
+00016b70: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
+00016b80: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00016b90: 610a 6060 600a 0a48 6572 6520 7765 206f  a.```..Here we o
+00016ba0: 6e6c 7920 696d 706c 656d 656e 7465 6420  nly implemented 
+00016bb0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+00016bc0: 6e20 616e 616c 7973 6973 2075 7369 6e67  n analysis using
+00016bd0: 2070 726f 7465 696e 2073 6571 7565 6e63   protein sequenc
+00016be0: 6520 6279 2061 6464 696e 6720 7468 6520  e by adding the 
+00016bf0: 666c 6167 2060 2d2d 7072 6f74 6461 7469  flag `--protdati
+00016c00: 6e67 6020 616e 6420 7765 2073 6574 2074  ng` and we set t
+00016c10: 6865 2070 6172 616d 6574 6572 2066 6f72  he parameter for
+00016c20: 2060 6d63 6d63 7472 6565 6020 7669 6120   `mcmctree` via 
+00016c30: 7468 6520 6f70 7469 6f6e 2060 2d64 7360  the option `-ds`
+00016c40: 2e20 4e6f 7465 2074 6861 7420 6f74 6865  . Note that othe
+00016c50: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
+00016c60: 2073 7563 6820 6173 2060 7238 7360 2061   such as `r8s` a
+00016c70: 6e64 2060 6265 6173 7460 2061 7265 2061  nd `beast` are a
+00016c80: 6c73 6f20 6176 6169 6c61 626c 6520 6769  lso available gi
+00016c90: 7665 6e20 736f 6d65 206d 616e 6461 746f  ven some mandato
+00016ca0: 7279 2070 6172 616d 6574 6572 732e 2054  ry parameters. T
+00016cb0: 6865 2066 696e 616c 206c 6f67 206f 6620  he final log of 
+00016cc0: 7468 6520 7375 6363 6573 7366 756c 2072  the successful r
+00016cd0: 756e 2069 7320 6173 2062 656c 6f77 2e0a  un is as below..
+00016ce0: 0a60 6060 0a31 363a 3034 3a32 3520 494e  .```.16:04:25 IN
+00016cf0: 464f 2020 2020 2052 756e 6e69 6e67 206d  FO     Running m
+00016d00: 636d 6374 7265 6520 7573 696e 6720 4865  cmctree using He
+00016d10: 7373 6961 6e20 6d61 7472 6978 206f 6620  ssian matrix of 
+00016d20: 4c47 2b47 616d 6d61 2020 636f 7265 2e70  LG+Gamma  core.p
+00016d30: 793a 3936 370a 2020 2020 2020 2020 2020  y:967.          
+00016d40: 2020 2020 2020 2020 666f 7220 7072 6f74          for prot
+00016d50: 6569 6e20 6d6f 6465 6c0a 3233 3a34 393a  ein model.23:49:
+00016d60: 3337 2049 4e46 4f20 2020 2020 506f 7374  37 INFO     Post
+00016d70: 6572 696f 7220 6d65 616e 2066 6f72 2074  erior mean for t
+00016d80: 6865 2061 6765 7320 6f66 2077 6764 2069  he ages of wgd i
+00016d90: 7320 3131 322e 3839 3435 206d 636d 6374  s 112.8945 mcmct
+00016da0: 7265 652e 7079 3a32 3936 0a20 2020 2020  ree.py:296.     
+00016db0: 2020 2020 2020 2020 2020 2020 206d 696c               mil
+00016dc0: 6c69 6f6e 2079 6561 7273 2066 726f 6d20  lion years from 
+00016dd0: 436f 6e63 6174 656e 6174 6564 2070 6570  Concatenated pep
+00016de0: 7469 6465 0a20 2020 2020 2020 2020 2020  tide.           
+00016df0: 2020 2020 2020 2061 6c69 676e 6d65 6e74         alignment
+00016e00: 2061 6e64 2039 3525 2063 7265 6469 6269   and 95% credibi
+00016e10: 6c69 7479 2069 6e74 6572 7661 6c73 2028  lity intervals (
+00016e20: 4349 290a 2020 2020 2020 2020 2020 2020  CI).            
+00016e30: 2020 2020 2020 6973 2031 3031 2e32 3234        is 101.224
+00016e40: 2d31 3233 2e31 3231 206d 696c 6c69 6f6e  -123.121 million
+00016e50: 2079 6561 7273 0a20 2020 2020 2020 2020   years.         
+00016e60: 494e 464f 2020 2020 2054 6f74 616c 2072  INFO     Total r
+00016e70: 756e 2074 696d 653a 2032 3931 3735 7320  un time: 29175s 
+00016e80: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016e90: 2020 2020 2020 2020 2020 2020 2063 6c69               cli
+00016ea0: 2e70 793a 3234 310a 2020 2020 2020 2020  .py:241.        
+00016eb0: 2049 4e46 4f20 2020 2020 446f 6e65 2020   INFO     Done  
+00016ec0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016ed0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00016ee0: 2020 2020 2020 2020 2020 2020 2020 636c                cl
+00016ef0: 692e 7079 3a32 3432 0a60 6060 0a0a 546f  i.py:242.```..To
+00016f00: 2076 6973 7561 6c69 7a65 2074 6865 2064   visualize the d
+00016f10: 6174 652c 2077 6520 616c 736f 2070 726f  ate, we also pro
+00016f20: 7669 6465 6420 6120 7079 7468 6f6e 2073  vided a python s
+00016f30: 6372 6970 7420 746f 2070 6c6f 7420 7468  cript to plot th
+00016f40: 6520 5747 4420 6461 7465 7320 696e 2074  e WGD dates in t
+00016f50: 6865 2060 7767 6460 2066 6f6c 6465 722e  he `wgd` folder.
+00016f60: 2055 7365 7273 206e 6565 6420 746f 2065   Users need to e
+00016f70: 7874 7261 6374 2074 6865 2072 6177 2064  xtract the raw d
+00016f80: 6174 6573 2066 726f 6d20 7468 6520 606d  ates from the `m
+00016f90: 636d 632e 7478 7460 2066 6f72 2074 6865  cmc.txt` for the
+00016fa0: 2057 4744 206e 6f64 6520 6669 7273 7420   WGD node first 
+00016fb0: 616e 6420 7361 7665 2069 7420 6173 2066  and save it as f
+00016fc0: 696c 6520 6064 6174 6573 2e74 7874 6020  ile `dates.txt` 
+00016fd0: 286f 7220 7768 6174 6576 6572 2070 7265  (or whatever pre
+00016fe0: 6665 7272 6564 206e 616d 6529 2e20 416e  ferred name). An
+00016ff0: 2065 7861 6d70 6c65 2063 6f6d 6d61 6e64   example command
+00017000: 2069 7320 6173 2062 656c 6f77 2e0a 0a60   is as below...`
+00017010: 6060 0a70 7974 686f 6e20 2450 4154 482f  ``.python $PATH/
+00017020: 706f 7374 706c 6f74 2e70 7920 706f 7374  postplot.py post
+00017030: 6469 7320 6461 7465 732e 7478 7420 2d2d  dis dates.txt --
+00017040: 7065 7263 656e 7469 6c65 2039 3020 2d2d  percentile 90 --
+00017050: 7469 746c 6520 2257 4744 2064 6174 6522  title "WGD date"
+00017060: 202d 2d68 7064 202d 6f20 2252 616e 756e   --hpd -o "Ranun
+00017070: 6375 6c61 6c65 735f 5747 445f 6461 7465  culales_WGD_date
+00017080: 2e73 7667 220a 6060 600a 0a21 5b5d 2864  .svg".```..![](d
+00017090: 6174 612f 5261 6e75 6e63 756c 616c 6573  ata/Ranunculales
+000170a0: 5f57 4744 5f64 6174 652e 7376 6729 0a0a  _WGD_date.svg)..
+000170b0: 5468 6520 706f 7374 6572 696f 7220 6d65  The posterior me
+000170c0: 616e 2c20 6d65 6469 616e 2061 6e64 206d  an, median and m
+000170d0: 6f64 6520 6f66 2074 6865 2052 616e 756e  ode of the Ranun
+000170e0: 6375 6c61 6c65 7320 5747 4420 6167 6520  culales WGD age 
+000170f0: 6973 2031 3132 2e39 322c 2031 3133 2e34  is 112.92, 113.4
+00017100: 3420 616e 6420 3131 322e 3534 206d 7961  4 and 112.54 mya
+00017110: 2c20 7769 7468 2039 3025 2048 5044 2031  , with 90% HPD 1
+00017120: 3035 2e30 3720 2d20 3132 322e 3332 206d  05.07 - 122.32 m
+00017130: 7961 2061 7320 6d61 6e69 6665 7374 6564  ya as manifested
+00017140: 2061 626f 7665 2e0a 0a23 2323 204b 7374   above...### Kst
+00017150: 7265 650a 0a49 6e20 6164 6469 7469 6f6e  ree..In addition
+00017160: 2074 6f20 7061 6972 7769 7365 202a 4b2a   to pairwise *K*
+00017170: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+00017180: 696d 6174 696f 6e2c 2061 202a 4b2a 3c73  imation, a *K*<s
+00017190: 7562 3e53 3c2f 7375 623e 2074 7265 6520  ub>S</sub> tree 
+000171a0: 7769 7468 2062 7261 6e63 6820 6c65 6e67  with branch leng
+000171b0: 7468 2069 6e20 2a4b 2a3c 7375 623e 533c  th in *K*<sub>S<
+000171c0: 2f73 7562 3e20 756e 6974 2063 616e 2061  /sub> unit can a
+000171d0: 6c73 6f20 6265 2064 6572 6976 6564 2066  lso be derived f
+000171e0: 726f 6d20 7468 6520 7072 6f67 7261 6d20  rom the program 
+000171f0: 6077 6764 206b 7364 6020 6769 7665 6e20  `wgd ksd` given 
+00017200: 7468 6520 6f70 7469 6f6e 2060 2d2d 6b73  the option `--ks
+00017210: 7472 6565 6020 616e 6420 602d 2d73 7065  tree` and `--spe
+00017220: 6369 6573 7472 6565 602e 204e 6f74 6520  ciestree`. Note 
+00017230: 7468 6174 2074 6865 2061 6464 6974 696f  that the additio
+00017240: 6e61 6c20 6f70 7469 6f6e 2060 2d2d 6f6e  nal option `--on
+00017250: 6c79 636f 6e63 6174 6b73 7472 6565 6020  lyconcatkstree` 
+00017260: 7769 6c6c 206f 6e6c 7920 6361 6c6c 2074  will only call t
+00017270: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+00017280: 623e 2065 7374 696d 6174 696f 6e20 666f  b> estimation fo
+00017290: 7220 7468 6520 636f 6e63 6174 656e 6174  r the concatenat
+000172a0: 6564 2061 6c69 676e 6d65 6e74 2072 6174  ed alignment rat
+000172b0: 6865 7220 7468 616e 2061 6c6c 2074 6865  her than all the
+000172c0: 2061 6c69 676e 6d65 6e74 732e 2055 7365   alignments. Use
+000172d0: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
+000172e0: 6465 2061 2070 7265 7365 7420 7370 6563  de a preset spec
+000172f0: 6965 7320 7472 6565 2066 6f72 2074 6865  ies tree for the
+00017300: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00017310: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+00017320: 6f66 2074 6865 2063 6f6e 6361 7465 6e61  of the concatena
+00017330: 7465 6420 616c 6967 6e6d 656e 7420 7768  ted alignment wh
+00017340: 696c 6520 7468 6520 7265 6d61 696e 696e  ile the remainin
+00017350: 6720 616c 6967 6e6d 656e 7473 2077 696c  g alignments wil
+00017360: 6c20 6265 2061 6761 696e 7374 2061 6e20  l be against an 
+00017370: 6175 746f 6d61 7465 6c79 2069 6e66 6572  automately infer
+00017380: 7265 6420 7472 6565 2066 726f 6d20 6066  red tree from `f
+00017390: 6173 7474 7265 6560 206f 7220 6069 7174  asttree` or `iqt
+000173a0: 7265 6560 2e20 496e 2074 6865 2065 6e64  ree`. In the end
+000173b0: 2c20 7573 6572 7320 7769 6c6c 2067 6574  , users will get
+000173c0: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
+000173d0: 623e 2074 7265 652c 2061 202a 4b2a 3c73  b> tree, a *K*<s
+000173e0: 7562 3e41 3c2f 7375 623e 2074 7265 6520  ub>A</sub> tree 
+000173f0: 616e 6420 6120 cf89 2074 7265 6520 7065  and a .. tree pe
+00017400: 7220 6661 6d20 616e 6420 666f 7220 7468  r fam and for th
+00017410: 6520 636f 6e63 6174 656e 6174 6564 2061  e concatenated a
+00017420: 6c69 676e 6d65 6e74 2e0a 0a60 6060 0a77  lignment...```.w
+00017430: 6764 206b 7364 2064 6174 612f 6b73 7472  gd ksd data/kstr
+00017440: 6565 5f64 6174 612f 6661 6d2e 7473 7620  ee_data/fam.tsv 
+00017450: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
+00017460: 2f41 636f 7275 735f 7461 7461 7269 6e6f  /Acorus_tatarino
+00017470: 7769 6920 6461 7461 2f6b 7374 7265 655f  wii data/kstree_
+00017480: 6461 7461 2f41 6d62 6f72 656c 6c61 5f74  data/Amborella_t
+00017490: 7269 6368 6f70 6f64 6120 6461 7461 2f6b  richopoda data/k
+000174a0: 7374 7265 655f 6461 7461 2f41 7175 696c  stree_data/Aquil
+000174b0: 6567 6961 5f63 6f65 7275 6c65 6120 6461  egia_coerulea da
+000174c0: 7461 2f6b 7374 7265 655f 6461 7461 2f41  ta/kstree_data/A
+000174d0: 7269 7374 6f6c 6f63 6869 615f 6669 6d62  ristolochia_fimb
+000174e0: 7269 6174 6120 6461 7461 2f6b 7374 7265  riata data/kstre
+000174f0: 655f 6461 7461 2f43 7963 6173 5f70 616e  e_data/Cycas_pan
+00017500: 7a68 6968 7561 656e 7369 202d 2d6b 7374  zhihuaensi --kst
+00017510: 7265 6520 2d2d 7370 6563 6965 7374 7265  ree --speciestre
+00017520: 6520 6461 7461 2f6b 7374 7265 655f 6461  e data/kstree_da
+00017530: 7461 2f73 7065 6369 6573 5f74 7265 6531  ta/species_tree1
+00017540: 2e6e 7720 2d2d 6f6e 6c79 636f 6e63 6174  .nw --onlyconcat
+00017550: 6b73 7472 6565 202d 6f20 7767 645f 6b73  kstree -o wgd_ks
+00017560: 7472 6565 5f74 6f70 6f6c 6f67 7931 0a60  tree_topology1.`
+00017570: 6060 0a0a 215b 5d28 6461 7461 2f6b 7374  ``..![](data/kst
+00017580: 7265 655f 7265 7375 6c74 732f 6b73 7472  ree_results/kstr
+00017590: 6565 2e73 7667 290a 0a41 626f 7665 2077  ee.svg)..Above w
+000175a0: 6520 7573 6564 2074 6872 6565 2061 6c74  e used three alt
+000175b0: 6572 6e61 7469 7665 2074 6f70 6f6c 6f67  ernative topolog
+000175c0: 6965 7320 746f 2069 6e66 6572 2074 6865  ies to infer the
+000175d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000175e0: 2074 7265 6520 7768 6963 6820 6c65 6420   tree which led 
+000175f0: 746f 2064 6966 6665 7265 6e74 2062 7261  to different bra
+00017600: 6e63 6820 6c65 6e67 7468 2065 7374 696d  nch length estim
+00017610: 6174 696f 6e2e 204e 6f74 6520 7468 6174  ation. Note that
+00017620: 2074 6865 2066 616d 696c 6965 7320 7765   the families we
+00017630: 2075 7365 6420 7765 7265 206f 6e6c 7920   used were only 
+00017640: 7477 6f20 676c 6f62 616c 204d 5242 4820  two global MRBH 
+00017650: 6661 6d69 6c69 6573 2066 6f72 2074 6865  families for the
+00017660: 2070 7572 706f 7365 206f 6620 696c 6c75   purpose of illu
+00017670: 7374 7261 7469 6f6e 2e20 546f 2061 6371  stration. To acq
+00017680: 7569 7265 2061 6e20 6163 6375 7261 7465  uire an accurate
+00017690: 2070 726f 6669 6c65 206f 6620 7468 6520   profile of the 
+000176a0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+000176b0: 6520 7661 7269 6174 696f 6e2c 206f 7274  e variation, ort
+000176c0: 686f 6c6f 6775 6573 2061 7420 7468 6520  hologues at the 
+000176d0: 7768 6f6c 6520 6765 6e6f 6d65 2073 6361  whole genome sca
+000176e0: 6c65 2073 686f 756c 6420 6265 2075 7365  le should be use
+000176f0: 642e 0a0a 2323 2043 6974 6174 696f 6e0a  d...## Citation.
+00017700: 200a 506c 6561 7365 2063 6974 6520 7573   .Please cite us
+00017710: 2061 7420 6874 7470 733a 2f2f 646f 692e   at https://doi.
+00017720: 6f72 672f 3130 2e31 3030 372f 3937 382d  org/10.1007/978-
+00017730: 312d 3037 3136 2d32 3536 312d 335f 3120  1-0716-2561-3_1 
+00017740: 616e 6420 6874 7470 733a 2f2f 646f 692e  and https://doi.
+00017750: 6f72 672f 3130 2e31 3039 332f 6269 6f69  org/10.1093/bioi
+00017760: 6e66 6f72 6d61 7469 6373 2f62 7461 6532  nformatics/btae2
+00017770: 3732 2e0a 0a60 6060 0a48 656e 6763 6869  72...```.Hengchi
+00017780: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
+00017790: 6165 6e65 706f 656c 2028 3230 3233 292e  aenepoel (2023).
+000177a0: 2049 6e66 6572 656e 6365 206f 6620 416e   Inference of An
+000177b0: 6369 656e 7420 506f 6c79 706c 6f69 6479  cient Polyploidy
+000177c0: 2066 726f 6d20 4765 6e6f 6d69 6320 4461   from Genomic Da
+000177d0: 7461 2e20 496e 3a20 5661 6e20 6465 2050  ta. In: Van de P
+000177e0: 6565 722c 2059 2e20 2865 6473 2920 506f  eer, Y. (eds) Po
+000177f0: 6c79 706c 6f69 6479 2e20 4d65 7468 6f64  lyploidy. Method
+00017800: 7320 696e 204d 6f6c 6563 756c 6172 2042  s in Molecular B
+00017810: 696f 6c6f 6779 2c20 766f 6c20 3235 3435  iology, vol 2545
+00017820: 2e20 4875 6d61 6e61 2c20 4e65 7720 596f  . Humana, New Yo
+00017830: 726b 2c20 4e59 2e20 6874 7470 733a 2f2f  rk, NY. https://
+00017840: 646f 692e 6f72 672f 3130 2e31 3030 372f  doi.org/10.1007/
+00017850: 3937 382d 312d 3037 3136 2d32 3536 312d  978-1-0716-2561-
+00017860: 335f 310a 4865 6e67 6368 6920 4368 656e  3_1.Hengchi Chen
+00017870: 2c20 4172 7468 7572 205a 7761 656e 6570  , Arthur Zwaenep
+00017880: 6f65 6c2c 2059 7665 7320 5661 6e20 6465  oel, Yves Van de
+00017890: 2050 6565 7220 2832 3032 3429 2e20 7767   Peer (2024). wg
+000178a0: 6420 7632 3a20 6120 7375 6974 6520 6f66  d v2: a suite of
+000178b0: 2074 6f6f 6c73 2074 6f20 756e 636f 7665   tools to uncove
+000178c0: 7220 616e 6420 6461 7465 2061 6e63 6965  r and date ancie
+000178d0: 6e74 2070 6f6c 7970 6c6f 6964 7920 616e  nt polyploidy an
+000178e0: 6420 7768 6f6c 652d 6765 6e6f 6d65 2064  d whole-genome d
+000178f0: 7570 6c69 6361 7469 6f6e 2e20 4269 6f69  uplication. Bioi
+00017900: 6e66 6f72 6d61 7469 6373 2c20 6874 7470  nformatics, http
+00017910: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
+00017920: 3039 332f 6269 6f69 6e66 6f72 6d61 7469  093/bioinformati
+00017930: 6373 2f62 7461 6532 3732 0a60 6060 0a0a  cs/btae272.```..
+00017940: 466f 7220 6369 7461 7469 6f6e 206f 6620  For citation of 
+00017950: 7468 6520 746f 6f6c 7320 7573 6564 2069  the tools used i
+00017960: 6e20 7767 642c 2070 6c65 6173 6520 636f  n wgd, please co
+00017970: 6e73 756c 7420 7468 6520 646f 6375 6d65  nsult the docume
+00017980: 6e74 6174 696f 6e20 6174 0a68 7474 7073  ntation at.https
+00017990: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
+000179a0: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
+000179b0: 742f 6369 7461 7469 6f6e 2e68 746d 6c2e  t/citation.html.
+000179c0: 0a0a                                     ..
```

### Comparing `wgd-2.0.32/wgd.egg-info/requires.txt` & `wgd-2.0.33/wgd.egg-info/requires.txt`

 * *Files identical despite different names*

