# Comparing `tmp/DAJIN2-0.4.3.zip` & `tmp/dajin2-0.4.4.zip`

## zipinfo {}

```diff
@@ -1,81 +1,81 @@
-Zip file size: 87340 bytes, number of entries: 79
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/
--rw-r--r--  2.0 unx      270 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/requirements.txt
--rw-r--r--  2.0 unx     1189 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/setup.py
--rw-r--r--  2.0 unx       38 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/setup.cfg
--rw-r--r--  2.0 unx    14424 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/README.md
--rw-r--r--  2.0 unx     1069 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/LICENSE
--rw-r--r--  2.0 unx      139 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/MANIFEST.in
--rw-r--r--  2.0 unx    15508 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/PKG-INFO
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/
--rw-r--r--  2.0 unx       47 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/entry_points.txt
--rw-r--r--  2.0 unx        1 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/dependency_links.txt
--rw-r--r--  2.0 unx      236 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/requires.txt
--rw-r--r--  2.0 unx        7 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/top_level.txt
--rw-r--r--  2.0 unx     2272 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/SOURCES.txt
--rw-r--r--  2.0 unx    15508 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2.egg-info/PKG-INFO
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/static/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/templates/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/
--rw-r--r--  2.0 unx        0 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/__init__.py
--rw-r--r--  2.0 unx     2860 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/view.py
--rw-r--r--  2.0 unx     9634 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/main.py
--rw-r--r--  2.0 unx     2947 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/gui.py
--rw-r--r--  2.0 unx     1901 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/template_igvjs.html
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/report/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/classification/
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/
--rw-r--r--  2.0 unx        0 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/__init__.py
--rw-r--r--  2.0 unx    10663 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/core.py
--rw-r--r--  2.0 unx      699 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/__init__.py
--rw-r--r--  2.0 unx    12789 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/mutation_extractor.py
--rw-r--r--  2.0 unx    21347 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/insertions_to_fasta.py
--rw-r--r--  2.0 unx     3585 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/input_formatter.py
--rw-r--r--  2.0 unx     1032 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/directory_manager.py
--rw-r--r--  2.0 unx      790 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/cache_checker.py
--rw-r--r--  2.0 unx     1636 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/knockin_handler.py
--rw-r--r--  2.0 unx     1656 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/homopolymer_handler.py
--rw-r--r--  2.0 unx     1931 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/genome_fetcher.py
--rw-r--r--  2.0 unx     4630 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/mapping.py
--rw-r--r--  2.0 unx     8891 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/preprocess/midsv_caller.py
--rw-r--r--  2.0 unx      402 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/__init__.py
--rw-r--r--  2.0 unx     5690 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/mutation_extractor.py
--rw-r--r--  2.0 unx     4120 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/similarity_searcher.py
--rw-r--r--  2.0 unx     2847 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/name_handler.py
--rw-r--r--  2.0 unx      374 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/clust_formatter.py
--rw-r--r--  2.0 unx     4202 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/consensus/consensus.py
--rw-r--r--  2.0 unx      142 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/report/__init__.py
--rw-r--r--  2.0 unx     4801 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/report/mutation_exporter.py
--rw-r--r--  2.0 unx     2176 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/report/insertion_reflector.py
--rw-r--r--  2.0 unx     2859 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/report/sequence_exporter.py
--rw-r--r--  2.0 unx     6248 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/report/bam_exporter.py
--rw-r--r--  2.0 unx       67 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/classification/__init__.py
--rw-r--r--  2.0 unx     3715 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/classification/allele_merger.py
--rw-r--r--  2.0 unx     1795 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/classification/classifier.py
--rw-r--r--  2.0 unx     2298 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/kmer_generator.py
--rw-r--r--  2.0 unx     1193 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_updator.py
--rw-r--r--  2.0 unx      210 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/__init__.py
--rw-r--r--  2.0 unx     6522 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/score_handler.py
--rw-r--r--  2.0 unx     3708 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_merger.py
--rw-r--r--  2.0 unx     3215 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_extractor.py
--rw-r--r--  2.0 unx     4291 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/strand_bias_handler.py
--rw-r--r--  2.0 unx     1460 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/appender.py
--rw-r--r--  2.0 unx     3421 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/core/clustering/clustering.py
-drwxr-xr-x  2.0 unx        0 b- stor 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/static/css/
--rw-r--r--  2.0 unx       27 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/static/css/style.css
--rw-r--r--  2.0 unx     1777 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/templates/index.html
--rw-r--r--  2.0 unx     2374 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/multiprocess.py
--rw-r--r--  2.0 unx     7985 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/cssplits_handler.py
--rw-r--r--  2.0 unx     3184 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/fastx_handler.py
--rw-r--r--  2.0 unx      245 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/dna_handler.py
--rw-r--r--  2.0 unx     4775 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/report_generator.py
--rw-r--r--  2.0 unx     5343 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/io.py
--rw-r--r--  2.0 unx     6579 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/input_validator.py
--rw-r--r--  2.0 unx     2112 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/config.py
--rw-r--r--  2.0 unx     7420 b- defN 24-Mar-29 02:05 DAJIN2-0.4.3/src/DAJIN2/utils/sam_handler.py
-79 files, 245276 bytes uncompressed, 74280 bytes compressed:  69.7%
+Zip file size: 87134 bytes, number of entries: 79
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/
+-rw-r--r--  2.0 unx     1069 b- defN 24-Apr-23 08:29 dajin2-0.4.4/LICENSE
+-rw-r--r--  2.0 unx     1189 b- defN 24-Apr-23 08:29 dajin2-0.4.4/setup.py
+-rw-r--r--  2.0 unx    12234 b- defN 24-Apr-23 08:29 dajin2-0.4.4/README.md
+-rw-r--r--  2.0 unx       38 b- defN 24-Apr-23 08:29 dajin2-0.4.4/setup.cfg
+-rw-r--r--  2.0 unx      270 b- defN 24-Apr-23 08:29 dajin2-0.4.4/requirements.txt
+-rw-r--r--  2.0 unx    13318 b- defN 24-Apr-23 08:29 dajin2-0.4.4/PKG-INFO
+-rw-r--r--  2.0 unx      139 b- defN 24-Apr-23 08:29 dajin2-0.4.4/MANIFEST.in
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/
+-rw-r--r--  2.0 unx        1 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/dependency_links.txt
+-rw-r--r--  2.0 unx     2272 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/SOURCES.txt
+-rw-r--r--  2.0 unx        7 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/top_level.txt
+-rw-r--r--  2.0 unx    13318 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/PKG-INFO
+-rw-r--r--  2.0 unx      236 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/requires.txt
+-rw-r--r--  2.0 unx       47 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2.egg-info/entry_points.txt
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/static/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/templates/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/
+-rw-r--r--  2.0 unx     2947 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/gui.py
+-rw-r--r--  2.0 unx     1901 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/template_igvjs.html
+-rw-r--r--  2.0 unx        0 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/__init__.py
+-rw-r--r--  2.0 unx     2860 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/view.py
+-rw-r--r--  2.0 unx     9634 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/main.py
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/static/css/
+-rw-r--r--  2.0 unx       27 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/static/css/style.css
+-rw-r--r--  2.0 unx     1777 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/templates/index.html
+-rw-r--r--  2.0 unx      245 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/dna_handler.py
+-rw-r--r--  2.0 unx     2374 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/multiprocess.py
+-rw-r--r--  2.0 unx     7420 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/sam_handler.py
+-rw-r--r--  2.0 unx     7985 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/cssplits_handler.py
+-rw-r--r--  2.0 unx     3184 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/fastx_handler.py
+-rw-r--r--  2.0 unx     2112 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/config.py
+-rw-r--r--  2.0 unx     5411 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/io.py
+-rw-r--r--  2.0 unx     4775 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/report_generator.py
+-rw-r--r--  2.0 unx     6579 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/utils/input_validator.py
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/classification/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/report/
+drwxr-xr-x  2.0 unx        0 b- stor 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/
+-rw-r--r--  2.0 unx    10663 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/core.py
+-rw-r--r--  2.0 unx        0 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/__init__.py
+-rw-r--r--  2.0 unx     8891 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/midsv_caller.py
+-rw-r--r--  2.0 unx    21347 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/insertions_to_fasta.py
+-rw-r--r--  2.0 unx    13202 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/mutation_extractor.py
+-rw-r--r--  2.0 unx     1032 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/directory_manager.py
+-rw-r--r--  2.0 unx     1656 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/homopolymer_handler.py
+-rw-r--r--  2.0 unx      699 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/__init__.py
+-rw-r--r--  2.0 unx     3585 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/input_formatter.py
+-rw-r--r--  2.0 unx     1931 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/genome_fetcher.py
+-rw-r--r--  2.0 unx     1636 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/knockin_handler.py
+-rw-r--r--  2.0 unx     4630 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/mapping.py
+-rw-r--r--  2.0 unx      790 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/preprocess/cache_checker.py
+-rw-r--r--  2.0 unx     3715 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/classification/allele_merger.py
+-rw-r--r--  2.0 unx     1795 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/classification/classifier.py
+-rw-r--r--  2.0 unx       67 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/classification/__init__.py
+-rw-r--r--  2.0 unx     3215 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/label_extractor.py
+-rw-r--r--  2.0 unx     6522 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/score_handler.py
+-rw-r--r--  2.0 unx     2298 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/kmer_generator.py
+-rw-r--r--  2.0 unx     3516 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/clustering.py
+-rw-r--r--  2.0 unx      210 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/__init__.py
+-rw-r--r--  2.0 unx     1460 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/appender.py
+-rw-r--r--  2.0 unx     1193 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/label_updator.py
+-rw-r--r--  2.0 unx     3708 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/label_merger.py
+-rw-r--r--  2.0 unx     4999 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/clustering/strand_bias_handler.py
+-rw-r--r--  2.0 unx     6248 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/report/bam_exporter.py
+-rw-r--r--  2.0 unx     4801 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/report/mutation_exporter.py
+-rw-r--r--  2.0 unx      142 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/report/__init__.py
+-rw-r--r--  2.0 unx     2176 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/report/insertion_reflector.py
+-rw-r--r--  2.0 unx     2859 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/report/sequence_exporter.py
+-rw-r--r--  2.0 unx     4202 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/consensus.py
+-rw-r--r--  2.0 unx     2847 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/name_handler.py
+-rw-r--r--  2.0 unx     5690 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/mutation_extractor.py
+-rw-r--r--  2.0 unx      402 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/__init__.py
+-rw-r--r--  2.0 unx      374 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/clust_formatter.py
+-rw-r--r--  2.0 unx     4120 b- defN 24-Apr-23 08:29 dajin2-0.4.4/src/DAJIN2/core/consensus/similarity_searcher.py
+79 files, 239990 bytes uncompressed, 74074 bytes compressed:  69.1%
```

## zipnote {}

```diff
@@ -1,238 +1,238 @@
-Filename: DAJIN2-0.4.3/
+Filename: dajin2-0.4.4/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/
+Filename: dajin2-0.4.4/src/
 Comment: 
 
-Filename: DAJIN2-0.4.3/requirements.txt
+Filename: dajin2-0.4.4/LICENSE
 Comment: 
 
-Filename: DAJIN2-0.4.3/setup.py
+Filename: dajin2-0.4.4/setup.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/setup.cfg
+Filename: dajin2-0.4.4/README.md
 Comment: 
 
-Filename: DAJIN2-0.4.3/README.md
+Filename: dajin2-0.4.4/setup.cfg
 Comment: 
 
-Filename: DAJIN2-0.4.3/LICENSE
+Filename: dajin2-0.4.4/requirements.txt
 Comment: 
 
-Filename: DAJIN2-0.4.3/MANIFEST.in
+Filename: dajin2-0.4.4/PKG-INFO
 Comment: 
 
-Filename: DAJIN2-0.4.3/PKG-INFO
+Filename: dajin2-0.4.4/MANIFEST.in
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/
+Filename: dajin2-0.4.4/src/DAJIN2/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/entry_points.txt
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/dependency_links.txt
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/dependency_links.txt
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/SOURCES.txt
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/requires.txt
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/top_level.txt
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/top_level.txt
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/PKG-INFO
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/SOURCES.txt
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/requires.txt
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2.egg-info/PKG-INFO
+Filename: dajin2-0.4.4/src/DAJIN2.egg-info/entry_points.txt
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/
+Filename: dajin2-0.4.4/src/DAJIN2/static/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/static/
+Filename: dajin2-0.4.4/src/DAJIN2/templates/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/templates/
+Filename: dajin2-0.4.4/src/DAJIN2/utils/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/
+Filename: dajin2-0.4.4/src/DAJIN2/core/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/gui.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/view.py
+Filename: dajin2-0.4.4/src/DAJIN2/template_igvjs.html
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/main.py
+Filename: dajin2-0.4.4/src/DAJIN2/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/gui.py
+Filename: dajin2-0.4.4/src/DAJIN2/view.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/template_igvjs.html
+Filename: dajin2-0.4.4/src/DAJIN2/main.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/
+Filename: dajin2-0.4.4/src/DAJIN2/static/css/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/
+Filename: dajin2-0.4.4/src/DAJIN2/static/css/style.css
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/report/
+Filename: dajin2-0.4.4/src/DAJIN2/templates/index.html
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/classification/
+Filename: dajin2-0.4.4/src/DAJIN2/utils/dna_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/
+Filename: dajin2-0.4.4/src/DAJIN2/utils/multiprocess.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/sam_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/core.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/cssplits_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/fastx_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/mutation_extractor.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/config.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/insertions_to_fasta.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/io.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/input_formatter.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/report_generator.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/directory_manager.py
+Filename: dajin2-0.4.4/src/DAJIN2/utils/input_validator.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/cache_checker.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/knockin_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/classification/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/homopolymer_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/genome_fetcher.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/report/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/mapping.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/preprocess/midsv_caller.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/core.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/mutation_extractor.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/midsv_caller.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/similarity_searcher.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/insertions_to_fasta.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/name_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/mutation_extractor.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/clust_formatter.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/directory_manager.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/consensus/consensus.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/homopolymer_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/report/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/report/mutation_exporter.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/input_formatter.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/report/insertion_reflector.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/genome_fetcher.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/report/sequence_exporter.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/knockin_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/report/bam_exporter.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/mapping.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/classification/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/preprocess/cache_checker.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/classification/allele_merger.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/classification/allele_merger.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/classification/classifier.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/classification/classifier.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/kmer_generator.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/classification/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_updator.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/label_extractor.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/__init__.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/score_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/score_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/kmer_generator.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_merger.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/clustering.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_extractor.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/strand_bias_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/appender.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/appender.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/label_updator.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/core/clustering/clustering.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/label_merger.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/static/css/
+Filename: dajin2-0.4.4/src/DAJIN2/core/clustering/strand_bias_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/static/css/style.css
+Filename: dajin2-0.4.4/src/DAJIN2/core/report/bam_exporter.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/templates/index.html
+Filename: dajin2-0.4.4/src/DAJIN2/core/report/mutation_exporter.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/multiprocess.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/report/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/cssplits_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/report/insertion_reflector.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/fastx_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/report/sequence_exporter.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/dna_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/consensus.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/report_generator.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/name_handler.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/io.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/mutation_extractor.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/input_validator.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/__init__.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/config.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/clust_formatter.py
 Comment: 
 
-Filename: DAJIN2-0.4.3/src/DAJIN2/utils/sam_handler.py
+Filename: dajin2-0.4.4/src/DAJIN2/core/consensus/similarity_searcher.py
 Comment: 
 
 Zip file comment:
```

## Comparing `DAJIN2-0.4.3/setup.py` & `dajin2-0.4.4/setup.py`

 * *Files 0% similar despite different names*

```diff
@@ -5,15 +5,15 @@
 long_description = (this_directory / "README.md").read_text()
 
 with open("requirements.txt") as requirements_file:
     install_requirements = requirements_file.read().splitlines()
 
 setuptools.setup(
     name="DAJIN2",
-    version="0.4.3",
+    version="0.4.4",
     author="Akihiro Kuno",
     author_email="akuno@md.tsukuba.ac.jp",
     description="One-step genotyping tools for targeted long-read sequencing",
     long_description=long_description,
     long_description_content_type="text/markdown",
     url="https://github.com/akikuno/DAJIN2",
     install_requires=install_requirements,
```

## Comparing `DAJIN2-0.4.3/README.md` & `dajin2-0.4.4/README.md`

 * *Files 11% similar despite different names*

```diff
@@ -129,15 +129,15 @@
 -v, --version             Display the version number and exit.
 ```
 
 #### Example
 
 ```bash
 # Download example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_single.tar.gz
+curl -LJO https://github.com/akikuno/DAJIN2/raw/main/examples/example_single.tar.gz
 tar -xf example_single.tar.gz
 
 # Run DAJIN2
 DAJIN2 \
     --control example_single/control \
     --sample example_single/sample \
     --allele example_single/stx2_deletion.fa \
@@ -193,81 +193,46 @@
   -h, --help                Display this help message and exit.
 ```
 
 #### Example
 
 ```bash
 # Donwload the example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
+curl -LJO https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
 tar -xf example_batch.tar.gz
 
 # Run DAJIN2
 DAJIN2 batch --file example_batch/batch.csv --threads 4
 ```
 
-<!-- ```bash
-# Donwload the example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
-tar -xf example_batch.tar.gz
-
-# Run DAJIN2
-DAJIN2 batch --file example-batch/batch.csv --threads 3
-
-# 2023-07-31 17:01:10: example-batch/tyr_control.fq.gz is now processing...
-# 2023-07-31 17:01:16: Preprocess example-batch/tyr_control.fq.gz...
-# 2023-07-31 17:01:48: Output BAM files of example-batch/tyr_control.fq.gz...
-# 2023-07-31 17:01:52: 🍵 example-batch/tyr_control.fq.gz is finished!
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_50%.fq.gz is now processing...
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_10%.fq.gz is now processing...
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_01%.fq.gz is now processing...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:17: Classify example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:19: Clustering example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:34: Classify example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:02:35: Classify example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:39: Clustering example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:02:39: Clustering example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:53: Consensus calling of example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:59: Output reports of example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:03:04: 🍵 example-batch/tyr_c230gt_50%.fq.gz is finished!
-# 2023-07-31 17:03:39: Consensus calling of example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:03:51: Output reports of example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:04:03: 🍵 example-batch/tyr_c230gt_01%.fq.gz is finished!
-# 2023-07-31 17:04:08: Consensus calling of example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:04:16: Output reports of example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:04:24: 🍵 example-batch/tyr_c230gt_10%.fq.gz is finished!
-# 🎉 Finished! Open DAJIN_Results/tyr-substitution to see the report.
-``` -->
 
 ## 📈 Report Contents
 
 Upon completion of DAJIN2 processing, a directory named **DAJIN_Results** is generated.  
 Inside the **DAJIN_Results** directory, the following files can be found:  
 
 ```
 DAJIN_Results/tyr-substitution
 ├── BAM
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   ├── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   ├── tyr_c230gt_50
 │   └── tyr_control
 ├── FASTA
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   └── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   └── tyr_c230gt_50
 ├── HTML
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   └── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   └── tyr_c230gt_50
 ├── MUTATION_INFO
-│   ├── tyr_c230gt_01%.csv
-│   ├── tyr_c230gt_10%.csv
-│   └── tyr_c230gt_50%.csv
+│   ├── tyr_c230gt_01.csv
+│   ├── tyr_c230gt_10.csv
+│   └── tyr_c230gt_50.csv
 ├── read_plot.html
 ├── read_plot.pdf
 └── read_summary.xlsx
 ```
 
 ### 1. BAM
```

## Comparing `DAJIN2-0.4.3/LICENSE` & `dajin2-0.4.4/LICENSE`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/PKG-INFO` & `dajin2-0.4.4/PKG-INFO`

 * *Files 14% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: DAJIN2
-Version: 0.4.3
+Version: 0.4.4
 Summary: One-step genotyping tools for targeted long-read sequencing
 Home-page: https://github.com/akikuno/DAJIN2
 Author: Akihiro Kuno
 Author-email: akuno@md.tsukuba.ac.jp
 Classifier: Environment :: Console
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: MIT License
@@ -162,15 +162,15 @@
 -v, --version             Display the version number and exit.
 ```
 
 #### Example
 
 ```bash
 # Download example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_single.tar.gz
+curl -LJO https://github.com/akikuno/DAJIN2/raw/main/examples/example_single.tar.gz
 tar -xf example_single.tar.gz
 
 # Run DAJIN2
 DAJIN2 \
     --control example_single/control \
     --sample example_single/sample \
     --allele example_single/stx2_deletion.fa \
@@ -226,81 +226,46 @@
   -h, --help                Display this help message and exit.
 ```
 
 #### Example
 
 ```bash
 # Donwload the example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
+curl -LJO https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
 tar -xf example_batch.tar.gz
 
 # Run DAJIN2
 DAJIN2 batch --file example_batch/batch.csv --threads 4
 ```
 
-<!-- ```bash
-# Donwload the example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
-tar -xf example_batch.tar.gz
-
-# Run DAJIN2
-DAJIN2 batch --file example-batch/batch.csv --threads 3
-
-# 2023-07-31 17:01:10: example-batch/tyr_control.fq.gz is now processing...
-# 2023-07-31 17:01:16: Preprocess example-batch/tyr_control.fq.gz...
-# 2023-07-31 17:01:48: Output BAM files of example-batch/tyr_control.fq.gz...
-# 2023-07-31 17:01:52: 🍵 example-batch/tyr_control.fq.gz is finished!
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_50%.fq.gz is now processing...
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_10%.fq.gz is now processing...
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_01%.fq.gz is now processing...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:17: Classify example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:19: Clustering example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:34: Classify example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:02:35: Classify example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:39: Clustering example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:02:39: Clustering example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:53: Consensus calling of example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:59: Output reports of example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:03:04: 🍵 example-batch/tyr_c230gt_50%.fq.gz is finished!
-# 2023-07-31 17:03:39: Consensus calling of example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:03:51: Output reports of example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:04:03: 🍵 example-batch/tyr_c230gt_01%.fq.gz is finished!
-# 2023-07-31 17:04:08: Consensus calling of example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:04:16: Output reports of example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:04:24: 🍵 example-batch/tyr_c230gt_10%.fq.gz is finished!
-# 🎉 Finished! Open DAJIN_Results/tyr-substitution to see the report.
-``` -->
 
 ## 📈 Report Contents
 
 Upon completion of DAJIN2 processing, a directory named **DAJIN_Results** is generated.  
 Inside the **DAJIN_Results** directory, the following files can be found:  
 
 ```
 DAJIN_Results/tyr-substitution
 ├── BAM
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   ├── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   ├── tyr_c230gt_50
 │   └── tyr_control
 ├── FASTA
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   └── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   └── tyr_c230gt_50
 ├── HTML
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   └── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   └── tyr_c230gt_50
 ├── MUTATION_INFO
-│   ├── tyr_c230gt_01%.csv
-│   ├── tyr_c230gt_10%.csv
-│   └── tyr_c230gt_50%.csv
+│   ├── tyr_c230gt_01.csv
+│   ├── tyr_c230gt_10.csv
+│   └── tyr_c230gt_50.csv
 ├── read_plot.html
 ├── read_plot.pdf
 └── read_summary.xlsx
 ```
 
 ### 1. BAM
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2.egg-info/SOURCES.txt` & `dajin2-0.4.4/src/DAJIN2.egg-info/SOURCES.txt`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2.egg-info/PKG-INFO` & `dajin2-0.4.4/src/DAJIN2.egg-info/PKG-INFO`

 * *Files 14% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: DAJIN2
-Version: 0.4.3
+Version: 0.4.4
 Summary: One-step genotyping tools for targeted long-read sequencing
 Home-page: https://github.com/akikuno/DAJIN2
 Author: Akihiro Kuno
 Author-email: akuno@md.tsukuba.ac.jp
 Classifier: Environment :: Console
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: MIT License
@@ -162,15 +162,15 @@
 -v, --version             Display the version number and exit.
 ```
 
 #### Example
 
 ```bash
 # Download example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_single.tar.gz
+curl -LJO https://github.com/akikuno/DAJIN2/raw/main/examples/example_single.tar.gz
 tar -xf example_single.tar.gz
 
 # Run DAJIN2
 DAJIN2 \
     --control example_single/control \
     --sample example_single/sample \
     --allele example_single/stx2_deletion.fa \
@@ -226,81 +226,46 @@
   -h, --help                Display this help message and exit.
 ```
 
 #### Example
 
 ```bash
 # Donwload the example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
+curl -LJO https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
 tar -xf example_batch.tar.gz
 
 # Run DAJIN2
 DAJIN2 batch --file example_batch/batch.csv --threads 4
 ```
 
-<!-- ```bash
-# Donwload the example dataset
-wget https://github.com/akikuno/DAJIN2/raw/main/examples/example_batch.tar.gz
-tar -xf example_batch.tar.gz
-
-# Run DAJIN2
-DAJIN2 batch --file example-batch/batch.csv --threads 3
-
-# 2023-07-31 17:01:10: example-batch/tyr_control.fq.gz is now processing...
-# 2023-07-31 17:01:16: Preprocess example-batch/tyr_control.fq.gz...
-# 2023-07-31 17:01:48: Output BAM files of example-batch/tyr_control.fq.gz...
-# 2023-07-31 17:01:52: 🍵 example-batch/tyr_control.fq.gz is finished!
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_50%.fq.gz is now processing...
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_10%.fq.gz is now processing...
-# 2023-07-31 17:01:52: example-batch/tyr_c230gt_01%.fq.gz is now processing...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:01:55: Preprocess example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:17: Classify example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:19: Clustering example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:34: Classify example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:02:35: Classify example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:39: Clustering example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:02:39: Clustering example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:02:53: Consensus calling of example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:02:59: Output reports of example-batch/tyr_c230gt_50%.fq.gz...
-# 2023-07-31 17:03:04: 🍵 example-batch/tyr_c230gt_50%.fq.gz is finished!
-# 2023-07-31 17:03:39: Consensus calling of example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:03:51: Output reports of example-batch/tyr_c230gt_01%.fq.gz...
-# 2023-07-31 17:04:03: 🍵 example-batch/tyr_c230gt_01%.fq.gz is finished!
-# 2023-07-31 17:04:08: Consensus calling of example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:04:16: Output reports of example-batch/tyr_c230gt_10%.fq.gz...
-# 2023-07-31 17:04:24: 🍵 example-batch/tyr_c230gt_10%.fq.gz is finished!
-# 🎉 Finished! Open DAJIN_Results/tyr-substitution to see the report.
-``` -->
 
 ## 📈 Report Contents
 
 Upon completion of DAJIN2 processing, a directory named **DAJIN_Results** is generated.  
 Inside the **DAJIN_Results** directory, the following files can be found:  
 
 ```
 DAJIN_Results/tyr-substitution
 ├── BAM
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   ├── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   ├── tyr_c230gt_50
 │   └── tyr_control
 ├── FASTA
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   └── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   └── tyr_c230gt_50
 ├── HTML
-│   ├── tyr_c230gt_01%
-│   ├── tyr_c230gt_10%
-│   └── tyr_c230gt_50%
+│   ├── tyr_c230gt_01
+│   ├── tyr_c230gt_10
+│   └── tyr_c230gt_50
 ├── MUTATION_INFO
-│   ├── tyr_c230gt_01%.csv
-│   ├── tyr_c230gt_10%.csv
-│   └── tyr_c230gt_50%.csv
+│   ├── tyr_c230gt_01.csv
+│   ├── tyr_c230gt_10.csv
+│   └── tyr_c230gt_50.csv
 ├── read_plot.html
 ├── read_plot.pdf
 └── read_summary.xlsx
 ```
 
 ### 1. BAM
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2/view.py` & `dajin2-0.4.4/src/DAJIN2/view.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/main.py` & `dajin2-0.4.4/src/DAJIN2/main.py`

 * *Files 6% similar despite different names*

```diff
@@ -16,15 +16,15 @@
 from itertools import groupby
 
 from DAJIN2 import gui, view
 from DAJIN2.core import core
 from DAJIN2.utils import io, config, report_generator, input_validator, multiprocess
 
 
-DAJIN_VERSION = "0.4.3"
+DAJIN_VERSION = "0.4.4"
 
 
 def generate_report(name: str) -> None:
     report_generator.report(name)
     print(
         f"\N{party popper} Finished! Open {config.DAJIN_RESULTS_DIR}/{name} to see the report.",
         file=sys.stderr,
@@ -54,29 +54,29 @@
 
 
 ################################################################################
 # Batch mode
 ################################################################################
 
 
-def validate_columns_of_batch_file(columns: list, filepath: str) -> None:
-    """Validate the columns of a batch file."""
-    required_columns = ["sample", "control", "allele", "name"]
-    accepted_columns = ["sample", "control", "allele", "name", "genome"]
+def validate_headers_of_batch_file(headers: list, filepath: str) -> None:
+    """Validate the headers of a batch file."""
+    required_headers = ["sample", "control", "allele", "name"]
+    accepted_headers = ["sample", "control", "allele", "name", "genome"]
 
-    if not set(required_columns).issubset(set(columns)):
-        raise ValueError(f"{filepath} must contain {', '.join(required_columns)} in the header")
+    if not set(required_headers).issubset(set(headers)):
+        raise ValueError(f"{filepath} must contain {', '.join(required_headers)} in the header")
 
-    if not set(columns).issubset(accepted_columns):
-        raise ValueError(f"Accepted header names of {filepath} are {', '.join(accepted_columns)}.")
+    if not set(headers).issubset(accepted_headers):
+        raise ValueError(f"Accepted header names of {filepath} are {', '.join(accepted_headers)}.")
 
 
-def create_argument_dict(columns: list, group: list, cache_urls_genome: dict, is_control: bool) -> dict:
-    """Create a dictionary of arguments from the given columns and group."""
-    args = dict(zip(columns, group))
+def create_argument_dict(headers: list, group: list, cache_urls_genome: dict, is_control: bool) -> dict:
+    """Create a dictionary of arguments from the given headers and group."""
+    args = dict(zip(headers, group))
     args["threads"] = 1  # Set the number of threads to 1 for batch mode
 
     # Assign the "sample" field depending on whether it's a control or not
     if is_control:
         args["sample"] = args["control"]
     else:
         if args["sample"] == args["control"]:
@@ -85,19 +85,19 @@
     if args.get("genome"):
         args.update(cache_urls_genome[args["genome"]])
 
     return args
 
 
 def run_DAJIN2(
-    groups: list, columns: list, cache_urls_genome: dict, is_control: bool = True, num_workers: int = 1
+    groups: list, headers: list, cache_urls_genome: dict, is_control: bool = True, num_workers: int = 1
 ) -> None:
     contents = []
     for group in groups:
-        args = create_argument_dict(columns, group, cache_urls_genome, is_control)
+        args = create_argument_dict(headers, group, cache_urls_genome, is_control)
         if args:  # Add args to contents only if it's not an empty dict
             contents.append(args)
 
     # Return a list of unique dictionaries
     contents_unique = [dict(item) for item in set(frozenset(d.items()) for d in contents)]
 
     contents_unique.sort(key=lambda x: x["sample"])
@@ -113,40 +113,40 @@
 
     if not Path(path_batchfile).exists():
         raise FileNotFoundError(f"'{path_batchfile}' does not exist.")
 
     inputs = io.load_batchfile(path_batchfile)
 
     # Validate Column of the batch file
-    columns = inputs[0]
-    validate_columns_of_batch_file(columns, path_batchfile)
+    headers = inputs[0]
+    validate_headers_of_batch_file(headers, path_batchfile)
 
     # Validate contents and fetch genome urls
     contents = inputs[1:]
     cache_urls_genome = dict()
-    index_of_name = columns.index("name")
+    index_of_name = headers.index("name")
     contents.sort(key=lambda x: x[index_of_name])
     for _, groups in groupby(contents, key=lambda x: x[index_of_name]):
         for group in groups:
-            args = dict(zip(columns, group))
+            args = dict(zip(headers, group))
             # validate contents in the batch file
             input_validator.validate_files(args["sample"], args["control"], args["allele"])
             # validate genome and fetch urls
             if args.get("genome") and args["genome"] not in cache_urls_genome:
                 urls_genome = input_validator.validate_genome_and_fetch_urls(args["genome"])
                 cache_urls_genome[args["genome"]] = urls_genome
     for name, groups in groupby(contents, key=lambda x: x[index_of_name]):
         # set logging to export log to stderr and file
         config.reset_logging()
         path_logfile = config.get_logfile()
         config.set_logging(path_logfile)
         groups = list(groups)
         # Run DAJIN2
-        run_DAJIN2(groups, columns, cache_urls_genome, is_control=True, num_workers=arguments["threads"])
-        run_DAJIN2(groups, columns, cache_urls_genome, is_control=False, num_workers=arguments["threads"])
+        run_DAJIN2(groups, headers, cache_urls_genome, is_control=True, num_workers=arguments["threads"])
+        run_DAJIN2(groups, headers, cache_urls_genome, is_control=False, num_workers=arguments["threads"])
         # Finish
         generate_report(name)
         shutil.move(path_logfile, Path("DAJIN_Results", name))
         if not arguments["debug"]:
             shutil.rmtree(Path("DAJIN_Results", ".tempdir", name))
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2/gui.py` & `dajin2-0.4.4/src/DAJIN2/gui.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/template_igvjs.html` & `dajin2-0.4.4/src/DAJIN2/template_igvjs.html`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/core.py` & `dajin2-0.4.4/src/DAJIN2/core/core.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/__init__.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/__init__.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/mutation_extractor.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/mutation_extractor.py`

 * *Files 7% similar despite different names*

```diff
@@ -85,36 +85,36 @@
 
 
 def cosine_similarity(x, y):
     return np.dot(x, y) / (np.linalg.norm(x) * np.linalg.norm(y))
 
 
 def identify_dissimilar_loci(values_sample, values_control, index: int, is_consensus: bool = False) -> int:
-    # If 'sample' has more than X% variation compared to 'control', unconditionally set it to "dissimilar loci"
-    threshold = 20 if is_consensus else 5
-    if values_sample[index] - values_control[index] > threshold:
+    # If 'sample' has more than 20% variation compared to 'control' in consensus mode, unconditionally set it to 'dissimilar loci'. This is set to counteract cases where, when evaluating cosine similarity during significant deletions, values exceedingly close to 1 can occur even if not observed in the control (e.g., control = [1,1,1,1,1], sample = [100,100,100,100,100] -> cosine similarity = 1).
+    if is_consensus and values_sample[index] - values_control[index] > 20:
         return True
 
-    x = values_sample[index - 5 : index + 6]
-    y = values_control[index - 5 : index + 6]
+    # Subset 10 bases around index and add 1e-6 to avoid division by zero when calculating cosine similarity.
+    x = np.array(values_sample[index - 5 : index + 6]) + 1e-6
+    y = np.array(values_control[index - 5 : index + 6]) + 1e-6
 
     return cosine_similarity(x, y) < 0.95
 
 
 def detect_anomalies(values_sample, values_control, threshold: float, is_consensus: bool = False) -> set[int]:
     """
     Detect anomalies and return indices of outliers.
     """
     values_subtract = values_sample - values_control
     values_subtract = np.where(values_subtract <= threshold, 0, values_subtract)
 
     values_subtract_reshaped = values_subtract.reshape(-1, 1)
     kmeans = MiniBatchKMeans(n_clusters=2, random_state=0, n_init="auto").fit(values_subtract_reshaped)
-    threshold = kmeans.cluster_centers_.mean()
-    candidate_loci = {i for i, v in enumerate(values_subtract_reshaped) if v > threshold}
+    threshold_kmeans = kmeans.cluster_centers_.mean()
+    candidate_loci = {i for i, v in enumerate(values_subtract_reshaped) if v > threshold_kmeans}
 
     return {i for i in candidate_loci if identify_dissimilar_loci(values_sample, values_control, i, is_consensus)}
 
 
 def extract_anomal_loci(
     indels_normalized_sample,
     indels_normalized_control,
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/insertions_to_fasta.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/insertions_to_fasta.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/input_formatter.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/input_formatter.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/directory_manager.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/directory_manager.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/cache_checker.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/cache_checker.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/knockin_handler.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/knockin_handler.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/homopolymer_handler.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/homopolymer_handler.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/genome_fetcher.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/genome_fetcher.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/mapping.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/mapping.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/preprocess/midsv_caller.py` & `dajin2-0.4.4/src/DAJIN2/core/preprocess/midsv_caller.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/consensus/mutation_extractor.py` & `dajin2-0.4.4/src/DAJIN2/core/consensus/mutation_extractor.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/consensus/similarity_searcher.py` & `dajin2-0.4.4/src/DAJIN2/core/consensus/similarity_searcher.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/consensus/name_handler.py` & `dajin2-0.4.4/src/DAJIN2/core/consensus/name_handler.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/consensus/consensus.py` & `dajin2-0.4.4/src/DAJIN2/core/consensus/consensus.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/report/mutation_exporter.py` & `dajin2-0.4.4/src/DAJIN2/core/report/mutation_exporter.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/report/insertion_reflector.py` & `dajin2-0.4.4/src/DAJIN2/core/report/insertion_reflector.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/report/sequence_exporter.py` & `dajin2-0.4.4/src/DAJIN2/core/report/sequence_exporter.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/report/bam_exporter.py` & `dajin2-0.4.4/src/DAJIN2/core/report/bam_exporter.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/classification/allele_merger.py` & `dajin2-0.4.4/src/DAJIN2/core/classification/allele_merger.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/classification/classifier.py` & `dajin2-0.4.4/src/DAJIN2/core/classification/classifier.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/kmer_generator.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/kmer_generator.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_updator.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/label_updator.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/score_handler.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/score_handler.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_merger.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/label_merger.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/label_extractor.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/label_extractor.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/strand_bias_handler.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/strand_bias_handler.py`

 * *Files 17% similar despite different names*

```diff
@@ -1,113 +1,115 @@
 from __future__ import annotations
 
+"""
+Nanopore sequencing results often results in strand specific mutations even though the mutation is not strand specific, thus they are considered as sequencing errors and should be removed.
+
+This module provides functions to determine whether each allele obtained after clustering is formed due to sequencing errors caused by strand bias.
+
+Re-allocates reads belonging to clusters with strand bias to clusters without strand bias.
+"""
+
 from pathlib import Path
-from collections import defaultdict, Counter
+from collections import defaultdict
 from sklearn.tree import DecisionTreeClassifier
 
 from DAJIN2.utils import io
 
 # Constants
 STRAND_BIAS_LOWER_LIMIT = 0.1
 STRAND_BIAS_UPPER_LIMIT = 0.9
 
 
 def is_strand_bias(path_control: Path) -> bool:
+    """
+    Determines whether there is a strand bias in sequencing data
+    based on the distribution of '+' and '-' strands.
+    """
     count_strand = defaultdict(int)
-    for m in io.read_jsonl(path_control):
-        count_strand[m["STRAND"]] += 1
+    for sample in io.read_jsonl(path_control):
+        count_strand[sample["STRAND"]] += 1
 
     total = count_strand["+"] + count_strand["-"]
-    percentage_plus = count_strand["+"] / total if total else 0
+    percentage_plus = count_strand["+"] / total if total > 0 else 0
 
     return not (STRAND_BIAS_LOWER_LIMIT < percentage_plus < STRAND_BIAS_UPPER_LIMIT)
 
 
 ###############################################################################
 # Handle Strand bias
 # # Clusters of reads with mutations with strand bias are merged into similar clusters without strand bias
 ###############################################################################
 
 
-def _count_strand(labels: list[int], samples: list[dict[str, str]]) -> tuple[defaultdict, defaultdict]:
+def count_strand(labels: list[int], samples: list[dict[str, str]]) -> tuple[dict[str, int], dict[str, int]]:
     """Count the occurrences of each strand type by label."""
-    count_strand_by_labels = defaultdict(int)
-    total_count_by_labels = defaultdict(int)
+    positive_strand_counts_by_labels = defaultdict(int)
+    total_counts_by_labels = defaultdict(int)
 
     for label, sample in zip(labels, samples):
-        total_count_by_labels[label] += 1
+        total_counts_by_labels[label] += 1
         if sample["STRAND"] == "+":
-            count_strand_by_labels[label] += 1
+            positive_strand_counts_by_labels[label] += 1
 
-    return count_strand_by_labels, total_count_by_labels
+    return dict(positive_strand_counts_by_labels), dict(total_counts_by_labels)
 
 
-def _calculate_strand_biases(
-    count_strand_by_labels: defaultdict, total_count_by_labels: defaultdict
+def determine_strand_biases(
+    positive_strand_counts_by_labels: defaultdict, total_counts_by_labels: defaultdict
 ) -> dict[int, bool]:
-    """Calculate strand biases based on strand counts."""
+    """Determine strand biases based on positive strand counts."""
     strand_biases = {}
-    for label, total in total_count_by_labels.items():
-        strand_count = count_strand_by_labels[label]
-        strand_ratio = strand_count / total
+    for label, total in total_counts_by_labels.items():
+        positive_strand_count = positive_strand_counts_by_labels[label]
+        strand_ratio = positive_strand_count / total
         strand_biases[label] = not (STRAND_BIAS_LOWER_LIMIT < strand_ratio < STRAND_BIAS_UPPER_LIMIT)
 
     return strand_biases
 
 
-def _get_strand_biases_on_each_label(labels: list[int], path_sample: Path | str) -> dict[int, bool]:
-    """Get strand biases for given labels and samples.
-    Args:
-        labels: A list of integer labels.
-        path_sample: The path to the sample file.
-    Returns:
-        A dictionary containing strand biases by label.
-    """
-    samples = io.read_jsonl(path_sample)
-    count_strand_by_labels, total_count_by_labels = _count_strand(labels, samples)
-    return _calculate_strand_biases(count_strand_by_labels, total_count_by_labels)
-
-
-def _prepare_training_testing_sets(labels, scores, strand_biases) -> tuple[list, list, list]:
-    x_train, y_train, x_test = [], [], []
+def prepare_training_testing_sets(labels, scores, strand_biases) -> tuple[list, list, list]:
+    """Prepare training and testing datasets based on strand biases."""
+    train_data, train_labels, test_data = [], [], []
     for label, score in zip(labels, scores):
         if strand_biases[label]:
-            x_test.append(score)
+            test_data.append(score)
         else:
-            x_train.append(score)
-            y_train.append(label)
-    return x_train, y_train, x_test
+            train_data.append(score)
+            train_labels.append(label)
+    return train_data, train_labels, test_data
 
 
-def _train_decision_tree(x_train, y_train) -> DecisionTreeClassifier:
+def train_decision_tree(train_data, train_labels) -> DecisionTreeClassifier:
+    """Train a decision tree classifier using the provided features and labels."""
     dtree = DecisionTreeClassifier(random_state=1)
-    dtree.fit(x_train, y_train)
+    dtree.fit(train_data, train_labels)
     return dtree
 
 
-def _allocate_labels(labels, strand_biases, dtree, x_test) -> list[int]:
-    label_predictions = dtree.predict(x_test)
-    label_predict_iter = iter(label_predictions)
+def allocate_labels(labels: list[int], strand_biases: dict[str, bool], dtree, test_data) -> list[int]:
+    """Re-allocates reads belonging to clusters with strand bias to clusters without strand bias."""
+    label_predictions = iter(dtree.predict(test_data))
     for i, label in enumerate(labels):
         if strand_biases[label]:
-            labels[i] = next(label_predict_iter)
+            labels[i] = next(label_predictions)
     return labels
 
 
-def _correct_clusters_with_strand_bias(path_score_sample, labels, strand_biases) -> list[int]:
-    scores = io.read_jsonl(path_score_sample)
-    x_train, y_train, x_test = _prepare_training_testing_sets(labels, scores, strand_biases)
-    dtree = _train_decision_tree(x_train, y_train)
-    return _allocate_labels(labels, strand_biases, dtree, x_test)
-
+def remove_biased_clusters(path_sample: Path, path_score_sample: Path, labels: list[int]) -> list[int]:
+    """Remove clusters with strand bias by re-labeling based on decision tree predictions.
+    Continue until at least one of the samples exhibits strand bias (i.e., do not calculate if all samples exhibit strand bias, or conversely, if none of the samples exhibit strand bias) or
+    1000 iterations are reached, which serves as a safeguard to prevent infinite loops.
+    """
+    samples = io.read_jsonl(path_sample)
+    positive_strand_counts_by_labels, total_counts_by_labels = count_strand(labels, samples)
+    strand_biases = determine_strand_biases(positive_strand_counts_by_labels, total_counts_by_labels)
 
-def remove_biased_clusters(path_sample, path_score_sample, labels) -> list[int]:
-    strand_biases = _get_strand_biases_on_each_label(labels, path_sample)
-    # Until there is at least one True and one False or
-    # 1000 iterations (1000 is a suitable number to exit an infinite loop just in case)
-    i = 0
+    iteration_count = 0
     labels_corrected = labels
-    while len(Counter(strand_biases.values())) > 1 and i < 1000:
-        labels_corrected = _correct_clusters_with_strand_bias(path_score_sample, labels_corrected, strand_biases)
-        strand_biases = _get_strand_biases_on_each_label(labels_corrected, path_sample)
-        i += 1
+    while len(set(strand_biases.values())) > 1 or iteration_count < 1000:
+        scores = io.read_jsonl(path_score_sample)
+        train_data, train_labels, test_data = prepare_training_testing_sets(labels, scores, strand_biases)
+        dtree = train_decision_tree(train_data, train_labels)
+        labels_corrected = allocate_labels(labels, strand_biases, dtree, test_data)
+        strand_biases = determine_strand_biases(labels_corrected, path_sample)
+        iteration_count += 1
     return labels_corrected
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/appender.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/appender.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/core/clustering/clustering.py` & `dajin2-0.4.4/src/DAJIN2/core/clustering/clustering.py`

 * *Files 16% similar despite different names*

```diff
@@ -35,43 +35,44 @@
 
         labels_sample = labels_all[:coverage_sample]
         labels_control = labels_all[coverage_sample:]
         labels_current = merge_labels(labels_control, labels_sample, labels_previous)
         # print(i, Counter(labels_sample), Counter(labels_control), Counter(labels_current))  # ! DEBUG
 
         num_labels_control = count_number_of_clusters(labels_control, coverage_control)
-        mutual_info = metrics.adjusted_rand_score(labels_previous, labels_current)
+        rand_index = metrics.adjusted_rand_score(labels_previous, labels_current)
 
         """
         Return the number of clusters when:
-        - the number of clusters in control is split into more than one.
-        - the mutual information between the current and previous labels is high enough (= similar).
+            - the number of clusters in control is split into more than one.
+            - the mutual information between the current and previous labels is high enough (= similar).
+        To reduce the allele number, previous labels are returned.
         """
-        if num_labels_control >= 2:
+        if num_labels_control >= 2 or rand_index >= 0.95:
             return labels_previous
-        if 0.95 <= mutual_info <= 1.0:
-            return labels_current
         labels_previous = labels_current
     return labels_previous
 
 
 def get_label_most_common(labels: list[int]) -> int:
     return Counter(labels).most_common()[0][0]
 
 
-def return_labels(path_score_sample: Path, path_score_control: Path, path_sample: Path, strand_bias: bool) -> list[int]:
+def return_labels(
+    path_score_sample: Path, path_score_control: Path, path_sample: Path, strand_bias_in_control: bool
+) -> list[int]:
     np.random.seed(seed=1)
     score_control = list(io.read_jsonl(path_score_control))
     X_control = csr_matrix(score_control)
-    # subset to 1000 reads of controls in the most common cluster to remove outliers and reduce computation time
+    """Subset to 1000 reads of controls in the most common cluster to remove outliers and reduce computation time"""
     labels_control = BisectingKMeans(n_clusters=2, random_state=1).fit_predict(X_control)
     label_most_common = get_label_most_common(labels_control)
     scores_control_subset = subset_scores(labels_control, io.read_jsonl(path_score_control), label_most_common, 1000)
     scores_sample = list(io.read_jsonl(path_score_sample))
     X = csr_matrix(list(chain(scores_sample, scores_control_subset)))
     coverage_sample = io.count_newlines(path_score_sample)
     coverage_control = len(scores_control_subset)
     labels = optimize_labels(X, coverage_sample, coverage_control)
-    # correct clusters with strand bias
-    if strand_bias is False:
+    """Re-allocate clusters with strand bias to clusters without strand bias"""
+    if strand_bias_in_control is False:
         labels = remove_biased_clusters(path_sample, path_score_sample, labels)
     return labels
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2/templates/index.html` & `dajin2-0.4.4/src/DAJIN2/templates/index.html`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/multiprocess.py` & `dajin2-0.4.4/src/DAJIN2/utils/multiprocess.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/cssplits_handler.py` & `dajin2-0.4.4/src/DAJIN2/utils/cssplits_handler.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/fastx_handler.py` & `dajin2-0.4.4/src/DAJIN2/utils/fastx_handler.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/report_generator.py` & `dajin2-0.4.4/src/DAJIN2/utils/report_generator.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/io.py` & `dajin2-0.4.4/src/DAJIN2/utils/io.py`

 * *Files 6% similar despite different names*

```diff
@@ -79,16 +79,16 @@
 
 
 ###########################################################
 # Load batch file
 ###########################################################
 
 
-def check_excel_or_csv(file_path: str) -> str | None:
-    """Check if the file is an Excel or CSV file. Raise error for other types."""
+def determine_file_type(file_path: str) -> str | None:
+    """Determine if the file is an Excel or CSV file. Raise error for other types."""
     file_extension = Path(file_path).suffix
     if file_extension in [".xlsx", ".xls"]:
         return "excel"
     elif file_extension == ".csv":
         return "csv"
     else:
         raise ValueError("The provided file must be either an Excel or CSV file.")
@@ -108,24 +108,26 @@
 
     return data
 
 
 def read_csv(file_path: str) -> list[dict[str, str]]:
     """Load data from a CSV file and return as a list."""
     with open(file_path, "r") as csvfile:
-        contents = []
+        inputs = []
         for row in csv.reader(csvfile):
+            if not row:  # Skip empty rows
+                continue
             trimmed_row = [field.strip() for field in row]
-            contents.append(trimmed_row)
-        return contents
+            inputs.append(trimmed_row)
+        return inputs
 
 
 def load_batchfile(batchfile_path: str) -> list[dict[str, str]]:
     """Load data from either an Excel or CSV file."""
-    file_type = check_excel_or_csv(batchfile_path)
+    file_type = determine_file_type(batchfile_path)
     if file_type == "excel":
         return read_xlsx(batchfile_path)
     elif file_type == "csv":
         return read_csv(batchfile_path)
 
 
 ###########################################################
```

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/input_validator.py` & `dajin2-0.4.4/src/DAJIN2/utils/input_validator.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/config.py` & `dajin2-0.4.4/src/DAJIN2/utils/config.py`

 * *Files identical despite different names*

## Comparing `DAJIN2-0.4.3/src/DAJIN2/utils/sam_handler.py` & `dajin2-0.4.4/src/DAJIN2/utils/sam_handler.py`

 * *Files identical despite different names*

